--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,8 @@ b2sums=('e49b21dadc06cd91ec71a7c48e6a731a29560dde91f53cea615ffcfc747f443b5db3bfc
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../add-riscv64-support.patch
+  patch -Np1 -i ../riscv-hotplug.patch
   ./bootstrap
 }
 
@@ -31,3 +33,7 @@ package() {
   install -D -m644 kexec/kexec.8 "$pkgdir"/usr/share/man/man8/kexec.8
 }
 
+source+=("add-riscv64-support.patch"
+         "riscv-hotplug.patch")
+b2sums+=('60f71a3da5945af3099f0359f695c7e8cb6875b10b223df32bd729d25baa7474914c9aa49c37fbc2af173ce9804696b17d3681060cba599e150b42c9daa32bbd'
+         '05600e0964fa35df04f5f25aee15745bce91f5b1ad9320880bc48bf56563e512b21ceb069bc3898069602b4629cbba89e3688887fb9d39c1438c031654c8016a')
