--- PKGBUILD
+++ PKGBUILD
@@ -39,10 +39,12 @@
 source=(
   "git+https://github.com/mypaint/mypaint.git?signed#tag=v$pkgver"
   mypaint-backport-fixes.patch
+  fix-swig-4.5.patch
 )
 b2sums=(
   0b6010d0c3d3ef1b2a8e9132ad80ba4bbb701c8dfa33d619157aa01d3d87baf9832defba2b141b9ed58ec8345165bb7c80edebdeb6945689217d22754ec23646
   9e858de9dbe6ddaa58955ab868357da87a3a7a32648fc49cca5eb58d991493992117382e3df27951d4e3bb3a0bd27517b9dc742fce427ab214d9aa2454c20737
+  060b846a78887e92aa4a2bf90696d58d617ea81b42119e22b2f5e0b85e5946466fe635cf9c6ac79adabf12e61a0565b41ac35c496accaac49cd4bc8b29615664
 )
 validpgpkeys=(A5EF2D077F35BBD2973418254803CB83E02F8A4D) # Jesper Lloyd (Stoneface) <jpl.lloyd@gmail.com>
 
@@ -52,6 +54,8 @@
   # Backport fixes
   # https://github.com/mypaint/mypaint/pull/1252
   git apply -3 ../mypaint-backport-fixes.patch
+
+  patch -Np1 -i ../fix-swig-4.5.patch
 }
 
 build() {
