--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,17 @@ url="https://github.com/martinpitt/umockdev/"
 license=('LGPL')
 depends=('systemd' 'libgudev')
 makedepends=('meson' 'gobject-introspection' 'gtk-doc' 'vala')
-source=($pkgname-$pkgver.tar.gz::https://github.com/martinpitt/$pkgname/archive/$pkgver.tar.gz)
-sha256sums=('9d23009b9aa196c5abff69bc52bd7fee16af16225f39d57645b1bc6a3336b2a1')
+source=($pkgname-$pkgver.tar.gz::https://github.com/martinpitt/$pkgname/archive/$pkgver.tar.gz
+        fix-race-condition-test.patch::https://github.com/martinpitt/umockdev/pull/171.patch)
+sha256sums=('9d23009b9aa196c5abff69bc52bd7fee16af16225f39d57645b1bc6a3336b2a1'
+            'f5029596f424999f663f51b37c18d677fff74cc48b67fd0556841314d0e03231')
+
+prepare() {
+  cd $pkgname-$pkgver
+
+  # see https://github.com/martinpitt/umockdev/pull/171
+  patch -Np1 -i ../fix-race-condition-test.patch
+}
 
 build() {
   cd $pkgname-$pkgver
