diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 7f0d323..7d4a61d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,10 +10,17 @@ arch=('x86_64')
 url="https://github.com/launchbadge/sqlx"
 depends=('gcc-libs' 'openssl')
 makedepends=('rust')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/launchbadge/sqlx/archive/refs/tags/v${pkgver}.tar.gz")
-sha512sums=('308410d5d9125d241e6141b254ba55dc964f3e9fcca0a4b0a9e26a96bad250c0c8e3400f7e689226f0558bfaaf04fb9ac450fc2abbb484ac94d615442b34429b')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/launchbadge/sqlx/archive/refs/tags/v${pkgver}.tar.gz"
+        "allow_let_underscore_lock.patch")
+sha512sums=('308410d5d9125d241e6141b254ba55dc964f3e9fcca0a4b0a9e26a96bad250c0c8e3400f7e689226f0558bfaaf04fb9ac450fc2abbb484ac94d615442b34429b'
+            '0b7672f636a057fd45bd379704c2c357639f7581af97dedc0f1a14d961b2a7a9bc263420f291a4b9022d4d96819bc6810b50b94ad926871c724ed1b50e88db88')
 options=('!lto')

+prepare() {
+    cd "sqlx-$pkgver"
+    patch -p1 -i ../allow_let_underscore_lock.patch # picked from https://github.com/launchbadge/sqlx/commit/b5312c3b6f8e34e757d1441216b6b34a18afd14b
+}
+
 build() {
     cd "sqlx-$pkgver/sqlx-cli"
     cargo build --locked --release
