--- PKGBUILD
+++ PKGBUILD
@@ -5,15 +5,10 @@ pkgname=(
   'bootconfig'
   'bpf'
   'cpupower'
-  'hyperv'
-  'intel-speed-select'
-  'kcpuid'
   'linux-tools-meta'
   'perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=6.12
 pkgrel=11
@@ -69,6 +64,8 @@ sha256sums=('267bab84f30e3ce4a88b6441aeee777b114fd58041b43cabfe50fdf0c0a97321'
 prepare() {
   cd linux
 
+  git cherry-pick -n 710fbca820c721cdd60fa8c5bbe9deb4c0788aae
+
   # apply patch from the source array (should be a pacman feature)
   local src
   for src in "${source[@]}"; do
@@ -100,7 +97,8 @@ build() {
     NO_LIBLLVM=1 \
     PYTHON_CONFIG=python-config \
     LIBPFM4=1 \
-    DESTDIR="$pkgdir"
+    DESTDIR="$pkgdir" \
+    SRCARCH=riscv
   popd
 
   echo ':: cpupower'
@@ -108,11 +106,6 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
-  echo ':: x86_energy_perf_policy'
-  pushd linux/tools/power/x86/x86_energy_perf_policy
-  make
-  popd
-
   echo ':: usbip'
   pushd linux/tools/usb/usbip
   # Fix gcc compilation
@@ -127,21 +120,11 @@ build() {
   make
   popd
 
-  echo ':: turbostat'
-  pushd linux/tools/power/x86/turbostat
-  make
-  popd
-
-  echo ':: hv'
-  pushd linux/tools/hv
-  CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hypervkvpd/\"' make
-  popd
-
   echo ':: bpf'
   pushd linux/tools/bpf
   # doesn't compile when we don't first compile bpftool in its own directory and
   # man pages require to be also launch from the subdirectory
-  make -C bpftool all doc
+  make -C bpftool all doc SRCARCH=riscv
   # runqslower, require kernel binary path to build, skip it
   make -W runqslower
   popd
@@ -150,16 +133,6 @@ build() {
   pushd linux/tools/bootconfig
   make
   popd
-
-  echo ':: intel-speed-select'
-  pushd linux/tools/power/x86/intel-speed-select
-  make
-  popd
-
-  echo ':: kcpuid'
-  pushd linux/tools/arch/x86/kcpuid
-  make
-  popd
 }
 
 package_linux-tools-meta() {
@@ -169,14 +142,9 @@ package_linux-tools-meta() {
     'bootconfig'
     'bpf'
     'cpupower'
-    'hyperv'
-    'intel-speed-select'
-    'kcpuid'
     'perf'
     'tmon'
-    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
