diff --git PKGBUILD PKGBUILD
index 2389983d..28d092e7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,7 @@ pkgname=(
   'perf'
 #  'python-perf'
   'tmon'
-  'turbostat'
   'usbip'
-  'x86_energy_perf_policy'
 )
 pkgver=6.0
 pkgrel=2
@@ -97,11 +95,6 @@ build() {
   make VERSION=$pkgver-$pkgrel
   popd
 
-  echo ':: x86_energy_perf_policy'
-  pushd linux/tools/power/x86/x86_energy_perf_policy
-  make
-  popd
-
   echo ':: usbip'
   pushd linux/tools/usb/usbip
   # Fix gcc compilation
@@ -121,11 +114,6 @@ build() {
   make
   popd
 
-  echo ':: turbostat'
-  pushd linux/tools/power/x86/turbostat
-  make
-  popd
-
   echo ':: hv'
   pushd linux/tools/hv
   CFLAGS+=' -DKVP_SCRIPTS_PATH=\"/usr/lib/hyperv/kvp_scripts/\"' make
@@ -158,9 +146,7 @@ package_linux-tools-meta() {
     'perf'
 #    'python-perf'
     'tmon'
-    'turbostat'
     'usbip'
-    'x86_energy_perf_policy'
   )
   conflicts=(
     'acpidump'
