diff --git PKGBUILD PKGBUILD
index e836a63..0cd1512 100644
--- PKGBUILD
+++ PKGBUILD
@@ -67,6 +67,9 @@ prepare() {
   # https://github.com/llvm/llvm-project/issues/82431
   sed '/^diff.*inline-asm-memop.ll/,$d' ../$pkgname-SelectionDAG.patch | patch -Np2
 
+  patch -Np2 -i ../JITLink-RISCV-Use-hashmap-to-find-PCREL_HI20-edge-78.patch
+  patch -Np2 -i ../JITLink-Always-unmap-standard-segments-in-InProcessM.patch
+
   # Remove CMake find module for zstd; breaks if out of sync with upstream zstd
   rm cmake/modules/Findzstd.cmake
 }
@@ -158,4 +161,9 @@ package_llvm-libs() {
     "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
+source+=(JITLink-RISCV-Use-hashmap-to-find-PCREL_HI20-edge-78.patch
+         JITLink-Always-unmap-standard-segments-in-InProcessM.patch)
+sha256sums+=('dd0d5904bdd69aea6e4937cc11336dc7a9d9c64584575c183da306a9baf93199'
+             '3fccc34270fa1cc0645fe8e6eac74c410ec108c63d653beca26434bd93dfd159')
+
 # vim:set ts=2 sw=2 et:
