diff -uprN hping3-20051105/hping2.h hping3-20051105-patch/hping2.h
--- hping3-20051105/hping2.h	2021-10-01 04:41:04.143459520 +0800
+++ hping3-20051105-patch/hping2.h	2021-10-01 04:41:16.686999354 +0800
@@ -353,7 +353,7 @@ struct delaytable_element {
 	int status;
 };
 
-volatile struct delaytable_element delaytable[TABLESIZE];
+extern volatile struct delaytable_element delaytable[TABLESIZE];
 
 /* protos */
 void	nop(void);				/* nop */
