--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,9 @@ sha256sums=('6069cff79a31dfdfc97a03ce1ee1d7df94ae649b4703394000936724cf88cf44')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
   mkdir -p utils/completions/
 }
 
