--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,15 @@ url="https://github.com/nukesor/pueue"
 license=('MIT')
 depends=('gcc-libs')
 makedepends=('cargo')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha256sums=('5d880731b7911dcc01c84ad547d703d4d438a95a64396d3610829d0c05bb1e37')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
+        "fix-ring.patch::https://patch-diff.githubusercontent.com/raw/Nukesor/pueue/pull/484.diff")
+sha256sums=('5d880731b7911dcc01c84ad547d703d4d438a95a64396d3610829d0c05bb1e37'
+            'SKIP')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  patch -Np1 -i ../fix-ring.patch
+  cargo fetch --locked
   mkdir -p utils/completions/
 }
 
