--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,9 @@ sha256sums=('cd5c6500e65960f6a102db5d0f0544f49eec8f74b2cc0df16ded8e2525a545f6')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
   mkdir -p utils/completions/
 }
 
