--- a/crates/biome_cli/src/service/mod.rs
+++ b/crates/biome_cli/src/service/mod.rs
@@ -226,7 +226,7 @@
                         Err(_) => Err(TransportError::ChannelClosed),
                     }
                 }
-                _ = sleep(Duration::from_secs(15)) => {
+                _ = sleep(Duration::from_secs(120)) => {
                     Err(TransportError::Timeout)
                 }
             }
