--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,7 @@
 
 prepare() {
 	_srcenv
+	patch -Np1 -i "$srcdir/increase-workspace-timeout.patch"
 	cargo fetch --locked --target host-tuple
 }
 
@@ -57,3 +58,6 @@
 	install -Dm0755 -t "$pkgdir/usr/bin/" "target/release/$pkgname"
 	install -Dm0644 -t "$pkgdir/usr/share/licenses/$pkgname/" LICENSE-*
 }
+
+source+=(increase-workspace-timeout.patch)
+sha256sums+=('bd734ad62d9cfffee67564f2257a564db4fb48e40c8aeb545f48b6487e7fc8a6')
