--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,7 @@
 
 prepare() {
 	_srcenv
+	patch -Np1 -i "$srcdir/increase-workspace-timeout.patch"
 	cargo fetch --locked --target host-tuple
 }
 
@@ -57,3 +58,6 @@
 	install -Dm0755 -t "$pkgdir/usr/bin/" "target/release/$pkgname"
 	install -Dm0644 -t "$pkgdir/usr/share/licenses/$pkgname/" LICENSE-*
 }
+
+source+=(increase-workspace-timeout.patch)
+sha256sums+=('a02f5e3ff6011a5af29afe42a298f719bb8994c08ce118498b1b237e4837b63a')
