diff --git PKGBUILD PKGBUILD
index 80722b8f..ce6b0baf 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,17 @@ makedepends=('go' 'pandoc' 'man-db')
 source=(
     "${pkgname}-${pkgver}.tar.gz::https://github.com/${_pkgauthor}/${pkgname}/releases/download/v${pkgver}/${pkgname}_v${pkgver}_src-deps.tar.gz"
     "${pkgname}-${pkgver}.tar.gz.asc::https://github.com/${_pkgauthor}/${pkgname}/releases/download/v${pkgver}/${pkgname}_v${pkgver}_src-deps.tar.gz.asc"
+    "add-riscv64.patch"
 )
 sha256sums=('945e3287311547f9227f4a5b5d051fd6df8b8b41ce2a65f424de9829cc785129'
-            'SKIP')
+            'SKIP'
+            'b189e1c7bffb2587c45ba97a66c52d4ca592adbc348d7b54d13df0a0df811755')
 validpgpkeys=('FFF3E01444FED7C316A3545A895F5BC123A02740')
 
+prepare() {
+    patch -Np1 -d "${srcdir}/${pkgname}_v${pkgver}_src-deps" < "${srcdir}/add-riscv64.patch"
+}
+
 build() {
     cd "${srcdir}/${pkgname}_v${pkgver}_src-deps"
 
