--- tests/legacy/utils.py.orig	2023-06-14 15:22:03.463707446 +0800
+++ tests/legacy/utils.py	2023-06-14 15:22:59.248160752 +0800
@@ -88,7 +88,7 @@
 
 # Unit for timeouts. May be increased on slow machines by setting the
 # WEBSOCKETS_TESTS_TIMEOUT_FACTOR environment variable.
-MS = 0.001 * int(os.environ.get("WEBSOCKETS_TESTS_TIMEOUT_FACTOR", 1))
+MS = 0.001 * float(os.environ.get("WEBSOCKETS_TESTS_TIMEOUT_FACTOR", 1))
 
 # asyncio's debug mode has a 10x performance penalty for this test suite.
 if os.environ.get("PYTHONASYNCIODEBUG"):  # pragma: no cover
