diff --git PKGBUILD PKGBUILD
index 3d0ba83..8288477 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,6 @@ depends=(
 )
 makedepends=(
   cmake
-  cuda
   gcc-fortran
   gtest
   ninja
@@ -66,10 +65,6 @@ build() {
     -G Ninja
     -W no-dev
     -DCMAKE_INSTALL_PREFIX=/usr
-    -DCMAKE_CUDA_ARCHITECTURES="$cuda_archs"
-    # Compile source code for supported GPU archs in parallel
-    -DCMAKE_CUDA_FLAGS="--threads $(nproc)"
-    -DADIOS2_USE_CUDA=ON
     -DADIOS2_USE_EXTERNAL_DEPENDENCIES=ON
     #-DADIOS2_HAVE_HDF5_VOL needs hdf5-openmpi
     -DADIOS2_HAVE_HDF5_VOL=OFF
