--- a/test/parallel/test-worker-messaging.js
+++ b/test/parallel/test-worker-messaging.js
@@ -1,4 +1,5 @@
 'use strict';
+// Flags: --disable-wasm-trap-handler
 
 const common = require('../common');
 const Countdown = require('../common/countdown');
--- a/test/sequential/test-watch-mode.mjs
+++ b/test/sequential/test-watch-mode.mjs
@@ -73,7 +73,7 @@
       future.resolve();
       return { stdout, stderr };
     },
-    restart(timeout = 1000) {
+    restart(timeout = 4000) {
       if (!child) {
         run();
       }
--- a/test/parallel/test-child-process-fork-closed-channel-segfault.js
+++ b/test/parallel/test-child-process-fork-closed-channel-segfault.js
@@ -78,6 +78,7 @@
           // may already have exited.
           if (err && err.code !== 'ERR_IPC_CHANNEL_CLOSED' &&
                      err.code !== 'ECONNRESET' &&
+                     err.code !== 'EPIPE' &&
                      err.code !== 'ECONNREFUSED' &&
                      err.code !== 'EMFILE') {
             throw err;
