--- a/deps/v8/src/compiler/backend/riscv/instruction-selector-riscv64.cc
+++ b/deps/v8/src/compiler/backend/riscv/instruction-selector-riscv64.cc
@@ -2009,26 +2009,6 @@
 
 namespace {
 
-#ifndef V8_COMPRESS_POINTERS
-bool IsNodeUnsigned(Node* n) {
-  NodeMatcher m(n);
-
-  if (m.IsLoad() || m.IsUnalignedLoad() || m.IsProtectedLoad()) {
-    LoadRepresentation load_rep = LoadRepresentationOf(n->op());
-    return load_rep.IsUnsigned();
-  } else if (m.IsWord32AtomicLoad() || m.IsWord64AtomicLoad()) {
-    AtomicLoadParameters atomic_load_params = AtomicLoadParametersOf(n->op());
-    LoadRepresentation load_rep = atomic_load_params.representation();
-    return load_rep.IsUnsigned();
-  } else {
-    return m.IsUint32Div() || m.IsUint32LessThan() ||
-           m.IsUint32LessThanOrEqual() || m.IsUint32Mod() ||
-           m.IsUint32MulHigh() || m.IsChangeFloat64ToUint32() ||
-           m.IsTruncateFloat64ToUint32() || m.IsTruncateFloat32ToUint32();
-  }
-}
-#endif
-
 // Shared routine for multiple word compare operations.
 template <typename Adapter>
 void VisitFullWord32Compare(InstructionSelectorT<Adapter>* selector,
@@ -2049,78 +2029,11 @@
   VisitCompare(selector, opcode, leftOp, rightOp, cont);
 }
 
-#ifndef V8_COMPRESS_POINTERS
-template <typename Adapter>
-void VisitOptimizedWord32Compare(InstructionSelectorT<Adapter>* selector,
-                                 Node* node, InstructionCode opcode,
-                                 FlagsContinuationT<Adapter>* cont) {
-  if (v8_flags.debug_code) {
-    RiscvOperandGeneratorT<Adapter> g(selector);
-    InstructionOperand leftOp = g.TempRegister();
-    InstructionOperand rightOp = g.TempRegister();
-    InstructionOperand optimizedResult = g.TempRegister();
-    InstructionOperand fullResult = g.TempRegister();
-    FlagsCondition condition = cont->condition();
-    InstructionCode testOpcode = opcode |
-                                 FlagsConditionField::encode(condition) |
-                                 FlagsModeField::encode(kFlags_set);
-
-    selector->Emit(testOpcode, optimizedResult, g.UseRegister(node->InputAt(0)),
-                   g.UseRegister(node->InputAt(1)));
-
-    selector->Emit(kRiscvShl64, leftOp, g.UseRegister(node->InputAt(0)),
-                   g.TempImmediate(32));
-    selector->Emit(kRiscvShl64, rightOp, g.UseRegister(node->InputAt(1)),
-                   g.TempImmediate(32));
-    selector->Emit(testOpcode, fullResult, leftOp, rightOp);
-
-    selector->Emit(kRiscvAssertEqual, g.NoOutput(), optimizedResult, fullResult,
-                   g.TempImmediate(static_cast<int>(
-                       AbortReason::kUnsupportedNonPrimitiveCompare)));
-  }
-
-  VisitWordCompare(selector, node, opcode, cont, false);
-}
-#endif
 template <typename Adapter>
 void VisitWord32Compare(InstructionSelectorT<Adapter>* selector,
                         typename Adapter::node_t node,
                         FlagsContinuationT<Adapter>* cont) {
-  if constexpr (Adapter::IsTurboshaft) {
-    VisitFullWord32Compare(selector, node, kRiscvCmp, cont);
-  } else {
-    // RISC-V doesn't support Word32 compare instructions. Instead it relies
-    // that the values in registers are correctly sign-extended and uses
-    // Word64 comparison instead. This behavior is correct in most cases,
-    // but doesn't work when comparing signed with unsigned operands.
-    // We could simulate full Word32 compare in all cases but this would
-    // create an unnecessary overhead since unsigned integers are rarely
-    // used in JavaScript.
-    // The solution proposed here tries to match a comparison of signed
-    // with unsigned operand, and perform full Word32Compare only
-    // in those cases. Unfortunately, the solution is not complete because
-    // it might skip cases where Word32 full compare is needed, so
-    // basically it is a hack.
-    // When call to a host function in simulator, if the function return a
-    // int32 value, the simulator do not sign-extended to int64 because in
-    // simulator we do not know the function whether return a int32 or int64.
-    // so we need do a full word32 compare in this case.
-#ifndef V8_COMPRESS_POINTERS
-#ifndef USE_SIMULATOR
-    if (IsNodeUnsigned(node->InputAt(0)) != IsNodeUnsigned(node->InputAt(1))) {
-#else
-    if (IsNodeUnsigned(node->InputAt(0)) != IsNodeUnsigned(node->InputAt(1)) ||
-        node->InputAt(0)->opcode() == IrOpcode::kCall ||
-        node->InputAt(1)->opcode() == IrOpcode::kCall) {
-#endif
-      VisitFullWord32Compare(selector, node, kRiscvCmp, cont);
-    } else {
-      VisitOptimizedWord32Compare(selector, node, kRiscvCmp, cont);
-    }
-#else
-    VisitFullWord32Compare(selector, node, kRiscvCmp, cont);
-#endif
-  }
+  VisitFullWord32Compare(selector, node, kRiscvCmp, cont);
 }
 
 template <typename Adapter>
