diff --git a/test/parallel/test-strace-openat-openssl.js b/test/parallel/test-strace-openat-openssl.js
index 13882e67aec0f2..8e4a38fc6e7eaf 100644
--- a/test/parallel/test-strace-openat-openssl.js
+++ b/test/parallel/test-strace-openat-openssl.js
@@ -19,9 +19,13 @@ if (spawnSync('strace').error !== undefined) {
   const allowedOpenCalls = new Set([
     '/etc/ssl/openssl.cnf',
   ]);
+  const syscalls = ['openat'];
+  if (process.arch !== 'riscv64' && process.arch !== 'riscv32') {
+    syscalls.push('open');
+  }
   const strace = spawn('strace', [
     '-f', '-ff',
-    '-e', 'trace=open,openat',
+    '-e', `trace=${syscalls.join(',')}`,
     '-s', '512',
     '-D', process.execPath, '-e', 'require("crypto")',
   ]);
