diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 3ea5564d54..a09ca8d798 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -2284,6 +2284,9 @@
           ['v8_target_arch=="arm64"', {
             'defines': ['HWY_BROKEN_TARGETS=HWY_ALL_SVE',],
           }],
+          ['v8_target_arch=="riscv64"', {
+            'defines': ['HWY_BROKEN_TARGETS=HWY_RVV',],
+          }],
           ['v8_target_arch=="ppc64" or v8_target_arch=="s390x"', {
             'defines': ['TOOLCHAIN_MISS_ASM_HWCAP_H',],
           }],
@@ -2308,6 +2311,9 @@
         ['v8_target_arch=="arm64"', {
           'defines': ['HWY_BROKEN_TARGETS=HWY_ALL_SVE',],
         }],
+        ['v8_target_arch=="riscv64"', {
+            'defines': ['HWY_BROKEN_TARGETS=HWY_RVV',],
+        }],
         ['v8_target_arch=="ppc64" or v8_target_arch=="s390x"', {
           'defines': ['TOOLCHAIN_MISS_ASM_HWCAP_H',],
         }],
