diff --git PKGBUILD PKGBUILD
index ff2ec7e..c07eb5a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -61,6 +61,17 @@ _set_flags() {
   # /usr/lib/libnode.so uses malloc_usable_size, which is incompatible with fortification level 3
   CFLAGS="${CFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
   CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
+
+  # https://github.com/riscv-forks/electron/issues/7
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
+prepare() {
+  cd node
+
+  patch -Np1 -i ../hwy-broken-rvv.diff
+  patch -Np1 -i ../v8-riscv-fix-sp.patch
 }
 
 build() {
@@ -94,6 +105,8 @@ build() {
 check() {
   _set_flags
   cd node
+  rm test/parallel/test-snapshot-reproducible.js
+  # TODO: figure out why the above test is failing on RISC-V
   rm test/parallel/test-http2-client-set-priority.js
   rm test/parallel/test-http2-client-unescaped-path.js
   rm test/parallel/test-http2-max-invalid-frames.js
@@ -114,4 +127,10 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
 
+
+makedepends+=(clang)
+source+=("hwy-broken-rvv.diff"
+         v8-riscv-fix-sp.patch)
+b2sums+=('4e770adcd1d4edc71f1ebced7573ca61b6e7424f780baa15bd12aae8f37656ce518fe5b1a9cc4c838f41464421a25490d4d3c8c88b441a8dab50481df94f08d5'
+         'f1ad5e9efed34da853427b8aa1857d08f8856f0ec2c60feeb1d810dcc1f0d60f03570c0e40ee066e3a995e9be3de5115284b34aa2301bd7b15f100efb7caff23')
 # vim:set ts=2 sw=2 et:
