--- PKGBUILD
+++ PKGBUILD
@@ -58,10 +58,11 @@ pkgver() {
 }
 
 prepare() {
-  mkdir -p glibc-build lib32-glibc-build
+  mkdir -p glibc-build
 
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
+  patch -Np1 -i ../mremap-fix-test.patch
   # Fix build with linux 7.0+
   patch -Np1 < ../0001-include-isolate-__O_CLOEXEC-flag-for-sys-mount.h-and.patch
   patch -Np1 < ../0002-Linux-Only-define-OPEN_TREE_-macros-in-sys-mount.h-i.patch
@@ -76,7 +77,7 @@ build() {
   local _configure_flags=(
       --prefix=/usr
       --with-headers=/usr/include
-      --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues
+      --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
@@ -92,9 +93,13 @@ build() {
   # actual builds (support is built-in via --enable-fortify-source).
   CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
 
-  # locale-gen segfaults without this on glibc 2.42
+  # check-localplt requires malloc family symbols to go through PLT.
+  # -fno-plt causes compiler generated memset calls in early code to use GOT,
+  # leading to segfault due to uninitialized GOT entry. Removing -fno-plt restores
+  # the old happy path of generating R_RISCV_CALL_PLT for such memset calls and
+  # then optimizing them away.
+  CFLAGS=${CFLAGS/-fno-plt/}
   if [[ ${CARCH} = "aarch64" ]]; then
-     CFLAGS=${CFLAGS/-fno-plt/}
      _configure_flags+=(--enable-memory-tagging)
   fi
 
@@ -110,7 +115,6 @@ build() {
         --libdir=/usr/lib \
         --libexecdir=/usr/lib \
         --enable-cet \
-        --enable-sframe \
         "${_configure_flags[@]}"
 
     make -O
@@ -177,7 +181,7 @@ check() (
   _skip_test tst-shstk-legacy-1g     sysdeps/x86_64/Makefile
   _skip_test tst-adjtime             time/Makefile
 
-  make -O check
+  TIMEOUTFACTOR=20 make -O check
 )
 
 package_glibc() {
@@ -262,3 +266,6 @@ package_glibc-locales() {
   # deduplicate locale data
   hardlink -c "${pkgdir}"/usr/lib/locale
 }
+
+source+=("mremap-fix-test.patch::https://inbox.sourceware.org/libc-alpha/20250208-fix-mremap-tst-v1-1-ad7ee617ec65@coelacanthus.name/raw")
+b2sums+=('3ba09c775c0d58bf9e5b08985ea5fa73509433df7ff284f035f7f73357e8f35a79c8e7f3a0fecfe4b8759831e9f0d38e8524dcf2b8132066a47cb7e234880a6e')
