--- PKGBUILD
+++ PKGBUILD
@@ -44,21 +44,21 @@
 }
 
 prepare() {
-  mkdir -p glibc-build lib32-glibc-build
+  mkdir -p glibc-build
 
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc
   cd glibc
+  patch -Np1 -i ../mremap-fix-test.patch
 }
 
 build() {
   local _configure_flags=(
       --prefix=/usr
       --with-headers=/usr/include
-      --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues
+      --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues
       --enable-bind-now
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -70,9 +70,13 @@
   # actual builds (support is built-in via --enable-fortify-source).
   CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
 
-  # locale-gen segfaults without this on glibc 2.42
+  # check-localplt requires malloc family symbols to go through PLT.
+  # -fno-plt causes compiler generated memset calls in early code to use GOT,
+  # leading to segfault due to uninitialized GOT entry. Removing -fno-plt restores
+  # the old happy path of generating R_RISCV_CALL_PLT for such memset calls and
+  # then optimizing them away.
+  CFLAGS=${CFLAGS/-fno-plt/}
   if [[ ${CARCH} = "aarch64" ]]; then
-     CFLAGS=${CFLAGS/-fno-plt/}
      _configure_flags+=(--enable-memory-tagging)
   fi
 
@@ -154,7 +158,7 @@
   _skip_test tst-shstk-legacy-1g     sysdeps/x86_64/Makefile
   _skip_test tst-adjtime             time/Makefile
 
-  make -O check
+  TIMEOUTFACTOR=20 make -O check
 )
 
 package_glibc() {
@@ -239,3 +243,6 @@
   # deduplicate locale data
   hardlink -c "${pkgdir}"/usr/lib/locale
 }
+
+source+=("mremap-fix-test.patch::https://inbox.sourceware.org/libc-alpha/20250208-fix-mremap-tst-v1-1-ad7ee617ec65@coelacanthus.name/raw")
+b2sums+=('3ba09c775c0d58bf9e5b08985ea5fa73509433df7ff284f035f7f73357e8f35a79c8e7f3a0fecfe4b8759831e9f0d38e8524dcf2b8132066a47cb7e234880a6e')
