--- PKGBUILD
+++ PKGBUILD
@@ -7,14 +7,14 @@
 # NOTE: valgrind requires rebuilt with each major glibc version
 
 pkgbase=glibc
-pkgname=(glibc lib32-glibc glibc-locales)
+pkgname=(glibc glibc-locales)
 pkgver=2.38
 _commit=750a45a783906a19591fb8ff6b7841470f1f5701
 pkgrel=7
 arch=(x86_64)
 url='https://www.gnu.org/software/libc'
 license=(GPL LGPL)
-makedepends=(git gd lib32-gcc-libs python)
+makedepends=(git gd python)
 options=(staticlibs !lto)
 source=(git+https://sourceware.org/git/glibc.git#commit=${_commit}
         locale.gen.txt
@@ -54,7 +54,6 @@ build() {
       --enable-cet
       --enable-fortify-source
       --enable-kernel=4.4
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-systemtap
       --disable-nscd
@@ -86,25 +85,7 @@ build() {
     make info
   )
 
-  (
-    cd lib32-glibc-build
-    export CC="gcc -m32 -mstackrealign"
-    export CXX="g++ -m32 -mstackrealign"
-
-    echo "slibdir=/usr/lib32" >> configparms
-    echo "rtlddir=/usr/lib32" >> configparms
-    echo "sbindir=/usr/bin" >> configparms
-    echo "rootsbindir=/usr/bin" >> configparms
-
-    "${srcdir}"/glibc/configure \
-        --host=i686-pc-linux-gnu \
-        --libdir=/usr/lib32 \
-        --libexecdir=/usr/lib32 \
-        "${_configure_flags[@]}"
-
-    make -O
-  )
-
+  
   # pregenerate locales here instead of in package
   # functions because localedef does not like fakeroot
   make -C "${srcdir}"/glibc/localedata objdir="${srcdir}"/glibc-build \
@@ -140,7 +121,7 @@ check() (
   _skip_test tst-process_mrelease    sysdeps/unix/sysv/linux/Makefile
   _skip_test tst-adjtime             time/Makefile
 
-  make -O check
+  TIMEOUTFACTOR=20 make -O check
 )
 
 package_glibc() {
