--- PKGBUILD
+++ PKGBUILD
@@ -4,16 +4,15 @@
 # toolchain build order: linux-api-headers->glibc->binutils->gcc->binutils->glibc
 # NOTE: valgrind requires rebuilt with each major glibc version
 
-pkgbase=glibc
-pkgname=(glibc lib32-glibc)
+pkgname=glibc
 pkgver=2.33
-pkgrel=5
+pkgrel=5.1
 arch=(x86_64)
 url='https://www.gnu.org/software/libc'
 license=(GPL LGPL)
-makedepends=(git gd lib32-gcc-libs python)
+makedepends=(git gd python)
 optdepends=('perl: for mtrace')
-options=(!strip staticlibs)
+options=(!strip staticlibs !lto)
 #_commit=3de512be7ea6053255afed6154db9ee31d4e557a
 #source=(git+https://sourceware.org/git/glibc.git#commit=$_commit
 source=(https://ftp.gnu.org/gnu/glibc/glibc-$pkgver.tar.xz{,.sig}
@@ -40,7 +39,7 @@ md5sums=('390bbd889c7e8e8a7041564cb6b27cca'
          '7a09f1693613897add1791e7aead19c9')
 
 prepare() {
-  mkdir -p glibc-build lib32-glibc-build
+  mkdir -p glibc-build
 
   [[ -d glibc-$pkgver ]] && ln -s glibc-$pkgver glibc 
   cd glibc
@@ -68,13 +67,12 @@ build() {
       --enable-cet
       --enable-kernel=4.4
       --enable-lock-elision
-      --enable-multi-arch
       --enable-stack-protector=strong
       --enable-stackguard-randomization
-      --enable-static-pie
-      --enable-systemtap
+      #--enable-static-pie
       --disable-profile
       --disable-werror
+      --disable-experimental-malloc
   )
 
   cd "$srcdir/glibc-build"
@@ -86,6 +84,8 @@ build() {
 
   # remove fortify for building libraries
   CPPFLAGS=${CPPFLAGS/-D_FORTIFY_SOURCE=2/}
+  # https://bugs.archlinux.org/task/73530
+  CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=2/}
 
   #
   CFLAGS=${CFLAGS/-fno-plt/}
@@ -97,51 +97,17 @@ build() {
       --libexecdir=/usr/lib \
       ${_configure_flags[@]}
 
+  # TODO: figure out fortify
   # build libraries with fortify disabled
-  echo "build-programs=no" >> configparms
-  make
-
-  # re-enable fortify for programs
-  sed -i "/build-programs=/s#no#yes#" configparms
-
-  echo "CC += -D_FORTIFY_SOURCE=2" >> configparms
-  echo "CXX += -D_FORTIFY_SOURCE=2" >> configparms
-  make
-
-  # build info pages manually for reprducibility
-  make info
-
-  cd "$srcdir/lib32-glibc-build"
-  export CC="gcc -m32 -mstackrealign"
-  export CXX="g++ -m32 -mstackrealign"
-
-  echo "slibdir=/usr/lib32" >> configparms
-  echo "rtlddir=/usr/lib32" >> configparms
-  echo "sbindir=/usr/bin" >> configparms
-  echo "rootsbindir=/usr/bin" >> configparms
-
-  # remove fortify for building libraries
-  CPPFLAGS=${CPPFLAGS/-D_FORTIFY_SOURCE=2/}
-  CFLAGS=${CFLAGS/-fno-plt/}
-  CXXFLAGS=${CXXFLAGS/-fno-plt/}
-
-  "$srcdir/glibc/configure" \
-      --host=i686-pc-linux-gnu \
-      --libdir=/usr/lib32 \
-      --libexecdir=/usr/lib32 \
-      ${_configure_flags[@]}
-
-  # build libraries with fortify disabled
-  echo "build-programs=no" >> configparms
-  make
+  #echo "build-programs=no" >> configparms
+  #make
 
   # re-enable fortify for programs
-  sed -i "/build-programs=/s#no#yes#" configparms
+  #sed -i "/build-programs=/s#no#yes#" configparms
 
-  echo "CC += -D_FORTIFY_SOURCE=2" >> configparms
-  echo "CXX += -D_FORTIFY_SOURCE=2" >> configparms
+  #echo "CC += -D_FORTIFY_SOURCE=2" >> configparms
+  #echo "CXX += -D_FORTIFY_SOURCE=2" >> configparms
   make
-
 }
 
 check() {
@@ -154,7 +120,7 @@ check() {
   make check || true
 }
 
-package_glibc() {
+package() {
   pkgdesc='GNU C Library'
   depends=('linux-api-headers>=4.10' tzdata filesystem)
   optdepends=('gd: for memusagestat')
@@ -202,48 +168,6 @@ package_glibc() {
       -name '*-*.so' -type f -exec strip $STRIP_SHARED {} + 2> /dev/null || true
   fi
 
-  # Provide tracing probes to libstdc++ for exceptions, possibly for other
-  # libraries too. Useful for gdb's catch command.
-  install -Dm644 "$srcdir/sdt.h" "$pkgdir/usr/include/sys/sdt.h"
-  install -Dm644 "$srcdir/sdt-config.h" "$pkgdir/usr/include/sys/sdt-config.h"
-
   # Provided by libxcrypt; keep the old shared library for backwards compatibility
   rm -f "$pkgdir"/usr/include/crypt.h "$pkgdir"/usr/lib/libcrypt.{a,so}
 }
-
-package_lib32-glibc() {
-  pkgdesc='GNU C Library (32-bit)'
-  depends=("glibc=$pkgver")
-  options+=('!emptydirs')
-
-  cd lib32-glibc-build
-
-  make install_root="$pkgdir" install
-  rm -rf "$pkgdir"/{etc,sbin,usr/{bin,sbin,share},var}
-
-  # We need to keep 32 bit specific header files
-  find "$pkgdir/usr/include" -type f -not -name '*-32.h' -delete
-
-  # Dynamic linker
-  install -d "$pkgdir/usr/lib"
-  ln -s ../lib32/ld-linux.so.2 "$pkgdir/usr/lib/"
-
-  # Add lib32 paths to the default library search path
-  install -Dm644 "$srcdir/lib32-glibc.conf" "$pkgdir/etc/ld.so.conf.d/lib32-glibc.conf"
-
-  # Symlink /usr/lib32/locale to /usr/lib/locale
-  ln -s ../lib/locale "$pkgdir/usr/lib32/locale"
-
-  if check_option 'debug' n; then
-    find "$pkgdir"/usr/lib32 -name '*.a' -type f -exec strip $STRIP_STATIC {} + 2> /dev/null || true
-    find "$pkgdir"/usr/lib32 \
-      -not -name 'ld-*.so' \
-      -not -name 'libc-*.so' \
-      -not -name 'libpthread-*.so' \
-      -not -name 'libthread_db-*.so' \
-      -name '*-*.so' -type f -exec strip $STRIP_SHARED {} + 2> /dev/null || true
-  fi
-
-  # Provided by lib32-libxcrypt; keep the old shared library for backwards compatibility
-  rm -f "$pkgdir"/usr/lib32/libcrypt.{a,so}
-}
