--- PKGBUILD
+++ PKGBUILD
@@ -78,6 +78,10 @@
 }
 
 build() {
+  # https://github.com/apache/arrow/issues/12681
+  CFLAGS=${CFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+  CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+
   # workaround for cmake 4.0 - arrow bundles several old dependencies https://github.com/apache/arrow/issues/45985
   # (note that ExternalProject does not propagate this as a CMake variable so we need to export an environment variable)
   export CMAKE_POLICY_VERSION_MINIMUM=3.5
@@ -139,7 +143,7 @@
     # execute tests in parallel
     --parallel $(nproc)
     # limit execution time for each test
-    --timeout 200
+    --timeout 600
     # exclude problematic tests
     --exclude-regex "$excluded_tests"
   )
