diff --git PKGBUILD PKGBUILD
index 3cd0f68..26aa4ea 100644
--- PKGBUILD
+++ PKGBUILD
@@ -71,6 +71,10 @@ prepare() {
 }
 
 build() {
+  # https://github.com/apache/arrow/issues/12681
+  CFLAGS=${CFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+  CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+
   # workaround for cmake 4.0 - arrow bundles several old dependencies https://github.com/apache/arrow/issues/45985
   # (note that ExternalProject does not propagate this as a CMake variable so we need to export an environment variable)
   export CMAKE_POLICY_VERSION_MINIMUM=3.5
