diff --git a/PKGBUILD b/PKGBUILD
index d381f84..f378721 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,7 @@ depends=(apache-orc boost-libs brotli bzip2 double-conversion c-ares gflags
 provides=(parquet-cpp)
 conflicts=(parquet-cpp)
 makedepends=(boost cmake flatbuffers gmock python-numpy git clang)
+options=(!lto)
 source=(https://archive.apache.org/dist/${pkgname}/${pkgname}-${pkgver}/apache-${pkgname}-${pkgver}.tar.gz{,.asc}
         git+https://github.com/apache/parquet-testing.git
         git+https://github.com/apache/arrow-testing.git)
@@ -58,7 +59,8 @@ build(){
     -DARROW_WITH_SNAPPY=ON \
     -DARROW_WITH_ZLIB=ON \
     -DARROW_WITH_ZSTD=ON \
-    -DPARQUET_REQUIRE_ENCRYPTION=ON
+    -DPARQUET_REQUIRE_ENCRYPTION=ON \
+    -DARROW_CPU_FLAG=riscv64
   make -C build
 }
 
