--- PKGBUILD
+++ PKGBUILD
@@ -72,6 +72,11 @@ prepare() {
 }
 
 build() {
+  # https://github.com/apache/arrow/issues/12681
+  CFLAGS=${CFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+  CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}
+
+
   # Arrow options are defined in https://github.com/apache/arrow/blob/main/cpp/cmake_modules/DefineOptions.cmake
   # Documentation for building from source: https://arrow.apache.org/docs/developers/cpp/building.html
   local cmake_options=(
