--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,12 @@ depends=('glibc' 'lirc')
 source=("https://downloads.sourceforge.net/libirman/${pkgname}-${pkgver}.tar.gz")
 sha256sums=('43e58d7cd22b3a4c4dc8dcf8542a269ebcb4d8f6ceea0577b9fc882898f09a47')

+prepare() {
+  cd $pkgname-$pkgver
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var --disable-static --with-plugin
