diff --git PKGBUILD PKGBUILD
index 5ba99cb..926afba 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ _arch='x64'
 
 prepare() {
   cd $pkgbase-$pkgver
+  patch -Np1 -i ../add_riscv_support.patch
   # remove the path prefix from the css reference, so that the css can live
   # in the same directory
   sed -e 's|../Styles/||g' -i docs/$pkgbase/*.html
@@ -33,7 +34,7 @@ build() {
   cd $pkgname-$pkgver
   make
   make gptsync
-  make fs
+  make fs OMIT_SBAT=1
 }
 
 package_refind() {
@@ -103,3 +104,8 @@ package_refind-docs() {
   install -vDm 644 docs/Styles/*.css -t "$pkgdir/usr/share/doc/$pkgbase/html/"
   install -vDm 644 images/$pkgbase-banner.{png,svg} -t "$pkgdir/usr/share/doc/$pkgbase/html/"
 }
+
+_arch='riscv64'
+source+=("add_riscv_support.patch")
+sha512sums+=('35bfa50a1aecd572238dcb2439286ba9ee0470c190b047fc7277baa26980e27695351f47f91d0620c5d8b49d54ae95d0d25f9716ba2e850b8476ef8c91231eef')
+b2sums+=('2bdcb1f9fcec61db9498b6befc467f9a659a7f5c2c25d4d436c6aba734968ef4e3dec4a87c5eeef9c352a5ecde7a9d3123de0488b19902f47b709767c5b14645')
