diff --git PKGBUILD PKGBUILD
index df03030c..7a120600 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,13 +18,21 @@ optdepends=('gptfdisk: for finding non-vfat ESP with refind-install'
 )
 # the drivers don't build with many jobs
 options=(!makeflags)
-source=("https://sourceforge.net/projects/refind/files/${pkgver}/${pkgname}-src-${pkgver}.tar.gz")
-sha512sums=('e671d44385297295c143824dd95593f2e7bc8e282745d127b041a645aa27003467a3a23f368fd30297f33faa06bd914540d7bdc3c71ec42f6b7a38d7ee8e0ddb')
-b2sums=('2c3b1daf5fa100dcd4e1c801aa625e269ae9c1a99b567a271d7cdd30f95fa1e041e22984623b9a0e61f8af45966919f4b9f3a5b6c2026271d1bc37c6925f1e70')
-_arch='x64'
+source=("https://sourceforge.net/projects/refind/files/${pkgver}/${pkgname}-src-${pkgver}.tar.gz"
+        "add_riscv_support.patch"
+        "remove_multiple_definition.patch")
+sha512sums=('e671d44385297295c143824dd95593f2e7bc8e282745d127b041a645aa27003467a3a23f368fd30297f33faa06bd914540d7bdc3c71ec42f6b7a38d7ee8e0ddb'
+            '7ca4bcd8f88315a5caa2a665e33ca837c561ec614ebe2bfe6a772f078d01dc16ae363360175a7e24f62172617ec9d66580236e24d74cea925fd146a3536db37a'
+            '5a4bfabc5134c0fdd34dd1309dcce48d5aaf23f39e57611d42a8fc2d9ef483caa2d6738a4900e4c7ee155ab581d949bdaf2b127cc224a455781372c3559223b3')
+b2sums=('2c3b1daf5fa100dcd4e1c801aa625e269ae9c1a99b567a271d7cdd30f95fa1e041e22984623b9a0e61f8af45966919f4b9f3a5b6c2026271d1bc37c6925f1e70'
+        '9f3d0591e39047471476f636d01fb4443a4fc572488398e161b05340c11553dcae4f7a720d72a330478c10012906a8e01f44c038b43802ad480c7fbacea29f28'
+        '9b3ebb717aaa90d3c5a7d0dc8ee2784a5a706ca1f3da8950454c957b294be9db9f6f47e0551513e50dda073e066edc103a73a18643d1cfead8c294ebeef1185c')
+_arch='riscv64'
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../add_riscv_support.patch
+  patch -Np1 -i ../remove_multiple_definition.patch
   # remove the path prefix from the css reference, so that the css can live
   # in the same directory
   sed -e 's|../Styles/||g' -i "docs/${pkgname}/"*.html
