--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,11 @@ sha256sums=('4f4222820f97ca08c7ea707e4c53e5a3556af4d8f1ab51e0da6ff1627ff433ab'
             'SKIP'
             '2d8d5a287ebaf857b76c5c3fb2da5f25c8c0c38995bb96291ba42d6d3d1d53ba')
 
+prepare() {
+  cd $pkgname-$pkgver
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} .
+}
+
 build() {
   cd $pkgname-$pkgver
   patch -p0 -i ../plotutils-2.6-libpng-1.5.patch
