diff --color -Naur caps-0.9.26-orig/configure.py caps-0.9.26/configure.py
--- caps-0.9.26-orig/configure.py	2014-11-10 18:34:29.000000000 +0800
+++ caps-0.9.26/configure.py	2021-12-16 20:31:00.004827968 +0800
@@ -39,6 +39,7 @@
 	os.system ("make clean")
 
 if __name__ == '__main__':
-	if we_have_sse(): CFLAGS += ('-msse', '-mfpmath=sse')
-	if we_have_ssse3(): CFLAGS += ('-msse3',)
+	# systemd-nspawn gives host cpuinfo, pretending we have sse
+	# if we_have_sse(): CFLAGS += ('-msse', '-mfpmath=sse')
+	# if we_have_ssse3(): CFLAGS += ('-msse3',)
 	store()
