--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,10 @@ license=(APACHE)
 depends=(podman bash flatpak)
 makedepends=(go shellcheck go-md2man bash-completion ninja git meson podman)
 _commit=f7c4c7dedb34be33333065296275869e77464400	#refs/tags/0.0.99.4^{}
-source=("git+$url#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+$url#commit=$_commit"
+        "add_dynamic_linker.patch::https://patch-diff.githubusercontent.com/raw/containers/toolbox/pull/1159.patch")
+sha256sums=('SKIP'
+            'a9178677c13d15559320ef8e02af8f6961b06d801db091b2c95f741384edd8c7')
 
 pkgver() {
   cd toolbox
@@ -25,7 +27,7 @@ prepare() {
   git cherry-pick -n 816a7cab9e2f02f2eb828062be54ffef2e3666dc
   git cherry-pick -n f555029304415a06bde410deb879cd2cc7dad444
   git cherry-pick -n fc5f568c5d82f4a16982268fa67092e52be91fbe
-    
+  patch -p1 -Ni "${srcdir}/add_dynamic_linker.patch"
 }
 
 build() {
