diff --git a/rocclr/include/top.hpp b/rocclr/include/top.hpp
--- a/rocclr/include/top.hpp
+++ b/rocclr/include/top.hpp
@@ -28,6 +28,8 @@
     defined(__x86__) || defined(__x86_64) || defined(__x86_64__) || defined(__amd64) ||            \
     defined(__amd64__) || defined(_M_X64) || defined(_M_AMD64)
 #define ATI_ARCH_X86
+#elif defined(__riscv)
+#define ATI_ARCH_RISCV
 #endif
 
 #if defined(ATI_ARCH_ARM)
diff --git a/rocclr/os/os_posix.cpp b/rocclr/os/os_posix.cpp
--- a/rocclr/os/os_posix.cpp
+++ b/rocclr/os/os_posix.cpp
@@ -679,6 +679,9 @@
 #elif defined(ATI_ARCH_ARM)
       "mov %0,sp"
       : "=r"(value)
+#elif defined(ATI_ARCH_RISCV)
+      "mv %0,sp"
+      : "=r"(value)
 #else
       ""
 #endif
