diff --git PKGBUILD PKGBUILD
index bc30e54..e76ce52 100644
--- PKGBUILD
+++ PKGBUILD
@@ -1,7 +1,7 @@
 # Maintainer: Torsten Keßler <tpkessler at archlinux dot org>
 # Contributor: acxz <akashpatel2008 at yahoo dot com>
 pkgbase=hip-runtime
-pkgname=(hip-runtime-amd hip-runtime-nvidia)
+pkgname=(hip-runtime-amd)
 pkgver=6.4.0
 pkgrel=1
 _pkgdesc="Heterogeneous Interface for Portability"
@@ -10,9 +10,8 @@ url='https://rocm.docs.amd.com/projects/HIP/en/latest/'
 license=('MIT')
 _amd_depends=('rocm-core' 'bash' 'perl' 'glibc' 'gcc-libs' 'numactl'
          'mesa' 'comgr' 'rocminfo' 'rocm-llvm' 'libelf' 'rocprofiler-register')
-_nvidia_depends=('cuda')
 makedepends=('git' 'cmake' 'python' 'python-cppheaderparser'
-             "${_amd_depends[@]}" "${_nvidia_depends[@]}")
+             "${_amd_depends[@]}")
 _tag="tag=rocm-$pkgver"
 # Common HIP dir (AMD or nVidia)
 _hip='https://github.com/ROCm/HIP'
@@ -31,6 +30,14 @@ sha256sums=('2044f7736ea777e258c2956c89a65bea62102eb1db94edd9e0414d44a93e3c93'
             'b23d1485533028ea7689d76ccb8a7b17b30765a04883e0330af22bc8a8e1acf9'
             '09f332aa964079647a9407406305f6159937c72c0bc2b675f841ba0ab1b63c43')
 
+prepare() {
+  cd "$srcdir/$pkgbase-clr"
+
+  patch -Np1 -i "../$pkgbase-clr-riscv-hard-float.patch"
+  patch -Np1 -i "../$pkgbase-clr-riscv-fence.patch"
+  patch -Np1 -i "../$pkgbase-clr-riscv-currentStackPtr.patch"
+}
+
 build() {
   local hipcc_common_args=(
     -Wno-dev
@@ -59,41 +66,16 @@ build() {
     -DHIP_CATCH_TEST=0
     -DCLR_BUILD_HIP=ON
     -DCLR_BUILD_OCL=OFF
+    -D__HIP_ENABLE_PCH=OFF
   )
   cmake "${hip_amd_args[@]}"
   cmake --build build-amd
-
-  local hipcc_nvidia_args=(
-    "${hipcc_common_args[@]}"
-    -B build-nvidia-hipcc
-    -D CMAKE_INSTALL_PREFIX=/usr
-  )
-  cmake "${hipcc_nvidia_args[@]}"
-  cmake --build build-nvidia-hipcc
-
-  local hip_nvidia_args=(
-    -Wno-dev
-    -S "$srcdir/$pkgbase-clr"
-    -B build-nvidia
-    -DCMAKE_BUILD_TYPE=None
-    -DCMAKE_INSTALL_PREFIX=/usr
-    -DHIP_PLATFORM=nvidia
-    -DHIP_COMMON_DIR="$srcdir/$pkgbase"
-    -DHIPCC_BIN_DIR="$srcdir/build-nvidia-hipcc"
-    -DHIPNV_DIR="$srcdir/$pkgbase-hipother/hipnv"
-    -DHIP_CATCH_TEST=0
-    -DCLR_BUILD_HIP=ON
-    -DCLR_BUILD_OCL=OFF
-  )
-  cmake "${hip_nvidia_args[@]}"
-  cmake --build build-nvidia
 }
 
 package_hip-runtime-amd() {
   pkgdesc="$_pkgdesc (AMD runtime)"
   depends=("${_amd_depends[@]}")
-  optdepends=('inetutils: Print hostname in hipconfig'
-              'cuda: Cross compile for nvidia')
+  optdepends=('inetutils: Print hostname in hipconfig')
   replaces=("hip")
   provides=("hip=${pkgver}")
   DESTDIR="$pkgdir" cmake --install build-amd
@@ -106,3 +88,10 @@ package_hip-runtime-nvidia() {
   DESTDIR="$pkgdir" cmake --install build-nvidia
   install -Dm644 "$srcdir/$pkgbase/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("$pkgbase-clr-riscv-hard-float.patch"
+         "$pkgbase-clr-riscv-fence.patch"
+         "$pkgbase-clr-riscv-currentStackPtr.patch::https://github.com/ROCm/clr/pull/117.diff")
+sha256sums+=('SKIP'
+             'SKIP'
+             'df7a04af6cc8fd9145f50d56e0d6b4c4eb78a569227af1f6106b61bc322ca4c7')
