--- PKGBUILD	2026-04-24 01:04:33.516123078 +0800
+++ PKGBUILD.riscv2	2026-04-24 01:22:32.830808448 +0800
@@ -3,7 +3,7 @@
 # Contributor: acxz <akashpatel2008 at yahoo dot com>
 
 pkgbase=hip-runtime
-pkgname=(hip-runtime-amd hip-runtime-nvidia)
+pkgname=(hip-runtime-amd)
 pkgver=7.2.2
 pkgrel=1
 _pkgdesc="Heterogeneous Interface for Portability"
@@ -12,9 +12,8 @@
 license=('MIT')
 _amd_depends=('rocm-core' 'bash' 'perl' 'glibc' 'libgcc' 'numactl'
          'mesa' 'comgr' 'rocminfo' 'rocm-llvm' 'libelf' 'rocprofiler-register')
-_nvidia_depends=('cuda')
 makedepends=('git' 'cmake' 'python' 'python-cppheaderparser'
-             "${_amd_depends[@]}" "${_nvidia_depends[@]}")
+             "${_amd_depends[@]}")
 _tag="tag=rocm-$pkgver"
 # HIPCC compiler wrapper
 _hipcc='https://github.com/ROCm/llvm-project'
@@ -26,6 +25,14 @@
 
 options=(!lto)
 
+prepare() {
+  cd "$srcdir/$_projectBaseDir/clr"
+
+  patch -Np1 -i "$srcdir/$pkgbase-clr-riscv-hard-float.patch"
+  patch -Np1 -i "$srcdir/$pkgbase-clr-riscv-fence.patch"
+  patch -Np1 -i "$srcdir/$pkgbase-clr-riscv-currentStackPtr.patch"
+}
+
 build() {
   local hipcc_common_args=(
     -Wno-dev
@@ -54,50 +61,26 @@
     -DHIP_CATCH_TEST=0
     -DCLR_BUILD_HIP=ON
     -DCLR_BUILD_OCL=OFF
+    -D__HIP_ENABLE_PCH=OFF
   )
   cmake "${hip_amd_args[@]}"
   cmake --build build-amd
-
-  local hipcc_nvidia_args=(
-    "${hipcc_common_args[@]}"
-    -B build-nvidia-hipcc
-    -D CMAKE_INSTALL_PREFIX=/usr
-  )
-  cmake "${hipcc_nvidia_args[@]}"
-  cmake --build build-nvidia-hipcc
-
-  local hip_nvidia_args=(
-    -Wno-dev
-    -S "$srcdir/$_projectBaseDir/clr"
-    -B build-nvidia
-    -DCMAKE_BUILD_TYPE=None
-    -DCMAKE_INSTALL_PREFIX=/usr
-    -DHIP_PLATFORM=nvidia
-    -DHIP_COMMON_DIR="$srcdir/$_projectBaseDir/hip"
-    -DHIPCC_BIN_DIR="$srcdir/build-nvidia-hipcc"
-    -DHIPNV_DIR="$srcdir/$_projectBaseDir/hipother/hipnv"
-    -DHIP_CATCH_TEST=0
-    -DCLR_BUILD_HIP=ON
-    -DCLR_BUILD_OCL=OFF
-  )
-  cmake "${hip_nvidia_args[@]}"
-  cmake --build build-nvidia
 }
 
 package_hip-runtime-amd() {
   pkgdesc="$_pkgdesc (AMD runtime)"
   depends=("${_amd_depends[@]}")
-  optdepends=('inetutils: Print hostname in hipconfig'
-              'cuda: Cross compile for nvidia')
+  optdepends=('inetutils: Print hostname in hipconfig')
   replaces=("hip")
   provides=("hip=${pkgver}")
   DESTDIR="$pkgdir" cmake --install build-amd
   install -Dm644 "$srcdir/$_projectBaseDir/hip/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
-package_hip-runtime-nvidia() {
-  pkgdesc="$_pkgdesc (Nvidia runtime)"
-  depends=("${_nvidia_depends[@]}")
-  DESTDIR="$pkgdir" cmake --install build-nvidia
-  install -Dm644 "$srcdir/$_projectBaseDir/hip/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
-}
+
+source+=("$pkgbase-clr-riscv-hard-float.patch"
+         "$pkgbase-clr-riscv-fence.patch"
+         "$pkgbase-clr-riscv-currentStackPtr.patch")
+sha256sums+=('88c1531459b25e7c8903ef13c81e6f1ec72fcce5dfeac5af657db3fc2f200086'
+             '33c026a400d7258a7ded2de53dc1282f7edcde6935340d8fb59b0c7758781db2'
+             '667d8cbddb62b2f85c855967b97f9bac5bf720a1d08018002608a740488a0bc7')
