diff --git PKGBUILD PKGBUILD
index 516c09d..e0677df 100644
--- PKGBUILD
+++ PKGBUILD
@@ -3,7 +3,7 @@
 # Contributor: acxz <akashpatel2008 at yahoo dot com>
 
 pkgbase=hip-runtime
-pkgname=(hip-runtime-amd hip-runtime-nvidia)
+pkgname=(hip-runtime-amd)
 pkgver=7.1.1
 pkgrel=1
 _pkgdesc="Heterogeneous Interface for Portability"
@@ -12,9 +12,8 @@ url='https://rocm.docs.amd.com/projects/HIP/en/latest/'
 license=('MIT')
 _amd_depends=('rocm-core' 'bash' 'perl' 'glibc' 'gcc-libs' 'numactl'
          'mesa' 'comgr' 'rocminfo' 'rocm-llvm' 'libelf' 'rocprofiler-register')
-_nvidia_depends=('cuda')
 makedepends=('git' 'cmake' 'python' 'python-cppheaderparser'
-             "${_amd_depends[@]}" "${_nvidia_depends[@]}")
+             "${_amd_depends[@]}")
 _tag="tag=rocm-$pkgver"
 # HIPCC compiler wrapper
 _hipcc='https://github.com/ROCm/llvm-project'
@@ -26,6 +25,14 @@ _projectBaseDir="rocm-systems-rocm-$pkgver/projects"
 
 options=(!lto)
 
+prepare() {
+  cd "$srcdir/$_projectBaseDir/clr"
+
+  patch -Np1 -i "$srcdir/$pkgbase-clr-riscv-hard-float.patch"
+  patch -Np3 -i "$srcdir/$pkgbase-clr-riscv-fence.patch"
+  patch -Np1 -i "$srcdir/$pkgbase-clr-riscv-currentStackPtr.patch"
+}
+
 build() {
   local hipcc_common_args=(
     -Wno-dev
@@ -54,41 +61,16 @@ build() {
     -DHIP_CATCH_TEST=0
     -DCLR_BUILD_HIP=ON
     -DCLR_BUILD_OCL=OFF
+    -D__HIP_ENABLE_PCH=OFF
   )
   cmake "${hip_amd_args[@]}"
   cmake --build build-amd
-
-  local hipcc_nvidia_args=(
-    "${hipcc_common_args[@]}"
-    -B build-nvidia-hipcc
-    -D CMAKE_INSTALL_PREFIX=/usr
-  )
-  cmake "${hipcc_nvidia_args[@]}"
-  cmake --build build-nvidia-hipcc
-
-  local hip_nvidia_args=(
-    -Wno-dev
-    -S "$srcdir/$_projectBaseDir/clr"
-    -B build-nvidia
-    -DCMAKE_BUILD_TYPE=None
-    -DCMAKE_INSTALL_PREFIX=/usr
-    -DHIP_PLATFORM=nvidia
-    -DHIP_COMMON_DIR="$srcdir/$_projectBaseDir/hip"
-    -DHIPCC_BIN_DIR="$srcdir/build-nvidia-hipcc"
-    -DHIPNV_DIR="$srcdir/$_projectBaseDir/hipother/hipnv"
-    -DHIP_CATCH_TEST=0
-    -DCLR_BUILD_HIP=ON
-    -DCLR_BUILD_OCL=OFF
-  )
-  cmake "${hip_nvidia_args[@]}"
-  cmake --build build-nvidia
 }
 
 package_hip-runtime-amd() {
   pkgdesc="$_pkgdesc (AMD runtime)"
   depends=("${_amd_depends[@]}")
-  optdepends=('inetutils: Print hostname in hipconfig'
-              'cuda: Cross compile for nvidia')
+  optdepends=('inetutils: Print hostname in hipconfig')
   replaces=("hip")
   provides=("hip=${pkgver}")
   DESTDIR="$pkgdir" cmake --install build-amd
@@ -101,3 +83,10 @@ package_hip-runtime-nvidia() {
   DESTDIR="$pkgdir" cmake --install build-nvidia
   install -Dm644 "$srcdir/$_projectBaseDir/hip/LICENSE.md" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("$pkgbase-clr-riscv-hard-float.patch"
+         "$pkgbase-clr-riscv-fence.patch"
+         "$pkgbase-clr-riscv-currentStackPtr.patch::https://github.com/ROCm/clr/pull/117.diff")
+sha256sums+=('SKIP'
+             'SKIP'
+             '933ddbc531a652a3ada6322df6eb0d8adcedc214ca01bd882fc82c704c69cdc8')
