--- a/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc
+++ b/third_party/wasm2c/src/prebuilt/wasm2c_simd_source_declarations.cc
@@ -16,10 +16,14 @@
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"w"(var));
 )w2c_template"
-R"w2c_template(#elif defined(__riscv)
+R"w2c_template(#elif defined(__riscv_vector)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"vr"(var));
 )w2c_template"
+R"w2c_template(#elif defined(__riscv)
+)w2c_template"
+R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"m"(var));
+)w2c_template"
 R"w2c_template(#elif defined(__s390x__)
 )w2c_template"
 R"w2c_template(#define SIMD_FORCE_READ(var) __asm__("" ::"d"(var));
