Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 465075)
+++ PKGBUILD	(working copy)
@@ -9,8 +9,10 @@
 license=('custom')
 depends=('ncurses')
 makedepends=('git' 'cmake' 'ninja' 'python')
-source=("directx-shader-compiler-${pkgver}::git+https://github.com/microsoft/DirectXShaderCompiler.git#tag=v${pkgver}")
-sha256sums=('SKIP')
+source=("directx-shader-compiler-${pkgver}::git+https://github.com/microsoft/DirectXShaderCompiler.git#tag=v${pkgver}"
+        "riscv.patch::https://github.com/microsoft/DirectXShaderCompiler/commit/a0b6fc1028b673228e61dba413606de5d4b4c458.patch")
+sha256sums=('SKIP'
+            'c2d8258d79956d995258e437ad8845fca2aab4ef903225e3d9e5ced9808ae753')
 
 prepare() {
   cd "${pkgname}-${pkgver}"
@@ -18,6 +20,8 @@
   git submodule update --init
 
   rm -rf build && mkdir build
+
+  patch -Np1 -i ../riscv.patch
 }
 
 build() {
