Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 456835)
+++ PKGBUILD	(working copy)
@@ -10,8 +10,10 @@
 depends=('ncurses')
 makedepends=('git' 'cmake' 'ninja' 'python')
 _commit=de70ea29bc8734624c71e0078fbd8d532d20bdea
-source=("directx-shader-compiler::git+https://github.com/microsoft/DirectXShaderCompiler.git#commit=${_commit}")
-sha256sums=('SKIP')
+source=("directx-shader-compiler::git+https://github.com/microsoft/DirectXShaderCompiler.git#commit=${_commit}"
+        "0001-Port-to-RISC-V.patch")
+sha256sums=('SKIP'
+            '28f210b7895b16764471032908ea77e54d411480bfc37b88656d570b9295d334')
 
 pkgver() {
   cd "${pkgname}"
@@ -25,6 +27,8 @@
   git submodule update --init
 
   rm -rf build && mkdir build
+
+  patch -Np1 -i ../0001-Port-to-RISC-V.patch
 }
 
 build() {
