--- PKGBUILD
+++ PKGBUILD
@@ -22,18 +22,22 @@ source=(
   https://brltty.app/archive/$pkgname-$pkgver.tar.bz2
   $pkgname-6.2-systemd_sysusers_groups.patch
   $pkgname-6.4-x11_autostart.patch
+  $pkgname-ac-riscv64.patch::https://github.com/brltty/brltty/pull/389.patch
 )
 sha512sums=('7aef484f8f719fe183ea4d83b75962ff1e1fb0002d38ee89bd41808c3cae768cbd65c63cbdcb9a27beecaddc442fb65d8fb9219e48c87a1a02dda4ecc55cc24f'
             '32ba91271e2247b4a330cd213ed75b591268cb99a79c2efd9ae675804faee027c6b2f782768cb2329a65fc914ca2400b2901f35ce1fc2522c6691b343799eb02'
-            '4871512affefbc178f4204a1b285fc2b5a05ea2d181163195d695b760e9729b3d2d00b5f052abd71379df609c3859d7cbd64128bdefd16e898bbc4368500a9a0')
+            '4871512affefbc178f4204a1b285fc2b5a05ea2d181163195d695b760e9729b3d2d00b5f052abd71379df609c3859d7cbd64128bdefd16e898bbc4368500a9a0'
+            '4a162069cefdb0737854b870a0070ec213a8261f51ce787e0204a0071871a6fa3aeeb5fa1b54d5e915e0b98d1306672a8ec2142738732c90ff64747e013a444e')
 b2sums=('9d431c033d0963f6afe59477d2d109029c1bededf4ea333a3347ff8c4c17073af3439a827ff4db44dce08dd2245e1bc57cc2fe443bf44df6225f75ffab2fa244'
         '036e36d558594bb06af1eb41ff5fc9ae52ccf4bad54556c3f4f81a5e2f31f574039835e5e756455527c327a73d563a3db54a9f32e3478545a2a22c2725aaeee2'
-        '4ebc07a725ef8362233a83118e93901e78943e8dae08f9358b668ff13ab88a65eb9e87c49d106a8c3d87eb62007b230e199107eacb01f92dc683335076c01309')
+        '4ebc07a725ef8362233a83118e93901e78943e8dae08f9358b668ff13ab88a65eb9e87c49d106a8c3d87eb62007b230e199107eacb01f92dc683335076c01309'
+        'e569a9ec00f0b035a2b458d4c365a7980480ffe53cb33192db0e187633b4d56b32ea4d343899b901f321d2e358300c2fc526f71d505925ed211dad7b6fcc050f')
 
 prepare() {
   cd $pkgbase-$pkgver
   patch -Np1 -i ../$pkgbase-6.2-systemd_sysusers_groups.patch
   patch -Np1 -i ../$pkgbase-6.4-x11_autostart.patch
+  patch -Np1 -i ../$pkgbase-ac-riscv64.patch
   ./autogen
 }
 
