--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,7 @@ depends=(
 )
 makedepends=(texlive-binextra)
 provides=(
-  libunwind-{coredump,ptrace,setjmp,x86_64}.so
+  libunwind-{coredump,ptrace,setjmp,riscv}.so
   libunwind.so
 )
 source=(
@@ -42,6 +42,7 @@ build() {
   )
 
   cd libunwind-$pkgver
+  autoreconf -fi
   ./configure "${configure_options[@]}"
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
