--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,6 @@ pkgname=(
   uwsgi-plugin-php
   uwsgi-plugin-php-legacy
   uwsgi-plugin-lua51
-  uwsgi-plugin-mono
   uwsgi-plugin-webdav
   uwsgi-plugin-zabbix
   uwsgi-plugin-notfound
@@ -35,7 +34,6 @@ makedepends=(
   libxcrypt
   libxml2
   lua51
-  mono
   openssl
   pam
   pcre2
@@ -250,20 +248,6 @@ package_uwsgi-plugin-php-legacy() {
   install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
-package_uwsgi-plugin-mono() {
-  pkgdesc+=" (mono plugin)"
-  depends=(
-    glibc
-    mono
-    uwsgi
-  )
-
-  cd $pkgbase-$pkgver
-  install -vDm 755 plugins/mono/uwsgi.dll -t "$pkgdir"/usr/lib/mono/2.0/
-  install -vDm 755 mono_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
-  install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
-}
-
 package_uwsgi-plugin-webdav() {
   pkgdesc+=" (WebDav plugin)"
   depends=(
@@ -297,3 +281,7 @@ package_uwsgi-plugin-notfound() {
   install -vDm 755 notfound_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
   install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+sha512sums[1]='ad32f18a6da01eb0ba06b04109f5f61ab27871dc87d899361ae7f3d1e67756d66484cdb15dbefb36668212dcd5704a2d6044be57ef79c096d7a828c9f166a00a'
+b2sums[1]='976603b37534903fb60f752c6210239fef3572ebc88f7fe593e27c040b2243fd87323c71673e1bbcff08c0a7b4730112a5be8352381f4b4836dab0a9e83cf1be'
+
diff --git archlinux.ini archlinux.ini
index f3d361c..47f2e93 100644
--- archlinux.ini
+++ archlinux.ini
@@ -1,6 +1,6 @@
 [uwsgi]
 plugin_build_dir = .
 plugin_dir = /usr/lib/uwsgi
-plugins = asyncio, cgi, fiber, gevent, greenlet, php, php_legacy, lua, mono, notfound, psgi, pypy, python, rack, rbthreads, webdav, zabbix
+plugins = asyncio, cgi, fiber, gevent, greenlet, php, php_legacy, lua, notfound, psgi, pypy, python, rack, rbthreads, webdav, zabbix
 main_plugin = systemd_logger,pam
 inherit = base
