--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,9 @@ pkgname=(
   uwsgi-plugin-psgi
   uwsgi-plugin-cgi
   uwsgi-plugin-python
-  uwsgi-plugin-pypy
   uwsgi-plugin-php
   uwsgi-plugin-php-legacy
   uwsgi-plugin-lua51
-  uwsgi-plugin-mono
   uwsgi-plugin-webdav
   uwsgi-plugin-zabbix
   uwsgi-plugin-notfound
@@ -34,14 +32,12 @@ makedepends=(
   libxcrypt
   libxml2
   lua51
-  mono
   openssl
   pam
   pcre
   perl
   php-embed
   php-legacy-embed
-  pypy
   python
   python-gevent
   python-greenlet
@@ -63,7 +59,7 @@ source=(
   uwsgi.tmpfiles
 )
 sha512sums=('28aaf786a88f034752ce980f6ae04ee5ea361443ffad7d7e9ec9a88a60dd63c649a44c2d41bb6ddc1572c321567da4c3985f907498bf250e2cd7cd11f4686303'
-            '912b2d85cee0907454f2b1eee79fdeb7d7205c7b6a9f2a05cef969205fd4f3ccb9f3260e71a1c19cc910f6408e1e94da93fcca9978e8432273659b7caed34d9a'
+            '5f93930bab659135f9b38e5287c26dac519a710fe26c58230c31fb5498c5be9c629e56d0313699101a870c85d206c9a5aedb594d30018cdc33f8ce1a6572542e'
             'd54c84838ddb2f389e115ebb81aa2583705c9e330f020e6583b496f9c271ed236b6820c2a065f2b55a79adbf13e262b9ff2428124a8044b8fa20ca29ca4930ac'
             '068ada6281fdeb504fed7e47dae2729d047ed810bb596e5268c1821ad275cd32110e801e93e7ba0ee2d92dbc39e1954180cf43d88d2990d42b802ceb54b50a9e'
             '592c85270f43b37e9c51aae126533830515b96fff9fcae969ee865fe173bd0550c7729530d5ff1e44ffee43cd5c78c96688ce091c677a4196379e24983c0180e'
@@ -73,7 +69,7 @@ sha512sums=('28aaf786a88f034752ce980f6ae04ee5ea361443ffad7d7e9ec9a88a60dd63c649a
             '9e9eab08199cf08810ec95b0697cd8817226986d1e3aeb54845c0140ea5887360580348f295060c0558fb2d6aaed26fec929ea36b28cfeaf2b3588ce40fec3df'
             '4def9dcd06cfc2c6ce554add9d9545a01f3bb2681f0a2a6fc4ba68f91011111803a955cc41b5e70832ca448b196109368fdf874e81b7ec0edee4b51f864e16f7')
 b2sums=('62d9b2fdbd597fbeb6d1a9652688075912510b25cacbcd8210176e7663a0bc9a99e9ab8a298ff56227a07fd7faebe6e27f06242abe987b46193af377c947100f'
-        '41a9213d55665c905f71f07c752a74fc7685b2f191381d24f3e1d9691a8dae538aa4344ebe771ffd77ba3c9cd7c52fd9eff54ff876bc399d92256cb86f5b7d53'
+        '00f0cd49e51b2283b027c9ad344bf44fc21d71a921e91e9ffaa9c34e0dadc50be39e0a65398c8c6f53960e3d63d60a64be806c2d107fe3c1b9ba557e9ebc3aac'
         '835a1036422e9821cd52ffdf5a6053bdd59fc700d473600bda60b6730746f6c0750c30c85f4139d3ea81c6b6dd10d6849f9406decee3179861c51aca90da6161'
         '80aa3c4d3f0310f5029ac01690d2ba718d8c86fa49cd1b4f3fc1bd5ae9eadaed6562bf0a48a0fd611e92f100a8594e08fb4bae58e25765fc2eb648cdf1c60b35'
         'b6838c0c91811d4b677bf0f558134c69015708c308ef7080df328f12b0b0d58ae6cdebe277c29b6d11c20cfeb0847a4934e3a8924323e57ed661d6e9f6656a34'
@@ -169,18 +165,6 @@ package_uwsgi-plugin-python() {
   install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
-package_uwsgi-plugin-pypy() {
-  pkgdesc+=" (PyPy plugin)"
-  depends=(glibc pypy uwsgi)
-
-  cd $pkgbase-$pkgver
-  install -vDm 755 pypy_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
-  install -vDm 644 uwsgidecorators.py -t "$pkgdir"/opt/pypy/site-packages/
-  pypy -m compileall "$pkgdir"/opt/pypy/site-packages/
-  pypy -O -m compileall "$pkgdir"/opt/pypy/site-packages/
-  install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
-}
-
 package_uwsgi-plugin-lua51() {
   pkgdesc+=" (LUA plugin)"
   depends=(glibc lua51 uwsgi)
@@ -208,16 +192,6 @@ package_uwsgi-plugin-php-legacy() {
   install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
-package_uwsgi-plugin-mono() {
-  pkgdesc+=" (mono plugin)"
-  depends=(glibc mono uwsgi)
-
-  cd $pkgbase-$pkgver
-  install -vDm 755 plugins/mono/uwsgi.dll -t "$pkgdir"/usr/lib/mono/2.0/
-  install -vDm 755 mono_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
-  install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
-}
-
 package_uwsgi-plugin-webdav() {
   pkgdesc+=" (WebDav plugin)"
   depends=(glibc libxml2 uwsgi)
diff --git archlinux.ini archlinux.ini
index f3d361c..4857c73 100644
--- archlinux.ini
+++ archlinux.ini
@@ -1,6 +1,6 @@
 [uwsgi]
 plugin_build_dir = .
 plugin_dir = /usr/lib/uwsgi
-plugins = asyncio, cgi, fiber, gevent, greenlet, php, php_legacy, lua, mono, notfound, psgi, pypy, python, rack, rbthreads, webdav, zabbix
+plugins = asyncio, cgi, fiber, gevent, greenlet, php, php_legacy, lua, notfound, psgi, python, rack, rbthreads, webdav, zabbix
 main_plugin = systemd_logger,pam
 inherit = base
