--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,9 @@ pkgname=(
   uwsgi-plugin-psgi
   uwsgi-plugin-cgi
   uwsgi-plugin-python
-  uwsgi-plugin-pypy
   uwsgi-plugin-php
   uwsgi-plugin-php7
   uwsgi-plugin-lua51
-  uwsgi-plugin-mono
   uwsgi-plugin-webdav
   uwsgi-plugin-zabbix
   uwsgi-plugin-notfound
@@ -34,14 +32,12 @@ makedepends=(
   libxcrypt
   libxml2
   lua51
-  mono
   openssl
   pam
   pcre
   perl
   php-embed
   php7-embed
-  pypy
   python
   python-gevent
   python-greenlet
@@ -65,7 +61,7 @@ source=(
   $pkgbase-1.9.13-ruby2.0.patch
 )
 sha512sums=('36a9c1b87a4c3d08482b9045d2227f56b006acb53f38ddf1b510880ae5fc24c0177a077338ec8af3ef0b8f8e220bc4fc7f8311dab8066e13cbcbb616f736c795'
-            '51c474a5ae025c0fae17dec309bea29550a8ca39815b7495189394ae5ce30eb264f77a45f48f9f7529c59d907baf139cdf7c488d1b13fcb862cc7b9bdc268804'
+            '2fdd81de1d4a9baf01c5a6f5958017263a0320fd091b666beb7b25f3e5e0e80567cbb3953cebf887f408291aaa1cfb1260ee1adef25be34b1400b51b9e1acf2b'
             'd54c84838ddb2f389e115ebb81aa2583705c9e330f020e6583b496f9c271ed236b6820c2a065f2b55a79adbf13e262b9ff2428124a8044b8fa20ca29ca4930ac'
             '068ada6281fdeb504fed7e47dae2729d047ed810bb596e5268c1821ad275cd32110e801e93e7ba0ee2d92dbc39e1954180cf43d88d2990d42b802ceb54b50a9e'
             '592c85270f43b37e9c51aae126533830515b96fff9fcae969ee865fe173bd0550c7729530d5ff1e44ffee43cd5c78c96688ce091c677a4196379e24983c0180e'
@@ -76,7 +72,7 @@ sha512sums=('36a9c1b87a4c3d08482b9045d2227f56b006acb53f38ddf1b510880ae5fc24c0177
             '4def9dcd06cfc2c6ce554add9d9545a01f3bb2681f0a2a6fc4ba68f91011111803a955cc41b5e70832ca448b196109368fdf874e81b7ec0edee4b51f864e16f7'
             '162ca1be96282b32e8e6be919b6315bfd08954c491056958985bf99e7f5bdad3511665f059329d6cb6ef5d222f9aa11d4dc43c72e6a963ab941ada959fe8d964')
 b2sums=('3c5fc3ce3a6a7f8f915bb302c0b775a75227f859dd2b141034e8b1aa5cb867b447ed532b3e8b537db80acc908fc26121b752e35ebc0bf9a542678843fd02ff54'
-        '46f833e5d179d950bcbab51b539e6431959c9e3618c0d66b791a4b4d866b6195707704b394224b4fea97e1f65188b8b474499d70e5c2bb44da32c60447b80806'
+        '16e0f25e7f9c423f61114bcbb98b5f94250539764375f626e8da2ed662715373461274f4125e5d2390c4b512762e8dad89b902bcd87dff30113c59216c3b72c4'
         '835a1036422e9821cd52ffdf5a6053bdd59fc700d473600bda60b6730746f6c0750c30c85f4139d3ea81c6b6dd10d6849f9406decee3179861c51aca90da6161'
         '80aa3c4d3f0310f5029ac01690d2ba718d8c86fa49cd1b4f3fc1bd5ae9eadaed6562bf0a48a0fd611e92f100a8594e08fb4bae58e25765fc2eb648cdf1c60b35'
         'b6838c0c91811d4b677bf0f558134c69015708c308ef7080df328f12b0b0d58ae6cdebe277c29b6d11c20cfeb0847a4934e3a8924323e57ed661d6e9f6656a34'
@@ -171,17 +167,6 @@ package_uwsgi-plugin-python() {
   python -O -m compileall "${pkgdir}${site_packages}"
 }
 
-package_uwsgi-plugin-pypy() {
-  pkgdesc+=" (PyPy plugin)"
-  depends=(glibc pypy uwsgi)
-
-  cd $pkgbase-$pkgver
-  install -vDm 755 pypy_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
-  install -vDm 644 uwsgidecorators.py -t "$pkgdir"/opt/pypy/site-packages/
-  pypy -m compileall "$pkgdir"/opt/pypy/site-packages/
-  pypy -O -m compileall "$pkgdir"/opt/pypy/site-packages/
-}
-
 package_uwsgi-plugin-lua51() {
   pkgdesc+=" (LUA plugin)"
   depends=(glibc lua51 uwsgi)
@@ -206,15 +191,6 @@ package_uwsgi-plugin-php7() {
   install -vDm 755 php7_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
 }
 
-package_uwsgi-plugin-mono() {
-  pkgdesc+=" (mono plugin)"
-  depends=(glibc mono uwsgi)
-
-  cd $pkgbase-$pkgver
-  install -vDm 755 plugins/mono/uwsgi.dll -t "$pkgdir"/usr/lib/mono/2.0/
-  install -vDm 755 mono_plugin.so -t "$pkgdir"/usr/lib/uwsgi/
-}
-
 package_uwsgi-plugin-webdav() {
   pkgdesc+=" (WebDav plugin)"
   depends=(glibc libxml2 uwsgi)
diff --git archlinux.ini archlinux.ini
index e93e11838..18fd3a758 100644
--- archlinux.ini
+++ archlinux.ini
@@ -1,6 +1,6 @@
 [uwsgi]
 plugin_build_dir = .
 plugin_dir = /usr/lib/uwsgi
-plugins = asyncio, cgi, fiber, gevent, greenlet, php, lua, mono, notfound, psgi, pypy, python, rack, rbthreads, webdav, zabbix
+plugins = asyncio, cgi, fiber, gevent, greenlet, php, lua, notfound, psgi, python, rack, rbthreads, webdav, zabbix
 main_plugin = systemd_logger,pam
 inherit = base
