diff --git PKGBUILD PKGBUILD
index bed4450..dc4f4a4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,15 +14,17 @@ depends=(packagekit-qt5 kcmutils appstream-qt qt5-xmlpatterns)
 makedepends=(extra-cmake-modules kdoctools plasma-workspace)
 optdepends=('plasma-workspace: Updates daemon')
 source=("https://download.kde.org/stable/$pkgname/$pkgver/$pkgname-$pkgver.tar.xz"{,.sig} apper-rpath.patch
-        apper-qt5.11.patch::"https://cgit.kde.org/apper.git/patch/?id=d8960ea9"
-        apper-qt5.14.patch::"https://cgit.kde.org/apper.git/patch/?id=1815d19f"
-        kdebug-401461.patch::"https://cgit.kde.org/apper.git/patch/?id=d486706f")
+        apper-qt5.11.patch::https://invent.kde.org/system/apper/-/commit/d8960ea9.patch
+        apper-qt5.14.patch::https://invent.kde.org/system/apper/-/commit/1815d19f.patch
+        kdebug-401461.patch::https://invent.kde.org/system/apper/-/commit/d486706f.patch
+        apper-qt5.15.patch::https://invent.kde.org/system/apper/-/commit/f2fb923e6920f567033e0954195e5c59e38aa164.patch)
 sha256sums=('1a30be92aab8bd258c2a8824f533c5646b934e06b4268edbd11724ea450f4923'
             'SKIP'
             '17ce82a3f9ba8888febc5d2b2dc63e5c73d7be4b216e9110f0ca73050be0010d'
-            '05d9f54d788256ec658cadec6571f3ec649ab3e82adb8daa2e7209aa9c1ff062'
-            '1275d8af4d61d50e3d39a6507395bb4142572720937e9ebfe9679c69b53ce402'
-            '2f66e12f40c68fb1b8c7a1d4fcf71adfdd4dceb9ecf0806c636fbf20d55e2154')
+            '667301c14bc6b4c83eaf78b5c5724f1a6cfe7f49aadd3a3fde2e3e1192f9374e'
+            'a662281b518929c5ae1d91bc829d60a02a7cf20fb434a1a3809017c5dc971312'
+            'fedc93b12289136b777d3b4441ce7997bea089c996463eec491328f2e6c1f390'
+            'b91e81fc476c443f6e20920476f5907e5e1695e5551595b42b866607ba2bcb9c')
 validpgpkeys=(70C26659D184ABC01FFAF45228DDEDC6E4480AD0) # Daniel Nicoletti <dantti12@gmail.com>
 
 prepare() {
@@ -33,6 +35,7 @@ prepare() {
   patch -p1 -i ../apper-qt5.11.patch # fix build with Qt 5.11
   patch -p1 -i ../apper-qt5.14.patch # fix build with Qt 5.14
   patch -p1 -i ../kdebug-401461.patch # fix crash
+  patch -p1 -i ../apper-qt5.15.patch # fix build with Qt 5.15
 }
 
 build() {
