--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@ backup=("etc/smartdns/smartdns.conf")
 depends=('gcc-libs' 'openssl')
 makedepends=('systemd')  # systemd.pc
 checkdepends=('gtest' 'bind')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/pymumu/smartdns/archive/Release$pkgver.tar.gz")
-sha512sums=('ab4ce7ac5fe05173a660f8dce61e7cc0ec8375ff7d64d08e854f73e5cfd78fec702220f076b75efe9518334a384b9689898d5c2856cc506d294b5325733bb9c6')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/pymumu/smartdns/archive/Release$pkgver.tar.gz"
+        "test-SpeedCheck-none.patch::https://github.com/pymumu/smartdns/pull/1538.patch")
+sha512sums=('ab4ce7ac5fe05173a660f8dce61e7cc0ec8375ff7d64d08e854f73e5cfd78fec702220f076b75efe9518334a384b9689898d5c2856cc506d294b5325733bb9c6'
+            '0177c8357dbd4d2834918f0e358c2abb0a1815324144a7d3772788525a9355c58335e541cba60fff21b9e48e4cc1d262b54cbd2f7581beed005b09e7c25b3caa')
 
 prepare() {
   # Unfortunately upstream does not provide a way to reuse .o files
+  patch -Np1 -d smartdns-Release$pkgver < test-SpeedCheck-none.patch
   cp -a smartdns-Release$pkgver smartdns-Release$pkgver-test
 }
 
