--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,15 @@ backup=("etc/smartdns/smartdns.conf")
 depends=('gcc-libs' 'openssl')
 makedepends=('systemd')  # systemd.pc
 checkdepends=('gtest' 'bind')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/pymumu/smartdns/archive/Release$pkgver.tar.gz")
-sha512sums=('ab4ce7ac5fe05173a660f8dce61e7cc0ec8375ff7d64d08e854f73e5cfd78fec702220f076b75efe9518334a384b9689898d5c2856cc506d294b5325733bb9c6')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/pymumu/smartdns/archive/Release$pkgver.tar.gz"
+        "test-SpeedCheck-none.patch")
+sha512sums=('ab4ce7ac5fe05173a660f8dce61e7cc0ec8375ff7d64d08e854f73e5cfd78fec702220f076b75efe9518334a384b9689898d5c2856cc506d294b5325733bb9c6'
+            'b098ebefd7f84d6090b7a52eb7817949c4c3f218b76444ef5242a6840ab9c32d6df42973d06e62d1976eef8f6067a9a1b831a017fc6db3d6472ef9a5bb7bf037')
 
 prepare() {
   # Unfortunately upstream does not provide a way to reuse .o files
+  patch -Np1 -d smartdns-Release$pkgver < test-SpeedCheck-none.patch
+  # former patch is https://github.com/pymumu/smartdns/pull/1538.patch
   cp -a smartdns-Release$pkgver smartdns-Release$pkgver-test
 }
 
