--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,7 @@
 
 prepare() {
   cd $_name
+  patch -Np1 -i "$srcdir/generator-lockfile.patch"
   export RUSTUP_TOOLCHAIN=stable
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
@@ -57,3 +58,7 @@
   install -vDm 644 $_name/target/shell_completions/_$pkgname -t "$pkgdir/usr/share/zsh/site-functions/"
   install -vDm 644 $_name/target/shell_completions/$pkgname.fish -t "$pkgdir/usr/share/fish/vendor_completions.d/"
 }
+
+source+=(generator-lockfile.patch)
+sha512sums+=('84e781297673a9178539a76b58db96ef00d8a498e6b597558015ff4559430fb7533a8db0bdcbbb8e6945ca7cab0a46492783b00406bc30f5ec4130e7f678797a')
+b2sums+=('3de251109010d5d71dd1fe3eea706f52ccdcbed849a46e33ac4bcf5498884d7415387debdc00c7854e9a76d051ffdb306afc2f3fbbe5d63e3a31b3b87a879c5f')
