diff --git PKGBUILD PKGBUILD
index ec31b36..15c2e4a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ arch=('any')
 license=('MIT')
 depends=('python' 'python-jinja' 'python-lxml' 'python-setuptools')
 makedepends=('python-build' 'python-installer' 'python-wheel')
-checkdepends=('python-pytest' 'xorgproto' 'libx11' 'glfw-x11' 'xorg-server-xvfb' 'mingw-w64-gcc' 'wine' 'rust' 'libxinerama')
+checkdepends=('python-pytest' 'xorgproto' 'libx11' 'glfw-x11' 'xorg-server-xvfb' 'rust' 'libxinerama')
 source=(${url}/archive/refs/tags/v${pkgver}/${pkgname}-${pkgver}.tar.gz)
 sha512sums=('ad7e0544fdd07ba7e091ed8a985deb27f05affb0a830b8132d7c03d6690b57139b25433303260e3affba7b11c0dad38358fcd4ff5b7b30e7ca193e8beed8db48')
 b2sums=('d18a9ee4445a603d1f277db24bca9e21bb87f919fa366ee307dba21ffbfef7be690c3e924ab405c54a9cd8dc1d487297586020a04e04afe48779fdb39f9b81dd')
