Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,7 @@
              'libxmu' 'libxrender' 'libxi' 'libxaw' 'libxtst' 'libxres'
              'xorg-xkbcomp' 'xorg-util-macros' 'xorg-font-util' 'libepoxy'
              'xcb-util' 'xcb-util-image' 'xcb-util-renderutil' 'xcb-util-wm' 'xcb-util-keysyms'
-             'libxshmfence' 'libunwind' 'systemd' 'wayland-protocols' 'egl-wayland' 'meson') # 'git')
+             'libxshmfence' 'systemd' 'wayland-protocols' 'egl-wayland' 'meson') # 'git')
 source=(https://xorg.freedesktop.org/releases/individual/xserver/${pkgbase}-${pkgver}.tar.bz2{,.sig}
         xserver-autobind-hotplug.patch
         0001-v2-FS-58644.patch
@@ -106,7 +106,7 @@
 
 package_xorg-server() {
   pkgdesc="Xorg X server"
-  depends=(libepoxy libxfont2 pixman xorg-server-common libunwind
+  depends=(libepoxy libxfont2 pixman xorg-server-common
            dbus libgl xf86-input-libinput nettle
            libpciaccess libdrm libxshmfence) # FS#52949
 
@@ -135,7 +135,7 @@
 
 package_xorg-server-xephyr() {
   pkgdesc="A nested X server that runs as an X application"
-  depends=(libxfont2 libgl libepoxy libunwind systemd-libs libxv pixman xorg-server-common
+  depends=(libxfont2 libgl libepoxy systemd-libs libxv pixman xorg-server-common
            xcb-util-image xcb-util-renderutil xcb-util-wm xcb-util-keysyms
            nettle libtirpc)
 
@@ -148,7 +148,7 @@
 
 package_xorg-server-xvfb() {
   pkgdesc="Virtual framebuffer X server"
-  depends=(libxfont2 libunwind pixman xorg-server-common xorg-xauth 
+  depends=(libxfont2 pixman xorg-server-common xorg-xauth 
            libgl nettle libtirpc)
 
   _install fakeinstall/usr/bin/Xvfb
@@ -174,7 +174,7 @@
 
 package_xorg-server-xwayland() {
   pkgdesc="run X clients under wayland"
-  depends=(libxfont2 libepoxy libunwind systemd-libs libgl pixman xorg-server-common
+  depends=(libxfont2 libepoxy systemd-libs libgl pixman xorg-server-common
            nettle libtirpc)
 
   _install fakeinstall/usr/bin/Xwayland
