--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,15 @@ source=(https://github.com/Exiv2/exiv2/archive/v${pkgver}/${pkgname}-${pkgver}.t
 sha512sums=('7d7121770a9394efd31acb74709583e587351fc48c485a2f684cacce4ed19a2d07a6cb1cea986c16bf64851fd53b22a7277c1e1ae1a6cd69a5761afae3c5c731')
 b2sums=('ce80a1f4252b7d12a3fb9b6e17d81703a02dfaf58cb096b5b6d65146e3c6470b7d1120430f2802b679e1648c1eac74a05381b2cf21d054758086b9ac0eacc556')
 
+source+=(exiv2-fix_test_failure.patch)
+sha512sums+=('e77c9963a2c4a02ae02ef6d56d2b678b48f33ed5ce77aff5b68aba79995573004510214f391fd06f55226c42d3d973aaf3320227999acb72d346a23676a986c4')
+b2sums+=('037735e95c4129f469e530ffa52f9d906fa416c5a54bb651e04bcccf9021291f276b8c1c1048f763027a9a030d05858067950e57553a4b4dd77d2c8146c40e56')
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  patch -Np1 < $srcdir/exiv2-fix_test_failure.patch
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   cmake -B build \
