diff --git PKGBUILD PKGBUILD
index b96eff4..4b2209b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,12 +9,16 @@ url='https://github.com/awslabs/aws-c-common'
 license=(Apache)
 depends=(glibc)
 makedepends=(cmake)
-source=(aws-c-common-$pkgver.zip::https://github.com/awslabs/aws-c-common/archive/v$pkgver.zip)
-sha256sums=('bce7be44bf327a3b666d6ac4e58288ad38c1f7561692c68dac62ffe28e682551')
+source=(aws-c-common-$pkgver.zip::https://github.com/awslabs/aws-c-common/archive/v$pkgver.zip
+        Wno-error-maybe-uninitialized.patch)
+sha256sums=('bce7be44bf327a3b666d6ac4e58288ad38c1f7561692c68dac62ffe28e682551'
+            'b139d87f463455d07559b4fde74ce665ee42300148121d81b3e33a50abae829c')
 
 build() {
   cd aws-c-common-$pkgver
 
+  patch -Np1 < ../Wno-error-maybe-uninitialized.patch
+
   cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SHARED_LIBS=ON -S . -B build
   cmake --build build
 }
