diff --git PKGBUILD PKGBUILD
index 0111bb43b..6f7a418f3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,7 +40,7 @@ prepare() {
   git submodule update
 
   # Only LDC have -O5, and add DC_FLAGS to linking
-  sed -i -e "s/-O5/-O5 -flto=full -link-defaultlib-shared=false -L=\"$LDFLAGS\"/g" -e 's/OBJ_BY_DC} \${WRITE_TO_TARGET_NAME}/OBJ_BY_DC} ${WRITE_TO_TARGET_NAME} ${DC_FLAGS}/' makefile
+  sed -i -e "s/-O5/-O5 -link-defaultlib-shared=false -L=\"$LDFLAGS\"/g" -e 's/OBJ_BY_DC} \${WRITE_TO_TARGET_NAME}/OBJ_BY_DC} ${WRITE_TO_TARGET_NAME} ${DC_FLAGS}/' makefile
 }
 
 build() {
