--- PKGBUILD
+++ PKGBUILD
@@ -42,7 +42,7 @@ prepare() {
   # Only LDC have -O5, and add DC_FLAGS to linking
   export _ldFlags="$(echo -ne $LDFLAGS | sed -e 's/-Wl,/-L=/g' -e 's/=auto/=full')"
   sed -i \
-      -e "s/-O5/-O5 -flto=full -link-defaultlib-shared=false $_ldFlags/g" \
+      -e "s/-O5/-O5 -link-defaultlib-shared=false $_ldFlags/g" \
       -e 's/OBJ_BY_DC} \${WRITE_TO_TARGET_NAME}/OBJ_BY_DC} ${WRITE_TO_TARGET_NAME} ${DC_FLAGS}/' \
       makefile
 }
