diff --git PKGBUILD PKGBUILD
index a5325d1..d77d38b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@
 url="https://pipewire.org"
 license=(MIT)
 arch=(x86_64)
-makedepends=(git meson doxygen graphviz python-docutils valgrind libpulse
+makedepends=(git meson doxygen graphviz python-docutils libpulse
              alsa-lib libusb gst-plugins-base rtkit dbus sdl2 ncurses libcanberra libsndfile
              bluez-libs sbc libldac libfreeaptx libfdk-aac avahi lilv
