Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -12,7 +12,7 @@
 url="https://pipewire.org"
 license=(MIT)
 arch=(x86_64)
-makedepends=(git meson doxygen xmltoman valgrind libpulse alsa-lib libusb
+makedepends=(git meson doxygen xmltoman libpulse alsa-lib libusb
              gst-plugins-base rtkit dbus sdl2 ncurses libsndfile bluez-libs sbc
              libldac libfreeaptx libfdk-aac avahi webrtc-audio-processing)
 checkdepends=(desktop-file-utils)
