--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ pkgdesc="Low-latency audio/video router and processor"
 url="https://pipewire.org"
 license=(MIT)
 arch=(x86_64)
-makedepends=(git meson doxygen graphviz python-docutils valgrind libpulse
+makedepends=(git meson doxygen graphviz python-docutils libpulse
              alsa-lib libusb gst-plugins-base rtkit dbus sdl2 ncurses
              libcanberra libsndfile bluez-libs sbc libldac libfreeaptx
              libfdk-aac avahi lilv webrtc-audio-processing)
