diff --git PKGBUILD PKGBUILD
index 4788ab0c..87973dfe 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ pkgdesc="Low-latency audio/video router and processor"
 url="https://pipewire.org"
 license=(MIT)
 arch=(x86_64)
-makedepends=(git meson doxygen graphviz python-docutils valgrind libpulse
+makedepends=(git meson doxygen graphviz python-docutils libpulse cmake
              alsa-lib libusb gst-plugins-base rtkit dbus sdl2 ncurses libsndfile
              bluez-libs sbc libldac libfreeaptx libfdk-aac avahi lilv
              webrtc-audio-processing)
