--- PKGBUILD
+++ PKGBUILD
@@ -67,7 +67,6 @@ build() {
                 --enable-checking=release \
                 --enable-clocale=gnu \
                 --enable-gnu-unique-object \
-                --enable-gold \
                 --enable-languages=c,c++ \
                 --enable-ld=default \
                 --enable-lto \
@@ -83,7 +82,7 @@ build() {
                 --with-gnu-as \
                 --with-gnu-ld \
                 --with-ld=/usr/bin/$_target-ld \
-                --with-plugin-ld=ld.gold \
+                --with-plugin-ld=ld.bfd \
                 --with-system-zlib \
                 --with-isl \
                 --enable-gnu-indirect-function
