--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,12 @@ validpgpkeys=('9456AFC0814A8139E99483517FE6B095B582B0D4') # Timothy Gu
 sha256sums=('1e572a0735b92aca5746c4528f9bebd35aa0ccf8619b22fa2756137a8cc9f912'
             'SKIP')
 
+prepare() {
+  cd schroedinger-${pkgver}
+  cp /usr/share/autoconf/build-aux/config.guess config.guess
+  cp /usr/share/autoconf/build-aux/config.sub config.sub
+}
+
 build() {
   cd schroedinger-${pkgver}
 
