diff --git PKGBUILD PKGBUILD
index 83cd4e1..d040129 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,6 +7,7 @@ pkgdesc="Farstream (formerly Farsight) - Audio/Video Communications Framework"
 arch=('x86_64')
 url="https://www.freedesktop.org/wiki/Software/Farstream"
 license=('LGPL')
+options=(!lto)
 depends=('gst-plugins-base-libs' 'libnice')
 makedepends=('gobject-introspection' 'python' 'gtk-doc')
 optdepends=('gst-plugins-good' 'gst-plugins-bad')
