--- PKGBUILD.orig	2021-12-14 16:01:23.794119881 +0000
+++ PKGBUILD	2021-12-14 16:44:30.590927436 +0000
@@ -13,17 +13,22 @@
 makedepends=(boost glu rtaudio rtmidi)
 source=("https://archive.org/download/dinisnoise_source_code/${pkgname}-${pkgver}.tar.gz"
         "${pkgname}-52-devendor_rtaudio_rtmidi.patch"
+	"fix-line-ce.patch"
 )
-sha512sums=('fef0e4dd6dbd691393b4211b9609879945b7359c62105559292b2cf04387012c295b6f03ef7c82309319a945927f5de0d3b5e476c516988519a59c10f8efc441'
-            '56aaa4f2e39297d3b64d254ddbf53ae7c7d0ea7df462a7f9ab3071ce3a21c997874b8a39065422b012a10e8ecd90b5373c887e924a8843134a6b095f588ffd26')
-b2sums=('52bdf721abcd0dfdf8ab678d25d2063e2eecfea5c5e66426e451bc07145543738be59d607a0a5c47d290d6ca5d6535bf8660af3eadd5794e3bfba1ede9b834dc'
-        'f38ddc9c92a025fe1e361e8c1e48fd2fced13e19894c117f5e79f00eb5444fd4861562895dbef9884bd9996e2248923b511e3879655bb6cb87a1242b7fbb6042')
+sha512sums=('c2ec779746d07ad6579aeebadff45e218fd29715ac446f00840b513d3f9c8d1ef79f556a1f5c7d931adb3451b547f58fb58d6da699170755ae62f25d70b95158'
+            '56aaa4f2e39297d3b64d254ddbf53ae7c7d0ea7df462a7f9ab3071ce3a21c997874b8a39065422b012a10e8ecd90b5373c887e924a8843134a6b095f588ffd26'
+            '7eb853f0cdde657acafec882405b0d80cba40b4853f4b55b26847d84a215fa8fcef032ba31a4cc870c0447fe6fb86f7f77c323e996bf2db1584955f300f01bf4')
+b2sums=('054d3857bed1b4f4affbeffa0fc5c27ef356f200a60f9d6c5145327311accc5f0b1c8deaf425b745ea4cd2c1accee1195b2e840e9125d105edf67e09062b8574'
+        'f38ddc9c92a025fe1e361e8c1e48fd2fced13e19894c117f5e79f00eb5444fd4861562895dbef9884bd9996e2248923b511e3879655bb6cb87a1242b7fbb6042'
+        '81ab450dc59b95bcb8c2b50c19c40af7de36d510cc8067fe720fc3ce4d678b85b4c68fadfba6d174f7a9f11220dbbd15ec91c4f7bd7a3049d3181cc6f5c94512')
 
 prepare() {
   cd "${pkgname}-${pkgver}"
   # use system rtaudio/rtmidi, instead of vendored versions
   patch -Np1 -i "../${pkgname}-52-devendor_rtaudio_rtmidi.patch"
   rm -fv src/{RtAudio,RtMidi}.*
+
+  patch -Np0 -d .. -i "fix-line-ce.patch"
   autoreconf -fiv
 }
 
