--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ _source_base=https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkg
 source=($_source_base/$pkgname-$pkgver.src.tar.xz{,.sig}
         $_source_base/clang-tools-extra-$pkgver.src.tar.xz{,.sig}
         $_source_base/llvm-$pkgver.src.tar.xz{,.sig}
+        llvm-D69869.patch::https://reviews.llvm.org/D69869?download=true
         enable-SSP-and-PIE-by-default.patch)
 sha256sums=('885b062b00e903df72631c5f98b9579ed1ed2790f74e5646b4234fa084eacb21'
             'SKIP'
@@ -27,6 +28,7 @@ sha256sums=('885b062b00e903df72631c5f98b9579ed1ed2790f74e5646b4234fa084eacb21'
             'SKIP'
             'df83a44b3a9a71029049ec101fb0077ecbbdf5fe41e395215025779099a98fdf'
             'SKIP'
+            '9a8565c61f685f8a3fffc4de10de66e0adece39535cc468d763f4781698b8da3'
             '9e557a18f3ca96cc5cc9b62d2e308b993025523b2aca0735248118a928c8c3ff')
 validpgpkeys+=('B6C8F98282B944E3B0D5C2530FC3042E345AD05D') # Hans Wennborg <hans@chromium.org>
 validpgpkeys+=('474E22316ABF4785A88C6E8EA2C794A986419D8A') # Tom Stellard <tstellar@redhat.com>
@@ -55,6 +57,7 @@ prepare() {
   mkdir build
   mv "$srcdir/clang-tools-extra-$pkgver.src" tools/extra
   patch -Np2 -i ../enable-SSP-and-PIE-by-default.patch
+  patch -d tools/extra -Np2 -i "$srcdir"/llvm-D69869.patch
 
   if [[ $pkgver != 10.* ]]; then
     echo "Remove libclangHandleLLVM hack!"
