--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,8 @@ source=($_source_base/clang-$pkgver.src.tar.xz{,.sig}
         $_source_base/llvm-$pkgver.src.tar.xz{,.sig}
         clang-coroutines-ubsan.patch
         clang-tidy-fix-standalone-build.patch
-        enable-fstack-protector-strong-by-default.patch)
+        enable-fstack-protector-strong-by-default.patch
+        add-R_RISCV_SUB6-relocation.patch::https://reviews.llvm.org/D120001?download=true)
 sha256sums=('2b5847b6a63118b9efe5c85548363c81ffe096b66c3b3675e953e26342ae4031'
             'SKIP'
             '7cf3b8ff56c65c4d1eae3c56883fc4a6cbc3ff9f3a1530a74d66e45d27271866'
@@ -31,7 +32,8 @@ sha256sums=('2b5847b6a63118b9efe5c85548363c81ffe096b66c3b3675e953e26342ae4031'
             'SKIP'
             '2c25ddf0ba6be01949842873fef4d285456321aaccd4ba95db61b69a4c580106'
             '081a7ebc1ae524b13fc6be3dc73feb2c9eb7cf4b99f7f13d9ed37a688311f58a'
-            '7a9ce949579a3b02d4b91b6835c4fb45adc5f743007572fb0e28e6433e48f3a5')
+            '7a9ce949579a3b02d4b91b6835c4fb45adc5f743007572fb0e28e6433e48f3a5'
+            '0b8056b7786e7426cedcc9a5aa13e67d44fc360d22f3400f3048dfea53344c2f')
 validpgpkeys=('474E22316ABF4785A88C6E8EA2C794A986419D8A') # Tom Stellard <tstellar@redhat.com>
 
 # Utilizing LLVM_DISTRIBUTION_COMPONENTS to avoid
@@ -57,6 +59,10 @@ _get_distribution_components() {
 }
 
 prepare() {
+  pushd llvm-$pkgver.src
+  patch -Np2 -i ../add-R_RISCV_SUB6-relocation.patch
+  popd
+
   cd clang-$pkgver.src
   mkdir build
   mv "$srcdir/clang-tools-extra-$pkgver.src" tools/extra
