Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422631)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,7 @@
 license=(LGPL)
 arch=(x86_64)
 depends=(expat libdaemon glib2 libcap gdbm dbus)
-makedepends=(git gobject-introspection gtk3 qt5-base xmltoman python-dbus
+makedepends=(git gobject-introspection qt5-base xmltoman python-dbus
              python-gobject doxygen graphviz libevent)
 optdepends=('gtk3: avahi-discover, avahi-discover-standalone, bshell, bssh, bvnc'
             'qt5-base: qt5 bindings'
@@ -21,7 +21,7 @@
             'python-dbus: avahi-bookmarks, avahi-discover')
 provides=(libavahi-client.so libavahi-common.so libavahi-core.so
           libavahi-glib.so libavahi-gobject.so libavahi-libevent.so
-          libavahi-qt5.so libavahi-ui-gtk3.so libdns_sd.so)
+          libavahi-qt5.so libdns_sd.so)
 backup=(etc/avahi/{hosts,avahi-daemon.conf,avahi-{autoip,dnsconf}d.action}
         usr/lib/avahi/service-types.db)
 _commit=d1e71b320d96d0f213ecb0885c8313039a09f693  # master
@@ -60,7 +60,8 @@
     --with-avahi-priv-access-group=network \
     --with-autoipd-user=avahi \
     --with-autoipd-group=avahi \
-    --with-systemdsystemunitdir=/usr/lib/systemd/system
+    --with-systemdsystemunitdir=/usr/lib/systemd/system \
+    --disable-gtk3
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
 
   make
