diff --git PKGBUILD PKGBUILD
index 47314d4..2ec2430 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,14 @@ arch=(x86_64)
 url='https://github.com/danvratil/qcoro'
 license=(MIT)
 makedepends=(cmake qt5-base qt6-base)
-source=(https://github.com/danvratil/qcoro/archive/v$pkgver/$pkgname-$pkgver.tar.gz)
-sha256sums=('0e68b3f0ce7bf521ffbdd731464d2d60d8d7a39a749b551ed26855a1707d86d1')
+source=(https://github.com/danvratil/qcoro/archive/v$pkgver/$pkgname-$pkgver.tar.gz
+        qcoro-fix-atomic.patch::https://github.com/danvratil/qcoro/pull/52.patch)
+sha256sums=('0e68b3f0ce7bf521ffbdd731464d2d60d8d7a39a749b551ed26855a1707d86d1'
+            '53c3aba5dbb273ab2fdce8021b418a292846da2cfd66739c93e5e5cb5b3124ab')
+
+prepare(){
+  patch -Np1 -d "$srcdir/$pkgbase-$pkgver" < qcoro-fix-atomic.patch
+}
 
 build() {
   cmake -B build5 -S $pkgbase-$pkgver \
