--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@ license=(Apache)
 depends=(lz4 protobuf snappy zlib zstd)
 makedepends=(cmake)
 checkdepends=(gtest)
-source=(https://dlcdn.apache.org/${_pkg}/${_pkg}-${pkgver}/${_pkg}-${pkgver}.tar.gz{,.asc})
+source=(https://archive.apache.org/dist/${_pkg}/${_pkg}-${pkgver}/${_pkg}-${pkgver}.tar.gz{,.asc})
 sha256sums=('7f46f2c184ecefd6791f1a53fb062286818bd8710c3f08b94dd3cac365e240ee'
             'SKIP')
 validpgpkeys=(F28C9C925C188C35E345614DEDA00CE834F0FC5C) # Dongjoon Hyun (CODE SIGNING KEY) <dongjoon@apache.org>
