--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,15 @@ license=('LGPL')
 depends=('glibmm' 'gst-plugins-base')
 makedepends=('mm-common' 'glibmm-docs' 'cairomm' 'cairomm-docs' 'pangomm' 'pangomm-docs')
 changelog=$pkgname.changelog
-source=(https://ftp.gnome.org/pub/GNOME/sources/gstreamermm/1.10/$pkgname-$pkgver.tar.xz)
-sha256sums=('be58fe9ef7d7e392568ec85e80a84f4730adbf91fb0355ff7d7c616675ea8d60')
+source=(https://ftp.gnome.org/pub/GNOME/sources/gstreamermm/1.10/$pkgname-$pkgver.tar.xz
+        $pkgname-fix-build-against-glib-2.68.patch::https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/4.patch)
+sha256sums=('be58fe9ef7d7e392568ec85e80a84f4730adbf91fb0355ff7d7c616675ea8d60'
+            'e02fcf5d6c2d658066dd3515025b6cb2d492dbe4944c91c2475100d9c9e97ac4')
 
 prepare() {
   cd $pkgbase-$pkgver
   NOCONFIGURE=1 ./autogen.sh
+  patch -Np1 -i ../$pkgname-fix-build-against-glib-2.68.patch
 }
 
 build() {
