Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1013499)
+++ PKGBUILD	(working copy)
@@ -13,14 +13,17 @@
 backup=('etc/sniffglue.conf')
 source=(https://github.com/kpcyrd/${pkgname}/archive/v${pkgver}/${pkgname}-${pkgver}.tar.gz
         https://github.com/kpcyrd/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz.asc
+        https://github.com/felixonmars/sniffglue/commit/660e19fb28f956690694e26c664ca48ed8d71455.patch
         sniffglue-sysuser.conf
         sniffglue.conf)
 sha512sums=('c13a10459fc0c022d253ece0ab01ddaa23547286a50a0763ee363d74333babbd18fd3fb252bec1e3182ae5640145f32ab41f5023168c4dbb87b120bd40ce509d'
             'SKIP'
+            'd41e7e4f2fe15823e44e48e3f9d3a28f917d48fc63d09a5761a7a81ab6d3402fc19cd9a4a98fc47c21d60e432f9653a6a3507f07c7bc9cf9c1031b8ad843ec41'
             '2595eb4856bc7a59c078255bf2aeeaa44a4f6a69b37cc8982e5d70a8ef14cb29b83b4d191ab3baabfbcf209f3effd232c10ec5baabc6ce52990969526d6fda98'
             '66e6b87e832ba8424334d24e3457a214e60e67c731471098329d0ca54c574518f7856ba780bb6bffde53bdefe2c8f789a242138ae67e2fef3e562fa083b362b1')
 b2sums=('d9be3390f9e6a083fbd9be5abe5abbf167577ed4ef598093826ffc1ce5aa4edb1e922508b8959dc183fc9409fb7322fb07796c5a1553516cde22d594cd8519fe'
         'SKIP'
+        '81584110a03e0df93927d3423b706082f6c33be280881b665820333adc509f65b45e37c3324a4a15603954ec595648592915d1f43e085084f3197a151d10dcfa'
         '69f5ba2cddca6878b776714c5015c1573df31fc4958432291baa4537349e5bdd1ab4307fd1c00bc20484213c118ecfc2ccfab88b942c0311e4d72872bf4eeeda'
         'ad68689ed08e882f734a613300c14ddfb6d84656dec8dbbed818302b0fe69907f7fc76306f49c0bac019acf4dcb46fec984e7194b16a481f7cc4636ca9aaf303')
 
@@ -28,6 +31,7 @@
 
 build() {
   cd ${pkgname}-${pkgver}
+  patch -p1 -i ../660e19fb28f956690694e26c664ca48ed8d71455.patch
   cargo build --release --locked
 }
 
