--- PKGBUILD
+++ PKGBUILD
@@ -21,11 +21,11 @@ prepare() {
 build() {
   cd "$srcdir"/$pkgname-$pkgver
 
-  make PLATFORM=PM3GENERIC SKIPQT=0 all
+  make PLATFORM=PM3GENERIC SKIPQT=0 DONT_BUILD_NATIVE=y all
 
   cd "$srcdir"/$pkgname-$pkgver-rdv4
 
-  make PLATFORM=PM3RDV4 FWTAG=rdv4 common bootrom recovery fullimage
+  make PLATFORM=PM3RDV4 FWTAG=rdv4 DONT_BUILD_NATIVE=y common bootrom recovery fullimage
 }
 
 package() {
