--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ prepare() {
   sed -i 's/-interim git//g' src/version.h
   setconf scripts/genGitHdr.sh GIT_URL "'""${source:4:34}""'"
   setconf scripts/genGitHdr.sh GIT_REV "${source#*=}"
+  patch -p0 < ../timestamp.patch
 }
 
 build() {
@@ -40,3 +41,6 @@ package() {
   install -Dm644 $pkgname/changelog.txt \
     "$pkgdir/usr/share/doc/$pkgname/changelog.txt"
 }
+
+source+=("timestamp.patch")
+b2sums+=("aa4aa12225ac85670d798ae6a4b0f4ad9449b42ee422cb12f50b66cf6aec0ef44dab83b43922cac9fb3c975cf2b8ce8d4e4f617f9753b94bff7e45a733b56c5f")
