diff --git src/utils/timeStamp.cpp src/utils/timeStamp.cpp
index f62d019f..dfc70dba 100644
--- src/utils/timeStamp.cpp
+++ src/utils/timeStamp.cpp
@@ -17,13 +17,19 @@
 #if defined(WIN32)
 #include <intrin.h>
 #pragma intrinsic(__rdtsc)
-#else
+#elif defined(__x86_64__)
 #include <x86intrin.h>
 #endif
 
 static uint64_t rdtsc()
 {
+#if (defined(__riscv) && __riscv_xlen == 64)
+	uint64_t current_time;
+	asm volatile ("rdtime %0" : "=r"(current_time));
+	return current_time;
+#else
 	return __rdtsc();
+#endif
 }
 
 namespace FCEU
