diff --git a/makefile b/makefile
index 13b7b535989..cf1d485ac2f 100644
--- a/makefile
+++ b/makefile
@@ -344,6 +344,9 @@ else ifeq ($(firstword $(filter powerpc64,$(UNAME))),powerpc64)
 ARCHITECTURE := _x64
 else ifeq ($(firstword $(filter s390x,$(UNAME))),s390x)
 ARCHITECTURE := _x64
+else ifeq ($(firstword $(filter riscv64,$(UNAME))),riscv64)
+# riscv64 compiler does not support -m64
+ARCHITECTURE :=
 else
 ARCHITECTURE := _x86
 endif
