--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,12 @@ sha256sums=('e8a98b0ee20c495a6ab894398a065ef580272dbd5a15b1b19e8bd1bc89d9f9fa'
             'SKIP')
 validpgpkeys=('9804BBF566E198FCA15D5C215EC2A77F36ECC523') # "Patrice Dumas (Pertusus) <pertusus@free.fr>"
 
+prepare() {
+  cd "$srcdir/$pkgname-$pkgver"
+  cp /usr/share/autoconf/build-aux/config.guess config.guess
+  cp /usr/share/autoconf/build-aux/config.sub config.sub
+}
+
 build() {
   cd "$srcdir/$pkgname-$pkgver"
   ./configure --prefix=/usr
