diff --git a/PKGBUILD b/PKGBUILD
index 6f04ac5..c4d0bfe 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,8 @@ sha256sums=('9611344733c0cdf14395f60880950ea4c3c7d6b765565b6493ad3e1afbe216de')
 
 build() {
 	cd "$srcdir"/${pkgname}-${pkgver}
-
+	cp /usr/share/autoconf/build-aux/config.guess aclocal/
+	cp /usr/share/autoconf/build-aux/config.sub aclocal/
 	./configure \
 		--prefix=/usr \
 		--bindir=/usr/bin \
