--- PKGBUILD.orig	2021-12-02 20:04:06.645111928 +0000
+++ PKGBUILD	2021-12-02 21:30:51.195395421 +0000
@@ -0,0 +1,16 @@
+pkgname=fcron-bootstrap
+pkgver=1.0.0
+pkgrel=1
+pkgdesc='Feature-rich cron implementation - for bootstrap'
+arch=(any)
+url='http://fcron.free.fr'
+license=('GPL')
+depends=()
+source=("fcron-bootstrap.sysusers")
+md5sums=('aef8c3887202cbc1252cdcfecc2db1dd')
+
+package() {
+  install -Dm644 "$srcdir/fcron-bootstrap.sysusers" "$pkgdir/usr/lib/sysusers.d/fcron.conf"
+}
+
+# vim:set ts=2 sw=2 et:
