diff --git PKGBUILD PKGBUILD
index 24bc40b..72a0be2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,6 +12,11 @@ makedepends=(qt6-tools)
 source=(https://github.com/Qalculate/qalculate-qt/releases/download/v$pkgver/$pkgname-$pkgver.tar.gz)
 sha256sums=('f14c2d8677b650fd03d8b96d82bcedbae406015f2f871222f94b827c5a0aeef6')
 
+prepare() {
+  cd $pkgname-$pkgver
+  rm .qmake.stash
+}
+
 build() {
   cd $pkgname-$pkgver
   qmake6 PREFIX=/usr
