diff --git PKGBUILD PKGBUILD
index 6c15998..4388bc1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,16 @@ makedepends=(gcc-fortran)
 depends=(lapack)
 arch=(x86_64)
 license=(GPL3)
-source=(https://downloads.sourceforge.net/project/qrupdate/qrupdate/1.2/$pkgname-$pkgver.tar.gz)
-sha256sums=('e2a1c711dc8ebc418e21195833814cb2f84b878b90a2774365f0166402308e08')
+source=(https://downloads.sourceforge.net/project/qrupdate/qrupdate/1.2/$pkgname-$pkgver.tar.gz
+        "fix_test_utils.patch") # This patch was generated from https://sourceforge.net/p/qrupdate/code/33
+sha256sums=('e2a1c711dc8ebc418e21195833814cb2f84b878b90a2774365f0166402308e08'
+            'd1c4e67891a771a9c1ac8aed0944ea6b0ef3ca7cb2b259ff1a8fe9f7019bb044')
 
 prepare() {
 # Honor system build flags
   sed -e 's|^FFLAGS.*|FFLAGS=${CFLAGS} ${LDFLAGS}|' -i $pkgname-$pkgver/Makeconf
+  cd $pkgname-$pkgver
+  patch -Np0 -i "../fix_test_utils.patch"
 }
 
 build() {
