--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ makedepends=('python' 'python-markdown' 'python-dnspython' 'docbook-xsl' 'pkg-co
              'readline' 'tevent' 'acl' 'libldap' 'libcap' 'ldb' 'krb5' 'pam'
              'systemd' 'gnutls' 'talloc' 'tdb' 'dbus' 'libaio'
              'perl-parse-yapp' 'libnsl' 'libtirpc' 'rpcsvc-proto' 'jansson'
-             'ceph-libs' 'liburing' 'perl-json' 'glusterfs')
+             'liburing' 'perl-json' 'glusterfs')
 optdepends=(
              'python-dnspython: samba_dnsupdate and samba_upgradedns in AD setup'
              'python-markdown: for samba-tool domain schemeupgrade'
@@ -156,7 +156,7 @@ package_samba() {
 pkgdesc="SMB Fileserver and AD Domain server"
 depends=('db>=4.7' 'popt' 'libcups' 'libcap>=2.16' 'gnutls>=2.4.1'
          'talloc' 'ldb' 'libbsd' 'python' 'iniparser' 'tdb' 'libaio' 'perl-parse-yapp' "smbclient>=$pkgver" 'gpgme'
-         'ceph-libs' 'libldb.so' 'libtdb.so' 'libtevent.so' 'liburing' 'libunwind')
+         'libldb.so' 'libtdb.so' 'libtevent.so' 'liburing' 'libunwind')
 backup=(etc/logrotate.d/samba
         etc/pam.d/samba
         etc/conf.d/samba)
