diff --git a/setup.py b/setup.py
index f35f6f8..c70eec6 100644
--- a/setup.py
+++ b/setup.py
@@ -3,10 +3,6 @@
 from setuptools import setup
 import sys
 
-extra = {}
-if sys.version_info >= (3,):
-    extra['use_2to3'] = True
-
 
 setup(
     name='funcparserlib',
@@ -17,5 +13,4 @@ setup(
     description='Recursive descent parsing library based on functional '
         'combinators',
     license='MIT',
-    url='http://code.google.com/p/funcparserlib/',
-    **extra)
+    url='http://code.google.com/p/funcparserlib/')
