diff --git PKGBUILD PKGBUILD
index 8307d39ab..0c8da532c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,16 @@ arch=('any')
 depends=('python')
 makedepends=('python-setuptools')
 checkdepends=('python-nose')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/vlasovskikh/funcparserlib/archive/$pkgver.tar.gz")
-sha512sums=('0fe23dc471f2de4588204d09379ef1400231bbe370e71045ddfad82c0f3a7bd13350954dde7210a370084c7bb7eb8c02ebdb04fb3727bee3e768cebe54e5ed99')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/vlasovskikh/funcparserlib/archive/$pkgver.tar.gz"
+        "remove_2to3.patch")
+sha512sums=('0fe23dc471f2de4588204d09379ef1400231bbe370e71045ddfad82c0f3a7bd13350954dde7210a370084c7bb7eb8c02ebdb04fb3727bee3e768cebe54e5ed99'
+            '54386d3b1e400b7dfa139ee42ba1c600b28745093a86fb7c873f1faf2a0613a7810826c93b3ffa81d10768bd92c9ad21d973ea6b644027eeab37250ee929a8a4')
+
+prepare() {
+  cd funcparserlib-$pkgver
+  patch -Np1 -i "../remove_2to3.patch"
+  2to3 -wn funcparserlib
+}
 
 build() {
   cd funcparserlib-$pkgver
