--- a/test/vspec/signatures.vim
+++ b/test/vspec/signatures.vim
@@ -111,7 +111,7 @@ describe 'signatures'
         let funcname = repeat('a', &columns - (30 + (&ruler ? 18 : 0)))
         put = 'def '.funcname.'(arg1, arg2, arg3, a, b, c):'
         put = '    pass'
-        execute "normal o\<BS>".funcname."( "
+        execute "normal o".funcname."( "
         Expect Signature() == "\n".funcname."(arg1, …)"
 
         exe 'normal sarg1, '
@@ -126,7 +126,7 @@ describe 'signatures'
         g/^/d
         put = 'def '.funcname.'('.repeat('b', 20).', arg2):'
         put = '    pass'
-        execute "normal o\<BS>".funcname."( "
+        execute "normal o".funcname."( "
         Expect Signature() == "\n".funcname."(…)"
     end
 
