diff --git PKGBUILD PKGBUILD
index 9b1cd5df..62dd8a3b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,6 +9,12 @@
 license=('GPL3')
 source=("https://ftp.gnu.org/gnu/gss/gss-${pkgver}.tar.gz")
 b2sums=('3f535fc152c72490a536b55dfdf175d6df6f4da7f9eb7454457d1e098763be604506006681f82d2d49325cb38e9a6f5454cb5b20462ff06b1b1084e79e95621e')
+prepare() {
+  cd ${pkgname}-${pkgver}
+  _sharedir=/usr/share/autoconf/build-aux
+  cp ${_sharedir}/config.guess build-aux
+  cp ${_sharedir}/config.sub build-aux
+}
 
 build() {
   cd ${pkgname}-${pkgver}
