--- PKGBUILD
+++ PKGBUILD
@@ -25,6 +25,8 @@ b2sums=('32a9818e13607ed81d0132935617e6b29d6b44ff34e07e37e31f56e41b7b116ad933947
 prepare() {
   mv "$pkgname-$pkgname-v$pkgver" "$pkgname-$pkgver"
   cd "$pkgname-$pkgver"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
