Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1322794)
+++ PKGBUILD	(working copy)
@@ -137,14 +137,14 @@
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/http \
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/http.rb \
     "${pkgdir}"/usr/lib/ruby/${rubyver}/net/https.rb \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/cgi \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/date_core.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/digest \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/console.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/nonblock.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/io/wait.so \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/json \
-    "${pkgdir}"/usr/lib/ruby/${rubyver}/x86_64-linux/racc
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/cgi \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/date_core.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/digest \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/io/console.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/io/nonblock.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/io/wait.so \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/json \
+    "${pkgdir}"/usr/lib/ruby/${rubyver}/$CARCH-linux/racc
 }
 
 package_ruby-docs() {
