--- a/test/net/http/test_https.rb	2024-11-24 21:17:41.800000000 +0000
+++ b/test/net/http/test_https.rb	2024-11-24 21:16:57.370000000 +0000
@@ -169,6 +169,7 @@
     omit if OpenSSL::OPENSSL_LIBRARY_VERSION.include?('OpenSSL 1.1.0h')
     omit if OpenSSL::OPENSSL_LIBRARY_VERSION.include?('OpenSSL 3.2.')
     omit if OpenSSL::OPENSSL_LIBRARY_VERSION.include?('OpenSSL 3.3.')
+    omit if OpenSSL::OPENSSL_LIBRARY_VERSION.include?('OpenSSL 3.4.')
 
     http = Net::HTTP.new(HOST, config("port"))
     http.use_ssl = true
