--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,9 @@ b2sums=('d6fad287e30fe60197f2c5f1eaf2715601cfbdae3de6675ce9c1fcccab91c31a43b7a3e

 prepare() {
   cd spotifyd-${pkgver}
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }

 build() {
