--- PKGBUILD
+++ PKGBUILD
@@ -12,13 +12,18 @@ license=('BSD')
 depends=('glibc' 'libogg')
 makedepends=('libpng' 'libvorbis' 'sdl')
 source=("https://downloads.xiph.org/releases/theora/$pkgname-$pkgver.tar.xz"
-        'libtheora-1.1.1-libpng16.patch')
+        'libtheora-1.1.1-libpng16.patch'
+        'fix-linking-issue.patch')
 sha256sums=('f36da409947aa2b3dcc6af0a8c2e3144bc19db2ed547d64e9171c59c66561c61'
-            'e4c9a8dc798c596ed32a2a720020ae27a0e72f5add1a47cb8fadebe0e7180d7e')
+            'e4c9a8dc798c596ed32a2a720020ae27a0e72f5add1a47cb8fadebe0e7180d7e'
+            'd115a230ff1407d340cb295d7ab63015fe952174bede035c069560dc4a88da2d')
 
 prepare() {
   cd $pkgname-$pkgver
   patch -Np0 -i ../libtheora-1.1.1-libpng16.patch
+  patch -Np0 -i ../fix-linking-issue.patch
+  find . -name config.guess -exec cp -f /usr/share/libtool/build-aux/config.guess {} \;
+  find . -name config.sub -exec cp -f /usr/share/libtool/build-aux/config.sub {} \;
 }
 
 build() {
