--- PKGBUILD
+++ PKGBUILD
@@ -13,18 +13,22 @@ makedepends=('go' 'systemd')
 options=('!lto')
 source=($pkgname-$pkgver.tar.gz::https://github.com/grafana/loki/archive/v$pkgver.tar.gz
         promtail.sysusers promtail.service promtail.tmpfiles
-        loki.sysusers loki.service loki.tmpfiles)
+        loki.sysusers loki.service loki.tmpfiles
+        fix-retention-expiration_test.patch::https://github.com/grafana/loki/pull/5137.patch)
 sha512sums=('56e39c17540be1d167ac63cc136fddc54d8d5a0f6c244e6c1013f0452eb887878d99351b3986f3b4495b650c2ca08ec21432bab21400ac293c8844e3f17df79f'
             '2b6c44b18ea3c9f955a7450222180d0b20b5fc551d0b7e5d0d8949e40adc847c4166829146260f87a75732cc5473eab0347dd56fc2125517698bac0652738c74'
             '7ff5c7324a4c883f5156b55cd025d2df8d8be134c18b1ecc007f89cafe20b4bed6c612b81588c9c660dd2fd3b6091fd952fcd3ad69a44e1ffa02e1eaeb05b290'
             '598042c40673a7914c5a1eeccfb78f832379a61f4360212c5d86f667343cf2fc78e98d9025f9717ea64f3e16e0a28f08cd7709706d811656722019f6167dd788'
             'f00b3cb64b71d3ca5a422a2bdff1f81a3e2707c4f73bcb27e24338eabff039b96125c847aaae43c9b22ccc47f89585118c1d0e0b29c4eb7b6f9260c68f8a8324'
             '03904b67617e4064a1eb39b6edfaec5d55bc85f142e867d115d3d0e12494a052fddf13bb6f5c541aaca29a18582df2332bb0cc6b8f29928cf6a0d5d2710ead74'
-            'de40d1e6752edbf8c21317ce1ee10f98dfc869e569c07092c613f7144e261e9438683a145a2e0e37e2a9fc758c5c2f02e1d0ac1c60347e98b147e4a550ec6040')
+            'de40d1e6752edbf8c21317ce1ee10f98dfc869e569c07092c613f7144e261e9438683a145a2e0e37e2a9fc758c5c2f02e1d0ac1c60347e98b147e4a550ec6040'
+            'b55a178dce29e04c2901659d7523571760cef8f6a84090bf24e5123fe98530f79d13d9c94b3086152ce7980a44ad7c6fc7f287aa31ed767cb6c28e318f73ae29')

 prepare() {
   cd "$pkgname-$pkgver"

+  patch -Np1 < ../fix-retention-expiration_test.patch
+
   go mod download
 }
