--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,19 @@ options=('!makeflags')
 source=(https://github.com/prusa3d/PrusaSlicer/archive/version_${pkgver}/${pkgname}-${pkgver}.tar.gz)
 sha256sums=('18b4e9e656a03bc0ae5f382b207ace07bc8944c3db7c252a6c1c5b67da169f64')
 
+source+=(onetbb-fix.patch::https://patch-diff.githubusercontent.com/raw/oneapi-src/oneTBB/pull/550.patch
+         $pkgname-tbbfix.patch)
+sha256sums+=('7f44a282449ff108a5f40856e9c93fbe2dd57e5a0815edaf2d41376f6bdf3bda'
+             '39c97b951f2a9e5f022983194737e6e68acbe3c0c5ed9238db6429faf8d37255')
+
+
+prepare() {
+  cd PrusaSlicer-version_${pkgver}/deps/+TBB
+
+  cp ${srcdir}/onetbb-fix.patch ./
+  patch -Np0 -i ${srcdir}/$pkgname-tbbfix.patch
+}
+
 build() {
   cd PrusaSlicer-version_${pkgver}
 
