diff --git a/tests/test_main.py b/tests/test_main.py
index f1e6871..246746c 100644
--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -298,7 +298,7 @@ class TestStuff(unittest.TestCase):
                 self.assertGreater(len(source.text), 7000)
             else:
                 self.assertIs(source, new_source)
-        self.assertLess(time.time() - start, 1)
+        self.assertLess(time.time() - start, 5)
 
     def test_decode_source(self):
         def check(source, encoding, exception=None, matches=True):
