diff --git PKGBUILD PKGBUILD
index 45cd750..d894963 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,8 +17,15 @@ makedepends=(
   'python-wheel'
 )
 checkdepends=('ipython' 'python-asttokens' 'python-littleutils' 'python-pytest')
-source=("git+$url.git#tag=v$pkgver")
-b2sums=('SKIP')
+source=("git+$url.git#tag=v$pkgver"
+        "fix-test-many-source-for-filename-calls-timeout.patch")
+b2sums=('SKIP'
+        '152545e095b4ed4401a199a22196f97ee7f7cca220063c4734ea33d70e0640f234e95f0a52cbf3aa8c15a8cc940e03f97ed636193b33db4892d4b73e55071c52')
+
+prepare() {
+  cd "$_name"
+  patch -Np1 -i ../fix-test-many-source-for-filename-calls-timeout.patch
+}
 
 build() {
   cd "$_name"
