--- PKGBUILD
+++ PKGBUILD
@@ -25,6 +25,11 @@ checkdepends=(
 source=("git+$url.git#tag=v$pkgver")
 b2sums=('38dfe34190ff9296b95d992cbe2d55cc481b05dd6b39c5ee4400e04fdf6c86d181304035f6b5b822936a74ce250089d23d5f82431f903244649235167ac18486')
 
+prepare() {
+  cd "$_name"
+  patch -Np1 -i "$srcdir/fix-test-many-source-for-filename-calls-timeout.patch"
+}
+
 build() {
   cd "$_name"
   python -m build --wheel --skip-dependency-check --no-isolation
@@ -44,3 +49,6 @@ package() {
   cd "$_name"
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(fix-test-many-source-for-filename-calls-timeout.patch)
+b2sums+=('152545e095b4ed4401a199a22196f97ee7f7cca220063c4734ea33d70e0640f234e95f0a52cbf3aa8c15a8cc940e03f97ed636193b33db4892d4b73e55071c52')
