diff --git PKGBUILD PKGBUILD
index 839470c52..4a4107622 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,15 @@ url='https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn'
 depends=(networkmanager nm-connection-editor openfortivpn libsecret)
 makedepends=(git intltool python)
 _commit=9688d63ecacec97d0c9217281eb7b85053f1f4a1
-source=("git+https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn.git#commit=$_commit"
+        "https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn/-/commit/701f6f6f66f10e0b2ec6b0d6af80d1a8ec226a55.patch")
+sha256sums=('SKIP'
+            '5ce59ee655ce4398f64cdbfb1df9667ced9ff97367f2268872d15d1f5eee168e')
 
 prepare() {
   cd NetworkManager-fortisslvpn
   autoreconf -fi
+  patch -Np1 -i "${srcdir}/701f6f6f66f10e0b2ec6b0d6af80d1a8ec226a55.patch"
 }
 
 build() {
