diff --git PKGBUILD PKGBUILD
index 9375286..9ecc28a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,8 @@
 prepare() {
   # Ensure system libs are used
   rm -r Firebird-$pkgver-0-source/extern/{editline,libtommath,libtomcrypt,zlib}
+
+  patch -d Firebird-$pkgver-0-source -Np1 -i ../add-riscv64-def.patch
 }
 
 build() {
@@ -46,3 +48,6 @@
   install -m755 gen/Release/firebird/bin/fb_config "$pkgdir"/usr/bin
   install -m644 "$srcdir"/LICENSE "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 }
+
+source+=(add-riscv64-def.patch::https://github.com/FirebirdSQL/firebird/commit/0e7947518aea1a48eeda4bc87f6fc4aa7e8140a1.diff)
+sha512sums+=('ee012e4d187b88e255a2d855c264f68602d86945983a528a1b540ca43ae143a49f9ace45f699b31a6a2109ec719e3250c202990e28b37322fcec8313ac9b23c8')
