diff --color -Naur xtmsplit-orig/src/xtmsplit.cpp xtmsplit/src/xtmsplit.cpp
--- xtmsplit-orig/src/xtmsplit.cpp	2006-12-21 04:33:51.000000000 +0800
+++ xtmsplit/src/xtmsplit.cpp	2021-10-08 08:57:18.748422706 +0800
@@ -190,7 +190,7 @@
 	header[21] = strlen("1.1");
 	sprintf(&header[22],"1.1");
 	header[40] = strlen(orig_file);
-	sprintf(&header[41],orig_file);
+	sprintf(&header[41],"%s",orig_file);
 	header[91] = md5;
 	long int nb = ceil(filesize /(size*1024.0*1024.0));
 	memcpy(&header[92], &nb, 4);
