diff --git PKGBUILD PKGBUILD
index 0ea38f1..5327da3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@
 license=('LGPL3')
 depends=('python' 'python-packaging')
 makedepends=('python-build' 'python-installer' 'python-wheel' 'python-pdm-pep517')
-checkdepends=('python-docutils' 'python-pytest' 'python-sphinx' 'python-tabulate')
+checkdepends=('python-docutils' 'python-pytest' 'python-sphinx' 'python-tabulate' 'python-platformdirs')
 optdepends=(
 	'python-platformdirs: for finding configuration dirs'
 	'python-pydantic: for validation'
