Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1187003)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,7 @@
 	 'xorg-xinit')
 makedepends=('cmake' 'nasm' 'xorg-font-util' 'xorg-util-macros'
 	     'xtrans' 'xorgproto'
-	     'mesa' 'imagemagick' 'java-environment=8')
+	     'mesa' 'imagemagick' 'java-environment=11')
 optdepends=('mesa: for OpenGL functionality in Xvnc')
 conflicts=('tightvnc')
 backup=(etc/pam.d/tigervnc
