diff --git PKGBUILD PKGBUILD
index b6a84dce..e9392f91 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ url="https://gitlab.freedesktop.org/pipewire/media-session"
 license=(MIT)
 arch=(x86_64)
 makedepends=(git meson doxygen graphviz 'pipewire>=0.3.39' systemd dbus
-             alsa-lib)
+             alsa-lib cmake)
 _commit=e5d5cf2404786af8bcc40bdb8a2962bef4ec18b6  # tags/0.4.1
 source=("$pkgbase::git+https://gitlab.freedesktop.org/pipewire/media-session.git#commit=$_commit")
 sha256sums=('SKIP')
