diff --git a/src/core.h b/src/core.h
index fe495c6..31c4d7f 100644
--- a/src/core.h
+++ b/src/core.h
@@ -46,8 +46,6 @@
 	#define B2_CPU_ARM
 #elif defined( __EMSCRIPTEN__ )
 	#define B2_CPU_WASM
-#else
-	#error Unsupported CPU
 #endif
 
 // Define compiler
@@ -70,7 +68,7 @@
 #elif defined( B2_COMPILER_GCC ) || defined( B2_COMPILER_CLANG )
 	#if defined( B2_CPU_X64 )
 		#define B2_BREAKPOINT __asm volatile( "int $0x3" )
-	#elif defined( B2_CPU_ARM )
+	#else
 		#define B2_BREAKPOINT __builtin_trap()
 	#endif
 #else
