--- PKGBUILD
+++ PKGBUILD
@@ -33,11 +33,14 @@ makedepends=(
 )
 optdepends=('power-profiles-daemon: power profiles support')
 _tag=648c4e24ab825444af4e84bf713dc37c6250dbee
-source=(git+https://github.com/pop-os/cosmic-settings.git#tag=${_tag})
-b2sums=('210c175a3282bee335434b88d466c053d5ba3fd56c5c2b0989c1afafe90f6465d99aac70915b7fe956e7c35e8b39d811b97e2cf28c8f43b292376638811b9f16')
+source=(git+https://github.com/pop-os/cosmic-settings.git#tag=${_tag}
+        remove-generator-075.diff)
+b2sums=('210c175a3282bee335434b88d466c053d5ba3fd56c5c2b0989c1afafe90f6465d99aac70915b7fe956e7c35e8b39d811b97e2cf28c8f43b292376638811b9f16'
+        'cd6dcc70767e5021a12690a8f44b13de62e24fd0ad31d35a42fccb97c28c5ac41701eda1af4e10af850236b212028cadf563a9a79fc518d5c173536241ba6a8f')
 
 prepare() {
   cd cosmic-settings
+  patch -Np1 -i ../remove-generator-075.diff
   cargo fetch --locked
 }
 
