diff --git PKGBUILD PKGBUILD
index eca9cf4ec..fb2fb868e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,14 +15,21 @@ provides=("gnome-tweak-tool=$pkgver")
 conflicts=(gnome-tweak-tool)
 replaces=(gnome-tweak-tool)
 _commit=7a16849f498c8a57d525eb9184a8b0163e2ddcbb  # tags/40.0^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-tweaks.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-tweaks.git#commit=$_commit"
+        "fix_positional_arg.patch")
+sha256sums=('SKIP'
+            '4f590ed68d0f289073bf814cce50d891fc72068b375d04c17e91a152be8f0704')
 
 pkgver() {
   cd $pkgname
   git describe --tags | sed 's/-/+/g'
 }
 
+prepare() {
+  cd $pkgname
+  patch -Np1 -i "../fix_positional_arg.patch"
+}
+
 build() {
   arch-meson $pkgname build
   meson compile -C build
