--- PKGBUILD
+++ PKGBUILD
@@ -8,8 +8,8 @@ pkgdesc="Platform to write and share markdown"
 arch=('any')
 url='https://github.com/hedgedoc/hedgedoc'
 license=('AGPL3')
-depends=('nodejs-lts-jod')
-makedepends=('nodejs-lts-jod' 'sqlite' 'python' 'node-gyp' 'git' 'yarn' 'npm')
+depends=('nodejs-lts-iron')
+makedepends=('nodejs-lts-iron' 'sqlite' 'python' 'python-setuptools' 'node-gyp' 'git' 'yarn' 'npm')
 optdepends=('postgresql: postgresql database support'
             'mariadb: mariadb database support'
             'sqlite: sqlite database support')
