diff --git PKGBUILD PKGBUILD
index 107f85b8d..22543a3c5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,13 +9,17 @@ url='https://www.digikam.org/'
 license=(GPL)
 depends=(kcoreaddons)
 makedepends=(extra-cmake-modules)
-source=("https://download.kde.org/stable/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig})
+source=("https://download.kde.org/stable/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig}
+        "https://github.com/KDE/libmediawiki/commit/7c79dae079d20b46d05462ed9f5b3cfae78f420a.patch")
 sha256sums=('da167259c49d012359ad65175c365613f0c42699fb99ac7fb9d1f1c8c48292c1'
-            'SKIP')
+            'SKIP'
+            '917f44c1e53309c0a794eb29aec8575debf68b590f928ccab204150c951141d5')
 validpgpkeys=(7C35920F1CE2899E8EA9AAD02E7C0367B9BFA089) # Scarlett Clark <sgclark@kde.org>
 
 prepare() {
-  mkdir -p build 
+  mkdir -p build
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i "${srcdir}/7c79dae079d20b46d05462ed9f5b3cfae78f420a.patch"
 }
 
 build() {
