diff --git PKGBUILD PKGBUILD
index 39e8d1cfa..db0d6c4fe 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,15 +27,22 @@ makedepends=(
 )
 provides=(librhonabwy.so)
 _tag=e4898061b3f99f6cdd8993f1d31ddd66301cf33b
-source=(git+https://github.com/babelouest/rhonabwy.git#tag=${_tag})
+source=(git+https://github.com/babelouest/rhonabwy.git#tag=${_tag}
+        "fix_type_warning.patch")
 validpgpkeys=(8405B02FCC28EF9744C8F253FE82139440BD22B9) # Nicolas Mora <nicolas@babelouest.org>
-b2sums=(SKIP)
+b2sums=('SKIP'
+        '006c46768c8d7b25eda5222c9e803b6738c894d6d8ae481cd7919d96c0a48f05784893a8dde0a3f3687af44a9b59976f806fc0d507789ec8aaf70da4eab73d01')
 
 pkgver() {
   cd rhonabwy
   git describe --tags | sed 's/^v//'
 }
 
+prepare() {
+  cd rhonabwy
+  patch -Np1 -i "../fix_type_warning.patch"
+}
+
 build() {
   export CFLAGS+=" -Wno-maybe-uninitialized"
   cmake -S rhonabwy -B build -G Ninja \
