diff --git PKGBUILD PKGBUILD
index 7741d1b..c3d53e6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@ pkgdesc='Applet for GNOME Panel to display readings from hardware sensors, inclu
 arch=('x86_64')
 url='http://sensors-applet.sourceforge.net/'
 license=('GPL')
-depends=('gnome-panel' 'libatasmart' 'libxnvctrl' 'lm_sensors')
+depends=('gnome-panel' 'libatasmart' 'lm_sensors')
 makedepends=('git' 'gnome-common' 'intltool' 'yelp-tools')
 optdepends=('hddtemp: get HDD temperatures')
 _commit=07284262768e3a7066da09845370d78526be6bc1  # master
