--- PKGBUILD
+++ PKGBUILD
@@ -42,6 +42,8 @@ b2sums=('980411d0ae08f95c2666b6bf45d978cbcd0fb47469b3d4cef1fe1ee4560c3f5d4463899
         '5e4e453eaa99b930d018ca200de791be7de2e5c3da60e5a6f84f0d61e1b18e5f7a6a7700503445a7f39fba0baac67c0042f8d69e2529374c6913458c465bc0e5')
 
 prepare() {
+  sed -i 's/$(LDFLAGS)/"$(LDFLAGS)"/' $pkgname-1.23.1-pinns_ldflags.patch
+
   cd $pkgname-$pkgver
   # make sure that /run instead of /var/run is used
   sed -e 's|/var/run|/run|g' \
