diff --git PKGBUILD PKGBUILD
index 30c0623..adb63dd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,14 @@ makedepends=(git)
 source=("git+https://github.com/riscv/riscv-isa-sim#commit=530af85d83781a3dae31a4ace84a573ec255fefa") # tag: v1.1.0
 b2sums=(SKIP)
 
+prepare(){
+  cd riscv-isa-sim
+
+  # Commits from https://github.com/riscv-software-src/riscv-isa-sim/pull/966
+  git cherry-pick -n e52327deeefb29908a822a9eb2f6fc5c87e968af
+  git cherry-pick -n dba7efaf9e2e8d5251820c8555a184f715bb4d46
+}
+
 build() {
   cd riscv-isa-sim
   ./configure --prefix=/usr
