diff --git PKGBUILD PKGBUILD
index 1bd6efa..c65320b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,11 @@
 
 prepare() {
   sed -i '/#include <cstring>/a #include <cstdint>' riscv-isa-sim/fesvr/device.h
+
+  cd riscv-isa-sim
+  # Commits from https://github.com/riscv-software-src/riscv-isa-sim/pull/966
+  git cherry-pick -n e52327deeefb29908a822a9eb2f6fc5c87e968af
+  git cherry-pick -n dba7efaf9e2e8d5251820c8555a184f715bb4d46
 }
 
 build() {
