--- PKGBUILD
+++ PKGBUILD
@@ -89,9 +89,14 @@
   done
 
   echo "Setting config..."
-  cp ../config.$CARCH .config
+  cp ../config.x86_64 .config
+  make olddefconfig
+
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  make mrproper
+  mv .config.1 .config
   make olddefconfig
-  diff -u ../config.$CARCH .config || :
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -103,7 +108,8 @@
   make htmldocs SPHINXOPTS=-QT &
   local pid_docs=$!
 
-  make all
+  # Generate symbols for dtbs to support overlays
+  make DTC_FLAGS="-@" all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
   wait $pid_docs
 }
@@ -190,6 +196,9 @@
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  make INSTALL_DTBS_PATH="$pkgdir/usr/lib/modules/$(<version)/dtb" dtbs_install
+
   # remove build link
   rm "$modulesdir"/build
 
@@ -272,6 +281,7 @@
   local karch
   case $CARCH in
     x86_64) karch=x86 ;;
+    riscv64) karch=riscv ;;
     *) echo "Unknown CARCH $CARCH"; exit 1 ;;
   esac
 
@@ -426,3 +436,9 @@
 done
 
 # vim:set ts=8 sts=2 sw=2 et:
+
+source_riscv64=(config.x86_64 riscv64.config-patch)
+sha256sums_riscv64=('404a54483535cabf1391999ecf66df168461a9ef4be1d2ef9c9048e5529e1714'
+                    '409ddd51f8bdf314fd304bd1afc8752a5cd0a5396f39ab41f4d4bf9bd74ac34d')
+b2sums_riscv64=('d801aa851700faaa5410b99ede568d6b5d2f4f98d047c3b2b8f26259437011e0c1dfe036a74fb37268540e0e6d54e4a562aab873727573736cad82016c242cb9'
+                '2c12a7d2509f010cf77cde12b1a88313e485b0d3536861843dc8c431648c92f24e2c3475010f844d63bdf3294537f1c92da2989d400cffc981a1827fcc7bef73')
