--- a/daemon/parted.c
+++ b/daemon/parted.c
@@ -74,6 +74,22 @@
     return NULL;
 }
 
+static int
+recover_blkrrpart (const char *device, const char *err)
+{
+  if (!strstr (err, "unable to inform the kernel of the change"))
+    return -1;
+
+  udev_settle ();
+
+  if (command (NULL, NULL, "blockdev", "--rereadpt", device, NULL) == -1)
+    return -1;
+
+  udev_settle ();
+
+  return 0;
+}
+
 int
 do_part_init (const char *device, const char *parttype)
 {
@@ -248,8 +264,10 @@
                 "mkpart", STREQ (parttype, "gpt") ? "p1" : "primary",
                 startstr, endstr, NULL);
   if (r == -1) {
-    reply_with_error ("parted: %s: %s", device, err);
-    return -1;
+    if (recover_blkrrpart (device, err) == -1) {
+      reply_with_error ("parted: %s: %s", device, err);
+      return -1;
+    }
   }
 
   udev_settle ();
