From 1523fdfedc11d0516d2d39fe5f9ccdee4a25366c Mon Sep 17 00:00:00 2001
From: Coelacanthus <coelacanthus@outlook.com>
Date: Sun, 26 Dec 2021 00:45:52 +0800
Subject: [PATCH] fix: OpenMP is not available

---
 PKGBUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git PKGBUILD PKGBUILD
index c30531a..dd36314 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,7 @@ build() {
     -DCMAKE_C_STANDARD=11 \
     -DENABLE_MPI=ON \
     -DENABLE_PTHREAD=ON	\
-    -DENABLE_OPENMP=ON \
+    -DENABLE_OPENMP=OFF \
     -DENABLE_KLU=ON \
     -DKLU_LIBRARY_DIR=/usr/lib \
     -DEXAMPLES_INSTALL_PATH=/usr/share/sundials/examples
-- 
2.34.1

