--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,7 @@ prepare() {
   cd $_pkgname-$pkgver
 
   patch -p1 < ../add-dependency.patch
+  patch -p1 < ../support-riscv64.patch
 }
 
 build() {
@@ -26,7 +27,7 @@ build() {
   cd $_pkgname-$pkgver/build
 
   export DC=ldc
-  export LDFLAGS="$(echo -ne $LDFLAGS | sed -e 's/-flto=auto/-flto=full/')"
+  export LDFLAGS="$(echo -ne $LDFLAGS | sed -e 's/-flto=auto//')"
 
   arch-meson ..
 
@@ -44,3 +45,6 @@ package() {
 
   DESTDIR="$pkgdir" ninja install
 }
+
+source+=("support-riscv64.patch")
+sha512sums+=('3da013d6027527180f529155ab90e23a6c0e2f218dcf237de040d2bfb75acb9b95a94e417240bdfd058dfaf859cb87b3cdeacce0af8916b62526a8052d00fa7f')
