--- PKGBUILD
+++ PKGBUILD
@@ -17,22 +17,31 @@
 depends=(libtorrent-rasterbar qt6-base)
 makedepends=(cmake boost qt6-tools qt6-svg)
 optdepends=('python: needed for torrent search tab')
-source=(https://downloads.sourceforge.net/sourceforge/qbittorrent/${pkgname}-${pkgver}.tar.xz{,.asc})
+source=(https://downloads.sourceforge.net/sourceforge/qbittorrent/${pkgname}-${pkgver}.tar.xz{,.asc}
+        prefer_pthread.patch)
 sha256sums=('efa580924e96605bae916b9a8ae1f3fce82a5130647ae41d74d689761262463d'
-            'SKIP')
+            'SKIP'
+            '96a36b520700895f6bae43fe055364e4725b476b751b00aa80749cd2c22f366c')
 validpgpkeys=('D8F3DA77AAC6741053599C136E4A2D025B7CC9A2') # sledgehammer999 <sledgehammer999@qbittorrent.org>
 
+prepare() {
+  cd $pkgbase-$pkgver
+  patch -Np1 -i "../prefer_pthread.patch"
+}
+
 build() {
   cmake -B build -S $pkgbase-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
-    -DQT6=ON
+    -DQT6=ON \
+    -DCMAKE_EXE_LINKER_FLAGS="-latomic"
   cmake --build build
 
   cmake -B build-nox -S $pkgbase-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DQT6=ON \
     -DGUI=OFF \
-    -DSYSTEMD=ON
+    -DSYSTEMD=ON \
+    -DCMAKE_EXE_LINKER_FLAGS="-latomic"
   cmake --build build-nox
 }
 
