--- PKGBUILD
+++ PKGBUILD
@@ -17,11 +17,18 @@ license=(custom GPL)
 depends=(libtorrent-rasterbar qt6-base)
 makedepends=(cmake boost qt6-tools qt6-svg)
 optdepends=('python: needed for torrent search tab')
-source=(https://downloads.sourceforge.net/sourceforge/qbittorrent/${pkgname}-${pkgver}.tar.xz{,.asc})
+source=(https://downloads.sourceforge.net/sourceforge/qbittorrent/${pkgname}-${pkgver}.tar.xz{,.asc}
+        prefer_pthread.patch)
 sha256sums=('1386f000ce1d791469c3ea03e3951ca25f67f534e66896592bd12357dda9a8ec'
-            'SKIP')
+            'SKIP'
+            '96a36b520700895f6bae43fe055364e4725b476b751b00aa80749cd2c22f366c')
 validpgpkeys=('D8F3DA77AAC6741053599C136E4A2D025B7CC9A2') # sledgehammer999 <sledgehammer999@qbittorrent.org>
 
+prepare() {
+  cd $pkgbase-$pkgver
+  patch -Np1 -i "../prefer_pthread.patch"
+}
+
 build() {
   cmake -B build -S $pkgbase-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
