diff --git PKGBUILD PKGBUILD
index b2799e0..7d7f930 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,14 +33,14 @@ build() {
   mkdir -p ${pkgbase}
   pushd ${pkgbase}
   ../configure --prefix=/usr
-  make
+  make CXXFLAGS="$CXXFLAGS -latomic"
   popd

   # Build nox variant
   mkdir -p "${pkgbase}-nox"
   pushd "${pkgbase}-nox"
   ../configure --prefix=/usr --disable-gui --enable-systemd
-  make
+  make CXXFLAGS="$CXXFLAGS -latomic -DDISABLE_GUI"
 }

 package_qbittorrent() {
