diff --git PKGBUILD PKGBUILD
index 35ff39b..5581979 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,16 +18,33 @@ makedepends=('boost' 'mesa')
 source=("https://github.com/mupen64plus/mupen64plus-core/releases/download/${pkgver}/mupen64plus-bundle-src-${pkgver}.tar.gz"
         '010-mupen64plus-ui-console-pie.patch'
         '020-mupen64plus-core-gcc10-fix.patch'
-        '030-mupen64plus-sdl12-compat-fix.patch'::'https://github.com/mupen64plus/mupen64plus-core/commit/5a5ffc0c78b0e255f54b15d2480f617d5cbc33ec.patch')
+        '030-mupen64plus-sdl12-compat-fix.patch'::'https://github.com/mupen64plus/mupen64plus-core/commit/5a5ffc0c78b0e255f54b15d2480f617d5cbc33ec.patch'
+        '040-mupen64plus-video-glide64mk2-add-aarch64-support.patch'::'https://github.com/mupen64plus/mupen64plus-video-glide64mk2/pull/102.patch'
+        '050-mupen64plus-core-add-riscv-support.patch'::'https://github.com/mupen64plus/mupen64plus-core/pull/898.patch'
+        '051-mupen64plus-rsp-hle-add-riscv-support.patch'::'https://github.com/mupen64plus/mupen64plus-rsp-hle/pull/81.patch'
+        '052-mupen64plus-video-rice-add-riscv-support.patch'::'https://github.com/mupen64plus/mupen64plus-video-rice/pull/99.patch'
+        '053-mupen64plus-video-glide64mk2-add-riscv-support.patch'::'https://github.com/mupen64plus/mupen64plus-video-glide64mk2/pull/117.patch')
 sha256sums=('9c75b9d826f2d24666175f723a97369b3a6ee159b307f7cc876bbb4facdbba66'
             '0b674779949ca90db9f156b56dd9d7be9847354f5b07ec73aa3a89dde79d6b6d'
             'f84f262bcf2b748ccded4443735caba92926241f9238f545a621009f6ae64ef7'
-            '1de522155742fd493075d051855e420a9bc98af2f4774ecb66b4fdef69a7adc4')
+            '1de522155742fd493075d051855e420a9bc98af2f4774ecb66b4fdef69a7adc4'
+            '686a7ac7df984bc0c9a09309c89c99aca56fd5ad185819e79560ccdb9f3be02e'
+            '23242b5e22bdc6c1947db35e0af7507b2005788ae2473cb75df3c0b8a9d58c24'
+            '5624db9bc9271eaa82a562735b236c0a7c7564f64edf9fabd1c887b76ccd577e'
+            '522828678587e5eed34e0c389ba4e628954ec1afb9a132504824b898e63d1926'
+            '30bf15d2ded73448a4e283e4bb3b85b22154f4a16b46d8f9fd3a905cc1771ccf')
 
 prepare() {
     patch -d "mupen64plus-bundle-src-${pkgver}" -Np1 -i "${srcdir}/010-mupen64plus-ui-console-pie.patch"
     patch -d "mupen64plus-bundle-src-${pkgver}" -Np1 -i "${srcdir}/020-mupen64plus-core-gcc10-fix.patch"
     patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-core" -Np1 -i "${srcdir}/030-mupen64plus-sdl12-compat-fix.patch"
+
+    patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-video-glide64mk2" -Np1 -i "${srcdir}/040-mupen64plus-video-glide64mk2-add-aarch64-support.patch" # Prepare for patch 043
+    patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-core" -Np1 -i "${srcdir}/050-mupen64plus-core-add-riscv-support.patch"
+    patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-rsp-hle" -Np1 -i "${srcdir}/051-mupen64plus-rsp-hle-add-riscv-support.patch"
+    patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-video-rice" -Np1 -i "${srcdir}/052-mupen64plus-video-rice-add-riscv-support.patch"
+    patch -d "mupen64plus-bundle-src-${pkgver}/source/mupen64plus-video-glide64mk2" -Np1 -i "${srcdir}/053-mupen64plus-video-glide64mk2-add-riscv-support.patch"
+
 }
 
 build() {
