diff --git PKGBUILD PKGBUILD
index 5e9162a..73dce9f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@ depends=('gcc-libs')
 source=(https://github.com/ivmai/bdwgc/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz
         gc-missing-header.patch::https://patch-diff.githubusercontent.com/raw/ivmai/bdwgc/pull/389.patch)
 sha512sums=('ff781360bca667f1f95fbfed073e9035f63134cac1bda4f7e4217664a3713c2846f750a3ce92732972c0435c69355b051f6e8eb6f2698b0d48d9107ff960bacf'
-            'a89d46b6826b90184e1a2724c2a1a57a4a2a215245938500aa7a652bf7eee3e0205c36fbd27e450ac59824c07b16fd38987ea77a33a41eb38878701fb70e0ad9')
+            'fc1bb2c8b412f22d6c861c2f89f8cf8ed3b8f053f935a766aad80ea05016401b72580250f37afd180283090fedc4e6257272c8fab610130d13b67cacbd445142')

 prepare() {
   cd $pkgname-$pkgver
