diff --git PKGBUILD PKGBUILD
index 065adc30..e9def0e5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -47,7 +47,9 @@ prepare() {
   git config submodule."freetype2".url "${srcdir}/${pkgname}"-freetype2
 
   git submodule update --init --recursive
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
