--- PKGBUILD
+++ PKGBUILD
@@ -48,7 +48,10 @@ prepare() {
   git config submodule."freetype2".url "${srcdir}/${pkgname}"-freetype2
 
   git -c protocol.file.allow=always submodule update --init --recursive
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+
+  echo -e "\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
