Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1013499)
+++ PKGBUILD	(working copy)
@@ -43,6 +43,9 @@
   git config submodule."freetype2".url "${srcdir}/${pkgname}"-freetype2
 
   git submodule update --init --recursive
+  sed -i "/\[patch.crates-io\]/a zbus = { git = 'https://gitlab.freedesktop.org/felixonmars/zbus', branch = '1.9.1' }" Cargo.toml
+  sed -i "/\[patch.crates-io\]/a ring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" Cargo.toml
+  cargo update -p zbus -p nix:0.17.0 -p ring
   cargo fetch --locked
 }
 
