diff --git PKGBUILD PKGBUILD
index 43942b3..2db9363 100644
--- PKGBUILD
+++ PKGBUILD
@@ -44,7 +44,9 @@
   git config submodule."freetype2".url "${srcdir}/${pkgname}"-freetype2
 
   git submodule update --init --recursive
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
