Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422452)
+++ PKGBUILD	(working copy)
@@ -11,8 +11,10 @@
 makedepends=(git)
 provides=(libwebrtc_audio_processing.so)
 _commit=e882a5442ac22c93648e12837248d651d18b9247  # tags/v0.3.1^0
-source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit"
+        https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/merge_requests/19.patch)
+sha256sums=('SKIP'
+            'dd0fb223c138df9894268201d7577be423f1c93c3114db3f152fb1a1b03fd44c')
 
 pkgver() {
   cd $pkgname
@@ -21,6 +23,7 @@
 
 prepare() {
   cd $pkgname
+  patch webrtc/typedefs.h -i ../19.patch
   NOCONFIGURE=1 ./autogen.sh
 }
 
