diff --git PKGBUILD PKGBUILD
index a0dc899..c2d2499 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,10 @@ depends=(gcc-libs)
 makedepends=(git)
 provides=(libwebrtc_audio_processing.so)
 _commit=e882a5442ac22c93648e12837248d651d18b9247  # tags/v0.3.1^0
-source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit")
-b2sums=('SKIP')
+source=("git+https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing.git#commit=$_commit"
+        https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/merge_requests/19.patch)
+b2sums=('SKIP'
+        'baedc74af9bef7afa79389f5cf9b444b72e56dd71766fb64678c0b13414edbb922497e86afbb10d9fe162779dfa47d4e391e493183d5df93556a1d4c3c81d478')
 
 pkgver() {
   cd $pkgname
@@ -21,6 +23,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch webrtc/typedefs.h -i ../19.patch
   NOCONFIGURE=1 ./autogen.sh
 }
 
