diff -Naur pypy2.7-v7.3.18-src/rpython/rlib/rvmprof/src/shared/vmprof_config.h pypy2.7-v7.3.18-src_/rpython/rlib/rvmprof/src/shared/vmprof_config.h
--- pypy2.7-v7.3.18-src/rpython/rlib/rvmprof/src/shared/vmprof_config.h	2025-02-14 12:43:44.000000000 +0800
+++ pypy2.7-v7.3.18-src_/rpython/rlib/rvmprof/src/shared/vmprof_config.h	2025-02-14 13:09:35.595334146 +0800
@@ -32,6 +32,8 @@
   #define PC_FROM_UCONTEXT uc_mcontext.pc
 #elif defined(__powerpc64__)
   #define PC_FROM_UCONTEXT uc_mcontext.gp_regs[PT_NIP]
+#elif defined(__riscv)
+  #define PC_FROM_UCONTEXT uc_mcontext.__gregs[REG_PC]
 #else
   /* linux, gnuc */
   #define PC_FROM_UCONTEXT uc_mcontext.gregs[REG_RIP]
