--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,7 @@ sha256sums=('7945658af4478d744b5d8e4afa60ef7f791fe16055070e5604cebec7f4c80c2c')
 prepare() {
   cd rpy2
   git cherry-pick -n c31ec06cb913ae2bebf46753a2751ca1a1b5660b # Fix tests with numpy 2.0
+  patch -Np1 -i ../rpy2-fix-Rcomplex.diff
 }
 
 build() {
@@ -48,3 +49,6 @@ package() {
 
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(rpy2-fix-Rcomplex.diff)
+sha256sums+=(303f40e56c6f5b06ef3fd53307ad25a51c4547679681945d9b6176a506626638)
