--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,7 @@ source=("https://github.com/PurpleI2P/i2pd/archive/${pkgver}/${pkgname}-${pkgver
         '040-i2pd-tunnels-d-readme.patch'
         'i2pd.sysusers'
         'i2pd.tmpfiles')
-sha256sums=('76b41d02a41a03d627fcd7fe695cad7f521b66e99a04ec9678f132a1eb052bb8'
+sha256sums=('83bb5e2494d566d41ff3a85e4c93078d290d10e3cf1fbe7807bec9f0d024513e'
             '2ddf15f1c1cdf5d747a0af667145238023fd126ab00c65f2897cacae935015b1'
             '45dae1e2f798d23df92c996c233fccb07349d62992d0f625be7fd913719875af'
             'e98eaa783fcd8e1ab84980f68158e3bb9eb5ec101f26c748946a313152643f11'
