--- PKGBUILD
+++ PKGBUILD
@@ -62,9 +62,9 @@
             'de428b496a825772ef49ec5555a386a4b23ffa10b9dfd9907f0671d5c1dc2178'
             '9a734e5d75be603d58aa92e8d29a15466b3ac2af6b7d6f6728dd5c3e5ed8758d'
             'd6f6ca86104df026462ab634efa1ecc850bccfa748ed5a6a88283e7659ec90c5'
-            '20497c90cbb678a9ca1a71eca66cf2a1d0c991c3e695f7b7d10fb68fc0ac4f12'
+            'faa3e8d07212d5edc09f24ac7aad549681e9884ca6c0c16afa1d86d1bdc768b2'
             '9c14afbc7dabee5a563202c833a043bd3b80c4522f8943c33d550641d7d1bbf1'
-            '933b0b878fa611bf24b92f655040a3bcb4a1b67841d929013802abbb09b2ccf4'
+            '29648518de7167dba0785051f4e1e5a390409d173a73493564fbce65be453819'
             '6fe03e6ea3f69d99d59a48847a8ae97c2160fca847c7aedf7b89d05e4aa9386d')
 
 pkgver() {
--- makepkg.conf
+++ makepkg.conf
@@ -35,20 +35,19 @@
 # ARCHITECTURE, COMPILE FLAGS
 #########################################################################
 #
-CARCH="x86_64"
-CHOST="x86_64-pc-linux-gnu"
+CARCH="riscv64"
+CHOST="riscv64-unknown-linux-gnu"
 
 #NPROC=2
 
 #-- Compiler and Linker Flags
 #CPPFLAGS=""
-CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
+CFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe -fno-plt -fexceptions \
         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security \
-        -fstack-clash-protection -fcf-protection \
-        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"
+        -fstack-clash-protection \
+        -fno-omit-frame-pointer"
 CXXFLAGS="$CFLAGS -Wp,-D_GLIBCXX_ASSERTIONS"
-LDFLAGS="-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now \
-         -Wl,-z,pack-relative-relocs"
+LDFLAGS="-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now"
 LTOFLAGS="-flto=auto"
 #-- Make Flags: change this for DistCC/SMP systems
 #MAKEFLAGS="-j2"
--- fortran.conf
+++ fortran.conf
@@ -11,9 +11,9 @@
 
 # Flags used for the Fortran compiler, similar in spirit to CFLAGS. Read
 # linkman:gfortran[1] for more details on the available flags.
-FFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt \
-        -Wp,-D_FORTIFY_SOURCE=3 -fstack-clash-protection -fcf-protection \
-        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"
+FFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe -fno-plt \
+        -Wp,-D_FORTIFY_SOURCE=3 -fstack-clash-protection \
+        -fno-omit-frame-pointer"
 FCFLAGS="$FFLAGS"
 
 # Additional compiler flags appended to `FFLAGS` and `FCFLAGS` for use in debugging. Usually
