--- PKGBUILD
+++ PKGBUILD
@@ -60,9 +60,9 @@ sha256sums=('74cfd4b6d2b10c742f5a4a7cdaeffb01405ab897b11b80681a9063245113a308'
             'b3bce9d662e189e8e49013b818f255d08494a57e13fc264625f852f087d3def2'
             'de428b496a825772ef49ec5555a386a4b23ffa10b9dfd9907f0671d5c1dc2178'
             'd6f6ca86104df026462ab634efa1ecc850bccfa748ed5a6a88283e7659ec90c5'
-            '20497c90cbb678a9ca1a71eca66cf2a1d0c991c3e695f7b7d10fb68fc0ac4f12'
+            'faa3e8d07212d5edc09f24ac7aad549681e9884ca6c0c16afa1d86d1bdc768b2'
             'c8760d7ebb6c9817d508c691c67084be251cd9c8811ee1ccf92c1278bad74c1c'
-            '933b0b878fa611bf24b92f655040a3bcb4a1b67841d929013802abbb09b2ccf4'
+            '29648518de7167dba0785051f4e1e5a390409d173a73493564fbce65be453819'
             '6fe03e6ea3f69d99d59a48847a8ae97c2160fca847c7aedf7b89d05e4aa9386d')
 
 pkgver() {
diff --git fortran.conf fortran.conf
index 7928473..5dd1830 100644
--- fortran.conf
+++ fortran.conf
@@ -11,9 +11,9 @@
 
 # Flags used for the Fortran compiler, similar in spirit to CFLAGS. Read
 # linkman:gfortran[1] for more details on the available flags.
-FFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt \
-        -Wp,-D_FORTIFY_SOURCE=3 -fstack-clash-protection -fcf-protection \
-        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"
+FFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe -fno-plt \
+        -Wp,-D_FORTIFY_SOURCE=3 -fstack-clash-protection \
+        -fno-omit-frame-pointer"
 FCFLAGS="$FFLAGS"
 
 # Additional compiler flags appended to `FFLAGS` and `FCFLAGS` for use in debugging. Usually
diff --git makepkg.conf makepkg.conf
index 7b2b1f4..509a9a0 100644
--- makepkg.conf
+++ makepkg.conf
@@ -35,20 +35,19 @@ VCSCLIENTS=('bzr::breezy'
 # ARCHITECTURE, COMPILE FLAGS
 #########################################################################
 #
-CARCH="x86_64"
-CHOST="x86_64-pc-linux-gnu"
+CARCH="riscv64"
+CHOST="riscv64-unknown-linux-gnu"
 
 #NPROC=2
 
 #-- Compiler and Linker Flags
 #CPPFLAGS=""
-CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
+CFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe -fno-plt -fexceptions \
         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security \
-        -fstack-clash-protection -fcf-protection \
-        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"
+        -fstack-clash-protection \
+        -fno-omit-frame-pointer"
 CXXFLAGS="$CFLAGS -Wp,-D_GLIBCXX_ASSERTIONS"
-LDFLAGS="-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now \
-         -Wl,-z,pack-relative-relocs"
+LDFLAGS="-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now"
 LTOFLAGS="-flto=auto"
 #-- Make Flags: change this for DistCC/SMP systems
 #MAKEFLAGS="-j2"
