diff --git PKGBUILD PKGBUILD
index 8cfba72..dfadae4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,7 @@ sha256sums=('7d8e3e8c5121aec0965df71f59bedf46052c6cf14f96365c4411ec3de0a4c1a5'
             'f3d4f39ef24e312f610cbb3439fb02bc6b8829e37bcf1a50ae50cd0a69bde5d0'
             'd87d0c9957c613fda272553bee58140349d151ae399f346ddaf6d75ee5916312'
             '656c4d4cb8cb12adbf178fc8cb2fd25f8c285d6572bbdbb24d865d00e0d5a85a'
-            '072020e34f2c55b94a9a486829a7eadab0a830ddb4d8e759b0c4e6cf1bde73a6')
+            'd119cf71f6ba52aa75dbbe58ba980075f9850486375b5d285c6eb306393691f6')
 
 prepare() {
   cd "${pkgname}-${pkgver}"
diff --git makepkg.conf makepkg.conf
index f270d8e..5ef7243 100644
--- makepkg.conf
+++ makepkg.conf
@@ -33,14 +33,14 @@ VCSCLIENTS=('bzr::bzr'
 # ARCHITECTURE, COMPILE FLAGS
 #########################################################################
 #
-CARCH="x86_64"
-CHOST="x86_64-pc-linux-gnu"
+CARCH="riscv64"
+CHOST="riscv64-unknown-linux-gnu"
 
 #-- Compiler and Linker Flags
 #CPPFLAGS=""
-CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
+CFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe -fno-plt -fexceptions \
         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security \
-        -fstack-clash-protection -fcf-protection"
+        -fstack-clash-protection"
 CXXFLAGS="$CFLAGS -Wp,-D_GLIBCXX_ASSERTIONS"
 LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now"
 LTOFLAGS="-flto=auto"
