diff --git PKGBUILD PKGBUILD
index 5a663f885..de81ee90e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,16 @@ depends=('python-mistune1' 'python-docutils')
 makedepends=('python-setuptools')
 checkdepends=('python-pygments' 'python-pytest-runner')
 source=("https://github.com/miyakogi/m2r/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
-        python310.patch)
+        python310.patch
+        "fix-docutils.patch")
 sha512sums=('847f04538ee52f0b0a88cee9474ed889e8114a3a7b1fb221cdaa867ab70ca52f517250f2c8bbf795ae598c7b6614e6d03d9f6f05b7e3d56fb8087d63eb52f1bc'
-            '3eeba63f27c0a4441948bfe3d55cf1db82ecc136a2ceafc054e68920f1da394df79d76c3fa8216effa516d1653f2bfda9f2f980cdca96d3f38468d18818c8bc1')
+            '3eeba63f27c0a4441948bfe3d55cf1db82ecc136a2ceafc054e68920f1da394df79d76c3fa8216effa516d1653f2bfda9f2f980cdca96d3f38468d18818c8bc1'
+            '51af1b8e26da84e04bb131b61817dd10a26f34ca3753cef250bc376fb8a2854bf60e6e35e4ca3ffa84ff312ed5b93401ef0d29ee3467c83e813784d9addec77b')
 
 prepare() {
   cd m2r-$pkgver
   patch -Np1 -i ../python310.patch
+  patch -Np1 -i "../fix-docutils.patch"
 }
 
 build() {
