--- PKGBUILD
+++ PKGBUILD
@@ -14,18 +14,24 @@ makedepends=('cargo' 'python-sphinx')
 install=sn0int.install
 options=(!lto)
 source=(https://github.com/kpcyrd/${pkgname}/archive/v${pkgver}/${pkgname}-${pkgver}.tar.gz
-        https://github.com/kpcyrd/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz.asc)
+        https://github.com/kpcyrd/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz.asc
+        $pkgname-support-riscv64.patch::https://github.com/kpcyrd/sn0int/pull/234.patch)
 sha512sums=('d932a55cbacb5da45d62f08df81179d49d1c974cd213fb629aa080cd9c8ca67c42ea8db63cfdaea62b49eaa1980711533844f1f744a8cd6177e4df1f19b118c4'
-            'SKIP')
+            'SKIP'
+            '18e7f81bd414e5d5f369fa89dc4ba1a290098dbaa9670a97f66dba4c83c4c5af3a52345f9d15454f184ef667df28bb835f96638a3bd60e4654817c35df58c14d')
 b2sums=('af11dc3b2b774b4929381d34c0458df05f61c3d1325027ada522add20a8ed75f2ed5784ab6444ebecd398206a860315b5831a7ff2bc604991e43087c56fe6a46'
-        'SKIP')
+        'SKIP'
+        '0f702a3f750d2c10842e0d8fe21ac4b2584ce0546446ccb53e3ccfaadfd0b2a6aa86a1aebe8104d5824c625e0e157d0e85ed167b07a4bc1bd0786562f115b550')
 
 validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  patch -Np1 < ../$pkgname-support-riscv64.patch
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
