diff --git PKGBUILD PKGBUILD
index 414cff7..cb6fe54 100644
--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,8 @@ validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
 prepare() {
   cd "${pkgname}-${pkgver}"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
