--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,8 @@ url="https://mojikiban.ipa.go.jp/"
 license=('custom')
 groups=('ipa-fonts')
 #source=("https://oscdl.ipa.go.jp/IPAmjMincho/ipamjm${_pkgver}.zip")
-source=("https://mojikiban.ipa.go.jp/OSCDL/IPAmjMincho/ipamjm${_pkgver}.zip")
+#source=("https://mojikiban.ipa.go.jp/OSCDL/IPAmjMincho/ipamjm${_pkgver}.zip")
+source=("https://dforest.watch.impress.co.jp/library/i/ipamjfont/10750/ipamjm${_pkgver}.zip")
 sha1sums=('efbd663a9e6ddfdc6b43002d29b9dbd191878b69')
 
 package() {
