Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1019676)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 url='https://github.com/HewlettPackard/netperf/'
 license=('custom')
 arch=('x86_64')
-depends=('glibc' 'libsmbios' 'lksctp-tools')
+depends=('glibc' 'lksctp-tools')
 install=netperf.install
 source=("https://github.com/HewlettPackard/netperf/archive/$pkgname-$pkgver.tar.gz"
         'netserver.service'
