diff --git PKGBUILD.orig PKGBUILD
index 973f8b5..5dafbbe 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -12,9 +12,16 @@
 license=('GPL3')
 groups=('deepin')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/linuxdeepin/deepin-desktop-base/archive/$pkgver.tar.gz"
-        distribution.info)
+        distribution.info
+        add_riscv_support.patch)
 sha512sums=('9fc903c6ae8f41d07fd6c4afd70f613788c5adfb4381e0c9e72d5c303883b2c8ab541c12ed448641174860dcda098ca5c2729397dd9cd214f7ee839b4623a4ed'
-            '27625e6d0786b8adacdb7c52806d4faa28d2ab6b319a593b3ea9bcb69f0cc18ea19b258d629e3a0069ef9a69503589b0285289caef39a1e85bbd99e915c7cd7d')
+            '27625e6d0786b8adacdb7c52806d4faa28d2ab6b319a593b3ea9bcb69f0cc18ea19b258d629e3a0069ef9a69503589b0285289caef39a1e85bbd99e915c7cd7d'
+            '5271d55e92a2b98db716ffdcc88b53d4cbb3939a782d0343d59c7d3de0610ca9d42446b368e33235237600108ed5a6701f62e7ef8780b9f65c1b9ad9492e6fdb')
+
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../add_riscv_support.patch
+}
 
 build() {
   cd $pkgname-$pkgver
