diff --git a/PKGBUILD b/PKGBUILD
index 09d5686..3325383 100644
--- PKGBUILD
+++ PKGBUILD
@@ -90,6 +90,11 @@ pkgver() {
 prepare() {
   # adjust group used for udev rules from sudo to wheel
   sed 's/sudo/wheel/g' -i $pkgname/resources/99-$pkgname.rules
+
+  # Backport https://github.com/qtile/qtile/pull/4991
+  # See https://github.com/qtile/qtile/issues/4987
+  cd $pkgname
+  git cherry-pick -n -1 52d920ad24b53374a78b614856978312cefa1591
 }
 
 build() {
