--- PKGBUILD
+++ PKGBUILD
@@ -89,6 +89,13 @@
   'A6BAA1E17D2664ADB97B2C6F58A9AA7C86727DF7' # elParaguayo <elparaguayocode@gmail.com>
 )
 
+prepare() {
+  patch -Np1 -d $pkgname -i "$srcdir/wait-for-widget-initialization.patch"
+
+  # Allow startup and hook checks more time on slower builders.
+  sed -i 's/^max_sleep = 5\.0$/max_sleep = 15.0/' $pkgname/test/helpers.py
+}
+
 build() {
   export CFLAGS="$CFLAGS -I/usr/include/wlroots0.20"
   export LDFLAGS="$LDFLAGS -L/usr/lib/wlroots0.20"
@@ -125,3 +132,8 @@
   install -vDm 644 "resources/$pkgname-"{generic,wayland}.desktop -t "$pkgdir/usr/share/wayland-sessions/"
   install -vDm 644 "resources/99-$pkgname.rules" -t "$pkgdir/usr/lib/udev/rules.d/"
 }
+
+source+=(wait-for-widget-initialization.patch)
+sha512sums+=('e869a8fbefa4d0e03c4f192bc0fdad4156603620ad0454d2e77c9ad7f47ef85e8d0753111fa0808dfeac79ead24180d17119344f6d3922571039a559a4c5f34c')
+b2sums+=('583bb4f3430ff0acfc1d155335136c1f105b0a7d8160b5e26c2086859e32a64c7975db0fd9755d264059224767d6adba9d1bc9382c44a28d43e2952e0d1994e0')
+checkdepends+=(python-pytest-timeout)
