From 6f4771b2b2a1ba8f30795caca8f550b40f33b185 Mon Sep 17 00:00:00 2001
From: Eric Long <i@hack3r.moe>
Date: Mon, 18 Nov 2024 11:36:33 +0800
Subject: [PATCH 2/3] Increase test timeout

---
 scripts/build/options.mjs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/build/options.mjs b/scripts/build/options.mjs
index 0d3a2093359..c9c8a9d467e 100644
--- a/scripts/build/options.mjs
+++ b/scripts/build/options.mjs
@@ -24,7 +24,7 @@ const parsed = minimist(process.argv.slice(2), {
         inspect: process.env.inspect || process.env["inspect-brk"] || process.env.i,
         host: process.env.TYPESCRIPT_HOST || process.env.host || "node",
         browser: process.env.browser || process.env.b || (os.platform() === "win32" ? "edge" : "chrome"),
-        timeout: +(process.env.timeout ?? 0) || 40000,
+        timeout: +(process.env.timeout ?? 0) || 4000000,
         tests: process.env.test || process.env.tests || process.env.t,
         runners: process.env.runners || process.env.runner || process.env.ru,
         light: process.env.light === undefined || process.env.light !== "false",
-- 
2.47.0

