From 8a80022158ff8e7d5c6997a4e631066921b1cbb6 Mon Sep 17 00:00:00 2001
From: Eric Long <i@hack3r.moe>
Date: Mon, 18 Nov 2024 11:37:09 +0800
Subject: [PATCH 3/3] Remove typescript.d.ts baseline check

---
 src/testRunner/unittests/publicApi.ts | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/testRunner/unittests/publicApi.ts b/src/testRunner/unittests/publicApi.ts
index 011f32ae285..9e5f6ca8abd 100644
--- a/src/testRunner/unittests/publicApi.ts
+++ b/src/testRunner/unittests/publicApi.ts
@@ -23,10 +23,6 @@ describe("unittests:: Public APIs", () => {
         });
     }
 
-    describe("for the language service and compiler", () => {
-        verifyApi("typescript.d.ts");
-    });
-
     describe("for the language server", () => {
         verifyApi("tsserverlibrary.d.ts");
     });
-- 
2.47.0

