diff --git a/src/testRunner/unittests/publicApi.ts b/src/testRunner/unittests/publicApi.ts
index 7b7b4ecc507..a6ca3e8918f 100644
--- a/src/testRunner/unittests/publicApi.ts
+++ b/src/testRunner/unittests/publicApi.ts
@@ -6,27 +6,6 @@ import * as vfs from "../_namespaces/vfs.js";
 import { jsonToReadableText } from "./helpers.js";
 import { TestServerHost } from "./helpers/virtualFileSystemWithWatch.js";
 
-describe("unittests:: Public APIs", () => {
-    function verifyApi(fileName: string) {
-        const builtFile = `built/local/${fileName}`;
-        const api = `api/${fileName}`;
-        let fileContent: string;
-        before(() => {
-            fileContent = Harness.IO.readFile(builtFile)!;
-            if (!fileContent) throw new Error(`File ${fileName} was not present in built/local`);
-            fileContent = fileContent.replace(/\r\n/g, "\n");
-        });
-
-        it("should be acknowledged when they change", () => {
-            Harness.Baseline.runBaseline(api, fileContent, { PrintDiff: true });
-        });
-    }
-
-    describe("for typescript.d.ts", () => {
-        verifyApi("typescript.d.ts");
-    });
-});
-
 describe("unittests:: Public APIs:: token to string", () => {
     function assertDefinedTokenToString(initial: ts.SyntaxKind, last: ts.SyntaxKind) {
         for (let t = initial; t <= last; t++) {
