diff --git a/src/testRunner/unittests/publicApi.ts b/src/testRunner/unittests/publicApi.ts
index 1b5ab93f0a1..bff42f0b028 100644
--- a/src/testRunner/unittests/publicApi.ts
+++ b/src/testRunner/unittests/publicApi.ts
@@ -20,10 +20,6 @@ describe("unittests:: Public APIs", () => {
         });
     }
 
-    describe("for the language service and compiler", () => {
-        verifyApi("typescript.d.ts");
-    });
-
     describe("for the language server", () => {
         verifyApi("tsserverlibrary.d.ts");
     });
