--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,20 @@ b2sums=('c0361e8140820040a5a10adc43dbd2aaf29bff8a1da04993a33c967b11b9ac841b7c3ff
 
 prepare() {
   cd $_name
+
+  # Remove dprint from dependencies and disable formatting when building code
+  # dprint itself requires prebuilt binary available, and some of its plugins uses prebuilts as well
+  # Building dprint from source from NPM package is upstreamed to https://github.com/dprint/dprint/pull/820
+  patch -Np1 -i ../0001-Remove-dprint.patch
+  sed -i -e 's/dprint fmt/:/' -e '/dprint/ d' package.json
+
+  # Increase test timeout from 40s to 4000s
+  patch -Np1 -i ../0002-Increase-test-timeout.patch
+
+  # Disable baseline check for typescript.d.ts
+  # This should not affect functionality since the diffs are solely code format variations
+  patch -Np1 -i ../0003-Remove-typescript.d.ts-baseline-check.patch
+
   npm ci
 }
 
@@ -46,3 +60,10 @@ package() {
   install -Dm644 -t "$pkgdir"/usr/share/licenses/$pkgname \
     ThirdPartyNoticeText.txt
 }
+
+source+=("0001-Remove-dprint.patch"
+         "0002-Increase-test-timeout.patch"
+         "0003-Remove-typescript.d.ts-baseline-check.patch")
+b2sums+=('ccdf17199b8e5ea2b0c16750e73666609712d2bb6df2d662a4324e1ce157ec5125a544b2b72ee2532db9feb89df30c046c74fdf3f5e618b95847e0bb3e9aef8c'
+         '1d42d2d94803fcec7665b98a70f502cdd90d0bb89aeb52e516bb1615c6f3d24deb79d05b6821908be408dbf63043ef59c8a34ec1426d80898d2f4f14cd985a13'
+         '8b4542a9bc35bc7927a757ca6653b4a1640424424d94a2fc32bee67984269d7f128550063998211bda3602d3fb9afebee4a1add48dd98d678e384f9e6f4606e6')
