--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,30 @@ url=http://www.typescriptlang.org
 license=('Apache-2.0')
 depends=('nodejs')
 makedepends=('git' 'npm' 'rsync')
-source=("git+https://github.com/microsoft/$_name.git#tag=v$pkgver")
-b2sums=('91c73cbadce2c602dc3fe78c60d6eef17b157af4aa9ee8cb535b78bfc1ae71a346d202133541872b0436a756b537550c5f0d883e036178776e09f9445e82a817')
+source=("git+https://github.com/microsoft/$_name.git#tag=v$pkgver"
+        "remove-dprint.patch"
+        "increase-timeout.patch"
+        "remove-typescript.d.ts-baseline-check.patch")
+b2sums=('91c73cbadce2c602dc3fe78c60d6eef17b157af4aa9ee8cb535b78bfc1ae71a346d202133541872b0436a756b537550c5f0d883e036178776e09f9445e82a817'
+        '03824f41aaa38cae3965cf2285465e0590d638ffbfe5019070c8a75d4e58ef4d44c92a1915f55bfa8f6bd0addddb99b8abc08c9d50cc5239faafe171e0740328'
+        '216abfce117ae52b4dbc49bd94eac6096c1606a1b50fab90af94f6611acad8e33e4ef497c6f8002af078caa9297ac84a2c4291b620836b4007b02f4add28bc98'
+        '38dc4bfd4a53613eb6406b4966d39a7c26f49fdb3eee626739773a0f16bd177e0ba735e70a34b8fb4bfbdd745392a0ac09429d2e21086e04bd28ba6cbaa080ef')
 
 prepare() {
   cd $_name
+
+  # Remove dprint from dependencies and disable formatting when building code
+  # dprint itself requires prebuilt binary available, and some of its plugins uses prebuilts as well
+  # Building dprint from source from NPM package is upstreamed to https://github.com/dprint/dprint/pull/820
+  patch -Np1 -i ../remove-dprint.patch
+
+  # Increase test timeout from 40s to 4000s
+  patch -Np1 -i ../increase-timeout.patch
+
+  # Disable baseline check for typescript.d.ts
+  # This should not affect functionality since the diffs are solely code format variations
+  patch -Np1 -i ../remove-typescript.d.ts-baseline-check.patch
+
   npm ci
 }
 
