--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,19 @@ b2sums=('d2e131993d301e5ef623fe3f87e999f8514e060cf52583003ecc2247bb416d3aad4798d
 
 prepare() {
   cd $_name
+
+  # Remove dprint from dependencies and disable formatting when building code
+  # dprint itself requires prebuilt binary available, and some of its plugins uses prebuilts as well
+  # Building dprint from source from NPM package is upstreamed to https://github.com/dprint/dprint/pull/820
+  patch -Np1 -i ../0001-Remove-dprint.patch
+
+  # Increase test timeout from 40s to 4000s
+  patch -Np1 -i ../0002-Increase-test-timeout.patch
+
+  # Disable baseline check for typescript.d.ts
+  # This should not affect functionality since the diffs are solely code format variations
+  patch -Np1 -i ../0003-Remove-typescript.d.ts-baseline-check.patch
+
   npm ci
 }
 
@@ -46,3 +59,10 @@ package() {
   install -Dm644 -t "$pkgdir"/usr/share/licenses/$pkgname \
     ThirdPartyNoticeText.txt
 }
+
+source+=("0001-Remove-dprint.patch"
+         "0002-Increase-test-timeout.patch"
+         "0003-Remove-typescript.d.ts-baseline-check.patch")
+b2sums+=('d33e1a50820247890f96bcdc94d670e52a9fbb4e765cb6740aef14c84f13e5490ad44b5e8d212b982c84eaf39d97da9d863e7a9597a66f73d92fa8f439de306a'
+         '1d42d2d94803fcec7665b98a70f502cdd90d0bb89aeb52e516bb1615c6f3d24deb79d05b6821908be408dbf63043ef59c8a34ec1426d80898d2f4f14cd985a13'
+         'd1b9786699d3861bacc4bf8ff8022fbc75c79a6598cb5209dc5024714870d6a6c9132b2e53a876b68a6812c062ecff65d6e158962f397aa332eb019d24907164')
