--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ license=(GPL3)
 depends=(kitemviews kitemmodels ki18n kcompletion kguiaddons kcrash qt5-svg qt5-multimedia quazip
          gsl libraw exiv2 openexr fftw openjpeg2 opencolorio libwebp hicolor-icon-theme)
 makedepends=(extra-cmake-modules kdoctools boost eigen poppler-qt5 python-pyqt5 libheif
-             qt5-tools sip kseexpr libmypaint libjxl xsimd)
+             qt5-tools sip kseexpr libmypaint libjxl)
 optdepends=('poppler-qt5: PDF filter' 'ffmpeg: to save animations'
             'python-pyqt5: for the Python plugins' 'libheif: HEIF filter'
             'kseexpr: SeExpr generator layer' 'kimageformats: PSD support' 'libmypaint: support for MyPaint brushes'
