--- PKGBUILD
+++ PKGBUILD
@@ -19 +19,2 @@ optdepends=('poppler-qt5: PDF filter' 'ffmpeg: to save animations'
-source=(https://download.kde.org/stable/krita/$_pkgver/$pkgname-$_pkgver.tar.gz{,.sig})
+source=(https://download.kde.org/stable/krita/$_pkgver/$pkgname-$_pkgver.tar.gz{,.sig}
+        $pkgname-support-compiling-on-risc-v.patch::https://invent.kde.org/graphics/krita/-/merge_requests/1416.patch)
@@ -21 +22,2 @@ sha256sums=('ed1d1fa5073ee0c4150e1fde9e109382980ea94d32e5845286bbff20b6bf5e45'
-            'SKIP')
+            'SKIP'
+            '475048d165fed3408e90382aabe846fef4a63a1eb1ea1639caf9f67512c062ad')
@@ -26,0 +29,5 @@ options=(debug)
+prepare() {
+  cd $pkgname-$_pkgver
+  patch -Np1 -i ../$pkgname-support-compiling-on-risc-v.patch
+}
+
