diff --git PKGBUILD PKGBUILD
index 8e931eb4..63b7140c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,10 @@ prepare() {
   # create folder for build output
   mkdir -p build
 
+  go mod edit -replace github.com/pganalyze/pg_query_go/v2=github.com/r-value/pg_query_go/v2@875303a22c
+  go mod tidy
+  go mod edit -replace github.com/remyoudompheng/bigfft=github.com/r-value/bigfft@905acca3ec
+  go mod tidy
   # download dependencies
   go mod download
 }
