diff -urN a/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.guess b/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.guess
--- a/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.guess	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.guess	2026-08-24 15:32:56.455255309 +0000
@@ -1000,6 +1000,9 @@
     ppc:Linux:*:*)
 	echo powerpc-unknown-linux-gnu
 	exit ;;
+    riscv64:Linux:*:*)
+	echo riscv64-unknown-linux-gnu
+	exit ;;
     s390:Linux:*:* | s390x:Linux:*:*)
 	echo ${UNAME_MACHINE}-ibm-linux
 	exit ;;
diff -urN a/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.sub b/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.sub
--- a/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.sub	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/common/autoconf/build-aux/autoconf-config.sub	2026-08-24 15:32:56.455356856 +0000
@@ -302,6 +302,7 @@
 	| pdp10 | pdp11 | pj | pjl \
 	| powerpc | powerpc64 | powerpc64le | powerpcle | ppcbe \
 	| pyramid \
+	| riscv32 | riscv64 \
 	| score \
 	| sh | sh[1234] | sh[24]a | sh[23]e | sh[34]eb | sheb | shbe | shle | sh[1234]le | sh3ele \
 	| sh64 | sh64le \
@@ -383,6 +384,7 @@
 	| pdp10-* | pdp11-* | pj-* | pjl-* | pn-* | power-* \
 	| powerpc-* | powerpc64-* | powerpc64le-* | powerpcle-* | ppcbe-* \
 	| pyramid-* \
+	| riscv32-* | riscv64-* \
 	| romp-* | rs6000-* \
 	| sh-* | sh[1234]-* | sh[24]a-* | sh[23]e-* | sh[34]eb-* | sheb-* | shbe-* \
 	| shle-* | sh[1234]le-* | sh3ele-* | sh64-* | sh64le-* \
diff -urN a/jdk8u-jdk8u504-b01/common/autoconf/platform.m4 b/jdk8u-jdk8u504-b01/common/autoconf/platform.m4
--- a/jdk8u-jdk8u504-b01/common/autoconf/platform.m4	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/common/autoconf/platform.m4	2026-08-24 15:33:26.380554511 +0000
@@ -393,6 +393,7 @@
     sparc*)  ZERO_ARCHDEF=SPARC ;;
     x86_64*) ZERO_ARCHDEF=AMD64 ;;
     x86)     ZERO_ARCHDEF=IA32  ;;
+    riscv*)  ZERO_ARCHDEF=RISCV  ;;
     *)      ZERO_ARCHDEF=$(echo "${OPENJDK_TARGET_CPU_LEGACY_LIB}" | tr a-z A-Z)
   esac
   AC_SUBST(ZERO_ARCHDEF)
diff -urN a/jdk8u-jdk8u504-b01/hotspot/src/os/linux/vm/os_linux.cpp b/jdk8u-jdk8u504-b01/hotspot/src/os/linux/vm/os_linux.cpp
--- a/jdk8u-jdk8u504-b01/hotspot/src/os/linux/vm/os_linux.cpp	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/hotspot/src/os/linux/vm/os_linux.cpp	2026-08-24 15:33:26.470554436 +0000
@@ -363,7 +363,7 @@
   //        1: ...
   //        ...
   //        7: The default directories, normally /lib and /usr/lib.
-#if defined(AMD64) || defined(_LP64) && (defined(SPARC) || defined(PPC) || defined(S390))
+#if defined(AMD64) || defined(_LP64) && (defined(SPARC) || defined(PPC) || defined(S390) || defined(RISCV))
   #define DEFAULT_LIBPATH "/usr/lib64:/lib64:/lib:/usr/lib"
 #else
 #if defined(AARCH64)
@@ -2058,10 +2058,12 @@
   }
 
 #ifndef S390
+#ifndef RISCV
   if (lib_arch.elf_class != arch_array[running_arch_index].elf_class) {
     ::snprintf(diag_msg_buf, diag_msg_max_length-1," (Possible cause: architecture word width mismatch)");
     return NULL;
   }
+#endif // RISCV
 #endif // !S390
 
   if (lib_arch.compat_class != arch_array[running_arch_index].compat_class) {
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/share/bin/java.c b/jdk8u-jdk8u504-b01/jdk/src/share/bin/java.c
--- a/jdk8u-jdk8u504-b01/jdk/src/share/bin/java.c	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/share/bin/java.c	2026-08-24 15:32:56.456166513 +0000
@@ -145,7 +145,7 @@
 static int knownVMsCount = 0;
 static int knownVMsLimit = 0;
 
-static void GrowKnownVMs();
+static void GrowKnownVMs(int minimum);
 static int  KnownVMIndex(const char* name);
 static void FreeKnownVMs();
 static jboolean IsWildCardEnabled();
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/share/native/common/jni_util.h b/jdk8u-jdk8u504-b01/jdk/src/share/native/common/jni_util.h
--- a/jdk8u-jdk8u504-b01/jdk/src/share/native/common/jni_util.h	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/share/native/common/jni_util.h	2026-08-24 15:32:56.456268668 +0000
@@ -407,7 +407,7 @@
 
 int getFastEncoding();
 
-void initializeEncoding();
+void initializeEncoding(JNIEnv *env);
 
 void* getProcessHandle();
 
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.c b/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.c
--- a/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.c	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.c	2026-08-24 15:32:56.456868818 +0000
@@ -427,7 +427,7 @@
         return 0;
     }
 
-    if ((*sMlibFns[MLIB_CONVKERNCVT].fptr)(kdata, &scale, dkern, w, h,
+    if (((MlibConvKernelConvertFP_t)sMlibFns[MLIB_CONVKERNCVT].fptr)(kdata, &scale, dkern, w, h,
                                     mlib_ImageGetType(src)) != MLIB_SUCCESS) {
         freeArray(env, srcImageP, src, sdata, dstImageP, dst, ddata);
         awt_freeParsedImage(srcImageP, TRUE);
@@ -455,7 +455,7 @@
     }
 
     cmask = (1<<src->channels)-1;
-    status = (*sMlibFns[MLIB_CONVMxN].fptr)(dst, src, kdata, w, h,
+    status = ((MlibConvMxNFP_t)sMlibFns[MLIB_CONVMxN].fptr)(dst, src, kdata, w, h,
                                (w-1)/2, (h-1)/2, scale, cmask,
                                getMlibEdgeHint(edgeHint));
 
@@ -674,7 +674,7 @@
         return 0;
     }
 
-    if ((*sMlibFns[MLIB_CONVKERNCVT].fptr)(kdata, &scale, dkern, w, h,
+    if (((MlibConvKernelConvertFP_t)sMlibFns[MLIB_CONVKERNCVT].fptr)(kdata, &scale, dkern, w, h,
                                     mlib_ImageGetType(src)) != MLIB_SUCCESS) {
         freeDataArray(env, srcRasterP->jdata, src, sdata,
                       dstRasterP->jdata, dst, ddata);
@@ -703,7 +703,7 @@
     }
 
     cmask = (1<<src->channels)-1;
-    status = (*sMlibFns[MLIB_CONVMxN].fptr)(dst, src, kdata, w, h,
+    status = ((MlibConvMxNFP_t)sMlibFns[MLIB_CONVMxN].fptr)(dst, src, kdata, w, h,
                                (w-1)/2, (h-1)/2, scale, cmask,
                                getMlibEdgeHint(edgeHint));
 
@@ -919,7 +919,7 @@
                mlib_ImageGetWidth(dst)*mlib_ImageGetHeight(dst));
     }
     /* Perform the transformation */
-    if ((status = (*sMlibFns[MLIB_AFFINE].fptr)(dst, src, mtx, filter,
+    if ((status = ((MlibAffineFP_t)sMlibFns[MLIB_AFFINE].fptr)(dst, src, mtx, filter,
                                   MLIB_EDGE_SRC_EXTEND) != MLIB_SUCCESS))
     {
         printMedialibError(status);
@@ -1136,7 +1136,7 @@
     }
 
     /* Perform the transformation */
-    if ((status = (*sMlibFns[MLIB_AFFINE].fptr)(dst, src, mtx, filter,
+    if ((status = ((MlibAffineFP_t)sMlibFns[MLIB_AFFINE].fptr)(dst, src, mtx, filter,
                                   MLIB_EDGE_SRC_EXTEND) != MLIB_SUCCESS))
     {
         printMedialibError(status);
@@ -1493,8 +1493,7 @@
             }
         }
         /* How about ddata == null? */
-    }
-    else if ((status = (*sMlibFns[MLIB_LOOKUP].fptr)(dst, src,
+    } else if ((status = ((MlibLookUpFP_t)sMlibFns[MLIB_LOOKUP].fptr)(dst, src,
                                       (void **)tbl) != MLIB_SUCCESS)) {
         printMedialibError(status);
         retStatus = 0;
@@ -1732,7 +1731,7 @@
             }
         }
         /* How about ddata == null? */
-    } else if ((status = (*sMlibFns[MLIB_LOOKUP].fptr)(dst, src,
+    } else if ((status = ((MlibLookUpFP_t)sMlibFns[MLIB_LOOKUP].fptr)(dst, src,
                                       (void **)mlib_lookupTable) != MLIB_SUCCESS)) {
         printMedialibError(status);
         retStatus = 0;
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.h b/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.h
--- a/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.h	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/share/native/sun/awt/medialib/awt_ImagingLib.h	2026-08-24 15:32:56.456736868 +0000
@@ -32,10 +32,21 @@
 
 /* Struct that holds the mlib function ptrs and names */
 typedef struct {
-    mlib_status (*fptr)();
+    void *fptr;
     char *fname;
 } mlibFnS_t;
 
+typedef mlib_status (*MlibConvMxNFP_t)(mlib_image *, const mlib_image *,
+                                       const mlib_s32 *, mlib_s32, mlib_s32,
+                                       mlib_s32, mlib_s32, mlib_s32, mlib_s32,
+                                       mlib_edge);
+typedef mlib_status (*MlibAffineFP_t)(mlib_image *, const mlib_image *,
+                                      const mlib_d64 *, mlib_filter, mlib_edge);
+typedef mlib_status (*MlibLookUpFP_t)(mlib_image *, const mlib_image *,
+                                      const void **);
+typedef mlib_status (*MlibConvKernelConvertFP_t)(mlib_s32 *, mlib_s32 *,
+                                                 const mlib_d64 *, mlib_s32, mlib_s32, mlib_type);
+
 typedef mlib_image *(*MlibCreateFP_t)(mlib_type, mlib_s32, mlib_s32,
                                        mlib_s32);
 typedef mlib_image *(*MlibCreateStructFP_t)(mlib_type, mlib_s32, mlib_s32,
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/awt_GraphicsEnv.h b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/awt_GraphicsEnv.h
--- a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/awt_GraphicsEnv.h	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/awt_GraphicsEnv.h	2026-08-24 15:32:56.456322414 +0000
@@ -50,8 +50,6 @@
 #define MITSHM_PERM_COMMON (0666)
 #define MITSHM_PERM_OWNER  (0600)
 
-extern int XShmQueryExtension();
-
 void TryInitMITShm(JNIEnv *env, jint *shmExt, jint *shmPixmaps);
 void resetXShmAttachFailed();
 jboolean isXShmAttachFailed();
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk2_interface.c b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk2_interface.c
--- a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk2_interface.c	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk2_interface.c	2026-08-24 15:32:56.456419410 +0000
@@ -1468,7 +1468,7 @@
             if (init_result = (NULL == gtk2_widgets[_GTK_NOTEBOOK_TYPE]))
             {
                 gtk2_widgets[_GTK_NOTEBOOK_TYPE] =
-                    (*fp_gtk_notebook_new)(NULL);
+                    (*fp_gtk_notebook_new)();
             }
             result = gtk2_widgets[_GTK_NOTEBOOK_TYPE];
             break;
@@ -1476,7 +1476,7 @@
             if (init_result = (NULL == gtk2_widgets[_GTK_TOGGLE_BUTTON_TYPE]))
             {
                 gtk2_widgets[_GTK_TOGGLE_BUTTON_TYPE] =
-                    (*fp_gtk_toggle_button_new)(NULL);
+                    (*fp_gtk_toggle_button_new)();
             }
             result = gtk2_widgets[_GTK_TOGGLE_BUTTON_TYPE];
             break;
@@ -1485,7 +1485,7 @@
             if (init_result = (NULL == gtk2_widgets[_GTK_TOOLBAR_TYPE]))
             {
                 gtk2_widgets[_GTK_TOOLBAR_TYPE] =
-                    (*fp_gtk_toolbar_new)(NULL);
+                    (*fp_gtk_toolbar_new)();
             }
             result = gtk2_widgets[_GTK_TOOLBAR_TYPE];
             break;
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk3_interface.c b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk3_interface.c
--- a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk3_interface.c	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/awt/gtk3_interface.c	2026-08-24 15:32:56.456605382 +0000
@@ -1241,7 +1241,7 @@
             if (init_result = (NULL == gtk3_widgets[_GTK_NOTEBOOK_TYPE]))
             {
                 gtk3_widgets[_GTK_NOTEBOOK_TYPE] =
-                    (*fp_gtk_notebook_new)(NULL);
+                    (*fp_gtk_notebook_new)();
             }
             result = gtk3_widgets[_GTK_NOTEBOOK_TYPE];
             break;
@@ -1249,7 +1249,7 @@
             if (init_result = (NULL == gtk3_widgets[_GTK_TOGGLE_BUTTON_TYPE]))
             {
                 gtk3_widgets[_GTK_TOGGLE_BUTTON_TYPE] =
-                    (*fp_gtk_toggle_button_new)(NULL);
+                    (*fp_gtk_toggle_button_new)();
             }
             result = gtk3_widgets[_GTK_TOGGLE_BUTTON_TYPE];
             break;
@@ -1258,7 +1258,7 @@
             if (init_result = (NULL == gtk3_widgets[_GTK_TOOLBAR_TYPE]))
             {
                 gtk3_widgets[_GTK_TOOLBAR_TYPE] =
-                    (*fp_gtk_toolbar_new)(NULL);
+                    (*fp_gtk_toolbar_new)();
             }
             result = gtk3_widgets[_GTK_TOOLBAR_TYPE];
             break;
diff -urN a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/net/spi/DefaultProxySelector.c b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/net/spi/DefaultProxySelector.c
--- a/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/net/spi/DefaultProxySelector.c	2026-08-10 15:49:49.000000000 +0000
+++ b/jdk8u-jdk8u504-b01/jdk/src/solaris/native/sun/net/spi/DefaultProxySelector.c	2026-08-24 15:32:56.457051175 +0000
@@ -95,11 +95,11 @@
 typedef struct _GSocketConnectable GSocketConnectable;
 typedef struct GError GError;
 typedef GProxyResolver* g_proxy_resolver_get_default_func();
-typedef char** g_proxy_resolver_lookup_func();
-typedef GSocketConnectable* g_network_address_parse_uri_func();
-typedef const char* g_network_address_get_hostname_func();
-typedef unsigned short g_network_address_get_port_func();
-typedef void g_strfreev_func();
+typedef char** g_proxy_resolver_lookup_func(GProxyResolver*, const char*, void*, GError**);
+typedef GSocketConnectable* g_network_address_parse_uri_func(const char*, unsigned short, GError**);
+typedef const char* g_network_address_get_hostname_func(GSocketConnectable*);
+typedef unsigned short g_network_address_get_port_func(GSocketConnectable*);
+typedef void g_strfreev_func(char**);
 
 static g_proxy_resolver_get_default_func* g_proxy_resolver_get_default = NULL;
 static g_proxy_resolver_lookup_func* g_proxy_resolver_lookup = NULL;
