--- PKGBUILD
+++ PKGBUILD
@@ -38,6 +38,7 @@
 case "${CARCH}" in
   'x86_64') _JARCH=amd64 ; _DOC_ARCH=x86_64 ;;
   'i686'  ) _JARCH=i386  ; _DOC_ARCH=x86    ;;
+  'riscv64')_JARCH=riscv64;_DOC_ARCH=riscv64;;
 esac
 
 _jdkname=openjdk8
@@ -54,6 +55,10 @@
 
   # Fix build with C++17 (Fedora)
   patch -Np1 -i "${srcdir}"/gcc11.patch
+
+  # RISC-V Support
+  patch -Np1 -i "${srcdir}"/java8-openjdk-add-riscv-support.patch
+  (cd common/autoconf && bash ./autogen.sh)
 }
 
 build() {
@@ -79,7 +84,9 @@
     --with-zlib=system \
     --with-extra-cflags="${CFLAGS}" \
     --with-extra-cxxflags="${CXXFLAGS}" \
-    --with-extra-ldflags="${LDFLAGS}"
+    --with-extra-ldflags="${LDFLAGS}" \
+    --with-jvm-variants=zero \
+    --disable-jfr
 
   # These help to debug builds: LOG=trace HOTSPOT_BUILD_JOBS=1
   # Without 'DEBUG_BINARIES', i686 won't build: http://mail.openjdk.java.net/pipermail/core-libs-dev/2013-July/019203.html
@@ -275,8 +282,11 @@
   pkgdesc='OpenJDK Java 8 documentation'
 
   install -d -m 755 "${pkgdir}/usr/share/doc/${pkgbase}/"
-  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-server-release/docs/* \
+  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-zero-release/docs/* \
     "${pkgdir}/usr/share/doc/${pkgbase}/"
 }
 
+source+=(java8-openjdk-add-riscv-support.patch::https://gitee.com/misaka00251/openjdk-1.8.0/raw/e4b9d135939e2263146905911ad100424101e305/add-riscv-support.patch)
+b2sums+=('86cd7d1079944b29441227da00df2c4ee45ccf6c95a5c20eb9cc5e9d02e9fa6b6c8d26f69216b42d8e7832ad0f592a689464521765622f9355f4184ff87b6f06')
+
 # vim: ts=2 sw=2 et:
