--- PKGBUILD
+++ PKGBUILD
@@ -49,6 +49,7 @@ case "${CARCH}" in
   'x86_64'  ) _JARCH=amd64   ; _DOC_ARCH=x86_64  ;;
   'i686'    ) _JARCH=i386    ; _DOC_ARCH=x86     ;;
   'aarch64' ) _JARCH=aarch64 ; _DOC_ARCH=aarch64 ;;
+  'riscv64' ) _JARCH=riscv64 ; _DOC_ARCH=riscv64 ;;
 esac
 
 _jdkname=openjdk8
@@ -65,6 +66,10 @@ _nonheadless=(
 prepare() {
   cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}
 
+  # RISC-V Support
+  patch -Np2 -i "${srcdir}"/jdk8-riscv.patch
+  (cd common/autoconf && bash ./autogen.sh)
+
   # Do not treats warnings as errors
   sed -E -i 's/(^WARNINGS_ARE_ERRORS = -W)(error)/\1no-\2/' \
     hotspot/make/linux/makefiles/gcc.make
@@ -121,7 +126,9 @@ build() {
     --with-extra-cflags="${CFLAGS}" \
     --with-extra-cxxflags="${CXXFLAGS}" \
     --with-extra-ldflags="${LDFLAGS}" \
-    --with-vendor-name="Arch Linux"
+    --with-vendor-name="Arch Linux" \
+    --with-jvm-variants=zero \
+    --disable-jfr
 
   # These help to debug builds: LOG=trace HOTSPOT_BUILD_JOBS=1
   # Without 'DEBUG_BINARIES', i686 won't build: http://mail.openjdk.java.net/pipermail/core-libs-dev/2013-July/019203.html
@@ -347,8 +354,11 @@ package_openjdk8-doc() {
   pkgdesc='OpenJDK Java 8 documentation'
 
   install -d -m 755 "${pkgdir}/usr/share/doc/${pkgbase}/"
-  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-server-release/docs/* \
+  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-zero-release/docs/* \
     "${pkgdir}/usr/share/doc/${pkgbase}/"
 }
 
+source+=(jdk8-riscv.patch)
+b2sums+=('f4b11b447bc36cdd26d2b157d7c6c2311e5b98fcae0c9e8d0507345c31a8a8f20c6c2af356ae87594e4d37fd5e7ba9d93441f273ca0578132c7ffb24a9b983f9')
+
 # vim: ts=2 sw=2 et:
