--- PKGBUILD
+++ PKGBUILD
@@ -50,6 +50,7 @@
   'x86_64'  ) _JARCH=amd64   ; _DOC_ARCH=x86_64  ;;
   'i686'    ) _JARCH=i386    ; _DOC_ARCH=x86     ;;
   'aarch64' ) _JARCH=aarch64 ; _DOC_ARCH=aarch64 ;;
+  'riscv64' ) _JARCH=riscv64 ; _DOC_ARCH=riscv64 ;;
 esac
 
 _jdkname=openjdk8
@@ -66,6 +67,10 @@
 prepare() {
   cd jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}
 
+  # RISC-V Support
+  patch -Np2 -i "${srcdir}"/jdk8-riscv.patch
+  (cd common/autoconf && bash ./autogen.sh)
+
   # Do not treats warnings as errors
   sed -E -i 's/(^WARNINGS_ARE_ERRORS = -W)(error)/\1no-\2/' \
     hotspot/make/linux/makefiles/gcc.make
@@ -112,7 +117,9 @@
     --with-extra-cflags="${CFLAGS}" \
     --with-extra-cxxflags="${CXXFLAGS}" \
     --with-extra-ldflags="${LDFLAGS}" \
-    --with-vendor-name="Arch Linux"
+    --with-vendor-name="Arch Linux" \
+    --with-jvm-variants=zero \
+    --disable-jfr
 
   # These help to debug builds: LOG=trace HOTSPOT_BUILD_JOBS=1
   # Without 'DEBUG_BINARIES', i686 won't build: http://mail.openjdk.java.net/pipermail/core-libs-dev/2013-July/019203.html
@@ -338,8 +345,13 @@
   pkgdesc='OpenJDK Java 8 documentation'
 
   install -d -m 755 "${pkgdir}/usr/share/doc/${pkgbase}/"
-  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-server-release/docs/* \
+  cp -r "${srcdir}"/jdk8u-jdk${_majorver}u${_minorver}-b${_updatever}/build/linux-${_DOC_ARCH}-normal-zero-release/docs/* \
     "${pkgdir}/usr/share/doc/${pkgbase}/"
 }
 
 # vim: ts=2 sw=2 et:
+
+source+=(jdk8-riscv.patch)
+b2sums+=('640cdcb5f2e1df972d7561b1f76013a29c97cab3fb3b78996da5522c493904fffd965e6aada50215906d27c4c50d93facf82260036bec8345814a7ad9d0def60')
+
+# vim: ts=2 sw=2 et:
