--- PKGBUILD
+++ PKGBUILD
@@ -67,6 +67,9 @@
 
 prepare() {
   cd $pkgbase
+
+  # Allow more time for the four-thread summary save/reload stress test.
+  patch -Np1 -i ../increase-stress-test-timeout.patch
 }
 
 build() {
@@ -157,3 +160,6 @@
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(increase-stress-test-timeout.patch)
+b2sums+=('6941ae2d24d1353aac30435a177a113c65e98e45569c579399284e227afc4f42b059ba1bd27915d2591b4bb18415df0706122c0282522ea2ef00e1dd72ca46d1')
