diff --unified --recursive --text a/plessey.c b/plessey.c
--- a/plessey.c	2021-09-07 22:28:50.150100945 +0000
+++ b/plessey.c	2021-09-07 22:29:50.963439996 +0000
@@ -149,7 +149,7 @@
                         checkptr[i+j] ^= check[j];
     }
     for (i = 0; i < 8; i++) {
-        sprintf(ptr, patterns[checkptr[strlen(text) * 4 + i]]);
+        sprintf(ptr, "%s", patterns[checkptr[strlen(text) * 4 + i]]);
 	ptr += 2;
     }
     fprintf(stderr, _("CRC: "));
