diff --git PKGBUILD PKGBUILD
index d9c653330..aabf8d4d0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,8 +8,15 @@ arch=('x86_64')
 url="https://www.gnu.org/software/barcode/"
 license=('GPL')
 depends=('glibc')
-source=("ftp://ftp.gnu.org/gnu/${pkgname}/${pkgname}-${pkgver}.tar.xz")
-md5sums=('cdc504ee1020e27fbfeebcb0718de054')
+source=("ftp://ftp.gnu.org/gnu/${pkgname}/${pkgname}-${pkgver}.tar.xz"
+        "fix-format-not-a-string-literal-and-no-format-arguments.patch")
+md5sums=('cdc504ee1020e27fbfeebcb0718de054'
+         'f9608182155a4205fd490c5dae66cd11')
+
+prepare(){
+  cd "${srcdir}/${pkgname}-${pkgver}"
+  patch -Np1 -i ../fix-format-not-a-string-literal-and-no-format-arguments.patch
+}
 
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
