--- PKGBUILD
+++ PKGBUILD
@@ -9,12 +9,15 @@ url="https://gitlab.archlinux.org/archlinux/pyalpm"
 license=('GPL')
 makedepends=('git' 'python-setuptools' 'python-pytest' 'python-pkgconfig' 'python-pytest-pacman')
 depends=('python' 'pacman')
-source=("git+https://gitlab.archlinux.org/archlinux/pyalpm.git#tag=$pkgver")
+source=("git+https://gitlab.archlinux.org/archlinux/pyalpm.git#tag=$pkgver"
+        pyalpm-fix-test.patch::https://gitlab.archlinux.org/archlinux/pyalpm/-/merge_requests/19.patch)
 validpgpkeys=('E499C79F53C96A54E572FEE1C06086337C50773E')
-sha512sums=('SKIP')
+sha512sums=('SKIP'
+            'f3549eddfef20d86757b5a425caba703b1d696da3752cd84597a61e4926bf9e367a88dc1e22b1894d210b5342a38259c770b4f89f6749c451fd2c50725361d02')
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../pyalpm-fix-test.patch
 }
 
 build() {
