--- PKGBUILD
+++ PKGBUILD
@@ -9,12 +9,15 @@ url="https://gitlab.archlinux.org/archlinux/pyalpm"
 license=('GPL')
 makedepends=('git' 'python-setuptools' 'python-pytest' 'python-pkgconfig' 'python-pytest-pacman')
 depends=('python' 'pacman')
-source=("git+https://gitlab.archlinux.org/archlinux/pyalpm.git#tag=$pkgver")
+source=("git+https://gitlab.archlinux.org/archlinux/pyalpm.git#tag=$pkgver"
+        pyalpm-fix-test.patch::https://gitlab.archlinux.org/archlinux/pyalpm/-/merge_requests/19.patch)
 validpgpkeys=('E499C79F53C96A54E572FEE1C06086337C50773E')
-sha512sums=('SKIP')
+sha512sums=('SKIP'
+            'b6d1ad51402d2b75f427938f33c1a89cc95a90294ffe2d8956b5c40aadb9a101dd0a2b15f3918b79b14f9b3ad2d54b01fb03efbf758d1d807bfccb9fcc5f7c78')
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../pyalpm-fix-test.patch
 }
 
 build() {
