diff --git PKGBUILD PKGBUILD
index 3f30931a9..81ff8166b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ pkgdesc='Lightweight C++ template library for vector and matrix math, a.k.a. lin
 arch=(any)
 url='https://eigen.tuxfamily.org'
 license=(GPL3 LGPL2.1 BSD custom:MPL2 custom:MINPACK)
-makedepends=(cmake freeglut gcc-fortran fftw suitesparse boost)
+makedepends=(cmake freeglut gcc-fortran fftw boost)
 source=(https://gitlab.com/libeigen/eigen/-/archive/$pkgver/$pkgname-$pkgver.tar.gz)
 sha256sums=('8586084f71f9bde545ee7fa6d00288b264a2b7ac3607b974e54d13e7162c1c72')
 
