--- PKGBUILD
+++ PKGBUILD
@@ -29,9 +29,6 @@
 _dirname="rocm-libraries-rocm-$pkgver/projects/$pkgname"
 
 build() {
-    # Compile source code for supported GPU archs in parallel
-    export HIPCC_COMPILE_FLAGS_APPEND="-parallel-jobs=$(nproc)"
-    export HIPCC_LINK_FLAGS_APPEND="-parallel-jobs=$(nproc)"
     # -fcf-protection is not supported by HIP, see
     # https://rocm.docs.amd.com/projects/llvm-project/en/latest/reference/rocmcc.html#support-status-of-other-clang-options
     local cmake_args=(
