diff --git a/ODIN_II/CMakeLists.txt b/ODIN_II/CMakeLists.txt
index 5bdc9ce..6c9ef8d 100644
--- a/ODIN_II/CMakeLists.txt
+++ b/ODIN_II/CMakeLists.txt
@@ -133,6 +133,7 @@ endif()
 add_executable(odin_II ${EXEC_SOURCES})
 
 target_link_libraries(odin_II
+                        -pthread
                         libodin_ii)
 
 #Supress IPO link warnings if IPO is enabled
diff --git a/libs/librtlnumber/Makefile b/libs/librtlnumber/Makefile
index 660e143..4647347 100644
--- a/libs/librtlnumber/Makefile
+++ b/libs/librtlnumber/Makefile
@@ -35,7 +35,7 @@ SRC =src/*.cpp
 
 BIN = rtl_number
 
-C = clang++ -std=c++14 -lpthread
+C = clang++ -std=c++14 -pthread
 
 cleanup_flags=\
 -ferror-limit=1000 \
