diff --git a/ODIN_II/SRC/include/Hashtable.hpp b/ODIN_II/SRC/include/Hashtable.hpp
index 228f41e..4e4de26 100644
--- a/ODIN_II/SRC/include/Hashtable.hpp
+++ b/ODIN_II/SRC/include/Hashtable.hpp
@@ -26,6 +26,7 @@ OTHER DEALINGS IN THE SOFTWARE.
 #include <stdlib.h>
 #include <stdint.h>
 #include <unordered_map>
+#include <string>
 
 class Hashtable
 {
diff --git a/libs/EXTERNAL/libargparse/src/argparse.cpp b/libs/EXTERNAL/libargparse/src/argparse.cpp
index 72071f1..3e144b1 100644
--- a/libs/EXTERNAL/libargparse/src/argparse.cpp
+++ b/libs/EXTERNAL/libargparse/src/argparse.cpp
@@ -4,6 +4,7 @@
 #include <cassert>
 #include <string>
 #include <set>
+#include <limits>
 
 #include "argparse.hpp"
 #include "argparse_util.hpp"
diff --git a/libs/libvtrutil/src/vtr_geometry.tpp b/libs/libvtrutil/src/vtr_geometry.tpp
index 0075a81..159b22c 100644
--- a/libs/libvtrutil/src/vtr_geometry.tpp
+++ b/libs/libvtrutil/src/vtr_geometry.tpp
@@ -1,3 +1,4 @@
+#include <limits>
 namespace vtr {
 /*
  * Point
