--- PKGBUILD
+++ PKGBUILD
@@ -97,3 +97,12 @@
   python -m installer --destdir="$pkgdir" dist/*.whl
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i "$srcdir/ftp-server-readiness.patch"
+}
+
+source+=(ftp-server-readiness.patch)
+sha512sums+=('91684a82e256987e4f44a7edbae2a96c148d3e9ed60e2f70d4a264d2e7cc7f1efd8683e70ef68659f89a1743ce3e7ede5451415b890e52523a346657db69952c')
+b2sums+=('30ef84f8fefa65bdc793a8b442e8d658f9b3da63e57ac1fc2066bf023e571e7be5bf37246aa11c88d7f153516c4a13e5cfc8a14cc55d02c6f2e700c103db1ef0')
