diff --git a/tools/jdk/BUILD.tools b/tools/jdk/BUILD.tools
index 85cb7b0..50d1b4e 100755
--- a/tools/jdk/BUILD.tools
+++ b/tools/jdk/BUILD.tools
@@ -106,8 +106,8 @@ cc_library(
         "//src/conditions:linux_aarch64": ["include/linux"],
         "//src/conditions:linux_ppc64le": ["include/linux"],
         "//src/conditions:linux_s390x": ["include/linux"],
-        "//src/conditions:linux_mips64": [":include/linux"],
-        "//src/conditions:linux_riscv64": [":include/linux"],
+        "//src/conditions:linux_mips64": ["include/linux"],
+        "//src/conditions:linux_riscv64": ["include/linux"],
         "//src/conditions:linux_x86_64": ["include/linux"],
         "//src/conditions:darwin": ["include/darwin"],
         "//src/conditions:freebsd": ["include/freebsd"],
