diff --git a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE
index a507f04..3be7b84 100755
--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE
+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE
@@ -83,6 +83,21 @@ maybe(
     version = "11",
 )
 
+# This must be kept in sync with the top-level WORKSPACE file.
+maybe(
+    remote_java_repository,
+    name = "remotejdk11_linux_riscv64",
+    exec_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//cpu:riscv64",
+    ],
+    strip_prefix = "java-11-openjdk",
+    urls = [
+        "file:///build/bazel/src/remotejdk11_linux_riscv64.tar.gz",
+    ],
+    version = "11",
+)
+
 # This must be kept in sync with the top-level WORKSPACE file.
 maybe(
     remote_java_repository,
diff --git a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl
index 7040eae..b783308 100755
--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl
+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl
@@ -66,6 +66,21 @@ maybe(
     version = "11",
 )
 
+# This must be kept in sync with the top-level WORKSPACE file.
+maybe(
+    remote_java_repository,
+    name = "remotejdk11_linux_riscv64",
+    exec_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//cpu:riscv64",
+    ],
+    strip_prefix = "java-11-openjdk",
+    urls = [
+        "file:///build/bazel/src/remotejdk11_linux_riscv64.tar.gz",
+    ],
+    version = "11",
+)
+
 # This must be kept in sync with the top-level WORKSPACE file.
 maybe(
     remote_java_repository,
diff --git a/tools/jdk/BUILD.tools b/tools/jdk/BUILD.tools
index ecddfe0..85cb7b0 100755
--- a/tools/jdk/BUILD.tools
+++ b/tools/jdk/BUILD.tools
@@ -447,6 +447,7 @@ _JDKS = [
     "remotejdk11_linux",
     "remotejdk11_linux_ppc64le",
     "remotejdk11_linux_s390x",
+    "remotejdk11_linux_riscv64",
     "remotejdk15_macos",
     "remotejdk15_macos_aarch64",
     "remotejdk15_win",
