--- repositories.bzl.orig	2025-02-07 23:47:06.934119986 -0500
+++ repositories.bzl	2025-02-07 23:46:35.347392689 -0500
@@ -97,6 +97,12 @@
         url = "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.5%2B11/OpenJDK21U-jdk_ppc64le_linux_hotspot_21.0.5_11.tar.gz",
     )
     http_file(
+        name = "openjdk_linux_riscv64_vanilla",
+        integrity = "sha256-Lxs+QB423oAzmN+5gYhh+fFMqK59tlDqCUarBI/v47k=",
+        downloaded_file_path = "adoptopenjdk-riscv64-vanilla.tar.gz",
+        url = "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.5%2B11/OpenJDK21U-jdk_riscv64_linux_hotspot_21.0.5_11.tar.gz",
+    )
+    http_file(
         name = "openjdk_macos_x86_64_vanilla",
         integrity = "sha256-zXTl63OMkUWdS45eEOrJGK4qBdIan8fKXcLaZOZbzr0=",
         downloaded_file_path = "zulu-macos-vanilla.tar.gz",
--- MODULE.bazel.orig	2025-02-07 23:43:20.230351495 -0500
+++ MODULE.bazel	2025-02-07 23:44:09.823780023 -0500
@@ -289,6 +289,7 @@
     "openjdk_linux_aarch64_vanilla",
     "openjdk_linux_ppc64le_vanilla",
     "openjdk_linux_s390x_vanilla",
+    "openjdk_linux_riscv64_vanilla",
     "openjdk_linux_vanilla",
     "openjdk_macos_aarch64_vanilla",
     "openjdk_macos_x86_64_vanilla",
--- src/minimize_jdk.sh.orig	2025-02-07 23:42:09.306882024 -0500
+++ src/minimize_jdk.sh	2025-02-07 23:56:12.061833025 -0500
@@ -40,8 +40,8 @@
 fi
 fulljdk=$1
 out=$3
-ARCH=`uname -p`
-if [[ "${ARCH}" == 'ppc64le'  ]] || [[ "${ARCH}" == 's390x' ]]; then
+ARCH=`uname -m`
+if [[ "${ARCH}" == 'ppc64le'  ]] || [[ "${ARCH}" == 's390x' ]] || [[ "${ARCH}" == 'riscv64' ]]; then
   FULL_JDK_DIR="jdk*"
   DOCS=""
 else
--- src/BUILD.orig	2025-02-07 23:43:09.086996772 -0500
+++ src/BUILD	2025-02-07 23:25:57.775017168 -0500
@@ -173,6 +173,9 @@
         "//src/conditions:linux_s390x": [
             "@openjdk_linux_s390x_vanilla//file",
         ],
+        "//src/conditions:linux_riscv64": [
+            "@openjdk_linux_riscv64_vanilla//file",
+        ],
         "//conditions:default": [
             "@openjdk_linux_vanilla//file",
         ],
