diff --git a/MODULE.bazel b/MODULE.bazel
index b2d5b2d3d144b8..bae117635f702c 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -296,6 +296,7 @@ use_repo(
     "openjdk_linux_aarch64_vanilla",
     "openjdk_linux_ppc64le_vanilla",
     "openjdk_linux_s390x_vanilla",
+    "openjdk_linux_riscv64_vanilla",
     "openjdk_linux_vanilla",
     "openjdk_macos_aarch64_vanilla",
     "openjdk_macos_x86_64_vanilla",
diff --git a/repositories.bzl b/repositories.bzl
index 62512117f865c3..2fc4b6090fcf0d 100644
--- a/repositories.bzl
+++ b/repositories.bzl
@@ -100,6 +100,12 @@ def embedded_jdk_repositories():
         downloaded_file_path = "adoptopenjdk-ppc64le-vanilla.tar.gz",
         url = "https://github.com/adoptium/temurin23-binaries/releases/download/jdk-23.0.1%2B11/OpenJDK23U-jdk_ppc64le_linux_hotspot_23.0.1_11.tar.gz",
     )
+    http_file(
+        name = "openjdk_linux_riscv64_vanilla",
+        integrity = "sha256-gNe6uflhS9+TTGvEQQMb0f6tOuqF8WdwEjvYprzcUrY=",
+        downloaded_file_path = "adoptopenjdk-riscv64-vanilla.tar.gz",
+        url = "https://github.com/adoptium/temurin23-binaries/releases/download/jdk-23.0.1%2B11/OpenJDK23U-jdk_riscv64_linux_hotspot_23.0.1_11.tar.gz",
+    )
     http_file(
         name = "openjdk_macos_x86_64_vanilla",
         integrity = "sha256-Ha2a94Z+QAc367voEEMXGo8Lw92vcvvu9ilbPR2+cow=",
diff --git a/src/BUILD b/src/BUILD
index fe2adbb918d8bb..155ac21b6fff04 100644
--- a/src/BUILD
+++ b/src/BUILD
@@ -173,6 +173,9 @@ filegroup(
         "//src/conditions:linux_s390x": [
             "@openjdk_linux_s390x_vanilla//file",
         ],
+        "//src/conditions:linux_riscv64": [
+            "@openjdk_linux_riscv64_vanilla//file",
+        ],
         "//conditions:default": [
             "@openjdk_linux_vanilla//file",
         ],
diff --git a/src/minimize_jdk.sh b/src/minimize_jdk.sh
index 06e0ad17438015..691d9580f3306e 100755
--- a/src/minimize_jdk.sh
+++ b/src/minimize_jdk.sh
@@ -40,8 +40,8 @@ else
 fi
 fulljdk=$1
 out=$3
-ARCH=`uname -p`
-if [[ "${ARCH}" == 'ppc64le'  ]] || [[ "${ARCH}" == 's390x' ]]; then
+ARCH=`uname -m`
+if [[ "${ARCH}" == 'ppc64le'  ]] || [[ "${ARCH}" == 's390x' ]] || [[ "${ARCH}" == 'riscv64' ]]; then
   FULL_JDK_DIR="jdk*"
   DOCS=""
 else
