--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,17 @@ makedepends=('git' 'protobuf' 'python')
 options=('!debug' '!strip')
 source=(
   "https://github.com/bazelbuild/bazel/releases/download/${pkgver}/bazel-${pkgver}-dist.zip"{,.sig}
+  bazel-riscv64.diff
 )
 b2sums=('f99c7a37f7552bb6a8f44d4ab44863590e31101cf7c0b749789c3f6fb8787ad694d0712ca90dea97fcc49c123378b6f7caaccf0e545ee722dafff7dd19262ede'
-        'SKIP')
+        'SKIP'
+        'd77de555c8f738d087d55f710e4502f9e3642e60a207bba23dd4b1176704ce0f272d804bccd3d61b535cdf0fe291864f6079e5467d412d220d74de16faf2c073')
 validpgpkeys=('71A1D0EFCFEB6281FD0437C93D5919B448457EE0')
 
+prepare() {
+  patch -Np1 -i "bazel-riscv64.diff"
+}
+
 build() {
   EMBED_LABEL=$pkgver EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" ./compile.sh
   ./output/bazel build scripts:bazel-complete.bash
