--- PKGBUILD
+++ PKGBUILD
@@ -22,9 +22,7 @@
 
 build() {
   EMBED_LABEL=$pkgver EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" ./compile.sh
-  ./output/bazel build scripts:bazel-complete.bash
-  cd output
-  ./bazel shutdown
+  ./output/bazel --batch help completion bash > output/bazel-complete.bash
 }
 
 package() {
@@ -32,7 +30,7 @@
   install -Dm755 "${srcdir}/output/bazel" "${pkgdir}/usr/bin/bazel-real"
   local _bazel_major=${pkgver%%.*}
   ln -s bazel-real "${pkgdir}/usr/bin/bazel-$_bazel_major.x-linux-$CARCH"
-  install -Dm644 "${srcdir}/bazel-bin/scripts/bazel-complete.bash" "${pkgdir}/usr/share/bash-completion/completions/bazel"
+  install -Dm644 "${srcdir}/output/bazel-complete.bash" "${pkgdir}/usr/share/bash-completion/completions/bazel"
   install -Dm644 "${srcdir}/scripts/zsh_completion/_bazel" "${pkgdir}/usr/share/zsh/site-functions/_bazel"
   mkdir -p "${pkgdir}/usr/share/bazel"
   for d in third_party tools; do
