--- PKGBUILD
+++ PKGBUILD
@@ -15,14 +15,20 @@ makedepends=('git' 'protobuf' 'python')
 options=('!debug' '!strip')
 source=(
   "https://github.com/bazelbuild/bazel/releases/download/${pkgver}/bazel-${pkgver}-dist.zip"{,.sig}
+  bazel-riscv.diff
 )
 b2sums=('76da69aa2ee53db5c2151d02cfc165489207883245b3e5b16a44020babac8eb8441beceef47f970e27b9897a1add3f4954e3ca5d3b2bed18b8493fe9ab036775'
-        'SKIP')
+        'SKIP'
+        '5434f4013aeb112ea2bcb0dc6620ad7539b2f7953d899e71129bdf00b3ce5c6d3cc10a96f294338dcda5232d2389c6da46c44cd2160f49a1f5827081cd5dddac')
 validpgpkeys=('71A1D0EFCFEB6281FD0437C93D5919B448457EE0')
 
+prepare() {
+  patch -Np1 -i bazel-riscv.diff
+}
+
 build() {
   EMBED_LABEL=$pkgver EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" ./compile.sh
-  ./output/bazel build scripts:bazel-complete.bash
+  ./output/bazel build scripts:bazel-complete.bash --extra_toolchains=@local_jdk//:all
   cd output
   ./bazel shutdown
 }
