--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,17 @@ makedepends=('git' 'protobuf' 'python')
 options=('!debug' '!strip')
 source=(
   "https://github.com/bazelbuild/bazel/releases/download/${pkgver}/bazel-${pkgver}-dist.zip"{,.sig}
+  bazel-riscv64.diff
 )
 b2sums=('cc23e09222ee4e5ae0cc2efe9f2f1772a657a81dc8bb255c95d5b2c6db84cda92542b3fe3d4bedd209ce693b927b05ba99290995139f349c764cd8394358101d'
-        'SKIP')
+        'SKIP'
+        '9b34d6d77a09bd83d4d755bdaaefcf64cfa7b4f1bbbfd59a0660e6d6c7c9f64643ed8bc2cbf34a879f103b2893f252586b3f79327cd2835d7df9a3e015c2104f')
 validpgpkeys=('71A1D0EFCFEB6281FD0437C93D5919B448457EE0')
 
+prepare() {
+  patch -Np1 -i "bazel-riscv64.diff"
+}
+
 build() {
   EMBED_LABEL=$pkgver EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" ./compile.sh
   ./output/bazel build scripts:bazel-complete.bash
