diff --git PKGBUILD PKGBUILD
index 225274f9e..f202c4f82 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,10 +14,18 @@ depends=('tk' 'libxpm' 'zlib' 'cairo')
 makedepends=('chrpath' 'imagemagick')
 source=("http://opencircuitdesign.com/xcircuit/archive/$pkgname-$pkgver.tgz"
         "xcircuit.desktop"
-        "http://opencircuitdesign.com/xcircuit/xcicon_sm.gif")
+        "http://opencircuitdesign.com/xcircuit/xcicon_sm.gif"
+        "https://github.com/IEAST/XCircuit/commit/7f57ebbf602713156be73a226cd067581e5021a5.patch")
 sha256sums=('b2f63cba605e79cc2a08714bf3888f7be7174384ed724db3c70f8bf25c36f554'
             '3c29f55ee6196b554d6fc5791f29ac7cab2fecc184c699e80ee265be0b609456'
-            '8e7b132d9803443e6f3ae65ed1990362a5a977e7856b1e54f2edf8cd4a3d7259')
+            '8e7b132d9803443e6f3ae65ed1990362a5a977e7856b1e54f2edf8cd4a3d7259'
+            '5f56153f37fa7764148e498e17542388fa4fe6a3ba3af3c4f87f370cf41c4973')
+
+prepare(){
+  cd "$srcdir/$pkgname-$pkgver"
+  # https://github.com/RTimothyEdwards/XCircuit/pull/11
+  patch -Np1 -i ../7f57ebbf602713156be73a226cd067581e5021a5.patch
+}
 
 build() {
   cd "$srcdir/$pkgname-$pkgver"
