diff --git a/ext/node/polyfills/_process/process.ts b/ext/node/polyfills/_process/process.ts
index 6dfb897ddb..45688791c2 100644
--- a/ext/node/polyfills/_process/process.ts
+++ b/ext/node/polyfills/_process/process.ts
@@ -19,6 +19,8 @@ export function arch(): string {
     return "x64";
   } else if (core.build.arch == "aarch64") {
     return "arm64";
+  } else if (core.build.arch == "riscv64gc") {
+    return "riscv64";
   } else {
     throw Error("unreachable");
   }
