--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,35 @@ license=('MIT')
 options=('!lto')
 depends=('gcc-libs')
 makedepends=('git' 'python' 'rust' 'nodejs' 'gn' 'ninja' 'clang' 'lld' 'cmake' 'protobuf')
-source=("git+https://github.com/denoland/deno.git#commit=$_commit")
-sha512sums=('SKIP')
+source=("git+https://github.com/denoland/deno.git#commit=$_commit"
+        "git+https://github.com/denoland/rusty_v8.git#commit=2825ced8bea16732ff280f71e1b487fd5159a644"  # 0.85.0
+        "add-rvb-flags.patch"
+        "arch-add-riscv64.patch")
+sha512sums=('SKIP'
+            'SKIP'
+            '262a2976faf3dc94fd0183c0b47e52e241bfaffaf16e2a1a54d079993c8cdacce5e9bd979aff41f9de3a7243444fa24cd6e9ea3ee69867bf4236d498203e15e0'
+            '07aba81e9ad488c60388afd1584a28323c7186268d3ea727117d6b04d5c4658a72b8cfab497ee71052cbdab7e7f4696f543861bac8c58a494146b495c8954261')
+
+prepare() {
+  cd rusty_v8
+  git config -f .gitmodules submodule.v8.shallow true
+  git submodule update --init --recursive
+  patch -Np1 -i ../add-rvb-flags.patch
+
+  cd ../deno
+
+  # 0.8.0 -> (at least) 0.8.1
+  # which contains https://github.com/simd-lite/value-trait/commit/136603dcc6c6313b437cc42410aa4c85c709ea58
+  cargo update -p value-trait
+
+  # 0.13.4 -> (at least) 0.13.8
+  # https://github.com/simd-lite/simd-json/commit/f46dc2ab7ca5eab0d531cb7ad7ae11bcac8c2d06
+  cargo update -p simd-json
+
+  echo -e "\n[patch.crates-io]\nv8 = { path = '../rusty_v8' }" >> Cargo.toml
+
+  patch -Np1 -i ../arch-add-riscv64.patch
+}
 
 build() {
   cd $pkgname
