diff --git PKGBUILD PKGBUILD
index fb789e4..7739488 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,8 @@ prepare() {
   git submodule update --init --recursive
   # Disable a warning. upx believes this is not a release version if the git revision is set.
   sed -i 's/bool warn = true/bool warn = false/' src/main.cpp
+  cd src/lzma-sdk
+  git checkout c6c2fc49e2ac6f3fc82762c6d3703969274d48de # fix "this 'if' clause does not guard"
 }
 
 build() {
