diff --git a/form_wrap.c b/form_wrap.c
index 346d443..c9e20c0 100644
--- a/form_wrap.c
+++ b/form_wrap.c
@@ -1131,7 +1131,7 @@ static void* make_arg(va_list* ap) {
 		  char msg[500];
 		  snprintf(msg, 500, "The validation functions for this field type need %d additional arguments.",(int)(NUM2INT(argc)-1));
 		  msg[499]=0;
-		  rb_raise(rb_eArgError, msg);	
+		  rb_raise(rb_eArgError, "%s", msg);	
 		}
 	 } 
   }
