--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,8 @@ depends=('c-ares' 'gtk3' 'libevent' 'nss' 'wayland')
 makedepends=('clang' 'git' 'gn' 'gperf' 'harfbuzz-icu' 'http-parser'
              'qt5-base' 'java-runtime-headless' 'libnotify' 'lld' 'llvm'
              'ninja' 'npm' 'pciutils' 'pipewire' 'python' 'python-httplib2'
-             'python-pyparsing' 'python-six' 'wget' 'yarn' 'patchutils')
+             'python-pyparsing' 'python-six' 'wget' 'yarn' 'patchutils'
+             'esbuild' 'go' 'p7zip' 'bazel' 'jdk11-openjdk' 'llvm' 'llvm-libs')
 # shellcheck disable=SC2034
 optdepends=('kde-cli-tools: file deletion support (kioclient5)'
             'libappindicator-gtk3: StatusNotifierItem support'
@@ -87,6 +88,15 @@ source=("git+https://github.com/electron/electron.git#tag=v$pkgver"
         'REVERT-roll-src-third_party-ffmpeg-m102.patch'
         'REVERT-roll-src-third_party-ffmpeg-m106.patch'
         'angle-wayland-include-protocol.patch'
+        "$pkgname-riscv-angle.patch"
+        "$pkgname-riscv-sandbox.patch"
+        "$pkgname-riscv-crashpad.patch"
+        "$pkgname-riscv-dav1d.patch"
+        "$pkgname-swiftshader-use-system-llvm.patch"
+        "$pkgname-swiftshader-LLVMJIT-AddressSanitizerPass-dead-code-remove.patch"
+        "$pkgname-gclient-ignore-prebuilt-platform-specific-deps.patch"
+        "$pkgname-grab-commit.py"
+        "git+https://chromium.googlesource.com/infra/infra"
        )
 # shellcheck disable=SC2034
 sha256sums=('SKIP'
@@ -124,7 +134,16 @@ sha256sums=('SKIP'
             'bfafedebd915e1824562329a3afc8589401342eff87dba53c78502b869023b7e'
             '30df59a9e2d95dcb720357ec4a83d9be51e59cc5551365da4c0073e68ccdec44'
             '4c12d31d020799d31355faa7d1fe2a5a807f7458e7f0c374adf55edb37032152'
-            'cd0d9d2a1d6a522d47c3c0891dabe4ad72eabbebc0fe5642b9e22efa3d5ee572')
+            'cd0d9d2a1d6a522d47c3c0891dabe4ad72eabbebc0fe5642b9e22efa3d5ee572'
+            'a191ef5190e5f823e6be078df53ed1a888558673040985d79347a26644c39b63'
+            '378ab9fd608cf6a9b2f3631798001a836997f13bcd296940afb538448e0f8b9d'
+            'cf233d17326e0aa59bd395971478df9982059b18d237068c3906e4716903d533'
+            '32811636eecd187dd386bbdd0fa1216728d6010f96fae8f1714dded04a95a5f4'
+            'd901e905a9b4303e6334bf39475bcbcdf22959796954de66507857108d53570f'
+            'cf80c0d70f8933a4495c71d7be681c1457a69d26e95d2ad41da4bb02b7cbee4c'
+            'b28b64181c46549e94e8c26d5401b0036991c18c60040c444a262484addd0d0d'
+            '13c47f8e085ba44739760b94dd0170a6ec5bd7e32645ba2f9e1467079dfe612e'
+            'SKIP')
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
 # Keys are the names in the above script; values are the dependencies in Arch
@@ -188,11 +207,32 @@ EOF
   echo "Linking chromium from sources..."
   ln -s chromium-mirror src
 
+  echo "Building build tools that doesn't have prebuilt binaries for riscv64..."
+  export GOBIN=/build/bin
+  export PATH="$PATH:/build/bin"
+  
+  # Install the build tools
+  local infra_rev=$(python "$pkgname-grab-commit.py" infra chromium-mirror/DEPS)
+  git -C infra checkout "$infra_rev"
+  local luci_go_rev=$(python "$pkgname-grab-commit.py" luci_go infra/DEPS)
+  go install "go.chromium.org/luci/swarming/cmd/...@$luci_go_rev"
+  go install "go.chromium.org/luci/client/cmd/...@$luci_go_rev"
+  go install "go.chromium.org/luci/cipd/client/cmd/...@$luci_go_rev"
+
+  # Fix .cipd-bin problem
+  mkdir depot_tools/.cipd_bin
+  GOBIN=$(realpath depot_tools/.cipd_bin) go install "go.chromium.org/luci/auth/client/cmd/...@$luci_go_rev"
+
+  patch -Np0 -i $pkgname-gclient-ignore-prebuilt-platform-specific-deps.patch
   depot_tools/gclient.py sync -D \
       --nohooks \
       --with_branch_heads \
       --with_tags
 
+  # Replace the bundled x86_64 JDK with system JDK11
+  rm -rf src/third_party/jdk/current
+  ln -s /usr/lib/jvm/java-11-openjdk src/third_party/jdk/current
+
   (
     cd src/electron || exit
     patch -Np1 -i ../../std-vector-non-const.patch
@@ -254,6 +294,15 @@ EOF
   patch -Np1 -i ../chromium-icu72.patch
   patch -Np1 -d v8 <../v8-enhance-Date-parser-to-take-Unicode-SPACE.patch
 
+  # RISC-V Patches
+  patch -Np1 -i ../$pkgname-riscv-angle.patch
+  patch -Np1 -i ../$pkgname-riscv-sandbox.patch
+  patch -Np1 -i ../$pkgname-riscv-crashpad.patch
+  patch -Np1 -i ../$pkgname-riscv-dav1d.patch
+  # Build failes with the bundled LLVM 10.
+  patch -Np1 -i ../$pkgname-swiftshader-use-system-llvm.patch
+  patch -Np1 -i ../$pkgname-swiftshader-LLVMJIT-AddressSanitizerPass-dead-code-remove.patch
+
   # Revert ffmpeg roll requiring new channel layout API support
   # https://crbug.com/1325301
   patch -Rp1 -i ../REVERT-roll-src-third_party-ffmpeg-m102.patch
