diff --git PKGBUILD PKGBUILD
index 79c486a8c..1a3f9451e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,9 @@
 
 prepare() {
   cd $_name-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  sed -i '/hubcaps = { git="https:\/\/github.com\/softprops\/hubcaps.git" }/a\ring = { git = "https:\/\/github.com\/felixonmars\/ring", branch = "0.16.20" }' Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
