diff --git PKGBUILD PKGBUILD
index 0624277c..a0c989e7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,11 @@ b2sums=('3ec7b82f2ac491828194c4515ca6861eb9697fee6d6051e549030ce52cd9a2f95b3d9bb
 
 prepare() {
   cd $_name-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  sed -i '/hubcaps = { git="https:\/\/github.com\/softprops\/hubcaps.git" }/a\ring = { git = "https:\/\/github.com\/felixonmars\/ring", branch = "0.16.20" }' Cargo.toml
+  sed -i '/debug = true/a\opt-level = 1' Cargo.toml
+  echo -e "" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
