--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,9 @@ sha256sums=('e6ccbca1656f4e616b4657f8514e33a70f6697e9d7294356577839322a3c5d2c'
 
 prepare() {
   cd "${srcdir}/${pkgname}-${pkgver}-stable"
+  # https://github.com/llvm/llvm-project/blob/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess
+  # llvm/cmake/config.guess: add support for riscv32 and riscv64
+  mv -f ${srcdir}/config.guess ./
   # [PATCH] Add pcre2 support
   patch -Np1 -i ../53748ca8ca3c893025be34dd4f104546fcbd0602.patch
 
@@ -68,3 +71,6 @@ package() {
   find user-manual developer-manual faq man-page \( -name '*.html' -o -name '*.jpg' \) \
     -exec install -Dm644 '{}' "$d"/'{}' \;)
 }
+
+source+=('config.guess::https://raw.githubusercontent.com/llvm/llvm-project/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess')
+sha256sums+=('41ad5c3afd53f21e4bdbe9fdaec99523f1d9783f38c7dad964264b350977ec64')
