diff --git PKGBUILD PKGBUILD
index 9baebf9..37e3bef 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,11 +16,19 @@ backup=('etc/privoxy/'{config,trust,match-all.action,{default,user}.{action,filt
 source=("https://downloads.sourceforge.net/ijbswa/${pkgname}-${pkgver}-stable-src.tar.gz"
         'privoxy.logrotate.d'
         'privoxy.sysusers'
-        'privoxy.service')
+        'privoxy.service'
+        'config.guess::https://raw.githubusercontent.com/llvm/llvm-project/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess')
 sha256sums=('e6ccbca1656f4e616b4657f8514e33a70f6697e9d7294356577839322a3c5d2c'
             '769740ea3c15228f24b559192e7b3f45b95dcf9040e2b25f15bfdfae8af86ce3'
             '6681231552f431962dda2ac49187df833b2b57544eac97ca94d1f4207b27b04c'
-            '30237681b6b117ce0d21efb3eeb896202744a32f9740fede1011b5dc84ff9f23')
+            '30237681b6b117ce0d21efb3eeb896202744a32f9740fede1011b5dc84ff9f23'
+            '41ad5c3afd53f21e4bdbe9fdaec99523f1d9783f38c7dad964264b350977ec64')
+
+prepare() {
+  # https://github.com/llvm/llvm-project/blob/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess
+  # llvm/cmake/config.guess: add support for riscv32 and riscv64
+  mv -f ${srcdir}/config.guess "${srcdir}/${pkgname}-${pkgver}-stable"
+}

 build() {
   cd "${srcdir}/${pkgname}-${pkgver}-stable"
