--- PKGBUILD
+++ PKGBUILD
@@ -25,6 +25,11 @@ sha256sums=('f2621a95a004f2869ed57053e11c0875ce930f152049fd0aed008a13c5b326fc'
 
 prepare() {
   cd privoxy
+
+  # https://github.com/llvm/llvm-project/blob/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess
+  # llvm/cmake/config.guess: add support for riscv32 and riscv64
+  mv -f ${srcdir}/config.guess ./
+
   autoheader
   autoconf
 }
@@ -62,3 +67,6 @@ package() {
   find user-manual developer-manual faq man-page \( -name '*.html' -o -name '*.jpg' \) \
     -exec install -Dm644 '{}' "$d"/'{}' \;)
 }
+
+source+=('config.guess::https://raw.githubusercontent.com/llvm/llvm-project/de1c2877a9ff12899ef50e179ade748fba8ab0c0/llvm/cmake/config.guess')
+sha256sums+=('41ad5c3afd53f21e4bdbe9fdaec99523f1d9783f38c7dad964264b350977ec64')
