--- PKGBUILD
+++ PKGBUILD
@@ -4,9 +4,9 @@
 # Contributor: schuay <jakob.gruber@gmail.com>
 
 pkgname=puzzles
-pkgver=20230101
+pkgver=20230110
 pkgrel=1
-_commit=14c025d
+_commit=e66d027
 pkgdesc="Simon Tatham's Portable Puzzle Collection"
 arch=(x86_64)
 url='https://www.chiark.greenend.org.uk/~sgtatham/puzzles/'
@@ -14,7 +14,7 @@ license=(MIT)
 depends=(gtk3)
 makedepends=(cmake imagemagick)
 source=(https://www.chiark.greenend.org.uk/~sgtatham/puzzles/puzzles-$pkgver.$_commit.tar.gz)
-sha256sums=('970ac7db42d8066eca11e3cb03c5b07973b9ba47887bc2b08808423672cb4c85')
+sha256sums=('7fbd9b22b05c34955f28d840b980cfd9c58c158b03020a2fdaf74fc928edb768')
 
 build() {
   cmake -B build -S puzzles-$pkgver.$_commit \
