diff --git PKGBUILD PKGBUILD
index dd84dadd..b7ae8df2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,8 +22,11 @@
 prepare() {
   cd "$pkgname-$pkgver"
 
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+
   # download dependencies
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked
 }
 
 build() {
