diff --color -Naur typespeed-0.6.5-orig/src/typespeed.c typespeed-0.6.5/src/typespeed.c
--- typespeed-0.6.5-orig/src/typespeed.c	2008-08-13 22:36:51.000000000 +0800
+++ typespeed-0.6.5/src/typespeed.c	2021-10-08 11:35:26.086796425 +0800
@@ -78,6 +78,11 @@
 #include "pathnames.h"
 #include "typespeed.h"
 
+struct stats now;
+struct opt opt;
+struct rules rules;
+struct rawdata words;
+
 #define _(string)	gettext(string)
 
 #define DEFAULT_CHEAT		0
@@ -1016,4 +1021,3 @@
 	exit(0);
 }
 #endif /* WIN32 */
-
diff --color -Naur typespeed-0.6.5-orig/src/typespeed.h typespeed-0.6.5/src/typespeed.h
--- typespeed-0.6.5-orig/src/typespeed.h	2008-08-13 22:36:51.000000000 +0800
+++ typespeed-0.6.5/src/typespeed.h	2021-10-08 11:35:23.446774704 +0800
@@ -56,7 +56,8 @@
 	clock_t duration;
 	unsigned int sinit;
 	char name[21];
-} now;
+};
+extern struct stats now;
 
 struct opt {
 	int cheat;
@@ -66,7 +67,8 @@
 	int usecolors;
 	char name[21];
 	char order[3];
-} opt;
+};
+extern struct opt opt;
 
 struct rules {
 	int misses;
@@ -85,12 +87,13 @@
 	int maxtime;
 	char name[31];
 	char fname[FILENAME_MAX + 1];
-} rules;
+};
+extern struct rules rules;
 
 struct rawdata {
 	char *bulk;
 	char **word;
 	size_t n;
 	size_t max;
-} words;
-
+};
+extern struct rawdata words;
