diff --git PKGBUILD PKGBUILD
index 7ad5e22..664416b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,15 @@ arch=('x86_64')
 url="http://typespeed.sourceforge.net"
 license=('GPL')
 depends=('ncurses')
-source=(https://downloads.sourceforge.net/sourceforge/$pkgname/$pkgname-$pkgver.tar.gz)
-md5sums=('578102b418c7df84903d3e90df2e7483')
+source=(https://downloads.sourceforge.net/sourceforge/$pkgname/$pkgname-$pkgver.tar.gz
+        fix-multiple-definition.patch)
+md5sums=('578102b418c7df84903d3e90df2e7483'
+         '0e041559cc5693c6dd68aa00449d3b0c')
+
+prepare() {
+  cd "$srcdir/$pkgname-$pkgver"
+  patch -Np1 -i ../fix-multiple-definition.patch
+}
 
 build() {
   cd "$srcdir/$pkgname-$pkgver"
