--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@
 
 _pkgname=magma
 pkgbase=$_pkgname
-pkgname=(magma-cuda magma-hip)
+pkgname=(magma-hip)
 pkgver=2.9.0
 pkgrel=1
 _pkgdesc="Matrix Algebra on GPU and Multicore Architectures"
@@ -16,7 +16,6 @@ url="https://icl.utk.edu/magma/"
 license=('BSD-3-Clause')
 depends=('blas' 'lapack')
 makedepends=('git' 'cmake' 'ninja' 'python' 'gcc-fortran'
-             'cuda'
              'rocm-core' 'hip-runtime-amd' 'hipblas' 'hipsparse')
 optdepends=('python: for examples and tests'
             'gcc-fortran: Fortran interface')
@@ -42,13 +41,8 @@ prepare() {
 
   cd "${srcdir}"
 
-  cp -r "${_pkgname}" "${_pkgname}-${pkgver}-cuda"
   cp -r "${_pkgname}" "${_pkgname}-${pkgver}-hip"
 
-  cd "${_pkgname}-${pkgver}-cuda"
-  echo -e "BACKEND = cuda\nFORT = true\nGPU_TARGET=$(_valid_sm)" > make.inc
-  make generate
-
   cd "${srcdir}"
 
   cd "${_pkgname}-${pkgver}-hip"
@@ -57,18 +51,6 @@ prepare() {
 }
 
 build() {
-  echo "Build with cuda backend"
-  cd "${srcdir}/${_pkgname}-${pkgver}-cuda"
-  cmake \
-    -Bbuild \
-    -GNinja \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=/opt/cuda/targets/x86_64-linux \
-    -DBUILD_SHARED_LIBS=ON \
-    -DGPU_TARGET="$(_valid_sm)"
-  ninja -C build
-
-
   echo "Build with rocm/hip backend"
   cd "${srcdir}/${_pkgname}-${pkgver}-hip"
   local _rocm_ver=$(./tools/get-rocm-version.sh)
