diff --git PKGBUILD PKGBUILD
index d83692c..680ec49 100644
--- PKGBUILD
+++ PKGBUILD
@@ -6,7 +6,7 @@
 
 _pkgname=magma
 pkgbase=$_pkgname
-pkgname=(magma-cuda magma-hip)
+pkgname=(magma-hip)
 pkgver=2.8.0
 pkgrel=2
 _pkgdesc="Matrix Algebra on GPU and Multicore Architectures"
@@ -15,7 +15,6 @@ url="https://icl.utk.edu/magma/"
 license=('BSD-3-Clause')
 depends=('blas' 'lapack')
 makedepends=('git' 'cmake' 'ninja' 'python' 'gcc-fortran'
-             'cuda'
              'rocm-core' 'hip-runtime-amd' 'hipblas' 'hipsparse')
 optdepends=('python: for examples and tests'
             'gcc-fortran: Fortran interface')
@@ -40,11 +39,6 @@ prepare() {
   cp -r "${_pkgname}" "${_pkgname}-${pkgver}-cuda"
   cp -r "${_pkgname}" "${_pkgname}-${pkgver}-hip"
 
-  cd "${_pkgname}-${pkgver}-cuda"
-  echo -e "BACKEND = cuda\nFORT = true\nGPU_TARGET=$(_valid_sm)" > make.inc
-  make generate
-  patch -Np1 -i "${srcdir}/add_cuda_archs.patch"
-
   cd "${srcdir}"
 
   cd "${_pkgname}-${pkgver}-hip"
@@ -54,18 +48,6 @@ prepare() {
 }
 
 build() {
-  echo "Build with cuda backend"
-  cd "${srcdir}/${_pkgname}-${pkgver}-cuda"
-  cmake \
-    -Bbuild \
-    -GNinja \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=/opt/cuda/targets/x86_64-linux \
-    -DBUILD_SHARED_LIBS=ON \
-    -DGPU_TARGET="$(_valid_sm)"
-  ninja -C build
-
-
   echo "Build with rocm/hip backend"
   cd "${srcdir}/${_pkgname}-${pkgver}-hip"
   local _rocm_ver=$(./tools/get-rocm-version.sh)
