Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 405399)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,6 @@
 license=('GPL2')
 depends=('gmp')
 provides=('libnettle.so' 'libhogweed.so')
-checkdepends=('valgrind')
 source=(https://ftp.gnu.org/gnu/$pkgname/$pkgname-$pkgver.tar.gz{,.sig})
 sha256sums=('f001f64eb444bf13dd91bceccbc20acbc60c4311d6e2b20878452eb9a9cec75a'
             'SKIP')
