--- a/libappstream-glib/as-app.c
+++ b/libappstream-glib/as-app.c
@@ -2665,7 +2665,7 @@ as_app_set_update_contact (AsApp *app, const gchar *update_contact)
 			if (tmp != NULL) {
 				*tmp = replacements[i].replace;
-				(void)g_strlcpy (tmp + 1,
-					   tmp + strlen (replacements[i].search),
-					   len);
+				memmove (tmp + 1,
+					 tmp + strlen (replacements[i].search),
+					 strlen (tmp + strlen (replacements[i].search)) + 1);
 				done_replacement = TRUE;
 			}
