Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1012920)
+++ PKGBUILD	(working copy)
@@ -22,6 +22,7 @@
 
 build() {
     cd "$pkgname-${pkgver}"
+    echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
     cargo build --release --features no-self-update --bin rustup-init
 }
 
