--- PKGBUILD
+++ PKGBUILD
@@ -42,6 +42,7 @@ prepare() {
 		-i scripts/systemd/mysqlrouter.service.in
 
         patch -p1 -i ../gcc-14.patch # Fix build with GCC 14
+	patch -p1 -d extra/coredumper -i "$srcdir"/fix-missing-include.patch
 }
 
 build() {
@@ -50,8 +51,8 @@ build() {
 	cd build
 
 	cmake "../$pkgbase-$_pkgver" \
-		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG -ffat-lto-objects -msse4.2 -mpclmul" \
-		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG -ffat-lto-objects -msse4.2 -mpclmul" \
+		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG -ffat-lto-objects" \
+		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG -ffat-lto-objects" \
 		-Wno-dev \
 		-DCMAKE_BUILD_TYPE=RelWithDebInfo \
 		-DCMAKE_POSITION_INDEPENDENT_CODE=ON \
@@ -196,3 +197,6 @@ package_percona-server() {
 	# not needed
 	rm -r usr/share/mysql-test
 }
+
+source+=("fix-missing-include.patch::https://github.com/Percona-Lab/coredumper/pull/8.diff")
+sha256sums+=('81e3293d9b032d0dfed281067958e5f10306214dfac92a0b321f072bfc5f9910')
