diff --git PKGBUILD PKGBUILD
index 02bb539..5ea90b9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,14 +18,16 @@ source=("https://www.percona.com/downloads/Percona-Server-${pkgver%.*_*}/Percona
         'mysql-user.conf'
         'mysqlrouter-user.conf'
         'mysql-build-fix.patch'
-        'protobuf-23.patch')
+        'protobuf-23.patch'
+		'add-riscv-support.patch::https://patch-diff.githubusercontent.com/raw/percona/percona-server/pull/5030.patch')
 sha256sums=('9871cac20c226bba7607f35c19ee23516a38c67573dd48618727c74eae22912e'
             '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131'
             'b467b04d6d06152b2abc33f2a6de63fef0fc922dd5119d2ee1d07d3c1a489731'
             '5d7710fe88ec6d298175a309c0b776142397b119c468830b2865980292ed5da6'
             '4ca7ffdcb2d1716d4f31e4c7dd314e5d76e64f13fdc67c5d81c53650b793f5e0'
             'ea17c754d5595bf053d733373b7c4ae6d608a9f85a990dbe5779b499dd398bef'
-            '2d73c79f355eaf3c0bced0f0da6ad099323f4f489d5cddb609cf01af5a617305')
+            '2d73c79f355eaf3c0bced0f0da6ad099323f4f489d5cddb609cf01af5a617305'
+            '465d90e49cf781b200e8d7155d7273dfde2781438f268f1dea8978259b3d4053')
 
 prepare() {
 	cd $pkgbase-$_pkgver
@@ -46,6 +48,8 @@ prepare() {
 
 	sed '/^PrivateTmp=/ a StateDirectory=mysqlrouter\nRuntimeDirectory=mysqlrouter' \
 		-i scripts/systemd/mysqlrouter.service.in
+
+	patch -Np1 -i ../add-riscv-support.patch
 }
 
 build() {
@@ -54,8 +58,8 @@ build() {
 	cd build
 
 	cmake "../$pkgbase-$_pkgver" \
-		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG -ffat-lto-objects -msse4.2 -mpclmul" \
-		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG -ffat-lto-objects -msse4.2 -mpclmul" \
+		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG -ffat-lto-objects" \
+		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG -ffat-lto-objects" \
 		-Wno-dev \
 		-DCMAKE_BUILD_TYPE=RelWithDebInfo \
 		-DCMAKE_POSITION_INDEPENDENT_CODE=ON \
