--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ _myver=${pkgver/_rel*}
 pkgrel=1
 arch=('x86_64')
 makedepends=('cmake' 'zlib' 'lz4' 'zstd' 'libaio' 'systemd-tools' 'pam' 'numactl' 'jemalloc' 'openssl'
-             'rpcsvc-proto' 'doxygen' 'graphviz' 'libevent' 'protobuf') # 'boost'
+             'rpcsvc-proto' 'doxygen' 'graphviz' 'libevent' 'protobuf' 'clang') # 'boost'
 options=('debug')
 license=('GPL')
 url="https://www.percona.com/software/mysql-database/percona-server"
@@ -29,6 +29,10 @@ sha256sums=('91a7ad8fc0e3931f2d2bcb616bca8a55b6b7cac45f735aba4188c45edcf85da8'
             '92d70b75a32517f598bbffbaf5db18b0d14be504c31f531e35c8572b330785f3'
             '1537fdbb92dd1c135c1eb9f4d10c44fd02e652db66c933d731990a1196f1397c'
             '2343a191c452b91caa458b03b0c1ef3f5afb0e7031816c68467af5c6a6ffe253')
+# Can't use -Wl,-plugin-opt=-target-abi=lp64d, because both CFLAGS and CXXFLAGS need it,
+# but specifying both will cause LLVMgold to complain about multiple -target-abi.
+# Also, -ffat-lto-objects can't resolve the compile error here. We have to set !lto
+options+=(!lto)
 
 prepare() {
 	cd $pkgbase-$_pkgver
@@ -56,9 +60,11 @@ build() {
 	mkdir -p build
 	cd build
 
+	export CC=clang CXX=clang++ CFLAGS="${CFLAGS//-fstack-clash-protection/}" CXXFLAGS="${CXXFLAGS//-fstack-clash-protection/}"
+	# -ffat-lto-objects not supported & -Werror
 	cmake ../$pkgbase-$_pkgver \
-		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG -ffat-lto-objects" \
-		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG -ffat-lto-objects" \
+		-DCMAKE_C_FLAGS_RELWITHDEBINFO="${CFLAGS} -DNDEBUG" \
+		-DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${CXXFLAGS} -DNDEBUG" \
 		-Wno-dev \
 		-DCMAKE_BUILD_TYPE=RelWithDebInfo \
 		-DCMAKE_POSITION_INDEPENDENT_CODE=ON \
