--- PKGBUILD
+++ PKGBUILD
@@ -38,13 +38,15 @@
         "$pkgname-fix-failing-requests-tests.patch::$url/commit/929cc89c2afced0c42bd08e3b9440ab1f4ec22a8.patch"
         # https://github.com/gabrielfalcao/HTTPretty/pull/485
         "$pkgname-urllib3-2.3-compatibility.patch::$url/commit/8e96b1e312d473429fbd08bc867376e9932ad42a.patch"
-         python-3.14.patch)
+         python-3.14.patch
+        increase-test-timeouts.patch)
 sha512sums=('58c733ba4719f97e06e2313bb6b94f1c6609d3facb2e0262ac37be97f1c3430eac661611ba9b3712c9c1809846e177b61f3fc2945f3770d475f70b81bea2aced'
             '5520594ddb2e73d75c6eb8476dc1984464306614d915627c0c9e69815a5671d65e286e90f3ee926daa040f9f24a81835eb5c7993e620a5b67aaaefa06571dcb2'
             'c9f1668be03210ef5efc7fe6f4783af31549aa812ebb32154a27034b88866e7e81df59648a3bb68f8f6c08c2734abc4b012c9e8b90d19d8570940c20391294e9'
             '0b890bfd1dad70cf27aef5720e3158225697a4ccbe735fd7dd7e2b0887b60cc158750b9724d3e192311a6f32400480ce6ded4b01f01ed885085753ef13fc939e'
             '483cb20c64872b68dbef35a3325fc7f93d5e550c3e4cb266425a0d0ecac172b97674de8f837b2b48d63145ff0a24d3a89af1871cb84bfec9eeb70b1bea803a34'
-            '1c05ee336e3d0f039699d633a02eeeef62cf4fcba2bdbd9ec35d65d1480390337bcf64111cfa6019a546ca0f7aac8fe6d3d44892e446ef18203337791ce528ed')
+            '1c05ee336e3d0f039699d633a02eeeef62cf4fcba2bdbd9ec35d65d1480390337bcf64111cfa6019a546ca0f7aac8fe6d3d44892e446ef18203337791ce528ed'
+            '1856426410ce7af9810db5e1e20295bed24fc2404789e612f136757c02f0d2f059496edfe4d7ae56fdda1c7c478acf1bad14c3c000d363cfaf657ec93b436c34')
 
 prepare() {
   cd HTTPretty-$pkgver
@@ -54,6 +56,7 @@
   patch -p1 -i ../$pkgname-urllib3-2.3-compatibility.patch
   sed -i -e '/rednose/d' -e '/cover/d' setup.cfg
   patch -p1 -i ../python-3.14.patch # Fedora patch
+  patch -p1 -i ../increase-test-timeouts.patch
 }
 
 build() {
