diff -r -u lldb-13.0.0.src-orig/cmake/modules/LLDBConfig.cmake lldb-13.0.0.src/cmake/modules/LLDBConfig.cmake
--- lldb-13.0.0.src-orig/cmake/modules/LLDBConfig.cmake	2022-02-07 16:44:21.365759406 +0100
+++ lldb-13.0.0.src/cmake/modules/LLDBConfig.cmake	2022-02-07 16:54:49.100210127 +0100
@@ -297,7 +297,11 @@
 # Figure out if lldb could use lldb-server.  If so, then we'll
 # ensure we build lldb-server when an lldb target is being built.
 if (CMAKE_SYSTEM_NAME MATCHES "Android|Darwin|FreeBSD|Linux|NetBSD|Windows")
-  set(LLDB_CAN_USE_LLDB_SERVER ON)
+  if (CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+    set(LLDB_CAN_USE_LLDB_SERVER OFF)
+  else()
+    set(LLDB_CAN_USE_LLDB_SERVER ON)
+  endif()
 else()
   set(LLDB_CAN_USE_LLDB_SERVER OFF)
 endif()
