--- PKGBUILD
+++ PKGBUILD
@@ -19,10 +19,16 @@ sha256sums=('f64d5533661c766a2ff97062af5a0523b9e2cd56aa3d95d9624c418e120d1427'
             'SKIP')
 validpgpkeys=('474E22316ABF4785A88C6E8EA2C794A986419D8A'  # Tom Stellard <tstellar@redhat.com>
               'D574BD5D1D0E98895E3BF90044F2485E45D59042') # Tobias Hieta <tobias@hieta.se>
+source+=(0000-disable-lldb-server.patch)
+sha256sums+=('d3a53fc2e6626269923cebc848b6705efd5b2678e75a6f85c9366963208fa614')

 prepare() {
   mv cmake{-$pkgver.src,}
   cd lldb-$pkgver.src
+
+  # lldb-server does not support riscv64
+  patch -Np1 -i ../0000-disable-lldb-server.patch
+
   mkdir build
 }
