--- PKGBUILD
+++ PKGBUILD
@@ -11,20 +11,28 @@ license=('custom:Apache 2.0 with LLVM Exception')
 depends=('llvm-libs' 'clang' 'python' 'python-six')
 makedepends=('llvm' 'cmake' 'ninja' 'swig' 'python-sphinx')
 _source_base=https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver
-source=($_source_base/$pkgname-$pkgver.src.tar.xz{,.sig})
+source=($_source_base/$pkgname-$pkgver.src.tar.xz{,.sig}
+        0000-disable-lldb-server.patch)
 sha256sums=('4a0c590bccbb3ce8fcbc1997aaed89a0ddbfd8be4b7f393230a867b2de62db16'
-            'SKIP')
+            'SKIP'
+            'd3a53fc2e6626269923cebc848b6705efd5b2678e75a6f85c9366963208fa614')
 validpgpkeys+=('B6C8F98282B944E3B0D5C2530FC3042E345AD05D') # Hans Wennborg <hans@chromium.org>
 validpgpkeys+=('474E22316ABF4785A88C6E8EA2C794A986419D8A') # Tom Stellard <tstellar@redhat.com>
 
 prepare() {
   cd "$srcdir/$pkgname-$pkgver.src"
+
+  # lldb-server does not support riscv64
+  patch -Np1 -i ../0000-disable-lldb-server.patch
+
   mkdir build
 }
 
 build() {
   cd "$srcdir/$pkgname-$pkgver.src/build"
 
+  CFLAGS="$CFLAGS -pthread"
+  CXXFLAGS="$CXXFLAGS -pthread"
   cmake .. -G Ninja \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_INSTALL_PREFIX=/usr \
