--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,20 @@ license=('custom:Apache 2.0 with LLVM Exception')
 depends=('llvm-libs' 'clang' 'python' 'python-six')
 makedepends=('llvm' 'cmake' 'ninja' 'swig' 'python-sphinx')
 _source_base=https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver
-source=($_source_base/lldb-$pkgver.src.tar.xz{,.sig})
+source=($_source_base/lldb-$pkgver.src.tar.xz{,.sig}
+        0000-disable-lldb-server.patch)
 sha256sums=('2ac41160b0bdc74bf72d4f9a3702f290f7c09998da6fb73281cc497f4ddbe130'
-            'SKIP')
+            'SKIP'
+            'd3a53fc2e6626269923cebc848b6705efd5b2678e75a6f85c9366963208fa614')
 validpgpkeys+=('B6C8F98282B944E3B0D5C2530FC3042E345AD05D') # Hans Wennborg <hans@chromium.org>
 validpgpkeys+=('474E22316ABF4785A88C6E8EA2C794A986419D8A') # Tom Stellard <tstellar@redhat.com>
 
 prepare() {
   cd lldb-$pkgver.src
+
+  # lldb-server does not support riscv64
+  patch -Np1 -i ../0000-disable-lldb-server.patch
+
   mkdir build
 }
 
