--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,13 @@
 provides=('libstoken.so')
 source=(https://downloads.sourceforge.net/stoken/${pkgname}-${pkgver}.tar.gz{,.asc}
         https://github.com/stoken-dev/stoken/commit/8fc6f0dd.patch
-        nettle-4.patch)
+        nettle-4.patch
+        0001-stc-nettle-initialize-base64-decode-buffer-length.patch)
 sha256sums=('aa2b481b058e4caf068f7e747a2dcf5772bcbf278a4f89bc9efcbf82bcc9ef5a'
             'SKIP'
             'dc1bfc755a3b2b368398a4623817b4093840469b6fb45be65d53754c4a6042f7'
-            'ac2e4a9506c9e70b4c200ad1d56799fdb36f15a7bf474b492939ca9f5d7e693c')
+            'ac2e4a9506c9e70b4c200ad1d56799fdb36f15a7bf474b492939ca9f5d7e693c'
+            '03166e85f099d8a32c984d337d0962a1f78f63b80be3238404687864d8ddae7b')
 validpgpkeys=('45DFF2D5205FE8CD74C2EE6C63B81599BC0B0D65')
 
 prepare() {
@@ -26,6 +28,8 @@
 # Fix build with nettle 4
   patch -p1 -i ../8fc6f0dd.patch
   patch -p1 -i ../nettle-4.patch
+  # Fix SDTID base64 decoding with nettle
+  patch -p1 -i ../0001-stc-nettle-initialize-base64-decode-buffer-length.patch
   autoreconf -fiv
 }
 
