diff --git PKGBUILD PKGBUILD
index 0c96862..f3902de 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,12 @@
   ruby-getoptlong # webkit
   unzip
 )
+makedepends_riscv64=(
+  bun
+  clang20
+  lld20
+  llvm20
+)
 options=('!lto')
 source=(
   "git+$url.git#tag=bun-v$pkgver"
@@ -71,15 +77,45 @@
   case "$CARCH" in
     x86_64)  _zigzip="$srcdir/$pkgname-zig-$_zigver-x86_64.zip" ;;
     aarch64) _zigzip="$srcdir/$pkgname-zig-$_zigver-aarch64.zip" ;;
+    riscv64)
+      patch -Np1 -d vendor/WebKit < "$srcdir/$pkgname-webkit-riscv64.patch"
+      patch -Np1 < "$srcdir/$pkgname-riscv64.patch"
+      ;;
   esac
-  # unzip can't strip dirs but we want to get rid of the first contained dir so we have to do a little dance
-  mkdir -p vendor/zig.extract
-  unzip -q "$_zigzip" -d vendor/zig.extract
-  mv vendor/zig.extract/* vendor/zig
-  printf '%s\n' "$_zigver" > vendor/zig/.zig-commit
+  if [[ -n "${_zigzip:-}" ]]; then
+    # unzip can't strip dirs but we want to get rid of the first contained dir so we have to do a little dance
+    mkdir -p vendor/zig.extract
+    unzip -q "$_zigzip" -d vendor/zig.extract
+    mv vendor/zig.extract/* vendor/zig
+    printf '%s\n' "$_zigver" > vendor/zig/.zig-commit
+  fi
 }
 
 build() {
+  if [[ "$CARCH" == riscv64 ]]; then
+    local cmake_vars=(
+      CMAKE_INSTALL_PREFIX=/usr
+      CMAKE_BUILD_TYPE=None
+      ZIG_PIE=ON
+      ZIG_SHARED_LLVM=ON
+      ZIG_USE_LLVM_CONFIG=ON
+      ZIG_TARGET_TRIPLE=native-linux.6.6-gnu.2.40
+      ZIG_TARGET_MCPU=baseline
+    )
+
+    (
+      export PATH="/usr/lib/llvm20/bin:$PATH"
+      cmake -B "$srcdir/$pkgname-zig-build" "${cmake_vars[@]/#/-D}" "$srcdir/$pkgname-zig"
+      cmake --build "$srcdir/$pkgname-zig-build"
+      DESTDIR="$srcdir/$pkgname-zig-install" cmake --install "$srcdir/$pkgname-zig-build"
+    )
+
+    mkdir -p "$srcdir/$pkgname/vendor/zig"
+    install -Dm755 "$srcdir/$pkgname-zig-install/usr/bin/zig" "$srcdir/$pkgname/vendor/zig/zig"
+    cp -a "$srcdir/$pkgname-zig-install/usr/lib/zig" "$srcdir/$pkgname/vendor/zig/lib"
+    printf '%s\n' "$_zigver" > "$srcdir/$pkgname/vendor/zig/.zig-commit"
+  fi
+
   export PATH="/usr/lib/llvm21/bin:$PATH"
 
   case "$CARCH" in
@@ -87,7 +123,9 @@
     aarch64) _bundir="bun-linux-aarch64" ;;
   esac
 
-  export PATH="$srcdir/$_bundir:$PATH"
+  if [[ -n "${_bundir:-}" ]]; then
+    export PATH="$srcdir/$_bundir:$PATH"
+  fi
 
   cd $pkgname
   bun scripts/build.ts \
@@ -112,3 +150,12 @@
   install -vDm644 completions/bun.fish "$pkgdir/usr/share/fish/vendor_completions.d/bun.fish"
   install -vDm644 completions/bun.zsh "$pkgdir/usr/share/zsh/site-functions/_bun"
 }
+
+source_riscv64=(
+  "$pkgname-riscv64.patch"
+  "$pkgname-webkit-riscv64.patch"
+  "$pkgname-zig::git+https://github.com/oven-sh/zig.git#tag=autobuild-$_zigver"
+)
+b2sums_riscv64=('d815fa6f73e37b3c2d20183fc3418a313592752b8e381fc7d3217952a5ba7b65c278bcca63173eb083a7a3f04d00ea25ccc67ec75418b7d6d75de4183ca05fb7'
+                'b64777e7bf3fb030235f3a53ad035c835a9c4335e3508ab9d94c93804041e5d3380a0c63803c6dd5612ecbef765a998cf9496ca8d612c9c361cff4cf8c187292'
+                '45421c507deaf962313ab0e7fb94768bf8aad6313597e372f965149409f0a5432f1f6b75e7dc57e66c5ad8180aa59f4292db508dbb6ed52d2c0cb72989462ba8')
