--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,13 @@ sha256sums=('dfcd9bcae364ce09b6ae59ea598f9dcad3e27a980b12c4b245acd336fa02c5a2'
             'SKIP')
 
 prepare() {
+  pushd "$srcdir"/$pkgbase-$pkgver/src/cxx_supportlib/vendor-copy/libuv/
+  autoreconf -fiv
+  popd
+  pushd "$srcdir"/$pkgbase-$pkgver/src/cxx_supportlib/vendor-modified/libev/
+  autoreconf -fiv
+  popd
+
   mkdir -p build
   cd build
   ln -sf /usr/src/nginx/auto
