--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,12 @@ sha256sums=('73fa22da5a11e4bc4ad6b95c13a0e393ba18109e6e07bd1953c45b2f0c0aae80'
             'SKIP')
 
 prepare() {
+	pushd "$srcdir"/$pkgbase-$pkgver/src/cxx_supportlib/vendor-copy/libuv/
+	autoreconf -fiv
+	popd
+	pushd "$srcdir"/$pkgbase-$pkgver/src/cxx_supportlib/vendor-modified/libev/
+	autoreconf -fiv
+	popd
 	mkdir -p build
 	cd build
 	ln -sf /usr/src/nginx/auto
