Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 423605)
+++ PKGBUILD	(working copy)
@@ -48,6 +48,9 @@
 	elif [ "${CARCH}" == 'i686' ]; then
 		openssltarget='linux-elf'
 		optflags=''
+	elif [ "${CARCH}" == 'riscv64' ]; then
+		openssltarget='linux-generic64'
+		optflags=''
 	fi
 
 	# mark stack as non-executable: http://bugs.archlinux.org/task/12434
