--- PKGBUILD
+++ PKGBUILD
@@ -43,6 +43,7 @@ b2sums=('cd1ad877f36a9b1a403f1c8d789a60aaef9d719e160c0aa40c9cc9b29bbaa7e84e8bdd4
 
 prepare() {
   cd Aegisub
+  patch -Np1 -i ../use-plct-luajit.patch
   meson subprojects download luajit
   meson subprojects packagefiles --apply luajit
   sed "/subdir('tests')/d" -i meson.build
@@ -69,4 +70,7 @@ package() {
   install -Dm 644 Aegisub/LICENCE -t "${pkgdir}"/usr/share/licenses/aegisub/
 }
 
+source+=(use-plct-luajit.patch)
+b2sums+=('6e56712104026c9cd6f4040722d71091d705f8093c39eb8140bebe0ef0cfb44939f543e8c39ce8313c927e7cae5c9dbb901d909ecf7003b15a806b47403e0d1d')
+
 # vim: ts=2 sw=2 et:
