diff --git a/subprojects/luajit.wrap b/subprojects/luajit.wrap
index e1daad5ab..b2f94eef3 100644
--- a/subprojects/luajit.wrap
+++ b/subprojects/luajit.wrap
@@ -1,7 +1,7 @@
 [wrap-git]
 directory = luajit
-url = https://github.com/LuaJIT/LuaJIT.git
-revision = v2.1
+url = https://github.com/plctlab/LuaJIT.git
+revision = 7ff9f3eb0f2d1b01a38fed6c5d57ca3d69d0c9b0
 patch_directory = luajit
 
 [provide]
