--- PKGBUILD
+++ PKGBUILD
@@ -54,9 +54,10 @@ build() {
     -DFEATURE_openssl_linked=ON \
     -DFEATURE_system_sqlite=ON \
     -DFEATURE_system_xcb_xinput=ON \
-    -DFEATURE_no_direct_extern_access=ON \
+    -DFEATURE_reduce_relocations=OFF \
     -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
     -DCMAKE_MESSAGE_LOG_LEVEL=STATUS
+# -DFEATURE_reduce_relocations=OFF for https://bugreports.qt.io/browse/QTBUG-112332
   cmake --build build
 }
 
