diff --git PKGBUILD PKGBUILD
index 82ebbb5..175041e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -44,9 +44,9 @@ source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pk
         https://invent.kde.org/pim/kleopatra/-/commit/10b61870.patch)
 sha256sums=('7c9ad9d10cd368d58cdadd41f78b8a41853b8cd2b15eb55472f3b5ce707ad1c2'
             'SKIP'
-            '0120d812f9a57cbd4a5a2718fbbc9acef090841701cb2c6b4f11a596986840e6'
-            'c2796b58683b07eabeae48a517192460c474a1722d7022e796b94df6553d933d'
-            '75f5c94d85eadad74dcef21f66c019946217a051a9e8b90803d5eceaadb3fcd4')
+            '7b10385b68d92b8bf728b3a3262f26f44e5b74043eb38dfc4b1ff35709a0a771'
+            '2aad7b88dcd18d3ada7b7d0288975345fc56c3bfe24b2e94ff8ebc1b0288db4c'
+            '889fb8e16a5388f743af58b35009dd7836e2a079cdc099fa818b2d0d2b4cfe00')
 validpgpkeys=(CA262C6C83DE4D2FB28A332A3A6A4DB839EAA6D7  # Albert Astals Cid <aacid@kde.org>
               F23275E4BF10AFC1DF6914A6DBD2CE893E2D1C87  # Christoph Feck <cfeck@kde.org>
               D81C0CB38EB725EF6691C385BB463350D6EF31EF) # Heiko Becker <heiko.becker@kde.org>
