diff --git PKGBUILD PKGBUILD
index ef6413e..9d0a346 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ url='https://www.postgresql.org/'
 arch=('x86_64')
 license=('custom:PostgreSQL')
 makedepends=('krb5' 'libxml2' 'python' 'perl' 'tcl>=8.6.0' 'openssl>=1.0.0'
-             'pam' 'zlib' 'icu' 'systemd' 'libldap' 'llvm' 'clang' 'libxslt'
+             'pam' 'zlib' 'icu' 'systemd' 'libldap' 'libxslt'
              'util-linux')
 source=(https://ftp.postgresql.org/pub/source/v${pkgver}/postgresql-${pkgver}.tar.bz2
         postgresql-run-socket.patch
@@ -78,7 +78,6 @@ build() {
     --with-icu
     --with-systemd
     --with-ldap
-    --with-llvm
     --with-libxslt
     --with-lz4
     --with-zstd
@@ -168,7 +167,7 @@ package_postgresql() {
   pkgdesc='Sophisticated object-relational DBMS'
   backup=('etc/pam.d/postgresql' 'etc/logrotate.d/postgresql')
   depends=("postgresql-libs>=${pkgver}" 'krb5' 'libxml2' 'readline>=6.0'
-           'openssl>=1.0.0' 'pam' 'icu' 'systemd-libs' 'libldap' 'llvm-libs'
+           'openssl>=1.0.0' 'pam' 'icu' 'systemd-libs' 'libldap'
            'libxslt' 'lz4' 'zstd')
   optdepends=('python: for PL/Python 3 support'
               'perl: for PL/Perl support'
