diff --git PKGBUILD PKGBUILD
index 1abfbd1..e0c6559 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@
 arch=('x86_64')
 license=('custom:PostgreSQL')
 makedepends=('krb5' 'libxml2' 'python' 'perl' 'tcl' 'openssl' 'pam' 'zlib'
-             'icu' 'systemd' 'libldap' 'llvm' 'clang' 'libxslt' 'util-linux')
+             'icu' 'systemd' 'libldap' 'libxslt' 'util-linux')
 source=(https://ftp.postgresql.org/pub/source/v${pkgver}/postgresql-${pkgver}.tar.bz2
         0001-Set-DEFAULT_PGSOCKET_DIR-to-run-postgresql.patch
         0002-Force-RPATH-to-be-used-for-the-PL-Perl-plugin.patch
@@ -87,7 +87,6 @@
     --with-icu
     --with-systemd
     --with-ldap
-    --with-llvm
     --with-libxslt
     --with-lz4
     --with-zstd
@@ -178,7 +177,7 @@
   pkgdesc='Sophisticated object-relational DBMS'
   backup=('etc/pam.d/postgresql' 'etc/logrotate.d/postgresql')
   depends=("postgresql-libs>=${pkgver}" 'krb5' 'libxml2' 'readline' 'openssl'
-           'pam' 'icu' 'systemd-libs' 'libldap' 'llvm-libs' 'libxslt' 'lz4'
+           'pam' 'icu' 'systemd-libs' 'libldap' 'libxslt' 'lz4'
            'zstd')
   optdepends=('python: for PL/Python 3 support'
               'perl: for PL/Perl support'
