diff --git PKGBUILD PKGBUILD
index 91ce359..4a0146c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ url='https://www.postgresql.org/'
 arch=('x86_64')
 license=('custom:PostgreSQL')
 makedepends=('krb5' 'libxml2' 'python' 'perl' 'tcl>=8.6.0' 'openssl>=1.0.0'
-             'pam' 'zlib' 'icu' 'systemd' 'libldap' 'llvm15' 'clang15' 'libxslt'
+             'pam' 'zlib' 'icu' 'systemd' 'libldap' 'libxslt'
              'util-linux')
 source=(https://ftp.postgresql.org/pub/source/v${pkgver}/postgresql-${pkgver}.tar.bz2
         0001-Set-DEFAULT_PGSOCKET_DIR-to-run-postgresql.patch
@@ -78,7 +78,6 @@ build() {
     --with-icu
     --with-systemd
     --with-ldap
-    --with-llvm
     --with-libxslt
     --with-lz4
     --with-zstd
@@ -90,7 +89,6 @@ build() {
   # Fix static libs
   CFLAGS+=" -ffat-lto-objects"
 
-  LLVM_CONFIG=llvm-config-15 CLANG=/usr/lib/llvm15/bin/clang \
   ./configure "${configure_options[@]}"
   make world
 }
@@ -170,7 +168,7 @@ package_postgresql() {
   pkgdesc='Sophisticated object-relational DBMS'
   backup=('etc/pam.d/postgresql' 'etc/logrotate.d/postgresql')
   depends=("postgresql-libs>=${pkgver}" 'krb5' 'libxml2' 'readline>=6.0'
-           'openssl>=1.0.0' 'pam' 'icu' 'systemd-libs' 'libldap' 'llvm15-libs'
+           'openssl>=1.0.0' 'pam' 'icu' 'systemd-libs' 'libldap'
            'libxslt' 'lz4' 'zstd')
   optdepends=('python: for PL/Python 3 support'
               'perl: for PL/Perl support'
