diff --git PKGBUILD PKGBUILD
index 7eb38fb..270d699 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,7 +23,6 @@ depends=(
   'libldap'
   'libxml2'
   'libxslt'
-  'llvm-libs'
   'lz4'
   'openssl'
   'pam'
@@ -34,8 +33,6 @@ depends=(
   'zstd'
 )
 makedepends=(
-  'clang'
-  'llvm'
   'perl'
   'perl-ipc-run'
   'python'
@@ -106,7 +103,6 @@ build() {
     --with-ldap
     --with-libxml
     --with-libxslt
-    --with-llvm
     --with-lz4
     --with-openssl
     --with-pam
