--- PKGBUILD
+++ PKGBUILD
@@ -112,7 +112,6 @@ build() {
     --with-ldap
     --with-libxml
     --with-libxslt
-    --with-llvm
     --with-lz4
     --with-openssl
     --with-pam
@@ -279,4 +278,7 @@ package_postgresql-docs() {
   install -Dm 644 COPYRIGHT -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+depends=(${depends[@]/llvm-libs})
+makedepends=(${makedepends[@]/clang})
+makedepends=(${makedepends[@]/llvm})
 # vim:set sw=2 sts=-1 et:
