diff --git a/Cargo.toml b/Cargo.toml
index f44a1061f4..60692f76d0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -764,6 +764,12 @@ windows-capture = { git = "https://github.com/zed-industries/windows-capture.git
 # Makes the workspace hack crate refer to the local one, but only when you're building locally
 workspace-hack = { path = "tooling/workspace-hack" }
 
+crash-context = { git = "https://github.com/hack3ric/crash-handling", rev = "f2c74406171235c960907fb0f3acb2770dcc4a35" }
+crash-handler = { git = "https://github.com/hack3ric/crash-handling", rev = "f2c74406171235c960907fb0f3acb2770dcc4a35" }
+minidumper = { git = "https://github.com/hack3ric/crash-handling", rev = "f2c74406171235c960907fb0f3acb2770dcc4a35" }
+minidump-common = { git = "https://github.com/hack3ric/rust-minidump", rev = "56e5539e3b3cf38d0ab07e3a08d34c2d27e1d31a" }
+minidump-writer = { git = "https://github.com/hack3ric/minidump-writer", rev = "da548e48906a4fd497d7d08a30757a37221004cb" }
+
 [profile.dev]
 split-debuginfo = "unpacked"
 codegen-units = 16
