--- PKGBUILD
+++ PKGBUILD
@@ -37,7 +37,14 @@ makedepends=(cargo
              cmake
              protobuf
              vulkan-headers
-             vulkan-validation-layers)
+             vulkan-validation-layers
+             # needed when building libwebrtc
+             cpio
+             git
+             gn
+             ninja
+             pulseaudio # not required at runtime
+             python-httplib2)
 optdepends=('clang: improved C/C++ language support'
             'eslint: improved Javascript language support'
             'pyright: improved Python language support'
@@ -53,6 +60,8 @@ _appid=dev.zed.Zed
 
 prepare() {
 	cd "$_archive"
+	patch -Np1 -i ../minidumper-riscv64.patch
+	cargo update -p crash-context -p crash-handler -p minidumper -p minidump-common -p minidump-writer
 	cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 	export DO_STARTUP_NOTIFY="true"
 	export APP_ICON="zed"
@@ -69,10 +78,20 @@ _srcenv() {
 	CFLAGS+=' -ffat-lto-objects'
 	CXXFLAGS+=' -ffat-lto-objects'
 	RUSTFLAGS+=" --remap-path-prefix $PWD=/"
+	CFLAGS="${CFLAGS/-fstack-clash-protection}"
+	CXXFLAGS="${CXXFLAGS/-fstack-clash-protection}"
 }
 
 build() {
 	_srcenv
+
+	pushd "$srcdir/livekit-rust-sdks/webrtc-sys/libwebrtc"
+	export CC=clang CXX=clang++ AR=ar NM=nm
+	sed -i '2i set -e' build_linux.sh
+	./build_linux.sh --arch riscv64 --toolchain host
+	export LK_CUSTOM_WEBRTC="$(pwd)/linux-riscv64-release"
+	popd
+
 	export ZED_UPDATE_EXPLANATION='Updates are handled by pacman'
 	export RELEASE_VERSION="$pkgver"
 	export PROTOC=/usr/bin/protoc
@@ -95,3 +114,8 @@ package() {
 	install -Dm0644 -t "$pkgdir/usr/share/applications/" "$_appid.desktop"
 	install -Dm0644 crates/zed/resources/app-icon.png "$pkgdir/usr/share/icons/$pkgname.png"
 }
+
+source+=("git+https://github.com/hack3ric/livekit-rust-sdks.git#commit=0563cf43c285f61b7889922b917128bec5dc552f"
+         "minidumper-riscv64.patch")
+sha256sums+=('8262058073b7fb6f28ce30a701629986abcd8987745e040b1710801573502d75'
+             '18b4bed2367ce1932b6b8669299629d3d1623206ae649d74941a32ba6ea5917b')
