--- a/call/rtp_config.cc
+++ b/call/rtp_config.cc
@@ -258,9 +258,7 @@
   stream_config.raw_payload = raw_payload;
   if (!rtx.ssrcs.empty()) {
     RTC_DCHECK_EQ(ssrcs.size(), rtx.ssrcs.size());
-    auto& stream_config_rtx = stream_config.rtx.emplace();
-    stream_config_rtx.ssrc = rtx.ssrcs[index];
-    stream_config_rtx.payload_type = rtx.payload_type;
+    stream_config.rtx = RtpStreamConfig::Rtx{rtx.ssrcs[index], rtx.payload_type};
   }
 
   return stream_config;
