--- PKGBUILD
+++ PKGBUILD
@@ -47,6 +47,8 @@ prepare(){
   patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-2.2.0-libdir.patch
   # fix generation of instruments in package()
   patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-2.2.0-libgig_package.patch
+  # add time stamp support for riscv64, and fix atomic arithmetic operation
+  patch -Np1 -d $pkgname-$pkgver -i ../0000-fix-riscv-atomic.patch
 
   # extract custom license exception
   sed -n '11,15p' $pkgname-$pkgver/README > LinuxSampler-Non-Commercial.txt
@@ -79,3 +81,7 @@ package() {
   mv -v "$pkgdir/usr/share/man/man1/lscp.1" "$pkgdir/usr/share/man/man1/lscp-$pkgname.1"
 }
 # vim:set ts=2 sw=2 et:
+
+source+=(0000-fix-riscv-atomic.patch)
+sha512sums+=('fb477be65873735cc066ed5389a6e43222691ab775bceeb1565f825aa1cf66c1dcfb2fd7bb0feff81dd1c0d2adf1535759cf062ddbd59c048bfd0b3c27831534')
+b2sums+=('bfce471e5c0415975033f5c1d66eb25b217c4e5a57e88cc3b4ac1fa0d6ba5431861bea092da00b59b7e251c18caca3f83ee9b954aa7bf21db6ebf61e6cea99fc')
