--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,9 @@ prepare(){
   # extract custom license exception
   sed -n '11,15p' $pkgname-$pkgver/README > LinuxSampler-Non-Commercial.txt
 
+  # add time stamp support for riscv64, and fix atomic arithmetic operation
+  patch -Np1 -d $pkgname-$pkgver -i ../0000-fix-riscv-atomic.patch
+
   cd $pkgname-$pkgver
   autoreconf -fiv
 }
@@ -77,3 +80,7 @@ package() {
   mv -v "$pkgdir/usr/share/man/man1/lscp.1" "$pkgdir/usr/share/man/man1/lscp-$pkgname.1"
 }
 # vim:set ts=2 sw=2 et:
+
+source+=(0000-fix-riscv-atomic.patch)
+sha512sums+=('4b041d9d473e388819a7d84732fc20b5d892c1be9f874e4fb9d73765b26f32573ac679e8aea8d7d0b05e09c1587bfa2ae43c08a397d6c824c328519509871a02')
+b2sums+=('751c9dd34e9c5c20dd465f45cdd8058d02bc2a64929ac1c9ffe8dbafa58ba2c554be581773b78e53a9f9b5db561f876fdf1a5ba13e6e0574adbdeead5fb9bd97')
