diff --git PKGBUILD PKGBUILD
index dca2c3d..e72fea1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -31,7 +31,10 @@ prepare() {
 
 build() {
     cd "$pkgname-$pkgver"
-    ./configure --prefix=/usr \
+    ./configure \
+        --build=riscv64-unknown-linux-gnu \
+        --host=riscv64-unknown-linux-gnu \
+        --prefix=/usr \
         --libexecdir=/usr/lib/telepathy \
         --disable-plugins --disable-Werror \
         --with-tls=gnutls \
