--- PKGBUILD
+++ PKGBUILD
@@ -18,16 +18,20 @@ makedepends=('vala' 'intltool' 'git' 'gnome-common')
 install=$pkgname.install
 source=("git+https://bitbucket.org/shuerhaaken/xnoise.git#tag=$pkgver"
         xnoise-ListStore.patch
-        xnoise-compilation-fix.patch)
+        xnoise-compilation-fix.patch
+        "fix-invalid-handler.patch::https://bitbucket.org/!api/2.0/repositories/shuerhaaken/xnoise/diff/xiejiss/xnoise:476f3867b2ee%0D842107306375?from_pullrequest_id=26")
 sha512sums=('SKIP'
             'b47297bda23fc4dcd22bf44a84ad2c5e3b5ed0d648ef78a3bbe4de4d481c82e8ab0c52193dc8c4373e74cd389537faf66bd434cd8d695eeff4b2820c67bf1423'
-            'fababde2f5ccd7c7074d8c7cd684ab36924e6781af05293736227c15d864f3fc726a50215ca84b5f4776e47cad54e1b5723932e98c042ad226c853b9efc2a5ad')
+            'fababde2f5ccd7c7074d8c7cd684ab36924e6781af05293736227c15d864f3fc726a50215ca84b5f4776e47cad54e1b5723932e98c042ad226c853b9efc2a5ad'
+            '94778fba770c7a12acf87727ac9aae4fce5e003ccc07110f13411281f3654a0e0ac2059f361bca7debce772107441b124c696e2a9013074aca025fda9f34ab9c')
 
 prepare() {
   cd $pkgname
   # Patches from openSUSE. Thanks!
   patch -p1 -i ../xnoise-ListStore.patch
   patch -p1 -i ../xnoise-compilation-fix.patch
+  # see https://bitbucket.org/shuerhaaken/xnoise/pull-requests/26/fix-make-xnoise-build-for-vala-044
+  patch -p1 -i ../fix-invalid-handler.patch
 }
 
 build() {
