diff --git PKGBUILD PKGBUILD
index 61150a6f..bdb3bf09 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,15 @@ makedepends=(gobject-introspection gtk-doc git meson yelp-tools systemd)
 optdepends=('eog-plugins: Additional features')
 groups=(gnome)
 _commit=4a315fd3dc4ba5ca913d7bb08edffad10d7fbede  # tags/41.0^0
-source=("git+https://gitlab.gnome.org/GNOME/eog.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/eog.git#commit=$_commit"
+        "fix-meson-build.patch::https://gitlab.gnome.org/GNOME/eog/-/merge_requests/113.patch")
+sha256sums=('SKIP'
+            '4e13af43a2c2fa131605c62f7b6650a931513eb69569f9c497a718a1fecbbebf')
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i $srcdir/fix-meson-build.patch
+}
 
 pkgver() {
   cd $pkgname
