diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index ec670afe15b..7abf9df6e8b 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -275,10 +275,8 @@ gcc_toolchain("clang_riscv32") {
 }
 
 gcc_toolchain("riscv64") {
-  prefix = "riscv64-linux-gnu-"
-  if (riscv64_toolchain_prefix != "") {
-    prefix = riscv64_toolchain_prefix
-  }
+  # Not cross-compiling
+  prefix = ""
 
   cc = "${gcc_compiler_prefix}${prefix}gcc"
   cxx = "${gcc_compiler_prefix}${prefix}g++"
