--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -276,10 +276,8 @@ gcc_toolchain_suite("clang_riscv32") {
 }
 
 gcc_toolchain_suite("riscv64") {
-  prefix = "riscv64-linux-gnu-"
-  if (riscv64_toolchain_prefix != "") {
-    prefix = riscv64_toolchain_prefix
-  }
+  # Not cross-compiling
+  prefix = ""
 
   cc = "${gcc_compiler_prefix}${prefix}gcc"
   cxx = "${gcc_compiler_prefix}${prefix}g++"
