--- gclient.py.orig	2025-11-01 08:31:34.496028596 +0000
+++ gclient.py	2025-11-01 08:32:52.295977141 +0000
@@ -79,6 +79,9 @@
 #   To specify a target CPU, the variables target_cpu and target_cpu_only
 #   are available and are analogous to target_os and target_os_only.
 
+import os, sys
+os.system(f"{sys.executable} -m pip install httplib2==0.13.1 six==1.10.0 requests==2.31.0")
+
 __version__ = '0.7'
 
 import copy
