--- PKGBUILD
+++ PKGBUILD
@@ -53,6 +53,7 @@ EOF
   cd sdk
 
   patch -Np 1 --input=$srcdir/DEPS.patch
+  patch -Np 1 --input=$srcdir/dart-riscv-no-cross.patch
 
   python ../depot_tools/gclient.py sync -D \
       --nohooks \
@@ -72,7 +73,7 @@ build() {
   # gn args --list out
 
   /usr/bin/gn gen -qv out --args='
-                        target_cpu = "x64"
+                        target_cpu = "riscv64"
                         is_debug = false
                         is_release = true
                         is_clang = false
@@ -104,3 +105,6 @@ package() {
 }
 
 # vim:set ts=2 sw=2 et:
+
+source+=("dart-riscv-no-cross.patch")
+sha256sums+=('8c0d3a27a86aea34d7b932181111a9f660d2e7bf67d5e45055163e6b980b187f')
