--- PKGBUILD
+++ PKGBUILD
@@ -58,8 +58,17 @@
 
   export PATH+=":$PWD/depot_tools" DEPOT_TOOLS_UPDATE=0
 
+  # Bundled dependencies are not available for riscv64
+  sed -i '/^dependencies = \[/,/^]/c\dependencies = []' \
+    depot_tools/vpython.toml depot_tools/gsutil.vpython.toml
+  # Hack
+  patch -Np0 -d depot_tools < HACK-gclient-fix-dep.diff
+
   cd sdk
 
+  patch -Np 1 --input=$srcdir/dart-riscv-no-cross.patch
+  patch -Np 1 --input=$srcdir/gcc-warning-errors.patch
+
   patch -Np 1 --input="$srcdir/DEPS.patch"
 
   # Fix for https://github.com/dart-lang/sdk/issues/63406
@@ -127,3 +136,10 @@
 }
 
 # vim:set ts=2 sw=2 et:
+
+source+=("dart-riscv-no-cross.patch"
+         "gcc-warning-errors.patch"
+         "HACK-gclient-fix-dep.diff")
+sha256sums+=('22430d5b1369357bc741de4a53a42d8c3dc6b2944eea349314f8ed69e8f5682c'
+             'c3bac7dfdc594e53b6b29525b2df88f91774d2c65b2047d941a6db81a32577e3'
+             '97de9eb2a2fe19f177399013ce8057f27fd981c07b9567558e4a70e45bb047c3')
