diff --git PKGBUILD PKGBUILD
index b1ebb4c..05dd584 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,15 +17,18 @@ source=(
   'gortr.env'
   'gortr.service'
   'gortr.sysusers'
+  "update_module_version.patch"
 )
 
 sha512sums=('02cd38da2af1af11a3384578c81cc07098b6e0a5abdf17cd70dcc7256c7a0ec687a60d590e7ce7a133faa28d0a768ef206c7a59672a10998e52b8a2dc2752a44'
             '3b48c5aa6e4ee074df4e7087f664a114f9fd9b5d586f17cad9cc5af6a6bff788f46012dbec37b5c08e5b858bcbe7f59d7fc65e013104a57a5606e1c3d0b9ee61'
             'c5a1bfb1957ef42d73811d087151c565beaec46b0d533dcd27661304e26e394040f0c4f3d75b1eedd6728e3c4a27c07db5a626cb8d8e8fe99bc6851f858df955'
-            'be46844e548f7eaf46c7f7b07db42b192c88f7c2cb980ab0dca952838ad0c80854f21414f12d47f62f51128ddd5d971e243a8846eece604228c2a4d13d64f1d7')
+            'be46844e548f7eaf46c7f7b07db42b192c88f7c2cb980ab0dca952838ad0c80854f21414f12d47f62f51128ddd5d971e243a8846eece604228c2a4d13d64f1d7'
+            'e524575f8dd7e311edcebe9f957a5b4a95f015d48aeb331ea8f7e0cb5254f0ef0f90545f7f51662772e850c2627e30a39717dc630ef6180203c6569ddd3ce21e')
 
 prepare(){
   cd "$pkgname-$pkgver"
+  patch -Np1 -i "../update_module_version.patch"
   mkdir -p build/
 }
 
