Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -11,7 +11,7 @@
 license=(LGPL MPL)
 depends=(lzo zlib libpng fontconfig freetype2 libx11 libxext libxrender libxcb
          glib2 pixman)
-makedepends=(valgrind git meson)
+makedepends=(git meson)
 _commit=156cd3eaaebfd8635517c2baf61fcf3627ff7ec2  # tags/1.17.4^0
 source=("git+https://gitlab.freedesktop.org/cairo/cairo.git#commit=$_commit")
 sha256sums=('SKIP')
