--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,7 @@ libfftw3_threads.so libfftw3_omp.so libfftw3_mpi.so libfftw3.so)
 source=("http://www.fftw.org/${pkgname}-${pkgver}.tar.gz")
 sha512sums=('2d34b5ccac7b08740dbdacc6ebe451d8a34cf9d9bfec85a5e776e87adf94abfd803c222412d8e10fbaa4ed46f504aa87180396af1b108666cde4314a55610b40')
 b2sums=('3663872bd95a01d2c79af185f53918e1d1a9c91c620082402772c07f25b9b77d4920a78dc686fac6b51aa961b8d62b7e2cef3f3031e096bed40ced80b59b29d5')
-_build_types=(single double long-double quad)
+_build_types=(single double long-double)
 
 prepare() {
   local _i
@@ -45,21 +45,13 @@ build() {
     --enable-openmp
   )
   local _configure_single=(
-    --enable-sse
-    --enable-avx
     --enable-single
   )
   local _configure_double=(
-    --enable-sse2
-    --enable-avx
   )
   local _configure_long_double=(
     --enable-long-double
   )
-  local _configure_quad=(
-    --disable-mpi
-    --enable-quad-precision
-  )
   local _cmake_options=(
     -B build
     -S $pkgname-$pkgver-$_build_types
@@ -69,11 +61,6 @@ build() {
     -D ENABLE_THREADS=ON
     -D ENABLE_FLOAT=ON
     -D ENABLE_LONG_DOUBLE=ON
-    -D ENABLE_QUAD_PRECISION=ON
-    -D ENABLE_SSE=ON
-    -D ENABLE_SSE2=ON
-    -D ENABLE_AVX=ON
-    -D ENABLE_AVX2=ON
   )
 
   # create missing FFTW3LibraryDepends.cmake
@@ -84,7 +71,7 @@ build() {
 
   export F77='gfortran'
   # use upstream default CFLAGS while keeping our -march/-mtune
-  CFLAGS+=" -O3 -fomit-frame-pointer -malign-double -fstrict-aliasing -ffast-math"
+  CFLAGS+=" -O3 -fomit-frame-pointer -fstrict-aliasing"
 
   for _name in "${_build_types[@]}"; do
     (
@@ -99,9 +86,6 @@ build() {
         long-double)
         "${_configure[@]}" "${_configure_long_double[@]}"
         ;;
-        quad)
-        "${_configure[@]}" "${_configure_quad[@]}"
-        ;;
       esac
       # fix overlinking because of libtool
       sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
