--- PKGBUILD
+++ PKGBUILD
@@ -24,7 +24,7 @@
 source=(http://www.fftw.org/$pkgname-$pkgver.tar.gz)
 sha512sums=('ca1bf80490dc6955a0ab49b1af05d6658c2ecc0968b3bde5b4af22271e47d30cd38f6f8347e8e6124091b6a17717447942bee95f94ca29574ba71c4d167af351')
 b2sums=('4bd2ae0ada612243f0681b185230969da2bca6cfccb1e16d17f84459cbdd4ae84dc8a9ddb39ce5e4dfc7d0308709d1dc56688826299ee1017fc6115d16b2bd38')
-_build_types=(single double long-double quad)
+_build_types=(single double long-double)
 _soname=3.7.11
 
 _pick() {
@@ -61,21 +61,13 @@
     MPILIBS="$(mpicc --showme:link)"
   )
   local _configure_single=(
-    --enable-sse
-    --enable-avx
     --enable-single
   )
   local _configure_double=(
-    --enable-sse2
-    --enable-avx
   )
   local _configure_long_double=(
     --enable-long-double
   )
-  local _configure_quad=(
-    --disable-mpi
-    --enable-quad-precision
-  )
   local _cmake_options=(
     -B build
     -S $pkgname-$pkgver-$_build_types
@@ -86,11 +78,6 @@
     -D ENABLE_THREADS=ON
     -D ENABLE_FLOAT=ON
     -D ENABLE_LONG_DOUBLE=ON
-    -D ENABLE_QUAD_PRECISION=ON
-    -D ENABLE_SSE=ON
-    -D ENABLE_SSE2=ON
-    -D ENABLE_AVX=ON
-    -D ENABLE_AVX2=ON
   )
 
   # create missing FFTW3LibraryDepends.cmake
@@ -101,7 +88,7 @@
 
   export F77='gfortran'
   # use upstream default CFLAGS while keeping our -march/-mtune
-  CFLAGS+=" -O3 -fomit-frame-pointer -malign-double -fstrict-aliasing -ffast-math"
+  CFLAGS+=" -O3 -fomit-frame-pointer -fstrict-aliasing"
 
   for _name in "${_build_types[@]}"; do
     (
@@ -116,9 +103,6 @@
         long-double)
         "${_configure[@]}" "${_configure_long_double[@]}"
         ;;
-        quad)
-        "${_configure[@]}" "${_configure_quad[@]}"
-        ;;
       esac
       # fix overlinking because of libtool
       sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
