--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@ pkgdesc="Detection of CCTag markers made up of concentric circles"
 arch=('x86_64')
 url="https://github.com/alicevision/CCTag"
 license=('MPL2')
-depends=('boost-libs' 'tbb' 'cuda' 'qt6-base' 'opencv')
+depends=('boost-libs' 'tbb' 'qt6-base' 'opencv')
 makedepends=('boost' 'cmake' 'eigen' 'ninja')
 options=('!lto')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/alicevision/CCTag/archive/refs/tags/v${pkgver}.tar.gz")
@@ -23,11 +23,10 @@ build() {
   cmake \
     -Bbuild \
     -GNinja \
-    -DCUDA_HOST_COMPILER=/opt/cuda/bin/gcc \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_BUILD_TYPE=Release \
     -DBUILD_SHARED_LIBS=ON \
-    -DCCTAG_CUDA_CC_LIST_INIT="5.2 5.3 6.0 6.1 6.2 7.0 7.2 7.5 8.0 8.6 8.7 8.9 9.0"
+    -DCCTAG_WITH_CUDA:BOOL=OFF
   ninja -C build
 }
 
