--- PKGBUILD
+++ PKGBUILD
@@ -12,10 +12,9 @@ pkgdesc="A Vulkan overlay layer for monitoring FPS, temperatures, CPU/GPU load a
 arch=('x86_64')
 url="https://github.com/flightlessmango/MangoHud"
 license=('MIT')
-makedepends=('appstream' 'cmocka' 'git' 'glfw-x11' 'glslang' 'libxnvctrl' 'meson' 'nlohmann-json' 'python-mako')
+makedepends=('appstream' 'cmocka' 'git' 'glfw-x11' 'glslang' 'meson' 'nlohmann-json' 'python-mako')
 depends=('dbus' 'fmt' 'gcc-libs' 'glew' 'hicolor-icon-theme' 'libglvnd' 'libx11' 'python' 'python-matplotlib' 'python-numpy' 'vulkan-icd-loader')
-optdepends=('libxnvctrl: NVIDIA GPU stats by XNVCtrl'
-            'glfw-x11: Required for MangoApp'
+optdepends=('glfw-x11: Required for MangoApp'
             'gamescope: Use MangoApp as an overlay within gamescope')
 replaces=("$pkgname-common" "$pkgname-common-wayland" "$pkgname-common-x11" "mangoapp" "$pkgname-x11" "$pkgname-wayland")
 conflicts=("$pkgname-common" "$pkgname-common-wayland" "$pkgname-common-x11" "mangoapp" "$pkgname-x11" "$pkgname-wayland")
@@ -34,7 +33,8 @@ build() {
     arch-meson "$_pkgname-v$_tarpkgver" build \
         -Dmangoapp=true \
         -Dmangohudctl=true \
-        -Dmangoapp_layer=true
+        -Dmangoapp_layer=true \
+	 -Dwith_xnvctrl=disabled
 
     meson compile -C build
 }
