diff --git PKGBUILD PKGBUILD
index 7e5d315a9..3d4678297 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ arch=('any')
 depends=('python')
 makedepends=('python-setuptools')
 checkdepends=('python-pytest-asyncio' 'python-pytest-cov' 'python-pytest-repeat')
-source=("https://github.com/vxgmichel/aioconsole/archive/v$pkgver/$pkgname-$pkgver.tar.gz.tar.gz")
-sha512sums=('70ea0cca07711f31e261ec45d96a0901f9cf4d007b3195389a1b81e87054b6cb20429f9bb47faac66022824f412957b68e7fad928c78e65ce18b0e9db822ec94')
+source=("https://github.com/vxgmichel/aioconsole/archive/v$pkgver/$pkgname-$pkgver.tar.gz.tar.gz"
+        "fix_test_interact.patch"::"https://github.com/vxgmichel/aioconsole/commit/ad4cbf8aa2e2f1af4c9f598a82a42c2e4c5be77b.patch")
+sha512sums=('70ea0cca07711f31e261ec45d96a0901f9cf4d007b3195389a1b81e87054b6cb20429f9bb47faac66022824f412957b68e7fad928c78e65ce18b0e9db822ec94'
+            '841917cc520bf270f2a6fdf7adf007c704f0357e0e9778ea17d47c8e549662452fb52f8e7d9580454392a9f2630e9a443ac306755a0e07929b3cb626f7cccc3e')
+
+prepare() {
+  cd aioconsole-$pkgver
+  patch -Np1 -i "../fix_test_interact.patch"
+}
 
 build() {
   cd aioconsole-$pkgver
