diff --git PKGBUILD PKGBUILD
index 619a504..96044c0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,7 @@
 arch=('x86_64')
 url='https://wiki.gnome.org/Projects/GnomeApplets'
 license=('GPL')
-depends=('cpupower' 'gnome-panel' 'libgtop' 'tracker3' 'upower')
+depends=('cpupower' 'gnome-panel' 'libgtop' 'tracker3' 'upower' 'libnotify')
 makedepends=('itstool')
 optdepends=('tracker3-miners: Search bar')
 source=("https://download.gnome.org/sources/$pkgname/${pkgver%.*}/$pkgname-$pkgver.tar.xz")
