diff --git PKGBUILD PKGBUILD
index fb7186b..2673750 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,7 @@ license=(
   GPL-2.0-or-later
   LGPL-2.0-or-later
 )
-_glibver=2.82.2
+_glibver=2.82.4
 makedepends=(
   "glib2=$_glibver"
   cairo
@@ -39,7 +39,7 @@ source=(
   "git+https://gitlab.gnome.org/GNOME/gobject-introspection-tests.git"
 )
 b2sums=('37852497c4e75a64e7d856e361cda2b5f0edf689ff3128dbae5231d81978211bfabb03c6089965141510643eb513f428f507e548bc1bef3f4db2a4180a291c2b'
-        '3ff7ea88d69b945524b28211cd11373856e78813ea2795c6aff040fb264d6323edcd2fd2782d827936bdee34238a6995b2c7a6b51eeac49de583a21a0072a2d2'
+        '7611f78a903db16b954727124fb5d86dbacd5f89b6c672819d05cee3c012f3c23d3f1e1c1ebb63f176ca1b50e3d1a76390b7a7ea5552a74609f3acfad6abb16c'
         'SKIP')
 validpgpkeys=(
   923B7025EE03C1C59F42684CF0942E894B2EAFA0  # Philip Withnall <philip@tecnocode.co.uk>
