diff --git PKGBUILD PKGBUILD
index 0efb3ab..b0d729e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,18 +9,23 @@
 url="https://github.com/s-yata/marisa-trie"
 license=('BSD' 'LGPL')
 makedepends=('python' 'ruby' 'perl' 'swig')
-source=("marisa-trie-$pkgver.tar.gz::https://github.com/s-yata/marisa-trie/archive/v$pkgver.tar.gz")
-sha512sums=('c094e4b22e1457efdd20f2b978ee421b53e36ed94e4fdbd8944136c0ba23da4f6ba9fe3a2c64729c1426aee4dbe8098bfa5eebb943ae7fdaa4eec760485c564d')
+source=("marisa-trie-$pkgver.tar.gz::https://github.com/s-yata/marisa-trie/archive/v$pkgver.tar.gz"
+        "fix-format-security.patch::https://github.com/s-yata/marisa-trie/pull/45.patch")
+sha512sums=('c094e4b22e1457efdd20f2b978ee421b53e36ed94e4fdbd8944136c0ba23da4f6ba9fe3a2c64729c1426aee4dbe8098bfa5eebb943ae7fdaa4eec760485c564d'
+            '3583f23c55ccd46cefbd757ef3f82dc7a90f14c64ecf69a99ab3467ca1e6aeddf9822be4c4dffcdbb8841d79fe116cfb8eff0e9b44abaadbcbf8d50a10ab1ec9')
 
 prepare() {
   cd marisa-trie-$pkgver
+  patch -Np1 -i ../fix-format-security.patch
+
   autoreconf -i
 }
 
 build() {
   cd marisa-trie-$pkgver
   # sse2 is part of amd64
-  ./configure --prefix=/usr --disable-static --enable-sse2
+  ./configure --prefix=/usr --disable-static
+    # --enable-sse2
     # --enable-popcnt
     # --enable-sse3
     # --enable-ssse3
