diff --git PKGBUILD PKGBUILD
index 4c6617d..1eeb278 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,9 +10,11 @@
 license=('BSD' 'LGPL')
 makedepends=('python' 'ruby' 'perl' 'swig')
 source=("marisa-trie-$pkgver.tar.gz::https://github.com/s-yata/marisa-trie/archive/v$pkgver.tar.gz"
-        fix-format-security.patch)
+        fix-format-security.patch
+        riscv64-marisa_word_size.patch)
 sha512sums=('c094e4b22e1457efdd20f2b978ee421b53e36ed94e4fdbd8944136c0ba23da4f6ba9fe3a2c64729c1426aee4dbe8098bfa5eebb943ae7fdaa4eec760485c564d'
-            '3583f23c55ccd46cefbd757ef3f82dc7a90f14c64ecf69a99ab3467ca1e6aeddf9822be4c4dffcdbb8841d79fe116cfb8eff0e9b44abaadbcbf8d50a10ab1ec9')
+            '3583f23c55ccd46cefbd757ef3f82dc7a90f14c64ecf69a99ab3467ca1e6aeddf9822be4c4dffcdbb8841d79fe116cfb8eff0e9b44abaadbcbf8d50a10ab1ec9'
+            'f22e2feb3db75d64a6ac9a60574450bf69cf970027edd4827aab132bf1ed827bcc1c7bf37137f055ea25eb06945cbd04811b19818a459985375b9b8d4bef3a51')
 
 prepare() {
   cd marisa-trie-$pkgver
@@ -20,12 +22,15 @@
 
   # https://github.com/s-yata/marisa-trie/pull/45
   patch -Np1 -i ../fix-format-security.patch
+  # https://github.com/s-yata/marisa-trie/issues/40
+  patch -Np1 -i ../riscv64-marisa_word_size.patch
 }
 
 build() {
   cd marisa-trie-$pkgver
   # sse2 is part of amd64
-  ./configure --prefix=/usr --disable-static --enable-sse2
+  ./configure --prefix=/usr --disable-static
+    # --enable-sse2
     # --enable-popcnt
     # --enable-sse3
     # --enable-ssse3
