diff --git a/Cargo.toml b/Cargo.toml
index 2b38e3a..729d482 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,8 +31,8 @@ ocl = { version = "0.19", optional = true }
 "sha2" = { version = "0.10", "default-features" = false, features = ["compress"] }
 
 [target.'cfg(not(target_env = "msvc"))'.dependencies]
-"sha-1" = { version = "0.10", "default-features" = false, features = ["asm", "compress"] }
-"sha2" = { version = "0.10", "default-features" = false, features = ["asm", "compress"] }
+"sha-1" = { version = "0.10", "default-features" = false, features = ["compress"] }
+"sha2" = { version = "0.10", "default-features" = false, features = ["compress"] }
 
 [profile.release]
 lto = true
