diff --git PKGBUILD PKGBUILD
index d0ecb1f0..426a7f28 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,17 @@
 license=('MIT')
 depends=('gcc-libs' 'ocl-icd')
 makedepends=('cargo')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('8d459e3aadbb5f1adbdb2ab03b5c8c3d666f76b2e72168c13c822c1b88ad54f100a75c7cdd1e6094b93f9080d383242932ed5bf997887f3c652e16e4e494490e')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
+        "disable-asm-feature.patch")
+sha512sums=('8d459e3aadbb5f1adbdb2ab03b5c8c3d666f76b2e72168c13c822c1b88ad54f100a75c7cdd1e6094b93f9080d383242932ed5bf997887f3c652e16e4e494490e'
+            '4f0e9724566d3b9e89f52d2ba37cbf289c88092bd625fde331a22f58e3b645ac6c2a77915c5617436bce83aaa55c79fb752cd2b0cc2fe0ad34a09affff94c564')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  patch -Np1 < ${srcdir}/disable-asm-feature.patch
+  cargo update --package sha-1:0.10.0
+  cargo update --package sha2:0.10.2
+  cargo fetch --locked
 }
 
 build() {
