diff --git PKGBUILD PKGBUILD
index f511d325..ef31546b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,15 @@ license=('LGPL')
 depends=('boost-libs' 'cairo' 'freetype2' 'gdal' 'harfbuzz' 'icu' 'libjpeg-turbo' 'libpng'
          'libtiff' 'libwebp' 'libxml2' 'postgresql-libs' 'proj' 'sqlite' 'zlib')
 makedepends=('boost' 'scons')
-source=(https://github.com/$pkgname/$pkgname/releases/download/v$pkgver/$pkgname-v$pkgver.tar.bz2)
-sha256sums=('43d76182d2a975212b4ad11524c74e577576c11039fdab5286b828397d8e6261')
+source=(https://github.com/$pkgname/$pkgname/releases/download/v$pkgver/$pkgname-v$pkgver.tar.bz2
+        "remove_deprecated_call.patch")
+sha256sums=('43d76182d2a975212b4ad11524c74e577576c11039fdab5286b828397d8e6261'
+            '8f3e7f2f9fdd91a066c07801bdf4b26eb499ed3f455a1c83c13f6ff1d4682d4c')
+
+prepare() {
+  cd "${srcdir}"/$pkgname-v$pkgver
+  patch -Np1 -i "../remove_deprecated_call.patch"
+}
 
 build() {
   cd "${srcdir}"/$pkgname-v$pkgver
