Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1329385)
+++ PKGBUILD	(working copy)
@@ -29,6 +29,7 @@
 
 prepare() {
     cd $pkgname-$pkgver
+    sed -i '/test-suite doctests/a \  buildable: False' $pkgname.cabal
     uusi -u aeson -u hspec -u HTTP -u jose -u lens -u lens-aeson -u optparse-applicative -u vector $pkgname.cabal
 
     # Hack LD_LIBRARY_PATH
