diff --git PKGBUILD PKGBUILD
index c1f4f18..ea3f6f3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -35,3 +35,11 @@
 	./build install man "$pkgdir"
 	./build install manual "$pkgdir"
 }
+
+source+=("$pkgname-fix-missing-stdbool.patch::https://gitlab.com/fbb-git/yodl/-/merge_requests/5.diff")
+sha512sums+=('95c60ca348d74ed181a51b09c4dca3b6b0219da6045fb4c6a9a1b74e505bb1c7504ced704d5def3a698e78d14bb5feb378ca361f2d7c0acb626e7a61d7f08b1a')
+
+prepare() {
+        cd "$_archive/$pkgname"
+        patch -Np2 -i "$srcdir/$pkgname-fix-missing-stdbool.patch"
+}
