diff --git PKGBUILD PKGBUILD
index cc612bd..e606ba4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,12 +25,12 @@ build() {
   export LINKFLAGS="$LDFLAGS"
   (
     cd ${pkgname[0]}-$pkgver
-    waf configure --prefix=/usr --project=mixer --disable-native-cpu-optimization
+    waf configure --prefix=/usr --project=mixer --disable-sse --disable-native-cpu-optimization
     waf build -vv
   )
   (
     cd ${pkgname[1]}-$pkgver
-    waf configure --prefix=/usr --project=timeline --disable-native-cpu-optimization
+    waf configure --prefix=/usr --project=timeline --disable-sse --disable-native-cpu-optimization
     waf build -vv
   )
 }
