diff --git PKGBUILD PKGBUILD
index 51a9f31f..a83afcf2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -24,12 +24,12 @@ prepare() {
 build() {
   (
     cd "${pkgname[0]}-${pkgver}"
-    waf configure --prefix=/usr --project=mixer
+    waf configure --prefix=/usr --project=mixer --disable-sse --disable-native-cpu-optimization
     waf build -vv
   )
   (
     cd "${pkgname[1]}-${pkgver}"
-    waf configure --prefix=/usr --project=timeline
+    waf configure --prefix=/usr --project=timeline --disable-sse --disable-native-cpu-optimization
     waf build -vv
   )
 }
