--- PKGBUILD
+++ PKGBUILD
@@ -122,7 +122,7 @@
 build() {
   cd ${pkgname}/build
   # -DCMAKE_POLICY_VERSION_MINIMUM=3.5 can be removed as soon as the project builds without
-  cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_POLICY_VERSION_MINIMUM=3.5
+  cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DCMAKE_CXX_FLAGS="$CXXFLAGS -Wno-error=stringop-overflow"
   make -j$(nproc)
 }
 
