diff --git PKGBUILD PKGBUILD
index 3a7b85ed7..8c963fd8c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,8 @@ source=(https://ftp.gnu.org/gnu/a2ps/${pkgname}-${pkgver}.tar.gz{,.sig}
 	CVE-2014-0466.patch
 	CVE-2015-8107.patch
 	a2ps-texinfo67.patch
-        a2ps-adhere-source-date-epoch.patch)
+        a2ps-adhere-source-date-epoch.patch
+        fix-format-not-a-string-literal-and-no-format-arguments.patch)
 sha1sums=('365abbbe4b7128bf70dad16d06e23c5701874852'
           'SKIP'
           '8783952d3410d8d59ed953e1db45e2ef1a0b8f65'
@@ -31,7 +32,8 @@ sha1sums=('365abbbe4b7128bf70dad16d06e23c5701874852'
           '0a04f4c145d3aef8f3c7537eb00cb4040c7209c9'
           '08c51a9361eaefe76af61b70392fa5ac1c4356c7'
           'f54b6b319a5c134e0ee9be8dc9c8f80a65dc98d6'
-          '6f37363785adf4357977de3d595d7938d63f0993')
+          '6f37363785adf4357977de3d595d7938d63f0993'
+          '09c42d36ba568d0958c9728f9f7621e623680a5c')
 validpgpkeys=('6EB39358D8328FE3CDC903A8CB6FA340E7075A54')
 
 prepare() {
@@ -51,6 +53,7 @@ prepare() {
   patch -p1 -i "${srcdir}/CVE-2015-8107.patch"
   patch -p1 -i "${srcdir}/a2ps-texinfo67.patch"
   patch -p1 -i "${srcdir}/a2ps-adhere-source-date-epoch.patch"
+  patch -p1 -i "${srcdir}/fix-format-not-a-string-literal-and-no-format-arguments.patch"
 }
 
 build() {
