--- configure.ac	2023-01-15 10:39:22.068662643 +0800
+++ configure.ac	2023-01-15 10:40:31.900283925 +0800
@@ -112,11 +112,6 @@ AC_PROG_EGREP
 
 AC_CHECK_HEADERS([strings.h libgen.h unistd.h direct.h sys/stat.h])
 
-# REQUIRE standard C headers
-if test "$ac_cv_header_stdc" != yes; then
-	AC_MSG_ERROR([Standard (ANSI/ISO C89) header files are required.])
-fi
-
 #
 # Checks for typedefs, structures, and compiler characteristics.
 #
