Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1014381)
+++ PKGBUILD	(working copy)
@@ -29,9 +29,10 @@
 b2sums=(SKIP)
 
 prepare() {
+  echo -e "\n[patch.crates-io]\nzbus = { git = 'https://gitlab.freedesktop.org/felixonmars/zbus', branch = '1.9.1' }" >> starship/Cargo.toml
+  cargo update -p zbus --manifest-path starship/Cargo.toml
   cargo fetch \
     --locked \
-    --target $CARCH-unknown-linux-gnu \
     --manifest-path starship/Cargo.toml
 }
 
