Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010008)
+++ PKGBUILD	(working copy)
@@ -11,13 +11,16 @@
 groups=('fcitx5-im')
 depends=('fcitx5')
 makedepends=('extra-cmake-modules' 'gobject-introspection' 'gtk2' 'gtk3' 'gtk4' 'ninja' 'vulkan-headers')
-source=("https://download.fcitx-im.org/fcitx5/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig})
+source=("https://download.fcitx-im.org/fcitx5/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig}
+        https://github.com/fcitx/fcitx5-gtk/commit/54995987451b01d3a47e50edcd4f3339c18eb49c.patch)
 sha512sums=('28226f95eec9cd19d9c51845ecb3a24e813631a0918292bfec01d605605f9e810a46bb42ad96f6e678944993788f995540bf86e87779df66fb7dac44dc42ed16'
-            'SKIP')
+            'SKIP'
+            '78023dc4a950cfd26eb5410e736dbd3378db9294cd1477837601f799c76ace4da915d192d69cb29740badfe9248fd419a8bf99b6379f10c33c51edb2ab52b34a')
 validpgpkeys=('2CC8A0609AD2A479C65B6D5C8E8B898CBF2412F9') # Weng Xuetian <wengxt@gmail.com>
 
 build(){
   cd $pkgname-$pkgver
+  patch -p1 -i ../54995987451b01d3a47e50edcd4f3339c18eb49c.patch
 
   cmake -GNinja -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib .
   ninja
