--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ license=('MIT')
 url="https://attrs.readthedocs.org/"
 depends=('python')
 makedepends=('python-setuptools')
-checkdepends=('python-pytest' 'python-cloudpickle' 'python-hypothesis' 'python-zope-interface')
+checkdepends=('python-pytest' 'python-cloudpickle' 'python-hypothesis' 'python-zope-interface' 'python-six')
 source=("https://pypi.io/packages/source/a/attrs/attrs-$pkgver.tar.gz")
 sha512sums=('5a52776eb505c8a354912fa037a9710e49830fa6ad37272cb6d8c3298f4179a5b237e36cd7db3b1c9af19fbf0194ad6b83c3db9a473f9349769180073273ba65')
 
