--- PKGBUILD
+++ PKGBUILD
@@ -6,9 +6,10 @@
 
 pkgname=pycharm-community-edition
 pkgver=2024.3.3
+_tag=${pkgver}+0
 _build=243.24978.54
-_jrever=17
-_jdkver=17
+_jrever=21
+_jdkver=21
 pkgrel=1
 pkgdesc='Python IDE for Professional Developers'
 arch=(x86_64)
@@ -32,42 +33,36 @@ makedepends=(
   git
   "java-environment-openjdk=${_jdkver}"
   python-setuptools
+  gcc
+  cargo
+  go
 )
 source=(
-  "git+https://github.com/JetBrains/intellij-community.git#tag=pycharm/${_build}"
-  idea-android::"git+https://github.com/JetBrains/android.git#tag=pycharm/${_build}"
+  "git+https://github.com/Glavo/JetBrains-IDE-Multiarch.git#tag=idea/${_tag}"
   pycharm.desktop
   pycharm.sh
   # enable-no-jdr.patch
 )
-sha256sums=('e7d762cae300a4849b39f0f770922d93280280752cdaa17df4747df90d059870'
-            'e1fabad30c867c3b9517cce7e9121b0c3dd5082a17ab606ddaae53833a64248f'
+sha256sums=('7ef194687767ebebd59783bcb94ece91c0682c25a5a5eac26d2bb2c822882f97'
             '70ae8df10243db2dbc454c0d8e273235707ecb1821a9e8a2e70669a1739fc1af'
             '447714bf41a215b38f8c31418e09bf65145bb3d9427d6966772c9b2b8ae16f1d')
 
 prepare() {
-  cd intellij-community
+  cd JetBrains-IDE-Multiarch
 
-  sed -e "s/buildNumber = null/buildNumber = \"${_build}\"/" \
-      -i "python/build/src/PyCharmCommunityInstallersBuildTarget.kt"
-  mv "${srcdir}"/idea-android android
-
-  sed -e 's/System.getProperty(TARGET_OS_PROPERTY, OS_ALL)/OS_LINUX/' \
-      -i "platform/build-scripts/src/org/jetbrains/intellij/build/BuildOptions.kt"
-  # Disabled until we package JBR outselves
-  #patch -Np1 < "${srcdir}/enable-no-jdr.patch"
+  # Disable fetch prebuild natives
+  sed -i 's/^natives\.linux\.riscv64\.url.*$//g' config/natives.properties
 }
 
 build() {
-  cd intellij-community
+  cd JetBrains-IDE-Multiarch
 
   export JAVA_HOME="/usr/lib/jvm/java-${_jdkver}-openjdk"
   export PATH="/usr/lib/jvm/java-${_jdkver}-openjdk/bin:$PATH"
-  export MAVEN_REPOSITORY=/build/.m2/repository
 
-  ./python/installers.cmd -Dintellij.build.use.compiled.classes=false -Dintellij.build.target.os=linux -Dbuild.number="${_build}"
+  ./gradlew transformPC-riscv64
 
-  tar -xf out/pycharm-ce/artifacts/pycharmPC-${_build}.tar.gz -C "${srcdir}"
+  tar -xf build/target/pycharm-community-${_tag}-riscv64.tar.gz -C "${srcdir}"
 }
 
 package() {
