diff --git PKGBUILD PKGBUILD
index 9084df4..a0ebfb0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@
   glibc
   nodejs
 )
-makedepends=(npm)
+makedepends=(npm python)
 source=("https://registry.npmjs.org/@google/$pkgname/-/$pkgname-$pkgver.tgz")
 noextract=("$pkgname-$pkgver.tgz")
 sha512sums=('a04aa6ab752e3ed0b9b6e017f22b890e35e9be637bdc78dbd2e7f32fff97a9bdc7f958f4524dccc4328239edc9968ab7b83912d97ae9d345d1c19f0e732e172d')
