diff --git PKGBUILD PKGBUILD
index 2ae2181..56a8712 100755
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,15 @@
 optdepends=('blas-openblas: faster linear algebra')
 makedepends=('python-build' 'python-installer' 'meson-python' 'cmake' 'gcc-fortran' 'cython')
 checkdepends=('python-pytest' 'python-hypothesis')
-source=("https://github.com/numpy/numpy/releases/download/v$pkgver/numpy-$pkgver.tar.gz")
-sha512sums=('0d500c623b274a219740c78ae2febb32a2f167016a9ff529678526e6b3e89a5b732c41defa23460a5da6f7f89d4a7d827f44fa9a1334c78e204b00ce164fb40c')
+source=("https://github.com/numpy/numpy/releases/download/v$pkgver/numpy-$pkgver.tar.gz"
+        "Fix-build-for-host-flags.patch::https://github.com/numpy/numpy/commit/38edef779f9cfa353a0f3ceb598895810ce405e6.diff")
+sha512sums=('0d500c623b274a219740c78ae2febb32a2f167016a9ff529678526e6b3e89a5b732c41defa23460a5da6f7f89d4a7d827f44fa9a1334c78e204b00ce164fb40c'
+            'afb1ff5af90302920ae0ad68848cc40867edc2c2644ff8b78b6a96326fd1a3715918167a8ac44feaf7100178232c5fc6165fe0dd26be522f3a4e1f10ab71552e')
+
+prepare() {
+  cd numpy-$pkgver
+  patch -Np1 -i ../Fix-build-for-host-flags.patch
+}
 
 build() {
   cd numpy-$pkgver
