--- PKGBUILD
+++ PKGBUILD
@@ -43,6 +43,7 @@ source=(
   git+https://github.com/KhronosGroup/SPIRV-Cross.git
   armips-tinyformat::git+https://github.com/Kingcom/tinyformat.git
   libretro-ppsspp-assets-path.patch
+  $pkgname-fix-const-pointer-conversion-error-for-risc-v.patch::https://github.com/hrydgard/ppsspp/pull/16027.patch
 )
 b2sums=('SKIP'
         'SKIP'
@@ -53,7 +54,8 @@ b2sums=('SKIP'
         'SKIP'
         'SKIP'
         'SKIP'
-        'c72640695e3e3b9512f13d71956872cea34b41cf28bd31c7dbd8062e1166d5ae8f248b01d6f5c73dd96d3ed5d5d59eab69c5619272525aefaa322cc2ca8b8541')
+        'c72640695e3e3b9512f13d71956872cea34b41cf28bd31c7dbd8062e1166d5ae8f248b01d6f5c73dd96d3ed5d5d59eab69c5619272525aefaa322cc2ca8b8541'
+        '8ae39d01baa60b74403bc1720c27bc6c7fe584d3c636d4973f5b01c30d271f995d795f778cf7d6bb32d3d80c21934567cd476e92f9375f84fdc33ab5fa0e7bd0')
 
 pkgver() {
   cd libretro-ppsspp
@@ -77,6 +79,8 @@ prepare() {
     git submodule update ${submodule}
   done
 
+  patch -Np1 -i $srcdir/$pkgname-fix-const-pointer-conversion-error-for-risc-v.patch
+
   cd ext/armips
 
   for submodule in ext/tinyformat; do
