diff --git PKGBUILD PKGBUILD
index 5c03c2e61..4e6d4cf57 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ sha512sums=('f851ceccb7aa1ee00560c64fca2cf4239f0b260c7c2eee5c3688a555f7f37ff7bc4
 build() {
 	cd "$pkgname-$pkgver"
 
-	DMD=ldmd2 ldmd2 -run build.d -release -O3 -linker=gold --flto=thin -link-defaultlib-shared=false -L--build-id
+	DMD=ldmd2 ldmd2 -run build.d -release -O3 -link-defaultlib-shared=false -L--build-id
 }
 
 package() {
