--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ sha512sums=('476d59bc56eedaff304a060b8dd798908f579b044f74a891c01ff7e1f13294c71e2
 build() {
 	cd "$pkgname-$pkgver"
 
-	GITVER=$pkgver DMD=ldmd2 ldmd2 -run build.d -release -O3 -linker=gold --flto=thin -link-defaultlib-shared=false -L--build-id
+	GITVER=$pkgver DMD=ldmd2 ldmd2 -run build.d -release -O3 -link-defaultlib-shared=false -L--build-id
 
 	# Build man pages
 	bin/dub --single scripts/man/gen_man.d
