--- PKGBUILD
+++ PKGBUILD
@@ -49,4 +49,13 @@ package() {
   install -Dm644 LICENSE "${pkgdir}"/opt/pypy3/LICENSE
   install -Dm644 LICENSE "${pkgdir}"/usr/share/licenses/pypy3/LICENSE
 }
+
+source+=('missing-riscv64-macro-defined.patch')
+sha512sums+=('5f3638ae94090595c7eb4b18dfba905a9e925150d814d6b520d9826edc4af9c8095a9ed19702a2ab2fd78dc9a2aac2e9f48f7e16fc7f4556c6fcb03827925262')
+
+prepare() {
+  pushd ${srcdir}/pypy${_pyversion}-v${pkgver}-src/
+  patch -Np1 < ${srcdir}/missing-riscv64-macro-defined.patch
+  popd
+}
 # vim: ts=2 sw=2 et:
