diff --git PKGBUILD PKGBUILD
index cd79c4808..c3c156965 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,7 @@ sha256sums=('9c75f193969a6654d1d4276b6f39cf6ecaef4fa2ba2b42ac162957c01eb2255e'
             '9e595176e63b301476982b1456d6ed065c479ff913b6743417ab8a9efdda0e3a'
             '43f07093ada8eb5c954f9e2e189610575906c690c0953fec041d4f2d31210d85'
             '1b9b3d5560f6633d4f31a521bc0ae17094d3afa28e703b4790eaefedcc9a92a8')
+options=(!lto)
 
 prepare() {
   cd "$srcdir"/unrealircd-$pkgver
