Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422631)
+++ PKGBUILD	(working copy)
@@ -7,12 +7,14 @@
 arch=(x86_64)
 license=(LGPL)
 url="https://www.freedesktop.org/wiki/Software/polkit/"
-depends=(glib2 pam expat systemd js78)
+depends=(glib2 pam expat systemd duktape)
 makedepends=(intltool gtk-doc gobject-introspection git autoconf-archive)
 backup=(etc/pam.d/polkit-1)
 _commit=2e5348bf4eb0ef984db32f7f96ec6722d441c6ca
-source=("git+https://gitlab.freedesktop.org/polkit/polkit.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.freedesktop.org/polkit/polkit.git#commit=$_commit"
+        polkit-duktape.patch::https://gitlab.freedesktop.org/polkit/polkit/-/merge_requests/35.patch)
+sha256sums=('SKIP'
+            'e15490209616c0e538611f27545fd169019c0973e6acf31f8bce0515068351e6')
 
 pkgver() {
   cd $pkgname
@@ -21,6 +23,7 @@
 
 prepare() {
   cd $pkgname
+  patch -p1 -i ../polkit-duktape.patch
   NOCONFIGURE=1 ./autogen.sh
 }
 
@@ -30,7 +33,7 @@
   ./configure --prefix=/usr --sysconfdir=/etc \
       --localstatedir=/var --libexecdir=/usr/lib \
       --enable-libsystemd-login=yes --disable-static \
-      --enable-gtk-doc --with-os-type=redhat
+      --enable-gtk-doc --with-os-type=redhat --with-duktape
 
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
 
