diff --git PKGBUILD PKGBUILD
index b84308d..2fb121d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,7 @@
         -D LLVM_ENABLE_RUNTIMES='compiler-rt;libunwind;libcxx;libcxxabi'
         -D LIBCXX_ENABLE_STATIC=ON
         -D LIBCXXABI_ENABLE_STATIC=ON
-        -D LLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;X86'
+        -D LLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;RISCV'
         -D CLANG_DEFAULT_LINKER=lld
         -D CLANG_DEFAULT_RTLIB=compiler-rt
         -D CLANG_DEFAULT_UNWINDLIB=libgcc
@@ -132,3 +132,9 @@
     cd "$pkgbase/amd/comgr"
     install -Dm644 LICENSE.txt "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+prepare() {
+  cd "$pkgbase"
+  # Fix llvm-mc abi
+  git cherry-pick -n 3acb0e9e600cbe3668b7db3956238a592ebadc0a
+}
