diff --git PKGBUILD PKGBUILD
index 3422600..49e34be 100644
--- PKGBUILD
+++ PKGBUILD
@@ -44,6 +44,9 @@ prepare() {
 
   # Fix build with glibc 2.42: https://github.com/llvm/llvm-project/issues/137321
   patch -Np1 < ../rocm-llvm-6.4-fix-glibc-2.42-termio-removal.patch
+
+  # Fix libraries_sha.inc:1:42: error: narrowing conversion of ‘-30’ from ‘int’ to ‘char’
+  git cherry-pick -n 74c3df6cb5540c937ad8b58f95275901efae8669
 }
 
 build() {
@@ -73,7 +76,7 @@ build() {
         -D LIBCXXABI_ENABLE_SHARED=OFF
         -D LIBCXXABI_ENABLE_STATIC=ON
         -D LIBCXXABI_INSTALL_STATIC_LIBRARY=OFF
-        -D LLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;X86'
+        -D LLVM_TARGETS_TO_BUILD='AMDGPU;NVPTX;RISCV'
         -D CLANG_DEFAULT_LINKER=lld
         -D ENABLE_LINKER_BUILD_ID=ON
         -D CLANG_DEFAULT_RTLIB=compiler-rt
