--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,7 @@ sha256sums=('11185e845b5a52813b9a7610505384813eb4ad3cfe3c3e7aee5ce10d75b91468')
 
 build() {
   arch-meson $pkgname build \
+    -Dxnvctrl=disabled \
     -Dhddtemp=enabled \
     -Dnetcat=enabled \
     --localstatedir=/var
@@ -29,4 +30,5 @@ package() {
   meson install -C build --destdir "$pkgdir"
 }
 
+depends=(${depends[@]/libxnvctrl})
 # vim:set ts=2 sw=2 et:
