--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,8 @@ sha256sums=('11185e845b5a52813b9a7610505384813eb4ad3cfe3c3e7aee5ce10d75b91468')
 
 build() {
   arch-meson $pkgname build \
-    --localstatedir=/var
+    --localstatedir=/var \
+    -Dxnvctrl=disabled
   meson compile -C build
 }
 
@@ -27,4 +28,6 @@ package() {
   meson install -C build --destdir "$pkgdir"
 }
 
+depends=(${depends[@]/libxnvctrl})
+
 # vim:set ts=2 sw=2 et:
