Index: qt-everywhere-src-6.6.1/qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni
===================================================================
--- qt-everywhere-src-6.6.1.orig/qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni
+++ qt-everywhere-src-6.6.1/qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni
@@ -14,7 +14,7 @@ if (is_apple) {
 if (is_nacl) {
   # NaCl targets don't use 64-bit pointers.
   has_64_bit_pointers = false
-} else if (current_cpu == "x64" || current_cpu == "arm64") {
+} else if (current_cpu == "x64" || current_cpu == "arm64" || current_cpu == "riscv64") {
   has_64_bit_pointers = true
 } else if (current_cpu == "x86" || current_cpu == "arm") {
   has_64_bit_pointers = false
Index: qt-everywhere-src-6.6.1/qtwebengine/src/3rdparty/chromium/third_party/angle/gni/angle.gni
===================================================================
--- qt-everywhere-src-6.6.1.orig/qtwebengine/src/3rdparty/chromium/third_party/angle/gni/angle.gni
+++ qt-everywhere-src-6.6.1/qtwebengine/src/3rdparty/chromium/third_party/angle/gni/angle.gni
@@ -104,7 +104,7 @@ declare_args() {
 
   if (current_cpu == "arm64" || current_cpu == "x64" ||
       current_cpu == "mips64el" || current_cpu == "s390x" ||
-      current_cpu == "ppc64" || current_cpu == "loong64") {
+      current_cpu == "ppc64" || current_cpu == "loong64" || current_cpu == "riscv64") {
     angle_64bit_current_cpu = true
   } else if (current_cpu == "arm" || current_cpu == "x86" ||
              current_cpu == "mipsel" || current_cpu == "s390" ||
