diff --git PKGBUILD PKGBUILD
index 4facc44..3921932 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,8 +33,15 @@ makedepends=(alsa-lib
              vulkan-headers)
 groups=(qt6)
 _pkgfn=qt-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/single/$_pkgfn.tar.xz)
-sha256sums=('3c1e42b3073ade1f7adbf06863c01e2c59521b7cc2349df2f74ecd7ebfcb922d')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/single/$_pkgfn.tar.xz
+        add-riscv64-support.patch)
+sha256sums=('3c1e42b3073ade1f7adbf06863c01e2c59521b7cc2349df2f74ecd7ebfcb922d'
+            '0a1a003334636bedaa5edeefe1cc7eb2ce4f964cee65f16e0956667a7f22f08f')
+
+prepare() {
+  cd $_pkgfn
+  patch -Np1 -i ../add-riscv64-support.patch
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
