diff --git PKGBUILD PKGBUILD
index c0f5f8e5b..ceb42459b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,10 +13,12 @@ options=('!makeflags')
 source=("https://downloads.sourceforge.net/sourceforge/${pkgname}/${pkgname}-${pkgver}.tar.xz"
         '0001-do-not-leave-stale-oplock-behind.patch'
         '0002-send-ConfigureNotify-using-root-coordinates.patch'
+        '0003-fix-ordered-comparison-of-pointer-with-integer-zero.patch'
         'fluxbox.desktop')
 sha256sums=('fc8c75fe94c54ed5a5dd3fd4a752109f8949d6df67a48e5b11a261403c382ec0'
             '9c486f454f46a51cb1b7cb65de812faa12e2a1cce13785c349b488f051a01470'
             '66f64c67e5c99b2c6bf139ba2f7a3ff9fcb613c5b49b0038c42cc55cc00153dc'
+            '82b511e0232b6bd00c1a4e3965ff202c385b012d472fa6fbe1b7251e2fb0fbfe'
             '2148a90aca653b596e9632264dbdbd8c7e673d732e4b04eee66a8ac1e68b3c5d')
 
 prepare() {
@@ -24,6 +26,7 @@ prepare() {
 
   patch -Np1 < ../0001-do-not-leave-stale-oplock-behind.patch
   patch -Np1 < ../0002-send-ConfigureNotify-using-root-coordinates.patch
+  patch -Np1 < ../0003-fix-ordered-comparison-of-pointer-with-integer-zero.patch
 }
 
 build() {
