diff --git PKGBUILD PKGBUILD
index 988c1f3bf..9622cdb43 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ arch=('x86_64')
 url="http://simul.iro.umontreal.ca/testu01/tu01.html"
 license=('Apache')
 depends=('glibc')
-source=(http://simul.iro.umontreal.ca/$pkgname/$_pkgname.zip)
-sha256sums=('bc1d1dd2aea7ed3b3d28eaad2c8ee55913f11ce67aec8fe4f643c1c0d2ed1cac')
+source=(http://simul.iro.umontreal.ca/$pkgname/$_pkgname.zip
+        "fix_not_literal_string.patch")
+sha256sums=('bc1d1dd2aea7ed3b3d28eaad2c8ee55913f11ce67aec8fe4f643c1c0d2ed1cac'
+            'd39e618fc2b0469d22340361392029c641e27634e8b66c60a660cd5f86ea43ab')
+
+prepare() {
+  cd $_pkgname-$pkgver
+  patch -Np1 -i "../fix_not_literal_string.patch"
+}
 
 build() {
   cd $_pkgname-$pkgver
