diff --git a/src/LibSupport.h b/src/LibSupport.h
index 5055560..4b41baa 100644
--- a/src/LibSupport.h
+++ b/src/LibSupport.h
@@ -71,7 +71,7 @@
     || defined(_WIN64) || defined(_M_X64)		 		\
     || defined(_M_ARM64) || defined(__aarch64__) 			\
     || defined(__powerpc64__) || defined(__PPC64__) || defined(__ppc64__) \
-    || defined(__s390x__)
+    || defined(__s390x__) || (defined(__riscv) && __riscv_xlen == 64)
 #       define RADIX_BITS                      64
 #   elif defined(__i386__) || defined(__i386) || defined(i386)		\
     || defined(_WIN32) || defined(_M_IX86)				\
