diff --git PKGBUILD PKGBUILD
index 782e1ef..1d06b42 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,8 +18,15 @@ makedepends=(
   git
 )
 _commit=79aaf908a602e03a16698976791c82baafa794d0
-source=(git+https://github.com/stenzek/duckstation.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(git+https://github.com/stenzek/duckstation.git#commit=${_commit}
+        "add_riscv64_support.patch")
+sha256sums=('SKIP'
+            '18a2f7647da74972a20c042b5c43658500e66c397ea2d5337f4c75ba58ea1722')
+
+prepare() {
+  cd duckstation
+  patch -Np1 -i "../add_riscv64_support.patch"
+}
 
 pkgver() {
   cd duckstation
