diff --git PKGBUILD PKGBUILD
index 98b372885..40d2bae9f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,6 +11,7 @@ depends=('bash' 'libpcap' 'openssl')
 makedepends=('python')
 source=(https://downloads.sourceforge.net/vde/$pkgname-$pkgver.tar.bz2
         vde_cryptcab-compile-against-openssl-1.1.0.patch
+        vde-compile-format.patch
         dhcpd.conf.sample
         iptables.rules.sample
         vde-config.sample
@@ -19,6 +20,7 @@ install=vde2.install
 options=(!makeflags)
 sha256sums=('cbea9b7e03097f87a6b5e98b07890d2275848f1fe4b9fcda77b8994148bc9542'
             '110370a5f48f1e241d43f8bb5e3ea6d2ca7d2c1949e1cf672d03bfc897f2e11f'
+            'f90242cf587b3bdd3fe59645907efde19ec99d6094fe3821c850debba2143994'
             'da0e2766dc63069da929c28126831ad5fdddcc4a04105a21217d78832c7ca1bc'
             '99076d7466cd99673dbe91ef83865187e7868177959b38e125df63eea957f83e'
             '5727c215646333c37b26388146cd3e6b3814b88d60d54051d7da99e00c0aef87'
@@ -27,6 +29,7 @@ sha256sums=('cbea9b7e03097f87a6b5e98b07890d2275848f1fe4b9fcda77b8994148bc9542'
 prepare() {
   cd $pkgname-$pkgver
   patch -Np1 -i ../vde_cryptcab-compile-against-openssl-1.1.0.patch
+  patch -Np1 -i ../vde-compile-format.patch
 }
 
 build() {
