diff --unified --recursive --text --strip-trailing-cr a/src/common/cmdparse.c b/src/common/cmdparse.c
--- a/src/common/cmdparse.c	2021-09-01 12:10:59.000000000 +0000
+++ b/src/common/cmdparse.c	2021-09-01 12:19:27.578945100 +0000
@@ -284,13 +284,13 @@
 								int i;
 								for (i=0;i<argc;i++) {
 									if (i) fprintf(mf," ");
-									fprintf(mf,argv[i]);
+									fprintf(mf,"%s",argv[i]);
 								}
 							} else {
 								int num=atoi(t);
 								while (*t >='0' && *t <= '9') t++;
 								if (num < argc) 
-									fprintf(mf,argv[num]);
+									fprintf(mf,"%s",argv[num]);
 							}
 						} else
 							fprintf(mf,"%c",*t);
diff --unified --recursive --text --strip-trailing-cr a/src/vde_switch/plugins/iplog.c b/src/vde_switch/plugins/iplog.c
--- a/src/vde_switch/plugins/iplog.c	2021-09-01 12:10:59.000000000 +0000
+++ b/src/vde_switch/plugins/iplog.c	2021-09-01 12:14:59.922132238 +0000
@@ -214,7 +214,7 @@
 				ctime_r(&ntime,stime);
 				writev(logfilefd,iov,3);
 			} else if (logfilefd != -1) 
-				syslog(LOG_INFO, msg);
+				syslog(LOG_INFO, "%s", msg);
 			DBGOUT(D_LOGIP_NEWIP,"%s",msg);
 		}
 	}
