diff --git PKGBUILD PKGBUILD
index cfe20b2b..7cdbe982 100644
--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,7 @@ b2sums=('240d102b5ca7b554e96af8ec927dd88804d6cb4f6a7eb22b5edf4231793747963d80583
 prepare() {
   # force LDFLAGS into all the libs to achieve full RELRO
   patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-0.2.8-full_relro.patch
+  cp -v "/usr/share/autoconf/build-aux/config.guess" "/usr/share/autoconf/build-aux/config.sub" $pkgname-$pkgver
 }
 
 build() {
