--- PKGBUILD
+++ PKGBUILD
@@ -37,14 +37,16 @@ prepare() {
   sed -i 's|/opt/schily|/usr|g' DEFAULTS/Defaults.linux
   sed -i 's|DEFINSGRP=.*|DEFINSGRP=root|' DEFAULTS/Defaults.linux
   sed -i 's|INSDIR=\s*sbin|INSDIR=bin|' rscsi/Makefile
+
+  ln -s i586-linux-cc.rul RULES/riscv64-linux-cc.rul
 }
 
 build() {
   cd $pkgname-$_pkgver
 
   export GMAKE_NOWARN=true
-  export CFLAGS="$CFLAGS -std=gnu89 -fno-strict-aliasing"
-  make INS_BASE=/usr INS_RBASE=/ VERSION_OS="_ArchLinux" LDOPTX="$LDFLAGS"
+  export CFLAGS="$CFLAGS -std=gnu89 -fno-strict-aliasing -Wno-implicit-int"
+  make INS_BASE=/usr INS_RBASE=/ VERSION_OS="_ArchLinux" LDOPTX="$LDFLAGS" COPTX="$CFLAGS"
 }
 
 check() {
