Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010493)
+++ PKGBUILD	(working copy)
@@ -22,7 +22,7 @@
   sed -i 's|DEFINSGRP=.*|DEFINSGRP=root|' DEFAULTS/Defaults.linux
   # install rscsi to /usr/bin instead of /usr/sbin
   sed -i 's|INSDIR=.*|INSDIR=bin|' rscsi/Makefile
-
+  ln -s i586-linux-cc.rul RULES/riscv64-linux-cc.rul
   export GMAKE_NOWARN=true
   make INS_BASE=/usr INS_RBASE=/ VERSION_OS="_ArchLinux" LDOPTX="$LDFLAGS"
 }
