--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,19 @@ license=('custom:NCSAOSL')
 depends=('hsa-rocr' 'python')
 makedepends=('cmake')
 _git='https://github.com/RadeonOpenCompute/rocm_smi_lib'
-source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz")
-sha256sums=('4d79cb0482b2f801cc7824172743e3dd2b44b9f6784d1ca2e5067f2fbb4ef803')
+source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
+        rocm-smi-lib_fix-cxx-flags.patch)
+sha256sums=('4d79cb0482b2f801cc7824172743e3dd2b44b9f6784d1ca2e5067f2fbb4ef803'
+            '98517844cbf9c0237b37b0bcedb2d29362d4307606d78898e9a19610ce923942')
 options=(!lto)
 _dirname="$(basename "$_git")-$(basename "${source[0]}" .tar.gz)"
 
+prepare() {
+  # remove '-m64 -msse -msse2' c++ flags on architectures that don't support it.
+  # backport from https://github.com/RadeonOpenCompute/rocm_smi_lib/commit/1cf05dd9c704f0fad0f535330ee1955914a4fe86
+  patch -d "$_dirname" -Np1 -i ../rocm-smi-lib_fix-cxx-flags.patch
+}
+
 build() {
   cmake \
     -Wno-dev \
