diff --git PKGBUILD PKGBUILD
index 151e307..6c146bd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,8 +20,19 @@ groups=(coin-or)
 source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/CoinUtils/archive/refs/tags/releases/$pkgver.tar.gz)
 sha256sums=('80c7c215262df8d6bd2ba171617c5df844445871e9891ec6372df12ccbe5bcfd')
 
+source+=("https://github.com/coin-or/CoinUtils/raw/aae9b0b807a920c41d7782d7bf2775afb17a12c6/config.guess"
+         "https://github.com/coin-or/CoinUtils/raw/aae9b0b807a920c41d7782d7bf2775afb17a12c6/config.sub")
+sha256sums+=('af8a1922c9b3c240bf2119d4ec0965a0b5ec36b1016017ba66db44b3b53e9cea'
+             'd611751fba98e807c9684d253bb02aa73d6825fe0e0b9ae3cbf258a59171c9b0')
+
 build() {
   cd CoinUtils-releases-$pkgver
+
+  cp -f ${srcdir}/config.guess ./CoinUtils/config.guess
+  cp -f ${srcdir}/config.guess ./config.guess
+  cp -f ${srcdir}/config.sub ./CoinUtils/config.sub
+  cp -f ${srcdir}/config.sub ./config.sub 
+
   ./configure \
     --prefix=/usr \
     --with-blas-lib='-lblas' \
