diff --git a/node_modules/app-builder-lib/out/linuxPackager.js b/node_modules/app-builder-lib/out/linuxPackager.js
index 4f179fc..13f0e6c 100644
--- a/node_modules/app-builder-lib/out/linuxPackager.js
+++ b/node_modules/app-builder-lib/out/linuxPackager.js
@@ -68,6 +68,8 @@ function toAppImageOrSnapArch(arch) {
             return "arm";
         case builder_util_1.Arch.arm64:
             return "arm_aarch64";
+        case builder_util_1.Arch.riscv64:
+            return "riscv64";
         default:
             throw new Error(`Unsupported arch ${arch}`);
     }
