diff --git a/apps/desktop/desktop_native/index.js b/apps/desktop/desktop_native/index.js
index 1cf7ea943..9c748d4b6 100644
--- a/apps/desktop/desktop_native/index.js
+++ b/apps/desktop/desktop_native/index.js
@@ -149,6 +149,20 @@ switch (platform) {
     break
   case 'linux':
     switch (arch) {
+      case 'riscv64':
+        localFileExisted = existsSync(
+          join(__dirname, 'desktop_native.linux-riscv64-gnu.node')
+        )
+        try {
+          if (localFileExisted) {
+            nativeBinding = require('./desktop_native.linux-riscv64-gnu.node')
+          } else {
+            nativeBinding = require('@bitwarden/desktop-native-linux-riscv64-gnu')
+          }
+        } catch (e) {
+          loadError = e
+        }
+        break
       case 'x64':
         localFileExisted = existsSync(
           join(__dirname, 'desktop_native.linux-x64-musl.node')
