diff --git a/node_modules/electron-builder/out/builder.js b/node_modules/electron-builder/out/builder.js
index 1e1490e..8c4e10a 100644
--- a/node_modules/electron-builder/out/builder.js
+++ b/node_modules/electron-builder/out/builder.js
@@ -20,6 +20,9 @@ function normalizeOptions(args) {
     function processTargets(platform, types) {
         function commonArch(currentIfNotSpecified) {
             const result = Array();
+            if (args.riscv64) {
+                result.push(builder_util_1.Arch.riscv64);
+            }
             if (args.x64) {
                 result.push(builder_util_1.Arch.x64);
             }
@@ -94,6 +97,7 @@ function normalizeOptions(args) {
     delete r.p;
     delete r.pd;
     delete result.ia32;
+    delete result.riscv64;
     delete result.x64;
     delete result.armv7l;
     delete result.arm64;
@@ -206,6 +210,11 @@ function configureBuildCommand(yargs) {
         alias: ["w", "windows"],
         description: `Build for Windows, accepts target list (see ${chalk.underline("https://goo.gl/jYsTEJ")})`,
         type: "array",
+    })
+        .option("riscv64", {
+        group: buildGroup,
+        description: "Build for riscv64",
+        type: "boolean",
     })
         .option("x64", {
         group: buildGroup,
