--- PKGBUILD
+++ PKGBUILD
@@ -10,19 +10,32 @@ arch=('x86_64')
 url='https://github.com/bitwarden/clients/tree/master/apps/desktop'
 license=('GPL3')
 depends=("electron$_electronversion" 'libnotify' 'libsecret' 'libxtst' 'libxss' 'libnss_nis')
-makedepends=('git' 'npm' 'python' 'node-gyp' 'nodejs-lts-hydrogen' 'jq' 'rust')
+makedepends=('git' 'npm' 'python' 'node-gyp' 'nodejs-lts-hydrogen' 'jq' 'rust' 'go' 'p7zip')
 source=(bitwarden::git+https://github.com/bitwarden/clients.git#tag=desktop-v$pkgver
         messaging.main.ts.patch
         nativelib.patch
         ${pkgname}.sh
-        ${pkgname}.desktop)
+        ${pkgname}.desktop
+	bitwarden-napi-riscv64.patch
+	builder-util+23.6.0.patch
+	app-builder-lib+23.6.0.patch
+	electron-builder+23.6.0.patch
+	git+https://github.com/develar/app-builder.git#commit=c92c3a2899b5887662321878a0a8681d122742bb
+)
 sha512sums=('SKIP'
             'babcae0dba4d036e5d2cd04d8932b63253bc7b27b14d090932066e9d39383f7565c06d72dae9f96e741b494ef7e50a1fe7ec33905aa3124b427a8bf404df5762'
             '88610cba9dea99aefdfea51139f5770f04f1e877d75e86f2eea3470c99880282c5ff91060cb08d92cdf00d0a1b3bd40c5f3ee887cee11946dd31ca06da978272'
             '98d2860bef2283fd09710fbbc5362d7ef2cd8eca26f35805ea258f2dacba78bd6aab14c834388a5089a8150eb0f32a82577aab10f8ad68e1a6371959b2802ad4'
-            'fdc047aadc1cb947209d7ae999d8a7f5f10ae46bf71687080bc13bc3082cc5166bbbe88c8f506b78adb5d772f5366ec671b04a2f761e7d7f249ebe5726681e30')
+            'fdc047aadc1cb947209d7ae999d8a7f5f10ae46bf71687080bc13bc3082cc5166bbbe88c8f506b78adb5d772f5366ec671b04a2f761e7d7f249ebe5726681e30'
+            '4087cd10bbaad8c44917eba6a74ea26ad9d38a3c5f6ad920cb6804e4526e5e66f75c71eab60ba48997daf2f1e199b2a170c070d900cba599e4947eb48474da0c'
+            'a7d87843bc31b94ebe47af10c7dbf0d7ed2b6507cc39f4140363d37d549eea940f4d9d5bb71b1a836751a37d20931faa80631aec511d5c1329803339c35a1b4c'
+            '2d76b117eba66fad8053270e18b75790aab09ba73ef03d48bce493ae2ac2d14bf229a6da888bbcd541408175635b3a3c1528bb4622d5aaf32a6d4b285cec89f0'
+            '6dc694e0c37c126419838622560380a5e5195c49e687ee5398305bd3c0b237c0d335ff97cfa10c3a65b3683d81ad8c56e350f67b01f679d17b7347dbdb2f46bf'
+            'SKIP')
 
 prepare() {
+	cp "${srcdir}"/{builder-util+23.6.0.patch,app-builder-lib+23.6.0.patch,electron-builder+23.6.0.patch} bitwarden/patches/
+	patch -Np1 -d bitwarden < bitwarden-napi-riscv64.patch
 	cd bitwarden/apps/desktop
 
 	export npm_config_build_from_source=true
@@ -43,6 +56,12 @@ prepare() {
 	cd ../../
 	patch --strip=1 apps/desktop/desktop_native/index.js "$srcdir/nativelib.patch"
 	npm ci
+	local _builder_bin=node_modules/app-builder-bin/linux/riscv64
+	mkdir "$_builder_bin"
+	go build -C ../app-builder
+	cp ../app-builder/app-builder "$_builder_bin"
+	mkdir -p  node_modules/7zip-bin/linux/riscv64
+	ln -s /usr/bin/7za node_modules/7zip-bin/linux/riscv64/7za
 }
 
 build() {
@@ -53,19 +72,20 @@ build() {
 	export npm_config_cache="$srcdir/npm_cache"
 	export ELECTRON_SKIP_BINARY_DOWNLOAD=1
 	pushd desktop_native/
-	npm run build
+	cargo build --release --locked
+	mv target/release/libdesktop_native.so desktop_native.linux-riscv64-gnu.node
 	popd
 	npm run build
 	npm run clean:dist 
-	npm exec -c "electron-builder --linux --x64 --dir -c.electronDist=$electronDist \
+	npm exec -c "electron-builder --linux --riscv64 --dir -c.electronDist=$electronDist \
 	             -c.electronVersion=$electronVer"
 }
 
 package(){
 	cd bitwarden/apps/desktop
-	install -vDm644 dist/linux-unpacked/resources/app.asar -t "${pkgdir}/usr/lib/${pkgname}"
+	install -vDm644 dist/linux-riscv64-unpacked/resources/app.asar -t "${pkgdir}/usr/lib/${pkgname}"
 	install -vDm644 build/package.json -t "${pkgdir}/usr/lib/${pkgname}"
-	cp -vr dist/linux-unpacked/resources/app.asar.unpacked -t "${pkgdir}/usr/lib/${pkgname}"
+	cp -vr dist/linux-riscv64-unpacked/resources/app.asar.unpacked -t "${pkgdir}/usr/lib/${pkgname}"
 
 	for i in 16 32 64 128 256 512 1024; do
 		install -vDm644 resources/icons/${i}x${i}.png "${pkgdir}/usr/share/icons/hicolor/${i}x${i}/apps/${pkgname}.png"
