diff --git a/VapourSynth/video_output.c b/VapourSynth/video_output.c
index 78bb595..c231162 100644
--- a/VapourSynth/video_output.c
+++ b/VapourSynth/video_output.c
@@ -39,7 +39,13 @@
 #include "video_output.h"
 #include <VSHelper.h>
 
+#ifdef __SSE2__
 #include <emmintrin.h>
+#else
+#define SIMDE_ENABLE_NATIVE_ALIASES 1
+#include <simde/x86/sse2.h>
+#undef SIMDE_ENABLE_NATIVE_ALIASES
+#endif
 
 typedef struct
 {
