--- PKGBUILD
+++ PKGBUILD
@@ -21,2 +21,7 @@
 
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i ../test-fixture-arch.patch
+}
+
 build() {
@@ -54 +59,5 @@
 # vim: ts=2 sw=2 et:
+
+source+=('test-fixture-arch.patch')
+sha256sums+=('c3432487cc099b63eee4864c18f96d5f7d8ac99ef14339d17b18007bf8bad174')
+b2sums+=('a4d95fd6520c0fdf928fb7d55bbbf231e9ef70a120a61efa002bd21793adfda2df2e2ecf763052ad9e0e7cd1815a142b953ffe2388cf7f8824848af3473804e2')
