Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 437720)
+++ PKGBUILD	(working copy)
@@ -11,7 +11,7 @@
 makedepends=(libasyncns libcap attr libxtst libsm libsndfile rtkit libsoxr
              speexdsp tdb systemd dbus avahi bluez bluez-libs jack2 sbc
              lirc openssl fftw orc gtk3 webrtc-audio-processing check git meson
-             xmltoman valgrind doxygen gst-plugins-bad)
+             xmltoman doxygen gst-plugins-bad)
 _commit=6329a2498eb038f8a9537888280a62b00a93f68e  # tags/v15.0^0
 source=("git+https://gitlab.freedesktop.org/pulseaudio/pulseaudio.git#commit=$_commit")
 sha256sums=('SKIP')
@@ -35,6 +35,7 @@
     -D udevrulesdir=/usr/lib/udev/rules.d
     -D elogind=disabled
     -D tcpwrap=disabled
+    -D valgrind=disabled
   )
 
   arch-meson pulseaudio build "${meson_options[@]}"
