diff --git PKGBUILD PKGBUILD
index 0348db1..3b4be1b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,8 +8,15 @@ url='https://github.com/slackhq/nebula'
 license=('MIT')
 depends=()
 makedepends=('go')
-source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/v${pkgver}.tar.gz")
-sha256sums=('e8d79231f6100a2cd240d6a092d0dcc2bfccadffa83cb40e99b7328f6c75c2ec')
+source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/v${pkgver}.tar.gz"
+        "fix-riscv-build-comments.patch::https://patch-diff.githubusercontent.com/raw/slackhq/nebula/pull/542.patch")
+sha256sums=('e8d79231f6100a2cd240d6a092d0dcc2bfccadffa83cb40e99b7328f6c75c2ec'
+            '804d981dd6e74b64c1cbe5305fcb7ad9bed785e1c9263afcd05ab3550a02b5ab')
+
+prepare(){
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 < ../fix-riscv-build-comments.patch
+}
 
 build() {
   cd "${pkgname}-${pkgver}"
