Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 423473)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 license=(LGPL2.1)
 arch=(x86_64)
 depends=(alsa-lib dbus)
-makedepends=(git meson)
+makedepends=(git meson systemd)
 provides=(libpipewire-${pkgver:0:3}.so)
 _commit=14c11c0fe4d366bad4cfecdee97b6652ff9ed63d  # tags/0.2.7
 source=("git+https://github.com/PipeWire/pipewire#commit=$_commit")
