--- PKGBUILD
+++ PKGBUILD
@@ -11,6 +11,11 @@ depends=('libtiff')
 source=(https://www.soft-switch.org/downloads/$pkgname/$pkgname-$pkgver.tar.gz)
 sha512sums=('16bb215ca89a39282e832403f69bc4c98ad3fe35ab3a6eb4731ee5029a6acd9b2df243c3701de845441cbdc16c88b3cd398ef15dc3502e45d5aeca8a161917d4')
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr
