--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ pkgdesc="cairocffi is a CFFI-based drop-in replacement for Pycairo, a set of Pyt
 arch=('any')
 url="https://cairocffi.readthedocs.io/"
 license=('BSD')
-depends=('python-cffi' 'cairo' 'python-xcffib')
+depends=('python-cffi' 'cairo')
 makedepends=('python-setuptools' 'gdk-pixbuf2' 'python-pytest-runner')
 checkdepends=('python-pytest-cov' 'python-pytest-flake8' 'python-pytest-isort'
               'python-numpy')
