Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1011867)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 arch=('any')
 url="https://cairocffi.readthedocs.io/"
 license=('BSD')
-depends=('python-cffi' 'cairo' 'python-xcffib')
+depends=('python-cffi' 'cairo')
 makedepends=('python-setuptools' 'gdk-pixbuf2')
 checkdepends=('python-pytest-runner' 'python-pytest-cov' 'python-pytest-flake8'
               'python-pytest-isort' 'python-numpy')
