--- PKGBUILD
+++ PKGBUILD
@@ -17,10 +17,15 @@
             'c00bce5471e912c7a2e0a4db2e96329dd4720b5f6813957bd5e0298ec5fe42ea362fcf0047b7a41e17aeab802bf36ef14e1b30826d87646c104f5f39002356ef'
             'ff442f4bf916703a4231607d8b79f16d71e8b03665deb9431f438edce12075eb1cc784413bdfd23e580e149c8f150cbe8ea137b3b1e6d046d83204e29264c470')
 
+# Backport https://github.com/vaidik/commentjson/pull/60 to v0.9.0.
+source+=(python314.patch)
+sha512sums+=('1258435ae78ebc167eaf5d7b3731622776b43f8a4e88077f0e4456cbe012dd127367fb362e5a9ed8cd3db8f86fc27a45c7e6d9ed99ab70d883e33de1b09cb20a')
+
 prepare() {
   cd commentjson-$pkgver
   patch -p1 -i ../$pkgname-rename-lark.patch
   patch -p1 -i ../python3-12-test-infinity-skip.patch
+  patch -p1 -i ../python314.patch
 }
 
 build() {
