diff --git PKGBUILD.orig PKGBUILD
index 0fecb71..ce497f9 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -12,7 +12,7 @@
 license=('GPL')
 depends=('python-psutil')
 makedepends=('python-setuptools')
-checkdepends=('python-pytest')
+checkdepends=('python-pytest' 'python-six')
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/${pkgver}.tar.gz")
 sha256sums=('37601b714b9a42b217765c6eb7ae20b36796b92c642feea6bd275e1c702799ed')
 
