--- PKGBUILD
+++ PKGBUILD
@@ -59,6 +59,9 @@
         'ae4d401c7375737b394654ab6d7cb9e61e34013f583ef05c57e74c6f6e9700ed6787afb83fab88e826e0f086e5e079258f0c1cdddf89b3e792ff67611e781052'
         'e2421669901b4549700ec5abccd07a169fe0a3d2c5ddadecf55bc911f68df64ce6154431f01e1ad16c4af0299b029f6df787bf74b3070fece270cec9b74c907d'
         '3d51383d407f7d9c5da8df55899e2ff6a0fdbbf07583af0e06c8bcefda73b1fcecd89ffea52f822ae605a14d498f11b0e492b686ad711d0e3dc97a2c601fdaa0')
+source+=(rust-1.98-riscv64-target-detection.patch)
+sha256sums+=('0236d5a8cc339096f287790f4f5015a79bd0a7f92477450b11eab88123afe142')
+b2sums+=('d47df95569d0fbc907fc631af56ca2ead47c06da533adcbc3b29ab9fa77faae0f1218884ca265a3dce6ce4c007f896eadb3739f45257066ee090b01a91dba522')
 
 # Make sure the duplication between bin and lib is found
 COMPRESSZST+=(--long)
@@ -76,6 +79,9 @@
   patch -Np1 -i ../0004-Bug-1983736-Patch-jsonschema-to-work-with-Python-3.1.patch
   patch -Np1 -i ../0005-Bug-1993797-Fix-AST-parsing-in-DecoratorVisitor-for-.patch
 
+  # https://bugzilla.mozilla.org/show_bug.cgi?id=2053518
+  patch -Np1 -i ../rust-1.98-riscv64-target-detection.patch
+
   cat >../mozconfig <<END
 ac_add_options --enable-application=js
 mk_add_options MOZ_OBJDIR=${PWD@Q}/obj
