Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 457546)
+++ PKGBUILD	(working copy)
@@ -14,10 +14,18 @@
 checkdepends=('dejagnu')
 provides=('libffi.so')
 options=('debug')
-source=(https://github.com/libffi/libffi/releases/download/v$pkgver/libffi-$pkgver.tar.gz)
-sha256sums=('4416dd92b6ae8fcb5b10421e711c4d3cb31203d77521a77d85d0102311e6c3b8')
-b2sums=('5e751c53a6b65316e438723810fbafe7f27732feb50466f1459d086c35a519f460b57968721212496a7502b0a5860546c84b22ec269e979728f18d0731fc918a')
+source=(https://github.com/libffi/libffi/releases/download/v$pkgver/libffi-$pkgver.tar.gz
+        fix-struct-reference.patch::https://github.com/libffi/libffi/commit/4b0c358e28fae22164bf0d423f183dfed8a1ba10.patch)
+sha256sums=('4416dd92b6ae8fcb5b10421e711c4d3cb31203d77521a77d85d0102311e6c3b8'
+            'b14d9a6fa17b46f85249c74fb6ddcd1c6a3af30526ea6400641566e39c17861f')
+b2sums=('5e751c53a6b65316e438723810fbafe7f27732feb50466f1459d086c35a519f460b57968721212496a7502b0a5860546c84b22ec269e979728f18d0731fc918a'
+        'cb41a382a327f7d59d155e49aeb47c8cb5a20212d0e7310387b8aecadd08d9afba180d351d04f267a4af1f626af8d5b22da6859b75ff3aa02447d7381585be5f')
 
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../fix-struct-reference.patch
+}
+
 build() {
   cd $pkgname-$pkgver
   # remove --disable-exec-static-tramp once ghc and gobject-introspection
