--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,12 @@ sha256sums=('329a4d9515d60bebdb657d070824933b993b85864b9d3e302e6361accab992da'
             '65200fa109a819f55843b864164812f6a8e5b5bf910e12732dbc507f8982f29a')
 _exec=('fillets')
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   gendesk -n --exec="fillets" --pkgname="fillets-ng" --pkgdesc="Port of the wonderful puzzle game Fish Fillets"
   export PATH=$PATH:$PWD