diff --git a/tests/nodeenv_test.py b/tests/nodeenv_test.py
index 0422cfa..eb6ca5d 100644
--- a/tests/nodeenv_test.py
+++ b/tests/nodeenv_test.py
@@ -23,7 +23,7 @@ def test_smoke(tmpdir):
     subprocess.check_call([
         # Enable coverage
         'coverage', 'run', '-p',
-        '-m', 'nodeenv', '--prebuilt', nenv_path,
+        '-m', 'nodeenv', '--node=system', nenv_path,
     ])
     assert os.path.exists(nenv_path)
     activate = pipes.quote(os.path.join(nenv_path, 'bin', 'activate'))
