--- systemc-2.3.3/src/sysc/datatypes/int/sc_nbdefs.h.orig	2021-10-30 01:09:47.566122411 +0000
+++ systemc-2.3.3/src/sysc/datatypes/int/sc_nbdefs.h	2021-10-30 01:09:52.989456035 +0000
@@ -177,7 +177,7 @@ typedef unsigned int sc_digit;	// 32-bit
 // Support for the long long type. This type is not in the standard
 // but is usually supported by compilers.
 #ifndef _WIN32
-#   if defined(__x86_64__) || defined(__aarch64__)
+#   if defined(__x86_64__) || defined(__aarch64__) || (defined(__riscv) && __riscv_xlen == 64)
         typedef long long          int64;
         typedef unsigned long long uint64;
 #   else
