diff --git PKGBUILD PKGBUILD
index 9635be4..bce0985 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,15 @@ license=('Apache')
 depends=('gcc-libs')
 makedepends=('cmake' 'ninja')
 replaces=('systemc-cmake')
-source=("http://accellera.org/images/downloads/standards/systemc/$pkgname-$pkgver.tar.gz")
-sha512sums=('831255f8e76e1cf776e0407c92c7a0b70ba63cf6b1785a2df372a6394440944540a8ae62264ef5293a7b951bd8b37de089ef0dbc0ee2d0354e69ce552c020ca2')
+source=("http://accellera.org/images/downloads/standards/systemc/$pkgname-$pkgver.tar.gz"
+        "fix-riscv64.patch")
+sha512sums=('831255f8e76e1cf776e0407c92c7a0b70ba63cf6b1785a2df372a6394440944540a8ae62264ef5293a7b951bd8b37de089ef0dbc0ee2d0354e69ce552c020ca2'
+            '1cd832830167eb7b37fc2c0ba596f091da84a012837d8329e8e6d764085f317484483dbc7817d9affb0d0b611fc45b28a06ebeab52552bc87c488a5a2a2930e9')
 
 prepare() {
   mkdir -p build
   sed -i 's/set (CMAKE_BUILD_TYPE Release/set (CMAKE_BUILD_TYPE ""/' $pkgname-$pkgver/CMakeLists.txt
+  patch -Np1 < $srcdir/fix-riscv64.patch -d $pkgname-$pkgver
 }
 
 build() {
