diff --git PKGBUILD PKGBUILD
index 6a209ef..409938c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,15 @@ license=('Apache')
 depends=('gcc-libs')
 makedepends=('cmake' 'ninja')
 replaces=('systemc-cmake')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/accellera-official/systemc/archive/$pkgver.tar.gz")
-sha512sums=('3ef4b5e9c05b8d03e856598ddc27ad50a0a39a7f9334cd00faefeacdf954b6527104d3238c4e8bfa88c00dc382f4da5a50efbd845fe0b6cc2f5a025c993deefd')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/accellera-official/systemc/archive/$pkgver.tar.gz"
+        "fix-riscv64.patch")
+sha512sums=('3ef4b5e9c05b8d03e856598ddc27ad50a0a39a7f9334cd00faefeacdf954b6527104d3238c4e8bfa88c00dc382f4da5a50efbd845fe0b6cc2f5a025c993deefd'
+            '1cd832830167eb7b37fc2c0ba596f091da84a012837d8329e8e6d764085f317484483dbc7817d9affb0d0b611fc45b28a06ebeab52552bc87c488a5a2a2930e9')

 prepare() {
   mkdir -p build
   sed -i 's/set (CMAKE_BUILD_TYPE Release/set (CMAKE_BUILD_TYPE ""/' $pkgname-$pkgver/CMakeLists.txt
+  patch -Np1 < $srcdir/fix-riscv64.patch -d $pkgname-$pkgver
 }

 build() {
