diff --git PKGBUILD PKGBUILD
index ebe3c626..9be1529f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,9 @@
 prepare() {
   # fix issues with tests: https://github.com/systemd/mkosi/commit/60289499675661169cd7bfda2faf331663a43c73
   patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-14-fix_tests.patch
+  # Upstream has dropped this test in HEAD, which hard-coded x86_64 and cannot pass on riscv64.
+  # https://github.com/systemd/mkosi/commit/f49d742f195a5b679716ca7c1e68b2ebc12f9bca
+  rm -f $pkgname-$pkgver/tests/{conftest.py,test_config_parser.py}
 }
 
 build() {
