diff --git PKGBUILD PKGBUILD
index f9010b6..566c9ff 100644
--- PKGBUILD
+++ PKGBUILD
@@ -135,12 +135,12 @@ build() {
     --enable-vaapi \
     --enable-amf \
     --enable-libvpl \
-    --enable-ffnvcodec \
-    --enable-cuda \
-    --enable-cuda-llvm \
-    --enable-cuvid \
-    --enable-nvdec \
-    --enable-nvenc
+    --disable-ffnvcodec \
+    --disable-cuda \
+    --disable-cuda-llvm \
+    --disable-cuvid \
+    --disable-nvdec \
+    --disable-nvenc
 
   make
 }
