diff --git PKGBUILD PKGBUILD
index 26fa20747..e943947cd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('x86_64')
 url='https://github.com/Philip-Scott/Spice-up'
 license=('GPL3')
 depends=('granite' 'libevdev')
-makedepends=('cmake' 'vala')
+makedepends=('cmake' 'vala' 'libgudev')
 checkdepends=('xorg-server-xvfb')
 source=("https://github.com/Philip-Scott/Spice-up/archive/$pkgver/Spice-up-$pkgver.tar.gz"
         '0001-Fix-build-with-vala-0.46.patch'
