--- PKGBUILD
+++ PKGBUILD
@@ -16,12 +16,10 @@ depends=(
 )
 makedepends=(
   rdma-core
-  cuda
   rocm-language-runtime
 )
 optdepends=(
   'rdma-core: for InfiniBand and RDMA support'
-  'cuda: for CUDA support'
   'rocm-language-runtime: for ROCm support'
 )
 provides=(
@@ -38,7 +36,6 @@ build() {
   local configure_options=(
     --prefix=/usr
     --sysconfdir=/etc
-    --with-cuda=/opt/cuda
     --with-rocm=/opt/rocm
     --with-verbs
     --with-rc
