diff --git PKGBUILD PKGBUILD
index fa0e670..79c0063 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,12 +16,10 @@
 )
 makedepends=(
   rdma-core
-  cuda
   rocm-language-runtime
 )
 optdepends=(
   'rdma-core: for InfiniBand and RDMA support'
-  'cuda: for CUDA support'
   'rocm-language-runtime: for ROCm support'
 )
 provides=(
@@ -37,7 +35,6 @@
 build() {
   local configure_options=(
     --prefix=/usr
-    --with-cuda=/opt/cuda
     --with-rocm=/opt/rocm
     --with-verbs
     --with-rc
