--- PKGBUILD	2026-03-06 00:54:10.632632013 +0800
+++ PKGBUILD.riscv64	2026-03-06 00:58:36.395091922 +0800
@@ -19,12 +19,10 @@
 )
 makedepends=(
   rdma-core
-  cuda
   rocm-language-runtime
 )
 optdepends=(
   'rdma-core: for InfiniBand and RDMA support'
-  'cuda: for CUDA support'
   'rocm-language-runtime: for ROCm support'
 )
 provides=(
@@ -35,7 +33,7 @@
   libuct.so
 )
 source=(
-  $pkgname-$pkgver.tar.gz::https://github.com/openucx/$_name/archive/refs/tags/$pkgver.tar.gz
+  $pkgname-$pkgver.tar.gz::https://github.com/openucx/$_name/archive/refs/tags/v$pkgver.tar.gz
   ucx-conf.patch
   c-gnu23.patch
 )
@@ -57,7 +55,6 @@
   local configure_options=(
     --prefix=/usr
     --sysconfdir=/etc
-    --with-cuda=/opt/cuda
     --with-rocm=/opt/rocm
     --with-verbs
     --with-rc
