diff -uprN peg-0.1.18/Makefile peg-0.1.18-patch/Makefile
--- peg-0.1.18/Makefile	2021-10-28 18:25:34.776195839 +0000
+++ peg-0.1.18-patch/Makefile	2021-10-28 18:30:54.379533539 +0000
@@ -19,18 +19,18 @@ PREFIX	= /usr/local
 BINDIR	= $(ROOT)$(PREFIX)/bin
 MANDIR	= $(ROOT)$(PREFIX)/man/man1
 
-install : $(BINDIR) $(BINDIR)/peg $(BINDIR)/leg $(MANDIR) $(MANDIR)/peg.1
+install : $(BINDIR)/peg $(BINDIR)/leg $(MANDIR)/peg.1
 
 $(BINDIR) :
 	mkdir -p $(BINDIR)
 
-$(BINDIR)/% : %
+$(BINDIR)/% : % $(BINDIR)
 	cp -p $< $@
 
 $(MANDIR) :
 	mkdir -p $(MANDIR)
 
-$(MANDIR)/% : src/%
+$(MANDIR)/% : src/% $(MANDIR)
 	cp -p $< $@
 
 uninstall : .FORCE
