diff --git PKGBUILD PKGBUILD
index 969bfe6..f5aa9ab 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,13 +9,16 @@ pkgdesc='recursive-descent parser generators for C'
 url='http://piumarta.com/software/peg/'
 license=('MIT')
 source=("http://piumarta.com/software/peg/peg-${pkgver}.tar.gz"
-        '0000-dont-strip.patch')
+        '0000-dont-strip.patch'
+        '0001-fix-make-dependency.patch')
 sha256sums=('20193bdd673fc7487a38937e297fff08aa73751b633a086ac28c3b34890f9084'
-            'b5ca5f4028651990d30735db1093be08c5cc5d214fc0f73f3a7c7314906c1c1e')
+            'b5ca5f4028651990d30735db1093be08c5cc5d214fc0f73f3a7c7314906c1c1e'
+            '1070a9e2b29eee997ba1113ea417f5e92d935ff2304b1611ffb272393f83cead')
 
 prepare() {
     cd "$pkgname-$pkgver"
     patch -Np1 -i "${srcdir}/0000-dont-strip.patch"
+    patch -Np1 -i "${srcdir}/0001-fix-make-dependency.patch"
 }
 
 build() {
