--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,12 @@ makedepends=('boost' 'cppunit' 'gperf' 'doxygen' 'glm' 'mdds')
 source=(https://dev-www.libreoffice.org/src/$pkgname/$pkgname-$pkgver.tar.xz)
 sha256sums=('b430435a6e8487888b761dc848b7981626eb814884963ffe25eb26a139301e9a')
 
+prepare(){
+  cd $pkgname-$pkgver
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr --with-mdds=2.0
