Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1097005)
+++ PKGBUILD	(working copy)
@@ -9,10 +9,17 @@
 url='https://github.com/pradyunsg/furo'
 license=('MIT')
 depends=('python-sphinx' 'python-pygments' 'python-beautifulsoup4')
-makedepends=('python-build' 'python-pip' 'python-sphinx-theme-builder' 'python-flit-core')
+makedepends=('python-build' 'python-pip' 'python-sphinx-theme-builder' 'python-flit-core'
+             'nodejs' 'npm' 'expac')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/$pkgver.tar.gz")
 sha512sums=('dcd2bae18c28a8623b566ef79a996a76d9157a05627b4e9c930b0ea82154992b9d2321ffe1c62fc2d014d499b7ca1f71117bcea9a1e8879a0ef12d3818cdd290')
 
+prepare() {
+  cd $_pkgname-$pkgver
+  # force using system nodejs, because there isn't a prebuilt nodejs for riscv64
+  sed -i "s/16.13.0/$(expac %v nodejs | cut -d - -f 1)/" pyproject.toml
+}
+
 build() {
   cd $_pkgname-$pkgver
 
