--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,7 @@ pkgver() {
 prepare() {
   cd "$pkgname"
 
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked
 }
 
 build() {
@@ -41,13 +41,13 @@ build() {
 
   CFLAGS+=" -ffat-lto-objects"
 
-  cargo build --release --frozen --workspace --features=extra,dataframe
+  cargo build --release --frozen --workspace --features=extra
 }
 
 check() {
   cd "$pkgname"
 
-  cargo test --frozen --workspace --features=extra,dataframe
+  cargo test --frozen --workspace --features=extra
 }
 
 package() {
