diff --git PKGBUILD PKGBUILD
index acec71c..d0d468d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,10 @@ pkgver() {
 
 prepare() {
   cd "$pkgname"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\narrow = { git = 'https://github.com/felixonmars/arrow-rs', branch = 'v5.1.0' }" >> Cargo.toml
+  echo -e "parquet = { git = 'https://github.com/felixonmars/arrow-rs', branch = 'v5.1.0' }" >> Cargo.toml
+  cargo update -p arrow -p parquet
+  cargo fetch --locked
 }
 
 build() {
