--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,7 @@ pkgver() {
 prepare() {
   cd "$pkgname"
 
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked
 }
 
 build() {
@@ -41,14 +41,14 @@ build() {
 
   CFLAGS+=" -ffat-lto-objects"
 
-  cargo build --release --frozen --workspace --features=extra,dataframe
+  cargo build --release --frozen --workspace --features=extra
 }
 
 check() {
   cd "$pkgname"
 
   # https://github.com/nushell/nushell/issues/10468
-  cargo test --frozen --workspace --features=extra,dataframe -- \
+  cargo test --frozen --workspace --features=extra -- \
     --skip commands::table::table_expand_exceed_overlap_0 \
     --skip commands::table::table_expand_padding_not_default \
     --skip commands::table::test_collapse_big_0 \
