Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1014381)
+++ PKGBUILD	(working copy)
@@ -18,7 +18,10 @@
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\narrow = { git = 'https://github.com/felixonmars/arrow-rs', branch = 'v5.1.0' }" >> Cargo.toml
+  echo -e "parquet = { git = 'https://github.com/felixonmars/arrow-rs', branch = 'v5.1.0' }" >> Cargo.toml
+  cargo update -p arrow -p parquet
+  cargo fetch --locked
 }
 
 build() {
