diff --git a/setup.py b/setup.py
index be45ccc..366d606 100755
--- a/setup.py
+++ b/setup.py
@@ -5,18 +5,13 @@
 
 try:
     from ez_setup import use_setuptools
+
     use_setuptools()
 except:
     pass
 
 from setuptools import setup
 
-try:
-    import six
-    py3 = six.PY3
-except:
-    py3 = sys.version_info[0] >= 3
-
 # metadata
 import re
 _version_re = re.compile(r'__version__\s*=\s*"(.*)"') 
