diff -uprN uudeview-0.5.20/inews/inews.c uudeview-0.5.20-patch/inews/inews.c
--- uudeview-0.5.20/inews/inews.c	2004-01-29 10:14:19.000000000 +0800
+++ uudeview-0.5.20-patch/inews/inews.c	2021-10-10 05:12:31.917312209 +0800
@@ -303,7 +303,7 @@ gen_frompath()
 			putc(*cp, ser_wr_fp);
 		else {		/* Stupid & hack.  God damn it. */
 			putc(toupper(passwd->pw_name[0]), ser_wr_fp);
-			fprintf(ser_wr_fp, passwd->pw_name+1);
+			fputs(passwd->pw_name+1, ser_wr_fp);
 		}
 
 	fprintf(ser_wr_fp, ")\r\n");
diff -uprN uudeview-0.5.20/unix/uuenview.c uudeview-0.5.20-patch/unix/uuenview.c
--- uudeview-0.5.20/unix/uuenview.c	2002-03-06 21:52:46.000000000 +0800
+++ uudeview-0.5.20-patch/unix/uuenview.c	2021-10-10 05:17:56.521710054 +0800
@@ -483,7 +483,7 @@ AttachFiles (char *towhom, char *subject
       if (_FP_stristr (input, "multipart") != NULL) {
 	/* it is already a multipart posting. grab the boundary */
 	if ((ptr = _FP_stristr (input, "boundary=")) != NULL) {
-	  fprintf(thepipe,  input);
+	  fputs(input, thepipe);
 	  strcpy (boundary, ParseValue (ptr));
 	  hadmulti = 1;
 	}
