diff --git PKGBUILD PKGBUILD
index fa13467..e573574 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ url="http://www.fpx.de/fp/Software/UUDeview/"
 license=('GPL')
 optdepends=('tk: for GUI')
 options=('staticlibs')
-source=(http://www.fpx.de/fp/Software/UUDeview/download/$pkgname-$pkgver.tar.gz)
-md5sums=('0161abaec3658095044601eae82bbc5b')
+source=(http://www.fpx.de/fp/Software/UUDeview/download/$pkgname-$pkgver.tar.gz
+        fix-format-string.patch)
+md5sums=('0161abaec3658095044601eae82bbc5b'
+         '6038a42ac14a5e42186ed954a5b38ea6')
+
+prepare() {
+  cd "$srcdir"/$pkgname-$pkgver
+  patch -Np1 -i ${srcdir}/fix-format-string.patch
+}
 
 build() {
   cd "$srcdir"/uudeview-$pkgver/uulib
