diff --git PKGBUILD PKGBUILD
index 5991799..87c2566 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,3 +30,11 @@
   cd deepin-voice-note
   DESTDIR="$pkgdir" ninja install
 }
+
+source+=("deepin-voice-note-include-QJsonDocument.patch::https://github.com/linuxdeepin/deepin-voice-note/pull/212.diff")
+sha512sums+=('eceda44216edfe956a780fb3e50f349df4c41d83ebf61d1afcbafdcd4033b5668a743e990e73c423ceac8caa2e5a1390d400af1466635e795645b1feb6c6aa36')
+
+prepare() {
+  cd deepin-voice-note
+  patch -Np1 -i ../deepin-voice-note-include-QJsonDocument.patch
+}
