diff -uprN beaver-0.4.1/src/editor.c beaver-0.4.1-patch/src/editor.c
--- beaver-0.4.1/src/editor.c	2010-08-29 15:29:16.000000000 +0000
+++ beaver-0.4.1-patch/src/editor.c	2021-10-13 00:43:54.929375084 +0000
@@ -487,7 +487,7 @@ static gboolean iter_is_on_line_comment
 /* preconditions:  iter points to possible start of interesting word  
 postconditions: end is set to the end of the word in the buffer if it was found.
 if word found, returns TRUE, else FALSE*/
-inline gboolean starts_string (gchar * text, gint i, gchar * word, gint * end)
+static inline gboolean starts_string (gchar * text, gint i, gchar * word, gint * end)
 {
 	gunichar ch;
 	gunichar k;
@@ -521,7 +521,7 @@ inline gboolean starts_string (gchar * t
 
 
 /* returns color index of match, or -1 if none */
-inline gint is_matching_keyword (gint Lg, const gchar * str)
+static inline gint is_matching_keyword (gint Lg, const gchar * str)
 {
 	gint rv = -1;
 	gpointer val;
@@ -537,7 +537,7 @@ inline gint is_matching_keyword (gint Lg
 }
 
 
-inline void refresh_markers (GtkTextBuffer * Buffer)
+static inline void refresh_markers (GtkTextBuffer * Buffer)
 {
 	gint CurrentPage;
 	gint Lg;
@@ -894,7 +894,7 @@ void search_for_keyword_correctly (gint
 END_FCN}
 
 
-inline void search_for_keyword_fastly (gint page, GtkTextBuffer * Buffer,
+static inline void search_for_keyword_fastly (gint page, GtkTextBuffer * Buffer,
 									   gchar * text, gint * i, gint offset,
 									   gint Lg)
 {
