diff --git PKGBUILD PKGBUILD
index 3156798..b51a9c8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,12 +15,19 @@ url='http://beaver-editor.sourceforge.net/'
 license=('GPL2')
 depends=('gtk2' 'hicolor-icon-theme')
 makedepends=('intltool')
-source=(https://sourceforge.net/projects/beaver-editor/files/${pkgver:0:3}/beaver-${pkgver}.tar.bz2)
-sha256sums=('844a327a59f2de21e48e55176d45207036837f08935c3341ea963dc62a1531e3')
+source=(https://sourceforge.net/projects/beaver-editor/files/${pkgver:0:3}/beaver-${pkgver}.tar.bz2
+        fix-static-inline.patch)
+sha256sums=('844a327a59f2de21e48e55176d45207036837f08935c3341ea963dc62a1531e3'
+            '50bd198000c8269e073007daebf725e6145dd21ae69ba1ebb40958265541af62')
+
+prepare() {
+  patch -Np1 < fix-static-inline.patch -d ${pkgname}-${pkgver}
+}
 
 build() {
   cd ${pkgname}-${pkgver}
 
+  CFLAGS+=" -fcommon"
   ./configure --prefix=/usr
   make
 }
