--- PKGBUILD
+++ PKGBUILD
@@ -5,6 +5,7 @@
 
 pkgname=intellij-idea-community-edition
 pkgver=2024.3.3
+_tag=${pkgver}+0
 _build=243.24978.46
 _jrever=21
 _jdkver=21
@@ -20,49 +21,24 @@ makedepends=('ant' 'git' maven)
 optdepends=(
   'lldb: lldb frontend integration'
 )
-source=("git+https://github.com/JetBrains/intellij-community.git#tag=idea/${_build}"
-        idea-android::"git+https://github.com/JetBrains/android.git#tag=idea/${_build}"
+source=("git+https://github.com/Glavo/JetBrains-IDE-Multiarch.git#tag=idea/${_tag}"
         idea.desktop
         idea.sh
-        enable-no-jdr.patch
-        # The class src/com/intellij/openapi/projectRoots/ex/JavaSdkUtil.java:56 (git commit 0ea5972cdad569407078fb27070c80e2b9235c53)
-        # assumes the user's maven repo is at {$HOME}/.m2/repository and it contains junit-3.8.1.jar
-        https://repo1.maven.org/maven2/junit/junit/3.8.1/junit-3.8.1.jar)
-noextract=('junit-3.8.1.jar')
-sha256sums=('519a2dd48f4804e37d89dd68c820152b5bb809b6d194f10f9e4b001280cc1056'
-            'e1fabad30c867c3b9517cce7e9121b0c3dd5082a17ab606ddaae53833a64248f'
+)
+sha256sums=('7ef194687767ebebd59783bcb94ece91c0682c25a5a5eac26d2bb2c822882f97'
             '049c4326b6b784da0c698cf62262b591b20abb52e0dcf869f869c0c655f3ce93'
             'c5787f6777fa292c6376f3fcddc89f1d0ae4b71f3284d9bfefcea0b0ab3d8ea9'
-            'b7858737346fb08423ee7fd177f9e59180d2173d988dd8622b84d58426fcb3a7'
-            'b58e459509e190bed737f3592bc1950485322846cf10e78ded1d065153012d70')
-
-prepare() {
-  cd intellij-community
-
-  # build system doesn't like symlinks
-  mv "${srcdir}"/idea-android android
-
-  export MAVEN_REPOSITORY=${srcdir}/.m2/repository
-  mvn install:install-file \
-    -Dfile="${srcdir}"/junit-3.8.1.jar \
-    -DgroupId=junit \
-    -DartifactId=junit \
-    -Dversion=3.8.1 \
-    -Dpackaging=jar \
-    -DgeneratePom=true
-
-  patch -Np1 < "${srcdir}/enable-no-jdr.patch"
-}
+)
 
 build() {
-  cd intellij-community
+  cd JetBrains-IDE-Multiarch
   
   export JAVA_HOME="/usr/lib/jvm/java-${_jdkver}-openjdk"
   export PATH="/usr/lib/jvm/java-${_jdkver}-openjdk/bin:$PATH"
-  export MAVEN_REPOSITORY=${srcdir}/.m2/repository
 
-  ./installers.cmd -Dintellij.build.use.compiled.classes=false -Dintellij.build.target.os=linux -Dbuild.number="${_build}"
-  tar -xf out/idea-ce/artifacts/ideaIC-${_build}-no-jbr.tar.gz -C "${srcdir}"
+  ./gradlew transformIC-riscv64
+
+  tar -xf build/target/ideaIC-${_tag}-riscv64.tar.gz -C "${srcdir}"
 }
 
 package() {
