--- PKGBUILD
+++ PKGBUILD
@@ -40,2 +40,9 @@
 
+prepare(){
+  cd $pkgname
+  patch -Np1 -i "$srcdir/relax-template-performance-limit.patch"
+  patch -Np1 -i "$srcdir/paho-legacy-api.patch"
+  patch -Np1 -i "$srcdir/relax-conversion-performance-limit.patch"
+}
+
 build(){
@@ -59 +66,6 @@
 }
+
+source+=(relax-template-performance-limit.patch paho-legacy-api.patch relax-conversion-performance-limit.patch)
+b2sums+=('d4b519cf54c2a4b1ce7f025c7b99e12caa5be9d253f6e3d8dc7e1b68865e7b1e85bd85b7ae430676342ead8126ec1581bce30ad71bad032e94fe00e48a9d9c42'
+         '20b3e5f19a17b887cd3051d4c543e465868c0c85ea1dfcc66673200e49599da0c8bfd27c30f84bc1adc7fe02368a017df708653df77b39d5b054e96d73826a2c'
+         'c8eef969d4323616a56eee92f5681d9a2bb8e6044a1bc6dce016e6bc7c543c20fcc9593662228f383e291c34877e2a68a804aa7bef334d922b8afe3027c16569')
