--- PKGBUILD
+++ PKGBUILD
@@ -51,6 +51,8 @@ options=(!lto)
 prepare() {
   cd cppyy-backend-$_name-$pkgver/cling
 
+  patch -Np2 -i "$srcdir"/add-riscv64-support.patch
+
   # patch out cmake dependency from pyproject.toml
   sed -i 's|requires = \["cmake", |requires = [|' pyproject.toml
 
@@ -83,3 +85,6 @@ package() {
   install -vDm 644 src/LGPL2_1.txt -t "$pkgdir"/usr/share/licenses/$pkgname/
   install -vDm 644 src/interpreter/cling/LICENSE.TXT "$pkgdir"/usr/share/licenses/$pkgname/LICENSE_cling.txt
 }
+
+source+=(add-riscv64-support.patch)
+b2sums+=('bf4798412a7b8b891fb63163e44a8c6bf4b972d6dadc912ec6db6c67608530c6fb9029ee3d47322df67278f34f712e8f1900ba9c14fc2bda907b596bf74b9558')
