--- PKGBUILD
+++ PKGBUILD
@@ -17,11 +17,14 @@ makedepends=('zlib' 'unzip' 'net-tools' 'python' 'git')
 backup=(etc/srs/srs.conf)
 options=('docs')
 source=("git+https://github.com/ossrs/srs/#commit=${_pkgcommit}"
-        "archlinux.patch")
+        "archlinux.patch"
+        "support-riscv.patch") # backport from https://github.com/ossrs/srs/pull/3115
 sha256sums=('SKIP'
-            '01af4fc5b393fe4c590215a8449ab10e7aefe1e377962ca35405baaa2ed43a90')
+            '01af4fc5b393fe4c590215a8449ab10e7aefe1e377962ca35405baaa2ed43a90'
+            'b4b5573cc52c66f7ad2194ce7c13d44b38bedc46ad537418a9c5903b83e9897f')
 
 prepare() {
+  patch -d srs -Np1 < ${srcdir}/support-riscv.patch
   cd "${srcdir}"/srs/trunk
   patch -p2 -i "${srcdir}"/archlinux.patch
 }
