diff --git PKGBUILD PKGBUILD
index 9129079..92ca09c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,8 +7,7 @@ pkgdesc="Curses based fullscreen BOINC manager"
 arch=('x86_64')
 url="https://sourceforge.net/projects/boinctui/"
 license=('GPL')
-depends=('expat' 'gcc-libs' 'openssl')
-makedepends=("cmake")
+depends=('expat' 'gcc-libs' 'openssl' 'ncurses')
 source=("https://sourceforge.net/projects/boinctui/files/${pkgname}_${pkgver}.tar.gz"{,.asc})
 sha512sums=('f884d4960ded1a6e9b6779077ddc85678d859c49c82f11b489e75edba81217947707d5d47cead7ea96f0ca43b6d35bb719db2766856a19a284f49f94e13b2131'
             'SKIP')
@@ -16,7 +15,10 @@ validpgpkeys=('4D2DBBB4F06EC2C977B26BFD80878E59832BEC15')  # Sergey Suslov
 
 build() {
   cd $pkgname-$pkgver
-  autoconf
+  # remove trailing space after a backslash
+  sed -i 's_\\ $_\\_' configure.in
+  # after autoconf 2.70
+  autoreconf -i
   ./configure --prefix=/usr --without-gnutls --mandir=/usr/share/man
   make 
 }
