diff --git PKGBUILD PKGBUILD
index 7c953e71..43c52406 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,15 +17,18 @@ makedepends=(cmake ninja clang llvm python-setuptools vulkan-headers python-nump
 _pkgfn=pyside-setup-opensource-src-$_qtver
 source=(https://download.qt.io/official_releases/QtForPython/pyside6/PySide6-$pkgver-src/${_pkgfn}.tar.xz
         designer-plugin-install-dir.patch
-        fix-includes.patch)
+        fix-includes.patch
+        "fix-shiboken6-with-new-qtbase.patch"::"https://code.qt.io/cgit/pyside/pyside-setup.git/patch/?id=a1581e167e1b669661ce360a0b4569679de2739d")
 sha256sums=('d9680ff298ee8b01a68de20911c60da04568a0918b3062d4c571ef170b4603ff'
             '66e895e07d5b01c64a94092353854c946fd7fc445b6181068dca290b5a3887e0'
-            'eee89b0ad67ac97724ea59a2d94225f3a348c194c01c58b34a757c00f42ce2d2')
+            'eee89b0ad67ac97724ea59a2d94225f3a348c194c01c58b34a757c00f42ce2d2'
+            'ce121f2b167e78a6203f917989866ef37548a1aad34cb58f71602212f658165a')
 options=(debug)
 
 prepare() {
   patch -d $_pkgfn -p1 < designer-plugin-install-dir.patch # Fix designer plugin install dir
   patch -d $_pkgfn -p1 < fix-includes.patch
+  patch -d $_pkgfn -p1 < "fix-shiboken6-with-new-qtbase.patch"
 }
 
 build() {
