diff --git a/wscript b/wscript
index 5ebb9c8..5605f68 100644
--- a/wscript
+++ b/wscript
@@ -75,7 +75,7 @@ def build_plugin(bld, bundle, name, source, cxxflags=[], cppflags=[], libs=[], a
     obj.target       = os.path.join(bundle, name)
     if cxxflags != []:
         obj.cxxflags = cxxflags
-    obj.cxxflags = obj.cxxflags + ['-msse', '-mfpmath=sse', '-ffast-math']
+    #obj.cxxflags = obj.cxxflags + ['-msse', '-mfpmath=sse', '-ffast-math']
     if cppflags != []:
         obj.cppflags = cppflags
     if libs != []:
@@ -98,7 +98,7 @@ def build_plugin_gui(bld, bundle, name, source, cxxflags=[], cppflags=[], libs=[
     obj.target       = os.path.join(bundle, name)
     if cxxflags != []:
         obj.cxxflags = cxxflags
-    obj.cxxflags = obj.cxxflags + ['-msse', '-mfpmath=sse', '-ffast-math']
+    #obj.cxxflags = obj.cxxflags + ['-msse', '-mfpmath=sse', '-ffast-math']
     if cppflags != []:
         obj.cppflags = cppflags
     if libs != []:
