--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,7 @@
   ./sbt \
     -Dsbt.build.version=${pkgver} \
     -Dsbt.build.offline=false \
+    -Dsbt.build.includesbtn=false \
     clean \
     launcherPackage/Universal/packageBin \
     launcherPackage/Universal/packageZipTarball
@@ -48,9 +49,9 @@
 
   mkdir -p "${pkgdir}"/usr/share/${pkgname}
   cp -r bin "${pkgdir}"/usr/share/${pkgname}
-  rm "${pkgdir}"/usr/share/${pkgname}/bin/*{.bat,.exe,-darwin}
+  rm -f "${pkgdir}"/usr/share/${pkgname}/bin/*{.bat,.exe,-darwin}
   chmod -x "${pkgdir}"/usr/share/${pkgname}/bin/*
-  chmod +x "${pkgdir}"/usr/share/${pkgname}/bin/{sbt,sbtn-${CARCH}-pc-linux}
+  chmod +x "${pkgdir}"/usr/share/${pkgname}/bin/sbt
   mkdir -p "${pkgdir}"/usr/bin
   ln -s /usr/share/${pkgname}/bin/sbt "${pkgdir}"/usr/bin/sbt
 
