diff --git PKGBUILD PKGBUILD
index 8d56793a..76c4b58a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,10 @@ makedepends=(gtk-doc gobject-introspection yelp-tools appdata-tools git meson
              unicode-character-database unzip vala)
 provides=(libgucharmap_2_90.so)
 _commit=3efb47b75b61d7c6db7e48d72f7aa8c4d4365011  # tags/14.0.1^0
-source=("git+https://gitlab.gnome.org/GNOME/gucharmap.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gucharmap.git#commit=$_commit"
+        "fix_positional_args.patch")
+sha256sums=('SKIP'
+            '155c46dc7a142c9e0a4dcad09210ae5c965907a6201f1f0f41e807660df6efea')
 
 pkgver() {
   cd $pkgname
@@ -23,6 +25,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../fix_positional_args.patch
 }
 
 build() {
