--- PKGBUILD
+++ PKGBUILD
@@ -74,6 +74,9 @@ prepare() {
   # See https://github.com/Microsoft/vscode/issues/31168 for details.
   patch -p0 -i ../product_json.diff
 
+  # Add missing riscv definition
+  sed -i "/BUILD_TARGETS = \[/a { platform: 'linux', arch: 'riscv64' }," build/gulpfile.vscode.js
+
   # Set the commit and build date
   sed -e "s/@COMMIT@/$(git rev-parse HEAD)/" -e "s/@DATE@/$(date -u -Is | sed 's/\+00:00/Z/')/" -i product.json
 
