diff --color -Naur Xdialog-2.3.1-orig/src/main.c Xdialog-2.3.1/src/main.c
--- Xdialog-2.3.1-orig/src/main.c	2006-08-14 02:22:58.000000000 +0800
+++ Xdialog-2.3.1/src/main.c	2021-10-08 08:47:35.688355644 +0800
@@ -268,7 +268,7 @@
 #endif
 
 	fprintf(stderr, "%s: %s !\n", cmd, errmsg);
-	fprintf(stderr, msg);
+	fprintf(stderr, "%s", msg);
 
 	if (strlen(msg) == HELP_MSG_SIZE-1)
 		fprintf(stderr, "\n\nHelp message truncated, please re-compile "\
