diff --git PKGBUILD PKGBUILD
index 1fe22f2..ba17b20 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,16 @@ arch=('x86_64')
 depends=('gtk2')
 license=('GPL')
 source=("http://xdialog.free.fr/Xdialog-${pkgver}.tar.bz2"
-	"xdialog-2.3.1-gtk2.patch")
+	"xdialog-2.3.1-gtk2.patch"
+  "fix-format.patch")
 sha512sums=('2a0f6e15297ae3ef54a41da1b1fe2ef376d28757d47433df2b8e9918cd4f95871a630b94ec544bd848e518df7f86934030a71bf0371a1466286bf7d912335525'
-            'e1a1d4e93946f89a9f4d2a619fcb0069cd5e6ec1b0c3a3ebcd87b1b601c07a7941ad1edeb1cbe89f22f993667c4431c496a6c9b35a13fd9e7d1b8dbd89546694')
+            'e1a1d4e93946f89a9f4d2a619fcb0069cd5e6ec1b0c3a3ebcd87b1b601c07a7941ad1edeb1cbe89f22f993667c4431c496a6c9b35a13fd9e7d1b8dbd89546694'
+            '346b928dd5a64a365ba5eccce8f98b8c16aefaac49f2f5247eccf3d233b59d54a195a1ddf61d98fdb7e070d54531aa548840dfeca7615ef21c9a61bf7593878c')
 
 build() {
   cd "$srcdir"/Xdialog-$pkgver
   patch -p1 <../xdialog-$pkgver-gtk2.patch
+  patch -p1 <../fix-format.patch
   ./configure --prefix=/usr \
     --mandir="$pkgdir"/usr/share/man --with-gtk2
   # Fix bugs in automake - see Xdialog.spec
