diff --git PKGBUILD PKGBUILD
index dc8d58d..7dbad00 100644
--- PKGBUILD
+++ PKGBUILD
@@ -102,13 +102,15 @@ source=(https://download.videolan.org/${pkgname}/${_vlcver}/${pkgname}-${_vlcver
         update-vlc-plugin-cache.hook
         caca-fix-to-newer-version.patch
         vlc-3.0.11.1-srt_1.4.2.patch
-        vlc-live-media-2021.patch)
+        vlc-live-media-2021.patch
+        riscv64-atomic.patch)
 sha512sums=('35cdf191071224d0cf1b5a83c00773ff87b9e5bfcf0f5523f7edd53f75b23eda6b27bb49ffa97d69a1d176b8fe4786d959aeeb00d4380beab71c9f7e6b7c7298'
             'SKIP'
             'b247510ffeadfd439a5dadd170c91900b6cdb05b5ca00d38b1a17c720ffe5a9f75a32e0cb1af5ebefdf1c23c5acc53513ed983a736e8fa30dd8fad237ef49dd3'
             'ef479a246dc98f882a05ca56a1c2872cc67ced154c625790070b887858ddc250d55b1295db82c9122e5ecd3c2c9c712ec9718e28d0a9d21ff6a230eb6c5010ce'
             'ac1d33d434aca2a0ad6e70800073deeaefc02b8fd72656b682ca833ee0cffe10dfa19a9355388700cab46ffbf9421c007d00ed04c7fa562698ff81e70db5f283'
-            'ad17d6f4f2cc83841c1c89623c339ec3ee94f6084ea980e2c8cbc3903854c85e5396e31bfd8dc90745b41794670903d854c4d282d8adec263087a9d47b226ccc')
+            'ad17d6f4f2cc83841c1c89623c339ec3ee94f6084ea980e2c8cbc3903854c85e5396e31bfd8dc90745b41794670903d854c4d282d8adec263087a9d47b226ccc'
+            'cc522a5538a3485c8973907ecfb13c98f9277023350de4598ceb18180f85f9bdb58dc1e8a7913f44b5e8885713197867bd74c86b5d0e40502e07d01f8ccdb8cc')
 validpgpkeys=('65F7C6B4206BD057A7EB73787180713BE58D1ADC') # VideoLAN Release Signing Key
 
 prepare() {
