diff --color -Naur '--exclude=.git' ssr-orig/glinject/elfhacks.h ssr-new/glinject/elfhacks.h
--- ssr-orig/glinject/elfhacks.h	2021-10-01 09:37:06.148835533 +0800
+++ ssr-new/glinject/elfhacks.h	2021-10-01 13:13:38.402293043 +0800
@@ -44,6 +44,11 @@
 #ifdef __i386__
 # define __elf32
 #endif
+#if __riscv_xlen == 64
+# define __elf64
+#elif __riscv_xlen == 32
+# define __elf32
+#endif
 
 #if defined(__elf64)
 # define ELFW_R_SYM ELF64_R_SYM
