--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,10 @@ depends=("qt5-base" "qt5-x11extras"
 optdepends=("lib32-simplescreenrecorder: OpenGL recording of 32-bit applications")
 makedepends=("git" "cmake" "qt5-tools")
 source=("git+https://github.com/MaartenBaert/ssr.git#tag=$pkgver"
+         riscv.patch::https://github.com/MaartenBaert/ssr/commit/a1962846e8d170e3bcd5f5fb61c432bbdbaa3855.patch
          ffmpeg5.patch::https://patch-diff.githubusercontent.com/raw/MaartenBaert/ssr/pull/934.patch)
 sha256sums=('SKIP'
+            'caddd85d23bb6013e10632dc31044a7a5f2512686ad4b9fea23940a08774d3e7'
             '4b01938615a34127236a21ee0ffa20bbb179c8bfcc5ecef872fc5a246727d2e1')

 install=simplescreenrecorder.install
@@ -25,6 +27,7 @@ prepare() {
   cd ssr
   mkdir -p build

+  patch -p1 -i ../riscv.patch   # Fix build with RISC-V
   patch -p1 -i ../ffmpeg5.patch # Fix build with ffmpeg 5
 }

