diff --color -Naur x42-plugins-20211016-src/balance.lv2/Makefile x42-plugins-20211016/balance.lv2/Makefile
--- x42-plugins-20211016-src/balance.lv2/Makefile	2019-01-24 06:16:51.000000000 +0800
+++ x42-plugins-20211016/balance.lv2/Makefile	2021-12-22 09:59:24.361516034 +0800
@@ -4,7 +4,7 @@
 #   make CXXFLAGS=-O2
 #   make install DESTDIR=$(CURDIR)/debian/balance_lv2 PREFIX=/usr
 #
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CXXFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/controlfilter.lv2/Makefile x42-plugins-20211016/controlfilter.lv2/Makefile
--- x42-plugins-20211016-src/controlfilter.lv2/Makefile	2020-11-10 05:30:18.000000000 +0800
+++ x42-plugins-20211016/controlfilter.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,6 +1,6 @@
 #!/usr/bin/make -f
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/convoLV2/Makefile x42-plugins-20211016/convoLV2/Makefile
--- x42-plugins-20211016-src/convoLV2/Makefile	2020-09-05 07:25:19.000000000 +0800
+++ x42-plugins-20211016/convoLV2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -7,7 +7,7 @@
 PREFIX ?= /usr/local
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CXXFLAGS ?= $(OPTIMIZATIONS) -Wall
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/darc.lv2/Makefile x42-plugins-20211016/darc.lv2/Makefile
--- x42-plugins-20211016-src/darc.lv2/Makefile	2021-10-12 07:19:26.000000000 +0800
+++ x42-plugins-20211016/darc.lv2/Makefile	2021-12-22 09:59:24.361516034 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/dpl.lv2/Makefile x42-plugins-20211016/dpl.lv2/Makefile
--- x42-plugins-20211016-src/dpl.lv2/Makefile	2021-10-12 07:19:37.000000000 +0800
+++ x42-plugins-20211016/dpl.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CXXFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/fat1.lv2/Makefile x42-plugins-20211016/fat1.lv2/Makefile
--- x42-plugins-20211016-src/fat1.lv2/Makefile	2021-10-12 07:19:44.000000000 +0800
+++ x42-plugins-20211016/fat1.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CXXFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/fil4.lv2/Makefile x42-plugins-20211016/fil4.lv2/Makefile
--- x42-plugins-20211016-src/fil4.lv2/Makefile	2021-07-14 06:34:25.000000000 +0800
+++ x42-plugins-20211016/fil4.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CXXFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/matrixmixer.lv2/Makefile x42-plugins-20211016/matrixmixer.lv2/Makefile
--- x42-plugins-20211016-src/matrixmixer.lv2/Makefile	2021-04-10 04:45:21.000000000 +0800
+++ x42-plugins-20211016/matrixmixer.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-optimized
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-optimized
 CFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/meters.lv2/Makefile x42-plugins-20211016/meters.lv2/Makefile
--- x42-plugins-20211016-src/meters.lv2/Makefile	2021-10-12 07:20:15.000000000 +0800
+++ x42-plugins-20211016/meters.lv2/Makefile	2021-12-22 10:07:14.835329614 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -Wall -Wno-unused-function
 
 PKG_CONFIG ?= pkg-config
diff --color -Naur x42-plugins-20211016-src/mididebug.lv2/Makefile x42-plugins-20211016/mididebug.lv2/Makefile
--- x42-plugins-20211016-src/mididebug.lv2/Makefile	2021-08-23 08:54:59.000000000 +0800
+++ x42-plugins-20211016/mididebug.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -4,7 +4,7 @@
 #   make CFLAGS=-O2
 #   make install DESTDIR=$(CURDIR)/debian/mididebug.lv2 PREFIX=/usr
 #
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 LIBDIR ?= lib
diff --color -Naur x42-plugins-20211016-src/midifilter.lv2/Makefile x42-plugins-20211016/midifilter.lv2/Makefile
--- x42-plugins-20211016-src/midifilter.lv2/Makefile	2021-07-27 03:30:05.000000000 +0800
+++ x42-plugins-20211016/midifilter.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/midigen.lv2/Makefile x42-plugins-20211016/midigen.lv2/Makefile
--- x42-plugins-20211016-src/midigen.lv2/Makefile	2020-07-13 09:02:18.000000000 +0800
+++ x42-plugins-20211016/midigen.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/midimap.lv2/Makefile x42-plugins-20211016/midimap.lv2/Makefile
--- x42-plugins-20211016-src/midimap.lv2/Makefile	2021-01-25 03:17:39.000000000 +0800
+++ x42-plugins-20211016/midimap.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/mixtri.lv2/Makefile x42-plugins-20211016/mixtri.lv2/Makefile
--- x42-plugins-20211016-src/mixtri.lv2/Makefile	2021-04-10 04:46:33.000000000 +0800
+++ x42-plugins-20211016/mixtri.lv2/Makefile	2021-12-22 09:59:24.361516034 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -Wall -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/nodelay.lv2/Makefile x42-plugins-20211016/nodelay.lv2/Makefile
--- x42-plugins-20211016-src/nodelay.lv2/Makefile	2020-01-08 12:19:19.000000000 +0800
+++ x42-plugins-20211016/nodelay.lv2/Makefile	2021-12-22 09:59:24.371501672 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/onsettrigger.lv2/Makefile x42-plugins-20211016/onsettrigger.lv2/Makefile
--- x42-plugins-20211016-src/onsettrigger.lv2/Makefile	2019-09-26 09:51:00.000000000 +0800
+++ x42-plugins-20211016/onsettrigger.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall -Wno-unused-function
 
diff --color -Naur x42-plugins-20211016-src/phaserotate.lv2/Makefile x42-plugins-20211016/phaserotate.lv2/Makefile
--- x42-plugins-20211016-src/phaserotate.lv2/Makefile	2021-09-20 12:25:22.000000000 +0800
+++ x42-plugins-20211016/phaserotate.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -2,7 +2,7 @@
 PREFIX ?= /usr/local
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-missed -fopt-info -fopt-info-loop-vec-all
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-missed -fopt-info -fopt-info-loop-vec-all
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 PKG_CONFIG ?= pkg-config
 STRIP ?= strip
diff --color -Naur x42-plugins-20211016-src/sisco.lv2/Makefile x42-plugins-20211016/sisco.lv2/Makefile
--- x42-plugins-20211016-src/sisco.lv2/Makefile	2021-04-10 04:39:04.000000000 +0800
+++ x42-plugins-20211016/sisco.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -5,7 +5,7 @@
 MANDIR ?= $(PREFIX)/share/man/man1
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -g -Wall -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/spectra.lv2/Makefile x42-plugins-20211016/spectra.lv2/Makefile
--- x42-plugins-20211016-src/spectra.lv2/Makefile	2021-04-10 04:46:41.000000000 +0800
+++ x42-plugins-20211016/spectra.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -6,7 +6,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-optimized
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG # -fopt-info-vec-optimized
 CFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/stepseq.lv2/Makefile x42-plugins-20211016/stepseq.lv2/Makefile
--- x42-plugins-20211016-src/stepseq.lv2/Makefile	2021-07-14 06:34:35.000000000 +0800
+++ x42-plugins-20211016/stepseq.lv2/Makefile	2021-12-22 09:59:24.361516034 +0800
@@ -9,7 +9,7 @@
 MANDIR ?= $(PREFIX)/share/man/man1
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/stereoroute.lv2/Makefile x42-plugins-20211016/stereoroute.lv2/Makefile
--- x42-plugins-20211016-src/stereoroute.lv2/Makefile	2019-09-03 07:04:07.000000000 +0800
+++ x42-plugins-20211016/stereoroute.lv2/Makefile	2021-12-22 09:59:24.371501672 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/testsignal.lv2/Makefile x42-plugins-20211016/testsignal.lv2/Makefile
--- x42-plugins-20211016-src/testsignal.lv2/Makefile	2021-06-01 03:26:03.000000000 +0800
+++ x42-plugins-20211016/testsignal.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/tuna.lv2/Makefile x42-plugins-20211016/tuna.lv2/Makefile
--- x42-plugins-20211016-src/tuna.lv2/Makefile	2021-04-10 04:46:58.000000000 +0800
+++ x42-plugins-20211016/tuna.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -10,7 +10,7 @@
 # see http://lv2plug.in/pages/filesystem-hierarchy-standard.html, don't use libdir
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CFLAGS ?= -Wall -g -Wno-unused-function
 
 PKG_CONFIG?=pkg-config
diff --color -Naur x42-plugins-20211016-src/xfade.lv2/Makefile x42-plugins-20211016/xfade.lv2/Makefile
--- x42-plugins-20211016-src/xfade.lv2/Makefile	2019-05-05 07:04:33.000000000 +0800
+++ x42-plugins-20211016/xfade.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -1,5 +1,5 @@
 #!/usr/bin/make -f
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 PREFIX ?= /usr/local
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 
diff --color -Naur x42-plugins-20211016-src/zconvo.lv2/Makefile x42-plugins-20211016/zconvo.lv2/Makefile
--- x42-plugins-20211016-src/zconvo.lv2/Makefile	2021-10-16 19:35:43.000000000 +0800
+++ x42-plugins-20211016/zconvo.lv2/Makefile	2021-12-22 09:59:24.374830217 +0800
@@ -2,7 +2,7 @@
 PREFIX ?= /usr/local
 LV2DIR ?= $(PREFIX)/lib/lv2
 
-OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
+OPTIMIZATIONS ?= -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only -DNDEBUG
 CXXFLAGS ?= $(OPTIMIZATIONS) -Wall
 
 PKG_CONFIG ?= pkg-config
