--- PKGBUILD
+++ PKGBUILD
@@ -38,6 +38,9 @@ _names=(darc dpl fat1 fil4 matrixmixer8x8 meter mixtri scope spectr stepseq tuna
 prepare() {
   cd $pkgname-$pkgver
 
+  # disable SSE
+  sed -i "s/-msse2//g; s/-msse//g; s/-mfpmath=sse//g" $(find . -type f)
+
   declare -A comments=(
     ["darc"]="A general purpose audio signal compressor"
     ["dpl"]="Look-ahead digital peak limiter intended but not limited to the final step of mastering or mixing"
