diff --git PKGBUILD PKGBUILD
index 14f7aba..a9c350c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,6 +10,7 @@ pkgdesc="General purpose DSP and SDR toolkit.  With drivers for usrp and fcd."
 arch=('x86_64')
 url="https://gnuradio.org"
 license=('GPL')
+options=(!lto)
 depends=('python-numpy' 'gsl' 'blas' 'libuhd' 'libvolk' 'log4cpp' 'python-yaml'
     'gmp' 'gsm' 'codec2' 'python-mako' 'python-click-plugins' 'soapysdr'
     'pybind11')
