Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1294153)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,7 @@
     'pybind11' 'libsndfile' 'spdlog')
 makedepends=('alsa-lib' 'boost' 'cmake' 'fftw' 'glu' 'gtk3' 'jack' 'pango'
     'portaudio' 'python-gobject' 'python-lxml' 'python-pyqt5' 'python-cairo'
-    'python-jsonschema' 'qwt' 'zeromq')
+    'python-jsonschema' 'qwt' 'zeromq' 'python-packaging')
 
 # todo
 # split the gui components?
