diff --git i/Source/JavaScriptCore/runtime/TypedArrayAdaptors.h w/Source/JavaScriptCore/runtime/TypedArrayAdaptors.h
index b6c443ca1833..4dd85f48d8b1 100644
--- i/Source/JavaScriptCore/runtime/TypedArrayAdaptors.h
+++ w/Source/JavaScriptCore/runtime/TypedArrayAdaptors.h
@@ -68,14 +68,7 @@ struct IntegralTypedArrayAdaptor {
     
     static Type toNativeFromDouble(double value)
     {
-#if HAVE(FJCVTZS_INSTRUCTION)
         return static_cast<Type>(toInt32(value));
-#else
-        int32_t result = static_cast<int32_t>(value);
-        if (static_cast<double>(result) != value)
-            result = toInt32(value);
-        return static_cast<Type>(result);
-#endif
     }
 
     static constexpr Type toNativeFromUndefined()
