--- PKGBUILD
+++ PKGBUILD
@@ -61,9 +61,6 @@ package() {
   # Remove file conflicting with host binutils and manpages for MS Windows tools
   rm "$pkgdir"/usr/share/man/man1/$_target-{dlltool,windres,windmc}*
   rm "$pkgdir"/usr/lib/bfd-plugins/libdep.so
-  rm "$pkgdir"/usr/etc/gprofng.rc
-  rm -r "$pkgdir"/usr/include
-  rm -r "$pkgdir"/usr/lib/gprofng/
 
   # Remove info documents that conflict with host version
   rm -r "$pkgdir"/usr/share/info
