Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1266539)
+++ PKGBUILD	(working copy)
@@ -26,6 +26,8 @@
   sed -e 's/hi64-apps-ffado/ffado-mixer/g' -i support/xdg/ffado.org-ffadomixer.desktop
   # fix id, so it's coherent with XDG desktop file name
   sed -e 's/ffado.org-ffadomixer.desktop/ffado-mixer.desktop/g' -i support/xdg/ffado-mixer.appdata.xml
+
+  cp -v /usr/share/autoconf/build-aux/config.guess admin/
 }
 
 build() {
