--- PKGBUILD
+++ PKGBUILD
@@ -17,9 +17,16 @@ sha512sums=('6fca0ce86aea84458021360bac3f6775135a1a5c1826194921e2d4ead7c12f6ac56
 validpgpkeys=('B6E73D84EA4FCC47166087253FAAD2CD5ECBB314'  # Rob Tompkins <chtompki@apache.org>
               '2DB4F1EF0FA761ECC4EA935C86FDC7E2A11262CB') # Gary David Gregory (Code signing key) <ggregory@apache.org>
 
+source+=("riscv-fix-archutils.patch::https://github.com/apache/commons-lang/pull/1128.diff"
+         "fastdateparser-test-hack.patch")
+sha512sums+=('153460e79ed0af3f81f3374fe4ece3e5b048c84305464c902a5aab28bcb2469e1c515186df85e74a89b8a51fef7a63e5bdf0b655e8d83ad8c1d4e4d184d3f47d'
+             '5cdbb39f21f3216659269a6d2a6cb85c64b3d337e6e5664928258b57099b13c214169c7ef25aaaecb9e705021d1bcc30fc7903b9f9a38e79921fb6d3f8857637')
+
 prepare() {
 	cd "commons-lang3-$pkgver-src"
 	patch -Np1 < '../skip-broken-locales.patch'
+	patch -Np0 < '../fastdateparser-test-hack.patch'
+	patch -Np1 < '../riscv-fix-archutils.patch'
 }
 
 build() {
