diff --git PKGBUILD PKGBUILD
index 501aebe..cfbfdbd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('x86_64')
 url="https://github.com/hrkfdn/ncspot"
 license=('BSD-2-Clause')
 depends=('glibc' 'gcc-libs' 'openssl' 'libpulse' 'libxcb' 'dbus' 'hicolor-icon-theme')
-makedepends=('cargo' 'python' 'pkgconf' 'ueberzug' 'pandoc-cli' 'portaudio')
+makedepends=('cargo' 'python' 'pkgconf' 'ueberzug' 'pandoc-cli' 'portaudio' 'cmake' 'clang')
 optdepends=('ueberzug: display album art in terminal (X11)')
 source=("${pkgname}-${pkgver}.tar.gz::https://github.com/hrkfdn/ncspot/archive/v${pkgver}.tar.gz")
 b2sums=('ad9d2ef9be11eb5f0efb6b7fd6c4583a04691972d03e8bb86f469f0a4dd6b9cf1bffc6d1c054bf547f95ec2b89c97082825f0cc13b63bd3db800d8b3d5c64bce')
@@ -26,6 +26,7 @@ build() {
   export RUSTUP_TOOLCHAIN=stable
   export CARGO_TARGET_DIR=target
   export CFLAGS="$CFLAGS -ffat-lto-objects"
+  export CMAKE_POLICY_VERSION_MINIMUM=3.5
   cargo build --frozen --release --features cover
   # generate docs
   pandoc README.md -t man -s --columns=500 | grep -vE "\[IMAGE:|Click to show/hide" > ncspot.1
