--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url="https://github.com/hrkfdn/ncspot"
 license=('BSD-2-Clause')
 depends=('glibc' 'gcc-libs' 'openssl' 'libpulse' 'libxcb' 'dbus' 'hicolor-icon-theme')
-makedepends=('cargo' 'python' 'pkgconf' 'ueberzug' 'pandoc-cli' 'portaudio')
+makedepends=('cargo' 'python' 'pkgconf' 'ueberzug' 'pandoc-cli' 'portaudio' 'cmake' 'clang')
 optdepends=('ueberzug: display album art in terminal (X11)')
 source=("${pkgname}-${pkgver}.tar.gz::https://github.com/hrkfdn/ncspot/archive/v${pkgver}.tar.gz")
 b2sums=('032d58508f86164fcdac9929bdf01b7fa5f51926c042d9d87f06f1bfd7436be885d0a48ff3b52bb403f2805a6df504d400cf4a64e955754eae8dd19c2cc581c9')
@@ -33,6 +33,7 @@ build() {
 check() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   export RUSTUP_TOOLCHAIN=stable
+  CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
   cargo test --frozen --features cover
 }
 
