diff --git PKGBUILD PKGBUILD
index 0d0cdfd..ec9a35e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ license=(Apache)
 arch=(x86_64)
 depends=(python-scipy python-qdldl)
 makedepends=(python-build python-installer python-setuptools python-wheel cmake)
-checkdepends=(python-pytest intel-oneapi-mkl)
+checkdepends=(python-pytest)
 source=(https://pypi.org/packages/source/${_pyname:0:1}/$_pyname/$_pyname-$pkgver.tar.gz)
 sha256sums=('16b03bc7c5db921de0d97a308ed874d21af9080660eddd5e7cadb2ba3428b0ff')
 
@@ -23,7 +23,7 @@ check() {
   cd $_pyname-$pkgver
   python -m venv --system-site-packages test-env
   test-env/bin/python -m installer dist/*.whl
-  test-env/bin/python -m pytest -v
+  test-env/bin/python -m pytest -v --deselect src/osqp/tests/mkl_pardiso_test.py
 }
 
 package() {
