--- /usr/bin/arch-nspawn	2023-09-27 07:21:56.000000000 +0800
+++ arch-nspawn	2024-01-02 03:22:33.106553417 +0800
@@ -70,7 +70,7 @@
 fi
 
 # shellcheck disable=2016
-host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)extra/os/.*#\1$repo/os/$arch#'))
+host_mirrors=($(pacman-conf --repo extra Server 2> /dev/null | sed -r 's#(.*/)extra.*#\1\$repo#'))
 
 for host_mirror in "${host_mirrors[@]}"; do
 	if [[ $host_mirror == *file://* ]]; then
