diff --unified --recursive --text a/meson.build b/meson.build
--- a/meson.build	2021-09-25 07:16:00.651373516 +0000
+++ b/meson.build	2021-09-25 07:16:41.454734788 +0000
@@ -27,8 +27,8 @@
 current_os = build_machine.system()
 current_arch = build_machine.cpu_family()
 
-# Add x86_64 optimization where appropriate (not for ARM)
-if current_arch != 'aarch64'
+# Add x86_64 optimization where appropriate (not for ARM, not for RISCV)
+if current_arch != 'aarch64' and current_arch != 'riscv64'
     cflags += ['-msse','-msse2','-mfpmath=sse']
 endif
 
