diff --git PKGBUILD PKGBUILD
index 1a8422266..6d051da23 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,17 @@ groups=('lv2-plugins' 'pro-audio')
 depends=('fftw' 'glibc' 'lv2-host')
 makedepends=('lv2' 'meson')
 checkdepends=('lv2lint')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/lucianodato/${pkgname}/archive/${pkgver}.tar.gz")
-sha512sums=('85fcafdaead400efa3ba9e87f2ecf96c02a41334f933902d456a2d5e714e72ef4124e68534f87b60421ef7139bea4e5e768a89d5ed13e6fcf95a9923cc89e383')
-b2sums=('b840df30a5b374be7df2797f970e115083aa97e1ba4219466a659d29ff5791f34c8d7f7352cfb615ea8d109b92a27b47b29ca1dd6b77031a6f7fd3f15e2ddf5f')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/lucianodato/${pkgname}/archive/${pkgver}.tar.gz"
+        "fix-unrecognized-command-line-option-msse.patch")
+sha512sums=('85fcafdaead400efa3ba9e87f2ecf96c02a41334f933902d456a2d5e714e72ef4124e68534f87b60421ef7139bea4e5e768a89d5ed13e6fcf95a9923cc89e383'
+            '77d6bce4cd28452cb50354573ecb21e6405df1e8f4f95e272e1d84ef6cfdd960c407d84b7dad10f78f8b09b72fbefc6b0b8afc3a6c5f1e5e87a55b79e9cad613')
+b2sums=('b840df30a5b374be7df2797f970e115083aa97e1ba4219466a659d29ff5791f34c8d7f7352cfb615ea8d109b92a27b47b29ca1dd6b77031a6f7fd3f15e2ddf5f'
+        '91c8ed7363f83b40e8c4a5b44b731dd87ce4a8e5b57990f0d053371e147b6e27c7cb9626758b64f3d8555bdab2662a72f867a571f3215cceea6fc6e3f8ce773a')
+
+prepare(){
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i "${srcdir}/fix-unrecognized-command-line-option-msse.patch"
+}
 
 build() {
   cd "$pkgname-$pkgver"
