From: Alberto Garcia <berto@igalia.com>
Subject: Fix build failure due to missing semicolon
Index: webkitgtk/Source/WebCore/platform/audio/DenormalDisabler.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/audio/DenormalDisabler.cpp
+++ webkitgtk/Source/WebCore/platform/audio/DenormalDisabler.cpp
@@ -91,7 +91,7 @@ DenormalDisabler::~DenormalDisabler()
     }
 }
 #else
-DenormalDisabler::DenormalDisabler() = default
+DenormalDisabler::DenormalDisabler() = default;
 DenormalDisabler::~DenormalDisabler() = default;
 #endif
 
