--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ url='https://github.com/delfick/alt-pytest-asyncio'
 depends=('python-pytest')
 makedepends=('python-build' 'python-installer' 'python-hatchling')
 checkdepends=('python-noseofyeti' 'python-nest-asyncio')
-source=("https://github.com/delfick/alt-pytest-asyncio/archive/release-$pkgver/$pkgname-$pkgver.tar.gz")
-sha512sums=('5e35f444ffa1cfbc00a74784f2bb01d4f9ca0852a88e9561afcbba79752575c9c1660d1f189799ab17fd1a31afcf20f590683149c5439c07299d8c3e6e6949d0')
+source=("https://github.com/delfick/alt-pytest-asyncio/archive/release-$pkgver/$pkgname-$pkgver.tar.gz"
+        $pkgname-fix-flaky-test.patch::https://patch-diff.githubusercontent.com/raw/delfick/alt-pytest-asyncio/pull/13.patch)
+sha512sums=('5e35f444ffa1cfbc00a74784f2bb01d4f9ca0852a88e9561afcbba79752575c9c1660d1f189799ab17fd1a31afcf20f590683149c5439c07299d8c3e6e6949d0'
+            '4275765c4013525fe9113984e76b5f03cfd1be237081d0669d7199720aa38451dbb6615afc0b59a25a6aed9d8f068d096c2eec6953c18e4552de395a377703e6')
+
+prepare() {
+  cd alt-pytest-asyncio-release-$pkgver
+  patch -Np1 -i ../$pkgname-fix-flaky-test.patch
+}
 
 build() {
   cd alt-pytest-asyncio-release-$pkgver
