diff --git a/data/meson.build b/data/meson.build
index 34e669da..306bb03f 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -23,7 +23,6 @@ resource_sources = gnome.compile_resources(
 desktop = cheese_namespace + '.desktop'
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop + '.in',
   output: '@BASENAME@',
@@ -35,7 +34,6 @@ i18n.merge_file(
 appdata = cheese_namespace + '.appdata.xml'
 
 i18n.merge_file(
-  appdata,
   input: appdata + '.in',
   output: '@BASENAME@',
   po_dir: po_dir,
