diff --git PKGBUILD PKGBUILD
index 49535c8..baae9c9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,10 @@ depends=(gtk3 gstreamer gst-plugins-bad gst-plugins-base gst-plugins-good clutte
 makedepends=(gobject-introspection vala git appstream-glib meson yelp-tools)
 checkdepends=(xorg-server-xvfb)
 _commit=92ef18589eb10fab08f0806846c66d1e1383c78d  # tags/41.0^0
-source=("git+https://gitlab.gnome.org/GNOME/cheese.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/cheese.git#commit=$_commit"
+        "fix_i18n_positional_args.patch")
+sha256sums=('SKIP'
+            "ae3d36fdab115e0f5e359723e8548d04dbe21a5da28a2cf99dceb7584cb9a2e9")
 
 pkgver() {
   cd cheese
@@ -24,6 +26,7 @@ pkgver() {
 
 prepare() {
   cd cheese
+  patch -Np1 -i "../fix_i18n_positional_args.patch"
 }
 
 build() {
