diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6d19e53..b4ea2bde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ IF(CMAKE_COMPILER_IS_GNUCC)
 	SET(CMAKE_BUILD_TYPE_DebugFast)
 	SET(CMAKE_CXX_FLAGS_DEBUGFAST "-g -O0")
 	SET(CMAKE_C_FLAGS_DEBUGFAST "-g -O0")
-	SET(CMAKE_C_FLAGS "-msse2")
+	#SET(CMAKE_C_FLAGS "-msse2")
 	if(NOT CMAKE_BUILD_TYPE)
 	        SET(CMAKE_BUILD_TYPE "DebugFast")
 	endif(NOT CMAKE_BUILD_TYPE)
