diff --git PKGBUILD PKGBUILD
index b11f1a9..d5f7177 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,12 +14,16 @@ depends=(freeglut freetype2 libedit libevent libjpeg-turbo libpng libsndfile lib
 makedepends=(cmake git)
 options=(!makeflags)
 source=("git+https://github.com/stevedekorte/io#commit=b8a18fc199758ed09cd2f199a9bc821f6821072a" # tag: 2017.09.06
-        'noexec.patch::https://github.com/IoLanguage/io/commit/5b88e9769c1d4629e9a4a01c78b57c0d80c7a70c.patch')
+        'noexec.patch::https://github.com/IoLanguage/io/commit/5b88e9769c1d4629e9a4a01c78b57c0d80c7a70c.patch'
+        "disable-sse.patch")
 b2sums=('SKIP'
-        'bf7bf212349a88d707ca99fff48dfc08b0b4fbdc91a64a090e53c388e161f210d42e1dff590c5cda022c0519f2469ad9be5d4788c226632ad3d516389782912d')
+        'bf7bf212349a88d707ca99fff48dfc08b0b4fbdc91a64a090e53c388e161f210d42e1dff590c5cda022c0519f2469ad9be5d4788c226632ad3d516389782912d'
+        '113eda5c1d79ee66ea4d867a0a4962dc448997d99ca93da53e70577e1d7138fee9378d63381c79c13ec2db3baab68ff340d75cc25b349294460c81aa87502aef')
 
 prepare() {
-  cd $pkgname/libs/iovm/source
+  cd $pkgname
+  patch -Np1 -i "../disable-sse.patch"
+  cd libs/iovm/source
   sed -ri "s,20[0-9]+,$pkgver," IoVersion.h
   sed -i 's,sys/sysctl.h,linux/sysctl.h,g' IoSystem.c
 }
