--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,7 @@
   cd "$pkgname"-$pkgver
   # https://github.com/OpenPrinting/libcupsfilters/issues/209
   patch -Np1 -i ../libcupsfilters-PR167.patch
+  patch -Np1 -i ../libcupsfilters-disable-generator-leaks.patch
   autoreconf -vfi
 }
 
@@ -51,3 +52,6 @@
   mkdir -p "${pkgdir}"/usr/share/licenses/${pkgname}
   install -m644 "${srcdir}"/${pkgname}-${pkgver}/{COPYING,NOTICE} "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+source+=(libcupsfilters-disable-generator-leaks.patch)
+sha256sums+=('a59b53709b68825faf7bd99ac3ce965971d4d2257dc9a8e1b3ed206564b21c7a')
