--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,7 @@
 )
 makedepends=(
   cargo
+  protobuf
 )
 source=($pkgname-$pkgver.tar.gz::$url/archive/refs/tags/v$pkgver.tar.gz)
 b2sums=('20fa1b7f78c639c2564df59f9ca5627b0a867efd151c6c00860b9954e1e56d80530f3f6fdfe380211ad1044eec349f358a4ada011e80fd929ce036e6ff13374a')
@@ -33,7 +34,8 @@
 
 check() {
   cd $_name-$pkgver
-  cargo test --frozen --release
+  # Avoid concurrent fat-LTO test links exhausting builder memory.
+  cargo test --frozen --release -j1
 }
 
 package() {
