--- PKGBUILD
+++ PKGBUILD
@@ -6,17 +6,16 @@ pkgname=libfreexl
 pkgver=2.0.0
 pkgrel=1
 pkgdesc="Library to extract valid data from within an Excel (.xls) spreadsheet"
-arch=('x86_64')
+arch=(x86_64)
 url="https://www.gaia-gis.it/fossil/freexl/index"
 license=('MPL' 'GPL' 'LGPL')
-depends=('glibc' 'minizip')
+depends=('glibc' 'minizip' 'gcc' 'autoconf' 'gdal' 'libspatialite')
 source=(https://www.gaia-gis.it/gaia-sins/${pkgname/lib/}-$pkgver.tar.gz)
 sha256sums=('176705f1de58ab7c1eebbf5c6de46ab76fcd8b856508dbd28f5648f7c6e1a7f0')
 
 build() {
   cd "${srcdir}"/${pkgname/lib/}-$pkgver
-
-  ./configure --prefix=/usr
+  CFLAGS=-O2 ./configure --prefix=/usr --build=riscv64-unknown-linux-gnu
   make
 }
 
