--- PKGBUILD
+++ PKGBUILD
@@ -49,2 +49,7 @@
 
+prepare() {
+  cd pre-commit-$pkgver
+  patch -Np1 -i ../fix-environment-bootstrap.patch
+}
+
 build() {
@@ -104 +109,5 @@
 }
+
+source+=(fix-environment-bootstrap.patch)
+sha512sums+=('22b844d161f65c601ca8605eb048acfe24a5a5ca0684618313cfe9888f57fc58a8b3a7bc95bb66c60df65a1b540cd9511ec3479ac288429a062f796ac865428d')
+b2sums+=('8497c6772cf75f09348eaec8ef1ec886c06e9de635b7b1cf7c29c9443648b1910f09fac6d753ae3b798c38c04858265a4c78ea9791f284097142419075aa38a1')
