--- PKGBUILD
+++ PKGBUILD
@@ -33,9 +33,17 @@ checkdepends=(
 )
 provides=('python-pre-commit')
 replaces=('python-pre-commit')
-source=("https://github.com/pre-commit/pre-commit/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-sha512sums=('f3f37bb3e363286682043af5ae804575f915877282ba986a23ffaf4777dc172c1a051f81e67fccd0f5d064c2f58801f920eb9449fe170856cc124a54d5588862')
-b2sums=('9fd611fd8d40e8de5a1a59ab50cf08dbb39f6d8938a03ab006045b7f0f031f79c2355a0f526e2ce6fa2452121322aeabe6a2d99c15e9b7d17f944ca695beb827')
+source=("https://github.com/pre-commit/pre-commit/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "fix-riscv64-build.patch::https://github.com/pre-commit/pre-commit/pull/3011.diff")
+sha512sums=('f3f37bb3e363286682043af5ae804575f915877282ba986a23ffaf4777dc172c1a051f81e67fccd0f5d064c2f58801f920eb9449fe170856cc124a54d5588862'
+            '9776900ad356a8ab3eecf1a1b2887617ab2f0c50b62180864109b5e160b8523a8f76dd565e9bc4e42c66c1bf3d69fdfd162c180dcdad28ee1c43919c96b6aeb6')
+b2sums=('9fd611fd8d40e8de5a1a59ab50cf08dbb39f6d8938a03ab006045b7f0f031f79c2355a0f526e2ce6fa2452121322aeabe6a2d99c15e9b7d17f944ca695beb827'
+        '20b00bb749053fae2dfa3cd9b420fa5b11e070dc1c556d5dd17b3002de9163b2a09afe687956718384527284dcd77412ab8122b4c96df9e868c8e8f04ee3f1b9')
+
+prepare() {
+  cd pre-commit-$pkgver
+  patch -Np1 -i ../fix-riscv64-build.patch
+}
 
 build() {
   cd pre-commit-$pkgver
