--- PKGBUILD
+++ PKGBUILD
@@ -9,13 +9,18 @@
 url="https://github.com/litex-hub/wishbone-utils"
 license=(Apache-2.0)
 depends=(glibc gcc-libs)
-makedepends=(rust)
+makedepends=(rust git)
 options=(!lto)
-source=($pkgname-$pkgver.tar.gz::https://github.com/litex-hub/wishbone-utils/archive/v$pkgver.tar.gz)
-sha512sums=('f02c376aefafc87ea8a5776764caa5f091ac635e630b60519517b488351d9f21a1428703c4daac357b393b3a424de0b0c119adf45622a556a70e5620717aa3a6')
+source=($pkgname-$pkgver.tar.gz::https://github.com/litex-hub/wishbone-utils/archive/v$pkgver.tar.gz
+        bump-dep-serialport.patch)
+sha512sums=('f02c376aefafc87ea8a5776764caa5f091ac635e630b60519517b488351d9f21a1428703c4daac357b393b3a424de0b0c119adf45622a556a70e5620717aa3a6'
+            '12a148c90656515d0126c372e7a0b6783287c3d803995bbe6d9c97efed4ab83b8f2860c88e9c10fd2e2b5a2c0058a21f0488b28a880628e248d5839952e7f50a')
 
 prepare() {
-  cd $pkgname-$pkgver/wishbone-tool
+  cd $pkgname-$pkgver
+  patch -Np0 -i ../bump-dep-serialport.patch
+  cd wishbone-tool
+  cargo update -p serialport
   cargo fetch --locked --target "$(rustc --print host-tuple)"
 }
 
