--- PKGBUILD
+++ PKGBUILD
@@ -9,13 +9,20 @@ arch=(x86_64)
 url="https://github.com/litex-hub/wishbone-utils"
 license=(Apache-2.0)
 depends=(glibc gcc-libs)
-makedepends=(rust)
+makedepends=(rust git)
 options=(!lto)
-source=($pkgname-$pkgver.tar.gz::https://github.com/litex-hub/wishbone-utils/archive/v$pkgver.tar.gz)
-sha512sums=('ca61e335f97c3e5ed0b02d71601c4d2102c87ea40051f7345514c1624c06bdaa47bb6acca66c0709968c9e39faf570f49e5ec81cb444909029498043240e506d')
+source=($pkgname-$pkgver.tar.gz::https://github.com/litex-hub/wishbone-utils/archive/v$pkgver.tar.gz
+        bump-dep-serialport.patch::https://github.com/litex-hub/wishbone-utils/pull/51.patch
+        Cargo.lock)
+sha512sums=('ca61e335f97c3e5ed0b02d71601c4d2102c87ea40051f7345514c1624c06bdaa47bb6acca66c0709968c9e39faf570f49e5ec81cb444909029498043240e506d'
+            'f95b7ed0cdcc1536e0e586ed4bac56df063914f6a6e123589f358e8db86a431f1c8af12fbd4dbb0434812c87b655255c601b8180ae9c1a591098e521e1b226a8'
+            'bb8bf6ca6f6c8c2d063378ea0eb8ffe464656afd8b88ab0bda8b749210565ffef285d2f7273104a66bf940c37cf2f1252ec0f499eef233ec5349f8af7ae4ebfa')
 
 prepare() {
-  cd $pkgname-$pkgver/wishbone-tool
+  cd $pkgname-$pkgver
+  mv ../Cargo.lock ./wishbone-tool/
+  git apply --exclude=wishbone-tool/Cargo.lock ../bump-dep-serialport.patch
+  cd wishbone-tool
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
