diff --git a/wishbone-tool/crates/bridge/Cargo.toml b/wishbone-tool/crates/bridge/Cargo.toml
index 6df6852..2569107 100644
--- a/wishbone-tool/crates/bridge/Cargo.toml
+++ b/wishbone-tool/crates/bridge/Cargo.toml
@@ -36,7 +36,7 @@ byteorder = { version = "1", optional = true }
 libusb-sys-wishbone-tool = { path="../libusb-sys", version = "0.2.6", optional = true }
 libusb-wishbone-tool = { path = "../libusb-rs", version = "0.3.1", optional = true }
 
-serialport = { version = "3.3", default-features = false, optional = true }
+serialport = { path = "../serialport-rs", version = "3.3", default-features = false, optional = true }
 
 # Enable GPIO access for SpiBone on Raspberry Pi
 [target.'cfg(all(target_os = "linux", any(target_arch = "arm", target_arch = "aarch64")))'.dependencies]
diff --git a/wishbone-tool/crates/serialport-rs/Cargo.toml b/wishbone-tool/crates/serialport-rs/Cargo.toml
index e555893..a6dfd54 100644
--- a/wishbone-tool/crates/serialport-rs/Cargo.toml
+++ b/wishbone-tool/crates/serialport-rs/Cargo.toml
@@ -23,7 +23,7 @@ default = ["libudev"]
 clap = "2.32"
 
 [target."cfg(unix)".dependencies]
-nix = "0.14.1"
+nix = "0.20.0"
 bitflags = "1.0"
 cfg-if = "0.1"
 
diff --git a/wishbone-tool/crates/serialport-rs/src/posix/poll.rs b/wishbone-tool/crates/serialport-rs/src/posix/poll.rs
index 1999660..3ed1557 100644
--- a/wishbone-tool/crates/serialport-rs/src/posix/poll.rs
+++ b/wishbone-tool/crates/serialport-rs/src/posix/poll.rs
@@ -29,7 +29,7 @@ fn wait_fd(fd: RawFd, events: PollFlags, timeout: Duration) -> io::Result<()> {
     #[cfg(target_os = "linux")]
     let wait_res = {
         let timespec = TimeSpec::milliseconds(milliseconds);
-        nix::poll::ppoll(fds.as_mut_slice(), timespec, SigSet::empty())
+        nix::poll::ppoll(fds.as_mut_slice(), Some(timespec), SigSet::empty())
     };
     #[cfg(not(target_os = "linux"))]
     let wait_res = nix::poll::poll(fds.as_mut_slice(), milliseconds as nix::libc::c_int);
