diff --git a/Cargo.toml b/Cargo.toml
index e555893..a6dfd54 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ default = ["libudev"]
 clap = "2.32"
 
 [target."cfg(unix)".dependencies]
-nix = "0.14.1"
+nix = "0.20.0"
 bitflags = "1.0"
 cfg-if = "0.1"
 
diff --git a/src/posix/poll.rs b/src/posix/poll.rs
index 1999660..3ed1557 100644
--- a/src/posix/poll.rs
+++ b/src/posix/poll.rs
@@ -29,7 +29,7 @@ fn wait_fd(fd: RawFd, events: PollFlags, timeout: Duration) -> io::Result<()> {
     #[cfg(target_os = "linux")]
     let wait_res = {
         let timespec = TimeSpec::milliseconds(milliseconds);
-        nix::poll::ppoll(fds.as_mut_slice(), timespec, SigSet::empty())
+        nix::poll::ppoll(fds.as_mut_slice(), Some(timespec), SigSet::empty())
     };
     #[cfg(not(target_os = "linux"))]
     let wait_res = nix::poll::poll(fds.as_mut_slice(), milliseconds as nix::libc::c_int);
