diff --git a/wishbone-tool/crates/bridge/Cargo.toml b/wishbone-tool/crates/bridge/Cargo.toml
index 6df6852..2569107 100644
--- a/wishbone-tool/crates/bridge/Cargo.toml
+++ b/wishbone-tool/crates/bridge/Cargo.toml
@@ -36,7 +36,7 @@ byteorder = { version = "1", optional = true }
 libusb-sys-wishbone-tool = { path="../libusb-sys", version = "0.2.6", optional = true }
 libusb-wishbone-tool = { path = "../libusb-rs", version = "0.3.1", optional = true }
 
-serialport = { version = "3.3", default-features = false, optional = true }
+serialport = { path = "../../../../serialport-rs", version = "3.3", default-features = false, optional = true }
 
 # Enable GPIO access for SpiBone on Raspberry Pi
 [target.'cfg(all(target_os = "linux", any(target_arch = "arm", target_arch = "aarch64")))'.dependencies]

