Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010008)
+++ PKGBUILD	(working copy)
@@ -10,10 +10,13 @@
 depends=('python')
 makedepends=('python-setuptools')
 checkdepends=('python-pytest-runner')
-source=("$pkgbase-$pkgver.tar.gz::https://github.com/workhorsy/py-cpuinfo/archive/v$pkgver.tar.gz")
-sha512sums=('a882516a062815782ebc2ad1edb8813ef566877444a6577d6a752b6e0bfce832010181ae20633d5338bdda3399c50fb670c650983b7ffa3064ff76c5cfb0f8d8')
+source=("$pkgbase-$pkgver.tar.gz::https://github.com/workhorsy/py-cpuinfo/archive/v$pkgver.tar.gz"
+        $pkgname-riscv.patch::https://github.com/workhorsy/py-cpuinfo/pull/167.patch)
+sha512sums=('a882516a062815782ebc2ad1edb8813ef566877444a6577d6a752b6e0bfce832010181ae20633d5338bdda3399c50fb670c650983b7ffa3064ff76c5cfb0f8d8'
+            'dd8121b70b4fc2b9b7b671a093f382f65e726161732d1d29d4cdd35b59f3ae13c7ef1d398602cfa1af6492cee09d5d2615fff375793297a7abbc57faa0c6769b')
 
 prepare() {
+  patch -d py-cpuinfo-$pkgver -p1 < $pkgname-riscv.patch
   cp -a py-cpuinfo-$pkgver{,-py2}
 }
 
