diff --unified --recursive --text a/libdm/getdents.c b/libdm/getdents.c
--- a/libdm/getdents.c	2021-09-05 17:47:06.838727885 +0000
+++ b/libdm/getdents.c	2021-09-05 17:48:43.112119349 +0000
@@ -225,7 +225,7 @@
     dp = (DIRENT_TYPE *) buf;
     skdp = kdp = alloca (red_nbytes);
 
-    retval = syscall(SYS_getdents, fd, kdp, red_nbytes);
+    retval = syscall(SYS_getdents64, fd, kdp, red_nbytes);
 
     if (retval == -1)
       return -1;
