--- PKGBUILD
+++ PKGBUILD
@@ -13,11 +13,15 @@ provides=('xfsdmapi')
 conflicts=('xfsdmapi')
 replaces=('xfsdmapi')
 #source=("ftp://oss.sgi.com/projects/xfs/previous/cmd_tars/$pkgname-$pkgver.tar.gz" dmapi-headers.patch)
-source=(https://sources.archlinux.org/other/packages/dmapi/dmapi-$pkgver.tar.gz dmapi-headers.patch)
+source=(https://sources.archlinux.org/other/packages/dmapi/dmapi-$pkgver.tar.gz dmapi-headers.patch
+        fix-SYS_getdents-undeclared.patch)
 
 prepare() {
   cd $pkgname-$pkgver
   patch -p1 -i ../dmapi-headers.patch # Fix build (Gentoo)
+  patch -p1 -i ../fix-SYS_getdents-undeclared.patch
+
+  autoreconf -fiv
 }
 
 build() {
@@ -42,4 +46,5 @@ package() {
   rm -rf "$pkgdir"/usr/libexec
 }
 sha256sums=('b18e34f47374f6adf7c164993c26df36986a009b86aa004ef9444102653aea69'
-            'c5935312dd4d77156da023776c93036fa7989ce78ff8b9f6d9421c31a5535647')
+            'c5935312dd4d77156da023776c93036fa7989ce78ff8b9f6d9421c31a5535647'
+            '309dbde7243fa09d375f62d05027e62b4d339bc34f0b107d71e87a723f8df757')
