diff --git PKGBUILD PKGBUILD
index 7ce24f243..1f1fcfdcb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,11 +13,12 @@ provides=('xfsdmapi')
 conflicts=('xfsdmapi')
 replaces=('xfsdmapi')
 #source=("ftp://oss.sgi.com/projects/xfs/previous/cmd_tars/$pkgname-$pkgver.tar.gz" dmapi-headers.patch)
-source=(https://sources.archlinux.org/other/packages/dmapi/dmapi-$pkgver.tar.gz dmapi-headers.patch)
+source=(https://sources.archlinux.org/other/packages/dmapi/dmapi-$pkgver.tar.gz dmapi-headers.patch fix-SYS_getdents-undeclared.patch)
 
 prepare() {
   cd $pkgname-$pkgver
   patch -p1 -i ../dmapi-headers.patch # Fix build (Gentoo)
+  patch -p1 -i ../fix-SYS_getdents-undeclared.patch
 }
 
 build() {
@@ -42,4 +43,5 @@ package() {
   rm -rf "$pkgdir"/usr/libexec
 }
 md5sums=('cd825d4e141c16011367e0a0dd98c9c5'
-         '6360c428bb89a1585669c3cf7a632208')
+         '6360c428bb89a1585669c3cf7a632208'
+         '720fbf608511ddf1c8e7349839a0fb17')
