--- PKGBUILD
+++ PKGBUILD
@@ -54,8 +54,6 @@ makedepends=(
   gcc-ada
   gcc-d
   git
-  lib32-glibc
-  lib32-gcc-libs
   libisl
   libmpc
   python
@@ -77,6 +75,7 @@ _libdir=usr/lib/gcc/$CHOST/${pkgver%%+*}
 source=(git+https://sourceware.org/git/gcc.git#commit=$_commit
         c89 c99
         gcc-ada-repro.patch
+	gcc-ada-adac.patch
 )
 validpgpkeys=(F3691687D867B81B51CE07D9BBE43771487328A9  # bpiotrowski@archlinux.org
               86CFFCA918CF3AF47147588051E8B148A9999C34  # foutrelis@archlinux.org
@@ -85,12 +84,19 @@ validpgpkeys=(F3691687D867B81B51CE07D9BBE43771487328A9  # bpiotrowski@archlinux.
 sha256sums=('e7641bfddebaada298ca0a631183ef8d81bb3c5dfd3aef30bc7d4223f1c0e450'
             '7b09ec947f90b98315397af675369a1e3dfc527fa70013062e6e85c4be0275ab'
             '44ea973558842f3f4bd666bdaf6e810fd7b7c7bd36b5cc4c69f93d2cd0124fc7'
-            '1773f5137f08ac1f48f0f7297e324d5d868d55201c03068670ee4602babdef2f')
+            '1773f5137f08ac1f48f0f7297e324d5d868d55201c03068670ee4602babdef2f'
+            '845c201c46c1f2a29e1ee848d3edfaf03e77506a1ff16f94c3b2af9d88120e11')
 pkgver() {
   cd gcc
   echo "$(cat gcc/BASE-VER)+$(git describe --abbrev=12 --tags | sed 's/[^-]*-[^-]*-//;s/[^-]*-/r&/;s/-/+/g;s/_/./')"
 }
 
+for i in "${!pkgname[@]}"; do
+  if [[ ${pkgname[i]} = "lib32-gcc-libs" ]]; then
+    unset 'pkgname[i]'
+  fi
+done
+
 prepare() {
   [[ ! -d gcc ]] && ln -s gcc-${pkgver/+/-} gcc
   cd gcc
@@ -103,6 +109,10 @@ prepare() {
 
   mkdir -p "$srcdir/gcc-build"
   mkdir -p "$srcdir/libgccjit-build"
+
+  # Replace hard-coded $(CC) with $(ADAC) in gcc-ada
+  sed -i 's:$(CC):$(ADAC):' gcc/ada/gcc-interface/Make-lang.in
+  patch -Np1 < "$srcdir/gcc-ada-adac.patch"
 }
 
 build() {
@@ -112,8 +122,7 @@ build() {
       --libexecdir=/usr/lib
       --mandir=/usr/share/man
       --infodir=/usr/share/info
-      --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/gcc/-/issues \
-      --with-build-config=bootstrap-lto
+      --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues \
       --with-linker-hash-style=gnu
       --with-system-zlib
       --enable-cet=auto
@@ -126,8 +135,7 @@ build() {
       --enable-libstdcxx-backtrace
       --enable-link-serialization=1
       --enable-linker-build-id
-      --enable-lto
-      --enable-multilib
+      --disable-multilib
       --enable-plugin
       --enable-shared
       --enable-threads=posix
@@ -145,10 +153,18 @@ build() {
   CFLAGS=${CFLAGS/-Werror=format-security/}
   CXXFLAGS=${CXXFLAGS/-Werror=format-security/}
 
+  # GCC does not respect ADAC in some code and uses CC to compile Ada files :(
+  # Use acx_cv_cc_gcc_supports_ada to bypass ada detection as that uses CC
+  export acx_cv_cc_gcc_supports_ada=yes
+  export ADAC=/opt/gcc-ada/bin/gcc
+  export PATH="$PATH:/opt/gcc-ada/bin"
+  export LIBRARY_PATH=/opt/gcc-ada/lib/
   "$srcdir/gcc/configure" \
     --enable-languages=ada,c,c++,d,fortran,go,lto,m2,objc,obj-c++,rust,cobol \
     --enable-bootstrap \
     "${_confflags[@]:?_confflags unset}"
+  # https://gcc.gnu.org/pipermail/gcc/2001-October/070278.html
+  env -C "$srcdir/gcc/gcc/ada" touch treeprs.ads [es]info.h nmake.ad[bs]
 
   # see https://bugs.archlinux.org/task/71777 for rationale re *FLAGS handling
   make -O STAGE1_CFLAGS="-O2" \
@@ -215,9 +231,6 @@ package_gcc() {
     zlib
     zstd
   )
-  optdepends=(
-    'lib32-gcc-libs: for generating code for 32-bit ABI'
-  )
   provides=(
     $pkgname-multilib
   )
@@ -230,6 +243,9 @@ package_gcc() {
   )
 
   cd gcc-build
+  export PATH="$PATH:/opt/gcc-ada/bin"
+  export ADAC=/opt/gcc-ada/bin/gcc
+  export LIBRARY_PATH=/opt/gcc-ada/lib/
 
   make -C gcc DESTDIR="$pkgdir" install-driver install-cpp install-gcc-ar \
     c++.install-common install-headers install-plugin install-lto-wrapper
@@ -238,24 +254,19 @@ package_gcc() {
   install -m755 -t "$pkgdir/$_libdir/" gcc/{cc1,cc1plus,collect2,lto1}
 
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libgcc DESTDIR="$pkgdir" install
 
   make -C $CHOST/libstdc++-v3/src DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/include DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/libsupc++ DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/python DESTDIR="$pkgdir" install
   make -C $CHOST/libatomic DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/src DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/include DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libstdc++-v3/libsupc++ DESTDIR="$pkgdir" install
-  make -C $CHOST/32/libatomic DESTDIR="$pkgdir" install
 
   make DESTDIR="$pkgdir" install-libcc1
   install -d "$pkgdir/usr/share/gdb/auto-load/usr/lib"
   mv "$pkgdir"/usr/lib/libstdc++.so.6.*-gdb.py \
     "$pkgdir/usr/share/gdb/auto-load/usr/lib/"
-  rm "$pkgdir"/usr/lib{,32}/libstdc++.so*
-  rm "$pkgdir"/usr/lib{,32}/libatomic.so*
+  rm "$pkgdir"/usr/lib/libstdc++.so*
+  rm "$pkgdir"/usr/lib/libatomic.so*
 
   make DESTDIR="$pkgdir" install-fixincludes
   make -C gcc DESTDIR="$pkgdir" install-mkheaders
@@ -272,10 +283,6 @@ package_gcc() {
   make -C $CHOST/libsanitizer/asan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libsanitizer/tsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libsanitizer/lsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libgomp DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libitm DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/32/libsanitizer DESTDIR="$pkgdir" install-nodist_{saninclude,toolexeclib}HEADERS
-  make -C $CHOST/32/libsanitizer/asan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
 
   make -C gcc DESTDIR="$pkgdir" install-man install-info
   rm "$pkgdir"/usr/share/man/man1/{gccgo,gfortran,lto-dump,gdc,gm2,gcobol}.1
@@ -291,7 +298,7 @@ package_gcc() {
   # create cc-rs compatible symlinks
   # https://github.com/rust-lang/cc-rs/blob/1.0.73/src/lib.rs#L2578-L2581
   for binary in {c++,g++,gcc,gcc-ar,gcc-nm,gcc-ranlib}; do
-    ln -s /usr/bin/$binary "$pkgdir"/usr/bin/x86_64-linux-gnu-$binary
+    ln -s /usr/bin/$binary "$pkgdir"/usr/bin/riscv64-linux-gnu-$binary
   done
 
   # POSIX conformance launcher scripts for c89 and c99
@@ -302,8 +309,7 @@ package_gcc() {
   make -C $CHOST/libstdc++-v3/doc DESTDIR="$pkgdir" doc-install-man
 
   # remove files provided by lib32-gcc-libs and libgcc
-  rm -f "$pkgdir"/usr/lib{,32}/libgcc_s{,_asneeded}.so*
-  rm -f "$pkgdir"/usr/lib32/libstdc++.so
+  rm -f "$pkgdir"/usr/lib/libgcc_s{,_asneeded}.so*
 
   # byte-compile python libraries
   python -m compileall "$pkgdir/usr/share/gcc-${pkgver%%+*}/"
@@ -336,6 +342,10 @@ package_gcc-ada() {
     staticlibs
   )
 
+  export PATH="$srcdir/ada-bin:$PATH"
+  export ADAC=/opt/gcc-ada/bin/gcc
+  export LIBRARY_PATH=/opt/gcc-ada/lib/
+
   cd gcc-build/gcc
   make DESTDIR="$pkgdir" ada.install-{common,info}
   install -m755 gnat1 "$pkgdir/$_libdir"
@@ -344,10 +354,6 @@ package_gcc-ada() {
   make DESTDIR="$pkgdir" INSTALL="install" \
     INSTALL_DATA="install -m644" install-libada
 
-  cd "$srcdir"/gcc-build/$CHOST/32/libada
-  make DESTDIR="$pkgdir" INSTALL="install" \
-    INSTALL_DATA="install -m644" install-libada
-
   ln -s gcc "$pkgdir/usr/bin/gnatgcc"
 
   # insist on dynamic linking, but keep static libraries because gnatmake complains
@@ -356,12 +362,6 @@ package_gcc-ada() {
   ln -s libgnat-${pkgver%%.*}.so "$pkgdir/usr/lib/libgnat.so"
   rm -f "$pkgdir"/$_libdir/adalib/libgna{rl,t}.so
 
-  install -d "$pkgdir/usr/lib32/"
-  mv "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}-${pkgver%%.*}.so "$pkgdir/usr/lib32"
-  ln -s libgnarl-${pkgver%%.*}.so "$pkgdir/usr/lib32/libgnarl.so"
-  ln -s libgnat-${pkgver%%.*}.so "$pkgdir/usr/lib32/libgnat.so"
-  rm -f "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}.so
-
   _install_runtime_library_exception
 }
 
@@ -425,8 +425,6 @@ package_gcc-fortran() {
   cd gcc-build
   make -C $CHOST/libgfortran DESTDIR="$pkgdir" install-cafexeclibLTLIBRARIES \
     install-{toolexeclibDATA,nodist_fincludeHEADERS,gfor_cHEADERS}
-  make -C $CHOST/32/libgfortran DESTDIR="$pkgdir" install-cafexeclibLTLIBRARIES \
-    install-{toolexeclibDATA,nodist_fincludeHEADERS,gfor_cHEADERS}
   make -C $CHOST/libgomp DESTDIR="$pkgdir" install-nodist_fincludeHEADERS
   make -C gcc DESTDIR="$pkgdir" fortran.install-{common,man,info}
   install -Dm755 gcc/f951 "$pkgdir/$_libdir/f951"
@@ -486,11 +484,10 @@ package_gcc-go() {
 
   cd gcc-build
   make -C $CHOST/libgo DESTDIR="$pkgdir" install-exec-am
-  make -C $CHOST/32/libgo DESTDIR="$pkgdir" install-exec-am
   make DESTDIR="$pkgdir" install-gotools
   make -C gcc DESTDIR="$pkgdir" go.install-{common,man,info}
 
-  rm -f "$pkgdir"/usr/lib{,32}/libgo.so*
+  rm -f "$pkgdir"/usr/lib/libgo.so*
   install -Dm755 gcc/go1 "$pkgdir/$_libdir/go1"
 
   _install_runtime_library_exception
