diff --git a/CMakeLists.txt b/CMakeLists.txt
index b025f3d..3f3630b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,8 +31,10 @@ set(archdetect_c_code "
     #error cmake_ARCH ppc
 #elif defined __ppc64__
     #error cmake_ARCH ppc64
-#elif defined __riscv
-    #error cmake_ARCH riscv
+#elif defined __riscv && __riscv_xlen == 32
+    #error cmake_ARCH riscv32
+#elif defined __riscv && __riscv_xlen == 64
+    #error cmake_ARCH riscv64
 #else
     #error cmake_ARCH unknown
 #endif
@@ -204,6 +206,12 @@ if(${ARCHITECTURE} STREQUAL arm OR ${ARCHITECTURE} STREQUAL aarch64)
         Source/MACLib/NNFilterNeon.cpp)
 endif()
 
+if(${ARCHITECTURE} STREQUAL riscv32 OR ${ARCHITECTURE} STREQUAL riscv64)
+    set(MAC_LIBRARY_SOURCE_FILES
+        ${MAC_LIBRARY_SOURCE_FILES}
+        Source/MACLib/NNFilterRVV.cpp)
+endif()
+
 #
 # Define library targets
 #
@@ -260,6 +268,12 @@ if(${ARCHITECTURE} STREQUAL x86 OR ${ARCHITECTURE} STREQUAL x86_64)
     endif()
 endif()
 
+if(${ARCHITECTURE} STREQUAL riscv64)
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+        set_source_files_properties(Source/MACLib/NNFilterRVV.cpp PROPERTIES COMPILE_FLAGS "-march=rv64gcv")
+    endif()
+endif()
+
 #
 # Set include folders
 #
