--- PKGBUILD
+++ PKGBUILD
@@ -42,3 +42,12 @@ package() {
 
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE.md
 }
+
+source+=(fix-rvv-build.patch)
+sha512sums+=('7831b22e068eca5c5f69fd27030cd0693a35fbe990f81a8c39bdd3fcf42967f8ba8c9ac69d03e5fc4d1097ad47d406a2514269fa9c4ae3632ac49ea6c4ff6f4e')
+b2sums+=('9b887c654844c36097aab1daf0aee91296c2ee0c4b266bc1fbdf2f65e6653ea73aa19ca2ef58eeef540936dcb75577583e62201a8d1b162116834f27fc6f1956')
+options+=(!lto)
+
+prepare() {
+  patch -Np1 -i fix-rvv-build.patch
+}
