--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,7 @@ pkgdesc='A PSP emulator written in C++'
 arch=(x86_64)
 url=https://www.ppsspp.org/
 license=(GPL-2.0-or-later)
+options=(!lto)
 makedepends=(
   clang
   cmake
@@ -92,6 +93,9 @@ prepare() {
     git config submodule.${submodule}.url ../../../armips-${submodule#*/}
     git -c protocol.file.allow=always submodule update ${submodule}
   done
+
+  cd $srcdir/ppsspp
+  patch -Np1 -i ../ppsspp-fix-build-for-sdl.patch
 }
 
 build() {
@@ -159,4 +163,7 @@ package_ppsspp-assets() {
   cp -dr --no-preserve=ownership build-sdl/assets "${pkgdir}"/usr/share/ppsspp/
 }
 
+source+=(ppsspp-fix-build-for-sdl.patch::https://github.com/hrydgard/ppsspp/pull/19840.patch)
+b2sums+=('e6632aad5077f353e30bf11d0d72d7a2d27cbd58f3658c4bfe82f0eb596927e3c269ff333a67eb1ebcab05e70e4aa45510ec14541ce881cd2d6d264977b29946')
+
 # vim: ts=2 sw=2 et:
