--- PKGBUILD
+++ PKGBUILD
@@ -56,9 +56,17 @@ replaces=(
 )
 source=(
   $url/archive/v$pkgver/$pkgname-v$pkgver.tar.gz
+  fix-tests.patch::https://github.com/ansible-community/molecule-plugins/commit/a7c74483bf7e36cf9c21d195a1917de38edf17fb.diff
 )
-sha512sums=('f9e50d0810ba8be3c393f48f101bf2db6f37374747204150237b3ba760abbbe1a562af4efb2c0804f40fd4635d1bfcdc324b02135e8954ec19614a05d4c1e53f')
-b2sums=('ee51a6f671043c75640a5b35c8207b8d0398cfd937d9ccd9bb1df2ae29a631c1fc2f72e58893605f75c88cdcad99fd75e85212d893583cc076751f778d1e2565')
+sha512sums=('f9e50d0810ba8be3c393f48f101bf2db6f37374747204150237b3ba760abbbe1a562af4efb2c0804f40fd4635d1bfcdc324b02135e8954ec19614a05d4c1e53f'
+            '18acf9cdced5fdb8ed5b7a2d4ea68ce2465a1cf4f4344260a9021e9e30b5d3cab6eec85899bf8e00091e3d995fcd58fc4bad7d26434f6e34a773be0215dea6a2')
+b2sums=('ee51a6f671043c75640a5b35c8207b8d0398cfd937d9ccd9bb1df2ae29a631c1fc2f72e58893605f75c88cdcad99fd75e85212d893583cc076751f778d1e2565'
+        '8ae7543c9cbe4aa7bb4f58701a7571c1060fd63a09c37665d879b0caaa3336b51d12bb908bbe2d675733e8fea26ea9e166f1ec0eaf93dffd72a32db169356462')
+
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../fix-tests.patch
+}
 
 build() {
   export SETUPTOOLS_SCM_PRETEND_VERSION=$pkgver
