--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('x86_64')
 license=('GPL')
 depends=('python-pycups' 'python-dbus' 'python-pycurl' 'libnotify'
          'python-gobject' 'gtk3' 'python-cairo')
-makedepends=('xmlto' 'docbook-xsl' 'desktop-file-utils' 'libsecret')
+makedepends=('xmlto' 'docbook-xsl' 'desktop-file-utils' 'libsecret' 'autoconf-archive')
 optdepends=('libsecret: password management'
             'python-pysmbc: SMB browser support'
             'cups-pk-helper: PolicyKit helper to configure cups with fine-grained privileges')
@@ -20,6 +20,12 @@ sha256sums=('dc9c8ad03f7983962ddf0ef05621c948370bd1763cd90c3dcff672280aa2d6e6')
 #            'SKIP')
 #validpgpkeys=('7ADB58203CA5F046F28025B215AA6A7F4D4227D7') # "Zdenek Dohnal (Associate Software Engineer) <zdohnal@redhat.com>"
 
+prepare() {
+  cd ${pkgname}-${pkgver}
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd ${pkgname}-${pkgver}
 
