diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -25,7 +25,7 @@ build() {
     -buildmode=pie \
     -mod=readonly \
     -modcacherw \
-    -ldflags "-linkmode external -extldflags ${LDFLAGS} \
+    -ldflags "-linkmode external -extldflags \"${LDFLAGS} -Wl,-plugin-opt=-target-abi=lp64d\" \
       -X github.com/prometheus/common/version.Version=$pkgver \
       -X github.com/prometheus/common/version.Revision=$pkgver \
       -X github.com/prometheus/common/version.Branch=tarball \
