diff --git PKGBUILD PKGBUILD
index de53c4d1..45eafa5f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,10 @@ depends=(gtk3 gjs libgweather geoclue2 gnome-desktop libhandy)
 makedepends=(gobject-introspection appstream-glib git meson)
 groups=(gnome)
 _commit=8eea7b1727c2b657e119f8729d7ebef77480326d  # tags/41.0^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-weather.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-weather.git#commit=$_commit"
+        "fix_positional_args.patch")
+sha256sums=('SKIP'
+            'b368ddd57eb03bde2d2e9b71a71314c3edb4d9c6b356ddcfd6f88c5fed122cef')
 
 pkgver() {
   cd $pkgname
@@ -21,6 +23,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../fix_positional_args.patch
 }
 
 build() {
