diff --git a/Sources/Core/System/detect_cpu_ext.cpp b/Sources/Core/System/detect_cpu_ext.cpp
index 5a1c73203..964c5d97e 100644
--- a/Sources/Core/System/detect_cpu_ext.cpp
+++ b/Sources/Core/System/detect_cpu_ext.cpp
@@ -32,7 +32,7 @@
 namespace clan
 {
 
-#if defined(ARM_PLATFORM) || defined(CL_ARM) || defined(__sun)
+#if defined(ARM_PLATFORM) || defined(CL_ARM) || defined(__sun) || defined(__riscv)
 	bool System::detect_cpu_extension(CPU_ExtensionPPC ext)
 	{
 		throw ("Congratulations, you've just been selected to code this feature!");
