--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,10 @@ install=ogre-next.install
 source=("$pkgname-$pkgver.tar.gz::https://github.com/OGRECave/ogre-next/archive/v${pkgver}.tar.gz")
 sha512sums=('52ed2d2a3375c0d35f0dc695b986514484ad1d47966c5c18351d3b09913123b2487b9729738c6b8b1219c1a992a8c8509a2303e097a6eb26497e152a14d48830')
 
+prepare() {
+  patch -Np1 -d ogre-next-${pkgver} < "${srcdir}/add-riscv64.patch"
+}
+
 build() {
   cd ogre-next-${pkgver}
 
@@ -25,6 +29,8 @@ build() {
     -DOGRE_CONFIG_ENABLE_JSON=ON \
     -DOGRE_CONFIG_THREADS=1 \
     -DOGRE_CONFIG_THREAD_PROVIDER="std" \
+    -DOGRE_SIMD_SSE2=OFF \
+    -DOGRE_SIMD_NEON=OFF \
     -DOGRE_BUILD_COMPONENT_PLANAR_REFLECTIONS=ON \
     -DOGRE_BUILD_COMPONENT_PROPERTY=ON \
     -DOGRE_BUILD_COMPONENT_SCENE_FORMAT=ON \
@@ -47,4 +53,7 @@ package() {
   install -Dm644 Docs/License.html "${pkgdir}"/usr/share/licenses/${pkgname}/License.html
 }
 
+source+=("add-riscv64.patch")
+sha512sums+=('e2c3e7abbbfc390accbb8cd783a2e45554cddee39f84c9894e6674bb31d6dc790971192958d65ef5c1ea54cd250328a05c18b569d3659936d7ea382f1d4277c8')
+
 # vim:set ts=2 sw=2 et:
