--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,10 @@ install=ogre-next.install
 source=("$pkgname-$pkgver.tar.gz::https://github.com/OGRECave/ogre-next/archive/v${pkgver}.tar.gz")
 sha512sums=('2ef8f16517c96cc7ddb31986857e4d0002e33c2eeff845b4af0b8e5848c3e92289dc3b10ededbe66fb63ef6234cbee88ed513466182bd4e70d710d0507f98418')
 
+prepare() {
+  patch -Np1 -d ogre-next-${pkgver} < "${srcdir}/add-riscv64.patch"
+}
+
 build() {
   cd ogre-next-${pkgver}
 
@@ -25,6 +29,8 @@ build() {
     -DOGRE_CONFIG_ENABLE_JSON=ON \
     -DOGRE_CONFIG_THREADS=1 \
     -DOGRE_CONFIG_THREAD_PROVIDER="std" \
+    -DOGRE_SIMD_SSE2=OFF \
+    -DOGRE_SIMD_NEON=OFF \
     -DOGRE_BUILD_COMPONENT_PLANAR_REFLECTIONS=ON \
     -DOGRE_BUILD_COMPONENT_PROPERTY=ON \
     -DOGRE_BUILD_COMPONENT_SCENE_FORMAT=ON \
@@ -47,4 +53,7 @@ package() {
   install -Dm644 Docs/License.html "${pkgdir}"/usr/share/licenses/${pkgname}/License.html
 }
 
+source+=("add-riscv64.patch::https://github.com/OGRECave/ogre-next/pull/478.diff")
+sha512sums+=('9859a828dc560e49b570ec887853cbbb5467998348fc1ea253ed6032bebeb41d51fc6f1d4e7ea667b23eeb9c7a44c3ba3c8708e01df02c173ac28086c7f94faa')
+
 # vim:set ts=2 sw=2 et:
