diff --color -Naur giada-0.18.2-src-orig/CMakeLists.txt giada-0.18.2-src/CMakeLists.txt
--- giada-0.18.2-src-orig/CMakeLists.txt	2021-09-12 20:56:28.000000000 +0800
+++ giada-0.18.2-src/CMakeLists.txt	2021-11-05 23:00:12.970247800 +0800
@@ -260,8 +260,9 @@
 
 # Threads (system)
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
-list(APPEND LIBRARIES ${Threads_LIBRARY})
+list(APPEND LIBRARIES Threads::Threads)
 
 # pkg-config/pkgconf, required to find some external dependencies on some
 # platforms
