diff --git PKGBUILD PKGBUILD
index d5c0f39..4cf0d9d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,15 +12,18 @@ arch=('x86_64')
 url='https://fbb-git.gitlab.io/icmake/'
 license=('GPL3')
 source=("https://gitlab.com/fbb-git/${pkgname}/-/archive/${pkgver}/${pkgname}-${pkgver}.tar.gz"
-        'disable-clear-screen.patch')
+        'disable-clear-screen.patch'
+        $pkgname-gcc13.patch::https://gitlab.com/fbb-git/icmake/-/commit/47c2d61ce2a3aaec7f7102423e4202667c8e76ff.patch)
 b2sums=('4711dee5d66882afeeb6e7206b97341690e2e1c0e511fcf9f43d41ec04bec2ecf5fee10113bba47475b3166dc8dc53f7b236c42dcf0969582a59bbf30b7ca618'
-        '6689ce74a985742e079f95bfaa22dcee22ad667f45cf21f79d4c3d2271f925f53bcd6729ca762a7cfa6a63af9b0fd48c2b9c024bd35ca580e83a6a54758a7bed')
+        '6689ce74a985742e079f95bfaa22dcee22ad667f45cf21f79d4c3d2271f925f53bcd6729ca762a7cfa6a63af9b0fd48c2b9c024bd35ca580e83a6a54758a7bed'
+        '8c80ee554855f3466ad89006a6d5f73413f069c456be47cf40bd50586de95bbb3a0f253a0aca99953e58cd27f1be5f076ecc3c9778d2f56997bdba8cc83591a5')
 makedepends=('icmake' 'yodl')
 depends=('libbobcat' 'libunwind')
 
 prepare() {
 	cd "${srcdir}/${pkgname}-${pkgver}"
     patch -Np1 -i "${srcdir}/disable-clear-screen.patch"
+    patch -Np1 -i "${srcdir}/${pkgname}-gcc13.patch"
 
     # fix paths
     sed -i 's|usr/libexec/icmake|usr/lib/icmake|' icmake/INSTALL.im
