--- PKGBUILD
+++ PKGBUILD
@@ -33,3 +33,10 @@ build() {
 package() {
     make -C "${pkgname}-${pkgver}" DESTDIR="$pkgdir" install
 }
+
+prepare(){
+    cd "${pkgname}-${pkgver}"
+    patch -Np1 -i ../fix-ISO-C++17-does-not-allow-dynamic-exception-specifications.patch
+}
+source+=('fix-ISO-C++17-does-not-allow-dynamic-exception-specifications.patch')
+sha256sums+=('aa65913c75abcaaa4734baf8a9774f76f7eee02497787cc2c9bd357de8737b01')
