diff --git PKGBUILD PKGBUILD
index d0d0886..77f9c71 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,15 @@
          'hicolor-icon-theme')
 optdepends=('fs-uae-launcher: for the graphical user interface')
 makedepends=('libxi' 'zip')
-source=("https://fs-uae.net/files/FS-UAE/Stable/${pkgver}/${pkgname}-${pkgver}.tar.xz")
-sha256sums=('606e1868b500413d69bd33bb469f8fd08d6c08988801f17b7dd022f3fbe23832')
+source=("https://fs-uae.net/files/FS-UAE/Stable/${pkgver}/${pkgname}-${pkgver}.tar.xz"
+        "fix-ISO-C++17-does-not-allow-dynamic-exception-specifications.patch")
+sha256sums=('606e1868b500413d69bd33bb469f8fd08d6c08988801f17b7dd022f3fbe23832'
+            'aa65913c75abcaaa4734baf8a9774f76f7eee02497787cc2c9bd357de8737b01')
+
+prepare(){
+    cd "${pkgname}-${pkgver}"
+    patch -Np1 -i ../fix-ISO-C++17-does-not-allow-dynamic-exception-specifications.patch
+}
 
 build() {
     cd "${pkgname}-${pkgver}"
