--- a/src/poetry/utils/env/mock_env.py
+++ b/src/poetry/utils/env/mock_env.py
@@ -16,7 +16,7 @@
 class MockEnv(NullEnv):
     def __init__(
         self,
-        version_info: tuple[int, int, int] | PythonVersion = (3, 7, 0),
+        version_info: tuple[int, int, int] | PythonVersion = (3, 9, 0),
         *,
         python_implementation: str = "CPython",
         platform: str = "darwin",
--- a/tests/console/commands/debug/test_info.py
+++ b/tests/console/commands/debug/test_info.py
@@ -23,7 +23,10 @@
     mocker.patch(
         "poetry.utils.env.EnvManager.get",
         return_value=MockEnv(
-            path=Path("/prefix"), base=Path("/base/prefix"), is_venv=True
+            version_info=(3, 7, 0),
+            path=Path("/prefix"),
+            base=Path("/base/prefix"),
+            is_venv=True,
         ),
     )
     mocker.patch(
--- a/tests/console/commands/env/test_info.py
+++ b/tests/console/commands/env/test_info.py
@@ -22,7 +22,10 @@
     mocker.patch(
         "poetry.utils.env.EnvManager.get",
         return_value=MockEnv(
-            path=Path("/prefix"), base=Path("/base/prefix"), is_venv=True
+            version_info=(3, 7, 0),
+            path=Path("/prefix"),
+            base=Path("/base/prefix"),
+            is_venv=True,
         ),
     )
 
