--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,7 @@
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -Np1 -i "$srcdir/wait-for-output-state.patch"
   sed 's|/usr/local|/usr|' -i resources/$pkgname.service
   cargo fetch --locked --target "$(rustc --print host-tuple)"
 }
@@ -45,3 +46,7 @@
   install -vDm 644 resources/$pkgname.service -t "$pkgdir/usr/lib/systemd/user/"
   install -vDm 644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgdir/"
 }
+
+source+=(wait-for-output-state.patch)
+sha512sums+=('548efbf646711a2b0053f769b6167904c4947701284d4653040fb895920e269ca4f1b364a5baac8d634fad0717721ca29f393f21598b47b2e0e73e0653330c1c')
+b2sums+=('651d1ae749a68829fd512f91545fda0e84fa33ed7a7868d2740403e43d21dfb92aa91e7ef275fae49a18d2051dcf93d4bba1e986d16c2fc18fae0d08e58633dd')
