Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1030926)
+++ PKGBUILD	(working copy)
@@ -9,11 +9,14 @@
 url='http://www.hyperrealm.com/main.php?s=talkfilters'
 depends=('glibc')
 license=('GPL2')
-source=("http://www.smallperturbation.com/sites/default/files/$pkgname-$pkgver.tar.gz")
-sha256sums=('4681e71170af06c6bffcd4e454eff67224cde06f0d678d26dc72da45f02ecca6')
+source=("http://www.smallperturbation.com/sites/default/files/$pkgname-$pkgver.tar.gz"
+        "fix-format-security.patch")
+sha256sums=('4681e71170af06c6bffcd4e454eff67224cde06f0d678d26dc72da45f02ecca6'
+            '27301a2d42de42f5949adb93afc82cd1c27fea07e7d7094c594e4f9eb5f1e3ae')
 
 build() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../fix-format-security.patch
 
   ./configure --prefix=/usr
   make
