--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,14 @@ depends=('glibc')
 license=('GPL2')
 source=("http://www.smallperturbation.com/sites/default/files/$pkgname-$pkgver.tar.gz")
 sha256sums=('4681e71170af06c6bffcd4e454eff67224cde06f0d678d26dc72da45f02ecca6')
 
+prepare() {
+  cd "$pkgname-$pkgver"
+
+  autoreconf -fiv
+}
+
 build() {
   cd "$pkgname-$pkgver"
 
   CFLAGS="-Wno-format-security" ./configure --prefix=/usr
