diff --git PKGBUILD PKGBUILD
index d6b529e..2ed09ae 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,11 +9,14 @@ arch=('x86_64')
 url='http://www.hyperrealm.com/main.php?s=talkfilters'
 depends=('glibc')
 license=('GPL2')
-source=("http://www.smallperturbation.com/sites/default/files/$pkgname-$pkgver.tar.gz")
-sha256sums=('4681e71170af06c6bffcd4e454eff67224cde06f0d678d26dc72da45f02ecca6')
+source=("http://www.smallperturbation.com/sites/default/files/$pkgname-$pkgver.tar.gz"
+        "fix-format-security.patch")
+sha256sums=('4681e71170af06c6bffcd4e454eff67224cde06f0d678d26dc72da45f02ecca6'
+            '7505aea9fe03e8e4fb8da9dc202a10d7593cfacf6cd0f3f0dda56d988856e9aa')
 
 build() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../fix-format-security.patch
 
   ./configure --prefix=/usr
   make
