diff --git PKGBUILD PKGBUILD
index a27dca2..54f1e89 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,12 +12,15 @@
 depends=('qt5-base' 'qwt')
 makedepends=('cmake' 'boost')
 source=("$_pkgname-$pkgver.tar.gz::$url/archive/$_tag.tar.gz"
-         srsgui-qwt-6.2.patch)
+         srsgui-qwt-6.2.patch
+         remove-deprecated-and-unused-boost-progress.patch::https://github.com/srsran/srsGUI/pull/21.diff)
 sha512sums=('9d766a80fc81b015f5c8dbed3fa97d417496a0e4db11bd23a66bae4c3a2d7e0206f2569af8884d01b40702416dab6add52cc0ef04f29b176aba99de00466a88d'
-            '1097681e57d0548db36db9ab1331e6e96554b44618b21ee08631e188e092796d57ec49c0456055b633026246eb652f75cd7470c60bb530ebf246d92cd9f28b5e')
+            '1097681e57d0548db36db9ab1331e6e96554b44618b21ee08631e188e092796d57ec49c0456055b633026246eb652f75cd7470c60bb530ebf246d92cd9f28b5e'
+            'c7b889bc87a1f86298289581f55edfff3144724ff23c22632ddda2e45ee18e7563585b047a7123261e7ed61bc49cb4575b9291ceaceea7618c2a916621da0ea8')
 
 prepare() {
   patch -d $_pkgname-$_tag -p1 < srsgui-qwt-6.2.patch # Fix build with qwt 6.2
+  patch -d $_pkgname-$_tag -p1 < remove-deprecated-and-unused-boost-progress.patch # Fix build with boost 1.71
 }
 
 build() {
@@ -37,4 +40,3 @@
 
   make DESTDIR="$pkgdir" install
 }
-
