diff --git a/PKGBUILD b/PKGBUILD
index 5ab7ba4..26f7b51 100644
--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,8 @@ sha256sums=('5f10d6d718a4a51c17ed1b32b031d4f9b80b061e8276535b2be31e5ac4b75e6f')
 
 build() {
     cd "${pkgname}-${pkgver}"
+    cp /usr/share/autoconf/build-aux/config.guess .
+    cp /usr/share/autoconf/build-aux/config.sub .
     ./configure \
         --prefix='/usr' \
         --enable-antialiasing \
