diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 2326561..e3f0523 100644
--- PKGBUILD
+++ PKGBUILD
@@ -37,10 +37,12 @@ makedepends=('mesa' 'libdvdnav' 'libdvdread' 'vcdimager' 'glu' 'sdl'
              'libmpcdec' 'wavpack' 'gdk-pixbuf2' 'imagemagick' 'libcaca'
              'libmng' 'libnfs' 'smbclient')
 source=("https://downloads.sourceforge.net/project/xine/xine-lib/${pkgver}/xine-lib-${pkgver}.tar.xz")
-sha256sums=('3def43d3c0597f0f23cbf73e9d4b1dcd328852f2100fcff89fe925ad7cdd0251')
+sha256sums=('5f10d6d718a4a51c17ed1b32b031d4f9b80b061e8276535b2be31e5ac4b75e6f')

 build() {
-    cd "${pkgname}.${pkgver}"
+    cd "${pkgname}-${pkgver}"
+    autoreconf -fi
+    autoupdate -f
     ./configure \
         --prefix='/usr' \
         --enable-antialiasing \
@@ -51,5 +53,5 @@ build() {
 }

 package() {
-    make -C "${pkgname}.${pkgver}" DESTDIR="$pkgdir" install
+    make -C "${pkgname}-${pkgver}" DESTDIR="$pkgdir" install
 }
