--- PKGBUILD
+++ PKGBUILD
@@ -35,8 +35,16 @@ makedepends=('mesa' 'libdvdnav' 'libdvdread' 'vcdimager' 'glu' 'sdl'
              'wayland' 'aalib' 'a52dec' 'faad2' 'flac' 'libdca' 'libmad'
              'libmpcdec' 'wavpack' 'gdk-pixbuf2' 'imagemagick' 'libcaca'
              'libmng' 'libnfs' 'smbclient')
-source=("https://downloads.sourceforge.net/project/xine/xine-lib/${pkgver}/xine-lib-${pkgver}.tar.xz")
-sha256sums=('d606270468e1540c2a89c0d7f5fdf11e17ecc0c2698cc0bcb1065ff26abee098')
+source=("https://downloads.sourceforge.net/project/xine/xine-lib/${pkgver}/xine-lib-${pkgver}.tar.xz"
+        fix-david.patch)
+sha256sums=('d606270468e1540c2a89c0d7f5fdf11e17ecc0c2698cc0bcb1065ff26abee098'
+            '6d9d12a85be0a2585ca124d4904b6f8a34cb574044e583e52eb104e646c15629')
+
+prepare() {
+    cd "${pkgname}-${pkgver}"
+
+    patch -Np0 -i ../fix-david.patch
+}
 
 build() {
     cd "${pkgname}-${pkgver}"
