--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,12 @@ makedepends=(git)
 source=(git+https://github.com/coin-or/Cgl#tag=releases/$pkgver)
 sha256sums=('b24a06071a18fd10bbda07b90aeb55a88863bd4c145240950869d527cc614097')
 
+prepare() {
+  cd Cgl
+  autoconf_file="/usr/share/autoconf/build-aux/config"
+  echo ". Cgl" | xargs -r -n1 cp ${autoconf_file}.{guess,sub}
+}
+
 build() {
   cd Cgl
   ./configure --prefix=/usr
