--- PKGBUILD
+++ PKGBUILD
@@ -48,6 +48,9 @@
 prepare() {
   cd "${pkgname}-${pkgver}"

+  # Select a foreign test target even when the host is riscv64.
+  patch -Np1 -i ../swift-foreign-target-test.patch
+
   cargo fetch --locked --target "$(rustc --print host-tuple)"
 }

@@ -93,4 +96,8 @@
   install --verbose -D --mode=0644 completions/_mise "${pkgdir}/usr/share/zsh/site-functions/_mise"
 }

+source+=(swift-foreign-target-test.patch)
+sha512sums+=('4aa160ba79549849840658fc5e0de75b5f364dcc66635dc80c6b2d2ee61205925b2f1055e2a8f4f5a39d51de9c00c37745b3c7a5e84c054aa4d61526b0622545')
+b2sums+=('f278718cf145213e9f9020a5a06777ba5d189967663030e04494b45e2dde23c741677c8b38220d347bfbcb44b7ac7c632a4fb97d9113dab76aad4f7cacc33140')
+
 # vim: tabstop=2 shiftwidth=2 expandtab:
