--- PKGBUILD
+++ PKGBUILD
@@ -48,6 +48,9 @@
 prepare() {
   cd "${pkgname}-${pkgver}"

+  # Select a foreign test target even when the host is riscv64.
+  patch -Np1 -i ../swift-foreign-target-test.patch
+
   cargo fetch --locked --target "$(rustc --print host-tuple)"
 }

@@ -93,4 +96,8 @@
   install --verbose -D --mode=0644 completions/_mise "${pkgdir}/usr/share/zsh/site-functions/_mise"
 }

+source+=(swift-foreign-target-test.patch)
+sha512sums+=('5a1c1fb9702931eef71e706c38637021ca37b3488b68c50d935fcdc3225a706106a577d86124d3fd97eb39daf3429d9980fc6b64b40acbde9d6d3933242f52db')
+b2sums+=('f50dda7c3e9f999e38b12127ab62dd096a5c68f06d146fb80a2042383e4ee71ddbe15b8749090d257b044aec532a38f7197863ce0f36c9423ec0616151d8bfa2')
+
 # vim: tabstop=2 shiftwidth=2 expandtab:
