--- a/tests/test_continuous_modulator.py
+++ b/tests/test_continuous_modulator.py
@@ -22,7 +22,7 @@
         self.assertTrue(continuous_modulator.ring_buffer.is_empty)
         continuous_modulator.start()
         self.assertTrue(continuous_modulator.process.is_alive())
-        time.sleep(2)
+        time.sleep(10)
         self.assertFalse(continuous_modulator.ring_buffer.is_empty)
         continuous_modulator.stop()
         self.assertFalse(continuous_modulator.process.is_alive())
--- a/tests/test_send_recv_dialog_gui.py
+++ b/tests/test_send_recv_dialog_gui.py
@@ -267,7 +267,7 @@
         process.daemon = True
         process.start()
         n = 0
-        while ready.value == 0 and n < 50:  # ensure server is up
+        while ready.value == 0 and n < 300:  # ensure server is up
             time.sleep(0.1)
             n += 1
 
