diff --git PKGBUILD PKGBUILD
index 10016b2ae..aa4572ece 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,10 @@ depends=(gjs gtk3 libunistring gnome-desktop emoji-font libhandy)
 makedepends=(meson gobject-introspection appstream-glib git gperf)
 groups=(gnome)
 _commit=05b7479685429c9a033c8fa801cb1e83402ae5c4  # tags/41.0^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-characters.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-characters.git#commit=$_commit"
+        "fix_positional_arg.patch")
+sha256sums=('SKIP'
+            '1b6bca2767e4d52235acdcb7538ae4a888412e75b3bc678fde16e5e5093461fb')
 
 pkgver() {
   cd $pkgname
@@ -22,6 +24,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "../fix_positional_arg.patch"
 }
 
 build() {
