Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 455744)
+++ PKGBUILD	(working copy)
@@ -78,7 +78,7 @@
  ./configure --prefix=/usr \
              --enable-qt5 \
              --disable-qt4 \
-             --enable-hpcups-install \
+             --disable-hpcups-install \
              --enable-cups-drv-install \
              --disable-imageProcessor-build \
              --enable-pp-build #--help
