--- PKGBUILD
+++ PKGBUILD
@@ -76,6 +76,9 @@ prepare() {
  # make gzip creation reproducible by removing the timestamp
  patch -Np1 -i ../reproducible-gzip.patch
 
+ # don't link unused ImageProcessor library because --disable-imageProcessor-build
+ patch -Np1 -i ../Prevent-linking-ImageProcessor.patch
+
  export AUTOMAKE='automake --foreign'
  autoreconf --force --install
 }
@@ -115,3 +118,6 @@ package() {
  python -m compileall -d /usr/share "$pkgdir/usr/share"
  python -O -m compileall -d /usr/share "$pkgdir/usr/share"
 }
+
+source+=("Prevent-linking-ImageProcessor.patch")
+sha512sums+=('f361fbf6ca84a99e17411716c0bfc9d283ea5ecc13517c70b89c2931cb2b0f54a423983383d0f3f25d8bb83813078c9d3aeb78178be77e80c7329046918c5c32')
