--- PKGBUILD
+++ PKGBUILD
@@ -31,12 +31,11 @@
 options=(!lto)
 
 build() {
-  export CFLAGS+=' -I/usr/include/minizip'
-  export CXXFLAGS+=' -I/usr/include/minizip'
+  export CFLAGS+=' -I/usr/include/minizip -DNO_ASM'
+  export CXXFLAGS+=' -I/usr/include/minizip -DNO_ASM'
   make \
-    WITH_DYNAREC=x86_64 \
     HAVE_PARALLEL_RDP=1 \
-    HAVE_PARALLEL_RSP=1 \
+    HAVE_PARALLEL_RSP=0 \
     HAVE_THR_AL=1 \
     SYSTEM_LIBPNG=1 \
     SYSTEM_MINIZIP=1 \
