--- build/config/clang/BUILD.gn.orig	2025-04-07 06:28:07.000000000 +0000
+++ build/config/clang/BUILD.gn	2025-04-07 06:41:04.761286863 +0000
@@ -197,6 +197,8 @@
           _dir = "armv7-unknown-linux-gnueabihf"
         } else if (current_cpu == "arm64") {
           _dir = "aarch64-unknown-linux-gnu"
+        } else if (current_cpu == "riscv64") {
+          _dir = "riscv64-unknown-linux-gnu"
         } else {
           assert(false)  # Unhandled cpu type
         }
