diff --git a/cmake/Functions.cmake b/cmake/Functions.cmake
index fef9c15e1..8c00d78ca 100644
--- a/cmake/Functions.cmake
+++ b/cmake/Functions.cmake
@@ -609,6 +609,10 @@ function(get_gn_arch result arch)
         set(${result} "mipsel" PARENT_SCOPE)
     elseif(arch STREQUAL "mipsel64")
         set(${result} "mips64el" PARENT_SCOPE)
+    elseif(arch STREQUAL "riscv32")
+        set(${result} "riscv32" PARENT_SCOPE)
+    elseif(arch STREQUAL "riscv64")
+        set(${result} "riscv64" PARENT_SCOPE)
     else()
         message(DEBUG "Unsupported architecture: ${arch}")
     endif()
