--- PKGBUILD
+++ PKGBUILD
@@ -91,6 +91,13 @@ prepare() {
 
 # Disable FFmpeg allow lists https://bugreports.qt.io/browse/QTBUG-132762
   patch -d src/3rdparty -p1 < ../disable-ffmpeg-allow-lists.patch
+
+  for _patch in sandbox dav1d; do
+    patch -d src/3rdparty/chromium -Np1 < ../riscv-$_patch.patch
+  done
+  patch -d src/3rdparty/chromium    -Np1 < ../unscaledcycleclock-remove-riscv-support.patch
+  patch -d src/3rdparty/chromium/v8 -Np1 < ../Skip-check-sv57-when-enable-pointer-compress.patch
+  patch -d src/3rdparty/chromium/v8 -Np1 < ../avoid-cpu-probing-in-li_ptr.patch
 }
 
 build() {
@@ -103,7 +110,10 @@ build() {
     -DQT_FEATURE_webengine_system_re2=ON \
     -DQT_FEATURE_webengine_proprietary_codecs=ON \
     -DQT_FEATURE_webengine_kerberos=ON \
-    -DQT_FEATURE_webengine_webrtc_pipewire=ON
+    -DQT_FEATURE_webengine_webrtc_pipewire=ON \
+    -DCMAKE_C_COMPILER=/usr/bin/clang \
+    -DCMAKE_CXX_COMPILER=/usr/bin/clang++
+  ninja -C build -t targets | grep '^runGn' | cut -d ':' -f 1 | xargs ninja -C build
   cmake --build build
 }
 
@@ -112,3 +122,14 @@ package() {
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+makedepends+=(clang)
+source+=(riscv-{sandbox,dav1d}.patch
+         "https://github.com/riscv-forks/electron/raw/1e16216b089b6aaed055d17608667e6582263cfd/patches/v8/avoid-cpu-probing-in-li_ptr.patch"
+         "https://github.com/riscv-forks/electron/raw/1e16216b089b6aaed055d17608667e6582263cfd/patches/v8/Skip-check-sv57-when-enable-pointer-compress.patch"
+         "https://github.com/riscv-forks/electron/raw/4eff53436a1a86ac548a107e21ca5078518833c0/patches/chromium/unscaledcycleclock-remove-riscv-support.patch")
+sha256sums+=('8d52d4da703c8a86059418d1a4ed63d2d6bc1134e9dfe569695a830479a9afae'
+             '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e'
+             'faaf1af670ab202f2e531b19c0af81e97b56afffdcd9f58afd33f0f65468f119'
+             '6a3ad2b09fe28be4bd3aa922d071b973cf52531c447b31a7415a02f467a795a0'
+             'bea6346ab4fc88061d3a657281e46eb14c1ce7fcfe108deda02b47bee50241aa')
