--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,20 @@ makedepends=(cmake ninja python-html5lib gperf jsoncpp qt6-tools pipewire nodejs
 optdepends=('pipewire: WebRTC desktop sharing under Wayland')
 groups=(qt6)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
-sha256sums=('e7c9438b56f502b44b4e376b92ed80f1db7c2c3881d68d319b0677afd5701d9f')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
+        riscv-{angle,breakpad,crashpad,dav1d,sandbox}.patch)
+sha256sums=('e7c9438b56f502b44b4e376b92ed80f1db7c2c3881d68d319b0677afd5701d9f'
+            'dfb1b19a25341c54bac3b761cec0c3853d9cd2b7b54c11bde97bf016a6dafd57'
+            'bbd83a325b3ee6a703386eac0dff5beeaaa0b2ecc2393ce4683b7bc7ca08fa7e'
+            '71db9c9fa85767a34baa12cb06580cf04c069ef0c9690dd7cfce09f11d1eec58'
+            '9de2326157f22a7d304cf7f70e74827cbffc69c5d2aecdbd30eb68c0b3cfecf1'
+            'd73e386237f9c068d24f3f18f87b17d5f39564f089242f976dd371de7cdf832e')
+
+prepare() {
+  for _patch in angle breakpad crashpad dav1d sandbox; do
+    patch -d $_pkgfn/src/3rdparty -p0 < riscv-$_patch.patch
+  done
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
