--- PKGBUILD
+++ PKGBUILD
@@ -76,6 +76,10 @@ sha256sums=('7a6ea228214bd66029ca90549b29021f30f7544abff997b7f831ceac2ce73691'
             '547e092f6a20ebd15e486b31111145bc94b8709ec230da89c591963001378845')
 
 prepare() {
+  for _patch in angle crashpad libgav1 sandbox base dav1d libyuv; do
+    patch -d $_pkgfn/src/3rdparty/chromium -Np1 < riscv-$_patch.patch
+  done
+
   patch -d $_pkgfn/src/3rdparty/chromium -p1 < libxml-2.12.patch
   patch -d $_pkgfn/src/3rdparty/chromium -p1 < icu-74.patch # Fix build with ICU 74 - patch from Alpine
 }
@@ -89,7 +93,10 @@ build() {
     -DQT_FEATURE_webengine_system_libevent=ON \
     -DQT_FEATURE_webengine_proprietary_codecs=ON \
     -DQT_FEATURE_webengine_kerberos=ON \
-    -DQT_FEATURE_webengine_webrtc_pipewire=ON
+    -DQT_FEATURE_webengine_webrtc_pipewire=ON \
+    -DCMAKE_C_COMPILER=/usr/bin/clang \
+    -DCMAKE_CXX_COMPILER=/usr/bin/clang++
+  ninja -C build -t targets | grep '^runGn' | cut -d ':' -f 1 | xargs ninja -C build
   cmake --build build
 }
 
@@ -98,3 +105,14 @@ package() {
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+makedepends=("${makedepends[@]/nodejs/nodejs-lts-iron}" clang)
+source+=(riscv-{angle,crashpad,libgav1,sandbox,base,dav1d,libyuv}.patch)
+sha256sums+=('fb0738fc32f228cef59e5516dbb4acd8818f1e2ba9db2cfbc21cf72a4ace8c3f'
+             '70f8bcd3dd68c13d1e074415952c0b7e3e3b6058c97d39ba7a78dc17daa30547'
+             '4b2dcfdeb8ab32130d220d9827f49a930cf748bf4d6c7aef97a7b36a98316430'
+             '3fc1c003675dfc336f5a61eadb2009d558d5ceef0113c46c1d38f39506835304'
+             'da17695cc437eb3f2a71719772d29ee9d0b6f80152536b061208ab29253fca7c'
+             '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e'
+             'd71a8dd17e3acb8f76032a35a6e633ddb12304194f36b6fc7bc24b4f0867f0bf')
+
