--- a/third_party/libyuv/BUILD.gn
+++ b/third_party/libyuv/BUILD.gn
@@ -32,6 +32,10 @@
   if (!libyuv_use_neon) {
     defines = [ "LIBYUV_DISABLE_NEON" ]
   }
+
+  if (libyuv_disable_rvv) {
+    defines += [ "LIBYUV_DISABLE_RVV" ]
+  }
 }
 
 # This target is built when no specific target is specified on the command line.
--- a/third_party/libyuv/libyuv.gni
+++ b/third_party/libyuv/libyuv.gni
@@ -13,6 +13,7 @@
 declare_args() {
   libyuv_include_tests = !build_with_chromium
   libyuv_disable_jpeg = false
+  libyuv_disable_rvv = true
   libyuv_use_neon =
       current_cpu == "arm64" ||
       (current_cpu == "arm" && (arm_use_neon || arm_optionally_use_neon))
