--- PKGBUILD
+++ PKGBUILD
@@ -93,6 +93,13 @@
   # Bump chromium to head of stable branch
   cd src/3rdparty
   [[ -n $_chromium ]] && git checkout $_chromium || true
+
+  # Qt's Chromium snapshot has RISC-V gaps in sandbox, dav1d, and V8.
+  for _patch in sandbox dav1d v8; do
+    patch -d chromium -Np1 < "$srcdir"/riscv-$_patch.patch
+  done
+  patch -d chromium -Np1 < "$srcdir"/fix-linux-syscall-ranges.patch
+  patch -d chromium -Np1 < "$srcdir"/blink-webaudio-delay-complete-upstream-span-fix.patch
 }
 
 build() {
@@ -105,7 +112,10 @@
     -DQT_FEATURE_webengine_system_openh264=ON \
     -DQT_FEATURE_webengine_proprietary_codecs=ON \
     -DQT_FEATURE_webengine_kerberos=ON \
-    -DQT_FEATURE_webengine_webrtc_pipewire=ON
+    -DQT_FEATURE_webengine_webrtc_pipewire=ON \
+    -DCMAKE_C_COMPILER=/usr/bin/clang \
+    -DCMAKE_CXX_COMPILER=/usr/bin/clang++
+  ninja -C build -t targets | grep '^runGn' | cut -d ':' -f 1 | xargs ninja -C build
   cmake --build build
 }
 
@@ -114,3 +124,13 @@
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+makedepends+=(clang)
+source+=(riscv-{sandbox,dav1d,v8}.patch
+         fix-linux-syscall-ranges.patch::https://github.com/rockos-riscv/chromium-129.0.6668.100/commit/f699f4ed684e413e508e1f3c5e30d806473ff13b.patch
+         blink-webaudio-delay-complete-upstream-span-fix.patch)
+sha256sums+=('5819a62edfc654c40e5d2f3d77e17d2a13b0c6faa7d2a6b1674c469657d5dd9a'
+             '15cca2c973fc989cd10b9ca4b7e555747b262b64adf9136ae2b3d1758cf88de5'
+             '11c25d3bff5ee801273515a93fb1dd82113c385124eee9263bcb0f8ad5ca8419'
+             '0ac16adaebe4e3c347ab67647ae0e1ff1db1d710d0b64b469ab77d27eb9f4192'
+             'a904766fcb502a30a8f50b9fc580d717310e56d5acd6f96312a8f890ede56fce')
