--- PKGBUILD
+++ PKGBUILD
@@ -105,6 +105,13 @@ prepare() {
   git -C src/3rdparty cherry-pick -n 416c7ff1efbb6565b109ff1462bc84a6870e13b9
 
   patch -d src/3rdparty/chromium/third_party/webrtc -p1 < ../pipewire-1.4.patch
+
+  for _patch in sandbox dav1d; do
+    patch -d src/3rdparty/chromium -Np1 < ../riscv-$_patch.patch
+  done
+  patch -d src/3rdparty/chromium/v8 -Np1 < ../0001-riscv-Flush-icache-in-both-local-and-remote-harts.patch
+  patch -d src/3rdparty/chromium -Np0 < ../compiler-rt-riscv.patch
+  patch -d src/3rdparty/chromium -Np1 < ../fix-linux-syscall-ranges.patch
 }
 
 build() {
@@ -117,7 +124,10 @@ build() {
     -DQT_FEATURE_webengine_system_re2=ON \
     -DQT_FEATURE_webengine_proprietary_codecs=ON \
     -DQT_FEATURE_webengine_kerberos=ON \
-    -DQT_FEATURE_webengine_webrtc_pipewire=ON
+    -DQT_FEATURE_webengine_webrtc_pipewire=ON \
+    -DCMAKE_C_COMPILER=/usr/bin/clang \
+    -DCMAKE_CXX_COMPILER=/usr/bin/clang++
+  ninja -C build -t targets | grep '^runGn' | cut -d ':' -f 1 | xargs ninja -C build
   cmake --build build
 }
 
@@ -126,3 +136,14 @@ package() {
 
   install -Dm644 "$srcdir"/${_pkgfn}/src/3rdparty/chromium/LICENSE "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.chromium
 }
+
+makedepends+=(clang)
+source+=(riscv-{sandbox,dav1d}.patch
+         https://raw.githubusercontent.com/riscv-forks/electron/refs/heads/v33.4.8-riscv/patches/v8/0001-riscv-Flush-icache-in-both-local-and-remote-harts.patch
+         compiler-rt-riscv.patch
+         fix-linux-syscall-ranges.patch::https://github.com/rockos-riscv/chromium-129.0.6668.100/commit/f699f4ed684e413e508e1f3c5e30d806473ff13b.patch)
+sha256sums+=('1713cfc3c73d7f33fd7a9daba9b642869632468bc1068b727827a6b5320a7f88'
+             '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e'
+             '72c4ee0fe374c6a8494c3c54494f551ce54957314b6e76f8b4c4a5b449fdf438'
+             'cc119391ef129ba85b636a5deb47c06464b76fcd3be14ac29c8ab78bf14fa780'
+             '0ac16adaebe4e3c347ab67647ae0e1ff1db1d710d0b64b469ab77d27eb9f4192')
