--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,18 @@ optdepends=('pipewire: WebRTC desktop sharing under Wayland')
 groups=(qt6)
 options=(debug)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
-sha256sums=('55f85af736a1dc79a41b8d95014ba27d8fce0be44293a69e64fece7fa12b2925')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
+        qt6-webengine-chromium-riscv64.patch
+        qt6-webengine-cmake-riscv64.patch)
+sha256sums=('55f85af736a1dc79a41b8d95014ba27d8fce0be44293a69e64fece7fa12b2925'
+            '9d745072a01417a39f1ec532fcee26cf8a214332e7cea29e94d61684d10de377'
+            'fdf8bb4f09adefa432ed7ab3b0e2f375f5b99fcb80d39415f3c574cbe6b71fb2')
+
+prepare() {
+  # ported from openSUSE (see https://build.opensuse.org/package/show/openSUSE:Factory:RISCV/chromium)
+  patch -d $_pkgfn/src/3rdparty -p0 < qt6-webengine-chromium-riscv64.patch # Fix riscv64 build for Chromium
+  patch -d $_pkgfn -p1 < qt6-webengine-cmake-riscv64.patch # Fix Qt's CMake architecture detection
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
