--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,18 @@ optdepends=('pipewire: WebRTC desktop sharing under Wayland')
 groups=(qt6)
 options=(debug)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
-sha256sums=('65b4397f451650226142f35c8d6c6ebc45cee461b1c2d17d688208d455e24426')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
+        qt6-webengine-chromium-riscv64.patch
+        qt6-webengine-cmake-riscv64.patch)
+sha256sums=('65b4397f451650226142f35c8d6c6ebc45cee461b1c2d17d688208d455e24426'
+            '7bef378ec8b0298b0f35895351762ab324f1ff258eeea162b0dde8818436acba'
+            'fdf8bb4f09adefa432ed7ab3b0e2f375f5b99fcb80d39415f3c574cbe6b71fb2')
+
+prepare() {
+  # ported from openSUSE (see https://build.opensuse.org/package/show/openSUSE:Factory:RISCV/chromium)
+  patch -d $_pkgfn/src/3rdparty -p1 < qt6-webengine-chromium-riscv64.patch # Fix riscv64 build for Chromium
+  patch -d $_pkgfn -p1 < qt6-webengine-cmake-riscv64.patch # Fix Qt's CMake architecture detection
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
