--- PKGBUILD
+++ PKGBUILD
@@ -17,12 +17,19 @@ groups=(qt6)
 options=(debug)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
 source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
-        qtbug-107144.patch::https://code.qt.io/cgit/qt/qtwebengine.git/patch/?id=64b7da9d)
+        qtbug-107144.patch::https://code.qt.io/cgit/qt/qtwebengine.git/patch/?id=64b7da9d
+        qt6-webengine-chromium-riscv64.patch
+        qt6-webengine-cmake-riscv64.patch)
 sha256sums=('662ae9ec599b00e902e2c2751ae4977fa95bedf1b035033547ea1b1b69e75303'
-            '8297f0375c8103828f3fd53bfd6005d009ae51cffa581c65cec93f098a6eb54e')
+            '8297f0375c8103828f3fd53bfd6005d009ae51cffa581c65cec93f098a6eb54e'
+            'd703b2f270cf225e84a4f2d71dd6da3eba58ab22f32cac309c2f925fe7fb4644'
+            'fdf8bb4f09adefa432ed7ab3b0e2f375f5b99fcb80d39415f3c574cbe6b71fb2')
 
 prepare() {
   patch -d $_pkgfn -p1 < qtbug-107144.patch
+  # ported from openSUSE (see https://build.opensuse.org/package/show/openSUSE:Factory:RISCV/chromium)
+  patch -d $_pkgfn/src/3rdparty -p0 < qt6-webengine-chromium-riscv64.patch # Fix riscv64 build for Chromium
+  patch -d $_pkgfn -p1 < qt6-webengine-cmake-riscv64.patch # Fix Qt's CMake architecture detection
 }
 
 build() {
