diff --git a/plugins/editor/external/vstgui4/vstgui/lib/vstguibase.h b/plugins/editor/external/vstgui4/vstgui/lib/vstguibase.h
index 4e1d953..5382a95 100644
--- a/plugins/editor/external/vstgui4/vstgui/lib/vstguibase.h
+++ b/plugins/editor/external/vstgui4/vstgui/lib/vstguibase.h
@@ -7,6 +7,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cstring>
+#include <limits>
 
 //-----------------------------------------------------------------------------
 // VSTGUI Version
diff --git a/plugins/vst/external/VST_SDK/VST3_SDK/public.sdk/source/main/moduleinit.h b/plugins/vst/external/VST_SDK/VST3_SDK/public.sdk/source/main/moduleinit.h
index 54a0013..f0e1527 100644
--- a/plugins/vst/external/VST_SDK/VST3_SDK/public.sdk/source/main/moduleinit.h
+++ b/plugins/vst/external/VST_SDK/VST3_SDK/public.sdk/source/main/moduleinit.h
@@ -40,6 +40,7 @@
 #include <algorithm>
 #include <functional>
 #include <numeric>
+#include <limits>
 
 //------------------------------------------------------------------------
 /** A replacement for InitModule and DeinitModule
