--- PKGBUILD
+++ PKGBUILD
@@ -61,7 +61,13 @@ _pick() {
   done
 }
 
+source+=($pkgbase-riscv.patch::https://github.com/sfztools/sfizz/pull/1023.patch)
+sha512sums+=('82898f4adbdc14a672d7c9be3204e965288a9aaa53ab9368a9fe39bdf9518b3b36753da2f275b06d68d320dc1c67b8a3ca581b98828cc513982f98f72c556d9f')
+b2sums+=('d6d8678728f358c02ee1b956caaa3432f2bbb675804347f2e7dc9c8f84c6792ab19183061eadb106aaf987b8de292bc24f0c1564942847eef1cd72a93e0a6bef')
+
 prepare() {
+  patch -d $pkgbase-$pkgver -Np1 < $pkgbase-riscv.patch
+
   # symlink tests data to top-level location so that tests can get to them (we build out of tree)
   ln -sv $pkgbase-$pkgver/tests .
 
