diff --git PKGBUILD PKGBUILD
index d790989..2423f28 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,12 +20,20 @@ optdepends=(
   'vst3-host: for the VST3 plugin'
 )
 provides=('libsfizz.so' 'soundfont-synthesizer')
-source=("https://github.com/sfztools/${pkgname}/releases/download/${pkgver}/${pkgname}-${pkgver}.tar.gz")
-sha512sums=('541f07c18e96ebf7dfa9977b4e6aa69b37e7146f3b2e88276363f7cc5c6f34d3d71a166854fcdc688f89368beeb8eb3a78edf5c8fedb3c89bd2f79f32b31d941')
-b2sums=('2d9b1cabc58d312ee0ac6837bb917369d8ec1785d1dd59a9d0e00d51b4371f98cfddb58fc713f9cafcec072129b8d5cf13870efbf00715fff44aa38a59ec3ddf')
+source=("https://github.com/sfztools/${pkgname}/releases/download/${pkgver}/${pkgname}-${pkgver}.tar.gz"
+		"add_riscv64_support.patch"
+		"add_missing_header.patch")
+sha512sums=('541f07c18e96ebf7dfa9977b4e6aa69b37e7146f3b2e88276363f7cc5c6f34d3d71a166854fcdc688f89368beeb8eb3a78edf5c8fedb3c89bd2f79f32b31d941'
+            "07713b5663d2532b1c54246675d1627f42ba802702ec55c9ac6e3d9229e444cee7f3c8b8f85c44454e36e4fa4800ea6409e34389a26650f8c0a284901ef8516d"
+            "aad8d252300a8d8f67e8203b59fd44412151e0f72440d897fc323b09cc80c6065f1019bf30e8872cd116c6c714a0b7d147a8bca1af00dbc16b628626df8c82f1")
+b2sums=('2d9b1cabc58d312ee0ac6837bb917369d8ec1785d1dd59a9d0e00d51b4371f98cfddb58fc713f9cafcec072129b8d5cf13870efbf00715fff44aa38a59ec3ddf'
+        "3948cf918d548f23e153f059589ac44c3c03d4f028b4aec1a7ba12d5f3d76e16f4e93a2882f30925d3824b15655bdf238a8d90080cab6ec8b7f97ab5feb985d8"
+        "22c4aaaeeef497ff596069b591b6089abb4d699c4480efbefaff7c399b355a82d0f90d4d825c13fed6327b3236a9bf05d2016a6cbc31aa9a5fe088309563d251")
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../add_riscv64_support.patch
+  patch -Np1 -i ../add_missing_header.patch
   # devendor lv2
   rm -rfv lv2/lv2
   sed -e 's|"lv2/core/lv2.h"|<lv2/core/lv2.h>|g' \
