--- a/src/testdir/test_plugin_termdebug.vim
+++ b/src/testdir/test_plugin_termdebug.vim
@@ -826,12 +826,12 @@
     let gdb_buf = winbufnr(1)
     wincmd b
     Break 9
-    sleep 100m
+    call term_wait(gdb_buf)
     redraw!
-    call assert_equal([
+    call WaitForAssert({-> assert_equal([
           \ {'lnum': 9, 'id': 1014, 'name': 'debugBreakpoint1.0',
           \  'priority': 110, 'group': 'TermDebug'}],
-          \ sign_getplaced('', #{group: 'TermDebug'})[0].signs)
+          \ sign_getplaced('', #{group: 'TermDebug'})[0].signs)})
     Run
     call term_wait(gdb_buf, 400)
     redraw!
