--- PKGBUILD
+++ PKGBUILD
@@ -45,6 +45,10 @@
 validpgpkeys=('4F19708816918E19AAE19DEEF3F92DA383FDDE09') # Christian Brabandt <cb@256bit.org>
 
 prepare() {
+  (cd vim
+    # Wait for asynchronous gdb output before checking breakpoint signs.
+    patch -Np1 -i ../fix-termdebug-remote-test-wait.patch
+  )
   (cd vim/src
     # define the place for the global (g)vimrc file (set to /etc/vimrc)
     sed -E 's|^.*(#define SYS_.*VIMRC_FILE.*").*$|\1|g' -i feature.h
@@ -214,4 +218,7 @@
   install -Dm 644 "${srcdir}"/vimdoc.hook "${pkgdir}"/usr/share/libalpm/hooks/vimdoc.hook
 }
 
+source+=(fix-termdebug-remote-test-wait.patch)
+sha256sums+=('dcf2ed45abf4e07117a95d5a619ca3ea4fb6018c35dceec1de589e3a67bbd8bc')
+
 # vim: ts=2 sw=2 et:
