diff --git a/bee/thread/spinlock.h b/bee/thread/spinlock.h
index a401f68..060ddae 100644
--- a/bee/thread/spinlock.h
+++ b/bee/thread/spinlock.h
@@ -10,6 +10,12 @@
     namespace bee { inline void cpu_relax() { _mm_pause(); }}
 #elif defined(__aarch64__)
     namespace bee { inline void cpu_relax() { asm volatile("yield" ::: "memory"); }}
+#elif defined(__riscv)
+    namespace bee { inline void cpu_relax() {
+        int dummy;
+        asm volatile ("div %0, %0, zero" : "=r" (dummy));
+        asm volatile ("" ::: "memory");
+    }}
 #else
     #error unsupport platform
 #endif
