diff --git PKGBUILD.orig PKGBUILD
index d20931d..92c51f8 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -18,7 +18,10 @@
         'git+https://github.com/sqmedeiros/lpeglabel.git'
         'git+https://github.com/actboy168/luamake.git'
         'git+https://github.com/electron/rcedit.git'
-        'wrapper')
+        'wrapper'
+        'add_riscv_support.patch'
+        'fix_pthread_linking.patch'
+        'fix_luamake_pthread_linking.patch')
 b2sums=('SKIP'
         'SKIP'
         'SKIP'
@@ -26,7 +29,10 @@
         'SKIP'
         'SKIP'
         'SKIP'
-        '2ec98b16514c8e416af10625fd582d081448dbe7889d3d9e1fee0540c79f26220bd4b91991c0479852c0102f4c35b63059fe176a59fffbbca8e01274171c7de2')
+        '2ec98b16514c8e416af10625fd582d081448dbe7889d3d9e1fee0540c79f26220bd4b91991c0479852c0102f4c35b63059fe176a59fffbbca8e01274171c7de2'
+        'c7ba1a167e63912a9274b08511b82f15e10d287eb50588cde2203e224f0b83c6f936f31d7cc81412128815f8b62248e6595c553c38ed8e5332be2277e8871d66'
+        'd79831ae242c16a69222f49a5ad36cfe24f211ee3bc465ea820e68a56cfe0a07f64e00b44f4456294134fa85c1f3bf39532c183afa3c033f6b78dd5ecea392ed'
+        '1766b4472813bd8c62cdc9cf8929237fb1d10e7415b2f2a81ed4c76385083fbd7791a118b163f10b00de6f9e664b4f8975faa98ef17a76af9822886c0cbd6443')
 
 prepare() {
   cd $pkgname
@@ -36,10 +42,18 @@
   done
   git config submodule.3rd/lovr-api.url ../lovr-docs
   git submodule update
+  patch -Np1 -i "${srcdir}/fix_pthread_linking.patch"
+  cd "3rd/bee.lua"
+  patch -Np1 -i "${srcdir}/add_riscv_support.patch"
+  cd -
   cd 3rd/luamake
   git submodule init
   git config submodule.3rd/bee.lua.url ../../../bee.lua
   git submodule update
+  patch -Np1 -i "${srcdir}/fix_luamake_pthread_linking.patch"
+  cd "3rd/bee.lua"
+  patch -Np1 -i "${srcdir}/add_riscv_support.patch"
+  cd -
 }
 
 build() {
