--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,7 @@
   mv -v "${_name}-${pkgver}" "$pkgname-$pkgver"
   cd "$pkgname-$pkgver"
   sed -e '/pytest.config/d' -i tests/test_vcr.py
+  patch -Np1 -i "${srcdir}/record-mode.patch"
 }
 
 build() {
@@ -46,3 +47,6 @@
   install -vDm 644 README.rst -t "${pkgdir}/usr/share/doc/${pkgname}"
   install -vDm 644 LICENSE -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
+
+source+=("record-mode.patch::https://github.com/ktosiek/pytest-vcr/commit/fa8ecb724805d73c6f2fd520d8a8d96e411215b6.patch")
+sha512sums+=('9a38eacf27ac5d20666fff82bb4ebad43975603b0dba8073c95023f6cdb0665915cfacb756eafc321a9c71a52997f453256442d931d9fc3f1a5466a353273296')
