--- PKGBUILD
+++ PKGBUILD
@@ -24,7 +24,7 @@ install="${pkgname}.install"
 conflicts=(
   grub-bios
   grub-common
-  grub-efi-x86_64
+  grub-efi-riscv64
   grub-emu
   grub-legacy
 )
@@ -32,12 +32,12 @@ replaces=(
   grub-common
   grub-bios
   grub-emu 
-  grub-efi-x86_64
+  grub-efi-riscv64
 )
 provides=(
   grub-bios
   grub-common
-  grub-efi-x86_64
+  grub-efi-riscv64
   grub-emu
 )
 makedepends=(
@@ -176,9 +176,7 @@ _configure_options=(
 )
 
 _platform=(
-  i386-pc
-  i386-efi
-  x86_64-efi
+  riscv64-efi
 )
 
 build() {
@@ -191,9 +189,10 @@ build() {
     [[ "${i}" == "i386-pc" ]] && _configure_options+=(--enable-efiemu --with-platform="pc" --target="i386")
     [[ "${i}" == "i386-efi" ]] && _configure_options+=(--disable-efiemu --with-platform="efi" --target="i386")
     [[ "${i}" == "x86_64-efi" ]] && _configure_options+=(--with-platform="efi" --target="x86_64")
+    [[ "${i}" == "riscv64-efi" ]] && _configure_options+=(--with-platform="efi" --target="riscv64")
     ./configure PACKAGE_VERSION="${epoch}:${pkgver}-${pkgrel}" \
                 ${_configure_options[@]}
-    if [[ "${i}" == "x86_64-efi" ]]; then
+    if [[ "${i}" == "riscv64-efi" ]]; then
       echo "Build language and doc files only for most common variant..."
       # language directory does not like -j option, build it first with -j1
       make -j1 po/
