Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 452982)
+++ PKGBUILD	(working copy)
@@ -11,9 +11,11 @@
 
 [[ "${CARCH}" == 'x86_64' ]] && _EFI_ARCH='x86_64'
 [[ "${CARCH}" == 'i686' ]] && _EFI_ARCH='i386'
+[[ "${CARCH}" == 'riscv64' ]] && _EFI_ARCH='riscv64'
 
 [[ "${CARCH}" == 'x86_64' ]] && _EMU_ARCH='x86_64'
 [[ "${CARCH}" == 'i686' ]] && _EMU_ARCH='i386'
+[[ "${CARCH}" == 'riscv64' ]] && _EMU_ARCH='riscv64'
 
 pkgname='grub'
 pkgdesc='GNU GRand Unified Bootloader (2)'
@@ -232,7 +234,7 @@
 	cd "${srcdir}/grub/"
 
 	echo "Build grub bios stuff..."
-	_build_grub-common_and_bios
+	#_build_grub-common_and_bios
 
 	echo "Build grub ${_EFI_ARCH} efi stuff..."
 	_build_grub-efi
@@ -249,10 +251,10 @@
 }
 
 _package_grub-common_and_bios() {
-	cd "${srcdir}/grub-bios/"
+	cd "${srcdir}/grub/"
 
 	echo "Run make install for bios build..."
-	make DESTDIR="${pkgdir}/" bashcompletiondir="/usr/share/bash-completion/completions" install
+	#make DESTDIR="${pkgdir}/" bashcompletiondir="/usr/share/bash-completion/completions" install
 
 	echo "Remove gdb debugging related files for bios build..."
 	rm -f "${pkgdir}/usr/lib/grub/i386-pc"/*.module || true
