--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,15 @@ arch=('x86_64')
 license=('GPL')
 depends=('sdl2' 'openal' 'libvorbis' 'libtheora' 'libsodium' 'physfs' 'sqlite' 'freetype2' 'opus')
 makedepends=('zip' 'unzip' 'libzip' 'asciidoc' 'mesa' 'vulkan-headers' 'shaderc' 'cmake' 'ninja')
-source=(${pkgname}-${pkgver}_src.tar.xz::https://github.com/Warzone2100/warzone2100/releases/download/${pkgver}/${pkgname}_src.tar.xz)
-sha256sums=('01d608f6f9638e14d7c857df40ad339c1bfc207a05daafe7157ad8652a3405c3')
+source=(${pkgname}-${pkgver}_src.tar.xz::https://github.com/Warzone2100/warzone2100/releases/download/${pkgver}/${pkgname}_src.tar.xz
+        "remove-curl-nss.patch::https://github.com/Warzone2100/warzone2100/commit/db1cf70950d4fa6630f37a7bf85f548b48ed53cd.diff")
+sha256sums=('01d608f6f9638e14d7c857df40ad339c1bfc207a05daafe7157ad8652a3405c3'
+            '1dcf4edb2741042f6d9abe76e6d8e1bf4255c04cba9ebafde42998d1fd0c9b5f')
 
 prepare() {
 # fix build with vulkan 1.3.260
   sed -e 's|vk::throwResultException|vk::detail::throwResultException|g' -i $pkgname/lib/ivis_opengl/gfx_api_vk.cpp
+  patch -d $pkgname -Np1 -i ../remove-curl-nss.patch
 }
 
 build() {
