--- PKGBUILD
+++ PKGBUILD
@@ -19,11 +19,15 @@ prepare() {

 build() {
   cd build
-
+  # ignore cast-align error https://github.com/Warzone2100/warzone2100/pull/1754
+  CXXFLAGS+=" -Wno-error=cast-align "
   cmake -G Ninja ../${pkgname} \
+    -DCMAKE_THREAD_PREFER_PTHREAD=TRUE \
+    -DTHREADS_PREFER_PTHREAD_FLAG=TRUE \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_INSTALL_LIBDIR=lib \
     -DWZ_DISTRIBUTOR="ArchLinux"
+  sed -i 's$-lpthread$-pthread$g' build.ninja
   ninja
 }
