--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,12 @@
 sha512sums=('11b244ab554385a697d14734757a031acbbe0c424f7bc824f25716ef98404882830c5d322e80582e5f7d4abdfef3d268956ad09da189d515d689242d1e3990e7')
 b2sums=('20935e58bf75783c4538722929cfea9656c103d0cadbb991786ee0ec247771155b42bd3008763c55bf8b141f1e832cb7388755917dc82ce6565917f60be4b486')
 
+prepare() {
+  cd "$pkgname"
+
+  patch -Np1 -i "$srcdir/pillow-get-flattened-data.patch"
+}
+
 build() {
   cd "$pkgname"
 
@@ -50,3 +56,7 @@
   # license
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
+
+source+=(pillow-get-flattened-data.patch)
+sha512sums+=('98716b4421adc13dfd9ab527e0b043e6ec12d42b1e62f47fe58bc50e7f703f5968bda3d5b01e6efc122424a1736d4763843b0ff47491275185c032f27b326f6d')
+b2sums+=('3063bdd50648b287d4d91ab9d2acd0703161f1c18a5025f2376dfe5679d5887aa91553e0716a7817ace4cb04be649c4f20e19001f3bcca73a5e2f4a159d8d511')
