--- a/numba/tests/test_parfors.py
+++ b/numba/tests/test_parfors.py
@@ -70,8 +70,8 @@
     _numba_parallel_test_ = False
 
     # Each test class can run for 30 minutes before time out. Extend this to an
-    # hour on aarch64 (some public CI systems were timing out).
-    _TIMEOUT = 1800 if platform.machine() != 'aarch64' else 3600
+    # hour on aarch64 and riscv64 (some builders were timing out).
+    _TIMEOUT = 3600 if platform.machine() in ('aarch64', 'riscv64') else 1800
 
     """This is the test runner for all the parfors tests, it runs them in
     subprocesses as described above. The convention for the test method naming
