Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1346480)
+++ PKGBUILD	(working copy)
@@ -13,7 +13,7 @@
 makedepends=('python-build' 'python-installer' 'python-poetry-core')
 checkdepends=('python-pytest' 'python-pytest-bdd<6' 'python-toml')
 source=("https://github.com/jrnl-org/jrnl/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-sha512sums=('9484fc275d36f580e2b7a9ac017a4f26da37e18d9a48d2bbd41a339e6377e7240ed3400502f10e8a6eba1d42bf4730023d7dad1218c11394516cdf7e72b4ff0a')
+sha512sums=('70e111846381b046f83e51c35aeb1ba6747862179c09ecdf39d04a7847723b1e69a2fbc60d7be9cccf53c1b5846dfff7dd0b9820859c93fae579d6be8117eacd')
 
 prepare() {
   cd jrnl-$pkgver
