--- PKGBUILD
+++ PKGBUILD
@@ -28,6 +28,12 @@
 
 prepare() {
 	cd "$pkgname-$pkgver"
+
+	# upgrade dependency
+	# upstreamed at https://github.com/hashicorp/terraform/pull/31103
+	go get -u github.com/Netflix/go-expect
+	go mod tidy
+
 	go mod download
 }
 
@@ -44,7 +48,7 @@ build() {
 
 check() {
 	cd "${pkgname}-${pkgver}"
-	go test -mod=readonly ./...
+	go test -mod=readonly -timeout 20m ./...
 }
 
 package() {
