--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,9 @@
 
 prepare() {
 	cd "$pkgname-$pkgver"
+	patch -Np1 -i ../terraform-e2e-tests.patch
+	patch -Np1 -i ../terraform-compression-tests.patch
+	patch -Np1 -i ../terraform-json-hook-tests.patch
 	go mod download
 }
 
@@ -56,3 +59,13 @@
 	install -Dm755 terraform-binary "$pkgdir/usr/bin/terraform"
 	install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("terraform-e2e-tests.patch"
+         "terraform-compression-tests.patch"
+         "terraform-json-hook-tests.patch")
+sha512sums+=('157caf7b6cadf46d9b5e27ebb1e2270ad7aaa03636d5a0734a15fcc46e9e82b9d847df50c9256867d9b54a1500aed0be99d11accfb0014bde6bfa7e6846b9b2c'
+             '9133f3c66afbd6a24fdbc18c781a939f093684f7d258c818ce3349dff011f087edc184050187e493d36cb7286f2ec87b9f7a492608c41075fa32a2cec57a6abe'
+             '34ef2dcb01fbfbd7a995ada6ddd72e7a0256541a369115ac68fb9faf35eb27556b7f97d259692f3ec4c0aca54e8616dec6e79a2f103f1cea4fca7f42c92624cd')
+b2sums+=('9e6c0433b19545d53be172ae4edab85a8ee222849c5e5026c168b40c947f14a98dd8ba89923855a4f0f5a507022dd1a45c20ef1376fa298b3d25061a48fb3f3b'
+         '2c1a2b72b16e3a98acb068b0656b8e463502cd2d3a6bdd345b58f6a7b19dc5d9a2ff17986d31c88c66b9edcf6079471ccbe385487020c302d2a297a3d1468924'
+         '8800a6fd7916c8a89668b7abfae1487c818bb34356d5801aef575cac6d154f62dee1a6106152365981c40a8d2aac0f1971f8cf9a01c5a7b9c4d2856d01646edd')
