--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,21 @@ b2sums=('fdaf424d6e833fb044fcc9fceea9c270b1900b89f988498f18e6cd56428ef150c6e6398
         'd3655f23ba8893d0f8c6cd5e8b42fae473ccfbc77d97b7424aa1f0d0057bfed6cb4d2505e74dd89099c39af6bc03b92eddedae5bb2821ff905d06b5e311be1f0'
         'd58712c8203e4a58ec68738f3d22457547a55d230018408400fd1ca433346171b72fa4a87f05efc93213c71f9757d3a8072149ca4e6894355e79f8cfae3bd55b')
 
+prepare() {
+       cd "$pkgname-$pkgver"
+
+       # current dependency tree:
+       #
+       #   terraform
+       #   `- github.com/Netflix/go-expect
+       #      `- github.com/kr/pty
+       #
+       # TODO: as soon as Netflix/go-expect#28 is merged and `go-expect` releases a new version, request `terraform`
+       # to update `go-expect`, then remove this patch
+       go mod edit -replace github.com/kr/pty=github.com/creack/pty@v1.1.17
+       go mod tidy
+}
+
 build() {
        cd "$pkgname-$pkgver"
        export CGO_CPPFLAGS="${CPPFLAGS}"
