diff --git PKGBUILD PKGBUILD
index 6779b316f..6c9ff6b1a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -35,7 +35,7 @@ prepare() {
 	git config submodule.stdx-allocator.url "$srcdir/stdx-allocator"
 	git submodule update
 
-	_extraFlags="-O5 -flto=full -linker=gold -link-defaultlib-shared=false -L=\"$LDFLAGS\""
+	_extraFlags="-O5 -link-defaultlib-shared=false -L=\"$LDFLAGS\""
 	sed -i "/LDC_FLAGS \:=/ a override LDC_FLAGS += $_extraFlags" makefile
 }
 
