--- PKGBUILD
+++ PKGBUILD
@@ -36,7 +36,7 @@ prepare() {
 	git -c protocol.file.allow=always submodule update
 
 	export _ldFlags="$(echo -ne $LDFLAGS | sed -e 's/-Wl,/-L=/g' -e 's/=auto/=full')"
-	_extraFlags="-O5 -linker=gold -link-defaultlib-shared=false $_ldFlags"
+	_extraFlags="-O5 -link-defaultlib-shared=false $_ldFlags"
 	sed -i "/LDC_FLAGS \:=/ a override LDC_FLAGS += $_extraFlags" makefile
 }
 
