Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1266036)
+++ PKGBUILD	(working copy)
@@ -12,9 +12,12 @@
 url="https://www.msweet.org/mxml"
 license=(LGPL)
 options=(debug)
-source=($pkgname-$pkgver.tar.gz::https://github.com/michaelrsweet/$pkgname/archive/v$pkgver.tar.gz)
-sha512sums=('43e6a92806d9c3f5db39fbf960c15ebfa6d92ef98274b7ce39b57724d6c26ad4362d6d8f3c1023efda92e6a815df068e5038a0cd479562b6be9dbdda8e827a41')
-b2sums=('e0d599fd947dfef72edae40c1e70ea76c92105128dd46e772a7c71f9293cab3339c980396dffe105246cad64d0474935cc1939005ad9a2bffc8e8107f025a466')
+source=($pkgname-$pkgver.tar.gz::https://github.com/michaelrsweet/$pkgname/archive/v$pkgver.tar.gz
+        0001-Update-config.guess-and-config.sub-Issue-296.patch)
+sha512sums=('43e6a92806d9c3f5db39fbf960c15ebfa6d92ef98274b7ce39b57724d6c26ad4362d6d8f3c1023efda92e6a815df068e5038a0cd479562b6be9dbdda8e827a41'
+            '365d9a541b31e3f11aa2f9b768eb21ff07925ff97494eed587a7e541f312e2f5963a84fd220ad7027bd0abaf3e7fa7e594b7caf97e4157943ac50cd7fa9bdb34')
+b2sums=('e0d599fd947dfef72edae40c1e70ea76c92105128dd46e772a7c71f9293cab3339c980396dffe105246cad64d0474935cc1939005ad9a2bffc8e8107f025a466'
+        'fa3f019c893351477ab9797277b4b0b4e98d475aace6d486f5e4b478245ea3a656085f3979b73acc9b37058a48bf518ce3f26b4057ddbe2e2aebfae5ea43ef5c')
 
 _pick() {
   local p="$1" f d; shift
@@ -26,6 +29,11 @@
   done
 }
 
+prepare() {
+  cd $pkgbase-$pkgver
+  patch -p1 -i ../0001-Update-config.guess-and-config.sub-Issue-296.patch
+}
+
 build() {
   cd $pkgbase-$pkgver
   ./configure \
