Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422631)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 arch=(x86_64)
 license=(LGPL)
 depends=(libxml2 glib2 libunwind libcap libelf python)
-makedepends=(gobject-introspection git valgrind bash-completion meson gmp gsl gtk3)
+makedepends=(gobject-introspection git bash-completion meson gmp gsl gtk3)
 _commit=eacb7aa8d12f66a3ff1970c149210972f6079b56  # tags/1.18.4^0
 install=gstreamer.install
 source=("git+https://gitlab.freedesktop.org/gstreamer/gstreamer.git#commit=$_commit")
