--- PKGBUILD
+++ PKGBUILD
@@ -10,11 +10,9 @@ pkgname=(
   gst-plugins-good
   gst-plugins-bad
   gst-plugin-gtk
-  gst-plugin-msdk
   gst-plugin-opencv
   gst-plugin-qml6
   gst-plugin-qmlgl
-  gst-plugin-qsv
   gst-plugin-va
   gst-plugin-wpe
   gst-plugins-ugly
@@ -53,7 +51,7 @@ makedepends=(
   libavtp libbs2b libdca libde265 libdvdnav libfdk-aac libfreeaptx libgme
   libkate libldac liblrdf libltc libmicrodns libmodplug libmpcdec libnice
   libopenmpt libsrtp libva libxkbcommon-x11 libxml2 lilv lv2 mjpegtools neon
-  openal opencv qrencode rtmpdump sbc shaderc soundtouch spandsp svt-hevc
+  openal opencv qrencode rtmpdump sbc shaderc soundtouch spandsp
   vulkan-headers vulkan-icd-loader vulkan-validation-layers
   webrtc-audio-processing wildmidi wpewebkit zbar zvbi zxing-cpp
 
@@ -133,10 +131,13 @@ build() {
     -D gst-plugins-bad:iqa=disabled
     -D gst-plugins-bad:isac=disabled
     -D gst-plugins-bad:magicleap=disabled
+    -D gst-plugins-bad:msdk=disabled
     -D gst-plugins-bad:onnx=disabled
     -D gst-plugins-bad:openh264=disabled
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
+    -D gst-plugins-bad:qsv=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -484,7 +485,7 @@ package_gst-plugins-bad() {
     libfreeaptx libgme libkate libldac liblrdf libltc libmicrodns libmodplug
     libmpcdec libopenmpt librsvg libsndfile libsrtp libwebp libxml2 lilv
     mjpegtools neon nettle openal openexr openjpeg2 openssl opus pango qrencode
-    rtmpdump sbc soundtouch spandsp srt svt-hevc webrtc-audio-processing
+    rtmpdump sbc soundtouch spandsp srt webrtc-audio-processing
     wildmidi x265 zbar zvbi zxing-cpp
   )
 
@@ -539,7 +540,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstspandsp.so
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
