diff --git PKGBUILD PKGBUILD
index db927c2..5ff1936 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,11 +10,9 @@ pkgname=(
   gst-plugins-good
   gst-plugins-bad
   gst-plugin-gtk
-  gst-plugin-msdk
   gst-plugin-opencv
   gst-plugin-qml6
   gst-plugin-qmlgl
-  gst-plugin-qsv
   gst-plugin-va
   gst-plugin-wpe
   gst-plugins-ugly
@@ -53,7 +51,7 @@ makedepends=(
   libltc bluez-libs libavtp libbs2b bzip2 chromaprint libdca faac faad2
   libfdk-aac fluidsynth libgme libkate liblrdf ladspa libde265 lilv libmodplug
   lv2 libmicrodns mjpegtools libmpcdec neon openal libdvdnav rtmpdump sbc
-  soundtouch spandsp libsrtp svt-hevc zvbi libnice webrtc-audio-processing
+  soundtouch spandsp libsrtp zvbi libnice webrtc-audio-processing
   wildmidi zxing-cpp zbar libxml2 gsm libopenmpt wpewebkit libldac libfreeaptx
   qrencode json-glib libva libxkbcommon-x11
 
@@ -131,10 +129,13 @@ build() {
     -D gst-plugins-bad:iqa=disabled
     -D gst-plugins-bad:isac=disabled
     -D gst-plugins-bad:magicleap=disabled
+    -D gst-plugins-bad:msdk=disabled
     -D gst-plugins-bad:onnx=disabled
     -D gst-plugins-bad:openh264=disabled
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
+    -D gst-plugins-bad:qsv=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -481,7 +482,7 @@ package_gst-plugins-bad() {
     openjpeg2 opus libdvdnav libdvdread librsvg rtmpdump sbc libsndfile libltc
     soundtouch spandsp srt libsrtp zvbi libwebp webrtc-audio-processing wildmidi
     x265 zbar gsm libopenmpt libldac libfreeaptx qrencode json-glib libavtp
-    libmicrodns svt-hevc zxing-cpp
+    libmicrodns zxing-cpp
   )
 
   cd root; local files=(
@@ -535,7 +536,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstspandsp.so
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
