Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 424292)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 arch=(x86_64)
 license=(LGPL)
 depends=(libxml2 glib2 libunwind libcap libelf python)
-makedepends=(gobject-introspection git valgrind bash-completion meson gmp gsl gtk3)
+makedepends=(gobject-introspection git bash-completion meson gmp gsl gtk3)
 _commit=71c213f99ad5c8fc8f36a26238483f2bdf7d846a  # tags/1.18.5^0
 install=gstreamer.install
 source=("git+https://gitlab.freedesktop.org/gstreamer/gstreamer.git#commit=$_commit")
