diff --git PKGBUILD PKGBUILD
index 9f09c67..5da2b4b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,10 @@ pkgname=(
   gst-plugins-good
   gst-plugins-bad
   gst-plugin-gtk
-  gst-plugin-msdk
   gst-plugin-onnx
   gst-plugin-opencv
   gst-plugin-qml6
   gst-plugin-qmlgl
-  gst-plugin-qsv
   gst-plugin-va
   gst-plugin-wpe
   gst-plugins-ugly
@@ -159,7 +157,6 @@ makedepends=(
   spandsp
   speex
   svt-av1
-  svt-hevc
   systemd-libs
   taglib
   twolame
@@ -224,10 +221,13 @@ build() {
     -D gst-plugins-bad:isac=disabled
     -D gst-plugins-bad:magicleap=disabled
     -D gst-plugins-bad:mfx_api=oneVPL
+    -D gst-plugins-bad:msdk=disabled
     -D gst-plugins-bad:openh264=disabled
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
+    -D gst-plugins-bad:qsv=disabled
     -D gst-plugins-bad:qt6d3d11=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -736,7 +736,6 @@ package_gst-plugins-bad() {
     spandsp
     srt
     svt-av1
-    svt-hevc
     webrtc-audio-processing-1
     wildmidi
     x265
@@ -799,7 +798,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
     usr/lib/gstreamer-1.0/libgstsvtav1.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
