diff --git PKGBUILD PKGBUILD
index c3548b4..8192a0a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,12 +12,10 @@ pkgname=(
   gst-plugins-ugly
   gst-libav
   gst-plugin-gtk
-  gst-plugin-msdk
   gst-plugin-onnx
   gst-plugin-opencv
   gst-plugin-qml6
   gst-plugin-qmlgl
-  gst-plugin-qsv
   gst-plugin-va
   gst-plugin-wpe
   gst-devtools-libs
@@ -179,7 +177,6 @@ makedepends=(
   speex
   srt
   svt-av1
-  svt-hevc
   systemd-libs
   taglib
   twolame
@@ -246,9 +243,12 @@ build() {
     -D gst-plugins-bad:isac=disabled
     -D gst-plugins-bad:magicleap=disabled
     -D gst-plugins-bad:mfx_api=oneVPL
+    -D gst-plugins-bad:msdk=disabled
     -D gst-plugins-bad:openni2=disabled
     -D gst-plugins-bad:opensles=disabled
+    -D gst-plugins-bad:qsv=disabled
     -D gst-plugins-bad:qt6d3d11=disabled
+    -D gst-plugins-bad:svthevcenc=disabled
     -D gst-plugins-bad:tinyalsa=disabled
     -D gst-plugins-bad:voaacenc=disabled
     -D gst-plugins-bad:voamrwbenc=disabled
@@ -758,7 +758,6 @@ package_gst-plugins-bad() {
     spandsp
     srt
     svt-av1
-    svt-hevc
     webrtc-audio-processing-1
     wildmidi
     x265
@@ -822,7 +821,6 @@ package_gst-plugins-bad() {
     usr/lib/gstreamer-1.0/libgstsrt.so
     usr/lib/gstreamer-1.0/libgstsrtp.so
     usr/lib/gstreamer-1.0/libgstsvtav1.so
-    usr/lib/gstreamer-1.0/libgstsvthevcenc.so
     usr/lib/gstreamer-1.0/libgstteletext.so
     usr/lib/gstreamer-1.0/libgsttimecode.so
     usr/lib/gstreamer-1.0/libgstttmlsubs.so
