--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,11 @@
 sha256sums=('d9d02c5f267d200afb9155e1c27af8ea40ec8015644585f985e13c9cf681af4e')
 
 
+prepare() {
+	cd "${srcdir}/${pkgname}-${pkgver}"
+	patch -Np1 -i "$srcdir/riscv64-system-libraries.patch"
+}
+
 build() {
 	cd "${srcdir}/${pkgname}-${pkgver}"
 	python -m build
@@ -49,3 +54,6 @@
 	cp -r sdl3/__doc__.py "${pkgdir}/usr/lib/python${_PYTHON_VERSION}/site-packages/sdl3/"
 	install -Dm644 "${srcdir}/${pkgname}-${pkgver}/LICENSE" "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
 }
+
+source+=(riscv64-system-libraries.patch)
+sha256sums+=('c6d74cae376418c65a3336cfa861b6af49ecbf2e2ffa11cd57f218f52ecf6fbf')
