diff --git PKGBUILD PKGBUILD
index 4a6226f..acff618 100644
--- PKGBUILD
+++ PKGBUILD
@@ -3,7 +3,7 @@
 pkgname=gn
 pkgver=0.1956.fc295f3a
 _commit=fc295f3ac7ca4fe7acc6cb5fb052d22909ef3a8f
-pkgrel=1
+pkgrel=1.1
 pkgdesc="Meta-build system that generates build files for Ninja"
 arch=('x86_64')
 url="https://gn.googlesource.com/gn/"
@@ -20,6 +20,7 @@ pkgver() {
 
 build() {
   cd $pkgname
+  export LDFLAGS="$LDFLAGS -Wl,-plugin-opt=-target-abi=lp64d"
   ./build/gen.py
   ninja -C out
 }
