diff --git PKGBUILD PKGBUILD
index 480965d..1007cd8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,6 +12,7 @@ depends=('gcc-libs')
 makedepends=('clang' 'ninja' 'python' 'git')
 source=(git+https://gn.googlesource.com/gn#commit=$_commit)
 sha256sums=('SKIP')
+options=(!lto)
 
 pkgver() {
   cd $pkgname
@@ -20,6 +21,9 @@ pkgver() {
 
 build() {
   cd $pkgname
+  # clang does not support it
+  CXXFLAGS="${CXXFLAGS//-fstack-clash-protection/}"
+  CFLAGS="${CFLAGS//-fstack-clash-protection/}"
   ./build/gen.py
   ninja -C out
 }
