diff --git a/Makefile b/Makefile
index b652a8b..9c6c6f2 100644
--- a/Makefile
+++ b/Makefile
@@ -24,10 +24,14 @@ ifneq (,$(findstring aarch,$(MACHINE)))
 else
 ifneq (,$(findstring arm,$(MACHINE)))
 	SIMDFLAGS := -march=armv8-a -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+else
+ifneq (,$(findstring riscv,$(MACHINE)))
+	SIMDFLAGS :=
 else
 	SIMDFLAGS := -msse2
 endif
 endif
+endif
 
 
 PROGRAM = helm
