--- PKGBUILD
+++ PKGBUILD
@@ -25,13 +25,19 @@ source=(
   https://github.com/mtytel/$_name/archive/v$pkgver/$_name-v$pkgver.tar.gz
   $pkgname-0.9.0-gcc91.patch
   $pkgname-0.9.0-rename_helm-synth.patch
+  disable_sse2_on_riscv.patch
+  fix_pthread_linking.patch
 )
 sha512sums=('241c6bf3aec422e9ccaf7c7113eec7e30a11baadd30fe8f05f0abfac9c41d0c8a9fb284f0cae49bcaf0d3d08aa952737f6ff3eb7d48c2b2e22c60932738aeaa4'
             '4c12575c6a58c6a03644b9c7d560e3a794fa3575ac10eb9301ca911a58a86eb5c721f0e4c8c1c5bfcf81b6efba25ed7bdba2a12fe68ec49eb18581067cec06bb'
-            '99df61242d6fd5629440c72210c541f3ce0e711190ab2472ebc066a9044d79499c657c9554b39570867a95e061da3c32ceefce3706e84fc6b4cb540e59289607')
+            '99df61242d6fd5629440c72210c541f3ce0e711190ab2472ebc066a9044d79499c657c9554b39570867a95e061da3c32ceefce3706e84fc6b4cb540e59289607'
+            '16e9423a3b941d755e840af793dd15d3efc3b5337b9b94399b88813f89c9f00270846abb5b71c9fc288ef6aee1c3848a523b98243fd721857311907f0e67fb2f'
+            'b4282be8fd49c4c93476ba8f611273d4729d68397b12c04f4277f89f251f950be88b572f482633c332bbd297c5576d9035675a4e0e0947871d0cbf6273a5b633')
 b2sums=('1b6811f32cf36a1c1f954cc071cd64f3dca656ee0bd420c01d0be3db516210a92adc3694ac49731cf2d7cf057a9eccf2c2dcefc14d24cf8582a3eb69efe1d0e1'
         '54c3c5baf336ee1c2739922be6040d69309db94394967be335103fe325c96afe69f05fbdde0516c153bc47763fc3686c9e0461c6b12dbec59476c85a003e4397'
-        'c95fb6c4080a47bf2a137d986c62f0510b021bafd89b1c9d2f8662f9ad9f086b7d5f7cb08a3dd65ef43f6bc119695ebbda132acc33972454a0d82a05578d48f5')
+        'c95fb6c4080a47bf2a137d986c62f0510b021bafd89b1c9d2f8662f9ad9f086b7d5f7cb08a3dd65ef43f6bc119695ebbda132acc33972454a0d82a05578d48f5'
+        '2d536cb27ce6644c2807aad367ae5a03b57db03cde3e1997c54212d821ce203320ea074fe45f26bbe95faf5a0241789fff96f06128965d9f5c0223e826d812ea'
+        '997f0f5eb948644ff554294bdee51f60bea17284c97b5df0f04fd4273a961a7c92c2b8505c28b757d555cbb29a368af24be834b5ebdedb322e814a6b4d7c00c0')
 
 _pick() {
   local p="$1" f d; shift
@@ -44,6 +50,8 @@ _pick() {
 }
 
 prepare() {
+  patch -Np1 -d $_name-$pkgver -i "../disable_sse2_on_riscv.patch"
+  patch -Np1 -d $_name-$pkgver -i "../fix_pthread_linking.patch"
   # gcc > 9.1 fixes:
   # https://github.com/mtytel/helm/pull/233
   patch -Np1 -d $_name-$pkgver -i ../$pkgname-0.9.0-gcc91.patch
