diff --git PKGBUILD PKGBUILD
index 425c4b9..69a2e40 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,8 @@
 prepare() {
   cd "$_pkgname-$pkgver"
 
+  patch -Np1 -i "../$pkgname-riscv-use-rdtime.patch"
+
   # Disable man page compression on install
   sed -e '/\tgzip .*\/man\// d' -i Makefile
   # Let the package manager handle stripping
@@ -38,4 +40,7 @@
   install -Dm644 LICENSE.bsd "$pkgdir/usr/share/licenses/$pkgname/LICENSE.bsd"
 }
 
+source_riscv64=("$pkgname-riscv-use-rdtime.patch")
+sha256sums_riscv64=('9bbe5ba93cbda062866ec51347b64f9ed746015166764d6dfe0e74ad723f04f1')
+
 # vim:set ts=2 sw=2 et:
