diff --git PKGBUILD PKGBUILD
index 05e5e9e..53be10e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -142,6 +142,8 @@ prepare() {
   sed -e '/add_test(PortChecker/d' -i $pkgname-$pkgver/src/Tests/CMakeLists.txt
   # add missing <cstdint> includes
   patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-fix-missing-cstdint-includes.patch
+  # fix architecture-sensitive test failures
+  patch -Np1 -d $pkgname-$pkgver -i ../$pkgname-3.0.6-test-fixes.patch
 
   (
     cd $pkgname-mruby-zest-build
@@ -227,3 +229,7 @@ package() {
   install -vDm 644 src/osc-bridge/schema/test.json -t "$pkgdir/usr/share/$pkgname/schema/"
   install -vDm 644 completions/zyn-fusion -t "$pkgdir/usr/share/bash-completion/completions/"
 }
+
+source+=($pkgname-3.0.6-test-fixes.patch)
+sha512sums+=('a2d007ded556827d02596bad583e611c02f7621d5e31a8e807dc298841ff4c347b27b9498babc1387bd81067553137c5c2f1160379c5875e1e783296b46790d3')
+b2sums+=('5ee4ebdb457c8f78f89478dd532c9c0f276c8fda7af84ff74064438df0019ac621163a89b3dda2d7a9c973fd83c6577ea9ed327815fadb10e808ef221c5e26fa')
