diff --git PKGBUILD PKGBUILD
index 6cfc65ee..6ce285a8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,10 @@
 makedepends=(gtk-doc gobject-introspection vala meson git bash-completion)
 checkdepends=(python-gobject python-dbus)
 _commit=ac6d31e463168e894269d2a7c949de4577026eea  # tags/1.18.4
-source=("git+https://gitlab.freedesktop.org/mobile-broadband/ModemManager.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.freedesktop.org/mobile-broadband/ModemManager.git#commit=$_commit"
+        "fix-meson-build.patch::https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/merge_requests/711.patch")
+sha256sums=('SKIP'
+            'db7260cb082f498ef8b5974aed2a9c81ba855bb00b3d72c1c2e4fa7654aa512c')
 
 pkgver() {
   cd ModemManager
@@ -24,6 +26,7 @@
 
 prepare() {
   cd ModemManager
+  patch -Np1 -i ../fix-meson-build.patch
 }
 
 build() {
