--- main.c	2021-10-29 19:39:09.439538047 +0800
+++ main.c	2021-10-29 19:40:29.686155961 +0800
@@ -54,7 +54,7 @@
   g_option_context_set_help_enabled(context, TRUE);
   g_option_context_add_main_entries(context, Options, NULL);
   if (!g_option_context_parse(context, &argc, &argv, &error)) {
-    g_warning(error->message);
+    g_warning("%s",error->message);
     g_option_context_free(context);
     return 0;
   }
--- preferences.c	2021-10-29 19:39:09.439538047 +0800
+++ preferences.c	2021-10-29 19:41:05.572827120 +0800
@@ -658,7 +658,7 @@
 
     if (!g_key_file_load_from_file(file, filename, 0, &error))
     {
-        g_message(error->message);
+        g_message("%s",error->message);
         g_error_free(error);
         return;
     }
