diff --git PKGBUILD PKGBUILD
index 0e3fdcf..db0affa 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,8 @@ build() {
   patch -Np0 -i ../../gtk-fix.patch
   # fix linking
   sed -i -e 's/@LIBXML_LIBS@/@LIBXML_LIBS@ -lm/g' Makefile.in
+  sed -i -e 's/g_warning(error->message);/g_warning("%s",error->message);/g' main.c
+  sed -i -e 's/g_message(error->message);/g_message("%s",error->message);/g' preferences.c
   cd ..
 
   ./configure --prefix=/usr
