--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,12 @@
 sha512sums=('678fa048fd001e098aa10bd8f0b9649ac7b635569f94de5a70a7a450f590a3503b25b8e8f4cb68ae14994f482d611b73695f8681b2d651df67862f10b9c03886')
 b2sums=('a9c4e69bf6555b9fb62f0fffeb450508bb18a2252cc4850d9f6dea496abcd806950460a05ea6dc5a7d62f342c9ffab9a2f68f8c9820023119119968708875214')
 
+prepare() {
+  cd $_name-$pkgver
+  # Allow slower builders more time for the 100-query benchmark.
+  sed -i 's/elapsed_time, 0\.5,/elapsed_time, 1.0,/' guardian/testapp/tests/test_indexes.py
+}
+
 build() {
   cd $_name-$pkgver
   python -m build --wheel --no-isolation
