--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@ depends=('ghc-libs' 'haskell-aeson' 'haskell-cassava' 'haskell-digits' 'haskell-
          'haskell-regex-tdfa' 'haskell-scientific' 'haskell-toml-parser' 'haskell-typst-symbols'
          'haskell-vector' 'haskell-xml-conduit' 'haskell-yaml')
 makedepends=('ghc' 'uusi' 'haskell-pretty-show' 'haskell-tasty' 'haskell-tasty-golden')
-source=("https://hackage.haskell.org/packages/archive/$_hkgname/$pkgver/$_hkgname-$pkgver.tar.gz")
-sha256sums=('6afcc3249062ea1ca47d9508c5224ffbe9f48b661553e08e793932a519a14a16')
+source=("https://hackage.haskell.org/packages/archive/$_hkgname/$pkgver/$_hkgname-$pkgver.tar.gz"
+        "increase-test-timeout.patch::https://github.com/hack3ric/typst-hs/commit/6e231ca9976db3f9b83909057f26300d966661db.diff")
+sha256sums=('6afcc3249062ea1ca47d9508c5224ffbe9f48b661553e08e793932a519a14a16'
+            'dff0903fc98744e692d58c840a81f855e5be0d3c19ced6d2c4e548bed4cce15b')
 
 prepare() {
   cd $_hkgname-$pkgver
+  patch -Np1 -i ../increase-test-timeout.patch
   gen-setup
   uusi -u toml-parser $_hkgname.cabal
 }
