diff --git a/src/lincity-ng/Dialog.cpp b/src/lincity-ng/Dialog.cpp
index 241a28e..8ff8fa0 100644
--- a/src/lincity-ng/Dialog.cpp
+++ b/src/lincity-ng/Dialog.cpp
@@ -378,7 +378,7 @@ void Dialog::gameStats(){
         setParagraphN( "statistic_text", line++, outf );
     }
     if (sustain_flag){
-	    snprintf (outf, maxlength, _("Economy is sustainable"));
+	    snprintf (outf, maxlength, "%s",  _("Economy is sustainable"));
         setParagraphN( "statistic_text", line++, outf );
     }
     snprintf (outf, maxlength, _("Population  %d  of which  %d  are not housed.")
diff --git a/src/lincity/fileutil.cpp b/src/lincity/fileutil.cpp
index fa6edbd..0983a03 100644
--- a/src/lincity/fileutil.cpp
+++ b/src/lincity/fileutil.cpp
@@ -498,7 +498,7 @@ void check_savedir(void)
 
 void malloc_failure(void)
 {
-    printf(_("Out of memory: malloc failure\n"));
+    printf("%s", _("Out of memory: malloc failure\n"));
     exit(1);
 }
 
