diff --git PKGBUILD PKGBUILD
index 87cdc9bf..66a0a5aa 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,15 @@ url="https://sourceforge.net/projects/lincity-ng.berlios/"
 license=('GPL' 'custom')
 depends=('sdl_mixer' 'sdl_ttf' 'sdl_image' 'physfs' 'libxml2' 'sdl_gfx' 'libgl')
 makedepends=('ftjam' 'mesa' 'glu')
-source=(https://sourceforge.net/projects/${pkgname}.berlios/files/${pkgname}-${pkgver}.tar.bz2)
-sha1sums=('2961a3f60f7481b33858d7d5ef7cf9c788d01077')
+source=(https://sourceforge.net/projects/${pkgname}.berlios/files/${pkgname}-${pkgver}.tar.bz2
+        "fix_not_literal_format.patch")
+sha1sums=('2961a3f60f7481b33858d7d5ef7cf9c788d01077'
+          'c79189abdc4d06f754b9e7c67811c6277e21d0f6')
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  patch -Np1 -i "../fix_not_literal_format.patch"
+}
 
 build() {
   cd ${pkgname}-${pkgver}
