diff --git a/Tests/VisualTests/CMakeLists.txt b/Tests/VisualTests/CMakeLists.txt
index a0b3703..eb51b89 100644
--- a/Tests/VisualTests/CMakeLists.txt
+++ b/Tests/VisualTests/CMakeLists.txt
@@ -80,6 +80,7 @@ function (TestRenderSystem render_system)
       "TEST_CONTEXT_RESULT_DIR=${USER_HOME_DIRECTORY}"
     "-P" # the script that carries this out
       "RunVTests.cmake")
+  set_tests_properties("TestContext_${render_system_nospace}" PROPERTIES TIMEOUT 4500)
       
 endfunction (TestRenderSystem)
 
