--- PKGBUILD
+++ PKGBUILD
@@ -94,3 +94,11 @@ package() {
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
   rm -r "${pkgdir}/${site_packages}/xarray/tests"
 }
+
+prepare() {
+  cd ${_pkg}-${pkgver}
+  patch -p1 < $srcdir/fix-test_roundtrip_mask_and_scale.diff
+}
+
+source+=('fix-test_roundtrip_mask_and_scale.diff::https://github.com/pydata/xarray/pull/9964.diff')
+sha256sums+=('d4830feddcc93e8aba46456bbd72c11e2a4740ced98ca76a50615c1aad006b62')
