--- PKGBUILD
+++ PKGBUILD
@@ -12,12 +12,17 @@ license=('MIT')
 depends=('glibc' 'v2ray-domain-list-community' 'v2ray-geoip')
 makedepends=('go' 'git')
 backup=(etc/v2ray/config.json)
-source=("git+https://github.com/v2fly/v2ray-core.git#commit=$_commit")
-sha512sums=('SKIP')
+source=("git+https://github.com/v2fly/v2ray-core.git#commit=$_commit"
+        "$pkgname-InitializeServerConfig-sleep.patch::https://github.com/v2fly/v2ray-core/pull/1786.patch")
+sha512sums=('SKIP'
+            '36553288b92bbf7d1ae65cc9efec95f1ca695f73e17804a72270e96382c685dea1dcae69ecfd9603bf7af0cf730080ac59c61c4b91cb724191e985032d5a5a03')
 
 prepare() {
   cd v2ray-core
   sed -i 's|/usr/local/bin|/usr/bin|;s|/usr/local/etc|/etc|' release/config/systemd/system/*.service
+
+  # wait enough time for v2ray to start
+  patch -Np1 -i $srcdir/$pkgname-InitializeServerConfig-sleep.patch
 }
 
 build() {
