diff --git a/testing/scenarios/common.go b/testing/scenarios/common.go
index ffa00006..c15a34a1 100644
--- a/testing/scenarios/common.go
+++ b/testing/scenarios/common.go
@@ -201,6 +201,8 @@ func testUDPConn(port net.Port, payloadSize int, timeout time.Duration) func() e
 }
 
 func testTCPConn2(conn net.Conn, payloadSize int, timeout time.Duration) func() error {
+	timeout *= 20
+
 	return func() (err1 error) {
 		start := time.Now()
 		defer func() {
