diff --git a/testing/scenarios/common.go b/testing/scenarios/common.go
--- a/testing/scenarios/common.go
+++ b/testing/scenarios/common.go
@@ -198,6 +198,8 @@ func testUDPConn(port net.Port, payloadSize int, timeout time.Duration) func() e
 }
 
 func testTCPConn2(conn net.Conn, payloadSize int, timeout time.Duration) func() error {
+	timeout *= 20
+
 	return func() error {
 		payload := make([]byte, payloadSize)
 		common.Must2(rand.Read(payload))
