diff --git a/testing/scenarios/common.go b/testing/scenarios/common.go
index 8bafee0e011..650022636e4 100644
--- a/testing/scenarios/common.go
+++ b/testing/scenarios/common.go
@@ -69,8 +69,6 @@ func InitializeServerConfigs(configs ...*core.Config) ([]*exec.Cmd, error) {
 		servers = append(servers, server)
 	}
 
-	time.Sleep(time.Second * 2)
-
 	return servers, nil
 }
 
@@ -91,6 +89,8 @@ func InitializeServerConfig(config *core.Config) (*exec.Cmd, error) {
 		return nil, err
 	}
 
+	time.Sleep(time.Second * 2)
+
 	return proc, nil
 }
 
