diff --git a/testing/scenarios/vmess_test.go b/testing/scenarios/vmess_test.go
index e1acd455..7dfc8466 100644
--- a/testing/scenarios/vmess_test.go
+++ b/testing/scenarios/vmess_test.go
@@ -806,6 +806,7 @@ func TestVMessKCP(t *testing.T) {
 
 	var errg errgroup.Group
 	for i := 0; i < 10; i++ {
+		time.Sleep(time.Second * 8)
 		errg.Go(testTCPConn(clientPort, 10240*1024, time.Minute*2))
 	}
 	if err := errg.Wait(); err != nil {
