diff --git a/transport/internet/kcp/kcp_test.go b/transport/internet/kcp/kcp_test.go
index d10534a2..43ea9db3 100644
--- a/transport/internet/kcp/kcp_test.go
+++ b/transport/internet/kcp/kcp_test.go
@@ -77,7 +77,7 @@ func TestDialAndListen(t *testing.T) {
 	}
 
 	for i := 0; i < 60 && listener.ActiveConnections() > 0; i++ {
-		time.Sleep(500 * time.Millisecond)
+		time.Sleep(1000 * time.Millisecond)
 	}
 	if v := listener.ActiveConnections(); v != 0 {
 		t.Error("active connections: ", v)
