--- configure.ac
+++ configure.ac
@@ -604,6 +604,12 @@
              AC_MSG_ERROR([libz is required and could not be found]))
 LIBS="$LIBS -lz"
 
+# We need atomic
+AC_CHECK_LIB([atomic], [__atomic_compare_exchange_1],
+             useatomic=1,
+             AC_MSG_ERROR([libatomic is required and could not be found]))
+LIBS="$LIBS -latomic"
+
 # We need threads
 PTHREAD_CFLAGS="-pthread"
 PTHREAD_LIBS="-lpthread"
