--- PKGBUILD
+++ PKGBUILD
@@ -36,15 +36,18 @@ backup=(
 install=kismet.install
 source=(https://www.kismetwireless.net/code/${pkgname}-${_realver}.tar.xz
         kismet.sysusers
-        python-install-flags.patch)
+        python-install-flags.patch
+        kismet.patch)
 sha512sums=('09292096614b9ca3569430ef196cd7cf6fbd2ab7212bdfc653cf5433bdc727ea48c63638220804799f9756a92108a85e8abf0f4e8e0383dbcbe0028bf91a6aee'
             '452b728c30c9932bdb91741af425267da87a66962ea9b736751948881f5477dee0bff8e2eeca2befdea2920da6d7d9e198bde69cef8c48bb49a0093a7f0ef897'
-            'c24bff8ecc0288c0584f60760a27b364bf7b013aad809e4050bb6ed703fc60f3ea93f7e897717a6924333ac6734b20b99166846dae4a9c5c303ae1de39f6692c')
+            'c24bff8ecc0288c0584f60760a27b364bf7b013aad809e4050bb6ed703fc60f3ea93f7e897717a6924333ac6734b20b99166846dae4a9c5c303ae1de39f6692c'
+            '054e268f445c70cebcdcbb401c87a9126021daa36eb982907330517f8c015c4289d4056ce12dd0daa6ce114d57390167256a088d92c75f88b599e727332982ba')
 validpgpkeys=('354689DF3C9DED803381A661D7B28822738BBDB1') # Michael Kershaw
 
 prepare() {
   cd ${pkgname}-${_realver}
   patch -Np0 < ../python-install-flags.patch
+  patch -Np0 < ../kismet.patch
   autoreconf -fiv
 }
 
