diff --git PKGBUILD PKGBUILD
index 1ae2e16..d312886 100644
--- PKGBUILD
+++ PKGBUILD
@@ -36,8 +36,13 @@
   cd $_name-$_pkgver
   # we do not want to package python-gp-libs: https://github.com/tmux-python/libtmux/issues/496
   sed '/addopts/d' -i pyproject.toml
+  patch -Np1 -i ../$_name-increase-test-time-bias-limit.patch
 }
 
+source+=($_name-increase-test-time-bias-limit.patch::https://github.com/tmux-python/libtmux/pull/529.diff)
+sha512sums+=('4ead7a3078466c8e85d15ee2ca2541a6cdd1012ab477fc3974d46bc894e00a64500fa1dd35461082e7707d91bd63acf5f8041ee46bf93dbe22456996e2d6a58c')
+b2sums+=('c73bfba52c7e3d84c6876514d2665f64df5e45a75142289a9958fedd7b2f892e6dcee6da5d183ab5f25be15bf0e52335bd8373c1f5a9c6545121b91d238709b7')
+
 build() {
   cd $_name-$_pkgver
   python -m build --wheel --skip-dependency-check --no-isolation
