diff --git PKGBUILD PKGBUILD
index 4a4c3ba2..48528c36 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,13 +13,16 @@
 depends=('qt5-base' 'fftw' 'libsamplerate')
 makedepends=('cmake')
 source=(https://github.com/lastfm/${pkgbase}/archive/${pkgver}/${pkgname}-${pkgver}.tar.gz
-        liblastfm-qt5.11.patch)
+        liblastfm-qt5.11.patch
+        rm-dynamic-exception-specification.patch::https://github.com/lastfm/liblastfm/commit/d51ad3e7.patch)
 sha1sums=('a84f00a42b3bb8c0f8845ce29c776d609a3b44ac'
-          '40d84032eaeae32072715732b9fe67295483ffad')
+          '40d84032eaeae32072715732b9fe67295483ffad'
+          '3c08a2f28202c1332132d2436016788451e50890')
 
 prepare() {
   cd ${pkgbase}-${pkgver}
   patch -p1 -i ../liblastfm-qt5.11.patch # Fix build with Qt 5.11
+  patch -p1 -i ../rm-dynamic-exception-specification.patch # remove dynamic exception specifications
 }
 
 build() {
