diff --git PKGBUILD PKGBUILD
index 995d5068..ff79e278 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,12 +21,16 @@
 prepare() {
   cd "$pkgbase-$pkgver"
   echo "Icon=drracket" >> share/pkgs/drracket/drracket/drracket.desktop
+  _sharedir=/usr/share/autoconf/build-aux
+  for _file in "config.guess" "config.sub"; do
+    find -name "${_file}" -exec  cp -v "${_sharedir}/${_file}" {} \;   
+  done
 }
 
 build() {
   cd "$pkgbase-$pkgver/src"
   [ "$CARCH" == "x86_64" ] && export CFLAGS+=" -fPIC"
-  ./configure --prefix=/usr --sysconfdir=/etc --enable-shared
+  ./configure --prefix=/usr --sysconfdir=/etc --enable-shared --enable-bcdefault
   make
 }
 
