--- PKGBUILD
+++ PKGBUILD
@@ -75,6 +75,7 @@ prepare() {
   patch -p1 -i ../boost-1.85.patch
   patch -p1 -i ../boost-1.87-1.patch
   patch -p1 -i ../boost-1.87-2.patch
+  patch -p1 -i ../only-enable-sse-x86.patch
 }
 
 build() {
@@ -118,3 +119,7 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -vDm 644 $_name-$pkgver-Source/{AUTHORS,{CHANGELOG,README,README_LINUX}.md} -t "$pkgdir/usr/share/doc/$pkgname/"
 }
+
+source+=(only-enable-sse-x86.patch::https://github.com/supercollider/supercollider/commit/637d2c744b62402fc54433349f16cdbddbae8272.diff)
+sha512sums+=(8b311b760b3ca588c78da8121c0a726c8844cc95defec0400e42570a619afd89c21cae8739fd2d8ddf45c93a9a6e10a3c2f2cbd45a69ec90690ca7386c0eca28)
+b2sums+=(412d427c16d6bdb52a3425355d35ae5045bb30441d2820206a372417de6d3923a07cc4c3a116da86983c3c77f6aaf492ae6b675437e60a6b4f315f2e69708136)
