--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,7 @@ depends=(
   sqlite
 )
 depends_x86_64=(mariadb-libs)
+depends_riscv64=(mariadb-libs clang)
 makedepends=(
   git
   rust
@@ -46,6 +47,8 @@ b2sums=('cc82f8ca695ed6152d8a99c178fda9a250c0ac233062dc0574ecfc26fa587eded894ed8
 
 prepare() {
   cd $pkgname
+  # generate bindgen at build time
+  cargo add mysqlclient-sys@0.4.1 --features buildtime_bindgen
   # download dependencies
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 
