diff --git PKGBUILD PKGBUILD
index 11a3ae5..74bf3ae 100644
--- PKGBUILD
+++ PKGBUILD
@@ -37,8 +37,12 @@ prepare() {
   s,/path/to/log,/var/log/$pkgname.log,
   /^# ROCKET_TLS/a ROCKET_LIMITS={json=10485760}" .env.template
 
+  # update dependencies
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+
   # download dependencies
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked
 }
 
 build() {
