Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1014267)
+++ PKGBUILD	(working copy)
@@ -33,6 +33,8 @@
   s,# WEB_VAULT_ENABLED=true,WEB_VAULT_ENABLED=false,
   s,/path/to/log,/var/log/$pkgname.log,
   /^# ROCKET_TLS/a ROCKET_LIMITS={json=10485760}" .env.template
+  sed -i "/patch.crates-io/a ring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" Cargo.toml
+  cargo update -p ring
   cargo fetch --locked
 }
 
