diff --git PKGBUILD PKGBUILD
index 28654af..ab200c1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,9 @@
   # https://github.com/undler/bundler/issues/6882
   sed -e '/BUNDLED WITH/,+1d' -i Gemfile.lock
 
+  # Fix incompatible-pointer-types
+  git cherry-pick -n 0b0c5d10a81c3fb912249f13465a1cd0a0c87844
+
   bundle config build.nokogiri --use-system-libraries
   bundle config set --local deployment 'true'
   sed 's|git ls-files|find -type f|' -i metasploit-framework.gemspec
