--- PKGBUILD
+++ PKGBUILD
@@ -47,9 +47,18 @@ provides=(
   'libcriu.so'
 )
 options=('!lto')
-source=("git+https://github.com/checkpoint-restore/criu#tag=v$pkgver")
-sha512sums=('cdfaebfe37c2e1111383444a0575499bb4a6b7618a944921d8c2bfa7c2a72831557bc568f8457c70ea0d94b0e70549ed5f4d43134e0c58baa5ffab3c1b3e40c6')
-b2sums=('4cf22e25a848007e5c7230aa1a60216f4ba98423be47442a00fd7d5c0079ac86901bfcd5c82d2ee7f4ab243bc1eb431328848317dc24c3c45d81250318bc9e21')
+source=("git+https://github.com/checkpoint-restore/criu#tag=v$pkgver"
+        'riscv64-support.patch')
+sha512sums=('cdfaebfe37c2e1111383444a0575499bb4a6b7618a944921d8c2bfa7c2a72831557bc568f8457c70ea0d94b0e70549ed5f4d43134e0c58baa5ffab3c1b3e40c6'
+            'e8ba686efde48aa1f8c8ea993151cefd31565d4746c63be6c466f1e4b6be186c325bdfb2766bdf2af872bad9505fd5ef25563074a760e17d26854d80ded800c4')
+b2sums=('4cf22e25a848007e5c7230aa1a60216f4ba98423be47442a00fd7d5c0079ac86901bfcd5c82d2ee7f4ab243bc1eb431328848317dc24c3c45d81250318bc9e21'
+        '5df6e1dc381fe662bf6efe08ad1071576afa1efb02f289c6d4171363f6d08f5c1d50c0aba1edc3b46da1224531a4a73ac6763d55dfb3b37729fc48e61c75eb8d')
+
+prepare() {
+  cd "$pkgname"
+
+  patch -Np1 -i "$srcdir/riscv64-support.patch"
+}
 
 build() {
   cd "$pkgname"
