diff --git a/packages/swipl-win/CMakeLists.txt b/packages/swipl-win/CMakeLists.txt
index 8e617d4..c091582 100644
--- a/packages/swipl-win/CMakeLists.txt
+++ b/packages/swipl-win/CMakeLists.txt
@@ -68,7 +68,7 @@ add_executable(swipl-win MACOSX_BUNDLE ${PLWIN_SRC})
 set_target_properties(swipl-win PROPERTIES
 		      OUTPUT_NAME "${SWIPL_APP_NAME}"
 		      MACOSX_BUNDLE_ICON_FILE swipl.icns)
-target_link_libraries(swipl-win libswipl Qt5::Widgets)
+target_link_libraries(swipl-win libswipl Qt5::Widgets -pthread)
 target_compile_options(swipl-win PRIVATE ${QT_DEFINES})
 
 install(TARGETS swipl-win
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5e9bc3c..7fa7ab3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -351,7 +351,7 @@ endif()
 
 # build swipl
 add_executable(swipl${PGO_SUFFIX} ${SWIPL_SRC})
-target_link_libraries(swipl${PGO_SUFFIX} ${SWIPL_LIBRARIES} libswipl${PGO_SUFFIX})
+target_link_libraries(swipl${PGO_SUFFIX} ${SWIPL_LIBRARIES} libswipl${PGO_SUFFIX} -pthread)
 target_c_stack(swipl${PGO_SUFFIX} 4000000)
 if(MINGW_ROOT)
   target_link_options(swipl${PGO_SUFFIX} PRIVATE -municode)
