diff --git PKGBUILD PKGBUILD
index 739ab78e..179cbcda 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,8 +20,15 @@
             'libxinerama: for using the pl2xpce library'
             'libxpm: for using the pl2xpce library'
             'unixodbc: for using the odbc4pl library')
-source=("$pkgname-$pkgver.tar.gz::https://www.swi-prolog.org/download/stable/src/swipl-$pkgver.tar.gz")
-b2sums=('bd53355abe63ffde348ed38b0e7fee4bcba4e74128c73904a72ca3ba2021a8b650d08480a643ea3336f3fc6d1d7bcf7651deb1d32608581ce43bc118288df6da')
+source=("$pkgname-$pkgver.tar.gz::https://www.swi-prolog.org/download/stable/src/swipl-$pkgver.tar.gz"
+        "fix_pthread_linking.patch::https://github.com/SWI-Prolog/swipl-devel/pull/921.patch")
+b2sums=('bd53355abe63ffde348ed38b0e7fee4bcba4e74128c73904a72ca3ba2021a8b650d08480a643ea3336f3fc6d1d7bcf7651deb1d32608581ce43bc118288df6da'
+        '8d35b05f36b107ab6a4e399262f0ae60101dfe4c60390287b65736d0b37fa3f01caae6a8fc4ef577a54e4c71168fbdb483a82b4a0c7977c233ff0695825df1b0')
+
+prepare() {
+  cd "swipl-$pkgver"
+  patch -Np1 -i "../fix_pthread_linking.patch"
+}
 
 build() {
   cmake \
