diff --unified --recursive --text a/src/goom_core.c b/src/goom_core.c
--- a/src/goom_core.c	2021-09-16 06:48:54.759707672 +0000
+++ b/src/goom_core.c	2021-09-16 06:50:39.623087914 +0000
@@ -834,7 +834,7 @@
     
     if (message) {
         int i=1,j=0;
-        sprintf (goomInfo->update_message.message, message);
+        sprintf (goomInfo->update_message.message, "%s", message);
         for (j=0;goomInfo->update_message.message[j];j++)
             if (goomInfo->update_message.message[j]=='\n')
                 i++;
@@ -849,7 +849,7 @@
         int pos;
         float ecart;
         message = msg;
-        sprintf (msg, goomInfo->update_message.message);
+        sprintf ("%s", msg, goomInfo->update_message.message);
         
         while (!fin) {
             while (1) {
