diff --git PKGBUILD PKGBUILD
index fed85a864..f0e021fc2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,16 @@ url="https://www.fovea.cc/?page=details_produits&id=goom"
 license=('LGPL')
 depends=('glibc')
 options=('!emptydirs')
-source=("https://downloads.sourceforge.net/project/goom/goom2k4/0/goom-2k4-0-src.tar.gz")
-sha512sums=('790e3ab8dee122320ad8b3ae15f6a1cd2780222d5ae97979f614f16ba73b4b853960e7945ef9053ad9eddf9e29e69d0c52f0875d07c5c6c7cee41d764bf4ea5f')
+source=("https://downloads.sourceforge.net/project/goom/goom2k4/0/goom-2k4-0-src.tar.gz"
+        "fix-format-not-a-string-literal-and-no-format-arguments.patch")
+sha512sums=('790e3ab8dee122320ad8b3ae15f6a1cd2780222d5ae97979f614f16ba73b4b853960e7945ef9053ad9eddf9e29e69d0c52f0875d07c5c6c7cee41d764bf4ea5f'
+            '217fdfd5ff7687e29b53e4e3b48ce4d55f7ff79e3f75c3eadcf55435fabcbf743d4a06c58286dc2041943381da6c623f8850b8454fa0d48dd4d3d3c5c9f53267')
+
+prepare(){
+  cd "${srcdir}/goom2k4-0"
+
+  patch -Np1 -i "${srcdir}/fix-format-not-a-string-literal-and-no-format-arguments.patch"
+}
 
 build() {
   cd "${srcdir}/goom2k4-0"
