diff --git a/game/ffmpeg.cc b/game/ffmpeg.cc
index 58c3e64..8a90163 100644
--- a/game/ffmpeg.cc
+++ b/game/ffmpeg.cc
@@ -87,7 +87,7 @@ void FFmpeg::open() {
 	if (avformat_find_stream_info(m_formatContext, nullptr) < 0) throw std::runtime_error("Cannot find stream information");
 	m_formatContext->flags |= AVFMT_FLAG_GENPTS;
 	// Find a track and open the codec
-	const AVCodec* codec = nullptr;
+	AVCodec* codec = nullptr;
 	m_streamId = av_find_best_stream(m_formatContext, (AVMediaType)m_mediaType, -1, -1, &codec, 0);
 	if (m_streamId < 0) throw std::runtime_error("No suitable track found");
 
