diff --git PKGBUILD PKGBUILD
index 1576b63..abf69d5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,8 +30,10 @@ optdepends=(
   'python-snappy: snappy compression suppport for WebSocket messages'
 )
 
-source=(https://files.pythonhosted.org/packages/source/a/autobahn/autobahn-$pkgver.tar.gz)
-sha256sums=('e126c1f583e872fb59e79d36977cfa1f2d0a8a79f90ae31f406faae7664b8e03')
+source=(https://files.pythonhosted.org/packages/source/a/autobahn/autobahn-$pkgver.tar.gz
+        judge-sse-header.patch)
+sha256sums=('e126c1f583e872fb59e79d36977cfa1f2d0a8a79f90ae31f406faae7664b8e03'
+            'a6505b22363d7b1bbd6c9b4d503118e16129abf91e65885b7ee7f47f74ed9d00')
 
 prepare() {
   cd "$srcdir/autobahn-$pkgver"
@@ -39,7 +41,8 @@ prepare() {
   # If Arch decides to increase CPU requirements [1], -march=x86-64 can be
   # replaced with -march=nehalem so that the SSE 4.1 implementation is built
   # [1] https://lists.archlinux.org/pipermail/arch-dev-public/2020-March/029914.html
-  sed -i 's#-march=native#-march=x86-64#' autobahn/nvx/_utf8validator.py
+  sed -i 's#-march=native#-march=rv64gc#' autobahn/nvx/_utf8validator.py
+  patch -Np1 -i "${srcdir}/judge-sse-header.patch"
 }
 
 build() {
