diff --git a/pycall.rb/spec/pycall_spec.rb b/pycall.rb/spec/pycall_spec.rb
index 536438b..3089cdd 100644
--- a/pycall.rb/spec/pycall_spec.rb
+++ b/pycall.rb/spec/pycall_spec.rb
@@ -152,7 +152,7 @@ require 'pycall'
 puts(PyCall.init ? 'true' : 'false')
 RUBY
       expect(status).to be_success
-      expect(out.chomp).to eq('true')
+      expect(out.lines.last.chomp).to eq('true')
     end
 
     it 'returns false if already initialized' do
