diff --git PKGBUILD PKGBUILD
index 001dd0c..50fa715 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,10 +27,13 @@ checkdepends=(
   ruby-test-unit
 )
 options=(!emptydirs)
-source=("git+${url}.git#tag=v${pkgver}")
-sha512sums=('SKIP')
+source=("git+${url}.git#tag=v${pkgver}"
+        init-return-test.patch)
+sha512sums=('SKIP'
+            '4abb802b714e7279381a554e94ad484d6d51319754f64fbc3518e8d75c3838813b543c838df6734ef4ff0ececf6396c5791003bd04c98665b261ff5624d2d21e')
 
 prepare() {
+  patch -Np1 -i ./init-return-test.patch
   cd pycall.rb
   sed -r -e '/rake-compiler-dock/d' -e '/launchy/d' -i pycall.gemspec
 }
