diff --git PKGBUILD PKGBUILD
index 1eb41d0..3a555a5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ _git='https://github.com/ROCm/hipBLASLt'
 source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
         "$pkgname-find-msgpack5.patch")
 sha256sums=('e281a1a7760fab8c3e0baafe17950cf43c422184e3226e3c14eb06e50c69d421'
-            '7943de413882065de7d840a4a0f48e8133c4977a04a3e9587ea7936cc8a0939d')
+            '8e6d134a1d45de814099dfe591296de5e1d47f9906ccf9c51d7f2228e992c5f4')
 _dirname="$(basename "$_git")-$(basename "${source[0]}" ".tar.gz")"
 
 prepare() {
diff --git hipblaslt-find-msgpack5.patch hipblaslt-find-msgpack5.patch
index 2d78089..f776e48 100644
--- hipblaslt-find-msgpack5.patch
+++ hipblaslt-find-msgpack5.patch
@@ -7,7 +7,7 @@ index 4320652..b51e04a 100644
  
  if(TENSILE_USE_MSGPACK)
 -    find_package(msgpack REQUIRED)
-+    find_package(msgpackc-cxx REQUIRED)
++    find_package(msgpack-cxx REQUIRED)
      target_compile_definitions(TensileHost PUBLIC -DTENSILE_MSGPACK=1)
  
      if(TARGET msgpackc-cxx)
