--- PKGBUILD
+++ PKGBUILD
@@ -48,11 +48,13 @@
 source=("rocm-libraries::git+$_git.git#tag=rocm-$pkgver"
         "hipblaslt-Don-t-fetch-deps-for-rocroller.patch"
         "rocroller-Always-fetch-libdivide-if-it-s-not-install.patch"
-        "hipblaslt-7.2.3-cstdint-include.patch")
+        "hipblaslt-7.2.3-cstdint-include.patch"
+        "rocroller-fix-hip-noinline.patch")
 sha256sums=('b476acbcd0f4017c800e4b05533e6dfb875bde32242729c8df557d4624379623'
             '81c0f9ac1872e34555c2f9d64eecec75c42fd8813c4acf54cd665b03e59cb752'
             'b845a96fee02651f2cad2417fc9e8522ab143849b2049b0010d17401e6000e52'
-            '337feb845674f8ec6015c902b27bd01bc51f530035cda769b7dea5ad12d41920')
+            '337feb845674f8ec6015c902b27bd01bc51f530035cda769b7dea5ad12d41920'
+            '867bce65900e560b7c6e03ff7aae84caebff26ac1b45019bd3b31f1b51203566')
 _dirname="rocm-libraries/projects/$pkgname"

 options=(!strip)
@@ -62,6 +64,7 @@
   patch -Np1 -i "$srcdir/hipblaslt-Don-t-fetch-deps-for-rocroller.patch"
   patch -Np1 -i "$srcdir/rocroller-Always-fetch-libdivide-if-it-s-not-install.patch"
   patch -Np1 -i "$srcdir/hipblaslt-7.2.3-cstdint-include.patch"
+  patch -Np1 -i "$srcdir/rocroller-fix-hip-noinline.patch"
 }

 build() {
