diff --git PKGBUILD PKGBUILD
index f94b12b..4d5410b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,8 @@ url='https://rocm.docs.amd.com/projects/hipBLASLt/en/latest/index.html'
 license=('MIT')
 depends=('rocm-core' 'glibc' 'gcc-libs' 'hip-runtime-amd' 'hipblas' 'rocblas')
 makedepends=('rocm-cmake' 'python' 'cmake' 'git' 'ninja'
-             'msgpack-cxx' 'python-msgpack' 'python-joblib' 'python-pyaml' 'hipblas-common')
+             'msgpack-cxx' 'python-msgpack' 'python-joblib' 'python-pyaml' 'hipblas-common'
+             'python-simplejson' 'python-ujson' 'python-orjson')
 _git='https://github.com/ROCm/hipBLASLt'
 source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
         "$pkgname-find-msgpack5.patch")
