diff --git PKGBUILD PKGBUILD
index 2738b5f44..3b343d2e2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ license=('MIT')
 depends=('python' 'python-six' 'python-engineio' 'python-bidict' 'python-msgpack')
 makedepends=('python-setuptools' 'python-sphinx' 'python-tornado')
 optdepends=('python-tornado: tornado asyncio webserver')
-checkdepends=('python-pytest' 'python-mock')
+checkdepends=('python-pip' 'python-pytest' 'python-mock')
 options=('!makeflags')
 source=(https://github.com/miguelgrinberg/python-socketio/archive/v${pkgver}/${pkgname}-${pkgver}.tar.gz)
 sha512sums=('cc5907c35428103b42bbd4bf95d59012211c3efaa925485e5dd31f8c0a8c626074f84f1a4d4e660abc02a019da90299f584c25f6a7c51fd41f5c7df126f2ef26')
