diff -uprN rxtx-2.2pre2/src/ParallelImp.c rxtx-2.2pre2-patch/src/ParallelImp.c
--- rxtx-2.2pre2/src/ParallelImp.c	2008-11-28 04:01:48.000000000 +0800
+++ rxtx-2.2pre2-patch/src/ParallelImp.c	2021-11-13 23:02:42.608071635 +0800
@@ -920,7 +920,7 @@ void throw_java_exception_system_msg( JN
 void report_error(char *msg)
 {
 #ifndef DEBUG_MW
-	fprintf(stderr, msg);
+	fprintf(stderr, "%s", msg);
 #else
 	mexWarnMsgTxt( msg );
 #endif /* DEBUG_MW */
@@ -938,7 +938,7 @@ void report_error(char *msg)
 void report(char *msg)
 {
 #ifdef DEBUG
-        fprintf(stderr, msg);
+        fprintf(stderr, "%s", msg);
 #endif /* DEBUG */
 }
 
diff -uprN rxtx-2.2pre2/src/SerialImp.c rxtx-2.2pre2-patch/src/SerialImp.c
--- rxtx-2.2pre2/src/SerialImp.c	2009-02-05 06:06:16.000000000 +0800
+++ rxtx-2.2pre2-patch/src/SerialImp.c	2021-11-13 22:56:22.439555164 +0800
@@ -5096,7 +5096,7 @@ void throw_java_exception( JNIEnv *env,
 void report_warning(char *msg)
 {
 #ifndef DEBUG_MW
-	fprintf(stderr, msg);
+	fprintf(stderr, "%s", msg);
 #else
 	mexWarnMsgTxt( (const char *) msg );
 #endif /* DEBUG_MW */
@@ -5117,7 +5117,7 @@ void report_verbose(char *msg)
 #ifdef DEBUG_MW
 	mexErrMsgTxt( msg );
 #else
-	fprintf(stderr, msg);
+	fprintf(stderr, "%s", msg);
 #endif /* DEBUG_MW */
 #endif /* DEBUG_VERBOSE */
 }
@@ -5133,7 +5133,7 @@ void report_verbose(char *msg)
 void report_error(char *msg)
 {
 #ifndef DEBUG_MW
-	fprintf(stderr, msg);
+	fprintf(stderr, "%s", msg);
 #else
 	mexWarnMsgTxt( msg );
 #endif /* DEBUG_MW */
@@ -5152,7 +5152,7 @@ void report(char *msg)
 {
 #ifdef DEBUG
 #	ifndef DEBUG_MW
-		fprintf(stderr, msg);
+		fprintf(stderr, "%s", msg);
 #	else
 		mexPrintf( msg );
 #	endif /* DEBUG_MW */
