diff -uprN rxtx-2.2pre2/src/RawImp.c rxtx-2.2pre2-patch/src/RawImp.c
--- rxtx-2.2pre2/src/RawImp.c	2007-11-19 06:32:42.000000000 +0800
+++ rxtx-2.2pre2-patch/src/RawImp.c	2021-11-13 23:08:57.066631988 +0800
@@ -100,9 +100,9 @@
 #	include <linux/version.h>
 #endif
 #ifndef __APPLE__  /* dima */
-#ifndef PPC
+#if !defined(PPC) && !defined(__riscv)
 #include <sys/io.h>
-#endif /* PPC */
+#endif /* PPC & RISC-V */
 #endif /* dima */
 
 extern int errno;
