--- PKGBUILD
+++ PKGBUILD
@@ -69,12 +69,16 @@ prepare() {
   # Fix format-security errors
   patch -p1 -i ../rxtx-2.2-format-security.patch
 
+  # Fix build on RISC-V 64
+  patch -Np1 -i ../fix-riscv64-build.patch
+
   rm *.m4
   autoreconf -fi
 }
 
 build() {
   cd $_pkgname-$pkgver
+  export CFLAGS="$CFLAGS -Wno-error=implicit-function-declaration"
   ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var
   make -j1
 }
@@ -88,3 +92,7 @@ package() {
   install -dm755 "$pkgdir"/usr/lib/tmpfiles.d
   echo 'd /run/lock/lockdev 0775 root lock -' > "$pkgdir/usr/lib/tmpfiles.d/$pkgname.conf"
 }
+
+source+=(fix-riscv64-build.patch)
+md5sums+=('a99178f7f50e2cfdb99d2eb3ed167421')
+sha256sums+=('b111b2b7395d74d5560839d73c67843f5422c44cf9c50fa3d35d4345e18c612b')
