--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,8 @@ source=(http://rxtx.qbang.org/pub/$_pkgname/$_pkgname-$pkgver.zip
         ttyACM_port.patch
         java10.patch
         java11.patch
-        rxtx-2.2-undefined_symbol.patch)
+        rxtx-2.2-undefined_symbol.patch
+        fix-riscv64-build.patch)
 md5sums=('7eedb18e3f33a427e2b0e9be8ce3f94c'
          '2f21ec5eb108f871815242698b6150f1'
          '1f7c43d582bfe9daea22d7f7057436da'
@@ -31,7 +32,8 @@ md5sums=('7eedb18e3f33a427e2b0e9be8ce3f94c'
          '903a3fe0067d0682dd5f64483c741df6'
          '683dd95e6e419b2b63851c08ede7ca86'
          '1db5c64e239c80294d00c932237889dd'
-         '4695fe9bb28a7c9b21447f998fb46b02')
+         '4695fe9bb28a7c9b21447f998fb46b02'
+         'a99178f7f50e2cfdb99d2eb3ed167421')
 
 prepare() {
   cd $_pkgname-$pkgver
@@ -60,6 +62,9 @@ prepare() {
   # Fix format-security errors
   patch -p1 -i ../rxtx-2.2-format-security.patch
 
+  # Fix build on RISC-V 64
+  patch -Np1 -i ../fix-riscv64-build.patch
+
   rm *.m4
   autoreconf -fi
 }
