--- PKGBUILD
+++ PKGBUILD
@@ -41,10 +41,15 @@ validpgpkeys=(
 # Make sure the duplication between bin and lib is found
 COMPRESSZST+=(--long)
 
+source+=(tests-skip-some-tests-on-rv64.patch)
+b2sums+=(986ec94fe2612f3fc234ad0db478f6ce0e95952fcce694b449f842233d56f657ea3c647d56331b6422670767fd3b23e1f98f25a6fc76fbd85767d8dabb358952)
+
 prepare() {
   mkdir mozbuild
   cd firefox-$pkgver
 
+  patch -Np1 -i ../tests-skip-some-tests-on-rv64.patch
+
   cat >../mozconfig <<END
 ac_add_options --enable-application=js
 mk_add_options MOZ_OBJDIR=${PWD@Q}/obj
@@ -54,12 +59,13 @@ ac_add_options --enable-release
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-ac_add_options --enable-linker=lld
+ac_add_options --enable-linker=bfd
 ac_add_options --disable-bootstrap
 ac_add_options --disable-debug
 ac_add_options --disable-jemalloc
 ac_add_options --disable-strip
 ac_add_options --disable-unified-build
+ac_add_options --disable-jit
 
 # System libraries
 ac_add_options --with-system-zlib
@@ -82,40 +88,7 @@ build() {
 
   # Do 3-tier PGO
   echo "Building instrumented JS..."
-  cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-profile-generate=cross
-END
-  ./mach build
-
-  echo "Profiling instrumented JS..."
-  (
-    local js="$PWD/obj/dist/bin/js"
-    export LLVM_PROFILE_FILE="$PWD/js-%p-%m.profraw"
-
-    cd js/src/octane
-    "$js" run.js
-
-    cd ../../../third_party/webkit/PerformanceTests/ARES-6
-    "$js" cli.js
-
-    cd ../SunSpider/sunspider-0.9.1
-    "$js" sunspider-standalone-driver.js
-  )
-
-  llvm-profdata merge -o merged.profdata *.profraw
-
-  stat -c "Profile data found (%s bytes)" merged.profdata
-  test -s merged.profdata
-
-  echo "Removing instrumented JS..."
-  ./mach clobber
-
-  echo "Building optimized JS..."
-  cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-lto=cross
-ac_add_options --enable-profile-use=cross
-ac_add_options --with-pgo-profile-path=${PWD@Q}/merged.profdata
-END
+  cat >.mozconfig ../mozconfig
   ./mach build
 }
 
