--- PKGBUILD	(revision 470897)
+++ PKGBUILD	(working copy)
@@ -12,9 +12,11 @@
 checkdepends=(mercurial git)
 options=(!lto)
 _relver=${pkgver}esr
-source=(https://archive.mozilla.org/pub/firefox/releases/$_relver/source/firefox-$_relver.source.tar.xz{,.asc})
+source=(https://archive.mozilla.org/pub/firefox/releases/$_relver/source/firefox-$_relver.source.tar.xz{,.asc}
+        tests-skip-some-tests-on-rv64.patch)
 sha256sums=('31b074623f09ca821a8a7dee0b3d2df1a45f1164264d31c26b4969a4d6a21dd1'
-            'SKIP')
+            'SKIP'
+            '6f9dc585367257e1d793b4cb6860d2b9ea00bc01e57d659202f5f5a27218dcfd')
 validpgpkeys=('14F26682D0916CDD81E37B6D61B7B526D98F0353') # Mozilla Software Releases <release@mozilla.com>
 
 # Make sure the duplication between bin and lib is found
@@ -24,6 +26,8 @@
   mkdir mozbuild
   cd firefox-$pkgver
 
+  patch -Np1 -i ../tests-skip-some-tests-on-rv64.patch
+
   cat >../mozconfig <<END
 ac_add_options --enable-application=js
 mk_add_options MOZ_OBJDIR=${PWD@Q}/obj
@@ -33,12 +37,13 @@
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-ac_add_options --enable-linker=lld
+ac_add_options --enable-linker=bfd
 ac_add_options --disable-bootstrap
 ac_add_options --disable-debug
 ac_add_options --disable-jemalloc
 ac_add_options --disable-strip
 ac_add_options --disable-unified-build
+ac_add_options --disable-jit
 
 # System libraries
 ac_add_options --with-system-zlib
@@ -62,40 +67,40 @@
   # Do 3-tier PGO
   echo "Building instrumented JS..."
   cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-profile-generate=cross
+#ac_add_options --enable-profile-generate=cross
 END
   ./mach build
 
-  echo "Profiling instrumented JS..."
-  (
-    local js="$PWD/obj/dist/bin/js"
-    export LLVM_PROFILE_FILE="$PWD/js-%p-%m.profraw"
+#   echo "Profiling instrumented JS..."
+#   (
+#     local js="$PWD/obj/dist/bin/js"
+#     export LLVM_PROFILE_FILE="$PWD/js-%p-%m.profraw"
 
-    cd js/src/octane
-    "$js" run.js
+#     cd js/src/octane
+#     "$js" run.js
 
-    cd ../../../third_party/webkit/PerformanceTests/ARES-6
-    "$js" cli.js
+#     cd ../../../third_party/webkit/PerformanceTests/ARES-6
+#     "$js" cli.js
 
-    cd ../SunSpider/sunspider-0.9.1
-    "$js" sunspider-standalone-driver.js
-  )
+#     cd ../SunSpider/sunspider-0.9.1
+#     "$js" sunspider-standalone-driver.js
+#   )
 
-  llvm-profdata merge -o merged.profdata *.profraw
+#   llvm-profdata merge -o merged.profdata *.profraw
 
-  stat -c "Profile data found (%s bytes)" merged.profdata
-  test -s merged.profdata
+#   stat -c "Profile data found (%s bytes)" merged.profdata
+#   test -s merged.profdata
 
-  echo "Removing instrumented JS..."
-  ./mach clobber
+#   echo "Removing instrumented JS..."
+#   ./mach clobber
 
-  echo "Building optimized JS..."
-  cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-lto=cross
-ac_add_options --enable-profile-use=cross
-ac_add_options --with-pgo-profile-path=${PWD@Q}/merged.profdata
-END
-  ./mach build
+#   echo "Building optimized JS..."
+#   cat >.mozconfig ../mozconfig - <<END
+# ac_add_options --enable-lto=cross
+# ac_add_options --enable-profile-use=cross
+# ac_add_options --with-pgo-profile-path=${PWD@Q}/merged.profdata
+# END
+#   ./mach build
 }
 
 check() {
