--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ url="https://guitarix.org"
 license=(GPL3)
 groups=(ladspa-plugins lv2-plugins pro-audio)
 depends=(atkmm bluez-libs cairo cairomm gcc-libs glibc glibmm gtkmm3 libsigc++
-libx11 pangomm ttf-roboto)
+libx11 pangomm ttf-roboto libsndfile.so)
 makedepends=(avahi boost curl gdk-pixbuf2 eigen faust fftw glib2 gperf gtk3
 intltool jack ladspa liblo liblrdf lilv lv2 pango sassc waf zita-convolver
 zita-resampler)
@@ -58,7 +58,7 @@ package() {
   depends+=(libavahi-common.so libavahi-gobject.so libboost_iostreams.so
   libcurl.so libfftw3f.so libgdk-3.so libgdk_pixbuf-2.0.so libgio-2.0.so
   libglib-2.0.so libgobject-2.0.so libjack.so liblilv-0.so liblo.so liblrdf.so
-  libpangocairo-1.0.so libpango-1.0.so libsndfile.so libzita-convolver.so
+  libpangocairo-1.0.so libpango-1.0.so libzita-convolver.so
   libzita-resampler.so)
 
   cd $pkgname-$pkgver
