diff --git PKGBUILD PKGBUILD
index 640ba10..9e5c1b0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,7 @@
   cd "$srcdir/OpenRCT2-$pkgver/build"
 
   # https://github.com/OpenRCT2/OpenRCT2/issues/17371
-  export CXXFLAGS+=" -Wno-error=maybe-uninitialized  -Wno-error=restrict  -Wno-error=null-dereference"
+  export CXXFLAGS+=" -Wno-error=maybe-uninitialized  -Wno-error=restrict  -Wno-error=null-dereference  -Wno-error=stringop-overflow"
   cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr \
     -DSTATIC=off -DWITH_TESTS=on -DCMAKE_INSTALL_LIBDIR=/usr/lib ..
   make all
