--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ url="https://github.com/kushview/element"
 license=(GPL3)
 groups=(pro-audio)
 depends=(gcc-libs glibc hicolor-icon-theme libx11 libxext)
-makedepends=(alsa-lib boost curl freetype2 git gtk3 imagemagick jack juce
+makedepends=(alsa-lib boost clang curl freetype2 git gtk3 imagemagick jack juce
 ladspa ldoc libxcomposite libxcursor libxinerama lilv lua lv2 pango readline
 serd sord sratom suil waf xorg-xrandr)
 checkdepends=(xorg-server-xvfb)
@@ -35,6 +35,10 @@ b2sums=('SKIP'
         'SKIP'
         'SKIP')
 validpgpkeys=('52CB000FABB9DBE345CD478980A5F4BE60360CDE') # Michael Fisher <mfisher@kushview.net>
+# Can't use -Wl,-plugin-opt=-target-abi=lp64d, because both CFLAGS and CXXFLAGS need it,
+# but specifying both will cause LLVMgold to complain about multiple -target-abi.
+# Also, -ffat-lto-objects can't resolve the compile error here. We have to set !lto
+options+=(!lto)
 
 prepare() {
   cd $pkgname
@@ -52,6 +56,7 @@ prepare() {
 build() {
   export LINKFLAGS="$LDFLAGS"
   CFLAGS+=" -I/usr/share/juce/modules/"
+  export CC=clang CXX=clang++
 
   cd $pkgname
   waf configure \
