--- PKGBUILD
+++ PKGBUILD
@@ -59,11 +59,19 @@
 )
 sha256sums=('2f897ebaca89b7090147e2eecd89e523ebf1f980206743e7f2177e7b9f374a09'
             '3c44a6fa1e3a8afc24754c90469404770b639cb960361988999a4cdd677699d8')
+source+=(increase-test-timeouts.patch)
+sha256sums+=('74bd6bb62012ac0faddd916664f022034c987332c896635fdd9f1be74cd07f37')
 validpgpkeys=(
   99B17F64FD5C94692E9EF8064968B2CC0208DCC8 # Matthias Bussonnier <bussonniermatthias@gmail.com>
   4D9D285A538F7C2E16D13ACE57982D11F0EA2247 # Michał Krassowski
 )
 
+prepare() {
+  cd $pkgname
+
+  patch -Np1 -i ../increase-test-timeouts.patch
+}
+
 build() {
   cd $pkgname
   python -m build --wheel --skip-dependency-check --no-isolation
