--- PKGBUILD
+++ PKGBUILD
@@ -58,11 +58,19 @@
 )
 b2sums=('714c30ebcc62d5b3a70d0f35caa0cc5f63652f07d782d93573fd06ed78554fd1589b46f1e6bb8946554229b16b6d5694ce7fac43837a10bfeadc5e2da82d9c89'
         'd445e2bc7a037db8715ea103611720e965987e155c32e445b0ef783e519fca8a0301b16c5763fd9a5d8d169c3b0d7b4db6c0bd0f9772842258b135dcb1d6d5a2')
+source+=(increase-test-timeouts.patch)
+b2sums+=('501052a04161289446663311f133ebb4df08ca4e59b79544dd08de5bfa3d4733ce2499d2cb6cf47de84ec949a52550e099963e74eb18ba8be84ca6b8e44ee1af')
 validpgpkeys=(
   99B17F64FD5C94692E9EF8064968B2CC0208DCC8 # Matthias Bussonnier <bussonniermatthias@gmail.com>
   4D9D285A538F7C2E16D13ACE57982D11F0EA2247 # Michał Krassowski
 )
 
+prepare() {
+  cd $pkgname
+
+  patch -Np1 -i ../increase-test-timeouts.patch
+}
+
 build() {
   cd $pkgname
   python -m build --wheel --skip-dependency-check --no-isolation
