--- PKGBUILD
+++ PKGBUILD
@@ -14,15 +14,20 @@ depends=('gcc-libs')
 makedepends=('clang' 'cmake' 'ninja' 'python')
 checkdepends=('llvm')
 options=(!lto)
-source=("https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver/llvm-project-$pkgver.src.tar.xz"{,.sig})
+source=("https://github.com/llvm/llvm-project/releases/download/llvmorg-$pkgver/llvm-project-$pkgver.src.tar.xz"{,.sig}
+        "${pkgname}abi-fix-demangler-for-long-double.patch")
 sha512sums=('6fc6eeb60fac698702d1aac495fc0161eb7216a1f8db2020af8fccec5837831f7cc20dc2a169bf4f0b5f520748280b4a86621f3697d622aa58faaa45dbfaad13'
-            'SKIP')
+            'SKIP'
+            'bac4a928f0dab60623f1931eb99d4d495a0c4bd40a03bff024b25c2c5b6a4d4ada21ffd5df351149cecafd0b26282156baf11b7dac66e6a66278b1d3df261745')
 validpgpkeys=('474E22316ABF4785A88C6E8EA2C794A986419D8A' # Tom Stellard <tstellar@redhat.com> (.1 releases)
               'B6C8F98282B944E3B0D5C2530FC3042E345AD05D') # Hans Wennborg <hans@chromium.org> (.0 releases)
  
 prepare() {
   mkdir -p build
   sed -i 's/CREDITS.TXT/CREDITS/' llvm-project-$pkgver.src/libcxx{,abi}/LICENSE.TXT
+
+  # patch modified from https://reviews.llvm.org/D126480
+  patch -d llvm-project-$pkgver.src -Np1 -i "${srcdir}/${pkgname}abi-fix-demangler-for-long-double.patch"
 }
  
 build() {
