diff --git PKGBUILD PKGBUILD
index 1997a0b..2783ad3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ pkgdesc="High-energy physics events generator"
 arch=('x86_64')
 url="https://pythia.org"
 license=('GPL2')
-makedepends=('fastjet' 'hepmc' 'hepmc2' 'lhapdf' 'openmp' 'python' 'root' 'zlib')
+makedepends=('fastjet' 'hepmc' 'hepmc2' 'lhapdf' 'openmp' 'python' 'zlib')
 source=("https://pythia.org/download/pythia${_pkgpre}/${_pkgid}.tgz"
         'pythia8.sh'
         'pythia8-python-3.11.patch'
@@ -73,9 +73,6 @@ build() {
         --with-python \
         --with-python-include="/usr/include/python${_pyver}" \
         --with-python-lib="/usr/lib/python${_pyver}" \
-        --with-root \
-        --with-root-include=/usr/include/root \
-        --with-root-lib=/usr/lib/root \
         --with-openmp \
         --with-openmp-include=${_inc} \
         --with-openmp-lib=${_lib}
@@ -97,7 +94,6 @@ package_pythia8() {
         'hepmc2: storing collisions from Monte Carlo (old interface)'
         'lhapdf: evaluate PDFs from discretised data files'
         'python-pythia8: for examples'
-        'root: integrated examples with CERN ROOT data analysis framework'
     )
 
     cd "${_pkgid}"
