--- a/vendor/github.com/dgraph-io/ristretto/v2/z/mmap_linux.go
+++ b/vendor/github.com/dgraph-io/ristretto/v2/z/mmap_linux.go
@@ -10,7 +10,6 @@
 
 import (
 	"os"
-	"unsafe"
 
 	"golang.org/x/sys/unix"
 )
@@ -26,24 +25,8 @@
 }
 
 // munmap unmaps a previously mapped slice.
-//
-// unix.Munmap maintains an internal list of mmapped addresses, and only calls munmap
-// if the address is present in that list. If we use mremap, this list is not updated.
-// To bypass this, we call munmap ourselves.
 func munmap(data []byte) error {
-	if len(data) == 0 || len(data) != cap(data) {
-		return unix.EINVAL
-	}
-	_, _, errno := unix.Syscall(
-		unix.SYS_MUNMAP,
-		uintptr(unsafe.Pointer(&data[0])),
-		uintptr(len(data)),
-		0,
-	)
-	if errno != 0 {
-		return errno
-	}
-	return nil
+	return unix.Munmap(data)
 }
 
 // madvise uses the madvise system call to give advise about the use of memory
--- a/vendor/github.com/dgraph-io/ristretto/v2/z/mremap_size.go
+++ b/vendor/github.com/dgraph-io/ristretto/v2/z/mremap_size.go
@@ -8,39 +8,9 @@
 
 package z
 
-import (
-	"fmt"
-	"reflect"
-	"unsafe"
-
-	"golang.org/x/sys/unix"
-)
+import "golang.org/x/sys/unix"
 
 // mremap is a Linux-specific system call to remap pages in memory. This can be used in place of munmap + mmap.
 func mremap(data []byte, size int) ([]byte, error) {
-	//nolint:lll
-	// taken from <https://github.com/torvalds/linux/blob/f8394f232b1eab649ce2df5c5f15b0e528c92091/include/uapi/linux/mman.h#L8>
-	const MREMAP_MAYMOVE = 0x1
-
-	header := (*reflect.SliceHeader)(unsafe.Pointer(&data))
-	mmapAddr, mmapSize, errno := unix.Syscall6(
-		unix.SYS_MREMAP,
-		header.Data,
-		uintptr(header.Len),
-		uintptr(size),
-		uintptr(MREMAP_MAYMOVE),
-		0,
-		0,
-	)
-	if errno != 0 {
-		return nil, errno
-	}
-	if mmapSize != uintptr(size) {
-		return nil, fmt.Errorf("mremap size mismatch: requested: %d got: %d", size, mmapSize)
-	}
-
-	header.Data = mmapAddr
-	header.Cap = size
-	header.Len = size
-	return data, nil
+	return unix.Mremap(data, size, unix.MREMAP_MAYMOVE)
 }
