--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,12 @@ makedepends=('gtkmm3-docs')
 source=("https://downloads.sourceforge.net/gtkspell/$pkgbase-$pkgver.tar.xz")
 sha256sums=('5b875a5753ce593274d0c6e803af6300973020c5443905999aba96ed3cef1545')
 
+prepare() {
+  cd $pkgname-$pkgver
+  cp /usr/share/autoconf/build-aux/config.guess build/config.guess
+  cp /usr/share/autoconf/build-aux/config.sub build/config.sub
+}
+
 build() {
   cd $pkgbase-$pkgver
   ./configure --prefix=/usr
