diff --git PKGBUILD PKGBUILD
index f21dd91..d5fbb92 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ depends=('browserpass')
 source=("${pkgname}-${pkgver}.xpi::https://addons.mozilla.org/firefox/downloads/file/4406417/browserpass_ce-${pkgver}.xpi"
         "https://raw.githubusercontent.com/browserpass/browserpass-extension/master/LICENSE")
 noextract=("${pkgname}-${pkgver}.xpi")
-b2sums=('cb03f212d32a06a12e663add29e50acc8348e375e31dcbcbb1be1252a01f892ee6a0261e1ca29fe54aa45112de44f3583a03b1ea81bea30b5a1ac22c238c653c'
+b2sums=('66ba736e23376c96b21579f2584864129a8ff6f29b1ccb5af63d7aaa5d09e5231632c464f89f81e0bcd839da5a49eee5ba33d6ebd02a1ee541dc7f66e372314b'
         '9e54d91167393d3f3159c3598aa729434c1e7d8ef39083aa40ba6241c9b4b560c64bac1af43efa7be2dac72d8c35413b057b9f67eb41934e10bd8c733437e407')
 
 package() {
