diff --git a/bridges/Library_cpp_uno.mk b/bridges/Library_cpp_uno.mk
index 5dc92c23c..d6c60ae7e 100644
--- a/bridges/Library_cpp_uno.mk
+++ b/bridges/Library_cpp_uno.mk
@@ -204,6 +204,13 @@ bridge_noopt_objects := except
 bridge_asm_objects := call
 endif
 
+else ifeq ($(OS)-$(CPUNAME),LINUX-RISCV64)
+
+bridges_SELECTED_BRIDGE := gcc3_linux_riscv64
+bridge_asm_objects := call
+bridge_noopt_objects := abi cpp2uno uno2cpp
+bridge_exception_objects := except
+
 endif
 
 $(eval $(call gb_Library_use_internal_comprehensive_api,$(CPPU_ENV)_uno,\
diff --git a/configure.ac b/configure.ac
index 03c8d6c6a..62fef427f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5135,6 +5135,11 @@ kfreebsd*)
         RTL_ARCH=X86_64
         PLATFORMID=kfreebsd_x86_64
         ;;
+    riscv64)
+        CPUNAME=RISCV64
+        RTL_ARCH=RISCV64
+        PLATFORMID=linux_riscv64
+        ;;
     *)
         AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
         ;;
diff --git a/external/firebird/UnpackedTarball_firebird.mk b/external/firebird/UnpackedTarball_firebird.mk
index 1f76870b8..65faa5090 100644
--- a/external/firebird/UnpackedTarball_firebird.mk
+++ b/external/firebird/UnpackedTarball_firebird.mk
@@ -51,6 +51,7 @@ $(eval $(call gb_UnpackedTarball_add_patches,firebird,\
     external/firebird/0001-extern-cloop-Missing-dependencies-of-compilations-on.patch.1 \
     external/firebird/configure-c99.patch \
     external/firebird/Wincompatible-function-pointer-types.patch \
+    external/firebird/firebird-riscv64-support.patch.1 \
 ))
 
 ifeq ($(OS),WNT)
