--- PKGBUILD
+++ PKGBUILD
@@ -149,6 +149,9 @@ prepare() {
         # fix build with icu 74
         patch -Np1 -i "${srcdir}"/libreoffice-7.5.8.2-icu-74-compatibility.patch
 
+        patch -Np1 -i "${srcdir}"/add-rv64-support.patch
+        cp "${srcdir}"/firebird-riscv64-support.patch.1 external/firebird
+
 	#use the CFLAGS but remove the LibO overridden ones
 	for i in $CFLAGS; do
 		case "$i" in
@@ -418,3 +421,8 @@ package_libreoffice-still() {
 	'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
 		> "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=(add-rv64-support.patch
+         firebird-riscv64-support.patch.1)
+sha256sums+=('b5624b9f2b44cea30e0d366d0e0d710ee39ae49e69fbcb248c084ef0669c8fa8'
+             '85919bb40aec323f760cdb0a1d0a1d14a32c10162426f1e213b985aca6d19d79')
