--- PKGBUILD
+++ PKGBUILD
@@ -142,6 +142,8 @@ prepare() {
     # remove breaking test breaking rule
     sed -i "109d" i18npool/source/breakiterator/data/sent.txt
 
+    patch -Np1 -i "${srcdir}/riscv64-support.patch"
+
     #use the CFLAGS but remove the LibO overridden ones
     for i in $CFLAGS; do
         case "$i" in
@@ -433,3 +435,6 @@ package_libreoffice-still() {
     'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
         > "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=(riscv64-support.patch)
+sha256sums+=('3f05bb4d781d331c5cfa2600866be7e10c6b23406ce890c3db741a2db76e79e2')
