diff --git PKGBUILD PKGBUILD
index 22adf65..6b54f84 100644
--- PKGBUILD
+++ PKGBUILD
@@ -46,3 +46,18 @@
   DESTDIR="$pkgdir" cmake --install build
   install -vDm 644 $pkgname-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source_riscv64=(
+  "$pkgname-riscv-rdtime.patch"
+  "$pkgname-riscv-address-bits.patch::https://github.com/microsoft/snmalloc/commit/bc3d5c09996c89dc9683014716203daa745f8a4a.patch?full_index=1"
+)
+sha512sums_riscv64=('04f5f4e070e94d3ec519e756d5c25b2a8f3c5cd5687895423f09deee5d17cd8bf000d0065181a3100547f41a6a690e8336c9bde1e1a3ada5b4b026bbb22c2c1a'
+                    '260d5031f1492e48df80689823563f2cc91a66092193d6f6a014e3b33aa34ba5ba223743b7eea0d3d2731efa9b95398f1f9e8fe92e16b718feaf161359ca52c9')
+b2sums_riscv64=('a27a8fcfe4279cf6e364d9414654861c35f3ec9492e2a1345a48d9ed61359771a43ca36cc2c74eb59e949b3921c889970f2e20c50b1d0189fa2cd84a879bc092'
+                '1c32ab6a076a1f866afbe8e95a1d20fac6e8335c18ba69cc360e673a44ab82ac65a9d672d9cdc39c361e461f1bc9ab136a5e05da304c26f63a174c1db6231bfb')
+
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../$pkgname-riscv-rdtime.patch
+  patch -Np1 -i ../$pkgname-riscv-address-bits.patch
+}
