--- PKGBUILD
+++ PKGBUILD
@@ -46,3 +46,12 @@
   DESTDIR="$pkgdir" cmake --install build
   install -vDm 644 $pkgname-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../$pkgname-riscv-rdtime.patch
+}
+
+source_riscv64=("$pkgname-riscv-rdtime.patch::${url}commit/481b6484ab8b1dd5cdf01a855d97152d1df15752.patch")
+sha512sums_riscv64=('b8974ff9904f413b905440b05ac860fc04557bf86b56812dd320faf161cc7a2416665bf4728614b176294b7c147125a39689d9243e3ca53d7c0f3a237e5c9e78')
+b2sums_riscv64=('cae908705a1bffcceea89cc2583aa12726d8438e87eb5e0c894b08035d46fd420a983cb6050d61ddd0762d34b07ba78830be51a8b14148f46e6c8c35c56f4aff')
