diff --git PKGBUILD PKGBUILD
index 22adf65..8c9710f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -46,3 +46,12 @@
   DESTDIR="$pkgdir" cmake --install build
   install -vDm 644 $pkgname-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source_riscv64=("$pkgname-riscv-rdtime.patch")
+sha512sums_riscv64=('04f5f4e070e94d3ec519e756d5c25b2a8f3c5cd5687895423f09deee5d17cd8bf000d0065181a3100547f41a6a690e8336c9bde1e1a3ada5b4b026bbb22c2c1a')
+b2sums_riscv64=('a27a8fcfe4279cf6e364d9414654861c35f3ec9492e2a1345a48d9ed61359771a43ca36cc2c74eb59e949b3921c889970f2e20c50b1d0189fa2cd84a879bc092')
+
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../$pkgname-riscv-rdtime.patch
+}
