diff --git PKGBUILD PKGBUILD
index 03d46c4db..46ebfa271 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,15 @@ depends=('libelf' 'zlib' 'llvm-libs' 'clang' 'bcc' 'libbpf')
 makedepends=('cmake' 'llvm' 'git' 'linux-headers' 'ninja' 'gtest' 'cereal'
              'asciidoctor' 'xxd')
 options=(!strip)
-source=("https://github.com/iovisor/bpftrace/archive/v$pkgver/bpftrace-$pkgver.tar.gz")
-sha512sums=('0be06c209ba12f70a0d1140b168dac94945c1b034dbd0c7dae5460fca43eec10bb436e28c122ec37b259256bf6b9bb23a3c5e7f8435feb67eae1b9277debaf73')
+source=("https://github.com/iovisor/bpftrace/archive/v$pkgver/bpftrace-$pkgver.tar.gz"
+        "add_riscv_support.patch")
+sha512sums=('0be06c209ba12f70a0d1140b168dac94945c1b034dbd0c7dae5460fca43eec10bb436e28c122ec37b259256bf6b9bb23a3c5e7f8435feb67eae1b9277debaf73'
+            'd586c6e1d94f6eb8e42a8dc66036e076cc822eff49d7cfaad8249be336d1b8b4554af89c13f16ea05bdf8a0bac2b51bceab8d2d5d648d412af78de76e0c7a54e')
+
+prepare() {
+  cd bpftrace-$pkgver
+  patch -Np1 -i "../add_riscv_support.patch"
+}
 
 build() {
   cd bpftrace-$pkgver
