diff --git PKGBUILD PKGBUILD
index 03d46c4db..46ebfa271 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,15 @@ depends=('libelf' 'zlib' 'llvm-libs' 'clang' 'bcc' 'libbpf')
 makedepends=('cmake' 'llvm' 'git' 'linux-headers' 'ninja' 'gtest' 'cereal'
              'asciidoctor' 'xxd')
 options=(!strip)
-source=("https://github.com/iovisor/bpftrace/archive/v$pkgver/bpftrace-$pkgver.tar.gz")
-sha512sums=('78f2d9a4d54c0dda1947b924a828777b402befc3a05031b609da4d256907df1e64f946454a9be7317449f49706f02a881febc4e63d2ef2f4e76924ea19e36ddb')
+source=("https://github.com/iovisor/bpftrace/archive/v$pkgver/bpftrace-$pkgver.tar.gz"
+        "add_riscv_support.patch")
+sha512sums=('78f2d9a4d54c0dda1947b924a828777b402befc3a05031b609da4d256907df1e64f946454a9be7317449f49706f02a881febc4e63d2ef2f4e76924ea19e36ddb'
+            'd586c6e1d94f6eb8e42a8dc66036e076cc822eff49d7cfaad8249be336d1b8b4554af89c13f16ea05bdf8a0bac2b51bceab8d2d5d648d412af78de76e0c7a54e')
+
+prepare() {
+  cd bpftrace-$pkgver
+  patch -Np1 -i "../add_riscv_support.patch"
+}
 
 build() {
   cd bpftrace-$pkgver
