--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,8 @@
 
 prepare() {
   cd ${pkgname}
+  patch -Np1 -i "$srcdir/no-vaddr-underrun.patch"
+  chmod +x tests/no-vaddr-underrun.sh
   autoreconf -fiv
 }
 
@@ -41,3 +43,7 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+source+=(no-vaddr-underrun.patch)
+sha512sums+=('ed34745a2e75c1c6973197981a7f4487548a406513e5fc098ebaa4638184488654d36ae2355312f438156980b0e89ad551fcc095a8a845f088924b037f0e9c24')
+b2sums+=('ce6c505321f48ae64ada44b44648a5bd8e0e27ca80b642ab9166b657a2e9d9a787f724ed7c7abda58e783ce8671ee18d851048aa587ce2bb76730dac58107157')
