diff --git a/libbeat/common/atomic/atomic64.go b/libbeat/common/atomic/atomic64.go
index 5d7de520..680d8bf4 100644
--- a/libbeat/common/atomic/atomic64.go
+++ b/libbeat/common/atomic/atomic64.go
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// +build amd64 arm64 ppc64 ppc64le mips64 mips64le s390x
+// +build amd64 arm64 ppc64 ppc64le mips64 mips64le s390x riscv64
 
 package atomic
 
diff --git a/vendor/github.com/elastic/go-concert/atomic/atomic64.go b/vendor/github.com/elastic/go-concert/atomic/atomic64.go
index 5d7de520..680d8bf4 100644
--- a/vendor/github.com/elastic/go-concert/atomic/atomic64.go
+++ b/vendor/github.com/elastic/go-concert/atomic/atomic64.go
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// +build amd64 arm64 ppc64 ppc64le mips64 mips64le s390x
+// +build amd64 arm64 ppc64 ppc64le mips64 mips64le s390x riscv64
 
 package atomic
 
diff --git a/go.mod b/go.mod
index 851226dc..49c78869 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/elastic/beats/v7
 
-go 1.14
+go 1.16
 
 require (
 	4d63.com/tz v1.1.1-0.20191124060701-6d37baae851b
