diff --git PKGBUILD PKGBUILD
index cd4152e4..4f281bfd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('x86_64')
 url='https://www.elastic.co/products/beats'
 license=('Apache')
 depends=('glibc')
-makedepends=('go' 'git' 'libpcap' 'rsync' 'python-virtualenv' 'audit' 'systemd')
+makedepends=('cargo' 'go' 'git' 'libpcap' 'rsync' 'python-virtualenv' 'audit' 'systemd')
 optdepends=('elasticsearch: for standalone installation'
             'python: for migration script')
 options=('!makeflags')
@@ -21,7 +21,8 @@ source=("https://github.com/elastic/beats/archive/v$pkgver/beats-$pkgver.tar.gz"
         "heartbeat.service"
         "auditbeat.service"
         "journalbeat.service"
-        "tmpfile.conf")
+        "tmpfile.conf"
+        "add_riscv_support.patch")
 sha512sums=('eba0cda8521068f4d15dfb39808695a8de47a568e4d11949746a8f698aaff6825848d5e524bd16b0e555786e277512014dfe33d4daee8c3acef9a9867653c1c3'
             '4d8b160482ba27bdc63c79592f310f2c9bcd2e8e5d3aec5ba9d953f37916bffef57c0f21e3776f4712f87e9a1b90e42dba6058f72bbc4c75380a959276183a59'
             'f1e6fe6b677db31326433f4e3eef72356573c6947d653dbe6bc2151581444f80e09343fbf8544952aae82a061b87705e39c8741ea8e402ad53ac3552f532cfea'
@@ -29,7 +30,8 @@ sha512sums=('eba0cda8521068f4d15dfb39808695a8de47a568e4d11949746a8f698aaff682584
             '5f20f160fea1d517d98cf220dc15ca7fa883fbbb3fde78255bbdd6f70ae556b4d750462af49599e27fefdc435761eedcc8bd06f4870211bc35bffa246e1e9dd3'
             'f9e015a0789946ff78a62839c9d0af40e557774e14df6de9008d5fa7786377131c74111332496ec7e6303467eb4151b185a2a4cfaa8a1e77c894be50fc22d69b'
             '45c986c3bca6fed329ff9c8f1f9c3f7de7d2a072227bed618b20cd03eb045f321995b97fa5a10a93cdbb699bfb4739d7e8256c6a99e985477532fa2dcfb58082'
-            'b6bf266c04395c6733f0e57c5acf1d8f385a2b898c36f0af201c0702ac8c8c77cbad925c01670d799a1f173e85745a17baa13fda48119b2e022f9c47f9ce4fc9')
+            'b6bf266c04395c6733f0e57c5acf1d8f385a2b898c36f0af201c0702ac8c8c77cbad925c01670d799a1f173e85745a17baa13fda48119b2e022f9c47f9ce4fc9'
+            '254144ab56dde2236b70910ff4bea099ef500fab558b52644d06e077e6a7092ebe661c1598c2b8b6e69ccb8a2cc536588b27d441380205491368756ec005a2ce')
 
 prepare() {
     export GOPATH="$srcdir"/go
@@ -51,6 +53,11 @@ prepare() {
 
     # Missing BEAT_NAME in metricbeat Makefile
     sed -i '1i BEAT_NAME?=metricbeat' metricbeat/Makefile
+
+    go mod vendor
+    patch -Np1 -i "../add_riscv_support.patch"
+    rm -rf 'vendor/github.com/godror/godror'
+    ln -s "${GOPATH}/pkg/mod/github.com/godror/godror@v0.10.4" 'vendor/github.com/godror/godror'
 }
 
 build() {
@@ -58,7 +65,7 @@ build() {
     export CGO_CFLAGS="${CFLAGS}"
     export CGO_CPPFLAGS="${CPPFLAGS}"
     export CGO_CXXFLAGS="${CXXFLAGS}"
-    export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+    export GOFLAGS="-buildmode=pie -trimpath -modcacherw"
     export GOPATH="$srcdir"/go
     export PATH="$GOPATH/bin:$PATH"
 
