--- PKGBUILD
+++ PKGBUILD
@@ -26,7 +26,7 @@ prepare() {
 build() {
   cd SuiteSparse-$pkgver
 
-  BLAS=-lblas LAPACK=-llapack TBB=-ltbb SPQR_CONFIG=-DHAVE_TBB MY_METIS_LIB=/usr/lib/libmetis.so make
+  BLAS=-lblas LAPACK=-llapack MY_METIS_LIB=/usr/lib/libmetis.so make
 }
 
 
@@ -34,7 +34,7 @@ package() {
   cd SuiteSparse-$pkgver
   install -dm755 "${pkgdir}"/usr/{include,lib}
 
-  BLAS=-lblas LAPACK=-llapack TBB=-ltbb SPQR_CONFIG=-DHAVE_TBB MY_METIS_LIB=/usr/lib/libmetis.so \
+  BLAS=-lblas LAPACK=-llapack MY_METIS_LIB=/usr/lib/libmetis.so \
   make INSTALL_LIB="${pkgdir}"/usr/lib INSTALL_INCLUDE="${pkgdir}"/usr/include install
 
   # fix RPATH
