--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,9 @@
     -e 's|~>|>=|g' \
     -e '/curb/d' \
     Gemfile
+
+  # Arch's curl build no longer enables the optional NTLM feature.
+  patch -Np1 -i ../check-ntlm-support.patch
 }
 
 build() {
@@ -99,3 +102,6 @@
 }
 
 # vim: tabstop=2 shiftwidth=2 expandtab:
+
+source+=("check-ntlm-support.patch")
+sha256sums+=('14a371dc608851e8760071a30889cfaf387e6799f5f5b3076221cd2a952f83c7')
