--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,7 @@
 
 prepare() {
   sed -e '/oldest-supported-numpy/d' -i $_pyname/pyproject.toml # Drop version constraints
+  patch -d $_pyname -Np1 -i "$srcdir/relax-nd-hypothesis-deadline.patch"
 }
 
 build() {
@@ -62,3 +63,6 @@
   cd $_pyname
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(relax-nd-hypothesis-deadline.patch)
+sha256sums+=('0acb03e3d4eb9b6b5c362cbd1a6e0f996e29790971f6f76bcd11f0fdfd4df5ce')
