Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 423444)
+++ PKGBUILD	(working copy)
@@ -19,7 +19,7 @@
              libnice lcms2 bluez-libs glu srt aom x265 liblrdf libde265 zbar
              librsvg fluidsynth lilv lv2 gst-plugins-good python git
              gobject-introspection vulkan-headers vulkan-validation-layers
-             wayland-protocols gtk3 meson shaderc libavtp svt-hevc libmicrodns
+             wayland-protocols gtk3 meson shaderc libavtp libmicrodns
              zxing-cpp opencv libva wpewebkit)
 checkdepends=(xorg-server-xvfb)
 options=(!emptydirs)
@@ -56,6 +56,7 @@
     -D voamrwbenc=disabled \
     -D wasapi2=disabled \
     -D wasapi=disabled \
+    -D svthevcenc=disabled \
     -D gobject-cast-checks=disabled \
     -D package-name="GStreamer Bad Plugins (Arch Linux)" \
     -D package-origin="https://www.archlinux.org/"
@@ -86,7 +87,7 @@
             neonhttpsrc ofa openal openexr openjpeg opusparse resindvd rsvg \
             rtmp sbc sctp smoothstreaming sndfile soundtouch spandsp srt srtp \
             teletext ttmlsubs vulkan waylandsink webp webrtc webrtcdsp \
-            wildmidi x265 zbar va svthevcenc avtp zxing microdns; do
+            wildmidi x265 zbar va avtp zxing microdns; do
     _x="lib/gstreamer-1.0/libgst${_x}.so"
     mv "$pkgdir/usr/$_x" "ext/$_x"
   done
@@ -111,7 +112,7 @@
            libmpcdec neon libofa openal openexr openjpeg2 opus libdvdnav
            libdvdread librsvg rtmpdump sbc libsndfile soundtouch spandsp srt
            libsrtp zvbi vulkan-icd-loader libxcb wayland libwebp libnice
-           webrtc-audio-processing wildmidi x265 zbar libavtp svt-hevc
+           webrtc-audio-processing wildmidi x265 zbar libavtp
            libmicrodns zxing-cpp libva)
   optdepends=('nvidia-utils: nvcodec plugin')
 
