diff --git PKGBUILD PKGBUILD
index 7dcc5164..401e551b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -24,10 +24,12 @@
              'libnet' 'bash' 'harfbuzz' 'gcc-libs' 'glibc')
 options=(!lto)
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
+        java17-riscv64.patch
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
         freedesktop-jshell.desktop)
 sha256sums=('b4587238b714764dd77720e5c71e6f44871b09468810c4e82f51408ae0172989'
+            '43c9247df000a9b2e35ff7b1c924f8685b075ce89f9e4c74a9a197f9b8ef7872'
             '85c4742764590783160af74587a47269e6797fbdf17ec485c7644bd239adf61d'
             'abac1ab09a33a3654378bc69be5a7cf075263ab3ae9efec1eb25cf388e711bb7'
             'e7cce0ecf868f656d8dc2eb25ab82ab665526a0a28aba20f02632dd29962dac3')
@@ -35,6 +37,7 @@
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -46,6 +49,13 @@
               lib/libjsound.{so,debuginfo}
               lib/libsplashscreen.{so,debuginfo})
 
+prepare() {
+  cd ${_jdkdir}
+  cp -fv /usr/share/autoconf/build-aux/config.guess make/autoconf/build-aux/
+  cp -fv /usr/share/autoconf/build-aux/config.sub make/autoconf/build-aux/
+  patch -Np1 -i ../java17-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
