--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,7 @@
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -46,6 +47,13 @@
               lib/libjsound.so
               lib/libsplashscreen.so)
 
+prepare() {
+  cd ${_jdkdir}
+  cp -fv /usr/share/autoconf/build-aux/config.guess make/autoconf/build-aux/
+  cp -fv /usr/share/autoconf/build-aux/config.sub make/autoconf/build-aux/
+  patch -Np1 -i ../java17-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -286,4 +294,7 @@
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+source+=("java17-riscv64.patch")
+sha256sums+=('43c9247df000a9b2e35ff7b1c924f8685b075ce89f9e4c74a9a197f9b8ef7872')
+
 # vim: ts=2 sw=2 et:
