--- PKGBUILD
+++ PKGBUILD
@@ -35,3 +35,11 @@ package() {
   install -Dm644 LICENSE-GPL-2.0-only "$pkgdir"/usr/share/licenses/$pkgname/LICENSE-GPL-2.0-only
   install -Dm644 LICENSE-LGPL-2.1-only "$pkgdir"/usr/share/licenses/$pkgname/LICENSE-LGPL-2.1-only
 }
+
+source+=("$pkgname-riscv64.patch::https://github.com/containers/libkrunfw/commit/79d9ff95ebc8e15e39b2296e65388f6e8aded4be.diff")
+sha256sums+=('60349d5f269d28bf487dfb90718f542300a6be7553d1ff49d5f46a6776015994')
+
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../"$pkgname-riscv64.patch"
+}
