diff --git PKGBUILD PKGBUILD
index ae45620..7fbf5ae 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ url="https://wiki.gnome.org/Apps/Vinagre"
 arch=(x86_64)
 license=(GPL)
 depends=(libsecret gtk-vnc vte3 telepathy-glib avahi dconf spice-gtk3 gnutls freerdp)
-makedepends=(docbook-xsl intltool openssh itstool spice-protocol vala gnome-common appstream-glib git)
+makedepends=(docbook-xsl intltool openssh itstool spice-protocol vala gnome-common appstream-glib git gobject-introspection)
 optdepends=('openssh: SSH plugin')
 _commit=adcb63cfad988a92126a751b6e8f7cfef0762955  # master
 source=("git+https://gitlab.gnome.org/GNOME/vinagre.git#commit=$_commit"
