diff --git PKGBUILD PKGBUILD
index e83d114..4c364a5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,11 @@ depends=(
 source=("https://sourceforge.net/projects/libemf/files/libemf/${pkgver}/${pkgname}-${pkgver}.tar.gz")
 sha256sums=('74d92c017e8beb41730a8be07c2c6e4ff6547660c84bf91f832d8f325dd0cf82')
 
+prepare(){
+    cd "${pkgname}-${pkgver}"
+    sed -i 's/^.*#.*error Unknown CPU architecture!.*/#undef WORDS_BIGENDIAN\n#undef BITFIELDS_BIGENDIAN\n#undef ALLOW_UNALIGNED_ACCESS/' include/libEMF/wine/winnt.h
+    sed -i 's/^.*#.*error You need to define a CONTEXT.*/typedef struct _CONTEXT { DWORD ContextFlags; } CONTEXT;/' include/libEMF/wine/winnt.h
+}
 build() {
     cd "${pkgname}-${pkgver}"
     ./configure \
