diff --git a/sim/common/cgen-trace.c b/sim/common/cgen-trace.c
index 43bf017..264415a 100644
--- a/sim/common/cgen-trace.c
+++ b/sim/common/cgen-trace.c
@@ -168,7 +168,7 @@ cgen_trace_insn (SIM_CPU *cpu, const struct cgen_insn *opcode,
   if (CGEN_INSN_VIRTUAL_P (opcode))
     {
       trace_prefix (CPU_STATE (cpu), cpu, NULL_CIA, pc, 0,
-		    NULL, 0, CGEN_INSN_NAME (opcode));
+		    NULL, 0, "%s", CGEN_INSN_NAME (opcode));
       return;
     }
 
diff --git a/sim/common/cgen-utils.c b/sim/common/cgen-utils.c
index 8e87ceb..f45b5e6 100644
--- a/sim/common/cgen-utils.c
+++ b/sim/common/cgen-utils.c
@@ -428,7 +428,7 @@ cgen_rtx_error (SIM_CPU *cpu, const char * msg)
 {
   SIM_DESC sd = CPU_STATE (cpu);
 
-  sim_io_printf (sd, msg);
+  sim_io_printf (sd, "%s", msg);
   sim_io_printf (sd, "\n");
 
   sim_engine_halt (sd, cpu, NULL, CPU_PC_GET (cpu), sim_stopped, SIM_SIGTRAP);
