--- PKGBUILD
+++ PKGBUILD
@@ -9,11 +9,14 @@ url="https://staticcheck.io"
 license=("MIT")
 depends=('glibc')
 makedepends=('go' 'git')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/dominikh/go-tools/archive/$pkgver.tar.gz")
-sha256sums=('c3fcadc203e20bc029abc9fc1d97b789de4e90dd8164e45489ec52f401a2bfd0')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/dominikh/go-tools/archive/$pkgver.tar.gz"
+        "fix-riscv-test.patch::https://patch-diff.githubusercontent.com/raw/dominikh/go-tools/pull/1158.patch")
+sha256sums=('c3fcadc203e20bc029abc9fc1d97b789de4e90dd8164e45489ec52f401a2bfd0'
+            '6671044a457a0e21696e5a223fdb8c0d653beaa66470fe6ad7cb4fd7a5ad8568')
 
 prepare(){
 	cd "go-tools-$pkgver"
+	patch -Np1 < ../fix-riscv-test.patch
 	mkdir -p bin
 }

