diff --git PKGBUILD PKGBUILD
index d4eac605..b103eee2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,12 @@ options=(!lto)
 source=("https://github.com/decathorpe/mitmproxy_wireguard/archive/$pkgver/$pkgname-$pkgver.tar.gz")
 sha512sums=('157f1e7b1d4a47920bee39c506643954c28395eeca58c4024f2bcdb1d170d2842c110652b28eec4307838f378c8187a79bb8c99b4fc9db8dd95cd2bda097097d')
 
+prepare() {
+  cd ${_pyname}-${pkgver}
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   cd ${_pyname}-${pkgver}
   maturin build --release --strip
