--- PKGBUILD
+++ PKGBUILD
@@ -10,16 +10,15 @@ license=('BSD')
 depends=('eigen' 'python-numpy' 'pybind11')
 makedepends=('cmake' 'git' 'ninja')
 source=("git+https://github.com/laurentkneip/opengv.git#commit=$_commit"
-        https://salsa.debian.org/science-team/opengv/-/raw/master/debian/patches/no-march-options.patch
+        "riscv64-no-march-native.patch"
         https://salsa.debian.org/science-team/opengv/-/raw/master/debian/patches/use-system-pybind11.patch)
 sha256sums=('SKIP'
-            'e483d0c1a4a76920ac3260999978154b12422c42e9991e69afc9230643d5c24a'
+            'eb4a32d838a64b58652da3249c9218a6028066f72865414d6c75069cfc88eadc'
             '20832cd52808bd7ef2f6bee8a5c85f69fc0fe6c82169e2b2297781f4bdcdd3ae')

 prepare() {
   cd opengv
-
-  patch -Np1 -i "${srcdir}/no-march-options.patch"
+  patch -Np1 -i "${srcdir}/riscv64-no-march-native.patch"
   patch -Np1 -i "${srcdir}/use-system-pybind11.patch"
 }

