--- PKGBUILD
+++ PKGBUILD
@@ -10,9 +10,11 @@ license=('BSD')
 depends=('eigen' 'python-numpy')
 makedepends=('cmake' 'git' 'ninja')
 source=("git+https://github.com/laurentkneip/opengv.git#commit=$_commit"
-        "git+https://github.com/pybind/pybind11.git")
+        "git+https://github.com/pybind/pybind11.git"
+        "riscv64-no-march-native.patch")
 sha256sums=('SKIP'
-            'SKIP')
+            'SKIP'
+            'eb4a32d838a64b58652da3249c9218a6028066f72865414d6c75069cfc88eadc')
 
 prepare() {
   cd opengv
@@ -20,6 +22,8 @@ prepare() {
   git submodule init
   git config submodule.pybind11.url "${srcdir}/pybind11"
   git submodule update
+
+  patch -p1 < ../riscv64-no-march-native.patch
 }
 
 build() {
