diff --git a/PKGBUILD b/PKGBUILD
index 0689f1b..1ae7866 100644
--- PKGBUILD
+++ PKGBUILD
@@ -49,11 +49,15 @@ build() {
 
   # Build Speech Tools first
   cd speech_tools
+  cp /usr/share/autoconf/build-aux/config.guess .
+  cp /usr/share/autoconf/build-aux/config.sub .
   ./configure --prefix=/usr --sysconfdir=/etc
   make OPTIMISE_CXXFLAGS="$CPPFLAGS $CXXFLAGS" OPTIMISE_CCFLAGS="$CPPFLAGS $CFLAGS $LDFLAGS" LINKFLAGS="$LDFLAGS" MAKE_SHARED_LIB='$(CXX) $(LINKFLAGS) -shared -o XXX -Wl,-soname,YYY'
 
   # Build Festival itself
   cd ../festival
+  cp /usr/share/autoconf/build-aux/config.guess .
+  cp /usr/share/autoconf/build-aux/config.sub .
   ./configure --prefix=/usr --sysconfdir=/etc
   make OPTIMISE_CXXFLAGS="$CPPFLAGS $CXXFLAGS" OPTIMISE_CCFLAGS="$CPPFLAGS $CFLAGS $LDFLAGS" LINKFLAGS="$LDFLAGS" MAKE_SHARED_LIB='$(CXX) $(LINKFLAGS) -shared -o XXX -Wl,-soname,YYY'
 }
