--- PKGBUILD
+++ PKGBUILD
@@ -8,11 +8,14 @@ arch=('x86_64')
 url="https://github.com/cargo-bins/cargo-binstall"
 license=('GPL3')
 depends=('cargo' 'gcc-libs')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha256sums=('a0a6f30203488045d47af9d6ae86fa7ef5fd793f7c5407804f24fe78c4e787c0')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
+        "fix-client-test.patch::https://github.com/cryolitia-forks/cargo-binstall/commit/40a0a0e572d365447b49e5b58c1acd0c64371644.patch")
+sha256sums=('a0a6f30203488045d47af9d6ae86fa7ef5fd793f7c5407804f24fe78c4e787c0'
+            '238bc33b3552fa58e9acf3142a1f1a4b9a78dbffb4bf70fe5d0e303a81606e46')
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i "$srcdir/fix-client-test.patch"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
