--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,9 @@ sha256sums=('82db29d3e990fefcbeab9447da1b13e4e62b31aceeff38b8860510f9acdd357b')
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
+  cargo fetch --locked 
 }
 
 build() {
