Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420421)
+++ PKGBUILD	(working copy)
@@ -17,11 +17,13 @@
 source=("https://deb.debian.org/debian/pool/main/f/$pkgname/${pkgname}_${pkgver}.orig.tar.gz"
         fakeroot-1.25.3-glibc-2.33-fix-1.patch
         fakeroot-1.25.3-glibc-2.33-fix-2.patch
-        fakeroot-1.25.3-glibc-2.33-fix-3.patch)
+        fakeroot-1.25.3-glibc-2.33-fix-3.patch
+        fakeroot-1.25.3-glibc-2.33-fix-4.patch::https://salsa.debian.org/clint/fakeroot/-/merge_requests/12.patch)
 sha256sums=('8e903683357f7f5bcc31b879fd743391ad47691d4be33d24a76be3b6c21e956c'
             '7b1ea49a4123a6d95329b370cc9edb51dc0c292ee3eeaa034757b555607026ea'
             '1d2b3c9de24a4249f71e4ed1465ccac462c4ae36bd0a72c796cbfc51fb88a5e6'
-            '669dc27fc4a63d265adffbd7fe04521d9e91f46fdba5465c55c6b2d281aae0dc')
+            '669dc27fc4a63d265adffbd7fe04521d9e91f46fdba5465c55c6b2d281aae0dc'
+            'ea4aa585e9c06e63fe57b12971e813e81fa7b4381f5fd84ece5395daf3ccc739')
 
 prepare() {
   cd $pkgname-$pkgver
@@ -29,6 +31,7 @@
   patch -p1 -i $srcdir/fakeroot-1.25.3-glibc-2.33-fix-1.patch
   patch -p1 -i $srcdir/fakeroot-1.25.3-glibc-2.33-fix-2.patch
   patch -p1 -i $srcdir/fakeroot-1.25.3-glibc-2.33-fix-3.patch
+  patch -p1 -i $srcdir/fakeroot-1.25.3-glibc-2.33-fix-4.patch
 
   ./bootstrap
 }
