Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 460916)
+++ PKGBUILD	(working copy)
@@ -13,9 +13,11 @@
 makedepends=('cmake' 'python' 'pybind11' 'ninja')
 optdepends=('python: python bindings')
 source=($pkgname-$pkgver.tar.gz::https://github.com/AcademySoftwareFoundation/OpenColorIO/archive/v$pkgver.tar.gz
-        use-system-minizip.patch)
+        use-system-minizip.patch
+        0001-CheckSupportSSE2-Fix-sse-flags-unexpected-propagatio.patch)
 sha512sums=('f9fc6991f99f28bb2117ecf6af6ea907310c9ae118d17e54c1bf642ec99e35bf899463d80bccdbaca6cd66bae62e17fdd0417e2fb42c8b8f80c6892e7cbe8770'
-            'de760fa88f9680e9bd02c3810957f68f82ef461591763de47e4ffa31739aebd1ebf0793dd7e93582aaef11afd9d4ba088f8911259c749dc6a74b9cf4b163470e')
+            'de760fa88f9680e9bd02c3810957f68f82ef461591763de47e4ffa31739aebd1ebf0793dd7e93582aaef11afd9d4ba088f8911259c749dc6a74b9cf4b163470e'
+            '614dc60b42a1f6ed9bfc72a4dc5f93d409040d9521462adad4ffbde346b4bafc438081e36a99f61c4b4d63260d01fe5ae722f51d4a022affefcb4e72cbf3765a')
 
 prepare() {
   cd OpenColorIO-$pkgver
@@ -22,6 +24,9 @@
 
   # We don't care about the failures of this patch in this current release
   patch -Np1 -i "$srcdir"/use-system-minizip.patch
+
+  # Fix sse flags unexpected propagation
+  patch -Np1 -i "$srcdir"/0001-CheckSupportSSE2-Fix-sse-flags-unexpected-propagatio.patch
 }
 
 build() {
