diff --git PKGBUILD PKGBUILD
index e0a4e2ec..140abf58 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,8 +17,10 @@ optdepends=('texlive-bin: DVI support')
 provides=(libev{document,view}3.so)
 groups=(gnome)
 _commit=e6d860a3058732c83b3774d705650bc516a294ae  # tags/41.2^0
-source=("git+https://gitlab.gnome.org/GNOME/evince.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/evince.git#commit=$_commit"
+        "fix-meson-build.patch::https://gitlab.gnome.org/GNOME/evince/-/merge_requests/398.patch")
+sha256sums=('SKIP'
+            '6d10a0ec01b885a6fadfc5a804ea8c3ee0aaf37a03088284bcfe8d2f8023a258')
 
 pkgver() {
   cd $pkgname
@@ -27,6 +29,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i $srcdir/fix-meson-build.patch
 }
   
 build() {
