--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,9 @@ makedepends=('perl-xml-xpath' 'python' 'wayland' 'libxrandr' 'xorg-server-devel'
 options=('!lto')
 source=("https://github.com/GPUOpen-Drivers/AMDVLK/archive/v-${pkgver}.tar.gz")
 sha256sums=('be2ebaf7f0df1076fd9bffc920cfa6a039d7b51f1ba2c3ae72a0511b644d7b17')
+
+source+=("fix-stb_sprintf.patch::https://github.com/nothings/stb/pull/1517.diff")
+sha256sums+=('adaac69864658e8af799cee6272ea6d3504c0021b8ba922e2c867d2cc9c0a62a')
             
 prepare() {
   local nrepos path name revision
@@ -32,6 +35,9 @@ prepare() {
       popd
     (( nrepos-- ))
   done
+
+  cd pal/shared/devdriver/third_party/stb_sprintf/inc/
+  patch -Np1 -i $srcdir/fix-stb_sprintf.patch
 }
 
 build() {
