--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,9 @@ prepare() {
   cd ${srcdir}/pal/shared/devdriver/third_party/rapidjson/
   patch -Np1 -i ${srcdir}/0001-Avoid-inheritance-from-std-iterator.patch
   patch -Np1 -i ${srcdir}/0001-Removed-non-compiling-assignment-operator.-Fixed-718.patch
+
+  cd ${srcdir}/pal/shared/devdriver/third_party/stb_sprintf/inc/
+  patch -Np1 -i $srcdir/fix-stb_sprintf.patch
 }
 
 build() {
@@ -69,3 +72,5 @@ package() {
   sed -i "s#/lib64#/lib#g" "${pkgdir}"/usr/share/vulkan/icd.d/amd_icd64.json
   sed -i "s#/lib64#/lib#g" "${pkgdir}"/usr/share/vulkan/implicit_layer.d/amd_icd64.json
 }
+source+=("fix-stb_sprintf.patch::https://github.com/nothings/stb/pull/1517.diff")
+sha256sums+=('9e97e130813bd058bae57aadc1c451d0ba29db5deff821d050449460f5e2901c')
