--- PKGBUILD
+++ PKGBUILD
@@ -32,6 +32,9 @@ prepare() {
       popd
     (( nrepos-- ))
   done
+
+  cd pal/shared/devdriver/third_party/stb_sprintf/inc/
+  patch -Np1 -i $srcdir/fix-stb_sprintf.patch
 }
 
 build() {
@@ -39,6 +42,7 @@ build() {
   cmake -H. -Bbuilds/Release64 \
     -DCMAKE_BUILD_TYPE=Release \
     -DBUILD_WAYLAND_SUPPORT=On \
+    -DCMAKE_POLICY_VERSION_MINIMUM=3.5 \
     -G Ninja
     
   ninja -C builds/Release64
@@ -60,3 +64,5 @@ package() {
   sed -i "s#/lib64#/lib#g" "${pkgdir}"/usr/share/vulkan/icd.d/amd_icd64.json
   sed -i "s#/lib64#/lib#g" "${pkgdir}"/usr/share/vulkan/implicit_layer.d/amd_icd64.json
 }
+source+=("fix-stb_sprintf.patch::https://github.com/nothings/stb/pull/1517.diff")
+sha256sums+=('9e97e130813bd058bae57aadc1c451d0ba29db5deff821d050449460f5e2901c')
