Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -30,7 +30,7 @@
 
   # ensure reproducibility of .pyc files
   touch -d @$SOURCE_DATE_EPOCH lang/python/version.py
-  touch -d @$SOURCE_DATE_EPOCH lang/python/python${_python_ver}-gpg/lib.linux-x86_64-${_python_ver}/gpg/gpgme.py
+  touch -d @$SOURCE_DATE_EPOCH lang/python/python${_python_ver}-gpg/lib.linux-$CARCH-${_python_ver}/gpg/gpgme.py
 
   # .pyc files will be created if check() is used - generate them here
 
