Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 405853)
+++ PKGBUILD	(working copy)
@@ -3,7 +3,7 @@
 # Contributor: Sarah Hay <sarah@archlinux.org>
 
 pkgbase=gpgme
-pkgname=(gpgme qgpgme python-gpgme)
+pkgname=(gpgme python-gpgme)
 pkgver=1.16.0
 pkgrel=1
 _python_ver=3.9
@@ -11,7 +11,7 @@
 arch=('x86_64')
 url='https://www.gnupg.org/related_software/gpgme/'
 license=('LGPL')
-makedepends=('libgpg-error' 'gnupg' 'qt5-base' 'python' 'swig')
+makedepends=('libgpg-error' 'gnupg' 'python' 'swig')
 validpgpkeys=('6DAA6E64A76D2840571B4902528897B826403ADA') # Werner Koch (dist signing 2020)
 source=("https://www.gnupg.org/ftp/gcrypt/${pkgbase}/${pkgbase}-${pkgver}.tar.bz2"{,.sig})
 sha256sums=('eebc3c1b27f1c8979896ff361ba9bb4778b508b2496c2fc10e3775a40b1de1ad'
@@ -30,7 +30,7 @@
 
   # ensure reproducibility of .pyc files
   touch -d @$SOURCE_DATE_EPOCH lang/python/version.py
-  touch -d @$SOURCE_DATE_EPOCH lang/python/python${_python_ver}-gpg/lib.linux-x86_64-${_python_ver}/gpg/gpgme.py
+  touch -d @$SOURCE_DATE_EPOCH lang/python/python${_python_ver}-gpg/lib.linux-$CARCH-${_python_ver}/gpg/gpgme.py
 
   # .pyc files will be created if check() is used - generate them here
 
@@ -59,8 +59,6 @@
   make DESTDIR="${pkgdir}" install
 
   # split qgpgme
-  rm -r "${pkgdir}"/usr/include/{qgpgme,QGpgME}/
-  rm -r "${pkgdir}"/usr/lib/{cmake/QGpgme/,libqgpgme.*}
   rm -r "${pkgdir}"/usr/lib/python*
 }
 
