--- PKGBUILD
+++ PKGBUILD
@@ -17,2 +17,7 @@
 
+prepare() {
+  cd "neural-search-${pkgver}"
+  patch -Np1 -i ../stable-collapse-scores.patch
+}
+
 build() {
@@ -23,3 +28,3 @@
   # integTest (Reaper) requires JDK 14
-  ./gradlew build \
+  gradle build \
     --exclude-task "integTest" \
@@ -36 +41,5 @@
 }
+
+makedepends+=('gradle')
+source+=('stable-collapse-scores.patch')
+sha256sums+=('075f59167e41c4e32b1b2a77ae53f79274c8df8fed99972e11474b77cb7b35dc')
