--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,8 @@ source=(https://github.com/MegaGlest/megaglest-source/releases/download/${pkgver
         https://github.com/MegaGlest/megaglest-source/commit/5a352054.patch
         https://github.com/MegaGlest/megaglest-source/commit/407b7729.patch
         fbd0cfb1.patch
-        fix-crash-on-network-game-start.patch        )
+        fix-crash-on-network-game-start.patch
+        bump-cmake-minver.patch)
 sha256sums=('09fef67bc6878b1afbb0b403fec9fcf7dc270d500fdc62ed98e5f210e398ccb9'
             'SKIP'
             'dbb8e5b6953bd2f86ecbb9c60ac00c0c25299f2738fbdc39996adaed844cd6ed'
@@ -29,7 +30,8 @@ sha256sums=('09fef67bc6878b1afbb0b403fec9fcf7dc270d500fdc62ed98e5f210e398ccb9'
             'b8104c81839e8282f624891b3e102f27de0852950e13f52b938a7c65d54bb357'
             '5ea849fd5f301367ddd68132e38e07ea66e5657d24cd241ccddf919fd43044bb'
             '5dc0096ea581ca9436b2de528f913ac99de7a1a5d00bc7e8d52117298fd1eeb5'
-            'eedc863bb1eb2cbdaa807b0cca37dfacd189f57e3018f4aabc949c321d13d0e8')
+            'eedc863bb1eb2cbdaa807b0cca37dfacd189f57e3018f4aabc949c321d13d0e8'
+            '20cf6738c0b22d82c024898e5eddb17a7a05a0819da0fafd5d3ad79ac517e958')
 validpgpkeys=('0DA321A44B30EF477E9A6DB8DE45EB9E71EEB59B')
 
 prepare() {
@@ -43,6 +45,7 @@ prepare() {
   patch -p1 < ../fbd0cfb1.patch
   patch -p1 < ../407b7729.patch
   patch -p1 < ../fix-crash-on-network-game-start.patch
+  patch -p1 < ../bump-cmake-minver.patch
 }
 
 build() {
