--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,19 @@ license=('GPL3')
 depends=('curl' 'megaglest-data' 'xerces-c' 'sdl2' 'libvorbis' 'openal' 'libgl' 'lua51' 'icu' 'ftgl' 'glew'
          'libircclient' 'miniupnpc' 'wxgtk3' 'glu' 'libx11' 'libminiupnpc.so')
 makedepends=('ftjam' 'cmake' 'mesa')
-source=(https://github.com/MegaGlest/megaglest-source/releases/download/${pkgver}/megaglest-source-${pkgver}.tar.xz{,.asc})
+source=(https://github.com/MegaGlest/megaglest-source/releases/download/${pkgver}/megaglest-source-${pkgver}.tar.xz{,.asc}
+        ftp-fixes.patch::https://github.com/MegaGlest/megaglest-source/commit/5a3520540276a6fd06f7c88e571b6462978e3eab.patch)
 sha256sums=('09fef67bc6878b1afbb0b403fec9fcf7dc270d500fdc62ed98e5f210e398ccb9'
-            'SKIP')
+            'SKIP'
+            'b8104c81839e8282f624891b3e102f27de0852950e13f52b938a7c65d54bb357')
 validpgpkeys=('0DA321A44B30EF477E9A6DB8DE45EB9E71EEB59B')
 
+prepare() {
+  cd megaglest-${pkgver}
+
+  patch -Np1 -i ../ftp-fixes.patch
+}
+
 build() {
   cd megaglest-${pkgver}
 
