Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 410243)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 arch=(x86_64)
 url='https://strace.io/'
 license=(BSD)
-depends=(perl libunwind)
+depends=(perl)
 source=(https://github.com/strace/strace/releases/download/v$pkgver/strace-$pkgver.tar.xz{,.asc})
 sha1sums=('1b7a533a45b9ca351d7a14702c044b917d11e979'
           'SKIP')
@@ -16,7 +16,7 @@
 
 build() {
   cd $pkgname-$pkgver
-  ./configure --prefix=/usr --with-libunwind --enable-mpers=no
+  ./configure --prefix=/usr --enable-mpers=no
   make
 }
 
