--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,13 @@ pkgver() {
 prepare() {
   cd charm
 
+  # replace dependency with its successor to support riscv64
+  # remove after https://github.com/charmbracelet/charm/pull/178 is merged
+  sed -i "/github.com\/jacobsa\/crypto/d" go.mod
+  go get -v github.com/aperturerobotics/jacobsa-crypto
+  sed -i 's|"github.com/jacobsa/crypto/siv"|"github.com/aperturerobotics/jacobsa-crypto/siv"|' crypt/crypt.go
+  go mod tidy
+
   # create directory for build output
   mkdir build
 
