--- PKGBUILD
+++ PKGBUILD
@@ -46,6 +46,11 @@ sha256sums=('a0f6bd102a8213df8fadc66e7ec1d6611e841f14d954524e31494ad9bd26c0b1'
             'SKIP')
 validpgpkeys=('C11C0A4DF702EDF6C04F458C18ADB31CF18AD6BB') # Antoine Martin <antoine@nagafix.co.uk>
 
+prepare() {
+  cd "${srcdir}/$pkgname-$pkgver"
+  sed -i 's/r = subprocess.Popen(cmd).wait(30)/r = subprocess.Popen(cmd).wait(300)/' setup.py
+}
+
 build() {
   cd "${srcdir}/$pkgname-$pkgver"
   python setup.py build
