diff --git a/setup.py b/setup.py
index 5c71e3d..5b9225a 100755
--- a/setup.py
+++ b/setup.py
@@ -2353,6 +2353,7 @@ if vpx_ENABLED:
 toggle_packages(enc_ffmpeg_ENABLED, "xpra.codecs.enc_ffmpeg")
 if enc_ffmpeg_ENABLED:
     ffmpeg_pkgconfig = pkgconfig("libavcodec", "libavformat", "libavutil")
+    add_to_keywords(ffmpeg_pkgconfig, 'extra_compile_args', "-Wno-deprecated-declarations")
     add_cython_ext("xpra.codecs.enc_ffmpeg.encoder",
                 ["xpra/codecs/enc_ffmpeg/encoder.pyx"],
                 **ffmpeg_pkgconfig)
