--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,10 @@
 sha512sums=('e57face42b62a6ae552ad2f3a20251bc7e4913d7114355535f1581899da9c45d33b27976fc09670452dbe870b3bac00d0831c0556756ae831c9121769b7e6a05')

+source+=(fix-signed-memset.patch)
+sha512sums+=('bfcb2ce78fb63d2e1ba7ae0082e538184bf6de37951deb482c7446fceab4b08b5f4ec85b1341fd55231886390a6dd60d549b2b843aef15c62c04296f8ee1b2ed')
+
 prepare() {
   cd primitive-$pkgver
+  patch -p1 -i ../fix-signed-memset.patch
   gen-setup
 }
