diff --git PKGBUILD PKGBUILD
index dda1cdc..29750a5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,7 @@
 install=stone-soup.install
 source=("stone-soup-$pkgver.tar.xz::https://github.com/crawl/crawl/releases/download/$pkgver/stone_soup-$pkgver-nodeps.tar.xz")
 md5sums=('ce8e390097fbfb3f56e8f6744ab64736')
+options=(!lto) # FS#73605
 
 package() {
     gendesk -n --pkgname "$pkgname" --pkgdesc "$pkgdesc" \
