--- PKGBUILD
+++ PKGBUILD
@@ -67,6 +67,9 @@
 
   # Fix LUA xml parser tests
   git apply -3 ../0002-lua-xml-Fix-lua-tests.patch
+
+  git apply --include=tests/lua-factory/sources/test_lua_theaudiodb.c \
+    ../grilo-plugins-variadic-terminator.patch
 }
 
 build() {
@@ -89,3 +92,6 @@
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=("grilo-plugins-variadic-terminator.patch::https://gitlab.gnome.org/GNOME/grilo-plugins/-/commit/d88caf1c500c3200f5eadac1ea24533ef9f392c1.patch")
+b2sums+=('c6630a32dc82f8d485f96a13ad07ea15924bcda762a721344aff075698ac00d4461a3536420b91d9828504ced983b14c0abea50f10ac516d705cfa7738d8f9d8')
