diff -uprN gupnp-tools-0.10.1.orig/src/universal-cp/main.c gupnp-tools-0.10.1/src/universal-cp/main.c
--- gupnp-tools-0.10.1.orig/src/universal-cp/main.c	2021-11-09 03:07:59.535922857 +0800
+++ gupnp-tools-0.10.1/src/universal-cp/main.c	2021-11-09 03:08:29.909665936 +0800
@@ -74,7 +74,7 @@ parse_uda_version (const gchar *option_n
                 g_set_error (error,
                              G_IO_ERROR,
                              G_IO_ERROR_INVALID_ARGUMENT,
-                             msg);
+                             "%s", msg);
                 g_free (msg);
 
                 return FALSE;
