diff --git PKGBUILD PKGBUILD
index 4467e4c..cb31779 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,14 @@ url="https://wiki.gnome.org/Projects/GUPnP"
 license=('GPL')
 depends=('gtksourceview4' 'gupnp' 'gupnp-av')
 makedepends=('meson')
-source=("https://download.gnome.org/sources/$pkgname/${pkgver%.*}/$pkgname-$pkgver.tar.xz")
-sha256sums=('4ea96d167462b3a548efc4fc4ea089fe518d7d29be349d1cce8982b9ffb53b4a')
+source=("https://download.gnome.org/sources/$pkgname/${pkgver%.*}/$pkgname-$pkgver.tar.xz"
+        "fix-format-string.patch")
+sha256sums=('4ea96d167462b3a548efc4fc4ea089fe518d7d29be349d1cce8982b9ffb53b4a'
+            '3f40f57207ef9d1259b6393444c45b10f0a8696bd416c57e71137931455110b4')
+
+prepare() {
+  patch -Np1 -d $pkgname-$pkgver -i $srcdir/fix-format-string.patch
+}
 
 build() {
   arch-meson $pkgname-$pkgver build
