--- PKGBUILD
+++ PKGBUILD
@@ -19,8 +19,6 @@ backup=(etc/$pkgname/$pkgname.conf)
 prepare() {
   echo '# Placeholder config. See https://pgbackrest.org/configuration.html for more info.' > example.conf
 
-  export CFLAGS="$CFLAGS -fcf-protection"
-  export CXXFLAGS="$CXXFLAGS -fcf-protection"
   export LDFLAGS="$LDFLAGS -Wl,-z,relro -Wl,-z,now -Wl,-z,shstk"
 
   arch-meson $pkgname-release-$pkgver build
