diff -Naur freehdl-0.0.8_0/v2cc/v2cc-explore.cc freehdl-0.0.8_1/v2cc/v2cc-explore.cc
--- freehdl-0.0.8_0/v2cc/v2cc-explore.cc	2010-04-13 02:40:40.000000000 +0800
+++ freehdl-0.0.8_1/v2cc/v2cc-explore.cc	2025-01-18 22:59:49.198191595 +0800
@@ -1633,6 +1633,7 @@
   choice_descriptor &operator=(const choice_descriptor &src) {
     choice_node = src.choice_node;
     choice_value = src.choice_value;
+    return *this;
   }
 };

diff -Naur freehdl-0.0.8_0/v2cc/v2cc-impl.cc freehdl-0.0.8_1/v2cc/v2cc-impl.cc
--- freehdl-0.0.8_0/v2cc/v2cc-impl.cc	2010-05-08 20:16:33.000000000 +0800
+++ freehdl-0.0.8_1/v2cc/v2cc-impl.cc	2025-01-18 23:05:49.575378359 +0800
@@ -1031,6 +1031,8 @@
   for (pIIR_ElementDeclarationList edl = rt->element_declarations;
        edl; edl = edl->rest)
     str += ".set(" + to_string(i++) + "," + get_type_info_obj(edl->first->subtype, rstack, static_info) + ")";
+
+  return IR_LOCALLY_STATIC;
 }
