--- PKGBUILD
+++ PKGBUILD
@@ -44,11 +44,18 @@ prepare() {
   patch -p1 -i "$srcdir"/declarative_region.patch
   patch -p1 -i "$srcdir"/gentoo-qa.patch
   patch -p1 -i "$srcdir"/gvhdl_tag_command.patch
+  patch -p1 -i "$srcdir"/fix-core-dump.patch
+
+  # The size is a member variable, not a member function.
+  sed -i 's|ainfo->element_type->size()|ainfo->element_type->size|g' freehdl/std-vhdl-types.hh
 }
 
 build() {
   cd "${srcdir}"/$pkgname-$pkgver
-  CXXFLAGS="-D_GLIBCXX_USE_CXX11_ABI=1" ./configure --prefix=/usr
+  # The configure is outdated, regenerate it.
+  autoreconf -fiv
+  # The register variable is not allowed in C++17, so use C++11 instead.
+  CXXFLAGS="-D_GLIBCXX_USE_CXX11_ABI=1 -std=c++11" ./configure --prefix=/usr
   make
 }
 
@@ -56,3 +63,6 @@ package() {
   cd "${srcdir}"/$pkgname-$pkgver
   make DESTDIR="${pkgdir}" install
 }
+
+source+=('fix-core-dump.patch')
+sha256sums+=('c8061f1340373d09fb3c4df8c4ce9d5c20685b88ab585266ecbf798277e7a9ee')
