diff --git PKGBUILD PKGBUILD
index 5553969..75e9981 100644
--- PKGBUILD
+++ PKGBUILD
@@ -39,13 +39,17 @@ makedepends=(
 source=(
   git+https://github.com/snes9xgit/snes9x.git#tag=${pkgver}
   git+https://github.com/KhronosGroup/SPIRV-Cross.git
+  "fix_comparator_not_const.patch"
 )
 sha256sums=('SKIP'
-            'SKIP')
+            'SKIP'
+            '8ec91456a2145dd2a6eff888d725f69532f4d497997961ee3b8f8f811b6d26d0')
 
 prepare() {
   cd snes9x
 
+  patch -Np1 -i ../fix_comparator_not_const.patch
+
   for submodule in shaders/SPIRV-Cross; do
     git submodule init ${submodule}
     git config submodule.${submodule}.url ../${submodule#*/}
