diff -u -r hylafax-6.0.7-orig/libhylafax/FaxClient.c++ hylafax-6.0.7/libhylafax/FaxClient.c++
--- hylafax-6.0.7-orig/libhylafax/FaxClient.c++	2021-11-29 17:00:00.603727539 +0000
+++ hylafax-6.0.7/libhylafax/FaxClient.c++	2021-11-29 17:05:32.713745619 +0000
@@ -116,7 +116,7 @@
 void
 FaxClient::vprintWarning(const char* fmt, va_list ap)
 {
-    fprintf(stderr, NLS::TEXT("Warning, "));
+    fprintf(stderr, "%s", NLS::TEXT("Warning, "));
     vfprintf(stderr, fmt, ap);
     fputs("\n", stderr);
 }
diff -u -r hylafax-6.0.7-orig/libhylafax/SNPPClient.c++ hylafax-6.0.7/libhylafax/SNPPClient.c++
--- hylafax-6.0.7-orig/libhylafax/SNPPClient.c++	2021-11-29 17:00:00.603727539 +0000
+++ hylafax-6.0.7/libhylafax/SNPPClient.c++	2021-11-29 17:06:01.807080538 +0000
@@ -103,7 +103,7 @@
 void
 SNPPClient::vprintWarning(const char* fmt, va_list ap)
 {
-    fprintf(stderr, NLS::TEXT("Warning, "));
+    fprintf(stderr, "%s", NLS::TEXT("Warning, "));
     vfprintf(stderr, fmt, ap);
     fputs("\n", stderr);
 }
diff -u -r hylafax-6.0.7-orig/libhylafax/TextFormat.c++ hylafax-6.0.7/libhylafax/TextFormat.c++
--- hylafax-6.0.7-orig/libhylafax/TextFormat.c++	2021-11-29 17:00:00.603727539 +0000
+++ hylafax-6.0.7/libhylafax/TextFormat.c++	2021-11-29 17:06:37.100415795 +0000
@@ -1371,7 +1371,7 @@
 {
     fxStr emsg;
     if (!decodeFontName(family, fontpath, emsg)) {
-	fprintf(stderr,emsg);
+	fprintf(stderr, "%s", emsg);
 	return NULL;
     }
     return Sys::fopen(fontpath, "r");
diff -u -r hylafax-6.0.7-orig/libhylafax/TypeRules.c++ hylafax-6.0.7/libhylafax/TypeRules.c++
--- hylafax-6.0.7-orig/libhylafax/TypeRules.c++	2021-11-29 17:00:00.603727539 +0000
+++ hylafax-6.0.7/libhylafax/TypeRules.c++	2021-11-29 17:04:50.700410004 +0000
@@ -101,7 +101,7 @@
 	    printf(" \"%s\"", value.s);
 	else if (type != ASCII && type != ASCIIESC) {
 	    if (op == ANY)
-		printf(NLS::TEXT(" <any value>"));
+		printf("%s", NLS::TEXT(" <any value>"));
 	    else
 		printf(" %#llx", (long long) value.v);
 	}
@@ -109,7 +109,7 @@
     }
     if (off > (off_t)size) {
 	if (verbose)
-	    printf(NLS::TEXT("failed (offset past data)\n"));
+	    printf("%s", NLS::TEXT("failed (offset past data)\n"));
 	return (false);
     }
     bool ok = false;
@@ -162,7 +162,7 @@
 	    break;
 	}
 	if (verbose)
-	    printf(NLS::TEXT("failed (insufficient data)\n"));
+	    printf("%s", NLS::TEXT("failed (insufficient data)\n"));
 	return (false);
     case LONG:
 	if (off + 4 < (off_t)size) {
@@ -171,7 +171,7 @@
 	    break;
 	}
 	if (verbose)
-	    printf(NLS::TEXT("failed (insufficient data)\n"));
+	    printf("%s", NLS::TEXT("failed (insufficient data)\n"));
 	return (false);
     }
     /*
@@ -195,7 +195,7 @@
 	    printf(NLS::TEXT("success (result %s, rule \"%s\")\n"),
 		resultNames[result], (const char*) cmd);
 	else
-	    printf(NLS::TEXT("failed (comparison)\n"));
+	    printf("%s", NLS::TEXT("failed (comparison)\n"));
     }
     return (ok);
 }
@@ -496,6 +496,6 @@
 	    return (&(*rules)[i + match2(i, data, size, verbose)]);
     }
     if (verbose)
-	printf(NLS::TEXT("no match\n"));
+	printf("%s", NLS::TEXT("no match\n"));
     return (NULL);
 }
