--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,8 @@ source=(https://ftp.mozilla.org/pub/mozilla.org/thunderbird/releases/$pkgver/sou
         vendor-prefs.js
         distribution.ini
         mozconfig.cfg
-        metainfo.patch)
+        metainfo.patch
+        makotokato-riscv64-support-and-zenithal-backported.patch)
 validpgpkeys=(
   14F26682D0916CDD81E37B6D61B7B526D98F0353 # Mozilla Software Releases <release@mozilla.com>
   4360FE2109C49763186F8E21EBE41E90F6F12F6D # Mozilla Software Releases <release@mozilla.com>
@@ -218,8 +219,9 @@ sha512sums=('1591f3e9c76c1f2ea7fa5e194a7d030c8657a7855a95c8a177e8067c5aa838f0d8c
             'a0061fcb2a7f66061e336a8d95948592f56f4752e56467f14ba63846720ebf845cce7511d1a2637e3b80d5a1ffdaa2fb783fa37195103425ef65222d45372012'
             '6918c0de63deeddc6f53b9ba331390556c12e0d649cf54587dfaabb98b32d6a597b63cf02809c7c58b15501720455a724d527375a8fb9d757ccca57460320734'
             '5cd3ac4c94ef6dcce72fba02bc18b771a2f67906ff795e0e3d71ce7db6d8a41165bd5443908470915bdbdb98dddd9cf3f837c4ba3a36413f55ec570e6efdbb9f'
-            '328422adac0cfe6af4ecdcd864004b7ada8f8171aa954fecc23a7e883e90a9bb0848372faa1100440dc754922f965e1e7b98c185aa88df190bff1051d2146c85'
+            '9ac3f37b7d21b8d09c43f736fadb0c2fc4fea0892d190d79661d1cab8939b0a8860c27f6a676b7ce1dafb679bf2652c87406e3e7ee4bbb3c1335629219481c24'
             '7e43b1f25827ddae615ad43fc1e11c6ba439d6c2049477dfe60e00188a70c0a76160c59a97cc01d1fd99c476f261c7cecb57628b5be48874be7cf991c22db290'
+            '4d69c70f036132a092f47a3beb62ba6a4068c0b00c2a9bf32318467e8541c426a14606e9077baaadcc9ff2e24ea1b5c8a450ce73695a9126f91582db9d52c9db'
             '698673f6afb6285806bae6c88a619d20ec654d2cd1fc525500022f0300b113a0601be53391ca714b49addb7a8d8be84382892aaf3427fd971925f68e0001bf11'
             '66c40dd5d43b84ab9aaf2f9bf01dd9e9855bad531022c0c03c018dab4be15d13098fcfcddae7e4b26dbff64f851c26f59985577e8d09cb7ba03a8b303befcb7c'
             '9477df1858b8b42ebb693681d57bffab925a02ba0be6815d82d65f273c5d52d4915ea129df88c9a992741099df770ac570bc6e169252ef265348ac972765eea6'
diff --git mozconfig.cfg mozconfig.cfg
index e19e3cf2..13571ee8 100644
--- mozconfig.cfg
+++ mozconfig.cfg
@@ -1,13 +1,16 @@
 ac_add_options --enable-application=comm/mail
 
 ac_add_options --prefix=/usr
-ac_add_options --enable-release
-ac_add_options --enable-linker=gold
+# enable-release incurs rustc to segfault when compiling neqo-transport
+ac_add_options --disable-release
+# can not use gold as linker
+ac_add_options --enable-linker=bfd
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-# https://bugzilla.mozilla.org/show_bug.cgi?id=1423822
-ac_add_options --disable-elf-hack
+ac_add_options --disable-bootstrap
+ac_add_options --disable-jit
+ac_add_options --disable-debug
 
 # Branding
 ac_add_options --enable-official-branding
