--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,8 @@ source=(https://ftp.mozilla.org/pub/mozilla.org/thunderbird/releases/$pkgver/sou
         vendor-prefs.js
         distribution.ini
         mozconfig.cfg
-        metainfo.patch)
+        metainfo.patch
+        makotokato-riscv64-support-and-zenithal-backported.patch)
 validpgpkeys=(
   14F26682D0916CDD81E37B6D61B7B526D98F0353 # Mozilla Software Releases <release@mozilla.com>
   4360FE2109C49763186F8E21EBE41E90F6F12F6D # Mozilla Software Releases <release@mozilla.com>
@@ -218,8 +219,9 @@ sha512sums=('474b5aca9c5e54fdc72eebff938f0d217bc039c3ac8d1caf965fb61bd1cf349f389
             'a0061fcb2a7f66061e336a8d95948592f56f4752e56467f14ba63846720ebf845cce7511d1a2637e3b80d5a1ffdaa2fb783fa37195103425ef65222d45372012'
             '6918c0de63deeddc6f53b9ba331390556c12e0d649cf54587dfaabb98b32d6a597b63cf02809c7c58b15501720455a724d527375a8fb9d757ccca57460320734'
             '5cd3ac4c94ef6dcce72fba02bc18b771a2f67906ff795e0e3d71ce7db6d8a41165bd5443908470915bdbdb98dddd9cf3f837c4ba3a36413f55ec570e6efdbb9f'
-            '328422adac0cfe6af4ecdcd864004b7ada8f8171aa954fecc23a7e883e90a9bb0848372faa1100440dc754922f965e1e7b98c185aa88df190bff1051d2146c85'
+            'a863919da17cd5ce7e1d74d5b90d2b6720751b58139aa12d37f177c456328bae0e28e9d0c118470bcbd4fcba629f4d9eeee5ecddc473ee7abdd35d4ab0c389b9'
             '7e43b1f25827ddae615ad43fc1e11c6ba439d6c2049477dfe60e00188a70c0a76160c59a97cc01d1fd99c476f261c7cecb57628b5be48874be7cf991c22db290'
+            '4d69c70f036132a092f47a3beb62ba6a4068c0b00c2a9bf32318467e8541c426a14606e9077baaadcc9ff2e24ea1b5c8a450ce73695a9126f91582db9d52c9db'
             '88ff48ab1881908070abe982cf3de1b6d2e956a7a8b12f2504369e27e1a6f7b00af3cdfccc6f6e6eaba916f05811b88b8ef36218373c1fd709ab5814ad94e67c'
             'ea28bb9b5b13c0d93429239e27f4166a1c408b21973c850482ddf717addfb900e418930dbbfdc86ba6f41a4b561584ca6d3ae4d9118a461ebd856b9b9fe6b488'
             '673575fc93a75631b443ed09df2c3c211f980059202494bd8320d9c4e7d2bdfe7611b83eda7d2c08695cfd7dc7adbbd7edf99e9419d36726434ec5bbd2f61a8e'
diff --git mozconfig.cfg mozconfig.cfg
index e19e3cf2..13571ee8 100644
--- mozconfig.cfg
+++ mozconfig.cfg
@@ -1,13 +1,16 @@
 ac_add_options --enable-application=comm/mail
 
 ac_add_options --prefix=/usr
-ac_add_options --enable-release
-ac_add_options --enable-linker=gold
+# enable-release incurs rustc to segfault when compiling neqo-transport
+ac_add_options --disable-release
+# can not use gold as linker
+ac_add_options --enable-linker=bfd
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-# https://bugzilla.mozilla.org/show_bug.cgi?id=1423822
-ac_add_options --disable-elf-hack
+ac_add_options --disable-bootstrap
+ac_add_options --disable-jit
+ac_add_options --disable-debug
 
 # Branding
 ac_add_options --enable-official-branding
