--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,8 @@ source=(https://ftp.mozilla.org/pub/mozilla.org/thunderbird/releases/$pkgver/sou
         distribution.ini
         mozconfig.cfg
         metainfo.patch
-        Add-missing-stub-for-wl_proxy_marshal_fl.patch)
+        Add-missing-stub-for-wl_proxy_marshal_fl.patch
+        makotokato-riscv64-support-and-zenithal-backported.patch)
 validpgpkeys=(
   14F26682D0916CDD81E37B6D61B7B526D98F0353 # Mozilla Software Releases <release@mozilla.com>
   4360FE2109C49763186F8E21EBE41E90F6F12F6D # Mozilla Software Releases <release@mozilla.com>
@@ -219,9 +220,10 @@ sha512sums=('5939e09b143e440efa906d95cda06826bd3a73b2edde5eed86229b8a0e4d1434519
             'a0061fcb2a7f66061e336a8d95948592f56f4752e56467f14ba63846720ebf845cce7511d1a2637e3b80d5a1ffdaa2fb783fa37195103425ef65222d45372012'
             '6918c0de63deeddc6f53b9ba331390556c12e0d649cf54587dfaabb98b32d6a597b63cf02809c7c58b15501720455a724d527375a8fb9d757ccca57460320734'
             '5cd3ac4c94ef6dcce72fba02bc18b771a2f67906ff795e0e3d71ce7db6d8a41165bd5443908470915bdbdb98dddd9cf3f837c4ba3a36413f55ec570e6efdbb9f'
-            '328422adac0cfe6af4ecdcd864004b7ada8f8171aa954fecc23a7e883e90a9bb0848372faa1100440dc754922f965e1e7b98c185aa88df190bff1051d2146c85'
+            'c9271c6a89c1037aa018d08695d0b41e6f0a2da3c731a40d372baa825c08b960fc95c8003a86fb0bc5155c953ae1740e510a1c270e3dd1e3c2a4b23f4b099b7e'
             '7e43b1f25827ddae615ad43fc1e11c6ba439d6c2049477dfe60e00188a70c0a76160c59a97cc01d1fd99c476f261c7cecb57628b5be48874be7cf991c22db290'
             'd2a6a14871eb227d7d2941936949ed5d0517f63fcdd135ff3fe9557b6cddd815c99babd326fcfd504fa9c10134e0db57b12984a8abe13c378f2823041b134e23'
+            '4d69c70f036132a092f47a3beb62ba6a4068c0b00c2a9bf32318467e8541c426a14606e9077baaadcc9ff2e24ea1b5c8a450ce73695a9126f91582db9d52c9db'
             '9982c851d4cb0e9ca391649f09d8521aba716e8e75b600b81e0810111ff0e0cfac08cf27e96c1f3a3cc096a8baf21c08bd63e4ff2f3a53338168ac35ad534c1d'
             '8b5595201dd5ab76de158d7b7c2685f2e50a6398a6fae41125ce572ed71e4ce83c00d595c18288cafc883d188373f3bfeb66ba970a60da7d4f82beabf5e8bcc8'
             '0b3e588e03fce7d7dcdd0d83bb647d95216e7fbbbcffce2e1d94be5a4ac1c139bfc970a4140e020700b1827152ca4d7f46542c61295f362dfcaf90f3049e4ada'
--- mozconfig.cfg
+++ mozconfig.cfg
@@ -1,13 +1,18 @@
 ac_add_options --enable-application=comm/mail
 
 ac_add_options --prefix=/usr
-ac_add_options --enable-release
-ac_add_options --enable-linker=gold
+# enable-release incurs rustc to segfault when compiling neqo-transport
+ac_add_options --disable-release
+# can not use gold as linker
+ac_add_options --enable-linker=bfd
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
 # https://bugzilla.mozilla.org/show_bug.cgi?id=1423822
-ac_add_options --disable-elf-hack
+
+ac_add_options --disable-bootstrap
+ac_add_options --disable-jit
+ac_add_options --disable-debug
 
 # Branding
 ac_add_options --enable-official-branding
