diff --git PKGBUILD PKGBUILD
index 13f1940..f31ac7f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,7 +33,9 @@ source=(https://ftp.mozilla.org/pub/mozilla.org/thunderbird/releases/$pkgver/sou
         vendor-prefs.js
         distribution.ini
         mozconfig.cfg
-        metainfo.patch)
+        metainfo.patch
+        cc-crate-riscv64-clang-support.patch
+        makotokato-riscv64-support-and-zenithal-backported.patch)
 validpgpkeys=(
   14F26682D0916CDD81E37B6D61B7B526D98F0353 # Mozilla Software Releases <release@mozilla.com>
   4360FE2109C49763186F8E21EBE41E90F6F12F6D # Mozilla Software Releases <release@mozilla.com>
@@ -218,8 +220,10 @@ sha512sums=('954be27795935e494d27d57da99b49ff61db8a2b26fa8e159a30d6c272033b01579
             'a0061fcb2a7f66061e336a8d95948592f56f4752e56467f14ba63846720ebf845cce7511d1a2637e3b80d5a1ffdaa2fb783fa37195103425ef65222d45372012'
             '6918c0de63deeddc6f53b9ba331390556c12e0d649cf54587dfaabb98b32d6a597b63cf02809c7c58b15501720455a724d527375a8fb9d757ccca57460320734'
             '5cd3ac4c94ef6dcce72fba02bc18b771a2f67906ff795e0e3d71ce7db6d8a41165bd5443908470915bdbdb98dddd9cf3f837c4ba3a36413f55ec570e6efdbb9f'
-            '328422adac0cfe6af4ecdcd864004b7ada8f8171aa954fecc23a7e883e90a9bb0848372faa1100440dc754922f965e1e7b98c185aa88df190bff1051d2146c85'
+            'c9271c6a89c1037aa018d08695d0b41e6f0a2da3c731a40d372baa825c08b960fc95c8003a86fb0bc5155c953ae1740e510a1c270e3dd1e3c2a4b23f4b099b7e'
             '7e43b1f25827ddae615ad43fc1e11c6ba439d6c2049477dfe60e00188a70c0a76160c59a97cc01d1fd99c476f261c7cecb57628b5be48874be7cf991c22db290'
+            'abdc9bb34b380840eb35ce2c79318a5dca8e6c842ee9df35c4ba85c4e78cf389193bf91c3c4e03d5582863ffde5a372ef7061d80ed45e7ed4aa686e7a0af8598'
+            '4d69c70f036132a092f47a3beb62ba6a4068c0b00c2a9bf32318467e8541c426a14606e9077baaadcc9ff2e24ea1b5c8a450ce73695a9126f91582db9d52c9db'
             '19352b7210a0edf2770a3d174fac12be53719e82adc42911e37736bdb8f9f5d5320dad1f583c79bbf067a2aa21e2d8f8152db8aef91406e571e6f709c569a781'
             '61555c72082338f7b1550a76809577edf0f3df67aa3c5fc66ba8cbca95272ff9cb15e70bc7e846936038fd7939460cb7dcda63878fbdca2ce44dd9a848ef5c78'
             'b05517ca2ff26b308c78e7e601533464fb8b1816b081553a99bc0daf9393af39ffe68a16bc28fc2b9ab6200ab81b889ba9005b500bd0258aa9ab7f4dcacc6f48'
diff --git mozconfig.cfg mozconfig.cfg
index e19e3cf..066b231 100644
mozconfig.cfg
mozconfig.cfg
@@ -1,13 +1,18 @@
 ac_add_options --enable-application=comm/mail
 
 ac_add_options --prefix=/usr
-ac_add_options --enable-release
-ac_add_options --enable-linker=gold
+# enable-release incurs rustc to segfault when compiling neqo-transport
+ac_add_options --disable-release
+# can not use gold as linker
+ac_add_options --enable-linker=bfd
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
 # https://bugzilla.mozilla.org/show_bug.cgi?id=1423822
-ac_add_options --disable-elf-hack
+
+ac_add_options --disable-bootstrap
+ac_add_options --disable-jit
+ac_add_options --disable-debug
 
 # Branding
 ac_add_options --enable-official-branding
