--- PKGBUILD
+++ PKGBUILD
@@ -47,6 +47,7 @@
 
   # disable VST2
   patch -p1 -i "$srcdir/disable-vst2.patch"
+  patch -p1 -i "$srcdir/riscv64-source.patch"
 }
 
 build() {
@@ -111,3 +112,9 @@
 
   install -vDm644 -t "$pkgdir/usr/lib/clap" build/JC303_artefacts/Release/CLAP/*.clap
 }
+
+_common_depends+=(libsamplerate)
+makedepends+=(libsamplerate)
+source+=(riscv64-source.patch riscv64-chowdsp.patch)
+sha512sums+=(SKIP SKIP)
+b2sums+=(SKIP SKIP)
