diff --git PKGBUILD PKGBUILD
index c58c40b..b90fcc0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,32 +20,30 @@ arch=('x86_64')
 license=('GPL')
 url='https://valgrind.org/'
 depends=('glibc' 'perl' 'debuginfod')
-makedepends=('gdb' 'lib32-glibc' 'lib32-gcc-libs' 'docbook-xml'
+makedepends=('gdb' 'docbook-xml'
              'docbook-xsl' 'docbook-sgml')
 checkdepends=('procps-ng')
-optdepends=('lib32-glibc: 32-bit ABI support')
 provides=('valgrind-multilib')
 replaces=('valgrind-multilib')
 options=('!emptydirs' '!strip')
-source=(https://sourceware.org/pub/valgrind/valgrind-${pkgver}.tar.bz2{,.asc}
+_commit='71272b252977fe52f03ea4fa8306b457b098cca5'
+source=(https://github.com/petrpavlu/valgrind-riscv64/archive/$_commit.tar.gz
         valgrind-3.7.0-respect-flags.patch
         fix-perl-errors.patch)
 validpgpkeys=(
   0E9FFD0C16A1856CF9C7C690BA0166E698FA6035 # Julian Seward <jseward@acm.org>
   EC3CFE88F6CA0788774F5C1D1AA44BE649DE760A # Mark Wielaard <mjw@gnu.org>
 )
-sha512sums=('3e86cda2f2d6cd30807fac7933ba2c46a97a2b4a868db989e6b0cceeadf029af7ee34ba900466a346948289aacb30f4399799bb83b97cc49a4d2d810441e5cfd'
-            'SKIP'
+sha512sums=('5e0634ed81970c0b9d274511e622b505798c601886f25ee86a2c81614c7aae8ff0d663e725f1a4418d8590e39ad85cc195f9015185f342e1a06608feb1e2b978'
             'e0cec39381cefeca09ae4794cca309dfac7c8693e6315e137e64f5c33684598726d41cfbb4edf764fe985503b13ff596184ca5fc32b159d500ec092e4cf8838c'
             '20b251bfc7bef8dfd232f9b679e907114c575299916164a608e2fe7fab5f30bf7241f25e37ab4194c56b0a21e682b3cea2fd892aab30fa2ce3863ef744f27f18')
-b2sums=('1f7306d288eb5ecfb2b2f0b2dc68960ecd4a7213b96a2e4ecb9990a755e8c233b0e9d190c0f807019bde669f44a631c8dafe9eb521d739561e6f889f7bdaca5c'
-        'SKIP'
+b2sums=('48ba35d9710cc321ad695031873089065658c7dc6f3a0f97ad79be559371fe1d8d029a3b33420780e37ff322b185884491850acee32feb24369aa793f043daa4'
         'af556fdf3c02e37892bfe9afebc954cf2f1b2fa9b75c1caacfa9f3b456ebc02bf078475f9ee30079b3af5d150d41415a947c3d04235c1ea8412cf92b959c484a'
         '78e5ebeda69302ad380923fe0e76ef8fc3443ffa29cc3104fe629335c8ceda1b4198cb5c72bdefb0e47c77ea02d2ca7bfb478cbf8731f8ded0e0c7c5d83981ee')
 options=(!lto) # https://bugs.kde.org/show_bug.cgi?id=338252
 
 prepare() {
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-$_commit
   patch -Np1 < ../valgrind-3.7.0-respect-flags.patch
   sed -i 's|sgml/docbook/xsl-stylesheets|xml/docbook/xsl-stylesheets-1.79.2-nons|' docs/Makefile.am
 
@@ -60,7 +58,7 @@ build() {
   CFLAGS=${CFLAGS/-fno-plt/}
   CXXFLAGS=${CXXFLAGS/-fno-plt/}
 
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-$_commit
   ./configure \
     --prefix=/usr \
     --sysconfdir=/etc \
@@ -75,7 +73,7 @@ check() {
   # only run if glibc-debug is supplied manually
   if ! pacman -Q glibc-debug; then echo -e "\033[1;31mcheck() not run, supply glibc-debug if unintended!\033[0m"; return 0; fi
 
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-$_commit
 
   # Make sure a basic binary runs. There should be no errors.
   ./vg-in-place --error-exitcode=1 /bin/true
@@ -113,7 +111,7 @@ check() {
 }
 
 package() {
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-$_commit
   make DESTDIR="${pkgdir}" install
 
   install -Dm644 docs/*.1 -t "$pkgdir/usr/share/man/man1"
