Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 457315)
+++ PKGBUILD	(working copy)
@@ -20,29 +20,26 @@
 license=('GPL')
 url='http://valgrind.org/'
 depends=('glibc' 'perl' 'debuginfod')
-makedepends=('gdb' 'lib32-glibc' 'lib32-gcc-libs' 'docbook-xml'
+makedepends=('gdb' 'docbook-xml'
              'docbook-xsl' 'docbook-sgml')
 checkdepends=('procps-ng')
-optdepends=('lib32-glibc: 32-bit ABI support')
 provides=('valgrind-multilib')
 replaces=('valgrind-multilib')
 options=('!emptydirs' '!strip')
-source=(https://sourceware.org/pub/valgrind/valgrind-${pkgver}.tar.bz2{,.asc}
+source=(https://github.com/petrpavlu/valgrind-riscv64/archive/013094fca0a039149aa574e8459fbe07c2359aba.tar.gz
         valgrind-3.7.0-respect-flags.patch)
 validpgpkeys=(
   0E9FFD0C16A1856CF9C7C690BA0166E698FA6035 # Julian Seward <jseward@acm.org>
   EC3CFE88F6CA0788774F5C1D1AA44BE649DE760A # Mark Wielaard <mjw@gnu.org>
 )
-sha512sums=('f720a89dc4c4989cc5714bff9efe97529f71990bcfad7a92b889ce099c4326d6da07fa4d5fbab2e9125e20f352354f6178471e49e419b613a3c82c2a1c667ab2'
-            'SKIP'
+sha512sums=('9e067d2eeeec3cd33fdf1e884804e7c4a51a77d8213968da21087c88328edd847f92ca9207eb75263b4cfc613870bc94c53748d97539a3754b71dfff6cff1621'
             'e0cec39381cefeca09ae4794cca309dfac7c8693e6315e137e64f5c33684598726d41cfbb4edf764fe985503b13ff596184ca5fc32b159d500ec092e4cf8838c')
-b2sums=('d3ccd25789ec4d87a0ddab58d8098246a33a083505b42adaa94ee3af659cbe073e516bc98163ef56233e6cd9de2125839b71ee443fbc0db6557dcb35a6c05280'
-        'SKIP'
+b2sums=('9e7069e9cfbb2aaf6b6233ab9d04783f4315dded7fbda1fe6d615002c4330550bdbc8a057ef5e5371ed64041e4d5b66ae73370a6304bacaba6346a101ed6881c'
         'af556fdf3c02e37892bfe9afebc954cf2f1b2fa9b75c1caacfa9f3b456ebc02bf078475f9ee30079b3af5d150d41415a947c3d04235c1ea8412cf92b959c484a')
 options=(!lto) # https://bugs.kde.org/show_bug.cgi?id=338252
 
 prepare() {
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-013094fca0a039149aa574e8459fbe07c2359aba
   patch -Np1 < ../valgrind-3.7.0-respect-flags.patch
   sed -i 's|sgml/docbook/xsl-stylesheets|xml/docbook/xsl-stylesheets-1.79.2-nons|' docs/Makefile.am
 
@@ -55,7 +52,7 @@
   CFLAGS=${CFLAGS/-fno-plt/}
   CXXFLAGS=${CXXFLAGS/-fno-plt/}
 
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-013094fca0a039149aa574e8459fbe07c2359aba
   ./configure \
     --prefix=/usr \
     --sysconfdir=/etc \
@@ -108,7 +105,7 @@
 }
 
 package() {
-  cd valgrind-${pkgver}
+  cd valgrind-riscv64-013094fca0a039149aa574e8459fbe07c2359aba
   make DESTDIR="${pkgdir}" install
 
   install -Dm644 docs/*.1 -t "$pkgdir/usr/share/man/man1"
