--- PKGBUILD
+++ PKGBUILD
@@ -20,15 +20,12 @@
 license=('GPL-2.0-or-later')
 url='https://valgrind.org/'
 depends=('glibc' 'perl' 'debuginfod')
-makedepends=('gdb' 'lib32-glibc' 'lib32-gcc-libs' 'docbook-xml'
+makedepends=('gdb' 'docbook-xml'
              'docbook-xsl' 'docbook-sgml')
 checkdepends=('procps-ng')
 optdepends=(
-  'lib32-glibc: 32-bit ABI support'
   'python: cg_* scripts'
 )
-provides=('valgrind-multilib')
-replaces=('valgrind-multilib')
 options=('!emptydirs' '!strip')
 source=(https://sourceware.org/pub/valgrind/valgrind-${pkgver}.tar.bz2{,.asc}
         valgrind-3.7.0-respect-flags.patch)
