diff --git PKGBUILD PKGBUILD
index cb9185b1a..5ccc7ee7b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,9 +15,11 @@
 groups=(gnome)
 _commit=884b1fd16e6ee505e216f38566c777131321a7ed  # gnome-40
 source=("git+https://gitlab.gnome.org/GNOME/gnome-photos.git#commit=$_commit"
-        "git+https://gitlab.gnome.org/GNOME/libgd.git")
+        "git+https://gitlab.gnome.org/GNOME/libgd.git"
+        "remove-incorrect-args-for-i18n.merge_file.diff::https://gitlab.gnome.org/GNOME/gnome-photos/-/commit/a0ac868a19d55dc52100e54fe4b2d29041bc6752.diff")
 sha256sums=('SKIP'
-            'SKIP')
+            'SKIP'
+            'da3d62b32c86646a3d86dc6ef9d2f0fec323ec8f58899b3508b58ddf9f8e2f58')
 
 pkgver() {
   cd $pkgname
@@ -30,6 +32,8 @@
   git submodule init
   git submodule set-url subprojects/libgd "$srcdir/libgd"
   git submodule update
+
+  patch -Np1 -i ../remove-incorrect-args-for-i18n.merge_file.diff
 }
 
 build() {
