--- pkg/surveyext/editor_test.go
+++ pkg/surveyext/editor_test.go
@@ -80,6 +80,7 @@ func Test_GhEditor_Prompt_skip(t *testing.T) {
 	}()
 
 	res, err := e.Prompt(defaultPromptConfig())
+	time.Sleep(time.Second)
 	assert.NoError(t, err)
 	assert.Equal(t, "initial value", res)
 	assert.Equal(t, "", normalizeANSI(pty.Output()))
@@ -111,6 +112,7 @@ func Test_GhEditor_Prompt_editorAppend(t *testing.T) {
 	}()
 
 	res, err := e.Prompt(defaultPromptConfig())
+	time.Sleep(time.Second)
 	assert.NoError(t, err)
 	assert.Equal(t, "initial value - added by vim", res)
 	assert.Equal(t, "", normalizeANSI(pty.Output()))
@@ -142,6 +144,7 @@ func Test_GhEditor_Prompt_editorTruncate(t *testing.T) {
 	}()
 
 	res, err := e.Prompt(defaultPromptConfig())
+	time.Sleep(time.Second)
 	assert.NoError(t, err)
 	assert.Equal(t, "", res)
 	assert.Equal(t, "", normalizeANSI(pty.Output()))
