--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ sha256sums=('8ba98b5e46526c9828507a587b429448fe9436ce1f875aa567d77ec3a8cae56c')
 
 prepare() {
     cd "cli-${pkgver}"
+    patch -Np0 -i $srcdir/githubcli-prompt-sleep.patch
     # TODO: These tests invoke the TTY and our container *really* does not like that
     rm pkg/cmd/auth/login/login_test.go
     # Drop tests that invoking 3rd party server processes
@@ -55,3 +56,6 @@ package() {
     install -Dm644 "LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
     install -Dm644 "README.md" "$pkgdir/usr/share/doc/$pkgname/README.md"
 }
+
+source+=("githubcli-prompt-sleep.patch")
+sha256sums+=('3d42812b87bdcb98981c18fd3bdbd4eff36b849c951dd059f95c26ba90699ea4')
