--- PKGBUILD
+++ PKGBUILD
@@ -1,5 +1,5 @@
 # Maintainer: Sven-Hendrik Haase <svenstaro@archlinux.org>
-pkgname=(clpeak clpeak-cuda)
+pkgname=(clpeak)
 pkgver=3.0.0
 pkgrel=1
 pkgdesc="A tool which profiles OpenCL devices to find their peak capacities (OpenCL/Vulkan support)"
@@ -7,7 +7,7 @@
 url="https://github.com/krrishnarraj/clpeak"
 license=("Unlicense")
 depends=('ocl-icd' 'libgcc' 'libstdc++' 'glibc' 'vulkan-icd-loader')
-makedepends=('cmake' 'opencl-headers' 'ninja' 'git' 'vulkan-headers' 'shaderc' 'cuda' 'clang')
+makedepends=('cmake' 'opencl-headers' 'ninja' 'git' 'vulkan-headers' 'shaderc' 'clang')
 source=(git+https://github.com/krrishnarraj/clpeak#tag=$pkgver)
 sha512sums=('4d6f51d371379215510aa35cf4d30349b9426d28c93bec24341382a0789f78882b92facf32de3017791e080ffcfe5f29d9086f580230e658060eb32b18d3ea29')
 
@@ -21,13 +21,6 @@
     -DCLPEAK_ENABLE_CUDA=OFF
   ninja -C build
 
-  cmake \
-    -Bbuild-cuda \
-    -GNinja \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=/usr \
-    -DCLPEAK_ENABLE_CUDA=ON
-  ninja -C build-cuda
 }
 
 package_clpeak() {
