--- PKGBUILD
+++ PKGBUILD
@@ -1,5 +1,5 @@
 # Maintainer: Sven-Hendrik Haase <svenstaro@archlinux.org>
-pkgname=(clpeak clpeak-cuda)
+pkgname=(clpeak)
 pkgver=2.1.4
 pkgrel=1
 pkgdesc="A tool which profiles OpenCL devices to find their peak capacities (OpenCL/Vulkan support)"
@@ -7,7 +7,7 @@
 url="https://github.com/krrishnarraj/clpeak"
 license=("Unlicense")
 depends=('ocl-icd' 'libgcc' 'libstdc++' 'glibc' 'vulkan-icd-loader')
-makedepends=('cmake' 'opencl-headers' 'ninja' 'git' 'vulkan-headers' 'shaderc' 'cuda' 'clang')
+makedepends=('cmake' 'opencl-headers' 'ninja' 'git' 'vulkan-headers' 'shaderc' 'clang')
 source=(git+https://github.com/krrishnarraj/clpeak#tag=$pkgver)
 sha512sums=('c68a63ae2ac9b22fe239d8aca16fe9f3c62da007a27acf3669ac0b618fb91e4b4e2f8c6c44e676cf3ccc046e7e5150d56998637baedba7ea65253aeedce40691')
 
@@ -21,13 +21,6 @@
     -DCLPEAK_ENABLE_CUDA=OFF
   ninja -C build
 
-  cmake \
-    -Bbuild-cuda \
-    -GNinja \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=/usr \
-    -DCLPEAK_ENABLE_CUDA=ON
-  ninja -C build-cuda
 }
 
 package_clpeak() {
