diff --git PKGBUILD PKGBUILD
index 5a426e03..b15718d7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,9 @@
   # python-milksnake is *not* a runtime dependency
   # https://github.com/pyocd/cmsis-pack-manager/issues/188
   sed '/milksnake/d' -i setup.cfg
+
+  cd rust
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
 }
 
 build() {
