--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,11 @@
 source=(git+https://github.com/pop-os/system76-scheduler.git#tag=${pkgver})
 b2sums=('4bf17c797bbd4fc65d8182709bbb21cc991d1233dd33c4b24fbc3b230b17725686c91874f394d2997051f381311268a05b7cd34c73b8c332b7e0edc7d16c763c')
 
+prepare() {
+  cd system76-scheduler
+  patch -Np1 -i ../bump-generator.patch
+}
+
 build() {
   cd system76-scheduler
   export CC=clang
@@ -39,3 +44,6 @@
   cd system76-scheduler
   just rootdir="${pkgdir}" install
 }
+
+source+=(bump-generator.patch::https://github.com/pop-os/system76-scheduler/commit/d130b5dfa2692fd2ebfd0f9dd45faf65bc0da1ba.patch)
+b2sums+=('960e69f80f8fad0019f7169fe2494abcf6fd79c696d4e251a5674f5ca07ab5cedef429e1610243fd853f533c353ce75586ed3a510282530b73b9970bd5a10bf2')
