--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('any')
 url="https://www.keycloak.org/"
 license=('Apache')
 depends=("java-runtime-headless=${_java}" 'grep' 'bash' 'coreutils' 'util-linux')
-makedepends=('maven' "java-environment=${_java}")
+makedepends=('maven' "java-environment=${_java}" 'nodejs' 'npm' 'pnpm')
 backup=(
   'etc/keycloak/keycloak.conf'
 )
@@ -34,6 +34,7 @@ prepare() {
   cd $pkgname-$pkgver
 
   patch -Np1 -i "$srcdir"/pin-java-version.patch
+  patch -Np1 -i "$srcdir"/use-system-node.patch
 }
 
 build() {
@@ -83,4 +84,7 @@ package() {
   install -Dm 644 LICENSE.txt "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE.txt"
 }
 
+source+=(use-system-node.patch)
+sha512sums+=('e7a912e90bfac2fbd880f4abfbb6cfc31ea877dfa67f1c13681b16cd3fddb4e87bd839dba10e318ddb1690d5b7894aa85179edcc9c022c6c94987a2482b88f74')
+
 # vim: ts=2 sw=2 et:
