--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('any')
 url="https://www.keycloak.org/"
 license=('Apache')
 depends=("java-runtime-headless=${_java}" 'grep' 'bash' 'coreutils' 'util-linux')
-makedepends=('maven' "java-environment=${_java}")
+makedepends=('maven' "java-environment=${_java}" 'nodejs' 'npm' 'pnpm')
 backup=(
   'etc/keycloak/keycloak.conf'
 )
@@ -22,18 +22,20 @@ source=(https://github.com/keycloak/keycloak/archive/$pkgver/$pkgname-$pkgver.ta
         keycloak.tmpfiles
         common.sh
         pin-java-version.patch
-        )
+        use-system-node.patch)
 sha512sums=('65e3e01d008f8935aaca9d32323d78491125fc794c25ec089e044f8a280255f1c28db9634c81892693fd73e82417af5a6644ff358a31820944330bd042fc57ab'
             '925ca021a9989a6d5181a90f42ec9e67a4957e98bb716acba75e13e1b01a03e4bd3a5939f5f0abe4cf57157be54dda5e373286041b22d84f5079eba94df4e6c9'
             '2e2ba147007ad74e38579a8838d79de47beac509b4bd1a14d7f80905953d79a7396d781f141b461ec688f5ceef9a1081a825a4ca8afc1ea12c178d8ae7f5a7dd'
             '155db40105c08d0aaa810ca5533dc16fc9f82060280541ede6fafd754d30b4844f6d10ace1417a5ad68d89fc54e1b9e6d906ce7ccf973f4ac964422211ed9a72'
             '27c2b0d8ef2f64a515410e5316524eed879d67c42a695642c23653f59bbdc37ff7aaff3414fa6779e97ae477c51014aadb2f0256effdf7abf65f1778fe8885bd'
-            '38f014de00db7ec1b5693529401f316f70474c97484cfdea0a7048da6bb9d1ace8fdc915bf8857edcd6aae0d0dd9c1e0b2c0be0aaa99d105be09bed975f049fd')
+            '38f014de00db7ec1b5693529401f316f70474c97484cfdea0a7048da6bb9d1ace8fdc915bf8857edcd6aae0d0dd9c1e0b2c0be0aaa99d105be09bed975f049fd'
+            'e7a912e90bfac2fbd880f4abfbb6cfc31ea877dfa67f1c13681b16cd3fddb4e87bd839dba10e318ddb1690d5b7894aa85179edcc9c022c6c94987a2482b88f74')
 
 prepare() {
   cd $pkgname-$pkgver
 
   patch -Np1 -i "$srcdir"/pin-java-version.patch
+  patch -Np1 -i "$srcdir"/use-system-node.patch
 }
 
 build() {
