diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index fcf9fee..fe637ac 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -497,7 +497,7 @@ endif(NOT CMAKE_MODULES_DIR)
 
 # UHDConfig.cmake needs UHD_LINK_LIST_STATIC set:
 list(APPEND UHD_LINK_LIST_STATIC "dl")
-list(APPEND UHD_LINK_LIST_STATIC "pthread")
+list(APPEND UHD_LINK_LIST_STATIC "-pthread")
 foreach(Boost_Comp ${UHD_BOOST_REQUIRED_COMPONENTS})
     if(NOT ${Boost_Comp} STREQUAL "unit_test_framework")
         list(APPEND UHD_LINK_LIST_STATIC "boost_${Boost_Comp}")
diff --git a/host/lib/utils/CMakeLists.txt b/host/lib/utils/CMakeLists.txt
index 639b3e9..bed5ca5 100644
--- a/host/lib/utils/CMakeLists.txt
+++ b/host/lib/utils/CMakeLists.txt
@@ -46,7 +46,7 @@ CHECK_CXX_SOURCE_COMPILES("
 if(HAVE_PTHREAD_SETSCHEDPARAM)
     message(STATUS "  Priority scheduling supported through pthread_setschedparam.")
     list(APPEND THREAD_PRIO_DEFS HAVE_PTHREAD_SETSCHEDPARAM)
-    LIBUHD_APPEND_LIBS(pthread)
+    LIBUHD_APPEND_LIBS("-pthread")
 elseif(HAVE_WIN_SETTHREADPRIORITY)
     message(STATUS "  Priority scheduling supported through windows SetThreadPriority.")
     set(THREAD_PRIO_DEFS HAVE_WIN_SETTHREADPRIORITY)
@@ -59,7 +59,7 @@ endif()
 # Setup defines for thread naming
 ########################################################################
 
-set(CMAKE_REQUIRED_LIBRARIES "pthread")
+set(CMAKE_REQUIRED_LIBRARIES "-pthread")
 CHECK_CXX_SOURCE_COMPILES("
     #include <pthread.h>
     int main(){
@@ -80,7 +80,7 @@ endif(CYGWIN)
 if(HAVE_PTHREAD_SETNAME)
     message(STATUS "  Setting thread names is supported through pthread_setname_np.")
     list(APPEND THREAD_PRIO_DEFS HAVE_PTHREAD_SETNAME)
-    LIBUHD_APPEND_LIBS(pthread)
+    LIBUHD_APPEND_LIBS("-pthread")
 else()
     message(STATUS "  Setting thread names is not supported.")
     list(APPEND THREAD_PRIO_DEFS HAVE_THREAD_SETNAME_DUMMY)
@@ -90,7 +90,7 @@ endif()
 # Setup defines for thread affinitizing
 ########################################################################
 
-set(CMAKE_REQUIRED_LIBRARIES "pthread")
+set(CMAKE_REQUIRED_LIBRARIES "-pthread")
 CHECK_CXX_SOURCE_COMPILES("
     #include <pthread.h>
     int main(){
@@ -119,7 +119,7 @@ CHECK_CXX_SOURCE_COMPILES("
 if(HAVE_PTHREAD_SETAFFINITY_NP)
     message(STATUS "  Setting thread affinity is supported through pthread_setaffinity_np.")
     list(APPEND THREAD_PRIO_DEFS HAVE_PTHREAD_SETAFFINITY_NP)
-    LIBUHD_APPEND_LIBS(pthread)
+    LIBUHD_APPEND_LIBS("-pthread")
 elseif(HAVE_WIN_SETTHREADPRIORITY)
     message(STATUS "  Setting thread affinity is supported through windows SetThreadAffinityMask.")
     list(APPEND THREAD_PRIO_DEFS HAVE_WIN_SETTHREADAFFINITYMASK)
