diff --git PKGBUILD PKGBUILD
index a361950ad..7f8272bfb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,14 @@ makedepends=('cmake' 'boost' 'python-mako' 'python-numpy')
 options=(!lto)
 # gpsd?  dpdk?
 
-source=("libuhd-$pkgver.tar.gz::https://github.com/EttusResearch/uhd/archive/v$pkgver.tar.gz")
-sha256sums=('73580bb8823338dbc0ffe2840088cb15d7496ef7d402a4816df17a8e5c8d0221')
+source=("libuhd-$pkgver.tar.gz::https://github.com/EttusResearch/uhd/archive/v$pkgver.tar.gz"
+        "fix_pthread_linking.patch")
+sha256sums=('73580bb8823338dbc0ffe2840088cb15d7496ef7d402a4816df17a8e5c8d0221'
+            '3212753606506ce5ac15159440e49bf53c5a21b1d1da3e8af1aa578418f77068')
 
 prepare() {
   cd "$srcdir/uhd-$pkgver/host"
+  patch -Np2 -i "../../fix_pthread_linking.patch"
   mkdir build
 }
 
