--- PKGBUILD
+++ PKGBUILD
@@ -68,7 +68,7 @@
   local gypfile
   for gypfile in "${gypfiles[@]}"; do
     pushd "${gypfile%/binding.gyp}"
-    node-gyp rebuild --nodedir=/usr
+    node-gyp rebuild --nodedir=/usr --jobs=1
     # Keep the addon, drop the intermediates.
     find build -mindepth 1 -maxdepth 1 ! -name Release -exec rm -rf {} +
     find build/Release -mindepth 1 -maxdepth 1 ! -name '*.node' -exec rm -rf {} +
