diff --git PKGBUILD PKGBUILD
index d506ce88..b95df70c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,14 @@ arch=('x86_64')
 license=("CCPL:by-sa")
 depends=('ncurses' 'hicolor-icon-theme' 'gettext')
 makedepends=('sdl2_image' 'sdl2_ttf' 'sdl2_mixer' 'freetype2' 'astyle')
-source=("$pkgname-$_pkgver.tar.gz::https://github.com/CleverRaven/Cataclysm-DDA/archive/$_pkgver.tar.gz")
-md5sums=('87a41c979475955dd5f9eda288120096')
+source=("$pkgname-$_pkgver.tar.gz::https://github.com/CleverRaven/Cataclysm-DDA/archive/$_pkgver.tar.gz"
+        "disable_gold_on_riscv.patch")
+md5sums=('87a41c979475955dd5f9eda288120096'
+         'c2eabb11274ef6ecb1ad9534ba52790f')
 
 prepare() {
   cd "Cataclysm-DDA-$_pkgver"
+  patch -Np1 -i "../disable_gold_on_riscv.patch"
   sed -i 's/-Werror//' Makefile
   sed -i 's/ncursesw5-config/ncursesw6-config/' Makefile
   sed -i 's/shell git/shell false/' Makefile
