--- PKGBUILD
+++ PKGBUILD
@@ -30,15 +30,15 @@ build() {
   export CXXFLAGS="${CXXFLAGS/-fcf-protection}"
   export CXXFLAGS="${CXXFLAGS/-fstack-clash-protection}"
 
-  make PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all
-  make PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all TILES=1 SOUND=1
+  make PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 GOLD=0 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all
+  make PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 GOLD=0 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all TILES=1 SOUND=1
   ./lang/compile_mo.sh
 }
 
 package_cataclysm-dda() {
   cd "Cataclysm-DDA-$_pkgver"
 
-  make DESTDIR="$pkgdir" PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all install
+  make DESTDIR="$pkgdir" PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 GOLD=0 RUNTESTS=0 LOCALIZE=1 LANGUAGES=all install
 
   # Docs
   install -d "$pkgdir/usr/share/doc/cataclysm-dda"
@@ -71,7 +71,7 @@ package_cataclysm-dda-tiles() {
   depends=('cataclysm-dda' 'sdl2_image' 'sdl2_ttf' 'freetype2' 'sdl2_mixer')
   cd "Cataclysm-DDA-$_pkgver"
 
-  make DESTDIR="$pkgdir" PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 LOCALIZE=1 LANGUAGES=all TILES=1 SOUND=1 install
+  make DESTDIR="$pkgdir" PREFIX=/usr PCH=0 RELEASE=1 USE_XDG_DIR=1 LTO=1 GOLD=0 LOCALIZE=1 LANGUAGES=all TILES=1 SOUND=1 install
 
   # Icons
   cd build-data/osx/AppIcon.iconset
