--- PKGBUILD
+++ PKGBUILD
@@ -59,6 +59,7 @@
 
   # use anonymous git
   patch -p1 -i "$srcdir/use-anonymous-git.patch"
+  patch -p1 -i "$srcdir/non-x86-simd.patch"
 
   # setup git submodules
   git submodule init
@@ -85,3 +86,7 @@
 
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(non-x86-simd.patch)
+sha512sums+=('21baeb137385151775ca17e64c1a49ae55ad9796e10a92726b244ddf716b2d1ceae1fd031fc0d851f2e946d9f863d69ee6f0bf53c7cc153d782afd178552b3f5')
+b2sums+=('64b6ee251a9c9cab4eac05cd5275813ab1aaee559333268645fc33cc4ad0ddb1335d5203d9480cd486a64f9d268a776f33a194bd853ce3966e56e812c16bb98c')
