diff --git PKGBUILD PKGBUILD
index 13a73fc..d4b513c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -59,6 +59,8 @@
 
   # use anonymous git
   patch -p1 -i "$srcdir/use-anonymous-git.patch"
+  # build baseline
+  patch -p1 -i "$srcdir/build-baseline.patch"
 
   # setup git submodules
   git submodule init
@@ -73,6 +75,7 @@
 build() {
   cd "$pkgname"
 
+  export USE_PORTABLE_SIMD=1
   python -m build --wheel --no-isolation
 }
 
@@ -81,3 +84,7 @@
 
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=('build-baseline.patch::https://github.com/spotify/pedalboard/commit/f9e5a6e75dedbdddece19b03e619621c18366feb.patch?full_index=1')
+sha512sums+=('a99eecf6a07a862945a67e7374f8d20afbfa1ed42f48c0dd9ee21c76b0ceeecf5368b55e4c549d5092747a9abd621ad545da0cc533a290f30ee4de04ef406cd2')
+b2sums+=('fc103ba4cb9e5b3d4b360e6ba04da95e905ec4e9808a183381b95dd5d56748410321302c5a53a06561676392c19b229349c64b36d7495cde3671ac4497614b96')
