diff --git .mocharc.json .mocharc.json
index 49e0f0e..8c1a379 100644
--- .mocharc.json
+++ .mocharc.json
@@ -2,5 +2,5 @@
 {
   "ignore": ["test/_bootstrap.js"],
   "require": ["test/_bootstrap.js"],
-  "timeout": "5s"
+  "timeout": "10s"
 }
\ No newline at end of file
diff --git test/_bootstrap.js test/_bootstrap.js
index 746d15d..e608d2b 100644
--- test/_bootstrap.js
+++ test/_bootstrap.js
@@ -62,7 +62,7 @@ Nominatim_Config.Reverse_Only = ${reverse_only};
   // 3. Create browser instance
   global.browser = await puppeteer.launch({
     defaultViewport: { width: 1024, height: 768 },
-    timeout: 10000,
+    timeout: 20000,
     // latency: 1000,
     headless: 'new',
     args: [
