--- PKGBUILD
+++ PKGBUILD
@@ -14,13 +14,18 @@ backup=('etc/webapps/nominatim-ui/config.theme.js')
 install=nominatim-ui.install
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/tags/v${pkgver}.tar.gz"
         'nginx.conf'
-)
+	'timeout1.patch::https://github.com/osm-search/nominatim-ui/commit/f48bd9485e7befbf3f4b055b5a6c67f3feb1b989.patch'
+	'timeout2.patch::https://github.com/osm-search/nominatim-ui/commit/2e5823e5ced30b6e83259a73fa16a3b2b5451ea0.patch')
 sha256sums=('31f793bafda4a3259e9b1a4be0cb4dd1a70d856c2fd96db760cbbf64462f8cd6'
-            '045dda10d978acedba43b071271731b2b63a8ad2743cdd6eda7dab2fd98a52ae')
+            '045dda10d978acedba43b071271731b2b63a8ad2743cdd6eda7dab2fd98a52ae'
+            'a26295d5c92ec5cae912702f5a388164cd9fa1370a61c6e474b683ee2f84ecf7'
+            'cf46a712d62265a9d4ab974a6bf3cc786a330bf950e4fc4b83d88bc91f7ccf2e')
 
 prepare() {
   cd "${pkgname}-tags-v${pkgver}"
 
+  patch -p1 -i ../timeout1.patch
+  patch -p1 -i ../timeout2.patch
   yarn install
 }
 
