--- PKGBUILD
+++ PKGBUILD
@@ -14,9 +14,10 @@ backup=('etc/webapps/nominatim-ui/config.theme.js')
 install=nominatim-ui.install
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/tags/v${pkgver}.tar.gz"
         'nginx.conf'
-)
+	'longer-timeout.patch')
 sha256sums=('31f793bafda4a3259e9b1a4be0cb4dd1a70d856c2fd96db760cbbf64462f8cd6'
-            '045dda10d978acedba43b071271731b2b63a8ad2743cdd6eda7dab2fd98a52ae')
+            '045dda10d978acedba43b071271731b2b63a8ad2743cdd6eda7dab2fd98a52ae'
+            '914520cf86ec3c212b798a4be3b3619ed1ae12f6ae8746b617069709bb83ceee')
 
 prepare() {
   cd "${pkgname}-tags-v${pkgver}"
