diff --git PKGBUILD PKGBUILD
index dc1e2c4f..ae084898 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ arch=('x86_64')
 url="https://linuxcontainers.org/lxd"
 license=('APACHE')
 depends=('lxc' 'lxcfs' 'squashfs-tools' 'dnsmasq' 'dqlite' 'libuv' 'ebtables')
-makedepends=('go' 'git' 'tcl' 'apparmor' 'libseccomp' 'systemd')
+makedepends=('go' 'git' 'tcl' 'apparmor' 'libseccomp' 'systemd' 'clang')
 optdepends=(
     'lvm2: for lvm2 support'
     'thin-provisioning-tools: for thin provisioning support'
@@ -53,7 +53,8 @@ build() {
   export CGO_LDFLAGS_ALLOW="-Wl,-z,now"
 
 	go build -v -tags "netgo" -o bin/ ./lxd-p2c/...
-	CGO_LDFLAGS="$CGO_LDFLAGS -static" go build -v -tags "agent" -o bin/ ./lxd-agent/...
+	# Using clang because gcc pthread will try to link libatomic.a on riscv which does not exist
+	CC="clang" CGO_LDFLAGS="$CGO_LDFLAGS -static" go build -v -tags "agent" -o bin/ ./lxd-agent/...
   for tool in fuidshift lxc lxc-to-lxd lxd lxd-benchmark; do
     go build -v -tags "libsqlite3" -o bin/ ./$tool/...
   done
