Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 460997)
+++ PKGBUILD	(working copy)
@@ -21,14 +21,17 @@
 provides=('initramfs')
 backup=('etc/mkinitcpio.conf')
 source=("https://sources.archlinux.org/other/$pkgname/$pkgname-$pkgver.tar.gz"{,.sig}
-        "fix-strip.patch::https://github.com/archlinux/mkinitcpio/commit/dece596d569ca90af902bf2403227b478b629c03.patch")
+        "fix-strip.patch::https://github.com/archlinux/mkinitcpio/commit/dece596d569ca90af902bf2403227b478b629c03.patch"
+        "mkinitcpio-generic-gzip-kernel.patch::https://github.com/archlinux/mkinitcpio/pull/112.patch")
 install=mkinitcpio.install
 sha512sums=('c180bef8af774305e8c9779b3405daaa5be33588bf89c6625a28eac1cde424f447782737d85d810fed4971d38b31086a072293d42aad08398c090f9538b56a4e'
             'SKIP'
-            'd19718f92916e00cece4e213bd7bd1cbf55b0094a39c459aaa995442342b64eb3fe8ec0bf023074f7ea26474a6fe2dc0370e0663823ed08ec5d33f6f645b833c')
+            '33bebb544b632066838a13a5d2d5adb09a74be37cee96e44e0f9e71b2ba5f4097d3642f98f299d936141692347504cc95a9cf35fdf444e27d6620871051d7ddc'
+            'b8077ab3b1cb21b1cdf839219596ec2ca080cd188d3f3404bc4072c22a6c59521dbf95d99409b718ec987e599286a1b5c33a2f597152b3c69642e8dd1c48b316')
 b2sums=('0c29659278e662eb58bc776f99e9579b5f0cf3268175382ecf2d635d54a6a8261ae173e387503a0c20a7492656c703a6699b52e1a6693c85c3211225d3e7abf8'
         'SKIP'
-        '2b404010d5edb4b7eac36e25f0bf1c7958b7bc7b9740a1fda859579de48dc47552d9eee1d40abb8d2ebf1dffc523a64797c834bba59a22bab9e537ad12e991c7')
+        'b1779f2d2fc0ce51b2ea347058b55845c00eb964d87013310528dcc0de661af9f0a2ef801531d66d17bf807ffe44cbdf7fbc34eb91734bfa973fc9b2fb08399b'
+        '9d23242387e24176cb93a8b7feca882430f95ca6d5deffc3721189073c650960824a7971415002912ea063366d17d25febca7fa8d43e44d38640113b981266b3')
 validpgpkeys=('ECCAC84C1BA08A6CC8E63FBBF22FB1D78A77AEAB'    # Giancarlo Razzolini
               'C100346676634E80C940FB9E9C02FF419FECBE16')   # Morten Linderud
 
@@ -35,6 +38,7 @@
 prepare() {
     cd "${pkgname}-${pkgver}"
     patch -Np1 < "$srcdir/fix-strip.patch"
+    patch -Np1 < "$srcdir/mkinitcpio-generic-gzip-kernel.patch"
 }
 
 check() {
