Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 437720)
+++ PKGBUILD	(working copy)
@@ -11,7 +11,7 @@
 license=(GPL LGPL)
 depends=(libxml2 gdk-pixbuf2 bzip2)
 makedepends=(gobject-introspection git gtk-doc autoconf-archive)
-checkdepends=(perl-xml-parser unzip valgrind) 
+checkdepends=(perl-xml-parser unzip) 
 provides=(libgsf-1.so)
 _commit=dc69cda017e7b6bdf74ec9af8d3aa79985efe64d  # tags/LIBGSF_1_14_48^0
 source=("git+https://gitlab.gnome.org/GNOME/libgsf.git#commit=$_commit")
