diff --git Cargo.toml Cargo.toml
index 83e0bb2..53264b6 100644
--- Cargo.toml
+++ Cargo.toml
@@ -23,3 +23,6 @@ winapi = { version = "0.3", features = ["securitybaseapi","debugapi"] }
 
 [dev-dependencies]
 insta = "0.16.0"
+
+[patch.crates-io]
+libc = { git = "https://github.com/rust-lang/libc", tag = "0.2.118" }
diff --git src/tests/fakes/fake_input.rs src/tests/fakes/fake_input.rs
index a8637e6..4193bc3 100644
--- src/tests/fakes/fake_input.rs
+++ src/tests/fakes/fake_input.rs
@@ -18,7 +18,7 @@ impl Iterator for TerminalEvents {
             Some(ev) => match ev {
                 Some(ev) => Some(ev),
                 None => {
-                    thread::sleep(time::Duration::from_millis(200));
+                    thread::sleep(time::Duration::from_secs(1));
                     self.next()
                 }
             },
