diff --git repos/community-x86_64/PKGBUILD repos/community-x86_64/PKGBUILD
index ca2f903..5e94fc6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,7 @@ prepare() {
   cp Cargo.lock "${_pkgname}-${pkgver}"
   # fetch dependencies
   cd "${_pkgname}-${pkgver}"
-  cargo fetch --locked --target="${CARCH}-unknown-linux-gnu"
+  cargo fetch --locked
   # rename man page
   mv "${_pkgname}.md" "${pkgname}.md"
   mkdir man
