diff --git PKGBUILD PKGBUILD
index 50a2f0c16..19897d19c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,16 +10,19 @@ arch=('any')
 url="https://github.com/klen/pylama"
 license=('GPL3')
 depends=('python-pycodestyle' 'python-pydocstyle' 'python-pyflakes' 'python-mccabe')
-checkdepends=('python-pytest' 'git' 'mypy')
+checkdepends=('python-pytest' 'git' 'mypy' 'python-tomli')
 optdepends=('python-radon: radon support')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/klen/pylama/archive/$pkgver.tar.gz"
-        pylama-pytest-6.patch::https://github.com/klen/pylama/pull/189.patch)
+        pylama-pytest-6.patch::https://github.com/klen/pylama/pull/189.patch
+        "strip_newline.patch")
 sha256sums=('acec2b80ad6a4781dc2626992b10ecac7b81be6c0145750c11688c281298f6fe'
-            '6a025bb41afe783841ecb28ab6faada4be9d7b3d152f0a58970757953d343cba')
+            '6a025bb41afe783841ecb28ab6faada4be9d7b3d152f0a58970757953d343cba'
+            'e4cba0a337f66515ff67d253157c9c3dd14d8337d74b47db2ce5dba52fc32f2b')
 
 prepare() {
   cd $pkgname-$pkgver
   patch -p1 -i ../pylama-pytest-6.patch
+  patch -Np1 -i "../strip_newline.patch"
 }
 
 build() {
