--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,11 @@ source=(https://downloads.sourceforge.net/hunspell/${pkgname}-${pkgver}.tar.gz)
 sha1sums=('0556c392beb59433e577e3517575801212201df6')
 sha256sums=('304636d4eccd81a14b6914d07b84c79ebb815288c76fe027b9ebff6ff24d5705')
 
+prepare() {
+  cd "$pkgname"-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd "$pkgname"-$pkgver
   ./configure --prefix=/usr --disable-static
