--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,18 @@ license=('Apache')
 depends=('typescript')
 makedepends=('jq' 'yarn')
 checkdepends=('npm')
-source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('df81d4074b2eb7bac72e49ad71c3d6fd77d61fc489074ceb123297547a8023b6f3e489b59180c85b17ae1c695689662bcb49a62c520e40fdb93cb167c1fed2cb')
+source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "typescript-language-server-mocha-timeout.patch")
+b2sums=('df81d4074b2eb7bac72e49ad71c3d6fd77d61fc489074ceb123297547a8023b6f3e489b59180c85b17ae1c695689662bcb49a62c520e40fdb93cb167c1fed2cb'
+        '730767552883bc2802aff2f43d93caa3da7314606070bcf571e324d00e552ada3702824986dafeefa360c199e25f3edcc159c6745e56759d96f659b53b243845')
 
 prepare() {
   cd $pkgname-$pkgver
   yarn --frozen-lockfile
+
+  # Increase tests' timeout
+  patch -Np1 -i $srcdir/typescript-language-server-mocha-timeout.patch
+  sed -i 's#})\.timeout(10000)#})\.timeout(40000)#' ./src/*.spec.ts ./src/**/*.spec.ts
 }
 
 build() {
