--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,18 @@ license=('Apache')
 depends=('typescript')
 makedepends=('jq' 'yarn')
 checkdepends=('npm')
-source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('94686ba5a02b04d0651f31c8f89decd379067f92c6c5a69a8da92d24dec7d26644e95bd0a3192bd843f54b8314a59ad8f9d0a0f8375373fcca98cc24e918bf02')
+source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "typescript-language-server-mocha-timeout.patch")
+b2sums=('8a6e163df0b3aa4aa1b15a98ad313a1871c8e71dd48870c8ed65dd9d8f362916687fc99a93bcba9feda4a6823a1bd91358efdbc5cb64fb627deb2e3f7661c4fe'
+        '730767552883bc2802aff2f43d93caa3da7314606070bcf571e324d00e552ada3702824986dafeefa360c199e25f3edcc159c6745e56759d96f659b53b243845')
 
 prepare() {
   cd $pkgname-$pkgver
   yarn --frozen-lockfile
+
+  # Increase tests' timeout
+  patch -Np1 -i $srcdir/typescript-language-server-mocha-timeout.patch
+  sed -i 's#})\.timeout(10000)#})\.timeout(40000)#' ./src/*.spec.ts ./src/**/*.spec.ts
 }
 
 build() {
