--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,17 @@ license=('Apache')
 depends=('typescript')
 makedepends=('jq' 'yarn')
 checkdepends=('npm')
-source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('4a5f32f8875c36975dedc756ebf7865374597f192d266c6991f307a7e1948cb596edc2b3fa6fdbcc65e0fb9d523763a7f4df22b0fbe8933c20f2c287dee7f14b')
+source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "typescript-language-server-mocha-timeout.patch")
+b2sums=('4a5f32f8875c36975dedc756ebf7865374597f192d266c6991f307a7e1948cb596edc2b3fa6fdbcc65e0fb9d523763a7f4df22b0fbe8933c20f2c287dee7f14b'
+        'f7fddd6ff5bf962c41aa3d45efa8c4c9b4d10cd05f6aef6d002eda6d2069d9c1dd43848e4e79f3e5a343ea96863aa65adc69c5756bb22163f2870af58a40399c')
 
 prepare() {
   cd $pkgname-$pkgver
   yarn --frozen-lockfile
+
+  # Increase tests' timeout
+  patch -Np1 -i $srcdir/typescript-language-server-mocha-timeout.patch
 }
 
 build() {
