--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,17 @@ license=('Apache')
 depends=('typescript')
 makedepends=('jq' 'yarn')
 checkdepends=('npm')
-source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('a1f0a852e3d5043085f1e958aa0ee0b585225107b7e774a4b6c062c50f4215de909331ca89ef12ae58193b32e1d81a1ed5c078960c33e7213ca8421c14ef932d')
+source=("$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "typescript-language-server-mocha-timeout.patch")
+b2sums=('a1f0a852e3d5043085f1e958aa0ee0b585225107b7e774a4b6c062c50f4215de909331ca89ef12ae58193b32e1d81a1ed5c078960c33e7213ca8421c14ef932d'
+        'f7fddd6ff5bf962c41aa3d45efa8c4c9b4d10cd05f6aef6d002eda6d2069d9c1dd43848e4e79f3e5a343ea96863aa65adc69c5756bb22163f2870af58a40399c')
 
 prepare() {
   cd $pkgname-$pkgver
   yarn --frozen-lockfile
+
+  # Increase tests' timeout
+  patch -Np1 -i $srcdir/typescript-language-server-mocha-timeout.patch
 }
 
 build() {
