--- PKGBUILD
+++ PKGBUILD
@@ -40,12 +40,13 @@
 	export CARGO_PROFILE_RELEASE_DEBUG=2
 	export LIBGIT2_NO_VENDOR=1
 	export OPENSSL_NO_VENDOR=true
-	cargo build --frozen --profile release-lto --all-features
+	export RUSTFLAGS="-C code-model=large"
+	cargo build --frozen --profile release --all-features
 }
 
 package() {
 	cd "$_archive"
-	install -Dm0755 -t "$pkgdir/usr/bin/" "target/release-lto/$pkgname"{,-proxy}
+	install -Dm0755 -t "$pkgdir/usr/bin/" "target/release/$pkgname"{,-proxy}
 	local lname=dev.lapce.lapce
 	install -Dm0644 -t "$pkgdir/usr/share/applications/" extra/linux/$lname.desktop
 	install -Dm0644 -t "$pkgdir/usr/share/metainfo/" extra/linux/$lname.metainfo.xml
