--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,6 @@
 license=(MIT)
 depends=(nodejs)
 makedepends=(
-  bun
   npm
   typescript
 )
@@ -19,11 +18,12 @@
 
 build() {
   cd $pkgname-$pkgver
+  # Use system TypeScript and install only the Node types needed below.
+  npm pkg delete devDependencies
   npm install --omit=dev --cache "$srcdir"/npm-cache
   # required for typescript when not installing dev dependencies
   npm install --save-dev @types/node --cache "$srcdir"/npm-cache
   tsc
-  bun build --compile --minify src/cli.ts --target=bun-linux-x64-modern --outfile=dist/linux-x64/dot-language-server
 }
 
 check() {
