--- PKGBUILD
+++ PKGBUILD
@@ -11,16 +11,28 @@ license=(MIT)
 # maybe revert back to openssl-1.1 or internal openssl
 # https://github.com/nodejs/node/issues/47852
 depends=(openssl zlib icu libuv c-ares brotli libnghttp2) # http-parser v8)
-makedepends=(python procps-ng)
+makedepends=(python procps-ng clang)
 optdepends=('npm: nodejs package manager')
 options=(!lto)
 provides=("nodejs=$pkgver")
 conflicts=(nodejs)
-source=(${url}/dist/v${pkgver}/node-v${pkgver}.tar.xz)
+source=(${url}/dist/v${pkgver}/node-v${pkgver}.tar.xz
+        nodejs-51495.diff::https://github.com/nodejs/node/pull/51495.diff)
 # https://nodejs.org/download/release/latest-iron/SHASUMS256.txt.asc
-sha256sums=('31807ebeeeb049c53f1765e4a95aed69476a4b696dd100cb539ab668d7950b40')
+sha256sums=('31807ebeeeb049c53f1765e4a95aed69476a4b696dd100cb539ab668d7950b40'
+            '33e47be086c60f8937fb027f3f07582ac2e40b22bec03427f140244d415c1289')
+
+_set_compilation_env() {
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
+prepare() {
+  patch -Np1 -d node-v$pkgver < nodejs-51495.diff
+}
 
 build() {
+  _set_compilation_env
   cd node-v${pkgver}
 
   ./configure \
@@ -46,6 +58,7 @@ check() {
 }
 
 package() {
+  _set_compilation_env
   cd node-v${pkgver}
   make DESTDIR="${pkgdir}" install
   install -Dm644 LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
