--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,15 @@ source=(https://nodejs.org/dist/v${pkgver}/node-v${pkgver}.tar.xz)
 # https://nodejs.org/download/release/latest-iron/SHASUMS256.txt.asc
 sha256sums=('6840d490ba4d1d51655e0fbe1209956a15db405510d7ea166bad98a8c9d37a4e')
 
+makedepends+=(clang)
+
+_set_compilation_env() {
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
 build() {
+  _set_compilation_env
   cd node-v${pkgver}
 
   # this uses malloc_usable_size, which is incompatible with fortification level 3
@@ -45,11 +53,13 @@ build() {
 }
 
 check() {
+  _set_compilation_env
   cd node-v${pkgver}
   make test-only || :
 }
 
 package() {
+  _set_compilation_env
   cd node-v${pkgver}
 
   # this uses malloc_usable_size, which is incompatible with fortification level 3
