diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 82c5ed8..8940b6d 100644
--- a/trunk/PKGBUILD
+++ b/trunk/PKGBUILD
@@ -15,6 +15,7 @@ sha1sums=('db4717a151742dbdb492318f104504a92075543a')
 build() {
   cd $pkgname-$pkgver

+  CFLAGS+=" -Wno-error=format-security"
   ./configure --prefix=/usr \
     --mandir=/usr/share/man --disable-static
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
