diff --git PKGBUILD PKGBUILD
index 8299c8c..c57a06a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,9 @@ depends=('bash' 'openssh' 'subversion' 'rsync' 'arch-install-scripts'
          'git' 'bzr' 'mercurial' 'diffutils' 'util-linux' 'awk')
 makedepends=('asciidoc')
 optdepends=('btrfs-progs: btrfs support')
-source=(https://sources.archlinux.org/other/${pkgname}/${pkgname}-${pkgver}.tar.gz{,.sig})
+source=(https://sources.archlinux.org/other/${pkgname}/${pkgname}-${pkgver}.tar.gz{,.sig}
+        https://github.com/archlinux/devtools/pull/64.patch
+        makepkg-riscv64.conf pacman-extra-riscv64.conf)
 validpgpkeys=('487EACC08557AD082088DABA1EB2638FF56C0C53'
               '4AA4767BBC9C4B1D18AE28B77F2D434B9741E8AC'
               '86CFFCA918CF3AF47147588051E8B148A9999C34'
@@ -23,10 +25,20 @@ validpgpkeys=('487EACC08557AD082088DABA1EB2638FF56C0C53'
               '6645B0A8C7005E78DB1D7864F99FFE0FEAE999BD'
               'E240B57E2C4630BA768E2F26FC1B547C8D8172C8')
 sha256sums=('9da0d997dff57be9193b8c0396ca0b8b17e97a6cb73f92a23d8b52a5fa941967'
-            'SKIP')
+            'SKIP'
+            '59abf3913a3fdc06bc02a61c51a1b1ac1373ec0c0ce0ad309053cc0102f289fc'
+            'b235f5367504aee159c0261ee296c1f80bfb309f12ce3c0e42fbf66d87b0de47'
+            'fc933f164d21774e7a1435d9fccf87cb05f7b601e89a2ba54b899b2ce1e809df')
 b2sums=('7305f2cd28f3970bb7ee175fd52b71dd65289078e2a5c1e1ba26b044de11c2ee5b00f5f1fb267be4fa5f5fd683ad820b529c41ca2f2de391bf902eabec38ba73'
-        'SKIP')
+        'SKIP'
+        '7d6bda662a96184262a55d0128da640b52346aa61d47de2e7b04ae85af851c77a9491b866fd28a6deaf49868ac3284ba601b214162ce14ca50bdbc9543268c84'
+        '0997b8e37a27edeb9142806d75ff2250134a4e158b3625e3cd93762c83b31bd05a05c33f93c1acf79e7d12125bd8921019ba46970df6626a3986fb3a48bee5f1'
+        'fbfb852a217879e37a4d19c6c3edc84e25591cfa15b51ef6c736d7fa924ad5b7d3b6063e4820c99eeab05e1aa57179bb31c83197858faca0a966c8fb515ff5cf')
 
+prepare(){
+  cd ${pkgname}-${pkgver}
+  patch -Np1 -i ../64.patch
+}
 build() {
   cd ${pkgname}-${pkgver}
   make BUILDTOOLVER="${pkgver}-${pkgrel}-${arch}" PREFIX=/usr
@@ -35,6 +47,15 @@ build() {
 package() {
   cd ${pkgname}-${pkgver}
   make PREFIX=/usr DESTDIR="${pkgdir}" install
+
+  ln -s archbuild "$pkgdir"/usr/bin/extra-riscv64-build
+  install -Dm644 ../makepkg-riscv64.conf ../pacman-extra-riscv64.conf -t "$pkgdir"/usr/share/devtools/
+
+  if [[ ! "$CARCH" =~ riscv ]]; then
+    install -dm755 "$pkgdir"/usr/share/devtools/setarch-aliases.d
+    echo "$CARCH" > "$pkgdir"/usr/share/devtools/setarch-aliases.d/riscv64
+  fi
+
 }
 
 # vim: ts=2 sw=2 et:
