--- PKGBUILD
+++ PKGBUILD
@@ -74,3 +74,14 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  # Normalize hostname label boundaries so wasi-libc++.build is valid for nspawn.
+  # https://gitlab.archlinux.org/archlinux/devtools/-/merge_requests/345#note_553787
+  patch -Np1 -i "$srcdir/normalize-machine-name.patch"
+}
+
+source+=('normalize-machine-name.patch')
+sha256sums+=('4e0f0a78c12748593b682659aac8e9aa81e9d0feec2410040f97df68cb6a0f4a')
+b2sums+=('9d8f41b853aa7489baa097a22eb5bbdb0980815aa28c7729a683018e1d7697544679290ac598374a215e0021e452f71208af5e5095ab90d8e5d523b35ae89679')
