Index: configure
===================================================================
--- configure	(revision 38322)
+++ configure	(working copy)
@@ -1853,6 +1860,7 @@
       nios2) host_arch=nios2 ;;
       vax) host_arch=vax ;;
       xtensa*) host_arch=xtensa ;;
+      riscv64) host_arch=riscv64 ;;
       *) host_arch=UNKNOWN ;;
   esac
   }
@@ -2777,6 +2785,11 @@
     iproc='nios2'
     ;;
 
+  riscv64)
+    arch='riscv64'
+    iproc='riscv64'
+    ;;
+
   mips*)
     arch='mips'
     iproc='mips'
