diff --git PKGBUILD PKGBUILD
index b1934b0..d137815 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=('x86_64')
 license=('GPL')
 url='https://code.google.com/archive/p/ibus-unikey/'
 depends=('ibus')
-makedepends=('intltool')
+makedepends=('intltool' 'gtk2')
 source=("https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/ibus-unikey/${pkgname}-${pkgver}.tar.gz"
         ibus-unikey-gcc6.patch)
 md5sums=('3bae6df0d4609a8c438c246030b9a61e'
