diff --git PKGBUILD PKGBUILD
index ee0b115a..6326cfb2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@
 depends=(dbus glib2 libsoup polkit libxau ostree json-glib libseccomp libarchive
          python bubblewrap appstream-glib xdg-dbus-proxy systemd)
 makedepends=(intltool gobject-introspection gtk-doc git docbook-xsl xmlto)
-checkdepends=(valgrind socat)
+checkdepends=(socat)
 provides=(libflatpak.so)
 _commit=103ed5c02c8453580fac19e5122d150f44f6774f  # tags/1.12.4^0
 source=("git+https://github.com/flatpak/flatpak#commit=$_commit"
