--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,11 @@ replaces=(
 source=("https://github.com/mongodb/mongo-c-driver/archive/$pkgver/$pkgname-$pkgver.tar.gz")
 sha256sums=('507414795dfb24ddf1a418b155b57459d8cea1191c7f0fcd8b826acf5400343c')
 
+prepare() {
+  cd $pkgname-$pkgver
+  sed -i 's/time.sleep(0.1)/time.sleep(1)/g' build/proc-ctl.py
+}
+
 build() {
   cd $pkgname-$pkgver
   # ENABLE_STATIC=ON is required to build tests.
@@ -48,6 +53,7 @@ build() {
 
 check() {
   cd $pkgname-$pkgver
+  CFLAGS+=' -mcmodel=medany'
   cmake --build build --target check
   export MONGOC_TEST_OFFLINE=ON
   export MONGOC_TEST_SKIP_LIVE=ON
