diff --git PKGBUILD PKGBUILD
index 36ab393..603ebfe 100644
--- PKGBUILD
+++ PKGBUILD
@@ -44,9 +44,17 @@
 )
 source=(
   "https://github.com/AFLplusplus/AFLplusplus/archive/refs/tags/v${pkgver}/${pkgname}-${pkgver}.tar.gz"
+  "${pkgname}-riscv64-WORD_SIZE_64.patch::https://github.com/AFLplusplus/AFLplusplus/pull/2348/commits/5842ba87e55c4a2a5842deee7f59a783260bca11.diff"
 )
-sha256sums=('8c6e9bef19b3d43020972701553734d1cb435c39a28b253f0dd6668e6ecb86bb')
-b2sums=('3c0b8cff5fcc5cdd10d29a98f227d9c9bd5fc16bbc08e8e501b385e1473a0e8347a9e5f81513d499725b23329a4cc3907d611597e14c8b312cfcab4a2116d382')
+sha256sums=('8c6e9bef19b3d43020972701553734d1cb435c39a28b253f0dd6668e6ecb86bb'
+            '3d17f282df92a607ee2ed1f9f4b7f133d896f90646fc438091a2526d511afd6a')
+b2sums=('3c0b8cff5fcc5cdd10d29a98f227d9c9bd5fc16bbc08e8e501b385e1473a0e8347a9e5f81513d499725b23329a4cc3907d611597e14c8b312cfcab4a2116d382'
+        '0b78a707d0795f84f89e17db4c15baf59871abc528c5aba6de73c84e9e1b885b826e56232d59b51576525d030dcb274d148dcfebfd9ce4cca0becf9f722c64bc')
+
+prepare() {
+  cd "AFLplusplus-${pkgver}"
+  patch -Np1 -i ../${pkgname}-riscv64-WORD_SIZE_64.patch
+}
 
 build() {
   cd "AFLplusplus-${pkgver}"
