diff --git PKGBUILD PKGBUILD
index d913cf68..bc2492ce 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,15 @@
             'poppler-data: correctly display certain CJK documents'
             'python: Python scripts support'
             'texlive-core: typeset TeX documents')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/TeXworks/texworks/archive/release-$pkgver.tar.gz")
-sha512sums=('3c1f09d3bb1a2a32947abc079c8def1a5f8a12203ccc6d25a2dc98ae7c4506f83653e8cac467ee74669d48a152c69110250c925c2bf9e40bafd6ae123f68a891')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/TeXworks/texworks/archive/release-$pkgver.tar.gz"
+        "fix-synchronizing-unit-tests-for-poppler.patch::https://github.com/TeXworks/texworks/commit/24a45c46665389bb0d4abe17d534adee0a169057.patch")
+sha512sums=('3c1f09d3bb1a2a32947abc079c8def1a5f8a12203ccc6d25a2dc98ae7c4506f83653e8cac467ee74669d48a152c69110250c925c2bf9e40bafd6ae123f68a891'
+            'f16715f260033af61130b2c05aae3a392eb900fd6d8f02506e09b525a857811d4569e6423b4a64a9b160d636b16dd9d4202806ec7efab365e6b67391d0d90fe1')
+
+prepare() {
+    cd "$srcdir/texworks-release-$pkgver"
+    patch -Np1 -i ../fix-synchronizing-unit-tests-for-poppler.patch
+}
 
 build() {
 	cmake -DCMAKE_INSTALL_PREFIX=/usr \
