--- PKGBUILD
+++ PKGBUILD
@@ -23,6 +23,13 @@
 source=("$pkgname::git+$url.git#tag=v$pkgver")
 b2sums=('77316382ba8a0534ec3de7e894e8abc6cd2a3a7cf453c9791d783a76e0b21ad20c7c260857f4c8f3e1e997e404d48ae4e1c47576163d1b7c1eb09f235d20ba75')
 
+source+=(symbolize-skip-tls.patch)
+b2sums+=('bebc0a4c8f2229f43641aee1e35125bf392a7da63124f879c676958b1da7ce7efbae9a2f5f7277e9e16db4c7eb0dcfb6a72f9abe074343996d5966043e380b91')
+
+prepare() {
+  patch -d $pkgname -p1 -i "$srcdir/symbolize-skip-tls.patch"
+}
+
 build() {
   cmake -B build -S $pkgname \
     -D CMAKE_INSTALL_PREFIX=/usr \
