Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 license=('MIT')
 depends=('glibc' 'libxext' 'libxrender' 'libx11' 'libgl' 'libxcursor' 'libibus' 'hidapi' 'libusb')
 makedepends=('alsa-lib' 'mesa' 'libpulse' 'libxrandr' 'libxinerama' 'wayland' 'libxkbcommon'
-             'wayland-protocols' 'ibus' 'fcitx' 'libxss' 'cmake' 'jack' 'ninja' 'pipewire')
+             'wayland-protocols' 'ibus' 'libxss' 'cmake' 'jack' 'ninja')
 optdepends=('alsa-lib: ALSA audio driver'
             'libpulse: PulseAudio audio driver'
             'jack: JACK audio driver'
@@ -36,9 +36,7 @@
       -DVIDEO_WAYLAND=ON \
       -DRPATH=OFF \
       -DCLOCK_GETTIME=ON \
-      -DJACK_SHARED=ON \
-      -DPIPEWIRE=ON \
-      -DPIPEWIRE_SHARED=ON
+      -DJACK_SHARED=ON
   ninja -C build
 }
 
