diff --git PKGBUILD PKGBUILD
index 2b8d915..1b41060 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,7 @@ makedepends=(
   cmake
   git
   libglvnd
+  simde
 )
 _commit=ca98a3ca28a863469861400da2b220aa45cb1243
 source=(
@@ -35,19 +36,19 @@ source=(
   git+https://github.com/SDWebImage/SDWebImage.git
   git+https://github.com/Cyan4973/xxHash.git
   git+https://github.com/facebook/zstd.git
+  add-riscv64-build.patch
 )
-b2sums=(
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-)
+b2sums=('a8a835b4d8fc94143f5c3c6869f96effc0c1fe1790aa719dbcb5456778bd70f6285cee573b3ea0226459236adb9c81a718c1ba381129e05ecc56b5c7c5ca06fc'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        '2d919fce01c0aa4704ffe8619b941b8ae1b2a3d0cb2d69b75ca9eb30e58ae0bb9b9868c3f77aa95028e94366c3434a41119248943e8ab14e632562b1d7fa7e3b')
 
 pkgver() {
   cd libretro-play
@@ -72,6 +73,9 @@ prepare() {
   git submodule init zstd
   git config submodule.zstd.url ../../../zstd
   git submodule update
+
+  cd ../..
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
