--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,7 @@ makedepends=(
   cmake
   git
   libglvnd
+  simde
 )
 _commit=8e1cc18be61f5baac0c78b9eeeb2f75d196217a5
 source=(
@@ -35,19 +36,19 @@ source=(
   git+https://github.com/SDWebImage/SDWebImage.git
   git+https://github.com/Cyan4973/xxHash.git
   git+https://github.com/facebook/zstd.git
+  add-riscv64-build.patch
 )
-b2sums=(
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-)
+b2sums=('SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        '55202fd99d778796ee0fcee0633f8d5fd9d8b974bb51b97f8fa9ccab00ea6732747288d2767f0804a5483c398a1d22ff890ef87e90e63fab19105051a031fcf4')
 
 pkgver() {
   cd libretro-play
@@ -82,6 +83,9 @@ prepare() {
   git submodule init zstd
   git config submodule.zstd.url ../../../zstd
   git submodule update zstd
+
+  cd ../..
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
