--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,7 @@ makedepends=(
   cmake
   git
   libglvnd
+  simde
 )
 _commit=1b631a6d5e5b31e802a39b093b65ebf0b697fb89
 source=(
@@ -35,19 +36,19 @@ source=(
   git+https://github.com/SDWebImage/SDWebImage.git
   git+https://github.com/Cyan4973/xxHash.git
   git+https://github.com/facebook/zstd.git
+  add-riscv64-build.patch
 )
-b2sums=(
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-)
+b2sums=('SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        '046baa04aeaf665ebe8d2f70f8eb18c660e6ae1d36c56b371679c1ae08bed77285f5865fdf93239a4cf433254e0257bb8083ab700fb1dd729ec6c33b5b9eae67')
 
 pkgver() {
   cd libretro-play
@@ -72,6 +73,9 @@ prepare() {
   git submodule init zstd
   git config submodule.zstd.url ../../../zstd
   git submodule update
+
+  cd ../..
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
