--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,7 @@ depends=(
   libgl
   libretro-core-info
   zlib
+  simde
 )
 makedepends=(
   cmake
@@ -35,19 +36,19 @@ source=(
   git+https://github.com/SDWebImage/SDWebImage.git
   git+https://github.com/Cyan4973/xxHash.git
   git+https://github.com/facebook/zstd.git
+  add-riscv64-build.patch
 )
-b2sums=(
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-  SKIP
-)
+b2sums=('SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'SKIP'
+        'fc9600299c2c84df487e5de0f2979c3a1a2c49708d55bd41f90a462567bc7b8370a50c536983932f1c418f85274eda6a9b1c5151a79444eee2d610d7e388532a')
 
 pkgver() {
   cd libretro-play
@@ -72,6 +73,9 @@ prepare() {
   git submodule init zstd
   git config submodule.zstd.url ../../../zstd
   git submodule update
+
+  cd ../..
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
