--- PKGBUILD
+++ PKGBUILD
@@ -63,6 +63,9 @@
   git submodule init zstd
   git config submodule.zstd.url ../../../zstd
   git -c protocol.file.allow=always submodule update
+
+  cd ../..
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
@@ -83,3 +86,7 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+makedepends+=(simde)
+source+=(add-riscv64-build.patch)
+b2sums+=('6f79fddf5a412f8ee6b7f68b0b3c1e499a841c56052d3f486fc1010e19794e65b4a312451889f1a3e153f2bec8289f80f7808eacae5331de4c086b4b4cad7b1c')
