Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422469)
+++ PKGBUILD	(working copy)
@@ -24,6 +24,8 @@
   ln -s /usr/bin/{rcc,uic,moc} qtbase/bin/
 
   sed -e 's|QMAKE_PYTHON2 = python|QMAKE_PYTHON2 = python2|' -i qtwebengine/mkspecs/features/functions.prf # Use python2
+
+  sed -i '/QT_BEGIN_NAMESPACE/i #include <limits>' qtbase/src/corelib/global/{qendian.h,qfloat16.h} qtbase/src/corelib/text/qbytearraymatcher.h
 }
 
 build() {
