--- PKGBUILD
+++ PKGBUILD
@@ -79,10 +79,11 @@
 )
 checkdepends=(
   'python-pytest' 'python-jsbeautifier' 'python-h5py' 'acl' 'binutils' 'bzip2' 'cdrtools' 'cpio' 'e2fsprogs' 'enjarify'
-  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'ghc' 'gnupg' 'mono' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
+  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'ghc' 'gnupg' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
   'libxmlb' 'lz4' 'unzip' 'gzip' 'tar' 'tcpdump' 'vim' 'xz' 'llvm' 'colord' 'fpc' 'openssh' 'openssl' 'odt2txt' 'docx2txt' 'r' 'dtc'
   'giflib' 'gnumeric' 'python-progressbar' 'binwalk' 'python-argcomplete' 'zstd' 'uboot-tools' 'asar')
 source=(https://diffoscope.org/archive/diffoscope-${pkgver}.tar.bz2{,.asc})
+source+=(ghc-snapshot-version.patch)
 sha512sums=('074c7c9f253d8d3b28521d675e0a0a55320863fe2fd2655aecba8daa1f0fcba6e8585e1b50337743ad682a710787b31ceb41c7449248c941b2d20b125816bf01'
             'SKIP')
 b2sums=('f3634737a13cefb3e3847703d1b8890d5c46aafc46bfd02cee7e1b0bc2ac24df40e124d971cdd3149f08086061a9e7d516e0771121bbdbc62c202438fecaee4a'
@@ -92,8 +93,12 @@
               # Mattia Rizzolo <mattia@mapreri.org>
               # Chris Lamb <chris@chris-lamb.co.uk>
 
+sha512sums+=('SKIP')
+b2sums+=('SKIP')
+
 prepare() {
   cd ${pkgname}-${pkgver}
+  patch -Np1 -i ../ghc-snapshot-version.patch
 }
 
 build() {
