--- PKGBUILD
+++ PKGBUILD
@@ -79,7 +79,7 @@
 )
 checkdepends=(
   'python-pytest' 'python-jsbeautifier' 'python-h5py' 'acl' 'binutils' 'bzip2' 'cdrtools' 'cpio' 'e2fsprogs' 'enjarify'
-  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'ghc' 'gnupg' 'mono' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
+  'hdf5' 'imagemagick' 'java-environment=11' 'fontforge' 'gettext' 'ghc' 'gnupg' 'pgpdump' 'poppler' 'sqlite' 'squashfs-tools'
   'libxmlb' 'lz4' 'unzip' 'gzip' 'tar' 'tcpdump' 'vim' 'xz' 'llvm' 'colord' 'fpc' 'openssh' 'openssl' 'odt2txt' 'docx2txt' 'r' 'dtc'
   'giflib' 'gnumeric' 'python-progressbar' 'binwalk' 'python-argcomplete' 'zstd' 'uboot-tools' 'asar')
 source=(https://diffoscope.org/archive/diffoscope-${pkgver}.tar.bz2{,.asc})
@@ -94,6 +94,7 @@
 
 prepare() {
   cd ${pkgname}-${pkgver}
+  patch -Np1 -i ../ghc-snapshot-version.patch
 }
 
 build() {
@@ -123,3 +124,7 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+source+=(ghc-snapshot-version.patch)
+sha512sums+=('8e83b9da16f97e00148a60bf270b1dbd0873e769f4df36bb0e24291b879d8151ed31750f7678eb806a5d858f76b37337f278adf85b1d1ffb076fce5744443a7a')
+b2sums+=('871ab9c7a6fb044b084a0b2e628b0c49901e84427e5e30f3ea76c67906ca98fc57d1e2a10c8c7467585a91f107867391b3de56a09620b30cefbee61209dc8fd9')
