diff --git PKGBUILD PKGBUILD
index 9129a644..5a340cb4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,10 @@ depends=(gtk3 libhandy)
 makedepends=(git meson appstream-glib)
 groups=(gnome)
 _commit=a867bea20ec621f42e8bd815311654cb687d6733  # tags/41.0^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-screenshot.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-screenshot.git#commit=$_commit"
+        "fix_positional_args.patch")
+sha256sums=('SKIP'
+            'b1bbe197e3e82ae0c11f8126da57050f615862d997c0d1f57c922f4c6e6f77f9')
 
 pkgver() {
   cd $pkgname
@@ -22,6 +24,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../fix_positional_args.patch
 }
 
 build() {
