diff --git PKGBUILD PKGBUILD
index 4645a58ac..901f76a01 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,9 +16,17 @@
 optdepends=(
   'bash: for a2j'
   'python-dbus: for a2j and a2j_control')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/linuxaudio/${pkgname}/archive/${pkgver}.tar.gz")
-sha512sums=('5bd13b6904ed68c5bfe40ca516fd49b7eb4d4a946b9908ee04687265848734c8e1a81579f0f1a5bd0752595be8858dc748da10487b7f366394c09a5ffc7d5e5c')
-b2sums=('ad079e76a54f56ee8f82150c306f91aead52f86632978ea6f3376a0bf163744379210b34701b39da4ee14fc4d8a1c6be82d1052abeb8ccf79410bc242d174d62')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/linuxaudio/${pkgname}/archive/${pkgver}.tar.gz"
+        "enable-riscv-build.diff::https://github.com/jackaudio/a2jmidid/pull/18.diff")
+sha512sums=('5bd13b6904ed68c5bfe40ca516fd49b7eb4d4a946b9908ee04687265848734c8e1a81579f0f1a5bd0752595be8858dc748da10487b7f366394c09a5ffc7d5e5c'
+            'c1af7c8491787f2bfe84f35f8691dc1a0e1cc198a06c50558e465222812b86680756463e394312f8a9f0b45a7199cf0ca97ceb49bbcd5bb763770f3a3c828c52')
+b2sums=('ad079e76a54f56ee8f82150c306f91aead52f86632978ea6f3376a0bf163744379210b34701b39da4ee14fc4d8a1c6be82d1052abeb8ccf79410bc242d174d62'
+        'a41719838d6447513c54fd37fe0ad503f68de2692fe7227d6d56534d73521e8d24372a935272d17f458183729a4fc268a090bd14688a2f36395269d0d79cc672')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i ../enable-riscv-build.diff
+}
 
 build() {
   cd "${pkgname}-${pkgver}"
