--- a/src/runtime_src/core/common/detail/linux/sysinfo.h
+++ b/src/runtime_src/core/common/detail/linux/sysinfo.h
@@ -11,7 +11,7 @@
 #include <sys/utsname.h>
 #include <thread>
 
-#if defined(__aarch64__) || defined(__arm__) || defined(__mips__)
+#if defined(__aarch64__) || defined(__arm__) || defined(__mips__) || defined(__riscv)
   #define MACHINE_NODE_PATH "/proc/device-tree/model"
 #elif defined(__PPC64__)
   #define MACHINE_NODE_PATH "/proc/device-tree/model-name"
