--- PKGBUILD
+++ PKGBUILD
@@ -61,6 +61,7 @@
 
   git -C src/runtime_src/core/common/aiebu/ checkout 4bf58d5a8d46089847130ecffb8952504482feb5
   patch -Np1 -i ../install-to-bin.patch
+  patch -Np1 -i ../riscv64-sysinfo.patch
 }
 
 build() {
@@ -86,3 +87,6 @@
   install -Dm644 "$pkgname/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
   rm -f "$pkgdir/usr/include/CL/cl_ext.h"
 }
+
+source+=(riscv64-sysinfo.patch)
+sha256sums+=('cebd4a8ddf23e457531393fe8c57f91d7d0c2798b001890c82f197f373e0bf78')
