diff --git PKGBUILD PKGBUILD
index 8f75207..9d053fc 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,11 +29,15 @@ optdepends=('faac: faac backend'
             'vorbis-tools: vorbis tools backend'
             'wavpack: wavpack backend'
             'sox: sox plugin (change sample rate, various effects)')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/dfaust/soundkonverter/archive/v${pkgver}.tar.gz")
-sha256sums=('92117f8aff85aac3a336cc0c94f3739aaa2d07ddd741d737dfd4b1571c60a0ce')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/dfaust/soundkonverter/archive/v${pkgver}.tar.gz"
+        "fix_cmake_module_priority.patch")
+sha256sums=('92117f8aff85aac3a336cc0c94f3739aaa2d07ddd741d737dfd4b1571c60a0ce'
+            "61b0e93486eb9d6b80337bf5158ec914bd379220980fbbe1a06bfdaeb091ab12")
 
 prepare() {
   mkdir -p build
+  cd $pkgname-$pkgver
+  patch -Np1 -i "../fix_cmake_module_priority.patch"
 }
 
 build() {
