diff --git a/system-tests/001/fs.test.js b/system-tests/001/fs.test.js
index e9cad9d4..653eb13b 100644
--- a/system-tests/001/fs.test.js
+++ b/system-tests/001/fs.test.js
@@ -14,4 +14,4 @@ it('fs - valid sanitize.css and their config', async () => {
 			}),
 		),
 	).toMatchSnapshot();
-}, 10000);
+}, 30000);
diff --git a/system-tests/003/fs.test.js b/system-tests/003/fs.test.js
index a03baea7..48201546 100644
--- a/system-tests/003/fs.test.js
+++ b/system-tests/003/fs.test.js
@@ -15,4 +15,4 @@ it('fs - zen garden CSS with standard config', async () => {
 			}),
 		),
 	).toMatchSnapshot();
-}, 10000);
+}, 30000);
