diff --git a/system-tests/001/fs.test.mjs b/system-tests/001/fs.test.mjs
index 4877ab5..04ac5fa 100644
--- a/system-tests/001/fs.test.mjs
+++ b/system-tests/001/fs.test.mjs
@@ -13,4 +13,4 @@ it('fs - valid sanitize.css and their config', async () => {
 			}),
 		),
 	).toMatchSnapshot();
-}, 10000);
+}, 30000);
diff --git a/system-tests/003/fs.test.mjs b/system-tests/003/fs.test.mjs
index 0927a62..ceab6db 100644
--- a/system-tests/003/fs.test.mjs
+++ b/system-tests/003/fs.test.mjs
@@ -14,4 +14,4 @@ it('fs - zen garden CSS with standard config', async () => {
 			}),
 		),
 	).toMatchSnapshot();
-}, 10000);
+}, 30000);
