diff --git PKGBUILD PKGBUILD
index d2a1c61..2d1020e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,14 @@ depends=('nodejs')
 makedepends=('git' 'npm' 'rsync')
 optdepends=('stylelint-config-standard: for the standard shareable config')
 options=('!emptydirs')
-source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver")
-b2sums=('SKIP')
+source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver"
+	"modify-time-limit.patch")
+b2sums=('SKIP'
+	'1f8720f4bd14847441d21d194cb8874bf0da7946a939feaac68b5d65b36bac9cbfc0fd4fe38ecf4f3541f16e9d8e628a6b5eb30b1b06821f5a837ad199623dcb')
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../modify-time-limit.patch
   npm ci
 }
 
