--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ b2sums=('6b5d7a9adbd027b33d74cd2045b17d39ff3fa465fc00dd76796744b1c55d5411674ac9e
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../modify-time-limit.patch
   npm ci
 }
 
@@ -42,3 +43,7 @@ package() {
   cp -r {CHANGELOG,CONTRIBUTING,README}.md docs "$pkgdir"/usr/share/doc/$pkgname
   install -Dm644 -t "$pkgdir"/usr/share/licenses/$pkgname LICENSE
 }
+
+source+=("modify-time-limit.patch")
+b2sums+=('1f8720f4bd14847441d21d194cb8874bf0da7946a939feaac68b5d65b36bac9cbfc0fd4fe38ecf4f3541f16e9d8e628a6b5eb30b1b06821f5a837ad199623dcb')
+
