diff --git PKGBUILD PKGBUILD
index 7c76472..aff2fa4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,14 @@ depends=('nodejs')
 makedepends=('git' 'npm' 'rsync')
 optdepends=('stylelint-config-standard: for the standard shareable config')
 options=('!emptydirs')
-source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver")
-b2sums=('d70a1c89f862d4de35de1be7860435b484071ec3bc29d308e3649d786f2631640264518b8997c0f0e8c8bb2f07b3fdc5bb470f6e2af258a45e8e365f5547b7b3')
+source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver"
+        "modify-time-limit.patch")
+b2sums=('d70a1c89f862d4de35de1be7860435b484071ec3bc29d308e3649d786f2631640264518b8997c0f0e8c8bb2f07b3fdc5bb470f6e2af258a45e8e365f5547b7b3'
+        '1f8720f4bd14847441d21d194cb8874bf0da7946a939feaac68b5d65b36bac9cbfc0fd4fe38ecf4f3541f16e9d8e628a6b5eb30b1b06821f5a837ad199623dcb')
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../modify-time-limit.patch
   npm ci
 }
 
