--- PKGBUILD
+++ PKGBUILD
@@ -10,11 +10,14 @@ license=('MIT')
 depends=('nodejs')
 makedepends=('git' 'npm' 'rsync')
 optdepends=('stylelint-config-standard: for the standard shareable config')
-source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver")
-b2sums=('SKIP')
+source=("git+https://github.com/stylelint/stylelint.git#tag=$pkgver"
+        "modify-time-limit.patch")
+b2sums=('SKIP'
+        'e2c91d195ec1caf4cd7206322c5c5df23580b6be53ab2c904201a1f0a3a617bdad5c6ca84adef03513ecb92b2fb182f3ecf2e7abbba0d108efb7cb8de75369dd')
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../modify-time-limit.patch
   npm ci
 }
 
