--- PKGBUILD
+++ PKGBUILD
@@ -36,7 +36,6 @@
   alsa-lib
   hicolor-icon-theme
   jack
-  lib32-gcc-libs
   libpulse
   libsndfile
   libspeechd
@@ -103,6 +102,8 @@
   local cmake_options_client=(
     -D update=OFF
     -D server=OFF
+    # No multilib on riscv64; keep the native overlay.
+    -D overlay-xcompile=OFF
     -B build-client
     -D bundled-json=OFF
     -D bundled-rnnoise=OFF
@@ -153,7 +154,6 @@
   )
   optdepends=(
     'bash: for mumble-overlay'
-    'lib32-glibc: for mumble-overlay'
     'espeak-ng: Text-to-speech support'
     'speech-dispatcher: Text-to-speech support'
   )
