--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,6 @@ makedepends+=(
   alsa-lib
   hicolor-icon-theme
   jack
-  lib32-gcc-libs
   libpulse
   libsndfile
   libspeechd
@@ -89,6 +88,9 @@ prepare() {
 }
 
 build() {
+  # fix signedness of char, see https://github.com/mumble-voip/mumble/issues/3845
+  CFLAGS+=" -fsigned-char "
+  CXXFLAGS+=" -fsigned-char "
   local default_options=(
     -D CMAKE_INSTALL_PREFIX=/usr
     -D CMAKE_BUILD_TYPE=None
