--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,16 @@ license=('Apache')
 makedepends=('go')
 depends=('pam')
 source=("fscrypt-$pkgver.zip::https://github.com/google/fscrypt/archive/v$pkgver.zip"
-        "pam_config")
+        "pam_config"
+        "disable_c_shared_build.patch")
 sha512sums=('0c2414b55e22b314aa771c3f205c45c2c7f45f51fed48d10006f60fb3371383d91d4c8a57f159bedfd4cf20e86ea4874531df511bafda531f5b9a25424768ecd'
-            '56f443cd2fbed9c164693c800cd8dc08bb35b88bb9c912fb35eb43cc6c6bc3b966d22d9810f43580d12b9faa8d753419a06336f08c47a0e602c48a103a705299')
+            '56f443cd2fbed9c164693c800cd8dc08bb35b88bb9c912fb35eb43cc6c6bc3b966d22d9810f43580d12b9faa8d753419a06336f08c47a0e602c48a103a705299'
+            '892f79a746cdf589a3bbae78f8b28087249d50a61f7749124f86e7563c39394154081a79fce9c9ad56cf55162900893b76cd4081eec07af0853811bd64dfbcd2')
+
+prepare() {
+  cd "fscrypt-$pkgver"
+  patch -Np1 -i "../disable_c_shared_build.patch"
+}
 
 build() {
   export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
