diff --git PKGBUILD PKGBUILD
index 2385567..bb757b5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -2,7 +2,7 @@
 
 _hkgname=ghc-lib-parser
 pkgname=haskell-ghc-lib-parser
-pkgver=9.6.6.20240701
+pkgver=9.6.7.20250325
 pkgrel=1
 pkgdesc="The GHC API, decoupled from GHC versions"
 url="https://github.com/digital-asset/ghc-lib"
@@ -11,11 +11,12 @@ arch=('x86_64')
 depends=('ghc-libs')
 makedepends=('ghc' 'alex' 'happy' 'uusi')
 source=("https://hackage.haskell.org/packages/archive/$_hkgname/$pkgver/$_hkgname-$pkgver.tar.gz")
-sha256sums=('8c9184efc89fe1ff2e5d2e34b27d7bece4196f9b7a67072b2ab1636e875e02b3')
+sha256sums=('ac031a3c16a2391327672a540bb42b8e3a38f490803390f2ff6537c3677cd6fd')
 
 prepare() {
   cd $_hkgname-$pkgver
   gen-setup
+  uusi -u happy
 }
 
 build() {
