diff --git PKGBUILD PKGBUILD
index 00384ae6..44f47b73 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,10 @@
 makedepends=(git meson)
 groups=(gnome)
 _commit=deb9ff18302457b91fb8013f98880c8e3a548c19  # tags/41.2^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-calendar.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-calendar.git#commit=$_commit"
+        "fix-meson-0.61-builds.patch::https://gitlab.gnome.org/GNOME/gnome-calendar/-/commit/e35dbc1acd03dd1751b755b6c8e1932a692ad4e1.patch")
+sha256sums=('SKIP'
+            'ded10dd0b1e9fce59414fd60a6ab42c560842745ef38774dc6d30f222293e71d')
 
 pkgver() {
   cd $pkgname
@@ -23,6 +25,7 @@
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../fix-meson-0.61-builds.patch
 }
 
 build() {
