--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,9 @@ b2sums=('a46cae0327a32c3b3c4e70bf0fc8d5ae11e3b01e87f3240fd125264c6cada25129a9979
 
 prepare() {
   cd ${pkgname}-${pkgver}
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
