--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,7 @@ build() {
   make \
     SYS_LIB_PATH=/usr/lib \
     TARGET_PLATFORM="$CARCH-gpp" \
-    INSTRUCTION_SET=sse2 \
+    INSTRUCTION_SET=none \
     VERSION_STR=$(git describe --tags) \
     VERSION_INT=$(git describe --tags | sed "s/[^0-9]//g") \
     vst3
@@ -45,7 +45,7 @@ package() {

   # vst3
   install -vDm644 \
-    "dist/js80p-dev-linux-$CARCH-sse2-vst3_single_file/js80p.vst3" \
+    "dist/js80p-dev-linux-$CARCH-none-vst3_single_file/js80p.vst3" \
     "$pkgdir/usr/lib/vst3/$pkgname.vst3/Contents/$CARCH-linux/$pkgname.so"

   # presets
