diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 7bbee18..fc53a9d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,8 @@ sha256sums=('b6b99dde5c8f0822ae362606a0429628ee478f4ec943a156723841b742954707')

 build() {
     cd "projectM-$pkgver"
+    autoreconf -fi
+    autoupdate -f
     ./configure --prefix=/usr --enable-gles --enable-sdl --enable-threading --enable-qt
     sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool # Fix overlinking
     make
