Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1346003)
+++ PKGBUILD	(working copy)
@@ -12,13 +12,17 @@
 optdepends=('octave: Octave plugin')
 provides=('soapylms7')
 conflicts=('soapylms7')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('4ff422d04bb8795463da1a3e04dd742701bca89cb9003e3a1af3a97f9aa13a167c6cafa4b36734c3c810d08cec96f4d8aced40413504660f42a9c7208bfa3264')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
+        "limesuite-march-native.patch")
+sha512sums=('4ff422d04bb8795463da1a3e04dd742701bca89cb9003e3a1af3a97f9aa13a167c6cafa4b36734c3c810d08cec96f4d8aced40413504660f42a9c7208bfa3264'
+            '004f2ad8165445f1119e10a403d166dab3a5033837b65ab5582bf526e3d9945eb4e3b5fce79e2df5f236f2328cdc1986bc6e833dd61ce7bc11d6b1fbb2940d2c')
 
 prepare() {
   cd LimeSuite-$pkgver
 
   sed -i 's|MODE="660", GROUP="plugdev"|MODE="666"|g' udev-rules/64-limesuite.rules
+
+  patch -Np1 -i ../limesuite-march-native.patch
 }
 
 build() {
