--- PKGBUILD
+++ PKGBUILD
@@ -69,9 +69,9 @@ package() {
   install -Dm644 grafana.sysusers "$pkgdir/usr/lib/sysusers.d/grafana.conf"
   install -Dm644 grafana.service "$pkgdir/usr/lib/systemd/system/grafana.service"
   cd $pkgname
-  install -Dsm755 bin/linux-amd64/grafana "$pkgdir/usr/bin/grafana"
-  install -Dsm755 bin/linux-amd64/grafana-server "$pkgdir/usr/bin/grafana-server"
-  install -Dsm755 bin/linux-amd64/grafana-cli "$pkgdir/usr/bin/grafana-cli"
+  install -Dsm755 bin/linux-riscv64/grafana "$pkgdir/usr/bin/grafana"
+  install -Dsm755 bin/linux-riscv64/grafana-server "$pkgdir/usr/bin/grafana-server"
+  install -Dsm755 bin/linux-riscv64/grafana-cli "$pkgdir/usr/bin/grafana-cli"
   install -Dm640 -o207 -g207 conf/sample.ini "$pkgdir/etc/$pkgname.ini"
   install -Dm644 conf/defaults.ini "$pkgdir/usr/share/$pkgname/conf/defaults.ini"
   install -dm755 "$pkgdir/usr/share/grafana/"
@@ -83,4 +83,7 @@ package() {
   rm -r "$pkgdir/usr/share/grafana/public/test"
 }
 
+source+=('upgrade-nx.patch')
+sha512sums+=('f67d4f2a49ba9758161ae20952fb73d7fdd505d8cd8a9b7d9bdaf808f4bd3c32d9c86af121b2d43ff2cf7056bf98b000b9974123fe9ef5041d8263d6d858a725')
+
 # vim:set ts=2 sw=2 et:
