diff --git PKGBUILD PKGBUILD
index ba484b95a..4013e9564 100644
--- PKGBUILD
+++ PKGBUILD
@@ -80,8 +80,8 @@ package() {
   install -Dm644 grafana.sysusers "$pkgdir/usr/lib/sysusers.d/grafana.conf"
   install -Dm644 grafana.service "$pkgdir/usr/lib/systemd/system/grafana.service"
   cd $pkgname
-  install -Dsm755 bin/linux-amd64/grafana-server "$pkgdir/usr/bin/grafana-server"
-  install -Dsm755 bin/linux-amd64/grafana-cli "$pkgdir/usr/bin/grafana-cli"
+  install -Dsm755 bin/linux-riscv64/grafana-server "$pkgdir/usr/bin/grafana-server"
+  install -Dsm755 bin/linux-riscv64/grafana-cli "$pkgdir/usr/bin/grafana-cli"
   install -Dm640 -o207 -g207 conf/sample.ini "$pkgdir/etc/$pkgname.ini"
   install -Dm644 conf/defaults.ini "$pkgdir/usr/share/$pkgname/conf/defaults.ini"
   install -dm755 "$pkgdir/usr/share/grafana/"
