--- wscript.orig	2025-02-24 04:39:04.642894481 -0500
+++ wscript	2025-02-24 04:57:45.745046300 -0500
@@ -47,6 +47,8 @@
             local_vamp_lib = local_vamp_lib_amd64
         elif platform.machine() == 'x86_64':
             local_vamp_lib = local_vamp_lib_i686
+        else:
+            local_vamp_lib = "/non-existent"
     elif sys.platform == 'darwin':
         local_vamp_lib = local_vamp_lib_osx
     elif sys.platform == 'win32':
@@ -78,8 +80,7 @@
         conf.env.CXXFLAGS += ['-g', '-Wall', '-Wextra']
 
     if sys.platform.startswith('linux'):
-        conf.env['CXXFLAGS'] += ['-O3', '-msse', '-msse2', '-mfpmath=sse',
-                '-ftree-vectorize']
+        conf.env['CXXFLAGS'] += ['-O3', '-ftree-vectorize']
         if 'mingw' in conf.env.CXX[0]:
             conf.env.append_value('LINKFLAGS', '-Wl,--enable-auto-import')
             conf.env.append_value('LINKFLAGS', '-Wl,--retain-symbols-file=../vamp-plugin.list')
