--- PKGBUILD
+++ PKGBUILD
@@ -17,11 +17,14 @@ options=('!lto')
 
 prepare() {
   cd "${pkgname%-cli}-$pkgver"
+  patch -Np1 -i ../test-cfg.patch
   cargo fetch --target "$(rustc -vV | sed -n 's/host: //p')" # --locked
 }
 
 build() {
   cd "${pkgname%-cli}-$pkgver"
+  export CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
+  export CXXFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
   cargo build -p "$pkgname" --release --frozen
 }
 
@@ -37,4 +40,8 @@ package() {
   install -Dm 644 LICENSE-MIT -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=("test-cfg.patch::https://github.com/DioxusLabs/dioxus/pull/3624.diff")
+b2sums+=('80c8b565762e4fc68cdb2213f193b55acdc211851b15ca3482fddecfa001b21e323dc83dedd9b78714e0cec2492889287472cd4f2ed9d04269757a19f14f81da')
+makedepends+=('clang' 'cmake')
+
 # vim: ts=2 sw=2 et:
