Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422794)
+++ PKGBUILD	(working copy)
@@ -12,7 +12,7 @@
          atk at-spi2-atk openjpeg2 woff2 libtasn1 libxslt libsystemd bubblewrap
          libseccomp xdg-dbus-proxy gstreamer gst-plugins-base-libs libxkbcommon
          wayland wpebackend-fdo)
-makedepends=(cmake ninja gtk-doc python ruby systemd gst-plugins-bad gperf
+makedepends=(cmake ninja gtk-doc python ruby systemd gperf
              wayland-protocols)
 optdepends=('geoclue: Geolocation support'
             'gst-plugins-good: media decoding'
