--- PKGBUILD
+++ PKGBUILD
@@ -113,6 +113,7 @@ validpgpkeys=(
 
 prepare() {
   cd wpewebkit-$pkgver
+  patch -Np1 -i ../fix-ftbfs-riscv64.patch
 }
 
 build() {
@@ -185,3 +186,7 @@ package_wpewebkit-docs() {
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(fix-ftbfs-riscv64.patch)
+sha256sums+=('b42d36c275e5fbd6da8e0761381d9eac13b2d103a366a61d9d450421948385a3')
+b2sums+=('cd80efecf12982e9e9335032a433e094614c8b95edce7114d37c6d651ee1f5bb3e331c3911d7901055cd59ca3ed5de38d14e15fab5488d3248bf5899623b2f8c')
