--- PKGBUILD
+++ PKGBUILD
@@ -64,9 +64,11 @@ makedepends=(
 )
 source=(
   $url/releases/wpewebkit-$pkgver.tar.xz{,.asc}
+  fix-riscv64-build.patch # bpo: https://github.com/WebKit/WebKit/commit/c07cdb6ae80b0847da58a6acb1022b8b3e170073
 )
 sha256sums=('a8ec2bcfa1613768ab3ce7f65cac4f214835266c2ff59440d72180baec3086b6'
-            'SKIP')
+            'SKIP'
+            '565e21e1e474ae1769ce2e3e8939193aa58bfe12a3c30f85c8bf2eb140007362')
 validpgpkeys=(
   'D7FCF61CF9A2DEAB31D81BD3F3D322D0EC4582C3'  # Carlos Garcia Campos <cgarcia@igalia.com>
   '5AA3BC334FD7E3369E7C77B291C559DBE4C9123B'  # Adrián Pérez de Castro <aperez@igalia.com>
@@ -74,6 +76,7 @@ validpgpkeys=(
 
 prepare() {
   cd wpewebkit-$pkgver
+  patch -Np1 -i ../fix-riscv64-build.patch
 }
 
 build() {
