--- PKGBUILD
+++ PKGBUILD
@@ -121,6 +121,8 @@ prepare() {
   # https://bugs.webkit.org/show_bug.cgi?id=278113
   # https://github.com/WebKit/WebKit/commit/9140ce712aa87091613874d802787ab476be0e39
   patch -Np1 -i ../revert-271175.patch
+
+  patch -Np1 -i ../fix-ftbfs-riscv64.patch
 }
 
 build() {
@@ -187,3 +189,7 @@ package_wpewebkit-docs() {
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(fix-ftbfs-riscv64.patch)
+sha256sums+=('b42d36c275e5fbd6da8e0761381d9eac13b2d103a366a61d9d450421948385a3')
+b2sums+=('cd80efecf12982e9e9335032a433e094614c8b95edce7114d37c6d651ee1f5bb3e331c3911d7901055cd59ca3ed5de38d14e15fab5488d3248bf5899623b2f8c')
