--- a/compiler/GHC/CmmToAsm/RV64/CodeGen.hs
+++ b/compiler/GHC/CmmToAsm/RV64/CodeGen.hs
@@ -849,13 +849,17 @@
     -- 1. Compute Reg +/- n directly.
     --    For Add/Sub we can directly encode 12bits, or 12bits lsl #12.
     CmmMachOp (MO_Add w) [CmmReg reg, CmmLit (CmmInt n _)]
-      | fitsIn12bitImm n -> return $ Any (intFormat w) (\d -> unitOL $ annExpr expr (ADD (OpReg w d) (OpReg w' r') (OpImm (ImmInteger n))))
+      | fitsIn12bitImm n -> return $ Any (intFormat w) $ \d ->
+          unitOL (annExpr expr (ADD (OpReg w d) (OpReg w' r') (OpImm (ImmInteger n))))
+            `appOL` truncateReg w w d
       where
         -- TODO: 12bits lsl #12; e.g. lower 12 bits of n are 0; shift n >> 12, and set lsl to #12.
         w' = formatToWidth (cmmTypeFormat (cmmRegType plat reg))
         r' = getRegisterReg plat reg
     CmmMachOp (MO_Sub w) [CmmReg reg, CmmLit (CmmInt n _)]
-      | fitsIn12bitImm n -> return $ Any (intFormat w) (\d -> unitOL $ annExpr expr (SUB (OpReg w d) (OpReg w' r') (OpImm (ImmInteger n))))
+      | fitsIn12bitImm n -> return $ Any (intFormat w) $ \d ->
+          unitOL (annExpr expr (SUB (OpReg w d) (OpReg w' r') (OpImm (ImmInteger n))))
+            `appOL` truncateReg w w d
       where
         -- TODO: 12bits lsl #12; e.g. lower 12 bits of n are 0; shift n >> 12, and set lsl to #12.
         w' = formatToWidth (cmmTypeFormat (cmmRegType plat reg))
