diff --git PKGBUILD PKGBUILD
index c7828b0..d763b43 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ url='https://www.haskell.org/ghc/'
 license=('custom')
 makedepends=('ghc-static' 'perl' 'libxslt' 'docbook-xsl' 'python-sphinx' 'haskell-hadrian'
              'haskell-hscolour' 'texlive-fontsrecommended' 'texlive-latexextra' 'texlive-xetex'
-             'ttf-dejavu' 'alex' 'happy' 'time' 'numactl' 'python-sphinx_rtd_theme')
+             'ttf-dejavu' 'alex' 'happy' 'time' 'numactl' 'python-sphinx_rtd_theme' 'llvm14')
 source=("https://downloads.haskell.org/~ghc/$pkgver/$pkgbase-${pkgver}-src.tar.xz"
         ghc-rebuild-doc-index.hook ghc-register.hook ghc-unregister.hook)
 sha512sums=('e5cfb30adc73dc0054f5db2921e5f255c8a980e005798882a2e9daa8df2409ddbe1ec6403e1f6862efc9e4700db4b68d5cae36999d77c7d3fd2fe0ab51cb9923'
@@ -75,7 +75,7 @@ package_ghc-static() {
 
 package_ghc() {
   pkgdesc='The Glasgow Haskell Compiler'
-  depends=('gcc' 'ghc-libs')
+  depends=('gcc' 'ghc-libs' 'llvm14')
   provides=('haskell-haddock=2.27.0'
             'haskell-hp2ps=0.1'
             'haskell-hpc-bin=0.68'
