diff --git PKGBUILD.orig PKGBUILD
index 7fac9d2..5b26d75 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -23,18 +23,30 @@ source=("https://downloads.haskell.org/~ghc/$pkgver/$pkgbase-${pkgver}-src.tar.x
         0001-Replace-more-autotools-obsolete-macros-19189.patch
         0002-configure-fix-the-use-of-some-obsolete-macros-19189.patch
         0003-llvmGen-Accept-range-of-LLVM-versions.patch
-        0004-Fix-autoconf-after-6d6edb1bbb.patch
+        0004-Fix-FIND_LLVM_PROG-arguments-after-6d6edb1bbb.patch
         0005-Set-min-LLVM-version-to-9-and-make-version-checking-.patch
-        0006-Optimiser-Correctly-deal-with-strings-starting-with-unicode.patch
+        0006-Optimiser-Correctly-deal-with-strings-starting-with-.patch
+        0007-Limit-upper-version-of-Happy-for-ghc-9.2-18620.patch
+        0008-Check-for-libatomic-dependency-for-atomic-operations.patch
+        0009-Implement-riscv64-LLVM-backend.patch
+        0010-Enable-tables-next-to-code-for-riscv64.patch
+        0011-Stop-leaking-defunct-llc-processes.patch
+        0012-Allow-bootstrapping-with-unregisterised-compiler.patch
         ghc-rebuild-doc-index.hook ghc-register.hook ghc-unregister.hook)
 noextract=("$pkgbase-${pkgver}-src.tar.xz")
 sha512sums=('bee7950a5118be8d8cefe0db5070139a5a93ca21c5bc6f8bf453429831f0c44f5e0fb5ee569865d6b8b92749044ee4123be06920928ac7a1ec9cffa9404a3e53'
-            '6390cb76b7786efb89acc854fdf68766c149412dddbf8c5f399aa90e204e704d6e476e6d850361d66e8e821e1a69dc55cb6b3b883dcd02e9ec5c6f5288ee280b'
-            'c1a4906f4a0a3abf5b2ed03a3d24c6ab097910f3fab5486c443dfc87ffbba399f95218663421ebf7e43d6d066ff0b364931d9bc3e5aa2162684f8b8511bc7531'
-            'cb90cf9c3af2704acb15e9c18b1b7d6b1de32bc4014c3d631e3d34bdf07ef91b03869b0ae3721ab73acb9f7dbd2e2805462e10c9c5431ed17464dd274121228a'
-            '775ef373ed60f752138d5ff728c2205b7d18ec72d8cf86152f6473a154003d0c9379d82310aeca2211123bfe48f6a5414af00d50a26680638b6db9718160199e'
-            'b3aeaf4c4f9e816350b97f1060cb2de3f9dcacd2c10c714cceb5e50d47ad94e845c2dae63eea22d806437c307ab6f561bfb385ec98087465c4172e22d755f402'
-            '8523a5fef22e391c668e315d115792c90de072e817f7a171e0c94e360684536e5c75eabe8a02ca73029431fb32462096bbdf6b8210dc862f72a79e28ec0ca27a'
+            '8ac40e409fd159ff30bfdd312c07a33d57630a30bfa28159f01c6e6024f1517ad4c426d7b565ff96b0bb99aab19a64461a58793e5f1000b15ca67b00280c1bae'
+            '720f740ac2339607319b5e253efb587c1a6520bc8eb87857206d00512f2a60b2b2f4f9c2681de476cfb1458354289fdd3a04b0d1ab4602cb87e8f2c1ae9ee555'
+            'c2a5744b998a472ca193cb84fd3db0c899e02a27722e5d27528bb61f7845d668ef9c298f084f4320eb74c304fc1685463cab51a28843b4612c35b0defabb8f7e'
+            'e424ad3cfec24cfc6aa1c20d44cda3ae069c6483e7f0b0fb27960a96d1923a42ff466a1f50c83b69bdd849208e02b1ab4f9ae013290f8544384e5b7083b92557'
+            '7a9ceed46da987bea55ea60c91fc82632d64115731ee61a65cf06037b50b75151f9b8954fddfcbf51bada31ea965950698901fdb80e06ccac789f76ddff6d358'
+            '739204d6806f69bf8e176a169fa448b036fc7ccd0d9fd42ce99373a174057fdc11d66983fd73b77c774a979e99d802ebc6d0cffd0dc1fdd9410ab6b367193441'
+            '7c04b24d8dce37a59abb05a87ca8a3e13340f0d4fb77aad6e54e4da83b631c652f52679f405ea8b50a7489ef17804c7224a3fac8077515beb562c46710939a91'
+            'bb99e27972d607df924682c315c8d29b90a57c08c020c7e276096e13bdb73bb01b3cf61b64c5f22dce4d73baba874759b1c01a0199fb7c4d16274fd9b307a4c2'
+            '7ba5e2d02302aacbfbce1b9f1395c3f4c18d2be5a508927629378351a67f2e2ff9406b4d2c85e55963636607ad8bc76759eab58b6d88a7b68feea8280556eceb'
+            '18983fd13faba1c7e4bebc930e3ffb0f88f17592f5f5d2cfbf5f016163bfa7b70a4f7f01d5a7a61118194e63717f2c5abea02b49136c5f3e5348cc71099725fc'
+            'ed7fb9a34f0b0b4428ef04fd0318bdc26aea85cb3d5b54eb2e1721a3de18390c1fcd566a94caf4fafe6a04ee77c2bf7618282e26e93b902cf54f517e38216601'
+            'd2c021dcd41402fe148e9b9e53e680d57478addaea11626032c94228a93f2214315d5d50ccecd3cd14bc7934f72ba90ed3810e6fc85505b45daf3cb9567910b5'
             'd69e5222d1169c4224a2b69a13e57fdd574cb1b5932b15f4bc6c7d269a9658dd87acb1be81f52fbcf3cb64f96978b9943d10cee2c21bff0565aaa93a5d35fcae'
             '5f659651d8e562a4dcaae0f821d272d6e9c648b645b1d6ab1af61e4dd690dc5a4b9c6846753b7f935963f001bb1ae1f40cd77731b71ef5a8dbc079a360aa3f8f'
             '3bdbd05c4a2c4fce4adf6802ff99b1088bdfad63da9ebfc470af9e271c3dd796f86fba1cf319d8f4078054d85c6d9e6a01f79994559f24cc77ee1a25724af2e6')
@@ -47,12 +59,18 @@ prepare() {
 
   cd ghc-$pkgver
 
-  patch -p1 -i ../0001-Replace-more-autotools-obsolete-macros-19189.patch
-  patch -p1 -i ../0002-configure-fix-the-use-of-some-obsolete-macros-19189.patch
-  patch -p1 -i ../0003-llvmGen-Accept-range-of-LLVM-versions.patch
-  patch -p1 -i ../0004-Fix-autoconf-after-6d6edb1bbb.patch
-  patch -p1 -i ../0005-Set-min-LLVM-version-to-9-and-make-version-checking-.patch
-  patch -p1 -i ../0006-Optimiser-Correctly-deal-with-strings-starting-with-unicode.patch
+  patch -Np1 -i ../0001-Replace-more-autotools-obsolete-macros-19189.patch
+  patch -Np1 -i ../0002-configure-fix-the-use-of-some-obsolete-macros-19189.patch
+  patch -Np1 -i ../0003-llvmGen-Accept-range-of-LLVM-versions.patch
+  patch -Np1 -i ../0004-Fix-FIND_LLVM_PROG-arguments-after-6d6edb1bbb.patch
+  patch -Np1 -i ../0005-Set-min-LLVM-version-to-9-and-make-version-checking-.patch
+  patch -Np1 -i ../0006-Optimiser-Correctly-deal-with-strings-starting-with-.patch
+  patch -Np1 -i ../0007-Limit-upper-version-of-Happy-for-ghc-9.2-18620.patch
+  patch -Np1 -i ../0008-Check-for-libatomic-dependency-for-atomic-operations.patch
+  patch -Np1 -i ../0009-Implement-riscv64-LLVM-backend.patch
+  patch -Np1 -i ../0010-Enable-tables-next-to-code-for-riscv64.patch
+  patch -Np1 -i ../0011-Stop-leaking-defunct-llc-processes.patch
+  patch -Np1 -i ../0012-Allow-bootstrapping-with-unregisterised-compiler.patch
 
   cp mk/build.mk{.sample,}
   sed -i '1iBuildFlavour = perf-llvm' mk/build.mk
