--- src/seccomp.rs.orig	2023-03-20 07:48:22.122003454 +0100
+++ src/seccomp.rs	2023-03-20 07:49:45.872647556 +0100
@@ -153,6 +153,7 @@
     allow_syscall!(ctx, libc::SYS_read);
     allow_syscall!(ctx, libc::SYS_readlinkat);
     allow_syscall!(ctx, libc::SYS_recvmsg);
+    #[cfg(not(target_arch = "riscv64"))]
     allow_syscall!(ctx, libc::SYS_renameat);
     allow_syscall!(ctx, libc::SYS_renameat2);
     allow_syscall!(ctx, libc::SYS_removexattr);
