--- PKGBUILD
+++ PKGBUILD
@@ -11,16 +11,21 @@ depends=(gcc-libs glibc libcap-ng libseccomp)
 makedepends=(cargo)
 provides=(virtiofsd-provider)
 conflicts=(virtiofsd-provider)
-source=($url/-/archive/v$pkgver/$pkgname-v$pkgver.tar.gz)
-sha512sums=('a50b3fac6d5eafc691ef4fc8808acb4dc03261f01623f6675a234d457231f96181e9cf6c8ac68b4fe1b1c97257faf56bc6d0c2776142c9dbfe82b4e9bde0e34a')
-b2sums=('52c86ab9d4842ecd496aed3f09a98be96d7ee0eda43a2519eb066be1f648963b155386f9eac1a1c878c12fd05d928f88700c6a7a840c0f34dfa6f86caab15fec')
+source=($url/-/archive/v$pkgver/$pkgname-v$pkgver.tar.gz
+        remove-unsupport-syscall.patch::https://gitlab.com/virtio-fs/virtiofsd/-/merge_requests/156.patch)
+sha512sums=('a50b3fac6d5eafc691ef4fc8808acb4dc03261f01623f6675a234d457231f96181e9cf6c8ac68b4fe1b1c97257faf56bc6d0c2776142c9dbfe82b4e9bde0e34a'
+            '0a1bcb4091a5738f72c53068d8268f38059e380db9daa40669399869088330787fb957e4233d4b34e05f229a88681075b8647978de671b42a133c4a7351f2e49')
+b2sums=('52c86ab9d4842ecd496aed3f09a98be96d7ee0eda43a2519eb066be1f648963b155386f9eac1a1c878c12fd05d928f88700c6a7a840c0f34dfa6f86caab15fec'
+        '0b479d9ecc7d793254f68dfe1e636297f0804a19a4279597dc9cf6be8d35d22462276619153df9f3507d5df20b69f0363569a08100f4dae73207f2fa2be039cb')
 
 prepare() {
   cd $pkgname-v$pkgver
   # use /usr/lib instead of /usr/libexec: https://gitlab.com/virtio-fs/virtiofsd/-/issues/86
   sed 's/libexec/lib/' -i 50-qemu-$pkgname.json
 
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  # remove riscv64 unsupport syscall
+  patch -p0 -Ni ${srcdir}/remove-unsupport-syscall.patch
+  cargo fetch --locked
 }
 
 build() {
