diff --git PKGBUILD PKGBUILD
index 6ebad7e..e098b3c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,9 +16,11 @@ license=('GPL3')
 depends=('boost-libs' 'wxgtk3' 'hunspell')
 makedepends=('boost' 'imagemagick' 'zip')
 source=("https://downloads.sourceforge.net/codeblocks/$pkgname-$pkgver.tar.xz"
-         codeblocks-gcc11.patch)
+         codeblocks-gcc11.patch
+         add_riscv_support.patch)
 sha256sums=('15eeb3e28aea054e1f38b0c7f4671b4d4d1116fd05f63c07aa95a91db89eaac5'
-            '10b2b08440adc447d1d23eb122fa37beff1f26b9d7588b83f8435721586f4ed1')
+            '10b2b08440adc447d1d23eb122fa37beff1f26b9d7588b83f8435721586f4ed1'
+            'bbff9f6c6d23c0fab1a02454f66580aaaa0b1fb9c9a47386c85c56f8f307f5e4')
 
 prepare() {
   cd $pkgname-$pkgver
@@ -27,11 +29,12 @@ prepare() {
   sed -i 's|$(datarootdir)/appdata|$(datarootdir)/metainfo|' Makefile.{am,in} src/plugins/contrib/appdata/Makefile.{am,in}
 
   patch -p1 < ../codeblocks-gcc11.patch # Fix build with GCC 11 (Fedora)
+  patch -p1 < ../add_riscv_support.patch
 }
 
 build() {
   cd $pkgname-$pkgver
-  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-wx-config=/usr/bin/wx-config-gtk3
+  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-wx-config=/usr/bin/wx-config-gtk3 --with-boost-libdir=/usr/lib
   sed -i 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
 }
