--- PKGBUILD
+++ PKGBUILD
@@ -30,7 +30,8 @@ source=("https://downloads.sourceforge.net/codeblocks/$pkgname-$pkgver.tar.xz"
          https://github.com/arnholm/codeblocks_sfmirror/commit/b2e4f1279804e1d11b71bc75eeb37072c3589296.patch
          https://github.com/arnholm/codeblocks_sfmirror/commit/29315df024251850832583f73e67e515dae10830.patch
          https://github.com/arnholm/codeblocks_sfmirror/commit/46720043319758cb0e798eb23520063583c40eaa.patch 
-         sc_wxtypes-normalize.patch)
+         sc_wxtypes-normalize.patch
+         add_riscv_support.patch)
 sha256sums=('15eeb3e28aea054e1f38b0c7f4671b4d4d1116fd05f63c07aa95a91db89eaac5'
             '10b2b08440adc447d1d23eb122fa37beff1f26b9d7588b83f8435721586f4ed1'
             '0f9a3ae205afd358eeb643d5553f57f43cb0494c54ebd0a03f2c07d08df7713a'
@@ -46,7 +47,8 @@ sha256sums=('15eeb3e28aea054e1f38b0c7f4671b4d4d1116fd05f63c07aa95a91db89eaac5'
             'c9723cac8ca690694b9db867b9b65736a1596958e153efd7672bf89ce832a6bd'
             '8a89646c875e2c4f39019a58c378531b3bbe5e0a21deb18f679a799da0e2321b'
             '5b80ef86fd6de408bfa77914cd5c6fc72c0f096bd40da0e02b376636cc18684d'
-            'a4d39d3016fe128e521c38a23b7b25ce94ea18263498a60d6224729dee528564')
+            'a4d39d3016fe128e521c38a23b7b25ce94ea18263498a60d6224729dee528564'
+            'bbff9f6c6d23c0fab1a02454f66580aaaa0b1fb9c9a47386c85c56f8f307f5e4')
 
 prepare() {
   cd $pkgname-$pkgver
@@ -69,11 +71,12 @@ prepare() {
   patch -p1 < ../29315df024251850832583f73e67e515dae10830.patch
   patch -p1 < ../46720043319758cb0e798eb23520063583c40eaa.patch
   patch -p1 < ../sc_wxtypes-normalize.patch
+  patch -p1 < ../add_riscv_support.patch
 }
 
 build() {
   cd $pkgname-$pkgver
-  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager
+  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-boost-libdir=/usr/lib
   sed -i 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
 }
