diff --git PKGBUILD PKGBUILD
index a214b16a8..8fa546716 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,17 @@ url='http://codeblocks.org/'
 license=('GPL3')
 depends=('boost-libs' 'wxgtk3' 'hunspell')
 makedepends=('boost' 'imagemagick' 'zip')
-source=("https://downloads.sourceforge.net/codeblocks/$pkgname-$pkgver.tar.xz")
-sha256sums=('15eeb3e28aea054e1f38b0c7f4671b4d4d1116fd05f63c07aa95a91db89eaac5')
+source=("https://downloads.sourceforge.net/codeblocks/$pkgname-$pkgver.tar.xz"
+        "add_riscv_support.patch"
+        "fix_nonconst_comparator.patch")
+sha256sums=('15eeb3e28aea054e1f38b0c7f4671b4d4d1116fd05f63c07aa95a91db89eaac5'
+            'bbff9f6c6d23c0fab1a02454f66580aaaa0b1fb9c9a47386c85c56f8f307f5e4'
+            '10b2b08440adc447d1d23eb122fa37beff1f26b9d7588b83f8435721586f4ed1')
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -Np1 -i "../add_riscv_support.patch"
+  patch -Np1 -i "../fix_nonconst_comparator.patch"
   convert src/mime/codeblocks.png +set date:create +set date:modify -background none -extent 64x64 src/mime/codeblocks.png
   sed -i 's|$(datadir)/pixmaps|$(datadir)/icons/hicolor/64x64/apps|' src/mime/Makefile.{am,in}
   sed -i 's|$(datarootdir)/appdata|$(datarootdir)/metainfo|' Makefile.{am,in} src/plugins/contrib/appdata/Makefile.{am,in}
@@ -27,7 +33,7 @@ prepare() {
 
 build() {
   cd $pkgname-$pkgver
-  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-wx-config=/usr/bin/wx-config-gtk3
+  ./configure --prefix=/usr --with-contrib-plugins=all,-FileManager --with-wx-config=/usr/bin/wx-config-gtk3 --with-boost-libdir=/usr/lib
   sed -i 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
 }
