--- PKGBUILD
+++ PKGBUILD
@@ -34,14 +34,41 @@ prepare() {
   # Specify electron version in launcher
   sed -i "s|@ELECTRON@|${_electron}|" element-desktop.sh
 
+  export ELECTRON_SKIP_BINARY_DOWNLOAD=1
+  export SENTRYCLI_USE_LOCAL=1 SENTRYCLI_SKIP_DOWNLOAD=1
+  export PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1
+
   cd element-web-${pkgver}
   yarn install --no-fund
 
   cd ../element-desktop-${pkgver}
   patch -p1 < ../autolaunch.patch
-  sed -i 's|"target": "deb"|"target": "dir"|' package.json
+  patch -Np0 -i ../element-desktop-riscv64-support.patch
+  jq '.build.linux.target=["dir"]
+    | .resolutions."electron-builder"="npm:@riscv-forks/electron-builder"
+    | .resolutions."app-builder-lib"="npm:@riscv-forks/app-builder-lib"
+    | .resolutions."builder-util"="npm:@riscv-forks/builder-util"' package.json > package.json.new
+  mv package.json.new package.json
   sed -i 's|"https://packages.element.io/desktop/update/"|null|' element.io/release/config.json
   yarn install --no-fund
+
+  yarn run hak fetch matrix-seshat
+  pushd .hak/matrix-seshat/riscv64gc-unknown-linux-gnu/build/
+  echo -e "[patch.crates-io]\nnix = { git = 'https://github.com/kxxt/nix.git', branch = 'archrv-element-fix' }" >> Cargo.toml
+  cargo update -p nix
+  popd
+
+  mkdir -p  node_modules/7zip-bin/linux/riscv64
+  ln -s /usr/bin/7za node_modules/7zip-bin/linux/riscv64/7za
+
+  local _builder_bin=node_modules/app-builder-bin/linux/riscv64
+  mkdir "$_builder_bin"
+  go build -C ../app-builder
+  cp ../app-builder/app-builder "$_builder_bin"
+
+  local _electron_ver=$(jq -r .version node_modules/electron/package.json)
+  local _electron_zip="electron-v$_electron_ver-linux-riscv64.zip"
+  cd "/usr/lib/$_electron" && zip -r "/tmp/$_electron_zip" ./
 }
 
 build() {
@@ -53,7 +80,7 @@ build() {
   export SQLCIPHER_BUNDLED=1
   export CFLAGS+=" -ffat-lto-objects"
   yarn run build:native
-  yarn run build
+  ELECTRON_CACHE=/tmp yarn run build
 }
 
 package_element-web() {
@@ -81,7 +108,7 @@ package_element-desktop() {
   install -d "${pkgdir}"{/usr/lib/element/,/etc/webapps/element}
 
   # Install the app content, replace the webapp with a symlink to the system package
-  cp -r dist/linux-unpacked/resources/* "${pkgdir}"/usr/lib/element/
+  cp -r dist/linux-riscv64-unpacked/resources/* "${pkgdir}"/usr/lib/element/
   ln -s /usr/share/webapps/element "${pkgdir}"/usr/lib/element/webapp
 
   # Config file
@@ -98,3 +125,10 @@ package_element-desktop() {
     install -Dm644 build/icons/${i}x${i}.png "${pkgdir}"/usr/share/icons/hicolor/${i}x${i}/apps/io.element.Element.png
   done
 }
+
+makedepends+=(sentry-cli go jq zip p7zip)
+source+=(element-desktop-riscv64-support.patch
+         git+https://github.com/develar/app-builder.git#commit=c92c3a2899b5887662321878a0a8681d122742bb)
+sha256sums+=('362020117ec482dc165b276046705468106316a769e39bfa408586ea72fefe45'
+             'SKIP')
+
