diff --git PKGBUILD trunk/PKGBUILD
index 2b929d3..18e15d0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,11 @@ sha256sums=('4655498a1b8e844e3d6f21f3b2c4e2b571effb5fd83199d428a6ba7ea4bf5204'
             'SKIP')
 validpgpkeys=('C41C985FDCF1E5364576638B687393EE37D128F8') # Matthieu Herrb <matthieu.herrb@laas.fr>
 
+prepare() {
+  cd "${srcdir}/libXtst-${pkgver}"
+  autoreconf -fiv
+}
+
 build() {
   cd "${srcdir}/libXtst-${pkgver}"
   ./configure --prefix=/usr --disable-static
