diff --git PKGBUILD PKGBUILD
index 9a588aa94..6c66fdb28 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,17 @@ arch=('x86_64')
 url="https://github.com/rogerbinns/apsw"
 license=('MIT')
 depends=('python' 'sqlite')
-source=("apsw-$pkgver-r1.tar.gz::https://github.com/rogerbinns/apsw/archive/$pkgver-r1.tar.gz")
-sha512sums=('22848effb6d4d69e486750b1706316551b593c055b40b32e897161d6d78642a36ccf72cf86ff5e7e1a2520b33fcdcf50116338aaa689630b00e1b6b3362744c0')
-b2sums=('fc942e9402488352b344e8e3a749194f84657d929b6e0258fe065aa3a9698e7e9744f61c328e0873f7c6bc1233cfbca0b019c4c801121c2d35d94c06f4766e37')
+source=("apsw-$pkgver-r1.tar.gz::https://github.com/rogerbinns/apsw/archive/$pkgver-r1.tar.gz"
+        "fix_type_case.patch::https://github.com/rogerbinns/apsw/commit/9a1891bd56ba2188316e221e99e4f3644c2dc779.patch")
+sha512sums=('22848effb6d4d69e486750b1706316551b593c055b40b32e897161d6d78642a36ccf72cf86ff5e7e1a2520b33fcdcf50116338aaa689630b00e1b6b3362744c0'
+            '6232e3a48932ceb1e6be501a06eadf2bfb876bc1c0a269def7f2ffbc28b43bdd6b1d26bc70c4364d5dc56922d627e4e330c37010624e9c6b378a2482a376a3bb')
+b2sums=('fc942e9402488352b344e8e3a749194f84657d929b6e0258fe065aa3a9698e7e9744f61c328e0873f7c6bc1233cfbca0b019c4c801121c2d35d94c06f4766e37'
+        'afbcc521a4ccde0e31dc3f95a5bb572c04abfaf2a4b251ed4e3205fd9018ffdb757cbf5ce19c6a94a8d9882f262b964ebee6c021039d1cbd3f86daa57a844efb')
+
+prepare() {
+  cd apsw-${pkgver}-r1
+  patch -Np1 -i "../fix_type_case.patch"
+}
 
 build() {
   cd apsw-${pkgver}-r1
