diff --git PKGBUILD PKGBUILD
index f80ec8c..1782691 100644
--- PKGBUILD
+++ PKGBUILD
@@ -47,6 +47,8 @@ build() {
     export CXX='clang++'
     export CFLAGS+=' -DNDEBUG'
     export CXXFLAGS+=' -DNDEBUG'
+    export CFLAGS+=' -Wno-error=unused-command-line-argument'
+    export CXXFLAGS+=' -Wno-error=unused-command-line-argument'
     cmake -B build -S libjxl \
         -DCMAKE_BUILD_TYPE:STRING='None' \
         -DCMAKE_INSTALL_PREFIX:PATH='/usr' \
