--- PKGBUILD	(revision 1403198)
+++ PKGBUILD	(working copy)
@@ -21,7 +21,8 @@
         'git+https://github.com/google/highway.git'
         'git+https://github.com/glennrp/libpng.git'
         'git+https://github.com/madler/zlib.git'
-        'libjxl-testdata'::'git+https://github.com/libjxl/testdata.git')
+        'libjxl-testdata'::'git+https://github.com/libjxl/testdata.git'
+        'fix-atomic.patch'::'https://github.com/libjxl/libjxl/pull/2211.patch')
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
@@ -31,7 +32,8 @@
             'SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            'eb686a1c6cc7ad36adb191631bb8e28b788a5ba28c47e98d302fae50fa35fcf2')
 
 prepare() {
     git -C libjxl submodule init
@@ -43,6 +45,8 @@
     git -C libjxl config --local submodule.third_party/lcms.url "${srcdir}/Little-CMS"
     git -C libjxl config --local submodule.third_party/testdata.url "${srcdir}/libjxl-testdata"
     git -C libjxl -c protocol.file.allow='always' submodule update
+
+    patch -Np1 -d libjxl < fix-atomic.patch
 }
 
 build() {
