--- PKGBUILD
+++ PKGBUILD
@@ -48,6 +48,8 @@
 prepare() {
   cd snapshot
 
+  patch -Np1 -i "$srcdir/fix-clippy-timeout.patch"
+
   # Match CARGO_HOME in src/meson.build
   CARGO_HOME="$srcdir/build/cargo-home" \
     cargo fetch --locked --target "$(rustc --print host-tuple)"
@@ -65,3 +67,6 @@
 package() {
   meson install -C build --destdir "$pkgdir" --no-rebuild
 }
+
+source+=(fix-clippy-timeout.patch)
+b2sums+=('e0b8007025f5a766e7a051fdb0a8d0cccb3e0690f340e5e7fa1061c655f1d475a8813694cccfabf7abdeebe7c080241115a2a1b22dc50b8caeb24289cc80e064')
