diff --git PKGBUILD PKGBUILD
index ddee02d..1b4f8c3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -39,11 +39,14 @@ checkdepends=(
 source=(
   https://github.com/DISTRHO/$pkgbase/releases/download/$pkgver/$pkgbase-$pkgver.tar.xz
   $pkgbase-$pkgver.svg::https://raw.githubusercontent.com/DISTRHO/Cardinal/eb95b5990cf734c10f2caab1a246eac893f0266d/plugins/Cardinal/orig/distrho.svg
+  $pkgname-gcc13.patch::https://github.com/surge-synthesizer/surge/commit/d2fba1c6ff02cbdfa9db2ce06508745c247629b5.patch
 )
 sha512sums=('60d6dcb2f3c90e41739c645f75408dc1603c5c999819577be26dbc7d05e2fbbce8dbf3d8d20d7d9e3075459f0fadb6f32c17d98c2186cf0ebed767f016410aec'
-            '31a7d1e548285af0ead1bc844fbb1d35b50f6284159047cb401a829d4068992fa328770e42c377c08d013f6787e55ac12b94eba4d5af4b90373b157971fc8835')
+            '31a7d1e548285af0ead1bc844fbb1d35b50f6284159047cb401a829d4068992fa328770e42c377c08d013f6787e55ac12b94eba4d5af4b90373b157971fc8835'
+            'f38872eee7264fff2f598a4aa100877d979c80c1c7796c4533fee7075aeffed2dc6758240413d07fcea32438749f5c676fc0083c3fa08be7da3abb1de3914433')
 b2sums=('c28b1aff34c3e1b5fff01cf74c2df38e4ed58e7c56ccb5e106ef8582675e2914e4973bca874053371521150a838007d4eaa0e8ea58899337c22a2f5b22dfd5cd'
-        'd82fefe15234c1ed3a4d487c6082a2e3ac73b60f11041314e99c5a5de5b3ea141efbd5afd18851510bd4f801fd71f7cd89d54d86258d1ba750bbfd762aa37a8d')
+        'd82fefe15234c1ed3a4d487c6082a2e3ac73b60f11041314e99c5a5de5b3ea141efbd5afd18851510bd4f801fd71f7cd89d54d86258d1ba750bbfd762aa37a8d'
+        'de2e6b293025275b8141c4b0aff37c3d4c170f67229f8abcd58e8bf2ed261ba8e74e29a310b8227042334f1855e0d4d13b9777b8d823be628fce5508b718a561')
 
 _common_depends=(
   cardinal-data
@@ -86,6 +89,8 @@ prepare() {
           --pkgdesc "$pkgdesc JACK standalone" \
           --icon $pkgbase \
           --genericname "Virtual modular synthesizer"
+  cd $pkgbase-$pkgver/plugins/surgext/surge
+  patch -p1 -i "$srcdir/$pkgname-gcc13.patch"
 }
 
 build() {
@@ -180,3 +185,5 @@ package_cardinal-vst3() {
 
   mv -v $pkgbase-vst3/* "$pkgdir"
 }
+
+makedepends+=(simde)
