diff --git a/AI/Skirmish/CircuitAI/src/circuit/util/MaskHandler.h b/AI/Skirmish/CircuitAI/src/circuit/util/MaskHandler.h
index 7bde5c3..78a3215 100644
--- a/AI/Skirmish/CircuitAI/src/circuit/util/MaskHandler.h
+++ b/AI/Skirmish/CircuitAI/src/circuit/util/MaskHandler.h
@@ -11,6 +11,7 @@
 #include <string>
 #include <vector>
 #include <unordered_map>
+#include <cstdint>
 
 namespace circuit {
 
diff --git a/rts/Sim/Misc/BuildingMaskMap.h b/rts/Sim/Misc/BuildingMaskMap.h
index 89351a3..c2608da 100644
--- a/rts/Sim/Misc/BuildingMaskMap.h
+++ b/rts/Sim/Misc/BuildingMaskMap.h
@@ -4,6 +4,7 @@
 #define BUILDINGMASKMAP_H
 
 #include <vector>
+#include <cstdint>
 
 #include "System/creg/creg_cond.h"
 
diff --git a/rts/System/CRC.h b/rts/System/CRC.h
index 1abde22..ee25b24 100644
--- a/rts/System/CRC.h
+++ b/rts/System/CRC.h
@@ -3,6 +3,7 @@
 #ifndef CRC_H
 #define CRC_H
 
+#include <cstdint>
 #include <string>
 
 /** @brief An updateable CRC-32 checksum. */
diff --git a/rts/System/Platform/Misc.h b/rts/System/Platform/Misc.h
index 37bb44f..e71e3d2 100644
--- a/rts/System/Platform/Misc.h
+++ b/rts/System/Platform/Misc.h
@@ -3,6 +3,7 @@
 #ifndef PLATFORM_MISC_H
 #define PLATFORM_MISC_H
 
+#include <cstdint>
 #include <string>
 #include <array>
 
diff --git a/rts/System/SpringHashSet.hpp b/rts/System/SpringHashSet.hpp
index 071a6c3..75d8174 100644
--- a/rts/System/SpringHashSet.hpp
+++ b/rts/System/SpringHashSet.hpp
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <cstdlib> // malloc
 #include <iterator>
 #include <utility>
diff --git a/rts/System/StringUtil.h b/rts/System/StringUtil.h
index e9ba35b..2298f4c 100644
--- a/rts/System/StringUtil.h
+++ b/rts/System/StringUtil.h
@@ -4,6 +4,7 @@
 #define STRING_UTIL_H
 
 #include <algorithm>
+#include <cstdint>
 #include <cstring>
 #include <string>
 #include <sstream>
diff --git a/rts/lib/luasocket/src/restrictions.h b/rts/lib/luasocket/src/restrictions.h
index a6bb1ba..83e0115 100644
--- a/rts/lib/luasocket/src/restrictions.h
+++ b/rts/lib/luasocket/src/restrictions.h
@@ -1,5 +1,6 @@
 /* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
 
+#include <cstdint>
 #include <list>
 #include <string>
 #include <utility>
