diff --git a/AI/Interfaces/Java/CMakeLists.txt b/AI/Interfaces/Java/CMakeLists.txt
index 7738190..decaae5 100644
--- a/AI/Interfaces/Java/CMakeLists.txt
+++ b/AI/Interfaces/Java/CMakeLists.txt
@@ -539,8 +539,10 @@ if    (BUILD_${myName}_AIINTERFACE)
 	include_directories(BEFORE "${rts}/lib/streflop" "${myNativeSourceDir}" "${myNativeGeneratedSourceDir}")
 	add_library(${myNativeTarget} MODULE ${myNativeSources} ${myNativeGeneratedSources} ${ai_common_SRC} ${myVersionDepFile})
 	add_dependencies(${myNativeTarget} generateVersionFiles)
-	target_link_libraries(${myNativeTarget} CUtils streflop)
-	set_target_properties(${myNativeTarget} PROPERTIES COMPILE_FLAGS "-DUSING_STREFLOP")
+	if (ENABLE_STREFLOP)
+		target_link_libraries(${myNativeTarget} CUtils streflop)
+		set_target_properties(${myNativeTarget} PROPERTIES COMPILE_FLAGS "-DUSING_STREFLOP")
+	endif()
 	set_target_properties(${myNativeTarget} PROPERTIES OUTPUT_NAME   "AIInterface")
 	fix_lib_name(${myNativeTarget})
 
