diff --git a/cmake/geo_detect_platform.cmake b/cmake/geo_detect_platform.cmake
index c5bb6be09..853464e1e 100644
--- a/cmake/geo_detect_platform.cmake
+++ b/cmake/geo_detect_platform.cmake
@@ -7,10 +7,18 @@
 # - Use CheckCXXCompilerFlag to enable compiler-specific flags
 #
 if(CMAKE_SYSTEM_NAME MATCHES Linux)
-    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-        set(VORPALINE_PLATFORM "Linux64-gcc-dynamic" CACHE STRING "")
-    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-        set(VORPALINE_PLATFORM "Linux64-clang-dynamic" CACHE STRING "")
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES x86_64)
+        if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+            set(VORPALINE_PLATFORM "Linux64-gcc-dynamic" CACHE STRING "")
+        elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+            set(VORPALINE_PLATFORM "Linux64-clang-dynamic" CACHE STRING "")
+        endif()
+    else()
+        if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+            set(VORPALINE_PLATFORM "Linux64-nonx86-gcc-dynamic" CACHE STRING "")
+        elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+            set(VORPALINE_PLATFORM "Linux64-nonx86-clang-dynamic" CACHE STRING "")
+        endif()
     endif()
 elseif(CMAKE_SYSTEM_NAME MATCHES Darwin)
     set(VORPALINE_PLATFORM "Darwin-clang-dynamic" CACHE STRING "")
diff --git a/cmake/platforms/Linux64-nonx86-clang-dynamic/config.cmake b/cmake/platforms/Linux64-nonx86-clang-dynamic/config.cmake
new file mode 100644
index 000000000..a053bb244
--- /dev/null
+++ b/cmake/platforms/Linux64-nonx86-clang-dynamic/config.cmake
@@ -0,0 +1,6 @@
+set(VORPALINE_ARCH_64 true)
+set(VORPALINE_BUILD_DYNAMIC true)
+include(${GEOGRAM_SOURCE_DIR}/cmake/platforms/Linux-clang.cmake)
+add_flags(CMAKE_CXX_FLAGS -m64)
+add_flags(CMAKE_C_FLAGS -m64)
+
diff --git a/cmake/platforms/Linux64-nonx86-clang-dynamic/setvars.sh b/cmake/platforms/Linux64-nonx86-clang-dynamic/setvars.sh
new file mode 100644
index 000000000..030414bce
--- /dev/null
+++ b/cmake/platforms/Linux64-nonx86-clang-dynamic/setvars.sh
@@ -0,0 +1,2 @@
+export CC=clang
+export CXX=clang++ 
diff --git a/cmake/platforms/Linux64-nonx86-gcc-dynamic/config.cmake b/cmake/platforms/Linux64-nonx86-gcc-dynamic/config.cmake
new file mode 100644
index 000000000..32a6fa296
--- /dev/null
+++ b/cmake/platforms/Linux64-nonx86-gcc-dynamic/config.cmake
@@ -0,0 +1,3 @@
+set(VORPALINE_ARCH_64 true)
+set(VORPALINE_BUILD_DYNAMIC true)
+include(${GEOGRAM_SOURCE_DIR}/cmake/platforms/Linux-gcc.cmake)
diff --git a/cmake/platforms/Linux64-nonx86-gcc-dynamic/setvars.sh b/cmake/platforms/Linux64-nonx86-gcc-dynamic/setvars.sh
new file mode 100644
index 000000000..e69de29bb
diff --git a/configure.sh b/configure.sh
index c1ef489d8..14acb24f6 100755
--- a/configure.sh
+++ b/configure.sh
@@ -128,6 +128,9 @@ if [ -z "$os" ]; then
         Linux*aarch64*)
             os=Linux64-gcc-aarch64
             ;;
+        Linux*riscv64*)
+            os=Linux64-nonx86-gcc-riscv64
+            ;;
         Darwin*)
             os=Darwin-clang-dynamic
             ;;
diff --git a/src/lib/geogram/basic/common.h b/src/lib/geogram/basic/common.h
index bf4e9ed7f..a43780a29 100644
--- a/src/lib/geogram/basic/common.h
+++ b/src/lib/geogram/basic/common.h
@@ -258,7 +258,7 @@ namespace GEO {
 #  error "Unsupported compiler"
 #endif
 
-#if defined(__x86_64) || defined(__ppc64__) || defined(__arm64__) || defined(__aarch64__)
+#if defined(__x86_64) || defined(__ppc64__) || defined(__arm64__) || defined(__aarch64__) || (defined(__riscv) && __riscv_xlen == 64)
 #  define GEO_ARCH_64
 #else
 #  define GEO_ARCH_32
