--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,21 @@ url="https://github.com/BrunoLevy/geogram"
 license=('BSD')
 depends=('glu' 'glfw-x11')
 makedepends=('cmake' 'ninja')
-source=("https://github.com/BrunoLevy/geogram/releases/download/v${pkgver}/geogram_${pkgver}.tar.gz")
-sha256sums=('698bc9ad9d58139fe9fdf3eab0596f5b418c4edd593eee960de98c0ab646d47e')
+source=("https://github.com/BrunoLevy/geogram/releases/download/v${pkgver}/geogram_${pkgver}.tar.gz"
+        geogram-riscv64.patch
+        impl-riscv-spinlocks.patch
+        libf2c-riscv-fpu-control.patch)
+sha256sums=('698bc9ad9d58139fe9fdf3eab0596f5b418c4edd593eee960de98c0ab646d47e'
+            '89b64fd88eba1f292d6ca50a11465baa078b983c9ed4437b239e37036bda3201'
+            'a4193f52ec3cd97e3a9e79dba1cbe41af18143477596ad50c52c97e6bcce15f3'
+            '1cf90f1e3d67fbdc47135dfdcabd8b79d1d4935f665467397e2decc3a25426c8')
+
+prepare() {
+  cd ${pkgname}_${pkgver}
+  patch -Np1 -i ../geogram-riscv64.patch
+  patch -Np1 -i ../libf2c-riscv-fpu-control.patch
+  patch -Np1 -i ../impl-riscv-spinlocks.patch
+}
 
 build() {
   cd ${pkgname}_${pkgver}
