--- PKGBUILD
+++ PKGBUILD
@@ -17,22 +17,25 @@ pkgrel=1
 arch=('x86_64')
 url='https://openjdk.java.net/'
 license=('LicenseRef-Java')
-makedepends=('java-environment=21' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
+makedepends=('java-environment=22' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
              'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib' 'pandoc'
              'graphviz' 'freetype2' 'libjpeg-turbo' 'giflib' 'libpng' 'lcms2'
              'libnet' 'bash' 'harfbuzz' 'gcc-libs' 'glibc')
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
-        freedesktop-jshell.desktop)
+        freedesktop-jshell.desktop
+        bootstrap-using-jdk22.patch)
 sha256sums=('b7a78e596b272d958843eab0c0412fd7ee874a3b6fff577584ebeed39dfef7ee'
             '72111743ab6ab36854b0c85a504172983715d0798fce10bc4e35689b7d15fd93'
             '8ecdf5c1605bafa58b3f7da615e6d8d3d943e3a2d3831930d6efa7815aacce07'
-            '50fc0d677489b73d549df2f08d759d5f057f200adbbab83ea5e87456152ee03e')
+            '50fc0d677489b73d549df2f08d759d5f057f200adbbab83ea5e87456152ee03e'
+            '32c21d9a6f125c3756b6253392849e015e0bee6193b1af104a8c33003fb88e7c')
 
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -48,6 +51,11 @@ _commondeps=('java-runtime-common>=3' 'ca-certificates-utils' 'nss' 'libjpeg-tur
            'lcms2' 'liblcms2.so' 'libnet' 'freetype2' 'libfreetype.so' 'harfbuzz' 'libharfbuzz.so'
            'glibc' 'gcc-libs')
 
+prepare() {
+  cd ${_jdkdir}
+  patch -Np1 -i ../bootstrap-using-jdk22.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -94,6 +102,8 @@ build() {
     jvm_features="zgc,shenandoahgc"
   fi
 
+  export JAVA_WARNINGS_ARE_ERRORS=' '
+
 
   bash configure \
     --with-version-build="${_updatever}" \
