diff --git a/src/main.c b/src/main.c
index e271ebe..bfec6f5 100644
--- a/src/main.c
+++ b/src/main.c
@@ -97,7 +97,7 @@ void error_dialog(char *message)
 				      GTK_DIALOG_MODAL|GTK_DIALOG_DESTROY_WITH_PARENT, /* GtkDialogFlags */
                                       GTK_MESSAGE_ERROR,
                                       GTK_BUTTONS_OK,
-				      message);
+				      "%s", message);
   g_assert(d);
   /*gint result =*/ gtk_dialog_run(GTK_DIALOG(d));
 #if 0
@@ -120,7 +120,7 @@ void info_dialog(char *message)
 				      GTK_DIALOG_MODAL|GTK_DIALOG_DESTROY_WITH_PARENT, /* GtkDialogFlags */
                                       GTK_MESSAGE_INFO,
                                       GTK_BUTTONS_OK,
-				      message);
+				      "%s", message);
   g_assert(d);
   gtk_dialog_run(GTK_DIALOG(d));
   gtk_widget_destroy(d);
