diff --git PKGBUILD PKGBUILD
index 87340bd..6959b08 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,13 +10,16 @@ arch=('x86_64')
 license=('GPL2')
 depends=('gtk3' 'expat')
 source=(http://www.rfdump.org/dl/${pkgname}-${pkgver}.tar.bz2
-        drop-libglade.patch)
+        drop-libglade.patch
+        "fix_not_literal_string.patch")
 sha512sums=('3402b9147b3e255b3c957d92f1fa4d7610fb6fde0aef2e3205343f64b252635bcf6345fb904b18ded063df775f15379a726b7b3b166b13d94efc9712373c7df7'
-            'SKIP')
+            'SKIP'
+            'baa4bb5d1a5e9852d450906336b9e86168dac95114f5ad98bd78feea1d9effe6562c769aeb8d345a27a55afce040666fef40f0a27801dda099da9816a79b82ac')
 
 prepare() {
   cd ${pkgname}-${pkgver}
   patch -Np1 -i ../drop-libglade.patch
+  patch -Np1 -i "../fix_not_literal_string.patch"
   autoreconf -fiv
 }
 
