--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,9 @@ sha512sums=('ab8a3b4ae855d5d66d384472ce3ecb2b10a479e1bab1e59327d7e4405636ef4be32
 
 prepare() {
   cd "$pkgname-$pkgver"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
   mkdir -p {completions,man}
 }
 
