Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1356136)
+++ PKGBUILD	(working copy)
@@ -14,6 +14,11 @@
 source=("https://hackage.haskell.org/packages/archive/${_hkgname}/${pkgver}/${_hkgname}-${pkgver}.tar.gz")
 sha512sums=('ba0a76b6649b0594c0359747ffa80ecb09297f8751f9eed59a17088780e989008d0a5c5771a9f4de0398232c7cdf83786f97a12c5eabaa596d0e00dd037506e1')
 
+prepare() {
+    cd $_hkgname-$pkgver
+    sed -i '/test-suite doctests/a \    buildable: False' $_hkgname.cabal
+}
+
 build() {
     cd $_hkgname-$pkgver
     
