--- PKGBUILD
+++ PKGBUILD
@@ -23,7 +23,6 @@ depends=(
   'abseil-cpp'
 )
 makedepends=(
-  'bazelisk'
   'cmake'
   'git'
   'gtest'
@@ -45,10 +44,13 @@ source=(
   protobuf-fix-build-type-none.patch
   # https://github.com/protocolbuffers/protobuf/issues/11935
   $pkgbase-25.3-ruby-disable-LTO.patch
+  # Keep it here instead of bottom of the file because this file updates along with the package
+  "$pkgbase-$pkgver-python.tar.gz::https://files.pythonhosted.org/packages/source/${pkgbase::1}/${pkgbase//-/_}/${pkgbase//-/_}-6.$pkgver.tar.gz"
 )
 sha512sums=('f3e58172d823e83a880e0e6aa600aa689cfd53e9199c3e676d6665c64cab606fc2f0569bca7b96a85087bbf13a4410decc397f2ccc7aa6d7653f4d357cbc54e8'
             '18bc71031bbcbc3810a9985fa670465040f06a6c104ab8079b56bdfc499bb6cec40805a0cefd455031142490a576dc60aa8000523877ac0353b93558e9beabbd'
-            '1ebdea4e533ee0f71baf1b3fe2623ca723b36a08c6b97475ea5996b10aeb6873cf94d9120596ddd1216bd2f6feb991f8c33078e8104008a5078ace5be5431efd')
+            '1ebdea4e533ee0f71baf1b3fe2623ca723b36a08c6b97475ea5996b10aeb6873cf94d9120596ddd1216bd2f6feb991f8c33078e8104008a5078ace5be5431efd'
+            'ad1b61e7cd9dacbcc5dfa181ed7ed55325eb582e5c104102977e4051c02293c163e16deb6593c8b7cc65ff7b1ee3a636c324c1b05eb6eec9960391b0499a04e0')
 
 options=(!lto)
 
@@ -56,7 +58,6 @@ _gemname=google-protobuf
 
 prepare() {
   cd "$pkgname"
-  bazel --version
 
   patch -p1 < ../protobuf-fix-build-type-none.patch # Fix cmake config compatibility mode
   patch -p1 < ../$pkgbase-25.3-ruby-disable-LTO.patch
@@ -79,13 +80,14 @@ build() {
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
 
-  cd "$pkgbase"
-  bazel build //python/dist:binary_wheel --noenable_bzlmod --enable_workspace
+  cd "$pkgbase-6.$pkgver"
+  python -m build --wheel --no-isolation
 
-  cd ruby
+  cd ../"$pkgbase"/ruby
   local _gemdir="$(gem env gemdir)"
   local _gemver=4.$pkgver
 
+  export PATH="$srcdir"/build:$PATH
   PROTOC="$srcdir"/build/protoc \
     rake
   rake gem
@@ -153,7 +155,7 @@ package_python-protobuf() {
     'python'
   )
 
-  python -m installer --destdir="$pkgdir" "$pkgbase"/bazel-bin/python/dist/*.whl
+  python -m installer --destdir="$pkgdir" "$pkgbase-6.$pkgver"/dist/*.whl
   install -vDm 644 $pkgbase/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
