diff --git PKGBUILD PKGBUILD
index af79868..a8391a2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,7 +23,6 @@ depends=(
   'abseil-cpp'
 )
 makedepends=(
-  'bazelisk'
   'cmake'
   'git'
   'gtest'
@@ -44,9 +43,12 @@ checkdepends=(
 source=(
   git+https://github.com/protocolbuffers/protobuf.git#tag=v$pkgver
   protobuf-fix-build-type-none.patch
+  # Keep it here instead of bottom of the file because this file updates along with the package
+  "$pkgbase-$pkgver-python.tar.gz::https://files.pythonhosted.org/packages/source/${pkgbase::1}/${pkgbase//-/_}/${pkgbase//-/_}-6.$pkgver.tar.gz"
 )
 sha512sums=('0efcb80ec3a1f04f0f2e53ccd629eace4f6b9a2cbe5dae4c1b82140f11e174f8d023b8e35855def7e19c35da838c5b4fcfaa54748ee3534886caf1d35f55cccb'
-            '18bc71031bbcbc3810a9985fa670465040f06a6c104ab8079b56bdfc499bb6cec40805a0cefd455031142490a576dc60aa8000523877ac0353b93558e9beabbd')
+            '18bc71031bbcbc3810a9985fa670465040f06a6c104ab8079b56bdfc499bb6cec40805a0cefd455031142490a576dc60aa8000523877ac0353b93558e9beabbd'
+            'c29d1b01984e0cc204bffa5c757e4df4eb8ee696b6f5557414e65310b15956192fc73db5daae176f2b4db03e98c36a46cc74ca8b377df193bc9461142c8269cf')
 
 options=(!lto)
 
@@ -54,7 +56,6 @@ _gemname=google-protobuf
 
 prepare() {
   cd "$pkgname"
-  bazel --version
 
   patch -p1 < ../protobuf-fix-build-type-none.patch # Fix cmake config compatibility mode
 }
@@ -76,7 +77,7 @@ build() {
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
 
-  cd "$pkgbase"
+  cd "$pkgbase-6.$pkgver"
   local bazel_options=(
     --noenable_bzlmod
     --enable_workspace
@@ -87,9 +88,9 @@ build() {
   # for aarch64
   [[ $CARCH == "aarch64" ]] && bazel_options+=(--cpu=linux-aarch_64)
 
-  bazel build //python/dist:binary_wheel "${bazel_options[@]}"
+  python -m build --wheel --no-isolation
 
-  cd ruby
+  cd ../"$pkgbase"/ruby
   local _gemdir="$(gem env gemdir)"
   local _gemver=4.$pkgver
 
@@ -160,7 +161,7 @@ package_python-protobuf() {
     'python'
   )
 
-  python -m installer --destdir="$pkgdir" "$pkgbase"/bazel-bin/python/dist/*.whl
+  python -m installer --destdir="$pkgdir" "$pkgbase-6.$pkgver"/dist/*.whl
   install -vDm 644 $pkgbase/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
 
