--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,6 @@ depends=(
   'abseil-cpp'
 )
 makedepends=(
-  'bazel'
   'cmake'
   'gtest'
   'python-build'
@@ -60,8 +59,8 @@ build() {
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
 
-  cd "$pkgbase-$pkgver"
-  bazel build //python/dist:binary_wheel
+  cd "$pkgbase-5.$pkgver"
+  python -m build --wheel --no-isolation
 }
 
 check() {
@@ -89,6 +88,9 @@ package_python-protobuf() {
     'python'
   )
 
-  python -m installer --destdir="$pkgdir" "$pkgbase-$pkgver"/bazel-bin/python/dist/*.whl
+  python -m installer --destdir="$pkgdir" "$pkgbase-5.$pkgver"/dist/*.whl
   install -vDm 644 $pkgbase-$pkgver/LICENSE -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source+=("$pkgbase-$pkgver-python.tar.gz::https://files.pythonhosted.org/packages/source/${pkgbase::1}/${pkgbase//-/_}/${pkgbase//-/_}-5.$pkgver.tar.gz")
+sha512sums+=('b3afcf13e223d7e855ac0aa000cc395e3ea1301b2e3dfe3c0d5435d8031be3726e6aec1fb6228c572aa1685f75aba2d97d2c9dfdb6d9e058a2e88321d6ba9180')
