diff -uprN vm.lv2-0.14.0/meson.build vm.lv2-0.14.0-patch/meson.build
--- vm.lv2-0.14.0/meson.build	2021-04-14 17:35:22.000000000 +0800
+++ vm.lv2-0.14.0-patch/meson.build	2021-11-10 16:09:46.170243248 +0800
@@ -68,7 +68,8 @@ mod = shared_module('vm', dsp_srcs,
 	name_prefix : '',
 	dependencies : dsp_deps,
 	install : true,
-	install_dir : inst_dir)
+	install_dir : inst_dir,
+	link_args: ['-pthread'])
 
 ui = shared_module('vm_ui', ui_srcs,
 	c_args : c_args,
