diff --git PKGBUILD PKGBUILD
index eaafce6..36ee1d0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,6 +9,7 @@ pkgdesc='Small and modern C++ library, offering some basic functionality to buil
 url='https://lib.filezilla-project.org/'
 arch=('x86_64')
 license=('GPL')
+options=(!lto) # https://github.com/llvm/llvm-project/issues/50591
 depends=('glibc' 'gcc-libs' 'nettle' 'gnutls' 'libnettle.so' 'libhogweed.so')
 makedepends=(clang)
 checkdepends=('cppunit')
