diff --git a/Makefile b/Makefile
index 68d5062..53c2012 100644
--- a/Makefile
+++ b/Makefile
@@ -73,9 +73,9 @@ ifeq ($(platform), unix)
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
    CFLAGS += -D_GNU_SOURCE
    ifneq ($(findstring Haiku,$(shell uname -s)),)
-      LIBS += -lpthread -lroot -lnetwork
+      LIBS += -pthread -lroot -lnetwork
    else
-      LIBS += -lpthread -lrt
+      LIBS += -pthread -lrt
    endif
    HAVE_THREADS=1
    ifneq ($(filter $(ARCH),x86 x86_64),)
@@ -92,7 +92,7 @@ else ifeq ($(platform), linux-portable)
    fpic := -fPIC -nostdlib
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T
 	LIBS =
-	#LIBS +=-lpthread
+	#LIBS +=-pthread
 else ifneq (,$(findstring osx,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.dylib
    fpic := -fPIC
@@ -168,7 +168,7 @@ else ifneq (,$(findstring qnx,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro_qnx.so
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-	#LIBS +=-lpthread
+	#LIBS +=-pthread
 else ifeq ($(platform), emscripten)
    TARGET := $(TARGET_NAME)_libretro_emscripten.bc
    fpic := -fPIC
@@ -236,7 +236,7 @@ else ifeq ($(platform), rpi4_64)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
@@ -249,7 +249,7 @@ else ifeq ($(platform), odroidc4)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
@@ -262,7 +262,7 @@ else ifeq ($(platform), odroidgoa)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
@@ -275,7 +275,7 @@ else ifeq ($(platform), odroidn2)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
@@ -288,7 +288,7 @@ else ifeq ($(platform), RK3399)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
@@ -301,7 +301,7 @@ else ifeq ($(platform), orangepizero2)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -pthread -lGLESv2
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
