diff --git PKGBUILD PKGBUILD
index 81d6339..a36053e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,8 +17,15 @@ makedepends=(
   libglvnd
 )
 _commit=2c5d2ecfe744647ce3055cd21c39620a0327cb95
-source=(libretro-melonds::git+https://github.com/libretro/melonDS.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(libretro-melonds::git+https://github.com/libretro/melonDS.git#commit=${_commit}
+        "fix_pthread_linking.patch")
+sha256sums=('SKIP'
+            'f0526d70d41a99602ad5b43d8bc6cd1f86b148e7f1bb70963051ffaad3e8465d')
+
+prepare() {
+  cd libretro-melonds
+  patch -Np1 -i "../fix_pthread_linking.patch"
+}
 
 pkgver() {
   cd libretro-melonds
