diff --git PKGBUILD.orig PKGBUILD
index 0bc7507..61cb9ed 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -11,9 +11,17 @@
 depends=('glibc')
 makedepends=('alsa-lib' 'cmake')
 provides=('libportmidi.so' 'libporttime.so')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/mixxxdj/portmidi/archive/refs/tags/${pkgver}.tar.gz")
-sha512sums=('901729440c4b8c654ab17c4c4f1e3986813bf1e4ad1a874f46e7b1ee0c9ef4ee9f1ecfdf71012fb56b055a6185194dd22c520ae87b9f2259c18af5d189ca57da')
-b2sums=('618c3778b8b2170a145ec956cd4721e731cc135d0192e4b1c35b2c93b6b0f0e74085d044beafe2095b3d8e02e1776f442ae174d342c3f391d04a2aab378b547a')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/mixxxdj/portmidi/archive/refs/tags/${pkgver}.tar.gz"
+        "0001-fix-build-with-Werror-format-security.patch")
+sha512sums=('901729440c4b8c654ab17c4c4f1e3986813bf1e4ad1a874f46e7b1ee0c9ef4ee9f1ecfdf71012fb56b055a6185194dd22c520ae87b9f2259c18af5d189ca57da'
+            'f45952d72a36cb0166e45529893e2ff8b9795f592f322d9ffef35a4c3f55dff206f3d470cdd071696f71f44270f5d6ff368e34eef3153d81b154edbe5df0998d')
+b2sums=('618c3778b8b2170a145ec956cd4721e731cc135d0192e4b1c35b2c93b6b0f0e74085d044beafe2095b3d8e02e1776f442ae174d342c3f391d04a2aab378b547a'
+        'ed782e2b208a2af19e1590bb59b5c7f5e62dd7f053b1f563e446c480c5ac8563196e0c1604b9ebe046f88c0e28835bd1875086b8a3184c659f994f7a23576f63')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i ../0001-fix-build-with-Werror-format-security.patch
+}
 
 build() {
   cd "${pkgname}-${pkgver}"
