diff --git PKGBUILD.orig PKGBUILD
index 0bc7507..40b2158 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -11,9 +11,17 @@
 depends=('glibc')
 makedepends=('alsa-lib' 'cmake')
 provides=('libportmidi.so' 'libporttime.so')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/mixxxdj/portmidi/archive/refs/tags/${pkgver}.tar.gz")
-sha512sums=('901729440c4b8c654ab17c4c4f1e3986813bf1e4ad1a874f46e7b1ee0c9ef4ee9f1ecfdf71012fb56b055a6185194dd22c520ae87b9f2259c18af5d189ca57da')
-b2sums=('618c3778b8b2170a145ec956cd4721e731cc135d0192e4b1c35b2c93b6b0f0e74085d044beafe2095b3d8e02e1776f442ae174d342c3f391d04a2aab378b547a')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/mixxxdj/portmidi/archive/refs/tags/${pkgver}.tar.gz"
+        "0001-fix-build-with-Werror-format-security.patch")
+sha512sums=('901729440c4b8c654ab17c4c4f1e3986813bf1e4ad1a874f46e7b1ee0c9ef4ee9f1ecfdf71012fb56b055a6185194dd22c520ae87b9f2259c18af5d189ca57da'
+            '7d633c16eae96dc111bba7908bf671505d5a152676c08bb0ac22e71ee739e906a12a8a3cdc7cea547635ec6156f0dcac5f8c6a23e20f12057b42fc0f32c54e12')
+b2sums=('618c3778b8b2170a145ec956cd4721e731cc135d0192e4b1c35b2c93b6b0f0e74085d044beafe2095b3d8e02e1776f442ae174d342c3f391d04a2aab378b547a'
+        '01014429ecea06411e312c5861eeb601e9c9e07b8716afac47dd469fefe76f30e737e654065fded11a30a6d872e201f1f2950a2d6cb2e5ea581ba374e8536098')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i ../0001-fix-build-with-Werror-format-security.patch
+}
 
 build() {
   cd "${pkgname}-${pkgver}"
