--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,9 @@ validpgpkeys=(
   '8777AB8F778EE89487A2F8E7F4ACA0183641E010' # luigi1111 <luigi1111w@gmail.com>
   '487277A8BD0A209C16B700F3C64552D877C32479' # Alexander Blair (Snipa / Snipa22) <snipa@jagtech.io>
 )
+# bypass error:
+# relocation truncated to fit: R_RISCV_JAL against symbol `mdb_stat' defined in text_env section in /tmp/ccWy6leF.ltrans22.ltrans.o
+options=(!lto)
 
 prepare() {
   cd "${pkgname}"
