diff --git PKGBUILD PKGBUILD
index 330135a7..6eacc47c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,7 @@ source=(
     "git+https://github.com/tevador/randomx.git"
     "git+https://github.com/monero-project/supercop.git"
     "monero-gui.desktop"
+    "fix-build.patch::https://github.com/monero-project/monero/pull/8117.patch"
 )
 sha512sums=('SKIP'
             'SKIP'
@@ -30,7 +31,8 @@ sha512sums=('SKIP'
             'SKIP'
             'SKIP'
             'SKIP'
-            '98b67aec60e492f0a220a156a110baf96862343d3802ec34347c70c7af671a1ae49ccaec218f2e8f81e153c433e9145ad2f55b8613ce6999fc2c87fbd81fd8e1')
+            '98b67aec60e492f0a220a156a110baf96862343d3802ec34347c70c7af671a1ae49ccaec218f2e8f81e153c433e9145ad2f55b8613ce6999fc2c87fbd81fd8e1'
+            'ab12eee1941b3f5b76bacf579914d74dabe69ebf7249afba808c84129b2d51247ed9581f4f214d53e4c2e0a6eea8fd515a7a8cffd50bf63a7312809fc48e6b5e')
 validpgpkeys=(
   '8777AB8F778EE89487A2F8E7F4ACA0183641E010' # luigi1111 <luigi1111w@gmail.com>
   '487277A8BD0A209C16B700F3C64552D877C32479' # Alexander Blair (Snipa / Snipa22) <snipa@jagtech.io>
@@ -49,6 +51,7 @@ prepare() {
   git config submodule.external/randomx.url "$srcdir/randomx"
   git config submodule.external/supercop.url "$srcdir/supercop"
   git submodule update
+  patch -Np1 -i "$srcdir/fix-build.patch"
 }
 
 build() {
