Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1269496)
+++ PKGBUILD	(working copy)
@@ -14,6 +14,11 @@
 source=(https://www.gaia-gis.it/gaia-sins/$pkgname-$pkgver.tar.gz)
 sha256sums=('eecbc94311c78012d059ebc0fae86ea5ef6eecb13303e6e82b3753c1b3409e98')
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr
