--- PKGBUILD
+++ PKGBUILD
@@ -20,9 +20,11 @@ optdepends=('apache: for extending Apache with mod_neko')
 options=(!strip)
 source=(2023-04-10.patch
         neko-mbedtls2.patch
+        "add-riscv64-support.patch::https://patch-diff.githubusercontent.com/raw/HaxeFoundation/neko/pull/285.diff"
         "git+https://github.com/HaxeFoundation/neko#commit=ff67a696a718cba0a6bc0ec583e355272b4a5923") # tag: v2-3-0
 b2sums=('83a08288938a294dbcd2e6c91e7912989332807375173a5f6c11ec818e3e8bad59e0f4826a6743a6a3802b4faa0c85837424a20a314c7cb9a596367283aaf564'
         'ea9c86b2f6ac8e943f0c70406a73a055468e71de8740adc2116e0cfd9e68d08ac0d071ce94119a52954fed08319176308395ae20715ce7bd0d38fb538c95f765'
+        '519796a02ffa2a0bf101339e0b1e643d6d78fe3a4dd555f8838fc44d575505752a6e711b99c1d2470a9d0546af9b0e039fce13c5a1de8b32e2a74d670febd512'
         'SKIP')
 
 prepare() {
@@ -30,6 +32,7 @@ prepare() {
   sed -i '/xlocale.h/d' libs/std/sys.c
   patch -p1 -i ../2023-04-10.patch
   patch -p1 -i ../neko-mbedtls2.patch
+  patch -p1 -i ../add-riscv64-support.patch
 }
 
 build() {
