diff -Naur naev-0.12.3/meson.build naev-0.12.3_/meson.build
--- naev-0.12.3/meson.build	2025-02-01 10:29:56.000000000 +0800
+++ naev-0.12.3_/meson.build	2025-02-18 23:25:31.006265834 +0800
@@ -275,9 +275,7 @@
    summary('tracy', have_tracy, section: 'Debug', bool_yn: true )
 
    # Standard library feature tests
-   config_data.set10('HAVE_FEENABLEEXCEPT', cc.has_header_symbol('fenv.h', 'feenableexcept', prefix: '#define _GNU_SOURCE'))
    config_data.set10('HAVE_ALLOCA_H', cc.has_header('alloca.h'))
-   config_data.set10('HAVE_FENV_H', cc.has_header('fenv.h'))
    config_data.set10('HAVE_MALLOC_H', cc.has_header('malloc.h'))
    # SDL_strndup and SDL_strnstr is in SDL3, so we can remove this once it is released
    # strndup() detection must work around this bug: https://github.com/mesonbuild/meson/issues/3672
