--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,7 @@ prepare() {
 
   patch -Np1 -i ../eb11709e3a38da52f93e5415cc7fc9152a081fef.patch
   patch -Np1 -i ../8cd181e8666e1250f39766dda742f00206f1edbf.patch
+  patch -Np1 -i ../disable-riscv64-fpu-exception.patch
 }
 
 build() {
@@ -40,3 +41,6 @@ package() {
   DESTDIR="$pkgdir" meson install
   rm -r "$pkgdir"/usr/{include,lib}
 }
+
+source+=('disable-riscv64-fpu-exception.patch')
+sha512sums+=('6746ff95362eb579e180ee3443a9f7ab1aa670333b5a67d7cdda4400237360c4bcaedc534b25ebe7d908f8914da6d15adba62ecd690a482c9a5ce47e14e173d9')
