diff --git PKGBUILD PKGBUILD
index 9ff67d5..8cb9b54 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,10 +13,17 @@ url='https://github.com/ROCm/rocminfo'
 license=('NCSA')
 depends=('rocm-core' 'glibc' 'gcc-libs' 'pciutils' 'python' 'hsa-rocr')
 makedepends=('cmake' 'rocm-cmake')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/rocm-$pkgver.tar.gz")
-sha256sums=('e616d364a48de18eaee661bdce999d095086905f49777663ca99312f40a63da1')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/rocm-$pkgver.tar.gz"
+        rocminfo_fix-cxx-flags.patch::https://github.com/RadeonOpenCompute/rocminfo/pull/63.diff)
+sha256sums=('e616d364a48de18eaee661bdce999d095086905f49777663ca99312f40a63da1'
+            '423baa1e51a14987f472286f86eabe00533ccaa3ccfd294d153545f9abc61880')
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 
+prepare() {
+  # remove '-m64' c++ flag on architectures that don't support it.
+  patch -d "$_dirname" -Np1 -i ../rocminfo_fix-cxx-flags.patch
+}
+
 build() {
   # ROCRTST_BLD_TYPE=Release fixes a build error regarding _FORTIFY_SOURCE=2
   local cmake_args=(
