--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,10 @@ b2sums=('e8eaa780f5f985419db81124a9d36470443e5c70eb6f8dc1d850c40c739cb0a957c1a71
         'SKIP')
 validpgpkeys=('71A1D0EFCFEB6281FD0437C93D5919B448457EE0')
 
+prepare() {
+  patch -Np1 -i bazel7-riscv.diff
+}
+
 build() {
   EMBED_LABEL=$pkgver EXTRA_BAZEL_ARGS="--tool_java_runtime_version=local_jdk" ./compile.sh
   ./output/bazel build scripts:bazel-complete.bash
@@ -38,4 +42,7 @@ package() {
     cp -r "${srcdir}/${d}" "${pkgdir}/usr/share/bazel/"
   done
 }
+
+source+=(bazel7-riscv.diff)
+b2sums+=('d65de7c58096d588be5964f52accbd65ee0f0c4e53c2021e4665089fed3f8a3a58e5d9fed5fa8c201ad391482a94a463eb8d33be53aa699f30cbfbc4afa3051c')
 # vim:set ts=2 sw=2 et:
