--- PKGBUILD
+++ PKGBUILD
@@ -97,6 +97,7 @@
   sed -E 's|(else )hostname(;)|\1uname -n\2|' -i deps/rabbitmq_ct_helpers/tools/tls-certs/Makefile
   patch -p1 < ../rabbitmq-customize-systemd-service.patch
   patch -p1 < ../rabbitmq-devendor-rebar3.patch
+  patch -p1 < ../slow-builder-tests.patch
 
   # The only two tests in deprecated_features_SUITE that start a full broker via
   # rabbit_ct_broker_helpers:setup_steps/0 rather than lightweight slave nodes,
@@ -194,3 +195,5 @@
   chown -vR 197:0 "$pkgdir/etc/rabbitmq"
 }
 
+source+=('slow-builder-tests.patch')
+sha512sums+=('a1d51e86e9938fb18a063da6a6366cd1ce609aae40e7fd9006617ba0fe652c4657934a6a83880d753836ed99fd9a43eb3725c3e537d14cb1ec8b0e57c0868d84')
