--- crates/glaz-proto/build.rs
+++ crates/glaz-proto/build.rs
@@ -1,7 +1,13 @@
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let protoc_path = protoc_bin_vendored::protoc_bin_path()?;
-    prost_build::Config::new()
-        .protoc_executable(protoc_path)
-        .compile_protos(&["service.proto"], &["."])?;
+    let mut config = prost_build::Config::new();
+
+    if let Ok(protoc) = std::env::var("PROTOC") {
+        config.protoc_executable(protoc);
+    } else {
+        let protoc = protoc_bin_vendored::protoc_bin_path()?;
+        config.protoc_executable(protoc);
+    }
+
+    config.compile_protos(&["service.proto"], &["."])?;
     Ok(())
 }
