--- PKGBUILD
+++ PKGBUILD
@@ -85,7 +85,15 @@
 _datadir=/var/lib/gitlab # directory with gitlab data and it also $HOME for 'gitlab' user
 _logdir=/var/log/gitlab
 
+_rbtrace_version=0.5.5
+_glaz_revision=af9019a186fbb42cede0ad5ba7496aff2f88cb41
+
 prepare() {
+	# Refresh rbtrace's bundled msgpack config.guess and config.sub for riscv64.
+	sed -Ei '/^([[:space:]]*)Dir.chdir\(dir\) do/a\      sys("autoreconf -fi")' rbtrace/ext/extconf.rb
+	mkdir -p .cargo
+	printf 'paths = ["%s"]\n' "$srcdir/glaz/crates/glaz-proto" > .cargo/config.toml
+
 	cd gitlab-foss
 
 	# GitLab tries to read its revision information from a file.
@@ -116,6 +124,12 @@
 }
 
 build() {
+	export PROTOC=/usr/bin/protoc
+
+	cd rbtrace
+	gem build rbtrace.gemspec
+	cd ..
+
 	cd gitlab-foss
 
 	# https://github.com/nodejs/node/issues/48444
@@ -139,13 +153,24 @@
 	bundle config --local force_ruby_platform true # some native gems are not available for newer ruby
 	bundle config --local deployment true
 	bundle config --local without 'development test aws kerberos'
+	bundle config --local build.nokogiri "--use-system-libraries --enable-shared --disable-static --with-cflags='-O2'"
+	bundle config --local build.ruby-magic --use-system-libraries
+
+	# Replace rbtrace with the copy patched above.
+	export GEM_HOME=$HOME/.gem
+	bundle config --local cache_path vendor/cache/
+	bundle cache --no-install
+	cp ../rbtrace/rbtrace-${_rbtrace_version}.gem vendor/cache/
+	rbtrace_checksum=$(sha256sum vendor/cache/rbtrace-${_rbtrace_version}.gem)
+	sed -Ei "s/(  rbtrace \\(${_rbtrace_version}\\) sha256=)[[:alnum:]]*/\\1${rbtrace_checksum:0:64}/" Gemfile.lock
+
 	# Disable make jobserver due to a race in prometheus-client-mmap
 	# Use clang-21 as clang-22 is incomaptible with rb-sys < 0.9.125 used by prometheus-client-mmap and magnus
 	env \
 		MAKEFLAGS= \
 		CLANG_PATH=/usr/lib/llvm21/bin/clang-21 \
 		BUNDLER_CHECKSUM_VERIFICATION_OPT_IN=1 \
-		bundle install --jobs=$(nproc) --no-cache
+		bundle install --local --jobs=$(nproc) --no-cache
 
 	export CGO_CPPFLAGS="${CPPFLAGS}"
 	export CGO_CFLAGS="${CFLAGS}"
@@ -256,3 +281,10 @@
 
 	find "$pkgdir$_appdir" -name man -type d -print0 | xargs -0 -I {} rm -rf {}
 }
+
+depends+=(file)
+makedepends+=(protobuf)
+source+=("git+https://github.com/tmm1/rbtrace.git#tag=v${_rbtrace_version}"
+         "git+https://gitlab.com/gitlab-org/auth/glaz.git#commit=${_glaz_revision}")
+sha256sums+=('a46a61f3c12ee54ed73a2d326009fc4de05eb69fd262008b44c9008cc492719a'
+             '69f6f1f87c5c559189ddb30c49b5dea03297a9ea5d71beeac90e38320707fc9c')
