--- PKGBUILD
+++ PKGBUILD
@@ -49,6 +49,7 @@
   sed -i 's|"libraries"|"/usr/share/materialx/libraries"|g' source/MaterialXGenShader/GenOptions.h
 
   dos2unix python/Scripts/*
+  patch -Np1 -i "$srcdir"/nanogui-do-not-fallback-to-nehalem.patch -d source/MaterialXView/NanoGUI
 }
 
 build() {
@@ -104,3 +105,6 @@
   # Remove empty dirs
   find ${pkgdir}/usr -empty -type d -delete
 }
+
+source+=("nanogui-do-not-fallback-to-nehalem.patch::https://github.com/mitsuba-renderer/nanogui/pull/156.diff")
+sha256sums+=('fe60c485ddc71df9be0bd79ddda8bdd127d1ff8beec7b6fc2ffc4e1dcc0bbb70')
