--- PKGBUILD
+++ PKGBUILD
@@ -26,15 +26,19 @@ optdepends=(
   'hivex: for repack-windows command'
   'wimlib: for repack-windows command'
 )
-source=("$url/releases/download/$pkgname-$pkgver/$pkgname-$pkgver.tar.gz"{,.asc})
+source=("$url/releases/download/$pkgname-$pkgver/$pkgname-$pkgver.tar.gz"{,.asc}
+        "fix-static-declaration-of-lxd.patch")
 sha512sums=('80184d27305f5659f1367563161220b966b1957810da61e8f3b9f32cf408795319b43dd650bc9f867b3ac73ef2b3251972bb11999d92feb34bfd9cdae7ff7b7b'
-            'SKIP')
+            'SKIP'
+            '04f5bbdcb4643de97fca51c7272f04cf5518d62ca7f09d4e559ba103c8639ac8cfcd84c5bb4405447364522e42cae9dac9c3b412bc89561472f0524cbd73969f')
 b2sums=('90e82a20a3ef61d13148951956bb1f74cc3494ca86e339c188a1c9009dcb27c0cb9a298c639c6de891df67915b48d107557f6b698a45cbe35c54d41a45bac1ab'
-        'SKIP')
+        'SKIP'
+        '1935f38101344ac803d70352afde9f5e0b95352cfd74cf95b87630562731bc2bd2704e73a82fb786e7beb28971d2bde74a7093c9b8b16ca36de4798a5e919b85')
 validpgpkeys=('602F567663E593BCBD14F338C638974D64792D67') # Stéphane Graber <stgraber@stgraber.org>
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -p1 -i ${srcdir}/fix-static-declaration-of-lxd.patch -d ${srcdir}/distrobuilder-2.1/vendor/github.com/lxc/lxd
 
   # create folder for build output
   mkdir build
