diff --git PKGBUILD PKGBUILD
index cb91150..3acc226 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,7 @@ arch=('x86_64')
 url="http://glhack.sourceforge.net/"
 license=('custom')
 depends=('sdl' 'libpng' 'libgl')
-makedepends=('mesa')
+makedepends=('mesa' 'systemd')
 options=('!makeflags')
 install=glhack.install
 source=(https://downloads.sourceforge.net/${pkgname}/${pkgname}-${pkgver}.tar.gz glhack-libpng15.patch)
