diff --git PKGBUILD PKGBUILD
index 616501a..b61571a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,7 @@
              'meson' 'libseccomp' 'pcre2' 'audit' 'kexec-tools' 'libxkbcommon'
              'bash-completion' 'p11-kit' 'systemd' 'libfido2' 'tpm2-tss' 'rsync'
              'bpf' 'libbpf' 'clang' 'llvm' 'curl' 'gnutls' 'python-pyelftools'
-             'libpwquality' 'qrencode' 'lib32-gcc-libs' 'python-pefile')
+             'libpwquality' 'qrencode' 'python-pefile')
 conflicts=("mkinitcpio<38-1")
 validpgpkeys=('63CDA1E5D3FC22B998D20DD6327F26951A015CC4'  # Lennart Poettering <lennart@poettering.net>
               'A9EA9081724FFAE0484C35A1A81CEA22BC8C7E2E'  # Luca Boccassi <luca.boccassi@gmail.com>
@@ -85,6 +85,8 @@
 fi
 
 _backports=(
+  # vmspawn: define QEMU_MACHINE_TYPE for riscv
+  5f7136a3690cd118eeef3c84dfbd9f5d0df556d6
 )
 
 _reverts=(
@@ -172,7 +174,7 @@
 }
 
 check() {
-  meson test -C build --print-errorlogs
+  meson test -C build --print-errorlogs -t 10
 }
 
 package_systemd() {
