diff --git PKGBUILD PKGBUILD
index 1109533..d1014fa 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,10 @@ url="https://gn.googlesource.com/gn/"
 license=('BSD')
 depends=('gcc-libs')
 makedepends=('clang' 'ninja' 'python' 'git')
-source=(git+https://gn.googlesource.com/gn#commit=$_commit)
-sha256sums=('SKIP')
+source=(git+https://gn.googlesource.com/gn#commit=$_commit
+        fix-build-for-riscv64-arch.patch) # https://gn.googlesource.com/gn/+/335dd456f93d9817210a537bb847ad2bc4f66803%5E%21/
+sha256sums=('SKIP'
+            '87a2ecb85a740983f75dc4fde936f39b632958fac965d36a095ab68a852718ca')
 
 pkgver() {
   cd gn
@@ -21,6 +23,9 @@ pkgver() {
 
 build() {
   cd gn
+
+  patch -Np1 -i ../fix-build-for-riscv64-arch.patch
+
   ./build/gen.py
   ninja -C out
 }
