Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -12,7 +12,7 @@
 depends=(
   'glibc' 'ncurses' 'libcap' 'libassuan' 'libsecret' 'glib2'
   libsecret-1.so libcap.so libncursesw.so libassuan.so libglib-2.0.so)
-makedepends=('gtk2' 'qt5-base' 'gcr')
+makedepends=()
 optdepends=(
   'gtk2: gtk2 backend'
   'qt5-base: qt backend'
@@ -39,9 +39,6 @@
     --enable-pinentry-curses \
     --enable-fallback-curses \
     --enable-pinentry-emacs \
-    --enable-pinentry-gtk2 \
-    --enable-pinentry-gnome3 \
-    --enable-pinentry-qt \
     --enable-libsecret \
 
   make
