diff --git PKGBUILD trunk/PKGBUILD
index 15fc4fd..2bfe92f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,11 @@ sha512sums=('c5f92d86e143db02ebb36bcd25618acaa2cb2831f5a23800a06dd431cd73b6702d9
             'SKIP')
 validpgpkeys=('4A193C06D35E7C670FA4EF0BA2FB9E081F2D130E') # Alan Coopersmith <alan.coopersmith@oracle.com>
 
+prepare() {
+  cd libXxf86vm-${pkgver}
+  autoreconf -fiv
+}
+
 build() {
   cd libXxf86vm-${pkgver}
   ./configure --prefix=/usr --disable-static
