--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ arch=(x86_64)
 url="https://github.com/curl/trurl"
 license=(custom)
 depends=(curl glibc)
-checkdepends=(python valgrind)
+checkdepends=(python)
 validpgpkeys=('27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2') # Daniel Stenberg
 source=(https://github.com/curl/$pkgname/releases/download/$pkgname-$pkgver/$pkgname-$pkgver.tar.gz{,.asc})
 sha256sums=('e58d9a698c009b2b1381f5636b5334ce7704ad2cd5ae8d30da97d483518a7f25'
@@ -24,7 +24,6 @@ build() {
 check() {
 	cd "$pkgname-$pkgver"
 	make test
-	make test-memory
 }
 
 package() {
