--- PKGBUILD	(revision 1263162)
+++ PKGBUILD	(working copy)
@@ -22,7 +22,7 @@
         --prefix=/usr --docdir=/usr/share/doc/$pkgname --datasubdir=$pkgname --enable-tests \
         --dynlibdir=/usr/lib --libsubdir=\$compiler/site-local/\$pkgid --ghc-option=-fllvm \
             -fsupport_deepseq -finteger-gmp -f-support_pclmuldq -fsupport_rdrand \
-            -fsupport_aesni -f-old_toolchain_inliner -f-check_alignment -fsupport_sse
+            -fsupport_aesni -f-old_toolchain_inliner -f-check_alignment
     runhaskell Setup build $MAKEFLAGS
     runhaskell Setup register --gen-script
     runhaskell Setup unregister --gen-script
