diff --git PKGBUILD PKGBUILD
index c5feb48..63134a3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,15 @@
              'gobject-introspection' 'boost' 'fmt' 'mm-common' 'perl-xml-parser' 'libsigc++-3.0')
 optdepends=('webkit2gtk: embedded browser features'
             'xdg-desktop-portal: desktop integration')
-source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz")
-sha512sums=('7ab1f8b93d4c1efcb08ca42f9b4604a17a22517d6616ba973018a4b7ef63de3d6650deeb74d752548ec7fce8b3581a73b974c6bc6d2c16e4f31049bc11d87b7a')
+source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz"
+        tdesktop-scudo-rv-fix.patch)
+sha512sums=('7ab1f8b93d4c1efcb08ca42f9b4604a17a22517d6616ba973018a4b7ef63de3d6650deeb74d752548ec7fce8b3581a73b974c6bc6d2c16e4f31049bc11d87b7a'
+            'bedfd923f64e66c7562afc6e346d32602d2a3f4190b0b2135940cbaf1014dcb993201bc4cecad56a1add576b32dba14d73cee2f019448182fbf7e7ce5223be0d')
+
+prepare() {
+    cd tdesktop-$pkgver-full
+    patch -p1 -i ../tdesktop-scudo-rv-fix.patch
+}
 
 build() {
     CXXFLAGS+=' -ffat-lto-objects'
