--- PKGBUILD
+++ PKGBUILD
@@ -16,14 +16,17 @@ optdepends=('webkit2gtk: embedded browser features'
             'xdg-desktop-portal: desktop integration')
 source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz"
         fix-tgcalls-cstdint.patch
+        $pkgname-add-signed-to-char.patch::https://github.com/telegramdesktop/tdesktop/pull/24706.patch
        )
 sha512sums=('11004d7a889e64da0f9a9d976fc9665974e000b9aefd909723606639d97df07c62e86fce6b496e40ebdb954acc410609959cd94ca5d7b18c1d89373a09e970f1'
-            'ba24a2f1524010b4891764aacee2e27a5ebae44cf7626ab2aaf9e6c48b0f8088bf920886ceeb497b3c463fa0c3b885dd63db273d4c29cab6c96c8193c0c5c888')
+            'ba24a2f1524010b4891764aacee2e27a5ebae44cf7626ab2aaf9e6c48b0f8088bf920886ceeb497b3c463fa0c3b885dd63db273d4c29cab6c96c8193c0c5c888'
+            '7c75d969e8771f4674b74ca38f1ec8818bd96566dcf0906ec6d19dded205a64204e3a603a176b6f024b74bf9dca03c8f612a6b4c21ca69be3e9666d9884deae3')
 
 prepare() {
     cd tdesktop-$pkgver-full
 
     patch -Np1 -i "$srcdir"/fix-tgcalls-cstdint.patch -d Telegram/ThirdParty/tgcalls
+    patch -Np1 -i $srcdir/$pkgname-add-signed-to-char.patch
 }
 
 build() {
