--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,14 @@ optdepends=('webkit2gtk: embedded browser features'
 source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz")
 sha512sums=('863e22e5d0809736a6f05e64ae7ceb5e93a1135c26f8e00d45a45cbb6bb90da3dafe43f9249aa07b6183e6e06aa0801613b72b033b7a05323c27556a25ca5854')
 
+source+=("tdesktop-scudo-rv-fix.patch")
+sha512sums+=("bedfd923f64e66c7562afc6e346d32602d2a3f4190b0b2135940cbaf1014dcb993201bc4cecad56a1add576b32dba14d73cee2f019448182fbf7e7ce5223be0d")
+
+prepare() {
+    cd tdesktop-$pkgver-full
+    patch -p1 -i ../tdesktop-scudo-rv-fix.patch
+}
+
 build() {
     CXXFLAGS+=' -ffat-lto-objects'
 
