--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,15 @@ makedepends=('cmake' 'git' 'ninja' 'python' 'range-v3' 'tl-expected' 'microsoft-
              'gobject-introspection' 'boost' 'fmt' 'mm-common' 'perl-xml-parser' 'libsigc++-3.0')
 optdepends=('webkit2gtk: embedded browser features'
             'xdg-desktop-portal: desktop integration')
-source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz")
-sha512sums=('a25d7ef6a1a810ac1c7267c5316b74714ad1c5f8d51cddb30f24b58bf4a5966ef0d252c985e962acfee2b8930567eda628b546844ee9eefad1c9bd9801897c44')
+source=("https://github.com/telegramdesktop/tdesktop/releases/download/v${pkgver}/tdesktop-${pkgver}-full.tar.gz"
+        tdesktop-scudo-rv-fix.patch)
+sha512sums=('a25d7ef6a1a810ac1c7267c5316b74714ad1c5f8d51cddb30f24b58bf4a5966ef0d252c985e962acfee2b8930567eda628b546844ee9eefad1c9bd9801897c44'
+            'bedfd923f64e66c7562afc6e346d32602d2a3f4190b0b2135940cbaf1014dcb993201bc4cecad56a1add576b32dba14d73cee2f019448182fbf7e7ce5223be0d')
+
+prepare() {
+    cd tdesktop-$pkgver-full
+    patch -p1 -i ../tdesktop-scudo-rv-fix.patch
+}
 
 build() {
     CXXFLAGS+=' -ffat-lto-objects'
