--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,7 @@ build() {
     cd tdesktop-$pkgver-full
 
     # Fix https://bugs.archlinux.org/task/73220
-    export CXXFLAGS+=" -Wp,-U_GLIBCXX_ASSERTIONS"
+    export CXXFLAGS+=" -Wp,-U_GLIBCXX_ASSERTIONS -latomic"
 
     export PKG_CONFIG_PATH='/usr/lib/ffmpeg4.4/pkgconfig'
     # Turns out we're allowed to use the official API key that telegram uses for their snap builds:
