diff --git PKGBUILD PKGBUILD
index 46b878b4..56546af0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,9 @@ b2sums=('2760600beeb00cdea417a21d4e0d9017c1c91e073e3569db7264e87107acf3ac2e657b5
 
 prepare() {
   cd $pkgname-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
