diff --unified --recursive --text a/patterns/dfa.c b/patterns/dfa.c
--- a/patterns/dfa.c	2021-09-03 19:56:30.720592244 +0000
+++ b/patterns/dfa.c	2021-09-03 19:57:01.100609820 +0000
@@ -288,19 +288,19 @@
   int i;
   int att, k;
 
-  fprintf(f, line);
+  fprintf(f, "%s", line);
   fprintf(f, " name : %s\n", pdfa->name);
   fprintf(f, " Nb states :  %7d, max= %d\n", pdfa->last_state + 1,
 	  pdfa->max_states);
   fprintf(f, " Nb Indexes : %7d, max= %d\n", pdfa->last_index,
 	  pdfa->max_indexes);
   fprintf(f, " memory needed : %d Mb\n", dfa_size(pdfa) / 1024);
-  fprintf(f, line);
+  fprintf(f, "%s", line);
 
   if (dfa_size(pdfa) > 10000) /* change this value if needed */
     return;
   fprintf(f, " state  |   .    |   O    |   X    |   #    |  att \n");
-  fprintf(f, line);
+  fprintf(f, "%s", line);
   for (i = 1; i != pdfa->last_state + 1; i++) {
     int *pnext = pdfa->states[i].next;
     fprintf(f, " %6d |", i);
@@ -318,7 +318,7 @@
       fprintf(f, " ...");
     fprintf(f, "\n");
   }
-  fprintf(f, line);
+  fprintf(f, "%s", line);
   fflush(f);
 }
 
diff --unified --recursive --text a/patterns/mkpat.c b/patterns/mkpat.c
--- a/patterns/mkpat.c	2021-09-03 19:56:30.720592244 +0000
+++ b/patterns/mkpat.c	2021-09-03 19:57:01.100609820 +0000
@@ -1419,22 +1419,22 @@
     /* A common case. Just use the labels as parameters. */
     switch (number_of_params) {
     case 0:
-      code_pos += sprintf(code_pos, autohelper_functions[funcno].code);
+      code_pos += sprintf(code_pos, "%d", autohelper_functions[funcno].code);
       break;
     case 1:
-      code_pos += sprintf(code_pos, autohelper_functions[funcno].code,
+      code_pos += sprintf(code_pos, "%d", autohelper_functions[funcno].code,
 			  varnames[0]);
       break;
     case 2:
-      code_pos += sprintf(code_pos, autohelper_functions[funcno].code,
+      code_pos += sprintf(code_pos, "%d", autohelper_functions[funcno].code,
 			  varnames[0], varnames[1]);
       break;
     case 3:
-      code_pos += sprintf(code_pos, autohelper_functions[funcno].code,
+      code_pos += sprintf(code_pos, "%d", autohelper_functions[funcno].code,
 			  varnames[0], varnames[1], varnames[2]);
       break;
     case 4:
-      code_pos += sprintf(code_pos, autohelper_functions[funcno].code,
+      code_pos += sprintf(code_pos, "%d", autohelper_functions[funcno].code,
 			  varnames[0], varnames[1], varnames[2],
 			  varnames[3]);
       break;
diff --unified --recursive --text a/patterns/uncompress_fuseki.c b/patterns/uncompress_fuseki.c
--- a/patterns/uncompress_fuseki.c	2021-09-03 19:56:30.720592244 +0000
+++ b/patterns/uncompress_fuseki.c	2021-09-03 19:57:01.100609820 +0000
@@ -224,9 +224,9 @@
 
   assert(boardsize > 0);
   if (boardsize > MAX_BOARD) {
-    printf(output_strings[PREAMBLE]);
+    printf("%s", output_strings[PREAMBLE]);
     printf(output_strings[HEADER], boardsize);
-    printf(output_strings[FOOTER]);
+    printf("%s", output_strings[FOOTER]);
     return EXIT_SUCCESS;
   }
   
@@ -251,7 +251,7 @@
     board[k][boardsize + 1] = '|';
   }
 
-  printf(output_strings[PREAMBLE]);
+  printf("%s", output_strings[PREAMBLE]);
   printf(output_strings[HEADER], boardsize);
   
 
@@ -308,7 +308,7 @@
   if (mode == C_OUTPUT)
     write_pattern_c_code(NULL, board1d, NO_MOVE, 0, boardsize, -1);
   
-  printf(output_strings[FOOTER]);
+  printf("%s", output_strings[FOOTER]);
 
   return EXIT_SUCCESS;
 }
