diff --git PKGBUILD PKGBUILD
index 1923663da..8d46b9fb2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,9 +12,16 @@ license=('GPL3')
 arch=('x86_64')
 depends=('ncurses')
 validpgpkeys=('67CCE2F5220D3E03024A175B68E7E09135C3EC3E')
-source=("https://ftp.gnu.org/gnu/gnugo/${pkgname}-${pkgver}.tar.gz"{,.sig})
+source=("https://ftp.gnu.org/gnu/gnugo/${pkgname}-${pkgver}.tar.gz"{,.sig}
+        "fix-format-not-a-string-literal-and-no-format-arguments.patch")
 sha256sums=('da68d7a65f44dcf6ce6e4e630b6f6dd9897249d34425920bfdd4e07ff1866a72'
-            'SKIP')
+            'SKIP'
+            '10c410eac93afe7590c20504c95f811fbc11aa0d46f24a186c62fa432f5b74d9')
+
+prepare(){
+	cd "${srcdir}/${pkgname}-${pkgver}"
+	patch -Np1 -i ../fix-format-not-a-string-literal-and-no-format-arguments.patch
+}
 
 build() {
 	cd "${srcdir}/${pkgname}-${pkgver}"
