diff --git PKGBUILD PKGBUILD
index 81182d2c..8f6b8a61 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,7 @@ pkgdesc='The IP Virtual Server administration utility'
 arch=('x86_64')
 url='http://www.linuxvirtualserver.org/software/ipvs.html'
 license=('GPL2')
-depends=('libnl1' 'popt' 'bash')
+depends=('libnl' 'popt' 'bash')
 backup=("etc/$pkgname.conf")
 options=('!makeflags')
 install=$pkgname.install
