--- PKGBUILD
+++ PKGBUILD
@@ -34,7 +34,7 @@ build() {
 	cd "$srcdir/$pkgname-$pkgver"
 
 	./Configure --prefix=/usr --openssldir=/etc/ssl --libdir=lib \
-		shared enable-ktls enable-ec_nistp_64_gcc_128 linux-${CARCH}
+		shared enable-ktls enable-ec_nistp_64_gcc_128 linux64-riscv64
 
 	make depend
 	make
@@ -47,7 +47,8 @@ check() {
 	# revert this patch for make test
 	patch -Rp1 -i "$srcdir/ca-dir.patch"
 
-	make HARNESS_JOBS=$(nproc) test
+	# https://github.com/openssl/openssl/issues/12242
+	make HARNESS_JOBS=$(nproc) TESTS=-test_afalg test
 
 	patch -Np1 -i "$srcdir/ca-dir.patch"
 	# re-run make to re-generate CA.pl from the patched .in file.
