--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,15 @@ depends=(
 )
 makedepends=(git)
 _tag=bb429dac1a686807c69b916f03dd843fa10b0927
-source=(git+https://github.com/wiseman/py-webrtcvad.git#tag=${_tag})
-b2sums=(SKIP)
+source=(git+https://github.com/wiseman/py-webrtcvad.git#tag=${_tag}
+        add-riscv64-build.patch)
+b2sums=('SKIP'
+        '7b1d6a6173cb50ca84b9339eb36f217fc4ab1cc134760821b7b62741055bae12bc96552a404526a7af03c72bf73835a1d3a20a5f3882a7f35480ba40097fa95a')
+
+prepare() {
+  cd py-webrtcvad
+  patch -Np1 -i ../add-riscv64-build.patch
+}
 
 build() {
   cd py-webrtcvad
