--- PKGBUILD
+++ PKGBUILD
@@ -19,14 +19,17 @@
   python-setuptools
   python-wheel
 )
-source=(git+https://github.com/wiseman/py-webrtcvad.git#tag=${pkgver})
-b2sums=('726b033c2cad75cdadb0822f9f6344d5504e0b8bf8a9725c048726cd3fa6663961419b7de1e3b26b1081797555dd77ceeda561785ce994e19a43dcd0daf5cf35')
+source=(git+https://github.com/wiseman/py-webrtcvad.git#tag=${pkgver}
+        add-riscv64-build.patch)
+b2sums=('726b033c2cad75cdadb0822f9f6344d5504e0b8bf8a9725c048726cd3fa6663961419b7de1e3b26b1081797555dd77ceeda561785ce994e19a43dcd0daf5cf35'
+        '7b1d6a6173cb50ca84b9339eb36f217fc4ab1cc134760821b7b62741055bae12bc96552a404526a7af03c72bf73835a1d3a20a5f3882a7f35480ba40097fa95a')
 
 prepare() {
   cd py-webrtcvad
   # replace pkg_resources with importlib.metadata for version lookup
   # https://github.com/wiseman/py-webrtcvad/pull/96
   git cherry-pick -n ca2377f2b6ac4c6e8d473e733dd8249302aef482
+  patch -Np1 -i ../add-riscv64-build.patch
 }
 
 build() {
