diff --git PKGBUILD PKGBUILD
index 3bdc4bd..fdc14e1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,12 @@ sha256sums=('0152d66a9d340b659fe9c880eb9190f3570fb477ac07cf52e8bcd134a1d30d70'
             'SKIP')
 validpgpkeys=('44BC9FD0D688EB007C4DD029E9CBDFC0ABC0A854') # Thomas Schmitt <scdbackup@gmx.net>
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd "${pkgname}"-${pkgver}
   ./configure --prefix=/usr \
