diff --git PKGBUILD PKGBUILD
index 87c88cf..b2d492a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,14 +16,17 @@ makedepends=('re2' 're2c' 'protobuf' 'php' 'php7' 'c-ares' 'openssl' 'chrpath' '
 checkdepends=('python-six')
 source=("https://github.com/grpc/grpc/archive/v$pkgver/$pkgbase-$pkgver.tar.gz"
         "https://github.com/google/googletest/archive/$_gtestver/googletest-$_gtestver.tar.gz"
+        cmake-pthread.patch::"https://github.com/grpc/grpc/commit/bf2ab4aa6e83c46b48f00be3a5fcd3a139f4fb62.patch"
         "python-grpcio-use-system-abseil.patch")
 sha256sums=('b74ce7d26fe187970d1d8e2c06a5d3391122f7bc1fdce569aff5e435fb8fe780'
             '443d383db648ebb8e391382c0ab63263b7091d03197f304390baac10f178a468'
+            '4aa5e60f49ff95ad9db682a108eeb298060a10bdb24d7a3604ddb04b7efc5c83'
             '3cfb289266f9d2697b917e4400f18a6592f95a99e658e401c1332870b219358b')
 
 prepare() {
   cd "$srcdir/$pkgbase-$pkgver"
   patch -Np1 -i ../python-grpcio-use-system-abseil.patch
+  patch -Np1 -i ../cmake-pthread.patch
 
   ln -sf "$srcdir/googletest-$_gtestver/"{googlemock,googletest} \
     third_party/googletest
