diff --git PKGBUILD PKGBUILD
index 54123ac..cf300fd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,7 @@ prepare() {
   cd "$srcdir/$pkgbase-$pkgver"
   patch -Np1 -i ../grpc-assert.patch
   patch -Np1 -i ../python-grpcio-use-system-abseil.patch
-  patch -Np1 -i ../benchmark-1.6.patch
+  # patch -Np1 -i ../benchmark-1.6.patch
 
   ln -sf "$srcdir/googletest-$_gtestver/"{googlemock,googletest} \
     third_party/googletest
