--- PKGBUILD
+++ PKGBUILD
@@ -15,14 +15,17 @@
              'cmake' 'benchmark' 'python-setuptools' 'abseil-cpp' 'ninja')
 source=("https://github.com/grpc/grpc/archive/v$pkgver/$pkgbase-$pkgver.tar.gz"
         "https://github.com/google/googletest/archive/$_gtestver/googletest-$_gtestver.tar.gz"
+        cmake-pthread.patch::"https://github.com/grpc/grpc/commit/bf2ab4aa6e83c46b48f00be3a5fcd3a139f4fb62.patch"
         "python-grpcio-use-system-abseil.patch")
 sha256sums=('9647220c699cea4dafa92ec0917c25c7812be51a18143af047e20f3fb05adddc'
             '443d383db648ebb8e391382c0ab63263b7091d03197f304390baac10f178a468'
+            '45edf5855ebc767e1882e27b85258a30e62d9881bd6dcece42058854d6fa78fe'
             '3cfb289266f9d2697b917e4400f18a6592f95a99e658e401c1332870b219358b')
 
 prepare() {
   cd "$srcdir/$pkgbase-$pkgver"
   patch -Np1 -i ../python-grpcio-use-system-abseil.patch
+  patch -Np1 -i ../cmake-pthread.patch
 
   ln -sf "$srcdir/googletest-$_gtestver/"{googlemock,googletest} \
     third_party/googletest
