--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,11 @@ depends=('glibc') # not because of compiling, but because of /usr/share/i18n/cha
 source=("https://download.savannah.gnu.org/releases/m17n/${pkgname}-${pkgver}.tar.gz")
 sha256sums=('657f23835b6655e7a63a362bac66260454ee356da4855eadb9366911d33fdc6d')
 
+prepare() {
+  cd ${pkgname}-${pkgver}
+  autoreconf -fiv
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   ./configure --prefix=/usr
