diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,10 @@ optdepends=('rpm: RHEL/CentOS based image support')
 makedepends=('go' 'btrfs-progs')
 provides=('trivy')
 conflicts=('trivy')
-source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/v${pkgver}.tar.gz")
-b2sums=('8267fb66afc7951b16d65d3abaab4af6cdf3f1c3d1dc7cbadeb47624b498e240341d558570912b41765a0e574ce93baf35af0480e29c500a816c158c255dcfd9')
+source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/v${pkgver}.tar.gz"
+        fix-libc.patch)
+b2sums=('a4e409b34d91fc8ea3d0efab069518179ed7195ee956064255d14044c7115ea7cfbff72a77a7bdd18c45fd13e6769bef31997079e67916fee16b34fc433e42a7'
+        '385c02c2e8d6e783b2718aa53340094ef7444a2567f294f0e0f1399c0982df87c12f2fccb6c35fe1cf8ae8a896268e1e30c8bc211b40ece0a712c1c40f7c4b27')
 
 prepare() {
   cd ${pkgname}-${pkgver}
@@ -23,6 +25,10 @@ prepare() {
     -exec sed "s|\(../\)*imgs/\(.*\.[a-z]\{3\}\)|${url}/raw/main/docs/imgs/\2|g" -i {} \;
 
   rm -Rf docs/{build,imgs}
+
+  patch -p0 -Ni ../fix-libc.patch
+  go get -d modernc.org/libc
+  go get -d modernc.org/sqlite
 }
 
 build() {
