diff --git PKGBUILD PKGBUILD
index 8e54d8f..37c63be 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,8 @@ arch=('x86_64')
 license=('LGPL')
 depends=('gcc-libs' 'openssl')
 source=(http://live555.com/liveMedia/public/live.$pkgver.tar.gz)
-sha256sums=('e6c2ebfec6772c1f37084271e4e5236abc7f0e1fc4489f9869763440200ddcc2')
-sha512sums=('62aeacea8f987731237c5924097a2aa4f650629f44d190637b2c960425f295b1360fa27ecdfe96636e7f29a23385ed872f54b66b402c38e3e76f6564f3cdc38f')
+sha256sums=('273d9cf77468015c2377e50e384e3eea67cf255c75fe60a175e9e0917afe847b')
+sha512sums=('e24427633dca4222d64606ef7a0723034c3c378745654b8adb39aef67da98e0f6ef2391e08270cb971f6f577d5a3b09060f38611f94193f143ddc58217ad5ab9')
 
 prepare() {
   cd live
