--- PKGBUILD
+++ PKGBUILD
@@ -138,6 +138,8 @@ prepare() {
 
     # fix build with Poppler 24.12 (LFS patch)
     patch -Np1 -i "${srcdir}"/poppler24.12-buildfix.patch
+
+    patch -p1 -i ../riscv64-support.patch
 }
 
 build() {
@@ -425,3 +427,6 @@ package_libreoffice-fresh() {
     'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
         > "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=(riscv64-support.patch)
+sha256sums+=('c5615aa1437a2a3ac7fbba749dc7ee0c0b3406346d501e500837ed1f68878c23')
