diff --git PKGBUILD PKGBUILD
index 9cd08f2f..fd89d794 100644
--- PKGBUILD
+++ PKGBUILD
@@ -62,7 +62,8 @@ source=(${_mirror}/libreoffice{,-help,-translations}-${_LOver}.tar.xz{,.asc}
 	make-pyuno-work-with-system-wide-module-install.diff
         623ea5c.diff
 	soffice-template.desktop.in 
-	libreoffice-fresh.sh libreoffice-fresh.csh)
+	libreoffice-fresh.sh libreoffice-fresh.csh
+	"riscv64-support.patch")
 noextract=(35c94d2df8893241173de1d16b6034c0-swingExSrc.zip
            798b2ffdc8bcfe7bca2cf92b62caf685-rhino1_5R5.zip
            a7983f859eafb2677d7ff386a023bc40-xsltml_2.1.2.zip
@@ -113,7 +114,8 @@ sha256sums=('f5120fa3ae80a7deb08150038a6e78335bcab578bd7c41ddc733c2057977003b'
             '440c9af5f3d1213d8ed7177282380f25cbc981cabc8b590dcb777aaae84178e5'
             'd0be8099cbee3c9dfda694a828149b881c345b204ab68826f317580aafb50879'
             'b43ed267643fc5ced803dca010427b12b1f10db485173ccb19efb3395e60c82e'
-            '66f2cb5d2ff9909ee9633aea73d5306fc8c4ff358fa526f45d9994210d3e23ff')
+            '66f2cb5d2ff9909ee9633aea73d5306fc8c4ff358fa526f45d9994210d3e23ff'
+            '3f05bb4d781d331c5cfa2600866be7e10c6b23406ce890c3db741a2db76e79e2')
 
 prepare() {
 
@@ -144,6 +146,8 @@ prepare() {
 		esac
 		ARCH_FLAGS="$ARCH_FLAGS $i"
 	done
+
+	patch -Np1 -i "${srcdir}"/riscv64-support.patch
 }
 
 build() {
@@ -184,7 +188,7 @@ build() {
 		--enable-gtk3 \
 		--enable-gtk4 \
 		--enable-introspection \
-		--enable-lto \
+		--disable-lto \
 		--enable-openssl \
 		--enable-odk\
 		--enable-python=system \
