--- PKGBUILD
+++ PKGBUILD
@@ -66,7 +66,8 @@ source=(${_mirror}/libreoffice{,-help,-translations}-${_LOver}.tar.xz{,.asc}
     libreoffice-7.5.8.2-icu-74-compatibility.patch
     kf6_buildfix.diff
     soffice-template.desktop.in 
-    libreoffice-fresh.sh libreoffice-fresh.csh)
+    libreoffice-fresh.sh libreoffice-fresh.csh
+    "riscv64-support.patch")
 noextract=(35c94d2df8893241173de1d16b6034c0-swingExSrc.zip
            798b2ffdc8bcfe7bca2cf92b62caf685-rhino1_5R5.zip
            a7983f859eafb2677d7ff386a023bc40-xsltml_2.1.2.zip
@@ -121,7 +122,8 @@ sha256sums=('3ccf577e8f665059ed5a06577b3b37278080be9f29cc4ad3352857a8f2549fa8'
             'f7905962a634ab4ab5c5e591c1816e6baf3f66db91adfeee25c631b6939110ef'
             'd0be8099cbee3c9dfda694a828149b881c345b204ab68826f317580aafb50879'
             '5e1dc35b9851cf2ef93a2fae151cf90258c81f3ab56f86e3e9ea6c56be4eed19'
-            '57c92c48e0d82a073a0095bfe44fb541c68bead6205436a5432879185f420768')
+            '57c92c48e0d82a073a0095bfe44fb541c68bead6205436a5432879185f420768'
+            '3f05bb4d781d331c5cfa2600866be7e10c6b23406ce890c3db741a2db76e79e2')
 prepare() {
 
     cd libreoffice-$_LOver
@@ -146,6 +148,8 @@ prepare() {
     # fix KF& build - https://gerrit.libreoffice.org/c/core/+/163237
     patch -Np1 -i "${srcdir}"/kf6_buildfix.diff
 
+    patch -Np1 -i "${srcdir}"/riscv64-support.patch
+
     #use the CFLAGS but remove the LibO overridden ones
     for i in $CFLAGS; do
         case "$i" in
@@ -202,7 +206,7 @@ build() {
         --enable-gtk3 \
         --enable-gtk4 \
         --enable-introspection \
-        --enable-lto \
+        --disable-lto \
         --enable-openssl \
         --enable-odk\
         --enable-python=system \
