diff --git PKGBUILD PKGBUILD
index 9863ade01..2e4a61c44 100644
--- PKGBUILD
+++ PKGBUILD
@@ -62,7 +62,8 @@ source=(${_mirror}/libreoffice{,-help,-translations}-${_LOver}.tar.xz{,.asc}
 	${_additional_source_url2}/185d60944ea767075d27247c3162b3bc-unowinreg.dll
 	make-pyuno-work-with-system-wide-module-install.diff
 	soffice-template.desktop.in 
-	libreoffice-fresh.sh libreoffice-fresh.csh)
+	libreoffice-fresh.sh libreoffice-fresh.csh
+	"riscv64-support.patch")
 noextract=(35c94d2df8893241173de1d16b6034c0-swingExSrc.zip
            798b2ffdc8bcfe7bca2cf92b62caf685-rhino1_5R5.zip
            a7983f859eafb2677d7ff386a023bc40-xsltml_2.1.2.zip
@@ -114,7 +115,8 @@ sha256sums=('82c1ffe02f2ed8ee1451a9f67c35335b27a638db591f9bc539eff86ab14dd95a'
             'c463654a73ecfbc242ff109726fb4faecdbfb3d91affafe919b24bea65afb563'
             'd0be8099cbee3c9dfda694a828149b881c345b204ab68826f317580aafb50879'
             'b43ed267643fc5ced803dca010427b12b1f10db485173ccb19efb3395e60c82e'
-            '66f2cb5d2ff9909ee9633aea73d5306fc8c4ff358fa526f45d9994210d3e23ff')
+            '66f2cb5d2ff9909ee9633aea73d5306fc8c4ff358fa526f45d9994210d3e23ff'
+            '74f7e8e5ef1a6e7db299310571ac4840efc593732611683c954bd373a901c7a7')
 
 prepare() {
 
@@ -142,6 +144,8 @@ prepare() {
 		esac
 		ARCH_FLAGS="$ARCH_FLAGS $i"
 	done
+
+	patch -Np1 -i "${srcdir}"/riscv64-support.patch
 }
 
 build() {
@@ -174,7 +178,7 @@ build() {
 		--enable-gtk3 \
 		--enable-gtk4 \
 		--enable-introspection \
-		--enable-lto \
+		--disable-lto \
 		--enable-openssl \
 		--enable-odk\
 		--enable-python=system \
