--- PKGBUILD
+++ PKGBUILD
@@ -137,6 +137,8 @@ prepare() {
     export CXXFLAGS="$ARCH_FLAGS"
 
     patch -p1 -i ../poppler-24.10.patch # Fix build with Poppler 24.10 (KaoS patch)
+
+    patch -p1 -i ../riscv64-support.patch
 }
 
 build() {
@@ -424,3 +426,6 @@ package_libreoffice-fresh() {
     'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
         > "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=(riscv64-support.patch)
+sha256sums+=('c5615aa1437a2a3ac7fbba749dc7ee0c0b3406346d501e500837ed1f68878c23')
