--- PKGBUILD
+++ PKGBUILD
@@ -131,6 +131,8 @@
     done
     popd
 
+    patch -Np1 -i ../riscv64-support.patch
+
     #use the CFLAGS but remove the LibO overridden ones
     for i in $CFLAGS; do
         case "$i" in
@@ -415,3 +417,6 @@
     'sid','sk','sl','sq','sv','ta','tg','tr','ug','uk','vi','zh-CN','zh-TW'])" \
         > "${pkgdir}"/usr/lib/libreoffice/help/languages.js
 }
+
+source+=(riscv64-support.patch)
+sha256sums+=('80862c36a73917df91aed2851e683701a3ac92f00be5d5bdc5b2d788d100195c')
