--- PKGBUILD
+++ PKGBUILD
@@ -50,6 +50,7 @@ prepare() {
   ln -s nemequ-munit munit
 
   cd libei
+  patch -Np1 -i ../increase-valgrind-timeout.patch
 }
 
 build() {
@@ -92,4 +93,7 @@ package_libei-docs() {
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 libei/COPYING
 }
 
+source+=(increase-valgrind-timeout.patch)
+b2sums+=('f904bd3e392c739f4ec330c6ae67d90a2c725182074ef38f3c2f4022e628f5daf69ea0c8967c26ebe18cfc62f5a49092a859b58a07964cdef88e779a4cd757fc')
+
 # vim:set sw=2 sts=-1 et:
