--- PKGBUILD
+++ PKGBUILD
@@ -50,6 +50,8 @@ prepare() {
   ln -s nemequ-munit munit
 
   cd libei
+  # Backport fix for a valgrind protocol test race.
+  patch -Np1 -i ../fix-valgrind-protocol-test-race.patch
 }
 
 build() {
@@ -92,4 +94,7 @@ package_libei-docs() {
   install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 libei/COPYING
 }
 
+source+=(fix-valgrind-protocol-test-race.patch)
+b2sums+=('93744120f6fa7087496fb86c216cc184e9cd3f897eb51ac3e629c3ffa4b4b660a366d91a17cd7acaa1c5bbb4109e9b32b9a0a30ec9aa1cbcbe5ae067b9301e06')
+
 # vim:set sw=2 sts=-1 et:
