--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,7 @@
     sed -e 's/LIB_SUFFIX=.*/LIB_SUFFIX=/g' -i install/linux/install.sh
     sed -e "s@=\$(which lazbuild)@=\"\$(which lazbuild) --lazarusdir=$srcdir/lazarus\"@" -i build.sh
     sed -e '/doublecmd.zdli/d' -i install/linux/install.sh
+    patch -Np1 -i "$srcdir/fix-riscv64-time-type.patch"
 }
 
 build() {
@@ -71,3 +72,6 @@
     cd "$srcdir/$pkgbase-qt6"
     ./install/linux/install.sh --install-prefix="$pkgdir"
 }
+
+source+=('fix-riscv64-time-type.patch')
+sha512sums+=('5299c0566506d087766f7ad9737352b3f408672795d89805634446eca82fb5635d4255c604bbf13c4830ef20f1cda967af4aaeb8b7bb9df8e17ab0b7ab3875d2')
