--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,11 @@ source=(https://sourceforge.net/projects/libwpd/files/libodfgen/libodfgen-$pkgve
 sha1sums=('9b36d313e64c6ce1026485d521fbdeff7f47594c')
 sha256sums=('55200027fd46623b9bdddd38d275e7452d1b0ff8aeddcad6f9ae6dc25f610625')
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr --disable-werror
