--- PKGBUILD
+++ PKGBUILD
@@ -51,7 +51,7 @@ source=(
   'add-generated-protobuf-files.patch'
   'use-system-python-packages.patch'
   'add-translations.patch'
-  'add-go.sum-file.patch'
+  'riscv-add-go.sum-file.patch'
   'template-version-strings.patch'
   'tmpfiles.conf'
 )
@@ -62,7 +62,7 @@ b2sums=('SKIP'
         '88f7d6a4cacffd0694a7346ceef5e7e81234bc7ef7c4870cd6f73c810a8f6e807134f798d8934458bdf011354adc79397a3b53654fd606a97bbef83d0dfadf5f'
         '5b1ed289ccfb6598941667065a4e533631f374a43793765044ee9f655ac2331595670e383821cf6b7a4114f117905de4895ba2bb430408011cc384821d53355c'
         'c2d2642c5de8b50aee3923dce09faedbe4db63ec2369894efceb8bc4237759190c8ea4cade375262623153c13d814b4196915687cd33886b22e0b33cc501b09d'
-        'f92122e9df08b720b63dbf023e81c1b0f660ff88149174efab8603ebabd5913dc4056a19b3e43202e1e4e68431ce52bee9eee2b0b32b8a22d327358deaf87a27'
+        '6efbd3bbd2e4f56cc5089d94802abee48950294bff2b31d53989fe3acf06ec8a4f3786ffe162e02c1edb0071542c81b9bbbc53b1fe945b803ead359700779d85'
         '093ecf4398ffcd9f4077b41174e93707ba90cef717fa7e35a44557a52969d356216b08f6a4ce38fc93a7ba4e8cb14c7c2a4368ba0b023527d1ee93879728b30d'
         '81dbb6ba72d5bf5a0d9fa3dea33b87f03604c1a2537a4ca47b45923ab186e68d5c24b043691a857948ae404409dc3cc66a7ef004ee20dc2451aa5190fe6479bb')
 
@@ -89,8 +89,9 @@ prepare() {
   # use system python packages
   patch -p1 -i "$srcdir/use-system-python-packages.patch"
 
-  # get go.sum file from 'go mod tidy'
-  patch -p1 -i "$srcdir/add-go.sum-file.patch"
+  # apply 'add-generated-protobuf-files.patch' first,
+  # then get go.sum file from 'go mod tidy'
+  patch -p1 -i "$srcdir/riscv-add-go.sum-file.patch"
 
   # add generated translation files
   # required for reproducible builds
