--- PKGBUILD
+++ PKGBUILD
@@ -81,6 +81,9 @@
 prepare() {
   cd "$pkgname"
 
+  # fix eBPF compilation with Linux 6.19+
+  patch -Np1 -i "$srcdir/kernel-6.19-build.patch"
+
   # TODO file an upstream bug
   # * fix an issue with setup.py installing to python's site-packages instead
   # of /usr
@@ -149,7 +152,8 @@
   local KDIR="/usr/src/linux"
   # we set -fno-stack-protector here to work around a clang regression
   # this is fine - bpf programs do not use stack protectors
-  CLANG="clang -fno-stack-protector" ARCH="$CARCH" KERNEL_DIR="$KDIR" KERNEL_HEADERS="$KDIR" EXTRA_FLAGS="$CFLAGS" make
+  # Pass EXTRA_FLAGS to make so its defaults do not replace Arch's CFLAGS.
+  CLANG="clang -fno-stack-protector" ARCH="$CARCH" KERNEL_DIR="$KDIR" KERNEL_HEADERS="$KDIR" make EXTRA_FLAGS="$CFLAGS -Wno-microsoft-anon-tag -fms-extensions"
   popd
 }
 
@@ -201,3 +205,7 @@
   local site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
   rm -rf "$pkgdir/$site_packages/tests"
 }
+
+source+=(kernel-6.19-build.patch::https://github.com/evilsocket/opensnitch/commit/614537c92ec82f54f76a45fb406ad2fb6e6fa618.patch)
+sha512sums+=('98def3c0158b9e5f9e3c3c1bed28d970cc3cdd8f6c33799cfc496105404f01af303e6e3cd583b643954276137be0a0eb95534722ede94ff83696d40ba869b8ea')
+b2sums+=('eca89e3e1bea84803294a4b9c01a6d1fbfa83c42ebb9ca8b63dc3a41d09fee26fb524fae82a92eb547959973451bbea2413aefd59fa7d15e91194b1b04065ea2')
