diff --git PKGBUILD PKGBUILD
index 4791ef8..ce598af 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,8 @@ sha256sums=('fd0d824fd94bef3aae55b95e5ef19342b653fe3ff4fb561364e942c809eb616b')

 build() {
   cd cxsc-${pkgver//./-}
+  # fix error: ISO C++17 does not allow dynamic exception specifications
+  find . -type f -print0 | xargs -0 sed -ri 's@([)]+ +|^[ \t]+)throw[(][^)]+[)]@\1noexcept(false)@g'
   autoreconf -vi
   ./configure --prefix=/usr --libexecdir=/usr/lib
   make
