--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@ depends=('boost-libs' 'openssl' 'mariadb-libs')
 optdepends=('ca-certificates: server certificate verification'
             'mariadb: advanced user management')
 makedepends=('cmake' 'boost' 'openssl' 'mariadb-libs')
-checkdepends=('openssl' 'python' 'curl' 'netcat')
+checkdepends=('openssl' 'python' 'curl' 'openbsd-netcat')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/trojan-gfw/$pkgname/archive/v$pkgver.tar.gz")
 backup=('etc/trojan.json'
         'etc/trojan/config.json')
