--- PKGBUILD
+++ PKGBUILD
@@ -37,7 +37,7 @@
   rapidjson
 )
 source=(
-  "$pkgname::git+https://git.dev.opencascade.org/repos/occt.git#tag=V${pkgver//./_}"
+  "$pkgname::git+https://github.com/Open-Cascade-SAS/OCCT.git#tag=V${pkgver//./_}"
   'opencascade.sh'
 )
 sha512sums=('8f87d0ecf7fe448ebd85cf2f6dce900f168e45ee0af30e6d03029c7a53a565307a958c5032c7d563a973c7410612f077b36dfbece89c165ea34eabccce2ca707'
