diff --unified --recursive --text a/dpf/dgl/Makefile.mk b/dpf/dgl/Makefile.mk
--- a/dpf/dgl/Makefile.mk	2021-10-12 13:01:09.573443248 +0000
+++ b/dpf/dgl/Makefile.mk	2021-10-12 13:04:00.486890824 +0000
@@ -25,6 +25,10 @@
 BASE_FLAGS = -Wall -Wextra -pipe -MD -MP
 BASE_OPTS  = -O2 -mtune=generic -msse -msse2 -fdata-sections -ffunction-sections
 
+ifneq ($(findstring riscv64, $(shell uname -a)),)
+# RISCV flags
+BASE_OPTS = -O2 -fdata-sections -ffunction-sections
+endif
 ifeq ($(MACOS),true)
 # MacOS linker flags
 LINK_OPTS  = -fdata-sections -ffunction-sections -Wl,-dead_strip -Wl,-dead_strip_dylibs
diff --unified --recursive --text a/Makefile.mk b/Makefile.mk
--- a/Makefile.mk	2021-10-12 13:39:37.316992538 +0000
+++ b/Makefile.mk	2021-10-12 13:41:26.783268125 +0000
@@ -40,6 +40,12 @@
 BASE_OPTS += -mfpmath=sse
 endif
 
+ifneq ($(findstring riscv64, $(shell uname -a)),)
+# RISCV support
+BASE_OPTS  = -O2 -ffast-math -fdata-sections -ffunction-sections
+endif
+
+
 ifeq ($(MACOS),true)
 # MacOS linker flags
 LINK_OPTS  = -fdata-sections -ffunction-sections -Wl,-dead_strip -Wl,-dead_strip_dylibs
