--- a/ghidra-Ghidra_12.1.2_build.orig/GPL/DemanglerGnu/build.gradle
+++ b/ghidra-Ghidra_12.1.2_build/GPL/DemanglerGnu/build.gradle
@@ -80,6 +80,7 @@
 			targetPlatform "win_arm_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_riscv_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
@@ -104,6 +105,7 @@
 			targetPlatform "win_arm_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_riscv_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
--- a/ghidra-Ghidra_12.1.2_build.orig/GPL/GnuDisassembler/buildGdis.gradle
+++ b/ghidra-Ghidra_12.1.2_build/GPL/GnuDisassembler/buildGdis.gradle
@@ -4,7 +4,7 @@
 
 defaultTasks 'assemble'
 
-ext.supportedPlatforms = ['mac_x86_64', 'mac_arm_64', 'linux_x86_64', 'linux_arm_64']
+ext.supportedPlatforms = ['mac_x86_64', 'mac_arm_64', 'linux_x86_64', 'linux_arm_64', 'linux_riscv_64']
 
 ext.binutilsResource = new File("${binutilsLocation}/${binutils}.tar.bz2")
 
@@ -57,6 +57,9 @@
 	tasks.compileGdisLinux_arm_64ExecutableGdisC {
 		dependsOn 'copyBinutilsArtifcats_linux_arm_64'
 	}
+	tasks.compileGdisLinux_riscv_64ExecutableGdisC {
+		dependsOn 'copyBinutilsArtifcats_linux_riscv_64'
+	}
 	
 }
 
--- a/ghidra-Ghidra_12.1.2_build.orig/GPL/nativePlatforms.gradle
+++ b/ghidra-Ghidra_12.1.2_build/GPL/nativePlatforms.gradle
@@ -24,6 +24,7 @@
 	[name: "win_arm_64", os: "windows", arch: "arm64"],
 	[name: "linux_x86_64", os: "linux", arch: "x86_64"],
 	[name: "linux_arm_64", os: "linux", arch: "arm64"],
+	[name: "linux_riscv_64", os: "linux", arch: "riscv64"],
 	[name: "mac_x86_64", os: "osx", arch: "x86_64"],
 	[name: "mac_arm_64", os: "osx", arch: "arm64"],
 	[name: "freebsd_x86_64", os: "freebsd", arch: "x86_64"],
@@ -77,6 +78,9 @@
 		case "arm64":
 			arch = "arm_64"
 			break
+		case "riscv64":
+			arch = "riscv_64"
+			break
 		default:
 			throw new GradleException("Unrecognized platform architecture: $arch" )
 	}
@@ -172,6 +176,14 @@
 	return isArm_64(getCurrentPlatformName())
 }
 
+ext.isRiscv_64 = { platform_name ->
+	return platform_name.contains("riscv")
+}
+
+ext.isCurrentRiscv_64 = {
+	return isRiscv_64(getCurrentPlatformName())
+}
+
 /****************************************************************************
  * Establish Visual Studio configuration environment for Windows native builds
  * NOTE: vsconfig.gradle path is relative to each GPL project module
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Debug/Debugger-rmi-trace/build.gradle
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Debug/Debugger-rmi-trace/build.gradle
@@ -41,10 +41,16 @@
 	dependsOn(configurations.protocArtifact)
 
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!isCurrentWindows()) {
-			exe.setExecutable(true)
+		def exe
+		if (isCurrentRiscv_64()) {
+			exe = System.env.PROTOBUF_DESTDIR + "/usr/local/bin/protoc"
+		} else {
+			exe = configurations.protocArtifact.first()
+			if (!isCurrentWindows()) {
+				exe.setExecutable(true)
+			}
 		}
+
 		generateProtoPy.commandLine exe
 		generateProtoPy.args "--python_out=${generateProtoPy.outdir}"
 		generateProtoPy.args "--pyi_out=${generateProtoPy.stubsOutdir}"
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Features/BSim/support/make-postgres.sh
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Features/BSim/support/make-postgres.sh
@@ -98,6 +98,8 @@
 	OSDIR="linux_x86_64"
 elif [ "$ARCH" = "aarch64" ]; then
 	OSDIR="linux_arm_64"
+elif [ "$ARCH" = "riscv64" ]; then
+	OSDIR="linux_riscv_64"
 else
 	echo "Unsupported platform: $OS $ARCH"
 	exit -1	
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Features/Decompiler/buildNatives.gradle
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Features/Decompiler/buildNatives.gradle
@@ -41,6 +41,7 @@
 			targetPlatform "win_arm_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_riscv_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Features/FileFormats/buildNatives.gradle
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Features/FileFormats/buildNatives.gradle
@@ -32,6 +32,7 @@
 			targetPlatform "win_arm_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_riscv_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java
@@ -21,6 +21,7 @@
 	POWERPC("ppc"),
 	POWERPC_64("ppc64"),
 	ARM_64("aarch64", "arm64"),
+	RISCV_64("riscv64"),
 	UNKNOWN("Unknown Architecture");
 
 	/**
--- a/ghidra-Ghidra_12.1.2_build.orig/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java
+++ b/ghidra-Ghidra_12.1.2_build/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java
@@ -56,6 +56,11 @@
 	LINUX_ARM_64(OperatingSystem.LINUX, Architecture.ARM_64, "linux_arm_64", ".so", ""),
 
 	/**
+	 * Identifies a Linux RISC-V 64-bit OS.
+	 */
+	LINUX_RISCV_64(OperatingSystem.LINUX, Architecture.RISCV_64, "linux_riscv_64", ".so", ""),
+
+	/**
 	 * Identifies a macOS x86 32-bit OS.
 	 */
 	MAC_X86_32(OperatingSystem.MAC_OS_X, Architecture.X86, "mac_x86_32", ".dylib", ""),
--- a/ghidra-Ghidra_12.1.2_build.orig/gradle/hasProtobuf.gradle
+++ b/ghidra-Ghidra_12.1.2_build/gradle/hasProtobuf.gradle
@@ -28,29 +28,31 @@
 
 
 dependencies {
-	allProtocArtifacts "com.google.protobuf:protoc:${version}:windows-x86_64@exe"
-	allProtocArtifacts "com.google.protobuf:protoc:${version}:linux-x86_64@exe"
-	allProtocArtifacts "com.google.protobuf:protoc:${version}:linux-aarch_64@exe"
-	allProtocArtifacts "com.google.protobuf:protoc:${version}:osx-x86_64@exe"
-	allProtocArtifacts "com.google.protobuf:protoc:${version}:osx-aarch_64@exe"
+	if (!isCurrentRiscv_64()) {
+		allProtocArtifacts "com.google.protobuf:protoc:${version}:windows-x86_64@exe"
+		allProtocArtifacts "com.google.protobuf:protoc:${version}:linux-x86_64@exe"
+		allProtocArtifacts "com.google.protobuf:protoc:${version}:linux-aarch_64@exe"
+		allProtocArtifacts "com.google.protobuf:protoc:${version}:osx-x86_64@exe"
+		allProtocArtifacts "com.google.protobuf:protoc:${version}:osx-aarch_64@exe"
 
-	if (isCurrentWindows()) {
-		protocArtifact "com.google.protobuf:protoc:${version}:windows-x86_64@exe"
-	}
-	if (isCurrentLinux()) {
-		if (platform.endsWith("x86_64")) {
-			protocArtifact "com.google.protobuf:protoc:${version}:linux-x86_64@exe"
-		}
-		else {
-			protocArtifact "com.google.protobuf:protoc:${version}:linux-aarch_64@exe"
+		if (isCurrentWindows()) {
+			protocArtifact "com.google.protobuf:protoc:${version}:windows-x86_64@exe"
 		}
-	}
-	if (isCurrentMac()) {
-		if (platform.endsWith("x86_64")) {
-			protocArtifact "com.google.protobuf:protoc:${version}:osx-x86_64@exe"
+		if (isCurrentLinux()) {
+			if (platform.endsWith("x86_64")) {
+				protocArtifact "com.google.protobuf:protoc:${version}:linux-x86_64@exe"
+			}
+			else {
+				protocArtifact "com.google.protobuf:protoc:${version}:linux-aarch_64@exe"
+			}
 		}
-		else {
-			protocArtifact "com.google.protobuf:protoc:${version}:osx-aarch_64@exe"
+		if (isCurrentMac()) {
+			if (platform.endsWith("x86_64")) {
+				protocArtifact "com.google.protobuf:protoc:${version}:osx-x86_64@exe"
+			}
+			else {
+				protocArtifact "com.google.protobuf:protoc:${version}:osx-aarch_64@exe"
+			}
 		}
 	}
 }
@@ -65,9 +67,14 @@
 	dependsOn(configurations.protocArtifact)
 
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!isCurrentWindows()) {
-			exe.setExecutable(true)
+		def exe
+		if (isCurrentRiscv_64()) {
+			exe = System.env.PROTOBUF_DESTDIR + "/usr/local/bin/protoc"
+		} else {
+			exe = configurations.protocArtifact.first()
+			if (!isCurrentWindows()) {
+				exe.setExecutable(true)
+			}
 		}
 		generateProto.commandLine exe, "--java_out=${generateProto.outdir}", "-I${generateProto.srcdir}"
 		generateProto.args generateProto.src
