--- PKGBUILD
+++ PKGBUILD
@@ -75,6 +75,12 @@
             'e611f0a84023aee72016a8bb73a627e4d3b3fc28201fe2b60ccc8d6628fc2e7005b9edc24e35e489311fd6fa9ed1a35d5e230e2657a77b331dbf7cd6773ae957')
 
 prepare() {
+  # prepare for protobuf
+  mkdir protobuf-dst
+  cd protobuf-${_protobuf_cpp}
+  ./configure --disable-shared
+  cd ..
+
   cd ghidra-Ghidra_${pkgver}_build
 
   # Check dependency expectations
@@ -108,8 +114,10 @@
   sed -i '/FileTree tree/a\\t\texclude "yajsw-stable-**.zip"\n\t\texclude "src/main/fidb/**.fidb"\n\t\texclude "os/linux64/**"' gradle/support/ip.gradle
 
   # z3 dependencies
-  install -Dm 644 /usr/lib/libz3* -t dependencies/SymbolicSummaryZ3/os/linux_x86_64
+  install -Dm 644 /usr/lib/libz3* -t dependencies/SymbolicSummaryZ3/os/linux_riscv_64
   install -Dm 644 /usr/share/java/com.microsoft.z3.jar -t dependencies/flatRepo
+
+  patch -Np2 -i ../add-riscv64-build.patch
 }
 
 _check_dep() {
@@ -121,6 +129,14 @@
 }
 
 build() {
+  # protoc
+  export PROTOBUF_DESTDIR=${srcdir}/protobuf-dst
+  cd protobuf-${_protobuf_cpp}
+  make
+  # make check
+  make DESTDIR=${PROTOBUF_DESTDIR} install
+  cd ..
+
   cd ghidra-Ghidra_${pkgver}_build
 
   export GRADLE_HOME="$(pwd)/usr/share/java/gradle/"
@@ -136,7 +152,7 @@
 
   # Ghidra build
   gradle yajswDevUnpack
-  gradle buildNatives_linux_x86_64
+  gradle buildNatives_linux_riscv_64
   gradle sleighCompile
   gradle buildGhidra
 }
@@ -163,7 +179,7 @@
   install -d "${pkgdir}"/{opt,usr/bin}
   _appver=$(grep -oP '(?<=^application.version=).*$' Ghidra/application.properties)
   _relname=$(grep -oP '(?<=^application.release.name=).*$' Ghidra/application.properties)
-  bsdtar xf "build/dist/ghidra_${_appver}_${_relname}"_*_linux_x86_64.zip -C "${pkgdir}"/opt
+  bsdtar xf "build/dist/ghidra_${_appver}_${_relname}"_*_linux_riscv_64.zip -C "${pkgdir}"/opt
 
   # Simplify some directory and binary names
   mv "${pkgdir}"/opt/ghidra{_*,}
@@ -181,4 +197,11 @@
   done
 }
 
+_protobuf_ver=21.8
+_protobuf_cpp=3.$_protobuf_ver
+source+=(https://github.com/protocolbuffers/protobuf/releases/download/v${_protobuf_ver}/protobuf-cpp-${_protobuf_cpp}.tar.gz
+         add-riscv64-build.patch)
+sha512sums+=('ef4939dcd7d10a602f95cb8d6fdb5d31c53bd39f18678572f97d443be01093e05a69b6756e8431a27200ad30decf2a007274a5b90ecaecbb5d8e341008bba175'
+             '6beeb1e840bbb97380fd7d81ddd893d5d395e655fb2b78e3835cba8afd37198f87c1f9821f8d6dd87d2f4b94523ed3cb8db3cbcfb5f96fee298178b574e2445b')
+
 # vim: ts=2 sw=2 et:
