diff --git PKGBUILD PKGBUILD
index a17236b..fd218ef 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,8 @@ source=("git+${url}.git#tag=${pkgver}")
 prepare() {
   # Fix build with GCC 15
   git -C ${pkgname} cherry-pick -n 7b469b4220f96fb3d036cf68cd7bd30bd39e61d2
+  # Fix narrowing conversion on platforms with unsigned char (riscv64)
+  git -C ${pkgname} cherry-pick -n fcbb8193b94921e058be7b563aea053531e5b2d9
   # Don't install embedded copy of GTest
   git -C ${pkgname} cherry-pick -n f878043f12a9434a2ef72de3992ec92e6845c889
 }
