--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@
 # NOTE: requires rebuilt with each new gcc version
 
 pkgbase=libtool
-pkgname=(libtool lib32-libltdl)
+pkgname=(libtool)
 pkgver=2.6.0+r23+gb08cb0a0
 _commit=b08cb0a06da153372dac8d184d24e7edfff2a209
 pkgrel=1
@@ -24,8 +24,6 @@
   "gcc>=$_gccver"
   git
   glibc
-  lib32-gcc-libs
-  lib32-glibc
   help2man
 )
 checkdepends=(
@@ -54,20 +52,12 @@
   git -c protocol.file.allow=always submodule update
 
   ./bootstrap
-
-  cp -ar "${srcdir}"/libtool "${srcdir}"/libtool32
 }
 
 build() {
   cd libtool
   ./configure --prefix=/usr lt_cv_sys_lib_dlsearch_path_spec="/usr/lib /usr/lib32"
   make
-
-  #lib32
-  cd "${srcdir}"/libtool32
-  export CC="gcc -m32" CXX="g++ -m32"
-  ./configure --prefix=/usr lt_cv_sys_lib_dlsearch_path_spec="/usr/lib /usr/lib32" --libdir=/usr/lib32
-  make
 }
 
 check() {
