diff --git PKGBUILD PKGBUILD
index 275d1f0..b52a37a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@
 # NOTE: requires rebuilt with each new gcc version
 
 pkgbase=libtool
-pkgname=(libtool lib32-libltdl)
+pkgname=(libtool)
 pkgver=2.6.0
 _commit=e40fdc22cf727fb885f9df7d9affa827e3253d1c
 pkgrel=4
@@ -24,8 +24,6 @@ makedepends=(
   "gcc>=$_gccver"
   git
   glibc
-  lib32-gcc-libs
-  lib32-glibc
   help2man
 )
 checkdepends=(
@@ -55,7 +53,6 @@ prepare() {
 
   ./bootstrap
 
-  cp -ar "${srcdir}"/libtool "${srcdir}"/libtool32
 }
 
 build() {
@@ -63,11 +60,6 @@ build() {
   ./configure --prefix=/usr lt_cv_sys_lib_dlsearch_path_spec="/usr/lib /usr/lib32"
   make
 
-  #lib32
-  cd "${srcdir}"/libtool32
-  export CC="gcc -m32" CXX="g++ -m32"
-  ./configure --prefix=/usr lt_cv_sys_lib_dlsearch_path_spec="/usr/lib /usr/lib32" --libdir=/usr/lib32
-  make
 }
 
 check() {
