diff --git PKGBUILD PKGBUILD
index fb79c87..9b0fda7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ license=('GPL-3.0-only')
 depends=('gcc-libs')
 makedepends=('cargo-nightly' 'bpf-linker')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('a654140c12410c6e8d9e5bd14016529526c01e04c9a6a3f5ae3fc0501c7bd1a23622e234eb977f8bfd3583c6d44d2fa36b1b0c8228a67c99352c0670d867e6f7')
+sha512sums=('6013eb0f9562de6410c0ed3380bf7c66ee77351255a3683365334b71af33635ff2b82a092ff3b8f20ff1e0ba9c74750a655de11a1e8661898a2a3234c49107b6')
 options=('!lto')
 
 prepare() {
