diff --git a/tests/isolated/patcher_fork_after_monkey_patch.py b/tests/isolated/patcher_fork_after_monkey_patch.py
index 6c5c445..746d825 100644
--- a/tests/isolated/patcher_fork_after_monkey_patch.py
+++ b/tests/isolated/patcher_fork_after_monkey_patch.py
@@ -18,7 +18,7 @@ if __name__ == '__main__':
     import eventlet.green.threading
 
     def target():
-        eventlet.sleep(0.1)
+        eventlet.sleep(1)
 
     threads = [
         threading.Thread(target=target, name='patched'),
