diff --git PKGBUILD PKGBUILD
index ee1f7053..b9a316e8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,17 +20,21 @@ optdepends=(
 )
 options=('!makeflags')
 source=(https://github.com/eventlet/eventlet/archive/v${pkgver}/${_pyname}-${pkgver}.tar.gz
-        python310.patch)
+        python310.patch
+        "increase-test-thread-sleep-time.patch")
 sha512sums=('455234ff08bd79e5a6b66f31f7ae87d93fde9a335f70244806cec80a4141316cd42aff91f6b8cff16f0bbe6fb673f19d6a76dc604281789e166bc41df853d8e5'
-            '07739075628ff9d140064e04189332b479f184e4647753f987b0818fa55aaca907d4880afb5cf31f980426f315e1014027dcd848551149000a12145f982cd883')
+            '07739075628ff9d140064e04189332b479f184e4647753f987b0818fa55aaca907d4880afb5cf31f980426f315e1014027dcd848551149000a12145f982cd883'
+            '86dffff4fee9dcd4d17cd7c7ad04908874a68e6328641f72903f2e92438b537344712957f45476b872a489ef530915fc4c67916830e557b17c6b9cdbfd5a36cd')
 b2sums=('6e740507307f4be34bc76d07aefb438e8562efe02220015bc0f029ada739058d9e5a5486e74e2786d44895fdea2e31ab57f0188e6bac081ea8705d08a6016b4e'
-        '783445f708c12586e026f7feac982a7b6c21f86468609c375568b51ad6055977df3d0bc740f3127b9f8bc95c1c50e81aae02ca0e0be674ed4627910d39b1ef47')
+        '783445f708c12586e026f7feac982a7b6c21f86468609c375568b51ad6055977df3d0bc740f3127b9f8bc95c1c50e81aae02ca0e0be674ed4627910d39b1ef47'
+        'b2415a0e47894115fd785e458cd4f8972c1c2128dd815e7e12479db4b6573a01d1ad2e73d7f36463c25c0b80eb789528ee94d4a86bfd24b50d094456178700ad')
 
 prepare() {
   cd ${_pyname}-${pkgver}
   # https://github.com/eventlet/eventlet/issues/730
   # https://github.com/eventlet/eventlet/issues/739
   patch -Np1 -i ../python310.patch
+  patch -Np1 -i "../increase-test-thread-sleep-time.patch"
   sed -r 's|(check_idle_cpu_usage\(.*,) .*\)|\1 0.8\)|g' -i tests/*_test.py
 }
 
