diff --git av1an-cli/Cargo.toml av1an-cli/Cargo.toml
index afbc8eb..b077419 100644
--- av1an-cli/Cargo.toml
+++ av1an-cli/Cargo.toml
@@ -32,7 +32,7 @@ default-features = false
 features = ["git", "build", "rustc", "cargo"]
 
 [dependencies.ffmpeg-next]
-version = "4.4.0"
+version = "5.0.3"
 
 [features]
 ffmpeg_static = ["ffmpeg-next/static", "ffmpeg-next/build"]
diff --git av1an-core/Cargo.toml av1an-core/Cargo.toml
index 5a44c58..e088ec8 100644
--- av1an-core/Cargo.toml
+++ av1an-core/Cargo.toml
@@ -58,7 +58,7 @@ default-features = false
 features = ["const_generics", "const_new", "union"]
 
 [dependencies.ffmpeg-next]
-version = "4.4.0"
+version = "5.0.3"
 
 [dependencies.plotters]
 version = "0.3.1"
