diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 15a676a..470c9f5 100644
--- a/trunk/PKGBUILD
+++ b/trunk/PKGBUILD
@@ -5,7 +5,7 @@ pkgname=av1an
 pkgver=0.2.0
 pkgrel=2
 pkgdesc='Cross-platform command-line encoding framework with per scene quality encoding'
-arch=('x86_64')
+arch=('any')
 url='https://github.com/master-of-zen/Av1an'
 license=('GPL3')
 makedepends=('rust' 'nasm' 'clang')
@@ -17,8 +17,16 @@ optdepends=('svt-av1: SVT-AV1 encoder support'
             'mkvtoolnix-cli: mkvmerge support'
             'ffms2: FFMS2 chunk detection support'
             'vapoursynth-plugin-lsmashsource: L-SMASH chunk detection support')
-source=("$pkgname-$pkgver.tar.gz"::https://github.com/master-of-zen/Av1an/archive/refs/tags/$pkgver.tar.gz)
-sha256sums=('68169f0c8760cb7d42ab484a969ca5f092551352998d89c602ac3527542e6538')
+source=("$pkgname-$pkgver.tar.gz"::https://github.com/master-of-zen/Av1an/archive/refs/tags/$pkgver.tar.gz
+        fix-dependencies.patch)
+sha256sums=('68169f0c8760cb7d42ab484a969ca5f092551352998d89c602ac3527542e6538'
+            '749e022ce23acca8452f604f2fcd9b248a9c7d2a1e90c2160487c4728216bbba')
+
+prepare() {
+  cd "Av1an-${pkgver}"
+  patch -p2 -i ./fix-dependencies.patch
+  cargo fetch
+}
 
 build() {
   cd "Av1an-${pkgver}"
