--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,12 @@ optdepends=('svt-av1: SVT-AV1 encoder support'
 source=("$pkgname-$pkgver.tar.gz"::https://github.com/master-of-zen/Av1an/archive/refs/tags/$pkgver.tar.gz)
 sha256sums=('c30d8d2ba9613d21daadb1c1a35781a36a979bc7aa90b95fc94c59b9be774866')
 
+prepare() {
+  cd "Av1an-${pkgver}"
+  echo -e "\n[patch.crates-io]\nffmpeg-the-third = { git = 'https://github.com/moui0/ffmpeg-the-third', tag = 'v2.0.1' }" >> Cargo.toml
+  cargo update -p ffmpeg-the-third
+}
+
 build() {
   cd "Av1an-${pkgver}"
   export CFLAGS+=" -ffat-lto-objects"
