--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,13 @@
 source=("git+https://github.com/master-of-zen/Av1an#tag=v$pkgver")
 sha256sums=('01c7e71296c60b1584fb1ecaf5cca9ecef849ba65bbd8a765be195432e445695')
 
+prepare() {
+  cd "Av1an"
+  # https://github.com/rust-av/av-scenechange/pull/209
+  printf '\n[patch.crates-io]\nav-scenechange = { git = "https://github.com/felixonmars/av-scenechange", branch = "cpufeatures-0.20" }\n' >> Cargo.toml
+  cargo update -p av-scenechange
+}
+
 build() {
   cd "Av1an"
   export CFLAGS+=" -ffat-lto-objects"
