diff --git PKGBUILD PKGBUILD
index 573ff2a..2c849b9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,16 @@ url="http://squeakvm.org"
 license=('custom:MIT' 'APACHE')
 depends=('libgl' 'libpulse' 'libxrender')
 makedepends=('cmake') 
-source=(http://squeakvm.org/unix/release/Squeak-$pkgver-src.tar.gz)
-md5sums=('8a5cb1d6899d55df56154fab8716e034')
+source=(http://squeakvm.org/unix/release/Squeak-$pkgver-src.tar.gz
+        "fix_not_literal_format.patch")
+md5sums=('8a5cb1d6899d55df56154fab8716e034'
+         'fcd5caf14353490280231a737c8a74b6')
+
+prepare() {
+  cd Squeak-$pkgver-src/
+
+  patch -Np1 -i "${srcdir}/fix_not_literal_format.patch"
+}
 
 build() {
   cd Squeak-$pkgver-src/
