diff --git a/vendor/golang.org/x/crypto/internal/syscall/unix/getrandom_linux_generic.go b/vendor/golang.org/x/crypto/internal/syscall/unix/getrandom_linux_generic.go
index 8425800..d6af3de 100644
--- a/vendor/golang.org/x/crypto/internal/syscall/unix/getrandom_linux_generic.go
+++ b/vendor/golang.org/x/crypto/internal/syscall/unix/getrandom_linux_generic.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.

-// +build arm64
+// +build arm64 riscv64

 package unix

