--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,7 @@
 
 prepare() {
   cd "$pkgname"
+  patch -Np1 -i "$srcdir/increase-test-timeouts.patch"
   GOFLAGS="-mod=readonly" go mod vendor -v
 
   # Fix CVE-2025-49466, see:
@@ -75,3 +76,6 @@
 
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE
 }
+
+source+=(increase-test-timeouts.patch)
+sha512sums+=('2ae7aeac3c944c3e7b753f88a7038de7976532ee19ef4a7ae0e9d197bc5dee23dac8971dbf19193c2e30ee545e312aa39a5c310449c00bb205a607087c5f7504')
