--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,21 @@ makedepends=('go' 'git' 'scdoc')
 optdepends=('w3m: html viewer support' 'dante: proxy support')
 provides=('aerc')
 conflicts=('aerc')
-source=("$pkgname-$pkgver.tar.gz::https://git.sr.ht/~rjarry/${pkgname}/archive/${pkgver}.tar.gz")
-sha512sums=('04c80d5c86cc1cf978ae2590a469f56241978253dab82dfa0a8b17860b496c2c9ee30ebf345953046968856da06beadebdced4bdee6e92d5faf85abe1ff5bcd3')
+source=("$pkgname-$pkgver.tar.gz::https://git.sr.ht/~rjarry/${pkgname}/archive/${pkgver}.tar.gz"
+        "update_go_version.patch"
+        "add_riscv_getrandom.patch")
+sha512sums=('04c80d5c86cc1cf978ae2590a469f56241978253dab82dfa0a8b17860b496c2c9ee30ebf345953046968856da06beadebdced4bdee6e92d5faf85abe1ff5bcd3'
+            'db27c9111f5486911e3970c50a97325dcfa10b8060b2e8525fd1a7668157d6cd546db190cdaca9bf1fddc8d74f04f873ab0de61514994e70930a4db2269b9d26'
+            '5a9afffdb4336f017ffde1909eb67482cfc41d64fdc23dc4697fa15da27fbf613f5532bbedf9deeb0ef1a8bd4addfff38f937b6374067ec4ea03d2d560f6befa')
+
+prepare() {
+    cd "$srcdir/$pkgname-$pkgver"
+    patch -Np1 -i ../update_go_version.patch
+    go mod vendor
+    patch -Np1 -i ../add_riscv_getrandom.patch
+    rm -rf 'vendor/github.com/ddevault/go-libvterm'
+    ln -s '/build/go/pkg/mod/github.com/ddevault/go-libvterm@v0.0.0-20190526194226-b7d861da3810' 'vendor/github.com/ddevault/go-libvterm'
+}
 
 build() {
     cd "$srcdir/$pkgname-$pkgver"
