diff --git PKGBUILD PKGBUILD
index 59067b2..65d9681 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,11 @@ checkdepends=('xorg-server-xvfb' 'python-pytest-forked' 'python-numpy')
 source=("https://files.pythonhosted.org/packages/source/w/wxPython/wxPython-$pkgver.tar.gz")
 sha512sums=('ee9ab8effb3c71de004c3d61ec583b9980c6b75fe9bb04cf2c76025ea70afc206bc3073c44d1c83c7ba604afed9fd38c01b9ef506617ba279a4ac58e3cf34438')
 
+prepare() {
+  cd $_pkgname-$pkgver
+  sed -i '/bool.cpp/d' wscript
+}
+
 build() {
   cd $_pkgname-$pkgver
 
