--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,9 @@ prepare() {
   cd $_pkgname-$pkgver
   patch -Np1 -i ../no-demos.patch # Don't build demos
   patch -Np1 -i ../add-wmluiltok-main-function.patch # Doesn't build without it
+  patch -Np1 -i ../fix_not_literal_format.patch
+  patch -Np1 -i ../add_option_main.patch
+  rm -f tools/wml/wmluiltok.c # force regenerating this file
   touch NEWS AUTHORS
   autoreconf -if
 }
@@ -60,3 +63,9 @@ package() {
   cd $_pkgname-$pkgver
   make DESTDIR="$pkgdir" install
 }
+
+source+=("fix_not_literal_format.patch"
+         "add_option_main.patch")
+sha256sums+=('23b6750d6c8c4ed3fb9bb448eaaf1f5b008065992b999895b77c12309ded3724'
+             '8a82c4e55d0a7855db50c91fe0daab49a01800d85462e92ffad1e0e4f0ce8d80')
+
