--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,12 @@
 )
 
 prepare() {
-  cd llvm-project-$pkgver.src/compiler-rt
+  cd llvm-project-$pkgver.src
+
+  # https://github.com/llvm/llvm-project/pull/185486
+  patch -Np1 -i "$srcdir/fix-cfloat128-detection.patch"
+
+  cd compiler-rt
   mkdir build
 }
 
@@ -53,4 +58,7 @@
   install -Dm644 "$srcdir/llvm-project-$pkgver.src/compiler-rt/LICENSE.TXT" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
+source+=("fix-cfloat128-detection.patch::https://github.com/llvm/llvm-project/commit/ea9f8b77467fdbfab0d5bb3342e97de6b54e4e2d.patch?full_index=1")
+sha256sums+=('63f2615dfa909333132d90a1cba48cd9c1d4eb8aba469cf48d1159987ce694ae')
+
 # vim:set ts=2 sw=2 et:
