Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 443903)
+++ PKGBUILD	(working copy)
@@ -25,8 +25,7 @@
   # NOTE: apply only minimal CFLAGS, as gnu-efi does not provide userspace
   # libs, but may be used in unitialized machine state and should therefore not
   # be architecture optmized
-  # NOTE: fat-lto-objects is required for non-mangled (static) object files
-  CFLAGS="-O2 -flto -ffat-lto-objects"
+  CFLAGS="-O2"
   make
   make -C lib
   make -C gnuefi
