diff --git a/scripts/lib/babel.js b/scripts/lib/babel.js
index 8fedf51..30ea003 100644
--- a/scripts/lib/babel.js
+++ b/scripts/lib/babel.js
@@ -34,8 +34,8 @@ export function isBuilt() {
 
 export default () => {
   const res = spawnSync(
-    'babel',
-    ['--source-maps', 'true', 'src/', '-d', 'lib/'],
+    'node',
+    ['--max-opt=0', 'node_modules/.bin/babel', '--source-maps', 'true', 'src/', '-d', 'lib/'],
     {
       stdio: 'inherit',
       shell: true,
diff --git a/scripts/lib/eslint.js b/scripts/lib/eslint.js
index 94df352..82e9a8d 100644
--- a/scripts/lib/eslint.js
+++ b/scripts/lib/eslint.js
@@ -3,7 +3,7 @@ import { spawnSync } from 'child_process';
 import config from './config.js';
 
 export default () => {
-  const res = spawnSync('eslint', config.eslint.files, {
+  const res = spawnSync('node', ['--max-opt=0', 'node_modules/.bin/eslint', ...config.eslint.files], {
     stdio: 'inherit',
     shell: true,
   });
diff --git a/scripts/lib/mocha.js b/scripts/lib/mocha.js
index da1e67c..cc70e34 100644
--- a/scripts/lib/mocha.js
+++ b/scripts/lib/mocha.js
@@ -14,6 +14,8 @@ const runMocha = (args, execMochaOptions = {}, coverageEnabled) => {
   const binArgs = coverageEnabled ? [mochaPath, ...args] : args;
   const binPath = coverageEnabled ? which('nyc') : mochaPath;
 
+  binArgs.push('-n', 'max-opt=0');
+
   if (process.env.MOCHA_TIMEOUT) {
     const { MOCHA_TIMEOUT } = process.env;
     binArgs.push('--timeout', MOCHA_TIMEOUT);
