diff --git PKGBUILD PKGBUILD
index 4ea0e4790..dd07bdaa9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,12 +17,14 @@ source=(https://www.monkey.org/~dugsong/${pkgname}/beta/${pkgname}-${pkgver}.tar
         http://ftp.de.debian.org/debian/pool/main/d/${pkgname}/${pkgname}_2.4b1+debian-29.debian.tar.xz
         dsniff-rpc.patch
         dsniff-macof-size-calculation.patch
-        dsniff-httppostfix.patch)
+        dsniff-httppostfix.patch
+        "fix_conflict_func_name.patch")
 sha256sums=('a9803a7a02ddfe5fb9704ce86f0ffc48453c321e88db85810db411ba0841152a'
             '3f2263452facf9f0a402497b34c7a2573da0700005bb3c7940df9c5e099b5835'
             '21b37ba4c386aa576d6829c298bdd62df6fa227e44164d9e96675e66b93bb134'
             'd8f5cc5d14a614410a84a8eaba5a4212ee03466c4fe2fd4a634f5d7c3f688ec7'
-            '3efd6ca2267540016e843af3e8d7720d888fd17a73cba410e2c68022fd3a7baf')
+            '3efd6ca2267540016e843af3e8d7720d888fd17a73cba410e2c68022fd3a7baf'
+            '32559e08a49581acf9082bbb5c72a635228a6fd28cfe1638560ed5d2398609be')
 
 prepare() {
   cd "${pkgname}-2.4"
