diff --color -Naur configure configure
--- configure	2004-04-19 19:13:01.000000000 +0800
+++ configure	2021-10-03 22:34:17.894666855 +0800
@@ -25144,7 +25144,7 @@
 	CFLAGS="$CFLAGS -g"
 fi
 
-CFLAGS="$CFLAGS -W -Wall -Wno-unused -Wno-switch -Wno-format"
+CFLAGS="$CFLAGS -W -Wall -Wno-unused -Wno-switch"
 if test x$enable_Werror = xyes; then
 	CFLAGS="$CFLAGS -Werror"
 fi
diff --color -Naur progs/libprogs/tools.c progs/libprogs/tools.c
--- progs/libprogs/tools.c	2004-04-19 03:48:47.000000000 +0800
+++ progs/libprogs/tools.c	2021-10-03 22:57:32.952475151 +0800
@@ -67,7 +67,7 @@
 	vsprintf(mess, format, args);
 	va_end(args);
 	
-	fprintf(stderr, mess);
+	fprintf(stderr, "%s", mess);
 	fflush(stderr);
 	
 	do {
@@ -78,7 +78,7 @@
 			
 		if (!check_choose(valid_chooses, choose)) {
 			if (prompt_count < 2) {
-				fprintf(stderr, error_msg);
+				fprintf(stderr, "%s", error_msg);
 				fflush(stderr);
 			}	
 		} else
