diff --git PKGBUILD PKGBUILD
index ced7734..0007bae 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,16 @@ depends=('e2fsprogs')
 source=(#ftp://ftp.namesys.com/pub/libreiserfs/${pkgname}-${pkgver}.tar.gz
 	https://sources.archlinux.org/other/progsreiserfs/${pkgname}-${pkgver}.tar.gz
         fix-underquoted-definition.patch
+        fix-format-security.patch
 	)
 sha512sums=('e2b94c778796492e66578b3dde8601d8670c11344dca2082eccf648f8bf26a226d07c26333d91357b081c50b09aed4b7f9a5bb1a6c985808c55db02c83e24ce3'
-            '2450b032af7141a90c69584ca4f4d0830a12c35525829d0850207734c218dd83e1d4f49236e835fb93992ac8d4d310d4d63db7038aa49bac32ec60d828cc0bbf')
+            '2450b032af7141a90c69584ca4f4d0830a12c35525829d0850207734c218dd83e1d4f49236e835fb93992ac8d4d310d4d63db7038aa49bac32ec60d828cc0bbf'
+            '8f1e4d7e5dc43b6a5a4eabb94b5f2335eb987639b62f1f2aabdb8a53af8554ad1bc8e97e9d55e5f83baefab22baec7253de55a2292dda67da1ab7a9ca9150687')
 
 build()        {
         cd ${pkgname}-${pkgver}
         patch -Np0 -i ../fix-underquoted-definition.patch
+        patch -Np0 -i ../fix-format-security.patch
         ./configure --prefix=/usr --mandir=/usr/share/man 
         make
 }
