--- PKGBUILD
+++ PKGBUILD
@@ -30,7 +30,7 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DTARGET=CORE2 \
+    -DTARGET=RISCV64_GENERIC \
     -DNUM_THREADS=64 \
     -DDYNAMIC_ARCH=ON
   cmake --build build
@@ -42,7 +42,8 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DTARGET=CORE2 \
+    -DTARGET=RISCV64_GENERIC \
+    -DFCOMMON_OPT=-fdefault-integer-8 \
     -DNUM_THREADS=64 \
     -DDYNAMIC_ARCH=ON \
     -DINTERFACE64=1
