--- PKGBUILD
+++ PKGBUILD
@@ -34,8 +34,10 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DCORE=CORE2 \
+    -DTARGET=RISCV64_GENERIC \
     -DNUM_THREADS=64 \
+    -DNO_BINARY_MODE=1 \
+    -DBINARY_DEFINED=1 \
     -DDYNAMIC_ARCH=ON 
   cmake --build build
 
@@ -48,9 +50,11 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DCORE=CORE2 \
+    -DTARGET=RISCV64_GENERIC \
     -DNUM_THREADS=64 \
     -DDYNAMIC_ARCH=ON \
+    -DNO_BINARY_MODE=1 \
+    -DBINARY_DEFINED=1 \
     -DINTERFACE64=1
   cmake --build build64
 }
