--- PKGBUILD
+++ PKGBUILD
@@ -30,9 +30,8 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DTARGET=CORE2 \
-    -DNUM_THREADS=64 \
-    -DDYNAMIC_ARCH=ON
+    -DTARGET=RISCV64_GENERIC \
+    -DNUM_THREADS=64
   cmake --build build
 
   ASMFLAGS=$CFLAGS FFLAGS=${CFLAGS/ -Wformat -Werror=format-security/} FC=gfortran cmake -B build64 -S $_pkgname \
@@ -42,9 +41,9 @@ build() {
     -DNO_AFFINITY=ON \
     -DUSE_OPENMP=1 \
     -DNO_WARMUP=1 \
-    -DTARGET=CORE2 \
+    -DTARGET=RISCV64_GENERIC \
+    -DFCOMMON_OPT=-fdefault-integer-8 \
     -DNUM_THREADS=64 \
-    -DDYNAMIC_ARCH=ON \
     -DINTERFACE64=1
   cmake --build build64
 }
