diff --git PKGBUILD PKGBUILD
index 314e5adb7..889ae9916 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,12 +12,15 @@ license=('MIT')
 depends=('libgl' 'glfw')
 makedepends=('cmake' 'mesa')
 checkdepends=('python')
-source=($pkgname-$pkgver.tar.gz::"https://github.com/cginternals/glbinding/archive/v$pkgver.tar.gz")
-sha256sums=('6729b260787108462ec6d8954f32a3f11f959ada7eebf1a2a33173b68762849e')
+source=($pkgname-$pkgver.tar.gz::"https://github.com/cginternals/glbinding/archive/v$pkgver.tar.gz"
+        "Compatibility-for-g++-9.patch")
+sha256sums=('6729b260787108462ec6d8954f32a3f11f959ada7eebf1a2a33173b68762849e'
+            '1b87f338a75fd1612febaf516b2b9b183f90c32b058648a4478254c82e7f6dfb')
 
 prepare() {
   cd $pkgname-$pkgver
 
+  patch -Np1 -i ../Compatibility-for-g++-9.patch
   mkdir build
 }
 
