Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 arch=(x86_64)
 license=('custom')
 depends=('libpciaccess')
-makedepends=('valgrind' 'python-docutils' 'meson')
+makedepends=('python-docutils' 'meson')
 checkdepends=('cairo')
 replaces=('libdrm-new' 'libdrm-nouveau')
 source=(https://dri.freedesktop.org/$pkgname/$pkgname-$pkgver.tar.xz{,.sig}
