diff --unified --recursive --text a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
--- a/yabause/src/libretro/Makefile	2021-09-16 14:20:15.416804308 +0000
+++ b/yabause/src/libretro/Makefile	2021-09-16 14:22:54.340224687 +0000
@@ -81,6 +81,11 @@
 		FLAGS += -DARM
 	endif
 
+	# riscv64
+	ifneq ($(findstring riscv64,$(shell uname -a)),)
+		HAVE_SSE = 0
+	endif
+
 ifneq ($(findstring Linux,$(shell uname -s)),)
 	HAVE_CDROM = 1
 endif
