diff --git PKGBUILD PKGBUILD
index 7c58c82..1e70a01 100644
--- PKGBUILD
+++ PKGBUILD
@@ -4,7 +4,7 @@ pkgname=libretro-kronos
 pkgver=7016
 pkgrel=1
 pkgdesc='Sega Saturn core'
-arch=(x86_64)
+arch=(riscv64)
 url=https://github.com/libretro/yabause
 license=(GPL2)
 groups=(libretro)
@@ -18,8 +18,16 @@ makedepends=(
   libglvnd
 )
 _commit=146f4295eb7f5f76a2e6e6c84518c9bdf6a8398f
-source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit}
+        fix-unrecognized-command-line-option-msse.patch)
+sha256sums=('SKIP'
+            '24b71edbf79b5536a82a11072549dc56689d48745a2ca486004309b47410ef9f')
+
+prepare(){
+  cd libretro-yabause
+
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch
+}
 
 pkgver() {
   cd libretro-yabause
