--- PKGBUILD
+++ PKGBUILD
@@ -28,3 +28,10 @@
   install -Dm 644 libretro-yabause/yabause/src/libretro/kronos_libretro.so -t "${pkgdir}"/usr/lib/libretro/
 }
 
+prepare() {
+  cd libretro-yabause
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch
+}
+
+source+=(fix-unrecognized-command-line-option-msse.patch::https://github.com/libretro/yabause/commit/e81b147872927f06528150e4d06207cdb08e88b8.patch)
+b2sums+=('b1e0ba926a18f34cceb7015f424b50f5dceec7c0e1bcaa5071ffb5d7d1351a3fa8a946f0bcc401e36031a4b0a228237e1534861717237182e4b7a74f7271a391')
