diff --git PKGBUILD PKGBUILD
index 96e0f7735..2d2707ca7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,8 +17,16 @@ makedepends=(
   git
   libglvnd
 )
-source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=95a6176d9c059f8cce87a8aff8e25e98ef9d94a7)
-sha256sums=(SKIP)
+source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=95a6176d9c059f8cce87a8aff8e25e98ef9d94a7
+        fix-unrecognized-command-line-option-msse.patch)
+sha256sums=('SKIP'
+            '24b71edbf79b5536a82a11072549dc56689d48745a2ca486004309b47410ef9f')
+
+prepare(){
+  cd libretro-yabause
+
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch
+}
 
 pkgver() {
   cd libretro-yabause
