--- PKGBUILD
+++ PKGBUILD
@@ -7,9 +7,9 @@
 
 pkgname=luajit
 # LuaJIT has a "rolling release" where you should follow git HEAD
-_commit=e826d0c101d750fac8334d71e221c50d8dbe236c
+_commit=4da850743b197a7e864ab811ec3dce7204040a7b
 # The patch version is the timestamp of the above git commit, obtain via `git show -s --format=%ct`
-_ct=1697887905
+_ct=1698135991
 pkgver="2.1.${_ct}"
 pkgrel=1
 pkgdesc='Just-in-time compiler and drop-in replacement for Lua 5.1'
@@ -17,27 +17,27 @@ arch=('x86_64')
 url='https://luajit.org/'
 license=('MIT')
 depends=('gcc-libs')
-source=("LuaJIT-${_commit}.tar.gz::https://repo.or.cz/luajit-2.0.git/snapshot/${_commit}.tar.gz")
-md5sums=('c4530f5b223752e8da4f25f8503a195f')
-sha256sums=('76e9aa90717cf6ba5c8991fa31b03a7c30a2b29ad48f04907199ea53b7078fbe')
-b2sums=('9096c3b787728697d5b07a1e7c1f1b120952b48129bfd46e8e8f4db15f3c31c3d9d002f17db4a00fa5ffceb5024309cfdd335ccae17fa8d30b3ea46e16672f4c')
+source=("LuaJIT-${_commit}.tar.gz::https://github.com/infiWang/LuaJIT/archive/${_commit}.tar.gz")
+md5sums=('e92ce6c7b86c452a1be3f014ac57c685')
+sha256sums=('24369db7e0be21ff9f70a43ef3ef4622fdc45883c526cb717f7b7281ea66c065')
+b2sums=('31ffed44719ff45b6bab028e1cd26cc7a977feec68585e1bfaeb9fa5cfeca15eaa80576881cb40bd0453b8f344a5e3a2613a2e0a0be9f7247e717ff83a1510ac')
 
 build() {
-  cd "luajit-2.0-${_commit::7}"
+  cd "LuaJIT-${_commit}"
 
   # Avoid early stripping
   make amalg PREFIX=/usr BUILDMODE=dynamic TARGET_STRIP=" @:"
 }
 
 check() {
-  cd "luajit-2.0-${_commit::7}"
+  cd "LuaJIT-${_commit}"
 
   # Make sure that _ct was updated
   test "${_ct}" == "$(cat .relver)"
 }
 
 package() {
-  cd "luajit-2.0-${_commit::7}"
+  cd "LuaJIT-${_commit}"
 
   make install DESTDIR="$pkgdir" PREFIX=/usr
   install -Dm644 COPYRIGHT "$pkgdir/usr/share/licenses/$pkgname/COPYRIGHT"
