--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,13 @@
 sha256sums=('6e5fec07750add912e7c3eae0c194d24cd6d023714e1f04a0298a5b4819e4457')
 b2sums=('1dd16b18f2310fd61ee6c3b8352c7a79d09b02da5ac13a8e00f6b67fdac9adcd40c7333428a2f6b3cf22b20797061bdfdba9fa211b9b10ee63a99546831e7da1')

+prepare() {
+  cd "LuaJIT-${_commit}"*/
+
+  # Add the RISC-V port to the packaged upstream revision.
+  patch -Np1 -i ../riscv64-port.patch
+}
+
 build() {
   cd "LuaJIT-${_commit}"*/

@@ -40,3 +47,7 @@
   make install DESTDIR="$pkgdir" PREFIX=/usr
   install -Dm644 COPYRIGHT "$pkgdir/usr/share/licenses/$pkgname/COPYRIGHT"
 }
+
+source+=("riscv64-port.patch::https://github.com/IgnotaYun/LuaJIT/compare/c6ffc141a8762b41703f9287d63d93622a13dd8f...fb8569e1e11b598c34ca16988682d8286219dfd5.diff")
+sha256sums+=('b49737da996b165698bb81e5883a575999de4b03c760aa0746ddd208d68fecc4')
+b2sums+=('7920441985c53aa7d3fd9c61763a826197e234fc699a44bccf63e32012c8ebb09497411b2772267da365735536206b5e477fba8f66d9aa463f096a787d5dd253')
