--- PKGBUILD
+++ PKGBUILD
@@ -7,9 +7,11 @@
 
 pkgname=luajit
 # LuaJIT has a "rolling release" where you should follow git HEAD
-_commit=ff204d0350575cf710f6f4af982db146cb454e1a
+# note to riscv: choose commit close to (but newer than) Arch upstream from `riscv` branch of infiWang/LuaJIT
+# make sure the selected commit contains necessary changes to RISC-V and update $_ct below
+_commit=5938da937b13790dc3705ee5823405f0ff8c31ed
 # The patch version is the timestamp of the above git commit, obtain via `git show -s --format=%ct`
-_ct=1702233742
+_ct=1702376626
 pkgver="2.1.${_ct}"
 pkgrel=1
 pkgdesc='Just-in-time compiler and drop-in replacement for Lua 5.1'
@@ -17,10 +19,10 @@ arch=('x86_64')
 url='https://luajit.org/'
 license=('MIT')
 depends=('gcc-libs')
-source=("LuaJIT-${_commit}.tar.gz::https://github.com/LuaJIT/LuaJIT/archive/${_commit}.tar.gz")
-md5sums=('97486356d223510a6e3c31a20bcd32ed')
-sha256sums=('3ec37f78ab3b1afd4c3af0fde743c332da3da32eadc8500489c1cc2e4f0ec7eb')
-b2sums=('6ba03fa107baadf0ac980d515debd638b1a166014ee46c6fa95865a12678a831fbae04d14ccb737723a69874af2b0637bbaa516973830ca4c7e5311aa3f91b76')
+source=("LuaJIT-${_commit}.tar.gz::https://github.com/infiWang/LuaJIT/archive/${_commit}.tar.gz")
+md5sums=('aec4671279cacf32c09dab2fe9ff9aa1')
+sha256sums=('0de7c45601a9052c4a37963f343c6963c84871be48be8893c9bace2c959b699e')
+b2sums=('9f210fe44b7966051a3317dedf15f54252978eb953c297558dba265aa7852f6d715f26f46693271b726211da4ab49ad60b2b7c55315a9ac3372fac00fb34bf44')
 
 build() {
   cd "LuaJIT-${_commit}"
