--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,9 @@ b2sums=('bfd66c9cbe77edbceb617d6acc83dad9a42bdc541be381b87883fedb0e185d4c4459862
 
 prepare() {
   cd $_name-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
