diff --git PKGBUILD PKGBUILD
index 761fdf2b2..2b00b0823 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ license=('MIT')
 url="http://pyicu.osafoundation.org/"
 depends=('python' 'icu')
 makedepends=('python-setuptools')
-checkdepends=('python-fonttools' 'python-pytest')
+checkdepends=('python-fonttools' 'python-pytest' 'python-six')
 source=("https://pypi.io/packages/source/P/PyICU/PyICU-$pkgver.tar.gz")
 sha512sums=('9edd2f5f8755d06e70a9889f1129bdf53eec50d2948a54edfd2142573447774b0767a329092436c9bc7045c8e8e81043bf4e68d0990a0d8427b18542c5b44727')
 
