--- PKGBUILD
+++ PKGBUILD
@@ -18,9 +18,9 @@ provides=('deepin-screenshot')
 conflicts=('deepin-screenshot')
 groups=('deepin-extra')
 source=("https://github.com/linuxdeepin/deepin-screen-recorder/archive/$pkgver/$pkgname-$pkgver.tar.gz"
-        $pkgname-unqualified-id.patch::https://github.com/linuxdeepin/deepin-screen-recorder/pull/416.patch)
+        "$pkgname-unqualified-id.patch")
 sha512sums=('9e9281918b27b34e27f9f7c2ef7b490eda272adc63370fb2908fd98735983467195cc24a466ca3b9a04e7e6a1b622eb905158da513947b633b434882fe797fa2'
-            '5b26c7127cde0ae041ca53ab87edb3ea341d883f7d204f03e697175245c5ad564f9f0e30448fb291cff34fa26d03e1ee2dec4d368527f0bb217b086d7f1dde78')
+            '5624a25292b0d4f74aa73c1374c51d9098db8a6984e9586da1f88349c0315e2dde1cb6f3b26ac7db6fd04044b9283238ab93a607be6883f746b0e1a57e8c3ce3')
 
 prepare() {
   cd deepin-screen-recorder-$pkgver
