diff --git PKGBUILD PKGBUILD
index d35ccdc..b90c921 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,8 @@ url='https://www.graphviz.org/'
 license=('EPL')
 arch=('x86_64')
 depends=('libltdl' 'gd' 'librsvg' 'ghostscript' 'pango' 'gts' 'gsfonts')
-makedepends=('swig' 'mono' 'guile' 'lua' 'perl' 'python' 'r' 'tk' 'qt6-base')
-optdepends=('mono: sharp bindings'
-            'guile: guile bindings'
+makedepends=('swig' 'guile' 'lua' 'perl' 'python' 'r' 'tk' 'qt6-base')
+optdepends=('guile: guile bindings'
             'lua: lua bindings'
             'perl: perl bindings'
             'python: python bindings'
