--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ url='https://www.graphviz.org/'
 license=('EPL')
 arch=('x86_64')
 depends=('libltdl' 'gd' 'librsvg' 'libxaw' 'ghostscript' 'pango' 'gts' 'gsfonts' 'libnsl' 'libpng')
-makedepends=('swig' 'mono' 'guile' 'lua' 'ocaml>=4.14.0' 'perl' 'python' 'r' 'tk' 'qt5-base' 'gtk2')
+makedepends=('swig' 'guile' 'lua' 'ocaml>=4.14.0' 'perl' 'python' 'r' 'tk' 'qt5-base' 'gtk2')
 optdepends=('mono: sharp bindings'
             'guile: guile bindings'
             'lua: lua bindings'
