diff --git PKGBUILD PKGBUILD
index 157b517..1af8d10 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,7 @@ optdepends=('ffmpeg: additional import/export capabilities')
 source=("https://github.com/${pkgname}/${pkgname}/archive/Audacity-${pkgver}.tar.gz"
         "${pkgname}-2.4.1-resize_crash.patch::https://github.com/dvzrv/audacity/commit/d7e769040b508aaf15ce7b51cc1a565d41cdb079.patch")
 sha512sums=('2f0a3f7e24dce91ae520e285dbb05778b68e0fe37b8ded55ac568b5f8135069951b3986764adc5d0791c165b283a3c19596f1d6f51ba17028e0eb04558c63f25'
-            '9c111b6d74ff0cd76f5f2db0ce513bf1c5400c41497220c87ea694f5b2b75c1e6bc1a817dd1fb2e81ed5ddcf2c610bb334a091c2cd8f7bae1111a911f47cab4b')
+            'SKIP')
 
 prepare() {
   mv -v "${pkgname}-Audacity-${pkgver}" "${pkgname}-${pkgver}"
@@ -53,7 +53,8 @@ build() {
     --with-soundtouch \
     --with-libtwolame \
     --with-libvorbis \
-    --with-lv2
+    --with-lv2 \
+    --disable-sse
   make
 }
 
