--- PKGBUILD
+++ PKGBUILD
@@ -96,9 +96,11 @@ build() {
     -W no-dev
   )
 
+  # https://github.com/audacity/audacity/issues/8726 (-std=gnu17)
+  # https://github.com/audacity/audacity/issues/8872 (-ffp-contract=off)
   export VST3SDK='/usr/src/vst3sdk'
-  export CFLAGS+=" -DNDEBUG -std=gnu17"
-  export CXXFLAGS+=" -DNDEBUG"
+  export CFLAGS+=" -DNDEBUG -std=gnu17 -ffp-contract=off"
+  export CXXFLAGS+=" -DNDEBUG -ffp-contract=off"
   cmake "${cmake_options[@]}"
   cmake --build build --verbose
 }
