--- a/ops/bootstrap.rs
+++ b/ops/bootstrap.rs
@@ -43,7 +43,10 @@
 
 impl Default for SnapshotOptions {
   fn default() -> Self {
-    let arch = std::env::consts::ARCH;
+    let arch = match std::env::consts::ARCH {
+      "riscv64" => "riscv64gc",
+      arch => arch,
+    };
     let platform = std::env::consts::OS;
     let target = match platform {
       "macos" => format!("{}-apple-darwin", arch),
