--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -198,6 +198,9 @@
           _dir = "armv7-unknown-linux-gnueabihf"
         } else if (current_cpu == "arm64") {
           _dir = "aarch64-unknown-linux-gnu"
+        } else if (current_cpu == "riscv64") {
+          _dir = "linux"
+          _suffix = "-riscv64"
         } else {
           assert(false)  # Unhandled cpu type
         }
