diff --git PKGBUILD PKGBUILD
index 2b49c56e..b4944279 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ license=('BSD')
 depends=('zlib' 'gcc-libs')
 makedepends=('cmake' 'ninja')
 url='https://assimp.org/'
-source=("$pkgname-$pkgver.tar.gz::https://github.com/assimp/assimp/archive/v${pkgver}.tar.gz")
-sha512sums=('414c2c4581a9c4e09b7aefed377c49e62b11d2de7a7dc976245fb2fb91c81012a8611d16f6bebb8315215b4be42d2a3f8817ab6e369ddd505e0ebf2ba3cb3f10')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/assimp/assimp/archive/v${pkgver}.tar.gz"
+        "fix_nonheap_free.patch")
+sha512sums=('414c2c4581a9c4e09b7aefed377c49e62b11d2de7a7dc976245fb2fb91c81012a8611d16f6bebb8315215b4be42d2a3f8817ab6e369ddd505e0ebf2ba3cb3f10'
+            '82b30f82f765dce78c810eb817f219bda5ac203eb27f0ee95033215991397ba105900b55edb3476afc4219d4d7a8bed30c5b262004c7239983a717a78618bfdb')
+
+prepare() {
+    cd ${pkgname}-${pkgver}
+    patch -Np1 -i ../fix_nonheap_free.patch
+}
 
 build() {
     cd ${pkgname}-${pkgver}
