--- PKGBUILD
+++ PKGBUILD
@@ -12,10 +12,13 @@ url='https://gitlab.gnome.org/GNOME/libgovirt'
 license=(LGPL)
 depends=(librest)
 makedepends=(gobject-introspection meson)
-source=("https://gitlab.gnome.org/GNOME/libgovirt/-/archive/v${pkgver}/libgovirt-v${pkgver}.tar.bz2")
-b2sums=('383660540bcde90e4406e086f8acc98af233b83cd8a8d7f3634fa1de4e6cc43ceeae2f476a16a65732c3cf5c302a2f108b9c5b1d67b39b9999d1de48af22da24')
+source=("https://gitlab.gnome.org/GNOME/libgovirt/-/archive/v${pkgver}/libgovirt-v${pkgver}.tar.bz2"
+        "${pkgname}-suppress-error-cast-align.patch::https://gitlab.gnome.org/GNOME/libgovirt/-/merge_requests/21.patch")
+b2sums=('383660540bcde90e4406e086f8acc98af233b83cd8a8d7f3634fa1de4e6cc43ceeae2f476a16a65732c3cf5c302a2f108b9c5b1d67b39b9999d1de48af22da24'
+        '8e8e978f5ab3543381205acdf4f8b27d954d3dcc52ca20100dcabc74d1d4eef8c4c5a5e537c18a8e6a602f7c774af3e761d8c6f9824b129679937666b8ce7c77')
 
 build() {
+  patch -d ${pkgname}-v${pkgver} < ${pkgname}-suppress-error-cast-align.patch
   arch-meson ${pkgname}-v${pkgver} build
   meson compile -C build
 }
