--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,7 @@ b2sums=('383660540bcde90e4406e086f8acc98af233b83cd8a8d7f3634fa1de4e6cc43ceeae2f4
 
 prepare() {
   cd ${pkgname}-v${pkgver}
+  patch -Np1 -i ../${pkgname}-suppress-error-cast-align.patch
 }
   
 build() {
@@ -31,3 +32,6 @@ check() {
 package() {
   meson install -C build --destdir "${pkgdir}"
 }
+
+source+=("${pkgname}-suppress-error-cast-align.patch::https://gitlab.gnome.org/GNOME/libgovirt/-/merge_requests/21.diff")
+b2sums+=('SKIP')
