diff --git a/dist/targets.js b/dist/targets.js
index 54faae8..ae9248c 100644
--- a/dist/targets.js
+++ b/dist/targets.js
@@ -7,11 +7,11 @@ exports.validateListFromOptions = exports.allOfficialArchsForPlatformAndVersion
 const common_1 = require("./common");
 const get_1 = require("@electron/get");
 const semver_1 = __importDefault(require("semver"));
-exports.officialArchs = ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'universal'];
+exports.officialArchs = ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'universal', 'riscv64'];
 exports.officialPlatforms = ['darwin', 'linux', 'mas', 'win32'];
 exports.officialPlatformArchCombos = {
     darwin: ['x64', 'arm64', 'universal'],
-    linux: ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el'],
+    linux: ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'riscv64'],
     mas: ['x64', 'arm64', 'universal'],
     win32: ['ia32', 'x64', 'arm64'],
 };
@@ -24,6 +24,7 @@ const buildVersions = {
         arm64: '>= 1.8.0',
         ia32: '<19.0.0-beta.1',
         mips64el: '^1.8.2-beta.5',
+        riscv64: '>= 22'
     },
     mas: {
         arm64: '>= 11.0.0-beta.1',
@@ -136,4 +137,4 @@ function validateListFromOptions(opts, name) {
     return list;
 }
 exports.validateListFromOptions = validateListFromOptions;
-//# sourceMappingURL=targets.js.map
\ No newline at end of file
+//# sourceMappingURL=targets.js.map
