diff --git a/src/targets.js b/src/targets.js
index 39b50dd7e34dca393b083b488485eaed30e2e7f6..1a7b229bce4dba5c4e533b357dae53a8c0cff1df 100644
--- a/src/targets.js
+++ b/src/targets.js
@@ -4,11 +4,11 @@ const common = require('./common')
 const { getHostArch } = require('@electron/get')
 const semver = require('semver')
 
-const officialArchs = ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'universal']
+const officialArchs = ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'universal', 'riscv64']
 const officialPlatforms = ['darwin', 'linux', 'mas', 'win32']
 const officialPlatformArchCombos = {
   darwin: ['x64', 'arm64', 'universal'],
-  linux: ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el'],
+  linux: ['ia32', 'x64', 'armv7l', 'arm64', 'mips64el', 'riscv64'],
   mas: ['x64', 'arm64', 'universal'],
   win32: ['ia32', 'x64', 'arm64']
 }
