--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@
             'docker: To use docker as a container runtime'
             'podman: To use podman as a container runtime')
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/refs/tags/v${pkgver}.tar.gz")
-sha256sums=('a90f6b9f1571be4e694cd90beee9e80e7bd85a8bfdff197abc012a952f3df99b')
+sha256sums=('ed20a74b8c821f0954a53efd387cae4b8507f7f5f1e7365a29cf89de8c6d2d60')
 
 build() {
 	cd "${pkgname}-${pkgver}"
