diff --git PKGBUILD PKGBUILD
index 800169f..1267f9f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@ optdepends=('ansible: check official ansible collections'
             'docker: To use docker as a container runtime'
             'podman: To use podman as a container runtime')
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/refs/tags/v${pkgver}.tar.gz")
-sha256sums=('ee7f4ae8e7a3164de95d963867e32e345153ab95ca83b9a8fb94504df10394e6')
+sha256sums=('13063a3913be65cb80d3df35458f3dba2fdceb5c3fb3ea8cc583328e4debb3a9')
 
 build() {
 	cd "${pkgname}-${pkgver}"
