diff --git PKGBUILD PKGBUILD
index 9d7fa0d6..bcf055ca 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,20 +11,24 @@ makedepends=('xorgproto' 'libdrm>=2.4.26' 'libxxf86vm>=1.1.1' 'libxdamage>=1.1.3
 url="http://mesa3d.sourceforge.net"
 license=('custom')
 source=(ftp://ftp.freedesktop.org/pub/mesa/older-versions/7.x/${pkgver}/MesaLib-${pkgver}.tar.bz2
-	LICENSE mesa-dri1-symbol-visibility.patch mesa-dri1-python3.patch)
+	LICENSE mesa-dri1-symbol-visibility.patch mesa-dri1-python3.patch
+	riscv-mmio-barrier.patch)
 md5sums=('0837c52698fe3252369c3fdb5195afcc'
          '5c65a0fe315dd347e09b1f2826a1df5a'
          '489feaf524440ff2e4fa32e65e7037ce'
-         'b1da35125d638bb3ea673cc18ce9f9e3')
+         'b1da35125d638bb3ea673cc18ce9f9e3'
+         'f84eec40136ddecf5bdbe1e488c2964c')
 sha256sums=('57746aa8c60bc6f5596ec6c6c7274de40fa12edce5ec33fe899e704fa00b5357'
             '7fdc119cf53c8ca65396ea73f6d10af641ba41ea1dd2bd44a824726e01c8b3f2'
             '68570caa94de6956aa942c34889e3cf98198d934faa29419cdd90f2bdb9801b6'
-            'bdce7f2e8890a3f2a25c885005b43ab732ab6b9ec48504f813a8d5c1ff568e6e')
+            'bdce7f2e8890a3f2a25c885005b43ab732ab6b9ec48504f813a8d5c1ff568e6e'
+            '862ae5e087b860312be5d20c924aa73d191360c8568e3d360f8f81627778c063')
 
 build() {
     cd "${srcdir}"/Mesa-7.11.2
     patch -Np1 -i ../mesa-dri1-symbol-visibility.patch
     patch -Np1 -i ../mesa-dri1-python3.patch
+    patch -Np1 -i ../riscv-mmio-barrier.patch
     autoreconf -vfi
     ./configure --prefix=/usr \
     --with-dri-driverdir=/usr/lib/xorg/modules/dri \
