diff -uprN Mesa-7.11.2/src/mesa/drivers/dri/sis/sis_context.h Mesa-7.11.2.fix/src/mesa/drivers/dri/sis/sis_context.h
--- Mesa-7.11.2/src/mesa/drivers/dri/sis/sis_context.h	2011-10-15 02:43:58.000000000 +0200
+++ Mesa-7.11.2.fix/src/mesa/drivers/dri/sis/sis_context.h	2023-03-08 03:18:47.689528831 +0100
@@ -404,6 +404,8 @@ struct sis_context
 #define MMIO_WMB()	__asm __volatile("" : : : "memory")
 #elif defined(__ia64__)
 #define MMIO_WMB()	__asm __volatile("mf" : : : "memory")
+#elif defined(__riscv)
+#define MMIO_WMB()	__asm __volatile("fence io, iorw" : : : "memory")
 #else
 #error platform needs WMB
 #endif
