diff --git PKGBUILD PKGBUILD
index 6979a2a..d9e630f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,18 @@ license=('BSD')
 depends=('readline' 'zlib' 'bzip2' 'bash')
 provides=(libpcre2-{8,16,32,posix}.so)
 options=(staticlibs)
-source=("https://github.com/PhilipHazel/pcre2/releases/download/$pkgname-$pkgver/$pkgname-$pkgver.tar.bz2"{,.sig})
+source=("https://github.com/PhilipHazel/pcre2/releases/download/$pkgname-$pkgver/$pkgname-$pkgver.tar.bz2"{,.sig}
+        sljit-sv48-sv57.patch)
 sha512sums=('72fbde87fecec3aa4b47225dd919ea1d55e97f2cbcf02aba26e5a0d3b1ffb58c25a80a9ef069eb99f9cf4e41ba9604ad06a7ec159870e1e875d86820e12256d3'
-            'SKIP')
+            'SKIP'
+            'd643c79da00da625b6f8cb798e09ee49ed5900fc7eb969d801f1cdd82edc6f6dc75d75faf6f0cadf13ab2f4e96c4c80381e9df264b0885c84851770367812302')
 validpgpkeys=('45F68D54BBE23FB3039B46E59766E084FB0F43D8')  # Philip Hazel <ph10@hermes.cam.ac.uk>
 
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i "$srcdir"/sljit-sv48-sv57.patch
+}
+
 build() {
   cd $pkgname-$pkgver
 
