--- PKGBUILD
+++ PKGBUILD
@@ -27,7 +27,7 @@ sha512sums=('2a0d6d0879f8845329315947c06180e8a428c2670daad370b5f5c231eef48ab7039
             'ec5a800529a297c709fa383c094ecf106351cf0f8ac7b613b972d415d77fe001088902d7ab805e63e78a8e6360323fec1b795db5a4446df1e21b9b4ed31e7079'
             'ef872aedb2bc022a29292b7972a792b22e684c1ccb904a2b2cfec6d8966c28fb19be1452ce060821c419f1b646b236ba2e783175595e4bb6926d164c27a15c87'
             'c4292b8f56ee955ed7385a49843fd90d6434029891b3e1e724cb2fc841514c06e2554a26d3937c114371b18c2168c4e64319eb2cbd726ee8b35870df19089348'
-            '3279545adae32d0cdbac49e7ae36e6ace99b1cad12da4addd0c9fcca7dc405a4ebc2205fee7177490191069c4645da9276870abaf3a9704c64e62326fa4c4647')
+            '39aa5761a725fab8e9cf3eabd7c47d64b91489c03c059747849ec4a4b1032e44f2770a238338fb43c963516c6e941fd82157b91128c2c1899ba5448a9dc49979')
 
 export CGO_LDFLAGS="${LDFLAGS}"
 export CGO_CFLAGS="${CFLAGS}"
@@ -37,8 +37,8 @@ export CGO_CXXFLAGS="${CXXFLAGS}"
 export GOFLAGS="-trimpath -mod=readonly -modcacherw"
 
 export GOOS='linux'
-export GOARCH='amd64'
-export XC_OSARCH='linux/amd64'
+export GOARCH='riscv64'
+export XC_OSARCH='linux/riscv64'
 
 prepare() {
   cd "${srcdir}/${pkgname}"
@@ -51,6 +51,10 @@ prepare() {
   done
 
   mkdir -p build
+
+  go mod edit -replace=github.com/boltdb/bolt=go.etcd.io/bbolt@v1.3.6
+  go mod edit -replace=go.etcd.io/bbolt=github.com/coreos/bbolt@v1.3.6
+  go mod tidy
 }
 
 build() {
