diff --git PKGBUILD PKGBUILD
index 6bdeeb3d..ffceab60 100644
--- PKGBUILD
+++ PKGBUILD
@@ -34,8 +34,8 @@ export CGO_CXXFLAGS="${CXXFLAGS}"
 export GOFLAGS="-trimpath -mod=readonly -modcacherw"
 
 export GOOS='linux'
-export GOARCH='amd64'
-export XC_OSARCH='linux/amd64'
+export GOARCH='riscv64'
+export XC_OSARCH='linux/riscv64'
 
 prepare() {
   cd "${srcdir}/${pkgname}"
@@ -48,6 +48,9 @@ prepare() {
   done
 
   mkdir -p build
+
+  go mod edit -replace=github.com/boltdb/bolt=go.etcd.io/bbolt@v1.3.6
+  go mod tidy
 }
 
 build() {
