diff --unified --recursive --text a/libs/miniwebrtc/typedefs.h b/libs/miniwebrtc/typedefs.h
--- a/libs/miniwebrtc/typedefs.h	2021-10-06 09:29:07.129646004 +0000
+++ b/libs/miniwebrtc/typedefs.h	2021-10-06 09:31:16.666367836 +0000
@@ -99,6 +99,9 @@
 #define WEBRTC_ARCH_BIG_ENDIAN
 #define WEBRTC_BIG_ENDIAN
 #endif
+#elif defined(__riscv)
+#define WEBRTC_RISCV_64_BITS 
+#define WEBRTC_RISCV_LITTLE_ENDIAN 
 
 #else
 #error Please add support for your architecture in typedefs.h
