diff --git PKGBUILD PKGBUILD
index 9ba721593..e1cff6b2b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -86,9 +86,16 @@ backup=('etc/yate/accfile.conf'
 depends=('speex' 'gsm' 'postgresql-libs' 'mariadb-libs' 'alsa-lib' 'openssl' 'sqlite')
 options=(!makeflags)
 source=("http://yate.null.ro/tarballs/yate6/yate-${pkgver}-1.tar.gz"
-	"yate.service")
+	"yate.service"
+        "add-riscv-support.patch")
 sha256sums=('8c23dc6bffbf8d478db3a85964b5019771c8f6c9acf5220f3465516a748a03b0'
-            '73eab39a0ba884a4c36ca7c8f1faea4000301a0e795544b74cba78d90de5e302')
+            '73eab39a0ba884a4c36ca7c8f1faea4000301a0e795544b74cba78d90de5e302'
+            '697b7fdf9aa2e10eaefc02a95eec033f5e537cf47af9749cd3725511c1d01014')
+
+prepare(){
+  cd "$srcdir/${pkgname}"
+  patch -Np1 -i ../add-riscv-support.patch
+}
 
 build() {
   cd "$srcdir/${pkgname}"
