--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,11 @@ b2sums=('18432893bc3fa7bf94b7a32566ea548c5abd228e646a63e5c82f1af30f777980d6f6974
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+
+  cargo fetch --locked
 }
 
 build() {
