--- PKGBUILD
+++ PKGBUILD
@@ -25,7 +25,7 @@ build() {
     -D efi_sbat_distro_summary='Arch Linux' \
     -D efi_sbat_distro_pkgname=${pkgname} \
     -D efi_sbat_distro_version=${pkgver} \
-    -D efi_sbat_distro_url="https://archlinux.org/packages/community/x86_64/${pkgname}/" \
+    -D efi_sbat_distro_url="https://archlinux.org/packages/community/$CARCH/${pkgname}/" \
     -D genpeimg=disabled
   ninja -C build
 }
