diff --git PKGBUILD PKGBUILD
index 6bffec0..8c7b3ee 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +19,7 @@ source=(https://botan.randombit.net/releases/Botan-${pkgver}.tar.xz{,.asc})
 sha256sums=('e26e00cfefda64082afdd540d3c537924f645d6a674afed2cd171005deff5560'
             'SKIP')
 validpgpkeys=('621DAF6411E1851C4CF9A2E16211EBF1EFBADFBC') # Botan Distribution Key
+options=(!lto)
 
 build() {
   cd ${pkgname^}-$pkgver
