--- PKGBUILD
+++ PKGBUILD
@@ -59,6 +59,7 @@
 
 prepare() {
   cd orca
+  patch -Np1 -i "$srcdir/increase-unit-test-timeout.patch"
 }
 
 build() {
@@ -72,11 +73,14 @@
 
 check() {
   dbus-run-session xvfb-run -s '-nolisten local' \
-    meson test -C build --print-errorlogs
+    meson test -C build --print-errorlogs --timeout-multiplier 3
 }
 
 package() {
   meson install -C build --destdir "$pkgdir"
 }
 
+source+=("increase-unit-test-timeout.patch::https://gitlab.gnome.org/GNOME/orca/-/commit/6876083e1688ea4ab41919a82179b0bb13ebeb4c.patch")
+b2sums+=('c78dc29167bfc72d468648b7d9edfc79e0d8855656b6a451a68efb3a525b077150e3b7e4e9a724d810f027ef1feb582d353625131753d30e18fcebbdaf6936b8')
+
 # vim:set sw=2 sts=-1 et:
