diff --git PKGBUILD PKGBUILD
index ffca360..10dbe4d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,7 @@ source=(git+https://github.com/fcitx/mozc.git#commit=${_mozc_commit}
         git+https://github.com/open-source-parsers/jsoncpp.git#commit=${_jsoncpp_commit}
         git+https://github.com/google/protobuf.git#commit=${_protobuf_commit}
         git+https://github.com/abseil/abseil-cpp.git#commit=${_abseil_cpp_commit}
+        "add_limits_header.patch"
 	)
 sha512sums=('SKIP'
             'cadb43138597371d13d21a2766ba38f0940a73bd961a1142f3713f700d1b8e75bdb6ccc0600ea57518ad5bf1931eed329cd11faeb87b191aa460e379ed1fed93'
@@ -51,7 +52,8 @@ sha512sums=('SKIP'
             'SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            '5b1d503865dbfdd0158d27877e5c899439933fe7576d72ae34e80e9c50495fab975c7d1b3f67491cd47b56e06d7f0b4521c85ca3048188159cc5ae3125e0dc65')
 validpgpkeys=('2CC8A0609AD2A479C65B6D5C8E8B898CBF2412F9')  # Weng Xuetian
 
 pkgver(){
@@ -85,6 +87,10 @@ prepare() {
   
   ## use libstdc++ instead of libc++
   sed "/stdlib=libc++/d;/-lc++/d" -i gyp/common.gypi
+
+  cd third_party/abseil-cpp
+  patch -Np1 -i "$srcdir/add_limits_header.patch"
+  cd -
 }
 
 build() {
