diff -r -u gcc-11.2.0-orig/config/bootstrap-lto-lean.mk gcc-11.2.0/config/bootstrap-lto-lean.mk
--- gcc-11.2.0-orig/config/bootstrap-lto-lean.mk	2022-02-03 00:48:51.968000000 +0000
+++ gcc-11.2.0/config/bootstrap-lto-lean.mk	2022-02-03 05:08:56.844000000 +0000
@@ -1,10 +1,7 @@
 # This option enables LTO for stage4 and LTO for generators in stage3 with profiledbootstrap.
 # Otherwise, LTO is used in only stage3.
 
-STAGE3_CFLAGS += -flto=jobserver
-override STAGEtrain_CFLAGS := $(filter-out -flto=jobserver,$(STAGEtrain_CFLAGS))
-STAGEtrain_GENERATOR_CFLAGS += -flto=jobserver
-STAGEfeedback_CFLAGS += -flto=jobserver
+override STAGEtrain_CFLAGS := $(filter-out -flto=jobserver,$(STAGEtrain_CFLAGS))
 
 # assumes the host supports the linker plugin
 LTO_AR = $$r/$(HOST_SUBDIR)/prev-gcc/gcc-ar$(exeext) -B$$r/$(HOST_SUBDIR)/prev-gcc/
diff -r -u gcc-11.2.0-orig/config/bootstrap-lto.mk gcc-11.2.0/config/bootstrap-lto.mk
--- gcc-11.2.0-orig/config/bootstrap-lto.mk	2022-02-03 00:48:51.940000000 +0000
+++ gcc-11.2.0/config/bootstrap-lto.mk	2022-02-03 05:09:22.652000000 +0000
@@ -1,10 +1,5 @@
 # This option enables LTO for stage2 and stage3 in slim mode
 
-STAGE2_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGE3_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGEprofile_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGEtrain_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGEfeedback_CFLAGS += -flto=jobserver -frandom-seed=1
 
 # assumes the host supports the linker plugin
 LTO_AR = $$r/$(HOST_SUBDIR)/prev-gcc/gcc-ar$(exeext) -B$$r/$(HOST_SUBDIR)/prev-gcc/
diff -r -u gcc-11.2.0-orig/config/bootstrap-lto-noplugin.mk gcc-11.2.0/config/bootstrap-lto-noplugin.mk
--- gcc-11.2.0-orig/config/bootstrap-lto-noplugin.mk	2022-02-03 00:48:51.932000000 +0000
+++ gcc-11.2.0/config/bootstrap-lto-noplugin.mk	2022-02-03 05:09:34.616000000 +0000
@@ -1,9 +1,4 @@
 # This option enables LTO for stage2 and stage3 on
 # hosts without linker plugin support.
 
-STAGE2_CFLAGS += -flto=jobserver -frandom-seed=1 -ffat-lto-objects
-STAGE3_CFLAGS += -flto=jobserver -frandom-seed=1 -ffat-lto-objects
-STAGEprofile_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGEtrain_CFLAGS += -flto=jobserver -frandom-seed=1
-STAGEfeedback_CFLAGS += -flto=jobserver -frandom-seed=1
 do-compare = /bin/true
