--- PKGBUILD
+++ PKGBUILD
@@ -55,8 +55,6 @@ makedepends=(
   gcc-ada
   gcc-d
   git
-  lib32-gcc-libs
-  lib32-glibc
   libisl
   libmpc
   python
@@ -92,6 +90,12 @@ pkgver() {
   echo "$(cat gcc/BASE-VER)+$(git describe --abbrev=12 --tags | sed 's/[^-]*-[^-]*-//;s/[^-]*-/r&/;s/-/+/g;s/_/./')"
 }

+for i in "${!pkgname[@]}"; do
+  if [[ ${pkgname[i]} = "lib32-gcc-libs" ]] || [[ ${pkgname[i]} = "libhwasan" ]] || [[ ${pkgname[i]} = "libquadmath" ]]; then
+    unset 'pkgname[i]'
+  fi
+done
+
 prepare() {
   [[ ! -d gcc ]] && ln -s gcc-${pkgver/+/-} gcc
   cd gcc
@@ -116,7 +120,7 @@ build() {
       --libexecdir=/usr/lib
       --mandir=/usr/share/man
       --infodir=/usr/share/info
-      --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/gcc/-/issues
+      --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues
       --with-build-config=bootstrap-lto
       --with-gcc-major-version-only
       --with-linker-hash-style=gnu
@@ -132,7 +136,7 @@ build() {
       --enable-link-serialization=1
       --enable-linker-build-id
       --enable-lto
-      --enable-multilib
+      --disable-multilib
       --enable-plugin
       --enable-shared
       --enable-threads=posix
@@ -212,7 +216,6 @@ package_gcc() {
   depends=(
     "libasan=$pkgver-$pkgrel"
     "libgcc=$pkgver-$pkgrel"
-    "libhwasan=$pkgver-$pkgrel"
     "liblsan=$pkgver-$pkgrel"
     "libstdc++=$pkgver-$pkgrel"
     "libtsan=$pkgver-$pkgrel"
@@ -226,9 +229,6 @@ package_gcc() {
     zlib
     zstd
   )
-  optdepends=(
-    'lib32-gcc-libs: for generating code for 32-bit ABI'
-  )
   provides=(
     $pkgname-multilib
   )
@@ -251,30 +251,29 @@ package_gcc() {
     # where we ALSO ship a matching static lib in the internal _libdir
     for _so in libatomic.so libstdc++.so libgdruntime.so libgphobos.so; do
       ln -srv usr/lib/$_so $_libdir/$_so
-      ln -srv usr/lib32/$_so $_libdir/32/$_so
     done

     _pick gcc-ada usr/bin/gnat{,bind,chop,clean,kr,link,ls,make,name,prep}
-    _pick gcc-ada $_libdir/{,32/}ada_target_properties
-    _pick gcc-ada $_libdir/{,32/}ada{include,lib}/
+    _pick gcc-ada $_libdir/ada_target_properties
+    _pick gcc-ada $_libdir/ada{include,lib}/
     _pick gcc-ada $_libdir/gnat1
     _pick gcc-ada usr/share/info/gnat{-style,_rm,_ugn}.info

     _pick gcc-d usr/bin/{gdc,$CHOST-gdc}
     _pick gcc-d $_libdir/d21
     _pick gcc-d $_libdir/include/d/
-    _pick gcc-d $_libdir/{,32/}libg{druntime,phobos}.so
-    _pick gcc-d usr/lib{,32}/libg{druntime,phobos}.a
-    _pick gcc-d usr/lib{,32}/libgphobos.spec
+    _pick gcc-d $_libdir/libg{druntime,phobos}.so
+    _pick gcc-d usr/lib/libg{druntime,phobos}.a
+    _pick gcc-d usr/lib/libgphobos.spec
     _pick gcc-d usr/share/info/gdc.info
     _pick gcc-d usr/share/man/man1/gdc.1

     _pick gcc-fortran usr/bin/{gfortran,$CHOST-gfortran}
-    _pick gcc-fortran $_libdir/{,32/}finclude/
-    _pick gcc-fortran $_libdir/{,32/}libcaf_{shmem,single}.a
+    _pick gcc-fortran $_libdir/finclude/
+    _pick gcc-fortran $_libdir/libcaf_{shmem,single}.a
     _pick gcc-fortran $_libdir/f951
     _pick gcc-fortran $_libdir/include/ISO_Fortran_binding.h
-    _pick gcc-fortran usr/lib{,32}/libgfortran.spec
+    _pick gcc-fortran usr/lib/libgfortran.spec
     _pick gcc-fortran usr/share/info/gfortran.info
     _pick gcc-fortran usr/share/man/man1/gfortran.1

@@ -287,8 +286,8 @@ package_gcc() {

     _pick gcc-go usr/bin/{gccgo,go,gofmt,$CHOST-gccgo}
     _pick gcc-go $_libdir/{buildid,cgo,go1,test2json,vet}
-    _pick gcc-go usr/lib{,32}/go/
-    _pick gcc-go usr/lib{,32}/lib{go,gobegin,golibbegin}.a
+    _pick gcc-go usr/lib/go/
+    _pick gcc-go usr/lib/lib{go,gobegin,golibbegin}.a
     _pick gcc-go usr/share/info/gccgo.info
     _pick gcc-go usr/share/man/man1/{gccgo,go,gofmt}.1

@@ -303,21 +302,6 @@ package_gcc() {
     _pick gcc-rust usr/bin/{gccrs,$CHOST-gccrs}
     _pick gcc-rust $_libdir/crab1

-    _pick lib32-gcc-libs usr/lib32/libasan.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libatomic.so*
-    _pick lib32-gcc-libs $_libdir/32/libatomic.so
-    _pick lib32-gcc-libs usr/lib32/libgcc_s.so.*
-    _pick lib32-gcc-libs usr/lib32/libgdruntime.so*
-    _pick lib32-gcc-libs usr/lib32/libgfortran.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libgo.so*
-    _pick lib32-gcc-libs usr/lib32/libgomp.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libgphobos.so*
-    _pick lib32-gcc-libs usr/lib32/libitm.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libobjc.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libquadmath.{a,so*}
-    _pick lib32-gcc-libs usr/lib32/libstdc++.so*
-    _pick lib32-gcc-libs usr/lib32/libubsan.{a,so*}
-
     _pick libasan usr/lib/libasan.{a,so*}

     _pick libatomic usr/lib/libatomic.so*
@@ -333,8 +317,8 @@ package_gcc() {

     _pick libgfortran usr/lib/libgfortran.{a,so*}

-    _pick libgm2 $_libdir/{,32/}m2/
-    _pick libgm2 usr/lib{,32}/libm2{cor,iso,log,min,pim}.{a,so*}
+    _pick libgm2 $_libdir/m2/
+    _pick libgm2 usr/lib/libm2{cor,iso,log,min,pim}.{a,so*}

     _pick libgo usr/lib/libgo.so*

@@ -344,8 +328,6 @@ package_gcc() {
     _pick libgphobos usr/lib/libgdruntime.so*
     _pick libgphobos usr/lib/libgphobos.so*

-    _pick libhwasan usr/lib/libhwasan.{a,so*}
-
     _pick libitm usr/lib/libitm.{a,so*}
     _pick libitm usr/share/info/libitm.info

@@ -353,9 +335,6 @@ package_gcc() {

     _pick libobjc usr/lib/libobjc.{a,so*}

-    _pick libquadmath usr/lib/libquadmath.{a,so*}
-    _pick libquadmath usr/share/info/libquadmath.info
-
     _pick libstdc++ usr/lib/libstdc++.so*
     _pick libstdc++ usr/share/locale/{de,fr}/LC_MESSAGES/libstdc++.mo

@@ -368,17 +347,11 @@ package_gcc() {

     # move specific files into the internal "_libdir"
     mv -v usr/lib/libgcc_s{,_asneeded}.so $_libdir/
-    mv -v usr/lib32/libgcc_s{,_asneeded}.so $_libdir/32/
     mv -v usr/lib/libatomic{,_asneeded}.a usr/lib/libatomic_asneeded.so $_libdir/
-    mv -v usr/lib32/libatomic{,_asneeded}.a usr/lib32/libatomic_asneeded.so $_libdir/32/
     mv -v usr/lib/*_preinit.o $_libdir/
-    mv -v usr/lib32/*_preinit.o $_libdir/32/
     mv -v usr/lib/*.spec $_libdir/
-    mv -v usr/lib32/*.spec $_libdir/32/
     mv -v usr/lib/libstdc++{,exp,fs}.a $_libdir/
-    mv -v usr/lib32/libstdc++{,exp,fs}.a $_libdir/32/
     mv -v usr/lib/libsupc++.a $_libdir/
-    mv -v usr/lib32/libsupc++.a $_libdir/32/
   )

   install -d "$pkgdir"/usr/share/gdb/auto-load/usr/lib
@@ -445,12 +418,6 @@ package_gcc-ada() {
   ln -s libgnat-${pkgver%%.*}.so "$pkgdir"/usr/lib/libgnat.so
   rm -f "$pkgdir"/$_libdir/adalib/libgna{rl,t}.so

-  install -d "$pkgdir"/usr/lib32/
-  mv "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}-${pkgver%%.*}.so "$pkgdir"/usr/lib32
-  ln -s libgnarl-${pkgver%%.*}.so "$pkgdir"/usr/lib32/libgnarl.so
-  ln -s libgnat-${pkgver%%.*}.so "$pkgdir"/usr/lib32/libgnat.so
-  rm -f "$pkgdir"/$_libdir/32/adalib/libgna{rl,t}.so
-
   _install_runtime_library_exception
 }

@@ -479,11 +446,8 @@ package_gcc-d() {
   )

   mv -v $pkgname/* "$pkgdir"
-  mkdir -pv "$pkgdir"/$_libdir/32
   mv -v "$pkgdir"/usr/lib/libg{druntime,phobos}.a "$pkgdir"/$_libdir/
-  mv -v "$pkgdir"/usr/lib32/libg{druntime,phobos}.a "$pkgdir"/$_libdir/32
   mv -v "$pkgdir"/usr/lib/libgphobos.spec "$pkgdir"/$_libdir/
-  mv -v "$pkgdir"/usr/lib32/libgphobos.spec "$pkgdir"/$_libdir/32
   _install_runtime_library_exception
 }

@@ -510,7 +474,6 @@ package_gcc-fortran() {
   mv -v $pkgname/* "$pkgdir"
   ln -s gfortran "$pkgdir"/usr/bin/f95
   mv -v "$pkgdir"/usr/lib/libgfortran.spec "$pkgdir"/$_libdir/
-  mv -v "$pkgdir"/usr/lib32/libgfortran.spec "$pkgdir"/$_libdir/32
   _install_runtime_library_exception
 }

@@ -570,10 +533,8 @@ package_gcc-libs() {
     libgcc
     libgfortran
     libgomp
-    libhwasan
     liblsan
     libobjc
-    libquadmath
     libstdc++
     libtsan
     libubsan
