Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -12,7 +12,7 @@
          libxkbcommon graphene iso-codes tracker3 libcolord wayland
          wayland-protocols libxrandr libx11 libxrender libxi libxext libxcursor
          libxdamage libxfixes fontconfig libxinerama libcloudproviders libcups
-         rest json-glib gst-plugins-bad-libs librsvg dconf shared-mime-info
+         rest json-glib librsvg dconf shared-mime-info
          desktop-file-utils adwaita-icon-theme cantarell-fonts)
 makedepends=(git meson gi-docgen shaderc sassc gobject-introspection
              docbook-xsl)
@@ -43,6 +43,7 @@
     -D colord=enabled
     -D gtk_doc=true
     -D man-pages=true
+    -D media-gstreamer=disabled
   )
 
   arch-meson gtk build "${meson_options[@]}"
