diff --git PKGBUILD.orig PKGBUILD
--- PKGBUILD.orig
+++ PKGBUILD
@@ -12,7 +12,7 @@ depends=(glib2 cairo pango fribidi gdk-pixbuf2 libepoxy libgl libegl harfbuzz
          libxkbcommon graphene iso-codes tracker3 libcolord wayland libxrandr
          libx11 libxrender libxi libxext libxcursor libxdamage libxfixes
          fontconfig libxinerama libcloudproviders libcups rest json-glib
-         gst-plugins-bad-libs librsvg dconf shared-mime-info desktop-file-utils
+         librsvg dconf shared-mime-info desktop-file-utils
          adwaita-icon-theme cantarell-fonts)
 makedepends=(git meson gi-docgen shaderc sassc gobject-introspection docbook-xsl
              wayland-protocols)
@@ -32,6 +32,7 @@ pkgver() {
 
 prepare() {
   cd gtk
+  git cherry-pick -n 49032b4b6349ee196289a237d9baa74f90ac589d
 }
 
 build() {
@@ -43,6 +44,7 @@ build() {
     -D colord=enabled
     -D gtk_doc=true
     -D man-pages=true
+    -D media-gstreamer=disabled
   )
 
   arch-meson gtk build "${meson_options[@]}"
