diff --git PKGBUILD PKGBUILD
index 7b0620a..252afb3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,13 @@
 sha512sums=('ebd326173756019ae23a9826735c000c80a70d780d00b82aa4ce1be0cc8970d779e006bb2f564ef5969da92866e3c499be32553823ccbe769400fa3578a913d1')
 b2sums=('a924d576880eb455edc3a1cb4d0300765eff7aa4456d05753741dc2c25dcbaf25ef002b88536390f04f8b8b3361a664f1e0561c88e5ff0b7bef9aed2d8e2da32')
 
+prepare() {
+  cd "$pkgname"
+
+  # rv64: use pc-relative addressing for globals
+  git cherry-pick -n e786f06032fefa2e3790d6b1c9e31ed138f475a6
+}
+
 build() {
   cd "$pkgname"
 
