diff --git a/util/pid_lock.c b/util/pid_lock.c
index 5982fea..01a2a77 100644
--- a/util/pid_lock.c
+++ b/util/pid_lock.c
@@ -26,7 +26,7 @@ RESULT pid_lock_init(const char* pidfile) {
         return SUCCESS;
     }
 
-    pid_lock_fd = open(pidfile, O_RDWR | O_CREAT);
+    pid_lock_fd = open(pidfile, O_RDWR | O_CREAT, 0600);
     if (pid_lock_fd < 0) {
         PR_ERRNO("无法打开 PID 文件");
         return FAILURE;
