diff --git PKGBUILD PKGBUILD
index c79d1dd..32f5be4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,15 @@ url="https://github.com/updateing/minieap"
 license=('GPL')
 depends=('glibc' 'coreutils')
 backup=(etc/minieap.conf)
-source=("$pkgname-$pkgver.tar.gz::https://github.com/updateing/minieap/archive/v$pkgver.tar.gz")
-sha512sums=('986ae4b031fe40ea4aafcb1fbc2a95f50f5fb7fa45a8a4092aef7275b8befd98ad702764f76749a2d5b350ed7c673b51de52306d0a8bdf19794afdf523f44869')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/updateing/minieap/archive/v$pkgver.tar.gz"
+        "fix_lock_mode.patch")
+sha512sums=('986ae4b031fe40ea4aafcb1fbc2a95f50f5fb7fa45a8a4092aef7275b8befd98ad702764f76749a2d5b350ed7c673b51de52306d0a8bdf19794afdf523f44869'
+            "59cb0c9e7b37e57532c10f665c7d9d60b7e2306b159c6b5cbd46ec60d2976045fccb41f627ff0fc4e6e7e67ea8482ad19b7cbaf97e0dca13418efbdab0b84d88")
+
+prepare() {
+  cd minieap-$pkgver
+  patch -Np1 -i "../fix_lock_mode.patch"
+}
 
 build() {
   cd minieap-$pkgver
