--- PKGBUILD	2023-09-10 00:55:26.455719410 -0700
+++ PKGBUILD-riscv	2023-09-10 00:52:21.902014518 -0700
@@ -6,13 +6,12 @@
 pkgver=1.15.0
 pkgrel=3
 pkgdesc='Reference frontend for the libretro API'
-arch=(x86_64)
+arch=('x86_64' 'riscv64')
 url=http://www.libretro.com/
 license=(GPL)
 groups=(libretro)
 depends=(
   alsa-lib
-  ffmpeg
   flac
   libass.so
   libdrm
@@ -31,7 +30,6 @@
   zlib
 )
 makedepends=(
-  gamemode
   git
   libdecor
   libx11
@@ -88,8 +86,10 @@
     --disable-cg \
     --disable-jack \
     --disable-oss \
-    --disable-sdl \
-    --enable-dbus
+    --enable-dbus \
+    --enable-opengles \
+    --enable-egl \
+    --disable-ffmpeg
   make
   make -C libretro-common/audio/dsp_filters
   make -C gfx/video_filters
