--- PKGBUILD
+++ PKGBUILD
@@ -30,3 +30,11 @@ package() {
   python setup.py install --prefix=/usr --root="${pkgdir}" --optimize=1 --skip-build
   install -Dm644 LICENSE.txt -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+source+=(python-zict-fix-map-size.patch)
+sha256sums+=('9758577d61a8ad48ef054a44e0ba8c4b04435e87f372837f46c9b176ca002b4f')
+
+prepare() {
+  cd ${_pkg}-${pkgver}
+  patch -Np1 -i ../python-zict-fix-map-size.patch
+}
