diff --git PKGBUILD PKGBUILD
index b579333..48818ad 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,11 @@
 
 build() {
   cd $pkgname/cmd/$pkgname
-  CGO_ENABLED=0 go build -v -mod=vendor -trimpath -buildmode=pie -ldflags="-s -w -extldflags '$LDFLAGS'"
+  export CGO_CPPFLAGS="${CPPFLAGS}"
+  export CGO_CFLAGS="${CFLAGS}"
+  export CGO_CXXFLAGS="${CXXFLAGS}"
+  export CGO_LDFLAGS="${LDFLAGS}"
+  go build -v -mod=vendor -trimpath -buildmode=pie -ldflags="-s -w -extldflags '$LDFLAGS'"
 }
 
 package() {
