--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,13 @@
 
 prepare() {
   cd $pkgname/$pkgname
-  pnpm install --frozen-lockfile
+  rm -r artifacts
+  sed -i '/@yao-pkg\/pkg/d' ../package.json
+  sed -i '/@yao-pkg\/pkg/d' ../pnpm-workspace.yaml
+  sed -i '/patchedDependencies/d' ../pnpm-workspace.yaml
+  sed -i '/__patches__/d' ../pnpm-workspace.yaml
+  sed -i '/patchesDir/d' ../pnpm-workspace.yaml
+  pnpm install
 }
 
 build() {
