--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,12 @@
 
 prepare() {
   cd $pkgname/$pkgname
-  pnpm install --frozen-lockfile
+  rm -r artifacts
+
+  # Upstream Node.js and tsgo binaries are unavailable for riscv64.
+  sed -i '/"runtime": {/,/^    }/s/"download"/"ignore"/' ../package.json
+  sed -i 's/tsgo --build/tsc --build/; s/pnx node@runtime:24.6.0/node/' package.json
+  pnpm install --no-frozen-lockfile
 }
 
 build() {
