--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ validpgpkeys=(7B74D1299568B586BA9962B5649E4D4AF74E7DEC) # Zoltan Kochan <z@kocha
 
 prepare() {
   cd $pkgname/$pkgname
+  patch -Np1 -d .. -i ../add-pnpm-linux-riscv64.patch
   pnpm install --frozen-lockfile
 }
 
@@ -40,3 +41,6 @@ package() {
   cd dist
   cp -r $pkgname.cjs pnpmrc templates worker.js "$pkgdir"/$mod_dir/dist
 }
+
+source+=(add-pnpm-linux-riscv64.patch)
+b2sums+=('ed9b2d1f81bac9606487339796de750dc047227a1b5be0471906663bb9f30ebca443015a671e7fb1b6e366dee9ec6f5c77bca3fe90dd12b76c6a6daac869c43b')
