diff --git PKGBUILD PKGBUILD
index 7c76792..44b25e4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,9 @@
 
 prepare() {
   cd "$pkgname-$pkgver"
+  # https://github.com/orhun/sixel-rs/pull/6
+  echo -e "\n[patch.crates-io]\nsixel-rs = { git = 'https://github.com/Xeonacid/sixel-rs', rev = '3d6c83716e3d10c95dfbabb9b8f5725219e9a73e' }" >> Cargo.toml
+  cargo update -p sixel-rs
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
