--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,11 @@ depends=('glibc')
 source=(https://downloads.sourceforge.net/sourceforge/mhash/${pkgname}-${pkgver}.tar.bz2)
 sha512sums=('3b063d258cb0e7c2fa21ed30abae97bd6f3630ecd1cb4698afb826aa747555f3cf884828f24ac5e2b203730d0c7c0ecc9ef1e724ad9d85769a2f66128f3072eb')
 
+prepare() {
+  cd ${pkgname}-${pkgver}
+  autoreconf -fi
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   ./configure --prefix=/usr
