--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,9 @@ b2sums=('10dcae7ef8aab28b26bebc39cecd4e1b3ed38195c51e9b50637451f778392d9fd15efb4
 prepare() {
   export RUSTUP_TOOLCHAIN=stable
   cd $_upstream-$pkgver
+  cargo update -p time@0.3.36
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 
 }
