diff --git PKGBUILD PKGBUILD
index 57fd7768..54e79dd7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -62,7 +62,7 @@ package() {
 	## Binaries
 	install -dm755 "$pkgdir"/usr/bin
 	install -m755 bin/fzf-tmux "$pkgdir"/usr/bin/
-	install -Dm755 target/fzf-linux_amd64 "$pkgdir"/usr/bin/fzf
+	install -Dm755 target/fzf-linux_riscv64 "$pkgdir"/usr/bin/fzf
 
 	## Completion and keybindings
 	install -dm755 "$pkgdir"/usr/share/fzf
