diff --git PKGBUILD PKGBUILD
index a241826..6fdc11f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,13 +20,16 @@ optdepends=('fish: fish keybindings'
             'tmux: fzf-tmux script for launching fzf in a tmux pane'
             'vim: plugin'
             'zsh: zsh keybindings')
-source=("git+https://github.com/junegunn/fzf.git#tag=${pkgver}?signed")
+source=("git+https://github.com/junegunn/fzf.git#tag=${pkgver}?signed"
+        "add-riscv64-support.patch::https://patch-diff.githubusercontent.com/raw/junegunn/fzf/pull/2626.patch")
 validpgpkeys=('B5FB4AB5F32EC1A0C9DE9637254BC280FEF9C627')
-sha512sums=('SKIP')
+sha512sums=('SKIP'
+            '4b58694bac9e4f18495338c50a6c1fbcd646cccd6a9410f7a602d610016d2198db1d83292d413d3a64c5026ea695fca45a59eaf9a651cc1439213341fa10e7a4')
 
 prepare() {
 	cd "${pkgname}"
 	sed -i 's/-w /-w -linkmode external /' Makefile
+	patch -Np1 -i ../add-riscv64-support.patch
 }
 
 build() {
@@ -61,7 +64,7 @@ package() {
 	## Binaries
 	install -dm755 "$pkgdir"/usr/bin
 	install -m755 bin/fzf-tmux "$pkgdir"/usr/bin/
-	install -Dm755 target/fzf-linux_amd64 "$pkgdir"/usr/bin/fzf
+	install -Dm755 target/fzf-linux_riscv64 "$pkgdir"/usr/bin/fzf
 
 	## Completion and keybindings
 	install -dm755 "$pkgdir"/usr/share/fzf
@@ -74,4 +77,3 @@ package() {
 	install -Dm644 doc/fzf.txt "$pkgdir"/usr/share/vim/vimfiles/doc/fzf.txt
 	install -Dm644 plugin/fzf.vim "$pkgdir"/usr/share/vim/vimfiles/plugin/fzf.vim
 }
-
