--- PKGBUILD
+++ PKGBUILD
@@ -25,10 +25,12 @@ build() {
 	cd "${srcdir}/${pkgname}-${pkgver}"
 	CFLAGS+=' -fcommon' # https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common
 	CFLAGS+=' -Wno-format-security' # fix build with gettext function
+	autoreconf -fi
 	./configure \
 		--prefix=/usr \
 		--mandir=/usr/share/man \
 		--infodir=/usr/share/info \
+		--disable-dependency-tracking
 
 	make
 }
