Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 467369)
+++ PKGBUILD	(working copy)
@@ -29,10 +29,12 @@
   "git+https://gitlab.gnome.org/GNOME/gnome-control-center.git#commit=$_commit"
   "git+https://gitlab.gnome.org/GNOME/libgnome-volume-control.git"
   pixmaps-dir.diff
+  fix-race.patch::"https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/1632.patch"
 )
 sha256sums=('SKIP'
             'SKIP'
-            'd3d97995f9e4d5bed2e81b196c1e086777fc33dd6b6f6b08bcc931944569ea4a')
+            'd3d97995f9e4d5bed2e81b196c1e086777fc33dd6b6f6b08bcc931944569ea4a'
+            'a54de9ed0efc6979a310b3d232bbda1f2ac4e384c8e9df29bc8488bd22f8d264')
 
 pkgver() {
   cd $pkgname
@@ -48,6 +50,8 @@
   git submodule init subprojects/gvc
   git submodule set-url subprojects/gvc "$srcdir/libgnome-volume-control"
   git -c protocol.file.allow=always submodule update
+
+  patch -Np1 -i ../fix-race.patch
 }
 
 
