--- PKGBUILD
+++ PKGBUILD
@@ -55,6 +55,14 @@
 source=(https://github.com/geopandas/geopandas/archive/refs/tags/v${pkgver}/${pkgname}-${pkgver}.tar.gz)
 sha256sums=('9c169dda2977bc6bd5d365cb5993aebcf3cef971ca713f9b6dea43b795369b03')
 
+source+=(geos-3.15-tests.patch)
+sha256sums+=('9e30aa8357d276301a4a1c07387e61ef6943589f8537f1eb866b29d02857aaf8')
+
+prepare() {
+  cd ${_pkg}-${pkgver}
+  patch -Np1 -i ../geos-3.15-tests.patch
+}
+
 build() {
   cd ${_pkg}-${pkgver}
   python -m build --wheel --no-isolation
