diff --git a/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp b/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp
index 2f40bdf..e7d2d22 100644
--- a/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp
+++ b/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/contrib/gizmos/msw/gizmos_wrap.cpp b/wxPython/contrib/gizmos/msw/gizmos_wrap.cpp
index 88d7d92..aeb4060 100644
--- a/wxPython/contrib/gizmos/msw/gizmos_wrap.cpp
+++ b/wxPython/contrib/gizmos/msw/gizmos_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/contrib/gizmos/osx_carbon/gizmos_wrap.cpp b/wxPython/contrib/gizmos/osx_carbon/gizmos_wrap.cpp
index ef43623..ade4a70 100644
--- a/wxPython/contrib/gizmos/osx_carbon/gizmos_wrap.cpp
+++ b/wxPython/contrib/gizmos/osx_carbon/gizmos_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/contrib/gizmos/osx_cocoa/gizmos_wrap.cpp b/wxPython/contrib/gizmos/osx_cocoa/gizmos_wrap.cpp
index ef43623..ade4a70 100644
--- a/wxPython/contrib/gizmos/osx_cocoa/gizmos_wrap.cpp
+++ b/wxPython/contrib/gizmos/osx_cocoa/gizmos_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/_controls_wrap.cpp b/wxPython/src/gtk/_controls_wrap.cpp
index 24d4dd5..5020b72 100644
--- a/wxPython/src/gtk/_controls_wrap.cpp
+++ b/wxPython/src/gtk/_controls_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/_core_wrap.cpp b/wxPython/src/gtk/_core_wrap.cpp
index b59952b..b43562b 100644
--- a/wxPython/src/gtk/_core_wrap.cpp
+++ b/wxPython/src/gtk/_core_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/_gdi_wrap.cpp b/wxPython/src/gtk/_gdi_wrap.cpp
index ec70a3f..6878cba 100644
--- a/wxPython/src/gtk/_gdi_wrap.cpp
+++ b/wxPython/src/gtk/_gdi_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/_misc_wrap.cpp b/wxPython/src/gtk/_misc_wrap.cpp
index 8be9135..228a437 100644
--- a/wxPython/src/gtk/_misc_wrap.cpp
+++ b/wxPython/src/gtk/_misc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/_windows_wrap.cpp b/wxPython/src/gtk/_windows_wrap.cpp
index 19ad1e6..0209308 100644
--- a/wxPython/src/gtk/_windows_wrap.cpp
+++ b/wxPython/src/gtk/_windows_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/animate_wrap.cpp b/wxPython/src/gtk/animate_wrap.cpp
index e65e2dd..7ac7f72 100644
--- a/wxPython/src/gtk/animate_wrap.cpp
+++ b/wxPython/src/gtk/animate_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/aui_wrap.cpp b/wxPython/src/gtk/aui_wrap.cpp
index 07723ef..034bf6b 100644
--- a/wxPython/src/gtk/aui_wrap.cpp
+++ b/wxPython/src/gtk/aui_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/calendar_wrap.cpp b/wxPython/src/gtk/calendar_wrap.cpp
index 30a817d..750806c 100644
--- a/wxPython/src/gtk/calendar_wrap.cpp
+++ b/wxPython/src/gtk/calendar_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/combo_wrap.cpp b/wxPython/src/gtk/combo_wrap.cpp
index cef1a7d..8c71028 100644
--- a/wxPython/src/gtk/combo_wrap.cpp
+++ b/wxPython/src/gtk/combo_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/dataview_wrap.cpp b/wxPython/src/gtk/dataview_wrap.cpp
index ec7e405..91dad00 100644
--- a/wxPython/src/gtk/dataview_wrap.cpp
+++ b/wxPython/src/gtk/dataview_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/glcanvas_wrap.cpp b/wxPython/src/gtk/glcanvas_wrap.cpp
index e67551d..55732ae 100644
--- a/wxPython/src/gtk/glcanvas_wrap.cpp
+++ b/wxPython/src/gtk/glcanvas_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/grid_wrap.cpp b/wxPython/src/gtk/grid_wrap.cpp
index de46a6b..92b65e2 100644
--- a/wxPython/src/gtk/grid_wrap.cpp
+++ b/wxPython/src/gtk/grid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/html2_wrap.cpp b/wxPython/src/gtk/html2_wrap.cpp
index 58a0332..45204e0 100644
--- a/wxPython/src/gtk/html2_wrap.cpp
+++ b/wxPython/src/gtk/html2_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/html_wrap.cpp b/wxPython/src/gtk/html_wrap.cpp
index 8611aa4..6aa9abc 100644
--- a/wxPython/src/gtk/html_wrap.cpp
+++ b/wxPython/src/gtk/html_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/media_wrap.cpp b/wxPython/src/gtk/media_wrap.cpp
index eb6fb7f..78cdc08 100644
--- a/wxPython/src/gtk/media_wrap.cpp
+++ b/wxPython/src/gtk/media_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/propgrid_wrap.cpp b/wxPython/src/gtk/propgrid_wrap.cpp
index 5711636..3b97640 100644
--- a/wxPython/src/gtk/propgrid_wrap.cpp
+++ b/wxPython/src/gtk/propgrid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/richtext_wrap.cpp b/wxPython/src/gtk/richtext_wrap.cpp
index ebfc730..d8ee767 100644
--- a/wxPython/src/gtk/richtext_wrap.cpp
+++ b/wxPython/src/gtk/richtext_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/stc_wrap.cpp b/wxPython/src/gtk/stc_wrap.cpp
index 1632b97..fcef6d3 100644
--- a/wxPython/src/gtk/stc_wrap.cpp
+++ b/wxPython/src/gtk/stc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/webkit_wrap.cpp b/wxPython/src/gtk/webkit_wrap.cpp
index b5deb77..467b4d2 100644
--- a/wxPython/src/gtk/webkit_wrap.cpp
+++ b/wxPython/src/gtk/webkit_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/wizard_wrap.cpp b/wxPython/src/gtk/wizard_wrap.cpp
index dea9e50..916783d 100644
--- a/wxPython/src/gtk/wizard_wrap.cpp
+++ b/wxPython/src/gtk/wizard_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/gtk/xrc_wrap.cpp b/wxPython/src/gtk/xrc_wrap.cpp
index afc28ac..abf5233 100644
--- a/wxPython/src/gtk/xrc_wrap.cpp
+++ b/wxPython/src/gtk/xrc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/_controls_wrap.cpp b/wxPython/src/msw/_controls_wrap.cpp
index 148892a..30842b7 100644
--- a/wxPython/src/msw/_controls_wrap.cpp
+++ b/wxPython/src/msw/_controls_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/_core_wrap.cpp b/wxPython/src/msw/_core_wrap.cpp
index dc7cb68..a04f5a9 100644
--- a/wxPython/src/msw/_core_wrap.cpp
+++ b/wxPython/src/msw/_core_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/_gdi_wrap.cpp b/wxPython/src/msw/_gdi_wrap.cpp
index 17d13b6..58a0ff4 100644
--- a/wxPython/src/msw/_gdi_wrap.cpp
+++ b/wxPython/src/msw/_gdi_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/_misc_wrap.cpp b/wxPython/src/msw/_misc_wrap.cpp
index 8d0d0b2..db04144 100644
--- a/wxPython/src/msw/_misc_wrap.cpp
+++ b/wxPython/src/msw/_misc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/_windows_wrap.cpp b/wxPython/src/msw/_windows_wrap.cpp
index fefbbd1..5f40c84 100644
--- a/wxPython/src/msw/_windows_wrap.cpp
+++ b/wxPython/src/msw/_windows_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/animate_wrap.cpp b/wxPython/src/msw/animate_wrap.cpp
index a1b5cae..7e57341 100644
--- a/wxPython/src/msw/animate_wrap.cpp
+++ b/wxPython/src/msw/animate_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/aui_wrap.cpp b/wxPython/src/msw/aui_wrap.cpp
index 07723ef..034bf6b 100644
--- a/wxPython/src/msw/aui_wrap.cpp
+++ b/wxPython/src/msw/aui_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/calendar_wrap.cpp b/wxPython/src/msw/calendar_wrap.cpp
index 30a817d..750806c 100644
--- a/wxPython/src/msw/calendar_wrap.cpp
+++ b/wxPython/src/msw/calendar_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/combo_wrap.cpp b/wxPython/src/msw/combo_wrap.cpp
index 0117893..119bdd1 100644
--- a/wxPython/src/msw/combo_wrap.cpp
+++ b/wxPython/src/msw/combo_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/dataview_wrap.cpp b/wxPython/src/msw/dataview_wrap.cpp
index ec7e405..91dad00 100644
--- a/wxPython/src/msw/dataview_wrap.cpp
+++ b/wxPython/src/msw/dataview_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/glcanvas_wrap.cpp b/wxPython/src/msw/glcanvas_wrap.cpp
index 3668cb0..c4487b1 100644
--- a/wxPython/src/msw/glcanvas_wrap.cpp
+++ b/wxPython/src/msw/glcanvas_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/grid_wrap.cpp b/wxPython/src/msw/grid_wrap.cpp
index de46a6b..92b65e2 100644
--- a/wxPython/src/msw/grid_wrap.cpp
+++ b/wxPython/src/msw/grid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/html2_wrap.cpp b/wxPython/src/msw/html2_wrap.cpp
index 58a0332..45204e0 100644
--- a/wxPython/src/msw/html2_wrap.cpp
+++ b/wxPython/src/msw/html2_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/html_wrap.cpp b/wxPython/src/msw/html_wrap.cpp
index 8611aa4..6aa9abc 100644
--- a/wxPython/src/msw/html_wrap.cpp
+++ b/wxPython/src/msw/html_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/media_wrap.cpp b/wxPython/src/msw/media_wrap.cpp
index eb6fb7f..78cdc08 100644
--- a/wxPython/src/msw/media_wrap.cpp
+++ b/wxPython/src/msw/media_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/propgrid_wrap.cpp b/wxPython/src/msw/propgrid_wrap.cpp
index 87ae357..196f334 100644
--- a/wxPython/src/msw/propgrid_wrap.cpp
+++ b/wxPython/src/msw/propgrid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/richtext_wrap.cpp b/wxPython/src/msw/richtext_wrap.cpp
index b1e2b17..9d4c49a 100644
--- a/wxPython/src/msw/richtext_wrap.cpp
+++ b/wxPython/src/msw/richtext_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/stc_wrap.cpp b/wxPython/src/msw/stc_wrap.cpp
index 1632b97..fcef6d3 100644
--- a/wxPython/src/msw/stc_wrap.cpp
+++ b/wxPython/src/msw/stc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/webkit_wrap.cpp b/wxPython/src/msw/webkit_wrap.cpp
index b5deb77..467b4d2 100644
--- a/wxPython/src/msw/webkit_wrap.cpp
+++ b/wxPython/src/msw/webkit_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/wizard_wrap.cpp b/wxPython/src/msw/wizard_wrap.cpp
index dea9e50..916783d 100644
--- a/wxPython/src/msw/wizard_wrap.cpp
+++ b/wxPython/src/msw/wizard_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/msw/xrc_wrap.cpp b/wxPython/src/msw/xrc_wrap.cpp
index afc28ac..abf5233 100644
--- a/wxPython/src/msw/xrc_wrap.cpp
+++ b/wxPython/src/msw/xrc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/_controls_wrap.cpp b/wxPython/src/osx_carbon/_controls_wrap.cpp
index 671b425..8031e48 100644
--- a/wxPython/src/osx_carbon/_controls_wrap.cpp
+++ b/wxPython/src/osx_carbon/_controls_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/_core_wrap.cpp b/wxPython/src/osx_carbon/_core_wrap.cpp
index e8fc4d1..ab19eba 100644
--- a/wxPython/src/osx_carbon/_core_wrap.cpp
+++ b/wxPython/src/osx_carbon/_core_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/_gdi_wrap.cpp b/wxPython/src/osx_carbon/_gdi_wrap.cpp
index d976466..fd88948 100644
--- a/wxPython/src/osx_carbon/_gdi_wrap.cpp
+++ b/wxPython/src/osx_carbon/_gdi_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/_misc_wrap.cpp b/wxPython/src/osx_carbon/_misc_wrap.cpp
index 4443c1f..b15e5bf 100644
--- a/wxPython/src/osx_carbon/_misc_wrap.cpp
+++ b/wxPython/src/osx_carbon/_misc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/_windows_wrap.cpp b/wxPython/src/osx_carbon/_windows_wrap.cpp
index d9bd86a..23bc69d 100644
--- a/wxPython/src/osx_carbon/_windows_wrap.cpp
+++ b/wxPython/src/osx_carbon/_windows_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/animate_wrap.cpp b/wxPython/src/osx_carbon/animate_wrap.cpp
index a1b5cae..7e57341 100644
--- a/wxPython/src/osx_carbon/animate_wrap.cpp
+++ b/wxPython/src/osx_carbon/animate_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/aui_wrap.cpp b/wxPython/src/osx_carbon/aui_wrap.cpp
index eae6330..a94ceb9 100644
--- a/wxPython/src/osx_carbon/aui_wrap.cpp
+++ b/wxPython/src/osx_carbon/aui_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/calendar_wrap.cpp b/wxPython/src/osx_carbon/calendar_wrap.cpp
index 30a817d..750806c 100644
--- a/wxPython/src/osx_carbon/calendar_wrap.cpp
+++ b/wxPython/src/osx_carbon/calendar_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/combo_wrap.cpp b/wxPython/src/osx_carbon/combo_wrap.cpp
index c084b9d..1522acd 100644
--- a/wxPython/src/osx_carbon/combo_wrap.cpp
+++ b/wxPython/src/osx_carbon/combo_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/dataview_wrap.cpp b/wxPython/src/osx_carbon/dataview_wrap.cpp
index ccebf31..8cd5f19 100644
--- a/wxPython/src/osx_carbon/dataview_wrap.cpp
+++ b/wxPython/src/osx_carbon/dataview_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/glcanvas_wrap.cpp b/wxPython/src/osx_carbon/glcanvas_wrap.cpp
index 3a81afb..1f8890c 100644
--- a/wxPython/src/osx_carbon/glcanvas_wrap.cpp
+++ b/wxPython/src/osx_carbon/glcanvas_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/grid_wrap.cpp b/wxPython/src/osx_carbon/grid_wrap.cpp
index 094b506..baaea6f 100644
--- a/wxPython/src/osx_carbon/grid_wrap.cpp
+++ b/wxPython/src/osx_carbon/grid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/html2_wrap.cpp b/wxPython/src/osx_carbon/html2_wrap.cpp
index 58a0332..45204e0 100644
--- a/wxPython/src/osx_carbon/html2_wrap.cpp
+++ b/wxPython/src/osx_carbon/html2_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/html_wrap.cpp b/wxPython/src/osx_carbon/html_wrap.cpp
index ebc4229..9602d16 100644
--- a/wxPython/src/osx_carbon/html_wrap.cpp
+++ b/wxPython/src/osx_carbon/html_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/media_wrap.cpp b/wxPython/src/osx_carbon/media_wrap.cpp
index eb6fb7f..78cdc08 100644
--- a/wxPython/src/osx_carbon/media_wrap.cpp
+++ b/wxPython/src/osx_carbon/media_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/propgrid_wrap.cpp b/wxPython/src/osx_carbon/propgrid_wrap.cpp
index 237b548..a0ede04 100644
--- a/wxPython/src/osx_carbon/propgrid_wrap.cpp
+++ b/wxPython/src/osx_carbon/propgrid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/richtext_wrap.cpp b/wxPython/src/osx_carbon/richtext_wrap.cpp
index 31db770..b61f603 100644
--- a/wxPython/src/osx_carbon/richtext_wrap.cpp
+++ b/wxPython/src/osx_carbon/richtext_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/stc_wrap.cpp b/wxPython/src/osx_carbon/stc_wrap.cpp
index 1632b97..fcef6d3 100644
--- a/wxPython/src/osx_carbon/stc_wrap.cpp
+++ b/wxPython/src/osx_carbon/stc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/webkit_wrap.cpp b/wxPython/src/osx_carbon/webkit_wrap.cpp
index b5deb77..467b4d2 100644
--- a/wxPython/src/osx_carbon/webkit_wrap.cpp
+++ b/wxPython/src/osx_carbon/webkit_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/wizard_wrap.cpp b/wxPython/src/osx_carbon/wizard_wrap.cpp
index ae01880..c4a8f54 100644
--- a/wxPython/src/osx_carbon/wizard_wrap.cpp
+++ b/wxPython/src/osx_carbon/wizard_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_carbon/xrc_wrap.cpp b/wxPython/src/osx_carbon/xrc_wrap.cpp
index afc28ac..abf5233 100644
--- a/wxPython/src/osx_carbon/xrc_wrap.cpp
+++ b/wxPython/src/osx_carbon/xrc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/_controls_wrap.cpp b/wxPython/src/osx_cocoa/_controls_wrap.cpp
index 671b425..8031e48 100644
--- a/wxPython/src/osx_cocoa/_controls_wrap.cpp
+++ b/wxPython/src/osx_cocoa/_controls_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/_core_wrap.cpp b/wxPython/src/osx_cocoa/_core_wrap.cpp
index e8fc4d1..ab19eba 100644
--- a/wxPython/src/osx_cocoa/_core_wrap.cpp
+++ b/wxPython/src/osx_cocoa/_core_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/_gdi_wrap.cpp b/wxPython/src/osx_cocoa/_gdi_wrap.cpp
index d976466..fd88948 100644
--- a/wxPython/src/osx_cocoa/_gdi_wrap.cpp
+++ b/wxPython/src/osx_cocoa/_gdi_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/_misc_wrap.cpp b/wxPython/src/osx_cocoa/_misc_wrap.cpp
index 4443c1f..b15e5bf 100644
--- a/wxPython/src/osx_cocoa/_misc_wrap.cpp
+++ b/wxPython/src/osx_cocoa/_misc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/_windows_wrap.cpp b/wxPython/src/osx_cocoa/_windows_wrap.cpp
index d9bd86a..23bc69d 100644
--- a/wxPython/src/osx_cocoa/_windows_wrap.cpp
+++ b/wxPython/src/osx_cocoa/_windows_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/animate_wrap.cpp b/wxPython/src/osx_cocoa/animate_wrap.cpp
index a1b5cae..7e57341 100644
--- a/wxPython/src/osx_cocoa/animate_wrap.cpp
+++ b/wxPython/src/osx_cocoa/animate_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/aui_wrap.cpp b/wxPython/src/osx_cocoa/aui_wrap.cpp
index eae6330..a94ceb9 100644
--- a/wxPython/src/osx_cocoa/aui_wrap.cpp
+++ b/wxPython/src/osx_cocoa/aui_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/calendar_wrap.cpp b/wxPython/src/osx_cocoa/calendar_wrap.cpp
index 30a817d..750806c 100644
--- a/wxPython/src/osx_cocoa/calendar_wrap.cpp
+++ b/wxPython/src/osx_cocoa/calendar_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/combo_wrap.cpp b/wxPython/src/osx_cocoa/combo_wrap.cpp
index c084b9d..1522acd 100644
--- a/wxPython/src/osx_cocoa/combo_wrap.cpp
+++ b/wxPython/src/osx_cocoa/combo_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/dataview_wrap.cpp b/wxPython/src/osx_cocoa/dataview_wrap.cpp
index ccebf31..8cd5f19 100644
--- a/wxPython/src/osx_cocoa/dataview_wrap.cpp
+++ b/wxPython/src/osx_cocoa/dataview_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/glcanvas_wrap.cpp b/wxPython/src/osx_cocoa/glcanvas_wrap.cpp
index 3a81afb..1f8890c 100644
--- a/wxPython/src/osx_cocoa/glcanvas_wrap.cpp
+++ b/wxPython/src/osx_cocoa/glcanvas_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/grid_wrap.cpp b/wxPython/src/osx_cocoa/grid_wrap.cpp
index 094b506..baaea6f 100644
--- a/wxPython/src/osx_cocoa/grid_wrap.cpp
+++ b/wxPython/src/osx_cocoa/grid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/html2_wrap.cpp b/wxPython/src/osx_cocoa/html2_wrap.cpp
index 58a0332..45204e0 100644
--- a/wxPython/src/osx_cocoa/html2_wrap.cpp
+++ b/wxPython/src/osx_cocoa/html2_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/html_wrap.cpp b/wxPython/src/osx_cocoa/html_wrap.cpp
index ebc4229..9602d16 100644
--- a/wxPython/src/osx_cocoa/html_wrap.cpp
+++ b/wxPython/src/osx_cocoa/html_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/media_wrap.cpp b/wxPython/src/osx_cocoa/media_wrap.cpp
index eb6fb7f..78cdc08 100644
--- a/wxPython/src/osx_cocoa/media_wrap.cpp
+++ b/wxPython/src/osx_cocoa/media_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/propgrid_wrap.cpp b/wxPython/src/osx_cocoa/propgrid_wrap.cpp
index 237b548..a0ede04 100644
--- a/wxPython/src/osx_cocoa/propgrid_wrap.cpp
+++ b/wxPython/src/osx_cocoa/propgrid_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/richtext_wrap.cpp b/wxPython/src/osx_cocoa/richtext_wrap.cpp
index 31db770..b61f603 100644
--- a/wxPython/src/osx_cocoa/richtext_wrap.cpp
+++ b/wxPython/src/osx_cocoa/richtext_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/stc_wrap.cpp b/wxPython/src/osx_cocoa/stc_wrap.cpp
index 1632b97..fcef6d3 100644
--- a/wxPython/src/osx_cocoa/stc_wrap.cpp
+++ b/wxPython/src/osx_cocoa/stc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/webkit_wrap.cpp b/wxPython/src/osx_cocoa/webkit_wrap.cpp
index b5deb77..467b4d2 100644
--- a/wxPython/src/osx_cocoa/webkit_wrap.cpp
+++ b/wxPython/src/osx_cocoa/webkit_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/wizard_wrap.cpp b/wxPython/src/osx_cocoa/wizard_wrap.cpp
index ae01880..c4a8f54 100644
--- a/wxPython/src/osx_cocoa/wizard_wrap.cpp
+++ b/wxPython/src/osx_cocoa/wizard_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
diff --git a/wxPython/src/osx_cocoa/xrc_wrap.cpp b/wxPython/src/osx_cocoa/xrc_wrap.cpp
index afc28ac..abf5233 100644
--- a/wxPython/src/osx_cocoa/xrc_wrap.cpp
+++ b/wxPython/src/osx_cocoa/xrc_wrap.cpp
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg)
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
