diff --git PKGBUILD PKGBUILD
index 4659fa6..f5a0f3a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,10 +14,12 @@ replaces=(wxpython)
 provides=(wxpython=$pkgver)
 source=("https://downloads.sourceforge.net/wxpython/wxPython-src-${pkgver}.tar.bz2"
         "fix-plot.patch"
-        "fix-editra-removal.patch")
+        "fix-editra-removal.patch"
+        "fix_not_literal_string.patch")
 sha256sums=('d54129e5fbea4fb8091c87b2980760b72c22a386cb3b9dd2eebc928ef5e8df61'
             '2d8b2bdbf55172738c7bf93955e552aac61842800c3b8b0a2e1f07f4314abc65'
-            '559b076f206c44e66b33857f49068d66f59a3fc15b97972486e32a8105885554')
+            '559b076f206c44e66b33857f49068d66f59a3fc15b97972486e32a8105885554'
+            '6b45f398d168fa8ae0c3efb05ab4e4cf188fe78ae8c8f53fc8577c1603068456')
 
 prepare() {
     find . -type f -exec sed -i 's/env python/env python2/' {} \;
@@ -28,6 +30,9 @@ prepare() {
 
     # Fix editra removal (FS#63563)
     patch -Np2 -i ../../fix-editra-removal.patch
+
+    # Fix non-literal format strings
+    patch -Np2 -i ../../fix_not_literal_string.patch
 }
 
 build() {
