diff --git PKGBUILD PKGBUILD
index 1204ef7..65277b5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,12 +7,12 @@ _pkgver=${pkgver//./}
 pkgrel=2
 pkgdesc="Japanese outline fonts following the tradition of Japanese printing font"
 arch=('any')
-url="https://ipafont.ipa.go.jp/"
+url="https://moji.or.jp/ipafont/"
 license=('custom')
 conflicts=('ttf-ipaex')
 replaces=('ttf-ipaex')
 groups=('ipa-fonts')
-source=(https://ipafont.ipa.go.jp/IPAexfont/IPAexfont${_pkgver}.zip)
+source=(https://moji.or.jp/wp-content/ipafont/IPAexfont/IPAexfont${_pkgver}.zip)
 sha1sums=('57583c2be5dbfa06648ab0ae4937d7903b32595c')
 
 package() {
