diff --git a/packages/language-server/package.json b/packages/language-server/package.json
index 3e5f7ac4..b92ee5ee 100644
--- a/packages/language-server/package.json
+++ b/packages/language-server/package.json
@@ -5,7 +5,7 @@
     "main": "dist/src/index.js",
     "typings": "dist/src/index",
     "scripts": {
-        "test": "cross-env TS_NODE_TRANSPILE_ONLY=true mocha --require ts-node/register \"test/**/*.ts\" --exclude \"test/**/*.d.ts\"",
+        "test": "cross-env TS_NODE_TRANSPILE_ONLY=true mocha --require ts-node/register \"test/**/*.ts\" --exclude \"test/**/*.d.ts\" --timeout 200000",
         "build": "tsc",
         "prepublishOnly": "npm run build",
         "watch": "tsc -w"
diff --git a/packages/language-server/test/plugins/typescript/features/CompletionProvider.test.ts b/packages/language-server/test/plugins/typescript/features/CompletionProvider.test.ts
index c8b559bb..bfdf5410 100644
--- a/packages/language-server/test/plugins/typescript/features/CompletionProvider.test.ts
+++ b/packages/language-server/test/plugins/typescript/features/CompletionProvider.test.ts
@@ -1040,7 +1040,7 @@ describe('CompletionProviderImpl', function () {
             const item = completions?.items?.[0];
             assert.strictEqual(item?.label, 'abc');
         }
-    }).timeout(4000);
+    }).timeout(400000);
 
     it('provides default slot-let completion for components with type definition', async () => {
         const { completionProvider, document } = setup('component-events-completion-ts-def.svelte');
diff --git a/packages/language-server/test/plugins/typescript/features/DiagnosticsProvider.test.ts b/packages/language-server/test/plugins/typescript/features/DiagnosticsProvider.test.ts
index eca5a832..5209b577 100644
--- a/packages/language-server/test/plugins/typescript/features/DiagnosticsProvider.test.ts
+++ b/packages/language-server/test/plugins/typescript/features/DiagnosticsProvider.test.ts
@@ -55,7 +55,7 @@ describe('DiagnosticsProvider', function () {
 
         const diagnostics3 = await plugin.getDiagnostics(document);
         assert.deepStrictEqual(diagnostics3.length, 1);
-    }).timeout(5000);
+    }).timeout(500000);
 
     it('notices update of imported module', async () => {
         const { plugin, document, lsAndTsDocResolver } = setup(
@@ -81,7 +81,7 @@ describe('DiagnosticsProvider', function () {
         const diagnostics2 = await plugin.getDiagnostics(document);
         assert.deepStrictEqual(diagnostics2.length, 0);
         await lsAndTsDocResolver.deleteSnapshot(newFilePath);
-    }).timeout(5000);
+    }).timeout(500000);
 
     it('notices file changes in all services that reference that file', async () => {
         // Hacky but ensures that this tests is not interfered with by other tests
@@ -134,5 +134,5 @@ describe('DiagnosticsProvider', function () {
         assert.deepStrictEqual(diagnostics3.length, 0);
         const diagnostics4 = await plugin.getDiagnostics(otherDocument);
         assert.deepStrictEqual(diagnostics4.length, 0);
-    }).timeout(5000);
+    }).timeout(500000);
 });
diff --git a/packages/language-server/test/plugins/typescript/test-utils.ts b/packages/language-server/test/plugins/typescript/test-utils.ts
--- a/packages/language-server/test/plugins/typescript/test-utils.ts
+++ b/packages/language-server/test/plugins/typescript/test-utils.ts
@@ -169,7 +169,7 @@
         if (existsSync(inputFile)) {
             const _it = dir.endsWith('.only') ? it.only : it;
             _it(dir.substring(__dirname.length), () => executeTest(inputFile, testOptions)).timeout(
-                5000
+                500000
             );
         } else {
             const _describe = dir.endsWith('.only') ? describe.only : describe;
diff --git a/packages/language-server/test/plugins/typescript/typescript-performance.test.ts b/packages/language-server/test/plugins/typescript/typescript-performance.test.ts
index a94e5dee..0a37cb9a 100644
--- a/packages/language-server/test/plugins/typescript/typescript-performance.test.ts
+++ b/packages/language-server/test/plugins/typescript/typescript-performance.test.ts
@@ -56,5 +56,5 @@
         const end = performance.now();

         console.log(`Performance test took ${end - start}ms`);
-    }).timeout(15000);
+    }).timeout(1500000);
 });
