--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@ depends=('nodejs' 'typescript')
 makedepends=('git' 'npm' 'pnpm')
 conflicts=("nodejs-$pkgname")
 replaces=("nodejs-$pkgname")
-source=("$_name::git+https://github.com/sveltejs/language-tools.git#tag=language-server-$pkgver")
-b2sums=('SKIP')
+source=("$_name::git+https://github.com/sveltejs/language-tools.git#tag=language-server-$pkgver"
+        "increase-test-timeout.patch::https://github.com/sveltejs/language-tools/pull/2200.diff")
+b2sums=('SKIP'
+        'cb81621aee9e9c4b244c90e3bff9f68b592fd10455c79c8d8ff2af63c3ac6ed5675453fab739c12a71acf9f998106460f693ede1019ad02ba1ac6275fb190005')
 
 prepare() {
   cd $_name
+  patch -Np1 -i ../increase-test-timeout.patch
   pnpm install --frozen-lockfile
 }
 
