--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ depends=('nodejs')
 makedepends=('yarn')
 conflicts=("nodejs-$pkgname")
 replaces=("nodejs-$pkgname")
-source=("https://github.com/sveltejs/language-tools/archive/language-server-$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('f4dccd3a1b0d2d162a4d50c88035e7dee6cf6a3f70b9cfcb7e819643b327bd2ff796a7a1eb650d0fdea0a01668c206847f7d51d557bed70558b22555ff41827c')
+source=("https://github.com/sveltejs/language-tools/archive/language-server-$pkgver/$pkgname-$pkgver.tar.gz"
+        "increase-test-timeout.patch")
+b2sums=('f4dccd3a1b0d2d162a4d50c88035e7dee6cf6a3f70b9cfcb7e819643b327bd2ff796a7a1eb650d0fdea0a01668c206847f7d51d557bed70558b22555ff41827c'
+        'c842cb404a84e6c74baf349ef36c035d517ce4fa9a74484bed1367477810c0afa0adf347a6a25035c33856e872c240c6cfb10ed0c75a4553dcfd89154b09ac44')
+
+prepare() {
+  cd language-tools-language-server-$pkgver
+  patch -Np1 -i ../increase-test-timeout.patch
+}
 
 build() {
   cd language-tools-language-server-$pkgver
