--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,11 @@ depends=(gcc-libs)
 source=(https://download.osgeo.org/laszip/${pkgname%2}-src-${pkgver}.tar.bz2)
 sha512sums=('da523243b93296e308eff9121adf9c2a917fc934b21a92a64d192aaa5a1dfbfededc347f5d5ce7b51ea7a5eb5972dc0a5724d50bef26406b359a85c578c60ef8')
 
+prepare() {
+  cd ${pkgname%2}-src-${pkgver}
+  autoreconf -fi
+}
+
 build() {
   cd ${pkgname%2}-src-${pkgver}
   ./configure --prefix=/usr --libdir=/usr/lib/${pkgname} --includedir=/usr/include/${pkgname}
