--- PKGBUILD
+++ PKGBUILD
@@ -27,3 +27,11 @@ package() {
   python setup.py install --root="$pkgdir" --optimize=1
   install -Dm644 LICENSE.txt -t "$pkgdir"/usr/share/licenses/$pkgname
 }
+
+prepare() {
+  cd plop-$pkgver
+  patch -Np1 -i ../relax-time-restrictions.patch
+}
+
+source+=(relax-time-restrictions.patch)
+sha256sums+=('42a9eb6b6c684a752cdab9e3fc45c0a79c143c2df45e7c53ccf9affc354d58c3')
