--- PKGBUILD
+++ PKGBUILD
@@ -14,12 +14,17 @@ checkdepends=('cargo-nextest')
 conflicts=('lychee-link-checker' 'lychee-rs')
 replaces=('lychee-link-checker' 'lychee-rs')
 options=('!lto')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/lycheeverse/lychee/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-b2sums=('0d57e1f47cf45f3926b1b3972bda6e3cac7934bc24d1b6a2061af4d407e18f95dda7c72ad3a031543f1e093fd53725c6b08354d37ed4d64904d27c40d1b49646')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/lycheeverse/lychee/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "$pkgname-fix-flaky-test.patch::https://github.com/lycheeverse/lychee/pull/1079.patch")
+b2sums=('0d57e1f47cf45f3926b1b3972bda6e3cac7934bc24d1b6a2061af4d407e18f95dda7c72ad3a031543f1e093fd53725c6b08354d37ed4d64904d27c40d1b49646'
+        'ec21122e8995df656aca2b90d4950d87c26ff304191243ebfc2710221ebdcf8ff0831e74c50455bde07003b8c3f28d075880c945842f0983b46e3050aa262536')
 
 prepare() {
   cd $pkgname-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  patch -Np1 -i ../$pkgname-fix-flaky-test.patch
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
