diff --git a/test/__init__.py b/test/__init__.py
index 4b2d44e..50d9313 100644
--- a/test/__init__.py
+++ b/test/__init__.py
@@ -14,7 +14,7 @@ from awscrt.io import init_logging, LogLevel
 import unittest
 import sys
 
-TIMEOUT = 10.0
+TIMEOUT = 30.0
 
 
 class NativeResourceTest(unittest.TestCase):
diff --git a/test/test_http_client.py b/test/test_http_client.py
index a9cb459..b9f413a 100644
--- a/test/test_http_client.py
+++ b/test/test_http_client.py
@@ -359,7 +359,7 @@ class TestClient(NativeResourceTest):
         stream.activate()
 
         # wait for stream to complete (use long timeout, it's a big file)
-        stream_completion_result = stream.completion_future.result(60)
+        stream_completion_result = stream.completion_future.result(80)
 
         # check result
         self.assertEqual(200, response.status_code)
