--- a/test/test_http_client.py
+++ b/test/test_http_client.py
@@ -373,7 +373,7 @@
         stream.activate()
 
         # wait for stream to complete (use long timeout, it's a big file)
-        stream_completion_result = stream.completion_future.result(80)
+        stream_completion_result = stream.completion_future.result(300)
 
         # check result
         self.assertEqual(200, response.status_code)
