--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,8 @@ source=("git+https://github.com/awslabs/aws-crt-python.git#tag=$_tag"
         "git+https://github.com/awslabs/aws-c-sdkutils"
         "git+https://github.com/awslabs/aws-checksums"
         "git+https://github.com/awslabs/aws-lc"
-        "git+https://github.com/awslabs/s2n")
+        "git+https://github.com/awslabs/s2n"
+        "fix-timeout.patch")
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
@@ -43,7 +44,8 @@ sha256sums=('SKIP'
             'SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            'add2574f4e5a6a079a8513be38a4372da1f6d0e2450f79022e6430f2894c2620')
 
 pkgver() {
   cd $_pkgname
@@ -52,6 +54,7 @@ pkgver() {
 
 prepare() {
   cd $_pkgname
+  patch -Np1 -i $srcdir/fix-timeout.patch
 
   # https://github.com/awslabs/aws-crt-python/blob/v$pkgver/.gitmodules
   git submodule init
