diff --git PKGBUILD PKGBUILD
index de66f93..aa3c768 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,9 +22,13 @@ replaces=(binutils-multilib)
 backup=('etc/gprofng.rc')
 options=(staticlibs !distcc !ccache)
 source=(git+https://sourceware.org/git/binutils-gdb.git#commit=${_commit}
-        gold-warn-unsupported.patch)
+        gold-warn-unsupported.patch
+        riscv-dynamic-tls-reloc-pie.patch
+        riscv-pr22263-1.patch)
 sha256sums=('SKIP'
-            '2d430b66f84a19c154725ec535280c493be1d34f1a90f95208dce02fecd0e4e4')
+            '2d430b66f84a19c154725ec535280c493be1d34f1a90f95208dce02fecd0e4e4'
+            '2cc818549e86f56046bf6ee8ab9d2d067b81232c14c1f43c65075a7566da23cb'
+            '36135dd42bfd1c13796611e755f2d4e643b8db8871e6e4d8ed4f65fd19b4f4aa')
 validpgpkeys=(3A24BC1E8FB409FA9F14371813FCEF89DD9E3C4F)
 
 prepare() {
@@ -41,6 +45,12 @@ prepare() {
   # unsupported targets.  This allows the binutils to be built with
   # BPF support enabled.
   patch -Np1 -i "${srcdir}"/gold-warn-unsupported.patch
+
+  # Avoid dynamic TLS relocs in RISC-V
+  patch -Np1 -i "${srcdir}"/riscv-dynamic-tls-reloc-pie.patch
+
+  # Avoid spurious R_RISCV_NONE for pr22263-1 test
+  patch -Np1 -i "${srcdir}"/riscv-pr22263-1.patch
 }
 
 build() {
