diff --git PKGBUILD PKGBUILD
index db8ef95..427c387 100644
--- PKGBUILD
+++ PKGBUILD
@@ -60,7 +60,7 @@
     --prefix=/usr \
     --sysconfdir="${pkgdir}"/etc \
     --with-lib-path=/usr/lib:/usr/local/lib \
-    --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/binutils/-/issues \
+    --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues \
     --enable-cet \
     --enable-colored-disassembly \
     --enable-default-execstack=no \
