--- PKGBUILD
+++ PKGBUILD
@@ -46,6 +46,9 @@ prepare() {
   # Turn off development mode (-Werror, gas run-time checks, date in sonames)
   sed -i '/^development=/s/true/false/' bfd/development.sh
 
+  # https://sourceware.org/bugzilla/show_bug.cgi?id=33029
+  git cherry-pick -n f601ffb52199a883f16df385b73a14e756b3e19a
+
   # Creds @Fedora
   # Change the gold configuration script to only warn about
   # unsupported targets.  This allows the binutils to be built with
@@ -60,7 +63,7 @@ build() {
     --prefix=/usr \
     --sysconfdir="${pkgdir}"/etc \
     --with-lib-path=/usr/lib:/usr/local/lib \
-    --with-bugurl=https://gitlab.archlinux.org/archlinux/packaging/packages/binutils/-/issues \
+    --with-bugurl=https://github.com/felixonmars/archriscv-packages/issues \
     --enable-cet \
     --enable-colored-disassembly \
     --enable-default-execstack=no \
