--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,9 @@ b2sums=('af82ef776ad153d76a919c6c28bd5c03ac89aa7d8b13d2215204b071dd734990bc87ee8
 
 prepare() {
   cd $pkgname-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
