diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilterCoefficients.cpp vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilterCoefficients.cpp
--- vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilterCoefficients.cpp	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilterCoefficients.cpp	2025-01-21 12:45:43.308594135 +0800
@@ -4,6 +4,9 @@
 #include "../Utils/VASTSynthfunctions.h"  //for pow
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilter.cpp vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilter.cpp
--- vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilter.cpp	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilter.cpp	2025-01-21 12:45:13.438593513 +0800
@@ -9,6 +9,9 @@
 #include "../VASTVcf.h"
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilter.h vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilter.h
--- vaporizer2/VASTvaporizer/Source/Engine/Filter/VASTQFilter.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Filter/VASTQFilter.h	2025-01-21 12:44:45.675259603 +0800
@@ -8,6 +8,9 @@
 #include "VASTVCFCombFilter.h"
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.cpp vaporizer2_/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.cpp
--- vaporizer2/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.cpp	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.cpp	2025-01-21 12:47:42.248596519 +0800
@@ -12,6 +12,9 @@
 #include <math.h>
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.h vaporizer2_/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.h
--- vaporizer2/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/FX/VASTWaveshaper.h	2025-01-21 12:46:12.888594721 +0800
@@ -10,6 +10,9 @@
 #include "VASTEffect.h"
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.cpp vaporizer2_/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.cpp
--- vaporizer2/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.cpp	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.cpp	2025-01-21 12:42:50.458590691 +0800
@@ -7,6 +7,9 @@
 
 #ifdef __aarch64__ //arm64
     #include "../../sse2neon.h"
+#elif defined(__riscv)
+    #define SIMDE_ENABLE_NATIVE_ALIASES
+    #include "simde/x86/sse2.h"
 #else
     #if defined _MACOSX || defined JUCE_LINUX
         #include <pmmintrin.h>
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.h vaporizer2_/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.h
--- vaporizer2/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Oversampler/VASTDecimator.h	2025-01-21 12:47:10.658595869 +0800
@@ -7,6 +7,9 @@
 #include "VASTRateConvertor.h"
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Utils/VASTSSEHelper.h vaporizer2_/VASTvaporizer/Source/Engine/Utils/VASTSSEHelper.h
--- vaporizer2/VASTvaporizer/Source/Engine/Utils/VASTSSEHelper.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Utils/VASTSSEHelper.h	2025-01-21 12:42:20.418590090 +0800
@@ -5,6 +5,9 @@
 #include "../VASTEngineHeader.h"
 #ifdef __aarch64__ //arm64
 	#include "../../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+	#include "simde/x86/sse3.h"
 #else
 	#include "immintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/Utils/VASTSynthfunctions.h vaporizer2_/VASTvaporizer/Source/Engine/Utils/VASTSynthfunctions.h
--- vaporizer2/VASTvaporizer/Source/Engine/Utils/VASTSynthfunctions.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/Utils/VASTSynthfunctions.h	2025-01-21 12:41:49.595256138 +0800
@@ -14,6 +14,9 @@
 #include <stdio.h>
 #ifdef __aarch64__ //arm64
     #include "../../sse2neon.h"
+#elif defined(__riscv)
+    #define SIMDE_ENABLE_NATIVE_ALIASES
+    #include "simde/x86/sse2.h"
 #else
     #include "immintrin.h"
 #endif
diff -Naur vaporizer2/VASTvaporizer/Source/Engine/VASTVcf.h vaporizer2_/VASTvaporizer/Source/Engine/VASTVcf.h
--- vaporizer2/VASTvaporizer/Source/Engine/VASTVcf.h	2025-01-20 22:39:15.000000000 +0800
+++ vaporizer2_/VASTvaporizer/Source/Engine/VASTVcf.h	2025-01-21 12:46:44.558595373 +0800
@@ -18,6 +18,9 @@
 /**************************/
 #ifdef __aarch64__ //arm64
 	#include "../sse2neon.h"
+#elif defined(__riscv)
+  #define SIMDE_ENABLE_NATIVE_ALIASES
+  #include "simde/x86/sse2.h"
 #else
 	#include "emmintrin.h"
 #endif
