--- PKGBUILD
+++ PKGBUILD
@@ -74,6 +74,9 @@ prepare() {
   git config submodule.clap-libs/clap-helpers.url "$srcdir/github.com-free-audio-clap-helpers"
   git -c protocol.file.allow=always submodule update --init
   popd
+
+  patch -p1 < $srcdir/add-riscv-simd-support.patch
+
 }
 
 build() {
@@ -150,3 +153,8 @@ package_vaporizer2-standalone() {
 
   install -vDm755 -t "$pkgdir/usr/bin" build/VASTvaporizer2_artefacts/Release/Standalone/VASTvaporizer2
 }
+
+makedepends+=('simde')
+source+=('add-riscv-simd-support.patch')
+sha512sums+=("c7bd9d886f94725a80faf914db0c632abb66aedf1f7ce0e9d70517ffc98bd88cae8cc29e8903ee1ddf948176d079f96b89082c7993f4e9fc4d30436dcfa22efb")
+b2sums+=("e002391122e3f62cbfea521d012102259db1f97278bdf45bf15fb83299f99941ad24b2fab2f72f8012b958efc49f0e3bdc89426222265863a39eebc28a979d7a")
