--- PKGBUILD
+++ PKGBUILD
@@ -74,6 +74,9 @@ prepare() {
   git config submodule.clap-libs/clap-helpers.url "$srcdir/github.com-free-audio-clap-helpers"
   git -c protocol.file.allow=always submodule update --init
   popd
+
+  patch -p1 < $srcdir/add-riscv-simd-support.patch
+
 }
 
 build() {
@@ -150,3 +153,8 @@ package_vaporizer2-standalone() {
 
   install -vDm755 -t "$pkgdir/usr/bin" build/VASTvaporizer2_artefacts/Release/Standalone/VASTvaporizer2
 }
+
+makedepends+=('simde')
+source+=('add-riscv-simd-support.patch::https://github.com/VASTDynamics/Vaporizer2/pull/94.patch')
+sha512sums+=("d9144d7950decd78b6d6dcd7cfb581788e9b55d012deaf7b4363191212094599c681a7df240f4f4671657ddb9453ed812d2de561d2cfb82b44c65fa68000ea50")
+b2sums+=("ed5185528d88920b2f5871e8ee2d65381318e7342fc9eb356515446cb9df7c1af9298a36b2c142c829552061d0bd6f47df6e95552b517f4f788313a665ddf34d")
