diff --git PKGBUILD PKGBUILD
index dd4af1716..45b68d52c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ arch=('any')
 depends=('python-click' 'python-yaml')
 makedepends=('python-setuptools' 'flake8')
 #checkdepends=('python-pytest-runner')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/$pkgver.tar.gz")
-sha512sums=('d8954d231bff6945ac0e8350be05775a8a0b852dd0284e3446b3f9c8bc5861870ed1bc058e132c144fe8d543aee87d7a7a54894abdb4006664e5994c8cfb507f')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/$pkgver.tar.gz"
+        "remove_unused_dep.patch"::"https://codeberg.org/hjacobs/python-clickclick/pulls/1.patch")
+sha512sums=('d8954d231bff6945ac0e8350be05775a8a0b852dd0284e3446b3f9c8bc5861870ed1bc058e132c144fe8d543aee87d7a7a54894abdb4006664e5994c8cfb507f'
+            '9d8d466b3bf3d08c7ba07aa0865be2318297fd7b6483d4cff2da53f711465da69332510bb77a990fc788878a910ebf6544177f95c4fd15a8f9f3990b57855905')
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i "../remove_unused_dep.patch"
+}
 
 build() {
   cd $pkgname
