diff --color -Naur unarj-2.65-orig/unarj.c unarj-2.65/unarj.c
--- unarj-2.65-orig/unarj.c	2002-06-06 00:28:06.000000000 +0800
+++ unarj-2.65/unarj.c	2021-10-08 19:23:08.525911557 +0800
@@ -692,7 +692,7 @@
     }
     if ((arj_flags & GARBLE_FLAG) != 0)
     {
-        printf(M_ENCRYPT);
+        printf("%s", M_ENCRYPT);
         printf(M_SKIPPED, filename);
         skip();
         return -1;
@@ -756,7 +756,7 @@
     }
     printf(M_EXTRACT, name);
     if (host_os != OS && file_type == BINARY_TYPE)
-        printf(M_DIFFHOST);
+        printf("%s", M_DIFFHOST);
     printf("  ");
 
     crc = CRC_MASK;
@@ -772,10 +772,10 @@
     set_ftime_mode(name, time_stamp, file_mode, (uint) host_os);
 
     if ((crc ^ CRC_MASK) == file_crc)
-        printf(M_CRCOK);
+        printf("%s", M_CRCOK);
     else
     {
-        printf(M_CRCERROR);
+        printf("%s", M_CRCERROR);
         error_count++;
     }
     return 1;
@@ -801,10 +801,10 @@
         decode_f();
 
     if ((crc ^ CRC_MASK) == file_crc)
-        printf(M_CRCOK);
+        printf("%s", M_CRCOK);
     else
     {
-        printf(M_CRCERROR);
+        printf("%s", M_CRCERROR);
         error_count++;
     }
     return 1;
@@ -951,7 +951,7 @@
     int i;
 
     for (i = 0; M_USAGE[i] != NULL; i++)
-        printf(M_USAGE[i]);
+        printf("%s", M_USAGE[i]);
 }
 
 int
@@ -966,7 +966,7 @@
     argc = ccommand(&argv);
 #endif
 
-    printf(M_VERSION);
+    printf("%s", M_VERSION);
 
     if (argc == 1)
     {
