--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,7 @@
         git+https://github.com/open-source-parsers/jsoncpp.git#commit=${_jsoncpp_commit}
         git+https://github.com/google/protobuf.git#commit=${_protobuf_commit}
         git+https://github.com/abseil/abseil-cpp.git#commit=${_abseil_cpp_commit}
+        "add_limits_header.patch"
 	)
 sha512sums=('SKIP'
             '606f45d48a9dad0e80a566cab0001910de3c6b2f634ec52c6ef6f44745b55ae8e181b3e3cdf90525a08be1f180eb35900672c90c6ab4f43679a178e863378bbc'
@@ -51,7 +52,8 @@
             'SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            '5b1d503865dbfdd0158d27877e5c899439933fe7576d72ae34e80e9c50495fab975c7d1b3f67491cd47b56e06d7f0b4521c85ca3048188159cc5ae3125e0dc65')
 validpgpkeys=('2CC8A0609AD2A479C65B6D5C8E8B898CBF2412F9')  # Weng Xuetian
 
 pkgver(){
@@ -88,6 +90,10 @@
 
   # adjust QT_BASE_PATH
   sed 's|path = QT_BASE_PATH|path = "/usr/include/qt"|' -i WORKSPACE.bazel
+  
+  cd third_party/abseil-cpp
+  patch -Np1 -i "$srcdir/add_limits_header.patch"
+  cd -
 }
 
 build() {
