--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,7 @@ options=('!lto')
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../test-timeout.patch
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
   mkdir completions/
 }
@@ -32,7 +33,7 @@ build() {
 
 check() {
   cd "$pkgname-$pkgver"
-  cargo test --frozen --all-features --workspace --lib
+  ATUIN_TEST_SQLITE_STORE_TIMEOUT=1.0 cargo test --frozen --all-features --workspace --lib
 }
 
 package() {
@@ -45,4 +46,7 @@ package() {
   install -Dm 644 "completions/_$pkgname" -t "${pkgdir}/usr/share/zsh/site-functions"
 }
 
+source+=("test-timeout.patch::https://github.com/atuinsh/atuin/pull/1703.diff")
+sha256sums+=('6eff082fdb4fbd09f342d6a98b8b72e8404899f458f7be7ebd531a6990e18b82')
+
 # vim: ts=2 sw=2 et:
