--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,9 @@ b2sums=('d7b3fd9c35c81be1bf971768758b3510747e7b4276714466700d832f1c7c426166ce605
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
