--- PKGBUILD
+++ PKGBUILD
@@ -23,6 +23,8 @@ prepare() {
   cd $pkgname-$pkgver
   patch -Np1 -i ../fix-tests.patch
   patch -Np1 -i ../nodlopen.patch
+  patch -Np1 -i ../fix-rvv-tests.patch
+  patch -Np1 -i ../kyber-rvv-fix.patch
 }
 
 build() {
@@ -45,3 +47,7 @@ package() {
   install -Dm644 -t "$pkgdir/usr/share/licenses/$pkgname/" $pkgname-$pkgver/LICENSE{,.bsd,.gplv2}
 }
 
+source+=(fix-rvv-tests.patch
+         kyber-rvv-fix.patch)
+sha256sums+=('1ef9ceac0b5bbb080f0545ab7822d9036b393a66eb94ea4ee06ab846f42b29e8'
+             '468d9ab4a34b99249b420f9a189d1f5b7686e5692c31745aa4625fc2a1bdffc4')
