diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index 8b7c17d..15ee1df 100644
--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,9 @@ validpgpkeys=(
 build() {
   cd $pkgname
   export CARGO_TARGET_DIR=../target
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
   cargo build --release --locked --features 'crypto-nettle compression-bzip2 autocrypt'
 }

