diff --color -Naur chmlib-0.40-orig/src/chm_lib.c chmlib-0.40/src/chm_lib.c
--- chmlib-0.40-orig/src/chm_lib.c	2009-05-23 22:43:31.000000000 +0800
+++ chmlib-0.40/src/chm_lib.c	2021-10-05 22:37:55.450952995 +0800
@@ -162,9 +162,10 @@
 typedef long long               Int64;
 typedef unsigned long long      UInt64;
 
-/* x86-64 */
+/* x86-64  */
+/* riscv64 */
 /* Note that these may be appropriate for other 64-bit machines. */
-#elif __x86_64__ || __ia64__
+#elif __x86_64__ || __ia64__ || (__riscv && (__riscv_xlen == 64))
 typedef unsigned char           UChar;
 typedef short                   Int16;
 typedef unsigned short          UInt16;
