diff --git PKGBUILD PKGBUILD
index 0b418e4..4cba9ef 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,16 @@ arch=('x86_64')
 url="http://www.jedrea.com/chmlib/"
 license=('LGPL')
 depends=('glibc')
-source=("http://www.jedrea.com/chmlib/chmlib-0.40.tar.bz2")
-md5sums=('7ea49ed8c335215c1edc6fae83e6b912')
+source=("http://www.jedrea.com/chmlib/chmlib-0.40.tar.bz2"
+        "define-riscv64-sized-types.patch")
+md5sums=('7ea49ed8c335215c1edc6fae83e6b912'
+         'd3ff2aa4ad18180b9955020b05396e6f')
 
 build() {
   cd "${srcdir}"/${pkgname}-${pkgver}
 
+  patch -Np1 -i ../define-riscv64-sized-types.patch
+
   ./configure --prefix=/usr \
               --enable-examples=yes
   make
