diff --git PKGBUILD PKGBUILD
index e53893aa7..c66140c65 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,11 @@ sha256sums=('4d17d4c93df2a95da3e3172c45b7a5bf317dd31dafd1c7a340169728c7089d1d'
             'SKIP')
 validpgpkeys=('6C44DB3E0BF3EAF5B433239A5FEE05E6A56E15A3') # Hubert Figuiere
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr # --disable-static
