diff -r -u ledger-3.2.1-orig/test/regress/BF3C1F82-2.test ledger-3.2.1/test/regress/BF3C1F82-2.test
--- ledger-3.2.1-orig/test/regress/BF3C1F82-2.test	2022-02-07 06:04:06.616079032 +0100
+++ ledger-3.2.1/test/regress/BF3C1F82-2.test	2022-02-07 06:05:34.372866932 +0100
@@ -4,11 +4,11 @@
 test -f - reg -> 1
 __ERROR__
 While parsing file "", line 2:
-Error: File to include was not found: "./non-existent-ledger-file-BF3C1F82"
+Error: File to include was not found: "non-existent-ledger-file-BF3C1F82"
 end test
 
 test -f /dev/stdin reg -> 1
 __ERROR__
 While parsing file "", line 2:
-Error: File to include was not found: "./non-existent-ledger-file-BF3C1F82"
+Error: File to include was not found: "non-existent-ledger-file-BF3C1F82"
 end test
