--- PKGBUILD
+++ PKGBUILD
@@ -19,8 +19,17 @@ optdepends=(
 	'emacs: emacs interface'
 	'python: python library'
 )
-source=("ledger-$pkgver.tar.gz::https://github.com/ledger/ledger/archive/v$pkgver.tar.gz")
-sha256sums=('92bf09bc385b171987f456fe3ee9fa998ed5e40b97b3acdd562b663aa364384a')
+source=("ledger-$pkgver.tar.gz::https://github.com/ledger/ledger/archive/v$pkgver.tar.gz"
+        "0000-fix-regress-test.patch")
+sha256sums=('92bf09bc385b171987f456fe3ee9fa998ed5e40b97b3acdd562b663aa364384a'
+            '17df250b2cb0ac5474cf2dd2cb663867dc63deda447381efa46f0886d7b2aa86')
+
+prepare() {
+	cd "ledger-$pkgver"
+
+	# https://github.com/ledger/ledger/issues/2075
+	patch -Np1 -i ../0000-fix-regress-test.patch
+}
 
 build() {
 	cd "ledger-$pkgver"
