diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1588aa7..d34b28f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,6 @@ endif()
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     string(TOLOWER ${CMAKE_SYSTEM_NAME} DIST_PLATFORM)
     add_compile_options(-std=c++11 -fopenmp )
-    add_compile_definitions(USE_SSE)
     set (PREPROCESSOR_COMMAND gcc -E -P -I${PROJECT_SOURCE_DIR}/src/main/shotcut/bigsh0t_transform_360/ - <)
 endif()
 
