--- PKGBUILD	(revision 1237843)
+++ PKGBUILD	(working copy)
@@ -9,11 +9,14 @@
 license=(GPL)
 depends=(gcc-libs)
 makedepends=(cmake python)
-source=($pkgname-$pkgver::https://bitbucket.org/leo_sutic/bigsh0t/get/$pkgver.tar.gz)
-sha256sums=('7808e1ebd6329f3f52f66c76f3db65571f2422247a1e654f1617efcad3aaa484')
+source=($pkgname-$pkgver::https://bitbucket.org/leo_sutic/bigsh0t/get/$pkgver.tar.gz
+        disable-sse.patch)
+sha256sums=('7808e1ebd6329f3f52f66c76f3db65571f2422247a1e654f1617efcad3aaa484'
+            '01a5a279263ba0572bf3b8db4b97f7087847eba24726693d7c99c078623d0b71')
 
 prepare() {
   sed -e 's|-march=native||' -i leo_sutic-$pkgname-*/CMakeLists.txt # Don't use -march=native
+  patch -Np1 leo_sutic-$pkgname-*/CMakeLists.txt 'disable-sse.patch'
 }
 
 build() {
