diff --git PKGBUILD PKGBUILD
index 36195e6..c1657bb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,10 @@ depends=(clutter-gtk gsound librsvg libgnome-games-support libgee)
 makedepends=(gobject-introspection yelp-tools appstream-glib vala git meson)
 groups=(gnome-extra)
 _commit=ca7d44e7a54a018b95d8eec843a701f1739be21f  # tags/3.38.2^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-nibbles.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-nibbles.git#commit=$_commit"
+        "fix_invalid_ref.patch")
+sha256sums=('SKIP'
+            '2fe351f8c26400aa81e9bd98504ef3b22c93307b9d99b82f1b17d727faa9b85d')
 
 pkgver() {
   cd $pkgname
@@ -22,6 +24,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "../fix_invalid_ref.patch"
 }
 
 build() {
