--- PKGBUILD
+++ PKGBUILD
@@ -24,8 +24,15 @@ optdepends=(
   'python: for atari-convert-dir'
   'sh: for atari-hd-image, zip2st, hatari-prg-args'
 )
-source=("https://download.tuxfamily.org/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.bz2")
-sha256sums=('2a5da1932804167141de4bee6c1c5d8d53030260fe7fe7e31e5e71a4c00e0547')
+source=("https://download.tuxfamily.org/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.bz2"
+        "fix-rega0-for-riscv.patch")
+sha256sums=('2a5da1932804167141de4bee6c1c5d8d53030260fe7fe7e31e5e71a4c00e0547'
+            '97aa3956079f3636865cb26b40e82a7c97fda92341adbef9d2e6878a5cde5ab4')
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  patch -Np1 -i ../fix-rega0-for-riscv.patch
+}
 
 build() {
   cd ${pkgname}-${pkgver}
