diff --git PKGBUILD PKGBUILD
index 10a5b24..88d373e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,14 @@ url='http://hatari.sourceforge.net/'
 license=('GPL')
 depends=('sdl2' 'libpng' 'portaudio' 'systemd-libs' 'hicolor-icon-theme')
 makedepends=('cmake' 'systemd')
-source=("https://download.tuxfamily.org/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.bz2")
-sha256sums=('44a2f62ca995e38d9e0874806956f0b9c3cc84ea89e0169a63849b63cd3b64bd')
+source=("https://download.tuxfamily.org/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.bz2"
+        "fix-name-conflicts.patch")
+sha256sums=('44a2f62ca995e38d9e0874806956f0b9c3cc84ea89e0169a63849b63cd3b64bd'
+            '7756052ef6254eb210e2b808c168b15980bc3675cb4aed50e6b5184f2f98efe6')
+
+prepare() {
+  patch -Np1 < fix-name-conflicts.patch -d ${pkgname}-${pkgver}
+}
 
 build() {
   cd ${pkgname}-${pkgver}
