--- a/python/private/pypi/pip_repository.bzl
+++ b/python/private/pypi/pip_repository.bzl
@@ -85,6 +85,7 @@
                 "linux_aarch64",
                 "linux_arm",
                 "linux_ppc",
+                "linux_riscv64",
                 "linux_s390x",
                 "linux_x86_64",
                 "osx_aarch64",

--- a/python/private/pypi/whl_installer/platform.py
+++ b/python/private/pypi/whl_installer/platform.py
@@ -45,6 +45,7 @@
     ppc64le = 5
     s390x = 6
     arm = 7
+    riscv64 = 8
     amd64 = x86_64
     arm64 = aarch64
     i386 = x86_32
@@ -269,6 +270,8 @@
             return "ppc"
         elif self.arch == Arch.ppc64le:
             return "ppc64le"
+        elif self.arch == Arch.riscv64:
+            return "riscv64"
         elif self.arch == Arch.s390x:
             return "s390x"
         else:

--- a/python/private/pypi/whl_target_platforms.bzl
+++ b/python/private/pypi/whl_target_platforms.bzl
@@ -30,6 +30,7 @@
     "ppc": "ppc",
     "ppc64": "ppc",
     "ppc64le": "ppc64le",
+    "riscv64": "riscv64",
     "s390x": "s390x",
     "arm": "arm",
     "armv6l": "arm",
