diff --git PKGBUILD PKGBUILD
index 4b7d8dc..95e7e1b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -28,6 +28,9 @@ sha256sums=('93421bd169d71ff5e7d2db95b62b030bfa205a12010b6468dcdef80337d6fbd8'
 prepare() {
   cd "$pkgname/1.0.0.rc16-3/$pkgname"
   patch -Np1 -i ${srcdir}/dmraid-format-security.patch
+
+  # get current config.guess and config.sub to recognize riscv64
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} autoconf/
 }
 
 build() {
