diff --git PKGBUILD PKGBUILD
index 06c0afe19..7ef785d26 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,17 +12,24 @@ arch=('x86_64')
 license=('GPL')
 source=(#https://sources.archlinux.org/other/dmraid/$pkgname-$pkgver.tar.bz2
         https://people.redhat.com/~heinzm/sw/dmraid/src/$pkgname-1.0.0.rc16-3.tar.bz2
+        dmarid-fix-format-not-a-string.patch
         dmraid_install
         dmraid_hook
         dmraid_tmpfiles
         dmraid.service)
 install=dmraid.install
 md5sums=('819338fcef98e8e25819f0516722beeb'
+         'e84fa11b93a8afe5424a94b8be5b61ae'
          '7a040ebcba305aba1e47dfe6ca8323b5'
          'faec669dc85f87187b45b5d3968efe2c'
          '56a8bb0ece8d206cd8efb504ee072ddd'
          'ea6d280fc6f63fb799abcd882bca53c8')
 
+prepare() {
+  cd "$pkgname"
+  patch -Np1 -i ../dmarid-fix-format-not-a-string.patch
+}
+
 build() {
   cd "$pkgname/1.0.0.rc16-3/$pkgname"
   ./configure --enable-led --enable-intel_led
