--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,7 @@ license=(GPL-3.0-only)
 depends=(vaultwarden)
 makedepends=(
   git
-  nodejs
+  nodejs-lts-iron
   npm
 )
 install=$pkgname.install
@@ -35,6 +35,8 @@ pkgver() {
 prepare() {
   cd bitwarden-clients
 
+  export ELECTRON_SKIP_BINARY_DOWNLOAD=1
+
   # Copy Vaultwarden images
   cp -vr ../bw_web_builds/resources/src/images/{logo-{dark,white}@2x,icon-white}.png \
     apps/web/src/images
