diff --git PKGBUILD PKGBUILD
index 36701de..72a8c14 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,9 @@ b2sums=('09494adf7a8ac8922124be1eea0f9dd1cbe9d629522d2e8f3446f7e7d73e239c900bea0
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
