diff --git PKGBUILD PKGBUILD
index d146c9d1..9f6c926d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,16 +10,20 @@ license=('GPL')
 url='https://v3.sk/~lkundrak/dev86/'
 depends=('glibc')
 source=("https://sources.archlinux.org/other/community/dev86/${pkgname}-${pkgver}.tar.gz"
-        'bin86-0.16.17-x86_64-1.patch')
+        'bin86-0.16.17-x86_64-1.patch'
+        'bin86-riscv64.patch')
 sha1sums=('35a1222350adce5b6d62dd56f3477d9416c59174'
-          '2c967c062fbee4e40ca1bf55feec2033e132e0a9')
+          '2c967c062fbee4e40ca1bf55feec2033e132e0a9'
+          'a25d5b2739db3a5489e6c463e87a5c965606472a')
 sha256sums=('021e37cde3a20632c4c9000993cb4aa9f58cb82b1d3c26b9aeb62d6566925738'
-            'f5174f7f703bb21486196c7d9ce027ba8d501c51f6099f02c6c13ee48fc861db')
+            'f5174f7f703bb21486196c7d9ce027ba8d501c51f6099f02c6c13ee48fc861db'
+            'ed783b983d688cffa3f1a9772b6916044e47083b3792e87ab0c4f7f294d74b51')
 
 prepare() {
   cd "${pkgname}-${pkgver}"
 
   patch -Np1 < ../bin86-0.16.17-x86_64-1.patch
+  patch -Np1 < ../bin86-riscv64.patch
 }
 
 build() {
