diff --git PKGBUILD PKGBUILD
index 8215dac..3c04ecd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -57,6 +57,9 @@ _commondeps=('java-runtime-common>=3' 'ca-certificates-utils' 'nss' 'libjpeg-tur
 build() {
   cd ${_jdkdir}
 
+  # Fix conflict with glibc 2.41+ uabs() declaration
+  sed -i '/^static inline unsigned int uabs(int n)/d' src/hotspot/share/utilities/globalDefinitions.hpp
+
   NUM_PROC_OPT=''
   MAKEFLAG_J=$(echo ${MAKEFLAGS} | sed -En 's/.*-j([0-9]+).*/\1/p')
   if [ -n "${MAKEFLAG_J}" ]; then
