Index: jdk18u-jdk-18.0.1-10/src/hotspot/os_cpu/linux_riscv/orderAccess_linux_riscv.hpp
===================================================================
--- jdk18u-jdk-18.0.1-10.orig/src/hotspot/os_cpu/linux_riscv/orderAccess_linux_riscv.hpp
+++ jdk18u-jdk-18.0.1-10/src/hotspot/os_cpu/linux_riscv/orderAccess_linux_riscv.hpp
@@ -54,6 +54,7 @@ inline void OrderAccess::fence() {
 }
 
 inline void OrderAccess::cross_modify_fence_impl() {
+  asm volatile (".option arch, +zifencei");
   asm volatile("fence.i" : : : "memory");
   if (UseConservativeFence) {
     asm volatile("fence ir, ir" : : : "memory");
