diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -17,15 +17,17 @@
 arch=('x86_64')
 url='https://openjdk.java.net/'
 license=('custom')
-makedepends=('java-environment>=11' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
-             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib' 'pandoc'
+makedepends=('java-environment>=15' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
+             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib'
              'graphviz' 'freetype2' 'libjpeg-turbo' 'giflib' 'libpng' 'lcms2'
              'libnet' 'bash' 'harfbuzz' 'gcc-libs' 'glibc')
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
+        java17-riscv64.patch
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
         freedesktop-jshell.desktop)
 sha256sums=('8c076203a6f85ab916b3e54de1992bcbcc5ffe580c52b1ac8d52ca7afb9f02d1'
+            'bbf96674558a801a7db691bff53bceb53f7ac39cd589c0c7f700f4522266436e'
             '85c4742764590783160af74587a47269e6797fbdf17ec485c7644bd239adf61d'
             'abac1ab09a33a3654378bc69be5a7cf075263ab3ae9efec1eb25cf388e711bb7'
             'e7cce0ecf868f656d8dc2eb25ab82ab665526a0a28aba20f02632dd29962dac3')
@@ -33,6 +35,7 @@
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac

 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -44,6 +47,13 @@
               lib/libjsound.{so,debuginfo}
               lib/libsplashscreen.{so,debuginfo})

+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i ../java17-riscv64.patch
+
+  sed -i '/zLargePages.inline.hpp/a #include "gc/z/zList.inline.hpp"' src/hotspot/share/gc/z/zPhysicalMemory.cpp
+}
+
 build() {
   cd ${_jdkdir}

@@ -96,6 +106,12 @@
     ${NUM_PROC_OPT}
     #--disable-javac-server

+    # build system does not recognize qemu-system-riscv64
+    # --build="riscv64-unknown-linux-gnu"
+
+  # Runtime.exec fails strangely if qemu binfmt does not have P flag
+  # export _JAVA_OPTIONS="-Djdk.lang.Process.launchMechanism=vfork"  # or fork
+
   make images legacy-jre-image docs

   # https://bugs.openjdk.java.net/browse/JDK-8173610
