diff --git PKGBUILD PKGBUILD
index 98a1992..a9e365b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,11 @@ depends=('glibc')
 makedepends=('patch')
 backup=(etc/foremost.conf)
 source=(http://foremost.sourceforge.net/pkg/$pkgname-$pkgver.tar.gz
-        $pkgname-makefile.patch)
+        $pkgname-makefile.patch
+        fix-multiple-definition.patch)
 sha512sums=('8827c29d52496783be26374f3943eb26a154d842f34e50fb489f87b3a5045bf85f1e44d7d8d8b12b2355ba3fe4b06a0db979cc22c0f431593c5976001eb931ab'
-            'eb3fc70590cafa3fb953946f6d80051ab3d976bcb3687cadb141a72dbd5bf317de68f39db7e28e27b49dd1629cab7e4c0ff5bc83c962c33026e9fe4d4ad2dc96')
+            'eb3fc70590cafa3fb953946f6d80051ab3d976bcb3687cadb141a72dbd5bf317de68f39db7e28e27b49dd1629cab7e4c0ff5bc83c962c33026e9fe4d4ad2dc96'
+            '8dcd5248e56dc33b0414b1595ad8856f08e51c72a23ba6d12f6ec503de53309ff0ed335fd6c0fc54f9688db5db360b6e8b8f506abfddf0815c3334f4035746c8')
 
 prepare() {
   cd "${srcdir}"/$pkgname-$pkgver
@@ -22,6 +24,9 @@ prepare() {
 
 build() {
   cd "${srcdir}"/$pkgname-$pkgver
+
+  patch -Np1 -i "${srcdir}"/fix-multiple-definition.patch
+
   make
 }
 
