diff --git PKGBUILD PKGBUILD
index bc73548b2..34bde106c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,17 @@ license=('GPL')
 depends=('gtk2')
 makedepends=('intltool')
 source=("https://download.savannah.gnu.org/releases/${pkgname}/${pkgname}-${pkgver}.tar.gz"
-        '0001-truncate.patch')
+        '0001-truncate.patch'
+        'https://raw.githubusercontent.com/Oleh-Kravchenko/portage/master/app-editors/leafpad/files/leafpad-0.8.18.1-format-security.patch')
 sha256sums=('959d22ae07f22803bc66ff40d373a854532a6e4732680bf8a96a3fbcb9f80a2c'
-            '9d9073ec4684e77cd1a6dd0cec4351750f9f61f2a6bab0e97f06ae6d00b7d2c4')
+            '9d9073ec4684e77cd1a6dd0cec4351750f9f61f2a6bab0e97f06ae6d00b7d2c4'
+            'd611eeb73aba8eaa76c4e31116a0697d75bcda9727f665abec866de66c102b5d')
 
 prepare() {
   cd "${srcdir}/${pkgname}-${pkgver}"
 
   patch -Np1 < "${srcdir}/"0001-truncate.patch
+  patch -Np1 < "${srcdir}/"leafpad-0.8.18.1-format-security.patch
 }
 
 build() {
