diff --git PKGBUILD PKGBUILD
index 79e7defc8..24f05eea0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,8 +8,15 @@ arch=('x86_64')
 pkgdesc="Recompression utilities for .zip .png .mng and .gz files using the 7-zip algorithm"
 url='https://www.advancemame.it/'
 license=('GPL')
-source=("https://github.com/amadvance/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz")
-sha256sums=('3ac0875e86a8517011976f04107186d5c60d434954078bc502ee731480933eb8')
+source=("https://github.com/amadvance/${pkgname}/releases/download/v${pkgver}/${pkgname}-${pkgver}.tar.gz"
+        "https://github.com/amadvance/advancecomp/commit/7b08f7a2af3f66ab95437e4490499cebb20e5e41.patch")
+sha256sums=('3ac0875e86a8517011976f04107186d5c60d434954078bc502ee731480933eb8'
+            '1638bf37068d4c37f2c03f1263538143eb5285546e6bace439e9ad4985eabee4')
+
+prepare(){
+    cd "${srcdir}"/${pkgname}-${pkgver}
+    patch -Np1 -i "${srcdir}/7b08f7a2af3f66ab95437e4490499cebb20e5e41.patch"
+}
 
 build() {
     cd "${srcdir}"/${pkgname}-${pkgver}
