--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,9 @@
 
   # -Werror, not even once
   sed -e 's/-Werror\b//g' -i Makefile Make.defaults
+  patch -Np1 -i ../shim-riscv64.patch
+  patch -Np1 -i ../shim-riscv64-check.patch
+  patch -Np1 -d gnu-efi -i ../../gnu-efi-riscv64.patch
 }
 
 build() {
@@ -53,3 +56,11 @@
   install -vDm 644 COPYRIGHT -t "$pkgdir/usr/share/licenses/$pkgname/"
   install -vDm 644 {BUILDING,README.{md,fallback,tpm},TODO} -t "$pkgdir/usr/share/doc/$pkgname/"
 }
+
+source+=(shim-riscv64.patch shim-riscv64-check.patch gnu-efi-riscv64.patch)
+sha512sums+=('6f74ef2a43e9faf1c7a034bf25085b56acbdbfdd1457362dadcfb35165560c01fe29afc8395e43d63c68f8d7d4ced7bbe5c558c5dbdb2849ce6b5c2434503bf9'
+             '526c9c907dcc9eefe024eaf27c0949018e74f494412632e1f9e8f0fa1cf385461ce7bcc5cfdd33c3c1e1610396a6c87f383aa945b22b30ad2040a7330e8cd5a8'
+             '303c67e487a083c442beaa5587ec8858e7bd1ecc471fe728e67d25eba9ce262ddf8bc13183089914b49c9709b534b6084409cd39dc05e944fa5a3ef09f6809bb')
+b2sums+=('ffd4036effc91b968e17be9d876e6b70fa1b9e0caed795c3da8bf6747ea73924c38b97b5ff74eddbaa71ec073a7612888215b86950f17218b64dea8f4e27d45e'
+         'f273ff8526324223e9a284df04fb9ef8325df00c41c178c04b7ab358b3fbdf6b56437d91f205168d0ea50d76a2d14f48fb58697b53f3443a05b2ccb7289ca78a'
+         '144f84b8eff9b43b080e357b06c62dadb2ee28434b958f9d484b9d49a9bd0a6e8f4a349fc7011c946afa1cb65ac895672e64b46cb7dedab7aabe09cb3ecca2a0')
