diff --git a/include/test.h b/include/test.h
index 5261dbd..f8e3271 100644
--- a/include/test.h
+++ b/include/test.h
@@ -25,6 +25,8 @@
 #include <ia32/efibind.h>
 #elif defined(__x86_64__)
 #include <x86_64/efibind.h>
+#elif defined(__riscv) && __riscv_xlen == 64
+#include <riscv64/efibind.h>
 #else
 #error what arch is this
 #endif
diff --git a/test-str.c b/test-str.c
index 9cb831d..dee9d10 100644
--- a/test-str.c
+++ b/test-str.c
@@ -719,7 +719,7 @@ test_strncmp(void)
 	 * This is still broken, and fails the test case as written on arm.
 	 * We no longer use this, so we do not strictly need to run it.
 	 */
-#if !defined(__arm__) && !defined(__aarch64__)
+#if !defined(__arm__) && !defined(__aarch64__) && !defined(__riscv)
 	/*
 	 * gnu-efi's broken strncmpa with the return type fixed
 	 */
