diff --git PKGBUILD PKGBUILD
index b0d1d8b92..7d286dfa3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@ pkgdesc="A D Compiler based on the LLVM Compiler Infrastructure including D runt
 arch=('x86_64')
 url="https://github.com/ldc-developers/ldc"
 license=('BSD')
-makedepends=('git' 'cmake' 'llvm' 'ldc' 'ninja')
+makedepends=('git' 'cmake' 'llvm' 'ninja')
 # Disable lto as linking the ldc2 binary fails
 options=(!lto)
 
@@ -23,22 +23,35 @@ source=(
     "ldc-druntime::git+https://github.com/ldc-developers/druntime.git"
     "ldc-phobos::git+https://github.com/ldc-developers/phobos.git"
     "ldc-testsuite::git+https://github.com/ldc-developers/dmd-testsuite.git"
+    "ldc.patch"::"https://github.com/ldc-developers/ldc/pull/4007.patch"
+    "rt.patch"::"https://github.com/ldc-developers/druntime/pull/204.patch"
+    "ph.patch"::"https://github.com/ldc-developers/phobos/pull/71.patch"
+    "disable-static-NaN-tests.patch"
 )
 
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            '95085d2f7a3b1cc8d97ce567a013967f88305ff21d0928c256ec5db19ec059af'
+            'd7105748df3f2888f79ead8740a19a01a416566d95e0ada5d61bcf0d36f6ad7b'
+            'fa5caceed1cb9530f8fe6a7312e1b226c48aa2086e93e403cccf1d31545eac58'
+            '22b9132b58dde320d6da3c22d2eeabbc0c4d6a079348e9e0fbe5172ef4b86aba')
 
 prepare() {
     cd "$srcdir/ldc"
 
+    patch -Np1 -i "$srcdir/ldc.patch"
+
     git submodule init
     git config submodule.druntime.url "$srcdir/ldc-druntime"
     git config submodule.phobos.url "$srcdir/ldc-phobos"
     git config submodule.tests/d2/dmd-testsuite.url "$srcdir/ldc-testsuite"
     git submodule update
 
+    patch -Np1 -d runtime/druntime -i "$srcdir/rt.patch"
+    patch -Np1 -d runtime/phobos -i "$srcdir/ph.patch"
+
     # Set version used for path construction in getFullClangCompilerRTLibPath()
     sed -i "s/ldc::llvm_version_base/\"$_clangversion\"/" driver/linker-gcc.cpp
 }
@@ -56,14 +69,13 @@ build() {
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=gold --flto=thin" \
-    -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"" \
     ..
     ninja
 }
 
 check() {
     cd "$srcdir/ldc/build"
+    patch -Np1 -d ../runtime/phobos -i "$srcdir/disable-static-NaN-tests.patch"
     ninja all-test-runners
 }
 
