--- PKGBUILD
+++ PKGBUILD
@@ -179,6 +179,10 @@ prepare() {
   git -c protocol.file.allow=always submodule update
   popd
 
+  pushd libs/JUCE
+  patch -Np1 -i "$srcdir/$pkgbase-fix-immintrin.h"
+  popd
+
   # first shot at improving reproducibility:
   # https://github.com/surge-synthesizer/surge/issues/2389
   patch -p1 -i "$srcdir/improve-reproducibility.patch"
@@ -201,7 +205,8 @@ build() {
     -D CMAKE_INSTALL_LIBDIR='/usr/lib' \
     -D CMAKE_BUILD_TYPE=Release \
     -W no-dev \
-    -D SURGE_BUILD_TESTRUNNER=OFF
+    -D SURGE_BUILD_TESTRUNNER=OFF \
+    -D SURGE_SKIP_LUA=ON
 
   cmake --build build
 }
@@ -284,3 +289,7 @@ package_surge-xt-clap(){
   install -vd "$pkgdir/usr/lib"
   mv usr-lib/clap "$pkgdir/usr/lib"
 }
+
+source+=("$pkgbase-fix-immintrin.h::https://github.com/surge-synthesizer/JUCE/pull/5.diff")
+sha512sums+=('e125b0310af3dce9a62b12a065391c0aa4d7b02fd3a64881c37d25d87ac5e386ae5880cfe2e4ff5232c403085510e3daec252acec953dfaf77a25527a525b4ad')
+b2sums+=('607aac6ab7afea065036546c42eb4a7448b68eb8310643574b67cbc0c4612bbe6fe643ed2dc564865b6ee7c5b741b798da6a4ee90d3a6b332f882be2be77c400')
