--- PKGBUILD
+++ PKGBUILD
@@ -105,6 +105,8 @@ pkgver() {
   git describe --tags | sed 's/^release_xt_//'
 }
 
+source+=("$pkgbase-fix-immintrin.h::https://github.com/surge-synthesizer/JUCE/pull/5.diff")
+b2sums+=('607aac6ab7afea065036546c42eb4a7448b68eb8310643574b67cbc0c4612bbe6fe643ed2dc564865b6ee7c5b741b798da6a4ee90d3a6b332f882be2be77c400')
 prepare() {
   cd "$pkgbase"
 
@@ -160,6 +162,10 @@ prepare() {
   git -c protocol.file.allow=always submodule update
   popd
 
+  pushd libs/JUCE
+  patch -Np1 -i "$srcdir/$pkgbase-fix-immintrin.h"
+  popd
+
   # first shot at improving reproducibility:
   # https://github.com/surge-synthesizer/surge/issues/2389
   patch -p1 -i "$srcdir/improve-reproducibility.patch"
@@ -186,7 +192,8 @@ build() {
     -D CMAKE_INSTALL_LIBDIR='/usr/lib' \
     -D CMAKE_BUILD_TYPE=Release \
     -W no-dev \
-    -D SURGE_BUILD_TESTRUNNER=OFF
+    -D SURGE_BUILD_TESTRUNNER=OFF \
+    -D SURGE_SKIP_LUA=ON
 
   cmake --build build
 }
