Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 437194)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 url="https://www.claws-mail.org"
 depends=('gtk3' 'gnutls' 'startup-notification' 'enchant' 'gpgme'
          'libetpan' 'compface' 'libsm' 'dbus-glib')
-makedepends=('spamassassin' 'bogofilter' 'networkmanager' 'valgrind'
+makedepends=('spamassassin' 'bogofilter' 'networkmanager'
              # dependencies for plugins
              'libgdata' 'libnotify' 'libcanberra' 'poppler-glib' 'libytnef' 'libical'
              'dillo' 'python' 'gumbo-parser'
