--- PKGBUILD
+++ PKGBUILD
@@ -28,12 +28,16 @@ prepare() {
 
   cd boinc
   ./_autosetup
+  # no -mavx for CC
+  sed -i 's/ -mavx//' configure
 
   # Prepare boinc-nox
   cd "$srcdir"/boinc-nox
   # Don't force xss
   sed -i 's/^  enable_xss="yes"$/  enable_xss="no"/' configure.ac
   ./_autosetup
+  # no -mavx for CC
+  sed -i 's/ -mavx//' configure
 }
 
 build() {
@@ -53,7 +57,8 @@ build() {
     --with-ssl \
     --with-x \
     --with-wxdir=/usr/lib \
-    --with-wx-config=$(which wx-config)
+    --with-wx-config=$(which wx-config) \
+    --with-boinc-alt-platform=riscv64-unknown-linux-gnu
   make
 
   # Build boinc-nox
@@ -71,7 +76,8 @@ build() {
     --disable-manager \
     --with-ssl \
     --without-wxdir \
-    --without-x
+    --without-x \
+    --with-boinc-alt-platform=riscv64-unknown-linux-gnu
   make
 }
 
