diff --git PKGBUILD PKGBUILD
index 1d6f61995..5c69844c2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,14 @@ arch=('x86_64')
 url="https://www.coker.com.au/bonnie++/"
 license=('GPL')
 depends=('perl')
-source=(https://www.coker.com.au/$pkgname/$pkgname-$pkgver.tgz)
-md5sums=('5bb79f393b4172a0d7014ce4125454cd')
+source=(https://www.coker.com.au/$pkgname/$pkgname-$pkgver.tgz
+        https://gitweb.gentoo.org/repo/gentoo.git/plain/app-benchmarks/bonnie++/files/bonnie++-2.00a-gcc11.patch)
+md5sums=('5bb79f393b4172a0d7014ce4125454cd'
+         '3f663deaba1d67fb55661554aa5d0d52')
+prepare(){
+  cd "$srcdir"/${pkgname}-${pkgver}
+  patch -Np1 -i ../bonnie++-2.00a-gcc11.patch
+}
 build() {
   cd "$srcdir"/${pkgname}-${pkgver}
 
