diff --git PKGBUILD PKGBUILD
index 1d6f61995..5c69844c2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,14 @@ arch=('x86_64')
 url="https://www.coker.com.au/bonnie++/"
 license=('GPL')
 depends=('perl')
-source=(https://www.coker.com.au/$pkgname/$pkgname-$pkgver.tgz)
-sha256sums=('a8d33bbd81bc7eb559ce5bf6e584b9b53faea39ccfb4ae92e58f27257e468f0e')
+source=(https://www.coker.com.au/$pkgname/$pkgname-$pkgver.tgz
+        https://gitweb.gentoo.org/repo/gentoo.git/plain/app-benchmarks/bonnie++/files/bonnie++-2.00a-gcc11.patch)
+sha256sums=('a8d33bbd81bc7eb559ce5bf6e584b9b53faea39ccfb4ae92e58f27257e468f0e'
+            'fdc202e07610728e4dbf55284c42e5d8e183e6175685bcf72f4f868cdfef71d1')
+prepare(){
+  cd "$srcdir"/${pkgname}-${pkgver}
+  patch -Np1 -i ../bonnie++-2.00a-gcc11.patch
+}
 build() {
   cd "$srcdir"/${pkgname}-${pkgver}
 
