diff --git a/src/platform.h b/src/platform.h
index cc32dae..98eedd1 100644
--- a/src/platform.h
+++ b/src/platform.h
@@ -133,6 +133,15 @@
 	#define BAM_ARCH_STRING "armv7l"
 #endif
 
+#if defined(__riscv) && __riscv_xlen == 32
+	#define BAM_ARCH_RISCV32
+	#define BAM_ARCH_STRING "riscv32"
+#endif
+
+#if defined(__riscv) && __riscv_xlen == 64
+	#define BAM_ARCH_RISCV64
+	#define BAM_ARCH_STRING "riscv64"
+#endif
 
 #ifndef BAM_FAMILY_STRING
 #define BAM_FAMILY_STRING "unknown"
