diff --git a/bam.lua b/bam.lua
index d879883ca..d3a48ff71 100644
--- a/bam.lua
+++ b/bam.lua
@@ -185,6 +185,8 @@ function GenerateLinuxSettings(settings, conf, arch, compiler)
 		settings.link.flags:Add("-m64")
 	elseif arch == "armv7l" then
 		-- arm 32 bit
+	elseif arch == "riscv64" then
+		-- RISC-V 64 bit
 	else
 		print("Unknown Architecture '" .. arch .. "'. Supported: x86, x86_64")
 		os.exit(1)
