--- PKGBUILD
+++ PKGBUILD
@@ -24,11 +24,16 @@
 prepare() {
   convert +set date:create +set date:modify "${pkgname}/other/icons/teeworlds.ico" "${srcdir}/${pkgname}.png"
 
+  cd bam-0.5.1
+  patch -Np1 -i ../add-riscv-defines.patch
+  cd ..
+
   cd teeworlds
   git submodule init
   git config submodule."datasrc/languages".url "${srcdir}"/teeworlds-translation
   git config submodule."datasrc/maps".url "${srcdir}"/teeworlds-maps
   git -c protocol.file.allow=always submodule update
+  patch -Np1 -i ../allow-riscv64-build.patch
 }
 
 build() {
@@ -45,10 +50,10 @@
 
   # Install data files
   mkdir -p "${pkgdir}"/usr/share/${pkgname}/data
-  cp -r build/x86_64/release/data/* "${pkgdir}"/usr/share/${pkgname}/data
+  cp -r build/$CARCH/release/data/* "${pkgdir}"/usr/share/${pkgname}/data
 
-  install -Dm755 build/x86_64/release/${pkgname} "${pkgdir}"/usr/bin/${pkgname}
-  install -Dm755 build/x86_64/release/${pkgname}_srv "${pkgdir}"/usr/bin/${pkgname}_srv
+  install -Dm755 build/$CARCH/release/${pkgname} "${pkgdir}"/usr/bin/${pkgname}
+  install -Dm755 build/$CARCH/release/${pkgname}_srv "${pkgdir}"/usr/bin/${pkgname}_srv
 
   install -Dm644 other/${pkgname}.desktop "${pkgdir}"/usr/share/applications/${pkgname}.desktop
   install -Dm644 other/${pkgname}.appdata.xml "${pkgdir}"/usr/share/metainfo/${pkgname}.appdata.xml
@@ -63,3 +68,8 @@
 }
 
 # vim:set ts=2 sw=2 et:
+
+source+=('add-riscv-defines.patch'
+         'allow-riscv64-build.patch')
+sha512sums+=('62d4e6ec2addeb350f9772029881161e4bcbecca7c445c0ad1d983a8a97d8c3025658da9d5d6dd63fa0604bf448522574c06999495551eb0add5a43d86781e6d'
+             'dd70871eefda84250d8ae8a6800447a9c887a45e28e8593f759eabcc9a9b4a935cb1bf74310c0e6922d4eb7136c295621d2cd116d6420dfff50d74348c34bb27')
