diff --git a/age/Cargo.toml b/age/Cargo.toml
--- a/age/Cargo.toml
+++ b/age/Cargo.toml
@@ -101,7 +101,7 @@ test-case = "2"
 tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
 
 [target.'cfg(unix)'.dev-dependencies]
-pprof = { version = "0.10", features = ["criterion", "flamegraph"] }
+pprof = { git = "https://github.com/tikv/pprof-rs.git", rev = "0760534", features = ["criterion", "flamegraph"] }
 
 [target.'cfg(any(target_arch = "x86", target_arch = "x86_64"))'.dev-dependencies]
 criterion-cycles-per-byte = "0.1"
