diff --git PKGBUILD PKGBUILD
index 7be6aa1..8691cd5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,16 @@ depends=(bzip2
          fuse3)
 makedepends=(cargo)
 _archive="$_pkgname-$pkgver"
-source=("$url/archive/v$pkgver/$_archive.tar.gz")
-sha256sums=('dca3b57cf32cef70d6085764cd72de379536ac63cec02593bf50e42fc9bc287c')
+source=("$url/archive/v$pkgver/$_archive.tar.gz"
+        "fix_rage_deps.patch")
+sha256sums=('3bd287372eb6226b246459c1b5c39ecdb36b3495d7af4d2bee93bb3aad9ccf65'
+            '3c38c189646bdb712527a3ae4118dc72c5746576b1e90510e914342a0225de03')
 
 prepare() {
 	cd "$_archive"
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+	patch -Np1 -i ../fix_rage_deps.patch # pprof-rs add riscv64 support, but not published to crates.io
+	cargo update -p pprof
+	cargo fetch --locked
 }
 
 build() {
