--- PKGBUILD
+++ PKGBUILD
@@ -70,6 +70,7 @@
     -vv
     --deselect tests/test_docs.py  # we are not interested in code formatting
     --deselect tests/pydantic_core/test_docstrings.py::test_readme  # we are not interested in code formatting
+    --deselect tests/pydantic_core/test_docstrings.py::test_docstrings  # we are not interested in code formatting
     --deselect tests/test_types.py::test_string_fails
     --deselect 'tests/test_networks.py::test_address_invalid[foobar-An email address must have an @-sign.]'
   )
