diff --git PKGBUILD PKGBUILD
index f53c458..9dab2bc 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,7 @@ prepare() {
     # https://gitlab.archlinux.org/archlinux/packaging/packages/hsa-rocr/-/issues/5
     cd "rocm-systems-rocm-$pkgver"
     patch -Np1 -i ../hsa-rocr-7.1.1-fix-vgpr-count.patch
+    patch -Np1 -i ../rocr-unsupport.patch
 }
 
 build() {
@@ -52,3 +53,6 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 "$_dirname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=(rocr-unsupport.patch)
+sha256sums+=('e3a4655a2cc01bdb6fb82ee26c6ede4dab31e20281fe3ab7436768859b35d4a1')
