--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,11 @@
 _dirname="$(basename "$url")-rocm-$pkgver"/projects/rocr-runtime/
 options=(!lto)
 
+prepare() {
+  cd "rocm-systems-rocm-$pkgver"
+  patch -Np1 -i ../rocr-unsupport.patch
+}
+
 build() {
   # Silence warnings on optional libraries with -DNDEBUG,
   # https://github.com/RadeonOpenCompute/ROCR-Runtime/issues/89#issuecomment-613788944
@@ -44,3 +49,6 @@
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 "$_dirname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=(rocr-unsupport.patch)
+sha256sums+=('e3a4655a2cc01bdb6fb82ee26c6ede4dab31e20281fe3ab7436768859b35d4a1')
