diff --git PKGBUILD PKGBUILD
index f3d711f..74375dd 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,18 @@ license=('NCSA')
 depends=('rocm-core' 'glibc' 'gcc-libs' 'libelf' 'libdrm'
          'hsakmt-roct' 'rocm-device-libs')
 makedepends=('cmake' 'rocm-llvm' 'xxd')
-source=("${pkgname}-${pkgver}.tar.gz::$url/archive/rocm-$pkgver.tar.gz")
-sha256sums=('dbe477b323df636f5e3221471780da156c938ec00dda4b50639aa8d7fb9248f4')
+source=("${pkgname}-${pkgver}.tar.gz::$url/archive/rocm-$pkgver.tar.gz"
+        https://raw.githubusercontent.com/loongarch-moe/rocm-loongarch/refs/heads/rocm-6.2.x/stage2/3.rocm-rocr-runtime/rocr-unsupport.patch)
+sha256sums=('dbe477b323df636f5e3221471780da156c938ec00dda4b50639aa8d7fb9248f4'
+            '523d7f93daca6e7272578ff0fced94a79421fef0ccd7f202425881543d4bcb56')
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 options=(!lto)
 
+prepare() {
+  cd "$_dirname"
+  patch -Np1 -i ../rocr-unsupport.patch
+}
+
 build() {
   # Silence warnings on optional libraries with -DNDEBUG,
   # https://github.com/RadeonOpenCompute/ROCR-Runtime/issues/89#issuecomment-613788944
