diff --git PKGBUILD PKGBUILD
index 9f7989e..a685789 100644
--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,7 @@ options=(!lto)
 
 prepare() {
     patch -d "${_dirname}" -Np1 < "${srcdir}/hsa-rocr-6.4-fix-missing-include.patch"
+    patch -d "${_dirname}" -Np1 < "${srcdir}/rocr-unsupport.patch"
 }
 
 build() {
@@ -49,3 +50,6 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 "$_dirname/LICENSE.txt" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=(rocr-unsupport.patch)
+sha256sums+=('1abb08ae47ee0a6a18cecee962624f4e0c2d9e26701385d23f5e413b4daf4390')
