diff --git PKGBUILD PKGBUILD
index ce5d4ca..3dec932 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@
 license=('custom')
 depends=('zlib')
 provides=('common-lisp' 'cl-asdf')
-makedepends=('sbcl')
+makedepends=('clisp')
 source=("https://downloads.sourceforge.net/project/sbcl/sbcl/$pkgver/$pkgname-$pkgver-source.tar.bz2"
   "arch-fixes.lisp")
 sha256sums=('698144fb805a31919a610de999ab98391d52eb698adaa8ca19b28fc6ed0d99f7'
@@ -28,7 +28,7 @@
   export LINKFLAGS="$LDFLAGS"
   unset LDFLAGS
   unset MAKEFLAGS
-  sh make.sh sbcl --prefix=/usr --fancy
+  sh make.sh sbcl --prefix=/usr --fancy --xc-host='env LC_ALL=C.utf8 clisp -q -norc'
   make -C doc/manual info
 }
 
