Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010223)
+++ PKGBUILD	(working copy)
@@ -13,7 +13,7 @@
 url='https://launchpad.net/libdbusmenu'
 arch=('x86_64')
 license=('GPL3' 'LGPL2.1' 'LGPL3')
-makedepends=('gnome-common' 'gobject-introspection' 'gtk2' 'gtk3' 'intltool' 'vala' 'valgrind')
+makedepends=('gnome-common' 'gobject-introspection' 'gtk2' 'gtk3' 'intltool' 'vala')
 options=('!emptydirs')
 source=(https://launchpad.net/${pkgbase}/${pkgver%.?}/${pkgver}/+download/${pkgbase}-${pkgver}.tar.gz{,.asc})
 sha512sums=('ee9654ac4ed94bdebc94a6db83b126784273a417a645b2881b2ba676a5f67d7fc95dd2bb37bfb0890aa47299ed73cb21ed7de8b75f3fed6b69bfd39065062241'
