diff --git PKGBUILD PKGBUILD
index 4fa9a4ec2..f8aa3f8d6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,11 @@ depends=('util-linux')
 source=(http://nilfs.sourceforge.net/download/$pkgname-$pkgver.tar.bz2)
 md5sums=('914cd9b876b090377504e1184c7b8ca0')
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd $pkgname-$pkgver
   sed -i -e 's#root_sbindir=/sbin#root_sbindir=/usr/bin#' configure
