Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 988897)
+++ PKGBUILD	(working copy)
@@ -16,9 +16,8 @@
 url='https://asciidoc.org/'
 license=('GPL')
 depends=('python' 'libxslt' 'docbook-xsl')
-checkdepends=('dblatex' 'graphviz' 'lilypond' 'source-highlight')
+checkdepends=('dblatex' 'graphviz' 'source-highlight')
 optdepends=('graphviz: graphviz-filter'
-            'lilypond: music-filter'
             'imagemagick: music-filter'
             'source-highlight: source-highlight-filter'
             'dblatex: pdf generation'
@@ -26,17 +25,17 @@
             'w3m: text generation'
             'lynx: alternative text generation')
 source=("https://github.com/asciidoc/asciidoc-py3/archive/${pkgver}/${pkgname}-${pkgver}.tar.gz")
-sha256sums=('8a6e3ae99785d9325fba0856e04dbe532492af3cb20d49831bfd757166d46c6b')
-b2sums=('dd2376a3b9ffa1d7a8016c9eb1030a9d0d3a01cde587d3179cb425a5be5ae120ea7fc1fa971cb1bb53faa2a7e93b4a1c2e8cf854e7e960c1299398a02b8aa42e')
+sha256sums=('5056c20157349f8dc74f005b6e88ccbf1078c4e26068876f13ca3d1d7d045fe7')
+b2sums=('379c4d4ee1b10e09d39192ab570be10f8b51f31c355866c494c3ab3d79961b81d0fb1c5c0a63e89f671644a55ff25627250597b0a05dfe78e55760480fe60765')
 
 prepare() {
-  cd ${pkgname}-py3-${pkgver}
+  cd ${pkgname}-py-${pkgver}
 
   autoconf
 }
 
 build() {
-  cd ${pkgname}-py3-${pkgver}
+  cd ${pkgname}-py-${pkgver}
 
   ./configure \
     --prefix=/usr \
@@ -45,13 +44,13 @@
 }
 
 check() {
-  cd ${pkgname}-py3-${pkgver}
+  cd ${pkgname}-py-${pkgver}
 
   make test
 }
 
 package() {
-  cd ${pkgname}-py3-${pkgver}
+  cd ${pkgname}-py-${pkgver}
 
   make install DESTDIR="${pkgdir}"
   make docs DESTDIR="${pkgdir}"
