diff --git PKGBUILD PKGBUILD
index fd6f004..968dbd6 100755
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,9 @@ arch=('x86_64')
 url="https://yabause.org/"
 license=('GPL')
 depends=('freeglut' 'glew' 'openal' 'qt5-base' 'qt5-multimedia' 'sdl2')
-makedepends=('cmake' 'glu' 'libxmu')
+makedepends=('cmake' 'glu' 'libxmu' 'clang')
 conflicts=('yabause-gtk')
+options=('!lto')
 source=("https://download.tuxfamily.org/yabause/releases/${pkgver}/yabause-${pkgver}.tar.gz"
         'qt-5.11.patch'
         'rwx.patch')
@@ -38,6 +39,7 @@ build() {
   mkdir build && cd build

   cmake .. \
+    -DCMAKE_C_COMPILER=clang \
     -DCMAKE_BUILD_TYPE='Release' \
     -DCMAKE_INSTALL_PREFIX='/usr' \
     -DYAB_PORTS='qt' \
