--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,19 @@ options=('!lto')
 
 prepare() {
   cd "$pkgname-$pkgver"
+
+  cat << EOF >> Cargo.toml
+[patch.'https://github.com/fishfolk/bones']
+bones_framework     = { git = "https://github.com/hack3ric/bones", branch = "archrv-0.3" }
+bones_bevy_renderer = { git = "https://github.com/hack3ric/bones", branch = "archrv-0.3" }
+
+[patch.crates-io]
+quinn = { git = "https://github.com/hack3ric/quinn", branch = "archrv-0.10" }
+quinn-udp = { git = "https://github.com/hack3ric/quinn", branch = "archrv-0.10" }
+EOF
+
+  cargo update -p bones_framework -p bones_bevy_renderer -p quinn -p quinn-udp
+
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
