diff --git PKGBUILD.orig PKGBUILD
--- PKGBUILD.orig
+++ PKGBUILD
@@ -11,14 +11,20 @@ depends=('gcc-libs' 'hicolor-icon-theme')
 optdepends=('devil: for ImageSeq plugin')
 makedepends=('cmake' 'devil')
 source=("https://github.com/AviSynth/AviSynthPlus/archive/v${pkgver}/${pkgname}-${pkgver}.tar.gz"
+         avisynthplus-riscv-1.patch::https://github.com/AviSynth/AviSynthPlus/commit/3ec0d81e7b6d8750f233122046c2692f20dd0643.patch
+         avisynthplus-riscv-2.patch::https://github.com/AviSynth/AviSynthPlus/commit/7596b6232626922d1b89ef5fc853cc8d1556a7b3.patch
         '010-avisynthplus-gcc11-fix.patch'::'https://github.com/AviSynth/AviSynthPlus/commit/a1e33be91dfa48b8db5ee358ea4fcea952e80e4f.patch'
         'avisynthplus.xml')
 sha256sums=('8906d9e46dc90a194413d69b710e3106c0aafddc0c5c62004885d0c3beb79862'
+            '41d542525b25bec37c24589e5a412c187e914ad34e0e3dd5243459e13a1f505a'
+            '1a267b76f797306380795ac0f8bbd752cdfd022ca5a449c6f4948bac576623dc'
             '6eb2b108c4689a5ca71e6c3da5af62fd2fb717bdcf4835a3a38eb67159d80072'
             'c4b270a3df7fbe1c153400215169c4ae4cae3b7a8710c843393e3a6ed0fd8a3e')
 
 prepare() {
     patch -d "AviSynthPlus-${pkgver}" -Np1 -i "${srcdir}/010-avisynthplus-gcc11-fix.patch"
+    patch -d "AviSynthPlus-${pkgver}" -Np1 -i "${srcdir}/avisynthplus-riscv-1.patch"
+    patch -d "AviSynthPlus-${pkgver}" -Np1 -i "${srcdir}/avisynthplus-riscv-2.patch"
 }
 
 build() {
