diff --git PKGBUILD.orig PKGBUILD
index df454b4..29ff975 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -11,9 +11,18 @@
 license=('GPL')
 depends=('python' 'bluez-libs')
 makedepends=('python-setuptools')
-source=("https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/${_pkgname}/${_pkgname}-${pkgver}.tar.gz")
-sha256sums=('c8f04d2e78951eaa9de486b4d49381704e8943d0a6e6e58f55fcd7b8582e90de')
-b2sums=('6a234d5c41dc0bb9f08559431eafcf1d1fb1c881d2cb912cc048e2717bfe8a06f09aae045d2961c9ba5e087422cfc8249a1907368eb6dc45942175bcf97b1841')
+source=("https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/${_pkgname}/${_pkgname}-${pkgver}.tar.gz"
+        "remove-2to3-command.patch::https://github.com/pybluez/pybluez/pull/414.patch")
+sha256sums=('c8f04d2e78951eaa9de486b4d49381704e8943d0a6e6e58f55fcd7b8582e90de'
+            '03d7f0ae3480a9e4d3b19322e046439c2a152b4840b49398d3fb9adab1e8bbee')
+b2sums=('6a234d5c41dc0bb9f08559431eafcf1d1fb1c881d2cb912cc048e2717bfe8a06f09aae045d2961c9ba5e087422cfc8249a1907368eb6dc45942175bcf97b1841'
+        '31a039653211d54143dcb11fa29919296c250a26d71599bab9de3a9a9531daff5cb0845e31dcb09f62ec888c504b19a2356ff2eb06d5464c581f68851c5dcc86')
+
+prepare() {
+  cd "$srcdir"/$_pkgname-$pkgver
+  patch -Np1 -i ../remove-2to3-command.patch
+}
+
 
 build() {
   cd "$srcdir"/$_pkgname-$pkgver

