--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,8 @@
 
 prepare() {
   cd "$pkgbase"
+
+  patch -Np1 -i "$srcdir/firecracker-riscv64.patch"
 
   # download dependencies
   cargo fetch --locked --target host-tuple
@@ -65,3 +67,7 @@
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" \
     LICENSE NOTICE THIRD-PARTY
 }
+
+source+=(firecracker-riscv64.patch)
+sha512sums+=('e8e91ed477ab2927a5b202470dafadadd45c6ff2a99200a76d548484c6c4192ecb64f5ea5cd90aa4a5ef2df9681594a49eaea87f6313dab3d93c0393594b4964')
+b2sums+=('77e6d9d0265e3ddee1bc24fabe804aae4480a16806c3ce0c57d9843260cc2c869603df8870b61d9636263ff360586e0bf1981cd10bed552acbe19ee6be55710f')
