diff --git PKGBUILD PKGBUILD
index c010411..800c6f6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,15 +40,18 @@
   "https://gnupg.org/ftp/gcrypt/${pkgname}/${pkgname}-${pkgver}.tar.bz2"{,.sig}
   'drop-import-clean.patch'
   'avoid-beta-warning.patch'
+  'update-config.guess.patch::https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=patch;h=7e44f883664e23503b5507852a342939c9aced8a'
 )
 sha256sums=('13f3291007a5e8546fcb7bc0c6610ce44aaa9b3995059d4f8145ba09fd5be3e1'
             'SKIP'
             '02d375f0045f56f7dd82bacdb5ce559afd52ded8b75f6b2673c39ec666e81abc'
-            '22fdf9490fad477f225e731c417867d9e7571ac654944e8be63a1fbaccd5c62d')
+            '22fdf9490fad477f225e731c417867d9e7571ac654944e8be63a1fbaccd5c62d'
+            'c2464ba8177bcd34c15d65ae44adcd6e8156d76d8b443a86474e3740d68bf322')
 b2sums=('0be2965a646a8636a127f89329030860908b0bbc447381782527459aed85f5276c29e7a2c89f87cb715407d9f1aabbf3ae1765073764d05e422035e8d5962569'
         'SKIP'
         'd598015e7f27b27840667d1656c083b4ad85d6acdd312e9929854067313a5f28415ee7eecf4d84a4b8da0385b667aaa01a9743461f5c785402a56c238274e376'
-        '7ea069e81e2d91a3154bcb62516b4b599f34596de277f95ad1ccaba73869c4f84f94f063b65026ba0bc8a72c0fd8e8e182b82346964f67ea78166b6399c923c5')
+        '7ea069e81e2d91a3154bcb62516b4b599f34596de277f95ad1ccaba73869c4f84f94f063b65026ba0bc8a72c0fd8e8e182b82346964f67ea78166b6399c923c5'
+        'a633489116c4e0f752c6091f0da6d815a8efce176c42bda62bceb524627fc3c2e5cd39531bef79a2cbcd6d8a0f04a0ff90b68df049a18a56d6c1a33b22df3be5')
 validpgpkeys=(
   '5B80C5754298F0CB55D8ED6ABCEF7E294B092E28' # Andre Heinecke (Release Signing Key)
   '6DAA6E64A76D2840571B4902528897B826403ADA' # Werner Koch (dist signing 2020)
