--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@
 url="https://opensearch.org/docs/latest/security-plugin"
 license=('Apache-2.0')
 depends=("opensearch=${_opensearchver}")
-makedepends=("java-environment=${_jdkver}" 'unzip' 'git')
+makedepends=("java-environment=${_jdkver}" 'unzip' 'git' 'gradle')
 source=("git+https://github.com/opensearch-project/security.git#tag=${pkgver}") # Build requires git
 sha256sums=('37124563bde238a393b9bb49c8e72b4d4d0e5e640b14088dd6d3942951bd0f6f')
 
@@ -22,7 +22,7 @@
 
   # Currently the tests don't provide a way to skip the integration tests
   # … so we skip them entirely for now
-  ./gradlew assemble -Pcrypto.standard=default
+  gradle assemble -Pcrypto.standard=default
 }
 
 package() {
