--- depot_tools/gclient.py.orig	2023-06-26 16:33:06.394443514 +0200
+++ depot_tools/gclient.py	2023-06-26 16:33:06.394443514 +0200
@@ -916,6 +916,9 @@
       hooks_cwd = self.root.root_dir
 
     for dep in deps_to_add:
+      if '${arch}' in dep.name or '${platform}' in dep.name:
+        print("WARN: ignoring platform-specific dep:", dep.name)
+        continue
       if dep.verify_validity():
         self.add_dependency(dep)
     self._mark_as_parsed([
