--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,12 @@ b2sums=('3385221b9e12dc452cce8d7ea2e4bd2ff20fd4beffceea9e8084ff45f2817ff9102bf83
 
 validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
+prepare() {
+  cd ${pkgname}-${pkgver}
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   cargo build --release --locked
