--- PKGBUILD.orig	2021-11-30 18:20:28.179427579 +0800
+++ PKGBUILD	2021-11-30 18:27:50.272288824 +0800
@@ -22,7 +22,9 @@
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked 
 }
 
 build() {
