diff --git PKGBUILD PKGBUILD
index 298e9f1..10fbc9c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -2,18 +2,24 @@
 # Contributor: rebel <novakpetya at yahoo dot co dot uk>
 
 pkgname=qt5pas
-_svnrevision=65219
+_pkgname=lazarus
+#_svnrevision=65219
+_commit=703f57c9f2a9ba3a38b91b3f38a21776aab164cb
 pkgver=2.6.r65219
 pkgrel=1
 pkgdesc='Free Pascal Qt5 binding library updated by lazarus IDE'
 arch=('x86_64')
-url='https://svn.freepascal.org/svn/lazarus/trunk/lcl/interfaces/qt5/cbindings'
+url='https://gitlab.com/freepascal.org/lazarus/lazarus/-/tree/main/lcl/interfaces/qt5/cbindings'
 license=('LGPL3')
 depends=('qt5-base' 'qt5-x11extras')
-makedepends=('subversion')
-source=("$pkgname::svn+https://svn.freepascal.org/svn/lazarus/trunk/lcl/interfaces/qt5/cbindings#revision=$_svnrevision")
+makedepends=('git' 'subversion')
+source=("$_pkgname::git+https://gitlab.com/freepascal.org/lazarus/lazarus#commit=$_commit")
 sha512sums=('SKIP')
 
+prepare() {
+	mv "$_pkgname/lcl/interfaces/qt5/cbindings" "$pkgname"
+}
+
 build() {
     cd "$pkgname"
 	qmake-qt5 "QT += x11extras"
