--- a/configure.ac      2022-06-17 14:04:00.757308838 +0200
+++ b/configure.ac      2022-06-17 13:55:30.352253509 +0200
@@ -17,7 +17,8 @@
 AC_DEFINE_UNQUOTED([CANONICAL_TARGET], "$target",
   [Define to canonical target])
 
-AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror dist-xz no-dist-gzip tar-ustar])
+AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror dist-xz no-dist-gzip tar-ustar
+                  subdir-objects])
 AM_SILENT_RULES([yes])
 
 # GNU Automake 1.12 requires this macro. Earlier versions do not
