--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,11 @@ options=('docs')
 source=("xmlstarlet-$pkgver.tar.gz::http://sourceforge.net/projects/xmlstar/files/xmlstarlet/$pkgver/xmlstarlet-$pkgver.tar.gz/download")
 sha512sums=('4228df812caec7059d7a76986c4d9a4262bd861cc53dca05f341ae6c062be05f1c39fc637918ab00f60f40587c6c556e3c9bfaf8a18b149e3c321a92214dbe8b')
 
+prepare() {
+  cd "$srcdir/$pkgname-$pkgver"
+  autoreconf -fi
+}
+
 build() {
   cd "$srcdir/$pkgname-$pkgver"
   LIBXSLT_PREFIX=/usr LIBXML_PREFIX=/usr ./configure --prefix=/usr --enable-build-docs
