diff --git PKGBUILD PKGBUILD
index 84f931c..3a36fc6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@
             'git: For downloading packages with git'
             'mercurial: For downloading packages with mercurial'
             'rsync: For downloading packages with rsync')
-source=("https://github.com/ocaml/opam/releases/download/$pkgver/opam-full-${pkgver/_/-}.tar.gz")
+source=("https://github.com/ocaml/opam/releases/download/${pkgver%_*}/opam-full-${pkgver/_/-}.tar.gz")
 b2sums=('808f896c07462f69d31c84266be5e430bea0084f2c463b60724ce5b861e0565dae72fae10a88fb0ff33b11a290606321360c378b9b9b0e274f04c8869b933557')
 
 build() {
