Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1356221)
+++ PKGBUILD	(working copy)
@@ -8,18 +8,25 @@
 arch=('x86_64')
 url="https://github.com/sayanarijit/xplr"
 license=('MIT')
-depends=('gcc-libs' 'hicolor-icon-theme')
+depends=('gcc-libs' 'hicolor-icon-theme' 'lua51')
 makedepends=('cargo')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
-        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc")
+        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc"
+        remove-luajit.patch)
 sha512sums=('d6f09610e3a2e2d29d391c6350734e26d435ab72be629c6590b09f3991b337ad7d086bc82024815cfb934eb2c1f69dd0e46da63f4b90a1305edc879668ad5f26'
-            'SKIP')
+            'SKIP'
+            'dfb99e1343cce69da5d1839e2133e1ef3b69383eb80b1856507d1b6acc1bb0b560f53002f0922e22230b16d78734450340d8e2de17a5109d059f228a013b3c43')
 validpgpkeys=('D59CA14710C17C6B24717AF90F8EF5258DC38077') # Arijit Basu (June 3, 2021)
 options=('!lto')
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -p0 -i ../remove-luajit.patch
+}
+
 build() {
   cd "$pkgname-$pkgver"
-  cargo build --locked --release
+  cargo build --release
 }
 
 check() {
