--- PKGBUILD
+++ PKGBUILD
@@ -8,15 +8,23 @@ pkgdesc="A hackable, minimal, fast TUI file explorer"
 arch=('x86_64')
 url="https://github.com/sayanarijit/xplr"
 license=('MIT')
-depends=('gcc-libs' 'hicolor-icon-theme')
+depends=('gcc-libs' 'hicolor-icon-theme' 'luajit')
 makedepends=('cargo')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
-        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc")
+        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc"
+        use-system-luajit.patch)
 sha512sums=('75cb1781106516ba53b937577e6c05de46eb3f770505808c72e4aae7ed68fb53a1c4079d4f42724ad9765c8ba11eb149c53c9785c9144d19816891f4762aa883'
-            'SKIP')
+            'SKIP'
+            '9c2154f3023da17ce1edab91913308f06fd0d6e618dc477a2f1b2742cc4ffb78459263df7df3cf6217967e2b2bbbb5334e725c4778fdd91d8415039dd067860e')
 validpgpkeys=('D59CA14710C17C6B24717AF90F8EF5258DC38077') # Arijit Basu (June 3, 2021)
 options=('!lto')
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -p0 -i ../use-system-luajit.patch
+  cargo update -p mlua
+}
+
 build() {
   cd "$pkgname-$pkgver"
   cargo build --locked --release
