--- PKGBUILD
+++ PKGBUILD
@@ -8,18 +8,25 @@ pkgdesc="A hackable, minimal, fast TUI file explorer"
 arch=('x86_64')
 url="https://github.com/sayanarijit/xplr"
 license=('MIT')
-depends=('gcc-libs' 'hicolor-icon-theme')
+depends=('gcc-libs' 'hicolor-icon-theme' 'luajit')
 makedepends=('cargo')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
-        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc")
+        "$pkgname-$pkgver.tar.gz.asc::$url/releases/download/v${pkgver}/source.tar.gz.asc"
+        use-system-luajit.patch)
 sha512sums=('d0082ded7d41f200c71b526cb7881f3885c4d3d96d7beaa91222ee8bd0d134926ac8a8a22b495e582ee586b99d907d0b5d9d27dfcc923be81abf40a93b501349'
-            'SKIP')
+            'SKIP'
+            '62bb79b3b05e836720ed1713e262b8638404d12fe749fa5f90e3e81c8f6e255cc9fd5a1bd82c5e5e9d0fd1a1c4e1e606aba3ac88abb23126829cc9e57a8ec984')
 validpgpkeys=('D59CA14710C17C6B24717AF90F8EF5258DC38077') # Arijit Basu (June 3, 2021)
 options=('!lto')
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -p0 -i ../use-system-luajit.patch
+}
+
 build() {
   cd "$pkgname-$pkgver"
-  cargo build --locked --release
+  cargo build --release
 }
 
 check() {
