--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,12 @@ source=("https://github.com/${_pkgname}/${_pkgname}/archive/${pkgver}/${pkgname}
 b2sums=('932855a8d81801a5063e0085f4b0419acc81a70f93668efaa0a874295f2df9c73e51d9b25aa86346806ebd39f1ba3e15896c408ccc936e60364324b2662bd179')
 options=(!lto)
 
+prepare() {
+  cd ${_pkgname}-${pkgver}
+  sed -i "/patch.crates-io/a ring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   cd ${_pkgname}-${pkgver}
   OPENSSL_NO_VENDOR=1 cargo build --release --locked
