--- PKGBUILD
+++ PKGBUILD
@@ -23,6 +23,7 @@
 prepare() {
   cd libretro-blastem
   patch -Np1 -i ../libretro-blastem-flags.patch
+  patch -Np1 -i ../libretro-blastem-generic-cores.patch
 }
 
 build() {
@@ -34,3 +35,7 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+makedepends+=(python)
+source+=("libretro-blastem-generic-cores.patch::https://github.com/libretro/blastem/commit/d4efbf0932fec6ba6a2f43f72fa9570a36a9ae55.patch")
+b2sums+=('cf4af03163f098ffb610e0e4a136395017bd9014baef58448a143a73fb2371249734ff175ff1f624a12540c42446c7cbcad5bb1f5403b95b87dfdb2abce44411')
