diff --git PKGBUILD PKGBUILD
index afa85c9..d050adf 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ groups=(libretro)
 depends=(libretro-core-info)
 makedepends=(git)
 _commit=e70b05bf5b5954140cba227fc996ab24c064d494
-source=(libretro-blastem::git+https://github.com/libretro/blastem.git#commit=${_commit})
-b2sums=(SKIP)
+source=(libretro-blastem::git+https://github.com/libretro/blastem.git#commit=${_commit}
+        "fix_abi_flags.patch")
+b2sums=('SKIP'
+        '37c6b7dff36ccb32cf9d9033d617d8fb7bb3b8893c79d775ad6233faea656bc0a3cb68ae5d1c01898f2fe789ae29c4218fdf7acc572311e69ada1b15fcf32f57')
+
+prepare() {
+  cd libretro-blastem
+  patch -Np1 -i "../fix_abi_flags.patch"
+}
 
 pkgver() {
   cd libretro-blastem
