--- PKGBUILD.orig	2021-11-18 12:52:03.644547226 +0800
+++ PKGBUILD	2021-11-18 12:52:40.311717339 +0800
@@ -8,7 +8,7 @@
 arch=('x86_64')
 license=('Apache')
 depends=('glibc')
-makedepends=('go')
+makedepends=('go' 'systemd')
 source=($pkgname-$pkgver.tar.gz::https://github.com/grafana/agent/archive/v$pkgver.tar.gz
         grafana-agent.sysusers grafana-agent.service grafana-agent.tmpfiles)
 sha512sums=('f65e238f2370ced07888b1003d6eeda95c40b0c51a9b27cc07a5344e85db8ee147117311c02014e253a507433ac17d35c0c352a60dd248b4575aa2984b14b816'
