diff --git PKGBUILD PKGBUILD
index 4f2422a..4f099a3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,13 +10,21 @@ license=('MIT')
 depends=('glibc')
 makedepends=('go' 'git')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/ginuerzh/gost/archive/v$pkgver.tar.gz"
-        LICENSE)
+        LICENSE
+        "update_go_version.patch"
+        "fix_riscv64_type.patch")
 sha512sums=('99223207c84f3ba899d0682e4a6f1c50a810b0fe08c60e11fda9b3f2adda136f67a84f93112d51e13f7928a0d8a42ef8ec7994b816ec4045ad5e7123df997811'
-            '8815c27a49d747e90fbcbf03c6011a76b95bc72a6c963ee8df9490e6aba9a3f4f9895e6a7759988aef123160ef8a6d2de85e8fce01c5980f8744e212792ad07c')
+            '8815c27a49d747e90fbcbf03c6011a76b95bc72a6c963ee8df9490e6aba9a3f4f9895e6a7759988aef123160ef8a6d2de85e8fce01c5980f8744e212792ad07c'
+            '6ba9eecfcbb9af9cd69457ed4b31c6a92c7ffac9230212a6c539b02ef69834f90c2448ee2cd568caf88c47fac928d8bda07b34a3aa27c9a9de70eb4d7c4080cb'
+            '064d0d204ed0cb2e9f6e86f3c91a655475335c609f9722266fae343fe59543919c2dbd95b4f7303699ecf82defce050bec754df1b6a4e0e6a307e5bfcbd33a0f')
 
 prepare() {
   mkdir -p .gopath/src/github.com/ginuerzh
   ln -s "$PWD/gost-$pkgver" .gopath/src/github.com/ginuerzh/gost
+  cd gost-$pkgver
+  patch -Np1 -i ../update_go_version.patch
+  go mod vendor
+  patch -Np1 -i ../fix_riscv64_type.patch
 }
 
 build() {
@@ -25,7 +33,7 @@ build() {
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -trimpath -modcacherw"
 
   cd gost-$pkgver/cmd/gost
   go build -ldflags "-extldflags ${LDFLAGS}" # -gccgoflags "$CFLAGS $LDFLAGS"
