diff --git PKGBUILD PKGBUILD
index 06877faa..685cfad7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,16 +10,24 @@
 depends=('glibc')
 makedepends=('go' 'git')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/ginuerzh/gost/archive/v$pkgver.tar.gz"
-        LICENSE)
+        LICENSE
+        "fix_riscv64_type.patch")
 sha512sums=('e734d62442231561c42296a441f0151cf746c5760afccd5a13fd564eb3d3fa96b61b6dd351311411814e43840dc3a30ff9d60bbc682ae9aad500414cd3cc6ff1'
-            '8815c27a49d747e90fbcbf03c6011a76b95bc72a6c963ee8df9490e6aba9a3f4f9895e6a7759988aef123160ef8a6d2de85e8fce01c5980f8744e212792ad07c')
+            '8815c27a49d747e90fbcbf03c6011a76b95bc72a6c963ee8df9490e6aba9a3f4f9895e6a7759988aef123160ef8a6d2de85e8fce01c5980f8744e212792ad07c'
+            '064d0d204ed0cb2e9f6e86f3c91a655475335c609f9722266fae343fe59543919c2dbd95b4f7303699ecf82defce050bec754df1b6a4e0e6a307e5bfcbd33a0f')
+
+prepare() {
+  cd gost-$pkgver
+  go mod vendor
+  patch -Np1 -i ../fix_riscv64_type.patch
+}
 
 build() {
   export CGO_CPPFLAGS="${CPPFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external -modcacherw"
 
   cd gost-$pkgver/cmd/gost
   go build
