diff --git PKGBUILD PKGBUILD
index 8bb9d05..4618d73 100644
--- PKGBUILD
+++ PKGBUILD
@@ -26,7 +26,8 @@ build() {
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DBUILD_TESTING=OFF \
     -DCRYFS_UPDATE_CHECKS=OFF \
-    -DDEPENDENCY_CONFIG="cmake-utils/DependenciesFromLocalSystem.cmake"
+    -DDEPENDENCY_CONFIG="cmake-utils/DependenciesFromLocalSystem.cmake" \
+    -DDISABLE_OPENMP=ON
   cmake --build build
 }
 
