--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,8 @@
 prepare() {
   cd "$pkgname"
 
+  patch -Np1 -i "$srcdir/shared-cooldown-test.patch"
+
   # download dependencies
   cargo fetch --locked --target host-tuple
 }
@@ -49,3 +51,7 @@
   # license
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSES/*
 }
+
+source+=(shared-cooldown-test.patch)
+sha512sums+=('ef7a3deefd5c0a18fe524484da496f1a4fd7330957dbb13c55e6a3bc35897ca4b3fc6e1490452b6646c8251ba47d5785207ec489d5954b21c8a206c60dc88f2f')
+b2sums+=('3503da8cca8aa96d172b635a184b6012d9da45a4a3c721693e46b59d899aa3038dec8be4a8af896919d51e3c6e9569cee9bcdffe043f50d1b30cdcf20fb13f40')
