--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,8 @@
   cd ${pkgname}
   # not sure why this works lol
   patch -Np1 -i ../http-service-async-mock.patch
+  # setuptools-scm 10 rejects absolute version-file paths.
+  patch -Np1 -i ../setuptools-scm-relative-version-file.patch
 }
 
 build() {
@@ -46,4 +48,7 @@
   install -Dm 644 README.md -t "${pkgdir}/usr/share/doc/${pkgname}"
 }
 
+source+=(setuptools-scm-relative-version-file.patch)
+sha512sums+=('b382dab8ec591ec5236f60f135fed821a9028c774390a7141b4fc15d6b8d496c651d813896bf9ea1d67113f2b73b9af2df55207d4de01ec63df245e51b21bee7')
+
 # vim: ts=2 sw=2 et:
