--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,7 @@
             'xdg-desktop-portal: for wayland support, you will need the implementation for your wayland desktop environment'
             'qt6-imageformats: for additional export image formats (e.g. tiff, webp, and more)')
 source=("${url}/archive/refs/tags/v${pkgver}/${pkgname}-${pkgver}.tar.gz")
-sha256sums=('31a9419089aee8570fa0a996515b97d165a23672b2da1e3a0b5ed2f7f68a3631')
+sha256sums=('810c399f3b9fbfd72e24e61417ede24243925f9c0d03040a8aba0d4866676d93')
 
 build() {
 	cmake -B build -S "${pkgname}-${pkgver}" \
