diff --git PKGBUILD PKGBUILD
index 0c3f94a2..f64251d4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@
 license=(BSD)
 depends=(python)
 makedepends=(python-build python-installer python-setuptools python-wheel)
-checkdepends=(python-pytest-cov python-pytest-flake8)
+checkdepends=(python-pytest-cov python-pytest-flake8 python-six)
 source=(https://pypi.io/packages/source/${_pipname:0:1}/$_pipname/$_pipname-$pkgver.tar.gz)
 sha256sums=('751cde3b76b176d25f961b90c423a11a4d5ede9bd09ab0d64a85abb738c190d8')
 
