--- PKGBUILD
+++ PKGBUILD
@@ -18,13 +18,20 @@ depends=('python-html5lib')
 makedepends=('python-setuptools')
 checkdepends=('python-zope-interface' 'python-twisted')
 optdepends=('python-html5-parser: faster parsing of HTML')
-source=("https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/${_pkgname}/${_pkgname}-${pkgver}.tar.gz"{,.asc})
+source=("https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/${_pkgname}/${_pkgname}-${pkgver}.tar.gz"
+        "fix-test_Cookie_iterator.patch")
 sha256sums=('5e86ac0777357e006eb04cd28f7ed9f811d48dffa603d3891ac6d2b92280dc91'
-            'SKIP')
+            '24f66263b788a554f7900aab416cc7a8e80be67f015787fcf00b5a425d3600c6')
 b2sums=('bd91738092ac473ee6e65ee5f175479515f735a0433b112241f17061adf7e60ef6fa410fa549f088390b6ef6c8b2ff95e00335a626322ad170c88923c96e8e44'
-        'SKIP')
+        'd5fb359773eb09807e405ba51474b16604d4cd86b89e99c1035e513d4a879280f703d36250878287bab3cd43d56faa5a7b7d0477123f848863280b60dcbcffff')
 validpgpkeys=('3CE1780F78DD88DF45194FD706BC317B515ACE7C') # Kovid Goyal (New longer key) <kovid@kovidgoyal.net>
 
+prepare() {
+  cd "mechanize-$pkgver"
+
+  patch -Np1 -i ../fix-test_Cookie_iterator.patch
+}
+
 build() {
   cd "mechanize-$pkgver"
 
