diff --git a/PKGBUILD b/PKGBUILD
index 1608047..9f8e282 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,8 @@ sha256sums=('1753736c31feea0085f5cfac33143743204f8a7e66b81ccd17e249ecafba802f')
 
 build() {
   cd $pkgname-$pkgver
+  cp /usr/share/autoconf/build-aux/config.guess .
+  cp /usr/share/autoconf/build-aux/config.sub .
   ./configure --prefix=/usr
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
   make
