--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,7 @@ depends=(qt6-base)
 makedepends=(cmake ninja qt6-declarative clang llvm litehtml)
 optdepends=('qt6-declarative: for qdoc' 'clang: for qdoc and lupdate' 'litehtml: for assistant')
 groups=(qt6)
+options=('!lto')
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
 source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
 sha256sums=('867df829cd5cd3ae8efe62e825503123542764b13c96953511e567df70c5a091')
