diff --git PKGBUILD PKGBUILD
index c02c6d0..493f497 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@
 url="https://github.com/veeso/termscp"
 license=('MIT')
 depends=('gcc-libs' 'zlib' 'openssl' 'dbus' 'smbclient')
-makedepends=('cargo')
+makedepends=('cargo' 'cmake' 'clang')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
 sha512sums=('5295298185c85f642d877ec9e4a6ace1ae321ea2f9eefac966570550f02df78987d3b2b25c6a589b9328dd08df36e74944bab7c5bb7bf17fcd57d612effccc8f')
 options=('!lto')
