diff --git PKGBUILD PKGBUILD
index 68f79ef..b856005 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,10 +16,12 @@ groups=(gnome-extra)
 _commit=678141fa4cdce0af0baccff965c27734b0c4546b  # master
 source=("git+https://gitlab.gnome.org/GNOME/gnome-documents.git#commit=$_commit"
         "git+https://gitlab.gnome.org/GNOME/libgd.git"
-        0001-getting-started-Fix-the-build-with-inkscape-1.0.patch)
+        0001-getting-started-Fix-the-build-with-inkscape-1.0.patch
+        "fix_positional_args.patch")
 sha256sums=('SKIP'
             'SKIP'
-            '9c7dea42e61b179c780795ae3441ec92a06d5660242649f39aaf441d0b1bb6c0')
+            '9c7dea42e61b179c780795ae3441ec92a06d5660242649f39aaf441d0b1bb6c0'
+            'e67ca36f75f99850a19ef78e2b101eec7cd435cccf3fcc5e0e172dbd2e60ede3')
 
 pkgver() {
   cd $pkgname
@@ -28,6 +30,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "../fix_positional_args.patch"
 
   # https://bugs.archlinux.org/task/67942
   git apply -3 ../0001-getting-started-Fix-the-build-with-inkscape-1.0.patch
