Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 460995)
+++ PKGBUILD	(working copy)
@@ -8,8 +8,7 @@
 arch=(x86_64)
 license=('LGPL')
 depends=('gdk-pixbuf2' 'sg3_utils' 'libimobiledevice' 'libplist' 'sqlite')
-makedepends=('intltool' 'swig' 'docbook-xsl' 'gtk-sharp-2' 'gtk-doc')
-optdepends=('gtk-sharp-2: Mono bindings')
+makedepends=('intltool' 'swig' 'docbook-xsl' 'gtk-doc')
 url="http://www.gtkpod.org/libgpod/"
 source=(https://downloads.sourceforge.net/sourceforge/gtkpod/${pkgname}-${pkgver}.tar.bz2
         libgpod-0.8.2-pkgconfig_overlinking.patch
@@ -30,7 +29,7 @@
   patch -Np1 -i ../0001-323-Segmentation-fault-when-opening-ipod.patch
   patch -Np0 -i ../libgpod-udev.patch
   sed -e 's|libplist >= 1.0|libplist-2.0 >= 2.2|' -i configure.ac # support libplist 2.2
-  autoreconf -vi
+  autoreconf -fvi
 }
 
 build() {
