Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1345653)
+++ PKGBUILD	(working copy)
@@ -20,6 +20,11 @@
 source=("https://hackage.haskell.org/packages/archive/${_hkgname}/${pkgver}/${_hkgname}-${pkgver}.tar.gz")
 sha512sums=('34fd648d329a9cd60d32b5eadb19bf254dcde8772474c97c58a719fe591d92b160680a7ba0c9527d1627f708dc29e372fcd176fac61f874c419808e0e97decdc')
 
+prepare() {
+    cd $_hkgname-$pkgver
+    sed -i '/test-suite doctests/a \  buildable: False' $_hkgname.cabal
+}
+
 build() {
     cd $_hkgname-$pkgver
 
