diff --git a/common.mak b/common.mak
index fb8ad0d..08d7126 100644
--- a/common.mak
+++ b/common.mak
@@ -160,14 +160,14 @@ ifeq ($(LV2AVAIL)$(HAVE_UI)$(HAVE_IDLE), yesyesyes)
       UILIBS+=-lws2_32 -lwinmm -lopengl32 -lglu32 -lgdi32 -lcomdlg32 -lpthread
     else
       UIDEPS+=../pugl/pugl_x11.c
-      override CFLAGS+=`pkg-config --cflags glu`
+      override CFLAGS+=`pkg-config --cflags gl glu`
       UILIBS=../pugl/pugl_x11.c -lX11
       ifeq ($(STATICBUILD), yes)
-        UILIBS+=`pkg-config --libs glu`
+        UILIBS+=`pkg-config --libs gl glu`
         UILIBS+=`pkg-config --variable=libdir ftgl`/libftgl.a `pkg-config --variable=libdir ftgl`/libfreetype.a
         UILIBS+=`pkg-config --libs zlib`
       else
-        UILIBS+=`pkg-config --libs glu ftgl`
+        UILIBS+=`pkg-config --libs gl glu ftgl`
       endif
       UICFLAGS+=-DFONTFILE=\"$(FONTFILE)\"
     endif
