diff --git PKGBUILD PKGBUILD
index 6371147f..d7cb1dd6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,9 +16,12 @@ makedepends=('gendesk' 'glib2' 'jack' 'liblo' 'lv2' 'pango' 'ttf-dejavu')
 optdepends=('jack: for standalone applications'
             'liblo: for standalone applications'
             'lv2-host: for LV2 plugin')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/pantherb/${_name}/archive/v${pkgver}.tar.gz")
-sha512sums=('1db2ee90032efee3c24ee99071e11c064a2fbc82f1126346ec0667e1e91f09ffab781bcdb8fa766613711779cdfd85b486e0043f99073d8072ac92e214241c88')
-b2sums=('63b89fb4865ef449d3e291c547f2e115b5fcb0e5de280353e58b3b25e9b9c4d7291df09b2c486d8819295e367a1f3128c0fb5a6bf3b144a999e8ae634eda5f4c')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/pantherb/${_name}/archive/v${pkgver}.tar.gz"
+        "fix_missing_lib.patch")
+sha512sums=('1db2ee90032efee3c24ee99071e11c064a2fbc82f1126346ec0667e1e91f09ffab781bcdb8fa766613711779cdfd85b486e0043f99073d8072ac92e214241c88'
+            'c5ec40afd8d9cf9bbf21c0a745deabd0337ceff3a81cd856c91390e304c37b3e6a4269faa83942f7dcc6e6ac0c55cebe3faecec5a1c4f9f3dd4e783049c9ec16')
+b2sums=('63b89fb4865ef449d3e291c547f2e115b5fcb0e5de280353e58b3b25e9b9c4d7291df09b2c486d8819295e367a1f3128c0fb5a6bf3b144a999e8ae634eda5f4c'
+        '37c92ebaaaf412ad4ee209c068fc31cc509029086f722c50d75e9725c8dc93f22b323587f50866afa44e23fef8d819d3b3d76c48934df3a278db5b59557a3ef9')
 
 prepare() {
   mv -v "${_name}-$pkgver" "$pkgname-$pkgver"
@@ -37,6 +40,8 @@ prepare() {
           --genericname "Leslie speaker emulation" \
           --pkgdesc "Leslie speaker emulation as JACK client" \
           --categories "AudioVideo;Audio"
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i "../fix_missing_lib.patch"
 }
 
 build() {
