diff --git PKGBUILD PKGBUILD
index 1bebd295..b0dd2e00 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,9 +12,16 @@ license=('GPL')
 depends=('libnet' 'libpcap' 'glib2' 'libnsl')
 makedepends=('pkgconfig')
 validpgpkeys=('67E00C8AE6DEE3486468F6C6E20D29536F5C037F')
-source=(https://downloads.sourceforge.net/${pkgname}/${pkgname}-${pkgver}.tar.gz{,.asc})
+source=(https://downloads.sourceforge.net/${pkgname}/${pkgname}-${pkgver}.tar.gz{,.asc}
+        "add_riscv_support.patch")
 sha256sums=('314b4793e0902fbf1fdb7fb659af37a3c1306ed1aad5d1c84de6c931b351d359'
-            'SKIP')
+            'SKIP'
+            '7ba6610936b19af3fcab9f04a2fb2282988758e13202b565e5a2bb149da6f00a')
+
+prepare() {
+  cd "$srcdir"/$pkgname-$pkgver
+  patch -Np1 -i ../add_riscv_support.patch
+}
 
 build() {
   cd "$srcdir"/$pkgname-$pkgver
