diff --git PKGBUILD PKGBUILD
index ee667cf..bee597a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ url="http://libnids.sourceforge.net/"
 license=('GPL')
 depends=('libnet' 'libpcap' 'glib2' 'libnsl')
 makedepends=('pkgconfig')
-source=(https://github.com/MITRECND/libnids/archive/$pkgver/$pkgname-$pkgver.tar.gz)
-sha256sums=('3f3e9f99a83cd37bc74af83d415c5e3a7505f5b190dfaf456b0849e0054f6733')
+source=(https://github.com/MITRECND/libnids/archive/$pkgver/$pkgname-$pkgver.tar.gz
+        "add_riscv_support.patch")
+sha256sums=('3f3e9f99a83cd37bc74af83d415c5e3a7505f5b190dfaf456b0849e0054f6733'
+        '7ba6610936b19af3fcab9f04a2fb2282988758e13202b565e5a2bb149da6f00a')
+
+prepare() {
+  cd "$srcdir"/$pkgname-$pkgver
+  patch -Np1 -i ../add_riscv_support.patch
+}

 build() {
   cd $pkgname-$pkgver
