--- PKGBUILD
+++ PKGBUILD
@@ -91,6 +91,8 @@ prepare(){
 
 	# Remove unneeded files
 	rm -f resources/$pkgname-portable.*
+
+	patch -Np1 -i "$srcdir"/fix-stbsp__uintptr.patch -d src/calibre/utils
 }
 
 build() {
@@ -135,3 +137,6 @@ package() {
 		python -O -m compileall -d "${_destdir}" "${_file}"
 	done < <(find "${pkgdir}"/usr/lib/ -name '*.py' -print0)
 }
+
+source+=("fix-stbsp__uintptr.patch")
+sha256sums+=('de1f5245c806aee957ae3dc4cb644e14a9be818e997bbc9f9f11929aa5b37ad3')
