diff --git PKGBUILD PKGBUILD
index ad9508b..51f0ac3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -44,7 +44,6 @@ depends=(
   'libxft'
   'libxinerama'
   'libxml2'
-  'libxnvctrl' libXNVCtrl.so
   'lua'
   'ncurses' libncursesw.so
   'pango'
@@ -87,7 +86,7 @@ build() {
     -D BUILD_IMLIB2=ON \
     -D BUILD_CURL=ON \
     -D BUILD_RSS=ON \
-    -D BUILD_NVIDIA=ON \
+    -D BUILD_NVIDIA=OFF \
     -D BUILD_PULSEAUDIO=ON \
     -D BUILD_JOURNAL=ON \
     -D BUILD_WAYLAND=ON \
