diff --git PKGBUILD PKGBUILD
index 658b1ffe..5d9f7c5b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,6 +23,11 @@ replaces=(hdf5-${_mpi}-java)
 source=(https://support.hdfgroup.org/ftp/HDF5/releases/${_pkgname}-${pkgver:0:4}/${_pkgname}-${pkgver/_/-}/src/${_pkgname}-${pkgver/_/-}.tar.bz2)
 sha256sums=('1a88bbe36213a2cea0c8397201a459643e7155c9dc91e062675b3fb07ee38afe')
 
+prepare() {
+    cd ${_pkgname}-${pkgver/_/-}
+    autoreconf -fiv
+}
+
 build() {
     # Crazy workaround: run CMake to generate pkg-config file
     #cmake -B build -S ${pkgname}-${pkgver/_/-} \
