--- PKGBUILD
+++ PKGBUILD
@@ -21,12 +21,13 @@ sha256sums=('SKIP'
             '7b0919ac447824199aff8c17b5a5799b46414818c6aed314506c5295d0ce9ccd')
 
 get_rust_target() {
-    RUSTC_BOOTSTRAP=1 rustc -Z unstable-options --print target-spec-json | python -c 'import json,sys;obj=json.load(sys.stdin);print(obj["llvm-target"])'
+    echo riscv64gc-unknown-linux-gnu
 }
 
 build() {
     cd "db-${_db_version}/build_unix"
 
+    cp /usr/share/autoconf/build-aux/config.{guess,sub} ../dist
     ../dist/configure \
         --prefix=/ \
         --disable-shared \
@@ -39,11 +40,16 @@ build() {
 
     cd "${srcdir}/${pkgname}"
 
+    echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+    cargo update -p ring
+
     CPPFLAGS="${CPPFLAGS} -I${srcdir}/db-root/include -I/usr/include/utf8cpp"
     LDFLAGS="${LDFLAGS} -L${srcdir}/db-root/lib"
 
     export CC=/usr/bin/clang
     export CXX=/usr/bin/clang++
+    export CFLAGS=${CFLAGS/-fstack-clash-protection}
+    export CXXFLAGS=${CXXFLAGS/-fstack-clash-protection}
 
     ./autogen.sh
     ./configure --prefix=/usr \
