--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,9 @@ prepare() {
   # Fix buld with Rust 1.80.0
   cargo update time --precise 0.3.36
 
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
+
   CARGO_HOME="$srcdir/build/cargo-home" \
     cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
