diff --git PKGBUILD PKGBUILD
index d0fc15b..8b9d158 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@ sha256sums=('6b186b76bc995ee12949a48a1041173459eb8050ef1dceb5f7ba43726327b1ea')
 
 build() {
     cd "$srcdir/$pkgname-$pkgver"
-    ./configure --prefix=/usr --with-openmp --with-libimagequant=/usr/lib
+    ./configure --prefix=/usr --with-openmp --with-libimagequant=/usr/lib --disable-sse
     make
 }
 
