diff --git configure.ac configure.ac
index 52f976c..419842f 100644
--- configure.ac
+++ configure.ac
@@ -201,7 +201,7 @@ case "$host" in
 # Linux.
 #
 # aarch64*-*-linux* is for RaspberryPi-4 on ARM-64 - otherwise http/tcp isnt found
-*86-*-gnu | *86_64-*-gnu | *86-*-linux* | *86_64-*-linux* | arm*-*-linux* | aarch*-*-linux*) 
+*86-*-gnu | *86_64-*-gnu | *86-*-linux* | *86_64-*-linux* | arm*-*-linux* | aarch*-*-linux* | riscv64-*-linux*)
   AC_DEFINE(BUILD_LINUX, 1, [Define this if LINUX was detected]) 
   AC_DEFINE(HAVE_IEEE_754, 1, [Define this if IEEE_754 compliant])
   AC_MSG_RESULT([linux])
