--- PKGBUILD
+++ PKGBUILD
@@ -12,9 +12,17 @@ depends=(gsl plotutils)
 optdepends=('postgresql-libs: for postgresql support')
 # This URL was removed:
 source=("https://jmvdveer.home.xs4all.nl/algol68g-$pkgver.tar.gz"
-        'https://ftp.tw.freebsd.org/distfiles/a68g-doc.pdf')
+        'https://ftp.tw.freebsd.org/distfiles/a68g-doc.pdf'
+        'fix-build-host.patch')
 b2sums=('37a0c0e7386f31dd9b986e9c6da0f21a762dc099d5ece48dbdd1bad7789f38bf131e23ffe20596e2d874f71a0bd3a8045739e794f75bd8b5316fd7bf8e99e91e'
-        'f8ca710d9a4f611873de41785e70359bd566645268ebb848803d3cd2e9a9f873f8ac423e3e08699998083444b7f6d98df32fdb7a4c6b36692462ee50ddb1e410')
+        'f8ca710d9a4f611873de41785e70359bd566645268ebb848803d3cd2e9a9f873f8ac423e3e08699998083444b7f6d98df32fdb7a4c6b36692462ee50ddb1e410'
+        'ca6affc2f633dd788a2b127256aaf47fb89bf9e954e408ae30595a2e01a1d6f5406d40109fe7ad081a7c452b38f0fdf9438f8455f36f5b75bc288777f5107c1b')
+
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Ni $srcdir/fix-build-host.patch
+  autoreconf -fiv
+}
 
 build() {
   cd $pkgname-$pkgver
