diff --git PKGBUILD PKGBUILD
index 6ae176f..7ac2312 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,13 +10,15 @@ license=(GPL2)
 source=($pkgname-$pkgver::"https://github.com/tremlin/SymPol/archive/v$pkgver.tar.gz"
         cddlib-0.94k.patch
         sympol-fix-headers.patch
-        sympol-lrs-071.patch)
+        sympol-lrs-071.patch
+        sympol-fix-bliss-interface.patch)
 depends=(boost-libs lrs cddlib)
 makedepends=(cmake permlib eigen bliss)
 sha256sums=('6753b8fb30745b66a0886e125c18b539417afcf62b804799eb220bd5a59ccc37'
             '23f85255ad1acbaf9c63134c6d331e0f3b8d06230a05e3f63e57399863579649'
             'f3ed90e6b9af5dae0728c52ce4bb9107f3040481cac6c08116dbf8c19bfe3971'
-            '0aa37a4b87021ed04e02fcdae525dc305a8b0490c98bfb82bba2d9af04ef1d35')
+            '0aa37a4b87021ed04e02fcdae525dc305a8b0490c98bfb82bba2d9af04ef1d35'
+            'dc68c20f8db35e544c2b5ee3b37cab2b7e6974facb78faaea38a87286d66a416')
 
 prepare() {
   cd SymPol-$pkgver
@@ -25,6 +27,7 @@ prepare() {
   patch -p1 -i ../cddlib-0.94k.patch # Fix build with cddlib 0.94k
   patch -p1 -i ../sympol-fix-headers.patch # Keep headers directory structure
   patch -p0 -i ../sympol-lrs-071.patch # Fix build with lrs 071 (Fedora)
+  patch -p1 -i ../sympol-fix-bliss-interface.patch # Adapt to libbliss update
 }
 
 build() {
