diff --git PKGBUILD PKGBUILD
index 5edafdcd..b7b2cd08 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,14 @@ arch=('any')
 depends=('python-numpy')
 makedepends=('git' 'python-build' 'python-installer' 'python-setuptools' 'python-wheel')
 checkdepends=('python-pytest')
-source=("git+https://github.com/pberkes/big_O.git#commit=$_commit")
-sha512sums=('SKIP')
+source=("git+https://github.com/pberkes/big_O.git#commit=$_commit"
+        "fix-big-o-test.patch"::"https://github.com/pberkes/big_O/commit/a4f9c983291571a1b8a80e9d6a94c36c8455cc4a.patch")
+sha512sums=('SKIP'
+            '9d0a0f2a310c9288827d512163d7424fdc0e338a9d7cd5ae58b79cee413f7fc09a8b23093992f8ec11ca362b55db2700eb2f2c282af4567e745a7c0064b9a0d2')
 
 prepare() {
   cd big_O
+  patch -Np1 -i "../fix-big-o-test.patch"
   # stablize tests on python 3.10+
   # TODO: doesn't apply
   # git cherry-pick -n 81a3a5483fd7804d95ece8cc8f18a1f840ac7a7b
