diff --git PKGBUILD PKGBUILD
index 12d086cb3..b0e775a0f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,10 +21,12 @@ groups=(gnome)
 _commit=87af8a6dcec4dd404e147919f53c036e233db2fc  # tags/41.0^0
 source=("git+https://gitlab.gnome.org/GNOME/gnome-music.git#commit=$_commit"
         "git+https://gitlab.gnome.org/GNOME/libgd.git"
-        "git+https://gitlab.gnome.org/mschraal/gfm.git")
+        "git+https://gitlab.gnome.org/mschraal/gfm.git"
+        "fix_positional_arg.patch")
 sha256sums=('SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            'fffac68272c4dbfe5387fa8e9fcce6371262a9ad87a930306c000054ec19c99b')
 
 pkgver() {
   cd $pkgname
@@ -33,6 +35,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "../fix_positional_arg.patch"
 
   git submodule init
   git submodule set-url subprojects/libgd "$srcdir/libgd"
