--- t_conduit_node_set.cpp
+++ t_conduit_node_set.cpp
@@ -1189,8 +1189,8 @@ TEST(conduit_node_set, set_cstyle_native_int)
   EXPECT_EQ(n.as_char(), char_v);
   EXPECT_EQ(conduit_datatype_is_number(n.c_dtype()), true);
   EXPECT_EQ(conduit_datatype_is_integer(n.c_dtype()), true);
-  EXPECT_EQ(conduit_datatype_is_signed_integer(n.c_dtype()), true);
-  EXPECT_EQ(conduit_datatype_is_unsigned_integer(n.c_dtype()), false);
+  EXPECT_EQ(conduit_datatype_is_signed_integer(n.c_dtype()), false);
+  EXPECT_EQ(conduit_datatype_is_unsigned_integer(n.c_dtype()), true);
   EXPECT_EQ(conduit_datatype_is_floating_point(n.c_dtype()), false);
   EXPECT_EQ(n.to_char(), 2);
 
@@ -2273,8 +2273,8 @@ TEST(conduit_node_set, set_path_cstyle_native_int)
   EXPECT_EQ(nc.as_char(), char_v);
   EXPECT_EQ(conduit_datatype_is_number(nc.c_dtype()), true);
   EXPECT_EQ(conduit_datatype_is_integer(nc.c_dtype()), true);
-  EXPECT_EQ(conduit_datatype_is_signed_integer(nc.c_dtype()), true);
-  EXPECT_EQ(conduit_datatype_is_unsigned_integer(nc.c_dtype()), false);
+  EXPECT_EQ(conduit_datatype_is_signed_integer(nc.c_dtype()), false);
+  EXPECT_EQ(conduit_datatype_is_unsigned_integer(nc.c_dtype()), true);
   EXPECT_EQ(conduit_datatype_is_floating_point(nc.c_dtype()), false);
   EXPECT_EQ(nc.to_char(), 2);
 
