--- PKGBUILD
+++ PKGBUILD
@@ -26,13 +26,18 @@ makedepends=(
 checkdepends=(
   python-mpi4py
 )
-source=("$url/-/archive/v$pkgver/$_name-v$pkgver.tar.gz")
-b2sums=('f5d81bbbe57ccbab692935f9ed45258be72e261af5c788ee0f4601a5ff1fe74be62a94f4a2992871b14a02ed56b5b9393ad02c939174eac292dc26bfded4e2a0')
+source=("$url/-/archive/v$pkgver/$_name-v$pkgver.tar.gz"
+        char-signed-test.patch)
+b2sums=('f5d81bbbe57ccbab692935f9ed45258be72e261af5c788ee0f4601a5ff1fe74be62a94f4a2992871b14a02ed56b5b9393ad02c939174eac292dc26bfded4e2a0'
+       '68daf1210b5c514d2efe627e71647721382d283aceead47b7677540471ca70fec07e80bc659a641704649e85ea0c57122b025193bec5c5fbd4f09bd0b736f302')
 
 # LTO breaks the ABI tests
 options=(!lto)
 
 prepare() {
+  cd ${_name}-v${pkgver}
+  patch -Np0 -i ../char-signed-test.patch
+  cd ..
   # Building the Fortran wrappers causes the ABI test to fail due to three
   # unexpected symbols: https://gitlab.kitware.com/paraview/catalyst/-/issues/42
   # Adding the extra symbols tricks the test suite to pass.
@@ -60,7 +65,7 @@ build() {
 }
 
 check() {
-  ctest --test-dir build --parallel 8 --output-on-failure
+  LD_LIBRARY_PATH=$srcdir/build/lib ctest --test-dir build --parallel 8 --output-on-failure
 }
 
 package() {
