diff --git PKGBUILD PKGBUILD
index 0d23806..625f4a0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,13 @@ makedepends=('git' 'boost')
 source=("git+https://github.com/simsong/$pkgname.git#tag=$pkgname-$pkgver"
         'git+https://github.com/simsong/be13_api.git'
         'git+https://github.com/simsong/dfxml.git'
-        'git+https://github.com/joyent/http-parser.git')
+        'git+https://github.com/joyent/http-parser.git'
+        'fix-werror-format-security.patch::https://patch-diff.githubusercontent.com/raw/simsong/tcpflow/pull/239.patch')
 sha512sums=('SKIP'
             'SKIP'
             'SKIP'
-            'SKIP')
+            'SKIP'
+            '316c9d21fdef948975e8fed3a3f1502990389dd8720fcb3e1ff8a9dc7a8eec758655cfb847c21e484330660056a944ffdb5d5ca8991c3888f264cdafaf97e782')
 
 prepare() {
   cd $pkgname
@@ -28,6 +30,8 @@ prepare() {
   git config submodule."src/http-parser".url "$srcdir/http-parser"
   git submodule update
 
+  patch -Np1 < ../fix-werror-format-security.patch
+
   autoreconf -vi
 }
 
