--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,6 @@ pkgname=(
   linux-firmware-other
 
   # Splits
-  amd-ucode
   linux-firmware-amdgpu
   linux-firmware-atheros
   linux-firmware-broadcom
@@ -67,21 +66,6 @@ prepare() {
   done
 }
 
-build() {
-  mkdir -p kernel/x86/microcode
-  cat ${pkgbase}/amd-ucode/microcode_amd*.bin > kernel/x86/microcode/AuthenticAMD.bin
-
-  # Reproducibility: set the timestamp on the bin file
-  if [[ -n ${SOURCE_DATE_EPOCH} ]]; then
-    touch -d @${SOURCE_DATE_EPOCH} kernel/x86/microcode/AuthenticAMD.bin
-  fi
-
-  # Reproducibility: strip the inode and device numbers from the cpio archive
-  echo kernel/x86/microcode/AuthenticAMD.bin |
-    bsdtar --uid 0 --gid 0 -cnf - -T - |
-    bsdtar --null -cf - --format=newc @- > amd-ucode.img
-}
-
 _pick() {
   local p="$1" f d; shift
   for f; do
