diff --git PKGBUILD PKGBUILD
index 83a7d51..b086a59 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,6 @@
   libdrm
   libfreetype.so
   libiec61883
-  libmfx
   libmodplug
   libpulse
   librav1e.so
@@ -80,13 +79,13 @@
   'avisynthplus: AviSynthPlus support'
   'intel-media-sdk: Intel QuickSync support'
   'ladspa: LADSPA filters'
-  'nvidia-utils: Nvidia NVDEC/NVENC support'
 )
 provides=(
   libavcodec.so
   libavdevice.so
   libavfilter.so
   libavformat.so
+  libavresample.so
   libavutil.so
   libpostproc.so
   libswresample.so
@@ -129,6 +128,7 @@
     --enable-ladspa \
     --enable-libaom \
     --enable-libass \
+    --enable-avresample \
     --enable-libbluray \
     --enable-libdav1d \
     --enable-libdrm \
@@ -137,7 +137,6 @@
     --enable-libgsm \
     --enable-libiec61883 \
     --enable-libjack \
-    --enable-libmfx \
     --enable-libmodplug \
     --enable-libmp3lame \
     --enable-libopencore_amrnb \
@@ -164,8 +163,6 @@
     --enable-libxml2 \
     --enable-libxvid \
     --enable-libzimg \
-    --enable-nvdec \
-    --enable-nvenc \
     --enable-shared \
     --enable-version3
 
