diff --git PKGBUILD PKGBUILD
index 83a7d51..7bb6a2d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,6 @@
   libdrm
   libfreetype.so
   libiec61883
-  libmfx
   libmodplug
   libpulse
   librav1e.so
@@ -78,9 +77,7 @@
 )
 optdepends=(
   'avisynthplus: AviSynthPlus support'
-  'intel-media-sdk: Intel QuickSync support'
   'ladspa: LADSPA filters'
-  'nvidia-utils: Nvidia NVDEC/NVENC support'
 )
 provides=(
   libavcodec.so
@@ -137,7 +134,6 @@
     --enable-libgsm \
     --enable-libiec61883 \
     --enable-libjack \
-    --enable-libmfx \
     --enable-libmodplug \
     --enable-libmp3lame \
     --enable-libopencore_amrnb \
@@ -164,8 +160,6 @@
     --enable-libxml2 \
     --enable-libxvid \
     --enable-libzimg \
-    --enable-nvdec \
-    --enable-nvenc \
     --enable-shared \
     --enable-version3
 
