https://github.com/aio-libs/multidict/pull/1504

--- PKGBUILD
+++ PKGBUILD
@@ -49,3 +49,6 @@
   cd ${pkgname#python-}-$pkgver
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+sha512sums[0]='b3c14370c98a34c03c8aceec54c3f1784c1e294209db8bc5363c3db7ed2836fa8384e1ffceb959597e646ca275f47eaecc82352608863ba60e4fe69dc2a17e36'
+b2sums[0]='8962c996894570489321e312ea765d9fb2da84817b465adfbf27195b6e383c9bd1b0cfcd9b93f0a2e2824db7a9de070dd47e0ac2c99164fc4818ec52b24ee1e2'
