diff --git PKGBUILD PKGBUILD
index 4462eaa..460b1e1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,9 +13,13 @@ depends=('zlib' 'bzip2')
 makedepends=('gperf' 'git')
 options=('staticlibs')
 source=("git+https://github.com/steveicarus/iverilog#tag=v${pkgver/./_}"
-        fix-string-literal.patch::https://github.com/steveicarus/iverilog/commit/23e51ef7a8e8e4ba42208936e0a6a25901f58c65.patch)
+        fix-string-literal.patch::https://github.com/steveicarus/iverilog/commit/23e51ef7a8e8e4ba42208936e0a6a25901f58c65.patch
+        'config.guess::https://github.com/steveicarus/iverilog/raw/20753ee6c513f669c6b3133fbfc647fc5dceaf9c/config.guess'
+        'config.sub::https://github.com/steveicarus/iverilog/raw/20753ee6c513f669c6b3133fbfc647fc5dceaf9c/config.sub')
 b2sums=('e84a7ebcb525a4ad3013f371df8871adf98beb4056d4e4b3502d7171820f59594e3ce892f566a74689adf442ecb6d39597c91f4e3e14d1d0338073544eee8830'
-        '1ec55d4ec133de284bcaa6c9cd9e6d90e6d9c65e499816826e1e8add5d02e25371cb065820179c66c50ca615176a6cb0952d0b86f25ef6d73db5645ef46da04f')
+        '1ec55d4ec133de284bcaa6c9cd9e6d90e6d9c65e499816826e1e8add5d02e25371cb065820179c66c50ca615176a6cb0952d0b86f25ef6d73db5645ef46da04f'
+        '849ad8f80457d0ae1eb85bcb28ea71c418a7ef0c1f38770b493864eca19a1a625db834b836da9b60ef6255c3f754a4352455a3920af0d1fe126d5945a85f91dc'
+        'f2ea4902ac590fc2a45bdd5243e82ee9a0e790fab1ce2e11a93f61295b474b45b6d9e843696b7b92f9185278da96d271efee9000fc11731ff743ae27c99a4b21')
 
 prepare() {
   cd "$pkgname"
@@ -24,6 +28,7 @@ prepare() {
 
   aclocal
   autoconf
+  cp -f ${srcdir}/config.{guess,sub} .
 }
 
 build() {
