diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index eacf13d..15c5eee 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,9 +13,13 @@ depends=('zlib' 'bzip2')
 makedepends=('gperf' 'git')
 options=('staticlibs')
 source=("git+https://github.com/steveicarus/iverilog#tag=v${pkgver/./_}"
-        fix-string-literal.patch::https://github.com/steveicarus/iverilog/commit/23e51ef7a8e8e4ba42208936e0a6a25901f58c65.patch)
+        fix-string-literal.patch::https://github.com/steveicarus/iverilog/commit/23e51ef7a8e8e4ba42208936e0a6a25901f58c65.patch
+        'config.guess::https://github.com/a1ca7raz/iverilog-ar5fix/raw/20753ee6c513f669c6b3133fbfc647fc5dceaf9c/config.guess'
+        'config.sub::https://github.com/a1ca7raz/iverilog-ar5fix/raw/20753ee6c513f669c6b3133fbfc647fc5dceaf9c/config.sub')
 md5sums=('SKIP'
-         '35359d78199f3093b99436e1370249b3')
+         '35359d78199f3093b99436e1370249b3'
+         'bda614c47291e896fef260bd5eb5af15'
+         'dd51eaa5277f305166495477b77b8ff1')
 
 prepare() {
   cd "$pkgname"
@@ -24,6 +28,7 @@ prepare() {
 
   aclocal
   autoconf
+  cp -f ${srcdir}/config.{guess,sub} .
 }
 
 build() {
