--- PKGBUILD
+++ PKGBUILD
@@ -2,7 +2,7 @@
 # Contributor: Adrian Perez de Castro <aperez@igalia.com>
 pkgname=libdispatch
 pkgver=5.5.0
-pkgrel=1
+pkgrel=1.1
 pkgdesc='Comprehensive support for concurrent code execution on multicore hardware'
 arch=('x86_64')
 url=https://apple.github.io/swift-corelibs-libdispatch
@@ -27,6 +27,8 @@ build () {
 	cd "${pkgname}"
     export CC=clang
     export CXX=clang
+    export CFLAGS="$CFLAGS -Wl,-plugin-opt=-target-abi=lp64d"
+    export CXXFLAGS="$CXXFLAGS -Wl,-plugin-opt=-target-abi=lp64d"
 	cmake \
         -Bbuild \
         -GNinja \
