Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 452604)
+++ PKGBUILD	(working copy)
@@ -20,7 +20,7 @@
   cd "${srcdir}/${pkgname}-${pkgver}"
   CFLAGS+=' -ffat-lto-objects'
   CXXFLAGS+=' -ffat-lto-objects'
-  ./configure --prefix /usr --mandir /usr/share/man --disable-force-safe-string --enable-frame-pointers
+  ./configure --prefix /usr --mandir /usr/share/man --disable-force-safe-string
   make --debug=v world.opt
 }
 
