Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -2,7 +2,7 @@
 
 pkgbase='ocaml'
 pkgname=('ocaml' 'ocaml-compiler-libs')
-pkgver=4.10.0
+pkgver=4.11.0+beta2
 pkgrel=1
 pkgdesc="A functional language with OO extensions"
 arch=('x86_64')
@@ -11,7 +11,7 @@
 makedepends=('ncurses>=5.6-7')
 optdepends=('ncurses: advanced ncurses features' 'tk: advanced tk features')
 source=(https://caml.inria.fr/distrib/ocaml-${pkgver%.*}/${pkgname}-${pkgver}.tar.xz)
-sha512sums=('d2ed8b6162898da45ccc231c97a1b46a330467b9b24390ed17cf3e5367ae6d198ecac8e0df11e5501cdf22cc6313ec23c6cf477d621d017f69f9744eb0050e2e')
+sha512sums=('424c1f6ae33fe7c757a06c8a5b7de052a8b887cc2b173b78fa7434b34d219da8148ff69d052197d16c135043616a8f2a3e40d24843158cd70f3a464fe47cbfdf')
 options=('!makeflags' '!emptydirs' 'staticlibs')
 
 
