--- PKGBUILD
+++ PKGBUILD
@@ -54,7 +54,7 @@ build() {
     -modcacherw \
     -ldflags "-compressdwarf=false \
     -linkmode external \
-    -extldflags ${LDFLAGS} \
+    -extldflags \"${LDFLAGS}\" \
       -X main.Version=${pkgver} \
       -X main.BuildTime=${build_time} \
       -X github.com/cloudflare/cloudflared/cmd/cloudflared/updater.BuiltForPackageManager=pacman" \
@@ -65,7 +65,7 @@ build() {
 check() {
   cd "$pkgname"
 
-  go test -v -mod=vendor -race ./...
+  go test -v -mod=vendor ./...
 }
 
 package() {
