diff --git a/build.rs b/build.rs
index b4c0266..612b197 100644
--- a/build.rs
+++ b/build.rs
@@ -331,6 +331,7 @@ fn gnu_target(target: &str) -> String {
         "i686-pc-windows-gnu" => "i686-w64-mingw32".to_string(),
         "x86_64-pc-windows-gnu" => "x86_64-w64-mingw32".to_string(),
         "armv7-linux-androideabi" => "arm-linux-androideabi".to_string(),
+        "riscv64gc-unknown-linux-gnu" => "riscv64-unknown-linux-gnu".to_string(),
         s => s.to_string(),
     }
 }
