diff --git a/src/bootstrap/Cargo.toml b/src/bootstrap/Cargo.toml
index c7a513d0890..a9fce106106 100644
--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -36,7 +36,7 @@ test = false
 # Most of the time updating these dependencies requires modifications
 # to the bootstrap codebase; otherwise, some targets will fail. That's
 # why these dependencies are explicitly pinned.
-cc = "=1.0.73"
+cc = { version = "=1.0.77", path = "../../../cc-1.0.77" }
 cmake = "=0.1.48"
 
 build_helper = { path = "../tools/build_helper" }
