diff --git a/src/bootstrap/Cargo.toml b/src/bootstrap/Cargo.toml
index e4d3591..3637247 100644
--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -34,7 +34,7 @@ test = false
 
 [dependencies]
 build_helper = { path = "../tools/build_helper" }
-cc = "1.0.69"
+cc = { version = "=1.0.77", path = "../../../cc-1.0.77" }
 clap = { version = "4.4.7", default-features = false, features = ["std", "usage", "help", "derive", "error-context"] }
 clap_complete = "4.4.3"
 cmake = "0.1.38"
