diff --git a/src/bootstrap/Cargo.toml b/src/bootstrap/Cargo.toml
index 85eb543e48e..648da1a25e2 100644
--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -33,7 +33,7 @@ test = false
 build_helper = { path = "../tools/build_helper" }
 cmake = "0.1.38"
 filetime = "0.2"
-cc = "1.0.69"
+cc = { version = "=1.0.77", path = "../../../cc-1.0.77" }
 libc = "0.2"
 hex = "0.4"
 object = { version = "0.31.1", default-features = false, features = ["archive", "coff", "read_core", "unaligned"] }
