diff --git a/src/bootstrap/Cargo.toml b/src/bootstrap/Cargo.toml
index 4c32547f059..491b62ac000 100644
--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -38,7 +38,7 @@ test = false
 cmake = "0.1.38"
 filetime = "0.2"
 getopts = "0.2.19"
-cc = "1.0.69"
+cc = { git = "https://github.com/Ast-x64/cc-rs", branch = "fix-flag-supported-compiler-1.0.69", version = "1.0.69" }
 libc = "0.2"
 serde = { version = "1.0.8", features = ["derive"] }
 serde_json = "1.0.2"
