diff --git PKGBUILD PKGBUILD
index bdd8497..604e96c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,6 @@
 pkgbase=rust
 pkgname=(
   rust
-  lib32-rust-libs
   rust-musl
   rust-wasm
   rust-src
@@ -37,7 +36,6 @@ depends=(
 )
 makedepends=(
   cmake
-  lib32-gcc-libs
   libffi
   lld
   llvm
@@ -47,6 +45,7 @@ makedepends=(
   python
   rust
   wasi-libc
+  git
 )
 checkdepends=(
   gdb
@@ -58,13 +57,17 @@ source=(
   0002-bootstrap-Change-bash-completion-dir.patch
   0003-compiler-Change-LLVM-targets.patch
   0004-compiler-Use-wasm-ld-for-wasm-targets.patch
+  pin-cc-to-1-0-77.patch
+  cc-1.0.77::git+https://github.com/rust-lang/cc-rs#tag=1.0.77
 )
 b2sums=('d25823e433ce66d3148359deab7fcf5b6e6aecf832359cf3d7cec55e7f45163c981d396f6b772fa12393896a71e69c91cdfd01d633e2c05d641d0f1b2174e264'
         'SKIP'
         'ec4a0ab6f92cecc7c8ced752a379609089664cbce7e53264167c800df38b672c625f493458547b3d27c643d5c22b4e173115457cb7e441d27795bd930d6ad714'
         'c8db828623b9bc2c9c34a4594f4c9409585c55815d0f3c0bf8b48ea2d3a01fe866c15850503ad5d3b6e04a6e68dd7e7f064d9745f4eb60a267fb90c7ad101d70'
         'cd5a2a5e237a935c236dffc3b0bbf7d468b693f19b63f9d1959f7b13c1c0cc6ec21b717d4f2874d7e357b861e9664fba3c2d22aa2f7f6c932d71adde7b9ac531'
-        'e0acea294146ae14ec18ac1f99cb9113dfe0dbe87cfd557fb093b56fc15cf036ac076af905fbe358a11a913d2fc845619c5fd8e18ac97127c232127a28666117')
+        'e0acea294146ae14ec18ac1f99cb9113dfe0dbe87cfd557fb093b56fc15cf036ac076af905fbe358a11a913d2fc845619c5fd8e18ac97127c232127a28666117'
+        '536662fb94b045efbd0b7b90743995db786bb0b72bb9ffac3081a97d151e0140b19a6104fa2cb744918cddfecd608412e225798dfd2722d445d101f7dd17b458'
+        'SKIP')
 validpgpkeys=(
   108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE  # Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>
 )
@@ -85,6 +88,10 @@ prepare() {
   # Use our wasm-ld
   patch -Np1 -i ../0004-compiler-Use-wasm-ld-for-wasm-targets.patch
 
+  # Bump bootstrap cc to v1.0.77
+  patch -Np1 -i ../pin-cc-to-1-0-77.patch
+  cd src/bootstrap && cargo update -p cc && cd -
+
   cat >config.toml <<END
 profile = "user"
 changelog-seen = 2
@@ -94,9 +101,8 @@ link-shared = true
 
 [build]
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
+  "riscv64gc-unknown-linux-gnu",
+  "riscv64gc-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32-wasi",
 ]
@@ -145,22 +151,18 @@ deny-warnings = false
 [dist]
 compression-formats = ["gz"]
 
-[target.x86_64-unknown-linux-gnu]
+[target.riscv64gc-unknown-linux-gnu]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
 llvm-config = "/usr/bin/llvm-config"
 
-[target.i686-unknown-linux-gnu]
-cc = "/usr/bin/gcc"
-cxx = "/usr/bin/g++"
-ar = "/usr/bin/gcc-ar"
-ranlib = "/usr/bin/gcc-ranlib"
-
-[target.x86_64-unknown-linux-musl]
+[target.riscv64gc-unknown-linux-musl]
 sanitizers = false
 musl-root = "/usr/lib/musl"
+cc = "/usr/bin/musl-gcc"
+cxx = "/usr/bin/musl-gcc"
 
 [target.wasm32-unknown-unknown]
 sanitizers = false
@@ -199,15 +201,12 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -p usr/lib32
-  ln -srft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srft usr/lib   usr/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/*.so
 
   mkdir -p usr/share/licenses/rust
   mv -t usr/share/licenses/rust usr/share/doc/rust/{COPYRIGHT,LICENSE*}
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
+  _pick dest-musl usr/lib/rustlib/riscv64gc-unknown-linux-musl
   _pick dest-wasm usr/lib/rustlib/wasm32-*
   _pick dest-src  usr/lib/rustlib/src
 }
@@ -236,22 +235,6 @@ package_rust() {
   cp -a dest-rust/* "$pkgdir"
 }
 
-package_lib32-rust-libs() {
-  pkgdesc="32-bit target and libraries for Rust"
-  depends=(
-    lib32-gcc-libs
-    rust
-  )
-  provides=(lib32-rust)
-  conflicts=(lib32-rust)
-  replaces=(lib32-rust)
-
-  cp -a dest-i686/* "$pkgdir"
-
-  mkdir -p "$pkgdir/usr/share/licenses"
-  ln -s rust "$pkgdir/usr/share/licenses/$pkgname"
-}
-
 package_rust-musl() {
   pkgdesc="Musl target for Rust"
   depends=(rust)
