--- PKGBUILD
+++ PKGBUILD
@@ -17,6 +17,8 @@
   # cross targets
   rust-aarch64-gnu
   rust-aarch64-musl
+  rust-x86_64-gnu
+  rust-x86_64-musl
 )
 pkgver=1.96.0
 pkgrel=1
@@ -63,12 +65,16 @@
   aarch64-linux-gnu-gcc
   aarch64-linux-gnu-glibc
   musl-aarch64
+  x86_64-linux-gnu-gcc
+  x86_64-linux-gnu-glibc
   musl-x86_64
 )
 makedepends_riscv64=(
   aarch64-linux-gnu-gcc
   aarch64-linux-gnu-glibc
   musl-aarch64
+  x86_64-linux-gnu-gcc
+  x86_64-linux-gnu-glibc
   musl-x86_64
 )
 makedepends_x86_64=(
@@ -130,8 +136,8 @@
         'd9b974154aa5615f4ca0c8956405974d3a8458a5b267e5d1105793b2bbc092be1fba5b579fd7fa97c7eac9d0840cb87090f2cce2091c62ec70a5e121bac84002'
         'd5fcda00c46e1e1ea9ed1d269d3ef549d4ad96148d7d0744acd673de802b3bf3747ceec1fa506d3e7c128cdff742410b1e69218d59ef20a5f246755c1bd02694')
 b2sums_aarch64=('a5e1bc83740f2dd188df333a19276ade5a2fb90707565550b58b74dfd810af31ce1123ddf80b23bd27a477c145f2a45e07d64db19de6f734a1d9449f5d411190')
-b2sums_loong64=('366b3db072ceba926b9bc6d015e917c3f3a51f2abe01ef91eae00502c157737d2f98839a825f2ebfaab0d2e3e724f94bb7150619e0233f84076367b1aab031ab')
-b2sums_riscv64=('d011f1ff717fdfd2d0589d1a2dd7074c0af712c385ff6c02e3f8f88353bdb3ed9c37b8471b23d3290adfa08320c3f61a11980c31aca5c16d126879e486b4105c')
+b2sums_loong64=('ba958de30e7e794f43df76056e77264531374c8d6c29e096fb6408201d9a3943280fdb760a71f563220c6fa5c3b0e0c18cc305066dd4362c5b0bce7afc79d25e')
+b2sums_riscv64=('f20ec5aa83c2d617a990c78c4d580c5e5f3ce839105e0790c649a1e4162188aa8cdc7e426b89789f43437285c9b0f8a11fee59933b8e33ea4ab156d181da6092')
 b2sums_x86_64=('bbc26877880cbdf0ddf100eb28f761d699885656c0650156a9c67ef7ccac639c3cd9c5f3da7c8516fd01e44f87dd88317cd7fc7b52e93da6736cbf9f45220c8b')
 validpgpkeys=(
   108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE  # Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>
@@ -265,6 +271,8 @@
     loong64|riscv64)
       _pick dest-aarch64-gnu usr/lib/rustlib/aarch64-unknown-linux-gnu
       _pick dest-aarch64-musl usr/lib/rustlib/aarch64-unknown-linux-musl
+      _pick dest-x86_64-gnu usr/lib/rustlib/x86_64-unknown-linux-gnu
+      _pick dest-x86_64-musl usr/lib/rustlib/x86_64-unknown-linux-musl
       ;;
   esac
 }
@@ -386,4 +394,35 @@
   _install_licenses
 }
 
+package_rust-x86_64-gnu() {
+  pkgdesc="x86_64 GNU target for Rust"
+  arch=(
+    loong64
+    riscv64
+  )
+  depends=(
+    x86_64-linux-gnu-gcc
+    x86_64-linux-gnu-glibc
+    rust
+  )
+
+  cp -a dest-x86_64-gnu/* "$pkgdir"
+  _install_licenses
+}
+
+package_rust-x86_64-musl() {
+  pkgdesc="x86_64 Musl target for Rust"
+  arch=(
+    loong64
+    riscv64
+  )
+  depends=(
+    x86_64-linux-gnu-gcc
+    rust
+  )
+
+  cp -a dest-x86_64-musl/* "$pkgdir"
+  _install_licenses
+}
+
 # vim:set ts=2 sw=2 et:
--- bootstrap.loong64.toml
+++ bootstrap.loong64.toml
@@ -18,6 +18,8 @@
   "loongarch64-unknown-linux-musl",
   "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
+  "x86_64-unknown-linux-gnu",
+  "x86_64-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32v1-none",
   "wasm32-wasip1",
@@ -107,6 +109,25 @@
 default-linker = "aarch64-linux-gnu-gcc"
 sanitizers = false
 
+[target.x86_64-unknown-linux-gnu]
+cc = "x86_64-linux-gnu-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
+[target.x86_64-unknown-linux-musl]
+musl-root = "/usr/x86_64-linux-musl/lib/musl"
+cc = "x86_64-linux-musl-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
 [target.wasm32-unknown-unknown]
 cc = "clang"
 cxx = "clang++"
--- bootstrap.riscv64.toml
+++ bootstrap.riscv64.toml
@@ -18,6 +18,8 @@
   "riscv64gc-unknown-linux-musl",
   "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
+  "x86_64-unknown-linux-gnu",
+  "x86_64-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32v1-none",
   "wasm32-wasip1",
@@ -107,6 +109,25 @@
 default-linker = "aarch64-linux-gnu-gcc"
 sanitizers = false
 
+[target.x86_64-unknown-linux-gnu]
+cc = "x86_64-linux-gnu-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
+[target.x86_64-unknown-linux-musl]
+musl-root = "/usr/x86_64-linux-musl/lib/musl"
+cc = "x86_64-linux-musl-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
 [target.wasm32-unknown-unknown]
 cc = "clang"
 cxx = "clang++"
