diff --git PKGBUILD PKGBUILD
index ff0256c3..534fc8a3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -5,7 +5,7 @@
 # Contributor: userwithuid <userwithuid@gmail.com>
 
 pkgbase=rust
-pkgname=(rust lib32-rust-libs rust-musl rust-wasm rust-src)
+pkgname=(rust rust-musl rust-wasm rust-src)
 epoch=1
 pkgver=1.66.1
 pkgrel=2
@@ -15,7 +15,7 @@ arch=(x86_64)
 license=(MIT Apache)
 options=(!emptydirs !strip !lto)
 depends=(gcc-libs llvm-libs curl libssh2 gcc)
-makedepends=(rust llvm libffi lib32-gcc-libs perl python cmake musl ninja
+makedepends=(rust llvm libffi perl python cmake musl ninja
              wasi-libc lld)
 checkdepends=(procps-ng gdb)
 source=(
@@ -24,13 +24,17 @@ source=(
   0001-bootstrap-Change-libexec-dir.patch
   0002-compiler-Change-LLVM-targets.patch
   0003-compiler-Use-wasm-ld-for-wasm-targets.patch
+  "cc-rs-pick.patch"::"https://github.com/rust-lang/cc-rs/pull/676.patch"
+  "jemalloc-sys-pick.patch"
 )
 sha256sums=('5b3c933a94c72187705d4ee293198babfdd09442f5937fbd685db3a81f4959ba'
             'SKIP'
             'c901a9bb036c29ca092f7dbc8b447efdd9aa1044a902a88f9d024cb22681dc19'
             '874c96cb43a5cc1170674771705deb1c72e95ec5f6ede860cd658111db3cda48'
             '4a4ff5c294acc0b1ca800856218770f6330a878acee72193aa456d82846a8a28'
-            'c9abbbc8e3a97b0e43fa5f948ae4df51320d9cc5243b5d0c575470271932f783')
+            'c9abbbc8e3a97b0e43fa5f948ae4df51320d9cc5243b5d0c575470271932f783'
+            '03368e7bf566fb5e658239489786e794fa2945a30ce69b7def41552e1f1bfa6a'
+            'dda34cb5300511aa0aff6bc39bdbb17215ea60bee81bc52f64e3474d6cbda516')
 validpgpkeys=(108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE  # Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>
               474E22316ABF4785A88C6E8EA2C794A986419D8A) # Tom Stellard <tstellar@redhat.com>
 
@@ -48,6 +52,18 @@ prepare() {
   # Use our wasm-ld
   patch -Np1 -i ../0003-compiler-Use-wasm-ld-for-wasm-targets.patch
 
+  cd vendor/cc
+  OLDSUM=`sha256sum src/lib.rs | head -c 64`
+  patch -Np1 -i ../../../cc-rs-pick.patch
+  sed -i "s/$OLDSUM/`sha256sum src/lib.rs | head -c 64`/g" .cargo-checksum.json
+  cd ../..
+
+  cd vendor/jemalloc-sys
+  OLDSUM=`sha256sum build.rs | head -c 64`
+  patch -Np1 -i ../../../jemalloc-sys-pick.patch
+  sed -i "s/$OLDSUM/`sha256sum build.rs | head -c 64`/g" .cargo-checksum.json
+  cd ../..
+
   cat >config.toml <<END
 changelog-seen = 2
 profile = "user"
@@ -57,9 +73,8 @@ link-shared = true
 
 [build]
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
+  "riscv64gc-unknown-linux-gnu",
+  "riscv64gc-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32-wasi",
 ]
@@ -99,22 +114,18 @@ deny-warnings = false
 [dist]
 compression-formats = ["gz"]
 
-[target.x86_64-unknown-linux-gnu]
+[target.riscv64gc-unknown-linux-gnu]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
 llvm-config = "/usr/bin/llvm-config"
 
-[target.i686-unknown-linux-gnu]
-cc = "/usr/bin/gcc"
-cxx = "/usr/bin/g++"
-ar = "/usr/bin/gcc-ar"
-ranlib = "/usr/bin/gcc-ranlib"
-
-[target.x86_64-unknown-linux-musl]
+[target.riscv64gc-unknown-linux-musl]
 sanitizers = false
 musl-root = "/usr/lib/musl"
+cc = "/usr/bin/musl-gcc"
+cxx = "/usr/bin/musl-gcc"
 
 [target.wasm32-unknown-unknown]
 sanitizers = false
@@ -152,9 +163,7 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -p usr/lib32
-  ln -srft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srft usr/lib   usr/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/*.so
 
   mkdir -p usr/share/bash-completion
   mv etc/bash_completion.d usr/share/bash-completion/completions
@@ -162,8 +171,7 @@ build() {
   mkdir -p usr/share/licenses/rust
   mv -t usr/share/licenses/rust usr/share/doc/rust/{COPYRIGHT,LICENSE*}
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
+  _pick dest-musl usr/lib/rustlib/riscv64gc-unknown-linux-musl
   _pick dest-wasm usr/lib/rustlib/wasm32-*
   _pick dest-src  usr/lib/rustlib/src
 }
@@ -178,19 +186,6 @@ package_rust() {
   cp -a dest-rust/* "$pkgdir"
 }
 
-package_lib32-rust-libs() {
-  pkgdesc="32-bit target and libraries for Rust"
-  depends=(rust lib32-gcc-libs)
-  provides=(lib32-rust)
-  conflicts=(lib32-rust)
-  replaces=(lib32-rust)
-
-  cp -a dest-i686/* "$pkgdir"
-
-  mkdir -p "$pkgdir/usr/share/licenses"
-  ln -s rust "$pkgdir/usr/share/licenses/$pkgname"
-}
-
 package_rust-musl() {
   pkgdesc="Musl target for Rust"
   depends=(rust)
