--- PKGBUILD
+++ PKGBUILD
@@ -185,6 +185,10 @@
   export RUST_BACKTRACE=1
   unset CFLAGS CXXFLAGS LDFLAGS
 
+  if [[ $CARCH = riscv64 ]]; then
+    export RUSTFLAGS+=" -C link-arg=-fuse-ld=lld"
+  fi
+
   local xpy_options=(
     -j "$(nproc)"
   )
@@ -280,6 +284,8 @@
     loong64|riscv64)
       _pick dest-aarch64-gnu usr/lib/rustlib/aarch64-unknown-linux-gnu
       _pick dest-aarch64-musl usr/lib/rustlib/aarch64-unknown-linux-musl
+      _pick dest-x86_64-gnu usr/lib/rustlib/x86_64-unknown-linux-gnu
+      _pick dest-x86_64-musl usr/lib/rustlib/x86_64-unknown-linux-musl
       ;;
   esac
 }
@@ -401,4 +407,44 @@
   _install_licenses
 }
 
+package_rust-x86_64-gnu() {
+  pkgdesc="x86_64 GNU target for Rust"
+  arch=(
+    loong64
+    riscv64
+  )
+  depends=(
+    x86_64-linux-gnu-gcc
+    x86_64-linux-gnu-glibc
+    rust
+  )
+
+  cp -a dest-x86_64-gnu/* "$pkgdir"
+  _install_licenses
+}
+
+package_rust-x86_64-musl() {
+  pkgdesc="x86_64 Musl target for Rust"
+  arch=(
+    loong64
+    riscv64
+  )
+  depends=(
+    x86_64-linux-gnu-gcc
+    rust
+  )
+
+  cp -a dest-x86_64-musl/* "$pkgdir"
+  _install_licenses
+}
+
+pkgname+=(rust-x86_64-gnu rust-x86_64-musl)
+makedepends_loong64+=(x86_64-linux-gnu-gcc x86_64-linux-gnu-glibc)
+makedepends_riscv64+=(libatomic x86_64-linux-gnu-gcc x86_64-linux-gnu-glibc)
+b2sums_loong64[0]='bd4d529837c1bf57131a68a54cdf448a0ef5fd79e7cd30ae3234d15f3ca47d6bf4f9d0a2feaede2a50523e888605cebd23be3783bba04b5abd143bc28d2b6179'
+b2sums_riscv64[0]='ff8ec67496b078e99b40c1fa62c807b2c535d931214e544e0a2047a0a0f0cef19d8d441aa30d17c6e8dac54068fe7228037494455a2be5c2f9070d2306d3b590'
+if [[ $CARCH = riscv64 ]]; then
+  makedepends=(${makedepends[@]/rust})
+fi
+
 # vim:set ts=2 sw=2 et:
--- bootstrap.loong64.toml
+++ bootstrap.loong64.toml
@@ -18,6 +18,8 @@
   "loongarch64-unknown-linux-musl",
   "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
+  "x86_64-unknown-linux-gnu",
+  "x86_64-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32v1-none",
   "wasm32-wasip1",
@@ -93,6 +95,25 @@
 ranlib = "aarch64-linux-gnu-gcc-ranlib"
 sanitizers = false
 
+[target.x86_64-unknown-linux-gnu]
+cc = "x86_64-linux-gnu-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
+[target.x86_64-unknown-linux-musl]
+musl-root = "/usr/x86_64-linux-musl/lib/musl"
+cc = "x86_64-linux-musl-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
 [target.wasm32-unknown-unknown]
 cc = "clang"
 cxx = "clang++"
--- bootstrap.riscv64.toml
+++ bootstrap.riscv64.toml
@@ -18,15 +18,14 @@
   "riscv64gc-unknown-linux-musl",
   "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
+  "x86_64-unknown-linux-gnu",
+  "x86_64-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32v1-none",
   "wasm32-wasip1",
   "wasm32-wasip1-threads",
   "wasm32-wasip2",
 ]
-cargo = "/usr/bin/cargo"
-rustc = "/usr/bin/rustc"
-rustfmt = "/usr/bin/rustfmt"
 locked-deps = true
 vendor = true
 tools = [
@@ -93,6 +92,25 @@
 ranlib = "aarch64-linux-gnu-gcc-ranlib"
 sanitizers = false
 
+[target.x86_64-unknown-linux-gnu]
+cc = "x86_64-linux-gnu-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
+[target.x86_64-unknown-linux-musl]
+musl-root = "/usr/x86_64-linux-musl/lib/musl"
+cc = "x86_64-linux-musl-gcc"
+cxx = "x86_64-linux-gnu-g++"
+ar = "x86_64-linux-gnu-gcc-ar"
+ranlib = "x86_64-linux-gnu-ranlib"
+linker = "x86_64-linux-gnu-gcc"
+default-linker = "x86_64-linux-gnu-gcc"
+sanitizers = false
+
 [target.wasm32-unknown-unknown]
 cc = "clang"
 cxx = "clang++"
