diff --git PKGBUILD PKGBUILD
index 33307ea..9a63d1a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -5,7 +5,7 @@
 # Contributor: userwithuid <userwithuid@gmail.com>
 
 pkgbase=rust
-pkgname=(rust lib32-rust-libs rust-musl rust-wasm rust-src)
+pkgname=(rust rust-musl rust-wasm rust-src)
 epoch=1
 pkgver=1.63.0
 pkgrel=1
@@ -15,7 +15,7 @@ arch=(x86_64)
 license=(MIT Apache)
 options=(!emptydirs !strip !lto)
 depends=(gcc-libs llvm-libs curl libssh2 gcc)
-makedepends=(rust llvm libffi lib32-gcc-libs perl python cmake musl ninja
+makedepends=(rust llvm libffi perl python cmake musl ninja
              wasi-libc lld)
 checkdepends=(procps-ng gdb)
 source=(
@@ -25,6 +25,7 @@ source=(
   0002-compiler-Change-LLVM-targets.patch
   0003-compiler-Use-wasm-ld-for-wasm-targets.patch
   remove-include-linux-fs.h.patch
+  replace_patched_libc_cc.patch
 )
 sha256sums=('1f9580295642ef5da7e475a8da2397d65153d3f2cb92849dbd08ed0effca99d0'
             'SKIP'
@@ -32,7 +33,8 @@ sha256sums=('1f9580295642ef5da7e475a8da2397d65153d3f2cb92849dbd08ed0effca99d0'
             '7080a88e44234aa363ba6321f319c8d9621382dd9581748a43847dae005e55bd'
             '5137df0a3399e279db4c5b83ae8e99d2c1977f62aec2f90521edb02f51597064'
             '26bf232d144b81a303ed91f6a331ddf004c5d9fefd41a11e36b61812a34459aa'
-            '34ed866e313e4580130a50118a4410d36fa0159123982521b6ef049439fc32ad')
+            '34ed866e313e4580130a50118a4410d36fa0159123982521b6ef049439fc32ad'
+            '908de2d6d84358289ff2667aca3360a3bc91cbe8d2687ee5b02474088282ff37')
 validpgpkeys=(108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE  # Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>
               474E22316ABF4785A88C6E8EA2C794A986419D8A) # Tom Stellard <tstellar@redhat.com>
 
@@ -53,6 +55,8 @@ prepare() {
   # https://github.com/llvm/llvm-project/issues/56421
   patch -Np1 -d src/llvm-project < ../remove-include-linux-fs.h.patch
 
+  patch -Np1 -i "../replace_patched_libc_cc.patch"
+
   cat >config.toml <<END
 changelog-seen = 2
 profile = "user"
@@ -62,9 +66,8 @@ link-shared = true
 
 [build]
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
+  "riscv64gc-unknown-linux-gnu",
+  "riscv64gc-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32-wasi",
 ]
@@ -90,7 +93,6 @@ description = "Arch Linux $pkgbase $epoch:$pkgver-$pkgrel"
 rpath = false
 backtrace-on-ice = true
 remap-debuginfo = true
-jemalloc = true
 
 # LLVM crashes when passing an object through ThinLTO twice.  This is triggered
 # when using rust code in cross-language LTO if libstd was built using ThinLTO.
@@ -104,12 +106,14 @@ deny-warnings = false
 [dist]
 compression-formats = ["gz"]
 
-[target.x86_64-unknown-linux-gnu]
+[target.riscv64gc-unknown-linux-gnu]
 llvm-config = "/usr/bin/llvm-config"
 
-[target.x86_64-unknown-linux-musl]
+[target.riscv64gc-unknown-linux-musl]
 sanitizers = false
 musl-root = "/usr/lib/musl"
+cc = "/usr/bin/musl-gcc"
+cxx = "/usr/bin/musl-gcc"
 
 [target.wasm32-unknown-unknown]
 sanitizers = false
@@ -120,6 +124,8 @@ sanitizers = false
 profiler = false
 wasi-root = "/usr/share/wasi-sysroot"
 END
+
+  cargo update -p bootstrap
 }
 
 _pick() {
@@ -147,9 +153,7 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -p usr/lib32
-  ln -srft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srft usr/lib   usr/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/*.so
 
   mkdir -p usr/share/bash-completion
   mv etc/bash_completion.d usr/share/bash-completion/completions
@@ -157,8 +161,7 @@ build() {
   mkdir -p usr/share/licenses/rust
   mv -t usr/share/licenses/rust usr/share/doc/rust/{COPYRIGHT,LICENSE*}
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
+  _pick dest-musl usr/lib/rustlib/riscv64gc-unknown-linux-musl
   _pick dest-wasm usr/lib/rustlib/wasm32-*
   _pick dest-src  usr/lib/rustlib/src
 }
@@ -173,19 +176,6 @@ package_rust() {
   cp -a dest-rust/* "$pkgdir"
 }
 
-package_lib32-rust-libs() {
-  pkgdesc="32-bit target and libraries for Rust"
-  depends=(rust lib32-gcc-libs)
-  provides=(lib32-rust)
-  conflicts=(lib32-rust)
-  replaces=(lib32-rust)
-
-  cp -a dest-i686/* "$pkgdir"
-
-  mkdir -p "$pkgdir/usr/share/licenses"
-  ln -s rust "$pkgdir/usr/share/licenses/$pkgname"
-}
-
 package_rust-musl() {
   pkgdesc="Musl target for Rust"
   depends=(rust)
