diff --git PKGBUILD PKGBUILD
index 522e5124..831c4254 100644
--- PKGBUILD
+++ PKGBUILD
@@ -4,7 +4,7 @@
 # Contributor: Daniel Micay <danielmicay@gmail.com>
 # Contributor: userwithuid <userwithuid@gmail.com>
 
-pkgname=('rust' 'lib32-rust-libs' 'rust-musl' 'rust-docs')
+pkgname=('rust' 'rust-docs')
 epoch=1
 pkgver=1.56.0
 pkgrel=3
@@ -16,8 +16,8 @@ url='https://www.rust-lang.org/'
 arch=('x86_64')
 license=('MIT' 'Apache')
 
-makedepends=('rust' "llvm=$_llvm_ver" 'libffi' 'lib32-gcc-libs' 'perl' 'python'
-             'curl' 'cmake' 'musl' 'ninja')
+makedepends=('rust' "llvm=$_llvm_ver" 'llvm12-libs' 'libffi' 'perl' 'python'
+             'curl' 'cmake' 'ninja')
 checkdepends=('procps-ng' 'gdb')
 
 options=('!emptydirs' '!strip')
@@ -59,7 +59,7 @@ profile = "user"
 link-shared = true
 
 [build]
-target = ["x86_64-unknown-linux-gnu", "i686-unknown-linux-gnu", "x86_64-unknown-linux-musl"]
+target = ["riscv64gc-unknown-linux-gnu"]
 cargo = "/usr/bin/cargo"
 rustc = "/usr/bin/rustc"
 rustfmt = "/usr/bin/rustfmt"
@@ -82,12 +82,8 @@ debuginfo-level-std = 2
 channel = "stable"
 rpath = false
 
-[target.x86_64-unknown-linux-gnu]
+[target.riscv64gc-unknown-linux-gnu]
 llvm-config = "/usr/bin/llvm-config"
-
-[target.x86_64-unknown-linux-musl]
-sanitizers = false
-musl-root = "/usr/lib/musl"
 END
 }
 
@@ -98,7 +94,7 @@ build() {
   export RUST_COMPILER_RT_ROOT="$srcdir/compiler-rt-$_llvm_ver.src"
   [[ -d $RUST_COMPILER_RT_ROOT ]]
 
-  DESTDIR="$PWD"/dest-rust python ./x.py install -j "$(nproc)"
+  DESTDIR="$PWD"/dest-rust python ./x.py install
 
   # Remove analysis data for libs that weren't installed
   # TODO: Find out where these come from
@@ -113,8 +109,6 @@ build() {
   done < <(find "dest-rust/usr/lib/rustlib"  -path '*/analysis/*.json' -print0)
 
   # move docs and cross targets out of the way for splitting
-  mv dest-rust/usr/lib/rustlib/i686-unknown-linux-gnu dest-i686
-  mv dest-rust/usr/lib/rustlib/x86_64-unknown-linux-musl dest-musl
   mv dest-rust/usr/share/doc dest-doc
 }
 
@@ -137,37 +131,12 @@ package_rust() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  ln -srft "$pkgdir"/usr/lib x86_64-unknown-linux-gnu/lib/*.so
+  ln -srft "$pkgdir"/usr/lib riscv64gc-unknown-linux-gnu/lib/*.so
 
   install -d "$pkgdir"/usr/share/bash-completion/
   mv "$pkgdir"/etc/bash_completion.d "$pkgdir"/usr/share/bash-completion/completions
 }
 
-package_lib32-rust-libs() {
-  pkgdesc='32-bit target and libraries for Rust'
-  depends=('lib32-gcc-libs')
-  provides=('lib32-rust')
-  conflicts=('lib32-rust')
-  replaces=('lib32-rust')
-
-  cd "rustc-$pkgver-src"
-  install -Dm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE*
-
-  install -d "$pkgdir"/usr/lib/rustlib/ "$pkgdir"/usr/lib32/
-  cp -a dest-i686 "$pkgdir"/usr/lib/rustlib/i686-unknown-linux-gnu
-  ln -srft "$pkgdir"/usr/lib32 "$pkgdir"/usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
-}
-
-package_rust-musl() {
-  pkgdesc='Musl target for Rust'
-
-  cd "rustc-$pkgver-src"
-  install -Dm644 -t "$pkgdir/usr/share/licenses/$pkgname" LICENSE*
-
-  install -d "$pkgdir"/usr/lib/rustlib/
-  cp -a dest-musl "$pkgdir"/usr/lib/rustlib/x86_64-unknown-linux-musl
-}
-
 package_rust-docs() {
   pkgdesc='Documentation for the Rust programming language'
 
