diff --git PKGBUILD PKGBUILD
index 0d6144b..18bf2ac 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,6 @@
 pkgbase=rust
 pkgname=(
   rust
-  lib32-rust-libs
   rust-musl
   rust-wasm
   rust-src
@@ -37,7 +36,6 @@ depends=(
 )
 makedepends=(
   cmake
-  lib32-gcc-libs
   libffi
   lld
   llvm
@@ -58,13 +56,15 @@ source=(
   0001-bootstrap-Change-libexec-dir.patch
   0002-compiler-Change-LLVM-targets.patch
   0003-compiler-Use-wasm-ld-for-wasm-targets.patch
+  bump-bootstrap-cc.patch
 )
 b2sums=('5b1fd44eaa10bf2ab76f64a3bb5c20ee4008031d05a49b0a32b3a9cb2e97d5684cf0c9a6b6883089afca6cab7a37a0a9cfa06d928ed3e6a3b6484b605e18a0f3'
         'SKIP'
         '6de8373ee24ad9a5ae12b44a1288d474c8b8a1461ba54ccadbecefc87d034b6ef763b2909e2f1452fca30488504cabacad972c6ae78b240f74e47c0e01f25136'
         '0b1c8f41144b2c9fc6528c67e6ac5ac0f4ccbbdddc3fd3ede0b83033f1745efc603c6dbc90eb64d0518832d8daf3f82e60bbcd042ff94effab1b5a499758dace'
         '972b67b9ed47b9ff0d9f5156232ed63103a9d40617325ab99be000753c420a228a89a1ca098fa978fef14dc7eb11bb222b85d63a5d2aec444b8ebfdfca07fc67'
-        '445802d26028848549781b9be7430f2f1cedcd0d1f960c61dbce870a66a867aff3b0c9905b4f81b6cefefd74d83a868742ec735dd9046943068ac98117d22987')
+        '445802d26028848549781b9be7430f2f1cedcd0d1f960c61dbce870a66a867aff3b0c9905b4f81b6cefefd74d83a868742ec735dd9046943068ac98117d22987'
+        '2a92f8e5190efd4dca0a36282259caeb7cdd5dcf408c6b90276e6b22cd7a5fce4d7a405e3ca024f84742671e0f98b2ba62b4dc6722bffaa20a76ec99d5d313c1')
 validpgpkeys=(
   108F66205EAEB0AAA8DD5E1C85AB96E6FA1BE5FE  # Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>
   474E22316ABF4785A88C6E8EA2C794A986419D8A  # Tom Stellard <tstellar@redhat.com>
@@ -84,6 +84,9 @@ prepare() {
   # Use our wasm-ld
   patch -Np1 -i ../0003-compiler-Use-wasm-ld-for-wasm-targets.patch
 
+  # Bump bootstrap cc to v1.0.77
+  patch -Np1 -i ../bump-bootstrap-cc.patch
+
   cat >config.toml <<END
 profile = "user"
 changelog-seen = 2
@@ -93,9 +96,8 @@ link-shared = true
 
 [build]
 target = [
-  "x86_64-unknown-linux-gnu",
-  "i686-unknown-linux-gnu",
-  "x86_64-unknown-linux-musl",
+  "riscv64gc-unknown-linux-gnu",
+  "riscv64gc-unknown-linux-musl",
   "wasm32-unknown-unknown",
   "wasm32-wasi",
 ]
@@ -144,22 +146,18 @@ deny-warnings = false
 [dist]
 compression-formats = ["gz"]
 
-[target.x86_64-unknown-linux-gnu]
+[target.riscv64gc-unknown-linux-gnu]
 cc = "/usr/bin/gcc"
 cxx = "/usr/bin/g++"
 ar = "/usr/bin/gcc-ar"
 ranlib = "/usr/bin/gcc-ranlib"
 llvm-config = "/usr/bin/llvm-config"
 
-[target.i686-unknown-linux-gnu]
-cc = "/usr/bin/gcc"
-cxx = "/usr/bin/g++"
-ar = "/usr/bin/gcc-ar"
-ranlib = "/usr/bin/gcc-ranlib"
-
-[target.x86_64-unknown-linux-musl]
+[target.riscv64gc-unknown-linux-musl]
 sanitizers = false
 musl-root = "/usr/lib/musl"
+cc = "/usr/bin/musl-gcc"
+cxx = "/usr/bin/musl-gcc"
 
 [target.wasm32-unknown-unknown]
 sanitizers = false
@@ -197,9 +195,7 @@ build() {
 
   # rustbuild always installs copies of the shared libraries to /usr/lib,
   # overwrite them with symlinks to the per-architecture versions
-  mkdir -p usr/lib32
-  ln -srft usr/lib   usr/lib/rustlib/x86_64-unknown-linux-gnu/lib/*.so
-  ln -srft usr/lib32 usr/lib/rustlib/i686-unknown-linux-gnu/lib/*.so
+  ln -srft usr/lib   usr/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/*.so
 
   mkdir -p usr/share/bash-completion
   mv etc/bash_completion.d usr/share/bash-completion/completions
@@ -207,8 +203,7 @@ build() {
   mkdir -p usr/share/licenses/rust
   mv -t usr/share/licenses/rust usr/share/doc/rust/{COPYRIGHT,LICENSE*}
 
-  _pick dest-i686 usr/lib/rustlib/i686-unknown-linux-gnu usr/lib32
-  _pick dest-musl usr/lib/rustlib/x86_64-unknown-linux-musl
+  _pick dest-musl usr/lib/rustlib/riscv64gc-unknown-linux-musl
   _pick dest-wasm usr/lib/rustlib/wasm32-*
   _pick dest-src  usr/lib/rustlib/src
 }
@@ -237,22 +232,6 @@ package_rust() {
   cp -a dest-rust/* "$pkgdir"
 }
 
-package_lib32-rust-libs() {
-  pkgdesc="32-bit target and libraries for Rust"
-  depends=(
-    lib32-gcc-libs
-    rust
-  )
-  provides=(lib32-rust)
-  conflicts=(lib32-rust)
-  replaces=(lib32-rust)
-
-  cp -a dest-i686/* "$pkgdir"
-
-  mkdir -p "$pkgdir/usr/share/licenses"
-  ln -s rust "$pkgdir/usr/share/licenses/$pkgname"
-}
-
 package_rust-musl() {
   pkgdesc="Musl target for Rust"
   depends=(rust)
