--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,11 @@ depends=(gcc-libs sh)
 source=(https://download.oracle.com/berkeley-db/db-${pkgver}.tar.gz)
 b2sums=('d3254fba1300d6c7dda5d872c5fa349e8704d71be9c4d37ccf3357782adeb879ce30e066cf083bd97554405c2e94071a96cee16090ad9f797a4fa657ff7e8c46')
 
+prepare() {
+  cd ${pkgname}-${pkgver}/dist
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} .
+}
+
 build() {
   cd ${pkgname}-${pkgver}/build_unix
   ../dist/configure \
