--- PKGBUILD	(revision 460901)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 url="https://www.abisource.com/projects/link-grammar/"
 license=('LGPL')
 depends=('hunspell' 'sqlite' 'libedit')
-makedepends=('python' 'swig' 'apache-ant' 'java-environment=11')
+makedepends=('python' 'swig' 'apache-ant' 'java-environment=11' 'autoconf-archive')
 options=('!makeflags')
 source=(https://www.abisource.com/downloads/${pkgname}/${pkgver}/${pkgname}-${pkgver}.tar.gz{,.asc})
 validpgpkeys=('6407453C98BECC19ADB03D82EB6AA534E0C0651C'
@@ -17,6 +17,13 @@
 sha256sums=('bdb9a359f877ff95d60f44d1780387324fa3763de5084ba1817dbf561a0ebed4'
             'SKIP')
 
+prepare(){
+  cd "${srcdir}/${pkgname}-${pkgver}"
+
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd ${pkgname}-${pkgver}
 
