diff --git PKGBUILD PKGBUILD
index 6f45b42d..a8a34132 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@
 license=('GPL3')
 url="https://bogofilter.sourceforge.io/"
 depends=('db' 'perl' 'gsl') # 'sqlite'
-makedepends=('valgrind' 'xmlto')
+makedepends=('xmlto')
 backup=('etc/bogofilter/bogofilter.cf')
 source=(https://downloads.sourceforge.net/project/${pkgname}/${pkgname}-stable/${pkgname}-${pkgver}.tar.xz{,.asc}) # asc file doesn't  work with pacman
 sha1sums=('c779c3afb3e57ae0208ee503a854aff716b0cafd'
@@ -44,7 +44,7 @@
 
 check() {
   cd "$pkgname"-$pkgver
-  BF_RUN_VALGRIND=1 make -k -C src check || /bin/true
+  BF_RUN_VALGRIND=0 make -k -C src check || /bin/true
 }
 
 package() {
