diff --git PKGBUILD PKGBUILD
index 2602c6c..4d1e164 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,16 +11,18 @@ license=('MIT')
 options=('!lto')
 depends=('clang' 'lld' 'llvm-libs')
 makedepends=('cmake' 'llvm')
-checkdepends=('lib32-glibc')
 source=("https://ziglang.org/download/$pkgver/zig-$pkgver.tar.xz"
-        "skip-localhost-test.patch")
+        "skip-localhost-test.patch"
+        "disable-macho-link-tests.patch")
 sha256sums=('ead029cfe474d9bf0413332d0e9d3bdfb5990cadce238f44f35ba32d92169295'
-            'eeb5f0f72035c52bf558ffc77a171a3ddf93eac7d663ef0c82826007763717a8')
+            'eeb5f0f72035c52bf558ffc77a171a3ddf93eac7d663ef0c82826007763717a8'
+            'e70f2a61c2543414d7f7a0fbcec4fc88fcca0b2f80368c4a6e6ad9ea76ed917d')
 
 prepare() {
     cd "$pkgname-$pkgver"
 
     patch -p1 -i ../skip-localhost-test.patch
+    patch -p1 -i ../disable-macho-link-tests.patch
 }
 
 build() {
