Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1142351)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,6 @@
 license=('MIT')
 depends=('clang' 'llvm-libs' 'lld')
 makedepends=('cmake' 'llvm')
-checkdepends=('lib32-glibc')
 source=("https://ziglang.org/download/$pkgver/zig-$pkgver.tar.xz"
         "resolve_DNS.patch")
 sha256sums=('cd1be83b12f8269cc5965e59877b49fdd8fa638efb6995ac61eb4cea36a2e381'
