--- PKGBUILD
+++ PKGBUILD
@@ -15,10 +15,12 @@ replaces=('gimp-ufraw')
 optdepends=('gimp: to use the gimp import plugin for raw images')
 source=("https://downloads.sourceforge.net/nufraw/nufraw-${_realver}.tar.gz"
         '0001-nufraw-glib-2.70.patch'
-        '0002-exiv2-error.patch')
+        '0002-exiv2-error.patch'
+        'use-signed-char.patch')
 sha256sums=('2c7dcc1590993ec267b3112562b2433010d6f94f6acc6506c76aa29cf8c6c32c'
             '042aa55a757c4ed1198fe4628c87933f958d768c2cd5270db1b2e4369daaf090'
-            '01fd29b180662da1a1510236bc64fac614229adaf1bc7b11df8b88745d6458da')
+            '01fd29b180662da1a1510236bc64fac614229adaf1bc7b11df8b88745d6458da'
+            '12b6856d70b8e8daa01954992765f7e37103b1cb780d3d6475b3caa5515a3731')
 
 prepare() {
   cd "nufraw-${_realver}"
@@ -29,6 +31,9 @@ prepare() {
   # fix build with exiv2 0.28.0
   patch -p1 < ../0002-exiv2-error.patch
 
+  # fix char signedness
+  patch -p1 < ../use-signed-char.patch
+
   autoreconf -fi
 
   mv nufraw.desktop{.desktop,}
