diff --git PKGBUILD PKGBUILD
index 8024f875..d046a25f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,12 +14,15 @@ conflicts=('gimp-ufraw')
 replaces=('gimp-ufraw')
 optdepends=('gimp: to use the gimp import plugin for raw images')
 source=("https://downloads.sourceforge.net/nufraw/nufraw-${_realver}.tar.gz"
-         nufraw-glib-2.70.patch)
+         nufraw-glib-2.70.patch
+        "fix_narrowing_conversion.patch")
 sha256sums=('2c7dcc1590993ec267b3112562b2433010d6f94f6acc6506c76aa29cf8c6c32c'
-            '0e422f77b277f3c38b454426ee155d1004e47ced8d5d05942448584557c9ed07')
+            '0e422f77b277f3c38b454426ee155d1004e47ced8d5d05942448584557c9ed07'
+            '7659ee99523618502da116796620b1f1a142d7dbe980796fef5c8dc1f8eff284')
 
 prepare() {
   cd "nufraw-${_realver}"
+  patch -Np1 -i "../fix_narrowing_conversion.patch"
 
   patch -p1 < ../nufraw-glib-2.70.patch # Fix build with glib 2.70
 
