--- PKGBUILD	(revision 439846)
+++ PKGBUILD	(working copy)
@@ -19,8 +19,10 @@
              'xorgproto' 'imake'
 )
 source=(https://github.com/ArcticaProject/nx-libs/archive/$pkgver/nx-libs-$pkgver.tar.gz
+        1023.patch
         0001-nx-libs-ar.patch)
 sha256sums=('3ce7ca4e6b57b3a2d7588b2d0f4009036d2566a8925ca2c62f08a8dc0df50357'
+            'b8ba8872a6647a73f84bebaa44b7f5a9b7f9b798ad05ac0966fb54f62f1cd275'
             '0a93a18591376cae1e4f0e9358c14cb905e5da9a7ec0c68ef4e2ad2c3741c306')
 validpgpkeys=('1AD23D1B8F087A35AB74BDE9F4A7678C9C6B0B2B' # X2go Git Administrator <git-admin@x2go.org>
               '9BFBAEE86C0AA5FFBF2207829AF46B3025771B31') # Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
@@ -29,6 +31,8 @@
   cd "${srcdir}/nx-libs-$pkgver"
   # binutils 2.36 buildfix 
   patch -Np1 -i ../0001-nx-libs-ar.patch
+  # ref to https://github.com/ArcticaProject/nx-libs/pull/1023
+  patch -Np1 -i ../1023.patch
 }
 
 build() {
