diff --git PKGBUILD PKGBUILD
index 561d8b8..e37ea14 100644
--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,8 @@ b2sums=('cd594be73fcf632544195d09518901b1055ae86dcf463a5d446a83beba66073c70a9dfb
 
 prepare() {
   cd "${pkgname}-${pkgver}"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
