diff --git PKGBUILD PKGBUILD
index cf62a933..1f1399ee 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,9 @@ b2sums=('40d2aaccbac02f37c107ac2a2b1f33cccdb9a4e80f03bb8f5f023a1748a3835ffbcf0a6
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
