diff --unified --recursive --text a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-09-15 05:54:24.794436886 +0000
+++ b/CMakeLists.txt	2021-09-15 05:59:42.637967929 +0000
@@ -91,10 +91,6 @@
 # enable warnings
 bb_enable_warnings( msvc warnings-as-errors "/wd4996" )
 
-# enable sse4.1 build for all source files for gcc and clang
-if( UNIX OR MINGW )
-  add_compile_options( "-msse4.1" )
-endif()
 
 # enable parallel build for Visual Studio
 if( MSVC )
