--- PKGBUILD
+++ PKGBUILD
@@ -12,10 +12,14 @@ makedepends=('cmake' 'lsb-release' 'openmp')
 source=("https://vcgit.hhi.fraunhofer.de/jct-vc/HM/-/archive/HM-${pkgver}/HM-HM-${pkgver}.tar.bz2"
         '010-hm-disable-werror.patch')
 sha256sums=('3cff991240cc51af7855231d4ca6437ecf4453708c005c7eacf5a0d8e19aa506'
-            '26ed4fb40812ecfce04b1301dc159fb2a575ece22a3532fe991f72e0f4e9488a')
+            '26ed4fb40812ecfce04b1301dc159fb2a575ece22a3532fe991f72e0f4e9488a'
+            'e83be15cb69359e7b6282ea2cd0db61635e3e5da1dc1318119451f2493bcc7b5')
+
+source+=('fix-unrecognized-command-line-option-msse4.1.patch')
 
 prepare() {
     patch -d "HM-HM-${pkgver}" -Np1 -i "${srcdir}/010-hm-disable-werror.patch"
+    patch -d "HM-HM-${pkgver}" -Np1 -i "${srcdir}/fix-unrecognized-command-line-option-msse4.1.patch"
     cp -a "HM-HM-${pkgver}"{,-highbit}
 }
 
