diff --git PKGBUILD PKGBUILD
index b42b760..990769c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,15 @@ license=('BSD')
 depends=('gcc-libs')
 makedepends=('cmake' 'lsb-release' 'openmp')
 source=("https://vcgit.hhi.fraunhofer.de/jct-vc/HM/-/archive/HM-${pkgver}/HM-HM-${pkgver}.tar.bz2"
-        '010-hm-disable-werror.patch')
+        '010-hm-disable-werror.patch'
+        'fix-unrecognized-command-line-option-msse4.1.patch')
 sha256sums=('560e750cefe2ad0edae0687735ae3a4d8f34d033809775af70efdbfcf85fcba2'
-            '26ed4fb40812ecfce04b1301dc159fb2a575ece22a3532fe991f72e0f4e9488a')
+            '26ed4fb40812ecfce04b1301dc159fb2a575ece22a3532fe991f72e0f4e9488a'
+            'e83be15cb69359e7b6282ea2cd0db61635e3e5da1dc1318119451f2493bcc7b5')
 
 prepare() {
     patch -d "HM-HM-${pkgver}" -Np1 -i "${srcdir}/010-hm-disable-werror.patch"
+    patch -d "HM-HM-${pkgver}" -Np1 -i "${srcdir}/fix-unrecognized-command-line-option-msse4.1.patch"
     cp -a "HM-HM-${pkgver}"{,-highbit}
 }
 
