Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1405129)
+++ PKGBUILD	(working copy)
@@ -34,6 +34,8 @@
   sed -i 's|png12|png16|g' configure.in
   sed -e '/CFLAGS=""/d' -e '/CXXFLAGS=""/d' -i configure.in
   export CXXFLAGS+=' -std=gnu++98'
+  cp -v "/usr/share/autoconf/build-aux/config.guess" "/usr/share/autoconf/build-aux/config.sub" .
+  cp -v "/usr/share/autoconf/build-aux/config.guess" "/usr/share/autoconf/build-aux/config.sub" curl
   ./autogen.sh
 
   ./configure --prefix=/usr --mandir=/usr/share/man
