diff --git a/common/scripts/setup_env.sh b/common/scripts/setup_env.sh
index 236035c..cd2defc 100755
--- a/common/scripts/setup_env.sh
+++ b/common/scripts/setup_env.sh
@@ -42,6 +42,8 @@ elif [[ ${LOCAL_ARCH} == s390x ]]; then
     export TARGET_ARCH=s390x
 elif [[ ${LOCAL_ARCH} == ppc64le ]]; then
     export TARGET_ARCH=ppc64le
+elif [[ ${LOCAL_ARCH} == riscv64 ]]; then
+    export TARGET_ARCH=riscv64
 else
     echo "This system's architecture, ${LOCAL_ARCH}, isn't supported"
     exit 1
