--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@ b2sums=('0104e6648c482b5229f456d760e455d1de711e560597ae42a73688199261a64da970352
 prepare() {
   cd $pkgname-$pkgver
   GOFLAGS="-mod=readonly" go mod vendor -v
+  patch -Np1 -i "${srcdir}/add_riscv64_support.patch"
 }
 
 build() {
@@ -56,3 +57,7 @@ package() {
   install -vDm644 istioctl.bash "$pkgdir/usr/share/bash-completion/completions/istioctl"
   install -vDm644 _istioctl "$pkgdir/usr/share/zsh/site-functions/_istioctl"
 }
+
+source+=('add_riscv64_support.patch')
+sha512sums+=('7d51979446415a3968ef538669cf563582745960d5b90aabe85226d5fd4b34ebe5c2f31f01126de778e8fca9102e04b5bc8ff9ceec9d3c059b8c51d377170e83')
+b2sums+=('8cfde63e74c40fa6834b59b17c05974013cc2ea8c578c246b265ef696fe76e74a08e9107969677577df816b2c4a0f81a0afe2c9ab4c8b96e652beeee032ed08d')
