diff --git PKGBUILD PKGBUILD
index 4179c7e..de3b088 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,9 +9,17 @@ arch=('x86_64')
 url='https://github.com/istio/istio'
 license=('Apache')
 makedepends=('go')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/istio/istio/archive/$pkgver.tar.gz")
-sha512sums=('0e52fa3d9f33ec4515bd105724fa1478312aaf834ae574e13041b96e6a8f2a5d1b0897ca0c96976f0ef4436e6cf924d257facce70471d71170593bf8f37a0e82')
-b2sums=('fba25cc96dc920ec89d072aa0f1b23e3699cdb10fa9d2b945e41153e5214952d52af3801146734bf9dfcae1d100d5096b81ef866bd1d72b81cba2592ac3e1f91')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/istio/istio/archive/$pkgver.tar.gz"
+        "add_riscv64_support.patch")
+sha512sums=('0e52fa3d9f33ec4515bd105724fa1478312aaf834ae574e13041b96e6a8f2a5d1b0897ca0c96976f0ef4436e6cf924d257facce70471d71170593bf8f37a0e82'
+            '783ab98ca7b8a5f908a615604bfef74a6928ecb70c369a3d869009469c52e205d636f4a4df894c9d32ccb125e331e9dd72abcb8fe721998d407a6de7d9242a3d')
+b2sums=('fba25cc96dc920ec89d072aa0f1b23e3699cdb10fa9d2b945e41153e5214952d52af3801146734bf9dfcae1d100d5096b81ef866bd1d72b81cba2592ac3e1f91'
+        '059ec840175fdee7694fd1f61f4fe7a2c22cb99686ac4055fc8a5cd4d22da20b03b19b99258a4e51a06bfd5c5fc8833a443b8aaffd450ad80f60c64a46abae9e')
+
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i "../add_riscv64_support.patch"
+}
 
 build() {
   cd $pkgname-$pkgver
@@ -22,7 +30,7 @@ build() {
 }
 
 package() {
-  install -Dm 755 "${srcdir}/$pkgname-$pkgver/out/linux_amd64/istioctl" "${pkgdir}/usr/bin/istioctl"
+  install -Dm 755 "${srcdir}/$pkgname-$pkgver/out/linux_riscv64/istioctl" "${pkgdir}/usr/bin/istioctl"
 
   # Populate bash and zsh completions
   install -dm 755 "${pkgdir}/usr/share/bash-completion/completions"
