--- PKGBUILD
+++ PKGBUILD
@@ -75,7 +75,7 @@ package() {
   install -Dm644 ${pkgname}.sysusers "${pkgdir}"/usr/lib/sysusers.d/${pkgname}.conf
 
   # Remove some cruft
-  rm -r "${pkgdir}"/usr/lib/couchdb/erts-13.1.1/{doc,include,lib,man,src}
+  rm -r "${pkgdir}"/usr/lib/couchdb/erts-13.2/{doc,include,lib,man,src}
   rm -rv "${pkgdir}"/usr/lib/couchdb/etc/
   rm -rv "${pkgdir}"/usr/lib/couchdb/lib/couch-${pkgver}/priv/couch_{ejson_compare,js}
 }
