diff --git PKGBUILD PKGBUILD
index e5e7620..ee5efa8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,10 @@ groups=(gnome)
 depends=(sane dconf gtk3 colord-sane libgusb libhandy)
 makedepends=(yelp-tools meson vala git)
 _commit=2d606befc51c19f1d9270915763e94f90dc0d71a  # tags/40.6^0
-source=("git+https://gitlab.gnome.org/GNOME/simple-scan.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/simple-scan.git#commit=$_commit"
+        "fix-meson-i18n_merge_file.patch::https://gitlab.gnome.org/GNOME/simple-scan/-/merge_requests/202.patch")
+sha256sums=('SKIP'
+            'f4aee4e295ae7da2caa1b7ef187b33e3de33d2bd1fb96d7765c605940451b418')
 
 pkgver() {
   cd $pkgname
@@ -24,6 +26,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i $srcdir/fix-meson-i18n_merge_file.patch
 }
 
 build() {
