--- PKGBUILD
+++ PKGBUILD
@@ -184,10 +184,6 @@ build() {
   ## ROOT
   cmake -B build "${common_cmake_options[@]}"
   cmake --build build
-
-  ## ROOT with CUDA
-  cmake -B build-cuda "${common_cmake_options[@]}" "${cuda_cmake_options[@]}"
-  cmake --build build-cuda
 }
 
 _package() {
@@ -303,4 +299,8 @@ package_root-cuda() {
   _package build-cuda
 }
 
+pkgname=(${pkgname[@]/root-cuda})
+makedepends=(${makedepends[@]/cuda})
+makedepends=(${makedepends[@]/cudnn})
+
 # vim: ts=2 sw=2 et:
