diff --git PKGBUILD PKGBUILD
index 9baac04..0685cf3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,7 +30,7 @@ checkdepends=(git
 _archive="$_pkgname-$pkgver"
 source=("$url/archive/$pkgver/$_archive.tar.gz"
         "$pkgname-1.9.0-devendor.patch")
-sha256sums=('54fac6008ceeb06ea137c62b7049d9d141c61741671eaa08b9447cba4ae12977'
+sha256sums=('ca398d26f666e09162847ddd7f8d6ff6b9fb07003c6d43dd68e80d0a78331aa7'
             'b6136544236192c80f3c34f266023fb285b753321679c29e61b69b26913c66d2')
 
 prepare() {
