diff --git PKGBUILD PKGBUILD
index 063b3b2d..a3545f14 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,10 @@ provides=(zeitgeist-datahub)
 conflicts=(zeitgeist-datahub)
 replaces=(zeitgeist-datahub)
 _commit=c9c41bc881ffba1f993e9391c853b1a772d5052b
-source=("git+https://anongit.freedesktop.org/git/zeitgeist/zeitgeist#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://anongit.freedesktop.org/git/zeitgeist/zeitgeist#commit=$_commit"
+        fix_genericarray_api.patch::"https://gitlab.freedesktop.org/zeitgeist/zeitgeist/-/commit/64ac3a6f94cd299e5e14945dc31b48f009dec152.patch")
+sha256sums=('SKIP'
+            '0c2ea6a7b2a614fc25be0ddbd22251cbe36d39b500d6f9f4b285824ca5024a77')
 validpgpkeys=('E4884AEEDE4CC02043C3D8045DECDBA89270E723') #Rico Tzschichholz <ricotz@ubuntu.com>
 
 pkgver() {
@@ -26,6 +28,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "../fix_genericarray_api.patch"
   autoreconf -fi
 }
 
