--- PKGBUILD
+++ PKGBUILD
@@ -9,9 +9,17 @@ url="https://github.com/batchcorp/plumber"
 license=('Apache')
 depends=('glibc')
 makedepends=('go' 'git')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz")
-sha512sums=('4db97f0d971540c0d17b46940b926031198e5ad5363367e5f5c9a38e093abb72c8e41f0bf444d8068d431ac27cbae67aae30cc2ea4e69a9eff1de88cc9c9faf2')
-b2sums=('a64214914fb33f0f763d7a1773ae85f81a159d4afb0b32fc83de2334bf63fe3862124198e84800f7522eae30df329d468e1ab80c070ec0e7d5099231bdac4146')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz"
+        "add_riscv_support.patch")
+sha512sums=('4db97f0d971540c0d17b46940b926031198e5ad5363367e5f5c9a38e093abb72c8e41f0bf444d8068d431ac27cbae67aae30cc2ea4e69a9eff1de88cc9c9faf2'
+            '4fe7a94c6e5740cf5073129a478c8f2d9d1a6be948defb5354affa1fe53bf56629f97e5a73c2bfc78f102bf19402bb3c426f41c58ba41635a003177a784f4854')
+b2sums=('a64214914fb33f0f763d7a1773ae85f81a159d4afb0b32fc83de2334bf63fe3862124198e84800f7522eae30df329d468e1ab80c070ec0e7d5099231bdac4146'
+        '7bef1dd55c324dbfd62407822b529f2b17309c6ef6d97624fb9a6e72a5e0b481965f0c3aecbcf913b63ce584fea83eeea06665b466cbb84544de1a5c5940fca2')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i "../add_riscv_support.patch"
+}
 
 build(){
   cd "${pkgname}-${pkgver}"
@@ -19,7 +27,7 @@ build(){
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=vendor -modcacherw"
   go build -o plumber-bin .
 }
 
