--- PKGBUILD
+++ PKGBUILD
@@ -13,13 +13,18 @@ source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/ar
 sha512sums=('e320e9e0aecf9eaee07e81e99ac263cf5a57fb8e03f0e93d6f2be00d9b60c82f2d4c465c2560e5833e574d78eb52bed1c83938e308539e9ee96e58449479f075')
 b2sums=('c2b78ece85934dbeae321ec7612c06334aecbd2e217b7e14d242c0bebd96142bae500e33556247d2c5fb99916490cedc5b6f512a0b9f80609e4f8b2033075961')
 
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i "../add_riscv_support.patch"
+}
+
 build(){
   cd "${pkgname}-${pkgver}"
   export CGO_CPPFLAGS="${CPPFLAGS}"
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=vendor -modcacherw"
   go build -o plumber-bin .
 }
 
@@ -33,3 +38,6 @@ package() {
   install -Dm755 plumber-bin "${pkgdir}/usr/bin/${pkgname}"
 }
 
+source+=("add_riscv_support.patch")
+sha512sums+=("4fe7a94c6e5740cf5073129a478c8f2d9d1a6be948defb5354affa1fe53bf56629f97e5a73c2bfc78f102bf19402bb3c426f41c58ba41635a003177a784f4854")
+b2sums+=("7bef1dd55c324dbfd62407822b529f2b17309c6ef6d97624fb9a6e72a5e0b481965f0c3aecbcf913b63ce584fea83eeea06665b466cbb84544de1a5c5940fca2")
