diff --git PKGBUILD PKGBUILD
index 626d9fa..4a67f84 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,9 +9,17 @@ url="https://github.com/batchcorp/plumber"
 license=('Apache')
 depends=('glibc')
 makedepends=('go' 'git')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz")
-sha512sums=('2a65beefcb3b01dbb7b9b1ad044627e9084103431966adb344b2276ac97e813a3d9194b52faa1cca24787ed3066877c8cb56e38861e3369be9e0855b198f7cbf')
-b2sums=('517004a40b018e5fd8e37aef7216ac48d2ba58261607ae12b21fce66d153975c0033b44d40e370a5356888e36f9fbda27f7faf0f6fa3f7a9d9dfc59953daeca6')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz"
+        "add_riscv_support.patch")
+sha512sums=('2a65beefcb3b01dbb7b9b1ad044627e9084103431966adb344b2276ac97e813a3d9194b52faa1cca24787ed3066877c8cb56e38861e3369be9e0855b198f7cbf'
+            '4fe7a94c6e5740cf5073129a478c8f2d9d1a6be948defb5354affa1fe53bf56629f97e5a73c2bfc78f102bf19402bb3c426f41c58ba41635a003177a784f4854')
+b2sums=('517004a40b018e5fd8e37aef7216ac48d2ba58261607ae12b21fce66d153975c0033b44d40e370a5356888e36f9fbda27f7faf0f6fa3f7a9d9dfc59953daeca6'
+        '7bef1dd55c324dbfd62407822b529f2b17309c6ef6d97624fb9a6e72a5e0b481965f0c3aecbcf913b63ce584fea83eeea06665b466cbb84544de1a5c5940fca2')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i "../add_riscv_support.patch"
+}
 
 build(){
   cd "${pkgname}-${pkgver}"
@@ -19,7 +27,7 @@ build(){
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=vendor -modcacherw"
   go build -o plumber-bin .
 }
 
