diff --git PKGBUILD PKGBUILD
index ce2cb88e2..e4e44770c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,9 +9,17 @@ url="https://github.com/batchcorp/plumber"
 license=('Apache')
 depends=('glibc')
 makedepends=('go' 'git')
-source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz")
-sha512sums=('4d397cb464e8ff5343ee373e0d6c612b7235645e79002f1bd66c7c081c9757076d65997e417707b59f220ef03274bc3eb8748787389d9740fca4cba2b65c23dd')
-b2sums=('a6526d54f5340fdf650993462bbba7a0f7a5d5f3784cc2da5f92e99d57dd2d65ef402e3dfa018b22df61c73766943ef242771adaf19b7c92ca9d1e32cf56730a')
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/batchcorp/${pkgname}/archive/v${pkgver}.tar.gz"
+        "add_riscv_support.patch")
+sha512sums=('4d397cb464e8ff5343ee373e0d6c612b7235645e79002f1bd66c7c081c9757076d65997e417707b59f220ef03274bc3eb8748787389d9740fca4cba2b65c23dd'
+            '4fe7a94c6e5740cf5073129a478c8f2d9d1a6be948defb5354affa1fe53bf56629f97e5a73c2bfc78f102bf19402bb3c426f41c58ba41635a003177a784f4854')
+b2sums=('a6526d54f5340fdf650993462bbba7a0f7a5d5f3784cc2da5f92e99d57dd2d65ef402e3dfa018b22df61c73766943ef242771adaf19b7c92ca9d1e32cf56730a'
+        '7bef1dd55c324dbfd62407822b529f2b17309c6ef6d97624fb9a6e72a5e0b481965f0c3aecbcf913b63ce584fea83eeea06665b466cbb84544de1a5c5940fca2')
+
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i "../add_riscv_support.patch"
+}
 
 build(){
   cd "${pkgname}-${pkgver}"
@@ -19,7 +27,7 @@ build(){
   export CGO_CFLAGS="${CFLAGS}"
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
-  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=readonly -modcacherw"
+  export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -trimpath -mod=vendor -modcacherw"
   go build -o plumber-bin .
 }
 
