--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@
 license=('MIT')
 _java=21
 depends=("java-runtime=${_java}" 'ttf-dejavu' 'fontconfig')
-makedepends=("java-environment=${_java}" 'git' 'maven')
+makedepends=("java-environment=${_java}" 'git' 'maven' 'corepack')
 provides=('jenkins-ci')
 conflicts=('jenkins-ci')
 replaces=('jenkins-ci')
@@ -34,6 +34,8 @@
 prepare() {
 	cd "${pkgname}"
 	patch -Np1 -i "${srcdir}/skip-failing-test.patch"
+	patch -Np1 -i "${srcdir}/use-system-corepack.patch"
+	patch -Np1 -i "${srcdir}/use-rolldown-wasm.patch"
 }
 
 build() {
@@ -45,7 +47,7 @@
 check() {
 	cd "${pkgname}"
 	export PATH="/usr/lib/jvm/java-${_java}-openjdk/bin:${PATH}"
-	mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" test
+	mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" -Djunit.jupiter.execution.timeout.default=20000 test
 }
 
 package() {
@@ -57,3 +59,7 @@
 	install -Dm 644 "${srcdir}/${pkgname}.conf" "${pkgdir}/etc/conf.d/${pkgname}"
 	install -Dm 644 LICENSE.txt "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
 }
+
+source+=('use-system-corepack.patch' 'use-rolldown-wasm.patch')
+sha512sums+=('88d8eb21704b36833ffb912169acb7b02bd181764520c5a2aad0a7fd739fb0e6b2888ddd60c814a090b10788ddf9d77dec8b6a0dbdddf7a2fd483e6a1c46e7a2'
+            'd14417be40462e1f1eb859f2c3a73cf7d29815e1aa5abe0ad0d639501b33981aa75b0a6147958bc741953c92b10b845ad80acc62368b88141062278b303cdc51')
