diff --git PKGBUILD PKGBUILD
index fff3294..7a67e35 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ arch=('any')
 url='https://jenkins.io'
 license=('MIT')
 depends=("java-runtime=${_java}" 'ttf-dejavu' 'fontconfig')
-makedepends=("java-environment=${_java}" 'git' 'maven')
+makedepends=("java-environment=${_java}" 'git' 'maven' 'nodejs-lts-iron' 'yarn')
 provides=('jenkins-ci')
 conflicts=('jenkins-ci')
 replaces=('jenkins-ci')
@@ -35,6 +35,7 @@ sha512sums=('7119d62ce38e7879cb8b0406cdc7bfde081c29b22444d174a9c37a46a84127cb1ff
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   patch -Np1 <"${srcdir}"/skip-failing-test.patch
+  patch -Np1 <"${srcdir}"/use-system-yarn.patch
 }
 
 build() {
@@ -46,7 +47,7 @@ build() {
 check() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   export PATH="/usr/lib/jvm/java-${_java}-openjdk/bin:${PATH}"
-  mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" test
+  mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" -Djunit.jupiter.execution.timeout.default=20000 test
 }
 
 package() {
@@ -58,3 +59,6 @@ package() {
   install -Dm 644 ../jenkins.sysusers "${pkgdir}/usr/lib/sysusers.d/jenkins.conf"
   install -Dm 644 ../jenkins.conf "${pkgdir}/etc/conf.d/jenkins"
 }
+
+source+=('use-system-yarn.patch')
+sha512sums+=('ba7d22c9ff7005e3c51d32af25cd2a3db8536a29a160eefa39bc41057c585fedae24cc394afa399f0e703a537f75a023a651f3bc203db17a7e7a1fad4ec75458')
\ No newline at end of file
