--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,7 @@ sha512sums=('433bcefd436c2c2b45672f5375ced6b229b77d154fafdffc2e6e14fe35a72377ff1
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   patch -Np1 <"${srcdir}"/skip-failing-test.patch
+  patch -Np1 <"${srcdir}"/use-system-yarn.patch
 }
 
 build() {
@@ -58,3 +59,7 @@ package() {
   install -Dm 644 ../jenkins.sysusers "${pkgdir}/usr/lib/sysusers.d/jenkins.conf"
   install -Dm 644 ../jenkins.conf "${pkgdir}/etc/conf.d/jenkins"
 }
+
+makedepends+=('nodejs' 'yarn')
+source+=("use-system-yarn.patch")
+sha512sums+=('ba7d22c9ff7005e3c51d32af25cd2a3db8536a29a160eefa39bc41057c585fedae24cc394afa399f0e703a537f75a023a651f3bc203db17a7e7a1fad4ec75458')
