--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,7 @@ sha512sums=('6007b822927c56f3bc275f944333621b4a662237f3abe92e9ae766364af0a1387e0
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   patch -Np1 <"${srcdir}"/skip-failing-test.patch
+  patch -Np0 <"${srcdir}"/use-system-yarn.patch
 }
 
 build() {
@@ -58,3 +59,7 @@ package() {
   install -Dm 644 ../jenkins.sysusers "${pkgdir}/usr/lib/sysusers.d/jenkins.conf"
   install -Dm 644 ../jenkins.conf "${pkgdir}/etc/conf.d/jenkins"
 }
+
+makedepends+=('nodejs' 'yarn')
+source+=("use-system-yarn.patch")
+sha512sums+=('dc8507776b047d335fa160dd43809578c684c2f319d6dee81d55344d6b0d9c5e67b5e99b9348996a47b254b9b791dd63d4958f857b916af160072e5d38504643')
