--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@ arch=('any')
 url='https://jenkins.io'
 license=('MIT')
 depends=("java-runtime=${_java}" 'ttf-dejavu' 'fontconfig')
-makedepends=("java-environment=${_java}" 'git' 'maven')
+makedepends=("java-environment=${_java}" 'git' 'maven' 'nodejs-lts-iron' 'yarn')
 provides=('jenkins-ci')
 conflicts=('jenkins-ci')
 replaces=('jenkins-ci')
@@ -24,17 +24,20 @@ source=("https://github.com/jenkinsci/jenkins/archive/refs/tags/jenkins-${pkgver
         'jenkins.service'
         'jenkins.tmpfiles'
         'jenkins.sysusers'
-        'skip-failing-test.patch')
+        'skip-failing-test.patch'
+        'use-system-yarn.patch')
 sha512sums=('1dcea9a9d1d449e953db8da43c7cc8f57667c4afac001a5da8eeddceb4ad9a086ea3dd7398863396d8f5fad4bc8575c3d06f0408a4453297ff8c24ecc58c0df5'
             '482b642988027b623995e0488cc00011be5bbe9890a6d40dd6979197fa50794889ff731f798891106b367962cf60cb59b0260caadfcee9685b358a966f759553'
             'c1e9a542bfd3db54ea0ef57acdac47ad5180330f9ffab44296762b0e69e75ce595d85a77d46025586fb507aedce4f1a032379dcd16b7b886fe475a99849fa4b2'
             'a845a7147be54affc586dfce9a188ec24c92e673e88dec0b62da386d2e597de6ecda103429008562abd897f179a52c37cf2188ebc65b8c636efd07d707e18f90'
             '896cbc64eadb2f3bb9fa7fc26dbfc527b1823d8bff3158be7f1936cc250f55eeafe89079e1ceb8f3d269041662573ccbbb36248140b4ba24144a564366f8847a'
-            '876180dc1674ec56c32a4fceac7d9608f450520ace7d3ecb68cbe7ba8f1bcba31f2c063f52a9076c432b78fd31157da3febc65e98e7d9050a4134d4e0304e8bd')
+            '876180dc1674ec56c32a4fceac7d9608f450520ace7d3ecb68cbe7ba8f1bcba31f2c063f52a9076c432b78fd31157da3febc65e98e7d9050a4134d4e0304e8bd'
+            'ba7d22c9ff7005e3c51d32af25cd2a3db8536a29a160eefa39bc41057c585fedae24cc394afa399f0e703a537f75a023a651f3bc203db17a7e7a1fad4ec75458')
 
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   patch -Np1 <"${srcdir}"/skip-failing-test.patch
+  patch -Np1 <"${srcdir}"/use-system-yarn.patch
 }
 
 build() {
@@ -46,7 +49,7 @@ build() {
 check() {
   cd "${pkgname}-${pkgname}-${pkgver}"
   export PATH="/usr/lib/jvm/java-${_java}-openjdk/bin:${PATH}"
-  mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" test
+  mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" -Djunit.jupiter.execution.timeout.default=20000 test
 }
 
 package() {
@@ -58,3 +61,4 @@ package() {
   install -Dm 644 ../jenkins.sysusers "${pkgdir}/usr/lib/sysusers.d/jenkins.conf"
   install -Dm 644 ../jenkins.conf "${pkgdir}/etc/conf.d/jenkins"
 }
+
