--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@
 license=('MIT')
 _java=21
 depends=("java-runtime=${_java}" 'ttf-dejavu' 'fontconfig')
-makedepends=("java-environment=${_java}" 'git' 'maven')
+makedepends=("java-environment=${_java}" 'git' 'maven' 'corepack')
 provides=('jenkins-ci')
 conflicts=('jenkins-ci')
 replaces=('jenkins-ci')
@@ -35,6 +35,7 @@
 prepare() {
 	cd "${pkgname}"
 	patch -Np1 -i "${srcdir}/skip-failing-test.patch"
+	patch -Np1 -i "${srcdir}/use-system-corepack.patch"
 }
 
 build() {
@@ -46,7 +47,7 @@
 check() {
 	cd "${pkgname}"
 	export PATH="/usr/lib/jvm/java-${_java}-openjdk/bin:${PATH}"
-	mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" test
+	mvn -am -Plight-test -Dproject.build.outputTimestamp="${SOURCE_DATE_EPOCH}" -Djunit.jupiter.execution.timeout.default=20000 test
 }
 
 package() {
@@ -58,3 +59,6 @@
 	install -Dm 644 "${srcdir}/${pkgname}.conf" "${pkgdir}/etc/conf.d/${pkgname}"
 	install -Dm 644 LICENSE.txt "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
 }
+
+source+=('use-system-corepack.patch')
+sha512sums+=('b4e0aa2ff77c94fd3e63b54ff0549b0f4caf6ce40068d390f0fb58bc2449aed98e5fed0e69aff4cad5df963baadd53c0d7bfb53b4c0b32b817c6bc132211cfa2')
