Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422467)
+++ PKGBUILD	(working copy)
@@ -35,6 +35,7 @@
   cd "$srcdir/$pkgbase-$pkgver"
   # NOTE: if you wish to enforce SSL, use SSLTYPE=unix.nopwd
 
+  export CFLAGS=${CFLAGS/-Werror=format-security/}
   yes "y" | make lnp EXTRAAUTHENTICATORS=gss PASSWDTYPE=pam SPECIALAUTHENTICATORS=ssl SSLTYPE=unix EXTRACFLAGS="${CFLAGS} -fPIC -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lpam" EXTRALDFLAGS="${LDFLAGS}"
 
 }
