--- PKGBUILD
+++ PKGBUILD
@@ -33,6 +33,10 @@ prepare() {
   cd vault
   sed -i 's|/etc/vault.d/vault.hcl|/etc/vault.hcl|g' .release/linux/package/usr/lib/systemd/system/vault.service
   sed -i 's|/etc/vault.d/vault.env|/etc/default/vault|g' .release/linux/package/usr/lib/systemd/system/vault.service
+
+  go mod edit -replace go.etcd.io/bbolt=github.com/etcd-io/bbolt@v1.3.7
+  go mod edit -replace github.com/boltdb/bolt=go.etcd.io/bbolt@v1.3.7
+  go mod tidy
 }
 
 build() {
@@ -48,6 +52,7 @@ build() {
   export CGO_CXXFLAGS="${CXXFLAGS}"
   export CGO_LDFLAGS="${LDFLAGS}"
   export GOPATH="${srcdir}"
+  export NODE_OPTIONS="--max-old-space-size=8192"
   mkdir -p bin
   go build \
     -buildmode=pie \
