--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,9 @@ validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.'https://github.com/kpcyrd/clamav-rs']\nclamav-rs = { git = 'https://github.com/aimixsaka/clamav-rs', branch = 'riscv-clamav-1.2' }" >> Cargo.toml
+  cargo update -p clamav-rs
+  cargo fetch --locked
 }
 
 build() {
