--- PKGBUILD
+++ PKGBUILD
@@ -20,8 +20,9 @@ validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  patch -i ../clamav-1.0.patch
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e '\n[patch.crates-io]\nclamav-rs = { git = "https://github.com/losynix/clamav-rs.git", branch = "patches" }' >> Cargo.toml
+  cargo update -p clamav-rs
+  cargo fetch --locked
 }
 
 build() {
