diff --git PKGBUILD PKGBUILD
index 58754df..9bdae25 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,8 @@
 depends=(python
          "${_pydeps[@]/#/python-}")
 makedepends=(python-setuptools
-             python-sphinx)
+             python-sphinx
+             python-sphinx-furo)
 _archive="$_pyname-$pkgver"
 source=("$_archive.tar.gz::https://github.com/wireservice/$_pyname/archive/$pkgver.tar.gz")
 sha256sums=('5546c65742f3550ddee90e8c688e13ec495ec4190bba0e7c9cfa05e233e2aec9')
