From 9e3394dab39c09ced0141b4a5cca19001447f4ef Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Tue, 26 Mar 2024 12:50:15 +0800
Subject: [PATCH] Remove glibc compatibility hack

---
 src/nonFree/sift/vl/host.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/nonFree/sift/vl/host.h b/src/nonFree/sift/vl/host.h
index 6a816f400..ab88a0053 100644
--- a/src/nonFree/sift/vl/host.h
+++ b/src/nonFree/sift/vl/host.h
@@ -658,10 +658,5 @@ vl_swap_host_big_endianness_2 (void *dst, void* src)
 #endif
 }
 
-/* Linux: limit glibc to old versions for compatibility */
-#if defined(VL_COMPILER_GNUC) & defined(VL_OS_LINUX) & ! defined(__DOXYGEN__) & ! defined(ANDROID)
-__asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
-#endif
-
 /* VL_HOST_H */
 #endif
-- 
2.44.0

