--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ url="https://alicevision.github.io/"
 options=('!lto' '!debug') # debug package is kinda big -- needs investigation!
 license=('MPL-2.0' 'MIT')
 depends=('boost-libs' 'openimageio' 'flann' 'geogram' 'coin-or-clp' 'ceres-solver' 'cctag'
-         'alembic' 'opengv' 'opencv' 'popsift' 'uncertainty-framework' 'assimp')
+         'alembic' 'opengv' 'opencv' 'assimp')
 makedepends=('boost' 'ninja' 'eigen' 'freetype2' 'coin-or-coinutils' 'coin-or-lemon'
              'git' 'cmake' 'doxygen' 'python-sphinx') # 'cuda'
 source=("git+https://github.com/alicevision/AliceVision#tag=v${pkgver}"
@@ -22,6 +22,7 @@ source=("git+https://github.com/alicevision/AliceVision#tag=v${pkgver}"
         "FindOsi.cmake"
         "fix-computeUncertainty-build.patch"
         "fix-default-ocio-path.patch"
+        "0001-Remove-glibc-compatibility-hack.patch"
 )
 sha256sums=('SKIP'
             'SKIP'
@@ -30,7 +31,8 @@ sha256sums=('SKIP'
             '6523435334eec6e39a244371287504cd0a0e88aa0cbe5dcac38b819ea881074e'
             'fbb87c86bc0b2ee2c98abfbecb0d555f75f01ccf5d4c59c22bb598e7f2897bf9'
             '17546a6c362782f90d1d85eb7a4b42b96f5f51879c68efbcbcf07635e0f60cbc'
-            '3f02c715f27498ac8982edee3e3af151b0cd2a9cb83da37fef3b7fec1e34b169')
+            '3f02c715f27498ac8982edee3e3af151b0cd2a9cb83da37fef3b7fec1e34b169'
+            'ccb3cd7c0f1914754e59a9b830dbc2eab26dc9f07a7aa2111aa6c490bac9f8ac')
 
 prepare() {
   cd AliceVision
@@ -52,6 +54,9 @@ prepare() {
 
   # fix default OCIO config path
   patch -p1 -i ../fix-default-ocio-path.patch
+
+  # Upstream glibc compatibility hack doesn't compile on riscv64
+  patch -p1 -i ../0001-Remove-glibc-compatibility-hack.patch
   # fix doc build
   sed -i '/^ *install.*doc/s/doc/htmlDoc/' src/CMakeLists.txt
   ln -rs docs/sphinx{,/rst}
@@ -69,15 +74,10 @@ build() {
     -DCLP_INCLUDE_DIR_HINTS=/usr/include/coin \
     -DOSI_INCLUDE_DIR_HINTS=/usr/include/coin \
     -DLEMON_INCLUDE_DIR_HINTS=/usr/include/lemon \
-    -DPopSift_DIR=/usr \
     -DCCTag_DIR=/usr/lib/cmake/CCTag \
-    -DUNCERTAINTYTE_DIR=/usr \
-    -DMAGMA_ROOT=/opt/cuda/targets/x86_64-linux/ \
     -DALICEVISION_BUILD_EXAMPLES=OFF \
     -DALICEVISION_USE_CUDA=OFF \
     -DALICEVISION_USE_CCTAG=ON \
-    -DALICEVISION_USE_POPSIFT=ON \
-    -DALICEVISION_USE_UNCERTAINTYTE=ON \
     -DALICEVISION_USE_ALEMBIC=ON \
     -DALICEVISION_USE_OPENGV=ON \
     -DALICEVISION_USE_OPENCV=ON \
