diff --unified --recursive --text a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
--- a/yabause/src/libretro/Makefile	2021-09-16 11:49:27.528836451 +0000
+++ b/yabause/src/libretro/Makefile	2021-09-16 11:50:30.995544580 +0000
@@ -568,10 +568,6 @@
 	endif
 endif
 
-ifeq ($(HAVE_SSE),1)
-	FLAGS += -mfpmath=sse
-endif
-
 ifeq ($(DEBUG), 1)
 	ifneq (,$(findstring msvc,$(platform)))
 		ifeq ($(STATIC_LINKING),1)
