diff --git PKGBUILD PKGBUILD
index a9fd441df..7ab5b92a1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,15 @@ depends=(
 )
 makedepends=(git)
 _commit=64985d2cacde6ba8568c105e15052920f609cd86
-source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit}
+        'fix-unrecognized-command-line-option-msse.patch')
+sha256sums=('SKIP'
+            'f362d1e350fe9d7a10ad86b1a998f6981afe09218a12693e126b06f943c6e8ac')
+
+prepare(){
+  cd libretro-yabause
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch 
+}
 
 pkgver() {
   cd libretro-yabause
