diff --git PKGBUILD PKGBUILD
index a9fd441df..b075d48ae 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,15 @@ depends=(
 )
 makedepends=(git)
 _commit=12c869191cb3e7483ad63bf1d5a204d78ac996c5
-source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit}
+        'fix-unrecognized-command-line-option-msse.patch')
+sha256sums=('SKIP'
+            '884e76c3a0068be44e0ef731c9938b4adfebc931b92bb7ac23c679501af39003')
+
+prepare(){
+  cd libretro-yabause
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch 
+}
 
 pkgver() {
   cd libretro-yabause
