diff --git PKGBUILD PKGBUILD
index cf32bf00..d56f5f73 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,15 @@
 )
 makedepends=(git)
 _commit=c6275349a40303ae787f7170ae4e55d2e3b5ca5b
-source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit})
-sha256sums=(SKIP)
+source=(libretro-yabause::git+https://github.com/libretro/yabause.git#commit=${_commit}
+        'fix-unrecognized-command-line-option-msse.patch')
+sha256sums=('SKIP'
+            '884e76c3a0068be44e0ef731c9938b4adfebc931b92bb7ac23c679501af39003')
+
+prepare(){
+  cd libretro-yabause
+  patch -Np1 -i ../fix-unrecognized-command-line-option-msse.patch 
+}
 
 pkgver() {
   cd libretro-yabause
