From fc07563aa4a4fccc4232bc223df979ae8d6894c4 Mon Sep 17 00:00:00 2001
From: Coelacanthus <coelacanthus@outlook.com>
Date: Sun, 23 Jan 2022 19:48:42 +0800
Subject: [PATCH] disable OPENMP

---
 PKGBUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git PKGBUILD PKGBUILD
index 7e88d6b..d8aee5c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ build() {
     -DBUILD_FORTRAN=ON \
     -DBUILD_HDF5_FILTER=ON \
     -DBUILD_NETCDF_READER=ON \
-    -DBUILD_OPENMP=ON \
+    -DBUILD_OPENMP=OFF \
     -DBUILD_PYTHON_WRAPPER=ON
   make -C build
 }
-- 
2.34.1
