--- PKGBUILD
+++ PKGBUILD
@@ -9,11 +9,14 @@ url=https://github.com/stylelint/stylelint-config-recommended
 license=('MIT')
 depends=('stylelint')
 makedepends=('git' 'npm')
-source=("git+$url.git")
-b2sums=('SKIP')
+source=("git+$url.git"
+        "disable-v8-jit.patch")
+b2sums=('SKIP'
+        '19d227478e8502d95e599de2c3f595605e1c010a0ca37a19c194211cd2f404226ab0c67799842a0ea6cd8fb38f6b37ac81604c15dcdcba791158643f6b3f1780')
 
 prepare() {
   cd $pkgname
+  patch -Np0 -i ../disable-v8-jit.patch
   npm ci
 }
 
