Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422631)
+++ PKGBUILD	(working copy)
@@ -18,10 +18,12 @@
 prepare() {
   cd cdparanoia-III-$pkgver
   patch -p0 -i "$srcdir"/gcc.patch
+  cp /usr/share/autoconf/build-aux/config.guess ./configure.guess
 }
 
 build() {
   cd cdparanoia-III-$pkgver
+  export CFLAGS=${CFLAGS/-Werror=format-security/}
   ./configure --prefix=/usr --mandir=/usr/share/man
   make
 }
