diff --git PKGBUILD PKGBUILD
index a1ddd1d..52e537c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,7 +25,7 @@ source=(
 sha512sums=('c6198e5c9b8fe5ccd89e445c9252da44d8d7c9e0c8eb5a802fa0cabf89482fddf775ed383bac1acc9331bc3195d21df7ea02c4a73aa6ee163c2959f34175d650'
             'SKIP'
             'SKIP'
-            'e94510c97884b9334a6f1fb70ad5fb92920a3b0610b9d3c64770c41148059aaa3c8100ad385e4ebc610b256af5fb8d13f6a0efdf233cbb734f753f48ee7ba07a')
+            '936a6e9e73f6fcd1128d9caa23c0d9cde1b5ff8b54a97ac5ebcba3764c2d49688f9d24a17ed3391854f1cf7f68c8d4bf4b6bfb7a0dddc888422c2455dccb5254')
 validpgpkeys=(265F80AB84FE03127E14F01125BCCA5220D84079  # Krisztian Szucs (apache) <szucs.krisztian@gmail.com>
               08D3564B7C6A9CAFBFF6A66791D18FCF079F8007) # Kouhei Sutou <kou@cozmixng.org>
 
