diff --git PKGBUILD PKGBUILD
index f6b63c3a..364e977c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,10 +12,17 @@
 depends=('libelf' 'linux-api-headers' 'libbpf' 'libpcap.so')
 makedepends=('clang' 'llvm' 'gcc' 'make' 'm4' 'pkgconfig' 'libpcap' 'zlib' 'bpf')
 options=(!lto)
-source=("$pkgname-$pkgver.tar.gz::https://github.com/xdp-project/xdp-tools/archive/refs/tags/v${pkgver}.tar.gz")
-sha512sums=('ce05dc0eb6905e0c07b24ff649e8715402184cc00ca27b4864f112e37a3df39a0c24822a86091526648be14c71b0bcda0df2b85c47cb6664a3e326b1748c8a71')
-b2sums=('15f66766f72af4fa7b443d052f9f270d6e08da77ac2f313fd50ec912c809a70dcfa80750ea06257bccc0df38a1d23fa6efc6a5dc59936f83e7b4e85864658dba')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/xdp-project/xdp-tools/archive/refs/tags/v${pkgver}.tar.gz"
+        "link-with-lm.patch::https://github.com/xdp-project/xdp-tools/commit/3b4d6453b355cf98c20b4b07fa4c91b736b4e47b.patch")
+sha512sums=('ce05dc0eb6905e0c07b24ff649e8715402184cc00ca27b4864f112e37a3df39a0c24822a86091526648be14c71b0bcda0df2b85c47cb6664a3e326b1748c8a71'
+            '7321139236d538ed5e25bc50012598121455de73aa6c7f7b924a65e482df26c1ab47dd1fc01f666b91a23d3348d88d46a8d2e74313632b3f4b94687799df2402')
+b2sums=('15f66766f72af4fa7b443d052f9f270d6e08da77ac2f313fd50ec912c809a70dcfa80750ea06257bccc0df38a1d23fa6efc6a5dc59936f83e7b4e85864658dba'
+        '34cd3a609d6d30158d4bc35ab5556ce3937ea73a963a1d0fe4f82b16613db9e2c7d7b6a0178a92469ccb5db0b86b08749efdcb1f253d38b6eafd8ba53f89ad0c')
 
+prepare() {
+  cd "${pkgname}-${pkgver}"
+  patch -Np1 -i ../link-with-lm.patch
+}
 
 build() {
   cd "${pkgname}-${pkgver}"
