--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,17 @@ arch=('x86_64')
 url="https://premake.github.io/"
 license=('BSD')
 depends=('glibc')
-source=("https://github.com/premake/premake-core/releases/download/v${_pkgver}/premake-${_pkgver}-src.zip")
-sha512sums=('a349dabe3db5b503c283e836da89a5239007696f87756fe34e011bee655c86cd8ccc1c0c00f72a46d0d10302c2f5d67e37d566313e68a16406924a8ba70c5d8f')
+makedepends=('dos2unix')
+source=("https://github.com/premake/premake-core/releases/download/v${_pkgver}/premake-${_pkgver}-src.zip"
+        premake-riscv.patch)
+sha512sums=('a349dabe3db5b503c283e836da89a5239007696f87756fe34e011bee655c86cd8ccc1c0c00f72a46d0d10302c2f5d67e37d566313e68a16406924a8ba70c5d8f'
+            'e8d81d2f58e6760b2dc83cc754b5c7ab46c276f94cf7a60f23a10ca69db7bacd226c551dbcf4f82d1393b506e1c8786be67ac92d86afc9a4d61f2754a60dc452')
+
+prepare() {
+  cd "premake-$_pkgver-src"
+
+  unix2dos -O ../premake-riscv.patch | patch -Np1 --binary
+}
 
 build() {
   cd "premake-$_pkgver-src/build/gmake2.unix"
