--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ arch=('x86_64')
 url="https://premake.github.io/"
 license=('BSD-3-Clause')
 depends=('glibc' 'util-linux-libs')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/premake/premake-core/archive/refs/tags/v${_pkgver}.tar.gz")
-sha512sums=('3fd5cf526e28b36bf32d3015bcb5841b8dc3e3570029dd25442bba118369f104e7504682f7e1a409d879de562b19db55bccfed97939fef376c211cfb77ceff6e')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/premake/premake-core/archive/refs/tags/v${_pkgver}.tar.gz"
+        premake-riscv.patch)
+sha512sums=('3fd5cf526e28b36bf32d3015bcb5841b8dc3e3570029dd25442bba118369f104e7504682f7e1a409d879de562b19db55bccfed97939fef376c211cfb77ceff6e'
+            'e8d81d2f58e6760b2dc83cc754b5c7ab46c276f94cf7a60f23a10ca69db7bacd226c551dbcf4f82d1393b506e1c8786be67ac92d86afc9a4d61f2754a60dc452')
+
+prepare() {
+  cd "premake-core-$_pkgver"
+  patch -Np1 -i ../premake-riscv.patch
+}
 
 build() {
   cd "premake-core-$_pkgver"
