--- PKGBUILD
+++ PKGBUILD
@@ -40,2 +40,3 @@ source=(
   kube-scheduler.service
+  add-riscv64-support.patch
 )
@@ -58,1 +59,2 @@ sha512sums=('cdd85c10d931367e29d2cf3e99fd7eb0561e14a8c4a73c57cd3e497ecbd4ae4c0b3
-            '2c25c0e11a7b2d6d61e03f9afe7ba21f9497495ab02e85f2623ce8c71019fb8a1af16197ab3968d5da050c2188c3e67372aa43322ac91af84f7da61bf73596bc')
+            '2c25c0e11a7b2d6d61e03f9afe7ba21f9497495ab02e85f2623ce8c71019fb8a1af16197ab3968d5da050c2188c3e67372aa43322ac91af84f7da61bf73596bc'
+            'faba6b03ea87469a9b4588e6dfcb7a193407759a867b2b6894a6d44c45dff0df72ac1a27abaacd76b393adc317c93e2f0a02e5e29d03a15e13d9d1934936893e')
@@ -72,11 +74,16 @@ b2sums=('f5076a5ec36cf108b363645b7e39dd9e9b7d56c26457a3a740f30fad29db2bf66877f71
         '082474a56525f7ea52315fbf9d765081d9f6bbd20ab4213bc2d2a6ddb8a3764987e365f08f157be6deec53f9c1bf2fb6c99595f5649c631e8610fcf81ed61eab'
         '5ce796468c442d76f311d1c620576dcd2c784fcd40ecaa68ca3a6d7c089e8703506f712ee918f89ddb4debfe061cea3939fa4f2d2a77553dd1cd7aa4fae17729'
         '4f35d3b5296839dd68885b924a41725a3479c2725c77887038c7d402a6aa754e37fe0d3697e746a1b65a5236f9e927df01e99a66d37d227cb801965575403788'
-        '95a5345e044b8fe9be01fc6230f20b3b12118c6f1fcc3e7ebbfd4d3a3d205dfb492ff0c11397d3dae9abc0cbba83c659ea23803b71c8e240dddc30bf4791dbc5')
+        '95a5345e044b8fe9be01fc6230f20b3b12118c6f1fcc3e7ebbfd4d3a3d205dfb492ff0c11397d3dae9abc0cbba83c659ea23803b71c8e240dddc30bf4791dbc5'
+        '4bb18835fe5c411362a2e990298543ee79544bf9d29efdd0d2880665386eee208c39f79a3c2e7166ec496c43c9d4a95be200755900c2a01e568488b25c0ced2a')
 
 prepare() {
   # set static builds CGO_ENABLED=1 for full RELRO
   patch -Np1 -d $pkgbase-$pkgver -i ../$pkgbase-1.24.0-static_cgo_enabled.patch
+  # add riscv64 support
+  # feedback from upstream:
+  # https://github.com/kubernetes/kubernetes/pull/86011#issuecomment-633312469
+  patch -Np1 -d $pkgbase-$pkgver -i ../add-riscv64-support.patch
 }
 
 build() {
@@ -112,7 +119,7 @@ package_kubeadm() {
   depends=(glibc crictl)
   install=$pkgname.install
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # service override for kubelet.service
@@ -127,7 +134,7 @@ package_kubectl() {
   depends=(glibc)
   groups=(kubernetes-tools)
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1"
   # shell completion
@@ -147,7 +154,7 @@ package_kubelet() {
   backup=(etc/$pkgbase/$pkgname.env)
   install=$pkgname.install
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # config
@@ -168,7 +175,7 @@ package_kube-apiserver() {
   groups=(kubernetes-control-plane)
   backup=(etc/$pkgbase/$pkgname.env)
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # config
@@ -183,7 +190,7 @@ package_kube-controller-manager() {
   groups=(kubernetes-control-plane)
   backup=(etc/$pkgbase/$pkgname.env)
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # config
@@ -198,7 +205,7 @@ package_kube-proxy() {
   groups=(kubernetes-control-plane kubernetes-node)
   backup=(etc/$pkgbase/$pkgname.env)
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin/"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin/"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # config
@@ -215,7 +222,7 @@ package_kube-scheduler() {
   groups=(kubernetes-control-plane)
   backup=(etc/$pkgbase/$pkgname.env)
 
-  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin"
+  install -vDm 755 $pkgbase-$pkgver/_output/local/bin/linux/riscv64/$pkgname -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 $pkgbase-$pkgver/docs/man/man1/$pkgname* -t "$pkgdir/usr/share/man/man1/"
   # config
