diff --git PKGBUILD PKGBUILD
index 824ca43..a98f332 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,7 @@ source=("https://github.com/kubernetes/kubernetes/archive/v${pkgver}/kubernetes-
         "kube-proxy.service"
         "kube-scheduler.env"
         "kube-scheduler.service"
+        "add-riscv64-support.patch"
 )
 sha512sums=('81045b0432f60a865c79f408921ff7c5865cdcb297c4cf17f741a08a5fa0f2af9e8c938e54e73c532dbeb643316e8acf983a0e9203be8ce920c2ae588dec6554'
             'a2c97738fb49c0e30eee44b6404af083bbaddbafb8a3fa92493802065f4ac06f0c09d50437a9fdda345e0365de984d9bc768fb9300a40c716e6dec53bd57b6de'
@@ -46,7 +47,8 @@ sha512sums=('81045b0432f60a865c79f408921ff7c5865cdcb297c4cf17f741a08a5fa0f2af9e8
             'c5c16d97afc0fa455981a56794547a4e6e8a710b1b686ccf84645c8001a601fa41b624ad0009bf21e56ec2da35874ac7808731b4a5b9b0fd80fc188714708f23'
             'dd4efa137462905f9e29a99d69b747ae35e58ba8152794bfa417325953dd5059bad96fabfabf73ce1ee2310dc4ab4d1b95c8d931d33b81c67addcb614a51be54'
             'ba277f765959ddb8aa0dee5a86cd9df1f40fb3f6ea1001f24825dbf21bd9342981d13894301170431729f76e710f70c23481e4061c64be29517ff497490f1ef3'
-            '2c25c0e11a7b2d6d61e03f9afe7ba21f9497495ab02e85f2623ce8c71019fb8a1af16197ab3968d5da050c2188c3e67372aa43322ac91af84f7da61bf73596bc')
+            '2c25c0e11a7b2d6d61e03f9afe7ba21f9497495ab02e85f2623ce8c71019fb8a1af16197ab3968d5da050c2188c3e67372aa43322ac91af84f7da61bf73596bc'
+            'faba6b03ea87469a9b4588e6dfcb7a193407759a867b2b6894a6d44c45dff0df72ac1a27abaacd76b393adc317c93e2f0a02e5e29d03a15e13d9d1934936893e')
 b2sums=('bcc5deb9f446032baeff60b3f75e11c75af0e69fd0e50df789d2b7b532f9e839875ef4227729166d667f022816b321bacf219b0fdeaa2d0da552587d4a9a4067'
         'a84c41a2b2be6ab0f5f6c3ede13b28456deb7098421a6f1af0e72e53a9214efa30a59709c5b1e6c35031c4dfd15e786a691b69e592796d769cf057fb1d21379a'
         'dbab30d7e1b566027fec9d6a95dd41ed8f64399c39aac07fb4513ce21050eeeb4a226adfd513f76921c305945dfa4a140602ede574dd5eb4cd287e0f2df21714'
@@ -63,12 +65,14 @@ b2sums=('bcc5deb9f446032baeff60b3f75e11c75af0e69fd0e50df789d2b7b532f9e839875ef42
         '082474a56525f7ea52315fbf9d765081d9f6bbd20ab4213bc2d2a6ddb8a3764987e365f08f157be6deec53f9c1bf2fb6c99595f5649c631e8610fcf81ed61eab'
         '5ce796468c442d76f311d1c620576dcd2c784fcd40ecaa68ca3a6d7c089e8703506f712ee918f89ddb4debfe061cea3939fa4f2d2a77553dd1cd7aa4fae17729'
         '4f35d3b5296839dd68885b924a41725a3479c2725c77887038c7d402a6aa754e37fe0d3697e746a1b65a5236f9e927df01e99a66d37d227cb801965575403788'
-        '95a5345e044b8fe9be01fc6230f20b3b12118c6f1fcc3e7ebbfd4d3a3d205dfb492ff0c11397d3dae9abc0cbba83c659ea23803b71c8e240dddc30bf4791dbc5')
+        '95a5345e044b8fe9be01fc6230f20b3b12118c6f1fcc3e7ebbfd4d3a3d205dfb492ff0c11397d3dae9abc0cbba83c659ea23803b71c8e240dddc30bf4791dbc5'
+        '4bb18835fe5c411362a2e990298543ee79544bf9d29efdd0d2880665386eee208c39f79a3c2e7166ec496c43c9d4a95be200755900c2a01e568488b25c0ced2a')
 
 prepare() {
   cd "${pkgbase}-${pkgver}"
   # the -tags can only be a space separated list
   patch -Np1 -i ../"${pkgbase}-1.20.0-gotags.patch"
+  patch -Np1 -i ../"add-riscv64-support.patch"
   # set CGO_ENABLED
   sed -E 's/(CGO_ENABLED)=0/\1=1/g' -i hack/lib/golang.sh
 }
@@ -103,7 +107,7 @@ package_kubeadm() {
   install="${pkgname}.install"
 
   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # service override for kubelet.service
   install -vDm 644 "../10-${pkgname}-kubelet.conf" \
     "${pkgdir}/usr/lib/systemd/system/kubelet.service.d/10-${pkgname}.conf"
@@ -119,7 +123,7 @@ package_kubectl() {
   groups=('kubernetes-tools')
 
   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 "docs/man/man1/${pkgname}"* -t "$pkgdir/usr/share/man/man1"
   # shell completion
@@ -138,7 +142,7 @@ package_kubelet() {
   install="${pkgname}.install"
 
   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "$pkgdir/etc/kubernetes/"
   # service
@@ -161,7 +165,7 @@ package_kube-apiserver() {
   backup=('etc/kubernetes/kube-apiserver.env')
 
   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -180,7 +184,7 @@ package_kube-controller-manager() {
   backup=("etc/kubernetes/${pkgname}.env")
 
   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -199,7 +203,7 @@ package_kube-proxy() {
   backup=("etc/kubernetes/${pkgname}.env")
 
   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin/"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin/"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -216,7 +220,7 @@ package_kube-scheduler() {
   backup=("etc/kubernetes/${pkgname}.env")
 
   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
