diff --git PKGBUILD PKGBUILD
index 2f4aeb3..2768496 100644
--- PKGBUILD
+++ PKGBUILD
@@ -32,6 +32,7 @@ makedepends=(
 source=(
   https://github.com/$pkgbase/$pkgbase/archive/v$pkgver/$pkgbase-$pkgver.tar.gz
   $pkgbase-1.24.0-static_cgo_enabled.patch
+  add-riscv64-support.patch
   10-kubeadm-kubelet.conf
   50-kubelet-sysctl.conf
   kubelet-modules.conf
@@ -50,6 +51,7 @@ source=(
 )
 sha512sums=('51cf0178c8a2a00798cc618c9918f556c418de137566db60a66a0c7556ee625b34cf86b1da241856599784588c0e3e8b81225dca627fea70a87c94adb073bb7a'
             'e98921d111fabf2c353df589dea0fab57b0b174079483b4092741cc50553a3b765cea4c784e42ad9ab90827581cac028af4fc7a2db5acde1877446882393ae78'
+            '93e12931be05ae304cd5ef1e01ac9c3b62875da008c15bdb6fe2bc02e12b02b9189d5d073aa736ecb005c5de04d95febbd238c75dc20b91d783bcd895babbdb2'
             '5f7132636b6afe9f00dc450c58073c0829942fa44070e7ec5a2c227c485c83f076bdea081d207f926b44d02700be65bf19a61f5d8d1472edd480f980e6ffbc3a'
             'ed5ba22b37eaa9f4950ff3b57d60dd7866fcd5b8bd5197eab3170470528e8d91379483d3eb724589e695184f9b0ed506ebaee73ecca0dc40afdb5f35e79d178a'
             'c318b64a03da07dfe435b2d8c368e55b0ab567da78c57ed814a7864fa75aeac52b28cf562b4afd8daa52168af93b318c1fead557ee676e950af25d422c276a17'
@@ -67,6 +69,7 @@ sha512sums=('51cf0178c8a2a00798cc618c9918f556c418de137566db60a66a0c7556ee625b34c
             '2c25c0e11a7b2d6d61e03f9afe7ba21f9497495ab02e85f2623ce8c71019fb8a1af16197ab3968d5da050c2188c3e67372aa43322ac91af84f7da61bf73596bc')
 b2sums=('69aab2d42fdbb5704d6bd248e6ced154664d20d66ad21e7cf3b6ed92a262f4d1a2dc5b22912b6cdca0a40fecdc262ae7124a2969955f234c49c0950ea68f171f'
         '94a21e3923396b5a97a1e713419bab1f897ce8a4eedbbc2c0ed82fd1fd909bb69183c349322d2a774afd26c4beeb9972a57c6d4f67ef527c7e0ca7ea20651d43'
+        '1b7d72d1dd66486ae81bd319ebf4727454f3801d915e6e7868a3e07bc6347337ecedfe91c02b402bd57a584e122f522795beb56e75c473dbe2c37c0ed12580be'
         'dbab30d7e1b566027fec9d6a95dd41ed8f64399c39aac07fb4513ce21050eeeb4a226adfd513f76921c305945dfa4a140602ede574dd5eb4cd287e0f2df21714'
         '27a8dcbbe06fa7aea122fc87ca663710ae2179c995270d94e22c905422e2639f3c9c81eef6723467c76366062381d8bd65e84ef79f47fed7c240973a55f5cb0e'
         'a03b8a2ce6a606068cad278c6b8039181968c132a935448bf45e1b3668357487da1528569ed582b107db0654a1614b784a73c726729aad42abed18a4c15ce5b6'
@@ -86,6 +89,8 @@ b2sums=('69aab2d42fdbb5704d6bd248e6ced154664d20d66ad21e7cf3b6ed92a262f4d1a2dc5b2
 prepare() {
   # set static builds CGO_ENABLED=1 for full RELRO
   patch -Np1 -d $pkgbase-$pkgver -i ../$pkgbase-1.24.0-static_cgo_enabled.patch
+  # add riscv64 support
+  patch -Np1 -d $pkgbase-$pkgver -i ../add-riscv64-support.patch
 }
 
 build() {
