diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,7 @@
   "https://github.com/kubernetes/kubernetes/archive/v${pkgver}/kubernetes-${pkgver}.tar.gz"
   "${pkgbase}-1.23.0-gotags.patch"
   "${pkgbase}-1.23.0-static_cgo_enabled.patch"
+  "add-riscv64-support.patch"
   "10-kubeadm-kubelet.conf"
   "50-kubelet-sysctl.conf"
   "kubelet-modules.conf"
@@ -35,6 +36,7 @@
 sha512sums=('95ccd27680cefa1f0e367259a373236866c85b8b0534bed364101b5177c718de0bcd6631136e84b910a2fce00f4c119aef043782228658f852d6753f6606c316'
             'da59a9d6e3fd9625d2803e441f6f06c8b272ee5a220eb32426f09245e62c5f19bda324a752ead05111471f7ccd11fe9777630ae9b7ae387fcd94f65a7f1ed5b8'
             'affcabbceadddb3f4178b9fba3e15f06bc6a21a6aa1c7b37c48defd1a81f674bc20198458ca7afdf1d979e2175d12648a70bb29d78ddaf7f5897241cd8d56dbe'
+            'faba6b03ea87469a9b4588e6dfcb7a193407759a867b2b6894a6d44c45dff0df72ac1a27abaacd76b393adc317c93e2f0a02e5e29d03a15e13d9d1934936893e'
             '5f7132636b6afe9f00dc450c58073c0829942fa44070e7ec5a2c227c485c83f076bdea081d207f926b44d02700be65bf19a61f5d8d1472edd480f980e6ffbc3a'
             'ed5ba22b37eaa9f4950ff3b57d60dd7866fcd5b8bd5197eab3170470528e8d91379483d3eb724589e695184f9b0ed506ebaee73ecca0dc40afdb5f35e79d178a'
             'c318b64a03da07dfe435b2d8c368e55b0ab567da78c57ed814a7864fa75aeac52b28cf562b4afd8daa52168af93b318c1fead557ee676e950af25d422c276a17'
@@ -53,6 +55,7 @@
 b2sums=('edfcd8649995cf944225cf3bc276a7b7cac5ae055e4255319c924b1b80776d939c6630dbd4f074c2b606e16d5abc5c45e72a11b42dda428c8e29411521b63bf7'
         '1f6e88cc5817584c72fd2166d50217f06bf29c66b9317c7b9d1a331d8a8192ad189bdfb0e3f0c467916432e9e6a54891753324cfb9ac12396aee66a65dd851da'
         'b06a21c5d4c349c8e9756022681085dc8a6bc4211a8e4700ccf10757a72a0da5e455f36adccfb41c1dd0d61d1df073c2a09e2074e10664f6de37a722c6d1401a'
+        '4bb18835fe5c411362a2e990298543ee79544bf9d29efdd0d2880665386eee208c39f79a3c2e7166ec496c43c9d4a95be200755900c2a01e568488b25c0ced2a'
         'dbab30d7e1b566027fec9d6a95dd41ed8f64399c39aac07fb4513ce21050eeeb4a226adfd513f76921c305945dfa4a140602ede574dd5eb4cd287e0f2df21714'
         '27a8dcbbe06fa7aea122fc87ca663710ae2179c995270d94e22c905422e2639f3c9c81eef6723467c76366062381d8bd65e84ef79f47fed7c240973a55f5cb0e'
         'a03b8a2ce6a606068cad278c6b8039181968c132a935448bf45e1b3668357487da1528569ed582b107db0654a1614b784a73c726729aad42abed18a4c15ce5b6'
@@ -75,6 +78,7 @@
   patch -Np1 -i ../"${pkgbase}-1.23.0-gotags.patch"
   # set static builds CGO_ENABLED=1 for full RELRO
   patch -Np1 -i ../"${pkgbase}-1.23.0-static_cgo_enabled.patch"
+  patch -Np1 -i ../"add-riscv64-support.patch"
 }

 build() {
@@ -108,7 +112,7 @@
   install="${pkgname}.install"

   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # service override for kubelet.service
   install -vDm 644 "../10-${pkgname}-kubelet.conf" \
     "${pkgdir}/usr/lib/systemd/system/kubelet.service.d/10-${pkgname}.conf"
@@ -124,7 +128,7 @@
   groups=(kubernetes-tools)

   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # man pages
   install -vDm 644 "docs/man/man1/${pkgname}"* -t "$pkgdir/usr/share/man/man1"
   # shell completion
@@ -144,7 +148,7 @@
   install=${pkgname}.install

   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "$pkgdir/etc/kubernetes/"
   # service
@@ -167,7 +171,7 @@
   backup=(etc/kubernetes/kube-apiserver.env)

   cd "${pkgbase}-${pkgver}"
-  install -vDm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -vDm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -186,7 +190,7 @@
   backup=(etc/kubernetes/${pkgname}.env)

   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -205,7 +209,7 @@
   backup=(etc/kubernetes/${pkgname}.env)

   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin/"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin/"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
@@ -222,7 +226,7 @@
   backup=(etc/kubernetes/${pkgname}.env)

   cd "${pkgbase}-${pkgver}"
-  install -Dm 755 "_output/local/bin/linux/amd64/${pkgname}" -t "$pkgdir/usr/bin"
+  install -Dm 755 "_output/local/bin/linux/riscv64/${pkgname}" -t "$pkgdir/usr/bin"
   # config
   install -vDm 644 "../${pkgname}.env" -t "${pkgdir}/etc/kubernetes/"
   # service
