diff --git a/openapi/commando_test.go b/openapi/commando_test.go
index 64101b84..2d39a9e4 100644
--- a/openapi/commando_test.go
+++ b/openapi/commando_test.go
@@ -90,7 +90,7 @@ func Test_main(t *testing.T) {
 	args = []string{"ecs", "DescribeRegions"}
 	err = command.main(ctx, args)
 	assert.NotNil(t, err)
-	assert.True(t, strings.Contains(err.Error(), "SDK.ServerError\nErrorCode: InvalidAction.NotFound\nRecommend: https://error-center.aliyun.com/status/search?Keyword=InvalidAction.NotFound&source=PopGw\nRequestId:"))
+	assert.True(t, strings.Contains(err.Error(), "SDK.ServerError\nErrorCode: InvalidAction.NotFound\nRecommend: https://troubleshoot.api.aliyun.com?q=InvalidAction.NotFound&product=Ecs\nRequestId:"))
 
 	ctx.Flags().Get("force").SetAssigned(false)
 	ctx.Flags().Get("version").SetAssigned(false)
