diff --git PKGBUILD PKGBUILD
index c67a6659..bc6afb74 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@
 depends=('glibc')
 makedepends=('git' 'go')
 source=("git+${url}#commit=${_gitcommit}"
-        git+https://github.com/aliyun/aliyun-openapi-meta)
+        git+https://github.com/aliyun/aliyun-openapi-meta
+        fix-commando-test.patch)
 sha512sums=('SKIP'
-            'SKIP')
+            'SKIP'
+            '21f26e58f5a6ea9e0087faa6994c6e5f7b3207e2dfc8c0bad7c8f766fac5e575c3ee47a60b72792eff7132f2a2d964d4bc25821be7351f9e79d7fd50f02cf0a9')
 b2sums=('SKIP'
-        'SKIP')
+        'SKIP'
+        '2090cc8b18f1185234179cbba7e71a4c748ddd98e2019f13722730d93ce2a0c92467f77f673135ef788d6032bed0280251c7e0edaa8ae28b38589fe7e316325a')
 
 pkgver() {
   cd ${pkgname}
@@ -28,6 +31,7 @@
   git submodule init
   git config submodule."aliyun-openapi-meta".url "${srcdir}/aliyun-openapi-meta"
   git submodule update --init --recursive
+  patch -Np1 -i ../fix-commando-test.patch
 }
 
 build() {
