From 1aa322b9e40d926dadcbb771f3ecdfa6bfbb9541 Mon Sep 17 00:00:00 2001
From: Rapiz1 <contact@rapiz.me>
Date: Sat, 23 Oct 2021 21:39:18 +0800
Subject: [PATCH] Fix gcc format string error

Signed-off-by: Rapiz1 <contact@rapiz.me>
---
 test/testutility.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/testutility.cpp b/test/testutility.cpp
index 76308a967..af09e3aeb 100644
--- a/test/testutility.cpp
+++ b/test/testutility.cpp
@@ -199,7 +199,7 @@ private slots:
 
         const auto add = [](const QString &child, const QString &parent, bool result, bool casePreserving = true) {
             const auto title = QStringLiteral("CasePreserving %1: %2 is %3 child of %4").arg(casePreserving ? QStringLiteral("yes") : QStringLiteral("no"), child, result ? QString() : QStringLiteral("not"), parent);
-            QTest::addRow(qUtf8Printable(title)) << child << parent << result << casePreserving;
+            QTest::addRow("%s", qUtf8Printable(title)) << child << parent << result << casePreserving;
         };
         add(QStringLiteral("/A/a"), QStringLiteral("/A"), true);
         add(QStringLiteral("/A/a"), QStringLiteral("/A/a"), true);
-- 
2.33.1

