--- PKGBUILD.orig	2021-12-14 16:40:20.548937755 +0800
+++ PKGBUILD	2021-12-14 16:40:09.232108910 +0800
@@ -12,11 +12,18 @@
 makedepends=(extra-cmake-modules python-sphinx qt5-tools doxygen kio)
 optdepends=('python-nautilus: integration with Nautilus' 'nemo-python: integration with Nemo' 'python-caja: integration with Caja')
 backup=('etc/ownCloud/sync-exclude.lst')
-source=(https://download.owncloud.com/desktop/ownCloud/stable/$pkgver/source/ownCloud-$pkgver.tar.xz{,.asc})
+source=(https://download.owncloud.com/desktop/ownCloud/stable/$pkgver/source/ownCloud-$pkgver.tar.xz{,.asc}
+	0001-Fix-gcc-format-string-error.patch)
 sha256sums=('8ba4e624512e1f80358d3ca82a2268554ef20be0176481f89a26129375f3544a'
-            'SKIP')
+            'SKIP'
+            '09fe477c042159de66bb6ddf0936470a1757b44968119f0446f1bbbaac484451')
 validpgpkeys=(F05F7DD7953A07DF36579DAA498C45EBE94E7B37) # ownCloud Client Team (Signing Key) <info@owncloud.com>
 
+prepare() {
+  cd ownCloud-$pkgver 
+  patch -Np1 < ../0001-Fix-gcc-format-string-error.patch
+}
+
 build() {
   cmake -B build -S ownCloud-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
