diff --git PKGBUILD PKGBUILD
index 1321a12f..18c2aa14 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,15 @@ license=('GPL3')
 groups=('arduino')
 depends=('glibc')
 makedepends=('go' 'git')
-source=("git+$url.git#tag=$pkgver")
-sha512sums=('SKIP')
+source=("git+$url.git#tag=$pkgver"
+        "update_go_vendor.patch")
+sha256sums=('SKIP'
+            '8239b7cd2e92a8f2daea1e4ab7a59f5e99bff9bf38cba2763f74083416d0c400')
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i ../update_go_vendor.patch
+}
 
 build(){
   cd $pkgname
