--- PKGBUILD
+++ PKGBUILD
@@ -14,11 +14,14 @@
   'git'
   'go'
 )
-source=("git+$url.git#tag=v$pkgver")
-b2sums=('9e57cabb66b018f3771f729e3c0004ff06cd8f8540d5fec3cc295679dcad2aeeae165f27c07222be872dfbe29e7de68209def7e97863cbb19ad2c7b7a7e291a3')
+source=("git+$url.git#tag=v$pkgver"
+        'dependency-test-host.patch')
+b2sums=('9e57cabb66b018f3771f729e3c0004ff06cd8f8540d5fec3cc295679dcad2aeeae165f27c07222be872dfbe29e7de68209def7e97863cbb19ad2c7b7a7e291a3'
+        '290f3ec524d7d28ae5b37ddfd6964b775cf60251b1106158e2a1983df8ada3722580a57677f29a8c0cbbb87c098ffdea47456be0a69029d208da8159293eb786')
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../dependency-test-host.patch
   GOFLAGS="-mod=readonly" go mod vendor -v
 }
 
