diff --git PKGBUILD PKGBUILD
index 6eb37e8..26058e3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,6 +11,7 @@ url="http://he.fi/bchunk/"
 arch=('x86_64')
 license=('GPL')
 depends=('glibc')
+makedepends=('systemd')
 source=(http://he.fi/$pkgname/$pkgname-$pkgver.tar.gz)
 sha512sums=('94fcc1dea3a38a907b3f209c20f3de54028cc34bf8964d1dccf94faf3ca3851270cc8e052c29a3f04c60e525c029ebfb1c3c04fa85525e20273ac8d72cb317c8')
 
