--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,14 @@ sha256sums=('de1847f7b59021c16bdc72abb4d8e2d9187cd6124d69156f3326dd34ee043681'
             '9ddf9f7581f5194d73ad908f3b26354c70d3079a7a87c8add8a0d0e4c6974fec')
 validpgpkeys=('F879F0DEDA780198DD08DC6443EC92504F71955A') # Dave Beckett
 
+prepare() {
+  cd "${srcdir}/${pkgname}-${pkgver}"
+  cp /usr/share/autoconf/build-aux/config.guess build/config.guess
+  cp /usr/share/autoconf/build-aux/config.guess libltdl/config/config.guess
+  cp /usr/share/autoconf/build-aux/config.sub build/config.sub
+  cp /usr/share/autoconf/build-aux/config.sub libltdl/config/config.sub
+}
+
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   ./configure --prefix=/usr \
