--- a/src/topology/test/backpressure.rs
+++ b/src/topology/test/backpressure.rs
@@ -170,6 +170,11 @@
 // Wait until the source has sent at least the expected number of events, plus a small additional
 // margin to ensure we allow it to run over the expected amount if it's going to.
 async fn wait_until_expected(source_counter: impl AsRef<AtomicUsize>, expected: usize) {
-    crate::test_util::wait_for_atomic_usize(source_counter, |count| count >= expected).await;
+    crate::test_util::wait_for_atomic_usize_timeout_ms(
+        source_counter,
+        |count| count >= expected,
+        30_000,
+    )
+    .await;
     tokio::time::sleep(Duration::from_millis(100)).await;
 }
