diff --git PKGBUILD PKGBUILD
index 7c59ec7..4e7fda0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,10 @@ makedepends=(gobject-introspection vala gi-docgen git meson)
 checkdepends=(xorg-server-xvfb)
 options=(debug)
 _commit=1e93d67061037c9eb241112c2b7424e31709d868  # tags/5.6.1^0
-source=("git+https://gitlab.gnome.org/GNOME/gtksourceview.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gtksourceview.git#commit=$_commit"
+        gsv-MR289.patch::https://gitlab.gnome.org/GNOME/gtksourceview/-/merge_requests/289.patch)
+sha256sums=('SKIP'
+            '6bb8c5b9ee445a7d085460f43404c89c7582929a0715fff0ac777cf6402f5f0a')
 
 pkgver() {
   cd gtksourceview
@@ -25,6 +27,7 @@ pkgver() {
 
 prepare() {
   cd gtksourceview
+  patch -Np1 < ../gsv-MR289.patch
 }
 
 build() {
@@ -34,7 +37,7 @@ build() {
 
 check() {
   dbus-run-session xvfb-run -s '-nolisten local' \
-    meson test -C build --print-errorlogs
+    meson test -C build --print-errorlogs -t 5
 }
 
 package_gtksourceview5() {
