--- PKGBUILD
+++ PKGBUILD
@@ -57,22 +57,54 @@
   https://cdn.kernel.org/pub/linux/kernel/v${_major}.x/${_srcname}.tar.{xz,sign}
   $url/-/releases/v${_distro_patch_ver}/downloads/$pkgbase-v${_rt_patch_ver}.patch.{sig,zst}
   $url/-/releases/v${_distro_patch_ver}/downloads/$pkgbase-v${_distro_patch_ver}.patch.{sig,zst}
+  # sg2042 write_once
+  0000-sg2042-write_once.patch::https://github.com/revyos/linux/commit/8b001d81419df1c78c58411a51dfd0475233d319.patch?full_index=1
+  # drm/ttm: support device w/o coherency
+  0100-ttm-coherency.patch::https://github.com/revyos/linux/commit/bcfe733064fbe0cb6dab76a73eabf100e73801f9.patch?full_index=1
+  0101-ttm-coherency2.patch::https://github.com/revyos/linux/commit/d5dad38f150b1e153b16cc610e9104bd1ef634b4.patch?full_index=1
+  # rust + gcc
+  0400-rust.patch::https://github.com/revyos/linux/commit/a0c1f2370d02c471f93578a7ce61cae18896acac.patch?full_index=1
+  0401-rust1.patch::https://github.com/revyos/linux/commit/edc216123add9833e039b1c741f2c2b2bde4487d.patch?full_index=1
+  0402-rust2.patch::https://github.com/revyos/linux/commit/71c8e98962c82e04b2e70c50a894032366651b89.patch?full_index=1
+  0403-rust3.patch::https://github.com/revyos/linux/commit/b60775af94fda606995ef848678c5e812951fcf7.patch?full_index=1
 )
 source_x86_64=(config.x86_64)
+source_riscv64=(
+  config.x86_64
+  riscv64.config-patch
+)
 sha512sums=('9a7ee3e35e1e4eea44fd2fadce7b51deb9cae8b1e19ed4d8dc1e59d2e310ffa6dae508a9b0919d2d3cd29d00ca79fd473e7540a3cfb1124e56c4de091915a9d9'
             'SKIP'
             'SKIP'
             'f4015d2d94395535f8358b0bbb8d93675482cfb7c3cb4f31cf546e8eb73b786d53e98c4b9168e2f7e63ee999a7096b62bba6310c422d28c264debc77d1360f38'
             'SKIP'
-            'e0f6771fc4922e33e0bd1555e992d79a109801cc37a3e6ad1abb5aad69633bc7356cb1e8b9e75ec1041471ed80e30a67b8eb120693332ca1daed00b31af8187d')
+            'e0f6771fc4922e33e0bd1555e992d79a109801cc37a3e6ad1abb5aad69633bc7356cb1e8b9e75ec1041471ed80e30a67b8eb120693332ca1daed00b31af8187d'
+            'a28825583c831f0ea2e2f644384a1b14a615d37eb49e0c768c531959ea985716dadd07c0283cfd50b6cda825f77a4b43151c02e4ea29bea14f49dccde5850d71'
+            '63292f705aa2c832cd2279e0c59d56b8f0583280eec3e4b95837f1a88b616c19d26a3185cc022a22bbff3cb6d22e501bbee24e33a208069957fabcdd759336a1'
+            '4c9ef074bf244d5c8cf2dfda67b32de738a8cc196a08b1086844202444650b53e4fd6030b605df365c3bad296aca1e2df4b8e9822c42a4691e1122f8c579d651'
+            'e398dc144a8465b3d356e7b674dd953896a741cfa473e5ba933caef6e2a5cb2af290f80df868eb44aecbb24bf3fd742b0ad894d8b180e5d92292e90abc935cf1'
+            '44a908dbd9ac63e8130c2c2818978cc412bc6bf77d66071170869c7bbb4c8b2cc98f664b249a77b8f6e89ae76b08b7673333f4981c3c60e2237576bdbe64c441'
+            '680e4166ad7e5a5370374dcc136e43a0ad6fbff2e493998392d345ee4b2f04ded99bf9fd4fac02ac69de07520844f1098e684ad1eb37c1a1f5c6d917e9bb94a5'
+            '214de0b7a140f7a84dd1e40623d2b42bd298ed091fbf16874436673669621a6464a39d0baa250d1b1b850e2e47279551b215beac960c5b6386a4619122391fd9')
 sha512sums_x86_64=('759c7f37d121cc7dbc4e691be83aa2341ee5c65080388bf8149c93e362d2e9801194c690e29283822d8d68b85bad613266c6a9343e85bfa571e4cb6f72e21d06')
+sha512sums_riscv64=('759c7f37d121cc7dbc4e691be83aa2341ee5c65080388bf8149c93e362d2e9801194c690e29283822d8d68b85bad613266c6a9343e85bfa571e4cb6f72e21d06'
+                    'ec917f5a1dbca1e7237d99bf9d19d4caaab7846a0234349d77e0b0b760eb858c7321213e7140dba7086f5391391b8dc79db535295be9efce95b90dec460921d9')
 b2sums=('0fa304e65b0d96d6082e3040db9c974ebdbcfd54a2bc22a8841db63291787f01bf63862d882ecf77cb810f51e91db4e44ac6e27616f93f945531b868b9a1a00d'
         'SKIP'
         'SKIP'
         '6a1cd2832b688690cfa42fc97b01cf29c00c72352797064738c4e5a8849eab9e22938d5c93926c3f1c73f3d7bc137aaaa5a8d7669ff2e8291f2052d2c78c325f'
         'SKIP'
-        '85e8f2660f38d4eb3041e274c6ba14e1efaae069ece331d0fc88feb833e95212b4e9c8bac2ed224d2de4408407701c144158eb68fa180492bbb34e0d54b5c3ac')
+        '85e8f2660f38d4eb3041e274c6ba14e1efaae069ece331d0fc88feb833e95212b4e9c8bac2ed224d2de4408407701c144158eb68fa180492bbb34e0d54b5c3ac'
+        '6748c004b274b187c3f3a8c0ab9548fd09dc4c962a7e59f74c4ce0181fe1dc1ff8bf0adb07688e4bb8776e21d63aa06b1494338ac6a9fcb2db5df881ef38135c'
+        '552afc1c296991adb14c46e56a96d40a01688aeb1e12f2e294bd18b9316ceebd527a16b4ab08a82c0aa0a4cd13517f6a2f4534c25627d375b96958dfac66f034'
+        '22cd5b8697894d8d4515f89e4053c05f99a6cb6de59b23aabc3f98fc4f80b4b1ad075399925ac67c81bbd5da5f18384d09171f778c8f4335357798965cafe6f1'
+        'a450e64f3ce9551e4c40fcce47d0054722511dbda2dd7132dcd1c8a7c5a21ebab46433ced3f030fb0ab84c58688208cc70a3b0c227cf2015d9a243abe47c61b4'
+        '36cdf842c02090ac7631d2fac6bc86960cdd51cb03d2b263e0f9a9aabbca118e4cff5ca12af20ea12460f413840b891e1d04119f63c3af59ad68a7b4e0ef4d83'
+        '75d3f910b61c852dcb4c3d4ade69dc0746e2e29c790fca31c38ba06afc5858f42ed6aed4f352a594892ff06555df598607f7a9693060dfa70947fe8ad4af8459'
+        '4027ad446a6da4b94f15a9928c8a7345022a647c240baefb61b523d90ba60a03e3ca7db9045aaa2b5a8fa91d37752852fd7f32c211d81ae1d1854159eb203fe6')
 b2sums_x86_64=('47588475723b353556168d59ef79a6b88941906a8e882074b48864ad11b92b0c299c0123672bd6b552d0800df9e3319d8bc9783d09b9182d210b6d34a83c04db')
+b2sums_riscv64=('47588475723b353556168d59ef79a6b88941906a8e882074b48864ad11b92b0c299c0123672bd6b552d0800df9e3319d8bc9783d09b9182d210b6d34a83c04db'
+                'cb4e1ffcec15404ec4dab21f784f861a13feb0b19e21b9fbbeab1a99e1e164e5e0a923c84034867bd7643eba88e836a242a45809045e8a1dd24179f087e5b360')
 validpgpkeys=(
   647F28654894E3BD457199BE38DBBDC86092693E  # Greg Kroah-Hartman <gregkh@kernel.org>
   64254695FFF0AA4466CC19E67B96E8162A8CF5D1  # Sebastian Andrzej Siewior
@@ -102,10 +134,14 @@
   done
 
   echo "Setting config..."
-  cp ../config.$CARCH .config
+  cp ../config.x86_64 .config
+  make olddefconfig
+
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  make mrproper
+  mv .config.1 .config
   make olddefconfig
-  # make nconfig
-  diff -u ../config.$CARCH .config || :
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -117,7 +153,8 @@
   make htmldocs SPHINXOPTS=-QT &
   local pid_docs=$!
 
-  make all
+  # Generate symbols for dtbs to support overlays
+  make DTC_FLAGS="-@" all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
   wait $pid_docs
 }
@@ -192,6 +229,9 @@
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  make INSTALL_DTBS_PATH="$modulesdir/dtb" dtbs_install
+
   # remove build link
   rm "$modulesdir"/build
 
@@ -266,6 +306,7 @@
   local karch
   case $CARCH in
     x86_64) karch=x86 ;;
+    riscv64) karch=riscv ;;
     *) echo "Unknown CARCH $CARCH"; exit 1 ;;
   esac
 
