--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,7 @@ sha256sums=('d79c12eae1460803a1ce8b440ae213dc4df63a6f2bf39ebd49eea1d7a008bec6'
 
 prepare() {
     ln -sf "../${pkgname}-${pkgver}.Cargo.lock" "${pkgname}-${pkgver}/Cargo.lock"
-    cargo fetch --locked --target "${CARCH}-unknown-linux-gnu" --manifest-path="${pkgname}-${pkgver}/Cargo.toml"
+    cargo fetch --locked --manifest-path="${pkgname}-${pkgver}/Cargo.toml"
 }
 
 build() {
