diff --git PKGBUILD PKGBUILD
index c120756..204217a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -31,4 +31,12 @@
   install -Dm644 LICENSE.txt "$pkgdir"/usr/share/licenses/${pkgname}/LICENSE.txt
 }
 
+source+=("qaseprite-laf-riscv64.patch::https://github.com/aseprite/laf/pull/129.diff")
+sha256sums+=("2de4f8918f6e044d4e5de36a3bfb568318e59897b4d1f125fb35604993b9675f")
+
+prepare() {
+  cd "$srcdir/$pkgname-$pkgver"
+  patch -d aseprite/laf -Np1 -i "$srcdir/qaseprite-laf-riscv64.patch"
+}
+
 # vim:set ts=2 sw=2 et:
