--- PKGBUILD
+++ PKGBUILD
@@ -7,7 +7,7 @@ arch=('x86_64')
 url='https://rocm.docs.amd.com/projects/HIPIFY/en/latest'
 license=('MIT')
 depends=('glibc' 'gcc-libs' 'ncurses' 'bash' 'perl' 'zstd' 'zlib'
-         'rocm-llvm' 'cuda')
+         'rocm-llvm')
 makedepends=('cmake')
 _git='https://github.com/ROCm-Developer-Tools/HIPIFY'
 source=("${pkgname}-${pkgver}.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
