--- PKGBUILD
+++ PKGBUILD
@@ -18,14 +18,16 @@ source=(https://downloads.sourceforge.net/sourceforge/sc2/uqm-${pkgver}-src.tgz
         config.state
         uqm
         uqm.desktop
-        uqm.png)
+        uqm.png
+        0000-link-math.patch)
 noextract=(uqm-${pkgver}-content.uqm)
 md5sums=('63c4c300736aa3ac9b841e31b9a76389'
          '6cbc9d51fa63e07c0f4d1d061136d816'
          'ca497b754edf56aab8a5791f1d2c974a'
          '07e64632fce6323a14ab558d0214b885'
          '44c83b9b9d102b99210af0036749a318'
-         '06d0a72aa18e7edce6cd3c7a3a747c44')
+         '06d0a72aa18e7edce6cd3c7a3a747c44'
+         '78d55e6368748369fdddf478ad6a4556')
 
 prepare() {
   cd $pkgname-$pkgver
@@ -34,6 +36,8 @@ prepare() {
       -i config.state    
   ./build.sh uqm reprocess_config
 
+  patch -p1 -i ../0000-link-math.patch
+
   # fix build (Fedora)
   cd src/uqm/comm/
   for d in */ ; do
