--- a/third_party/swiftshader/src/Reactor/BUILD.gn	2022-06-21 22:03:49.111995951 +0200
+++ b/third_party/swiftshader/src/Reactor/BUILD.gn	2022-06-24 14:57:25.017656238 +0200
@@ -308,11 +263,9 @@
 
 if (supports_llvm) {
   swiftshader_source_set("swiftshader_llvm_reactor") {
-    llvm_dir = "../../third_party/llvm-10.0"
 
     deps = [
       ":swiftshader_reactor_base",
-      "$llvm_dir:swiftshader_llvm",
     ]
 
     sources = [
@@ -323,24 +276,7 @@
 
     configs = [ ":swiftshader_llvm_reactor_private_config" ]
 
-    include_dirs = [
-      "$llvm_dir/llvm/include/",
-      "$llvm_dir/configs/common/include/",
-    ]
-
-    if (is_linux || is_chromeos) {
-      include_dirs += [ "$llvm_dir/configs/linux/include/" ]
-    } else if (is_fuchsia) {
-      include_dirs += [ "$llvm_dir/configs/fuchsia/include/" ]
-    } else if (is_win) {
-      include_dirs += [ "$llvm_dir/configs/windows/include/" ]
-    } else if (is_android) {
-      include_dirs += [ "$llvm_dir/configs/android/include/" ]
-    } else if (is_mac) {
-      include_dirs += [ "$llvm_dir/configs/darwin/include/" ]
-    } else {
-      assert(false, "llvm not configured for target platform")
-    }
+    libs = ["LLVM"]
   }
 }
 
