--- PKGBUILD
+++ PKGBUILD
@@ -41,9 +41,35 @@
   "crystal-${pkgver}.tar.gz::https://github.com/crystal-lang/crystal/archive/${pkgver}.tar.gz"
 )
 sha256sums=('5e2d69f565553aa7287e76570d4540a231bfdf1a37f3be179f497ef1df8a0d9a')
+source_riscv64=(
+  'crystal-riscv64.patch'
+  'crystal-cpu-count.patch'
+  'crystal-thread-wait.patch::https://github.com/crystal-lang/crystal/commit/83eb03ae58bc812c8d86c95953622b870f1a034f.patch'
+  'crystal-once-bare-threads.patch::https://github.com/crystal-lang/crystal/commit/9d70354f21030227f208be20238f62b7a6b36de5.patch'
+  'crystal-thread-wait-eintr.patch::https://github.com/crystal-lang/crystal/commit/d1298fe366acb90da86398dcb9d94a7866cc5109.patch'
+)
+sha256sums_riscv64=(
+  'be4fb34992048ab81a6bcb92244c7fda5d1b7ed374fac9a70fc218e148eb35a4'
+  '80d977acb3f994eee1d8bdfb84878f4e1aa1887c2bcd5dea29bc897976679639'
+  'f21ad170e837d7de9b4c455efeda33d82c3f4f0746b4a469d84c637fbc90d541'
+  '73aceafaffe6232208504e0de5f5eb3f590a201ffc993e63df40c2dbf3d4d335'
+  '38d4b07466849d13157e9d8037d92bc37ac182d7749c48a4faecf8d47ffbcb81'
+)
 
 prepare() {
   cd "${pkgbase}-${pkgver}"
+
+  if [[ $CARCH == riscv64 ]]; then
+    patch -Np1 -i ../crystal-riscv64.patch
+    patch -Np1 -i ../crystal-cpu-count.patch
+    ln -s ../../linux src/lib_c/riscv64-linux-gnu/c/linux
+
+    # Constant initialization must also support the execution-context monitor's bare thread.
+    patch -Np1 -i ../crystal-thread-wait.patch
+    patch -Np1 -i ../crystal-once-bare-threads.patch
+    patch -Np1 -i ../crystal-thread-wait-eintr.patch
+    cp src/lib_c/aarch64-linux-gnu/c/semaphore.cr src/lib_c/riscv64-linux-gnu/c/semaphore.cr
+  fi
 }
 
 build() {
