--- PKGBUILD
+++ PKGBUILD
@@ -20,17 +20,22 @@ depends=(
 makedepends=(
   git
   cargo
+  cmake
+  clang
 )
 source=("git+https://github.com/ProtonVPN/local-agent-rs.git#tag=${pkgver}")
 sha256sums=('66224b7958f4532da9ae0962edfb8a07181b8db5f8ed40c90db110d65a538f7b')
 
 prepare() {
     cd "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent
+    cargo update -p bindgen
     cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
 build() {
     cd "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent
+    export CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
+    export CXXFLAGS=${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
     cargo build --frozen --release --all-features
 }
 
