--- PKGBUILD
+++ PKGBUILD
@@ -20,17 +20,24 @@
 makedepends=(
   git
   cargo
+  cmake
+  clang
 )
 source=("git+https://github.com/ProtonVPN/local-agent-rs.git#tag=${pkgver}")
 sha256sums=('9fb13cf0362c47915036825d3249ea4915eb76149ccab4f2dc765888655deb05')
 
 prepare() {
     cd "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent
+    cargo update -p bindgen
     cargo fetch --locked --target "$(rustc --print host-tuple)"
 }
 
 build() {
     cd "${srcdir}"/local-agent-rs/python-proton-vpn-local-agent
+    export CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
+    export CXXFLAGS=${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=3/}
+    export CFLAGS+=" -Wno-error=unterminated-string-initialization -Wno-error=discarded-qualifiers"
+    export CMAKE_POLICY_VERSION_MINIMUM=3.5
     cargo build --frozen --release --all-features
 }
 
