diff --git PKGBUILD PKGBUILD
index 3ac7331..8176c05 100644
--- PKGBUILD
+++ PKGBUILD
@@ -36,6 +36,7 @@
   git submodule init
   git config submodule.ruff.url ../ruff
   git -c protocol.file.allow=always submodule update
+  patch -d ruff -Np1 -i "$srcdir/$pkgname-symlink-inside-project-index.patch"
   cargo fetch --manifest-path ruff/crates/$pkgname/Cargo.toml --locked --target host-tuple
 }
 
@@ -80,3 +81,6 @@
   install -Dm644 nushell-completions "$pkgdir"/usr/share/nushell/vendor/autoload/$pkgname.nu
   install -Dm644 zsh-completions "$pkgdir"/usr/share/zsh/site-functions/_$pkgname
 }
+
+source_riscv64=("$pkgname-symlink-inside-project-index.patch::https://github.com/astral-sh/ruff/pull/28245.patch?full_index=1")
+b2sums_riscv64=('bf59b02951ff99ecbae84791fa6e599e8b45639bcbf59b09aa76ac0b689b5bf7bfc26620a096ab8783bf249e66ca17723752e2a144f53535f83b67c07a60bad5')
