Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 457315)
+++ PKGBUILD	(working copy)
@@ -20,7 +20,8 @@
         'dhcp-tmpfiles.conf'
         'dhcpd4.service'
         'dhcpd6.service'
-        'dhclient@.service')
+        'dhclient@.service'
+        'autoreconf-bind.patch')
 sha256sums=('0ac416bb55997ca8632174fd10737fd61cdb8dba2752160a335775bc21dc73c7'
             'SKIP'
             '559ed9ce6318f1a3700f8b879552bd22844c7c5b96622202dbf3164cd99c444e'
@@ -28,12 +29,16 @@
             'abcd30e9e8428e34d22ab4d3074ef4bd84c2b11f5868597111b47d6f56d204da'
             '03fce30efab819b2d928085b0bab962a33ce56fc376acae98ad9b30aa278c9c8'
             'f98a4438f4f69cab7cc5cce6927df4790ee993ebc8f88a169e63043c53d25625'
-            '86cd0b1e0ea1d47ab096f6ee925eee60545116fb887a155761eda589b30e4f0e')
+            '86cd0b1e0ea1d47ab096f6ee925eee60545116fb887a155761eda589b30e4f0e'
+            '6d133c27979a8358b391ec243b050f15ba1dd85f2a645357953373083c874bda')
 
 prepare() {
   cd "${srcdir}/${pkgbase}-${_realver}"
 
   patch -Np1 < ../0001-99-replace-hostname-with-uname-n.patch
+  patch -Np1 < ../autoreconf-bind.patch
+
+  autoreconf -fiv
 }
 
 build() {
