diff --git PKGBUILD PKGBUILD
index 28665eda0..53debad37 100644
--- PKGBUILD
+++ PKGBUILD
@@ -28,6 +28,12 @@
             'f98a4438f4f69cab7cc5cce6927df4790ee993ebc8f88a169e63043c53d25625'
             '86cd0b1e0ea1d47ab096f6ee925eee60545116fb887a155761eda589b30e4f0e')
 
+prepare() {
+  cd "${srcdir}/${pkgbase}-${_realver}/bind"
+  gunzip -c bind.tar.gz | tar xf -
+  cp /usr/share/libtool/build-aux/config.guess ./bind-*/
+}
+
 build() {
   cd "${srcdir}/${pkgbase}-${_realver}"
 
