diff --git PKGBUILD PKGBUILD
index 4b48591..3a5ac09 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,7 @@ url="https://gitlab.dkrz.de/k202009/libaec"
 license=(BSD-2-Clause)
 depends=(glibc)
 makedepends=(cmake)
-source=("${url}/archive/v${pkgver}/${pkgname}-v${pkgver}.tar.bz2")
+source=("${url}/-/archive/v${pkgver}/${pkgname}-v${pkgver}.tar.bz2")
 sha256sums=('cf869c166656a83857adf62a092311a0069855c6ced3446e3f090a6d52279f65')
 
 build() {
