diff --git PKGBUILD PKGBUILD
index 27837cf..9401a24 100644
--- PKGBUILD
+++ PKGBUILD
@@ -43,7 +43,10 @@ prepare() {
 
 build() {
   cd $pkgname-$pkgver
-  ./configure --prefix=/usr \
+  ./configure \
+    --build=riscv64-unknown-linux-gnu \
+    --host=riscv64-unknown-linux-gnu \
+    --prefix=/usr \
     --libexecdir=/usr/lib/telepathy \
     --disable-Werror --enable-gtk-doc
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
