--- PKGBUILD
+++ PKGBUILD
@@ -23,6 +23,9 @@ prepare() {
   cd "${pkgname}"-${pkgver}
   # Use gnutls_certificate_verify_peers2 to verify server - FS#80239
   patch -Np1 -i ../lftp-4.9.2-gnutls-peers2.patch
+  # since config.guess and config.sub are way too old for supporting riscv64, we update them here
+  find . -name config.guess -exec cp -f /usr/share/libtool/build-aux/config.guess {} \;
+  find . -name config.sub -exec cp -f /usr/share/libtool/build-aux/config.sub {} \;
 }
 
 build() {
