--- PKGBUILD
+++ PKGBUILD
@@ -28,10 +28,11 @@ build() {
         -DCMAKE_BUILD_TYPE='None' \
         -DCMAKE_INSTALL_PREFIX=/usr \
         -DNCNN_SHARED_LIB=ON \
-        -DNCNN_ENABLE_LTO=ON \
+        -DNCNN_ENABLE_LTO=OFF \
         -DNCNN_VULKAN=ON \
         -DNCNN_SYSTEM_GLSLANG=ON \
         -DNCNN_BUILD_EXAMPLES=OFF \
+        -DNCNN_XTHEADVECTOR=OFF \
         -DGLSLANG_TARGET_DIR=/usr/lib/cmake \
         -Wno-dev
     ninja -C build
@@ -41,3 +42,5 @@ package() {
     DESTDIR="$pkgdir" ninja -C build install
     install -Dm644 "$srcdir"/ncnn/LICENSE.txt -t "$pkgdir"/usr/share/licenses/$pkgname/
 }
+
+options+=(!lto)
