--- a/tests/test_basic_logic.py
+++ b/tests/test_basic_logic.py
@@ -856,7 +856,7 @@
         assert cookie_fields == expected_cookie_headers
 
     @given(frame_size=integers(min_value=2**14, max_value=(2**24 - 1)))
-    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+    @settings(deadline=None, suppress_health_check=[HealthCheck.function_scoped_fixture])
     def test_changing_max_frame_size(self, frame_factory, frame_size) -> None:
         """
         When the user changes the max frame size and the change is ACKed, the
