diff --git PKGBUILD PKGBUILD
index db4ea46..556ff79 100644
--- PKGBUILD
+++ PKGBUILD
@@ -48,10 +48,10 @@ optdepends=(
     "xorg-xrandr: xrandr: -G (X) monitors(s) resolution; -Ga monitor data"
 )
 source=("$pkgname-$pkgver.tar.gz::https://codeberg.org/smxi/$pkgname/archive/${_pkgver}.tar.gz")
-b2sums=('cfcafc643e7e18261fac622df3614e7ad296ec2233ebf6750243d9490e0ca5e8ba3742286a0630ae948d16ac4d09705ffc1b20df666f48ed2c5331a8320919d7')
+b2sums=('864258f35f50ed6d5e781c3ed6df5b11dc405ef6f4bf4e845f5c7dcf859ddd1f65eed0591d6295ae3ac24b3b43ac00d54c6b4e9fd5abef295a39d43b16ebb050')
 
 package() {
-    cd "$pkgname-${_pkgver}"
+    cd "$pkgname"
 
     install -Dm 00755 $pkgname "${pkgdir}/usr/bin/$pkgname"
     install -Dm 00644 $pkgname.1 "${pkgdir}/usr/share/man/man1/$pkgname.1"
