--- a/src/runtime/asm_riscv64.s
+++ b/src/runtime/asm_riscv64.s
@@ -118,11 +118,7 @@
 
 // func cputicks() int64
 TEXT runtime·cputicks<ABIInternal>(SB),NOSPLIT,$0-0
-	// RDTIME to emulate cpu ticks
-	// RDCYCLE reads counter that is per HART(core) based
-	// according to the riscv manual, see issue 46737
-	RDTIME	X10
-	RET
+	JMP	runtime·nanotime<ABIInternal>(SB)
 
 // systemstack_switch is a dummy routine that systemstack leaves at the bottom
 // of the G stack. We need to distinguish the routine that
