Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010008)
+++ PKGBUILD	(working copy)
@@ -29,7 +29,7 @@
             'SKIP')
 
 build() {
-  export GOARCH=amd64
+  export GOARCH=riscv64
   export GOROOT_FINAL=/usr/lib/go
   export GOROOT_BOOTSTRAP=/usr/lib/go
   export GOPATH="$srcdir/"
@@ -43,7 +43,7 @@
 }
 
 check() {
-  export GOARCH=amd64
+  export GOARCH=riscv64
   export GOROOT_FINAL=/usr/lib/go
   export GOROOT_BOOTSTRAP=/usr/lib/go
   export GOROOT="$srcdir/$pkgname"
@@ -59,7 +59,7 @@
   cd "$pkgname"
 
   install -d "$pkgdir/usr/bin" "$pkgdir/usr/lib/go" "$pkgdir/usr/share/doc/go" \
-    "$pkgdir/usr/lib/go/pkg/linux_amd64_"{dynlink,race}
+    "$pkgdir/usr/lib/go/pkg/linux_riscv64_"{dynlink,race}
 
   # TODO: Make split package for source files?
   # This saves around 100 MB.
