diff --git PKGBUILD PKGBUILD
index 4f2645b4..61fc88d9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,8 +30,7 @@
             'SKIP')
 
 build() {
-  export GOARCH=amd64
-  export GOAMD64=v1 # make sure we're building for the right x86-64 version
+  export GOARCH=riscv64
   export GOROOT_FINAL=/usr/lib/go
   export GOROOT_BOOTSTRAP=/usr/lib/go
 
@@ -50,7 +49,7 @@
   cd "$pkgname"
 
   install -d "$pkgdir/usr/bin" "$pkgdir/usr/lib/go" "$pkgdir/usr/share/doc/go" \
-    "$pkgdir/usr/lib/go/pkg/linux_amd64_"{dynlink,race}
+    "$pkgdir/usr/lib/go/pkg/linux_riscv64_"{dynlink,race}
 
   cp -a bin pkg src lib misc api test "$pkgdir/usr/lib/go"
   cp -r doc/* "$pkgdir/usr/share/doc/go"
