Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010008)
+++ PKGBUILD	(working copy)
@@ -19,19 +19,21 @@
 arch=(x86_64)
 url='https://golang.org/'
 license=(BSD)
-makedepends=(git go perl)
+makedepends=(git perl)
 replaces=(go-pie)
 provides=(go-pie)
 options=(!strip staticlibs)
-source=(https://storage.googleapis.com/golang/go$pkgver.src.tar.gz{,.asc})
+source=(https://storage.googleapis.com/golang/go$pkgver.src.tar.gz{,.asc}
+        go-linux-riscv64-bootstrap.tbz)
 validpgpkeys=('EB4C1BFD4F042F6DDDCCEC917721F63BD38B4796')
 sha256sums=('3a70e5055509f347c0fb831ca07a2bf3b531068f349b14a3c652e9b5b67beb5d'
-            'SKIP')
+            'SKIP'
+            'ef8a274da3ffb37601d7d3486945e0439a493ef72f0159459edd29934a793c4a')
 
 build() {
-  export GOARCH=amd64
+  export GOARCH=riscv64
   export GOROOT_FINAL=/usr/lib/go
-  export GOROOT_BOOTSTRAP=/usr/lib/go
+  export GOROOT_BOOTSTRAP="$srcdir/go-linux-riscv64-bootstrap"
   export GOPATH="$srcdir/"
   export GOROOT="$srcdir/$pkgname"
   export GOBIN="$GOROOT/bin"
@@ -43,9 +45,9 @@
 }
 
 check() {
-  export GOARCH=amd64
+  export GOARCH=riscv64
   export GOROOT_FINAL=/usr/lib/go
-  export GOROOT_BOOTSTRAP=/usr/lib/go
+  export GOROOT_BOOTSTRAP="$srcdir/go-linux-riscv64-bootstrap"
   export GOROOT="$srcdir/$pkgname"
   export GOBIN="$GOROOT/bin"
   export PATH="$srcdir/$pkgname/bin:$PATH"
@@ -59,7 +61,7 @@
   cd "$pkgname"
 
   install -d "$pkgdir/usr/bin" "$pkgdir/usr/lib/go" "$pkgdir/usr/share/doc/go" \
-    "$pkgdir/usr/lib/go/pkg/linux_amd64_"{dynlink,race}
+    "$pkgdir/usr/lib/go/pkg/linux_riscv64_"{dynlink,race}
 
   # TODO: Make split package for source files?
   # This saves around 100 MB.
