diff --git PKGBUILD PKGBUILD
index 7efb0405..ed18ec30 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ arch=('x86_64')
 url="https://gitlab.gnome.org/GNOME/filemanager-actions"
 license=('GPL')
 depends=('gtk3' 'libgtop')
-makedepends=('intltool' 'caja' 'libnautilus-extension' 'nemo' 'yelp-tools')
+makedepends=('gtk-doc' 'intltool' 'caja' 'libnautilus-extension' 'nemo' 'yelp-tools')
 optdepends=('caja: supported file manager'
             'nautilus: supported file manager'
             'nemo: supported file manager')
