diff --git PKGBUILD PKGBUILD
index 2f0a204..b00fb5a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -38,3 +38,18 @@
 
   install -Dm 644 LICENSE.txt "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 }
+
+source+=("$pkgname-numpy-2.5-row-stack.patch::https://github.com/pyqtgraph/pyqtgraph/commit/0c23cbe65ccddc36e377695396854b4fc28e44dd.patch?full_index=1"
+         "$pkgname-qt6-svg-closepath.patch::https://github.com/pyqtgraph/pyqtgraph/commit/1b84c2b2d5b6f22b41c91733bc4ace808d14998c.patch?full_index=1"
+         "$pkgname-makeargb-nan-reference.patch::https://github.com/pyqtgraph/pyqtgraph/commit/a24a0b6f2bde199913180feac26cadc32da4a5bd.patch?full_index=1")
+sha512sums+=('9a408f70dd55daea306d1a3f6bee2cbc4f776e2fbc6c175bd3404e3a272f9d6f8402fcb75d001ab5c2242ee28c8182dfa7b8aa9e8ed3b22bbd58462e9f5caf8b'
+             '373d57a90e74c53a59e8d946ef93280b3886d4d802b0473c4c2a9bcebb43b53f7d943fdf7ad560b14546f93a5d181d897f254df18fd4515a8c0e47240b0f2b67'
+             '0854fd17fbdb178c0a6c01d97ce26961ecd3b56d4f8cba1ef7eb4e0847946479bc1b458c50eed1b6be8151fd8200dc9bf0f8bbbfe2515dd2fa605dd45c676c87')
+
+prepare() {
+  cd $_pkgname-$_pkgname-$pkgver
+
+  patch -Np1 -i ../$pkgname-numpy-2.5-row-stack.patch
+  patch -Np1 -i ../$pkgname-qt6-svg-closepath.patch
+  patch -Np1 -i ../$pkgname-makeargb-nan-reference.patch
+}
