diff --git PKGBUILD PKGBUILD
index 4764afc4..be95fbe4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,15 @@
 source=(https://github.com/nzbget/$pkgname/releases/download/v$pkgver/$pkgname-$pkgver-src.tar.gz)
 sha256sums=('4e8fc1beb80dc2af2d6a36a33a33f44dedddd4486002c644f4c4793043072025')
 
+prepare() {
+  # https://github.com/nzbget/nzbget/issues/755
+  # https://github.com/nzbget/nzbget/pull/765
+  cd ${pkgname}-${pkgver}
+  _sharedir=/usr/share/autoconf/build-aux
+  cp ${_sharedir}/config.guess posix
+  cp ${_sharedir}/config.sub posix
+}
+
 build() {
   cd $pkgname-$pkgver
 
