--- PKGBUILD
+++ PKGBUILD
@@ -37,14 +37,22 @@ source=(
 sha256sums=('d8a26fef9f92d63258251c69af01f39073a479e48c14114dc96d285470312c83'
             'SKIP'
             '862cf5a0e04274f141d308a5ce3c8c52baff3804e7d34f153202e7d3fc5a32e4')
+options=(!lto)
 
 prepare() {
   cd $pkgname-$pkgver
   patch -Np1 -i ../use-par2cmdline-turbo-from-sources.patch
+  
+  # https://github.com/nzbget/nzbget/issues/755
+  # https://github.com/nzbget/nzbget/pull/765
+  _sharedir=/usr/share/autoconf/build-aux
+  cp ${_sharedir}/config.guess posix
+  cp ${_sharedir}/config.sub posix
 }
 
 build() {
   cd $pkgname-$pkgver
+  export CFLAGS+=" -fno-lto" CXXFLAGS+=" -fno-lto"
   cmake -S . -B build \
     -DCMAKE_BUILD_TYPE=None \
     -DCMAKE_INSTALL_PREFIX=/usr \
