--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,11 @@ prepare() {
   cd $pkgname-$pkgver
   # https://github.com/nzbget/nzbget/pull/793
   patch -Np1 -i ../nzbget_21.1_openssl3_update.patch
+  # https://github.com/nzbget/nzbget/issues/755
+  # https://github.com/nzbget/nzbget/pull/765
+  _sharedir=/usr/share/autoconf/build-aux
+  cp ${_sharedir}/config.guess posix
+  cp ${_sharedir}/config.sub posix
 }
 
 build() {
