diff --git PKGBUILD PKGBUILD
index fc99427..135fffb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,6 +9,7 @@ pkgrel=2
 arch=('x86_64')
 license=('GPL')
 url='https://mariadb.org/'
+options=(!lto)
 makedepends=('boost' 'bzip2' 'cmake' 'jemalloc' 'libaio' 'libxcrypt'
              'libxml2' 'lz4' 'lzo' 'openssl' 'systemd' 'zlib' 'zstd' 'curl'
              'krb5' 'cracklib' 'judy')
@@ -85,6 +86,7 @@ build() {
     -DWITH_SYSTEMD=yes
     -DWITH_UNIT_TESTS=OFF
     -DWITH_ZLIB=system
+    -DCMAKE_CXX_STANDARD_LIBRARIES=-latomic
   )
 
   mkdir build
