diff --git a/Cargo.toml b/Cargo.toml
index 2ea0d43..43ec27e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,3 +41,6 @@ open = "1.7.0"
 clap = "2.33"
 diffus = { version = "0.10", features = [ "derive" ] }
 xmltree = "0.10"
+
+[profile.release]
+codegen-units=1
diff --git a/data/meson.build b/data/meson.build
index fe417fc..729c4f7 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -2,7 +2,7 @@ subdir('icons')

 desktop_conf = configuration_data()
 desktop_conf.set('icon', application_id)
-desktop_file = i18n.merge_file ('desktop-file',
+desktop_file = i18n.merge_file (
   type: 'desktop',
   input: configure_file(
     input: files('com.gitlab.newsflash.desktop.in.in'),
@@ -27,7 +27,7 @@ endif

 appdata_conf = configuration_data()
 appdata_conf.set('appid', application_id)
-appdata_file = i18n.merge_file ('appdata-file',
+appdata_file = i18n.merge_file (
   input: configure_file(
     input: files('com.gitlab.newsflash.appdata.xml.in.in'),
     output: '@BASENAME@',
