--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,12 @@ source=("https://gitlab.com/news-flash/$_pkgname/-/archive/v.$pkgver/$_pkgname-v
 sha256sums=('a13f0bbc32c59a36b039737a374728fcacbacce1b3b4a1fa560e4eb733680511')
 options=(!lto)
 
+prepare() {
+  cd $_pkgname-v.$pkgver
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   export FEEDLY_CLIENT_ID="boutroue"
   export FEEDLY_CLIENT_SECRET="FE012EGICU4ZOBDRBEOVAJA1JZYH"
