--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,16 @@ url='https://gitlab.com/news-flash/news_flash_gtk'
 license=('GPL3')
 depends=('libhandy' 'webkit2gtk')
 makedepends=('appstream-glib' 'meson' 'rust')
-source=("https://gitlab.com/news-flash/$_pkgname/-/archive/$pkgver/$_pkgname-$pkgver.tar.gz")
-sha256sums=('099318a6bc57e5e695352b2f3c7dd149c9b889f39cb9cbd58dc3f10f41d4948d')
+source=("https://gitlab.com/news-flash/$_pkgname/-/archive/$pkgver/$_pkgname-$pkgver.tar.gz"
+        "fix-meson-build-error.patch")
+sha256sums=('099318a6bc57e5e695352b2f3c7dd149c9b889f39cb9cbd58dc3f10f41d4948d'
+            '1568b8be45513d7af6da12ab8d3af2df4d0dd99f0568eb0bc99e696156aed04e')
 
 prepare() {
   cd $_pkgname-$pkgver
+  # 1. fix config errors in meson.build (this package is outdated, and upstream had fixed this issue)
+  # 2. circumvent a toolchain bug (see https://github.com/rust-lang/cargo/issues/6489)
+  patch -Np1 -i ../fix-meson-build-error.patch
   # Update open to avoid xdg-utils dependency
   sed -i 's/open = "1.4.0"/open = "1.7.0"/' Cargo.toml
 }
