diff --git a/zellij-server/Cargo.toml b/zellij-server/Cargo.toml
index a0a5e07..244a199 100644
--- a/zellij-server/Cargo.toml
+++ b/zellij-server/Cargo.toml
@@ -18,8 +18,8 @@ daemonize = "0.4.1"
 serde_json = "1.0"
 unicode-width = "0.1.8"
 url = "2.2.2"
-wasmer = "1.0.0"
-wasmer-wasi = "1.0.0"
+wasmer = { path = "../../wasmer/lib/api/", version = "1.0.0" }
+wasmer-wasi = { path = "../../wasmer/lib/wasi/", version = "1.0.0" }
 cassowary = "0.3.0"
 zellij-utils = { path = "../zellij-utils/", version = "0.24.0" }
 zellij-tile = { path = "../zellij-tile/", version = "0.24.0" }
