--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,7 @@ backup=('etc/glusterfs/glusterd.vol'
         'etc/glusterfs/glusterd.vol'
         'etc/glusterfs/glusterfs-georep-logrotate'
         'etc/glusterfs/glusterfs-logrotate')
-depends=(fuse python libxml2 libaio liburcu attr rpcbind liburing gperftools)
+depends=(fuse python libxml2 libaio liburcu attr rpcbind liburing)
 makedepends=(rpcsvc-proto)
 optdepends=('glib2: qemu-block'
 	    'python-prettytable: gluster-georep-sshkey')
@@ -46,6 +46,7 @@ build() {
     --libexecdir=/usr/lib/$pkgname \
     --with-systemddir=/usr/lib/systemd/system \
     --with-tmpfilesdir=/usr/lib/tmpfiles.d \
+    --without-tcmalloc \
     --enable-gnfs \
     LEXLIB=
   make
