diff --unified --recursive --text a/src/lp_cdd.cpp b/src/lp_cdd.cpp
--- a/src/lp_cdd.cpp	2021-10-23 08:12:18.124150947 +0000
+++ b/src/lp_cdd.cpp	2021-10-23 08:13:18.134449619 +0000
@@ -1,9 +1,9 @@
 #include "lp_cdd.h"
 //extern "C"{
 #ifdef NOCDDPREFIX
-#include "setoper.h"
-#include "cdd.h"
-#include "cdd_f.h"
+#include "cddlib/setoper.h"
+#include "cddlib/cdd.h"
+#include "cddlib/cdd_f.h"
 #else
 #include "cdd/setoper.h"
 #include "cdd/cdd.h"
diff --unified --recursive --text a/src/gfanlib_zcone.cpp b/src/gfanlib_zcone.cpp
--- a/src/gfanlib_zcone.cpp	2021-10-23 08:19:14.359583144 +0000
+++ b/src/gfanlib_zcone.cpp	2021-10-23 08:19:47.809753950 +0000
@@ -13,8 +13,8 @@
 
 //extern "C"{
 #ifdef NOCDDPREFIX
-#include "setoper.h"
-#include "cdd.h"
+#include "cddlib/setoper.h"
+#include "cddlib/cdd.h"
 #else
 #include "cdd/setoper.h"
 #include "cdd/cdd.h"
diff --unified --recursive --text a/src/app_librarytest.cpp b/src/app_librarytest.cpp
--- a/src/app_librarytest.cpp	2021-10-23 08:30:28.723058050 +0000
+++ b/src/app_librarytest.cpp	2021-10-23 08:30:57.679874856 +0000
@@ -9,8 +9,8 @@
 #include "gfanlib.h"
 #include "printer.h"
 #ifdef NOCDDPREFIX
-#include "setoper.h"
-#include "cdd.h"
+#include "cddlib/setoper.h"
+#include "cddlib/cdd.h"
 #else
 #include "cdd/setoper.h"
 #include "cdd/cdd.h"
