diff --git PKGBUILD PKGBUILD
index ef586ebd6..fb7c539b7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,14 +10,17 @@ url="http://home.imf.au.dk/jensen/software/gfan/gfan.html"
 license=(GPL)
 depends=(cddlib)
 source=("http://home.imf.au.dk/jensen/software/gfan/gfan$pkgver.tar.gz"
-         gfan-cflags.patch)
+         gfan-cflags.patch
+         fix-no-such-file.patch)
 sha256sums=('a674d5e5dc43634397de0d55dd5da3c32bd358d05f72b73a50e62c1a1686f10a'
-            'a6130c0b576cc58a86cc1010b2fb3359666ac6c5de9ae2d939877b8f609c9dff')
+            'a6130c0b576cc58a86cc1010b2fb3359666ac6c5de9ae2d939877b8f609c9dff'
+            '826b3bcb812607e1054bbf3f3a7f5c403ed68b51867afb0e359d182283f2eeb1')
 
 prepare() {
   cd gfan$pkgver
 # Honor system build flags
   patch -p1 -i ../gfan-cflags.patch
+  patch -p1 -i ../fix-no-such-file.patch
 }
 
 build() {
