--- PKGBUILD
+++ PKGBUILD
@@ -16,15 +16,20 @@ depends=(
   zstd
 )
 makedepends=(git cmake)
-source=("$pkgname::git+https://github.com/Blosc/c-blosc#tag=v$pkgver")
-sha512sums=('ea5e14ed1ece973e8cb31544cc016ae01b4f59ae7b6e6425c8703f98bda68eff8fc6382fb6e2cb4ec406db22a9229de8148df9a9aca8570f218c431866f709cc')
-b2sums=('e9027864a8a2f67cda8d7f71f22cfd4c99ace91b57ca02c1de274e36c546cad7e9dcb387323ca2924c3daf006f60f1f8a918f1e8c4b5a1cc3b0e248bfd895d0d')
+source=("$pkgname::git+https://github.com/Blosc/c-blosc#tag=v$pkgver"
+	0001-Set-std-gnu99-for-gcc-on-all-architectures.patch)
+sha512sums=('ea5e14ed1ece973e8cb31544cc016ae01b4f59ae7b6e6425c8703f98bda68eff8fc6382fb6e2cb4ec406db22a9229de8148df9a9aca8570f218c431866f709cc'
+            '209bf6f4604f115e4e2ac33527947be3c4cd1c4418b36ca17977dd528c08628448e7cb96b73a6b4a5bc775ec62a36c7717c89112a8e7a0a424082e890bc835ba')
+b2sums=('e9027864a8a2f67cda8d7f71f22cfd4c99ace91b57ca02c1de274e36c546cad7e9dcb387323ca2924c3daf006f60f1f8a918f1e8c4b5a1cc3b0e248bfd895d0d'
+        'd9339c4c20b750d170cdf34107f77823995d8714fda9a994d8152bb331f05de49fe873e92913a454389dfc720dce0c1ee45cc52c7f08e4d03c5a32ba6c46f6e9')
 
 prepare() {
   cd "$pkgname"
 
   # update minimal cmake version
   git cherry-pick --no-commit 051b9d2cb9437e375dead8574f66d80ebce47bee
+
+  patch -Np1 -i ../0001-Set-std-gnu99-for-gcc-on-all-architectures.patch
 }
 
 build() {
