diff --git a/src/goocanvasitemsimple.c b/src/goocanvasitemsimple.c
index 19b3424..b603f93 100644
--- a/src/goocanvasitemsimple.c
+++ b/src/goocanvasitemsimple.c
@@ -1536,7 +1536,7 @@ goo_canvas_item_simple_set_model (GooCanvasItemSimple  *item,
   goo_canvas_item_simple_free_data (item->simple_data);
   g_slice_free (GooCanvasItemSimpleData, item->simple_data);
 
-  item->model = g_object_ref (model);
+  item->model = (GooCanvasItemModelSimple*) g_object_ref (model);
   item->simple_data = &item->model->simple_data;
 
   if (accessibility_enabled)
