diff --git PKGBUILD PKGBUILD
index 48f9d46..2157721 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,8 +19,12 @@ checkdepends=(python-pytest)
 source=(https://github.com/cs01/pygdbmi/archive/v$pkgver/$pkgname-$pkgver.tar.gz)
 sha256sums=('5c52a37211d55eeeaf5c06eacc8e8d49e9dacfd36a526d3534b4a5d5f5ee94d9')
 
+source+=("add-test-timeout.patch::https://github.com/cs01/pygdbmi/pull/103.patch")
+sha256sums+=('2e87f3397521fec15bb1421c9618e65e6c11e9a6d2bd628c53bd89e33930c78b')
+
 build() {
   cd $_pyname-$pkgver
+  patch -Np1 -i ../add-test-timeout.patch
   python -m build --wheel --no-isolation
 }
 
