diff --git PKGBUILD PKGBUILD
index 3b4fe91..1d0e512 100644
--- PKGBUILD
+++ PKGBUILD
@@ -42,6 +42,7 @@
   chmod -R u+w . # WAT award
   patch -Np1 < "${srcdir}/no-makeself.patch"
   patch -p1 < ../wx-3.2-size-warnings.patch # https://github.com/veracrypt/VeraCrypt/issues/952
+  patch -Np2 < ../veracrypt-enable-shani-x86-only.patch
 }
 
 build() {
@@ -60,4 +61,7 @@
   install -Dm 644 License.txt -t "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+source+=(veracrypt-enable-shani-x86-only.patch::https://github.com/veracrypt/VeraCrypt/pull/1479.diff)
+sha512sums+=('6a827d2e934e843ec194f93a3a08e8b081a1df18391957a7b23aa83ae5f088aba13086fe43debd0f6c304c5b92335b882891bca543a3c4220b1e71b1d32424d8')
+
 # vim: ts=2 sw=2 et:
