diff --git PKGBUILD PKGBUILD
index 50d5f4b..608617f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -32,10 +32,11 @@ prepare() {

 build() {
   cd src
+  # https://github.com/veracrypt/VeraCrypt/pull/811
   make PKG_CONFIG_PATH=/usr/lib/pkgconfig \
     WX_CONFIG=/usr/bin/wx-config-gtk3 \
     TC_EXTRA_LFLAGS+="-ldl ${LDFLAGS}" \
-    TC_EXTRA_CXXFLAGS="${CXXFLAGS} ${CPPFLAGS}" \
+    TC_EXTRA_CXXFLAGS="${CXXFLAGS} ${CPPFLAGS} -std=c++14" \
     TC_EXTRA_CFLAGS="${CFLAGS} ${CPPFLAGS}"
 }
