--- PKGBUILD
+++ PKGBUILD
@@ -30,8 +30,10 @@ optdepends=(
   'python-gobject: for nvchecker-notify'
   'libnotify: for nvchecker-notify'
 )
-source=("git+https://github.com/lilydjwg/nvchecker.git?signed#tag=$_tag")
-sha512sums=('SKIP')
+source=("git+https://github.com/lilydjwg/nvchecker.git?signed#tag=$_tag"
+        "bump-test-versions.patch::https://github.com/lilydjwg/nvchecker/pull/244.diff")
+sha512sums=('SKIP'
+            '72c1e7e622075273cefdbcf3d2d0bc2c97e7b0c5da8156e932bd83fc34d2f925c79b94cc296de970dea8ccbdf053522882e34c563831094e31080b9320a845eb')
 validpgpkeys=(
   # No direct trust chain. Some related stuffs:
   # https://api.github.com/users/lilydjwg/gpg_keys
@@ -75,6 +77,8 @@ prepare() {
     git log --oneline -1 "${_c}"
     git cherry-pick -n "${_c}"
   done
+
+  patch -Np1 -i ../bump-test-versions.patch
 }
 
 build() {
