diff --git tests/meson.build tests/meson.build
index 7c6cb11..a6393f9 100644
--- tests/meson.build
+++ tests/meson.build
@@ -79,7 +79,7 @@ installed_tests = {
   },
   'pixbuf-fail': { 'suites': ['conform', 'slow'], },
   'pixbuf-icon-serialize': { 'suites': ['conform'], },
-  'pixbuf-randomly-modified': { 'suites': ['slow'], },
+  'pixbuf-randomly-modified': { 'skip': true, },
   'pixbuf-threads': { 'suites': ['io'], },
   'pixbuf-gif': {
     'suites': ['io'],
