diff --git PKGBUILD PKGBUILD
index f80cf74..01ca8e6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,9 +25,11 @@ _commit=bca00032ad68d0b0aa2c1f7558db931e52bd9cd2  # tags/2.42.8^0
 source=(
   "git+https://gitlab.gnome.org/GNOME/gdk-pixbuf.git#commit=$_commit"
   gdk-pixbuf-query-loaders.hook
+  fix-sigkill.patch
 )
 sha256sums=('SKIP'
-            '9fb71d95e6a212779eb0f88dde5a3cee0df7f4d9183f8f9ce286f8cdc14428f0')
+            '9fb71d95e6a212779eb0f88dde5a3cee0df7f4d9183f8f9ce286f8cdc14428f0'
+            '112abe91db897d8b6220b3f047a6ac38508feb753fbedce9fc49dd256b0106b0')
 
 pkgver() {
   cd gdk-pixbuf
@@ -36,6 +38,7 @@ pkgver() {
 
 prepare() {
   cd gdk-pixbuf
+  patch -p0 -Ni ../fix-sigkill.patch
 }
 
 build() {
