diff --git PKGBUILD PKGBUILD
index 4203cc7..e310611 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,3 +25,10 @@
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+source+=(libultrahdr-riscv64.patch::https://github.com/google/libultrahdr/pull/273.diff)
+sha256sums+=('66e8aa251e812bec660845193497e6fbf635e9b95313ab5040798c82f024b293')
+
+prepare() {
+  patch -d $pkgname -Np1 -i ../libultrahdr-riscv64.patch
+}
