--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,14 @@
 sha512sums=('16795c7673d8661c20181e370b09034125dfc76de670323688c69f15c6016c6619520635593b0733e60a45ec27b824e661c5675fa4a4a2a8a899e9356af88184')
 b2sums=('a83db64f1f521a83920bc93fa976d21352202a8951fd2382a159588f66b126b41a362e0685eeedb6d80105a82a7c8c00b5fd6dcd38be06462e4a74501e5ca290')
 
+prepare() {
+  cd ${pkgname}-${pkgname}-v${pkgver}
+
+  echo -e '\n[patch.crates-io]\nring = { git = "https://github.com/felixonmars/ring", branch = "0.16.20" }' \
+    >> Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   cd ${pkgname}-${pkgname}-v${pkgver}
   export CROSS="true"
