diff --git PKGBUILD PKGBUILD
index d156c16c..73572270 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,12 +14,16 @@
 optdepends=('gpsbabel: import and export GPS tracks from various formats')
 changelog=$pkgname.changelog
 source=(https://downloads.sourceforge.net/$pkgname/$pkgname-$pkgver.tar.bz2
-        https://raw.githubusercontent.com/viking-gps/$pkgname/$pkgname-$pkgver/org.viking.Viking.appdata.xml)
+        https://raw.githubusercontent.com/viking-gps/$pkgname/$pkgname-$pkgver/org.viking.Viking.appdata.xml
+        fix-format-security.patch::https://github.com/viking-gps/viking/commit/a977e3d01f3fd3dad0a3daaccf153d218f8634f2.patch)
 sha256sums=('fe4acf3d24121d6f3097d17ea704b42c0336a8f693151886b8b26e308dbfcb3b'
-            '2a3d60bfed29db9ebc973f02a769c347cb04dafd474fb499448f27de0bbede0f')
+            '2a3d60bfed29db9ebc973f02a769c347cb04dafd474fb499448f27de0bbede0f'
+            'dd8a8b93c5cbdcb8644e11d287d2093fe08d98084080e19578855777704a2c84')
 
 prepare() {
   sed -i 's/org.viking.Viking.desktop/viking.desktop/' org.viking.Viking.appdata.xml
+
+  patch -Np1 -d $pkgname-$pkgver -i ../fix-format-security.patch
 }
 
 build() {
