--- PKGBUILD
+++ PKGBUILD
@@ -5,9 +5,7 @@
 pkgbase=opencv
 pkgname=(opencv
          opencv-samples
-         python-opencv
-         opencv-cuda
-         python-opencv-cuda)
+         python-opencv)
 pkgver=5.0.0
 pkgrel=12
 pkgdesc='Open Source Computer Vision Library'
@@ -42,8 +40,6 @@
          zlib)
 makedepends=(ant
              cmake
-             cuda
-             cudnn
              eigen
              fast_float
              fmt
@@ -146,6 +142,8 @@
     -DBUILD_WITH_DEBUG_INFO=ON
   cmake --build build
 
+  return
+
   # In general, we want to list all real archs (sm_XX) and the latest virtual arch (compute_XX) for future PTX compatibility.
   # Valid values can be discovered from nvcc --help
   local cuda_archs="75;80;86;87;88;89;90;100;103;110;120;121;121-virtual"
