Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -11,9 +11,8 @@
 license=(BSD)
 url='https://opencv.org/'
 depends=(tbb openexr gst-plugins-base libdc1394 cblas lapack libgphoto2 openjpeg2 ffmpeg)
-makedepends=(cmake python-numpy python-setuptools mesa eigen hdf5 lapacke qt5-base vtk glew ant java-environment)
+makedepends=(cmake python-numpy python-setuptools mesa eigen hdf5 lapacke qt5-base glew ant java-environment)
 optdepends=('opencv-samples: samples'
-            'vtk: for the viz module'
             'glew: for the viz module'
             'qt5-base: for the HighGUI module'
             'hdf5: for the HDF5 module'
@@ -48,7 +47,6 @@
     -DINSTALL_PYTHON_EXAMPLES=ON \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCPU_BASELINE_DISABLE=SSE3 \
-    -DCPU_BASELINE_REQUIRE=SSE2 \
     -DOPENCV_EXTRA_MODULES_PATH="$srcdir/opencv_contrib-$pkgver/modules" \
     -DOPENCV_SKIP_PYTHON_LOADER=ON \
     -DOPENCV_PYTHON3_INSTALL_PATH=$_pythonpath \
@@ -94,7 +92,7 @@
 
 package_python-opencv() {
   pkgdesc="Python bindings for OpenCV"
-  depends=(python-numpy opencv vtk glew qt5-base hdf5)
+  depends=(python-numpy opencv glew qt5-base hdf5)
   unset optdepends
 
   DESTDIR="$pkgdir" cmake --install build/modules/python3
