diff --git PKGBUILD PKGBUILD
index b9fa3ca..2dbd559 100644
--- PKGBUILD
+++ PKGBUILD
@@ -5,9 +5,7 @@
 pkgbase=opencv
 pkgname=(opencv
          opencv-samples
-         python-opencv
-         opencv-cuda
-         python-opencv-cuda)
+         python-opencv)
 pkgver=4.11.0
 pkgrel=1
 pkgdesc='Open Source Computer Vision Library'
@@ -40,8 +38,6 @@ depends=(abseil-cpp
          zlib)
 makedepends=(ant
              cmake
-             cuda
-             cudnn
              eigen
              fmt
              gcc13
@@ -100,7 +96,6 @@ build() {
          -DINSTALL_PYTHON_EXAMPLES=ON \
          -DCMAKE_INSTALL_PREFIX=/usr \
          -DCPU_BASELINE_DISABLE=SSE3 \
-         -DCPU_BASELINE_REQUIRE=SSE2 \
          -DOPENCV_EXTRA_MODULES_PATH=$srcdir/opencv_contrib/modules \
          -DOPENCV_SKIP_PYTHON_LOADER=ON \
          -DLAPACK_LIBRARIES=/usr/lib/liblapack.so;/usr/lib/libblas.so;/usr/lib/libcblas.so \
@@ -118,6 +113,8 @@ build() {
     -DBUILD_WITH_DEBUG_INFO=ON
   cmake --build build
 
+  return
+
   CFLAGS="${CFLAGS} -fno-lto" CXXFLAGS="${CXXFLAGS} -fno-lto" LDFLAGS="${LDFLAGS} -fno-lto" \
   cmake -B build-cuda -S $pkgname $_opts \
     -DBUILD_WITH_DEBUG_INFO=OFF \
