Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -48,7 +47,6 @@
     -DINSTALL_PYTHON_EXAMPLES=ON \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCPU_BASELINE_DISABLE=SSE3 \
-    -DCPU_BASELINE_REQUIRE=SSE2 \
     -DOPENCV_EXTRA_MODULES_PATH="$srcdir/opencv_contrib-$pkgver/modules" \
     -DOPENCV_SKIP_PYTHON_LOADER=ON \
     -DOPENCV_PYTHON3_INSTALL_PATH=$_pythonpath \
