--- PKGBUILD
+++ PKGBUILD
@@ -8,12 +8,18 @@ arch=('x86_64')
 url="https://github.com/YS-L/csvlens"
 license=('MIT')
 depends=('gcc-libs')
-makedepends=('cargo')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha256sums=('7d9062fa94a67fec121062bd45efb00a40bf26236f3cc635ec0945793344e097')
+makedepends=('cargo' git)
+source=("$pkgname-$pkgver::git+$url#tag=v$pkgver")
+sha256sums=('dbd32e0bcc3d9200ae2effb438e59994eef994dfc159e9dfad4b07deb52866d6')
 
 prepare() {
   cd "$pkgname-$pkgver"
+
+  # https://github.com/YS-L/csvlens/commit/002edebeda69b0b81b4fd4060e25e80c5a87f4da
+  git cherry-pick -n 002edebeda69b0b81b4fd4060e25e80c5a87f4da
+  # https://github.com/YS-L/csvlens/pull/108/
+  git cherry-pick -n e6d9e1dc6788ec5ac8a981b727e92784c69b4274
+
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
