diff --git PKGBUILD PKGBUILD
index bcd0630..a2b745a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,7 @@ pkgdesc="Contains the code for the PlottableHistogram Protocol, to be used in ty
 arch=('any')
 url='https://github.com/scikit-hep/uhi'
 license=('MIT')
-depends=('python' 'python-numpy' 'root')
+depends=('python' 'python-numpy')
 makedepends=('git' 'python-build' 'python-installer' 'python-hatch-vcs' 'python-hatchling')
 checkdepends=('python-pytest' 'python-boost-histogram')
 source=("${pkgname}-${pkgver}::git+${url}#tag=v${pkgver}")
