--- PKGBUILD
+++ PKGBUILD
@@ -42,11 +42,20 @@ build() {
 check() {
   cd "$pkgname"

+  for failing_test in lto-gcc gnu-unique package-metadata tls-common
+  do
+    rm -vf "test/elf/$failing_test.sh"
+  done
+
   make \
     PREFIX=/usr \
     LTO=1 \
     SYSTEM_MIMALLOC=1 \
     SYSTEM_TBB=1 \
+    TEST_CC=clang \
+    TEST_CXX=clang++ \
+    TEST_GCC=clang \
+    TEST_GXX=clang++ \
     check
 }
