--- a/packages/target-electron/build/gen-electron-builder-config.js
+++ b/packages/target-electron/build/gen-electron-builder-config.js
@@ -181,7 +181,12 @@
       StartupWMClass: appName,
     },
   },
-  files: [...files, PREBUILD_FILTERS.NOT_MAC, PREBUILD_FILTERS.NOT_WINDOWS],
+  files: [
+    ...files,
+    PREBUILD_FILTERS.NOT_MAC,
+    PREBUILD_FILTERS.NOT_WINDOWS,
+    PREBUILD_FILTERS.NOT_LINUX,
+  ],
   icon: 'build/icon.icns', // electron builder gets the icon out of the mac icon archive
   description: 'Decentralized Private Messenger (https://delta.chat)',
 }
