--- a/packages/target-electron/build/gen-electron-builder-config.js
+++ b/packages/target-electron/build/gen-electron-builder-config.js
@@ -155,7 +155,12 @@
       Keywords: 'delta chat;chat;deltachat;messaging;messenger;privacy',
     },
   },
-  files: [...files, PREBUILD_FILTERS.NOT_MAC, PREBUILD_FILTERS.NOT_WINDOWS],
+  files: [
+    ...files,
+    PREBUILD_FILTERS.NOT_MAC,
+    PREBUILD_FILTERS.NOT_WINDOWS,
+    PREBUILD_FILTERS.NOT_LINUX,
+  ],
   icon: 'build/icon.icns', // electron builder gets the icon out of the mac icon archive
   description: 'Decentralized Private Messenger (https://delta.chat)',
 }
--- a/packages/target-electron/node_modules/@deltachat/stdio-rpc-server/index.js
+++ b/packages/target-electron/node_modules/@deltachat/stdio-rpc-server/index.js
@@ -60,9 +60,7 @@
   if (takeVersionFromPATH) {
     return PATH_EXECUTABLE_NAME;
   }
-  // 3. check for prebuilds
-
-  return findRPCServerInNodeModules();
+  return "/usr/bin/deltachat-rpc-server";
 }
 
 import { StdioDeltaChat } from "@deltachat/jsonrpc-client";
