diff --git PKGBUILD PKGBUILD
index a0aafb51..73aebdb5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -6,6 +6,7 @@ pkgrel=10
 pkgdesc="tests performance of a remote host by sending HTTP, TCP and UDP requests"
 arch=('x86_64')
 url="http://echoping.sourceforge.net/"
+options=(!lto)
 license=('GPL')
 depends=(libidn popt libldap)
 #source=(https://sourceforge.net/projects/$pkgname/files/$pkgname/$pkgver/$pkgname-$pkgver.tar.gz)
