diff --git PKGBUILD PKGBUILD
index 0864895..57349a7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,14 +9,22 @@ arch=('x86_64')
 url='https://github.com/mitmproxy/mitmproxy_rs'
 license=('MIT')
 depends=('python')
-makedepends=('maturin' 'python-installer' 'cargo')
+makedepends=('maturin' 'python-installer' 'cargo' 'protobuf')
 options=(!lto)
 source=("https://github.com/mitmproxy/mitmproxy_rs/archive/$pkgver/$pkgname-$pkgver.tar.gz")
 sha512sums=('1ca6271fa875ed149f12f6ca5f34a308496b31a2a462db743d1d65b9895f9535f031e56d002cabf90e60c54e329a04596c6d8b5028d3bb564452676bbe75657e')
 
+prepare() {
+  cd ${_pyname}-${pkgver}
+  cargo fetch --locked
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+}
+
 build() {
   cd ${_pyname}-${pkgver}/mitmproxy-rs
-  maturin build --release --strip
+  # set PROTOC: path to protoc executable file.
+  PROTOC=/usr/bin/protoc maturin build --release --strip
 }
 
 check() {
