--- PKGBUILD
+++ PKGBUILD
@@ -43,4 +43,12 @@ package() {
   install -Dm0644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
 }
 
+prepare() {
+  cd ${_pyname}-${pkgver}
+  sed -i Cargo.toml \
+    -e 's/x25519-dalek = "=2.0.0-rc.3"/x25519-dalek = "2.0.0"/' \
+    -e '/\[patch.crates-io\]/a boringtun = { git = "https://github.com/cloudflare/boringtun", rev = "e3252d9c4f4c8fc628995330f45369effd4660a1" }'
+  cargo update -p boringtun
+}
+
 # vim: ts=2 sw=2 et:
