--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,7 @@
 makedepends=(
   git
   bpf-linker
-  rustup
+  rust
   maturin
   python-installer
   rust-src
@@ -29,14 +29,12 @@
 build() {
   cd "$pkgname"
 
-  export RUSTUP_TOOLCHAIN=nightly
-  rustup component add rust-src --toolchain "nightly-${CARCH}-unknown-linux-gnu"
-
   pushd mitmproxy-rs
   maturin build --release --strip
   popd
   
   pushd mitmproxy-linux
+  # Build eBPF's core library with stable Rust and the system LLVM.
   RUSTC_BOOTSTRAP=1 maturin build --release --strip
   popd
 }
