diff --git PKGBUILD PKGBUILD
index 0ddb442..d44d287 100644
--- PKGBUILD
+++ PKGBUILD
@@ -35,7 +35,6 @@ optdepends=('ghostscript: PS/PDF support'
             'libraw: DNG support'
             'librsvg: SVG support'
             'libtiff: TIFF support'
-            'libultrahdr: UHDR support'
             'libwebp: WEBP support'
             'libwmf: WMF support'
             'libzip: OpenRaster support'
@@ -59,7 +58,6 @@ makedepends=(chrpath
              libjxl
              libraw
              librsvg
-             libultrahdr
              libwebp
              libwmf
              libzip
