Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,7 @@
 arch=(x86_64)
 license=(custom)
 makedepends=(ghostscript openexr libwmf librsvg libxml2 openjpeg2 libraw opencl-headers libwebp libzip
-             chrpath ocl-icd glu ghostpcl ghostxps libheif jbigkit lcms2 libxext liblqr libraqm libpng djvulibre)
+             chrpath ocl-icd glu ghostpcl ghostxps jbigkit lcms2 libxext liblqr libraqm libpng djvulibre)
 checkdepends=(gsfonts ttf-dejavu)
 _relname=ImageMagick-${pkgver%%.*}
 _tarname=ImageMagick-${pkgver%.*}-${pkgver##*.}
