diff --git PKGBUILD PKGBUILD
index 635f9652..26db6306 100755
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,17 @@ license=('GPL2')
 depends=('google-glog' 'gtk3' 'jsoncpp' 'libsigc++')
 makedepends=('appstream' 'gflags' 'meson')
 checkdepends=('xorg-server-xvfb')
-source=("https://github.com/iptux-src/$pkgname/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
-sha256sums=('858fa04b987a0d576f6f67d5b5cbc543df01ed2b6c241ca890f93f89ebcb2331')
+source=("https://github.com/iptux-src/$pkgname/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "fix-i18n-merge_file.patch::https://github.com/iptux-src/iptux/pull/492.patch"
+        "fix-source-dependency.patch::https://github.com/iptux-src/iptux/pull/495.patch")
+sha256sums=('858fa04b987a0d576f6f67d5b5cbc543df01ed2b6c241ca890f93f89ebcb2331'
+            'd2da91a77d5ba3762039905966b9f1dcc850bbbbf15cf5b56a4a98e53c3f8369'
+            'db0eb3182b6bfedd9ab0f9b6ff2dac13f4be23754c546d96fe95c822afab1895')
+
+prepare() {
+  patch -Np1 -i $srcdir/fix-i18n-merge_file.patch -d $pkgname-$pkgver
+  patch -Np1 -i $srcdir/fix-source-dependency.patch -d $pkgname-$pkgver
+}
 
 build() {
   arch-meson $pkgname-$pkgver build
