--- PKGBUILD
+++ PKGBUILD
@@ -61,9 +61,9 @@
     -DGGML_BLAS=ON
     -DGGML_BLAS_VENDOR=OpenBLAS
     -DGGML_CPU_ALL_VARIANTS=ON
-    -DGGML_CUDA=ON
+    -DGGML_CUDA=OFF
     -DGGML_HIP=ON
-    -DGGML_OPENVINO=ON
+    -DGGML_OPENVINO=OFF
     -DGGML_VULKAN=ON
     "-DAMDGPU_TARGETS=$(rocm-supported-gfx)"
     )
@@ -97,9 +97,7 @@
   depends+=(libgomp)
   optdepends=(
     'ggml-blas: BLAS backend'
-    'ggml-cuda: CUDA backend'
     'ggml-hip: HIP backend'
-    'ggml-openvino: Openvino backend'
     'ggml-vulkan: Vulkan backend'
   )
   conflicts=(
@@ -194,3 +192,9 @@
   install -Dm 755 build-all/bin/libggml-vulkan.so -t "${pkgdir}"/usr/lib/ggml/
   install -Dm 644 ggml/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+pkgname=(${pkgname[@]/ggml-cuda})
+pkgname=(${pkgname[@]/ggml-openvino})
+makedepends=(${makedepends[@]/cuda})
+makedepends=(${makedepends[@]/nccl})
+makedepends=(${makedepends[@]/openvino})
