--- PKGBUILD
+++ PKGBUILD
@@ -16,13 +16,16 @@ checkdepends=('python-pytest')
 source=(
   "$url/archive/v$pkgver/$pkgname-$pkgver.tar.gz"
   "fix-libgit2-compatibility.patch::$url/commit/eba710e45bb40e18641c6531394bb46631e7f295.patch"
+  "https://salsa.debian.org/python-team/packages/python-pygit2/-/raw/aedf1461ed0f7292ed0af8961d44bc06a54afa23/debian/patches/0007-Fix-GenericIterator-class-interface.patch"
 )
 sha256sums=('72584d3bc24eb783229e5475eed41078815e90b03254451f3918bc80bba10513'
-            'e6d6c2cde580c163159bba8263c0fba4bc18af79781a6834e297cc2da242cbe9')
+            'e6d6c2cde580c163159bba8263c0fba4bc18af79781a6834e297cc2da242cbe9'
+            'b99873a3d40cb05848f001b10a12a6ccbd3ecb607dc8626dcf0331f28e6654d6')
 
 prepare() {
 	cd "pygit2-$pkgver"
   patch -Np1 -i ../fix-libgit2-compatibility.patch
+  patch -Np1 -i ../0007-Fix-GenericIterator-class-interface.patch
 	# Disable tests that do stuff online
 	sed -e '/has_network/s/True/False/' -i test/utils.py
 }
