From a418c651c88cd2682c4cfe61e9f57b5389078c09 Mon Sep 17 00:00:00 2001
From: Makoto Kato <m_kato@ga2.so-net.ne.jp>
Date: Thu, 17 Jun 2021 21:50:49 +0900
Subject: [PATCH] signal handler

---
 js/src/wasm/WasmSignalHandlers.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)
--- a/js/src/wasm/WasmSignalHandlers.cpp        2022-02-12 19:29:33.566924464 +0800
+++ b/js/src/wasm/WasmSignalHandlers.cpp        2022-02-12 19:50:29.499985612 +0800
@@ -163,6 +163,11 @@
 #      define R01_sig(p) ((p)->uc_mcontext.gp_regs[1])
 #      define R32_sig(p) ((p)->uc_mcontext.gp_regs[32])
 #    endif
+#  if defined(__linux__) && defined(__riscv) && __riscv_xlen == 64
+#    define EPC_sig(p) ((p)->uc_mcontext.__gregs[0])
+#    define X02_sig(p) ((p)->uc_mcontext.__gregs[2])
+#    define X08_sig(p) ((p)->uc_mcontext.__gregs[8])
+#  endif
 #  elif defined(__NetBSD__)
 #    define EIP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EIP])
 #    define EBP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EBP])
@@ -405,6 +410,10 @@
 #    define FP_sig(p) RFP_sig(p)
 #    define SP_sig(p) RSP_sig(p)
 #    define LR_sig(p) RRA_sig(p)
+#elif defined(__riscv) && __riscv_xlen == 64
+#  define PC_sig(p) EPC_sig(p)
+#  define SP_sig(p) X02_sig(p)
+#  define FP_sig(p) X08_sig(p)
 #  endif

 static void SetContextPC(CONTEXT* context, uint8_t* pc) {

