diff --unified --recursive --text a/.cargo/config.in b/.cargo/config.in
--- a/.cargo/config.in
+++ b/.cargo/config.in
@@ -22,11 +22,6 @@
 replace-with = "vendored-sources"
 rev = "3bfc47d9a571d0842676043ba60716318e946c06"
 
-[source."https://github.com/mozilla/midir.git"]
-git = "https://github.com/mozilla/midir.git"
-replace-with = "vendored-sources"
-rev = "e1b4dcb767f9e69afe95a860374aaa9635d81e3d"
-
 [source."https://github.com/mozilla/cubeb-pulse-rs"]
 git = "https://github.com/mozilla/cubeb-pulse-rs"
 replace-with = "vendored-sources"
@@ -37,6 +32,11 @@
 replace-with = "vendored-sources"
 rev = "44eca95823bb57e964cf7b6d9791ed2ccb4b2108"
 
+[source."https://github.com/mozilla/authenticator-rs"]
+git = "https://github.com/mozilla/authenticator-rs"
+replace-with = "vendored-sources"
+rev = "b85bccf0527e42c877573029e8d35ff13ef06f9d"
+
 [source."https://github.com/mozilla/audioipc"]
 git = "https://github.com/mozilla/audioipc"
 replace-with = "vendored-sources"

From a418c651c88cd2682c4cfe61e9f57b5389078c09 Mon Sep 17 00:00:00 2001
From: Makoto Kato <m_kato@ga2.so-net.ne.jp>
Date: Thu, 17 Jun 2021 21:50:49 +0900
Subject: [PATCH] signal handler

---
 js/src/wasm/WasmSignalHandlers.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)
--- a/js/src/wasm/WasmSignalHandlers.cpp        2022-02-12 19:29:33.566924464 +0800
+++ b/js/src/wasm/WasmSignalHandlers.cpp        2022-02-12 19:50:29.499985612 +0800
@@ -163,6 +163,11 @@
 #      define R01_sig(p) ((p)->uc_mcontext.gp_regs[1])
 #      define R32_sig(p) ((p)->uc_mcontext.gp_regs[32])
 #    endif
+#  if defined(__linux__) && defined(__riscv) && __riscv_xlen == 64
+#    define EPC_sig(p) ((p)->uc_mcontext.__gregs[0])
+#    define X02_sig(p) ((p)->uc_mcontext.__gregs[2])
+#    define X08_sig(p) ((p)->uc_mcontext.__gregs[8])
+#  endif
 #  elif defined(__NetBSD__)
 #    define EIP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EIP])
 #    define EBP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EBP])
@@ -405,6 +410,10 @@
 #    define FP_sig(p) RFP_sig(p)
 #    define SP_sig(p) RSP_sig(p)
 #    define LR_sig(p) RRA_sig(p)
+#elif defined(__riscv) && __riscv_xlen == 64
+#  define PC_sig(p) EPC_sig(p)
+#  define SP_sig(p) X02_sig(p)
+#  define FP_sig(p) X08_sig(p)
 #  endif

 static void SetContextPC(CONTEXT* context, uint8_t* pc) {

diff --git a/Cargo.lock b/Cargo.lock
index edc5ef5ff2d98..f6240163e1440 100644
--- a/Cargo.lock	2022-08-29 14:41:23.089464305 +0800
+++ b/Cargo.lock	2022-08-29 14:48:22.707351349 +0800
@@ -30,14 +30,14 @@
 
 [[package]]
 name = "alsa"
-version = "0.4.3"
+version = "0.6.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "eb213f6b3e4b1480a60931ca2035794aa67b73103d254715b1db7b70dcb3c934"
+checksum = "5915f52fe2cf65e83924d037b6c5290b7cee097c6b5c8700746e6168a343fd6b"
 dependencies = [
  "alsa-sys",
  "bitflags",
  "libc",
- "nix",
+ "nix 0.23.1",
 ]
 
 [[package]]
@@ -2220,6 +2219,7 @@
  "log",
  "mapped_hyph",
  "mdns_service",
+ "midir",
  "midir_impl",
  "mio 0.8.0",
  "moz_asserts",
@@ -3290,8 +3290,9 @@
 
 [[package]]
 name = "midir"
-version = "0.7.0"
-source = "git+https://github.com/mozilla/midir.git?rev=e1b4dcb767f9e69afe95a860374aaa9635d81e3d#e1b4dcb767f9e69afe95a860374aaa9635d81e3d"
+version = "0.8.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "2c1c68e2b589cce71b14a10d7d1599a845673f9decde80fa9e8500fdccd50dca"
 dependencies = [
  "alsa",
  "bitflags",
@@ -3299,10 +3300,9 @@
  "js-sys",
  "libc",
  "memalloc",
- "nix",
  "wasm-bindgen",
  "web-sys",
- "winapi",
+ "windows",
 ]

 [[package]]
@@ -3341,7 +3341,7 @@
  "libc",
  "memmap2 0.2.999",
  "memoffset 0.5.6",
- "nix",
+ "nix 0.15.0",
  "tempfile",
  "thiserror",
 ]
@@ -3752,6 +3752,19 @@
 ]

 [[package]]
+name = "nix"
+version = "0.23.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9f866317acbd3a240710c63f065ffb1e4fd466259045ccb504130b7f668f35c6"
+dependencies = [
+ "bitflags",
+ "cc",
+ "cfg-if 1.0.0",
+ "libc",
+ "memoffset 0.6.5",
+]
+
+[[package]]
 name = "nom"
 version = "6.1.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
@@ -6238,6 +6251,49 @@
 checksum = "712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f"

 [[package]]
+name = "windows"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "fbedf6db9096bc2364adce0ae0aa636dcd89f3c3f2cd67947062aaf0ca2a10ec"
+dependencies = [
+ "windows_aarch64_msvc",
+ "windows_i686_gnu",
+ "windows_i686_msvc",
+ "windows_x86_64_gnu",
+ "windows_x86_64_msvc",
+]
+
+[[package]]
+name = "windows_aarch64_msvc"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "d8e92753b1c443191654ec532f14c199742964a061be25d77d7a96f09db20bf5"
+
+[[package]]
+name = "windows_i686_gnu"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "6a711c68811799e017b6038e0922cb27a5e2f43a2ddb609fe0b6f3eeda9de615"
+
+[[package]]
+name = "windows_i686_msvc"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "146c11bb1a02615db74680b32a68e2d61f553cc24c4eb5b4ca10311740e44172"
+
+[[package]]
+name = "windows_x86_64_gnu"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c912b12f7454c6620635bbff3450962753834be2a594819bd5e945af18ec64bc"
+
+[[package]]
+name = "windows_x86_64_msvc"
+version = "0.32.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "504a2476202769977a040c6364301a3f65d0cc9e3fb08600b2bda150a0488316"
+
+[[package]]
 name = "wineventlog"
 version = "0.1.0"
 dependencies = [

diff --git a/Cargo.toml b/Cargo.toml
index 1c2437f1f5675..2923c7e5ea9bf 100644
--- a/Cargo.toml        2022-05-31 11:15:47.681941501 +0800
+++ b/Cargo.toml        2022-05-31 11:26:35.076132533 +0800
@@ -149,7 +149,6 @@
 fog = { path = "toolkit/components/glean/api" }
 libudev-sys = { path = "dom/webauthn/libudev-sys" }
 packed_simd = { package = "packed_simd_2", git = "https://github.com/hsivonen/packed_simd", rev="412f9a0aa556611de021bde89dee8fefe6e0fbbd" }
-midir = { git = "https://github.com/mozilla/midir.git", rev = "e1b4dcb767f9e69afe95a860374aaa9635d81e3d" }
 minidump_writer_linux = { git = "https://github.com/rust-minidump/minidump-writer.git", rev = "75ada456c92a429704691a85e1cb42fef8cafc0d" }

 # application-services overrides to make updating them all simpler.

diff --git a/python/mozbuild/mozbuild/vendor/vendor_rust.py b/python/mozbuild/mozbuild/vendor/vendor_rust.py
index 31baea4290..7394ccaf40 100644
--- a/python/mozbuild/mozbuild/vendor/vendor_rust.py
+++ b/python/mozbuild/mozbuild/vendor/vendor_rust.py
@@ -98,6 +98,7 @@ TOLERATED_DUPES = {
     "libloading": 2,
     "memoffset": 2,
     "mio": 2,
+    "nix": 2,
     # Transition from time 0.1 to 0.3 underway, but chrono is stuck on 0.1
     # and hasn't been updated in 1.5 years (an hypothetical update is
     # expected to remove the dependency on time altogether).
--- a/dom/midi/midir_impl/Cargo.toml	2022-08-30 01:32:46.905420566 +0800
+++ b/dom/midi/midir_impl/Cargo.toml	2022-08-30 01:33:08.125466847 +0800
@@ -7,7 +7,7 @@
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-midir = "0.7.0"
+midir = "0.8.0"
 nsstring = { path = "../../../xpcom/rust/nsstring/" }
 uuid = { version = "0.8", features = ["v4"] }
 thin-vec = { version = "0.2.1", features = ["gecko-ffi"] }
--- a/toolkit/library/rust/shared/Cargo.toml	2022-08-30 01:32:51.372097505 +0800
+++ b/toolkit/library/rust/shared/Cargo.toml	2022-08-30 01:33:57.192217018 +0800
@@ -75,6 +75,7 @@
 jog = { path = "../../../components/glean/bindings/jog" }
 dap_ffi = { path = "../../../components/telemetry/dap/ffi" }
 data-encoding-ffi = { path = "../../../../dom/fs/parent/rust/data-encoding-ffi" }
+midir = { version = "0.8.0" }
 uniffi-example-arithmetic = { git = "https://github.com/mozilla/uniffi-rs.git", rev = "bb2039f077a29dba0879372a67e764e6ace8e33f", optional = true }
 uniffi-example-geometry = { git = "https://github.com/mozilla/uniffi-rs.git", rev = "bb2039f077a29dba0879372a67e764e6ace8e33f", optional = true }
 uniffi-example-rondpoint = { git = "https://github.com/mozilla/uniffi-rs.git", rev = "bb2039f077a29dba0879372a67e764e6ace8e33f", optional = true }

 # Note: `modern_sqlite` means rusqlite's bindings file be for a sqlite with
 # version less than or equal to what we link to. This isn't a problem because we

