--- PKGBUILD	(revision 454659)
+++ PKGBUILD	(working copy)
@@ -12,8 +12,7 @@
 depends=(gtk3 libxt mime-types dbus-glib ffmpeg nss ttf-font libpulse)
 makedepends=(unzip zip diffutils yasm mesa imake inetutils xorg-server-xvfb
              autoconf2.13 rust clang llvm jack nodejs cbindgen nasm
-             python-setuptools python-zstandard lld dump_syms
-             wasi-compiler-rt wasi-libc wasi-libc++ wasi-libc++abi)
+             python-setuptools python-zstandard dump_syms)
 optdepends=('networkmanager: Location detection via available WiFi networks'
             'libnotify: Notification integration'
             'pulseaudio: Audio support'
@@ -23,11 +22,15 @@
 options=(!emptydirs !makeflags !strip !lto !debug)
 source=(https://archive.mozilla.org/pub/firefox/releases/$pkgver/source/firefox-$pkgver.source.tar.xz{,.asc}
         zstandard-0.18.0.diff arc4random.diff
+        makotokato-riscv64-support-and-zenithal-backported.patch
+        https://raw.githubusercontent.com/alexfanqi/riscv/master/www-client/firefox/files/firefox-disable-vet-hack.patch
         $pkgname.desktop identity-icons-brand.svg)
 sha256sums=('1a294a651dc6260f9a72a3ab9f10e7792a4ab41a9cfa8527ad3dd9979cdc98ce'
             'SKIP'
             'a6857ad2f2e2091c6c4fdcde21a59fbeb0138914c0e126df64b50a5af5ff63be'
             '714ca50b2ce0cac470dbd5a60e9a0101b28072f08a5e7a9bba94fef2058321c4'
+            '95d79a8dfdd2742ce34a4194804162bdeea7060121fb4e393c4384d50ba5bd98'
+            'cf1c1269b3dcd0f9c028d13ce11f000953c9ddafbecf53b6dfef785e8fb7347b'
             '298eae9de76ec53182f38d5c549d0379569916eebf62149f9d7f4a7edef36abf'
             'a9b8b4a0a1f4a7b4af77d5fc70c2686d624038909263c795ecc81e0aec7711e9')
 validpgpkeys=('14F26682D0916CDD81E37B6D61B7B526D98F0353') # Mozilla Software Releases <release@mozilla.com>
@@ -54,6 +57,9 @@
   # Unbreak build with glibc 2.36
   patch -Np1 -i ../arc4random.diff
 
+  patch -Np1 -i ../firefox-disable-vet-hack.patch
+  patch -Np1 -i ../makotokato-riscv64-support-and-zenithal-backported.patch
+
   echo -n "$_google_api_key" >google-api-key
   echo -n "$_mozilla_api_key" >mozilla-api-key
 
@@ -62,15 +68,22 @@
 mk_add_options MOZ_OBJDIR=${PWD@Q}/obj
 
 ac_add_options --prefix=/usr
-ac_add_options --enable-release
+# release mode incurs TEST-UNEXPECTED-FAIL | check_networking
+# refer to config/makefiles/rust.mk#L408
+ac_add_options --disable-release
 ac_add_options --enable-hardening
-ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-ac_add_options --enable-linker=lld
-ac_add_options --disable-elf-hack
+# lld is broken
+ac_add_options --enable-linker=bfd
 ac_add_options --disable-bootstrap
-ac_add_options --with-wasi-sysroot=/usr/share/wasi-sysroot
+ac_add_options --disable-jit
+ac_add_options --without-wasm-sandboxed-libraries
 
+ac_add_options --enable-optimize
+ac_add_options --disable-debug
+# rustc with opt-level 2 would segfault when compiling neqo-transport
+export RUSTC_OPT_LEVEL=1
+
 # Branding
 ac_add_options --enable-official-branding
 ac_add_options --enable-update-channel=release
@@ -77,7 +90,8 @@
 ac_add_options --with-distribution-id=org.archlinux
 ac_add_options --with-unsigned-addon-scopes=app,system
 ac_add_options --allow-addon-sideload
-export MOZILLA_OFFICIAL=1
+# see check_network above
+# export MOZILLA_OFFICIAL=1
 export MOZ_APP_REMOTINGNAME=${pkgname//-/}
 
 # Keys
@@ -92,7 +106,7 @@
 # Features
 ac_add_options --enable-alsa
 ac_add_options --enable-jack
-ac_add_options --enable-crashreporter
+# crashreporter not ported to riscv64
 ac_add_options --disable-updater
 ac_add_options --disable-tests
 END
@@ -110,39 +124,42 @@
   ulimit -n 4096
 
   # Do 3-tier PGO
-  echo "Building instrumented browser..."
-  cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-profile-generate=cross
-END
+  # disable PGO
+#  echo "Building instrumented browser..."
+   cat >.mozconfig ../mozconfig
+#  cat >.mozconfig ../mozconfig - <<END
+#ac_add_options --enable-profile-generate=cross
+#END
+  ./mach vendor rust --ignore-modified
   ./mach build
 
-  echo "Profiling instrumented browser..."
-  ./mach package
-  LLVM_PROFDATA=llvm-profdata \
-    JARLOG_FILE="$PWD/jarlog" \
-    xvfb-run -s "-screen 0 1920x1080x24 -nolisten local" \
-    ./mach python build/pgo/profileserver.py
-
-  stat -c "Profile data found (%s bytes)" merged.profdata
-  test -s merged.profdata
-
-  stat -c "Jar log found (%s bytes)" jarlog
-  test -s jarlog
-
-  echo "Removing instrumented browser..."
-  ./mach clobber
-
-  echo "Building optimized browser..."
-  cat >.mozconfig ../mozconfig - <<END
-ac_add_options --enable-lto=cross
-ac_add_options --enable-profile-use=cross
-ac_add_options --with-pgo-profile-path=${PWD@Q}/merged.profdata
-ac_add_options --with-pgo-jarlog=${PWD@Q}/jarlog
-END
-  ./mach build
-
-  echo "Building symbol archive..."
-  ./mach buildsymbols
+#  echo "Profiling instrumented browser..."
+#  ./mach package
+#  LLVM_PROFDATA=llvm-profdata \
+#    JARLOG_FILE="$PWD/jarlog" \
+#    xvfb-run -s "-screen 0 1920x1080x24 -nolisten local" \
+#    ./mach python build/pgo/profileserver.py
+#
+#  stat -c "Profile data found (%s bytes)" merged.profdata
+#  test -s merged.profdata
+#
+#  stat -c "Jar log found (%s bytes)" jarlog
+#  test -s jarlog
+#
+#  echo "Removing instrumented browser..."
+#  ./mach clobber
+#
+#  echo "Building optimized browser..."
+#  cat >.mozconfig ../mozconfig - <<END
+#ac_add_options --enable-lto=cross
+#ac_add_options --enable-profile-use=cross
+#ac_add_options --with-pgo-profile-path=${PWD@Q}/merged.profdata
+#ac_add_options --with-pgo-jarlog=${PWD@Q}/jarlog
+#END
+#  ./mach build
+#
+#  echo "Building symbol archive..."
+#  ./mach buildsymbols
 }
 
 package() {
@@ -210,12 +227,12 @@
     ln -srfv "$pkgdir/usr/lib/libnssckbi.so" "$nssckbi"
   fi
 
-  export SOCORRO_SYMBOL_UPLOAD_TOKEN_FILE="$startdir/.crash-stats-api.token"
-  if [[ -f $SOCORRO_SYMBOL_UPLOAD_TOKEN_FILE ]]; then
-    make -C obj uploadsymbols
-  else
-    cp -fvt "$startdir" obj/dist/*crashreporter-symbols-full.tar.zst
-  fi
+  #export SOCORRO_SYMBOL_UPLOAD_TOKEN_FILE="$startdir/.crash-stats-api.token"
+  #if [[ -f $SOCORRO_SYMBOL_UPLOAD_TOKEN_FILE ]]; then
+  #  make -C obj uploadsymbols
+  #else
+  #  cp -fvt "$startdir" obj/dist/*crashreporter-symbols-full.tar.zst
+  #fi
 }
 
 # vim:set sw=2 et:
