--- PKGBUILD
+++ PKGBUILD
@@ -149,10 +149,18 @@ ac_add_options --with-system-nss
 # Features
 ac_add_options --enable-alsa
 ac_add_options --enable-jack
-ac_add_options --enable-crashreporter
 ac_add_options --disable-updater
 ac_add_options --disable-tests
 END
+
+  case "$CARCH" in
+    x86_64 | i*86 | arm* | aarch64)
+      echo 'ac_add_options --enable-crashreporter' >>../mozconfig
+      ;;
+    *)
+      echo 'ac_add_options --disable-crashreporter' >>../mozconfig
+      ;;
+  esac
 }
 
 build() {
