--- a/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp
+++ b/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp
@@ -139,7 +139,7 @@
       ".err   \"stub number "#n" >= 1000 not yet supported\"\n"                     \
       ".endif \n\t"                                                                 \
       "li      t0, "#n" \n\t"                                                       \
-      "j       SharedStub \n"                                                       \
+      "tail    SharedStub \n"                                                       \
       ".if "#n" < 10\n\t"                                                           \
       ".size   _ZN14nsXPTCStubBase5Stub"#n"Ev,.-_ZN14nsXPTCStubBase5Stub"#n"Ev\n\t" \
       ".elseif "#n" < 100\n\t"                                                      \
