Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 423737)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 url="https://github.com/falconindy/pkgfile"
 license=('MIT')
 depends=('libarchive' 'curl' 'pcre' 'pacman')
-makedepends=('git' 'meson')
+makedepends=('git' 'meson' 'clang' 'systemd')
 source=("git+https://github.com/falconindy/pkgfile?signed#tag=v$pkgver")
 validpgpkeys=('487EACC08557AD082088DABA1EB2638FF56C0C53')  # Dave Reisner
 install=pkgfile.install
