diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index e8a1b44..33e9811 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,7 @@ arch=('x86_64')
 license=('LGPL2.1' 'LGPL3')
 makedepends=('bzr' 'dbus-glib' 'gnome-common' 'gobject-introspection' 'gtk-doc' 'libdbusmenu-gtk'{2,3} 'libindicator-gtk'{2,3} 'vala')
 options=('!emptydirs')
-source=(bzr+lp:libappindicator#revision=$_bzrrev)
+source=($pkgbase::bzr+$url/trunk#revision=$_bzrrev)
 sha512sums=('SKIP')
 validpgpkeys=('6FC05581A37D71FCECE165DB5BE41E162CD6358E')  # Charles Kerr <charles.kerr@canonical.com>
 
@@ -60,7 +60,7 @@ package_libappindicator-gtk2() {
   depends=('libdbusmenu-gtk2' 'libindicator-gtk2')
 
   cd ${pkgbase}-gtk2
-  make DESTDIR="${pkgdir}" install
+  make DESTDIR="${pkgdir}" install -j1
 }
 
 package_libappindicator-gtk3() {
@@ -68,7 +68,7 @@ package_libappindicator-gtk3() {
   depends=('libdbusmenu-gtk3' 'libindicator-gtk3')
 
   cd ${pkgbase}
-  make DESTDIR="${pkgdir}" install
+  make DESTDIR="${pkgdir}" install -j1
 }
 
 # vim: ts=2 sw=2 et:
