--- PKGBUILD
+++ PKGBUILD
@@ -9,16 +9,17 @@ pkgdesc="Utilities for rescue and embedded systems"
 arch=("x86_64")
 url="https://www.busybox.net"
 license=('GPL')
-makedepends=("ncurses" "musl" "kernel-headers-musl")
+depends=('glibc')
+makedepends=("ncurses" "linux-api-headers")
 validpgpkeys=('C9E9416F76E610DBD09D040F47B70C55ACC9965B')
 source=("$url/downloads/$pkgname-$pkgver.tar.bz2"{,.sig}
         "config")
 sha256sums=('b8cc24c9574d809e7279c3be349795c5d5ceb6fdf19ca709f80cde50e47de314'
             'SKIP'
-            '239001693b58e471c594952b9805c24b0d8174ba2cab245cd624e4423d993cd3')
+            '67a5d71bfc0cb33f9cc428fecf5cef6ff19f694a4da0ed2e3e803e811953d2bf')
 b2sums=('e515825cb3ab1c520e16b9c2512e9fc72947366a72a0466bff59b507fdffbc78fc9d16b44a26116175fc7a429d849ad944b1bc379d36c6d3a0eb20969997336e'
         'SKIP'
-        '7d4759036db05d7e0c06571e8fd24db488edf87a9b82a5382eee3fb43fc5318bbd75398e43c9b30c88b78630e910ee9a5316aaa824b5ab26f93fc1b3f834f07d')
+        'bf88f8bb38bb8ceb6393722255af9434652464958817123b6b8f1802f7c22e1e37a6cbe950a74bd483c627f7df7d4e5ddaaaedf5f0d7ca23d449a35f2111d4bd')
 
 build() {
   cd "$srcdir/$pkgname-$pkgver"
@@ -26,7 +27,7 @@ build() {
   cp "$srcdir"/config .config
   # reproducible build
   export KCONFIG_NOTIMESTAMP=1
-  make CC=musl-gcc
+  make
 }
 
 package() {
--- config
+++ config
@@ -39,8 +39,8 @@ CONFIG_FEATURE_SYSLOG=y
 #
 # Build Options
 #
-CONFIG_STATIC=y
-# CONFIG_PIE is not set
+# CONFIG_STATIC is not set
+CONFIG_PIE=y
 # CONFIG_NOMMU is not set
 # CONFIG_BUILD_LIBBUSYBOX is not set
 # CONFIG_FEATURE_LIBBUSYBOX_STATIC is not set
@@ -53,7 +53,7 @@ CONFIG_EXTRA_LDFLAGS=""
 CONFIG_EXTRA_LDLIBS=""
 # CONFIG_USE_PORTABLE_CODE is not set
 CONFIG_STACK_OPTIMIZATION_386=y
-CONFIG_STATIC_LIBGCC=y
+# CONFIG_STATIC_LIBGCC is not set
 
 #
 # Installation Options ("make install" behavior)
