diff -r -u anyjson-0.3.3-orig/setup.py anyjson-0.3.3/setup.py
--- anyjson-0.3.3-orig/setup.py	2022-01-30 16:21:11.281194683 +0800
+++ anyjson-0.3.3/setup.py	2022-01-30 16:22:08.782484123 +0800
@@ -1,10 +1,6 @@
 import os
 import sys
 
-extra = {}
-if sys.version_info >= (3, 0):
-    extra.update(use_2to3=True)
-
 try:
     from setuptools import setup, find_packages
 except ImportError:
@@ -96,5 +92,4 @@
       zip_safe=False,
       platforms=["any"],
       test_suite = 'nose.collector',
-      **extra
 )
