--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,11 @@ validpgpkeys=(
   '890C08DB8579162FEE0DF9DB8BEAB4DFCF555EF4' # RafaelGSS <rafael.nunu@hotmail.com>
 )
 
+prepare() {
+  cd node
+  patch -Np1 -i ../fix-trap-handler.patch
+}
+
 build() {
   cd node
 
@@ -76,4 +81,7 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/nodejs/
 }
 
+source+=("fix-trap-handler.patch")
+sha512sums+=('f2ff6da8cf5dcc994a7a20342e2928dc1821fbbf42891009a6234b6051277e0200d7e3fbba63b9a2773887591d0ad5ceb1bb3d25e5efeb557f6d00109a80253c')
+
 # vim:set ts=2 sw=2 et:
