--- PKGBUILD
+++ PKGBUILD
@@ -19,8 +19,14 @@ options=(!lto)
 depends=('brotli' 'openssl' 'zlib' 'icu' 'libuv' 'libnghttp2' 'c-ares') # 'http-parser' 'v8')
 makedepends=('git' 'python' 'procps-ng')
 optdepends=('npm: nodejs package manager')
-source=("git+https://github.com/nodejs/node.git#commit=$_commit")
-sha512sums=('SKIP')
+source=("git+https://github.com/nodejs/node.git#commit=$_commit"
+        "node21-fix-v8.patch::https://github.com/nodejs/node/pull/50551.diff")
+sha512sums=('SKIP'
+            '97b47d493b3364695c150c5ba8e5991f5009260cc8c4931da7fd324bb0129b0f04b6fad73ffe156150a76a621b6f13d13dab8640fc4aa9499d19db2ac66411e0')
+
+prepare() {
+  patch -Np1 -d node < node21-fix-v8.patch
+}
 
 build() {
   cd node
