--- PKGBUILD
+++ PKGBUILD
@@ -52,6 +52,16 @@ _set_flags() {
   # /usr/lib/libnode.so uses malloc_usable_size, which is incompatible with fortification level 3
   CFLAGS="${CFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
   CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
+  # https://github.com/riscv-forks/electron/issues/7
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
+prepare() {
+  cd node
+  patch -Np1 -i ../hwy-broken-rvv.diff
+  patch -Np1 -i ../v8-riscv-fix-trampoline.patch
+  patch -Np1 -i ../v8-riscv-fix-trampoline-release.patch
 }
 
 build() {
@@ -103,4 +113,12 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/nodejs/
 }
 
+
+makedepends+=(clang)
+source+=("hwy-broken-rvv.diff"
+         "v8-riscv-fix-trampoline.patch"
+         "v8-riscv-fix-trampoline-release.patch")
+sha512sums+=('DE07b0d9c3481036ee97a22941ff444fee86c78abbc26afef36f17508bb479ce3ab83ca160109fbf4f0b9b3266dcce30860873dc8ffbcac1a70e98d17638ca55'
+             '531ed581212fad93782a5c38a3077be898c74b27b0211313a807cadb7fd6dbadb959c36d380c30ff597d0d0e9da9e68b9f8769d8a1ac6928fb27441f0e9812f5'
+             '1bd56f854cb50eecdeced4a007335e407def69fabe14305f053e4d33c226740bb0bbaa861880df6b45d7a761150b77ab8dca28ad0489c75b52cd4c4cad61ddc7')
 # vim:set ts=2 sw=2 et:
