--- PKGBUILD
+++ PKGBUILD
@@ -53,10 +53,15 @@ _set_flags() {
   # /usr/lib/libnode.so uses malloc_usable_size, which is incompatible with fortification level 3
   CFLAGS="${CFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
   CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
+  # https://github.com/riscv-forks/electron/issues/7
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
 }
 
+
 prepare() {
   cd node
+  patch -Np1 -i ../hwy-broken-rvv.diff
   # Update ICU tests https://github.com/nodejs/node/pull/60523
   patch -Np1 -i ${srcdir}/update-icu-tests.patch
 }
@@ -89,6 +96,8 @@ build() {
 check() {
   _set_flags
   cd node
+  rm test/parallel/test-snapshot-reproducible.js
+  # TODO: figure out why the above test is failing on RISC-V
   rm test/parallel/test-http-outgoing-end-cork.js
   rm test/parallel/test-http2-client-set-priority.js
   rm test/parallel/test-http2-client-unescaped-path.js
@@ -113,4 +122,8 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
 
+
+makedepends+=(clang)
+source+=("hwy-broken-rvv.diff")
+sha512sums+=('de07b0d9c3481036ee97a22941ff444fee86c78abbc26afef36f17508bb479ce3ab83ca160109fbf4f0b9b3266dcce30860873dc8ffbcac1a70e98d17638ca55')
 # vim:set ts=2 sw=2 et:
