--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,16 @@ optdepends=('npm: nodejs package manager')
 source=("git+https://github.com/nodejs/node.git#commit=$_commit")
 sha512sums=('c863de81eb5d45d0e271d8828c52395216bbfc9ac736c90d561234b5724454b95dd759cd96ff5435e43c2debd356f966c14cf8b68a81421601d21b611e2e1f80')
 
+makedepends+=(clang)
+
+_set_compilation_env() {
+  # https://bugs.chromium.org/p/v8/issues/detail?id=14547
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
 build() {
+  _set_compilation_env
   cd node
 
   # /usr/lib/libnode.so uses malloc_usable_size, which is incompatible with fortification level 3
@@ -48,11 +57,13 @@ build() {
 }
 
 check() {
+  _set_compilation_env
   cd node
   make test || :
 }
 
 package() {
+  _set_compilation_env
   cd node
   make DESTDIR="$pkgdir" install
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/nodejs/
