Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1003856)
+++ PKGBUILD	(working copy)
@@ -18,11 +18,14 @@
 makedepends=('python' 'procps-ng')
 optdepends=('npm: nodejs package manager')
 source=("https://github.com/nodejs/node/archive/v$pkgver/nodejs-$pkgver.tar.gz"
+        v8-riscv.patch
         system-c-ares.patch)
 sha512sums=('37840bc9bed1ce99b33f3307ecc9ab8b7f5de90ac46ce01250d080cae2cfcc7cd0f2c55fa74d862a7e48d5fbd3696581bbb31cc46a847f2901fe70e2f67224e1'
+            '48bb61eee7cd43c4c19a11428e48be96783b212b0a964459b1491f9f44524d8db8a9ecdde7e1d234ee5e945cbdd3b219e419b1511563ed729ed87454b3d20948'
             'bca85b5a4622b38f20f3787bdb2784db76cd4b213fadf8a5813a375080e7481c4aff7556712d59df3e19849a7eb6b2f048a69b4e2162cb717ab1b3a791d2558e')
 
 prepare() {
+  patch -d node-$pkgver -Np1 < v8-riscv.patch
   patch -d node-$pkgver -Rp1 < system-c-ares.patch
 }
 
