--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,12 @@ validpgpkeys=(
   'C82FA3AE1CBEDC6BE46B9360C43CEC45C17AB93C' # Richard Lau <rlau@redhat.com>
 )
 
+prepare() {
+  cd node
+  patch -Np1 -i ../fix-trap-handler.patch
+  patch -Np1 -i ../v8-disable-trap-handler.patch
+}
+
 build() {
   cd node
 
@@ -77,4 +83,8 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/nodejs/
 }
 
+source+=("fix-trap-handler.patch"
+         "v8-disable-trap-handler.patch")
+sha512sums+=('f2ff6da8cf5dcc994a7a20342e2928dc1821fbbf42891009a6234b6051277e0200d7e3fbba63b9a2773887591d0ad5ceb1bb3d25e5efeb557f6d00109a80253c'
+             'b6495aefd36915969ee848cca350a565317c74864cd33e6a69a310ed9cbc71dfbd91f31e8c6176667f6f72daa1762eb4d519700a024cdbe8b18049100a9e3c80')
 # vim:set ts=2 sw=2 et:
