--- PKGBUILD
+++ PKGBUILD
@@ -39,7 +39,10 @@ makedepends=(
 optdepends=('npm: nodejs package manager')
 options=('!lto')
 source=("git+https://github.com/nodejs/node.git#tag=v$pkgver?signed")
-sha512sums=('288b819aa331c2a59a5c07f1eef4be5a647c0363b340afc29504b254aacb86e4a23a2600e32ef87c4aa9e6d29c0023583a159de9853c047129ffb94489c66a68')
+sha512sums=('288b819aa331c2a59a5c07f1eef4be5a647c0363b340afc29504b254aacb86e4a23a2600e32ef87c4aa9e6d29c0023583a159de9853c047129ffb94489c66a68'
+            'de07b0d9c3481036ee97a22941ff444fee86c78abbc26afef36f17508bb479ce3ab83ca160109fbf4f0b9b3266dcce30860873dc8ffbcac1a70e98d17638ca55'
+            '4d972e4c65a30fca42c83c4c481bb622586cb8ad8070788e55bdc44f0e8d957bf901cbbb20db4ef4dc6ae07e26f2e8ca572be7d740262279fb6e57ca04d5e023'
+            '5f8467db1a40a5c4102dc740fbf817751c0fbb34de8dfa63e4716aae39d803fc4f0298e87a060fb9d27c484a09a1bc555f7cd07131181aa916066781b00ced43')
 validpgpkeys=(
   '8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600' # Michaël Zasso (Targos) <targos@protonmail.com>
   '890C08DB8579162FEE0DF9DB8BEAB4DFCF555EF4' # RafaelGSS <rafael.nunu@hotmail.com>
@@ -52,6 +55,16 @@ _set_flags() {
   # /usr/lib/libnode.so uses malloc_usable_size, which is incompatible with fortification level 3
   CFLAGS="${CFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
   CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
+  # https://github.com/riscv-forks/electron/issues/7
+  export CC=/usr/bin/clang
+  export CXX=/usr/bin/clang++
+}
+
+prepare() {
+  cd node
+  patch -Np1 -i ../hwy-broken-rvv.diff
+  patch -Np1 -i ../v8-riscv-fix-trampoline.patch
+  patch -Np1 -i ../v8-riscv-fix-trampoline-release.patch
 }
 
 build() {
@@ -96,4 +109,9 @@ package() {
   install -Dm644 LICENSE -t "$pkgdir"/usr/share/licenses/nodejs/
 }
 
+
+makedepends+=(clang)
+source+=("hwy-broken-rvv.diff"
+         "v8-riscv-fix-trampoline.patch"
+         "v8-riscv-fix-trampoline-release.patch")
 # vim:set ts=2 sw=2 et:
