--- PKGBUILD
+++ PKGBUILD
@@ -54,6 +54,8 @@
     --deselect 'test/test_main.py::test_call_from_outside_venv[isolated]'
     --deselect 'test/test_main.py::test_call_from_outside_venv[in]'
     --deselect 'test/test_main.py::test_call_from_outside_venv[missing]'
+    # Downloads cryptography from PyPI, which has no riscv64 wheels.
+    --deselect 'test/test_main.py::test_ro_venv'
     --deselect 'test/test_rules_collection.py::test_rules_id_format'
     --deselect 'test/test_main.py::test_get_version_warning[v1.2.2-True-pre-release-1]'
     --deselect 'test/test_main.py::test_get_version_warning[v1.2.3-False--1]'
