--- a/lut/Test.lua
+++ b/lut/Test.lua
@@ -467,9 +467,9 @@
       for k, info in pairs(coverage) do
         if info ~= true and not self.ignore[k] then
           if lib.mock then
-            k = string.upper(string.sub(k, 1, 1))..string.sub(k, 2, -1)
+            local _k = string.upper(string.sub(k, 1, 1))..string.sub(k, 2, -1)
             lub.insertSorted(not_tested, {
-              text = 'function should.respondTo'..k..'()\n'..lib.mock..'end',
+              text = 'function should.respondTo'.._k..'()\n'..lib.mock..'end',
               line = tonumber(debug.getinfo(info).linedefined),
             }, 'line')
           else
