--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ url='http://sympy.org/'
 arch=(x86_64)
 license=(MIT)
 depends=(arb libmpc gmp-ecm primesieve gperftools zlib)
-makedepends=(cmake boost llvm cereal)
+makedepends=(cmake boost cereal)
 source=(https://github.com/symengine/symengine/releases/download/v$pkgver/$pkgname-$pkgver.tar.gz)
 sha256sums=('dcf174ac708ed2acea46691f6e78b9eb946d8a2ba62f75e87cf3bf4f0d651724')
 
@@ -19,6 +19,7 @@ prepare() {
 }
 
 build() {
+  # LLVM's MCJIT doesn't support riscv64
   cmake -B build -S $pkgname-$pkgver \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DBUILD_SHARED_LIBS=ON \
@@ -29,7 +30,7 @@ build() {
     -DWITH_ECM=ON \
     -DWITH_MPFR=ON \
     -DWITH_MPC=ON \
-    -DWITH_LLVM=ON \
+    -DWITH_LLVM=OFF \
     -DWITH_PRIMESIEVE=ON \
     -DWITH_BOOST=ON \
     -DWITH_COTIRE=OFF \
