--- PKGBUILD
+++ PKGBUILD
@@ -41,7 +41,7 @@ build() {
     -DWITH_SYMENGINE_THREAD_SAFE=ON \
     -DWITH_ARB=ON \
     -DWITH_ECM=ON \
-    -DWITH_LLVM=ON \
+    -DWITH_LLVM=OFF \
     -DWITH_MPFR=ON \
     -DWITH_MPC=ON \
     -DWITH_PRIMESIEVE=ON \
@@ -59,3 +59,6 @@ package() {
   DESTDIR="$pkgdir" cmake --install build
   install -Dm644 $pkgname/LICENSE -t "$pkgdir"/usr/share/licenses/$pkgname
 }
+
+depends=(${depends[@]/llvm-libs})
+makedepends=(${makedepends[@]/llvm})
