diff --git a/include/boost/lockfree/detail/freelist.hpp b/include/boost/lockfree/detail/freelist.hpp
index 31c8c98..2ac6019 100644
--- a/include/boost/lockfree/detail/freelist.hpp
+++ b/include/boost/lockfree/detail/freelist.hpp
@@ -10,6 +10,7 @@
 #define BOOST_LOCKFREE_FREELIST_HPP_INCLUDED
 
 #include <array>
+#include <cstdint>
 #include <cstring>
 #include <limits>
 #include <memory>
diff --git a/include/boost/lockfree/stack.hpp b/include/boost/lockfree/stack.hpp
index 19d5e7d..d6646e3 100644
--- a/include/boost/lockfree/stack.hpp
+++ b/include/boost/lockfree/stack.hpp
@@ -29,6 +29,7 @@
 #include <boost/lockfree/detail/uses_optional.hpp>
 #include <boost/lockfree/lockfree_forward.hpp>
 
+#include <cstdint>
 #include <tuple>
 #include <type_traits>
 
