diff --git PKGBUILD PKGBUILD
index dde029a..b7e3397 100644
--- PKGBUILD
+++ PKGBUILD
@@ -31,6 +31,8 @@ prepare() {
   patch -d $_srcname/boost -p3 < 9ac89e99.patch
   # https://gitlab.archlinux.org/archlinux/packaging/packages/boost/-/issues/8
   patch -d $_srcname/boost -p3 < boost-1.88.0-fix-exception-state-performance.patch
+  # fix missing <cstdint> (https://github.com/boostorg/lockfree/pull/109)
+  patch -d $_srcname/boost -p3 < $pkgname-fix-missing-cstdint.patch
 }
 
 build() {
@@ -160,4 +162,7 @@ package_boost-libs() {
   cp fakeinstall/lib/boost-python*/mpi.so "$pkgdir"$site_packages/boost/mpi.so
 }
 
+source+=($pkgname-fix-missing-cstdint.patch)
+sha256sums+=('450fbb4f0d42a4fb098ea5ab72ea87e5b4a0921fc80618cded24ebd4b17c9c4a')
+
 # vim:set ts=2 sw=2 et:
