--- PKGBUILD
+++ PKGBUILD
@@ -27,9 +27,14 @@ sha256sums=('6478edfe2f3305127cffe8caf73ea0176c53769f4bf1585be237eb30798c3b8e'
             '1b5998ee8fb389dd6df55a3684d29ffa37246bc007e8e6712bf2be6c7f745036'
             'aa38addb40d5f44b4a8472029b475e7e6aef1c460509eb7d8edf03491dc1b5ee')
 
+source+=(b2-fix-OSPLAT.patch::https://github.com/bfgroup/b2/pull/341.patch)
+sha256sums+=('6c03f4bded48e7d9e24cbbd0fafc9c451326ff015a0d99ff3c783e8209b9509b')
+
 prepare() {
   cd $_srcname
 
+  patch -Np1 -d tools/build < ../b2-fix-OSPLAT.patch
+
   # https://github.com/boostorg/phoenix/issues/111
   patch -Np1 -i ../boost-1.81.0-phoenix-multiple-definitions.patch
 
