diff --git PKGBUILD PKGBUILD
index 6559602..965b842 100644
--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,7 @@ sha256sums=('46d9d2c06637b219270877c9e16155cbd015b6dc84349af064c088e9b5b12f7b'
 
 prepare() {
   patch -d $_srcname/boost -p3 < 9ac89e99.patch # Add missing include
+  patch -d $_srcname/boost -p3 < $pkgname-fix-missing-cstdint.patch # fix missing <cstdint> (https://github.com/boostorg/lockfree/pull/109)
 }
 
 build() {
@@ -155,4 +156,7 @@ package_boost-libs() {
   cp fakeinstall/lib/boost-python*/mpi.so "$pkgdir"$site_packages/boost/mpi.so
 }
 
+source+=($pkgname-fix-missing-cstdint.patch)
+sha256sums+=('450fbb4f0d42a4fb098ea5ab72ea87e5b4a0921fc80618cded24ebd4b17c9c4a')
+
 # vim:set ts=2 sw=2 et:
