diff --git PKGBUILD PKGBUILD
index a316ffb..777b117 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,9 @@
   # fix the smartpointer prints (https://github.com/boostorg/smart_ptr/issues/115)
   # also see https://github.com/luceneplusplus/LucenePlusPlus/issues/208
   patch -Np4 -d boost/smart_ptr < ../$pkgname-fix-smart-pointer-output.patch
+
+  # fix missing <cstdint> (https://github.com/boostorg/lockfree/pull/109)
+  patch -Np2 < ../$pkgname-fix-missing-cstdint.patch
 }
 
 build() {
@@ -159,4 +162,7 @@
   cp fakeinstall/lib/boost-python*/mpi.so "$pkgdir"$site_packages/boost/mpi.so
 }
 
+source+=($pkgname-fix-missing-cstdint.patch)
+sha256sums+=('450fbb4f0d42a4fb098ea5ab72ea87e5b4a0921fc80618cded24ebd4b17c9c4a')
+
 # vim:set ts=2 sw=2 et:
