--- PKGBUILD
+++ PKGBUILD
@@ -81,7 +81,15 @@ prepare() {
   )
   rm -r "${to_remove[@]}"
 
-  yarn install --frozen-lockfile
+  export ELECTRON_SKIP_BINARY_DOWNLOAD=1
+  jq '.devDependencies."electron-builder"="npm:@riscv-forks/electron-builder@25.1.8"
+   |  .overrides."app-builder-lib"="npm:@riscv-forks/app-builder-lib@25.1.8"
+   |  .overrides."builder-util"="npm:@riscv-forks/builder-util@25.1.7"' package.json > package.json.new
+  mv package.json{.new,}
+
+  yarn install
+  mkdir -p  node_modules/7zip-bin/linux/riscv64
+  ln -s /usr/bin/7za node_modules/7zip-bin/linux/riscv64/7za
 }
 
 build() {
@@ -93,7 +101,7 @@ build() {
 
 package() {
   cd $pkgname
-  install -vDm644 -t "$pkgdir/usr/lib/$pkgname" dist/linux-unpacked/resources/app.asar
+  install -vDm644 -t "$pkgdir/usr/lib/$pkgname" dist/linux-riscv64-unpacked/resources/app.asar
   install -vDm755 "$srcdir/drawio.sh" "$pkgdir/usr/bin/drawio"
 
   install -vDm644 "$srcdir/drawio.xml" "$pkgdir/usr/share/mime/packages/drawio.xml"
@@ -104,3 +112,5 @@ package() {
     install -vDm644 "build/$size.png" "$pkgdir/usr/share/icons/hicolor/$size/apps/drawio.png"
   done
 }
+
+makedepends+=(jq p7zip)
