diff --git PKGBUILD PKGBUILD
index 6a2fc15eb..7d3bb9ae8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -41,8 +41,10 @@ optdepends=(
   'webkit2gtk: Browser Preview plugin'
 )
 _tag=4a111ecbdc3060b36ef6ffb23510a0f0b52129be
-source=(pantheon-code::git+https://github.com/elementary/code.git#tag=${_tag})
-sha256sums=(SKIP)
+source=(pantheon-code::git+https://github.com/elementary/code.git#tag=${_tag}
+        "fix_positional_args.patch")
+sha256sums=('SKIP'
+            '98581ab4a54a6ba1ffaafe628d135f2cb1965df4f25ad6a41fadc02ef0a6b343')
 
 pkgver() {
   cd pantheon-code
@@ -50,6 +52,11 @@ pkgver() {
   git describe --tags
 }
 
+prepare() {
+  cd pantheon-code
+  patch -Np1 -i "../fix_positional_args.patch"
+}
+
 build() {
   arch-meson pantheon-code build
   ninja -C build
