diff --git PKGBUILD PKGBUILD
index 9f11265..2e8e85a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,9 +18,11 @@ provides=(libtracker-{control,miner,sparql}-2.0.so)
 options=(debug)
 _commit=b2739625209c564192b339814264084046e1cf17  # tracker-2.3
 source=("git+https://gitlab.gnome.org/GNOME/tracker.git#commit=$_commit"
-        0001-libtracker-data-Workaround-SQLite-3.35.x-bug.patch)
+        0001-libtracker-data-Workaround-SQLite-3.35.x-bug.patch
+        tracker-fix-slow-tests.patch)
 sha256sums=('SKIP'
-            '2dc030fb047e99420bd330d5eee1fd70bd26a7ee0adf31f90c724bd0732ed67d')
+            '2dc030fb047e99420bd330d5eee1fd70bd26a7ee0adf31f90c724bd0732ed67d'
+            'b3478c1860a61174ce70e4d3aab54ccac3e073f47bdf6b14aa1a8af0376b2794')
 
 pkgver() {
   cd tracker
@@ -32,6 +34,7 @@ prepare() {
 
   # https://gitlab.gnome.org/GNOME/tracker/-/merge_requests/382
   git apply -3 ../0001-libtracker-data-Workaround-SQLite-3.35.x-bug.patch
+  git apply -Np1 ../tracker-fix-slow-tests.patch
 }
 
 build() {
@@ -40,7 +43,7 @@ build() {
 }
 
 check() {
-  dbus-run-session meson test -C build --print-errorlogs -t 3
+  dbus-run-session meson test -C build --print-errorlogs -t 10
 }
 
 package() {
