diff --git a/utils/trackertestutils/helpers.py b/utils/trackertestutils/helpers.py
index 2b218e5d0..cbd9450bd 100644
--- a/utils/trackertestutils/helpers.py
+++ b/utils/trackertestutils/helpers.py
@@ -240,6 +240,9 @@ class StoreHelper (Helper):
     def start(self, command_args=None, extra_env=None):
         Helper.start(self, command_args, extra_env)

+        import time
+        time.sleep(5)
+
         self.resources = Gio.DBusProxy.new_sync(
             self.bus, Gio.DBusProxyFlags.DO_NOT_AUTO_START, None,
             self.TRACKER_BUSNAME, self.TRACKER_OBJ_PATH, self.RESOURCES_IFACE)

diff --git a/tests/libtracker-sparql/tracker-sparql-test.c b/tests/libtracker-sparql/tracker-sparql-test.c
index 3b60d18be..3deaa5ae6 100644
--- a/tests/libtracker-sparql/tracker-sparql-test.c
+++ b/tests/libtracker-sparql/tracker-sparql-test.c
@@ -304,7 +304,7 @@ test_tracker_sparql_nb237150 (void)
 	 * never returns
 	 */
 	g_test_trap_subprocess ("/libtracker-sparql/tracker-sparql/nb237150/subprocess",
-	                        G_USEC_PER_SEC * 2,
+	                        G_USEC_PER_SEC * 20,
 	                        G_TEST_SUBPROCESS_INHERIT_STDOUT |
 	                        G_TEST_SUBPROCESS_INHERIT_STDERR);

