--- PKGBUILD
+++ PKGBUILD
@@ -32,6 +32,7 @@ makedepends=(clang
              lld
              llvm
              ninja
+             nodejs-lts-iron
              npm
              pciutils
              pipewire
@@ -42,7 +43,10 @@ makedepends=(clang
              python-six
              qt5-base
              wget
-             yarn)
+             yarn
+             go
+             p7zip
+             jdk17-openjdk)
 optdepends=('kde-cli-tools: file deletion support (kioclient5)'
             'pipewire: WebRTC desktop sharing under Wayland'
             'qt5-base: enable Qt5 with --enable-features=AllowQt'
@@ -62,7 +66,20 @@ source=("git+https://github.com/electron/electron.git#tag=v$pkgver"
         std-vector-non-const.patch
         use-system-libraries-in-node.patch
         libxml2-2.12.patch
-        icu-74.patch)
+        icu-74.patch
+        'REVERT-problematic-signal-handling-workaround.patch'
+        "$pkgname-support-riscv64-in-electron_runtime_api_delegate.cc.patch"
+        "$pkgname-extensions-common-api-runtime.json-support-riscv64.patch"
+        "$pkgname-riscv-angle.patch"
+        "$pkgname-riscv-sandbox.patch"
+        "$pkgname-riscv-crashpad.patch"
+        "$pkgname-riscv-dav1d.patch"
+        "$pkgname-riscv-base.patch"
+        "$pkgname-swiftshader-use-system-llvm.patch"
+        "$pkgname-swiftshader-LLVMJIT-AddressSanitizerPass-dead-code-remove.patch"
+        "$pkgname-gclient-ignore-prebuilt-platform-specific-deps.patch"
+        "$pkgname-deps-parser.py"
+        "$pkgname-partition_alloc.gni-add-missing-riscv64-definition.patch")
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
@@ -76,7 +93,20 @@ sha256sums=('SKIP'
             '893bc04c7fceba2f0a7195ed48551d55f066bbc530ec934c89c55768e6f3949c'
             'ff588a8a4fd2f79eb8a4f11cf1aa151298ffb895be566c57cc355d47f161f53f'
             'bfae9e773edfd0ddbc617777fdd4c0609cba2b048be7afe40f97768e4eb6117e'
-            '547e092f6a20ebd15e486b31111145bc94b8709ec230da89c591963001378845')
+            '547e092f6a20ebd15e486b31111145bc94b8709ec230da89c591963001378845'
+            '87d2c69292ac55ea1784f812c30acdbc341d4333d790fab3439a6b60393bb6c9'
+            '93b2d190448ed94b089eb6c577dba2b81e07d377c8195f76cba6995305d9ed14'
+            '840947515347933da383408711696150cf8020b35d7abe1265b87084dfe1f3df'
+            'a191ef5190e5f823e6be078df53ed1a888558673040985d79347a26644c39b63'
+            'a89ab6edda8dcc6349bd357ccb0d5bcd8e2b92a2c5891437e14008da3f2e7f88'
+            '85644fd6b1a64e7cf76f690f1427010cdf773938987f7e1a93b2977873707a4f'
+            '32811636eecd187dd386bbdd0fa1216728d6010f96fae8f1714dded04a95a5f4'
+            'aab631f5b6ea316b08bd290b52f60ed7715fceabd5c8de67535271fbbe2ae794'
+            'd901e905a9b4303e6334bf39475bcbcdf22959796954de66507857108d53570f'
+            'cf80c0d70f8933a4495c71d7be681c1457a69d26e95d2ad41da4bb02b7cbee4c'
+            'b28b64181c46549e94e8c26d5401b0036991c18c60040c444a262484addd0d0d'
+            '4b04b17c997b0f22b4d9ea1940809fbc29dea230c14643cf97a65a4e154bcb3a'
+            '8d8141ad3dd6d4ddc79e1c29e613d551d70ed5d9bb67c79538e7e80d89470420')
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
 # Keys are the names in the above script; values are the dependencies in Arch
@@ -131,6 +161,10 @@ solutions = [
   },
 ]
 EOF
+ 
+  # Pin depot_tools
+  local _depot_tools_rev="$(python "$pkgname-deps-parser.py" depot_tools chromium-mirror/DEPS)"
+  git -C depot_tools checkout $_depot_tools_rev
 
   export PATH+=":$PWD/depot_tools" DEPOT_TOOLS_UPDATE=0
   export VPYTHON_BYPASS='manually managed python not supported by chrome operations'
@@ -138,11 +172,35 @@ EOF
   echo "Linking chromium from sources..."
   ln -s chromium-mirror src
 
+  echo "Building build tools that doesn't have prebuilt binaries for riscv64..."
+  export GOBIN="$HOME/bin"
+  export PATH="$PATH:$HOME/bin"
+
+  # Install the build tools
+  local infra_rev=$(python "$pkgname-deps-parser.py" infra src/DEPS)
+  curl "https://chromium.googlesource.com/infra/infra/+/$infra_rev/DEPS?format=text" | base64 -d > DEPS_infra
+  local luci_go_rev=$(python "$pkgname-deps-parser.py" luci_go DEPS_infra)
+  go install "go.chromium.org/luci/cipd/client/cmd/...@$luci_go_rev"
+
+  # Fix .cipd-bin problem
+  mkdir -p depot_tools/.cipd_bin
+  GOBIN=$(realpath depot_tools/.cipd_bin) go install "go.chromium.org/luci/auth/client/cmd/...@$luci_go_rev"
+
+  patch -Np0 -i $pkgname-gclient-ignore-prebuilt-platform-specific-deps.patch
+
   depot_tools/gclient.py sync -D \
       --nohooks \
       --with_branch_heads \
       --with_tags
 
+  # Install esbuild (version needs to be locked, not feasible to add it to make dependencies)
+  local esbuild_ver=$(python "$pkgname-deps-parser.py" esbuild src/DEPS)
+  GOBIN=$(realpath src/third_party/devtools-frontend/src/third_party/esbuild/) go install "github.com/evanw/esbuild/cmd/esbuild@v$esbuild_ver"
+
+  # Replace the bundled x86_64 JDK with system JDK17
+  rm -rf src/third_party/jdk/current
+  ln -s /usr/lib/jvm/java-17-openjdk src/third_party/jdk/current
+
   pushd src/electron
   patch -Np1 -i ../../std-vector-non-const.patch
   popd
@@ -194,6 +252,24 @@ EOF
   # Fix build with ICU 74
   patch -Np1 -i ../icu-74.patch
 
+  # Chromium Bug fixes
+  patch -R -p1 -i "${srcdir}/REVERT-problematic-signal-handling-workaround.patch"
+
+  # RISC-V Patches
+  patch -Np1 -d third_party/crashpad/crashpad < "${srcdir}/$pkgname-riscv-crashpad.patch"
+  patch -Np1 -i ../$pkgname-riscv-angle.patch
+  patch -Np1 -i ../$pkgname-riscv-sandbox.patch
+  patch -Np1 -i ../$pkgname-riscv-dav1d.patch
+  patch -Np1 -i ../$pkgname-riscv-base.patch
+  patch -Np1 -d electron < ../$pkgname-support-riscv64-in-electron_runtime_api_delegate.cc.patch
+  patch -Np0 -i ../$pkgname-extensions-common-api-runtime.json-support-riscv64.patch
+  patch -Np1 -i ../$pkgname-partition_alloc.gni-add-missing-riscv64-definition.patch
+
+  # Build failes with the bundled LLVM 10.
+  patch -Np1 -i ../$pkgname-swiftshader-use-system-llvm.patch
+  patch -Np1 -i ../$pkgname-swiftshader-LLVMJIT-AddressSanitizerPass-dead-code-remove.patch
+
+
   # Allow building against system libraries in official builds
   echo "Patching Chromium for using system libraries..."
   sed -i 's/OFFICIAL_BUILD/GOOGLE_CHROME_BUILD/' \
@@ -225,11 +301,19 @@ build() {
   export AR=ar
   export NM=nm
 
+  # Disable RVV because the build of libyuv at this commit is broken on riscv.
+  CFLAGS+=' -D LIBYUV_DISABLE_RVV'
+  CXXFLAGS+=' -D LIBYUV_DISABLE_RVV'
+
   # Facilitate deterministic builds (taken from build/config/compiler/BUILD.gn)
   CFLAGS+='   -Wno-builtin-macro-redefined'
   CXXFLAGS+=' -Wno-builtin-macro-redefined'
   CPPFLAGS+=' -D__DATE__=  -D__TIME__=  -D__TIMESTAMP__='
 
+  # Remove flags that are causing weird bugs on riscv64
+  CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=2}
+  CXXFLAGS=${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=2}
+
   # Let Chromium set its own symbol level
   CFLAGS=${CFLAGS/-g }
   CXXFLAGS=${CXXFLAGS/-g }
@@ -259,7 +343,6 @@ build() {
     host_toolchain = "//build/toolchain/linux/unbundle:default"
     clang_base_path = "/usr"
     clang_use_chrome_plugins = false
-    symbol_level = 0 # sufficient for backtraces on x86(_64)
     chrome_pgo_phase = 0
     treat_warnings_as_errors = false
     disable_fieldtrial_testing_config = true
@@ -273,6 +356,7 @@ build() {
     use_system_libffi = true
     icu_use_data_file = false
     is_component_ffmpeg = false
+    symbol_level = 1
   '
   gn gen out/Release \
       --args="import(\"//electron/build/args/release.gn\") ${GN_EXTRA_ARGS}"
