--- PKGBUILD
+++ PKGBUILD
@@ -41,7 +41,7 @@ source=(
   $pkgbase.desktop
   $pkgbase.xpm
 )
-sha256sums=('3a31afb734ec5c34bc913e63006665b7096c48bbeeda02759629b503b0fc5f3a'
+sha256sums=('6e5679cdfaaef9c7e89e296395220ce2c133ed3dcad51a478667336c6eaec706'
             'ccff66979249bd4ab4ba8918660f194eb90eb0ae231b16e36a6cecdcf471883f'
             'a435f44425f5432c074dee745d8fbaeb879038ec1f1ec64f037c74662f09aca8')
 
