--- PKGBUILD
+++ PKGBUILD
@@ -52,6 +52,7 @@
   sqlite
   systemd-libs
   thrift
+  util-linux-libs
   yara
   zlib
   zstd
@@ -166,3 +167,6 @@
   install -vDm644 tools/deployment/linux_packaging/osqueryd.sysconfig "$pkgdir/etc/sysconfig/osqueryd"
   install -vDm644 tools/deployment/linux_packaging/rpm/osqueryd.service "$pkgdir/usr/lib/systemd/system/osqueryd.service"
 }
+
+source+=(osquery-riscv64-support.patch)
+b2sums+=('0c79c7ffa38ff3a9bb6adbacb44b6d5b5f4151256c0f5e79abf678a238d08cac5df116677fc36b66f7716341b9f468698533920b7e157f9423963bab2ff66cd9')
