--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,7 @@
   util-linux-libs
   nasm
   python
-  riscv64-linux-gnu-gcc
+  x86_64-linux-gnu-gcc
 )
 options=(!makeflags)
 source=(
@@ -190,12 +190,9 @@
   )
 
   cd $pkgbase
-  # This is used as the prefix for both gcc and objcopy
-  # so it cannot be set to x86_64-linux-gnu-
-  # (binutils does not provide x86_64-linux-gnu-objcopy)
-  export GCC_BIN=""
+  # Cross-compile IA32/X64 binaries on non-x86 hosts.
+  export GCC_BIN="x86_64-linux-gnu-"
   export GCC_AARCH64_PREFIX="aarch64-linux-gnu-"
-  export GCC_RISCV64_PREFIX="riscv64-linux-gnu-"
   echo "Building base tools"
   make -C BaseTools
   # expose build tooling in PATH
