diff --git PKGBUILD PKGBUILD
index 3e20689b6..73314d0da 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,14 +23,17 @@ install=opensips.install
 options=('!emptydirs' 'zipman' '!makeflags' 'docs')
 source=(https://opensips.org/pub/opensips/${pkgver}/opensips-${pkgver}.tar.gz
         openldap-2.5.patch
-	opensips.service)
+	opensips.service
+	fix_nonliteral_format.patch)
 sha256sums=('557b33db1436d8212ac122f299481bee3f2957f060a2bc0b4256430faf8b3bbc'
             '48069c579676e6d38673152b22481624d98cf17ce7ae46bdff51f7df61c03c5e'
-            'c2fec4be085b108db10834fa9832e98d696c2de6408f85f96cf89c13bf6be819')
+            'c2fec4be085b108db10834fa9832e98d696c2de6408f85f96cf89c13bf6be819'
+            'da086acf6bd3a38577eb776f6305b0aee375bd4661136bd097956e8b92d6985a')
 
 prepare() {
   cd "$srcdir"/$pkgname-$pkgver/
 
+  patch -Np1 -i ../fix_nonliteral_format.patch
   patch -Np1 -i ../openldap-2.5.patch
 
   sed -i 's|sbin|bin|g' Makefile
