diff --git PKGBUILD PKGBUILD
index cbda150..86c2c15 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,6 +25,7 @@
   make \
     PREFIX=/usr \
     SHARED=1 \
+    MAINTAINER_MODE=1 \
     USE_SYSTEM_XXHASH=1 \
     USE_SYSTEM_OPENCL=1 \
     USE_SYSTEM_ZLIB=1
@@ -36,6 +37,7 @@
     DESTDIR="${pkgdir}" \
     PREFIX=/usr \
     SHARED=1 \
+    MAINTAINER_MODE=1 \
     USE_SYSTEM_XXHASH=1 \
     USE_SYSTEM_OPENCL=1 \
     USE_SYSTEM_ZLIB=1 \
