diff -r -u psiconv-0.9.9-orig/program/psiconv/psiconv.c psiconv-0.9.9/program/psiconv/psiconv.c
--- psiconv-0.9.9-orig/program/psiconv/psiconv.c	2014-10-22 19:45:57.000000000 +0000
+++ psiconv-0.9.9/program/psiconv/psiconv.c	2021-09-28 19:35:48.667605206 +0000
@@ -48,6 +48,8 @@
 static void print_version(void);
 static void strtoupper(char *str);
 
+psiconv_list fileformat_list; /* of struct psiconv_fileformat */
+
 void print_help(void)
 {
   fileformat ff;
diff -r -u psiconv-0.9.9-orig/program/psiconv/psiconv.h psiconv-0.9.9/program/psiconv/psiconv.h
--- psiconv-0.9.9-orig/program/psiconv/psiconv.h	2014-10-22 19:46:03.000000000 +0000
+++ psiconv-0.9.9/program/psiconv/psiconv.h	2021-09-28 19:35:01.340937897 +0000
@@ -52,7 +52,7 @@
   output_function *output;
 } *fileformat;
 
-psiconv_list fileformat_list; /* of struct psiconv_fileformat */
+extern psiconv_list fileformat_list; /* of struct psiconv_fileformat */
 
 
 #endif /* PSICONV_H */
