--- PKGBUILD
+++ PKGBUILD
@@ -12,12 +12,15 @@ license=('GPL')
 depends=('graphicsmagick')
 makedepends=('bc')
 backup=('etc/psiconv/psiconv.conf')
-source=("http://www.frodo.looijaard.name/system/files/software/${pkgname}/${pkgname}-${pkgver}.tar.gz")
-md5sums=('286e427b10f4d10aaeef1944210a2ea6')
+source=("http://www.frodo.looijaard.name/system/files/software/${pkgname}/${pkgname}-${pkgver}.tar.gz"
+        "psiconv.patch")
+md5sums=('286e427b10f4d10aaeef1944210a2ea6'
+         '4fb974d3ae3058de435050d1595f269b')
 
 prepare() {
   cd $pkgname-$pkgver
   sed -e 's|Magick-config|GraphicsMagick-config|g' -i configure.in
+  patch -i ../psiconv.patch --strip 1
   autoreconf -vi
 }
 
