--- a/Externals/cpp-ipc/cpp-ipc/include/libipc/imp/detect_plat.h
+++ b/Externals/cpp-ipc/cpp-ipc/include/libipc/imp/detect_plat.h
@@ -69,6 +69,8 @@
 # define LIBIPC_INSTR_ARM64
 #elif defined(_M_ARM) || defined(_ARM) || defined(__arm__) || defined(__arm)
 # define LIBIPC_INSTR_ARM32
+#elif defined(__riscv)
+# define LIBIPC_INSTR_RISCV
 #else
 # error "This instruction set is unsupported."
 #endif
