--- PKGBUILD
+++ PKGBUILD
@@ -65,6 +65,8 @@ pkgver() {
 build() {
   export CC=clang
   export CXX=clang++
+  export CFLAGS="$CFLAGS -Wl,-plugin-opt=-target-abi=lp64d"
+  export CXXFLAGS="$CXXFLAGS -Wl,-plugin-opt=-target-abi=lp64d"
   cmake -S libretro-dolphin -B build -G Ninja \
     -DCMAKE_BUILD_TYPE=None \
     -DCMAKE_INSTALL_PREFIX=/usr \
@@ -72,6 +74,7 @@ build() {
     -DENABLE_NOGUI=OFF \
     -DENABLE_QT=OFF \
     -DENABLE_TESTS=OFF \
+    -DENABLE_GENERIC=ON \
     -DLIBRETRO=ON \
     -DUSE_SHARED_ENET=ON \
     -Wno-dev
