--- PKGBUILD
+++ PKGBUILD
@@ -73,6 +73,7 @@
     git config submodule.${submodule}.url ../${submodule##*/}
     git -c protocol.file.allow=always submodule update ${submodule}
   done
+  patch -Np1 -i "${srcdir}/cpp-ipc-riscv.patch"
 }
 
 
@@ -83,6 +84,7 @@
     -DENABLE_ANALYTICS=OFF \
     -DENABLE_LTO=ON \
     -DENABLE_QT=OFF \
+    -DENABLE_GENERIC=ON \
     -DLIBRETRO=ON \
     -DLIBRETRO_STATIC=ON \
     -Wno-dev
@@ -94,3 +96,6 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+source+=(cpp-ipc-riscv.patch)
+b2sums+=('6d05049427adaef03e873a3debcfdc5e6b1d64145d7dd2b0a6e786c347992f4395ef63ae6e5eca3f572e94e62fb8e1bd9f430f775857201d53b165e3d612086e')
