Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 456595)
+++ PKGBUILD	(working copy)
@@ -13,11 +13,13 @@
 provides=('awk')
 options=('debug')
 source=(https://ftp.gnu.org/pub/gnu/${pkgname}/${pkgname}-${pkgver}.tar.gz{,.sig}
-        0001-Add-missing-UPREF.patch)
+        0001-Add-missing-UPREF.patch
+        0001-Fix-negative-NaN-issue-on-RiscV.patch)
 validpgpkeys=('D1967C63788713177D861ED7DF597815937EC0D2') # Arnold Robbins
 sha256sums=('ef5af4449cb0269faf3af24bf4c02273d455f0741bf3c50f86ddc09332d6cf56'
             'SKIP'
-            'd39fa487f89c743ba55ed0b5eeb9fe33db4bd7010bf5f61f8aa1a9541a18775e')
+            'd39fa487f89c743ba55ed0b5eeb9fe33db4bd7010bf5f61f8aa1a9541a18775e'
+            '530aad2d2155ff01c817cd491dfae6ccb57b9a6164bb178f25f8240d3fc4859a')
 
 prepare() {
   cd ${pkgname}-${pkgver}
@@ -24,6 +26,8 @@
 
   # https://bugs.gentoo.org/868567
   patch -Np1 -i ../0001-Add-missing-UPREF.patch
+
+  patch -Np1 -i ../0001-Fix-negative-NaN-issue-on-RiscV.patch
 }
 
 build() {
