diff --git PKGBUILD PKGBUILD
index 8ef52e5d3..cf3282445 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,9 +14,12 @@ optdepends=('dssi-host: for DSSI plugin'
             'jack: for standalone application'
             'lv2-host: for lv2 plugin'
             'vst-host: for VST plugin')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/pdesaulniers/${pkgname}/releases/download/v${pkgver}/${pkgname}-v${pkgver}-source.tar.gz")
-sha512sums=('d1bed1a59047340e4eab7a763ff2e29ab03ad95d74743262eefede4e34c9345351e8b6ffb9318bc2101017ea512366a18bfc0a4e40ad11ad098bd9a607590a4e')
-b2sums=('713e4cf82e508b6ba781cc40d2d2f46f1221514a11e4db0eeb0f1a00509b5032d1396e25b48e7e0b8128ba843098b9923bf135fd704c559fed8f80a3453e0f09')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/pdesaulniers/${pkgname}/releases/download/v${pkgver}/${pkgname}-v${pkgver}-source.tar.gz"
+        "https://raw.githubusercontent.com/felixonmars/archriscv-packages/master/wolf-spectrum/add-x86_64-support.patch")
+sha512sums=('d1bed1a59047340e4eab7a763ff2e29ab03ad95d74743262eefede4e34c9345351e8b6ffb9318bc2101017ea512366a18bfc0a4e40ad11ad098bd9a607590a4e'
+            '6b2ed11d081fdd1be63d6d68e395d10f6e740e433c9cc543744638b12c88f985af1a6c3b22f40d3da3d528614541183bf05bca59a9868493a5919a8d01623873')
+b2sums=('713e4cf82e508b6ba781cc40d2d2f46f1221514a11e4db0eeb0f1a00509b5032d1396e25b48e7e0b8128ba843098b9923bf135fd704c559fed8f80a3453e0f09'
+        '7aae8eb22d8fa5539c0f150ab75a737360010273374249e5fa2eb0b9576c2a714f34f050a724b65b443e1ad44263c458ce0b0d89881910f80777fe5ed8a6ab20')
 
 prepare() {
   mv -v "$pkgname" "$pkgname-$pkgver"
@@ -29,6 +32,8 @@ prepare() {
           --pkgdesc "${pkgdesc}" \
           --genericname "Wave Shaper" \
           --categories "AudioVideo;Audio"
+
+  patch -Np1 -i ../add-x86_64-support.patch
 }
 
 build() {
