diff --git PKGBUILD PKGBUILD
index 38ab1a15..153fbc11 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,14 @@
 conflicts=('intel-tbb')
 provides=("intel-tbb=$pkgver")
 replaces=('intel-tbb')
-source=(https://github.com/oneapi-src/oneTBB/archive/v$pkgver/$pkgname-$pkgver.tar.gz)
-sha512sums=('0e7b71022e397a6d7abb0cea106847935ae79a1e12a6976f8d038668c6eca8775ed971202c5bd518f7e517092b67af805cc5feb04b5c3a40e9fbf972cc703a46')
+source=(https://github.com/oneapi-src/oneTBB/archive/v$pkgver/$pkgname-$pkgver.tar.gz
+        tbb-riscv.patch::https://github.com/oneapi-src/oneTBB/pull/550.patch)
+sha512sums=('0e7b71022e397a6d7abb0cea106847935ae79a1e12a6976f8d038668c6eca8775ed971202c5bd518f7e517092b67af805cc5feb04b5c3a40e9fbf972cc703a46'
+            'dfba52da2bd6f463adf658e95a41912c60f12f33b3cdc8e15097dd88c0306fb09c8e269862fda74bbb07cd5091491f611086fb011c79ca762f2be8521cb2fc0b')
+
+prepare() {
+  patch -d oneTBB-$pkgver -p1 < tbb-riscv.patch
+}
 
 build() {
   cd oneTBB-$pkgver
