--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,7 @@ makedepends=('alsa-lib' 'cmake' 'dssi' 'fluidsynth' 'glib2' 'jack' 'ladspa'
 'liblo' 'liblrdf' 'libsamplerate' 'libsndfile' 'lilv' 'lv2' 'qt5-tools'
 'rtaudio' 'rubberband' 'serd')
 provides=('dssi-host' 'ladspa-host' 'lv2-host' 'vst-host')
+options=(!lto)
 source=("$pkgname-$pkgver.tar.gz::https://github.com/muse-sequencer/muse/archive/refs/tags/${pkgver}.tar.gz")
 sha512sums=('316d8afaa4412c1de3febd06f517245fcc0f380b78efab8a59041653c70828496e95f4ebd2e06ea080fb494c69ab73457e77264166be50b85abf093835d6c2d4')
 b2sums=('7b9fd19d5f9a5bf776d00f3c4695b7fb214de4e5be629c05edb82daa0113784b7cd42e2204ebca83fd7f6b82801a0bb9b815cc5618fa48d8d512a30af3e9b09f')
