diff --git PKGBUILD PKGBUILD
index 045345e1..0c093a1e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,10 @@ depends=(geocode-glib geoclue gjs libchamplain libgee folks gfbgraph libhandy)
 makedepends=(gobject-introspection git meson yelp-tools)
 groups=(gnome)
 _commit=47ac76371c1616f936d298053f03562f442d5c0d  # tags/v41.2^0
-source=("git+https://gitlab.gnome.org/GNOME/gnome-maps.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnome-maps.git#commit=$_commit"
+        "fix_positional_args.patch")
+sha256sums=('SKIP'
+            '698d36ba505cb0cb013574b4eaa2a0a121df91639932944064e4a8668ba0d06d')
 
 pkgver() {
   cd $pkgname
@@ -22,6 +24,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../fix_positional_args.patch
 }
 
 build() {
