--- PKGBUILD
+++ PKGBUILD
@@ -12,18 +12,18 @@ license=('BSD' 'GPL')
 depends=('gcc-libs' 'libogg')
 makedepends=('nasm' 'doxygen')
 source=(https://downloads.xiph.org/releases/flac/${pkgbase}-${pkgver}.tar.xz
-        https://github.com/xiph/flac/commit/2e7931c.patch
-        https://github.com/xiph/flac/commit/ced7f68.patch)
+        https://github.com/xiph/flac/commit/2e7931c.diff
+        https://github.com/xiph/flac/commit/ced7f68.diff)
 sha512sums=('d6417e14fab0c41b2df369e5e39ce62a5f588e491af4d465b0162f74e171e5549b2f061867f344bfbf8aaccd246bf5f2acd697e532a2c7901c920c69429b1a28'
-            '56182fb46cbefcd34137ec0e93ec540bc029e1e2cb4787444178ee941ae03cc73bd360c87ec3735a6ca524044d28f22253fe37a42fdb93aea7def65178ea663b'
-            'c5f17f6afa7358ba1dd610b7fe9d576f217ac7f71a8edf7833cfa2ed6a971a8d2c69468e96c35b97ffb6b019cb1be89dc23fe839ee19e445a6eef6cc74d2f435')
+            '65236c01d1252496940c10294edb5b964bf54cc6be2f4fd47c4f12e8d25a787b99e04b8cd01283c3f7040d27efce568d7a03568c6b972698ac4f58ff9a0041de'
+            '7bce5dd425a246265fc721d9688215618c0a59bcd68bccf399dd6ba77f848ce51f9169efd4564e6bdfedffd1d205eaf4ea119c40e5397f11ccd0ae2605b49f84')
 
 prepare() {
   cd ${pkgbase}-${pkgver}
   # CVE-2020-0499
-  patch -p1 -i ../2e7931c.patch
+  patch -p1 -i ../2e7931c.diff
   # Add some overflow checks for residual bits calculation
-  patch -p1 -i ../ced7f68.patch
+  patch -p1 -i ../ced7f68.diff
 }
 
 build() {
