--- PKGBUILD
+++ PKGBUILD
@@ -53,6 +53,8 @@
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i "$srcdir/fix-testtools-try-import.patch"
+  patch -Np1 -i "$srcdir/fix-rabbitmq-cli-nodename.patch"
   patch -Np1 -i ../$pkgname-skip-failing-tests.patch
   # Move tests out of package to avoid installing.
   mv pifpaf/tests tests
@@ -75,3 +77,8 @@
   cd $pkgname
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=("fix-testtools-try-import.patch::https://github.com/jd/pifpaf/commit/0dc95746e00fff298a421767fe27c058d1787128.patch"
+         'fix-rabbitmq-cli-nodename.patch')
+sha512sums+=('085cc63a421778dacba5f82cfb7125c9fd948212ddae2cd9562f860449be4604529325874404f72d2ddbd28a9c4682183de519891db724e6d420bcd5a4fa1581'
+             '0812d1062312cb8783a5e7c5bbac772e64d9d48d43cef41ea7fa735fc20c931582b836867382977e7f9c4f7f0cb915e79e8349df42203c3748b55e02de2b57ae')
