--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,8 @@ b2sums=('dbfce58ad418b0cae3ce063ca50286c3c9897a13dc868eb30a8b16c340748a8de576066
 prepare() {
   cd $_name-$pkgver
   export RUSTUP_TOOLCHAIN=stable
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
