diff --git PKGBUILD PKGBUILD
index 36d8b55..9c18304 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,9 @@ b2sums=('769f343978bc86bd3da51c6412dbbd500fa2f9597c656a8cbfb4b3160c261469d37ef8d
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
