--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,8 @@ sha512sums=('7766aec5cdf76e8dc39b3e2c6feb1e252545bab2999437fc45b2b42de475c0384e7
 prepare() {
   cd $pkgname-v$pkgver
 
+  patch -Np1 -i ../netdata-libbpf-path.patch
+
   sed -i "s/CMAKE_CXX_STANDARD 14/CMAKE_CXX_STANDARD 17/" CMakeLists.txt
   sed -i "s|usr/sbin|usr/bin|g" CMakeLists.txt
   sed -i "s|usr/libexec|usr/lib|g" CMakeLists.txt
@@ -82,3 +84,6 @@ local-listeners,network-viewer.plugin,ndsudo,freeipmi.plugin,nfacct.plugin}
   rm -rf "${pkgdir}/usr/lib/netdata/system"
   rm -rf "${pkgdir}/var/run"
 }
+
+source+=(netdata-libbpf-path.patch)
+sha512sums+=('f157fe75493623602d496ec2a6e13cd5a12501383471ec89446905eb124e4cc41f05e4c5319d7691450a665d8ee70d0b0731b7a0a4b4beb35818bb7b9c96213f')
