diff --git PKGBUILD PKGBUILD
index 57aeb4a..c1f1563 100644
--- PKGBUILD
+++ PKGBUILD
@@ -35,7 +35,7 @@ prepare() {
   sed -i "s|usr/libexec|usr/lib|g" CMakeLists.txt
 
   # libbpf sources for eBPF.plugin
-  CFLAGS="${CFLAGS} -fPIC" "packaging/bundle-libbpf.sh" "../netdata-v${pkgver}" "other"
+  CFLAGS="${CFLAGS} -fPIC" LIBDIR=/usr/lib "packaging/bundle-libbpf.sh" "../netdata-v${pkgver}" "other"
   CFLAGS="${CFLAGS} -fPIC" "packaging/bundle-ebpf-co-re.sh" "../netdata-v${pkgver}"
 }
 
