diff --git PKGBUILD PKGBUILD
index 3749a18..499a1b4 100644
--- PKGBUILD
+++ PKGBUILD
@@ -6,18 +6,18 @@ pkgver=1.3.7rc6
 pkgrel=2
 pkgdesc="Library to parse proxy auto-config (PAC) files"
 arch=(x86_64)
-url="https://github.com/pacparser/pacparser"
+url="https://github.com/manugarg/pacparser"
 license=('LGPL')
-source=("$pkgbase-$pkgver.tar.gz::https://github.com/pacparser/pacparser/archive/$pkgver.tar.gz")
+source=("$pkgname-$pkgver.tar.gz::https://codeload.github.com/manugarg/pacparser/tar.gz/1.3.7rc6")
 sha512sums=('4489d3ca1d2121013373cf473a449ee09d6eb33840241739c988657335e9c9be98a39dd1cb4d690bbc98f8e7a444a31829f8b99ab64685252b9aed750f5fa768')

 build() {
-  cd $pkgbase-$pkgver
+  cd $pkgname-$pkgver
   make all -C src
 }

 package() {
-  cd $pkgbase-$pkgver
+  cd $pkgname-$pkgver
   make -C src DESTDIR="$pkgdir/" install
 }
