Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1309110)
+++ PKGBUILD	(working copy)
@@ -13,6 +13,11 @@
 source=("https://hackage.haskell.org/packages/archive/$_hkgname/$pkgver/$_hkgname-$pkgver.tar.gz")
 sha256sums=('ea123532a46d24925e7364470ac899d521a9d6a00bb80c505b3696b0e21354cf')
 
+prepare() {
+  cd $_hkgname-$pkgver
+  sed -i 's/arch(ppc64)/arch(ppc64) || arch(riscv64)/' $_hkgname.cabal
+}
+
 build() {
   cd $_hkgname-$pkgver
 
