diff --unified --recursive --text a/textfunc.c b/textfunc.c
--- a/textfunc.c	2021-09-23 14:11:52.384471555 +0000
+++ b/textfunc.c	2021-09-23 14:13:10.714505036 +0000
@@ -227,7 +227,7 @@
 
 	while((percent=strchr(format,'%'))) {
 		*percent=0;
-		printf(format);
+		printf("%s", format);
 		*percent='%';
 		code=percent+1;
 		while(*code && (code[0] != '%' && !isalpha(*code))) code++;
@@ -354,7 +354,7 @@
 		}
 		
 	}
-	printf(format);
+	printf("%s", format);
 }
 
 
