diff --git PKGBUILD PKGBUILD
index 14f7c54aa..2000e828a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,9 +12,17 @@ depends=('ncurses')
 makedepends=('gtk2')
 optdepends=('gtk2: run gmp3info GUI')
 source=(https://ibiblio.org/pub/linux/apps/sound/mp3-utils/$pkgname/$pkgname-$pkgver.tgz
-        gmp3info.desktop)
+        gmp3info.desktop
+        fix-format-not-a-string-literal-and-no-format-arguments.patch)
 md5sums=('cb7b619a10a40aaac2113b87bb2b2ea2'
-         '70071d6503161a5f466d34ec9649b2fc')
+         '70071d6503161a5f466d34ec9649b2fc'
+         'c2241aeeed622c613022ac29264dacdf')
+
+prepare(){
+  cd "$srcdir"/$pkgname-$pkgver
+
+  patch  -Np1 -i "${srcdir}/fix-format-not-a-string-literal-and-no-format-arguments.patch"
+}
 
 build() {
   cd "$srcdir"/$pkgname-$pkgver
