--- PKGBUILD
+++ PKGBUILD
@@ -37,8 +37,14 @@ makedepends=(
   meson
 )
 _tag=08b6fade0bc95749acc9820cb9688aaf39d00c87
-source=(git+https://github.com/TASEmulators/desmume.git#tag=${_tag})
-b2sums=(SKIP)
+source=(git+https://github.com/TASEmulators/desmume.git#tag=${_tag}
+        fix-compiling-on-non-arm-arch.patch::https://github.com/TASEmulators/desmume/commit/d4afd4977ce8b7ba6293946c74914afe79536059.patch)
+b2sums=('SKIP'
+        '2c9ada4f8f6a9c61dceddf1c29c92e942c27e86edd73e1dc930cf9a50fe6fb6f3c4770634bd141464ce159a106a6cb13f56388eab9bc6353fcc808c70c1dfb67')
+
+prepare() {
+  patch -d desmume -Np1 -i ${srcdir}/fix-compiling-on-non-arm-arch.patch
+}
 
 pkgver() {
   cd desmume
