--- PKGBUILD
+++ PKGBUILD
@@ -37,8 +37,15 @@ makedepends=(
   meson
 )
 _tag=08b6fade0bc95749acc9820cb9688aaf39d00c87
-source=(git+https://github.com/TASEmulators/desmume.git#tag=${_tag})
-b2sums=(SKIP)
+source=(git+https://github.com/TASEmulators/desmume.git#tag=${_tag}
+        detect-arm-rather-than-exclude-x86-for-arm-specific-code.patch)
+b2sums=('SKIP'
+        'a8cca6cb154295150b1fc643c41107c665e39e9f6e9708a0b38eb303806374f4d9d43f52f3def6311e2ed795631492fc560ee661612e6f3876c03d38d9ef684d')
+
+prepare() {
+  cd desmume
+  patch -Np1 -i ../detect-arm-rather-than-exclude-x86-for-arm-specific-code.patch
+}
 
 pkgver() {
   cd desmume
