diff --git a/libdeja/tests/runner.vala b/libdeja/tests/runner.vala
index 95f6dbe1fe01bb25847c85037a0b03caf03533cf..b721632824783eb7394c7f11d181ef3159d4b2c9 100644
--- a/libdeja/tests/runner.vala
+++ b/libdeja/tests/runner.vala
@@ -1032,7 +1032,7 @@ int main(string[] args)
   if (keyfile.has_group("Duplicity")) {
     var suite = new TestSuite("duplicity");
     suite.add(new TestCase(testname, backup_setup, duplicity_run, backup_teardown));
-    TestSuite.get_root().add_suite(suite);
+    TestSuite.get_root().add_suite((owned)suite);
     found_group = true;
   }
 
@@ -1040,7 +1040,7 @@ int main(string[] args)
   if (keyfile.has_group("Restic")) {
     var suite = new TestSuite("restic");
     suite.add(new TestCase(testname, backup_setup, restic_run, backup_teardown));
-    TestSuite.get_root().add_suite(suite);
+    TestSuite.get_root().add_suite((owned)suite);
     found_group = true;
   }
 #endif
diff --git a/libdeja/tests/unit/unit-tests.vala b/libdeja/tests/unit/unit-tests.vala
index 8cce63178f23e6f0d2c2662c651acd39aa13f070..cbe85ea6cc85b91a201609801c8f6fef788376b3 100644
--- a/libdeja/tests/unit/unit-tests.vala
+++ b/libdeja/tests/unit/unit-tests.vala
@@ -138,7 +138,7 @@ int main(string[] args)
   unit.add(new TestCase("parse-dir", setup, parse_dir, teardown));
   unit.add(new TestCase("parse-version", setup, parse_version, teardown));
   unit.add(new TestCase("prompt", setup, prompt, teardown));
-  TestSuite.get_root().add_suite(unit);
+  TestSuite.get_root().add_suite((owned)unit);
 
   return Test.run();
 }
diff --git a/monitor/tests/scheduler.vala b/monitor/tests/scheduler.vala
index d83da81a46b0ce4b9c35b7d422858c78c571c3b5..beabc13412be5a1d353dc85b378fbeb3477a5637 100644
--- a/monitor/tests/scheduler.vala
+++ b/monitor/tests/scheduler.vala
@@ -242,7 +242,7 @@ int main(string[] args)
   unit.add(new TestCase("period-changes", setup, notices_period_changes, teardown));
   unit.add(new TestCase("period-overdue", setup, period_change_overdue, teardown));
   unit.add(new TestCase("backup-changes", setup, notices_last_backup_changes, teardown));
-  TestSuite.get_root().add_suite(unit);
+  TestSuite.get_root().add_suite((owned)unit);
 
   return Test.run();
 }
