--- PKGBUILD
+++ PKGBUILD
@@ -9,12 +9,16 @@ url="https://wiki.gnome.org/Apps/DejaDup"
 license=(GPL)
 depends=(duplicity gvfs libsecret org.freedesktop.secrets python-gobject python-pydrive libadwaita)
 makedepends=(appstream-glib itstool meson vala)
-source=(https://gitlab.gnome.org/World/$pkgname/-/archive/$pkgver/$pkgname-$pkgver.tar.gz)
+source=(https://gitlab.gnome.org/World/$pkgname/-/archive/$pkgver/$pkgname-$pkgver.tar.gz
+        fix-testsuite.patch::https://gitlab.gnome.org/World/deja-dup/-/commit/60856db2ca1b37079e9a7a2e1cbbbc9d2dcb43fa.diff)
 validpgpkeys=('A3A5C2FC56AE7341D308D8571B50ECA373F3F233') # Michael Terry
-sha256sums=('e1ca20ac02da49e9f0d102b921fa9061a037857d08db16c0ba129958c35ef047')
+sha256sums=('e1ca20ac02da49e9f0d102b921fa9061a037857d08db16c0ba129958c35ef047'
+            '3b2af8aafb165ae3ce531bfd1e0e56427714d9d1afbe11ff30d5d2ea6c745112')
 
 prepare() {
   find $pkgname-$pkgver -name meson.build | xargs sed -e 's|C.UTF-8|C|' -i
+
+  patch -p1 -d $pkgname-$pkgver -Ni ../fix-testsuite.patch
 }
 
 build() {
