--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,8 @@ sha256sums=('21c8eaa53cf3290d2b1405c8cb2cde5f39bc14ef597b328e76f1789b0ef3539a')
 prepare() {
 	cd "$_archive"
 	sed -r -i -e '/^incremental/a opt-level = 3' Cargo.toml
+	echo -e "\n[patch.crates-io]\nskia-bindings = { git = 'https://github.com/hack3ric/rust-skia', branch = 'archrv-0.75.0' }" >> Cargo.toml
+	cargo update -p skia-bindings
 	cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
