--- PKGBUILD
+++ PKGBUILD
@@ -18,17 +18,26 @@ depends=(gcc-libs
          neovim
          sndio)
 makedepends=(cargo
-             cmake)
+             cmake
+             python
+             git
+             gn
+             ninja
+             clang)
 optdepends=('vulkan-intel: vulkan support for intel'
             'libxkbcommon-x11: run on X11 (not needed for wayland)')
 _archive=("$pkgname-$pkgver")
 source=("$url/archive/$pkgver/$_archive.tar.gz")
 sha256sums=('8770dd6977605f9bafa990a60cf8f2ebeba7df16417dab2e8c5583d279ec86ef')
+options=(!lto)
 
 prepare() {
 	cd "$_archive"
 	sed -r -i -e '/^incremental/a opt-level = 3' Cargo.toml
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+	echo -e "\n[patch.crates-io]\nskia-bindings = { git = 'https://github.com/aimixsaka/rust-skia', branch = 'riscv-skia-bindings-0.68.0' }" >> Cargo.toml
+
+	cargo update -p skia-bindings
+	cargo fetch --locked
 }
 
 build() {
@@ -39,6 +48,9 @@ build() {
 	export CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
 	CFLAGS+=' -ffat-lto-objects'
 	export SKIA_USE_SYSTEM_LIBRARIES=true
+	export FORCE_SKIA_BUILD=true
+	export SKIA_GN_COMMAND=gn
+	export SKIA_NINJA_COMMAND=ninja
 	cargo build --frozen --release --features embed-fonts
 }
 
