--- PKGBUILD
+++ PKGBUILD
@@ -17,16 +17,33 @@ makedepends=(cargo
              cmake
              gtk3
              python
-             sdl2)
+             sdl2
+             git
+             gn
+             ninja
+             clang)
 optdepends=('vulkan-intel: vulkan support for intel')
 _archive=("$pkgname-$pkgver")
-source=("$url/archive/$pkgver/$_archive.tar.gz")
-sha256sums=('4926c7fe1033a6f529bac649921c66bf98a03d70d4f0f7a7f7cdd1c74bb33533')
+source=("$url/archive/$pkgver/$_archive.tar.gz"
+        "git+https://github.com/rust-skia/rust-skia.git#tag=0.62.0"
+        "skia_native_build.patch")
+sha256sums=('4926c7fe1033a6f529bac649921c66bf98a03d70d4f0f7a7f7cdd1c74bb33533'
+            'SKIP'
+            '7ec3b586aa492bc4e7417e55f46519e4d3dd40da51d758905b00f8c0913be387')
+options=(!lto)
 
 prepare() {
+	cd rust-skia
+	git submodule update --init --recursive
+	git apply ../skia_native_build.patch
+	cp /usr/bin/ninja skia-bindings/depot_tools/
+	cd ..
+
 	cd "$_archive"
 	sed -i -e '/^incremental/a opt-level = 3' Cargo.toml
-	cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+	echo -e "\n[patch.crates-io]\nskia-bindings = { path = '../rust-skia/skia-bindings' }\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+	cargo update -p skia-bindings -p ring
+	cargo fetch --locked
 }
 
 build() {
