--- PKGBUILD
+++ PKGBUILD
@@ -55,3 +55,11 @@
   cd ${pkgname#python-}-$pkgver
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+prepare() {
+  cd ${pkgname#python-}-$pkgver
+  patch -Np1 -i ../fix-test-expiration.patch
+}
+
+source+=(fix-test-expiration.patch)
+b2sums+=('6ef80abb505b66e659ef9fbcddc8bc5dfd05982c025cbb12e16759fab57d6b0cd957f13c9da219c6fc700036c44f1d28c477282dbd3b3b1a461475b3220e1dff')
