Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -16,13 +16,13 @@
         config)
 sha256sums=('d0f940a72f648943c1f2211e0e3117387c31d765137d92bd8284a3fb9752a998'
             'SKIP'
-            '2d724738dfb062b8676a8df0488a5e288fc7ceb305633fa6b8d672a722837e81')
+            'd97a1c31e658431de8f1bfdbd13ba6115211ac35d20d174239fb188015d3656f')
 validpgpkeys=('C9E9416F76E610DBD09D040F47B70C55ACC9965B') # Denis Vlasenko <vda.linux@googlemail.com>
 
 prepare() {
   cd "busybox-$pkgver"
 
-  local safeflags="-march=${CARCH/_/-} -mtune=generic -Os -pipe -fno-strict-aliasing"
+  local safeflags="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 
   sed 's|^\(CONFIG_EXTRA_CFLAGS\)=.*|\1="'"$safeflags"'"|' "$srcdir/config" > .config
 }
Index: config
===================================================================
--- config	(revision 391859)
+++ config	(working copy)
@@ -49,7 +49,7 @@
 # CONFIG_FEATURE_SHARED_BUSYBOX is not set
 CONFIG_CROSS_COMPILER_PREFIX=""
 CONFIG_SYSROOT=""
-CONFIG_EXTRA_CFLAGS="-march=x86-64 -mtune=generic -Os -pipe -fno-strict-aliasing"
+CONFIG_EXTRA_CFLAGS="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 CONFIG_EXTRA_LDFLAGS=""
 CONFIG_EXTRA_LDLIBS=""
 # CONFIG_USE_PORTABLE_CODE is not set
