Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -16,13 +16,13 @@
         config)
 sha256sums=('12cec6bd2b16d8a9446dd16130f2b92982f1819f6e1c5f5887b6db03f5660d28'
             'SKIP'
-            '8ec3581a99a0e28e3b08ae64f3187f6b28ceeacef34c6c98ec74bd07b4703c2e')
+            '50b7e0f38dcd922d050a922fe1f1a615dd806e8b2945f1cfde46413dc5f36881')
 validpgpkeys=('C9E9416F76E610DBD09D040F47B70C55ACC9965B') # Denis Vlasenko <vda.linux@googlemail.com>
 
 prepare() {
   cd "busybox-$pkgver"
 
-  local safeflags="-march=${CARCH/_/-} -mtune=generic -Os -pipe -fno-strict-aliasing"
+  local safeflags="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 
   sed 's|^\(CONFIG_EXTRA_CFLAGS\)=.*|\1="'"$safeflags"'"|' "$srcdir/config" > .config
 }
Index: config
===================================================================
--- config	(revision 420565)
+++ config	(working copy)
@@ -49,7 +49,7 @@
 # CONFIG_FEATURE_SHARED_BUSYBOX is not set
 CONFIG_CROSS_COMPILER_PREFIX=""
 CONFIG_SYSROOT=""
-CONFIG_EXTRA_CFLAGS="-march=x86-64 -mtune=generic -Os -pipe -fno-strict-aliasing"
+CONFIG_EXTRA_CFLAGS="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 CONFIG_EXTRA_LDFLAGS=""
 CONFIG_EXTRA_LDLIBS=""
 # CONFIG_USE_PORTABLE_CODE is not set
