diff --git PKGBUILD trunk/PKGBUILD
index 90172db..2bb47ba 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,13 +16,13 @@ source=(https://busybox.net/downloads/busybox-$pkgver.tar.bz2{,.sig}
         config)
 sha256sums=('faeeb244c35a348a334f4a59e44626ee870fb07b6884d68c10ae8bc19f83a694'
             'SKIP'
-            'fe7bd3c7150c25a8ad24190d305a012b84766f34847a7efc7126e19261b0c711')
+            'a5cb92c6336522659fec52be6903d8e40ae522c03a90a8a3b292b7000c93dd69')
 validpgpkeys=('C9E9416F76E610DBD09D040F47B70C55ACC9965B') # Denis Vlasenko <vda.linux@googlemail.com>
 
 prepare() {
   cd "busybox-$pkgver"
   
-  local safeflags="-march=${CARCH/_/-} -mtune=generic -Os -pipe -fno-strict-aliasing"
+  local safeflags="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 
   # use make oldconfig for updating the config file
   sed 's|^\(CONFIG_EXTRA_CFLAGS\)=.*|\1="'"$safeflags"'"|' "$srcdir/config" > .config
diff --git config trunk/config
index 304e15c..6fe9d96 100644
--- config
+++ config
@@ -49,7 +49,7 @@ CONFIG_PIE=y
 # CONFIG_FEATURE_SHARED_BUSYBOX is not set
 CONFIG_CROSS_COMPILER_PREFIX=""
 CONFIG_SYSROOT=""
-CONFIG_EXTRA_CFLAGS="-march=x86-64 -mtune=generic -Os -pipe -fno-strict-aliasing"
+CONFIG_EXTRA_CFLAGS="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 CONFIG_EXTRA_LDFLAGS=""
 CONFIG_EXTRA_LDLIBS=""
 # CONFIG_USE_PORTABLE_CODE is not set
