--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,7 @@ validpgpkeys=('C9E9416F76E610DBD09D040F47B70C55ACC9965B') # Denis Vlasenko <vda.
 prepare() {
   cd "busybox-$pkgver"
   
-  local safeflags="-march=${CARCH/_/-} -mtune=generic -Os -pipe -fno-strict-aliasing"
+  local safeflags="-march=rv64gc -mabi=lp64d -Os -pipe -fno-strict-aliasing"
 
   # use make oldconfig for updating the config file
   sed 's|^\(CONFIG_EXTRA_CFLAGS\)=.*|\1="'"$safeflags"'"|' "$srcdir/config" > .config
