--- PKGBUILD
+++ PKGBUILD
@@ -60,6 +60,7 @@
     ln -s "${srcdir}/opencl-clang" "${srcdir}/llvm-project/llvm/projects/opencl-clang"
     
     patch -d "intel-graphics-compiler-${pkgver}" -Np1 -i "${srcdir}/010-intel-graphics-compiler-disable-werror.patch"
+    patch -d "intel-graphics-compiler-${pkgver}" -Np1 -i "${srcdir}/intel-graphics-compiler-fix-igc_arch_normalize-arch-regex.patch"
 }
 
 build() {
@@ -73,6 +74,20 @@
     export CFLAGS="${CFLAGS/-Wp,-D_FORTIFY_SOURCE=?/}"
     export CXXFLAGS="${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=?/}"
 
+    # fix error: 'cdecl' attribute directive ignored [-Werror=attributes]
+    # note: https://github.com/intel/intel-graphics-compiler/pull/370
+    export CFLAGS+=' -Wno-error=attributes'
+    export CXXFLAGS+=' -Wno-error=attributes'
+
+    # fix error: 'void operator delete(void*, std::size_t)' called on pointer '<unknown>' with nonzero offset 40 [-Werror]
+    # note: false positive in STL
+    export CFLAGS+=' -Wno-error=free-nonheap-object'
+    export CXXFLAGS+=' -Wno-error=free-nonheap-object'
+
+    # fix error: #warning "<ciso646> is deprecated in C++17" [-Werror=cpp]
+    export CFLAGS+=' -Wno-error=cpp'
+    export CXXFLAGS+=' -Wno-error=cpp'
+
     export GIT_COMMITTER_NAME='builduser'
     export GIT_COMMITTER_EMAIL='builduser@archlinux.org'
     
@@ -83,6 +98,7 @@
         -DCMAKE_INSTALL_LIBDIR=lib \
         -DCMAKE_INSTALL_PREFIX=/usr \
         -DCMAKE_SKIP_RPATH=YES \
+        -DIGC_OPTION__ARCHITECTURE_TARGET=LinuxRISCV \
         -DIGC_OPTION__CLANG_MODE=Source \
         -DIGC_OPTION__LINK_KHRONOS_SPIRV_TRANSLATOR=ON \
         -DIGC_OPTION__LLD_MODE=Source \
@@ -106,3 +122,6 @@
     install -D -m644 opencl-clang/LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE-opencl-clang"
     ln -s "libopencl-clang.so.${_llvmver%.*}" "${pkgdir}/usr/lib/libopencl-clang.so"
 }
+
+source+=('intel-graphics-compiler-fix-igc_arch_normalize-arch-regex.patch::https://github.com/intel/intel-graphics-compiler/commit/0228c3cac90e35f094e6409ecdbb2787240f0b1b.diff')
+sha256sums+=('c7031d81fdcc838f6d8c139e65750da244efd0e004b2d583529fd10d8914d9b5')
