diff --git PKGBUILD PKGBUILD
index 3538322e..dce7ccd1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,14 @@ makedepends=('go' 'git')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/nadoo/glider/archive/v$pkgver.tar.gz")
 sha512sums=('7d54a52ffa6f0b7fa05d0188f475e67dd264cd545aca14252d134c2c1843310286a896b634534ae6534d37bbd07f74342390241a8e1cdaefaa086485bc453d03')
 
+prepare() {
+  cd glider-$pkgver
+
+  # Unreleased changes including fixes for riscv64
+  go mod edit -replace github.com/templexxx/cpu=github.com/templexxx/cpu@98168dcec14af3f8700bf309afe503bfe936e9f2
+  go mod tidy
+}
+
 build() {
   cd glider-$pkgver
   export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external"
