From 7954389e39292980323a986ed21eb47325bbddf8 Mon Sep 17 00:00:00 2001
Message-ID: <7954389e39292980323a986ed21eb47325bbddf8.1731734570.git.rsworktech@outlook.com>
From: Levi Zim <rsworktech@outlook.com>
Date: Sat, 16 Nov 2024 13:22:28 +0800
Subject: [PATCH] Enable relocate=1 for ff_h264_weight_funcs_8_rvv

---
 libavcodec/riscv/h264dsp_rvv.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/riscv/h264dsp_rvv.S b/libavcodec/riscv/h264dsp_rvv.S
index 079896b17b..05d077f44e 100644
--- a/libavcodec/riscv/h264dsp_rvv.S
+++ b/libavcodec/riscv/h264dsp_rvv.S
@@ -160,7 +160,7 @@ h264_weight 8, 16
 
         .global ff_h264_weight_funcs_8_rvv
         .hidden ff_h264_weight_funcs_8_rvv
-const ff_h264_weight_funcs_8_rvv
+const ff_h264_weight_funcs_8_rvv, relocate=1
         .irp    w, 16, 8, 4, 2
 #if __riscv_xlen == 32
         .word   ff_h264_weight_pixels\w\()_8_rvv
-- 
2.47.0

