diff --git a/DEPS b/DEPS
index aeeb3bf541..fc215c3480 100644
--- a/DEPS
+++ b/DEPS
@@ -1682,19 +1682,6 @@ deps = {
 
   'src/tools/luci-go': {
       'packages': [
-        {
-          'package': 'infra/tools/luci/cas/${{platform}}',
-          'version': Var('luci_go'),
-        },
-        # TODO(crbug.com/382506663): Remove after investigation/deprecation
-        {
-          'package': 'infra/tools/luci/isolate/${{platform}}',
-          'version': Var('luci_go'),
-        },
-        {
-          'package': 'infra/tools/luci/swarming/${{platform}}',
-          'version': Var('luci_go'),
-        },
       ],
       'condition': 'non_git_source',
       'dep_type': 'cipd',
@@ -2566,20 +2553,12 @@ deps = {
 
   'src/third_party/ninja': {
     'packages': [
-      {
-        'package': Var('ninja_package') + '${{platform}}',
-        'version': Var('ninja_version'),
-      }
     ],
     'condition': 'non_git_source',
     'dep_type': 'cipd',
   },
   'src/third_party/siso/cipd': {
     'packages': [
-      {
-        'package': 'build/siso/${{platform}}',
-        'version': Var('siso_version'),
-      }
     ],
     'condition': 'non_git_source',
     'dep_type': 'cipd',
