diff --git a/DEPS b/DEPS
index 213a9a623b..3b88468a24 100644
--- a/DEPS
+++ b/DEPS
@@ -1091,10 +1091,6 @@ deps = {
   },
   'src/buildtools/reclient': {
     'packages': [
-      {
-        'package': Var('reclient_package') + '${{platform}}',
-        'version': Var('reclient_version'),
-      }
     ],
     'condition': 'download_reclient and non_git_source',
     'dep_type': 'cipd',
@@ -1694,19 +1690,6 @@ deps = {
 
   'src/tools/luci-go': {
       'packages': [
-        {
-          'package': 'infra/tools/luci/cas/${{platform}}',
-          'version': Var('luci_go'),
-        },
-        # TODO(crbug.com/382506663): Remove after investigation/deprecation
-        {
-          'package': 'infra/tools/luci/isolate/${{platform}}',
-          'version': Var('luci_go'),
-        },
-        {
-          'package': 'infra/tools/luci/swarming/${{platform}}',
-          'version': Var('luci_go'),
-        },
       ],
       'condition': 'non_git_source',
       'dep_type': 'cipd',
@@ -2196,10 +2179,6 @@ deps = {
 
   'src/third_party/gperf/cipd': {
       'packages': [
-        {
-          'package': 'infra/3pp/tools/gperf/${{platform}}',
-          'version': 'version:3@3.2',
-        },
       ],
       'condition': 'host_os == "linux" and non_git_source',
       'dep_type': 'cipd',
@@ -2579,20 +2558,12 @@ deps = {
 
   'src/third_party/ninja': {
     'packages': [
-      {
-        'package': Var('ninja_package') + '${{platform}}',
-        'version': Var('ninja_version'),
-      }
     ],
     'condition': 'non_git_source',
     'dep_type': 'cipd',
   },
   'src/third_party/siso/cipd': {
     'packages': [
-      {
-        'package': 'build/siso/${{platform}}',
-        'version': Var('siso_version'),
-      }
     ],
     'condition': 'non_git_source',
     'dep_type': 'cipd',
@@ -3442,10 +3413,6 @@ deps = {
 
   'src/tools/resultdb': {
       'packages': [
-        {
-          'package': 'infra/tools/result_adapter/${{platform}}',
-          'version': Var('result_adapter_revision'),
-        },
       ],
       'condition': 'non_git_source',
       'dep_type': 'cipd',
