Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 447422)
+++ PKGBUILD	(working copy)
@@ -25,6 +25,7 @@
 source=(https://commondatastorage.googleapis.com/chromium-browser-official/$pkgname-$pkgver.tar.xz
         https://github.com/foutrelis/chromium-launcher/archive/v$_launcher_ver/chromium-launcher-$_launcher_ver.tar.gz
         https://github.com/stha09/chromium-patches/releases/download/chromium-${pkgver%%.*}-patchset-$_gcc_patchset/chromium-${pkgver%%.*}-patchset-$_gcc_patchset.tar.xz
+        riscv.patch
         iwyu-add-utility-for-std-exchange.patch
         enable-GlobalMediaControlsCastStartStop.patch
         roll-src-third_party-ffmpeg.patch
@@ -35,6 +36,7 @@
 sha256sums=('1a3797d36901fa3ba63744b9a870b65a8890c9a850442c160196bc64df886b1f'
             '213e50f48b67feb4441078d50b0fd431df34323be15be97c55302d3fdac4483a'
             '53de0f936fd571e578ba2fbf348c8741116cdcceac3ea6fae5008d8f054a7698'
+            '6d195f8140f2c21eb04aafd5cfd2e44c75d601bff6443f83ed6642ca5e575c89'
             '6f666ef0acb08704ca58cc0d5e97e7ce64d8fea51042e593adae1ce15a61231c'
             '779fb13f2494209d3a7f1f23a823e59b9dded601866d3ab095937a1a04e19ac6'
             '30df59a9e2d95dcb720357ec4a83d9be51e59cc5551365da4c0073e68ccdec44'
@@ -120,6 +122,9 @@
   patch -Np1 -i ../patches/chromium-102-fenced_frame_utils-include.patch
   patch -Np1 -i ../patches/chromium-102-regex_pattern-array.patch
 
+  # RISCV64
+  patch -Np1 -i ../riscv.patch
+
   # Link to system tools required by the build
   mkdir -p third_party/node/linux/node-linux-x64/bin
   ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
@@ -173,6 +178,10 @@
     'enable_widevine=true'
     'enable_nacl=false'
     "google_api_key=\"$_google_api_key\""
+    'use_thin_lto=false'
+    'is_clang=true'
+    'use_lld=false'
+    'enable_swiftshader=true'
   )
 
   if [[ -n ${_system_libs[icu]+set} ]]; then
