--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,10 @@ source=(https://commondatastorage.googleapis.com/chromium-browser-official/chrom
         https://github.com/foutrelis/chromium-launcher/archive/v$_launcher_ver/chromium-launcher-$_launcher_ver.tar.gz
         https://github.com/stha09/chromium-patches/releases/download/chromium-${pkgver%%.*}-patchset-$_gcc_patchset/chromium-${pkgver%%.*}-patchset-$_gcc_patchset.tar.xz
         fix-the-way-to-handle-codecs-in-the-system-icu.patch
+        riscv-angle.patch
+        riscv-crashpad.patch
+        riscv-dav1d.patch
+        riscv-sandbox.patch
         v8-move-the-Stack-object-from-ThreadLocalTop.patch
         REVERT-roll-src-third_party-ffmpeg-m102.patch
         REVERT-roll-src-third_party-ffmpeg-m106.patch
@@ -36,6 +40,10 @@ sha256sums=('e348ab2dc4311083e729d714a81e95dd9db108ff71437dde451c97ac939881ce'
             '213e50f48b67feb4441078d50b0fd431df34323be15be97c55302d3fdac4483a'
             '8c7f93037cc236024cc8be815b2c2bd84f6dc9e32685299e31d4c6c42efde8b7'
             'a5d5c532b0b059895bc13aaaa600d21770eab2afa726421b78cb597a78a3c7e3'
+            'f64e280d31e309970c3c5ea8110ff10ed17d5b0a2e42e78ceb0afcb19469c01e'
+            'fd64211e23ad2b474e51d4d30a7a2e7d0ebb7a665f7a20c5c49264e0fedeacf2'
+            'fd30767b8587338d1a38165aaaeef7e44a6ea91c4cfa644effaaa363e698796a'
+            '5881bcfc354c72a6e82fd340578192d1321cb5396e995b14c2e48855b3215d6b'
             '49c3e599366909ddac6a50fa6f9420e01a7c0ffd029a20567a41d741a15ec9f7'
             '30df59a9e2d95dcb720357ec4a83d9be51e59cc5551365da4c0073e68ccdec44'
             '4c12d31d020799d31355faa7d1fe2a5a807f7458e7f0c374adf55edb37032152'
@@ -125,6 +133,12 @@ prepare() {
   patch -Np1 -i ../patches/chromium-110-CredentialUIEntry-const.patch
   patch -Np1 -i ../patches/chromium-110-DarkModeLABColorSpace-pow.patch
 
+  # RISCV64
+  patch -Np1 -i ../riscv-angle.patch
+  patch -Np1 -i ../riscv-crashpad.patch
+  patch -Np1 -i ../riscv-dav1d.patch
+  patch -Np1 -i ../riscv-sandbox.patch
+
   # Link to system tools required by the build
   mkdir -p third_party/node/linux/node-linux-x64/bin
   ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
@@ -180,6 +194,10 @@ build() {
     'enable_widevine=true'
     'enable_nacl=false'
     "google_api_key=\"$_google_api_key\""
+    'use_thin_lto=false'
+    'is_clang=true'
+    'use_lld=false'
+    'enable_swiftshader=true'
   )
 
   if [[ -n ${_system_libs[icu]+set} ]]; then
