--- a/third_party/ffmpeg/libavcodec/riscv/bswapdsp_init.c
+++ b/third_party/ffmpeg/libavcodec/riscv/bswapdsp_init.c
@@ -33,10 +33,6 @@
     int flags = av_get_cpu_flags();
 
     if (flags & AV_CPU_FLAG_RVB_ADDR) {
-#if (__riscv_xlen >= 64)
-        if (flags & AV_CPU_FLAG_RVB_BASIC)
-            c->bswap_buf = ff_bswap32_buf_rvb;
-#endif
 #if HAVE_RVV
         if (flags & AV_CPU_FLAG_RVV_I32)
             c->bswap16_buf = ff_bswap16_buf_rvv;
--- a/third_party/ffmpeg/libavcodec/riscv/ac3dsp_init.c
+++ b/third_party/ffmpeg/libavcodec/riscv/ac3dsp_init.c
@@ -30,9 +30,4 @@
 av_cold void ff_ac3dsp_init_riscv(AC3DSPContext *c)
 {
     int flags = av_get_cpu_flags();
-
-    if (flags & AV_CPU_FLAG_RVB_ADDR) {
-        if (flags & AV_CPU_FLAG_RVB_BASIC)
-            c->extract_exponents = ff_extract_exponents_rvb;
-    }
 }
--- a/third_party/ffmpeg/ffmpeg_generated.gni
+++ b/third_party/ffmpeg/ffmpeg_generated.gni
@@ -669,32 +669,8 @@
     "libavutil/riscv/timer.h",
   ]
   ffmpeg_gas_sources += [
-    "libavcodec/riscv/aacpsdsp_rvv.S",
-    "libavcodec/riscv/ac3dsp_rvb.S",
-    "libavcodec/riscv/alacdsp_rvv.S",
     "libavcodec/riscv/audiodsp_rvf.S",
-    "libavcodec/riscv/audiodsp_rvv.S",
-    "libavcodec/riscv/bswapdsp_rvb.S",
-    "libavcodec/riscv/bswapdsp_rvv.S",
-    "libavcodec/riscv/exrdsp_rvv.S",
-    "libavcodec/riscv/flacdsp_rvv.S",
-    "libavcodec/riscv/fmtconvert_rvv.S",
-    "libavcodec/riscv/g722dsp_rvv.S",
-    "libavcodec/riscv/h264_mc_chroma.S",
-    "libavcodec/riscv/huffyuvdsp_rvv.S",
-    "libavcodec/riscv/idctdsp_rvv.S",
-    "libavcodec/riscv/jpeg2000dsp_rvv.S",
-    "libavcodec/riscv/llauddsp_rvv.S",
-    "libavcodec/riscv/llviddsp_rvv.S",
-    "libavcodec/riscv/opusdsp_rvv.S",
     "libavcodec/riscv/pixblockdsp_rvi.S",
-    "libavcodec/riscv/pixblockdsp_rvv.S",
-    "libavcodec/riscv/sbrdsp_rvv.S",
-    "libavcodec/riscv/utvideodsp_rvv.S",
-    "libavcodec/riscv/vorbisdsp_rvv.S",
     "libavutil/riscv/asm.S",
-    "libavutil/riscv/fixed_dsp_rvv.S",
-    "libavutil/riscv/float_dsp_rvv.S",
-    "libswscale/riscv/rgb2rgb_rvv.S",
   ]
 }
