--- PKGBUILD
+++ PKGBUILD
@@ -116,6 +116,16 @@ prepare() {
   # Fixes for building with libstdc++ instead of libc++
   patch -Np1 -i ../chromium-patches-*/chromium-117-material-color-include.patch
 
+  patch -Np0 -i ../swiftshader-use-llvm16.patch
+  patch -Np1 -i ../fix-rust-target.patch
+  patch -Np1 -i ../Debian-fix-rust-linking.patch
+
+  # riscv64
+  for rvpatch in riscv-{dav1d,ffmpeg,sandbox}.patch; do
+    patch -Np1 -i ../$rvpatch
+  done
+
+
   # Link to system tools required by the build
   mkdir -p third_party/node/linux/node-linux-x64/bin
   ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
@@ -312,4 +322,16 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+makedepends=("${makedepends[@]/nodejs/nodejs-lts-iron}")
+source+=(swiftshader-use-llvm16.patch
+         fix-rust-target.patch
+         riscv-{dav1d,ffmpeg,sandbox}.patch
+         Debian-fix-rust-linking.patch)
+sha256sums+=('2ea949ed1d20a1745ce72f760a7d9297dc0002a747c4bd53e243c4d58ba2c7ca'
+             'fb6a8d99f1eba27a546c391c66f9cacf182076255a2adafb9ebc716130ccbe2b'
+             '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e'
+             '0169a72eba136c8c036c44fd2fbb296b179f717a5ba1f5e9aeff3b589573cf64'
+             '8d52d4da703c8a86059418d1a4ed63d2d6bc1134e9dfe569695a830479a9afae'
+             '98b0fbe1318897954cb8891cafed65e985613c69192e65984ba6785579b29f80')
+
 # vim:set ts=2 sw=2 et:
