--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,8 @@
         chromium-93-ffmpeg-4.4.patch
         chromium-94-ffmpeg-roll.patch
         unexpire-accelerated-video-decode-flag.patch
-        use-oauth2-client-switches-as-default.patch)
+        use-oauth2-client-switches-as-default.patch
+        riscv.patch)
 sha256sums=('36a99d29c2e93a9975be53648f2cd3ffa4ee43730f217a2e7ed88c1901a671e8'
             '213e50f48b67feb4441078d50b0fd431df34323be15be97c55302d3fdac4483a'
             '090af7eab39aade15a1786273f2497d6b4abfaef24279fbf97ce0dd1c38c69aa'
@@ -38,7 +39,8 @@
             '1a9e074f417f8ffd78bcd6874d8e2e74a239905bf662f76a7755fa40dc476b57'
             '56acb6e743d2ab1ed9f3eb01700ade02521769978d03ac43226dec94659b3ace'
             '2a97b26c3d6821b15ef4ef1369905c6fa3e9c8da4877eb9af4361452a425290b'
-            'e393174d7695d0bafed69e868c5fbfecf07aa6969f3b64596d0bae8b067e1711')
+            'e393174d7695d0bafed69e868c5fbfecf07aa6969f3b64596d0bae8b067e1711'
+            '332ed1cfc9c7e8431dea5096b784e1e518c53bba83636ec7257652d559895cef')
 
 # Possible replacements are listed in build/linux/unbundle/replace_gn_files.py
 # Keys are the names in the above script; values are the dependencies in Arch
@@ -117,10 +119,14 @@
   patch -Np1 -i ../patches/chromium-96-DrmRenderNodePathFinder-include.patch
   patch -Np1 -i ../patches/chromium-96-CouponDB-include.patch
 
+  patch -Np1 -i ../riscv.patch
+
   # Link to system tools required by the build
   mkdir -p third_party/node/linux/node-linux-x64/bin
   ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
   ln -s /usr/bin/java third_party/jdk/current/bin/
+  rm buildtools/third_party/eu-strip/bin/eu-strip
+  ln -s /usr/bin/eu-strip buildtools/third_party/eu-strip/bin/eu-strip
 
   # Remove bundled libraries for which we will use the system copies; this
   # *should* do what the remove_bundled_libraries.py script does, with the
@@ -149,6 +155,11 @@
   export AR=ar
   export NM=nm
 
+  export CFLAGS="${CFLAGS//-fexceptions/}"
+  export CFLAGS="${CFLAGS//-fstack-clash-protection/}"
+  export CXXFLAGS="${CXXFLAGS//-fexceptions/}"
+  export CXXFLAGS="${CXXFLAGS//-fstack-clash-protection/}"
+
   local _flags=(
     'custom_toolchain="//build/toolchain/linux/unbundle:default"'
     'host_toolchain="//build/toolchain/linux/unbundle:default"'
@@ -168,6 +179,8 @@
     'enable_widevine=true'
     'enable_nacl=false'
     "google_api_key=\"$_google_api_key\""
+    'use_thin_lto=false'
+    'enable_swiftshader=true'
   )
 
   if [[ -n ${_system_libs[icu]+set} ]]; then
