--- PKGBUILD
+++ PKGBUILD
@@ -39,7 +39,16 @@ sha256sums=('eaf850183d32627ce0cde9f3d3f853bc11c217ef7f41540303214ed47803d96d'
             '713dab4f8c26790c0e4a4c5ce6a9269e90446df5370cc14214a01a363f7afe39'
             'b3de01b7df227478687d7517f61a777450dca765756002c80c4915f271e2d961'
             'd634d2ce1fc63da7ac41f432b1e84c59b7cceabf19d510848a7cff40c8025342'
-            'a9b417b96daec33c9059065e15b3a92ae1bf4b59f89d353659b335d9e0379db6')
+            'a9b417b96daec33c9059065e15b3a92ae1bf4b59f89d353659b335d9e0379db6'
+            '2ea949ed1d20a1745ce72f760a7d9297dc0002a747c4bd53e243c4d58ba2c7ca'
+            '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e'
+            '4865cd0267de649bda9c68ed62a342722c436a47ac14fef58f5f1da827e8faa7'
+            '1713cfc3c73d7f33fd7a9daba9b642869632468bc1068b727827a6b5320a7f88'
+            '9b03cd0430c70be9d90705f3d2ebe2d8a982b57bafb419371c0658d76f24f99e'
+            '3eb5e621757be3f2984acb76d16cf3571bfe5bbbc71ad230b21aa983041ff5ea'
+            'abc9d2f93d9104de6f6d82c89124cfe0bb5afc291a0b10e0fa8ad827210ae62c'
+            '53cbad1796b6ef78806b4babef8d29472895d9b23391f4259d092646173b4ccc'
+            '5898c517b6e78b5f2e67bd282b96107b58c79c1485efaa02ff6404dcaa788f36')
 
 if (( _manual_clone )); then
   source[0]=fetch-chromium-release
@@ -120,6 +129,20 @@ prepare() {
 
   # Fixes for building with libstdc++ instead of libc++
 
+  patch -Np0 -i ../swiftshader-use-llvm16.patch
+  patch -Np0 -i ../Debian-fix-rust-linking.patch
+
+  # riscv64
+  for rvpatch in riscv-{dav1d,sandbox}.patch; do
+    patch -Np1 -i ../$rvpatch
+  done
+  patch -Np0 -i ../compiler-rt-riscv.patch
+  patch -Np1 -i ../0001-chrome-runtime_api_delegate-add-riscv64-define.patch
+  patch -Np1 -i ../0001-extensions-common-api-runtime.json-riscv64-support.patch
+  patch -Np1 -i ../cargo_crate-riscv-pointer-width.patch
+  patch -Np1 -d third_party/ffmpeg < ../riscv-ffmpeg.patch
+
+
   # Link to system tools required by the build
   rm third_party/node/linux/node-linux-x64/bin/node
   ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
@@ -192,6 +215,7 @@ build() {
     'enable_hangout_services_extension=true'
     'enable_widevine=true'
     'enable_nacl=false'
+    'is_clang=true'
     'use_qt6=true'
     'moc_qt6_path="/usr/lib/qt6"'
     "google_api_key=\"$_google_api_key\""
@@ -325,4 +349,12 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=(swiftshader-use-llvm16.patch
+         riscv-{dav1d,ffmpeg,sandbox}.patch
+         compiler-rt-riscv.patch
+         Debian-fix-rust-linking.patch
+         https://github.com/riscv-forks/electron/raw/v32.0.1-riscv/patches/chromium/0001-extensions-common-api-runtime.json-riscv64-support.patch
+         0001-chrome-runtime_api_delegate-add-riscv64-define.patch
+         cargo_crate-riscv-pointer-width.patch)
+
 # vim:set ts=2 sw=2 et:
