--- PKGBUILD
+++ PKGBUILD
@@ -77,11 +77,15 @@ sha256sums=('57594966be592efdb9fe6491f5a834de237f5c7decdca5eb1f7d7a5d38dd54e9'
             'd634d2ce1fc63da7ac41f432b1e84c59b7cceabf19d510848a7cff40c8025342'
             '9c766b82d1143cb3413fe2057361bd2655e46287eacc2c6d6f8504b4c255647a'
             '9343afa1a4308a7cfb3317229f5aff7778688debcc03c4a74a85908aa1d0cc3a'
-            '1c1898f263eaacbc069a8e1a3e732852350350d1dad4cb1a6bba430e3b796cd0')
+            '1c1898f263eaacbc069a8e1a3e732852350350d1dad4cb1a6bba430e3b796cd0'
+            '2ea949ed1d20a1745ce72f760a7d9297dc0002a747c4bd53e243c4d58ba2c7ca'
+            '7b5e90a85835ed867103a9e3bf7b334d35a3ac50dab58b7642978f6ec5fb9821'
+            '3eb5e621757be3f2984acb76d16cf3571bfe5bbbc71ad230b21aa983041ff5ea'
+            'ddd02dbb1dafee30ea78c6694f064da3abce5230c84b27e12132c3a7f16b7a2f')
 
 if (( _manual_clone )); then
   source[0]=fetch-chromium-release
-  sha256sums[0]='2e2f36e3cd1ebc4ad57fd310774a5e5e9db77883d5f9374fedeaabd3c103b819'
+  sha256sums[0]='f0aed11f2dba184a4249094e33a1840be6705bba1cdb4e2a40b31e5d12fe80d3'
   makedepends+=('python-httplib2' 'python-pyparsing' 'python-six' 'npm' 'rsync')
 fi
 
@@ -148,6 +152,11 @@ prepare() {
   # runtime -- this allows signing into Chromium without baked-in values
   patch -Np1 -i ../use-oauth2-client-switches-as-default.patch
 
+  # RISC-V
+  patch -Np0 -i ../swiftshader-use-llvm16.patch
+  patch -Np1 -i ../riscv-sandbox.patch
+  patch -Np0 -i ../Debian-fix-rust-linking.patch
+
   # Upstream fixes
 
   # Fixes from Gentoo
@@ -395,4 +404,9 @@ package() {
   install -Dvm644 LICENSE "$pkgdir/usr/share/licenses/chromium/LICENSE"
 }
 
+source+=(swiftshader-use-llvm16.patch
+         riscv-sandbox.patch
+         Debian-fix-rust-linking.patch
+         DEPS-remove-native-binaries.diff)
+
 # vim:set ts=2 sw=2 et:
diff --git fetch-chromium-release fetch-chromium-release
index 4e15b8f..7902245 100755
--- fetch-chromium-release
+++ fetch-chromium-release
@@ -27,8 +27,48 @@ export PATH+=":$PWD/depot_tools" DEPOT_TOOLS_UPDATE=0
 # system python fails to load six.moves module
 #export VPYTHON_BYPASS='manually managed python not supported by chrome operations'
 
+# Python 3.12 breaks gsutils
+# Bundled wheels are not available for riscv64
+sed -i '/wheel: </,$d' depot_tools/.vpython3
+sed -i '/wheel: </,$d' depot_tools/gsutil.py.vpython3
+sed -i '/wheel: </,$d' depot_tools/recipes/recipe_modules/gitiles/resources/gerrit_client.py.vpython3
+mkdir -p rv-site-packages
+# Manually install required wheels
+vpython3 -m pip install httplib2==0.13.1 six==1.10.0 requests==2.31.0 crcmod==1.7 -t rv-site-packages
+export PYTHONPATH="$PWD/rv-site-packages:$PYTHONPATH"
+
+# Sync often fails on SG2042 with:
+# lockfile.LockError: Error locking /build/chromium/src/chromium-checkout/depot_tools/external_bin/gsutil/gsutil_5.35 (err: [Errno 11] Resource temporarily unavailable)
+retry() {
+  local retries=$1
+  shift
+
+  local count=0
+  until "$@"; do
+    exit=$?
+    wait=$((2 ** $count))
+    count=$(($count + 1))
+    if [ $count -lt $retries ]; then
+      echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
+      sleep $wait
+    else
+      echo "Retry $count/$retries exited $exit, no more retries left."
+      return $exit
+    fi
+  done
+  return 0
+}
+
+
 git clone -b $VERSION --depth=2 https://chromium.googlesource.com/chromium/src
-gclient sync --no-history --nohooks
+patch -Np1 -d src < ../DEPS-remove-native-binaries.diff
+retry 5 gclient sync --no-history --nohooks
+
+
+# Bundled wheels are not available for riscv64 (third_party/depot_tools)
+sed -i '/wheel: </,$d' src/third_party/depot_tools/.vpython3
+sed -i '/wheel: </,$d' src/third_party/depot_tools/gsutil.py.vpython3
+
 src/build/util/lastchange.py -o src/build/util/LASTCHANGE
 src/build/util/lastchange.py -m GPU_LISTS_VERSION \
   --revision-id-only --header src/gpu/config/gpu_lists_version.h
@@ -43,5 +83,6 @@ src/tools/update_pgo_profiles.py --target=linux update \
 src/third_party/node/update_npm_deps
 
 rm -rf src/third_party/node/linux/node-linux-x64/bin/
+mkdir -p src/third_party/jdk/current/bin/
 find src/third_party/jdk/current -type f -delete
 mv src ../chromium-$VERSION
