Index: chromium-114.0.5735.133/third_party/swiftshader/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
===================================================================
--- chromium-114.0.5735.133.orig/third_party/swiftshader/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
+++ chromium-114.0.5735.133/third_party/swiftshader/third_party/llvm-10.0/llvm/include/llvm/Support/Signals.h
@@ -14,6 +14,7 @@
 #ifndef LLVM_SUPPORT_SIGNALS_H
 #define LLVM_SUPPORT_SIGNALS_H
 
+#include <cstdint>
 #include <string>
 
 namespace llvm {
