diff --git PKGBUILD PKGBUILD
index 80ab9e24..b37a60a5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,20 @@
 license=(MIT)
 depends=(gcc-libs)
 source=(http://www.codesink.org/download/$pkgname-$pkgver.tar.gz
-        https://github.com/tat/mimetic/commit/bf84940f.patch)
+        https://github.com/tat/mimetic/commit/bf84940f.patch
+        0001-fix-signed-char.patch::https://github.com/tat/mimetic/commit/2f1bb2b93ecec3167a0aa3410ea4423d5b26320a.patch
+        0002-fix-signed-char.patch::https://github.com/tat/mimetic/commit/d83e0363e316836d1b8d587f4b89e8e09229abe4.patch)
 sha256sums=('3a07d68d125f5e132949b078c7275d5eb0078dd649079bd510dd12b969096700'
-            '877406aff5b11042df9e036ed9e58dcc3e2ce54b47f263fd31886a9c16d7a0ea')
+            '877406aff5b11042df9e036ed9e58dcc3e2ce54b47f263fd31886a9c16d7a0ea'
+            'cbb1af4be4d72a2bf5ecf7d52ff93b99f470fa60b89f478f2faf4b90490e7c79'
+            'a0a983728dc4a8ed26fe64a9b75f2b3397eec5117a52b01d2ec1d6e6ee2bbdbb')
 
 prepare() {
   patch -d $pkgname-$pkgver -p1 < bf84940f.patch # Fix build
+  patch -d $pkgname-$pkgver -p1 < 0001-fix-signed-char.patch
+  patch -d $pkgname-$pkgver -p1 < 0002-fix-signed-char.patch
+  cd $pkgname-$pkgver
+  autoreconf -fiv # https://github.com/tat/mimetic/pull/29
 }
 
 build() {
