diff --git PKGBUILD PKGBUILD
index 1424f0e90..717585f66 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,16 @@ license=('GPL2')
 depends=('gtk2' 'openssl' 'readline')
 makedepends=('intltool')
 #source=(https://www.gftp.org/$pkgname-$pkgver.tar.bz2)
-source=("gftp-$pkgver.tgz::https://github.com/masneyb/gftp/archive/refs/tags/$pkgver.tar.gz")
-md5sums=('5220e212959ae3e3273ee6493e64b312')
+source=("gftp-$pkgver.tgz::https://github.com/masneyb/gftp/archive/refs/tags/$pkgver.tar.gz"
+        "https://github.com/masneyb/gftp/commit/6672d8e98d18ec5c4ebfc7b04bdaf73a278e0eb1.patch")
+md5sums=('5220e212959ae3e3273ee6493e64b312'
+         '14c0767c9c0ddcb64962de4d2eafb304')
 
 prepare() {
    cd "$pkgname-$pkgver"
    # openssl 1.1 compatibility
    #sed -e 's|SSL_library_init|SSL_CTX_new|g' -i configure
+   patch -Np1 -i ../6672d8e98d18ec5c4ebfc7b04bdaf73a278e0eb1.patch
 }
 
 build() {
