--- PKGBUILD
+++ PKGBUILD
@@ -10,14 +10,17 @@ url="https://github.com/prefix-dev/pixi"
 license=('BSD-3-Clause')
 depends=('gcc-libs' 'openssl' 'bzip2')
 makedepends=('cargo')
-source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('c5c83670eff78297efd238e544e7d1046aa79dc27accb94512bdfa0fc128378da293f6023ce476a15f38fc975b9c2140c8b814f07ee557b846e42bd0ab5ee14a')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz"
+        "test-add-linux-riscv64.patch::https://github.com/prefix-dev/pixi/pull/1045.diff")
+sha512sums=('c5c83670eff78297efd238e544e7d1046aa79dc27accb94512bdfa0fc128378da293f6023ce476a15f38fc975b9c2140c8b814f07ee557b846e42bd0ab5ee14a'
+            'f76d2caefaf7e0d7b13de54dc27b620143dd2a3537c5cef7c01abcdbb000b2d5ac8f2125bc469e8f003f03f0bb899f9ad6dc3d10965c79cf71216210c3e1792f')
 options=('!lto')
 
 prepare() {
   cd "$pkgname-$pkgver"
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
   mkdir -p completions/
+  patch -Np1 -i ../test-add-linux-riscv64.patch
 }
 
 build() {
