--- PKGBUILD
+++ PKGBUILD
@@ -15,2 +15,7 @@
 
+prepare() {
+  cd vistir-$pkgver
+  patch -Np1 -i ../pytest9.patch
+}
+
 build() {
@@ -31 +36,5 @@
 }
+
+checkdepends+=(python-pytest-rerunfailures)
+source+=("pytest9.patch::https://github.com/sarugaku/vistir/commit/ca1369c4bb429479539c88884462fbae5103d3f6.patch")
+sha512sums+=('a0a14894bbb796483b09a46d5c9a4da25c0a83fb68ebbec5194080b6bf58b009d12088747d01d4f1e864cc2c03c4153a29c0f15ca4cdefc665cf94936be223a0')
