--- PKGBUILD
+++ PKGBUILD
@@ -34,6 +34,9 @@
 
     # LLVMDebuginfod required curl
     sed -i "s/-lLLVMDebuginfod/-lLLVMDebuginfod -lcurl/" tools/CMakeLists.txt
+
+    patch -Np1 -d runtime/phobos -i "$srcdir/disable-static-NaN-tests.patch"
+    patch -Np1 -d runtime/phobos -i "$srcdir/fix-riscv-floating-point-control.patch"
 }
 
 build() {
@@ -49,7 +52,7 @@
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld --flto=thin" \
+    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld" \
     -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"," \
     ..
     ninja
@@ -108,3 +111,8 @@
     # licenses
     install -D -m644 "$srcdir/ldc/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("disable-static-NaN-tests.patch"
+         "fix-riscv-floating-point-control.patch")
+sha256sums+=('22b9132b58dde320d6da3c22d2eeabbc0c4d6a079348e9e0fbe5172ef4b86aba'
+            '9d8f868594cee9e44aba17def72a734d6eafebe09a6c9ae4292b110449768ce8')
