--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,11 @@ prepare() {
 
     # Set version used for path construction in getRelativeClangCompilerRTLibPath()
     sed -i "s/ldc::llvm_version_base/\"$_clangversion\"/" driver/linker-gcc.cpp
+
+    patch -Np1 -d runtime/phobos -i "$srcdir/disable-static-NaN-tests.patch"
+
+    # runtime: Link libatomic on 64-bit RISC-V and LoongArch
+    git cherry-pick -n 9296fd6fccbc2a6ade38ded1358923d1a893caab
 }
 
 build() {
@@ -50,7 +55,7 @@ build() {
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld --flto=thin" \
+    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld" \
     -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"," \
     ..
     ninja
@@ -102,3 +107,6 @@ package_liblphobos() {
     # licenses
     install -D -m644 "$srcdir/ldc/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("disable-static-NaN-tests.patch")
+sha256sums+=('22b9132b58dde320d6da3c22d2eeabbc0c4d6a079348e9e0fbe5172ef4b86aba')
