--- PKGBUILD
+++ PKGBUILD
@@ -50,7 +50,8 @@ build() {
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=gold --flto=thin" \
+    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false" \
+    -DLD_FLAGS="-Wl,--no-as-needed -latomic -Wl,--as-needed" \
     -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"," \
     ..
     ninja
@@ -58,6 +59,7 @@ build() {
 
 check() {
     cd "$srcdir/ldc/build"
+    patch -Np1 -d ../runtime/phobos -i "$srcdir/disable-static-NaN-tests.patch"
     ninja all-test-runners
 }
 
@@ -102,3 +104,6 @@ package_liblphobos() {
     # licenses
     install -D -m644 "$srcdir/ldc/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=(disable-static-NaN-tests.patch)
+sha256sums+=('22b9132b58dde320d6da3c22d2eeabbc0c4d6a079348e9e0fbe5172ef4b86aba')
