diff --git PKGBUILD PKGBUILD
index 6a6cda4..77872ff 100644
--- PKGBUILD
+++ PKGBUILD
@@ -34,6 +34,8 @@ prepare() {
 
     # LLVMDebuginfod required curl
     sed -i "s/-lLLVMDebuginfod/-lLLVMDebuginfod -lcurl/" tools/CMakeLists.txt
+
+    patch -Np1 -d runtime/phobos -i "$srcdir/disable-static-NaN-tests.patch"
 }
 
 build() {
@@ -49,7 +51,7 @@ build() {
     -DBUILD_SHARED_LIBS=BOTH \
     -DBUILD_LTO_LIBS=ON \
     -DLDC_WITH_LLD=OFF \
-    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld --flto=thin" \
+    -DD_COMPILER_FLAGS="-link-defaultlib-shared=false -linker=lld" \
     -DADDITIONAL_DEFAULT_LDC_SWITCHES="\"-link-defaultlib-shared\"," \
     ..
     ninja
@@ -101,3 +103,6 @@ package_liblphobos() {
     # licenses
     install -D -m644 "$srcdir/ldc/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=("disable-static-NaN-tests.patch")
+sha256sums+=('22b9132b58dde320d6da3c22d2eeabbc0c4d6a079348e9e0fbe5172ef4b86aba')
