diff --git PKGBUILD PKGBUILD
index 75d322d..4c65c21 100644
--- PKGBUILD
+++ PKGBUILD
@@ -29,9 +29,10 @@ build() {
   cd ${pkgname}-${pkgver}
   make
   cd tools
-  make p0f-client
-  make p0f-sendsyn
-  make p0f-sendsyn6
+  CFLAGS=`echo $CFLAGS | sed 's/-Werror=format-security//g'`
+  make CFLAGS="$CFLAGS" p0f-client
+  make CFLAGS="$CFLAGS" p0f-sendsyn
+  make CFLAGS="$CFLAGS" p0f-sendsyn6
 }

 package() {
