--- PKGBUILD
+++ PKGBUILD
@@ -14,14 +14,25 @@ license=('GPL3')
 depends=('fftw' 'gtk3' 'glibmm' 'gtkmm3' 'lcms2' 'lensfun' 'libcanberra'
          'libiptcdata' 'desktop-file-utils' 'hicolor-icon-theme')
 makedepends=('cmake')
-source=("https://github.com/Beep6581/RawTherapee/releases/download/${pkgver}/rawtherapee-$pkgver.tar.xz")
-sha1sums=('2d8b7104c2f8dad1925a3ee3ac52204d424278d8')
-sha256sums=('360528a0aae922eb5af8742f415475fb91b6d62a739da5f2804828f04ec40853')
+source=("https://github.com/Beep6581/RawTherapee/releases/download/${pkgver}/rawtherapee-$pkgver.tar.xz"
+        "fix-fortify.patch::https://github.com/Beep6581/RawTherapee/commit/75f44d3f2beb4680ae6d69ed9fb56fbb48c710d8.diff")
+sha1sums=('2d8b7104c2f8dad1925a3ee3ac52204d424278d8'
+          'd30de292f3e99af7915e818316282280b3b593a0')
+sha256sums=('360528a0aae922eb5af8742f415475fb91b6d62a739da5f2804828f04ec40853'
+            'e9e7c46a70345924feda19f1b9d057fb48fca32c6ca4d337e1a03125b97f6db2')
+
+prepare() {
+  cd "$srcdir/$pkgname-$pkgver"
+  patch -Np1 -i ../fix-fortify.patch
+}
 
 build() {
   mkdir -p "$srcdir/$pkgname-build"
   cd "$srcdir/$pkgname-build"
 
+  CFLAGS="$CFLAGS -pthread"
+  CXXFLAGS="$CXXFLAGS -pthread"
+
   cmake "$srcdir/$pkgname-$pkgver" \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_BUILD_TYPE=Release \
