Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1016194)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,9 @@
 
 prepare() {
   cd zola-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
