diff --git PKGBUILD PKGBUILD
index f9103ed..0606780 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,7 +16,6 @@ depends=(
   gcc-libs
   glib2
   glibc
-  hyperscan
   icu
   jemalloc
   libelf
@@ -161,7 +160,6 @@ build() {
     -DRSPAMD_USER=rspamd \
     -DNO_SHARED=ON \
     -DWANT_SYSTEMD_UNITS=ON \
-    -DENABLE_HYPERSCAN=ON \
     -DENABLE_JEMALLOC=ON \
     -DENABLE_OPTIMIZATION=ON
   cmake --build build
@@ -175,7 +173,6 @@ package() {
   install -Dm 644 rspamd.logrotate "${pkgdir}"/etc/logrotate.d/rspamd
   install -Dm 644 rspamd-dmarc-report.service -t "${pkgdir}"/usr/lib/systemd/system/
   install -Dm 644 rspamd-dmarc-report.timer -t "${pkgdir}"/usr/lib/systemd/system/
-  install -Dm 644 rspamd-hyperscan.hook -t "${pkgdir}"/usr/share/libalpm/hooks/
 }
 
 # vim: ts=2 sw=2 et:
