--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,11 @@ b2sums=('02154afa3129c64d2dc3365a49860748271003c25441fd663a1abfc6e7ef3b324a00576
         '3aabf135c4f18e1ad745ae8800db782b25b15305dfeaaa031b4501408ab7e7d01f66e8ebb5be59fc813cfbff6788d08d2e48dcf24ecc480a40ec9db8dbce9fec'
         'e8f2b19e2474f30a4f984b45787950012668bf0acb5ad1ebb25cd9776925ab4a6aa927f8131ed53e35b1c71b32c504c700fe5b5145ecd25c7a8284373bb951ed')
 
+prepare() {
+  cd $pkgbase
+  patch -Np1 -i ../$pkgname-riscv-support.patch
+}
+
 build() {
   export CFLAGS+=' -w' CXXFLAGS+=' -w'
   export LDFLAGS+=' -L/opt/cuda/targets/x86_64-linux/lib/stubs/'
@@ -62,3 +67,6 @@ package_ollama-docs() {
   cp -r $pkgbase/docs "$pkgdir/usr/share/doc/$pkgbase"
   install -Dm644 $pkgbase/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+source+=($pkgname-riscv-support.patch)  # https://github.com/ollama/ollama/pull/6627
+b2sums+=('336d0ea08b05a3f4da075c80d79becadf72b5573cb7f74361d1a1513970e0907cc7d3d586c4b65f7b7086346bf77da3ff1234fcf9e3c1ff258fca069c33d0e1e')
