--- PKGBUILD
+++ PKGBUILD
@@ -37,7 +37,7 @@ build() {
 
   # Unset these otherwise somehow nvcc will try to use them.
   unset CFLAGS CXXFLAGS
-  make dist CUDA_12_PATH=/opt/cuda
+  make dist
   go build .
 }
 
@@ -50,7 +50,7 @@ check() {
 package_ollama() {
   install -Dm755 ollama/ollama "$pkgdir/usr/bin/ollama"
   mkdir -p "$pkgdir"/usr/lib/ollama/runners
-  cp -r ollama/dist/linux-amd64/lib/ollama/runners/cpu* "$pkgdir"/usr/lib/ollama/runners/
+  cp -r ollama/dist/linux-riscv64/lib/ollama/runners/cpu* "$pkgdir"/usr/lib/ollama/runners/
 
   install -Dm755 $pkgname/$pkgbase "$pkgdir/usr/bin/$pkgbase"
   install -dm755 "$pkgdir/var/lib/ollama"
@@ -67,7 +67,7 @@ package_ollama-rocm() {
   depends+=(ollama hipblas)
 
   mkdir -p "$pkgdir"/usr/lib/ollama/runners
-  cp -r ollama/dist/linux-amd64/lib/ollama/runners/rocm* "$pkgdir"/usr/lib/ollama/runners
+  cp -r ollama/dist/linux-riscv64/lib/ollama/runners/rocm* "$pkgdir"/usr/lib/ollama/runners
 }
 
 package_ollama-cuda() {
@@ -75,7 +75,7 @@ package_ollama-cuda() {
   depends+=(ollama cuda)
 
   mkdir -p "$pkgdir"/usr/lib/ollama/runners
-  cp -r ollama/dist/linux-amd64/lib/ollama/runners/cuda* "$pkgdir"/usr/lib/ollama/runners
+  cp -r ollama/dist/linux-riscv64/lib/ollama/runners/cuda* "$pkgdir"/usr/lib/ollama/runners
 }
 
 package_ollama-docs() {
@@ -85,3 +85,13 @@ package_ollama-docs() {
   cp -r $pkgbase/docs "$pkgdir/usr/share/doc/$pkgbase"
   install -Dm644 $pkgbase/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+pkgname=(${pkgname[@]/ollama-cuda})
+makedepends=(${makedepends[@]/cuda})
+source+=(remove-unsupported-flags.patch)
+b2sums+=('19f2aefbc5f4c4ba3102275f6e6e5376500cadc8df609873c290b29210ddd60926fadef315b2878ba3f97591469b862b9c7dd6e797017f3bc8cbd8bccc577086')
+
+prepare() {
+  cd ollama
+  patch -Np1 -i "$srcdir"/remove-unsupported-flags.patch
+}
