diff --git PKGBUILD PKGBUILD
index 42d060c..61f03b9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -2,7 +2,7 @@
 # Contributor: Matt Harrison <matt@harrison.us.com>
 
 pkgbase=ollama
-pkgname=(ollama ollama-cuda)
+pkgname=(ollama)
 pkgdesc='Create, run and share large language models (LLMs)'
 pkgver=0.1.3
 pkgrel=1
@@ -51,6 +51,8 @@
   sed -i 's,DLLAMA_CUBLAS=on,DLLAMA_CUBLAS=off,g' $pkgbase/llm/llama.cpp/generate_linux.go
 }
 
+makedepends=( ${makedepends[@]/cuda} )
+
 build() {
   export CGO_CFLAGS="$CFLAGS" CGO_CPPFLAGS="$CPPFLAGS" CGO_CXXFLAGS="$CXXFLAGS" CGO_LDFLAGS="$LDFLAGS"
 
@@ -58,11 +60,6 @@
   cd $pkgbase
   go generate ./...
   go build -buildmode=pie -trimpath -mod=readonly -modcacherw -ldflags=-linkmode=external -ldflags=-buildid=''
-
-  # with CUDA support (LLAMA_CUBLAS=on)
-  cd ../${pkgbase}-cuda
-  go generate ./...
-  go build -buildmode=pie -trimpath -mod=readonly -modcacherw -ldflags=-linkmode=external -ldflags=-buildid=''
 }
 
 check() {
