--- PKGBUILD
+++ PKGBUILD
@@ -55,7 +55,7 @@
     -W no-dev
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
-    -D OLLAMA_LLAMA_BACKENDS="cuda_v13;rocm_v7_2;vulkan"
+    -D OLLAMA_LLAMA_BACKENDS="rocm_v7_2;vulkan"
     -D AMDGPU_TARGETS="$(rocm-supported-gfx)"
     # Sync GPU targets from CMakePresets.json
     # For CUDA 12
@@ -80,7 +80,6 @@
 package_ollama() {
   cp -r ollama/staging-install/* "$pkgdir"
   rm -r "$pkgdir/usr/lib/ollama/rocm"*
-  rm -r "$pkgdir/usr/lib/ollama/cuda"*
   rm -r "$pkgdir/usr/lib/ollama/vulkan"*
 
   install -Dm755 $pkgname/ollama "$pkgdir/usr/bin/ollama"
@@ -125,3 +124,6 @@
   cp -r ollama/docs "$pkgdir/usr/share/doc/ollama"
   install -Dm644 ollama/LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+pkgname=(${pkgname[@]/ollama-cuda})
+makedepends=(${makedepends[@]/cuda})
