diff --git PKGBUILD PKGBUILD
index 5c8631ee5..e28fd7fe8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,10 +10,12 @@
 depends=('perl' 'python' 'lua' 'tcl' 'libxml2' 'libusb-compat' 'libtool' 'libuhd' 'boost-libs')
 makedepends=('swig' 'boost')
 source=("https://github.com/Hamlib/Hamlib/releases/download/$pkgver/$pkgname-$pkgver.tar.gz"{,.asc}
-        'fix-lua-5.4.patch::https://github.com/Hamlib/Hamlib/commit/038b87b29913b5264c00eafaf06a3cc13dc16db7.patch')
+        'fix-lua-5.4.patch::https://github.com/Hamlib/Hamlib/commit/038b87b29913b5264c00eafaf06a3cc13dc16db7.patch'
+        'fix-format-string.patch::https://github.com/Hamlib/Hamlib/commit/a915f1e69e1f363a9cb03a9400c9a80aea0aed86.diff')
 sha512sums=('4cf6c94d0238c8a13aed09413b3f4a027c8ded07f8840cdb2b9d38b39b6395a4a88a8105257015345f6de0658ab8c60292d11a9de3e16a493e153637af630a80'
             'SKIP'
-            '364b443c49aa40f5c30015f8a020545f23f05ddd0b8c25c3ff22516068a1cc92259ddd4063c7de37c31dde95012202f688115720c1c4f3f0b9bbfc4f3165653e')
+            '364b443c49aa40f5c30015f8a020545f23f05ddd0b8c25c3ff22516068a1cc92259ddd4063c7de37c31dde95012202f688115720c1c4f3f0b9bbfc4f3165653e'
+            '9010789da20fa4223ea4f5643136ae7fa1319db18a83150489f2023917c1fe44e8f68a3dbd7878fce7a43c6bd2ae3ed890d979eaa11d8985e7c5c5d2762267d8')
 validpgpkeys=('82D64F6B0E67CD41F689BBA6FB2C5130D55A8819') # Nate Bargmann <n0nb@n0nb.us>
 
 prepare() {
@@ -23,6 +25,7 @@
   sed -i 's|#include <usrp_standard.h>|#include <uhd.h>|' kit/usrp_impl.cc
 
   patch -p1 -i ../fix-lua-5.4.patch
+  patch -p1 -i ../fix-format-string.patch
 
   autoreconf -vif
 }
