diff --git PKGBUILD PKGBUILD
index 5936b13..e90abf7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,7 @@ prepare() {
   patch -Np1 -i ../$pkgname-python-3.13-compatibility.patch
   patch -Np1 -i ../$pkgname-fix-test.patch
   git cherry-pick -n 456bc10d94b57e254568e7ea9a8b3cffb856ebff # Python 3.14 fix
+  git cherry-pick -n 3cc22a6648e550daf37be4cdfe8955931f64a522 # https://github.com/pexpect/pexpect/pull/813
 }
 
 build() {
