diff --git PKGBUILD PKGBUILD
index 8acfbfd9..371f5b04 100644
--- PKGBUILD
+++ PKGBUILD
@@ -24,7 +24,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -36,18 +35,15 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
   itstool
-  java-environment=8
   ldc
   libelf
   libwmf
   llvm
   mercurial
-  mono
   nasm
   netcdf-fortran
   openmpi
@@ -61,7 +57,6 @@ checkdepends=(
   rust-bindgen
   sdl2
   vala
-  valgrind
   vulkan-validation-layers
   wxgtk3
 )
