Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 459439)
+++ PKGBUILD	(working copy)
@@ -10,12 +10,12 @@
 arch=('any')
 license=('Apache')
 depends=('python-setuptools' 'ninja')
-checkdepends=('gcc-objc' 'vala' 'rust' 'gcc-fortran' 'mono' 'boost' 'qt5-base' 'git' 'cython'
+checkdepends=('gcc-objc' 'vala' 'rust' 'gcc-fortran' 'boost' 'qt5-base' 'git' 'cython'
               'gtkmm3' 'gtest' 'gmock' 'protobuf' 'wxgtk3' 'python-gobject' 'gobject-introspection'
-              'itstool' 'gtk3' 'java-environment=8' 'gtk-doc' 'llvm' 'clang' 'sdl2' 'graphviz'
-              'doxygen' 'vulkan-validation-layers' 'openssh' 'mercurial' 'gtk-sharp-2' 'qt5-tools'
-              'libwmf' 'valgrind' 'cmake' 'netcdf-fortran' 'openmpi' 'nasm' 'gnustep-base' 'libelf'
-              'python-pytest-xdist' 'ldc' 'rust-bindgen' 'cuda' 'hotdoc')
+              'itstool' 'gtk3' 'gtk-doc' 'llvm' 'clang' 'sdl2' 'graphviz'
+              'doxygen' 'vulkan-validation-layers' 'openssh' 'mercurial' 'qt5-tools'
+              'libwmf' 'cmake' 'netcdf-fortran' 'openmpi' 'nasm' 'gnustep-base' 'libelf'
+              'python-pytest-xdist' 'ldc' 'rust-bindgen' 'hotdoc')
 source=(https://github.com/mesonbuild/meson/releases/download/${pkgver}/meson-${pkgver}.tar.gz{,.asc}
         0001-Skip-broken-tests.patch
         arch-meson)
