--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -31,18 +30,15 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
   itstool
-  java-environment=8
   ldc
   libelf
   libwmf
   llvm
   mercurial
-  mono
   nasm
   netcdf-fortran
   openmpi
@@ -56,7 +52,6 @@ checkdepends=(
   rust-bindgen
   sdl2
   vala
-  valgrind
   vulkan-validation-layers
   wxgtk3
 )
