--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -42,7 +41,6 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-2
   gtk3
   gtkmm3
   hotdoc
@@ -54,7 +52,6 @@ checkdepends=(
   libwmf
   llvm
   mercurial
-  mono
   nasm
   netcdf-fortran
   openmpi
@@ -72,6 +69,7 @@ checkdepends=(
   sdl2
   vala
   valgrind
+  vulkan-headers
   vulkan-validation-layers
   wayland-protocols
   wxgtk3
