--- PKGBUILD
+++ PKGBUILD
@@ -27,39 +27,35 @@
 checkdepends=(
   boost
   boost-libs
   byacc
   clang
   cmake
-  cuda
   cython
-  dmd
   doxygen
   gcc-fortran
   gcc-objc
   git
   glib2-devel
   glibc-locales
   gmock
   gnustep-base
   gobject-introspection
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-3
   gtk3
   gtkmm3
   hotdoc
   itstool
   java-environment=8
   libelf
   libpcap
   libwmf
   llvm
   mercurial
-  mono
   nasm
   netcdf-fortran
   openmpi
   openssh
   protobuf
   pypy3
