diff --git PKGBUILD PKGBUILD
index 94643a7..c7c1a64 100644
--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,6 @@ checkdepends=(
   boost
   clang
   cmake
-  cuda
   cython
   doxygen
   gcc-fortran
@@ -42,7 +41,6 @@ checkdepends=(
   graphviz
   gtest
   gtk-doc
-  gtk-sharp-3
   gtk3
   gtkmm3
   hdf5
@@ -54,7 +52,6 @@ checkdepends=(
   libwmf
   llvm
   mercurial
-  mono
   nasm
   netcdf-fortran
   openmpi
