diff --git PKGBUILD PKGBUILD
index c795f0c..f2152af 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,14 +16,17 @@ depends=(flint
 makedepends=(boost
              nauty)
 source=(https://github.com/Normaliz/Normaliz/releases/download/v$pkgver/Normaliz-$pkgver.tar.gz
-        flint-3.patch)
+        flint-3.patch
+        "pick-nolll.patch::https://github.com/Normaliz/Normaliz/commit/132415b96c57f6dc4d45fb4e8de62bdc8df18dae.patch")
 sha256sums=('365e1d1e2a338dc4df1947a440e606bb66dd261307e617905e8eca64eaafcf6e'
-            'adff63047e42a66aaf0801c3e01ff6bc5721f0e8da2f66391e88940e8cc44b81')
+            'adff63047e42a66aaf0801c3e01ff6bc5721f0e8da2f66391e88940e8cc44b81'
+            '51bc6ee071ea3777ef30db704e80d31d684883ed3c2c05e3c346f256332c22cc')
 
 prepare() {
   cd $pkgname-$pkgver
   patch -p1 -i ../flint-3.patch # Fix build with flint 3
   autoreconf -if
+  patch -Np1 -i ../pick-nolll.patch # https://github.com/Normaliz/Normaliz/issues/407
 }
 
 build() {
