--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,8 @@ b2sums=('ba7606cccd414203a2e492de2938c1b77450ad1b650940dd0e792bf63abbfb37a2630e6
 prepare() {
   cd "$pkgname"
 
+  patch -Np1 -i ../update-depends.patch
+
   # create directory for build output
   mkdir -p build
 
@@ -77,3 +79,7 @@ package() {
   # documentation
   install -vDm644 -t "$pkgdir/usr/share/doc/$pkgname" README.md CHANGELOG.md
 }
+
+source+=('update-depends.patch')
+sha512sums+=('08ef8cb903cec6299217fc5af948fb6f24343a2e5b8da4ae41606bf4b0feaa21126ffbb41c24d5f9c3fe03d44d98e2c86bf44c47c2a4ea23416840af5d6c457b')
+b2sums+=('6a0fbab35899730fd3399c214654abd160b7eadb5b6a9008661fb21b98fbab8fb9535299970b0a2cece44db32b8fb8ab2a1aa24aef730f17afad692170af443c')
