--- PKGBUILD
+++ PKGBUILD
@@ -23,10 +23,17 @@ optdepends=('git: for Git support'
 replaces=('passmenu')
 provides=('passmenu')
 conflicts=('passmenu')
-source=("git+https://git.zx2c4.com/password-store?signed#tag=${pkgver}")
-sha256sums=('SKIP')
+source=("git+https://git.zx2c4.com/password-store?signed#tag=${pkgver}"
+        "fix-outdated-tree.patch")
+sha256sums=('SKIP'
+            'c137e7cd415c29dd455243820913336bf3d0f38e79c1d051d36087a706da340d')
 validpgpkeys=('AB9942E6D4A4CFC3412620A749FC7012A5DE03AE') # Jason A. Donenfeld
 
+prepare() {
+  cd password-store
+  patch -Np1 -i ../fix-outdated-tree.patch
+}
+
 check() {
   cd "${srcdir}/password-store"
   make test
