diff --git PKGBUILD.orig PKGBUILD
index c4f1f0f..d60ebd8 100644
--- PKGBUILD.orig
+++ PKGBUILD
@@ -15,17 +15,21 @@
 source=($pkgname-$pkgver.tar.gz::https://github.com/$pkgname/$pkgname/archive/1.5.0-RC2.tar.gz
         goldendict-qt-5.11-1.patch::"https://github.com/goldendict/goldendict/commit/3d4a468b.patch"
         goldendict-qt-5.11-2.patch::"https://github.com/goldendict/goldendict/commit/a6596780.patch"
-        goldendict-ffmpeg4.patch::"https://github.com/goldendict/goldendict/commit/03bbe01b.patch")
+        goldendict-ffmpeg4.patch::"https://github.com/goldendict/goldendict/commit/03bbe01b.patch"
+        0001-Fix-gcc-7.3-compiler-warnings.patch)
 sha256sums=('bccee0a3d3902f8fa31e439e220a405065fff774e5f8c581be2b0743d3f83fde'
             'dbcd8f40c8a6d35bf00fdb80cd559e22068ea4a57633d3feed711babae8af8d4'
             '628eeec6b50a1f027347f799c2b06e01300412b94452bd72a611b3735e62bb98'
-            'c09649081dff26b7a4eb5d6828a077089eaedf0d8cc392d5bb8232fc64e2537f')
+            'c09649081dff26b7a4eb5d6828a077089eaedf0d8cc392d5bb8232fc64e2537f'
+            'aa9a6242a6bb2ece467614512a2432b349761d4dae6bee280da1f56f1b7f4dd1')
 
 prepare() {
   cd $pkgname-1.5.0-RC2
   patch -p1 -i ../goldendict-qt-5.11-1.patch # Fix build with Qt 5.11
   patch -p1 -i ../goldendict-qt-5.11-2.patch # Follow-up to the above
   patch -p1 -i ../goldendict-ffmpeg4.patch # Fix build with ffmpeg 4
+  # https://github.com/goldendict/goldendict/commit/fd00e9d1.patch
+  patch -p1 -i ../0001-Fix-gcc-7.3-compiler-warnings.patch # Fix gcc 7.3 compiler warnings (#issue 978) 
 }
 
 build(){
