--- PKGBUILD
+++ PKGBUILD
@@ -51,6 +51,7 @@ options=(!lto)
 prepare() {
   cd $pkgname-$pkgver
 
+  patch -Np1 -i ../rv-cpu.diff
 # Revert test that depends on patched gmp
   patch -Rp1 -i ../c12e8515.patch
 # Don't hardcode library names
@@ -103,6 +104,7 @@ _make() {
 
   [[ ${CARCH} == 'aarch64' ]] && make_options+=(MARCH=armv8.2-a JULIA_CPU_TARGET="generic;armv8.2-a,crypto,fullfp16,lse,rdm")
   [[ ${CARCH} == 'x86_64' ]] &&  make_options+=(MARCH=x86-64 JULIA_CPU_TARGET="generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)")
+  [[ ${CARCH} == 'riscv64' ]] &&  make_options+=(MARCH=rv64gc JULIA_CPU_TARGET="generic-rv64,+i,+m,+a,+f,+d,+zicsr,+zifencei,+c" USE_BINARYBUILDER=0)
 
   make "${make_options[@]}" "$@"
 }
@@ -153,3 +155,7 @@ package() {
 
   install -Dm644 contrib/julia.svg -t "$pkgdir"/usr/share/pixmaps
 }
+
+
+source+=(rv-cpu.diff)
+sha256sums+=('7b893f0453a4cd269c25f25cea332f5e7bfbe35ecb6a03863a049603a63e6f92')
