--- a/glib/glib.misc.lisp
+++ b/glib/glib.misc.lisp
@@ -122,14 +122,7 @@
 ;;; gsize
 ;;; ----------------------------------------------------------------------------
 
-(eval-when (:compile-toplevel :load-toplevel :execute)
-  #+x86-64 (defctype g-size :uint64)
-  #+x86    (defctype g-size :ulong)
-  #+ppc32  (defctype g-size :uint32)
-  #+ppc64  (defctype g-size :uint64)
-  #+arm64  (defctype g-size :uint64)
-  #-(or x86-64 x86 ppc32 ppc64 arm64)
-  (error "Can not define '~A', unknown CPU architecture" 'g-size))
+(defctype g-size :size)
 
 (export 'g-size)
 
