--- PKGBUILD
+++ PKGBUILD
@@ -12,6 +12,11 @@ depends=('libx11')
 source=(https://downloads.yoctoproject.org/releases/matchbox/optional-dependencies/Xsettings-client-$pkgver.tar.gz)
 sha512sums=('051eaafa3410f82afea8f5909e5ef46548737f5b7bb3cb9c31262426ebc8e39ce4e84cfa76629cfb6b608fc764b2bbc3e7345a0e8e4257c7f6ea640c3b419634')
 
+prepare() {
+  cd "$srcdir"/Xsettings-client-$pkgver
+  autoreconf -fiv
+}
+
 build() {
   cd "$srcdir"/Xsettings-client-$pkgver
   ./configure --prefix=/usr
