diff --git PKGBUILD PKGBUILD
index 7e0f236..76dbafa 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@
 license=('GPL' 'LGPL')
 depends=('cinnamon-desktop' 'libsm' 'libcanberra' 'xapps')
 optdepends=('cinnamon-translations: i18n')
-makedepends=('meson' 'samurai' 'xtrans' 'python-packaging')
+makedepends=('meson' 'samurai' 'xtrans' 'python-packaging' 'glib2-devel')
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/${pkgver}.tar.gz")
 sha256sums=('fd12d27652658f95644cf7a93adcbd6a985f3a6bc47c68660eed563b8ab34693')
 b2sums=('a856724a1a189aae5b9b6cd2bb4997bf8c4b9e39e084ce90c17463eece9e8615444e5e103d597e029cd0c0df15546474fd2ca787ddd7aadc6aa51856d67c1412')
