--- depot_tools/gclient.py.orig	2024-02-11 04:09:56.113914394 +0100
+++ depot_tools/gclient.py	2024-02-11 04:14:47.727897175 +0100
@@ -1025,6 +1025,9 @@
             hooks_cwd = self.root.root_dir
 
         for dep in deps_to_add:
+            if '${arch}' in dep.name or '${platform}' in dep.name:
+                print("WARN: ignoring platform-specific dep:", dep.name)
+                continue
             if dep.verify_validity():
                 self.add_dependency(dep)
         self._mark_as_parsed([
