--- PKGBUILD
+++ PKGBUILD
@@ -68,6 +68,9 @@
   # fix FTBFS due to redefined -D_FORTIFY_SOURCE
   # arch uses v3, upstream still uses v2
   patch -p1 -i "$srcdir/ftbfs-redefined-fortify_source.patch"
+
+  # fix build with Go 1.27
+  patch -p1 -i "$srcdir/go-1.27-x-net.patch"
 }
 
 build() {
@@ -112,3 +115,7 @@
   install -vDm644 "$srcdir/tmpfiles.conf" \
     "$pkgdir/usr/lib/tmpfiles.d/$pkgname.conf"
 }
+
+source+=("go-1.27-x-net.patch::https://github.com/apptainer/apptainer/commit/19cfb98cbdd32c98d29f1ac83e6013c4e1fd40ae.patch")
+sha512sums+=('3efce273743d6190c3debda322dafa620d3061533f31601e9e6afc56bbf2717c2930d65eb04026a93bcba88037788e33bdab86aa515d74afb2c0b4ae52c3e2f3')
+b2sums+=('bfb1a451e8275286170f90baf7e3e356e5ce702ddc4b73bad8af7353a02c2bf0ce89346f1138d72a45d44a7708f801ae47e78f2f4e92f7e2f1c565db182f5f0c')
