diff --git PKGBUILD PKGBUILD
index 620a620..db1994a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,6 +12,7 @@ arch=('x86_64')
 url='http://www.libpng.org/pub/png/libpng.html'
 license=('libpng-2.0')
 depends=('zlib' 'sh')
+options=(!lto)
 makedepends=('git')
 provides=('libpng16.so')
 source=("git+https://github.com/pnggroup/libpng.git?signed#tag=v${pkgver}")
@@ -28,7 +29,13 @@ prepare() {
 
 build() {
   cd $pkgname
-
+  # Enable RVV runtime dispatch using HWCAPS
+  # https://github.com/pnggroup/libpng/blob/libpng16/contrib/riscv-rvv/linux.c#L30
+  # Some objects needs to be built with rv64gv but our rv64gc always overrides it,
+  # so removing ours as a compromise.
+  # LTO is disabled due to a gcc bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110812
+  CFLAGS="${CFLAGS/-march=rv64gc/}"
+  CXXFLAGS="${CXXFLAGS/-march=rv64gc/}"
   ./configure \
     --prefix=/usr \
     --disable-static
