--- PKGBUILD
+++ PKGBUILD
@@ -28,7 +28,12 @@ prepare() {
 
 build() {
   cd $pkgname
-
+  # Enable RVV runtime dispatch using HWCAPS
+  # https://github.com/pnggroup/libpng/blob/libpng16/contrib/riscv-rvv/linux.c#L30
+  # Some objects needs to be built with rv64gv but our rv64gc always overrides it,
+  # so removing ours as a compromise.
+  CFLAGS="${CFLAGS/-march=rv64gc/}"
+  CXXFLAGS="${CXXFLAGS/-march=rv64gc/}"
   ./configure \
     --prefix=/usr \
     --disable-static
