diff --git PKGBUILD PKGBUILD
index b446af8..3340dbf 100644
--- PKGBUILD
+++ PKGBUILD
@@ -99,7 +99,7 @@ package_elfutils() {
   make DESTDIR="$pkgdir" install -C $pkgbase
 
   # set the default DEBUGINFOD_URLS environment variable to the distribution's debuginfod URL
-  echo "https://debuginfod.archlinux.org" > "$pkgdir/etc/debuginfod/archlinux.urls"
+  echo "https://archriscv.felixc.at/debuginfod" > "$pkgdir/etc/debuginfod/archlinux.urls"
 
   (
     cd "$pkgdir"
