diff --git a/source/Makefile.deps.mk b/source/Makefile.deps.mk
index ae4f0051f..76a817625 100644
--- a/source/Makefile.deps.mk
+++ b/source/Makefile.deps.mk
@@ -81,6 +81,10 @@ ifneq (,$(filter aarch64%,$(TARGET_PROCESSOR)))
 CPU_AARCH64 = true
 CPU_ARM_OR_AARCH64 = true
 endif
+ifneq (,$(filter riscv64%,$(TARGET_PROCESSOR)))
+CPU_RISCV64 = true
+CPU_RISCV32_OR_RISCV64 = true
+endif
 
 # ---------------------------------------------------------------------------------------------------------------------
 # Set PKG_CONFIG (can be overridden by environment variable)
diff --git a/source/modules/ysfx/Makefile b/source/modules/ysfx/Makefile
index b8b910c51..d921e1ffd 100644
--- a/source/modules/ysfx/Makefile
+++ b/source/modules/ysfx/Makefile
@@ -34,6 +34,11 @@ YSFX_FLAGS += -DEEL_TARGET_PORTABLE
 endif
 endif
 
+# RISC-V needs portable code
+ifeq ($(CPU_RISCV64),true)
+YSFX_FLAGS += -DEEL_TARGET_PORTABLE
+endif
+
 ifeq ($(YSFX_FTS_LACKS_LFS_SUPPORT),true)
 YSFX_FLAGS += -DYSFX_FTS_LACKS_LFS_SUPPORT
 endif
