diff --git PKGBUILD PKGBUILD
index 917ed902a..6aa00e788 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,11 +25,19 @@ provides=(dssi-host ladspa-host lv2-host vst-host vst3-host)
 options=(debug)
 source=(
   git+https://github.com/falkTX/$pkgname#tag=$_commit?signed
+  carla-riscv64-portable.patch
 )
-sha512sums=('SKIP')
-b2sums=('SKIP')
+sha512sums=('SKIP'
+            '63f3aae704810f3b558a9280393948f249b8728aa9821e650e686c91b62aa3c79b1eb6d8a6eacf24c3f588f79d861106572716aecbf9b943da7538bb1d43293b')
+b2sums=('SKIP'
+        '382a63d0cf795f7984121e29ba9bb7276daed7d3ef843c831552c836e4abeb59b05f2ab904706002d0071c5061391c3ddd231c8eb7398adfc1c9240fb1eaaa01')
 validpgpkeys=('62B11043D2F6EB6672D93103CDBAA37ABC74FBA0')  # falkTX <falktx@falktx.com>
 
+prepare() {
+  cd $pkgname
+  patch -Np1 -i $srcdir/carla-riscv64-portable.patch
+}
+
 build() {
   make features -C $pkgname
   make -C $pkgname
