--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ arch=('x86_64')
 url='https://www.gegl.org/'
 license=('GPL3' 'LGPL3')
 depends=('babl' 'jasper' 'json-glib' 'lensfun' 'libgexiv2' 'libraw' 'librsvg' 'libspiro'
-         'libtiff' 'luajit' 'openexr' 'poppler-glib' 'suitesparse')
+         'libtiff' 'openexr' 'poppler-glib' 'suitesparse')
 makedepends=('exiv2' 'ffmpeg' 'git' 'glu' 'gobject-introspection' 'intltool' 'mesa'
              'meson' 'python-gobject' 'ruby' 'vala')
 options=('debug')
@@ -34,7 +34,8 @@ build() {
   arch-meson ../"${pkgname}" \
     -Dworkshop=true \
     -Dmrg=disabled \
-    -Dmaxflow=disabled
+    -Dmaxflow=disabled \
+    -Dlua=disabled
   ninja
 }
