Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 437827)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 arch=('x86_64')
 url="https://www.gegl.org/"
 license=('GPL3' 'LGPL3')
-depends=('babl' 'libspiro' 'json-glib' 'libgexiv2' 'libraw' 'openexr' 'librsvg' 'libtiff' 'luajit' 'jasper' 'lensfun' 'suitesparse' 'poppler-glib')
+depends=('babl' 'libspiro' 'json-glib' 'libgexiv2' 'libraw' 'openexr' 'librsvg' 'libtiff' 'jasper' 'lensfun' 'suitesparse' 'poppler-glib')
 makedepends=('intltool' 'ruby' 'mesa' 'glu' 'ffmpeg4.4' 'exiv2' 'vala' 'gobject-introspection' 'git' 'meson' 'python-gobject')
 optdepends=('ffmpeg4.4: ffmpeg plugin' 'graphviz: for gegl-introspect')
 source=("git+https://gitlab.gnome.org/GNOME/gegl.git#tag=$_tag")
@@ -26,7 +26,8 @@
   arch-meson ../$pkgname \
     -Dworkshop=true \
     -Dmrg=disabled \
-    -Dmaxflow=disabled
+    -Dmaxflow=disabled \
+    -Dlua=disabled
   ninja
 }
 
