--- PKGBUILD
+++ PKGBUILD
@@ -32,11 +32,13 @@ source=(
   "git+https://gitlab.gnome.org/World/pika-backup.git#tag=v$pkgver"
   pika-backup-outoftree.patch
   pika-backup-window-icon.patch
+  longer-timeout.patch
 )
 b2sums=(
   ac5e583ef98818d5ab265b64b3660c4b24582c8c3a6a00e41c0979403509f2b6a0cc6465fda487d4c3637f5e56746e635e9ab1af97c384e980e1a54dbb57ec5d
   ed2f520d0aa463826176e7e7edeb0e94f6f03a62d84f6e845c378baf422f6e5d5577eb8ec5f0ef7d8633d2a8e2344e70ef5a0a7345ac298277dd8aa3add3608b
   ae856a2cfe627b9bc7b4b8a0643a134543d4bb52265a8f48fb67a4b41de8a3765d5b73734916cf46ab19ff81c87f6109aa2ea52bfcfdc2cc99e1e78028653372
+  2b3d2f46da744ebd745649b00e582c5fc23d3390519319ee1a68bb58ed2a182ff0ab06f13ab378f7c59a8620b2540b3062313ab9edf3236cce3693c7f1ea1230
 )
 
 prepare() {
@@ -50,6 +52,8 @@ prepare() {
 
   # https://gitlab.gnome.org/World/pika-backup/-/merge_requests/311
   git apply -3 ../pika-backup-window-icon.patch
+
+  patch -Np0 -i ../longer-timeout.patch
 }
 
 build() {
