diff --git a/PKGBUILD b/PKGBUILD
index aada5f4..0038f5a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -25,7 +25,7 @@ build() {
   #   2. https://github.com/magefile/mage/blob/fe9f9420/.goreleaser.yml#L10
   export CGO_ENABLED=0
   local BUILD_DATE="$(date --rfc-3339=date --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}")"
-  go build \
+  CGO_ENABLED=1 go build \
     -trimpath \
     -buildmode=pie \
     -mod=readonly \
