diff --git a/PKGBUILD b/PKGBUILD
index aada5f4..fd10a0f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -23,7 +23,7 @@ build() {
   # References:
   #   1. https://github.com/magefile/mage/blob/fe9f9420/.goreleaser.yml#L27
   #   2. https://github.com/magefile/mage/blob/fe9f9420/.goreleaser.yml#L10
-  export CGO_ENABLED=0
+  export CGO_ENABLED=1
   local BUILD_DATE="$(date --rfc-3339=date --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}")"
   go build \
     -trimpath \
