--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,12 @@ depends=(coin-or-cbc)
 source=(https://www.coin-or.org/download/source/CoinMP/CoinMP-$pkgver.tgz)
 sha256sums=('3459fb0ccbdd39342744684338984ac4cc153fb0434f4cae8cf74bd67490a38d')
 
+prepare() {
+  cd CoinMP-$pkgver
+  autoconf_file="/usr/share/autoconf/build-aux/config"
+  echo ". CoinMP" | xargs -r -n1 cp ${autoconf_file}.{guess,sub}
+}
+
 build() {
   cd CoinMP-$pkgver
   COIN_SKIP_PROJECTS="Sample" \
