--- PKGBUILD
+++ PKGBUILD
@@ -19,6 +19,12 @@
 source=(git+https://github.com/coin-or/CoinMP#tag=releases/$pkgver)
 sha256sums=('ca675f20c3008618ba8d6827191a0d64f9f08995b613adfd8aa2a40f47d3d2e2')
 
+prepare() {
+  cd CoinMP
+  autoconf_file="/usr/share/autoconf/build-aux/config"
+  echo ". CoinMP" | xargs -r -n1 cp ${autoconf_file}.{guess,sub}
+}
+
 build() {
   cd CoinMP
   export CXXFLAGS+=" -Wp,-U_GLIBCXX_ASSERTIONS"
