diff --git PKGBUILD PKGBUILD
index f153be5..6191609 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,12 @@ sha256sums=('7295491b4be5eeac5e7a3fb2067e236e2955ffdc6bbd45f546466edee321644b'
             'SKIP')
 validpgpkeys=('44BC9FD0D688EB007C4DD029E9CBDFC0ABC0A854') # Thomas Schmitt <scdbackup@gmx.net>
 
+prepare() {
+  cd $pkgname-$pkgver
+  autoupdate
+  autoreconf -fiv
+}
+
 build() {
   cd "${pkgname}"-${pkgver/.pl01/}
   ./configure --prefix=/usr --disable-static
