--- PKGBUILD
+++ PKGBUILD
@@ -60,7 +60,7 @@ makedepends=(
   yelp-tools
 )
 source=("git+https://gitlab.gnome.org/GNOME/geary.git#tag=${pkgver/[a-z]/.&}")
-b2sums=('ffec538e394146773e094d3cdabacaa6d54911b759c284e6433abeeb3921e5b8f587532286e146ef087db26876a3b59bc1f6e88d614b53ccadeda10827acad3b')
+b2sums=('7e424cc2f5964390881aefdc6becf527a1e7e05f4edaac373149a73a20b22605d907bb465a0da7456431b2bd6dbb4e1dfe85dccbb5fc8a50e2cee2ea3f335e32')
 validpgpkeys=(
   3A2EF7F138557A145F2866E99FAB18747A8FC649 # Michael James Gratton <mike@vee.net>
   71BFA565F77720034F44E0B00CF8493F76A7AE36 # Cédric Bellegarde <cedric.bellegarde@adishatz.org>
