--- PKGBUILD
+++ PKGBUILD
@@ -11,10 +11,12 @@ arch=('x86_64')
 license=('GPL3')
 url='https://www.gnu.org/software/coreutils/'
 depends=('glibc' 'acl' 'attr' 'gmp' 'libcap' 'openssl')
-source=("https://ftp.gnu.org/gnu/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig})
+source=("https://ftp.gnu.org/gnu/$pkgname/$pkgname-$pkgver.tar.xz"{,.sig}
+        "fix-test-getlogin.patch")
 validpgpkeys=('6C37DC12121A5006BC1DB804DF6FD971306037D9') # Pádraig Brady
 sha256sums=('61a1f410d78ba7e7f37a5a4f50e6d1320aca33375484a3255eddf17a38580423'
-            'SKIP')
+            'SKIP'
+            '5df5e81f5c859b7baca415a6435ea48ae5f09fa111ecd59b0d74c7fe696c3e92')
 
 prepare() {
   cd $pkgname-$pkgver
