--- PKGBUILD
+++ PKGBUILD
@@ -8,17 +8,20 @@ pkgdesc="Command-line based RSS enclosure downloader primarily intended for down
 arch=('x86_64')
 url="https://johndal.com/castget/"
 license=('GPL-2.0-or-later' 'LGPL-2.1-or-later')
-depends=('glib2' 'libxml2' 'curl' 'id3lib')
+depends=('glib2' 'libxml2' 'curl' 'taglib')
+makedepends=('ruby-ronn-ng')
 changelog=$pkgname.changelog
 source=(
   "https://download.savannah.gnu.org/releases/$pkgname/$pkgname-$pkgver.tar.bz2"
   "https://download.savannah.gnu.org/releases/$pkgname/$pkgname-$pkgver.tar.bz2.sig"
   "fix-build.patch"
+  "port-to-taglib.patch"
 )
 sha256sums=(
   '438b5f7ec7e31a45ed3756630fe447f42015acda53ec09202f48628726b5e875'
   'SKIP'
   '48cce3303601ee28e06bdf6b55b7c5023405cacd7c913f8bc6616b3277395b80'
+  '51b690cf0f2ec8526e1ebc14fb01ae537ce8acf8c176b9487c008b3d83b32e8c'  
 )
 validpgpkeys=('561BCBC3DF8A53503A62416EA7DE43E1D881FF60') # Marius Larsen Jøhndal <mariuslj@ifi.uio.no>
 
@@ -33,6 +36,10 @@ prepare() {
 build() {
   cd "${srcdir}"/$pkgname-$pkgver
 
+  patch -Np1 -i ../port-to-taglib.patch
+
+  autoreconf -fi
+
   ./configure --prefix=/usr
   make
 }
