diff --git PKGBUILD PKGBUILD
index 2c1bb45..3217a1f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,8 @@
   rm -f src/sv_parser/{addr,header,sieve}-lex.{c,h}
   patch -p1 <"$srcdir"/glibc-regex.patch
   patch -p1 <"$srcdir"/crash-fix.patch
+  patch -p1 <"$srcdir/$pkgname-unsigned-char-lexers.patch"
+  patch -p1 <"$srcdir/$pkgname-header-body-yystype.patch"
 }
 
 build() {
@@ -64,3 +66,8 @@
   cd "$srcdir/$pkgname"
   make DESTDIR="$pkgdir/" install
 }
+
+source_riscv64=("$pkgname-unsigned-char-lexers.patch::https://github.com/sodabrew/libsieve/pull/9.patch?full_index=1"
+                "$pkgname-header-body-yystype.patch")
+sha512sums_riscv64=('cb8b1945cbdf16436ce97800c48c557f3546faa0670a07bd598b1158e8b75add0437d93cd8ffd7a4254fbe2c6fff83b4793ccec2953cc79f16668cd090520ccc'
+                    '99bf0a9ccc062665ba71057541827ed6cd45ce91c0c46159a1f9e9e10d7fabe8755699e12377984e19526b35998f5eb94bc763d67e6b2dd6da0b5305f859e662')
