--- PKGBUILD
+++ PKGBUILD
@@ -45,6 +45,10 @@ prepare() {
   git submodule set-branch --branch 87-based src/3rdparty
   git -c protocol.file.allow=always submodule update
 
+  patch -Np1 -i "$srcdir"/$pkgname-v8.patch
+  patch -Np1 -i "$srcdir"/$pkgname-riscv.patch
+  patch -Np0 -d src/3rdparty/chromium/v8 < "$srcdir"/riscv-v8-security-backport.patch
+
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-ffmpeg5.patch # Fix build with ffmpeg 5
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-pipewire-0.3.patch # Port to pipewire 0.3
   patch -p2 -d src/3rdparty/chromium -i "$srcdir"/qt5-webengine-icu-75.patch # Fix build with ICU 75
@@ -91,3 +95,11 @@ package() {
   # Fix cmake dependency versions
   sed -e "s|$pkgver\ |$_basever |" -i "$pkgdir"/usr/lib/cmake/*/*Config.cmake
 }
+
+source+=($pkgname-v8.patch
+         $pkgname-riscv.patch
+         riscv-v8-security-backport.patch)
+sha256sums+=('f0394a75373d319fbe5826862a59c5b266a901887a8ef32e7d05fb8efa23992c'
+             '1b04f07fe50a6312f829354c8f1efa9af6914d25d00898bd69d5196cb844b7ba'
+             'd70a78f884171ae8131ffb4fcc074f3f56ce78627ac788e5b21b8963870a5294')
+options+=(!lto)
