--- PKGBUILD
+++ PKGBUILD
@@ -20,13 +20,17 @@
         git+https://chromium.googlesource.com/catapult#commit=5eedfe23148a234211ba477f76fc2ea2e8529189
         qt5-webengine-python3.patch
         qt5-webengine-chromium-python3.patch
-        qt5-webengine-ffmpeg5.patch)
+        qt5-webengine-ffmpeg5.patch
+        v8.patch
+        riscv.patch)
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
             '398c996cb5b606695ac93645143df39e23fa67e768b09e0da6dbd37342a43f32'
             'fda4ff16790799fb285847918a677f4f3f7c0f513d4751f846ffc5aa5d873932'
-            'c50d3019626183e753c53a997dc8a55938847543aa3178d4c51f377be741c693')
+            'c50d3019626183e753c53a997dc8a55938847543aa3178d4c51f377be741c693'
+            '0d4f8f0ac4675469241f3fdcf95a252f53279151da978e46cb5ee4455e6e8b23'
+            '1b04f07fe50a6312f829354c8f1efa9af6914d25d00898bd69d5196cb844b7ba')
 options=(debug)
 
 prepare() {
@@ -41,7 +45,10 @@
   patch -p1 -i "$srcdir"/qt5-webengine-python3.patch # Fix build with Python 3
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-chromium-python3.patch
 
-  patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-ffmpeg5.patch # Fix build with ffmpeg 5
+  # patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-ffmpeg5.patch # Fix build with ffmpeg 5
+
+  patch -Np1 -i "$srcdir"/v8.patch
+  patch -Np1 -i "$srcdir"/riscv.patch
 
 # Update catapult for python3 compatibility
   rm -r src/3rdparty/chromium/third_party/catapult
