diff --git PKGBUILD PKGBUILD
index 67454ec1..2160a853 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,9 @@ source=(git+https://code.qt.io/qt/qtwebengine.git#tag=v${pkgver}-lts
         qt5-webengine-chromium-python3.patch
         qt5-webengine-ffmpeg5.patch
         qt5-webengine-pipewire-0.3.patch
-        qt5-webengine-gcc12.patch)
+        qt5-webengine-gcc12.patch
+        v8.patch
+        riscv.patch)
 sha256sums=('SKIP'
             'SKIP'
             'SKIP'
@@ -30,7 +32,9 @@ sha256sums=('SKIP'
             'fda4ff16790799fb285847918a677f4f3f7c0f513d4751f846ffc5aa5d873932'
             'c50d3019626183e753c53a997dc8a55938847543aa3178d4c51f377be741c693'
             '5e3a3c4711d964d5152a04059a2b5c1d14bb13dd29bce370120f60e85b476b6f'
-            'cf9be3ffcc3b3cd9450b1ff13535ff7d76284f73173412d097a6ab487463a379')
+            'cf9be3ffcc3b3cd9450b1ff13535ff7d76284f73173412d097a6ab487463a379'
+            'f0394a75373d319fbe5826862a59c5b266a901887a8ef32e7d05fb8efa23992c'
+            '1b04f07fe50a6312f829354c8f1efa9af6914d25d00898bd69d5196cb844b7ba')
 
 prepare() {
   mkdir -p build
@@ -49,6 +53,9 @@ prepare() {
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-pipewire-0.3.patch # Port to pipewire 0.3
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-gcc12.patch # Fix build with GCC 12
 
+  patch -Np1 -i "$srcdir"/v8.patch
+  patch -Np1 -i "$srcdir"/riscv.patch
+
 # Update catapult for python3 compatibility
   rm -r src/3rdparty/chromium/third_party/catapult
   mv "$srcdir"/catapult src/3rdparty/chromium/third_party
