--- PKGBUILD
+++ PKGBUILD
@@ -43,6 +43,11 @@ prepare() {
   git submodule set-branch --branch 87-based src/3rdparty
   git -c protocol.file.allow=always submodule update
 
+  patch -Np1 -i "$srcdir"/$pkgname-v8.patch
+  patch -Np0 -i "$srcdir"/$pkgname-riscv.patch
+  patch -Np0 -d src/3rdparty/chromium/v8 -i "$srcdir"/riscv-v8-security-backport.patch
+  patch -Np1 -d src/3rdparty/chromium -i "$srcdir"/riscv-dav1d.patch
+
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-ffmpeg5.patch # Fix build with ffmpeg 5
   patch -p1 -d src/3rdparty -i "$srcdir"/qt5-webengine-pipewire-0.3.patch # Port to pipewire 0.3
   patch -p2 -d src/3rdparty/chromium -i "$srcdir"/qt5-webengine-icu-75.patch # Fix build with ICU 75
@@ -90,3 +95,13 @@ package() {
   # Fix cmake dependency versions
   sed -e "s|$pkgver\ |$_basever |" -i "$pkgdir"/usr/lib/cmake/*/*Config.cmake
 }
+
+source+=($pkgname-v8.patch
+         $pkgname-riscv.patch
+         riscv-v8-security-backport.patch
+         riscv-dav1d.patch)
+sha256sums+=('f0394a75373d319fbe5826862a59c5b266a901887a8ef32e7d05fb8efa23992c'
+             'b0879b207c6a239d49f6db3fc17994b06b21a1cae20ece4c6d8a4649f19e8271'
+             'd70a78f884171ae8131ffb4fcc074f3f56ce78627ac788e5b21b8963870a5294'
+             '5689e9422624c8725509b6fdc277e20c3e8862cf515656faef7507978489bc4e')
+options+=(!lto)
