Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1356836)
+++ PKGBUILD	(working copy)
@@ -16,6 +16,11 @@
 source=("https://hackage.haskell.org/packages/archive/${_hkgname}/${pkgver}/${_hkgname}-${pkgver}.tar.gz")
 sha512sums=('4831aacbe44f0541c2c638a501513a6c2c6814ec71291ed00ca2a0728a190626e551c54d682d1cf1d931833cfaaa48bb4a29f5138ab6a3963863b36af64a8964')
 
+prepare() {
+    cd $_hkgname-$pkgver
+    sed -i '/Test-Suite doctest/a \    buildable: False' $_hkgname.cabal
+}
+
 build() {
     cd $_hkgname-$pkgver
 
