--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,12 @@ makedepends=(nauty)
 source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz)
 sha256sums=('1fb591dd88336fdaf096b8e42e46111e41671a5eb85d4ee36e45baff1678bd33')
 
+prepare() {
+  cd Cbc-releases-$pkgver
+  autoconf_file="/usr/share/autoconf/build-aux/config"
+  echo ". Cbc" | xargs -r -n1 cp ${autoconf_file}.{guess,sub}
+}
+
 build() {
   cd Cbc-releases-$pkgver
   ./configure --prefix=/usr \
