--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,12 @@ makedepends=(git)
 source=(git+https://github.com/coin-or/Cbc#tag=releases/$pkgver)
 sha256sums=('36ce629474dd1c7bfff0e12c1a405d8046f6f794cdb403a94e9b3aeb09b8ccfa')
 
+prepare() {
+  cd Cbc
+  autoconf_file="/usr/share/autoconf/build-aux/config"
+  echo ". Cbc" | xargs -r -n1 cp ${autoconf_file}.{guess,sub}
+}
+
 build() {
   cd Cbc
   CFLAGS+=" -Wno-implicit-function-declaration" \
