diff --git PKGBUILD PKGBUILD
index b86cea7a..6adc2d86 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,19 @@ license=(EPL)
 groups=(coin-or)
 depends=(coin-or-cgl)
 makedepends=(nauty)
-source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz)
-sha256sums=('96d02593b01fd1460d421f002734384e4eb1e93ebe1fb3570dc2b7600f20a27e')
+source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.guess
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.sub)
+sha256sums=('96d02593b01fd1460d421f002734384e4eb1e93ebe1fb3570dc2b7600f20a27e'
+            'af8a1922c9b3c240bf2119d4ec0965a0b5ec36b1016017ba66db44b3b53e9cea'
+            'd611751fba98e807c9684d253bb02aa73d6825fe0e0b9ae3cbf258a59171c9b0')

 build() {
   cd Cbc-releases-$pkgver
+  cp -f ${srcdir}/config.guess ./config.guess
+  cp -f ${srcdir}/config.guess ./Cbc/config.guess
+  cp -f ${srcdir}/config.sub ./config.sub
+  cp -f ${srcdir}/config.sub ./Cbc/config.sub
   ./configure --prefix=/usr \
               --enable-cbc-parallel \
               --with-nauty-lib=/usr/lib/libnauty.a --with-nauty-incdir=/usr/include/nauty
