diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index f0615f7..2c74a22 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,19 @@ license=(EPL)
 groups=(coin-or)
 depends=(coin-or-cgl)
 makedepends=(nauty)
-source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz)
-sha256sums=('f394efecccc40a51bf79fba2c2af0bc92561f3e6b8b6e4c6e36d5e70986f734f')
+source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.guess
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.sub)
+sha256sums=('f394efecccc40a51bf79fba2c2af0bc92561f3e6b8b6e4c6e36d5e70986f734f'
+            'af8a1922c9b3c240bf2119d4ec0965a0b5ec36b1016017ba66db44b3b53e9cea'
+            'd611751fba98e807c9684d253bb02aa73d6825fe0e0b9ae3cbf258a59171c9b0')
 
 build() {
   cd Cbc-releases-$pkgver
+  cp -f ${srcdir}/config.guess ./config.guess
+  cp -f ${srcdir}/config.guess ./Cbc/config.guess
+  cp -f ${srcdir}/config.sub ./config.sub
+  cp -f ${srcdir}/config.sub ./Cbc/config.sub
   ./configure --prefix=/usr \
               --enable-cbc-parallel \
               --with-nauty-lib=/usr/lib/libnauty.a --with-nauty-incdir=/usr/include/nauty
