--- PKGBUILD
+++ PKGBUILD
@@ -18,11 +18,19 @@ depends=(coin-or-asl
          glibc
          gcc-libs)
 makedepends=(nauty)
-source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz)
-sha256sums=('1fb591dd88336fdaf096b8e42e46111e41671a5eb85d4ee36e45baff1678bd33')
+source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Cbc/archive/refs/tags/releases/$pkgver.tar.gz
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.guess
+        https://github.com/coin-or/Cbc/raw/f30b1f4ad9326dc76cc8af715bb3cff109925447/config.sub)
+sha256sums=('1fb591dd88336fdaf096b8e42e46111e41671a5eb85d4ee36e45baff1678bd33'
+            'af8a1922c9b3c240bf2119d4ec0965a0b5ec36b1016017ba66db44b3b53e9cea'
+            'd611751fba98e807c9684d253bb02aa73d6825fe0e0b9ae3cbf258a59171c9b0')
 
 build() {
   cd Cbc-releases-$pkgver
+  cp -f ${srcdir}/config.guess ./config.guess
+  cp -f ${srcdir}/config.guess ./Cbc/config.guess
+  cp -f ${srcdir}/config.sub ./config.sub
+  cp -f ${srcdir}/config.sub ./Cbc/config.sub
   ./configure --prefix=/usr \
               --enable-cbc-parallel \
               --with-nauty-lib=/usr/lib/libnauty.a --with-nauty-incdir=/usr/include/nauty
