diff --color -Naur xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.cxx xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.cxx
--- xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.cxx	2014-07-16 16:03:14.000000000 +0800
+++ xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.cxx	2021-10-02 20:49:04.599107721 +0800
@@ -25,7 +25,7 @@
 //
 //
 void*
-operator new (size_t n, cutl::share s) throw (std::bad_alloc)
+operator new (size_t n, cutl::share s)
 {
   if (s == shared)
   {
diff --color -Naur xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.hxx xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.hxx
--- xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.hxx	2014-07-16 16:03:14.000000000 +0800
+++ xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.hxx	2021-10-02 20:49:04.599107721 +0800
@@ -31,7 +31,7 @@
 extern LIBCUTL_EXPORT cutl::share exclusive;
 
 LIBCUTL_EXPORT void*
-operator new (std::size_t, cutl::share) throw (std::bad_alloc);
+operator new (std::size_t, cutl::share);
 
 LIBCUTL_EXPORT void
 operator delete (void*, cutl::share) throw ();
@@ -61,7 +61,7 @@
     _ref_count () const;
 
     void*
-    operator new (std::size_t, share) throw (std::bad_alloc);
+    operator new (std::size_t, share);
 
     void
     operator delete (void*, share) throw ();
diff --color -Naur xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.ixx xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.ixx
--- xsd-4.0.0+dep-orig/libcutl/cutl/shared-ptr/base.ixx	2014-07-16 16:03:14.000000000 +0800
+++ xsd-4.0.0+dep/libcutl/cutl/shared-ptr/base.ixx	2021-10-02 20:49:04.599107721 +0800
@@ -59,7 +59,7 @@
   }
 
   inline void* shared_base::
-  operator new (std::size_t n, share) throw (std::bad_alloc)
+  operator new (std::size_t n, share)
   {
     return ::operator new (n);
   }
