--- PKGBUILD
+++ PKGBUILD
@@ -14,15 +14,18 @@
 license=(GPL-2.0-or-later
          LGPL-3.0-or-later)
 source=(https://gmplib.org/download/gmp/gmp-$pkgver.tar.lz{,.sig}
-        gmp-gcc-15.patch)
+        gmp-gcc-15.patch
+        gmp-riscv64-sec-tabselect-public-loop.patch)
 sha256sums=('be5c908a7a836c3a9bd9d62aa58563c5e9e7fef94c43a7f42dbc35bb6d02733c'
             'SKIP'
-            'fd0f1b48dbe72413efbd9c45f1ca7013d555f46c578446ce3ba202e3a9b96c6c')
+            'fd0f1b48dbe72413efbd9c45f1ca7013d555f46c578446ce3ba202e3a9b96c6c'
+            '7296df25d7fee3b1f3a361aba824c5fd7317e2883172342781cb31b28db24298')
 validpgpkeys=('343C2FF0FBEE5EC2EDBEF399F3599FF828C67298')   # Niels Möller
 
 prepare() {
   cd $pkgname-$pkgver
   patch -p1 < ../gmp-gcc-15.patch # Fix build with GCC 15
+  patch -p1 < ../gmp-riscv64-sec-tabselect-public-loop.patch # Keep sec_tabselect branches public on RISC-V
   autoreconf -vif
 }
 
