diff --git PKGBUILD PKGBUILD
index 03859b1..697419c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,6 +8,7 @@ arch=('x86_64')
 url="http://www.olofson.net/kobodl/"
 license=('GPL' 'LGPL')
 depends=('sdl_image' 'hicolor-icon-theme')
+makedepends=('systemd')
 install=kobodeluxe.install
 source=(http://www.olofson.net/kobodl/download/KoboDeluxe-${pkgver}.tar.gz
         kobodeluxe-gcc-4.3.patch kobodeluxe-0.5.1-glibc2.10.patch)
