From 018ebd2f2c2ef4d25034851c9dc37543795a1686 Mon Sep 17 00:00:00 2001
From: Letu Ren <fantasquex@gmail.com>
Date: Sat, 24 Sep 2022 12:56:41 +0200
Subject: [PATCH] increase test timeout limit

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 3aa1a10a2..a0d12d658 100644
--- a/meson.build
+++ b/meson.build
@@ -2371,8 +2371,8 @@ if want_systemtap and enable_dtrace
   enable_systemtap = true
 endif
 
-test_timeout = 60
-test_timeout_slow = 180
+test_timeout = 600
+test_timeout_slow = 1800
 
 pkg = import('pkgconfig')
 windows = import('windows')
-- 
2.37.3

