--- a/gio/tests/memory-monitor-psi-env-set.c
+++ b/gio/tests/memory-monitor-psi-env-set.c
@@ -285,9 +285,7 @@
   g_setenv ("MEMORY_PRESSURE_WATCH", "/dev/null", TRUE);
   g_setenv ("MEMORY_PRESSURE_WRITE", env_write, TRUE);
 
-  monitor = g_object_new (G_TYPE_MEMORY_MONITOR_PSI,
-                          "proc-path", setup->mock_proc_path,
-                          NULL);
+  monitor = g_object_new (G_TYPE_MEMORY_MONITOR_PSI, NULL);
   warning_id = g_signal_connect (monitor, "low-memory-warning",
                                  G_CALLBACK (memory_monitor_psi_signal_cb), &warning_level);
   g_initable_init (G_INITABLE (monitor), NULL, &local_error);
