Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422646)
+++ PKGBUILD	(working copy)
@@ -38,12 +38,6 @@
 source=(git+https://code.videolan.org/videolan/x264.git#commit=${_commit})
 sha256sums=(SKIP)
 
-pkgver() {
-  cd x264
-
-  ./version.sh | grep X264_POINTVER | sed -r 's/^#define X264_POINTVER "([0-9]+\.[0-9]+)\.([0-9]+) (.*)"$/\1.r\2.\3/'
-}
-
 build() {
   cd x264
 
