--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,9 @@ b2sums=('e7fb0b8abc7f0c6e0fd82b8b4b035174eb8ec56f9578d3d5b5f2477b5a1eac23cc68984

 prepare() {
   cd ${pkgbase}-${pkgver}
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }

 build() {
