--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,8 @@ b2sums=('f4361c5c8f7d31d10cf22c67723847b1597c6ca307c67aa76e9b1620e9f3bb0a18b9f03
 
 prepare() {
   cd ${pkgname}-${pkgver}
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
@@ -58,4 +60,3 @@ package() {
   install -Dm 644 -t "${pkgdir}/usr/share/licenses/${pkgname}" LICENSE-*
 }
 
-# vim: ts=2 sw=2 et:
