--- PKGBUILD
+++ PKGBUILD
@@ -24,6 +24,8 @@ prepare() {
   cd "$pkgname-$_pkgver"
 
   patch -Np1 -i "$srcdir"/0ad-fix-missing-includes.patch
+  patch -Np1 -i "$srcdir"/0ad-riscv64.patch
+  patch -Np1 -i "$srcdir"/nvtt-riscv64.patch
 }
 
 build() {
@@ -33,6 +35,8 @@ build() {
   ./build-source-libs.sh
   cd ../build/workspaces
 
+  export LDFLAGS="$LDFLAGS -Wl,--no-as-needed -latomic -Wl,--as-needed"
+
   ./update-workspaces.sh -j$(nproc) \
       --without-pch \
       --with-system-premake5 \
@@ -57,3 +61,8 @@ package() {
   install -Dm644 build/resources/${pkgname}.desktop "${pkgdir}/usr/share/applications/${pkgname}.desktop"
   install -Dm644 build/resources/${pkgname}.png "${pkgdir}/usr/share/pixmaps/${pkgname}.png"
 }
+
+source+=(0ad-riscv64.patch
+         nvtt-riscv64.patch)
+sha512sums+=('98fae7f06109b9d2cc58b7253a1a2e35c4c5b541d8d9dcae367d75ad57103f5f226a2350373c64d2c8d3c2b04e6208398caa6f2be979b9f235304722d2c475df'
+             'fdacbdeac2828db27c6053fe4ebade58e9823ea012017d197c05d70a8b0cecea3790c4cbd5bb2a7fab4ee26b97afdb295c1220f7f31b1c8eba7e80d6d3898f80')
