--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,13 @@
 validpgpkeys=('A035C8C19219BA821ECEA86B64E628F8D684696D')  # Pablo Galindo Salgado <pablogsal@gmail.com>
 sha512sums=('6b5d7893d27d5fa71ce4cad1b424a7d4b36e8bc438c590cbd86d9872148d5051cd3215f8f8f4b817e48d4fe386d8902ecc0dcd612ca779306ef35dfa2898a878')
 
+prepare() {
+  cd "$pkgname-$pkgver"
+
+  patch -Np1 -i "$srcdir"/0ad-riscv64.patch
+  patch -Np1 -i "$srcdir"/nvtt-riscv64.patch
+}
+
 build() {
   cd "$pkgname-$pkgver"
 
@@ -55,3 +62,8 @@
   install -Dm644 build/resources/${pkgname}.appdata.xml "${pkgdir}/usr/share/metainfo/${pkgname}.appdata.xml"
   install -Dm644 build/resources/${pkgname}.png "${pkgdir}/usr/share/pixmaps/${pkgname}.png"
 }
+
+source+=(0ad-riscv64.patch
+         nvtt-riscv64.patch)
+sha512sums+=('22b035909d112860612908e00d8a43911962ff3021c2a90aa014f0b0d8818d85d24d4977c63ff6d0561adaae805e9f6724c7ca6a2ac79a8ecd2f2adae1779ec9'
+             '9804123f21e2c100ba90e88282e821b3604c3326c9b4a6a5266abfb9bbe93798cb90b3bca37c760c0e22b435c99788ffddb0c24ca03551718d7fc817bd276f68')
