--- PKGBUILD
+++ PKGBUILD
@@ -59,6 +59,7 @@ prepare() {
 # https://github.com/valkey-io/valkey/issues/1882
 #  patch -Np1 < ../valkey-5.0-use-system-jemalloc.patch
   patch -Np1 < ../remove-deprecated-use-of-je_calloc.patch
+  patch -Np1 < ../zmalloc-fix-build-on-less-common-platforms.patch
 }
 
 build() {
@@ -77,3 +78,6 @@ package() {
   install -Dm644 "$srcdir"/valkey.sysusers "$pkgdir"/usr/lib/sysusers.d/valkey.conf
   install -Dm644 "$srcdir"/valkey.tmpfiles "$pkgdir"/usr/lib/tmpfiles.d/valkey.conf
 }
+
+source+=(zmalloc-fix-build-on-less-common-platforms.patch::https://github.com/valkey-io/valkey/commit/204097dac46730e9eb125c822c1dc2959c6572c5.diff)
+sha512sums+=('793095df109eccda326a5d5356194e8a3735061fadd6e6bfc5e6dfa6b277c1d71d03e29ee9a54dfd12d23ae26817e0797b958377d10f9b7ac8f1d88581621883')
