--- PKGBUILD
+++ PKGBUILD
@@ -121,6 +121,8 @@ validpgpkeys=(
 
 prepare() {
   cd webkitgtk-$pkgver
+  patch -Np1 -i ../fix-ftbfs-riscv64.patch
+  patch -Np1 -i ../fix-typo-denormaldisabler.patch
 }
 
 build() {
@@ -194,3 +196,10 @@ package_webkit2gtk-docs() {
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(fix-ftbfs-riscv64.patch
+         fix-typo-denormaldisabler.patch)
+sha256sums+=('b42d36c275e5fbd6da8e0761381d9eac13b2d103a366a61d9d450421948385a3'
+             '08a5fc35ed0d65f1c47a00f2bcf0e6c335321ee5db7f2eb1ad4cbfb1a563a36c')
+b2sums+=('cd80efecf12982e9e9335032a433e094614c8b95edce7114d37c6d651ee1f5bb3e331c3911d7901055cd59ca3ed5de38d14e15fab5488d3248bf5899623b2f8c'
+         '9320fb1e591d1254c8ccafa2d3b14aa2aa802eb90aaf2f564933003f4051dca834c35e059b6740a81d98aa19c1654e6dcbc77433933c1b9ca40e4cf5fec4adc1')
