--- PKGBUILD
+++ PKGBUILD
@@ -78,9 +78,11 @@ makedepends=(
 )
 source=(
   $url/releases/webkitgtk-$pkgver.tar.xz{,.asc}
+  fix-riscv64-build.patch # bpo: https://github.com/WebKit/WebKit/commit/c07cdb6ae80b0847da58a6acb1022b8b3e170073
 )
 sha256sums=('a4607ea1bf89669e89b1cb2c63faaec513f93de09b6ae60cc71d6a8aab7ab393'
-            'SKIP')
+            'SKIP'
+            '565e21e1e474ae1769ce2e3e8939193aa58bfe12a3c30f85c8bf2eb140007362')
 validpgpkeys=(
   'D7FCF61CF9A2DEAB31D81BD3F3D322D0EC4582C3'  # Carlos Garcia Campos <cgarcia@igalia.com>
   '5AA3BC334FD7E3369E7C77B291C559DBE4C9123B'  # Adrián Pérez de Castro <aperez@igalia.com>
@@ -88,6 +90,7 @@ validpgpkeys=(
 
 prepare() {
   cd webkitgtk-$pkgver
+  patch -Np1 -i ../fix-riscv64-build.patch
 }
 
 build() {
