diff --git PKGBUILD PKGBUILD
index 28a6250..9e4b93f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,17 +30,20 @@ source=(
   $pkgname-1.9.2-makefile-enable-harden-local-files.patch
   $pkgname-1.9.2-fix-make-rule-deps.patch
   $pkgname-1.9.2-fix-ftbfs.patch
+  $pkgname-1.9.2-riscv64_build.patch
 )
 sha512sums=('64cdfd05b219feb66f437c2d1d156bc4942950d4c94f1bfa343aa0cfb59b711a8c1c0e55533eef94ac2a78743699e7251281ff9e09d092251cf7b33203208665'
             '9e609eb4fb2a7116166626d15470d66e2eb66a25867618d4065d48636304f88549a71c5e827ac92750183f0fabaa3b84beea3dffa905031a2867939bfae955e7'
             '87c837ad59e89d575d1b5d403b2663acec6799dc469dc7e992e5b0760c74064082569fc8b6be471d877cfd6b62324d6b6482ef4174dbae5eced50f34a3870532'
             '78a57f2ceae92d50bf78c745accae209895c42bc5a670cee32aec108fd05c024c3f6e5282cb1756b06fcc477b24dbd87871ec48a523795cff097b7e732dd32d6'
-            'b6ff9b4fea609a6a26de86b4054fa01f5922c9c5f8ac5841fe77b89c95aab9b720b599219d091fe0add1df3e14c1ceaed8ad9f6bb8444abc77acb4e9f9578411')
+            'b6ff9b4fea609a6a26de86b4054fa01f5922c9c5f8ac5841fe77b89c95aab9b720b599219d091fe0add1df3e14c1ceaed8ad9f6bb8444abc77acb4e9f9578411'
+            'f8cee091497aca1979a6f7275a8b0f7ab8305482ee72f2c3e8f9da1f7da18ff523520dc9a09a99639fcbcfb7c00bda7f24537ce4be42a60ed8e2ca688727839c')
 b2sums=('b57752bb3c127204abf255cee2dc2b3f4ba0a6a76a2e5453eaec9bf381687bdbc67375a6e2f612bbba239f8946e32602557f3ac849f6e0afcb33529643cea404'
         '2972c141b9ddfcda5e240fae5920f752da4f4a425c70fd5edd1c1ff387320013709ea56b924ce9a8222e0d2f1f1ee4e5aaf83ae38a93c38cea633432c24e17f7'
         'c0d3cca4a05bdb006511ffa1c3868d2154926e2555e32f9dd953e8ac0b76c50f2bffbaefa0ee4f8e985447d72ee64652e38a72ca65b1403dc888ef76a484290c'
         '5e984abd6748ebd4d769f0107c1b161cbad78034b9e3073135b8a0c3ef9a517a1c11129b22864674a653a15ea5c0cbe0b17544220b1cfc9dd0c2900aecbeea03'
-        '34c45aa9869ddf03fc0953af607b965becde3d17f267cf6f1fdf7b8e994c4c5539f2ab76ee0ec25913197738b3eea36e61a792fe77de6553f52f7e7a73fa6ae4')
+        '34c45aa9869ddf03fc0953af607b965becde3d17f267cf6f1fdf7b8e994c4c5539f2ab76ee0ec25913197738b3eea36e61a792fe77de6553f52f7e7a73fa6ae4'
+        '3254d186811b75043e85119fddfb7d6a631099466bd41efcfc3e2525ef81768cc9e155732a5eaaff0be655069eb60a19b4cded12789d92802645b6f4a7f85cce')
 validpgpkeys=('D5606E73C8B46271BEAD9ADF814AE47C214854D6') # James Bottomley <jejb@kernel.org>
 
 prepare() {
@@ -51,6 +54,7 @@ prepare() {
   patch -Np1 -i ../$pkgname-1.9.2-fix-make-rule-deps.patch
   # fix various issues
   patch -Np1 -i ../$pkgname-1.9.2-fix-ftbfs.patch
+  patch -Np1 -i ../$pkgname-1.9.2-riscv64_build.patch
 }
 
 build() {
