--- PKGBUILD
+++ PKGBUILD
@@ -58,6 +58,9 @@
   export CFLAGS="${CFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
   export CXXFLAGS="${CXXFLAGS/_FORTIFY_SOURCE=3/_FORTIFY_SOURCE=2}"
 
+  # Rust uses riscv*gc triples, but clang expects canonical riscv* triples.
+  export BINDGEN_EXTRA_CLANG_ARGS="$BINDGEN_EXTRA_CLANG_ARGS --target=riscv64-unknown-linux-gnu -march=rv64gc"
+
   make \
     LIBEXECDIR=/usr/lib \
     DESTDIR="${pkgdir}" \
