--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,7 @@ sha512sums=('8d74eca342a61f7a020cc85253e5afc70a81c46c4936b927fd3158c65299a3b1f67
 
 prepare() {
   cd "$pkgname-$pkgver"
+  patch -Np1 -i ../riscv64-support.patch
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
@@ -35,4 +36,7 @@ package() {
   install -Dm 644 LICENSE-MIT -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=(riscv64-support.patch::https://github.com/JakWai01/lurk/pull/44.diff)
+sha512sums+=('SKIP')
+
 # vim: ts=2 sw=2 et:
