diff --git a/cmake/platform.cmake b/cmake/platform.cmake
index 84b69bf5f9..2dcb422f0a 100644
--- a/cmake/platform.cmake
+++ b/cmake/platform.cmake
@@ -174,12 +174,12 @@ if (DNNL_TARGET_ARCH STREQUAL "RV64")
         set(RV64_MARCH_FLAG "-march=rv64gc")
     endif()
 
-    set(DNNL_RISCV_USE_RVV_INTRINSICS ${CAN_COMPILE_RVV_INTRINSICS})
+    set(DNNL_RISCV_USE_RVV_INTRINSICS ${CAN_COMPILE_RVV_INTRINSICS} CACHE BOOLEAN "Enable RISC-V RVV")
     if (${DNNL_RISCV_USE_RVV_INTRINSICS})
         add_definitions(-DDNNL_RISCV_USE_RVV_INTRINSICS)
     endif()
 
-    set(DNNL_RISCV_USE_ZVFH_INTRINSICS ${CAN_COMPILE_ZVFH_INTRINSICS})
+    set(DNNL_RISCV_USE_ZVFH_INTRINSICS ${CAN_COMPILE_ZVFH_INTRINSICS} CACHE BOOLEAN "Enable RISC-V Zvfh")
     if (${DNNL_RISCV_USE_ZVFH_INTRINSICS})
         add_definitions(-DDNNL_RISCV_USE_ZVFH_INTRINSICS)
     endif()
