diff --git PKGBUILD PKGBUILD
index 4b15ebeff..f569d3103 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,15 @@ url=https://github.com/oneapi-src/oneDNN
 license=('APACHE')
 depends=('gcc-libs')
 makedepends=('cmake' 'ninja')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/oneapi-src/oneDNN/archive/v${pkgver}.tar.gz")
-sha256sums=('d7a47caeb28d2c67dc8fa0d0f338b11fbf25b473a608f04cfed913aea88815a9')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/oneapi-src/oneDNN/archive/v${pkgver}.tar.gz"
+        "disable_openmp_fatal.patch")
+sha256sums=('d7a47caeb28d2c67dc8fa0d0f338b11fbf25b473a608f04cfed913aea88815a9'
+            'f09e79d9e011af5f7ea5b7d75fbcb648c0d88913b5e1fe533bc728d3eaa0109f')
+
+prepare() {
+  cd "oneDNN-$pkgver"
+  patch -Np1 -i "../disable_openmp_fatal.patch"
+}
 
 build() {
   cd "oneDNN-$pkgver"
