--- PKGBUILD
+++ PKGBUILD
@@ -19,12 +19,14 @@ b2sums=('904438577139c2706c71f1601ea8ad0503e9048f948746e6179880ef267e88c705bc205
 
 prepare() {
   cd "$_archive"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
   cd "$_archive"
-  maturin build --locked --release --all-features --target "$CARCH-unknown-linux-gnu" --strip
+  maturin build --locked --release --all-features --strip
 }
 
 check() {
@@ -40,7 +42,7 @@ _package_common() {
 package_ruff() {
   cd "$_archive"
   _package_common
-  local _target="target/$CARCH-unknown-linux-gnu/release/ruff"
+  local _target="target/release/ruff"
   install -Dm0755 -t "$pkgdir/usr/bin/" "$_target"
   $_target --generate-shell-completion bash | install -Dm0644 /dev/stdin "$pkgdir/usr/share/bash-completion/completions/$pkgname.bash"
   $_target --generate-shell-completion fish | install -Dm0644 /dev/stdin "$pkgdir/usr/share/fish/vendor_completions.d/$pkgname.fish"
