diff --git PKGBUILD PKGBUILD
index f06107a7..308f7505 100644
--- PKGBUILD
+++ PKGBUILD
@@ -18,7 +18,9 @@ b2sums=('5656c83ea0351609a37c0f14faa66aa8240a42d06bd0129d7bbdb637247c71bcdc72d12
 
 prepare() {
   cd "$_archive"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
