diff --git PKGBUILD PKGBUILD
index 1b81af1..1014604 100644
--- PKGBUILD
+++ PKGBUILD
@@ -24,10 +24,12 @@ _tag=890f24b7e88f488eee464ed14c01fbceb276cf2a
 source=(
   git+https://github.com/OpenZWave/open-zwave.git#tag=${_tag}
   openzwave-system-libs.patch
+  fix_nullptr_mistakes.patch
 )
 b2sums=(
   SKIP
   56095ee16bb04184b2bf037d7601bc5431c654e1328cd1ad63f1fcb7d6918bcd65ef4ab30ee2c80f3bf3c50e838f5818d23e9d01569e41125e9e68028d021c59
+  d9a3179a7e9db4bde6acca44cb8523b85799668a8a22a37bc625c021af920bafd2cf29de803dd48fe685e67792c2a0cdc2b99c0f18852d1f037dec9f5b297ff2
 )
 
 pkgver() {
@@ -40,6 +42,7 @@ prepare() {
   cd open-zwave
 
   patch -Np1 -i ../openzwave-system-libs.patch
+  patch -Np1 -i ../fix_nullptr_mistakes.patch
 }
 
 build() {
