diff --git PKGBUILD PKGBUILD
index a30b1a9ef..105b57a25 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,14 +8,17 @@ pkgdesc="A background changing app"
 url="https://sourceforge.net/projects/fvwm-crystal/"
 license=('GPL2')
 depends=('imlib2')
-source=("http://archive.debian.org/debian/pool/main/h/habak/habak_$pkgver.orig.tar.gz")
+source=("http://archive.debian.org/debian/pool/main/h/habak/habak_$pkgver.orig.tar.gz"
+        "https://gitweb.gentoo.org/repo/gentoo.git/plain/x11-misc/habak/files/habak-0.2.5-fno-common.patch")
 arch=('x86_64')
 options=('!makeflags')
-md5sums=('6c1410a16a57d3483c41fad6ed2f6f49')
+md5sums=('6c1410a16a57d3483c41fad6ed2f6f49'
+         'a86e8dd8e2166020e449a2f11ed985d0')
 
 prepare() {
   cd "$srcdir/$pkgname-$pkgver"
   sed -i 's|^\(LDFLAGS.*$\)|\1 -lX11|' src/Makefile
+  patch -Np1 -i ../habak-0.2.5-fno-common.patch
 }
 
 build() {
