diff --git a/sample_app/forms.py b/sample_app/forms.py
index 059ff9e..cb3efc5 100644
--- a/sample_app/forms.py
+++ b/sample_app/forms.py
@@ -1,12 +1,12 @@
 from flask_wtf import Form
 from wtforms.fields import *
-from wtforms.validators import Required, Email
+from wtforms.validators import DataRequired, Email
 
 
 class SignupForm(Form):
-    name = TextField(u'Your name', validators=[Required()])
-    password = TextField(u'Your favorite password', validators=[Required()])
-    email = TextField(u'Your email address', validators=[Email()])
+    name = StringField(u'Your name', validators=[DataRequired()])
+    password = StringField(u'Your favorite password', validators=[DataRequired()])
+    email = StringField(u'Your email address', validators=[Email()])
     birthday = DateField(u'Your birthday')
 
     a_float = FloatField(u'A floating point number')
@@ -17,6 +17,6 @@ class SignupForm(Form):
                         description='...for no particular reason')
     sample_file = FileField(u'Your favorite file')
     eula = BooleanField(u'I did not read the terms and conditions',
-                        validators=[Required('You must agree to not agree!')])
+                        validators=[DataRequired('You must agree to not agree!')])
 
     submit = SubmitField(u'Signup')
