diff --git PKGBUILD PKGBUILD
index dd46158a..d3899f76 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ depends=('python-flask' 'python-dominate' 'python-visitor')
 makedepends=('python-setuptools')
 checkdepends=('python-pytest-runner' 'python-requests' 'python-flask-appconfig' 'python-flask-nav'
               'python-flask-wtf' 'python-flask-debug' 'python-email-validator')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/mbr/flask-bootstrap/archive/$pkgver.tar.gz")
-sha512sums=('010c71f69a95cb51afb449d1b43cb345ad23815ea89b368281f721a2af13cf2c143d67b586114898450d8e7fec413f45ff042cfca3dc1183cb0f50566e39fdc3')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/mbr/flask-bootstrap/archive/$pkgver.tar.gz"
+        "fix-sample-wtforms-3.patch")
+sha512sums=('010c71f69a95cb51afb449d1b43cb345ad23815ea89b368281f721a2af13cf2c143d67b586114898450d8e7fec413f45ff042cfca3dc1183cb0f50566e39fdc3'
+            '5d206bdb7bbef72d3864d7c6828188a6ef631741e8df7d3b819b0b0fba55bc6e9d1c804f9b6141a3b09a7844c9305bcc91cc8a17c7497d521d16846cd2f7e406')
+
+prepare() {
+  cd flask-bootstrap-$pkgver
+  patch -Np1 -i "../fix-sample-wtforms-3.patch"
+}
 
 build() {
   cd flask-bootstrap-$pkgver
