--- PKGBUILD
+++ PKGBUILD
@@ -72,11 +72,9 @@ makedepends=(
   libtheora               # libvtkIOOggTheora.so (1 direct lib, 2 total libs)
   mariadb-libs            # libvtkIOMySQL.so (1 direct lib, 2 total libs)
   opencascade             # libvtkIOOCCT.so (1 direct lib, 2 total libs)
-  openimagedenoise        # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   openvdb                 # libvtkIOOpenVDB.so (1 direct lib, 2 total libs)
   openvr                  # libvtkRenderingOpenVR.so (1 direct lib, 2 total libs)
   openxr                  # libvtkRenderingOpenXR.so (1 direct lib, 2 total libs)
-  ospray                  # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   pdal                    # libvtkIOPDAL.so (1 direct lib, 2 total libs)
   postgresql-libs         # libvtkIOPostgreSQL.so (1 direct lib, 2 total libs)
   unixodbc                # libvtkIOODBC.so (1 direct lib, 2 total libs)
@@ -113,8 +111,6 @@ optdepends=(
   'anari-sdk: ANARI rendering module'
   'openvr: rendering for virtual reality'
   'openxr: rendering for virtual and augmented reality'
-  'openimagedenoise: rendering with raytracing support'
-  'ospray: rendering with raytracing support'
   'openmpi: OpenMPI support'
   'adios2: IO module'
   'alembic: IO module'
@@ -165,6 +161,7 @@ build() {
   local _tkver=$(echo 'puts $tcl_version' | tclsh)
   cmake -B build -S ${pkgname^^}-${pkgver} \
     -DCMAKE_BUILD_TYPE=Release \
+    -DVTK_ENABLE_OSPRAY=OFF \
     -DCMAKE_CXX_FLAGS="$CXXFLAGS -ffat-lto-objects" \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DCMAKE_INSTALL_LICENSEDIR=share/licenses/vtk \
