Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1304604)
+++ PKGBUILD	(working copy)
@@ -14,8 +14,8 @@
 license=(BSD)
 depends=(gcc-libs double-conversion pugixml tbb)
 makedepends=(adios2 cmake boost doxygen ffmpeg gdal git gnuplot java-environment=11
-             liblas libxt mariadb-libs openimagedenoise openmp openmpi openvdb
-             openvr ospray pdal postgresql-libs python-matplotlib qt5-base
+             liblas libxt mariadb-libs openmp openmpi openvdb
+             openvr pdal postgresql-libs python-matplotlib qt5-base
              qt5-tools qt5-declarative tk unixodbc wget
              cgns cli11 eigen expat fmt freetype2 gl2ps glew hdf5 libjpeg
              jsoncpp libjsoncpp.so libharu proj libxml2 lz4 xz python-mpi4py
@@ -47,10 +47,8 @@
             lz4
             mariadb-libs
             netcdf
-            openimagedenoise
             openvdb
             openvr
-            ospray
             pdal
             postgresql-libs
             proj
@@ -115,6 +113,7 @@
     -DVTKm_ENABLE_HDF5_IO=ON \
     -DVTKm_ENABLE_MPI=ON \
     -DVTKm_ENABLE_OPENMP=ON \
+    -DVTK_ENABLE_OSPRAY=OFF \
     -DVTK_USE_EXTERNAL=ON \
     -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF \
     -DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF \
