diff --git PKGBUILD PKGBUILD
index 91928b807..c91c3e7d8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@
 license=(BSD)
 depends=(gcc-libs double-conversion pugixml tbb)
 makedepends=(adios2 cmake boost doxygen ffmpeg gdal git gnuplot java-environment=11
-             liblas libxt openimagedenoise openmp openmpi openvdb openvr ospray pdal
+             liblas libxt openmp openmpi openvdb openvr pdal
              postgresql python-matplotlib qt5-base qt5-tools qt5-webkit
              qt5-x11extras tk unixodbc wget
              cgns cli11 eigen expat fmt freetype2 gl2ps glew hdf5 libjpeg
@@ -46,10 +46,8 @@
             liblas
             lz4
             netcdf
-            openimagedenoise
             openvdb
             openvr
-            ospray
             pdal
             postgresql
             proj
@@ -102,6 +100,7 @@
     -DVTKm_ENABLE_HDF5_IO=ON \
     -DVTKm_ENABLE_MPI=ON \
     -DVTKm_ENABLE_OPENMP=ON \
+    -DVTK_ENABLE_OSPRAY=OFF \
     -DVTK_USE_EXTERNAL=ON \
     -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF \
     -DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF \
