diff --git PKGBUILD PKGBUILD
index 9c855f3..92fcbaf 100644
--- PKGBUILD
+++ PKGBUILD
@@ -51,13 +51,11 @@ makedepends=(
   lz4
   mariadb-libs
   netcdf
-  openimagedenoise
   openmp
   openmpi
   openvdb
   openvr
   openxr
-  ospray
   pdal
   postgresql-libs
   proj
@@ -98,12 +96,10 @@ optdepends=(
   lz4
   mariadb-libs
   netcdf
-  openimagedenoise
   'openmpi: OpenMPI support'
   openvdb
   openvr
   openxr
-  ospray
   pdal
   postgresql-libs
   proj
@@ -162,6 +158,7 @@ build() {
     -DVTKm_ENABLE_HDF5_IO=ON \
     -DVTKm_ENABLE_MPI=ON \
     -DVTKm_ENABLE_OPENMP=ON \
+    -DVTK_ENABLE_OSPRAY=OFF \
     -DVTK_USE_EXTERNAL=ON \
     -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF \
     -DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF \
