--- PKGBUILD
+++ PKGBUILD
@@ -78,11 +78,9 @@ makedepends=(
   mariadb-libs            # libvtkIOMySQL.so (1 direct lib, 2 total libs)
   onnxruntime             # libvtkFiltersONNX.so.1 (1 direct lib, 2 total libs)
   opencascade             # libvtkIOOCCT.so (1 direct lib, 2 total libs)
-  openimagedenoise        # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   openvdb                 # libvtkIOOpenVDB.so (1 direct lib, 2 total libs)
   openvr                  # libvtkRenderingOpenVR.so (1 direct lib, 2 total libs)
   openxr                  # libvtkRenderingOpenXR.so (1 direct lib, 2 total libs)
-  ospray                  # libvtkRenderingRayTracing.so (1 direct lib, 2 total libs)
   pdal                    # libvtkIOPDAL.so (1 direct lib, 2 total libs)
   postgresql-libs         # libvtkIOPostgreSQL.so (1 direct lib, 2 total libs)
   unixodbc                # libvtkIOODBC.so (1 direct lib, 2 total libs)
@@ -120,8 +118,6 @@ optdepends=(
   'onnxruntime: ONNX filter for AI-related computation'
   'openvr: rendering for virtual reality'
   'openxr: rendering for virtual and augmented reality'
-  'openimagedenoise: rendering with raytracing support'
-  'ospray: rendering with raytracing support'
   'openmpi: OpenMPI support'
   'viskores: accelerators support'
   'adios2: IO module'
@@ -155,13 +151,16 @@ source=(
   $url/files/release/${pkgver%.*}/VTK-$pkgver.tar.gz
   gdal-3.13.patch
   fmt-12.2.patch
+  anari-sdk-0.16.patch
 )
 sha256sums=('aed12cec12a9609179bf66329070266627ca64244a10856a452b2a17ffb04a1d'
             'b56f176de0fdf233f37d61bc41514ede03492987c50dc7cdb633adaa5d9f433c'
-            'b784782cac02e11a129da68cc9feef2175b1cabb9dcc7b6c036f60becf77d8f6')
+            'b784782cac02e11a129da68cc9feef2175b1cabb9dcc7b6c036f60becf77d8f6'
+            'd884ad3f2c9a6fb26552f0b92646b6b6906d89afcfc2ff7fcd111f4c82825345')
 b2sums=('343e3ee333ec0f5541014faf07fb66ac00f935e389af3dc588e1d75a023b77b2ecb2b57d8522ef46f77f0060811ee7c1c417becca5b605282704bcdf2bc750db'
         '3492d04286f15ed5978bb7c39fcdf0e9a4922b57c580da8056ffc10c38226c01abfb6dc5d1018db4fab51896ed1d298330a1ab96e40ce33803fdb59caf11d770'
-        'db47a96696f124de4588b289d01f01d1fb28c6b2408d68b505cb59f38244ae2c2e75787a9f43d4820441bf074e3f47246d7a5de9248cb8d3ff2dcaaa0b545640')
+        'db47a96696f124de4588b289d01f01d1fb28c6b2408d68b505cb59f38244ae2c2e75787a9f43d4820441bf074e3f47246d7a5de9248cb8d3ff2dcaaa0b545640'
+        '6e06621fdbdfa209c6ea9ca489b14dbd394c62497a6361c8944cf33a6e9c275f43afe0bfdd7f486a870984bee7504b2d569aca3f12429870f5146628cb351744')
 
 prepare() {
   cd ${pkgname^^}-${pkgver}
@@ -175,6 +174,8 @@ prepare() {
 
   # Fix build with fmt 12.2.0 https://gitlab.kitware.com/vtk/vtk/-/work_items/20092
   patch -p1 -i ../fmt-12.2.patch
+
+  patch -p1 -i ../anari-sdk-0.16.patch # Fix build with ANARI SDK 0.16
 }
 
 build() {
@@ -189,6 +190,7 @@ build() {
     -G Ninja
     -Wno-dev
     -DCMAKE_BUILD_TYPE=Release
+    -DVTK_ENABLE_OSPRAY=OFF
     -DCMAKE_CXX_FLAGS="$CXXFLAGS -ffat-lto-objects"
     -DCMAKE_INSTALL_PREFIX=/usr
     -DCMAKE_INSTALL_LICENSEDIR=share/licenses/vtk
@@ -221,6 +223,8 @@ build() {
     -DVTK_MODULE_ENABLE_VTK_IOCatalystConduit=YES
     # building with the usd package does not work: https://gitlab.archlinux.org/archlinux/packaging/packages/usd/-/issues/7
     -DVTK_MODULE_ENABLE_VTK_IOUSD=NO
+    # Depends on binary LookingGlass libraries not available for RISC-V
+    -DVTK_MODULE_ENABLE_VTK_RenderingLookingGlass=NO
     -DHDF5_NO_FIND_PACKAGE_CONFIG_FILE=ON
     -DHDF5_C_COMPILER_EXECUTABLE=h5hlcc
     -DHDF5_CXX_COMPILER_EXECUTABLE=h5hlc++
