Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010493)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,7 @@
 depends=(gcc-libs double-conversion tbb)
 makedepends=(adios2 cmake boost doxygen ffmpeg gdal git gnuplot java-environment=11
              liblas libxt openimagedenoise openmp openmpi openvdb openvr ospray pdal
-             postgresql python-matplotlib qt5-base qt5-tools qt5-webkit
+             postgresql python-matplotlib qt5-base qt5-tools
              qt5-x11extras tk unixodbc wget
              cgns double-conversion eigen expat fmt freetype2 gl2ps glew hdf5 libjpeg
              jsoncpp libharu proj libxml2 lz4 xz python-mpi4py netcdf libogg
@@ -29,7 +29,6 @@
             'python-matplotlib: for Matplotlib rendering'
             'openmpi: OpenMPI support'
             'qt5-x11extras'
-            'qt5-webkit: WebKit support'
             'tk: tcl bindings'
             adios2
             cgns
