diff --git PKGBUILD PKGBUILD
index 0f03f8bee..f7263916d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@
 license=(BSD)
 depends=(gcc-libs double-conversion tbb)
 makedepends=(adios2 cmake boost doxygen ffmpeg gdal git gnuplot java-environment=11
-             liblas libxt openimagedenoise openmp openmpi openvdb openvr ospray pdal
+             liblas libxt openmp openmpi openvdb openvr pdal
              postgresql python-matplotlib qt5-base qt5-tools qt5-webkit
              qt5-x11extras tk unixodbc wget
              cgns double-conversion eigen expat fmt freetype2 gl2ps glew hdf5 libjpeg
@@ -44,10 +44,8 @@
             liblas
             lz4
             netcdf
-            openimagedenoise
             openvdb
             openvr
-            ospray
             pdal
             postgresql
             pugixml
@@ -90,7 +88,7 @@
     -DVTK_BUILD_ALL_MODULES=ON \
     -DVTK_INSTALL_TCL_DIR=/usr/lib/tcl${_tkver}/vtk/ \
     -DVTK_LEGACY_REMOVE=ON \
-    -DVTK_SMP_ENABLE_OPENMP=ON \
+    -DVTK_SMP_ENABLE_OPENMP=OFF \
     -DVTK_SMP_IMPLEMENTATION_TYPE=TBB \
     -DVTK_PYTHON_VERSION=3 \
     -DVTK_USE_MPI=ON \
@@ -98,10 +96,11 @@
     -DVTK_VERSIONED_INSTALL=OFF \
     -DVTK_WRAP_JAVA=ON \
     -DVTK_WRAP_PYTHON=ON \
-    -DVTKOSPRAY_ENABLE_DENOISER=ON \
+    -DVTKOSPRAY_ENABLE_DENOISER=OFF \
     -DVTKm_ENABLE_HDF5_IO=ON \
     -DVTKm_ENABLE_MPI=ON \
-    -DVTKm_ENABLE_OPENMP=ON \
+    -DVTKm_ENABLE_OPENMP=OFF \
+    -DVTK_ENABLE_OSPRAY=OFF \
     -DVTK_USE_EXTERNAL=ON \
     -DVTK_MODULE_USE_EXTERNAL_VTK_cli11=OFF \
     -DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF \
