Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1010493)
+++ PKGBUILD	(working copy)
@@ -14,7 +14,7 @@
 depends=(gcc-libs double-conversion)
 makedepends=(cmake boost doxygen ffmpeg gdal gnuplot
              java-environment openmpi python-matplotlib qt5-base
-             qt5-tools qt5-webkit qt5-x11extras tk unixodbc wget
+             qt5-tools qt5-x11extras tk unixodbc wget
              double-conversion eigen expat freetype2 glew hdf5
              libjpeg jsoncpp libxml2 lz4 xz python-mpi4py netcdf
              libogg pdal libpng libtheora libtiff zlib
@@ -27,7 +27,6 @@
             'python-matplotlib: for Matplotlib rendering'
             'openmpi: OpenMPI support'
             'qt5-x11extras'
-            'qt5-webkit: WebKit support'
             'tk: tcl bindings'
             ffmpeg
             gdal
