diff --git PKGBUILD PKGBUILD
index 771216b30..22a9899d9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,8 +8,16 @@ arch=('x86_64')
 url='https://www.bunkus.org/videotools/ogmtools'
 license=('GPL')
 depends=('libvorbis' 'libdvdread' 'gcc-libs')
-source=("https://www.bunkus.org/videotools/${pkgname}/${pkgname}-${pkgver}.tar.bz2")
-sha256sums=('c8d61d1dbceb981dc7399c1a85e43b509fd3d071fb8d3ca89ea9385e6e40fdea')
+source=("https://www.bunkus.org/videotools/${pkgname}/${pkgname}-${pkgver}.tar.bz2"
+        "https://gitweb.gentoo.org/repo/gentoo.git/plain/media-sound/ogmtools/files/ogmtools-1.5-gcc11.patch")
+sha256sums=('c8d61d1dbceb981dc7399c1a85e43b509fd3d071fb8d3ca89ea9385e6e40fdea'
+            '5cbbe4406a3b04e3d333ddfaf09fb8c328c8df8a75543e216e1df317c3561c36')
+
+prepare(){
+  cd "${srcdir}/${pkgname}-${pkgver}"
+
+  patch -Np1 -i "${srcdir}/ogmtools-1.5-gcc11.patch"
+}
 
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
