diff --git PKGBUILD PKGBUILD
index 9de2de9..faeceb8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,7 +23,9 @@
 
 prepare() {
 	cd "$_archive"
-	cargo fetch --locked --target "$(rustc --print host-tuple)"
+	# https://github.com/str4d/rage/commit/6b097057f65aae50860ef1f0417df2bf4a26cbde
+	sed -i 's/created: 20\[\.\.\]-\[\.\.\]-\[\.\.\]T\[\.\.\]:\[\.\.\]:\[\.\.\]Z/created: 20[..]-[..]-[..]T[..]:[..]:[..]/' rage/tests/cmd/rage-keygen/gen-output{.toml,.out/key.txt}
+	cargo fetch --locked
 }
 
 build() {
