diff --git PKGBUILD PKGBUILD
index 33a2eb49..f1cabe21 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,7 +13,7 @@
 depends=(fuse3 glibc libgpg-error libsodium util-linux openssl xz zlib curl)
 makedepends=(avahi e2fsprogs git glib2 gobject-introspection gpgme gtk-doc
              libarchive libxslt mkinitcpio python systemd)
-checkdepends=(cpio elfutils gjs parallel python-yaml syslinux)
+checkdepends=(cpio elfutils gjs parallel python-yaml)
 options=(debug)
 source=(
   git+https://github.com/ostreedev/ostree#tag=$_commit
