diff --git PKGBUILD PKGBUILD
index 5790100a..a7dc8b11 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,8 @@
 license=(GPL)
 depends=(fuse2 glibc libgpg-error libsodium util-linux openssl xz zlib)
 makedepends=(avahi e2fsprogs git glib2 gobject-introspection gpgme gtk-doc
-             libarchive libsoup libxslt mkinitcpio python systemd)
-checkdepends=(cpio elfutils gjs parallel python-yaml syslinux)
+             libarchive libsoup libxslt python systemd)
+checkdepends=(cpio elfutils gjs parallel python-yaml)
 source=("git+https://github.com/ostreedev/ostree#commit=${_commit}"
         "git+https://github.com/mendsley/bsdiff"
         "git+https://gitlab.gnome.org/GNOME/libglnx.git")
@@ -46,10 +46,8 @@
     --localstatedir=/var \
     --sbindir=/usr/bin \
     --libexecdir=/usr/lib \
-    --with-builtin-grub2-mkconfig \
     --with-ed25519-libsodium \
     --with-dracut \
-    --with-mkinitcpio \
     --with-openssl \
     --enable-experimental-api \
     --disable-static \
