--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,8 @@ arch=(x86_64)
 license=(GPL)
 depends=(fuse2 glibc libgpg-error libsodium util-linux openssl xz zlib)
 makedepends=(avahi e2fsprogs git glib2 gobject-introspection gpgme gtk-doc
-             libarchive libsoup libxslt mkinitcpio python systemd)
-checkdepends=(cpio elfutils gjs parallel python-yaml syslinux)
+             libarchive libsoup libxslt python systemd)
+checkdepends=(cpio elfutils gjs parallel python-yaml)
 source=(
   git+https://github.com/ostreedev/ostree#commit=$_commit
   git+https://github.com/mendsley/bsdiff
@@ -47,10 +47,8 @@ build() {
     --localstatedir=/var \
     --sbindir=/usr/bin \
     --libexecdir=/usr/lib \
-    --with-builtin-grub2-mkconfig \
     --with-ed25519-libsodium \
     --with-dracut \
-    --with-mkinitcpio \
     --with-openssl \
     --enable-experimental-api \
     --disable-static \
