--- PKGBUILD
+++ PKGBUILD
@@ -79,3 +79,20 @@
   install -vDm 644 {CHANGES,MAINTAINERS,README}.rst -t "$pkgdir/usr/share/doc/$pkgname/"
   install -vDm 644 LICENSE.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+prepare() {
+  cd $pkgname
+  git apply ../$pkgname-matplotlib-3.11.patch
+  git apply ../$pkgname-svglib-2.patch
+  git apply ../$pkgname-plantuml-reference.patch
+}
+
+source+=("$pkgname-matplotlib-3.11.patch::https://github.com/rst2pdf/rst2pdf/commit/1809171972e26bb5db7c431252baf537bf30330c.patch?full_index=1"
+         "$pkgname-svglib-2.patch::https://github.com/rst2pdf/rst2pdf/commit/c31a234d88b9ea6ac312cd7480e2069b0cd463fc.patch?full_index=1"
+         "$pkgname-plantuml-reference.patch::https://github.com/rst2pdf/rst2pdf/commit/43e4b8435ba31c33edf68195be03db05e3d5457f.patch?full_index=1")
+sha512sums+=('848159e78170702e037025ac3f5ae81efe952803dc00e9122b533a3ede6d071f969ff0746ae91b1e6b2af1dd41612ce1003699d3f857440867a5dd01c4b89331'
+             'a11f88609a16ea4dbb39dacb140aa7d6b9c9a3b574a069e8ded650c56e936e7269193d7cd8d044e4d7cbeb80737272a13f3d919baee1b9c57f009e9508501be0'
+             '74d9721f66025342d4b63327cb0d1222c0fc012db5c37d2d2e99cef8dab1a55ad1802aff560c1ec778bbe2571c49d318688ec5d3ef87e6724556a1d082399220')
+b2sums+=('432b8b622c72c28e0f0ea49729a10dc3f5bffe09c9f8906b6add1a225234d14fb0d454d9903bfbf2a7a9bcde6a55a9c5a97ec344b30fc3bef711091ebfcd88f2'
+         '1e0bca5a5a9dfb8b74dea5e9eaff50f5298a968e7646d0c145c3fcb906681841f7c8347be58de01c732f6091e1cb22780d3597b9d40bcc74f9a90338556361fe'
+         '0d9f02f509a35f1592565f29dcf6889c213d1d2a93c438bfb7c0335fcb3403b57bc8711e0865d8c93fdeb74e2cd0d54f035497b2e590c37bc7c64f064f0864a2')
