--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,12 @@ makedepends=('intltool')
 source=("https://github.com/kdekorte/$pkgname/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
 sha256sums=('fc23c853deb35ee35db06dd3da5069ce3dc64faf3a0053324616fadf81a415db')
 
+prepare() {
+  cd $pkgname-$pkgver
+  cp /usr/share/autoconf/build-aux/config.guess config.guess
+  cp /usr/share/autoconf/build-aux/config.sub config.sub
+}
+
 build() {
   cd $pkgname-$pkgver
   ./configure --prefix=/usr
