--- a/toolchains/esbuild/esbuild_packages.bzl
+++ b/toolchains/esbuild/esbuild_packages.bzl
@@ -5,6 +5,7 @@
 _DARWIN_ARM64_SHA = "4a12abe17bc6d0438f8353862173d65570c70967f804ff383d6828b53befd4cc"
 _LINUX_AMD64_SHA = "4a8bad6e2d13e7edfef1351e4725f65a8828093b0d63bba3237fcab5e530afa2"
 _LINUX_ARM64_SHA = "893fb422f0c2759998a125cb23c3055712d5f3f4414ca75f3ec9dd55b03fcc8c"
+_LINUX_RISCV64_SHA = "5e50d8b1a4f64574e492d5d039506c19c0c4cce50eb3f9cd372ecf60037ee6de"
 _WINDOWS_AMD64_SHA = "00eb7f0b81591c2d842c8824cb9a948089566ffb629882298b640b425de45f2f"
 _LINUX_PPC64LE_SHA = "c6703bc387c5ad86073135d3837a4a1d47909737b7169b509c4f636bb492fd9f"
 _LINUX_S390X_SHA = "f27128ae64e19eae5c73c3b43ae11abbc4b0757c36e61df3cc043b7126fe9069"
@@ -56,6 +57,17 @@
                 "@platforms//cpu:arm64",
             ],
         ),
+        "linux_riscv64": struct(
+            sha = _LINUX_RISCV64_SHA,
+            urls = [
+                "https://registry.npmjs.org/esbuild-linux-riscv64/-/esbuild-linux-riscv64-%s.tgz" % _VERSION,
+            ],
+            binary_path = "bin/esbuild",
+            exec_compatible_with = [
+                "@platforms//os:linux",
+                "@platforms//cpu:riscv64",
+            ],
+        ),
         "windows_amd64": struct(
             sha = _WINDOWS_AMD64_SHA,
             urls = [
