diff --git PKGBUILD PKGBUILD
index cc4041b0..11ebebde 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,10 @@ makedepends=(docbook-xml startup-notification lsb-release gobject-introspection
 checkdepends=(xorg-server-xvfb aspell hspell hunspell nuspell libvoikko)
 groups=(gnome)
 _commit=ce3713dfd4f69275c2c46a6687c24deb8b8aa367  # tags/41.3^0
-source=("git+https://gitlab.gnome.org/GNOME/epiphany.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/epiphany.git#commit=$_commit"
+        "fix-meson-build.patch::https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/1031.patch")
+sha256sums=('SKIP'
+            'f7fa05bb177a8400f4c2f95dd2a3260ead32d2f321b29f89a8475699c4f4bf5a')
 
 pkgver() {
   cd $pkgname
@@ -24,6 +26,7 @@ pkgver() {
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i $srcdir/fix-meson-build.patch
 }
 
 build() {
