diff --git a/cmd/bpf2go/main.go b/cmd/bpf2go/main.go
index 99860d884..5b99fd5a9 100644
--- a/cmd/bpf2go/main.go
+++ b/cmd/bpf2go/main.go
@@ -58,7 +58,7 @@ var targetByGoArch = map[string]target{
 	"mips64le":    {"bpfel", ""},
 	"mips64p32le": {"bpfel", ""},
 	"ppc64le":     {"bpfel", "powerpc"},
-	"riscv64":     {"bpfel", ""},
+	"riscv64":     {"bpfel", "riscv"},
 	"armbe":       {"bpfeb", "arm"},
 	"arm64be":     {"bpfeb", "arm64"},
 	"mips":        {"bpfeb", ""},
