--- PKGBUILD
+++ PKGBUILD
@@ -21,18 +21,26 @@
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
-	export TARGET_GOARCH=amd64
+	if [[ $CARCH == "riscv64" ]]; then
+		export TARGET_GOARCH=riscv64
+	else
+		export TARGET_GOARCH=amd64
+	fi
 	go generate
 }
 
 build() {
 	cd "$pkgname-$pkgver"
+	export CGO_ENABLED=1
 	export CGO_CPPFLAGS="${CPPFLAGS}"
 	export CGO_CFLAGS="${CFLAGS}"
 	export CGO_CXXFLAGS="${CXXFLAGS}"
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
+	if [[ $CARCH == "riscv64" ]]; then
+		export TARGET_GOARCH=riscv64 LIBPCAP_ARCH=riscv64-unknown-linux-gnu
+	fi
 	make libpcap/libpcap.a
 	go build \
 		-ldflags "-compressdwarf=false -linkmode external -X \"github.com/cilium/pwru/internal/pwru.Version=${pkgver}\"" \
