--- PKGBUILD
+++ PKGBUILD
@@ -15,13 +15,15 @@ sha256sums=('77968e728e3bace015dd332e04c7f2cb73e3fd7cd4e6046f841973044b4da2ba')
 
 prepare() {
 	cd "$pkgname-$pkgver"
+	patch -Np1 -i ../pwru-add-vmlinux-riscv64.patch
+	patch -Np1 -i ../pwru-kprobe-add-riscv64-support.patch
 	export CGO_CPPFLAGS="${CPPFLAGS}"
 	export CGO_CFLAGS="${CFLAGS}"
 	export CGO_CXXFLAGS="${CXXFLAGS}"
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
-	export TARGET_GOARCH=amd64
+	export TARGET_GOARCH=riscv64
 	go generate
 }
 
@@ -33,6 +35,7 @@ build() {
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
+	export GOARCH=riscv64 TARGET_GOARCH=riscv64 LIBPCAP_ARCH=riscv64-unknown-linux-gnu
 	make libpcap/libpcap.a
 	go build \
 		-ldflags "-compressdwarf=false -linkmode external -X \"github.com/cilium/pwru/internal/pwru.Version=${pkgver}\"" \
@@ -43,3 +46,8 @@ package() {
 	cd "$pkgname-$pkgver"
 	install -Dm755 ${pkgname} -t "$pkgdir/usr/bin"
 }
+
+source+=("pwru-add-vmlinux-riscv64.patch::https://github.com/hack3ric/pwru/commit/8aa5e0f00c849ea0ef83bb8f491bb089a21e9820.diff"
+         "pwru-kprobe-add-riscv64-support.patch::https://github.com/hack3ric/pwru/commit/1d6b5db47aa46cc85f90a74249ba77cd03a8c91f.diff")
+sha256sums+=('cfc9a074c2af7b8b9cad0e7520cff11b7726626ca5cbf3683106402cadcf6fca'
+             '5bce8166d3a5f38c51e29ce33b22138169ef308b99b378b0eb9cb4f9ff2fe54e')
