--- PKGBUILD
+++ PKGBUILD
@@ -15,13 +15,17 @@ sha256sums=('19317ceb2ae330e28b5f969ab1ed64e6f665f89773437dfdca1d24b4dea811f7')
 
 prepare() {
 	cd "$pkgname-$pkgver"
+	patch -Np1 -i ../pwru-add-vmlinux-riscv64.patch
+	go mod edit -replace github.com/cilium/ebpf=github.com/cilium/ebpf@v0.12.0
+	go mod tidy
+	go mod vendor
 	export CGO_CPPFLAGS="${CPPFLAGS}"
 	export CGO_CFLAGS="${CFLAGS}"
 	export CGO_CXXFLAGS="${CXXFLAGS}"
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
-	export TARGET_GOARCH=amd64
+	export TARGET_GOARCH=riscv64
 	go generate
 }
 
@@ -33,13 +37,18 @@ build() {
 	export CGO_LDFLAGS="${LDFLAGS}"
 	export GOPATH="${srcdir}"
 	export GOFLAGS="-buildmode=pie -mod=vendor -modcacherw"
+	export GOARCH=riscv64 TARGET_GOARCH=riscv64 LIBPCAP_ARCH=riscv64-unknown-linux-gnu
 	make libpcap/libpcap.a
 	go build \
 		-ldflags "-compressdwarf=false -linkmode external -X \"github.com/cilium/pwru/internal/pwru.Version=${pkgver}\"" \
-		-o ${pkgname}
+		-o ${pkgname} \
+		*.go  # WORKAROUND: `riscv` is a reserved architecture in Go, so *_riscv.go is not included by default on linux/riscv64
 }
 
 package() {
 	cd "$pkgname-$pkgver"
 	install -Dm755 ${pkgname} -t "$pkgdir/usr/bin"
 }
+
+source+=("pwru-add-vmlinux-riscv64.patch")
+sha256sums+=('d7c7942dc3dd6874d5b35063e11231b2f99c418a1c3d5fd97f607c1c4aa956b7')
