diff --git PKGBUILD PKGBUILD
index bae51eb85..5ff20ed22 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,10 @@
 depends=(gtkmm3 gspell libxslt dconf libsecret)
 makedepends=(yelp-tools meson git)
 _commit=56074f03883f11fd91669b5dcd2145efa7440128  # tags/41.2^0
-source=("git+https://gitlab.gnome.org/GNOME/gnote.git#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/GNOME/gnote.git#commit=$_commit"
+        "drop-unused-argument-for-i18n.merge_file.patch::https://gitlab.gnome.org/GNOME/gnote/-/commit/32795e53681e3ecbaaa05817d4eb5fa1c124d35b.patch")
+sha256sums=('SKIP'
+            '7a91dd545b447adc2b955dcc3933a7d2bd424a037973f61420fb56685f3452a0')
 
 pkgver() {
   cd gnote
@@ -23,6 +25,7 @@
 
 prepare() {
   cd gnote
+  patch -Np1 -i ../drop-unused-argument-for-i18n.merge_file.patch
 }
 
 build() {
