--- PKGBUILD
+++ PKGBUILD
@@ -21,8 +21,15 @@
   glm
   meson
 )
-source=("git+https://github.com/WayfireWM/wf-config.git#tag=v$pkgver")
-b2sums=(d74cf44a8a33bd8225f2c6342e6536a4996aa55d3d2d6935f8d133b51b420294c6e8f7ab583dbc8f7c9536afb159ad885a543aba2ab8f413325edf771e5aae58)
+source=("git+https://github.com/WayfireWM/wf-config.git#tag=v$pkgver"
+        widen-simple-animation-timing.patch)
+b2sums=(d74cf44a8a33bd8225f2c6342e6536a4996aa55d3d2d6935f8d133b51b420294c6e8f7ab583dbc8f7c9536afb159ad885a543aba2ab8f413325edf771e5aae58
+        952701135865692bacdbdfac904b04d592fb5495f8aae80a071be93679e2c3167ec61e044ce949bbe2775ac52b86dc9a8d617bc482e5c299cd05fc9959d90e49)
+
+prepare() {
+  cd $pkgname
+  patch -Np1 -i ../widen-simple-animation-timing.patch
+}
 
 build() {
   arch-meson $pkgname build
