--- a/test/duration_test.cpp
+++ b/test/duration_test.cpp
@@ -126,7 +126,7 @@
 
 TEST_CASE("wf::animation::simple_animation_t")
 {
-    auto length = std::make_shared<option_t<int>>("length", 10);
+    auto length = std::make_shared<option_t<int>>("length", 100);
     simple_animation_t anim{length, smoothing::linear};
 
     auto cycle_through = [&] (double s, double e, bool x1, bool x2)
@@ -144,10 +144,10 @@
 
         CHECK(anim.running());
         CHECK((double)anim == doctest::Approx(s));
-        usleep(5000);
+        usleep(50000);
         CHECK((double)anim == doctest::Approx((s + e) / 2).epsilon(0.1));
         CHECK(anim.running());
-        usleep(5500);
+        usleep(60000);
         CHECK((double)anim == doctest::Approx(e));
         CHECK(anim.running());
         CHECK(!anim.running());
