Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422452)
+++ PKGBUILD	(working copy)
@@ -21,6 +21,7 @@
   cd "${pkgname}-${pkgver}"
   # FS#36248 - strip march=native
   sed -e '/native/d' \
+      -e '/ENABLE_SSE2/d' \
       -i {source,apps}/Makefile
   # removing local call to ldconfig
   sed -e 's/ldconfig/& -N $(DESTDIR)\/$(LIBDIR)/' \
