diff --git PKGBUILD PKGBUILD
index e1975ff..035c0d6 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,8 +13,15 @@ depends=(qt6-quick3d)
 makedepends=(cmake ninja qt6-shadertools)
 groups=(qt6)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
-sha256sums=('3291301d243ff39febc96d6641eb4b44feeef43087c475cbb0b22dc910a5a4ba')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
+        physx-rv64.patch)
+sha256sums=('3291301d243ff39febc96d6641eb4b44feeef43087c475cbb0b22dc910a5a4ba'
+            '404a2d7b532a5b78c13b2a110aebebdce1dbcc2f9138319bb53a6b9ca7dce785')
+
+prepare() {
+  cd $_pkgfn
+  patch -Np1 -i "$srcdir/physx-rv64.patch"
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
