diff --git PKGBUILD PKGBUILD
index adc90fa..f296a88 100644
--- PKGBUILD
+++ PKGBUILD
@@ -23,4 +23,12 @@ source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodu
 sha256sums=('74395b686f680cfa302e41530ff9bbe1b7e13c1d1aa9d68aa192318ecbfd7c03')
 
+source+=(physx-rv64.patch)
+sha256sums+=('404a2d7b532a5b78c13b2a110aebebdce1dbcc2f9138319bb53a6b9ca7dce785') 
+
+prepare() {
+  cd $_pkgfn
+  patch -Np1 -i "$srcdir/physx-rv64.patch"
+}
+
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
