diff --git PKGBUILD PKGBUILD
index 659838f..673f6b9 100644
--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,15 @@ makedepends=(cmake ninja qt6-shadertools)
 groups=(qt6)
 options=(debug)
 _pkgfn=${pkgname/6-/}-everywhere-src-$_qtver
-source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz)
-sha256sums=('1dd4a81e671436c8f468f84dfc52fa2fb54bde043598cdb7c6fc69d0511badbb')
+source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/$_qtver/submodules/$_pkgfn.tar.xz
+    physx-rv64.patch)
+sha256sums=('1dd4a81e671436c8f468f84dfc52fa2fb54bde043598cdb7c6fc69d0511badbb'
+            '76e430885958c5680818de9c4a0f9e4bf5ec43a3c537e3e7aa45fb7542853570')
+
+prepare() {
+  cd $_pkgfn
+  patch -Np1 < $srcdir/physx-rv64.patch
+}
 
 build() {
   cmake -B build -S $_pkgfn -G Ninja \
