diff --git PKGBUILD PKGBUILD
index b12ec8142..c025fd452 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,10 +9,18 @@ arch=('x86_64')
 url="http://xplc.sourceforge.net"
 license=('LGPL')
 depends=('gcc-libs')
-source=("https://downloads.sourceforge.net/${pkgname}/$pkgname-$pkgver.tar.gz")
-md5sums=('39da3270527c8712b8e8fcf03768d29f')
+source=("https://downloads.sourceforge.net/${pkgname}/$pkgname-$pkgver.tar.gz"
+        "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-libs/wvstreams/files/wvstreams-4.6.1_p14-xplc-module.patch")
+md5sums=('39da3270527c8712b8e8fcf03768d29f'
+         '9aa32c56494e82a84d92acdd6ef49d2b')
 options=('!buildflags')
 
+prepare(){
+  cd "$srcdir/$pkgname-$pkgver"
+
+  patch -Np1 -i ../wvstreams-4.6.1_p14-xplc-module.patch
+}
+
 build() {
   cd "$srcdir/$pkgname-$pkgver"
   
