--- go.mod
+++ go.mod
@@ -170,3 +170,10 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )
+
+exclude github.com/bugsnag/osext v0.0.0-20130617224835-0dd3f918b21b
+
+replace (
+       github.com/bugsnag/osext v0.0.0-20130617224835-0dd3f918b21b => github.com/kardianos/osext v0.0.0-20190222173326-2bc1f35cddc0
+       github.com/bugsnag/panicwrap v0.0.0-20151223152923-e2c28503fcd0 => github.com/hack3ric/panicwrap v0.0.0-20230905013234-d1739d73e061
+)
