--- PKGBUILD
+++ PKGBUILD
@@ -26,6 +26,7 @@ sha256sums=('77b3b7ce47e6a2ba79d38f5e6936ff94177ff401708421198bbbab8d34faa22f')
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../add-riscv64-build.patch
   go mod download -x
 }
 
@@ -56,3 +57,6 @@ package() {
   bin/helm completion zsh | install -Dm644 /dev/stdin "$pkgdir/usr/share/zsh/site-functions/_helm"
   bin/helm completion fish | install -Dm644 /dev/stdin "$pkgdir/usr/share/fish/vendor_completions.d/helm.fish"
 }
+
+source+=("add-riscv64-build.patch")
+sha256sums+=('7f9f010b8b91d9c692b9fe7d96cc8d870f601aec366454f23fc1f59d40b8e1c0')
