--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,7 @@ sha256sums=('cba84628e96df97809eb61aa2f70b5e61c157e5e1b075c4aadd9d62f1815320b')
 
 prepare() {
   cd $pkgname
+  patch -Np1 -i ../add-riscv64-build.patch
   GOFLAGS="-mod=readonly" go mod vendor -v
 }
 
@@ -66,3 +67,6 @@ package() {
   ./helm completion zsh | install -vDm644 /dev/stdin "$pkgdir/usr/share/zsh/site-functions/_helm"
   ./helm completion fish | install -vDm644 /dev/stdin "$pkgdir/usr/share/fish/vendor_completions.d/helm.fish"
 }
+
+source+=("add-riscv64-build.patch")
+sha256sums+=('7f9f010b8b91d9c692b9fe7d96cc8d870f601aec366454f23fc1f59d40b8e1c0')
