--- PKGBUILD
+++ PKGBUILD
@@ -16,12 +16,14 @@ makedepends=("go" "git")
 options=("!lto")
 # _commit=3547a4b5bf5edb5478ce352e18858d8a552a4110	#refs/tags/v3.13.1^{}
 # source=("git+https://github.com/helm/helm.git#commit=${_commit}?signed")
-source=("git+https://github.com/helm/helm.git#tag=v${pkgver}?signed")
+source=("git+https://github.com/helm/helm.git#tag=v${pkgver}?signed"
+        "add-riscv64-build.patch")
 validpgpkeys=('672C657BE06B4B30969C4A57461449C25E36B98E'
               'CABAA8D44DFACA14791FBE9892C44A3D421FF7F9'
               '967F8AC5E2216F9F4FD270AD92AA783CBAAE8E3B'
               'F1261BDE929012C8FF2E501D6EA5D7598529A53E')
-sha256sums=('SKIP')
+sha256sums=('SKIP'
+            '0708ba6d62eae84d74c290ad13092ec997aa6883ef80b9854ec8080a5479a2ed')
 
 pkgver() {
   cd "${pkgname}"
@@ -30,6 +32,7 @@ pkgver() {
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../add-riscv64-build.patch
   go mod tidy -compat=1.17
 }
 
@@ -39,7 +42,7 @@ build() {
     export CGO_CFLAGS="$CFLAGS"
     export CGO_CXXFLAGS="$CXXFLAGS"
     export CGO_CPPFLAGS="$CPPFLAGS"
-    make EXT_LDFLAGS="-linkmode external" GOFLAGS="-buildmode=pie -trimpath"
+    make EXT_LDFLAGS="-linkmode external" GOFLAGS="-buildmode=pie -trimpath" CGO_ENABLED=1
 }
 
 check(){
@@ -48,7 +51,7 @@ check(){
     export CGO_CFLAGS="$CFLAGS"
     export CGO_CXXFLAGS="$CXXFLAGS"
     export CGO_CPPFLAGS="$CPPFLAGS"
-    make LDFLAGS="-s -w -linkmode external" GOFLAGS="-buildmode=pie -trimpath" test-unit || true
+    make LDFLAGS="-s -w -linkmode external" GOFLAGS="-buildmode=pie -trimpath" CGO_ENABLED=1 test-unit || true
 }
 
 package(){
