--- PKGBUILD
+++ PKGBUILD
@@ -16,12 +16,14 @@ makedepends=("go" "git")
 options=("!lto")
 # _commit=3547a4b5bf5edb5478ce352e18858d8a552a4110	#refs/tags/v3.13.1^{}
 # source=("git+https://github.com/helm/helm.git#commit=${_commit}?signed")
-source=("git+https://github.com/helm/helm.git#tag=v${pkgver}?signed")
+source=("git+https://github.com/helm/helm.git#tag=v${pkgver}?signed"
+        "add-riscv64-build.patch")
 validpgpkeys=('672C657BE06B4B30969C4A57461449C25E36B98E'
               'CABAA8D44DFACA14791FBE9892C44A3D421FF7F9'
               '967F8AC5E2216F9F4FD270AD92AA783CBAAE8E3B'
               'F1261BDE929012C8FF2E501D6EA5D7598529A53E')
-sha256sums=('SKIP')
+sha256sums=('SKIP'
+            '3a7c7c5a32ddd9667a9c6f80278dec75c03706b9bd953430acd528e5663f253f')
 
 pkgver() {
   cd "${pkgname}"
@@ -30,6 +32,7 @@ pkgver() {
 
 prepare() {
   cd "${pkgname}"
+  patch -Np1 -i ../add-riscv64-build.patch
   go mod tidy -compat=1.17
 }
 
