--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,14 @@ arch=('x86_64')
 depends=('python-numpy')
 makedepends=('git' 'python-setuptools' 'cmake')
 checkdepends=('python-pytest-runner')
-source=("git+https://github.com/heitzmann/gdstk.git#commit=$_commit")
-sha512sums=('SKIP')
+source=("git+https://github.com/heitzmann/gdstk.git#commit=$_commit"
+        "fix-fma-test-failure.patch::https://github.com/heitzmann/gdstk/pull/177.diff")
+sha512sums=('SKIP'
+            '2e45858d939ab725184ee17093cb9c447fcad86f22864ea618fa002390983c17feb66d73b6f43a814e72389af58a7a8a647e3c68f100b1f568aa6afab3223a01')
 
 prepare() {
   cd gdstk
+  patch -Np1 -i ../fix-fma-test-failure.patch
   # Upstream wants to test CI :/
   sed -i 's/oldest-supported-numpy/numpy/' setup.py
 }
