--- PKGBUILD
+++ PKGBUILD
@@ -73,6 +73,7 @@
   (
     cd desktop
     npm clean-install --ignore-scripts
+    patch -Np1 -i "$srcdir/electron-builder-riscv64.patch"
     npm run build-typescript -w nseventforwarder
     npm run build-typescript -w windows-utils
   )
@@ -145,7 +146,7 @@
 
   cd mullvadvpn-app
   install -vDm644 -t "$pkgdir/usr/lib/$pkgbase" \
-    desktop/packages/mullvad-vpn/dist/linux-unpacked/resources/app.asar
+    desktop/packages/mullvad-vpn/dist/linux-riscv64-unpacked/resources/app.asar
   install -vDm755 "../$pkgbase.sh" "$pkgdir/usr/bin/$pkgname"
   install -vDm644 -t "$pkgdir/usr/share/applications" "../$pkgbase.desktop"
 
@@ -154,3 +155,6 @@
       "$pkgdir/usr/share/icons/hicolor/${icon_size}x${icon_size}/apps/$pkgname.png"
   done
 }
+
+source+=(electron-builder-riscv64.patch)
+b2sums+=('e599e12b48bc526ebe5fa4f246387e36d771bc68ae5373f38f0a253f34641295987ba7a6b98490a95413c27f37bd17163d8c8287893adad3c34a454d0231c8db')
