diff --git PKGBUILD PKGBUILD
index d6f3f1b..77e5839 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,9 +16,9 @@ source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pk
         https://download.geonames.org/export/dump/admin2Codes.txt)
 sha256sums=('09a5fdadeb3bd94ab5039a63fef6d47c534ec586f41af2517a389792e475e6d8'
             'SKIP'
-            'b524b9f1df159879e266be551ecac78e674c46a080ca6108d84f057e18b64799'
-            'b81ddb93812c3c7e0e26d5d7520fa224992f22c2ef3d6422f5f6ca0a6c8e2ffe'
-            '8fbf2f16ed3cb97ac62f48b9bfdf51af20fe0ea054c9b836c511a322e13b5782')
+            '9d776b57056d2c2c969d54c310bc3b6dc5afd4589baed81e45ecc4ea0f79a5d3'
+            '52f5125b29bcf8dd864ede32f8ed81fe189bc18cd21b7e20212f15686904d102'
+            'b29e85bea1b352850cb3ef04ab7c1d7df39d6467f7d2459a1748b263176ec865')
 validpgpkeys=(CA262C6C83DE4D2FB28A332A3A6A4DB839EAA6D7  # Albert Astals Cid <aacid@kde.org>
               F23275E4BF10AFC1DF6914A6DBD2CE893E2D1C87  # Christoph Feck <cfeck@kde.org>
               D81C0CB38EB725EF6691C385BB463350D6EF31EF) # Heiko Becker <heiko.becker@kde.org>
