--- PKGBUILD
+++ PKGBUILD
@@ -125,7 +125,7 @@ build() {
 	cmake "${srcdir}/mysql-workbench-community-${pkgver}-src" \
 		-Wno-dev \
 		-DCMAKE_INSTALL_PREFIX:PATH=/usr \
-		-DCMAKE_CXX_FLAGS="-std=c++17" \
+		-DCMAKE_CXX_FLAGS="-std=c++17 -Wno-error=narrowing" \
 		-DCMAKE_BUILD_TYPE=Release \
 		-DMySQL_CONFIG_PATH="${srcdir}/install-bundle/usr/bin/mysql_config" \
 		-DMySQLCppConn_LIBRARY="${srcdir}/install-bundle/usr/lib/libmysqlcppconn.so" \
