--- PKGBUILD
+++ PKGBUILD
@@ -74,6 +74,8 @@ prepare() {
 
 	# be happy with more recent antlr
 	sed -i "/set(ANTLR_JAR_FILENAME/s/4.9.1/${_antlr4_version}/" CMakeLists.txt
+
+	sed -i 's|unixodbc_conf.h|unixODBC/unixodbc_conf.h|' build/cmake/Modules/FindUNIXODBC.cmake plugins/migration/CMakeLists.txt
 }
 
 build() {
@@ -144,7 +146,7 @@ build() {
 	cmake "${srcdir}/mysql-workbench-community-${pkgver}-src" \
 		-Wno-dev \
 		-DCMAKE_INSTALL_PREFIX:PATH=/usr \
-		-DCMAKE_CXX_FLAGS="-std=c++17" \
+		-DCMAKE_CXX_FLAGS="-std=c++17 -Wno-error=narrowing" \
 		-DCMAKE_BUILD_TYPE=Release \
 		-DMySQL_CONFIG_PATH="${srcdir}/install-bundle/usr/bin/mysql_config" \
 		-DMySQLCppConn_LIBRARY="${srcdir}/install-bundle/usr/lib/libmysqlcppconn.so" \
