--- PKGBUILD
+++ PKGBUILD
@@ -14,8 +14,10 @@ makedepends=(glib2 freetype2 graphite cairo icu gobject-introspection gtk-doc
 checkdepends=(python-fonttools python-setuptools)
 options=(debug)
 _commit=f1f2be776bcd994fa9262622e1a7098a066e5cf7  # tags/5.1.0^0
-source=("git+https://github.com/harfbuzz/harfbuzz#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://github.com/harfbuzz/harfbuzz#commit=$_commit"
+        "fix-char-signedness.patch::https://github.com/harfbuzz/harfbuzz/pull/3771.patch")
+sha256sums=('SKIP'
+            'a4fe35709c291cf3f703b74eb7db35e290ecc6e90c7d10ffe35bfc562fb63a95')
 
 pkgver() {
   cd harfbuzz
@@ -24,6 +26,7 @@ pkgver() {
 
 prepare() {
   cd harfbuzz
+  patch -Np1 -i ../fix-char-signedness.patch
 }
 
 build() {