diff --git PKGBUILD PKGBUILD
index 75667c1..fedd4a2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,8 +10,14 @@ url="http://simonkagstrom.github.com/kcov/index.html"
 license=('GPL2')
 depends=('elfutils' 'curl')
 makedepends=('cmake' 'python' 'ninja')
-source=("https://github.com/SimonKagstrom/kcov/archive/v${pkgver}.tar.gz")
-sha512sums=('5abe1dd94fbe93fe73d658840593781216b8562a87d010a94d4520e29dd4d537e167de64b7f535347f894c68d5d8b9974ee4b9fd2cebf9291bac4ddbb9ac7cff')
+source=("https://github.com/SimonKagstrom/kcov/archive/v${pkgver}.tar.gz"
+        "kcov-add-basic-riscv.patch::https://github.com/SimonKagstrom/kcov/pull/385.patch")
+sha512sums=('5abe1dd94fbe93fe73d658840593781216b8562a87d010a94d4520e29dd4d537e167de64b7f535347f894c68d5d8b9974ee4b9fd2cebf9291bac4ddbb9ac7cff'
+            '998ac90533331efb9f6da43ade7a65a3db3368317bf777b9337ed1a547fb6f1fa1c86fbbe8c2763dc74965b4b6e4922cc7792cad58e2165c78930c7aa260272c')
+
+prepare() {
+  patch -Np1 -d $srcdir/$pkgname-$pkgver -i $srcdir/kcov-add-basic-riscv.patch
+}
 
 build() {
   cd $pkgname-$pkgver
