Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 460539)
+++ PKGBUILD	(working copy)
@@ -26,12 +26,15 @@
     -D a64-neon=disabled \
     -D iwmmxt=disabled \
     -D mips-dspr2=disabled \
-    -D gtk=disabled
+    -D gtk=disabled \
+    -D mmx=disabled \
+    -D sse2=disabled \
+    -D ssse3=disabled
   ninja -C build
 }
 
 check() {
-  meson test -C build
+  meson test -C build --timeout-multiplier 10
 }
 
 package() {
