diff --git a/packages/tsc/tests/index.spec.ts b/packages/tsc/tests/index.spec.ts
index b50b76984..9ae29c5bf 100644
--- a/packages/tsc/tests/index.spec.ts
+++ b/packages/tsc/tests/index.spec.ts
@@ -75,6 +75,6 @@ function runVueTsc(cwd: string) {
 
 describe(`vue-tsc`, () => {
 	for (const [path, isRoot] of tests) {
-		it(`vue-tsc no errors (${prettyPath(path, isRoot)})`, () => runVueTsc(path), 40_000);
+		it(`vue-tsc no errors (${prettyPath(path, isRoot)})`, () => runVueTsc(path), 40_0000);
 	}
 });
