--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@ license=('MIT')
 depends=('nodejs')
 makedepends=('git' 'npm' 'pnpm')
 optdepends=('typescript: for use in typescript.tsdk')
-source=("$_name::git+https://github.com/vuejs/language-tools.git#tag=v$pkgver")
-b2sums=('SKIP')
+source=("$_name::git+https://github.com/vuejs/language-tools.git#tag=v$pkgver"
+        "bump-rollup-for-riscv64.patch::https://github.com/vuejs/language-tools/pull/3825.diff")
+b2sums=('SKIP'
+        'b861be6c2a025182f7998fa9b6fd1de221b9e39eb3cb00a277a57289cf17f6637debc414a023b7531e9682974f6f9ddc480b3af5dad7ebbc08dc32e6bf6d33a5')
 
 prepare() {
   cd $_name
+  patch -Np1 -i ../bump-rollup-for-riscv64.patch
   pnpm install --frozen-lockfile
 }
 
