--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ arch=('x86_64')
 license=('MIT')
 provides=('xdg-desktop-portal-impl')
 depends=('xdg-desktop-portal' 'pipewire' 'pipewire-session-manager' 'libinih')
-makedepends=('meson' 'wayland-protocols' 'wayland' 'scdoc')
+makedepends=('meson' 'wayland-protocols' 'wayland' 'scdoc' 'gdm')
 optdepends=(
     'slurp: to choose which output to screencast using slurp'
     'wofi: to choose which output to screencast using wofi'
