diff --git PKGBUILD PKGBUILD
index a4a3f9f7..36fff8a5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -36,11 +36,11 @@ pkgver() {
 }
 
 build() {
-  make PREFIX=/usr DISPATCH=1 -C xxHash
+  make PREFIX=/usr -C xxHash
 }
 
 package() {
-  make PREFIX=/usr DISPATCH=1 DESTDIR="${pkgdir}" -C xxHash install
+  make PREFIX=/usr DESTDIR="${pkgdir}" -C xxHash install
   install -Dm 644 xxHash/LICENSE -t "${pkgdir}"/usr/share/licenses/xxhash
 }
 
