--- PKGBUILD
+++ PKGBUILD
@@ -41,17 +41,17 @@ build() {
   # Use LTO
   export CARGO_PROFILE_RELEASE_LTO=true CARGO_PROFILE_RELEASE_CODEGEN_UNITS=1
 
-  cargo build --release --frozen --features all,dist-server
+  cargo build --release --frozen --features all
 }
 
 #check() {
 #  cd sccache
-#  cargo test --frozen --features all,dist-server
+#  cargo test --frozen --features all
 #}
 
 package() {
   cd sccache
-  install -Dt "$pkgdir/usr/bin" target/release/sccache{,-dist}
+  install -Dt "$pkgdir/usr/bin" target/release/sccache
   install -Dt "$pkgdir/usr/share/doc/$pkgname" -m644 README.md docs/*
 }
 
