--- PKGBUILD
+++ PKGBUILD
@@ -46,7 +46,7 @@ build() {
     --release \
     --frozen \
     --manifest-path sccache/Cargo.toml \
-    --features all,dist-server,native-zlib
+    --features all,native-zlib
 }
 
 #check() {
@@ -66,7 +66,7 @@ package() {
     --no-track \
     --path sccache \
     --root "${pkgdir}"/usr \
-    --features all,dist-server,native-zlib
+    --features all,native-zlib
 }
 
 # vim: ts=2 sw=2 et:
