--- PKGBUILD
+++ PKGBUILD
@@ -34,6 +34,8 @@ pkgver() {
 }
 
 prepare() {
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> sccache/Cargo.toml
+  cargo update -p ring --manifest-path sccache/Cargo.toml
   cargo fetch \
     --locked \
     --manifest-path sccache/Cargo.toml
@@ -46,7 +48,7 @@ build() {
     --release \
     --frozen \
     --manifest-path sccache/Cargo.toml \
-    --features all,dist-server,native-zlib
+    --features all,native-zlib
 }
 
 #check() {
@@ -66,7 +68,7 @@ package() {
     --no-track \
     --path sccache \
     --root "${pkgdir}"/usr \
-    --features all,dist-server,native-zlib
+    --features all,native-zlib
 }
 
 # vim: ts=2 sw=2 et:
