--- PKGBUILD
+++ PKGBUILD
@@ -14,17 +14,20 @@ makedepends=('cmake' 'python-sphinx' 'vim' 'libedit' 'libevent' 'libfido2')
 optdepends=('qpress: for compressed backups')
 source=("https://www.percona.com/downloads/Percona-XtraBackup-LATEST/Percona-XtraBackup-${pkgver//_/-}/source/tarball/percona-xtrabackup-${pkgver//_/-}.tar.gz"
         "https://boostorg.jfrog.io/artifactory/main/release/$_boost_ver/source/boost_${_boost_ver//./_}.tar.gz"
-        "mysql-build-fix.patch")
+        "mysql-build-fix.patch"
+        "add-riscv-support.patch::https://patch-diff.githubusercontent.com/raw/percona/percona-xtrabackup/pull/1443.diff")
 
 sha256sums=('3a6f7077db8489eb00f25ad01daeaed4fa087100068fac79215c7d2e10424f87'
             '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131'
-            'e99733bad4d66ccb87cd87be4f030023184fba54bfc354b018773372bbf0ddf1')
+            'e99733bad4d66ccb87cd87be4f030023184fba54bfc354b018773372bbf0ddf1'
+            '08ffa60b8370a760d35744919729d66e1714c49322aaca810e9e57ff951c582b')
 
 prepare() {
   mkdir build
 
   cd percona-xtrabackup-${pkgver//_/-}
   patch -p1 -i ../mysql-build-fix.patch
+  patch -p1 -i ../add-riscv-support.patch
 }
 
 build() {
