--- PKGBUILD
+++ PKGBUILD
@@ -10,20 +10,26 @@ arch=('x86_64')
 url='https://www.percona.com/software/mysql-database/percona-xtrabackup'
 license=('GPL')
 depends=('libaio' 'libev' 'libgcrypt' 'curl' 'perl-dbd-mysql' 'numactl' 'protobuf' 'procps-ng')
-makedepends=('cmake' 'python-sphinx' 'zlib' 'vim' 'libedit' 'libevent' 're2')
+makedepends=('cmake' 'python-sphinx' 'zlib' 'vim' 'libedit' 'libevent' 're2' 'clang')
 optdepends=('qpress: for compressed backups')
+options=('!lto')
 source=("https://www.percona.com/downloads/Percona-XtraBackup-LATEST/Percona-XtraBackup-${pkgver//_/-}/source/tarball/percona-xtrabackup-${pkgver//_/-}.tar.gz"
-        "https://boostorg.jfrog.io/artifactory/main/release/$_boost_ver/source/boost_${_boost_ver//./_}.tar.gz")
+        "https://boostorg.jfrog.io/artifactory/main/release/$_boost_ver/source/boost_${_boost_ver//./_}.tar.gz"
+        "add-riscv-support.patch::https://patch-diff.githubusercontent.com/raw/percona/percona-xtrabackup/pull/1443.diff")
 
 sha256sums=('2a1c23497ffd5905d6dc20bdb5a801d1b8baeb3245ec11ed115dee0d78b7a5e2'
-            '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131')
+            '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131'
+            '08ffa60b8370a760d35744919729d66e1714c49322aaca810e9e57ff951c582b')
 
 prepare() {
   mkdir build
+  cd percona-xtrabackup-${pkgver//_/-}
+  patch -Np1 -i ../add-riscv-support.patch
 }
 
 build() {
   cd build
+  export CC=clang CXX=clang++ CFLAGS="${CFLAGS//-fstack-clash-protection/}" CXXFLAGS="${CXXFLAGS//-fstack-clash-protection/}"
   cmake -DBUILD_CONFIG=xtrabackup_release \
         -DCMAKE_BUILD_TYPE=Release \
         -DMYSQL_DATADIR=/var/lib/mysql \
