--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,16 @@ depends=('libaio' 'libev' 'libgcrypt' 'curl' 'perl-dbd-mysql' 'numactl' 'protobu
 makedepends=('cmake' 'python-sphinx' 'vim' 'libedit' 'libevent' 'libfido2')
 optdepends=('qpress: for compressed backups')
 source=("https://www.percona.com/downloads/Percona-XtraBackup-LATEST/Percona-XtraBackup-${pkgver//_/-}/source/tarball/percona-xtrabackup-${pkgver//_/-}.tar.gz"
-        "https://boostorg.jfrog.io/artifactory/main/release/$_boost_ver/source/boost_${_boost_ver//./_}.tar.gz")
+        "https://boostorg.jfrog.io/artifactory/main/release/$_boost_ver/source/boost_${_boost_ver//./_}.tar.gz"
+        "$pkgname-add-riscv-support.patch::https://patch-diff.githubusercontent.com/raw/percona/percona-xtrabackup/pull/1443.diff")
 
 sha256sums=('8a3632a5a7a91834800f4f83902468bccff1d979e82347c0372b39a97b0c85f0'
-            '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131')
+            '5347464af5b14ac54bb945dc68f1dd7c56f0dad7262816b956138fc53bcc0131'
+            '08ffa60b8370a760d35744919729d66e1714c49322aaca810e9e57ff951c582b')
 prepare() {
   mkdir build
+  cd percona-xtrabackup-${pkgver//_/-}
+  patch -p1 -i ../$pkgname-add-riscv-support.patch
 }
 
 build() {
