diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -24,14 +24,16 @@
 url='https://openjdk.java.net/'
 license=('custom')
 makedepends=('java-environment>=10' 'java-environment<12' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
-             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib' 'pandoc'
+             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib'
              'graphviz' 'freetype2' 'libjpeg-turbo' 'giflib' 'libpng' 'lcms2'
              'libnet' 'bash' 'harfbuzz' 'glibc' 'gcc-libs')
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
+        java11-riscv64.patch  # rv64gc support and hotspot jit impl (patch taken from opensuse)
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
         freedesktop-jshell.desktop)
 sha256sums=('119c6233fe7ff5670c590e2f9d6686ac4d80c97b17065506998b75c547b54f2c'
+            '3963927fee540f90a4bd7d2ae5492ccc93a58b5b597f3ed09cd61f0b971865e9'
             '575587ad58dfa9908f046d307b9afc7b0b2eb20a1eb454f8fdbbd539ea7b3d01'
             '2f57b7c7dd671eabe9fa10c4f1283573e99d7f7c36eccd82c95b705979a2e8cb'
             'f271618a8c2a892b554caf26857af41efdf0d8bcb95d57ce7ba535d6979e96da')
@@ -39,6 +41,7 @@
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac

 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -50,6 +53,11 @@
               lib/libjsound.{so,debuginfo}
               lib/libsplashscreen.{so,debuginfo})

+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i ../java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}

