diff --git PKGBUILD PKGBUILD
index 6a4c9f50..794c7273 100644
--- PKGBUILD
+++ PKGBUILD
@@ -32,15 +32,20 @@
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
-        freedesktop-jshell.desktop)
+        freedesktop-jshell.desktop
+
+        # see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+        java11-riscv64.patch)
 sha256sums=('d0bb1719468f2eb7264caa3aa6a104b4015a0cffb1995e280370d9a7de6c0509'
             '575587ad58dfa9908f046d307b9afc7b0b2eb20a1eb454f8fdbbd539ea7b3d01'
             '2f57b7c7dd671eabe9fa10c4f1283573e99d7f7c36eccd82c95b705979a2e8cb'
-            'f271618a8c2a892b554caf26857af41efdf0d8bcb95d57ce7ba535d6979e96da')
+            'f271618a8c2a892b554caf26857af41efdf0d8bcb95d57ce7ba535d6979e96da'
+            'a3adec091d04b56e08208e69904679359b3a5f0761f3cf803a67d7ce86b56207')
 
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -52,6 +57,11 @@
               lib/libjsound.{so,debuginfo}
               lib/libsplashscreen.{so,debuginfo})
 
+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -98,7 +108,7 @@
     --with-lcms=system \
     --with-zlib=system \
     --with-harfbuzz=system \
-    --with-jvm-features=zgc,shenandoahgc \
+    --with-jvm-features=zgc \
     --enable-unlimited-crypto \
     --disable-warnings-as-errors \
     ${NUM_PROC_OPT}
