--- PKGBUILD
+++ PKGBUILD
@@ -67,6 +67,7 @@
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   aarch64)_JARCH='aarch64';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -96,6 +97,11 @@
   nss
 )
 
+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -435,4 +441,8 @@
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+# see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+source+=(java11-riscv64.patch)
+sha256sums+=('244847ec6ea172f54bed88be14ce4e877ea931e78a8e60aa7d96f37fd7b9743e')
+
 # vim: ts=2 sw=2 et:
