--- PKGBUILD
+++ PKGBUILD
@@ -68,6 +68,7 @@
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
   aarch64)_JARCH='aarch64';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -97,6 +98,11 @@
   nss
 )
 
+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -442,4 +448,8 @@
   ln -s ${pkgbase} "${pkgdir}/usr/share/licenses/${pkgname}"
 }
 
+# see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+source+=(java11-riscv64.patch)
+sha256sums+=('862b3b865da51a210d0d1c83be7912e23246c99397b27ea312566be35f510499')
+
 # vim: ts=2 sw=2 et:
