--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,7 @@ sha256sums=('982322667a9e5cb6a2e063bbab3c0bda43907cf9c6099b80888d41499823f1a3'
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -52,6 +53,15 @@ _nonheadless=(lib/libawt_xawt.so
               lib/libjsound.so
               lib/libsplashscreen.so)
 
+# see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+source+=(java11-riscv64.patch)
+sha256sums+=('2eeedc01389f0301639a7daaaa27c90c798030a6a5e41d8c6b2eed6c36fa54de')
+
+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -98,7 +108,7 @@ build() {
     --with-lcms=system \
     --with-zlib=system \
     --with-harfbuzz=system \
-    --with-jvm-features=zgc,shenandoahgc \
+    --with-jvm-features=zgc \
     --with-native-debug-symbols=internal \
     --enable-unlimited-crypto \
     --disable-warnings-as-errors \
