--- PKGBUILD
+++ PKGBUILD
@@ -70,6 +70,7 @@ case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
   aarch64)_JARCH='aarch64';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -101,8 +102,14 @@ _commondeps=(
 
 prepare() {
   patch -Np1 -i ../0001-java11-Build-failure-with-glibc-2.42-due-to-uabs-na.patch -d ${_jdkdir}
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
 }
 
+# see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+source+=(java11-riscv64.patch)
+sha256sums+=('862b3b865da51a210d0d1c83be7912e23246c99397b27ea312566be35f510499')
+
 build() {
   cd ${_jdkdir}
 
