Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 443753)
+++ PKGBUILD	(working copy)
@@ -25,7 +25,7 @@
 url='https://openjdk.java.net/'
 license=('custom')
 makedepends=('java-environment>=10' 'java-environment<12' 'cpio' 'unzip' 'zip' 'libelf' 'libcups' 'libx11'
-             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib' 'pandoc'
+             'libxrender' 'libxtst' 'libxt' 'libxext' 'libxrandr' 'alsa-lib'
              'graphviz' 'freetype2' 'libjpeg-turbo' 'giflib' 'libpng' 'lcms2'
              'libnet' 'bash' 'harfbuzz' 'glibc' 'gcc-libs')
 options=(!lto)
@@ -32,15 +32,20 @@
 source=(https://github.com/openjdk/jdk${_majorver}u/archive/${_git_tag}.tar.gz
         freedesktop-java.desktop
         freedesktop-jconsole.desktop
-        freedesktop-jshell.desktop)
+        freedesktop-jshell.desktop
+
+        # see https://build.opensuse.org/package/show/home:Andreas_Schwab:riscv:java/java-11-openjdk
+        java11-riscv64.patch)
 sha256sums=('a125f0f2e34061ea1509c9f24caec51e6e23411552d6b660a6f60c054853dc63'
             '575587ad58dfa9908f046d307b9afc7b0b2eb20a1eb454f8fdbbd539ea7b3d01'
             '2f57b7c7dd671eabe9fa10c4f1283573e99d7f7c36eccd82c95b705979a2e8cb'
-            'f271618a8c2a892b554caf26857af41efdf0d8bcb95d57ce7ba535d6979e96da')
+            'f271618a8c2a892b554caf26857af41efdf0d8bcb95d57ce7ba535d6979e96da'
+            'a3adec091d04b56e08208e69904679359b3a5f0761f3cf803a67d7ce86b56207')
 
 case "${CARCH}" in
   x86_64) _JARCH='x86_64';;
   i686)   _JARCH='x86';;
+  riscv64)_JARCH='riscv64';;
 esac
 
 _jvmdir=/usr/lib/jvm/java-${_majorver}-openjdk
@@ -52,6 +57,11 @@
               lib/libjsound.{so,debuginfo}
               lib/libsplashscreen.{so,debuginfo})
 
+prepare() {
+  cd "$srcdir/$_jdkdir"
+  patch -Np1 -i "$srcdir"/java11-riscv64.patch
+}
+
 build() {
   cd ${_jdkdir}
 
@@ -104,6 +114,9 @@
     ${NUM_PROC_OPT}
     #--disable-javac-server
 
+  # riscv64: requires P flag on qemu-user, or
+  # export _JAVA_OPTIONS="-Djdk.lang.Process.launchMechanism=vfork"  # or fork
+
   make images legacy-jre-image docs
 
   # https://bugs.openjdk.java.net/browse/JDK-8173610
