--- PKGBUILD
+++ PKGBUILD
@@ -4,7 +4,7 @@ pkgbase=aida-x
 _name="${pkgbase^^}"
 pkgname=(
   aida-x
-  aida-x-{clap,lv2,standalone,vst,vst3}
+  aida-x-{clap,lv2,standalone,vst}
 )
 pkgver=1.1.0
 pkgrel=1
@@ -24,9 +24,12 @@ makedepends=(
 checkdepends=(
   lv2lint
 )
-source=($url/releases/download/$pkgver/$_name-$pkgver-src.tar.xz)
-sha512sums=('6725501a9335bac0829550289509d2ef22bcc8f70fb65ba5ac845c430c4bfef47000e47162427ff0699a28468f1988fdc6cf0bc6a8ffbe7f53daaaead8433478')
-b2sums=('86a4ddd9b37519ca6ed9ad6a36c2916c473e22eacfd724e6c95f301df2e03ca507ae0a247c4e0026478d79a5cd69a7793f52f05e0ebec1b41d7f94a6e8fd8711')
+source=($url/releases/download/$pkgver/$_name-$pkgver-src.tar.xz
+        disable-vst3.patch)
+sha512sums=('6725501a9335bac0829550289509d2ef22bcc8f70fb65ba5ac845c430c4bfef47000e47162427ff0699a28468f1988fdc6cf0bc6a8ffbe7f53daaaead8433478'
+            '1134eb6320dff5e4386ac1797b336bf4c17b2d00dff949e8e775b66e3b24f23e02d31e66e2d6e312f69dbc6c0ba5de9a0506fcb69570ac323a76255a857b7ef5')
+b2sums=('86a4ddd9b37519ca6ed9ad6a36c2916c473e22eacfd724e6c95f301df2e03ca507ae0a247c4e0026478d79a5cd69a7793f52f05e0ebec1b41d7f94a6e8fd8711'
+        '2f1e12730282a24ed3d90f0aafa5d331da27d13cd9960e6130108496d7c8f8a0e8f060984ab6fd33668399de9e41cbe37f506439e28cfa191bbeef1aec39b545')
 
 _pick() {
   local p="$1" f d; shift
@@ -38,6 +41,11 @@ _pick() {
   done
 }
 
+prepare() {
+  cd $_name-$pkgver
+  patch -Np1 -i ../disable-vst3.patch
+}
+
 build() {
   local cmake_options=(
     -B build
@@ -154,23 +162,3 @@ package_aida-x-vst() {
   # mv -v $pkgname/* "$pkgdir"
   install -vDm 755 build/bin/$_name-vst2.so -t "$pkgdir/usr/lib/vst/"
 }
-
-package_aida-x-vst3() {
-  pkgdesc+=" - VST3 plugin"
-  groups=(
-    pro-audio
-    vst3-plugins
-  )
-  depends=(
-    dbus
-    gcc-libs
-    glibc
-    libglvnd
-    libx11
-    libxext
-    vst3-host
-  )
-
-  # mv -v $pkgname/* "$pkgdir"
-  install -vDm 755 build/bin/$_name.vst3/Contents/$CARCH-linux/*.so -t "$pkgdir/usr/lib/vst3/$_name.vst3/Contents/$CARCH-linux/"
-}
