--- PKGBUILD
+++ PKGBUILD
@@ -62,7 +62,7 @@ build() {
 		-D ENABLE_UTILS=ON \
 		-D ENABLE_UTP=ON \
 		-D INSTALL_LIB=ON \
-		-D USE_SYSTEM_B64=ON \
+		-D USE_SYSTEM_B64=OFF \
 		-D USE_SYSTEM_DEFLATE=ON \
 		-D USE_SYSTEM_DHT=ON \
 		-D USE_SYSTEM_EVENT2=ON \
@@ -145,3 +145,5 @@ package_transmission-qt() {
 	_install_component qt
 	install -Dm644 COPYING "$pkgdir/usr/share/licenses/transmission-qt/COPYING"
 }
+
+makedepends=(${makedepends[@]/libb64})
