--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,6 @@ makedepends=(cmake
              gtkmm-4.0
              intltool
              libayatana-indicator
-             libb64
              libdeflate
              libevent
              libnatpmp
@@ -63,7 +62,7 @@ build() {
 		-D ENABLE_UTILS=ON \
 		-D ENABLE_UTP=ON \
 		-D INSTALL_LIB=ON \
-		-D USE_SYSTEM_B64=ON \
+		-D USE_SYSTEM_B64=OFF \
 		-D USE_SYSTEM_DEFLATE=ON \
 		-D USE_SYSTEM_DHT=ON \
 		-D USE_SYSTEM_EVENT2=ON \
