Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 461119)
+++ PKGBUILD	(working copy)
@@ -15,14 +15,18 @@
 source=("git+https://git.kernel.org/pub/scm/linux/kernel/git/jejb/sbsigntools.git#tag=v${pkgver}?signed"
         "git+https://git.ozlabs.org/ccan"
         openssl3.patch
+        0001-add-support-for-risc-v-64-bit-pe-coff-images.patch
         )
 sha256sums=('SKIP'
             'SKIP'
-            '631336ec0687150c80aef32f296b50d322a5c461187c8456d89ae66f32a92718')
+            '631336ec0687150c80aef32f296b50d322a5c461187c8456d89ae66f32a92718'
+            '094d1cd29147a93877750163b96bece754d9edf7a23b69a8bd7d989dbb8f5be8')
 validpgpkeys=('D5606E73C8B46271BEAD9ADF814AE47C214854D6') # James Bottomley <jejb@kernel.org>
 
 prepare() {
   cd ${pkgname}
+  # https://groups.io/g/sbsigntools/message/51
+  patch -Np1 -i ../0001-add-support-for-risc-v-64-bit-pe-coff-images.patch
   patch -Np1 -i ../openssl3.patch
   git submodule init
   git config submodule."lib/ccan.git".url "${srcdir}/ccan"
