Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1013433)
+++ PKGBUILD	(working copy)
@@ -15,6 +15,8 @@
 
 build() {
   cd "piping-server-rust-$pkgver"
+  echo -e "[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
   cargo build --release --locked --target-dir=target
 }
 
