diff --git PKGBUILD PKGBUILD
index a4795f6..6783f2a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,11 @@ makedepends=('ghc' 'haskell-doctest' 'haskell-foldl' 'haskell-tasty' 'haskell-ta
 source=("https://hackage.haskell.org/packages/archive/$pkgname/$pkgver/$pkgname-$pkgver.tar.gz")
 sha512sums=('683c19cb31a2c0d919347f65bf6fd04950a9613db8d61288bf86b5968207a52edf986feb105ac287581f88b9e6ff686ec55abb8f0c25a65cde1c4537422c2e89')
 
+prepare() {
+  cd $pkgname-$pkgver
+  sed -i '/Test-Suite doctest/a \    buildable: False' $pkgname.cabal
+}
+
 build() {
   cd $pkgname-$pkgver
 
