diff --git a/Namcap/tests/package/test_anyelf.py b/Namcap/tests/package/test_anyelf.py
index 029e832..95a6896 100644
--- a/Namcap/tests/package/test_anyelf.py
+++ b/Namcap/tests/package/test_anyelf.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_anyelf
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_dbus1location.py b/Namcap/tests/package/test_dbus1location.py
index 5a46b18..63a7777 100644
--- a/Namcap/tests/package/test_dbus1location.py
+++ b/Namcap/tests/package/test_dbus1location.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_dbus1location
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_elffiles.py b/Namcap/tests/package/test_elffiles.py
index 604cf08..eb616ab 100644
--- a/Namcap/tests/package/test_elffiles.py
+++ b/Namcap/tests/package/test_elffiles.py
@@ -2,6 +2,8 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 import os
+import platform
+from pytest import mark
 
 import Namcap.rules.elffiles
 from Namcap.tests.makepkg import MakepkgTest
@@ -13,7 +15,7 @@ pkgname=__namcap_test_elffiles
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -44,7 +46,7 @@ pkgname=__namcap_test_execstack
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -77,7 +79,7 @@ pkgname=__namcap_test_nopie
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -104,6 +106,7 @@ package() {
         self.assertEqual(r.infos, [])
 
 
+@mark.skipif("x86_64" not in platform.machine(), reason="-fcf-protection currently only works for x86 processors")
 class TestNoShadowStack(MakepkgTest):
     pkgbuild = """
 pkgname=__namcap_test_noshstk
diff --git a/Namcap/tests/package/test_emptydirs.py b/Namcap/tests/package/test_emptydirs.py
index 207844e..c71409a 100644
--- a/Namcap/tests/package/test_emptydirs.py
+++ b/Namcap/tests/package/test_emptydirs.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_emptydirs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_fhs.py b/Namcap/tests/package/test_fhs.py
index f219288..faf6dc7 100644
--- a/Namcap/tests/package/test_fhs.py
+++ b/Namcap/tests/package/test_fhs.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -58,7 +58,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -88,7 +88,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_filenames.py b/Namcap/tests/package/test_filenames.py
index 0c6bc2e..7ebf8d9 100644
--- a/Namcap/tests/package/test_filenames.py
+++ b/Namcap/tests/package/test_filenames.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_nonascii
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_fileownership.py b/Namcap/tests/package/test_fileownership.py
index 2c20a39..59c182d 100644
--- a/Namcap/tests/package/test_fileownership.py
+++ b/Namcap/tests/package/test_fileownership.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_fileownership
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_gnomemime.py b/Namcap/tests/package/test_gnomemime.py
index acb95a3..d37e0ee 100644
--- a/Namcap/tests/package/test_gnomemime.py
+++ b/Namcap/tests/package/test_gnomemime.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_gnomemime
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_hardlinks.py b/Namcap/tests/package/test_hardlinks.py
index a2693de..3ffecb9 100644
--- a/Namcap/tests/package/test_hardlinks.py
+++ b/Namcap/tests/package/test_hardlinks.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_hardlinks
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_hicoloricons.py b/Namcap/tests/package/test_hicoloricons.py
index dadeee1..da1a62a 100644
--- a/Namcap/tests/package/test_hicoloricons.py
+++ b/Namcap/tests/package/test_hicoloricons.py
@@ -14,7 +14,7 @@ pkgname=__namcap_test_hicoloricons
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=()
@@ -55,7 +55,7 @@ pkgname=__namcap_test_hicoloricons
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('hicolor-icon-theme')
diff --git a/Namcap/tests/package/test_infodirectory.py b/Namcap/tests/package/test_infodirectory.py
index 46a6bbd..9b6731d 100644
--- a/Namcap/tests/package/test_infodirectory.py
+++ b/Namcap/tests/package/test_infodirectory.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_infodirectory
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_libtool.py b/Namcap/tests/package/test_libtool.py
index a9fefa3..992310d 100644
--- a/Namcap/tests/package/test_libtool.py
+++ b/Namcap/tests/package/test_libtool.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_libtool
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_licensepkg.py b/Namcap/tests/package/test_licensepkg.py
index 46c941c..b8bb83e 100644
--- a/Namcap/tests/package/test_licensepkg.py
+++ b/Namcap/tests/package/test_licensepkg.py
@@ -240,7 +240,7 @@ class LicenseFileTest(MakepkgTest):
         pkgver=1.0
         pkgrel=1
         pkgdesc="A package"
-        arch=('i686' 'x86_64')
+        arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
         url="http://www.example.com/"
         license=('GPL-3.0-or-later')
         depends=('glibc')
@@ -269,7 +269,7 @@ class LicenseFileTest(MakepkgTest):
             pkgver=1.0
             pkgrel=1
             pkgdesc="A package"
-            arch=('i686' 'x86_64')
+            arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
             url="http://www.example.com/"
             license=('DWTFYWL')
             depends=('glibc')
@@ -303,7 +303,7 @@ class LicenseFileTest(MakepkgTest):
             pkgver=1.0
             pkgrel=1
             pkgdesc="A package"
-            arch=('i686' 'x86_64')
+            arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
             url="http://www.example.com/"
             license=('LicenseRef-DWTFYWL')
             depends=('glibc')
@@ -335,7 +335,7 @@ class LicenseFileTest(MakepkgTest):
             pkgver=1.0
             pkgrel=1
             pkgdesc="A package"
-            arch=('i686' 'x86_64')
+            arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
             url="http://www.example.com/"
             license=('LicenseRef-DWTFYWL')
             depends=('glibc')
@@ -372,7 +372,7 @@ class LicenseFileTest(MakepkgTest):
             pkgver=1.0
             pkgrel=1
             pkgdesc="A package"
-            arch=('i686' 'x86_64')
+            arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
             url="http://www.example.com/"
             license=('GPL-3.0-or-later with Bootloader-exception')
             depends=('glibc')
@@ -410,7 +410,7 @@ class LicenseFileTest(MakepkgTest):
             pkgver=1.0
             pkgrel=1
             pkgdesc="A package"
-            arch=('i686' 'x86_64')
+            arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
             url="http://www.example.com/"
             license=('() broken')
             depends=('glibc')
@@ -447,7 +447,7 @@ class LicenseFileTest(MakepkgTest):
         pkgver=1.0
         pkgrel=1
         pkgdesc="A package"
-        arch=('i686' 'x86_64')
+        arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
         url="http://www.example.com/"
         license=('MIT')
         depends=('glibc')
@@ -484,7 +484,7 @@ class LicenseFileTest(MakepkgTest):
         pkgver=1.0
         pkgrel=1
         pkgdesc="A package"
-        arch=('i686' 'x86_64')
+        arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
         url="http://www.example.com/"
         license=('MIT')
         depends=('glibc')
@@ -518,7 +518,7 @@ class LicenseFileTest(MakepkgTest):
         pkgver=1.0
         pkgrel=1
         pkgdesc="A package"
-        arch=('i686' 'x86_64')
+        arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
         url="http://www.example.com/"
         license=('MIT')
         depends=('licenses')
diff --git a/Namcap/tests/package/test_lotsofdocs.py b/Namcap/tests/package/test_lotsofdocs.py
index 22103d6..1c6983f 100644
--- a/Namcap/tests/package/test_lotsofdocs.py
+++ b/Namcap/tests/package/test_lotsofdocs.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_lotsofdocs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_missingbackups.py b/Namcap/tests/package/test_missingbackups.py
index de713eb..ae9840f 100644
--- a/Namcap/tests/package/test_missingbackups.py
+++ b/Namcap/tests/package/test_missingbackups.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_missingbackups
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_pathdepends.py b/Namcap/tests/package/test_pathdepends.py
index 40eb383..0af2105 100644
--- a/Namcap/tests/package/test_pathdepends.py
+++ b/Namcap/tests/package/test_pathdepends.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_pathdepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_perllocal.py b/Namcap/tests/package/test_perllocal.py
index 84bbac9..26a7727 100644
--- a/Namcap/tests/package/test_perllocal.py
+++ b/Namcap/tests/package/test_perllocal.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_perllocal
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_permissions.py b/Namcap/tests/package/test_permissions.py
index 9cd10b0..4be0b1b 100644
--- a/Namcap/tests/package/test_permissions.py
+++ b/Namcap/tests/package/test_permissions.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_permissions
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_rpath.py b/Namcap/tests/package/test_rpath.py
index 10412b9..54727f7 100644
--- a/Namcap/tests/package/test_rpath.py
+++ b/Namcap/tests/package/test_rpath.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_rpath
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_scrollkeeper.py b/Namcap/tests/package/test_scrollkeeper.py
index 6abc54e..7d3c78c 100644
--- a/Namcap/tests/package/test_scrollkeeper.py
+++ b/Namcap/tests/package/test_scrollkeeper.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_scrollkeeper
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_sodepends.py b/Namcap/tests/package/test_sodepends.py
index a478251..605ac02 100644
--- a/Namcap/tests/package/test_sodepends.py
+++ b/Namcap/tests/package/test_sodepends.py
@@ -26,7 +26,7 @@ pkgname=__namcap_test_sodepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_sphinxbuildcachefiles.py b/Namcap/tests/package/test_sphinxbuildcachefiles.py
index 5166387..e660713 100644
--- a/Namcap/tests/package/test_sphinxbuildcachefiles.py
+++ b/Namcap/tests/package/test_sphinxbuildcachefiles.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_sphinxbuildcachefiles
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_symlink.py b/Namcap/tests/package/test_symlink.py
index 40d961c..bd1ff6b 100644
--- a/Namcap/tests/package/test_symlink.py
+++ b/Namcap/tests/package/test_symlink.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_symlink
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_systemdlocation.py b/Namcap/tests/package/test_systemdlocation.py
index f7a9914..6d063ae 100644
--- a/Namcap/tests/package/test_systemdlocation.py
+++ b/Namcap/tests/package/test_systemdlocation.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_systemdlocation
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_unusedsodepends.py b/Namcap/tests/package/test_unusedsodepends.py
index abadf59..200e5d3 100644
--- a/Namcap/tests/package/test_unusedsodepends.py
+++ b/Namcap/tests/package/test_unusedsodepends.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_unusedsodepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_arrays.py b/Namcap/tests/pkgbuild/test_arrays.py
index 18897bd..5209dba 100644
--- a/Namcap/tests/pkgbuild/test_arrays.py
+++ b/Namcap/tests/pkgbuild/test_arrays.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license='GPL-3.0-or-later'
 depends='glibc'
diff --git a/Namcap/tests/pkgbuild/test_badbackups.py b/Namcap/tests/pkgbuild/test_badbackups.py
index 425dd82..02fc75e 100644
--- a/Namcap/tests/pkgbuild/test_badbackups.py
+++ b/Namcap/tests/pkgbuild/test_badbackups.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_extravars.py b/Namcap/tests/pkgbuild/test_extravars.py
index 6aaae0e..59c92d8 100644
--- a/Namcap/tests/pkgbuild/test_extravars.py
+++ b/Namcap/tests/pkgbuild/test_extravars.py
@@ -16,7 +16,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license='GPL-3.0-or-later'
 depends=('glibc')
@@ -52,7 +52,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A pac
 kage=1 package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -61,6 +61,7 @@ glibc=2.12
 )
 depends_i686=('zlib')
 depends_x86_64=('lib32-zlib')
+depends_riscv64=('lib32-zlib')
 optdepends=('gtk2')
 options=('!libtool')
 source=(ftp://ftp.example.com/pub/mypackage-0.1.tar.gz)
diff --git a/Namcap/tests/pkgbuild/test_invalidstartdir.py b/Namcap/tests/pkgbuild/test_invalidstartdir.py
index 031726d..aa72158 100644
--- a/Namcap/tests/pkgbuild/test_invalidstartdir.py
+++ b/Namcap/tests/pkgbuild/test_invalidstartdir.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="This program does foobar"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_makedepends.py b/Namcap/tests/pkgbuild/test_makedepends.py
index d87b306..30dfa09 100644
--- a/Namcap/tests/pkgbuild/test_makedepends.py
+++ b/Namcap/tests/pkgbuild/test_makedepends.py
@@ -15,7 +15,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://www.example.com/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('lib1' 'lib2' 'lib3')
 makedepends=('lib1' 'lib2' 'lib4')
 license=('GPL-3.0-or-later')
@@ -54,7 +54,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://www.example.com/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=()
 makedepends=()
 license=('GPL-3.0-or-later')
diff --git a/Namcap/tests/pkgbuild/test_makepkgfunctions.py b/Namcap/tests/pkgbuild/test_makepkgfunctions.py
index 2e2565e..236297c 100644
--- a/Namcap/tests/pkgbuild/test_makepkgfunctions.py
+++ b/Namcap/tests/pkgbuild/test_makepkgfunctions.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_missingvars.py b/Namcap/tests/pkgbuild/test_missingvars.py
index ea3b34b..5171334 100644
--- a/Namcap/tests/pkgbuild/test_missingvars.py
+++ b/Namcap/tests/pkgbuild/test_missingvars.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -38,7 +38,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -65,7 +65,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc' 'pacman')
@@ -92,7 +92,7 @@ pkgver=24.0.1312.5
 pkgrel=1
 pkgdesc="PDF Viewer for Chromium"
 url="http://wiki.archlinux.org/index.php/Chromium#libpdf.so"
-arch=("i686" "x86_64")
+arch=("i686" "x86_64" "riscv64")
 license=('custom:chrome')
 makedepends=('lynx')
 depends=('chromium')
@@ -102,6 +102,9 @@ if [ "$CARCH" = "i686" ]; then
 elif [ "$CARCH" = "x86_64" ]; then
 	_arch="amd64"
 	sha512sums=('1989976e6ca93099c92957e44948d7b73a6f9594304b9fe5313ff44c28c5261852919f2b5dacb49724c77d98667cc3991f8aaef9d3d9b2277eb375293c4a0395')
+elif [ "$CARCH" = "riscv64" ]; then
+	_arch="riscv64"
+	sha512sums=('abcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef12345678901234567890123456789012345678901234567890123456789012345678901234567890')
 fi
 source=(https://dl.google.com/linux/direct/google-chrome-unstable_current_$_arch.deb)
 
@@ -167,7 +170,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="The foobar program"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -210,7 +213,7 @@ pkgbase=base
 pkgname=(foo bar)
 pkgver=1.0
 pkgrel=1
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 license=('GPL-3.0-or-later')
 url="https://archlinux.org"
@@ -245,7 +248,7 @@ package_bar() {
 pkgname=foo
 pkgver=1.0
 pkgrel=1
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 license=('GPL-3.0-or-later')
 url="https://archlinux.org"
@@ -272,7 +275,7 @@ package() {
 pkgname=foo
 pkgver=1.0
 pkgrel=1
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 pkgdesc='this is the package foo'
 license=('GPL-3.0-or-later')
diff --git a/Namcap/tests/pkgbuild/test_pkginfo.py b/Namcap/tests/pkgbuild/test_pkginfo.py
index 82670f0..f3c5e9e 100644
--- a/Namcap/tests/pkgbuild/test_pkginfo.py
+++ b/Namcap/tests/pkgbuild/test_pkginfo.py
@@ -14,7 +14,7 @@ pkgname=MyPackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
@@ -57,7 +57,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 depends=('glibc')
 options=('!libtool')
@@ -98,7 +98,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 license=('GPL-3.0-or-later')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild/test_pkgnameindesc.py b/Namcap/tests/pkgbuild/test_pkgnameindesc.py
index 15109bc..bc7dc05 100644
--- a/Namcap/tests/pkgbuild/test_pkgnameindesc.py
+++ b/Namcap/tests/pkgbuild/test_pkgnameindesc.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="The mypackage program does foobar"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_sfurl.py b/Namcap/tests/pkgbuild/test_sfurl.py
index 0725092..1bec51b 100644
--- a/Namcap/tests/pkgbuild/test_sfurl.py
+++ b/Namcap/tests/pkgbuild/test_sfurl.py
@@ -15,7 +15,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 license=('GPL-3.0-or-later')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild/test_splitpkgbuild.py b/Namcap/tests/pkgbuild/test_splitpkgbuild.py
index 00d4a0d..fd49617 100644
--- a/Namcap/tests/pkgbuild/test_splitpkgbuild.py
+++ b/Namcap/tests/pkgbuild/test_splitpkgbuild.py
@@ -16,7 +16,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 depends=('glibc')
 license=('GPL-3.0-or-later')
 options=('!libtool')
@@ -60,7 +60,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 makedepends=('gtk2')
 license=('GPL-3.0-or-later')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild_test.py b/Namcap/tests/pkgbuild_test.py
index 6d84e62..8ac049e 100644
--- a/Namcap/tests/pkgbuild_test.py
+++ b/Namcap/tests/pkgbuild_test.py
@@ -71,7 +71,7 @@ pkgver=1.0
 pkgrel=1
 epoch=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc' 'pacman')
@@ -100,7 +100,7 @@ pkgname=('mypackage1' 'mypackage2' 'mypackage3')
 pkgver=1.0
 pkgrel=1
 epoch=2
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 makedepends=('make' 'python')
diff --git a/Namcap/tests/test_pacman.py b/Namcap/tests/test_pacman.py
index ebede93..29195c8 100644
--- a/Namcap/tests/test_pacman.py
+++ b/Namcap/tests/test_pacman.py
@@ -16,7 +16,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'aarch64' 'loong64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL-3.0-or-later')
 depends=('glibc' 'foobar')
