diff --git a/Namcap/tests/package/test_anyelf.py b/Namcap/tests/package/test_anyelf.py
index 97c7ad7..937feae 100644
--- a/Namcap/tests/package/test_anyelf.py
+++ b/Namcap/tests/package/test_anyelf.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_anyelf
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_elffiles.py b/Namcap/tests/package/test_elffiles.py
index 07e98ac..95f31f1 100644
--- a/Namcap/tests/package/test_elffiles.py
+++ b/Namcap/tests/package/test_elffiles.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_elffiles
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -43,7 +43,7 @@ pkgname=__namcap_test_execstack
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -76,7 +76,7 @@ pkgname=__namcap_test_nopie
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_emptydirs.py b/Namcap/tests/package/test_emptydirs.py
index adb8f00..adb8b60 100644
--- a/Namcap/tests/package/test_emptydirs.py
+++ b/Namcap/tests/package/test_emptydirs.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_emptydirs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_fhs.py b/Namcap/tests/package/test_fhs.py
index de88734..7a7d075 100644
--- a/Namcap/tests/package/test_fhs.py
+++ b/Namcap/tests/package/test_fhs.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -57,7 +57,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -87,7 +87,7 @@ pkgname=__namcap_test_nonfhs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_filenames.py b/Namcap/tests/package/test_filenames.py
index 377179f..1a66cc3 100644
--- a/Namcap/tests/package/test_filenames.py
+++ b/Namcap/tests/package/test_filenames.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_nonascii
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_fileownership.py b/Namcap/tests/package/test_fileownership.py
index acd989b..f9c7d5e 100644
--- a/Namcap/tests/package/test_fileownership.py
+++ b/Namcap/tests/package/test_fileownership.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_fileownership
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_gnomemime.py b/Namcap/tests/package/test_gnomemime.py
index a057b28..ac352f6 100644
--- a/Namcap/tests/package/test_gnomemime.py
+++ b/Namcap/tests/package/test_gnomemime.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_gnomemime
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_hardlinks.py b/Namcap/tests/package/test_hardlinks.py
index b334dbe..c0bcfd9 100644
--- a/Namcap/tests/package/test_hardlinks.py
+++ b/Namcap/tests/package/test_hardlinks.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_hardlinks
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_hicoloricons.py b/Namcap/tests/package/test_hicoloricons.py
index 9c9cee6..40cffa4 100644
--- a/Namcap/tests/package/test_hicoloricons.py
+++ b/Namcap/tests/package/test_hicoloricons.py
@@ -13,7 +13,7 @@ pkgname=__namcap_test_hicoloricons
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=()
@@ -54,7 +54,7 @@ pkgname=__namcap_test_hicoloricons
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('hicolor-icon-theme')
diff --git a/Namcap/tests/package/test_infodirectory.py b/Namcap/tests/package/test_infodirectory.py
index 5bc08b2..aacfe59 100644
--- a/Namcap/tests/package/test_infodirectory.py
+++ b/Namcap/tests/package/test_infodirectory.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_infodirectory
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_libtool.py b/Namcap/tests/package/test_libtool.py
index ac8c476..616a704 100644
--- a/Namcap/tests/package/test_libtool.py
+++ b/Namcap/tests/package/test_libtool.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_libtool
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_licensepkg.py b/Namcap/tests/package/test_licensepkg.py
index dc2ba62..a843946 100644
--- a/Namcap/tests/package/test_licensepkg.py
+++ b/Namcap/tests/package/test_licensepkg.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_licensepkg
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('custom:DWTFYWL')
 depends=('glibc')
@@ -43,7 +43,7 @@ pkgname=__namcap_test_licensepkg
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('CCPL:cc-by-sa-3.0')
 depends=('glibc')
@@ -73,7 +73,7 @@ pkgname=__namcap_test_licensepkg
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('DWTFYWL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_lotsofdocs.py b/Namcap/tests/package/test_lotsofdocs.py
index ec6b2dd..5a9bc9b 100644
--- a/Namcap/tests/package/test_lotsofdocs.py
+++ b/Namcap/tests/package/test_lotsofdocs.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_lotsofdocs
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_missingbackups.py b/Namcap/tests/package/test_missingbackups.py
index c85dbf3..a26a863 100644
--- a/Namcap/tests/package/test_missingbackups.py
+++ b/Namcap/tests/package/test_missingbackups.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_missingbackups
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_pathdepends.py b/Namcap/tests/package/test_pathdepends.py
index 9976291..b9cab58 100644
--- a/Namcap/tests/package/test_pathdepends.py
+++ b/Namcap/tests/package/test_pathdepends.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_pathdepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_perllocal.py b/Namcap/tests/package/test_perllocal.py
index 830cfc1..1640652 100644
--- a/Namcap/tests/package/test_perllocal.py
+++ b/Namcap/tests/package/test_perllocal.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_perllocal
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_permissions.py b/Namcap/tests/package/test_permissions.py
index 13ab5f0..9e4b033 100644
--- a/Namcap/tests/package/test_permissions.py
+++ b/Namcap/tests/package/test_permissions.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_permissions
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_rpath.py b/Namcap/tests/package/test_rpath.py
index a617a45..2d27285 100644
--- a/Namcap/tests/package/test_rpath.py
+++ b/Namcap/tests/package/test_rpath.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_rpath
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_scrollkeeper.py b/Namcap/tests/package/test_scrollkeeper.py
index 3d6bc9a..90b1633 100644
--- a/Namcap/tests/package/test_scrollkeeper.py
+++ b/Namcap/tests/package/test_scrollkeeper.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_scrollkeeper
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_sodepends.py b/Namcap/tests/package/test_sodepends.py
index 186f082..f5fdce5 100644
--- a/Namcap/tests/package/test_sodepends.py
+++ b/Namcap/tests/package/test_sodepends.py
@@ -25,7 +25,7 @@ pkgname=__namcap_test_sodepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_symlink.py b/Namcap/tests/package/test_symlink.py
index 14aacce..f030204 100644
--- a/Namcap/tests/package/test_symlink.py
+++ b/Namcap/tests/package/test_symlink.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_symlink
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_systemdlocation.py b/Namcap/tests/package/test_systemdlocation.py
index 51263d8..2521aaa 100644
--- a/Namcap/tests/package/test_systemdlocation.py
+++ b/Namcap/tests/package/test_systemdlocation.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_systemdlocation
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/package/test_unusedsodepends.py b/Namcap/tests/package/test_unusedsodepends.py
index fb9960e..45b2a0e 100644
--- a/Namcap/tests/package/test_unusedsodepends.py
+++ b/Namcap/tests/package/test_unusedsodepends.py
@@ -12,7 +12,7 @@ pkgname=__namcap_test_unusedsodepends
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_arrays.py b/Namcap/tests/pkgbuild/test_arrays.py
index 8c9a622..667a546 100644
--- a/Namcap/tests/pkgbuild/test_arrays.py
+++ b/Namcap/tests/pkgbuild/test_arrays.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license='GPL'
 depends='glibc'
diff --git a/Namcap/tests/pkgbuild/test_badbackups.py b/Namcap/tests/pkgbuild/test_badbackups.py
index 29302e1..977ab2a 100644
--- a/Namcap/tests/pkgbuild/test_badbackups.py
+++ b/Namcap/tests/pkgbuild/test_badbackups.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_extravars.py b/Namcap/tests/pkgbuild/test_extravars.py
index f0d4ddc..b994d7b 100644
--- a/Namcap/tests/pkgbuild/test_extravars.py
+++ b/Namcap/tests/pkgbuild/test_extravars.py
@@ -15,7 +15,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license='GPL'
 depends=('glibc')
@@ -51,7 +51,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A pac
 kage=1 package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -60,6 +60,7 @@ glibc=2.12
 )
 depends_i686=('zlib')
 depends_x86_64=('lib32-zlib')
+depends_riscv64=('lib32-zlib')
 optdepends=('gtk2')
 options=('!libtool')
 source=(ftp://ftp.example.com/pub/mypackage-0.1.tar.gz)
diff --git a/Namcap/tests/pkgbuild/test_invalidstartdir.py b/Namcap/tests/pkgbuild/test_invalidstartdir.py
index 591ec5a..75afbc2 100644
--- a/Namcap/tests/pkgbuild/test_invalidstartdir.py
+++ b/Namcap/tests/pkgbuild/test_invalidstartdir.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="This program does foobar"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_makedepends.py b/Namcap/tests/pkgbuild/test_makedepends.py
index 2578074..d2e3b43 100644
--- a/Namcap/tests/pkgbuild/test_makedepends.py
+++ b/Namcap/tests/pkgbuild/test_makedepends.py
@@ -15,7 +15,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://www.example.com/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 depends=('lib1' 'lib2' 'lib3')
 makedepends=('lib1' 'lib2' 'lib4')
 license=('GPL')
@@ -54,7 +54,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://www.example.com/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 depends=()
 makedepends=()
 license=('GPL')
diff --git a/Namcap/tests/pkgbuild/test_makepkgfunctions.py b/Namcap/tests/pkgbuild/test_makepkgfunctions.py
index f18f692..4255cb0 100644
--- a/Namcap/tests/pkgbuild/test_makepkgfunctions.py
+++ b/Namcap/tests/pkgbuild/test_makepkgfunctions.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_missingvars.py b/Namcap/tests/pkgbuild/test_missingvars.py
index c176dce..b8e3e7b 100644
--- a/Namcap/tests/pkgbuild/test_missingvars.py
+++ b/Namcap/tests/pkgbuild/test_missingvars.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -38,7 +38,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -65,7 +65,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc' 'pacman')
@@ -92,7 +92,7 @@ pkgver=24.0.1312.5
 pkgrel=1
 pkgdesc="PDF Viewer for Chromium"
 url="http://wiki.archlinux.org/index.php/Chromium#libpdf.so"
-arch=("i686" "x86_64")
+arch=("i686" "x86_64" "riscv64")
 license=('custom:chrome')
 makedepends=('lynx')
 depends=('chromium')
@@ -102,6 +102,9 @@ if [ "$CARCH" = "i686" ]; then
 elif [ "$CARCH" = "x86_64" ]; then
 	_arch="amd64"
 	sha512sums=('1989976e6ca93099c92957e44948d7b73a6f9594304b9fe5313ff44c28c5261852919f2b5dacb49724c77d98667cc3991f8aaef9d3d9b2277eb375293c4a0395')
+elif [ "$CARCH" = "riscv64" ]; then
+	_arch="riscv64"
+	sha512sums=('abcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef12345678901234567890123456789012345678901234567890123456789012345678901234567890')
 fi
 source=(https://dl.google.com/linux/direct/google-chrome-unstable_current_$_arch.deb)
 
@@ -167,7 +170,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="The foobar program"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_pkginfo.py b/Namcap/tests/pkgbuild/test_pkginfo.py
index 234949d..aea0f01 100644
--- a/Namcap/tests/pkgbuild/test_pkginfo.py
+++ b/Namcap/tests/pkgbuild/test_pkginfo.py
@@ -14,7 +14,7 @@ pkgname=MyPackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
@@ -57,7 +57,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 depends=('glibc')
 options=('!libtool')
@@ -98,7 +98,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 depends=('glibc')
 license=('GPL')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild/test_pkgnameindesc.py b/Namcap/tests/pkgbuild/test_pkgnameindesc.py
index 98b4605..ce0f2d4 100644
--- a/Namcap/tests/pkgbuild/test_pkgnameindesc.py
+++ b/Namcap/tests/pkgbuild/test_pkgnameindesc.py
@@ -14,7 +14,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="The mypackage program does foobar"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc')
diff --git a/Namcap/tests/pkgbuild/test_sfurl.py b/Namcap/tests/pkgbuild/test_sfurl.py
index 08bd6c6..97737c1 100644
--- a/Namcap/tests/pkgbuild/test_sfurl.py
+++ b/Namcap/tests/pkgbuild/test_sfurl.py
@@ -15,7 +15,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 depends=('glibc')
 license=('GPL')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild/test_splitpkgbuild.py b/Namcap/tests/pkgbuild/test_splitpkgbuild.py
index f5d3b7e..2202cfc 100644
--- a/Namcap/tests/pkgbuild/test_splitpkgbuild.py
+++ b/Namcap/tests/pkgbuild/test_splitpkgbuild.py
@@ -16,7 +16,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 depends=('glibc')
 license=('GPL')
 options=('!libtool')
@@ -60,7 +60,7 @@ pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
 url="http://mypackage.sf.net/"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 makedepends=('gtk2')
 license=('GPL')
 options=('!libtool')
diff --git a/Namcap/tests/pkgbuild_test.py b/Namcap/tests/pkgbuild_test.py
index ea7fece..6bb4f2b 100644
--- a/Namcap/tests/pkgbuild_test.py
+++ b/Namcap/tests/pkgbuild_test.py
@@ -68,7 +68,7 @@ pkgver=1.0
 pkgrel=1
 epoch=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc' 'pacman')
@@ -97,7 +97,7 @@ pkgname=('mypackage1' 'mypackage2' 'mypackage3')
 pkgver=1.0
 pkgrel=1
 epoch=2
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 makedepends=('make' 'python')
diff --git a/Namcap/tests/test_pacman.py b/Namcap/tests/test_pacman.py
index 8a2eb81..90b27b6 100644
--- a/Namcap/tests/test_pacman.py
+++ b/Namcap/tests/test_pacman.py
@@ -16,7 +16,7 @@ pkgname=mypackage
 pkgver=1.0
 pkgrel=1
 pkgdesc="A package"
-arch=('i686' 'x86_64')
+arch=('i686' 'x86_64' 'riscv64')
 url="http://www.example.com/"
 license=('GPL')
 depends=('glibc' 'foobar')
