--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,17 @@ license=('GPL')
 depends=('python' 'pyalpm' 'licenses' 'binutils' 'elfutils' 'python-pyelftools' 'pkgconf')
 checkdepends=('systemd' 'python-pytest' 'python-six')
 makedepends=('python-build' 'python-installer' 'python-setuptools' 'python-wheel')
-source=(https://gitlab.archlinux.org/pacman/namcap/-/archive/${pkgver}/namcap-${pkgver}.tar.bz2)
-sha512sums=('ab4d1658dbdf192d1d91e3cf3cd3e4cab0be6021561e6d14ad86603b95f6261eb97bd3087a003ce493b3dea7a240d8fdddc29f059613b3ff3fb718467cd8078e')
+source=(https://gitlab.archlinux.org/pacman/namcap/-/archive/${pkgver}/namcap-${pkgver}.tar.bz2
+        support-other-arches.patch
+        fix-riscv64-test.patch)
+sha512sums=('ab4d1658dbdf192d1d91e3cf3cd3e4cab0be6021561e6d14ad86603b95f6261eb97bd3087a003ce493b3dea7a240d8fdddc29f059613b3ff3fb718467cd8078e'
+            '552eb498a734bda37b6d596f8fca61e433ccb5d00b93b9015e996afb1715258d16fe7daf4f3737abd4638502a9ef63c645c620ed3c208533fe1a722dbd509603'
+            '7352ff5ce60845a074c1657feb5013f2a45489066636492e0f65bb63a95868b5cd859f838715d5691290f49b88600919484ccca2e939dcfa49b933ee5290f925')
 
 prepare() {
   cd ${pkgname}-${pkgver}
+  patch -Np1 -i "${srcdir}/support-other-arches.patch"
+  patch -Np1 -i "${srcdir}/fix-riscv64-test.patch"
 }
 
 build() {
