diff --git PKGBUILD PKGBUILD
index 4a08fc5..92df1ba 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,11 +22,21 @@ checkdepends=(python-pytest
 makedepends=(python-{build,installer,wheel}
             python-setuptools
     	    git)
-source=("git+https://gitlab.archlinux.org/pacman/namcap.git#tag=${pkgver}")
-sha256sums=('SKIP')
+source=("git+https://gitlab.archlinux.org/pacman/namcap.git#tag=${pkgver}"
+        "support-other-arches.patch"
+        "fix-riscv64-test.patch")
+sha256sums=('a611105de27d4260351bc6fe86d992511a6a78b8d53cacce8837ed660dcb949b'
+            'a3e7babe2fde4604aea83dc1e9f2148d445aa83cc0e91d72b3ec4329cd9e2885'
+            '373f5aed12b28e215aeae900001f39efbd32e8ed0a6da0e8cb5bfa7c8748ea28')
 validpgpkeys=(9F377DDB6D3153A48EB3EB1E63CC496475267693  # caleb@alerque.com
               CCB34EBBB9541EF3F7B366C1D4A753468A5A5B67) # alerque@archlinux.org
 
+prepare() {
+  cd namcap
+  patch -Np1 -i ../support-other-arches.patch
+  patch -Np1 -i ../fix-riscv64-test.patch
+}
+
 build() {
   cd namcap
   python -m build -wn
