--- PKGBUILD
+++ PKGBUILD
@@ -22,12 +22,22 @@ checkdepends=(python-pytest
 makedepends=(python-{build,installer,wheel}
             python-setuptools)
 _archive="$pkgname-$pkgver"
-source=("$url/-/releases/$pkgver/downloads/$_archive.tar.bz2"{,.asc})
+source=("$url/-/releases/$pkgver/downloads/$_archive.tar.bz2"{,.asc}
+        "support-other-arches.patch"
+        "fix-riscv64-test.patch")
 sha256sums=('2da8f2dc267dc9be053e4c5719a4eda24eb0227ae5f0387089b392cf01bd1d80'
-            'SKIP')
+            'SKIP'
+            '3807c7ce59a50efb7c9a3189409364f0ce78d8b07c558e4b9b5b715e94b8564d'
+            'fa3904f9320b4d897cf40c3551dca16a123f2bf6ab005caf0dd50e42ce62e11b')
 validpgpkeys=(9F377DDB6D3153A48EB3EB1E63CC496475267693  # caleb@alerque.com
               CCB34EBBB9541EF3F7B366C1D4A753468A5A5B67) # alerque@archlinux.org
 
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../support-other-arches.patch
+  patch -Np1 -i ../fix-riscv64-test.patch
+}
+
 build() {
   cd "$_archive"
   python -m build -wn
