--- PKGBUILD
+++ PKGBUILD
@@ -23,12 +23,22 @@ checkdepends=(python-pytest
 makedepends=(python-{build,installer,wheel}
             python-setuptools)
 _archive="$pkgname-$pkgver"
-source=("$url/-/releases/$pkgver/downloads/$_archive.tar.bz2"{,.asc})
+source=("$url/-/releases/$pkgver/downloads/$_archive.tar.bz2"{,.asc}
+        "support-other-arches.patch"
+        "fix-riscv64-test.patch")
 sha256sums=('fbd3b1f0777fe457afd3dbb1f55de8adbaeb50257492626bcffd1a3eef67d618'
-            'SKIP')
+            'SKIP'
+            '3807c7ce59a50efb7c9a3189409364f0ce78d8b07c558e4b9b5b715e94b8564d'
+            'bcb50b7b7f4145c3aeba80c8aca4229b5ceb43d6b31b97c5a71178254b809dd2')
 validpgpkeys=(9F377DDB6D3153A48EB3EB1E63CC496475267693  # caleb@alerque.com
               CCB34EBBB9541EF3F7B366C1D4A753468A5A5B67) # alerque@archlinux.org
 
+prepare() {
+  cd $pkgname-$pkgver
+  patch -Np1 -i ../support-other-arches.patch
+  patch -Np1 -i ../fix-riscv64-test.patch
+}
+
 build() {
   cd "$_archive"
   python -m build -wn
