--- PKGBUILD
+++ PKGBUILD
@@ -19,8 +19,17 @@ makedepends=(
 
 source=(
   "$pkgname-$pkgver.tar.gz::https://github.com/simdjson/simdjson/archive/v$pkgver.tar.gz"
+  rvv-check.diff
+  rvv-check-single-header.diff
 )
-sha256sums=('04f7e0381eab0eaad68d031b9438e7843e2f6b138151c1a9fd0fe4cc4bab3250')
+sha256sums=('04f7e0381eab0eaad68d031b9438e7843e2f6b138151c1a9fd0fe4cc4bab3250'
+            '0faaaafcc813e2fc0715188fa8e7b4be728ed38de422ddee8aeb7cdcd88a98db'
+            'a90bce2c8a6474a90e0900de7387712e77f8b28cf86cc85449335573bc9d6770')
+
+prepare() {
+   patch -Np1 -d "$pkgname-$pkgver" < rvv-check.diff
+   patch -Np0 -d "$pkgname-$pkgver" < rvv-check-single-header.diff
+}
 
 build() {
   cmake -B build -S "$pkgname-$pkgver" \
