diff -r -u Vulkan-Docs-1.2.175-orig/scripts/translate_math.js Vulkan-Docs-1.2.175/scripts/translate_math.js
--- Vulkan-Docs-1.2.175-orig/scripts/translate_math.js	2021-04-13 16:35:11.000000000 +0800
+++ Vulkan-Docs-1.2.175/scripts/translate_math.js	2021-10-04 13:33:02.796405362 +0800
@@ -5,10 +5,11 @@
 // Translates the latexmath in html on build time using KaTeX
 // Usage: nodejs translate_math.js katex/katex.min.js vkspec.html
 
-const katex = require(process.argv[2]);
-const fs = require("fs");
-const escapeRegex = require("escape-string-regexp");
-const he = require('he');
+(async () => {
+const katex = (await import(process.argv[2])).default;
+const fs = (await import("fs")).default;
+const escapeRegex = (await import("escape-string-regexp")).default;
+const he = (await import('he')).default;
 
 const filepath = process.argv[3];
 
@@ -31,3 +32,4 @@
 }
 
 fs.writeFileSync(filepath, html, 'utf8');
+})();
