--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,16 @@ url="https://www.khronos.org/vulkan/"
 license=('custom')
 makedepends=(asciidoctor python3 dblatex source-highlight cmake ghostscript nodejs npm git)
 groups=(vulkan-devel)
-source=("https://github.com/KhronosGroup/${_pkgname}/archive/v${pkgver}.tar.gz")
-sha256sums=('c7fd5d0ff12eb795c2dde959dbfc0b12394341aedc45f28d457196eda22bdc4b')
+source=("https://github.com/KhronosGroup/${_pkgname}/archive/v${pkgver}.tar.gz"
+        "escape-string-regexp-esm.patch")
+sha256sums=('c7fd5d0ff12eb795c2dde959dbfc0b12394341aedc45f28d457196eda22bdc4b'
+            'b81b5436f83ae566fd805f26653b8ace650822900e64c9fceae66a9f35b8e874')
 
 prepare() {
   npm install escape-string-regexp
   npm install he
+
+  patch -Np0 < escape-string-regexp-esm.patch
 }
 
 build() {
