--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,16 @@ depends=('deepin-qt5integration' 'dtkcore' 'dtkgui' 'dtkwidget' 'qt5-base' 'qt5-
          'gcc-libs' 'deepin-picker' 'adobe-source-han-sans-otc-fonts')
 makedepends=('cmake' 'qt5-tools')
 groups=('deepin-extra')
-source=("https://github.com/linuxdeepin/deepin-draw/archive/$pkgver/$pkgname-$pkgver.tar.gz")
-sha512sums=('e48ea2a8168932ced31dce925a2f040b58f03ffd2abf2b723060383af49d57e4e4202694e4456523a32c606082ec721c2d9735841f8c67bbc8b18894654c7785')
+source=("https://github.com/linuxdeepin/deepin-draw/archive/$pkgver/$pkgname-$pkgver.tar.gz"
+        "fix-build.patch::https://github.com/hack3ric/deepin-draw/commit/81044d3d104e0c86b59ae4db3f986efc46762e4a.diff")
+sha512sums=('e48ea2a8168932ced31dce925a2f040b58f03ffd2abf2b723060383af49d57e4e4202694e4456523a32c606082ec721c2d9735841f8c67bbc8b18894654c7785'
+            '46c4ebe4f494f26d55a4514d83ce1bcba3d2d112be2a8c23a32d38f55b0d9fa479562be06531d15b85bcc97993d29cfb6036815e522cf73839dfebc109640a48')
 
 prepare() {
   cd deepin-draw-$pkgver
   sed -e '/set(CMAKE_CXX_FLAGS/d' -e '/set(CMAKE_C_FLAGS/d' -e '/SET(CMAKE_EXE_LINKER_FLAGS/d' \
       -i src/CMakeLists.txt
+  patch -Np1 -i ../fix-build.patch
 }
 
 build() {
