--- PKGBUILD
+++ PKGBUILD
@@ -49,3 +49,16 @@
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+source+=("$pkgname-clr-riscv-hard-float.patch"
+         "$pkgname-clr-riscv-fence.patch"
+         "$pkgname-clr-riscv-currentStackPtr.patch")
+sha256sums+=('88c1531459b25e7c8903ef13c81e6f1ec72fcce5dfeac5af657db3fc2f200086'
+             '33c026a400d7258a7ded2de53dc1282f7edcde6935340d8fb59b0c7758781db2'
+             '667d8cbddb62b2f85c855967b97f9bac5bf720a1d08018002608a740488a0bc7')
+
+prepare() {
+  patch -Np1 -d "$srcdir/$_dir_name" -i "$srcdir/$pkgname-clr-riscv-hard-float.patch"
+  patch -Np1 -d "$srcdir/$_dir_name" -i "$srcdir/$pkgname-clr-riscv-fence.patch"
+  patch -Np1 -d "$srcdir/$_dir_name" -i "$srcdir/$pkgname-clr-riscv-currentStackPtr.patch"
+}
