--- PKGBUILD
+++ PKGBUILD
@@ -40,3 +40,17 @@ package() {
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+source+=("$pkgname-clr-riscv-hard-float.patch"
+         "$pkgname-clr-riscv-fence.patch"
+         "$pkgname-clr-riscv-currentStackPtr.patch::https://github.com/ROCm/clr/pull/117.diff")
+sha256sums+=(
+            'b8d9643df110fd016796fe5e3ffda0b13bfe6ba430304322684a691bc35b84ff'
+            '6f289613f91b54c67dcd7e9bc45e77c3a6645ad4bc0da1ad0a57c873aaaeb997'
+            'df7a04af6cc8fd9145f50d56e0d6b4c4eb78a569227af1f6106b61bc322ca4c7')
+
+prepare() {
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-hard-float.patch"
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-fence.patch"
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-currentStackPtr.patch"
+}
