diff --git PKGBUILD PKGBUILD
index 5b66968..49a70da 100644
--- PKGBUILD
+++ PKGBUILD
@@ -41,3 +41,10 @@
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+source+=("$pkgname-only-enable-SIMD-on-x86.patch::https://github.com/ROCm-Developer-Tools/clr/commit/4c3450cccd8f4064b1e011e6b84e7ca4567121a3.diff")
+sha256sums+=('b20e8678851331093bf99d9d202b221f5ffd144e9ceb153728434e750df5349e')
+
+prepare() {
+  patch -Np1 -d $_dirname -i ../$pkgname-only-enable-SIMD-on-x86.patch
+}
