diff --git PKGBUILD PKGBUILD
index 9f649e8..64e2fa5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,3 +40,13 @@
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+source+=("$pkgname-clr-riscv-hard-float.patch"
+         "$pkgname-clr-riscv-fence.patch")
+sha256sums+=('b8d9643df110fd016796fe5e3ffda0b13bfe6ba430304322684a691bc35b84ff'
+             '1ddd118ec6f2f285adca0bd2511d7c3cff8c9cbe66d84572228957f995e9525f')
+
+prepare() {
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-hard-float.patch"
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-fence.patch"
+}
