diff --git PKGBUILD PKGBUILD
index 603fc1f..26bec0a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -39,3 +39,16 @@ package() {
     echo '/opt/rocm/lib/libamdocl64.so' > 'amdocl64.icd'
     install -Dm644 'amdocl64.icd' "$pkgdir/etc/OpenCL/vendors/amdocl64.icd"
 }
+
+source+=("$pkgname-clr-riscv-hard-float.patch"
+         "$pkgname-clr-riscv-fence.patch"
+         "$pkgname-clr-riscv-currentStackPtr.patch::https://github.com/ROCm/clr/pull/117.diff")
+sha256sums+=('7a8cfbdca38394381c1894183758852137cf72909573d926bc144aa7c20d7ddd'
+             '6f289613f91b54c67dcd7e9bc45e77c3a6645ad4bc0da1ad0a57c873aaaeb997'
+             '933ddbc531a652a3ada6322df6eb0d8adcedc214ca01bd882fc82c704c69cdc8')
+
+prepare() {
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-hard-float.patch"
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-fence.patch"
+  patch -Np1 -d "$srcdir/$_dirname" -i "$srcdir/$pkgname-clr-riscv-currentStackPtr.patch"
+}
