--- a/data/tests/fwupd_test.py
+++ b/data/tests/fwupd_test.py
@@ -153,7 +153,7 @@
         )
 
         # wait until the daemon gets online
-        wait_time = 60 if "valgrind" in daemon_path[0] else 5
+        wait_time = 60
         self.assert_eventually(
             lambda: self.proxy and self.proxy.get_name_owner(),
             timeout=wait_time * 1000,
--- a/data/tests/meson.build
+++ b/data/tests/meson.build
@@ -156,6 +156,7 @@
       python3,
       args: [fwupd_mockdev_tests, ut],
       is_parallel: false,
+      timeout: 120,
       env: {
         'CACHE_DIRECTORY': join_paths(meson.project_build_root(), 'cache'),
         'DAEMON_BUILDDIR': join_paths(meson.project_build_root(), 'src'),
