--- a/data/tests/fwupd_test.py
+++ b/data/tests/fwupd_test.py
@@ -153,7 +153,7 @@
         )
 
         # wait until the daemon gets online
-        wait_time = 60 if "valgrind" in daemon_path[0] else 5
+        wait_time = 60 if "valgrind" in daemon_path[0] else 15
         self.assert_eventually(
             lambda: self.proxy and self.proxy.get_name_owner(),
             timeout=wait_time * 1000,
