--- PKGBUILD
+++ PKGBUILD
@@ -29,7 +29,6 @@ depends=(
   libpassim.so
   libprotobuf-c.so
   libqmi-glib.so
-  libsmbios
   libxmlb
   polkit
   python
@@ -81,7 +80,9 @@ build() {
   arch-meson ${pkgname}-${pkgver} build \
     -D b_lto=false \
     -D docs=enabled \
-    -D plugin_intel_spi=true \
+    -D plugin_intel_spi=false \
+    -D plugin_msr=disabled \
+    -D plugin_dell=disabled \
     -D supported_build=enabled \
     -D efi_binary=false \
     -D systemd_unit_user=fwupd
@@ -102,7 +103,4 @@ package() {
   rm -r "${pkgdir}"/usr/share/installed-tests/
   mv "${pkgdir}"/usr/bin/{,fwupd-}dbxtool
   mv "${pkgdir}"/usr/share/man/man1/{,fwupd-}dbxtool.1
-  # Remove msr module-load config as it is built-in
-  rm "${pkgdir}"/usr/lib/modules-load.d/fwupd-msr.conf
-  rmdir "${pkgdir}"/usr/lib/modules-load.d
 }
