--- PKGBUILD
+++ PKGBUILD
@@ -81,6 +81,7 @@ build() {
     -D docs=enabled
     -D efi_binary=false
     -D launchd=disabled
+    -D plugin_msr=disabled
     -D supported_build=enabled
     -D systemd_unit_user=fwupd
   )
@@ -136,9 +137,6 @@ package_fwupd() {
   # Conflicts with the dbxtool package
   mv "${pkgdir}"/usr/bin/{,fwupd-}dbxtool
   mv "${pkgdir}"/usr/share/man/man1/{,fwupd-}dbxtool.1
-  # Remove msr module-load config as it is built-in
-  rm "${pkgdir}"/usr/lib/modules-load.d/fwupd-msr.conf
-  rmdir "${pkgdir}"/usr/lib/modules-load.d
 
   _pick docs "${pkgdir}"/usr/share/doc/{,fwupd/}{libfwupdplugin,libfwupd}
 }
