--- PKGBUILD
+++ PKGBUILD
@@ -27,7 +27,6 @@ depends=(
   json-glib
   libmm-glib
   libqmi
-  libsmbios
   libxmlb
   passim
   polkit
@@ -70,7 +69,8 @@ build() {
     -D docs=enabled \
     -D plugin_amdgpu=disabled \
     -D launchd=disabled \
-    -D plugin_intel_spi=true \
+    -D plugin_intel_spi=false \
+    -D plugin_msr=disabled \
     -D supported_build=enabled \
     -D efi_binary=false \
     -D systemd_unit_user=fwupd
@@ -123,9 +123,6 @@ package_fwupd() {
   rm -r "${pkgdir}"/usr/share/installed-tests/
   mv "${pkgdir}"/usr/bin/{,fwupd-}dbxtool
   mv "${pkgdir}"/usr/share/man/man1/{,fwupd-}dbxtool.1
-  # Remove msr module-load config as it is built-in
-  rm "${pkgdir}"/usr/lib/modules-load.d/fwupd-msr.conf
-  rmdir "${pkgdir}"/usr/lib/modules-load.d
 
   _pick docs "${pkgdir}"/usr/share/doc/{,fwupd/}{libfwupdplugin,libfwupd}
 }
