--- PKGBUILD
+++ PKGBUILD
@@ -75,6 +75,13 @@
         'b9e1c1c66452adb29378cbb42bd18152a2760750696644ee8dd4c09c6f23116d3e66fd405fc69c4694a5ea0f47b75d5959356a12da56ab1cc8345fe926098b3b')
 validpgpkeys=(163EB50119225DB3DF8F49EA17ACBA8DFA970E17) # Richard Hughes <richard@hughsie.com>
 
+prepare() {
+  cd ${pkgbase}
+
+  # Allow time for initial hwdata indexing and key generation on riscv64.
+  patch -Np1 -i ../increase-daemon-start-timeout.patch
+}
+
 build() {
   local meson_options=(
     -D docs=enabled
@@ -149,3 +156,7 @@
   depends=()
   mv docs/* "${pkgdir}"
 }
+
+source+=(increase-daemon-start-timeout.patch)
+sha512sums+=('fbedb18265cdf76bf71abaef2c8434f12dab2c034b830ea8b2c02e6940f674aa371d23d946ef641eb8e0611815c64a5511075d5b156d088e0edb517d656dea55')
+b2sums+=('009d14d894df5afb161793fd229a34fd2364867d986770984ed969d9c62e28ccc90108a9702ce736395e6f58fcffef5926524ad05ab2d2495f8a999576537cba')
