diff -r -u rocksdb-6.28.2-orig/utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h rocksdb-6.28.2/utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h
--- rocksdb-6.28.2-orig/utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h	2022-02-20 11:41:40.763464088 +0100
+++ rocksdb-6.28.2/utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h	2022-02-20 11:43:35.093774434 +0100
@@ -137,6 +137,10 @@
   uint64_t result;
   asm volatile("stckf %0" : "=Q"(result) : : "cc");
   return result;
+#elif defined(__riscv) && __riscv_xlen == 64
+  uint64_t cycles;
+  asm volatile("rdcycle %0" : "=r"(cycles));
+  return cycles;
 #else
 #error No timer implementation for this platform
 #endif
