--- PKGBUILD
+++ PKGBUILD
@@ -29,6 +29,7 @@
 
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
+  sed -i 's/python3 -m build$/& --no-isolation/' Makefile.am
   ./autogen.sh
 }
 
@@ -90,3 +91,6 @@
   # Fix warning with nonexistent jemalloc, see #3.
   sed -i "/Environment=LD_PRELOAD/s/^/#/" "${pkgdir}/usr/lib/systemd/system/dirsrv@.service.d/custom.conf"
 }
+
+depends+=('python-distro' 'python-psutil')
+makedepends+=('python-setuptools')
