diff --git PKGBUILD PKGBUILD
index 014b41e..74d6791 100644
--- PKGBUILD
+++ PKGBUILD
@@ -22,15 +22,18 @@
              'npm' 'nodejs' 'systemd')
 source=(https://github.com/389ds/389-ds-base/archive/389-ds-base-${pkgver}.tar.gz
         389-ds-base.sysusers
-        389-ds-base.tmpfiles)
+        389-ds-base.tmpfiles
+        fix-borrow-double-ref.patch::https://github.com/389ds/389-ds-base/commit/96959cf7b67be8b544efa25b6ad813c0034841b7.diff)
 sha512sums=('382ab009b170d7c63999068d058a04a9dd949f9b2949eac4745ae14c13e6fa0b1a4733b85c353b7a80f1df72d29dd6262c502514df1f6a78da6aea5a361c811c'
             '8689b60f13517e6541e1faa62e61781654a8e4678990d14a3fe0cfe4673697cdde1c46763fdc7ef6203e4026286fa004c73b2857fd7fecff69483e6d1e8e6d68'
-            '69ed8b8f3bdbf9098088b0c92c41a238f16d14ba9f86ebc2b5debe5f001b4d8e235f7cff4731d72b30b5ac70486b0f4300b99646aa3926a3fa59515a64f16402')
+            '69ed8b8f3bdbf9098088b0c92c41a238f16d14ba9f86ebc2b5debe5f001b4d8e235f7cff4731d72b30b5ac70486b0f4300b99646aa3926a3fa59515a64f16402'
+            '2ca58ad9a5b3cb3df4bf7839ee146552efe50a16176d5a34de2e3fc13fd6cc94936e49a41546001832233b660de2b029457b51684bb18838746d4735b786ea4f')
 
 prepare() {
   cd "${pkgname}-${pkgname}-${pkgver}"
 
   ./autogen.sh
+  patch -Np1 -i ../fix-borrow-double-ref.patch
 }
 
 build() {
