--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ _bento4_version="1.6.0-641"
 source=("$pkgname-$pkgver.tar.gz::https://github.com/peak3d/inputstream.adaptive/archive/$pkgver-$_codename.tar.gz"
         "$pkgname-bento4-$_bento4_version.tar.gz::https://github.com/xbmc/Bento4/archive/refs/tags/$_bento4_version-$_codename.tar.gz")
 noextract=("$pkgname-bento4-$_bento4_version.tar.gz")
-sha512sums=('9e1535fcb9e749cbe9fbe367c802c6edd8c91983530edf4643bf60f2b97796b618e6c959f0b25e686e29d11800d858c3c6610daaff2a1f4991a975693eb3a22e'
+sha512sums=('2dc3dd139f5afd3356e29c3dd47528755d784f7112f39011b37303c03d50f34ae8aa63ac04d8a7d1526f7415c0950e9fa11a077d63481c209fdf689ce5baa53f'
             '6d08b8e028cc5a73fa634e42cb4c0e89cc4a7a700177df7ef8e9be18c02077baae25e5a0f1b8d9cb4ceb2a6ff3108e41a4a07aae5f64d9372e915efce0fbd861')
 
 #prepare() {
