diff --git PKGBUILD PKGBUILD
index 1115224b3..2ba528e7b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ url="http://gcc-libs.sourceforge.net/"
 license=('LGPL')
 depends=('glibc')
 provides=('htmlcxx' 'libhtmlcxx' 'libcss_parser')
-source=("https://sourceforge.net/projects/$pkgname/files/v$pkgver/$pkgname-$pkgver.tar.gz")
-sha256sums=('5d38f938cf4df9a298a5346af27195fffabfef9f460fc2a02233cbcfa8fc75c8')
+source=("https://sourceforge.net/projects/$pkgname/files/v$pkgver/$pkgname-$pkgver.tar.gz"
+        "https://sourceforge.net/p/htmlcxx/patches/8/attachment/htmlcxx-0.87-c++17.patch")
+sha256sums=('5d38f938cf4df9a298a5346af27195fffabfef9f460fc2a02233cbcfa8fc75c8'
+            'dedadc4893486c4ac607b6c80e089b1c7552d8b4700fcf2ee4d36566d9c74ca5')
+
+prepare(){
+  cd $pkgname-$pkgver
+  patch -Np1 -i "${srcdir}/htmlcxx-0.87-c++17.patch"
+}
 
 build() {
   cd $pkgname-$pkgver
