--- a/harness/cases/5.t
+++ b/harness/cases/5.t
@@ -40,12 +40,13 @@
 	buf = mmap(0, page_size, PROT_WRITE, MAP_SHARED, rwfd, 0);
 	assert(buf != (char *)-1);
 
+	status |= attempt_rw(rwfd, buf, SIZE,  0,  READ, SIZE);
+
 	/* Whether PROT_WRITE is readable is arch-dependent.  So compare
 	 * against write() result (to make the kernel read from buf). */
 	res = write(rwfd, buf, SIZE);
 	if (res < 0)
 		res = -errno;
-	status |= attempt_rw(rwfd, buf, SIZE,  0,  READ, SIZE);
 	status |= attempt_rw(rwfd, buf, SIZE,  0, WRITE, res);
 
 	return status;
