Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391938)
+++ PKGBUILD	(working copy)
@@ -15,7 +15,7 @@
 build() {
   # libaio is a thin wrapper around kernel syscalls, it does not use stdlib and
   # other helpers like stack protection libraries
-  CFLAGS="-march=${CARCH/_/-} -mtune=generic -O2 -pipe"
+  CFLAGS="-march=rv64gc -mabi=lp64d -O2 -pipe"
   make -C $pkgname-$pkgname-$pkgver
 }
 
