--- PKGBUILD
+++ PKGBUILD
@@ -19,4 +19,6 @@
 
 prepare() {
+  patch -Np1 -d "$pkgname" -i "$srcdir/prot-write-test.patch"
+
   # -Werror, not even once
   sed 's/-Werror//' -i $pkgname/harness/Makefile
@@ -41,2 +43,5 @@
 }
 
+source+=(prot-write-test.patch)
+sha512sums+=('83cbb617c7d7c341952a7b58561025f42dbee201058d9931f30e1d83184361f59ebba46cd4f2fd955f9688e9c87f2a1c7ba6c67cad535a850582b7fc874393a5')
+b2sums+=('6f7c5d25bc8e2500df1ce2d44110bd0ff1c3405e9421d41a4ff663efb9f65b2c8cc0f6f6a87741655fd0810a18bb6864976cf581765f52e719263b0df8fdf31f')
