diff --git PKGBUILD PKGBUILD
index d163c24..59c409a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -30,6 +30,8 @@
 source=(
   "$_srcname::git+https://github.com/archlinux/linux?signed#tag=$_srctag"
   config  # the main kernel config file
+  riscv64.config-patch
+  force-disable-sv57.patch::https://git.launchpad.net/~ubuntu-kernel/ubuntu/+source/linux/+git/kinetic/patch/?id=f38611903ae7808ab1b7427809f478fbe0d20917
 )
 validpgpkeys=(
   ABAF11C65A2970B130ABE3C479BE3E4300411886  # Linus Torvalds
@@ -38,7 +40,9 @@
   C7E7849466FE2358343588377258734B41C31549  # David Runge <dvzrv@archlinux.org>
 )
 b2sums=('SKIP'
-        '1537bd27339315c0a9e663ae422d3aa7d8cf25bdd5b1bad494d2b8da3d67e2ce72ccbb36e0b1f213b28d2f598957b111b6638c44334bec10ff7d420e3a6597dc')
+        '1537bd27339315c0a9e663ae422d3aa7d8cf25bdd5b1bad494d2b8da3d67e2ce72ccbb36e0b1f213b28d2f598957b111b6638c44334bec10ff7d420e3a6597dc'
+        '6b6abdda8e7390675ab6a433684d6d663ca9ef677f1dde52a71fea26e90cd3159750926b488a1c9a699b25da766b422142b1073b661acd2d5eba1d3d2f0a3e65'
+        'c259d46d3a394cf7074b7cf41a2b21cfaf2eed2506fda614ce51e8105d1cc5324206e5677f2748a6b1cc447035ebee0460d9ac19afc4a4f8a242fa88237f6c02')
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -73,6 +77,12 @@
   _make olddefconfig
   diff -u ../config .config || :
 
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  _make mrproper
+  mv .config.1 .config
+  _make olddefconfig
+
   echo "Prepared $pkgbase version $(<version)"
 }
 
@@ -118,6 +128,9 @@
   ZSTD_CLEVEL=19 _make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  _make INSTALL_DTBS_PATH="$pkgdir/usr/share/dtbs/$kernver" dtbs_install
+
   # remove build and source links
   rm "$modulesdir"/{source,build}
 }
@@ -133,19 +146,16 @@
   install -Dt "$builddir" -m644 .config Makefile Module.symvers System.map \
     localversion.* version vmlinux
   install -Dt "$builddir/kernel" -m644 kernel/Makefile
-  install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  install -Dt "$builddir/arch/riscv" -m644 arch/riscv/Makefile
   cp -t "$builddir" -a scripts
 
-  # required when STACK_VALIDATION is enabled
-  install -Dt "$builddir/tools/objtool" tools/objtool/objtool
-
   # required when DEBUG_INFO_BTF_MODULES is enabled
   install -Dt "$builddir/tools/bpf/resolve_btfids" tools/bpf/resolve_btfids/resolve_btfids
 
   echo "Installing headers..."
   cp -t "$builddir" -a include
-  cp -t "$builddir/arch/x86" -a arch/x86/include
-  install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
+  cp -t "$builddir/arch/riscv" -a arch/riscv/include
+  install -Dt "$builddir/arch/riscv/kernel" -m644 arch/riscv/kernel/asm-offsets.s
 
   install -Dt "$builddir/drivers/md" -m644 drivers/md/*.h
   install -Dt "$builddir/net/mac80211" -m644 net/mac80211/*.h
@@ -167,7 +177,7 @@
   echo "Removing unneeded architectures..."
   local arch
   for arch in "$builddir"/arch/*/; do
-    [[ $arch = */x86/ ]] && continue
+    [[ $arch = */riscv/ ]] && continue
     echo "Removing $(basename "$arch")"
     rm -r "$arch"
   done
