diff --git PKGBUILD PKGBUILD
index e914107..f3c223c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -38,6 +38,7 @@ source=(
   https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
   $url/releases/download/$_srctag/linux-$_srctag.patch.zst{,.sig}
   config  # the main kernel config file
+  riscv64.config-patch
 )
 validpgpkeys=(
   ABAF11C65A2970B130ABE3C479BE3E4300411886  # Linus Torvalds
@@ -49,12 +50,14 @@ sha256sums=('de0053cb413d408f20fd1d5788219951b8a4403e5dac1b1d9032090938acd199'
             'SKIP'
             '366f8c26f21cb5f6a16c163aecbaff9e232dd454fbd5db5a2ee7bff90e51dd34'
             'SKIP'
-            '9d02a908be9f0a408af5166c245509763864b0a037e7c5b14ba3a546ca597fe9')
+            '9d02a908be9f0a408af5166c245509763864b0a037e7c5b14ba3a546ca597fe9'
+            '553d07423779cd1c0549edcf2524fee5ab3f03ba5aae4426bf382b3e8bf34f14')
 b2sums=('5a216c56c77efaee1a3f5c9198ade9180e4640ffcde39662ccf85c2a5945a08c5f362220fb0906369c72a3ea8bdc16fdd24d3e1dbc0f51fc831f3f724ed73300'
         'SKIP'
         '5d915626940caff2be5b70958455a5d7e63f1c31f919302a74bdc21b90c8962433980ef6287d3cfe2b0d125ff8bcfd1bfcba8a7140a53789b2ef326120607f83'
         'SKIP'
-        '910bb62064169ff2f8fab0cdf3b19c2881f55bccafccb650ea03771db04239e434fc49729a9baab1472e1bb036daff897bfdee36b298e93476a11461ba4450e1')
+        '910bb62064169ff2f8fab0cdf3b19c2881f55bccafccb650ea03771db04239e434fc49729a9baab1472e1bb036daff897bfdee36b298e93476a11461ba4450e1'
+        '560ceebc84359322e7e82cc5e0092e8856aa6cfb3da6a271ccaee8f6304ba5c0a125eb34bc83ea34fd4c0beb5b797f30ef3dad6d2d563378c506e30f7d91da28')
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -82,13 +85,20 @@ prepare() {
   make olddefconfig
   diff -u ../config .config || :
 
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  make mrproper
+  mv .config.1 .config
+  make olddefconfig
+
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
 }
 
 build() {
   cd $_srcname
-  make all
+  # Generate symbols for dtbs to support overlays
+  make DTC_FLAGS="-@" all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
   make htmldocs
 }
@@ -130,6 +140,9 @@ _package() {
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  make INSTALL_DTBS_PATH="$pkgdir/usr/lib/modules/$(<version)/dtb" dtbs_install
+
   # remove build link
   rm "$modulesdir"/build
 }
@@ -145,20 +158,17 @@ _package-headers() {
   install -Dt "$builddir" -m644 .config Makefile Module.symvers System.map \
     localversion.* version vmlinux tools/bpf/bpftool/vmlinux.h
   install -Dt "$builddir/kernel" -m644 kernel/Makefile
-  install -Dt "$builddir/arch/x86" -m644 arch/x86/Makefile
+  install -Dt "$builddir/arch/riscv" -m644 arch/riscv/Makefile
   cp -t "$builddir" -a scripts
   ln -srt "$builddir" "$builddir/scripts/gdb/vmlinux-gdb.py"
 
-  # required when STACK_VALIDATION is enabled
-  install -Dt "$builddir/tools/objtool" tools/objtool/objtool
-
   # required when DEBUG_INFO_BTF_MODULES is enabled
   install -Dt "$builddir/tools/bpf/resolve_btfids" tools/bpf/resolve_btfids/resolve_btfids
 
   echo "Installing headers..."
   cp -t "$builddir" -a include
-  cp -t "$builddir/arch/x86" -a arch/x86/include
-  install -Dt "$builddir/arch/x86/kernel" -m644 arch/x86/kernel/asm-offsets.s
+  cp -t "$builddir/arch/riscv" -a arch/riscv/include
+  install -Dt "$builddir/arch/riscv/kernel" -m644 arch/riscv/kernel/asm-offsets.s
 
   install -Dt "$builddir/drivers/md" -m644 drivers/md/*.h
   install -Dt "$builddir/net/mac80211" -m644 net/mac80211/*.h
@@ -180,7 +190,7 @@ _package-headers() {
   echo "Removing unneeded architectures..."
   local arch
   for arch in "$builddir"/arch/*/; do
-    [[ $arch = */x86/ ]] && continue
+    [[ $arch = */riscv/ ]] && continue
     echo "Removing $(basename "$arch")"
     rm -r "$arch"
   done
