--- PKGBUILD
+++ PKGBUILD
@@ -46,24 +46,57 @@
 source=(
   https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
   $url/releases/download/$_srctag/linux-$_srctag.patch.zst{,.sig}
+  # sg2042 write_once
+  0000-sg2042-write_once.patch::https://github.com/revyos/linux/commit/8b001d81419df1c78c58411a51dfd0475233d319.patch?full_index=1
+  # drm/ttm: support device w/o coherency
+  0100-ttm-coherency.patch::https://github.com/revyos/linux/commit/bcfe733064fbe0cb6dab76a73eabf100e73801f9.patch?full_index=1
+  0101-ttm-coherency2.patch::https://github.com/revyos/linux/commit/d5dad38f150b1e153b16cc610e9104bd1ef634b4.patch?full_index=1
+  # rust + gcc
+  0400-rust.patch::https://github.com/revyos/linux/commit/a0c1f2370d02c471f93578a7ce61cae18896acac.patch?full_index=1
+  0401-rust1.patch::https://github.com/revyos/linux/commit/edc216123add9833e039b1c741f2c2b2bde4487d.patch?full_index=1
+  0402-rust2.patch::https://github.com/revyos/linux/commit/71c8e98962c82e04b2e70c50a894032366651b89.patch?full_index=1
+  0403-rust3.patch::https://github.com/revyos/linux/commit/b60775af94fda606995ef848678c5e812951fcf7.patch?full_index=1
 )
 source_x86_64=(config.x86_64)
+source_riscv64=(
+  config.x86_64
+  riscv64.config-patch
+)
 validpgpkeys=(
   ABAF11C65A2970B130ABE3C479BE3E4300411886  # Linus Torvalds
   647F28654894E3BD457199BE38DBBDC86092693E  # Greg Kroah-Hartman
   83BC8889351B5DEBBB68416EB8AC08600F108CDF  # Jan Alexander Steffens (heftig)
 )
+sha256sums=('01710ee01737dac492f1bae52becd057e08d20d11589089aa06accff415c28dd'
+            'SKIP'
+            '07b0526c8d8b9bae9dacf1a90aa77a92de8fca505ea35d3b523f28556361fc16'
+            'SKIP'
+            '5f1f62f1082a6299cf6456f12287613aa9ebfc5be8291181258e9690b94c0fb1'
+            'de1eaddab3130e9503c417489e793cd83a77f8a6453fb182667bdcfcd801b8ba'
+            'f062bd6a6bb7e987617a0e576a03420673b062293d948b91c94eb3b87b15276b'
+            '6fb42f5c26b5ef6e18c26720cb31c848de8efae4b4b236702685a2f02d53e793'
+            'b91d25d18796baa11544a61aced57d490b40e3808198595d9cf0c0062163d24a'
+            '35021cf475645d77515f1a09ae04bead566fcb8eb4d06d8f19a370b1ec1b4f07'
+            '37f05363fff17aa29edf7cb6bd026d25bb7a4c20a7f91aaec30aa09cf269fc8e')
+sha256sums_x86_64=('ea21308436ee9e93924516e6cd1287c72fc01554a56b568bc0a0a94ab8afc6ce')
+sha256sums_riscv64=('ea21308436ee9e93924516e6cd1287c72fc01554a56b568bc0a0a94ab8afc6ce'
+                    '2fecaa50107a7ce9fd407e968ee328ea886b095a720b976c91df250d077c794b')
 b2sums=('1dc0bee4d040846ac31672400339c179daf68919b2d923261205febbb4ff654d1ac21a4bc700f5bd46a7676335692565678b8142549d83c1d0a0da3b9db2e3d4'
         'SKIP'
         '911acab50d12d1b81980ae5b639d222e6c2b4471c08b2eb0bf6221036ca435b18a836aac8855834acc56a0a6b01eb60087775df4f83f33d060410327a4786f30'
-        'SKIP')
+        'SKIP'
+        '6748c004b274b187c3f3a8c0ab9548fd09dc4c962a7e59f74c4ce0181fe1dc1ff8bf0adb07688e4bb8776e21d63aa06b1494338ac6a9fcb2db5df881ef38135c'
+        '552afc1c296991adb14c46e56a96d40a01688aeb1e12f2e294bd18b9316ceebd527a16b4ab08a82c0aa0a4cd13517f6a2f4534c25627d375b96958dfac66f034'
+        '22cd5b8697894d8d4515f89e4053c05f99a6cb6de59b23aabc3f98fc4f80b4b1ad075399925ac67c81bbd5da5f18384d09171f778c8f4335357798965cafe6f1'
+        'a450e64f3ce9551e4c40fcce47d0054722511dbda2dd7132dcd1c8a7c5a21ebab46433ced3f030fb0ab84c58688208cc70a3b0c227cf2015d9a243abe47c61b4'
+        '36cdf842c02090ac7631d2fac6bc86960cdd51cb03d2b263e0f9a9aabbca118e4cff5ca12af20ea12460f413840b891e1d04119f63c3af59ad68a7b4e0ef4d83'
+        '75d3f910b61c852dcb4c3d4ade69dc0746e2e29c790fca31c38ba06afc5858f42ed6aed4f352a594892ff06555df598607f7a9693060dfa70947fe8ad4af8459'
+        '4027ad446a6da4b94f15a9928c8a7345022a647c240baefb61b523d90ba60a03e3ca7db9045aaa2b5a8fa91d37752852fd7f32c211d81ae1d1854159eb203fe6')
 b2sums_x86_64=('60a991bded2a41a9b3880445f5cefe3f19e593b2204a007a423d8a5f2858bbc7d3babab02ffebd9ff8fd37f3a1a32b3ca2512cbe768f5cb154deffacfbfee28a')
+b2sums_riscv64=('60a991bded2a41a9b3880445f5cefe3f19e593b2204a007a423d8a5f2858bbc7d3babab02ffebd9ff8fd37f3a1a32b3ca2512cbe768f5cb154deffacfbfee28a'
+                '7ca4077e60b28a9cb68627e21c46fad576c0dfcd1b4b6d91c34448f3ae24624d3f27c0b08bf7776146ad230b19181ed65d6be1f4a190049812ed1c7813105773')
 
 # https://www.kernel.org/pub/linux/kernel/v7.x/sha256sums.asc
-sha256sums=('01710ee01737dac492f1bae52becd057e08d20d11589089aa06accff415c28dd'
-            'SKIP'
-            '07b0526c8d8b9bae9dacf1a90aa77a92de8fca505ea35d3b523f28556361fc16'
-            'SKIP')
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -87,9 +120,14 @@
   done
 
   echo "Setting config..."
-  cp ../config.$CARCH .config
+  cp ../config.x86_64 .config
+  make olddefconfig
+
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  make mrproper
+  mv .config.1 .config
   make olddefconfig
-  diff -u ../config.$CARCH .config || :
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -101,7 +139,8 @@
   make htmldocs SPHINXOPTS=-QT &
   local pid_docs=$!
 
-  make all
+  # Generate symbols for dtbs to support overlays
+  make DTC_FLAGS="-@" all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
   wait $pid_docs
 }
@@ -145,6 +184,9 @@
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  make INSTALL_DTBS_PATH="$pkgdir/usr/lib/modules/$(<version)/dtb" dtbs_install
+
   # remove build link
   rm "$modulesdir"/build
 }
@@ -170,6 +212,7 @@
   local karch
   case $CARCH in
     x86_64) karch=x86 ;;
+    riscv64) karch=riscv ;;
     *) echo "Unknown CARCH $CARCH"; exit 1 ;;
   esac
 
