diff --git PKGBUILD PKGBUILD
index 4ecc97d..f01811e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -46,6 +46,16 @@ _srctag=v${pkgver%.*}-${pkgver##*.}
 source=(
   https://cdn.kernel.org/pub/linux/kernel/v${pkgver%%.*}.x/${_srcname}.tar.{xz,sign}
   $url/releases/download/$_srctag/linux-$_srctag.patch.zst{,.sig}
+  # sg2042 write_once
+  0000-sg2042-write_once.patch::https://github.com/revyos/linux/commit/8b001d81419df1c78c58411a51dfd0475233d319.patch?full_index=1
+  # drm/ttm: support device w/o coherency
+  0100-ttm-coherency.patch::https://github.com/revyos/linux/commit/4360470added82f87c49c75464c0126241f7455e.patch?full_index=1
+  0101-ttm-coherency2.patch::https://github.com/revyos/linux/commit/328c79fd1daefcadce1672ebee6bb88810664b06.patch?full_index=1
+  # rust + gcc
+  0400-rust.patch::https://github.com/revyos/linux/commit/62db2801ca6383d228342b4a78b842bb04614560.patch?full_index=1
+  0401-rust1.patch::https://github.com/revyos/linux/commit/e5dbd930828f455c0161a72a198c73a3c5b06972.patch?full_index=1
+  0402-rust2.patch::https://github.com/revyos/linux/commit/09539b5244d8a1f87d83ac5f53c0d3e5746902f7.patch?full_index=1
+  0403-rust3.patch::https://github.com/revyos/linux/commit/5545a0d3db17b6858d2311fa29d284963c846698.patch?full_index=1
 )
 source_x86_64=(config.x86_64)
 validpgpkeys=(
@@ -53,17 +63,32 @@ validpgpkeys=(
   647F28654894E3BD457199BE38DBBDC86092693E  # Greg Kroah-Hartman
   83BC8889351B5DEBBB68416EB8AC08600F108CDF  # Jan Alexander Steffens (heftig)
 )
-b2sums=('2c53f205a940b0f9f68653b92ef46d49f828cbef3cfa8cf94d050c8e6df05c4fcaa4f9b9681b9130b14e3c790d31208eb244d123249a93e35e8e6165f3d858c9'
-        'SKIP'
-        '26230d1a111b24fe9239273acdfacda37c5bf009f861c448ad25392dcca433514246d629a077ce5c66478c7e0f4e5477ce5f95c91d08b3a02cc87bb35b849bcf'
-        'SKIP')
-b2sums_x86_64=('7082013345352c95303ee87cd78bf5d93ab49ec9f270e6cb803a05cb7f9a67c554bbd260de922d6d44145fd3712b410c13d67c8f76dc2b9f4088be86aeaec835')
-
-# https://www.kernel.org/pub/linux/kernel/v7.x/sha256sums.asc
 sha256sums=('57edc9a41efc1ca6b797afa8f4a587a30da2af6bca7356eb56e1e1a4ada265da'
             'SKIP'
             'ce38af1268931b099993cf01c537d6c3b21007e08cad84d2f1e71f95cc5cb75b'
-            'SKIP')
+            'SKIP'
+            '5f1f62f1082a6299cf6456f12287613aa9ebfc5be8291181258e9690b94c0fb1'
+            'f21649265bb4b7b60535e9945ba0754a0065f31aed7d98d6853570e35c432706'
+            '5e421d0e8ed5f1053ad6295444acdbaa0fd41ee7a8d925623869c4e892797cce'
+            '469ed3f727e7648af25925bd0bb2b666334e256f8b51fffb4e77196882e7a869'
+            '3d6ef0a0ff2dacde9c698492bfdf3798560174f706d6cc1b0bb7ea83382a2d41'
+            '6a29cee605a2d4188d905266e588a0871c7206e67be22889cdf09277761dd622'
+            'f6b48feecacf0960bb630397a19bde9ad9bb1cff164358e9a16a41edbfa18011')
+sha256sums_x86_64=('0ed8c43b4ad6c3c3f3affe1317581992ba0eac6697d421a5c6d8210bf1e29ad7')
+b2sums=('2c53f205a940b0f9f68653b92ef46d49f828cbef3cfa8cf94d050c8e6df05c4fcaa4f9b9681b9130b14e3c790d31208eb244d123249a93e35e8e6165f3d858c9'
+        'SKIP'
+        '26230d1a111b24fe9239273acdfacda37c5bf009f861c448ad25392dcca433514246d629a077ce5c66478c7e0f4e5477ce5f95c91d08b3a02cc87bb35b849bcf'
+        'SKIP'
+        '6748c004b274b187c3f3a8c0ab9548fd09dc4c962a7e59f74c4ce0181fe1dc1ff8bf0adb07688e4bb8776e21d63aa06b1494338ac6a9fcb2db5df881ef38135c'
+        '2c6453510ac9c3f1ce65c620fa36f133adbd949bf9582222fbef32536de902e323781d8a79fdf0d84f75d2ff41abde3c995416cd74298aabfbb33e4e44bb9490'
+        '2f2e1291e3193d80fac8318e195632167679a301d98f6920e32da1236f309970b2ecdbb43f2f1985175e5f42ba5ed466efd19f4072597c56fb0b9b7e460c74a7'
+        '52544a7d684a00206812c16468c3dfc19eeb0d1caff8d3710002ef184557ff4ae190cce056c5a896c80d531213e8715348d7095be6fcc7375cd9839ed3c7567b'
+        '22bce836b92fbcc48032218139f3e75f84cd3fa6b4b0d51ea0b1460776db12ad35983aac7bed611303cc708094cc466522e6040172e5204dc720cbed593618e2'
+        '94be17594b686d08988d2fff12b94fea3a4773002efb178998b839df43066a7188a8081d337516bebd04c393743254f74091a27db51d31f6658599b6fc441b85'
+        '0ca0802103e018c6fc5f4697f89c3353afb38006a7e2684da3a5c12ac29b1335624ebb7b9b46b9d59a31cdaba9663830acbf93934badbda416ae5750c52b429a')
+b2sums_x86_64=('7082013345352c95303ee87cd78bf5d93ab49ec9f270e6cb803a05cb7f9a67c554bbd260de922d6d44145fd3712b410c13d67c8f76dc2b9f4088be86aeaec835')
+
+# https://www.kernel.org/pub/linux/kernel/v7.x/sha256sums.asc
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -87,9 +112,14 @@ prepare() {
   done
 
   echo "Setting config..."
-  cp ../config.$CARCH .config
+  cp ../config.x86_64 .config
+  make olddefconfig
+
+  patch -Np0 < ../riscv64.config-patch
+  mv .config .config.1
+  make mrproper
+  mv .config.1 .config
   make olddefconfig
-  diff -u ../config.$CARCH .config || :
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -97,7 +127,8 @@ prepare() {
 
 build() {
   cd $_srcname
-  make all
+  # Generate symbols for dtbs to support overlays
+  make DTC_FLAGS="-@" all
   make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
   make htmldocs SPHINXOPTS=-QT
 }
@@ -141,6 +172,9 @@ _package() {
   ZSTD_CLEVEL=19 make INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
     DEPMOD=/doesnt/exist modules_install  # Suppress depmod
 
+  echo "Installing dtbs..."
+  make INSTALL_DTBS_PATH="$pkgdir/usr/lib/modules/$(<version)/dtb" dtbs_install
+
   # remove build link
   rm "$modulesdir"/build
 }
@@ -166,6 +200,7 @@ _package-headers() {
   local karch
   case $CARCH in
     x86_64) karch=x86 ;;
+    riscv64) karch=riscv ;;
     *) echo "Unknown CARCH $CARCH"; exit 1 ;;
   esac
 
