--- PKGBUILD
+++ PKGBUILD
@@ -22,6 +22,8 @@
   sed 's|set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS|# set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS|' -i llvm/CMakeLists.txt
   sed -e 's|CMP0051 OLD|CMP0051 NEW|' -i CMakeLists.txt -i llvm/CMakeLists.txt
   patch -p1 -i ../gcc-15.patch
+  patch -Np1 -i ../llvm-config-guess-riscv.patch
+  patch -Np1 -i ../llvm-native-riscv.patch
 }
 
 build() {
@@ -75,3 +77,7 @@
 }
 
 # vim: ts=2 sw=2 et:
+
+source+=('llvm-config-guess-riscv.patch::https://github.com/llvm/llvm-project/commit/de1c2877a9ff12899ef50e179ade748fba8ab0c0.patch' 'llvm-native-riscv.patch')
+sha512sums+=('632a5b46558a5ffe783da91147d85f8f02e1f4b0a1c228a0e8e341aea5ab00771c10ef8a6661847b70fd34d68a51d1fb343a5bd56d1275737430b5258ae92a42' 'c08885de4c34f3d445ac8eecf49b65a626e3a07e6450c96f7a05ef61442fe234fe23da6b6e1cc9aa2f8158a748e5b757be705562eb8ad0e287e4cd30d828b37a')
+b2sums+=('29dd05cc0bd2e7a242e8cdb05063077f4346bbbc50a95c3f691dd21a0e3dd558fd8f767aa816dec058d95e6a433d870b1681c8679f3ac6a39a78299e0fa569dd' '93f2ec80a52741473b2d245314889a5299eae68724db7f31e7125c8877f7b15e42d0d09545717878a150f6c0b7d06ab2b3991e50538ae3c8d3163759ed6347b8')
