--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,15 @@ depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'lib
 makedepends=('qt6-base' 'python-sphinx' 'emacs')
 optdepends=('qt6-base: cmake-gui')
 source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
-         wxwidgets-3.2.patch)
+         wxwidgets-3.2.patch
+         fix-openmp-find-path.patch)
 sha512sums=('0925adf973d642fd76d4089b61b3882babb0a85050c4c57d5f5f3bd6b17564a9feb0beed236cd636e25f69072fa30b67ea3f80932380b6b6576f2dd78b8e6931'
-            '618a529fcd0a5029160fc6ef74439b7cb54ca74d13d52dd6cad005710bf1be263d0bf90bc4becc63a37929ac2f86c51278685e9da93fd82355ac646cb43b356f')
+            '618a529fcd0a5029160fc6ef74439b7cb54ca74d13d52dd6cad005710bf1be263d0bf90bc4becc63a37929ac2f86c51278685e9da93fd82355ac646cb43b356f'
+            'ae4accf6f776a1835ea981dece8c93ef7a3030c36d4343e1ee927c88967ba1393b2612eb784ecf9849bc596f32344c80d2d40e1947f5d0e0a0683be00281e725')
 
 prepare() {
   patch -d $pkgname-$pkgver -p1 < wxwidgets-3.2.patch # Support wxWidgets 3.2
+  patch -d $pkgname-$pkgver -Np1 < fix-openmp-find-path.patch
 }
 
 build() {
