Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,7 @@
 url="https://www.cmake.org/"
 license=('custom')
 depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'libuv' 'rhash')
-makedepends=('qt6-base' 'python-sphinx' 'emacs')
+makedepends=('qt6-base' 'python-sphinx')
 optdepends=('qt6-base: cmake-gui')
 source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
          cmake-cppflags.patch)
@@ -39,9 +37,6 @@
   cd ${pkgname}-${pkgver}
   make DESTDIR="${pkgdir}" install
 
-  emacs -batch -f batch-byte-compile \
-    "${pkgdir}"/usr/share/emacs/site-lisp/cmake-mode.el
-
   install -Dm644 Copyright.txt \
     "${pkgdir}"/usr/share/licenses/${pkgname}/LICENSE
 }
