Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 453000)
+++ PKGBUILD	(working copy)
@@ -12,9 +12,15 @@
 depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'libuv' 'rhash')
 makedepends=('qt6-base' 'python-sphinx' 'emacs')
 optdepends=('qt6-base: cmake-gui')
-source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz")
-sha512sums=('91507fc0643f2de20e37cd8d6a539d8cb38a80b604bf77de8e88c8446b6d7b6d02b7329d58966dc2412c6c75edcb9dfbdae02fe84965fefeb7ca47f87631fe17')
+source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
+        $pkgname-find-openmp-from-system-paths.patch::https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7526.patch)
+sha512sums=('91507fc0643f2de20e37cd8d6a539d8cb38a80b604bf77de8e88c8446b6d7b6d02b7329d58966dc2412c6c75edcb9dfbdae02fe84965fefeb7ca47f87631fe17'
+            '2e711b23e7a823374ba76434bb5fb870300edc40521e6dbea41604d16ca23e377a8f3e512ef20eb0a92993e6bddec7f40af3db9e488355ed5f4d9540537765bf')
 
+prepare() {
+  patch -d $pkgname-$pkgver -p1 < $pkgname-find-openmp-from-system-paths.patch
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   ./bootstrap --prefix=/usr \
