--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,14 @@ license=('custom')
 depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'libuv' 'rhash')
 makedepends=('qt6-base' 'python-sphinx' 'emacs')
 optdepends=('qt6-base: cmake-gui')
-source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz")
-sha512sums=('57edf2a8c1c9edeed7fd11271083f333d940b31e0e909d9c71671465961559c505e4e22f36e68bfe1a40a01eb05fe94bc8d37b062f9be613f959f864207e3764')
+source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
+        $pkgname-find-openmp-from-system-paths.patch::https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7526.patch)
+sha512sums=('57edf2a8c1c9edeed7fd11271083f333d940b31e0e909d9c71671465961559c505e4e22f36e68bfe1a40a01eb05fe94bc8d37b062f9be613f959f864207e3764'
+            '2e711b23e7a823374ba76434bb5fb870300edc40521e6dbea41604d16ca23e377a8f3e512ef20eb0a92993e6bddec7f40af3db9e488355ed5f4d9540537765bf')
+
+prepare() {
+  patch -d $pkgname-$pkgver -p1 < $pkgname-find-openmp-from-system-paths.patch
+}
 
 build() {
   cd ${pkgname}-${pkgver}
