Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 456080)
+++ PKGBUILD	(working copy)
@@ -12,9 +12,15 @@
 depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'libuv' 'rhash')
 makedepends=('qt6-base' 'python-sphinx' 'emacs')
 optdepends=('qt6-base: cmake-gui')
-source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz")
-sha512sums=('6f0e8e29bf0336f555ba72c4d83f35d820f8a5159cc999d48795dc57a6627b4ee3966dda84ca97d39906e35dd476ea00cf80023672cc0fad862e2996194c0674')
+source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
+        $pkgname-find-openmp-from-system-paths.patch::https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7526.patch)
+sha512sums=('6f0e8e29bf0336f555ba72c4d83f35d820f8a5159cc999d48795dc57a6627b4ee3966dda84ca97d39906e35dd476ea00cf80023672cc0fad862e2996194c0674'
+            '2e711b23e7a823374ba76434bb5fb870300edc40521e6dbea41604d16ca23e377a8f3e512ef20eb0a92993e6bddec7f40af3db9e488355ed5f4d9540537765bf')
 
+prepare() {
+  patch -d $pkgname-$pkgver -p1 < $pkgname-find-openmp-from-system-paths.patch
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   ./bootstrap --prefix=/usr \
