diff --git PKGBUILD PKGBUILD
index 3f72a03a..4bc7a77e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,15 @@ license=('custom')
 depends=('curl' 'libarchive' 'hicolor-icon-theme' 'jsoncpp' 'libjsoncpp.so' 'libuv' 'rhash')
 makedepends=('qt6-base' 'python-sphinx' 'emacs')
 optdepends=('qt6-base: cmake-gui')
-source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz")
-sha512sums=('94893f888c0cbfc58e54a0bd65d6c0697fe4a0e95c678b7cb35e7dc8854d57eb360bfc952750f97983348817f847f6df85903f21a5857b1a3880b2a7eb6cc029')
+source=("https://www.cmake.org/files/v${pkgver%.*}/${pkgname}-${pkgver}.tar.gz"
+        "fix-openmp-find-path.patch")
+sha512sums=('94893f888c0cbfc58e54a0bd65d6c0697fe4a0e95c678b7cb35e7dc8854d57eb360bfc952750f97983348817f847f6df85903f21a5857b1a3880b2a7eb6cc029'
+            'ae4accf6f776a1835ea981dece8c93ef7a3030c36d4343e1ee927c88967ba1393b2612eb784ecf9849bc596f32344c80d2d40e1947f5d0e0a0683be00281e725')
+
+prepare() {
+  cd ${pkgname}-${pkgver}
+  patch -Np1 -i "../fix-openmp-find-path.patch"
+}
 
 build() {
   cd ${pkgname}-${pkgver}
