--- PKGBUILD
+++ PKGBUILD
@@ -816,4 +816,8 @@
   install -Dm644 mesa-$_pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=("llvm23-jitlink.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/cda383625f5d1fd06657fb998b7d1370a7fabd8f.patch")
+sha256sums+=('fdd1049a560a0bac41bca17b0800372c2a54de6a90c9b298be2080b06dae09da')
+b2sums+=('3eb95c4742e8e7f4dfa67bbed724e4546ef4dde7066dc9229b72bb67e8bf4b66d31c478114f686f8ef605a4aca3d8b060c374f67832a1dbabe572b67d2238906')
+
 # vim:set sw=2 sts=-1 et:
