--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,6 @@ pkgname=(
   opencl-rusticl-mesa
   vulkan-dzn
   vulkan-gfxstream
-  vulkan-intel
   vulkan-nouveau
   vulkan-radeon
   vulkan-swrast
@@ -188,7 +187,7 @@ build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-opencl=icd
@@ -200,8 +199,8 @@ build() {
     -D osmesa=true
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,gfxstream,intel,intel_hasvk,nouveau,swrast,virtio,microsoft-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,vram-report-limit
+    -D vulkan-drivers=amd,gfxstream,nouveau,swrast,virtio,microsoft-experimental
+    -D vulkan-layers=device-select,overlay,screenshot,vram-report-limit
   )
 
   # Build only minimal debug info to reduce size
@@ -287,9 +286,6 @@ package_mesa() {
     _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
     _pick vkgfxstr $libdir/libvulkan_gfxstream.so
 
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
-
     _pick vknvidia $icddir/nouveau_icd.*.json
     _pick vknvidia $libdir/libvulkan_nouveau.so
 
