diff --git PKGBUILD PKGBUILD
index bc2252e..201ff5e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,6 @@ pkgname=(
   'vulkan-mesa-layers'
   'opencl-clover-mesa'
   'opencl-rusticl-mesa'
-  'vulkan-intel'
   'vulkan-radeon'
   'vulkan-swrast'
   'vulkan-virtio'
@@ -103,6 +102,11 @@ prepare() {
   # https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/27834
   patch -Np1 -i ../vulkan-dispatch-table-add-an-uncompacted-version.patch
 
+  # Add orcjit support
+  # https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/26018
+  patch -Np1 -i ../0002-add-llvm-orcjit.patch
+  patch -Np1 -i ../0004-llvmpipe-linkage.patch
+
   # Include package release in version string so Chromium invalidates
   # its GPU cache; otherwise it can cause pages to render incorrectly.
   # https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/2020604
@@ -115,7 +119,7 @@ build() {
     -D b_ndebug=true
     -D dri3=enabled
     -D egl=enabled
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,i915,iris,crocus,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-omx=bellagio
@@ -132,6 +136,7 @@ build() {
     -D intel-clc=enabled
     -D libunwind=enabled
     -D llvm=enabled
+    -D llvm-orcjit=true
     -D lmsensors=enabled
     -D microsoft-clc=disabled
     -D osmesa=true
@@ -140,8 +145,8 @@ build() {
     -D shared-glapi=enabled
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,intel,intel_hasvk,swrast,virtio
-    -D vulkan-layers=device-select,intel-nullhw,overlay
+    -D vulkan-drivers=amd,swrast,virtio
+    -D vulkan-layers=device-select,overlay
   )
 
   # Build only minimal debug info to reduce size
@@ -414,3 +419,10 @@ package_mesa() {
 
   install -m644 -Dt "${pkgdir}/usr/share/licenses/${pkgname}" LICENSE
 }
+
+source+=(0002-add-llvm-orcjit.patch
+         0004-llvmpipe-linkage.patch)
+sha256sums+=('21e92d5a85748418137ef3d81c1ae0c5b426e28c325f24e46fc46f239b831ae6'
+             '748bbfd1daa7935d6d9dc51f8dca65275af47c64e94cf451afcb90c887997f8d')
+b2sums+=('7a41b5fd100ae77af47e609258f79f6e8bfa81f23d4268f7d032fc64f09ab3678c75046d23247a77eec610e17fcd7039de248adaa1513972c33ea65cffed9f52'
+         'cb90168e04497aa23e433219856a6d877f6042b1d4f5fdbceb28b4d717e2630346efbe60551c71ed15befae3cc3b438dbbfb706baa22b674a63add805ab68e05')
