--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,6 @@ pkgname=(
   mesa-vdpau
   opencl-clover-mesa
   opencl-rusticl-mesa
-  vulkan-intel
   vulkan-mesa-layers
   vulkan-nouveau
   vulkan-radeon
@@ -187,7 +186,7 @@ build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-omx=bellagio
@@ -197,15 +196,16 @@ build() {
     -D glx=dri
     -D html-docs=enabled
     -D intel-clc=enabled
-    -D intel-rt=enabled
+    -D intel-rt=disabled
     -D libunwind=disabled
+    -D llvm-orcjit=true
     -D microsoft-clc=disabled
     -D osmesa=true
     -D platforms=x11,wayland
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,intel,intel_hasvk,nouveau,swrast,virtio
-    -D vulkan-layers=device-select,intel-nullhw,overlay
+    -D vulkan-drivers=amd,nouveau,swrast,virtio
+    -D vulkan-layers=device-select,overlay
   )
 
   # Build only minimal debug info to reduce size
@@ -277,9 +277,6 @@ package_mesa() {
     _pick clrust $libdir/libRusticlOpenCL*
     _pick clrust etc/OpenCL/vendors/rusticl.icd
 
-    _pick vkintel $icddir/intel_*.json
-    _pick vkintel $libdir/libvulkan_intel*.so
-
     _pick vklayer $libdir/libVkLayer_*.so
     _pick vklayer usr/bin/mesa-overlay-control.py
     _pick vklayer usr/share/vulkan/{ex,im}plicit_layer.d
