--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,6 @@ pkgname=(
   opencl-mesa
   vulkan-dzn
   vulkan-gfxstream
-  vulkan-intel
   vulkan-nouveau
   vulkan-radeon
   vulkan-swrast
@@ -169,13 +168,19 @@ prepare() {
   # its GPU cache; otherwise it can cause pages to render incorrectly.
   # https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/2020604
   echo "$pkgver-arch$epoch.$pkgrel" >VERSION
+
+  # Apply patches
+  for _patch in "$srcdir"/*.patch; do
+    echo "Applying patch: $_patch"
+    patch -Np1 < "$_patch"
+  done
 }
 
 build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-rusticl=true
     -D gallium-xa=disabled
@@ -185,8 +190,8 @@ build() {
     -D microsoft-clc=disabled
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,gfxstream,intel,intel_hasvk,nouveau,swrast,virtio,microsoft-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,vram-report-limit
+    -D vulkan-drivers=amd,gfxstream,nouveau,swrast,virtio,microsoft-experimental
+    -D vulkan-layers=device-select,overlay,screenshot,vram-report-limit
   )
 
   # Build only minimal debug info to reduce size
@@ -268,9 +273,6 @@ package_mesa() {
     _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
     _pick vkgfxstr $libdir/libvulkan_gfxstream.so
 
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
-
     _pick vknvidia $icddir/nouveau_icd.*.json
     _pick vknvidia $libdir/libvulkan_nouveau.so
 
@@ -541,4 +543,11 @@ package_mesa-docs() {
   install -Dm644 mesa-$pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=(mesa-zink-do-not-use-demote-on-img-blobs.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/82dda2180624dea25887af405840297b593af71f.patch
+         mesa-zink-workarounds-for-powervr-rogue-vulkan-blobs.patch::https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/34183.patch)
+sha256sums+=('88af2611624229ea1ddda0ea28e8581e2ab78dc88d1e2077cb69e43577478bfa'
+             '9a482fde2172cfd3f989d260590ba1e836dfc8bf63e52441991ff8773930e4be')
+b2sums+=('c7fbb0701255dbe15b16e11778ffb33c6f31612192ab6cfa05f5beffeaa998bc088654e6ff436da47d2684e5a09b8f294f08b712dd1d8a14928e6c0ada7219d3'
+         '5d8be90df4e3f9b27007e7c7791abc7012409de2ad50db6ef1cacd82777f02b8c6fac6deef7c595dba2591c60c2dc4cc2c6e6c09f59b9e6e1f1afbc64449ec99')
+
 # vim:set sw=2 sts=-1 et:
