--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,6 @@ pkgname=(
   'vulkan-mesa-layers'
   'opencl-clover-mesa'
   'opencl-rusticl-mesa'
-  'vulkan-intel'
   'vulkan-nouveau'
   'vulkan-radeon'
   'vulkan-swrast'
@@ -127,13 +126,17 @@ prepare() {
   # its GPU cache; otherwise it can cause pages to render incorrectly.
   # https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/2020604
   echo "$pkgver-arch$epoch.$pkgrel" >VERSION
+
+  # Add orcjit support
+  # https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/26018
+  patch -Np1 -i ../add-llvm-orcjit.patch
 }
 
 build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,i915,iris,crocus,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-omx=bellagio
@@ -144,13 +147,14 @@ build() {
     -D glx=dri
     -D intel-clc=enabled
     -D libunwind=disabled
+    -D llvm-orcjit=true
     -D microsoft-clc=disabled
     -D osmesa=true
     -D platforms=x11,wayland
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,intel,intel_hasvk,swrast,virtio,nouveau-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay
+    -D vulkan-drivers=amd,swrast,virtio,nouveau-experimental
+    -D vulkan-layers=device-select,overlay
   )
 
   # Build only minimal debug info to reduce size
@@ -501,3 +505,7 @@ package_mesa() {
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(add-llvm-orcjit.patch)
+sha256sums+=('50179d3a76ea0d40c968e0f2f0a48500103d3e626436bfc66d7d97ec09738c84')
+b2sums+=('3d955c5dc5da34b218b337ee536b1a9923c50f8a894f75df5d8474ef3f227bffb6a1fa644aacc00fe433dd06bbeb372f2b6bc4eb51598bb708910cc45047c12b')
