diff --git PKGBUILD PKGBUILD
index d395927..b1b959c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,7 +10,6 @@ pkgname=(
   opencl-mesa
   vulkan-dzn
   vulkan-gfxstream
-  vulkan-intel
   vulkan-nouveau
   vulkan-radeon
   vulkan-swrast
@@ -193,7 +192,7 @@ build() {
   local meson_options=(
     -D android-libbacktrace=disabled
     -D b_ndebug=true
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,iris,crocus,i915,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,llvmpipe,softpipe,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-rusticl=true
     -D gallium-xa=disabled
@@ -203,8 +202,8 @@ build() {
     -D microsoft-clc=disabled
     -D valgrind=enabled
     -D video-codecs=all
-    -D vulkan-drivers=amd,gfxstream,intel,intel_hasvk,nouveau,swrast,virtio,microsoft-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay,screenshot,vram-report-limit
+    -D vulkan-drivers=amd,gfxstream,nouveau,swrast,virtio,microsoft-experimental
+    -D vulkan-layers=device-select,overlay,screenshot,vram-report-limit
   )
 
   # Build only minimal debug info to reduce size
@@ -286,9 +285,6 @@ package_mesa() {
     _pick vkgfxstr $icddir/gfxstream_vk_icd.*.json
     _pick vkgfxstr $libdir/libvulkan_gfxstream.so
 
-    _pick vkintel $icddir/intel{,_hasvk}_icd.*.json
-    _pick vkintel $libdir/libvulkan_intel{,_hasvk}.so
-
     _pick vknvidia $icddir/nouveau_icd.*.json
     _pick vknvidia $libdir/libvulkan_nouveau.so
 
@@ -559,4 +555,11 @@ package_mesa-docs() {
   install -Dm644 mesa-$pkgver/docs/license.rst -t "$pkgdir/usr/share/licenses/$pkgname"
 }
 
+source+=(mesa-zink-do-not-use-demote-on-img-blobs.patch::https://gitlab.freedesktop.org/mesa/mesa/-/commit/82dda2180624dea25887af405840297b593af71f.patch
+         mesa-zink-workarounds-for-powervr-rogue-vulkan-blobs.patch::https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/34183.patch)
+sha256sums+=('88af2611624229ea1ddda0ea28e8581e2ab78dc88d1e2077cb69e43577478bfa'
+             '70234ead61987a9809d10645d8f051b1eaf701a4573f7cf14a4b5138f47ea058')
+b2sums+=('c7fbb0701255dbe15b16e11778ffb33c6f31612192ab6cfa05f5beffeaa998bc088654e6ff436da47d2684e5a09b8f294f08b712dd1d8a14928e6c0ada7219d3'
+         'e81f5e846d00708f6f9d2e4c86e06f864fd930af3ec28882b374ed360e9ba573d33503afb6da7e8dc7916a302f1ac658c68d46ea01af76481805c6f0bfe20d55')
+
 # vim:set sw=2 sts=-1 et:
