diff --git PKGBUILD PKGBUILD
index 24749b6..156ce9e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -7,9 +7,7 @@ pkgbase=mesa
 pkgname=(
   'vulkan-mesa-layers'
   'opencl-mesa'
-  'vulkan-intel'
   'vulkan-radeon'
-  'vulkan-swrast'
   'vulkan-virtio'
   'libva-mesa-driver'
   'mesa-vdpau'
@@ -96,13 +94,15 @@ prepare() {
 _libdir=usr/lib
 
 build() {
+  # Disable draw-use-llvm
+  # https://github.com/felixonmars/archriscv-packages/issues/1258
   local meson_options=(
     --libdir=/$_libdir
     -D android-libbacktrace=disabled
     -D b_ndebug=true
     -D dri3=enabled
     -D egl=enabled
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,i915,iris,crocus,zink,d3d12
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,zink,d3d12
     -D gallium-extra-hud=true
     -D gallium-nine=true
     -D gallium-omx=bellagio
@@ -127,8 +127,9 @@ build() {
     -D shared-glapi=enabled
     -D valgrind=enabled
     -D video-codecs=vc1dec,h264dec,h264enc,h265dec,h265enc
-    -D vulkan-drivers=amd,intel,intel_hasvk,swrast,virtio-experimental
-    -D vulkan-layers=device-select,intel-nullhw,overlay
+    -D vulkan-drivers=amd,virtio-experimental
+    -D vulkan-layers=device-select,overlay
+    -D draw-use-llvm=false
   )
 
   # Build only minimal debug info to reduce size
