Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 444750)
+++ PKGBUILD	(working copy)
@@ -6,3 +6,3 @@
 pkgbase=mesa
-pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-intel' 'vulkan-radeon' 'vulkan-swrast' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
+pkgname=('vulkan-mesa-layers' 'opencl-mesa' 'vulkan-radeon' 'libva-mesa-driver' 'mesa-vdpau' 'mesa')
 pkgdesc="An open-source implementation of the OpenGL specification"
@@ -14,3 +14,3 @@
              'libomxil-bellagio' 'libclc' 'clang' 'libglvnd' 'libunwind' 'lm_sensors' 'libxrandr'
-             'valgrind' 'glslang' 'vulkan-icd-loader' 'directx-headers' 'cmake' 'meson')
+             'valgrind' 'glslang' 'vulkan-icd-loader' 'cmake' 'meson')
 url="https://www.mesa3d.org/"
@@ -32,3 +32,4 @@
   cd mesa-$pkgver
-
+  # Workaround to disable llvmpipe without turning off draw-use-llvm
+  sed -i 's|if draw_with_llvm|if false|' src/gallium/meson.build
 }
@@ -41,4 +42,5 @@
   # https://gitlab.freedesktop.org/mesa/mesa/-/issues/6229
-  CFLAGS+=' -mtls-dialect=gnu'
-  CXXFLAGS+=' -mtls-dialect=gnu'
+  # riscv64 doesn't support -mtls-dialect=gnu
+  # CFLAGS+=' -mtls-dialect=gnu'
+  # CXXFLAGS+=' -mtls-dialect=gnu'
 
@@ -47,5 +49,5 @@
     -D platforms=x11,wayland \
-    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,svga,swrast,iris,crocus,zink,d3d12 \
-    -D vulkan-drivers=amd,intel,swrast \
-    -D vulkan-layers=device-select,intel-nullhw,overlay \
+    -D gallium-drivers=r300,r600,radeonsi,nouveau,virgl,swrast,zink \
+    -D vulkan-drivers=amd \
+    -D vulkan-layers=device-select,overlay \
     -D dri3=enabled \
@@ -53,3 +55,3 @@
     -D gallium-extra-hud=true \
-    -D gallium-nine=true \
+    -D gallium-nine=false \
     -D gallium-omx=bellagio \
@@ -199,3 +201,2 @@
   _install fakeinstall/usr/lib/bellagio
-  _install fakeinstall/usr/lib/d3d
   _install fakeinstall/usr/lib/lib{gbm,glapi}.so*
