--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,8 @@ b2sums=('85b9dffcd37ea72a2f49b0e2587cebb6f82ecd46f08b44b18da508c9dca352006f88937
 prepare() {
   cd $_name-$pkgver
   export RUSTUP_TOOLCHAIN=stable
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring@0.16.20
   cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
 }
 
