diff --git PKGBUILD PKGBUILD
index bd224af0..fbe6df3e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,15 +13,19 @@ checkdepends=(lv2lint)
 source=(
   "${pkgname}-${pkgver}.tar.gz::https://github.com/sadko4u/${pkgname}/archive/${pkgver}/${pkgver}.tar.gz"
   "$pkgbase-1.1.30-shared_object_ldflags.patch"
+  "$pkgbase-add-riscv-support.patch::https://github.com/sadko4u/lsp-plugins/pull/215.patch"
 )
 sha512sums=('fb3d63aa1bd86c716ae912a17ec3ed4c385af7c7c14788e86073e1076aae011962519ec3757f19af9cba9f2a4ee9c484c888935a77cd9a02d1bac3a10dfe17f4'
-            '4ffd405bccf5d86e9c88a8de86da804914fe955ecbfac56b6b323f2ed9a34e9a65bb27357ec9fa13fc3d00c5439efa44d4d60fbf212675aecc4282e5ec920b5c')
+            '4ffd405bccf5d86e9c88a8de86da804914fe955ecbfac56b6b323f2ed9a34e9a65bb27357ec9fa13fc3d00c5439efa44d4d60fbf212675aecc4282e5ec920b5c'
+            'f5515b6dc96f5f326f94215c610c70fa10df63eb056a7a0fa546385fcf902c17f3e0a2167d06ac34ccc590759b48f6921eb1a1aafd5f91c73c947daab2cacfa9')
 b2sums=('d94a0eb5289e5a2a6e925747bee32b59693c54e9ec2fa08c8dde0aff934141d799f3d305f492710c6080385305cd21cdbd85f46aa4736d3d6f472094bb4f811c'
-        'c63f14733a8c2ff31fc35a532c5572046d002a6ad812988694dc528c77d727a902d8022060b13a4ac93fbe3cbc144c89379c4496f9aa5840b4a3dfc27e9102ac')
+        'c63f14733a8c2ff31fc35a532c5572046d002a6ad812988694dc528c77d727a902d8022060b13a4ac93fbe3cbc144c89379c4496f9aa5840b4a3dfc27e9102ac'
+        '047a25a666238f226d0e9f2e17c2ab430ac581f51dd411218d334ea9ab47b2de2b34e41c6948aa58e0fc8b7924465e5a0cbe789241d4842946eda357524c0035')
 
 prepare() {
   cd "$pkgbase-$pkgver"
   patch -Np1 -i ../"$pkgbase-1.1.30-shared_object_ldflags.patch"
+  patch -Np1 -i ../"$pkgbase-add-riscv-support.patch"
 }
 
 build() {
