--- a/tests/test_patch.py
+++ b/tests/test_patch.py
@@ -1447,7 +1447,8 @@
         self.assertEqual(1000007, len(result[0].changes))
         # This is 2x the usual time for CI to allow for some slow tests
         # Really all we care about is that this parses faster than it used to (200s+)
-        self.assertGreater(20, time.time() - start_time)
+        # Note: riscv64 is still way too slow. Bump from 20s to 100s
+        self.assertGreater(100, time.time() - start_time)
 
 
 if __name__ == "__main__":
