diff --git PKGBUILD PKGBUILD
index c0d0661..6485876 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,16 @@ license=(MIT)
 depends=(python)
 makedepends=(python-build python-installer python-setuptools python-wheel)
 checkdepends=(python-pytest)
-source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz)
+source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz
+        increase-test-timeout.patch)
 #source=(${url}/archive/v${pkgver}rel/${pkgname}-${pkgver}.tar.gz)
-sha256sums=('9eefb4ebea5200408e02d413d2b4bc28daea6b78bb4b4d53431af7245f7d7edf')
+sha256sums=('9eefb4ebea5200408e02d413d2b4bc28daea6b78bb4b4d53431af7245f7d7edf'
+            'e82a1dbee6c952be0aa4cdbfa0001bd3208366ab73503f3856e640fadc22ab16')
+
+prepare() {
+  cd ${_pkg}-${pkgver}
+  patch -Np1 -i ../increase-test-timeout.patch
+}
 
 build() {
   cd ${_pkg}-${pkgver}
