diff --git a/trunk/PKGBUILD b/trunk/PKGBUILD
index ffb53e7a..a7cb32a0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,15 @@ license=(MIT)
 depends=(python)
 makedepends=(python-build python-installer python-setuptools python-wheel)
 checkdepends=(python-pytest)
-source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz)
+source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz
+        increase-test-timeout.patch)
 #source=(${url}/archive/v${pkgver}rel/${pkgname}-${pkgver}.tar.gz)
-sha256sums=('e95c108087845b09258ddfaf82aa13cf83ba8319475117c0909754ca8b54d742')
+sha256sums=('e95c108087845b09258ddfaf82aa13cf83ba8319475117c0909754ca8b54d742'
+            'e82a1dbee6c952be0aa4cdbfa0001bd3208366ab73503f3856e640fadc22ab16')
 
 build() {
   cd ${_pkg}-${pkgver}
+  patch -Np1 -i ../increase-test-timeout.patch
   python -m build --wheel --no-isolation
 }
 
