--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,15 @@ license=(MIT)
 depends=(python)
 makedepends=(python-build python-installer python-setuptools python-wheel)
 checkdepends=(python-pytest)
-source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz)
+source=(https://files.pythonhosted.org/packages/source/${_pkg::1}/${_pkg}/${_pkg}-${pkgver}.tar.gz
+        increase-test-timeout.patch)
 #source=(${url}/archive/v${pkgver}rel/${pkgname}-${pkgver}.tar.gz)
-sha256sums=('b5983a49f751158a7b5c62baf55aaf815728d3d80bf5dd0c5acb2d7d3d7391ee')
+sha256sums=('b5983a49f751158a7b5c62baf55aaf815728d3d80bf5dd0c5acb2d7d3d7391ee'
+            'e82a1dbee6c952be0aa4cdbfa0001bd3208366ab73503f3856e640fadc22ab16')
 
 build() {
   cd ${_pkg}-${pkgver}
+  patch -Np1 -i ../increase-test-timeout.patch
   python -m build --wheel --no-isolation
 }
 
