--- PKGBUILD
+++ PKGBUILD
@@ -78,6 +78,7 @@
 
 prepare() {
   cd ${pkgname#python-}
+  patch -Np1 -i ../fix-test-timing.patch
   python update_version_from_git.py $pkgver
 }
 
@@ -115,3 +116,6 @@
   cd ${pkgname#python-}
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(fix-test-timing.patch)
+b2sums+=('09eec179e35e8ef2d2bb7abd54ed676e6bd6563e6d9dc1688687d146f691a8d240ea3cddb66515659239c04dab15ecef8130370cc734eef5d456daa37376b093')
