--- PKGBUILD
+++ PKGBUILD
@@ -11,9 +11,8 @@ arch=(x86_64)
 url='https://www.open-mpi.org'
 license=('custom:OpenMPI')
 depends=(gcc-libs glibc hwloc libevent libnl openpmix zlib)
-makedepends=(cuda gcc-fortran inetutils valgrind)
+makedepends=(gcc-fortran inetutils valgrind)
 optdepends=(
-  'cuda: cuda support'
   'gcc-fortran: fortran support'
   'openssh: for offloading to remote hosts'
   'perl: for aggregate_profile.pl and profile2mat.pl'
@@ -50,7 +49,6 @@ build() {
     --enable-pretty-print-stacktrace \
     --libdir=/usr/lib \
     --sysconfdir=/etc/$pkgname \
-    --with-cuda=/opt/cuda \
     --with-hwloc=external \
     --with-libevent=external \
     --with-pmix=external \
