Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 437782)
+++ PKGBUILD	(working copy)
@@ -10,10 +10,9 @@
 arch=('x86_64')
 license=('custom:OpenMPI')
 depends=('glibc' 'libltdl' 'hwloc' 'openssh' 'zlib' 'libnl' 'perl' 'libevent')
-makedepends=('inetutils' 'valgrind' 'gcc-fortran' 'cuda')
+makedepends=('inetutils' 'valgrind' 'gcc-fortran')
 optdepends=(
   'gcc-fortran: fortran support'
-  'cuda: cuda support'
 )
 options=('staticlibs')
 source=(https://www.open-mpi.org/software/ompi/v${pkgver%.*}/downloads/${pkgname}-${pkgver}.tar.bz2)
@@ -36,7 +35,6 @@
     --with-hwloc=/usr \
     --with-libltdl=/usr  \
     --with-libevent=/usr  \
-    --with-cuda=/opt/cuda \
     FC=/usr/bin/gfortran \
     LDFLAGS="${LDFLAGS} -Wl,-z,noexecstack"
   make
