Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 420565)
+++ PKGBUILD	(working copy)
@@ -10,10 +10,9 @@
 arch=('x86_64')
 license=('custom:OpenMPI')
 depends=('glibc' 'libltdl' 'hwloc' 'openssh' 'zlib' 'libnl' 'perl' 'libevent')
-makedepends=('inetutils' 'valgrind' 'gcc-fortran' 'cuda')
+makedepends=('inetutils' 'gcc-fortran')
 optdepends=(
   'gcc-fortran: fortran support'
-  'cuda: cuda support'
 )
 options=('staticlibs')
 source=(https://www.open-mpi.org/software/ompi/v${pkgver%.*}/downloads/${pkgname}-${pkgver}.tar.bz2)
@@ -29,14 +28,11 @@
     --libdir=/usr/lib/${pkgname} \
     --enable-builtin-atomics \
     --enable-mpi-cxx \
-    --with-valgrind \
-    --enable-memchecker \
     --enable-pretty-print-stacktrace \
     --without-slurm \
     --with-hwloc=/usr \
     --with-libltdl=/usr  \
     --with-libevent=/usr  \
-    --with-cuda=/opt/cuda \
     FC=/usr/bin/gfortran \
     LDFLAGS="${LDFLAGS} -Wl,-z,noexecstack"
   make
