Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -10,7 +10,7 @@
 arch=('x86_64')
 license=('custom:OpenMPI')
 depends=('libltdl' 'hwloc' 'openssh' 'zlib' 'libnl')
-makedepends=('inetutils' 'valgrind' 'gcc-fortran')
+makedepends=('inetutils' 'gcc-fortran')
 optdepends=('gcc-fortran: fortran support')
 options=('staticlibs')
 source=(https://www.open-mpi.org/software/ompi/v${pkgver%.*}/downloads/${pkgname}-${pkgver}.tar.bz2)
@@ -25,8 +25,6 @@
     --libdir=/usr/lib/${pkgname} \
     --enable-builtin-atomics \
     --enable-mpi-cxx \
-    --with-valgrind \
-    --enable-memchecker \
     --enable-pretty-print-stacktrace \
     --without-slurm \
     --with-hwloc=/usr \
