--- PKGBUILD
+++ PKGBUILD
@@ -21,18 +21,15 @@ depends=(
   zlib
 )
 makedepends=(
-  cuda
   gcc-fortran
   inetutils
   valgrind
 )
 optdepends=(
-  'cuda: cuda support'
   'gcc-fortran: fortran support'
   'perl: for aggregate_profile.pl and profile2mat.pl'
 )
 provides=(
-  libmca_common_cuda.so
   libmca_common_monitoring.so
   libmca_common_ompio.so
   libmca_common_sm.so
@@ -70,7 +67,6 @@ build() {
     --enable-pretty-print-stacktrace
     --libdir=/usr/lib
     --sysconfdir=/etc/$pkgname
-    --with-cuda=/opt/cuda
     --with-hwloc=external
     --with-libevent=external
     --with-pmix=external
