diff --git PKGBUILD PKGBUILD
index aee52b1..2bdca7c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,21 @@ groups=('lv2-plugins' 'pro-audio')
 depends=('glibc' 'libglvnd' 'libvterm' 'libx11' 'lv2-host' 'ttf-fira-code' 'ttf-fira-sans')
 makedepends=('faust' 'fontconfig' 'glew' 'glu' 'lv2' 'meson')
 checkdepends=('lv2lint')
-source=("https://git.open-music-kontrollers.ch/lv2/${pkgname}/snapshot/${pkgname}-${pkgver}.tar.xz"{,.asc})
+source=("https://git.open-music-kontrollers.ch/lv2/${pkgname}/snapshot/${pkgname}-${pkgver}.tar.xz"{,.asc}
+        "fix_pthread_linking.patch")
 sha512sums=('d5b954303f36c5351bf301b499b555556dc141b13f612be9ecbbfea24c62ecc2afdfc5de868795c635ed5806efe189b3b174b76e01e3764ab4616a0d82ed2657'
-            'SKIP')
+            'SKIP'
+            '274c577a2d8f1e83463c86d25cb01f537addab45be39e94a1bcb6b114ecd30763b4db94d1d99fd606e11fe9829e32819a95da872c1481cadcf78d7af8bfef616')
 b2sums=('71bf6ecdbdaacf6bb100d93b8f36be68b3b09798955fd786e06b21bc7c49332aa84fd21cdcb412debf83312073db36773495ab55f9fe598c0651c2e616ed8fdb'
-        'SKIP')
+        'SKIP'
+        'f9a2d1e9d3dfe3e02229295584e97b2e5e0611d7073b93d22447d879bdb596afdc9a6f6ba34afafc2629404b54aab5da7a05af803004530ce7d4bcc9c7d31fc4')
 validpgpkeys=('5AE26CEBBF49705C19FB273A30258F99258CB76A') # Hanspeter Portner <dev@open-music-kontrollers.ch>
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i "../fix_pthread_linking.patch"
+}
+
 build() {
   cd "$pkgname-$pkgver"
   arch-meson build
