--- PKGBUILD
+++ PKGBUILD
@@ -26,7 +26,8 @@
         'kea-dhcp-ddns.service'
         'kea-ctrl-agent.service'
         'fix-build-with-boost-1.89.patch'
-        'fix-build-with-boost-1.90.patch')
+        'fix-build-with-boost-1.90.patch'
+        'replace-boost-static-assert.patch')
 b2sums=('993f4a9d8ebde89685376694bb3655c5e1e340d7d56726724ef72262021b899eed94fe52a5c2a39e0630f30c3c3849e857cd80208b1241bc04a31fa3c244f2bf'
         '630310bb2b544a00276a0fa0fff8b7bb93f6bf63e3e5f8ed38f2e1fd2d9747ea4b4cbaa7c0023eec1baddf5e8fe1966b71c6941b3a3cd2e7705e67b15543f2c7'
         'd62c9181b55956441c43414bc2a5a8cff143281931bd57fe584ed03e6035a87c610da2530c10233189fa5926e98e47d05c120b5218e77c9b842131668c9be1e9'
@@ -35,7 +36,8 @@
         '8c7d435ea80d47778e7ed9fc382e0aa47806e2ba9327e4a1b095d5bc4bd347286df1254bb06d319b6c056969ee88493894267e0a60e9e7b9f8428c20b895264e'
         '59097fb418829501de6296752bf3f984d1fcf4eb6fc6721f0f70bf09c5df6bb32757e337aee3a8bbec778bb9a7a30228685bafffd3175da4dc98e8f7eede0f51'
         '4cfe04b61d7884ed492c3f7f06cf3ea68ec231063a8ba868221bbeb17ab94315d80e05243db89a12b29d0b2b1e35e1c7a86f33a42874ddc85661420ea9347e07'
-        '1258a72d82eedd1d428b65079cbf66e1efb1bd7fbe26c0029151c9264afd4d526e11f6f7a648fe4165cab52107bfb40577f59e50178e6c82b98efaab328f7e3c')
+        '1258a72d82eedd1d428b65079cbf66e1efb1bd7fbe26c0029151c9264afd4d526e11f6f7a648fe4165cab52107bfb40577f59e50178e6c82b98efaab328f7e3c'
+        '6989fc5e76c4c646c1b1bbf7961ed740366993b2c1cafb3fb7d2a93bd79409cac5e7c4794a14b4519a3610e7534471d8527fde63b52b9c98d353cee336016e95')
 validpgpkeys=('BE0E9748B718253A28BB89FFF1B11BF05CF02E57'  # Internet Systems Consortium, Inc. (Signing key, 2017-2018) <codesign@isc.org>
               'AE3FAC796711EC59FC007AA474BB6B9A4CBB3D38'  # Internet Systems Consortium, Inc. (Signing key, 2019-2020) <codesign@isc.org>
               '7E1C91AC8030A5A59D1EFAB9750F3C87723E4012'  # Internet Systems Consortium, Inc. (Signing key, 2021-2022) <codesign@isc.org>
@@ -51,6 +53,10 @@
 	# Fix build with boost 1.90
 	# See https://gitlab.archlinux.org/archlinux/packaging/packages/kea/-/merge_requests/3#note_440881
 	patch -Np1 -i "${srcdir}/fix-build-with-boost-1.90.patch"
+	# Replace BOOST_STATIC_ASSERT for boost 1.90
+	# See https://gitlab.isc.org/isc-projects/kea/-/issues/4266
+	# See https://gitlab.isc.org/isc-projects/kea/-/commit/c54dfd47714fea7e79c16d99c09b896d9c8d44df
+	patch -Np1 -i "${srcdir}/replace-boost-static-assert.patch"
 }
 
 build() {
