--- a/nczarr_test/test_readcaching.c
+++ b/nczarr_test/test_readcaching.c
@@ -74,8 +74,8 @@
   {
     size_t mbused = getPeakRSS() / (1024 * 1024);
     printf("Max mem: %zu MB\n", mbused);
-    if(mbused > 100) {
-      fprintf(stderr,"*** Failed: used: %luMB expected: < 100MB\n",mbused);
+    if(mbused > 256) {
+      fprintf(stderr,"*** Failed: used: %luMB expected: < 256MB\n",mbused);
       return (1);
     }
   }
--- a/nczarr_test/test_writecaching.c
+++ b/nczarr_test/test_writecaching.c
@@ -87,11 +87,11 @@
   {
     size_t mbused = getPeakRSS() / (1024 * 1024);
     printf("Max mem: %zu MB\n", mbused);
-    if(mbused > 128) {
-      printf("*** Failed: used: %luMB expected: < 128\n",mbused);
+    if(mbused > 256) {
+      printf("*** Failed: used: %luMB expected: < 256\n",mbused);
       return (1);
     } else {
-      printf("*** Passed: used: %luMB (< 128)\n",mbused);
+      printf("*** Passed: used: %luMB (< 256)\n",mbused);
       return 0;
     }
   }
