--- PKGBUILD
+++ PKGBUILD
@@ -20,6 +20,7 @@
 prepare() {
   # https://github.com/Unidata/netcdf-c/issues/2188 https://github.com/Unidata/netcdf-c/issues/2242
   sed -i "/tst_remote3/d" ${pkgname}-c/ncdap_test/CMakeLists.txt
+  patch -Np1 -d ${pkgname}-c -i ../cachetest-rss.patch
 }
 
 build() {
@@ -49,3 +50,6 @@
   make -C build DESTDIR="${pkgdir}" install
   install -Dm644 ${pkgname}-c/COPYRIGHT -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
+
+source+=(cachetest-rss.patch)
+sha256sums+=('53b6b5729c7b3375b4ce36db79f6ea37d04870c370a047d0dcf3cda3dd2688d2')
