--- PKGBUILD
+++ PKGBUILD
@@ -12,13 +12,16 @@ url='http://tools.android.com/'
 license=(Apache MIT)
 depends=(libusb protobuf brotli zstd android-udev)
 makedepends=(pcre2 gtest cmake go ninja git)
-source=(https://github.com/nmeum/android-tools/releases/download/$_tag/android-tools-$_tag.tar.xz        gcc12.patch)
+source=(https://github.com/nmeum/android-tools/releases/download/$_tag/android-tools-$_tag.tar.xz        gcc12.patch
+        'add_boringssl_riscv_support.patch')
 sha256sums=('0ef69f919d58a2bdff2083d2e83a9ef38df079ec82651b2544e9e48086df5ab8'
-            'df27d803da03d0a2656c4feb3cf471a076c459f45004ec20ec86a5cfb36be862')
+            'df27d803da03d0a2656c4feb3cf471a076c459f45004ec20ec86a5cfb36be862'
+            'b71e966c2cd3f3657523fa71e01c2fb73e34afd66028c9476925bdd82d7abe1d')
 
 prepare() {
   cd android-tools-$_tag
   patch -Np1 -i ../gcc12.patch
+  patch -Np1 -i ../add_boringssl_riscv_support.patch
 }
 
 build() {
