--- PKGBUILD
+++ PKGBUILD
@@ -15,11 +15,18 @@ depends=("libgl" "libx11" "libxcb" "mesa" "qt5-base" "qt5-svg" "qt5-x11extras" "
 options=("!lto")
 source=("https://github.com/baldurk/$pkgname/archive/v${pkgver}.tar.gz"
         "https://github.com/baldurk/$pkgname/releases/download/v${pkgver}/v${pkgver}.tar.gz.asc"
-        "https://github.com/baldurk/swig/archive/${pkgname}-modified-7.zip")
+        "https://github.com/baldurk/swig/archive/${pkgname}-modified-7.zip"
+        "add-riscv64-support.patch::https://github.com/baldurk/renderdoc/pull/3069.diff")
 validpgpkeys=('1B039DB9A4718A2D699DE031AC612C3120C34695')
 b2sums=('492b32140c8ad1ab3e753158cfede64ea0b580a2df03094aeeff2d4eac659fbedc62971f65d3fe024e0282de39d5f122b512be2fc6a28ac7abf794b58f116322'
         'SKIP'
-        '136b0b48d5bddb2dfe6fffd54220a5b62bb9a725b800f1f7846d03469f0c859d0585a61118f64b2def81ad94ff650bdf82d1db7aeb7608a190792ce8033597dc')
+        '136b0b48d5bddb2dfe6fffd54220a5b62bb9a725b800f1f7846d03469f0c859d0585a61118f64b2def81ad94ff650bdf82d1db7aeb7608a190792ce8033597dc'
+        'fa368aeec786c67dc65b4004b6dc7fde55ad8ead1edcce9b3b5aeb43a26474a06e1bb234b1d4088fa32b92d720614843c930c755e576d526b6bce832acfaaf5b')
+
+prepare() {
+  cd "${pkgname}-$pkgver"
+  patch -Np1 -i ../add-riscv64-support.patch
+}
 
 build() {
     cmake -G Ninja \
