diff --git PKGBUILD PKGBUILD
index 8865515..3645d70 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,14 +17,17 @@
               # moto optdepends
               python-docker python-openapi-spec-validator python-yaml python-flask python-flask-cors)
 source=("https://github.com/aio-libs/aiobotocore/archive/$pkgver/$pkgname-$pkgver.tar.gz"
-        moto-4.1.7.diff)
+        moto-4.1.7.diff
+        increase-MotoService-start-timeout.patch::https://github.com/aio-libs/aiobotocore/pull/1053.diff)
 sha256sums=('a9a6f215c06f826bdbd9af25ea697bef0c9861e64e226a96cad519ab1a5548e7'
-            '06c2548e54067c15c192c48b72a29b04b987aa3b694e7676366e3cbc638df81f')
+            '06c2548e54067c15c192c48b72a29b04b987aa3b694e7676366e3cbc638df81f'
+            'b2e3fa55bd7d5d74ce642eb734562f6ea6181648249e4d39b4d931fccac562a6')
 
 prepare() {
   cd $_pkgname-$pkgver
 
   patch -Np1 -i ../moto-4.1.7.diff
+  patch -Np1 -i ../increase-MotoService-start-timeout.patch
 
   # For pytest-asyncio >= 0.19
   # https://github.com/aio-libs/aiobotocore/issues/965
