diff --git PKGBUILD PKGBUILD
index f2ac48c..b1e16cf 100644
--- PKGBUILD
+++ PKGBUILD
@@ -42,6 +42,7 @@
       -Bbuild \
       -GNinja \
       -DBUILD_TESTS=OFF \
+      -DENABLE_WERROR=OFF \
       -DCMAKE_INSTALL_PREFIX=/usr
   ninja -C build
 
@@ -63,7 +64,7 @@
     -DLLVM_INSTALL_TOOLCHAIN_ONLY=ON \
     -DLLVM_INCLUDE_EXAMPLES=OFF \
     -DLLVM_INCLUDE_TESTS=OFF \
-    -DLLVM_TARGETS_TO_BUILD="X86;WebAssembly" \
+    -DLLVM_TARGETS_TO_BUILD="RISCV;WebAssembly" \
     -DLLVM_ENABLE_PROJECTS="lld;clang" \
     -DCLANG_INCLUDE_TESTS=OFF
   ninja -C build
