--- PKGBUILD
+++ PKGBUILD
@@ -5,7 +5,7 @@ pkgname=python-$_pyname
 pkgver=0.6.0
 pkgrel=2
 pkgdesc='Interior-point solver for convex conic optimisation problems'
-arch=(x86_64)
+arch=(riscv64 x86_64)
 url='https://github.com/oxfordcontrol/Clarabel.rs'
 license=(Apache)
 depends=(gcc-libs
@@ -19,7 +19,7 @@ sha256sums=('ef909a393e72981ca10b1d866d9cc7fb6295ece20ae035def764338894961184')
 
 prepare() {
   cd $_pyname-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cargo fetch --locked
 }
 
 build() {
