diff --git src/meson.build src/meson.build
index 330e401..a116e5c 100644
--- src/meson.build
+++ src/meson.build
@@ -59,7 +59,7 @@ test (
     '--no-deps',
     cargo_options,
   ],
-  timeout: 300, # cargo might take a bit of time sometimes
+  timeout: 1800, # cargo might take a bit of time sometimes
   env: [
     cargo_env,
   ],
@@ -72,7 +72,7 @@ test (
     'test',
     cargo_options,
   ],
-  timeout: 300, # cargo might take a bit of time sometimes
+  timeout: 1800, # cargo might take a bit of time sometimes
   env: [
     cargo_env,
   ],
