--- PKGBUILD
+++ PKGBUILD
@@ -23,7 +23,7 @@ build() {
   cd "$srcdir/$pkgname-$pkgver"
   # systemd support is broken in mount.cifs
   # https://bugs.archlinux.org/task/30958
-  autoreconf -i
+  autoreconf -fiv
   # https://www.spinics.net/lists/linux-cifs/msg21550.html
   # change back to libcap-ng depend when upstream is fixed
   ./configure --prefix=/usr --with-libcap=yes --sbindir=/usr/bin --disable-systemd 
