Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422631)
+++ PKGBUILD	(working copy)
@@ -8,7 +8,7 @@
 arch=(x86_64)
 url="https://github.com/heftig/rtkit"
 license=(GPL3 'custom:BSD')
-depends=(dbus polkit systemd)
+depends=(dbus systemd)
 makedepends=(git meson vim)
 _commit=b9169402fe5e82d20efb754509eb0b191f214599  # tags/v0.13^0
 source=("git+https://github.com/heftig/rtkit?signed#commit=$_commit")
