diff --git PKGBUILD PKGBUILD
index 2f003e4..e10b1f5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -56,4 +56,12 @@
   install -m755 -d "${pkgdir}/etc/ananicy.d"
 }
 
+source+=("vmlinux.h")
+sha512sums+=('5362d81c85a7d81a985c1e71f951d7e0f8047d4cfceac5b73d167599f5ec5c4613bed3a7d1c7103b646d9821aa4d41c915074b918f102df31af5a1f2c0b86f71')
+
+prepare() {
+  cd "${srcdir}/${pkgname}"
+  install -Dm644 "${srcdir}/vmlinux.h" -t libananicycpp_bpf/riscv
+}
+
 # vim:set sw=2 sts=2 et:
