diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,9 @@ sha256sums=('b64ab8ce00af9de5ba1e384c623e0f12d0f01a399472b7b3aab3b5d1bb50606c')
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "${CARCH}-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build(){
