diff --git PKGBUILD PKGBUILD
index a0e6168..c5009b7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,11 +9,14 @@ arch=('x86_64')
 license=('BSD')
 url="https://github.com/nelhage/reptyr"
 depends=('glibc')
-source=($pkgname-$pkgver.tar.gz::https://github.com/nelhage/$pkgname/tarball/$pkgname-$pkgver)
-sha256sums=('5f8426b6332467b1e61373380d96f8b7e956e9e623617cde84cf76f360ecd3cc')
+source=($pkgname-$pkgver.tar.gz::https://github.com/nelhage/$pkgname/tarball/$pkgname-$pkgver
+        "add_riscv64_support.patch")
+sha256sums=('5f8426b6332467b1e61373380d96f8b7e956e9e623617cde84cf76f360ecd3cc'
+            '4d8c8fb337893ea193ca96ba806daa321270c29050890fbaca114a99c2dad7b2')
 
 prepare() {
   cd "${srcdir}"/nelhage-$pkgname-*
+  patch -Np1 -i ../add_riscv64_support.patch
 
   sed -i 's|etc/bash_completion.d|usr/share/bash-completion/completions|' Makefile
 }
