Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -49,6 +49,9 @@
      --disable-t1utils \
      --disable-bibtexu \
      --disable-xz \
+     --disable-luajittex \
+     --disable-luajithbtex \
+     --disable-mfluajit \
      --enable-shared \
      --disable-static \
      --with-system-zlib \
@@ -74,7 +77,6 @@
      --with-xdvi-x-toolkit=xaw \
      --disable-dump-share \
      --disable-aleph \
-     --enable-luatex \
      --with-clisp-runtime=default \
      --enable-xindy --disable-xindy-rules --disable-xindy-docs
    #############################################################
@@ -123,9 +125,6 @@
    mkdir -p "${pkgdir}"/usr/bin
    ln -s eptex "${pkgdir}"/usr/bin/platex
    ln -s euptex "${pkgdir}"/usr/bin/uplatex
-   ln -s luatex "${pkgdir}"/usr/bin/dvilualatex
-   ln -s luatex "${pkgdir}"/usr/bin/dviluatex
-   ln -s luahbtex "${pkgdir}"/usr/bin/lualatex
    ln -s pdftex "${pkgdir}"/usr/bin/amstex
    ln -s pdftex "${pkgdir}"/usr/bin/cslatex
    ln -s pdftex "${pkgdir}"/usr/bin/csplain
@@ -183,7 +182,6 @@
 chkweb
 cjk-gs-integrate
 cluttex
-cllualatex
 clxelatex
 context
 contextjit
@@ -238,8 +236,6 @@
 listings-ext.sh
 ltxfileinfo
 ltximg
-luaotfload-tool
-luatools
 lwarpmk
 make4ht
 match_parens
