--- PKGBUILD
+++ PKGBUILD
@@ -27,6 +27,7 @@
 
 prepare() {
   cd "$pkgbase"
+  patch -Np0 -i ../uv-platform-arch-riscv64.patch
   local tripple="$(rustc -vV | sed -n 's/host: //p')"
   cargo fetch --locked --target "$tripple"
   mkdir completions
@@ -82,4 +83,7 @@ package_python-uv() {
   rm -rf "$pkgdir/usr/bin"
 }
 
+source+=(uv-platform-arch-riscv64.patch)
+sha256sums+=('070e691ba1812c05bf8490bb344368693caf000205ff23dc36ad609e51b7d03a')
+
 # vim: ts=2 sw=2 et:
