--- crates/uv-platform-tags/src/platform.rs
+++ crates/uv-platform-tags/src/platform.rs
@@ -86,6 +86,7 @@ pub enum Arch {
     #[serde(alias = "amd64")]
     X86_64,
     S390X,
+    Riscv64,
 }
 
 impl fmt::Display for Arch {
@@ -99,6 +100,7 @@ impl fmt::Display for Arch {
             Self::X86 => write!(f, "i686"),
             Self::X86_64 => write!(f, "x86_64"),
             Self::S390X => write!(f, "s390x"),
+            Self::Riscv64 => write!(f, "riscv64"),
         }
     }
 }
@@ -116,6 +118,7 @@ impl Arch {
             Self::X86 | Self::X86_64 => Some(5),
             // unsupported
             Self::Armv6L => None,
+            Self::Riscv64 => None,
         }
     }
 }
--- crates/uv-python/src/platform.rs
+++ crates/uv-python/src/platform.rs
@@ -165,6 +165,9 @@ impl From<&uv_platform_tags::Arch> for Arch {
                 target_lexicon::X86_32Architecture::I686,
             )),
             uv_platform_tags::Arch::X86_64 => Self(target_lexicon::Architecture::X86_64),
+            uv_platform_tags::Arch::Riscv64 => Self(target_lexicon::Architecture::Riscv64(
+                target_lexicon::Riscv64Architecture::Riscv64,
+            )),
         }
     }
 }
