diff --git PKGBUILD PKGBUILD
--- PKGBUILD
+++ PKGBUILD
@@ -31,7 +31,7 @@ check() {
   fakeroot tar xf rdiff-backup-filesrepo/rdiff-backup_testfiles.tar
   cd $pkgname
   export PATH="$PWD/build/scripts-3.10:$PATH"
-  export PYTHONPATH="$PWD/build/lib.linux-x86_64-3.10"
+  export PYTHONPATH="$PWD/build/lib.linux-$CARCH-3.10"
   python testing/commontest.py
   python testing/ctest.py
   python testing/timetest.py
