diff --git a/go/vendor/github.com/qrtz/nativemessaging/endian_riscv64.go b/go/vendor/github.com/qrtz/nativemessaging/endian_riscv64.go
new file mode 100644
index 00000000..41bfa466
--- /dev/null
+++ b/go/vendor/github.com/qrtz/nativemessaging/endian_riscv64.go
@@ -0,0 +1,8 @@
+package nativemessaging
+
+import (
+	"encoding/binary"
+)
+
+// NativeEndian system native byte order
+var NativeEndian = binary.LittleEndian
