diff --git PKGBUILD PKGBUILD
index 7741435..13197c2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,22 +20,26 @@ _archive="$pkgbase-v$pkgver"
 source=("$_url/releases/download/v$pkgver/$_archive.tar.xz"{,.sig}
         "keybase-gui"
         "0001-Maintain-current-contextIsolation-behavior-across-el.patch"
-        "0001-Don-t-use-electron-to-build.patch")
+        "0001-Don-t-use-electron-to-build.patch"
+        "update_go_mod.patch"::"https://github.com/keybase/client/pull/24842.patch")
 sha512sums=('260b3c89a6834ef6285cdc746bcf582ef9376ef06f50fc410a61a002591a5dc1189270ddd4df8516bf24a3c8aa9c6c230339eefe42a5010acf8d4fbdfc551a93'
             'SKIP'
             'bdecad57f8703eb6011b0c9ff238b9f3d78192031adfb5a27a7e6910119c600726b87aaacada88877eaeddcd6134fbe72b014bcaeb6cc7563c3753fcb33c52c9'
             '5cb3d93c4bb468fd47e1670f73cc3b8be14977ccf563a47d776285b6b9d96042647062427e58bd981727362caa23d17acdc0152dfd915efeeb6c4e76ca76ca65'
-            '3d38410574ac9d3a09a9ed4ba82bffb3078b15986d8447fe53d7499f538f5c54093f0893667ee64ddd22609b83a8c9c15e50f11f61ddb48cff3ecdf5b8c4628e')
+            '3d38410574ac9d3a09a9ed4ba82bffb3078b15986d8447fe53d7499f538f5c54093f0893667ee64ddd22609b83a8c9c15e50f11f61ddb48cff3ecdf5b8c4628e'
+            '0fe48ffe82387a21d06fe60d5eb82b9c499b18f619e1d457cca8d25aae4626cd041bb2dc66b69781ad4bf5f843d33021a31f45aa4e3f6b8efdff09505ee1a0d2')
 b2sums=('0711762ed70c9977a92702c33e4cddc3eee5b5c7ee4bbc381ae07649b5cc635a811331b6492f853813b82e557bb4211677523e62f68d2e37267957bcafb9d844'
         'SKIP'
         '1ecf08e4a0b2836828fc4f9283a69e1fbda6f268195ae56433050d54070038aacf1ae3057202324c678b242c88345cc8f453c7798129753f083c33db256517a4'
         '67f416c59b549606d037167c796c5b99f0ca9158fb767d52c101f41db6d31927b57618efb4d6eaf590cbd3c7aab4dd71a64bc3b52580e2383f5389c95264a499'
-        'd556694526d973431012c36891d82578aeafdd159e7f7f9108325946e874cef4303dff8dc12f0ec7d2c5b731d5cf20bec26c7f18b6483d8f2910c93e3a1a7a68')
+        'd556694526d973431012c36891d82578aeafdd159e7f7f9108325946e874cef4303dff8dc12f0ec7d2c5b731d5cf20bec26c7f18b6483d8f2910c93e3a1a7a68'
+        'e75e8ac247245857f0bcfb875d217bdbfb0b281b58d06fd042b7ef518fe9bad39aafbfabd0c047f628c55993a53775de467b28a29be6517dccfa5b0628b01765')
 validpgpkeys=('222B85B0F90BE2D24CFEB93F47484E50656D16C7') # Keybase.io Code Signing (v1) <code@keybase.io>
 
 prepare() {
 	ln -sf "${_archive/$pkgbase/client}" "$_archive"
 	cd "$_archive"
+	patch -Np1 -i "../update_go_mod.patch"
 
 	export GOPATH="${srcdir}/.gopath"
 	mkdir -p "${GOPATH}"/src/github.com/keybase
