diff --git PKGBUILD PKGBUILD
index 2c2f1fef..3a3d951d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,21 +15,25 @@ makedepends=('git' 'go' 'yarn')
 source=("https://github.com/keybase/client/releases/download/v${pkgver}/${pkgbase}-v${pkgver}.tar.xz"{,.sig}
         "keybase-gui"
         "0001-Maintain-current-contextIsolation-behavior-across-el.patch"
-        "0001-Don-t-use-electron-to-build.patch")
+        "0001-Don-t-use-electron-to-build.patch"
+        "add_riscv_support.patch")
 sha512sums=('413b6e952eea8cfc4f7a8a32c62f0fb989a569284c35a3a1d6de9be42e3fe0bc2008899fb534b8b8a4bc2f0432783b7caa13ae3969568c4743f984dc8174a537'
             'SKIP'
             'bdecad57f8703eb6011b0c9ff238b9f3d78192031adfb5a27a7e6910119c600726b87aaacada88877eaeddcd6134fbe72b014bcaeb6cc7563c3753fcb33c52c9'
             '5cb3d93c4bb468fd47e1670f73cc3b8be14977ccf563a47d776285b6b9d96042647062427e58bd981727362caa23d17acdc0152dfd915efeeb6c4e76ca76ca65'
-            'f759f69b774200261687842c66902a3c45638bb012ced9bd83d27491c8a0bbc0bc744dbb83ad7672d443f32d7303c15aeb0ded7a1a0d7ab56e7b1c1b64e263c3')
+            'f759f69b774200261687842c66902a3c45638bb012ced9bd83d27491c8a0bbc0bc744dbb83ad7672d443f32d7303c15aeb0ded7a1a0d7ab56e7b1c1b64e263c3'
+            '2c6d2b40b63258d767a7a4ee3f5be915ce3c23f0b4ed492c762946fd33b6e90881584654010cd71b475122814aa3cb64589f7aa397e6cc22b98ed0c48fbb6816')
 b2sums=('8a2485f0307749fcdf1887abcba508bf36f1a71d09d4f0ba14f2f5323bfd2485f8ac41b25ede49445f7d89b704c9f92fb67d53b86fe90b9e0e07bebd2c0f6060'
         'SKIP'
         '1ecf08e4a0b2836828fc4f9283a69e1fbda6f268195ae56433050d54070038aacf1ae3057202324c678b242c88345cc8f453c7798129753f083c33db256517a4'
         '67f416c59b549606d037167c796c5b99f0ca9158fb767d52c101f41db6d31927b57618efb4d6eaf590cbd3c7aab4dd71a64bc3b52580e2383f5389c95264a499'
-        '02d7876fb0d68b05e9ee262925a063049bfa2e5e063789008c4c74e406084d4441d2c860ecb2b08596f9d9a8c9a6e9136a2cdbd8312bcbf376d9d6e6c811bcfc')
+        '02d7876fb0d68b05e9ee262925a063049bfa2e5e063789008c4c74e406084d4441d2c860ecb2b08596f9d9a8c9a6e9136a2cdbd8312bcbf376d9d6e6c811bcfc'
+        'cec31649325a6d6dcf13c42a639bf05691e4ccb678f2ea3476087277b2e5a2136c00e02b8ac29e6aaf1c0b6e80f093ba4fef182ea0a0726882f68c67a52bfbff')
 validpgpkeys=('222B85B0F90BE2D24CFEB93F47484E50656D16C7') # Keybase.io Code Signing (v1) <code@keybase.io>
 
 prepare() {
     cd client-v${pkgver}
+    patch -Np1 -i ../add_riscv_support.patch
 
     export GOPATH="${srcdir}/.gopath"
     mkdir -p "${GOPATH}"/src/github.com/keybase
