Index: VirtualGL-2.6.4/include/boost/endian.h
===================================================================
--- VirtualGL-2.6.4.orig/include/boost/endian.h
+++ VirtualGL-2.6.4/include/boost/endian.h
@@ -117,6 +117,7 @@
    || defined(__x86_64) || defined(__x86_64__) \
    || defined(_M_X64) || defined(__bfin__) \
    || defined(__ARMEL__) || defined(__aarch64__) \
+   || defined(__riscv) \
    || (defined(_WIN32) && defined(__ARM__) && defined(_MSC_VER)) /* ARM Windows CE don't define anything reasonably unique, but there are no big-endian Windows versions */
 
 # define BOOST_LITTLE_ENDIAN
