diff --git PKGBUILD PKGBUILD
index 8bcf1f2b..c5492cd8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,12 +11,19 @@ url="https://virtualgl.org"
 license=('LGPL')
 depends=('libxv' 'glu' 'libjpeg-turbo' 'xcb-util-keysyms' 'opencl-driver' 'ocl-icd' 'xorg-xhost')
 makedepends=('cmake' 'mesa' 'libxtst' 'opencl-headers' 'ninja')
-source=("https://downloads.sourceforge.net/project/virtualgl/${pkgver}/VirtualGL-${pkgver}.tar.gz"{,.sig})
+source=("https://downloads.sourceforge.net/project/virtualgl/${pkgver}/VirtualGL-${pkgver}.tar.gz"{,.sig}
+        "riscv.patch")
 sha512sums=('9d2f56d44dba936f653ecf0350c581298c4c51d6deb114570df32c044991a61313d59fb140517864e760bc6f7fd93bd27ffd35e2ff8a024243472a822d5cfd98'
-            'SKIP')
+            'SKIP'
+            '177b416c353f9c76e1c2fa9c4db9f0dfa2f2b9d1fc24c2cc98d8e9a829b66ed1504639088b6e90fa47cc333348181223d0c1e9dc854a614b01568992990e54b8')
 validpgpkeys=('AE1A7BA4EFFF9A9987E1474C4BACCAB36E7FE9A1') # The VirtualGL Project (Signing key for official binaries) <information@VirtualGL.org>
 
 prepare() {
+  cd VirtualGL-$pkgver
+
+  # Patch from openSUSE
+  patch -Np1 -i $srcdir/riscv.patch
+
   # FS#48403
   export LDFLAGS="${LDFLAGS/--as-needed,}"
 }
