diff --git PKGBUILD PKGBUILD
index 20705749a..1934567b0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ arch=('any')
 url='https://github.com/dtkav/swagger_ui_bundle'
 license=('Apache')
 depends=('python-jinja')
-makedepends=('python-setuptools')
+makedepends=('python-pip' 'python-setuptools')
 checkdepends=('python-pytest-runner' 'python-pytest-flake8')
 source=("$pkgname-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
 sha512sums=('2172bc039d1b4ad603412b1298211d738384d94df47612fb384da10fb69bf71f6b37c5afdcbe94dbe7b5c4df1495d43801cc11b2d4bebc8783f4d3442676f62f')
