diff --git PKGBUILD PKGBUILD
index 894b23b3..89a0914a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,12 +10,13 @@
 arch=(x86_64)
 url='https://gnucash.org/'
 license=(GPL)
-depends=(aqbanking gwenhywfar boost-libs guile libdbi webkit2gtk)
-makedepends=(boost cmake gtest libdbi-drivers mariadb-libs postgresql-libs swig)
+depends=(aqbanking gwenhywfar boost-libs guile libdbi libofx webkit2gtk)
+makedepends=(boost cmake gtest libdbi-drivers mariadb-libs postgresql-libs python swig)
 optdepends=('gnucash-docs: documentation'
             'libdbi-drivers: open and save data into SQL database'
             'perl-date-manip: stock information lookups'
-            'perl-finance-quote: stock information lookups')
+            'perl-finance-quote: stock information lookups'
+            'python: for python bindings')
 options=(!emptydirs)
 source=(https://downloads.sourceforge.net/$pkgname/$pkgname-$pkgver.tar.bz2)
 sha256sums=('7fd4bb919f6e39388ab47bfa11af6fa3f59e9b9bd695c7d4e9208af85cb9c93b')
