diff --git PKGBUILD PKGBUILD
index 2a55873..c25d730 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,13 @@ arch=(x86_64)
 license=(GPL LGPL FDL)
 groups=(kde-applications kde-education)
 depends=(analitza libspectre ktexteditor knewstuff libqalculate qt5-xmlpatterns qt5-tools qt5-webengine poppler-qt5)
-makedepends=(extra-cmake-modules python kdoctools luajit r julia)
+makedepends=(extra-cmake-modules python kdoctools luajit r)
 optdepends=('maxima: Maxima backend'
             'octave: Octave backend'
             'r: R backend'
             'luajit: LUA backend'
             'python: Python backend'
-            'sagemath: SageMath backend'
-            'julia: Julia backend')
+            'sagemath: SageMath backend')
 source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
 sha256sums=('f92de55a7ffe84f8e09e56a194e5cf6a2b1696893aa1d5892c47e1b6a8b23814'
             'SKIP')
