diff --git PKGBUILD PKGBUILD
index 8f72590..27f2b1a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,13 @@ arch=(x86_64)
 license=(GPL LGPL FDL)
 groups=(kde-applications kde-education)
 depends=(analitza libspectre ktexteditor5 knewstuff5 libqalculate qt5-xmlpatterns qt5-tools qt5-webengine poppler-qt5)
-makedepends=(extra-cmake-modules python kdoctools5 luajit r julia)
+makedepends=(extra-cmake-modules python kdoctools5 luajit r)
 optdepends=('maxima: Maxima backend'
             'octave: Octave backend'
             'r: R backend'
             'luajit: LUA backend'
             'python: Python backend'
-            'sagemath: SageMath backend'
-            'julia: Julia backend')
+            'sagemath: SageMath backend')
 source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
 sha256sums=('1cc1f68ad6277bbb22472bdf868966506a02098f0100ec1bfb0e0f88b31fb961'
             'SKIP')
