diff --git PKGBUILD PKGBUILD
index 67c6106..74936e5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,7 +40,6 @@ depends=(gcc-libs
          syntax-highlighting)
 makedepends=(analitza
              extra-cmake-modules
-             julia
              kdoctools
              luajit
              python
@@ -51,8 +50,7 @@ optdepends=('analitza: KAlgebra backend'
             'r: R backend'
             'luajit: LUA backend'
             'python: Python backend'
-            'sagemath: SageMath backend'
-            'julia: Julia backend')
+            'sagemath: SageMath backend')
 groups=(kde-applications
         kde-education)
 source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
