diff --git PKGBUILD PKGBUILD
index b8189148..2bc89ae8 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,13 @@
 license=(GPL LGPL FDL)
 groups=(kde-applications kde-education)
 depends=(analitza libspectre kpty ktexteditor knewstuff libqalculate qt5-xmlpatterns qt5-tools qt5-webengine poppler-qt5)
-makedepends=(extra-cmake-modules python kdoctools luajit r julia)
+makedepends=(extra-cmake-modules python kdoctools luajit r)
 optdepends=('maxima: Maxima backend'
             'octave: Octave backend'
             'r: R backend'
             'luajit: LUA backend'
             'python: Python backend'
-            'sagemath: SageMath backend'
-            'julia: Julia backend')
+            'sagemath: SageMath backend')
 source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
 sha256sums=('b677e2c0b68f7399617fb8ee010375ddf562d01b17faaaef1b00f8cdb1a5922b'
             'SKIP')
