--- PKGBUILD
+++ PKGBUILD
@@ -11,14 +11,13 @@ arch=(x86_64)
 license=(GPL LGPL FDL)
 groups=(kde-applications kde-education)
 depends=(analitza5 libspectre ktexteditor5 knewstuff5 libqalculate qt5-xmlpatterns qt5-tools qt5-webengine poppler-qt5)
-makedepends=(extra-cmake-modules python kdoctools5 luajit r julia)
+makedepends=(extra-cmake-modules python kdoctools5 luajit r)
 optdepends=('maxima: Maxima backend'
             'octave: Octave backend'
             'r: R backend'
             'luajit: LUA backend'
             'python: Python backend'
-            'sagemath: SageMath backend'
-            'julia: Julia backend')
+            'sagemath: SageMath backend')
 source=(https://download.kde.org/stable/release-service/$pkgver/src/$pkgname-$pkgver.tar.xz{,.sig})
 sha256sums=('9cade872dbe83c5fe134b2d4b0d4ac3cdfd2209686b1c13c61776657e4fd2a01'
             'SKIP')
