--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@
 license=('MIT')
 depends=('python' 'python-typing_extensions' 'python-fastapi' 'python-starlette' 'python-django' 'python-aiohttp' 'python-sanic')
 makedepends=('python-build' 'python-installer' 'python-hatchling' 'python-wheel')
-checkdepends=('python-pytest' 'python-pytest-asyncio' 'python-pytest-django' 'python-httpx' 'python-python-multipart')
+checkdepends=('python-pytest' 'python-pytest-asyncio' 'python-pytest-django' 'python-httpx' 'python-httpx2' 'python-python-multipart')
 source=("${pkgname}-${pkgver}.tar.gz::${url}/archive/refs/tags/${pkgver}.tar.gz")
 sha256sums=('df296d87baa3a3c388ffe47397c7b12494568f9069790a66ed55e9bfbea9d796')
 
