diff --git a/setup.py b/setup.py
index 0d99aae..7d777aa 100644
--- a/setup.py
+++ b/setup.py
@@ -338,29 +338,6 @@ PACKAGES = [
 ]
 
 
-# The purpose of this function is to follow the guidance suggested here:
-#
-#   http://packages.python.org/distribute/python3.html#note-on-compatibility-with-setuptools
-#
-# The guidance is for better compatibility when using setuptools (e.g. with
-# earlier versions of Python 2) instead of Distribute, because of new
-# keyword arguments to setup() that setuptools may not recognize.
-def get_extra_args():
-    """
-    Return a dictionary of extra args to pass to setup().
-
-    """
-    extra = {}
-    # TODO: it might be more correct to check whether we are using
-    #   Distribute instead of setuptools, since use_2to3 doesn't take
-    #   effect when using Python 2, even when using Distribute.
-    if py_version >= (3, ):
-        # Causes 2to3 to be run during the build step.
-        extra['use_2to3'] = True
-
-    return extra
-
-
 def main(sys_argv):
 
     # TODO: use the logging module instead of printing.
@@ -378,7 +355,6 @@ def main(sys_argv):
 
     long_description = read(RST_DESCRIPTION_PATH)
     template_files = ['*.mustache', '*.txt']
-    extra_args = get_extra_args()
 
     setup(name='pystache',
           version=VERSION,
@@ -405,7 +381,6 @@ def main(sys_argv):
             ],
           },
           classifiers = CLASSIFIERS,
-          **extra_args
     )
 
 
