Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -33,7 +33,7 @@
 provides=(qt5-python-bindings)
 makedepends=('sip' 'pyqt-builder' 'python-opengl' 'python-dbus'
              'qt5-connectivity' 'qt5-multimedia' 'qt5-tools' 'qt5-serialport' 'qt5-speech' 'qt5-svg'
-             'qt5-webkit' 'qt5-websockets' 'qt5-x11extras' 'qt5-xmlpatterns' 'qt5-remoteobjects' 'qt5-quick3d')
+             'qt5-websockets' 'qt5-x11extras' 'qt5-xmlpatterns' 'qt5-remoteobjects' 'qt5-quick3d')
 conflicts=('pyqt5-common')
 source=("https://pypi.python.org/packages/source/P/PyQt5/PyQt5-$pkgver.tar.gz")
 sha256sums=('2a69597e0dd11caabe75fae133feca66387819fc9bc050f547e5551bce97e5be')
