diff --git PKGBUILD PKGBUILD
index ed5fb62b..6a5c74d5 100644
--- PKGBUILD
+++ PKGBUILD
@@ -38,7 +38,7 @@
 
 build() {
   cd $pkgbase-$pkgver
-  ./configure --prefix=/usr --with-gui=qt5 --with-incompatible-bdb #LDFLAGS="-L$srcdir/db4/lib/ $LDFLAGS" CPPFLAGS="-I$srcdir/db4/include/ $CPPFLAGS"
+  ./configure --prefix=/usr --with-gui=qt5 --with-incompatible-bdb --with-asm=no #LDFLAGS="-L$srcdir/db4/lib/ $LDFLAGS" CPPFLAGS="-I$srcdir/db4/include/ $CPPFLAGS"
   make
 }
 
