--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ url='https://github.com/sebasguts/JuPyMake'
 license=(GPL2)
 depends=(python polymake)
 makedepends=(python-build python-installer python-setuptools python-wheel)
-source=(https://pypi.python.org/packages/source/${_name:0:1}/${_name}/${_name}-${pkgver}.tar.gz)
-sha256sums=('4c756cd6a6056a20cd71f7b5e463c79dad0257241bbd32a40b750882fafdb545')
+source=(https://pypi.python.org/packages/source/${_name:0:1}/${_name}/${_name}-${pkgver}.tar.gz
+        jupymake-fix-compiler-cmdline.patch::https://patch-diff.githubusercontent.com/raw/sebasguts/JuPyMake/pull/5.patch)
+sha256sums=('4c756cd6a6056a20cd71f7b5e463c79dad0257241bbd32a40b750882fafdb545'
+            'd379f8b7e61f535653d5a43d2084a51414d63185317ec4fdeec04425021f6b5c')
+
+prepare() {
+  cd $_name-$pkgver
+  patch -Np1 -i ../jupymake-fix-compiler-cmdline.patch
+}
 
 build() {
   cd $_name-$pkgver
