--- PKGBUILD
+++ PKGBUILD
@@ -15,8 +15,11 @@ sha256sums=('d509de3528a2d5c1ee78bf0750ba768f1c9c7cabf1e45d366cb1aed64be514fa')
 b2sums=('eda1d94a0819e34bb743c8c909ee27a8cd3988ad0e5731d61446c956e97d4158f385374f0c7280c04a587b7ae757f3a17158653407d851fed1073c69dc2d5a40')
 
 prepare() {
-  cd rustsec-${pkgname}-v${pkgver}/${pkgname}
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  cd rustsec-${pkgname}-v${pkgver}
+  echo -e "\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cd ${pkgname}
+  cargo fetch --locked
 }
 
 build() {
