diff --git a/meson.build b/meson.build
index 2c5f7df..246a846 100644
--- a/meson.build
+++ b/meson.build
@@ -70,6 +70,9 @@ ui_srcs = ['sherlock_nk.c',
 c_args = [
 	'-fvisibility=hidden']
 
+link_args = [
+	'-pthread']
+
 if host_machine.system() == 'windows'
 	conf_data.set('UI_TYPE', 'WindowsUI')
 elif host_machine.system() == 'darwin'
@@ -84,7 +87,8 @@ mod = shared_module('sherlock', dsp_srcs,
 	name_prefix : '',
 	dependencies : dsp_deps,
 	install : true,
-	install_dir : inst_dir)
+	install_dir : inst_dir,
+	link_args : link_args)
 
 ui = shared_module('sherlock_nk', ui_srcs,
 	c_args : c_args,
