diff --git PKGBUILD PKGBUILD
index 477b21d..09b55ec 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,9 +12,16 @@ license=('MIT')
 depends=('glibc' 'perl-image-exiftool')
 makedepends=('go')
 source=("${pkgname}-${pkgver}.tar.gz::https://gitlab.com/api/v4/projects/gitlab-org%2Fgitlab-workhorse/repository/archive?sha=v${pkgver}"
-        "gitlab-workhorse.service")
+        "gitlab-workhorse.service"
+        "update_go_mod.patch")
 sha512sums=('724f8901bc3fb93cbe40104d4b5e5072dcf24b81b5520fbe91b6b2c91b9ff869edf6398acbf9d04a2c95a19b50888c975552fd1f369cc4787eb8dc78bf2eded6'
-            '893527f270179fc78ff60e33f0b44abd984d1e9859dd5bf59fea77245ddd4c1c252d2c7882719366944d0f9341a90b5e077470b15afcc3df419b0d0a9f2e6d56')
+            '893527f270179fc78ff60e33f0b44abd984d1e9859dd5bf59fea77245ddd4c1c252d2c7882719366944d0f9341a90b5e077470b15afcc3df419b0d0a9f2e6d56'
+            'a969f307e1b74849a9e2436f5f85a778839c6cbdaab14f073705fe44f580f7f8c0c57b10f10536c8b5fcc90e6e17c2c4cba2c7f433aa4b51bc9a97adef45dcd6')
+
+prepare() {
+  cd "${pkgname}-v${pkgver}-"*
+  patch -Np1 -i "../update_go_mod.patch"
+}
 
 build() {
   cd "${pkgname}-v${pkgver}-"*
