diff --git a/modules/juce_core/juce_core.h b/modules/juce_core/juce_core.h
index 97c86bc..e4ebc2e 100644
--- a/modules/juce_core/juce_core.h
+++ b/modules/juce_core/juce_core.h
@@ -42,7 +42,7 @@
   dependencies:
   OSXFrameworks:      Cocoa Foundation IOKit
   iOSFrameworks:      Foundation
-  linuxLibs:          rt dl pthread
+  linuxLibs:          rt dl -pthread
   mingwLibs:          uuid wsock32 wininet version ole32 ws2_32 oleaut32 imm32 comdlg32 shlwapi rpcrt4 winmm
 
  END_JUCE_MODULE_DECLARATION
