--- PKGBUILD
+++ PKGBUILD
@@ -41,19 +41,20 @@ build() {
   export CGO_ENABLED=1
   export GOPATH="${srcdir}"
   export GOFLAGS="-buildmode=pie -mod=readonly -modcacherw"
+  export _GOARCH=$(go env GOARCH)
 
   make VERSION=$pkgver GO_LDFLAGS="${common_ldflags[*]}" -C $pkgbase-$pkgver
 
   mkdir -vp completions
-  $pkgbase-$pkgver/build/bin/linux/amd64/crictl completion bash > completions/crictl
-  $pkgbase-$pkgver/build/bin/linux/amd64/crictl completion zsh > completions/_crictl
-  $pkgbase-$pkgver/build/bin/linux/amd64/crictl completion fish > completions/crictl.fish
+  $pkgbase-$pkgver/build/bin/linux/$_GOARCH/crictl completion bash > completions/crictl
+  $pkgbase-$pkgver/build/bin/linux/$_GOARCH/crictl completion zsh > completions/_crictl
+  $pkgbase-$pkgver/build/bin/linux/$_GOARCH/crictl completion fish > completions/crictl.fish
 }
 
 package_crictl() {
   pkgdesc="A CLI for CRI-compatible container runtimes"
 
-  install -vDm 755 $pkgbase-$pkgver/build/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin/"
+  install -vDm 755 $pkgbase-$pkgver/build/bin/linux/$_GOARCH/$pkgname -t "$pkgdir/usr/bin/"
   # shell completion
   install -vDm 644 completions/$pkgname -t "$pkgdir/usr/share/bash-completion/completions/"
   install -vDm 644 completions/_$pkgname -t "$pkgdir/usr/share/zsh/site-functions/"
@@ -67,7 +68,7 @@ package_crictl() {
 package_critest() {
   pkgdesc="A benchmarking CLI for CRI-compatible container runtimes"
 
-  install -vDm 755 $pkgbase-$pkgver/build/bin/linux/amd64/$pkgname -t "$pkgdir/usr/bin/"
+  install -vDm 755 $pkgbase-$pkgver/build/bin/linux/$_GOARCH/$pkgname -t "$pkgdir/usr/bin/"
   # docs
   install -vDm 644 $pkgbase-$pkgver/docs/{benchmark,validation}.md -t "$pkgdir/usr/share/doc/$pkgname/"
   install -vDm 644 $pkgbase-$pkgver/{{CHANGELOG,CONTRIBUTING,README,code-of-conduct}.md,SECURITY_CONTACTS} -t "$pkgdir/usr/share/doc/$pkgname/"
