--- PKGBUILD
+++ PKGBUILD
@@ -14,7 +14,7 @@ checkdepends=('python-pytest-runner' 'python-sphinx' 'python-pytest-cov' 'python
               'python-responses' 'python-freezegun' 'python-sqlalchemy' 'python-sqlalchemy-utils'
               'python-flask-sqlalchemy' 'python-flask-login' 'python-flask-caching'
               'python-flask-sphinx-themes' 'python-sphinxcontrib-seqdiag' 'python-betamax'
-              'python-pyjwt' 'python-pytest-mock')
+              'python-pyjwt' 'python-pytest-mock' 'python-setuptools-rust')
 source=("https://github.com/singingwolfboy/flask-dance/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
 sha512sums=('3a72a193dca0db2a0a19a33cff68a7f479e56e6ac3652df32caf5629bf5794884f8e55e3e2976b4c2a969c5d2a019cf8fa1a4b6c0730247f2b0397d7531ee98f')
 
