Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 452991)
+++ PKGBUILD	(working copy)
@@ -15,6 +15,11 @@
 sha512sums=('3aa3cb46ddd456532a009fb9cfcd746971396be33e03e52a15b754a6d7683f4efd020edb0ec4eb36d22ba20f050aaac4ba6cdd3b69bb5701ea58ddb9a903c59d'
             'SKIP')
 
+prepare() {
+  cd "${srcdir}/${pkgname}-${pkgver}/"
+  cp -v "/usr/share/autoconf/build-aux/config.guess" "/usr/share/autoconf/build-aux/config.sub" autoconf/
+}
+
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   ./configure --prefix=/usr --sysconfdir=/etc
