--- PKGBUILD
+++ PKGBUILD
@@ -15,7 +15,7 @@ b2sums=('d2cdf274baf9cf8890ee4c5c434448a34bc6d3d8967df6e2e9334fe1eff66ce53715973
 prepare() {
   setconf config USE_OPENMP 0
   setconf config USE_SHARED 1
-  setconf config USE_SSE 1
+  setconf config USE_SSE 0
   setconf libsquish.pc.in Version "$pkgver"
 }
 
@@ -25,6 +25,7 @@ build() {
     -D CMAKE_BUILD_TYPE=Release \
     -D CMAKE_INSTALL_PREFIX=/usr \
     -D BUILD_SHARED_LIBS=ON \
+    -D BUILD_SQUISH_WITH_SSE2=OFF \
     -G Ninja \
     -S .
   ninja -C build
