--- PKGBUILD
+++ PKGBUILD
@@ -68,6 +68,9 @@
   mv "$srcdir/wxpdfdoc-${_wxpdfdoc_version}" wxpdfdoc-src
   mv "$srcdir/lobster-${_lobster_version}" lobster-src
   mv "$srcdir/Monocypher-${_monocypher_version}" monocypher-src
+
+  cp "$srcdir/tinycc-$_tinycc_commit"/riscv64* lobster-src/dev/external/libtcc/
+  patch -Np1 -d lobster-src -i "$srcdir/lobster-riscv64.patch"
 }
 
 build() {
@@ -94,3 +97,11 @@
   # license
   install -vDm644 -t "$pkgdir/usr/share/licenses/$pkgname" "$pkgname/ZLIB_LICENSE.txt"
 }
+
+_tinycc_commit=64cf0b816bb72ba9b1e6874ade3a3a3339bc4368
+source+=("tinycc-$_tinycc_commit.tar.gz::https://github.com/TinyCC/tinycc/archive/$_tinycc_commit.tar.gz"
+         lobster-riscv64.patch)
+sha512sums+=('1937990edf0c917943bc39fc2d1938cfd183ca2701f24036e1d6d991d4f618f74a88db939eb84fb82aa49a0b10a664adc8fc422ca5b5b887e7209d108138ea56'
+             '909724505e4d56bd32568dde32fb08a8c0e1e22a15627be4f5655a10a9d32834b5b878c3bb9086ee77d63cef36cb0c4ba59bc3d6c0a3270aed83864c35903ec5')
+b2sums+=('5f2327c40e7c4e714a2ed89dce816e86c98a30dddc96cd98865d49ea2194d1af2055aa44dd4a0c2cbddcdd50eb933ee3aea6a50b2f460cdec438cc8e8cf3310b'
+         '5a183ee7c71b351ee702fb0af62331039a18b1fcfe4f5ae1ab2374a558bceaf339e488da18503f01adcfd047adebbb65ae8c3b901fc4260cfef1b02365c41752')
