diff --git a/tests/test_requirements.py b/tests/test_requirements.py
index 8cc45687..f2c209c4 100644
--- a/tests/test_requirements.py
+++ b/tests/test_requirements.py
@@ -192,4 +192,6 @@ def test_sys_platform_linux_in(self):
     def test_parseexception_error_msg(self):
         with pytest.raises(InvalidRequirement) as e:
             Requirement("toto 42")
-        assert "Expected stringEnd" in str(e.value)
+        assert "Expected stringEnd" in str(e.value) or (
+            "Expected string_end" in str(e.value)  # pyparsing>=3.0.0
+        )
