diff --git PKGBUILD PKGBUILD
index d692b3f4..eeedb04a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,13 +11,15 @@ license=('Apache')
 makedepends=('python-setuptools' 'python2-setuptools' 'python-pyparsing' 'python2-pyparsing')
 checkdepends=('python-pytest' 'python-pretend')
 source=("https://github.com/pypa/packaging/archive/$pkgver/$pkgbase-$pkgver.tar.gz"
-        replace-distutils-usage-with-sysconfig.patch)
+        replace-distutils-usage-with-sysconfig.patch
+        "fix-test-with-pyparsing-3.patch")
 sha512sums=('fb71f1036cfaacbe94fdee663af31d6ad1960f73ecc95cba87b461c2d7d2ea90085853bb4682b146492d8c48f784b60ef082e3b1259269857166b143cd9a920b'
             '015ddcb799259190e3a4b97d386fb8e4cb6f76a22eed2ce97babf10116e886b82f6f3e3e74e3590dd14a8fce8e6ca5980a91205c61e29afa5dbdc387f4daa8dd')
 
 prepare() {
   cd packaging-$pkgver
   patch -Np1 -i ../replace-distutils-usage-with-sysconfig.patch
+  patch -Np1 -i ../fix-test-with-pyparsing-3.patch
 }
 
 build() {
