diff --git PKGBUILD PKGBUILD
index 55e891c..326ce8d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -19,7 +19,7 @@ validpgpkeys=(CE4A4D080F0D304F23B9EBDD972D5BF4DC613806  # John W. Linville <linv
 
 build() {
     cd $pkgname-$pkgver
-    ./configure --prefix=/usr --mandir=/usr/share/man --sbindir=/usr/bin
+    ./configure --prefix=/usr --mandir=/usr/share/man --sbindir=/usr/bin --enable-netlink=no
     make
 }
 
