--- PKGBUILD
+++ PKGBUILD
@@ -16,6 +16,14 @@ makedepends=(gcc-fortran
 source=(git+https://github.com/coin-or/Osi#tag=releases/$pkgver)
 sha256sums=('596856f51b4dc5296e40703e04f62fd64e088445866148fd24c3875a894b49b7')
 
+prepare() {
+  cd Osi
+
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} ./
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} ./Osi
+}
+
+
 build() {
   cd Osi
   ./configure --prefix=/usr --enable-dependency-linking \
