--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,12 @@ makedepends=(gcc-fortran)
 source=($pkgname-$pkgver.tar.gz::https://github.com/coin-or/Osi/archive/refs/tags/releases/$pkgver.tar.gz)
 sha256sums=('8b09802960d7d4fd9579b3e4320bfb36e7f8dca5e5094bf1f5edf1b7003f5562')
 
+prepare() {
+  cd Osi-releases-$pkgver
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} ./
+  cp /usr/share/autoconf/build-aux/config.{guess,sub} ./Osi
+}
+
 build() {
   cd Osi-releases-$pkgver
   ./configure --prefix=/usr --enable-dependency-linking \
