--- PKGBUILD
+++ PKGBUILD
@@ -21,7 +21,9 @@ b2sums=('eeb1a179ea361444551d9f618076704bbc5b0cca188fe7a48fac3df9f98246ae2b88a9e
 prepare() {
   cd "${pkgname}-${pkgver}"
   patch -Np1 -i ../$pkgname-fix-test-for-cargo-1.60.x.patch
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
