--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,9 @@ b2sums=('9c8662232b2296e1c12a3e303477b9588ac20cb41d950b13228ca45a73c13f4572b6ad0
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
