diff --git PKGBUILD trunk/PKGBUILD
index beb3f7c1..24e66b7f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,9 @@ b2sums=('0f63b41845ec0e3d4ba41fa3f1fdc4a4703103a1f89dc90bf91536b84b8079c6ffdd50a
 
 prepare() {
   cd "${pkgname}-${pkgver}"
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
