--- PKGBUILD
+++ PKGBUILD
@@ -17,7 +17,9 @@ b2sums=('4714269414ecc7509d8ee9dd28d705e56dc3b74486256ed95a8c93c86c6594ea1d708d5
 prepare() {
   cd "${pkgname}-${pkgver}"
   sed -i '/\"vendored-libgit2\"/d' Cargo.toml
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
