diff --git a/CMakeLists.txt b/CMakeLists.txt
index 149ba07..8e3bb2f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -330,11 +330,12 @@ TARGET_LINK_LIBRARIES(gimagereader
     ${ENCHANT_LDFLAGS}
     ${PODOFO_LDFLAGS}
     -ldl
+    "-pthread"
 )
 
-FIND_PACKAGE(OpenMP REQUIRED)
-SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+#FIND_PACKAGE(OpenMP REQUIRED)
+#SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+#SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 
 SET_TARGET_PROPERTIES(gimagereader PROPERTIES OUTPUT_NAME gimagereader-${INTERFACE_TYPE})
 IF("${srcdir}" STREQUAL "qt")
