diff --git PKGBUILD PKGBUILD
index bcdabe1..101254c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -13,12 +13,15 @@ depends=('djvulibre' 'gtksourceviewmm' 'gtkspellmm' 'libtesseract.so' 'libxml++'
          'poppler-qt6' 'qt6-imageformats' 'qtspell' 'quazip-qt6' 'sane' 'tesseract')
 makedepends=('cmake' 'intltool' 'ninja' 'python-gobject')
 source=("$url/releases/download/v$pkgver/$pkgbase-$pkgver.tar.xz"
-        'openmp.patch')
+        'openmp.patch'
+        "disable-openmp.patch")
 sha256sums=('b55cfe9212fa16bf2d5a7ca0fc549454812b265ac7d4aa34697947b20eecf425'
-            'ee84de7fee390ee54152b4e332e50d8f585b0c9a3aa8b40a04f865b5250607c2')
+            'ee84de7fee390ee54152b4e332e50d8f585b0c9a3aa8b40a04f865b5250607c2'
+            'cdfb56d9bfd8bbffa8e8298bf65531d6c50bcb9474a945033b789dd740eedcfd')
 
 prepare() {
   cd $pkgbase-$pkgver
+  patch -Np1 -i "../disable-openmp.patch"
 
   # Run OpenMP once before calling tesseract
   # https://github.com/manisandro/gImageReader/issues/285
