diff --git PKGBUILD PKGBUILD
index 0bfa05d..7042d49 100755
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,9 @@ pkgdesc='A Sega Saturn emulator'
 arch=('x86_64')
 url="https://yabause.org/"
 license=('GPL')
+options=(!lto)
 depends=('freeglut' 'glew' 'gtkglext' 'openal' 'sdl2')
-makedepends=('cmake')
+makedepends=('cmake' 'clang')
 conflicts=('yabause-qt')
 source=("https://download.tuxfamily.org/yabause/releases/${pkgver}/yabause-${pkgver}.tar.gz"
         'rwx.patch')
@@ -35,6 +36,7 @@ build() {
   mkdir build && cd build

   cmake .. \
+    -DCMAKE_C_COMPILER=clang \
     -DCMAKE_BUILD_TYPE='Release' \
     -DCMAKE_INSTALL_PREFIX='/usr' \
     -DYAB_PORTS='gtk' \
