diff --git PKGBUILD PKGBUILD
index 7bdf8a923..dd73145ea 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,8 +11,15 @@ depends=('gst-plugins-bad' 'gst-plugins-base' 'gst-plugins-good' 'libhandy')
 makedepends=('meson' 'rust')
 optdepends=('gst-plugins-ugly: Extra media codecs'
             'gst-libav: Extra media codecs')
-source=("https://gitlab.gnome.org/World/podcasts/-/archive/$pkgver/podcasts-$pkgver.tar.gz")
-sha256sums=('a3c12c835f00a952ebd40e83be172c7e45cad782f4b2a6d5c8c296a5e20b09ea')
+source=("https://gitlab.gnome.org/World/podcasts/-/archive/$pkgver/podcasts-$pkgver.tar.gz"
+        "fix_positional_arg.patch")
+sha256sums=('a3c12c835f00a952ebd40e83be172c7e45cad782f4b2a6d5c8c296a5e20b09ea'
+            '8a6e9d2e43311d636214529cc580de95fc1e955b619b998036b88d7e41a609a8')
+
+prepare() {
+  cd podcasts-$pkgver
+  patch -Np1 -i "../fix_positional_arg.patch"
+}
 
 build() {
   arch-meson podcasts-$pkgver build
