diff --git PKGBUILD PKGBUILD
index 14c38334..40feb74a 100644
--- PKGBUILD
+++ PKGBUILD
@@ -89,6 +89,9 @@ build() {
     mv ../run/john{,-non-xop}
     ./configure "${options[@]}" CFLAGS="${CFLAGS} -mxop"
     make clean; make
+  elif [[ "${CARCH}" == "riscv64" ]]; then
+    ./configure "${options[@]}" CFLAGS="${CFLAGS/-DCPU_FALLBACK}"
+    make clean; make
   else
     ./configure "${options[@]}" CFLAGS="${CFLAGS}"
     make clean; make
