diff --git PKGBUILD PKGBUILD
index e6d71a1..36485c0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -41,6 +41,9 @@ prepare() {
 	# Give ourselves a way to granularly disable test dependent on installed path
 	sed -i -e '/^markers =/a\    pathdependent: Does not play nice' setup.cfg
 	sed -i -e '/test_do_not_import_files_from_local_directory_with_pythonpath/i\    @pytest.mark.pathdependent' tests/test_self.py
+
+	# Give PyTest more time on this test case.
+	sed -i 's,@pytest.mark.timeout(25),@pytest.mark.timeout(60),g' tests/checkers/unittest_refactoring.py
 }
 
 build() {
