diff --git PKGBUILD PKGBUILD
index 607ff3b..b096670 100644
--- PKGBUILD
+++ PKGBUILD
@@ -43,6 +43,8 @@ prepare() {
 	sed -i -E '/^requires/s/~=[0-9.]+//g' pyproject.toml
 	# Remove overly restrictive dependency pinning that ends up in runtime checks
 	sed -i -e '/>=/{s/>.*//;p}' -e '/python_requires/d' setup.cfg
+	# Give PyTest more time on this test case.
+	sed -i 's,@pytest.mark.timeout(25),@pytest.mark.timeout(60),g' tests/checkers/unittest_refactoring.py
 }

 build() {
