--- PKGBUILD
+++ PKGBUILD
@@ -35,6 +35,7 @@ source=(
   'php-fpm.patch'
   'php.ini.patch'
   'php-legacy.patch'
+  'zend-concat-test.patch'
 )
 sha256sums=('cd450fb4ee50488c5bf5f08851f514e5a1cac18c9512234d9e16c3a1d35781a6'
             'SKIP'
@@ -43,7 +44,8 @@ sha256sums=('cd450fb4ee50488c5bf5f08851f514e5a1cac18c9512234d9e16c3a1d35781a6'
             '702b163c66c65af92dcad8d79f41bda84bcd5d863235fcf1497c33a86db9e4ca'
             '7322cc296780908b55fe6d601d403b0e506a2a7ce5359141cb4132f84778243d'
             '0d651c14cbd76bf430cd9b0c3b3ebf12fd19bf3ca4a18cb6dc329e0352aabe66'
-            'a1859005c9fad835079470e6fb3c0721b9bd024c8f3ae1f34cb082ec41c63194')
+            'a1859005c9fad835079470e6fb3c0721b9bd024c8f3ae1f34cb082ec41c63194'
+            '0a34ca9f1e74112cfecfab613efebca5671f4211d5ae521d5c54b4c9444f3231')
 validpgpkeys=('F1F692238FBC1666E5A5CCD4199F9DFEF6FFBAFD'
               '39B641343D8C104B2B146DC3F9C39DC0B9698544')
 _interpreter=${pkgver%.*}
