--- PKGBUILD
+++ PKGBUILD
@@ -69,6 +69,8 @@
 
 prepare() {
   cd $pkgname/$_name/rust/
+  patch -Np1 -d ../.. -i "$srcdir/suppress-stateful-too-slow-healthcheck.patch"
+  patch -Np1 -d ../.. -i "$srcdir/wait-for-test-threads.patch"
   cargo fetch --locked --target host-tuple
 }
 
@@ -108,3 +110,10 @@
   cd $pkgname/$_name
   python -m installer --destdir="$pkgdir" dist/*.whl
 }
+
+source+=(suppress-stateful-too-slow-healthcheck.patch
+         wait-for-test-threads.patch)
+sha512sums+=('e8df90b55134bfde8a24f7f0513b2f9d39b4c237c34a95fd74787a83f2196d70229d8d9d7ee6dfdb7a01fcb8a1767124772ac22f95fb1b09aacc51f4ec449070'
+             'ec2b8b191956ef7cf41358fa61e77444d7f288d7c2c340db63c72329a4cd4b55e94af4d7558c85f9b4a51f3cbaa6dad92caf7ff443bfcfaa9458ddfdf43568a9')
+b2sums+=('afcaae1f8513549a99d26f16a6e345c19080cc085aad19ed29fcc40b37f56156c5aa80696a9d63e47b5939ea35c833d87986b8b54b7e09b100bd73634ae3346d'
+         'd9e15eb941261e60b7da5d990eb10273eeac038250abaa6fe84abf3eec9538fca0ae22e19d34a64ee7bacd04cea05e8116a7688a5c9e2347e53d839703d6fc5f')
