--- PKGBUILD
+++ PKGBUILD
@@ -45,6 +45,9 @@
 
   # Fix building test with BUILD_STATIC_LIB=OFF
   git apply -3 ../0003-Fix-BUILD_STATIC_LIB-OFF.patch
+
+  # Prevent concurrent protoc invocations from overwriting sources during compilation.
+  patch -Np1 -i ../fix-protobuf-generation-race.patch
 }
 
 build() {
@@ -81,3 +84,6 @@
 }
 
 # vim:set sw=2 sts=-1 et:
+
+source+=(fix-protobuf-generation-race.patch)
+b2sums+=(2bdd3a2888fd570c53a858635ed4c5c6b3577653fdc1269986b936ed7010fb961b3f6507f0dcb59dcdd2be292fb062db20173db41cfd99b2bb395d0b097644ad)
