--- PKGBUILD
+++ PKGBUILD
@@ -57,3 +57,12 @@ package() {
   install -Dm0644 "${srcdir}/depmod.hook" "${pkgdir}/usr/share/libalpm/hooks/60-depmod.hook"
   install -Dm0755 "${srcdir}/depmod.script" "${pkgdir}/usr/share/libalpm/scripts/depmod"
 }
+
+source+=(fix-riscv64-tests.patch)
+sha256sums+=('3319d0e398276d0a5c8ada91e327ed2014041f5eb08d1ea890615dabd29b572a')
+
+prepare() {
+  cd kmod
+  # https://salsa.debian.org/md/kmod/-/blob/627679a9d9cbb8549cb84934ed7d7423d802f7a6/debian/patches/fix_riscv64_tests
+  patch -Np1 -i ../fix-riscv64-tests.patch
+}
