diff --git PKGBUILD PKGBUILD
index c78d0c6db..53f05d2fb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,11 +10,14 @@ license=('GPL2')
 depends=('gtk3' 'python-gobject')
 makedepends=('gobject-introspection' 'meson' 'python-pycodestyle' 'yelp-tools')
 checkdepends=('xorg-server-xvfb')
-source=(https://download.gnome.org/sources/$pkgname/${pkgver%.*}/$pkgname-$pkgver.tar.xz)
-sha512sums=('ff4db968cbd679e2065c614d538187fe3c56118b44638771ec9f7472bdba4a798dc682565f520a8680a8001ea39ce88555b07cda172e60f88eb6bdf1e00b5a6d')
+source=(https://download.gnome.org/sources/$pkgname/${pkgver%.*}/$pkgname-$pkgver.tar.xz
+        "fix_positional_arg.patch")
+sha512sums=('ff4db968cbd679e2065c614d538187fe3c56118b44638771ec9f7472bdba4a798dc682565f520a8680a8001ea39ce88555b07cda172e60f88eb6bdf1e00b5a6d'
+            '0a01fd35a4109f39bcc9192742773c7726bcb912fc7998fd7bf5ef84da2f71c3b44f22bffd506712d081029063be5c0b8cc5c94c2c9167a83c73a09da89ead1b')
 
 prepare() {
   cd $pkgname-$pkgver
+  patch -Np1 -i "../fix_positional_arg.patch"
 }
 
 build() {
