diff --git PKGBUILD PKGBUILD
index b4945693..b849db6c 100644
--- PKGBUILD
+++ PKGBUILD
@@ -17,9 +17,11 @@ depends=('ca-certificates' 'brotli' 'libbrotlidec.so' 'krb5' 'libgssapi_krb5.so'
          'libidn2' 'libidn2.so' 'libnghttp2' 'libpsl' 'libpsl.so' 'libssh2' 'libssh2.so'
          'openssl' 'zlib' 'zstd' 'libzstd.so')
 provides=('libcurl.so')
-source=("https://curl.haxx.se/download/${pkgname}-${pkgver}.tar.gz"{,.asc})
+source=("https://curl.haxx.se/download/${pkgname}-${pkgver}.tar.gz"{,.asc}
+        'curl-add-sched-include.patch::https://github.com/curl/curl/commit/e2e7f54b7bea521fa8373095d0f43261a720cda0.patch')
 sha512sums=('8133baf48dfd93531ce0a226b54cb153fd58bb0c1ffe8159cee0c0aa23ce210192c572e8ee01f3d75a87b609a580e76929df1e66635be59c177b0cb8076043b2'
-            'SKIP')
+            'SKIP'
+            'd7673d2fcb99ebc0d10157fb221499da213aa7a78e0ba44280d1ce7d316d394f1c184fb60214d046dc66e2fd7cc089ae5d3cc4c4d1fdbbec80a5d75eb644dd0e')
 validpgpkeys=('27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2') # Daniel Stenberg
 
 _configure_options=(
@@ -37,6 +39,11 @@ _configure_options=(
   --with-ca-bundle='/etc/ssl/certs/ca-certificates.crt'
 )
 
+prepare() {
+  cd "${srcdir}/${pkgbase}-${pkgver}"
+  patch -p1 < $srcdir/curl-add-sched-include.patch
+}
+
 build() {
   mkdir build-curl{,-compat,-gnutls}
 
