--- PKGBUILD
+++ PKGBUILD
@@ -64,6 +64,8 @@
     -e "/\WLIBCURL_TIMESTAMP\W/c #define LIBCURL_TIMESTAMP \"$(git log -1 --format=%cs "curl-${pkgver//./_}")\"" \
     include/curl/curlver.h
 
+  patch -Np1 -i "$srcdir/riscv64-fixes.patch"
+
   autoreconf -fi
 }
 
@@ -228,3 +230,6 @@
   install -dm 0755 "${pkgdir}"/usr/share/licenses
   ln -s curl "${pkgdir}"/usr/share/licenses/libcurl-gnutls
 }
+
+source+=(riscv64-fixes.patch)
+sha512sums+=('b032bb272d91675e47a00be72510e210241b17c083a7f7dbdf67682254d4c0bb3ffeb767a10e9dab69d1bfbc35a77d335a957f4e160cde706ee51b6332acc8af')
