--- PKGBUILD
+++ PKGBUILD
@@ -30,7 +30,7 @@ makedepends=(
   'nettle'
   'zlib'
   'zstd')
-checkdepends=('valgrind')
+#checkdepends=('valgrind')
 validpgpkeys=('27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2') # Daniel Stenberg
 source=("git+https://github.com/curl/curl.git#tag=curl-${pkgver//./_}?signed")
 sha512sums=('a10e4408f1e89462880646a29b4f947d6b8eaf81d323a86046fe5eb8622005e4499684609f7c352729f5f88933275b650e7285cf2e8299ff966df03f70a33637')
@@ -132,8 +132,9 @@ check() {
   # -am: automake style TAP output
   # -p: print logs if test fails
   # -j: parallelization
+  # -n: don't use valgrind
   # disable test 433, since it requires the glibc debug info
-  make TFLAGS="-v -a -k -p -j$(nproc) !433" test-nonflaky
+  make TFLAGS="-v -a -k -p -j$(nproc) -n !433" test-nonflaky
 }
 
 package_curl() {
