diff --git a/docs/conf.py b/docs/conf.py
index aa10620..0608988 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -297,13 +297,13 @@ pygments_style = 'flask_theme_support.FlaskyStyle'
 # fall back if theme is not there
 try:
     __import__('flask_theme_support')
-except ImportError, e:
-    print '-' * 74
-    print 'Warning: Flask themes unavailable.  Building with default theme'
-    print 'If you want the Flask themes, run this command and build again:'
+except ImportError:
+    print ('-' * 74)
+    print ('Warning: Flask themes unavailable.  Building with default theme')
+    print ('If you want the Flask themes, run this command and build again:')
     print
-    print '  git submodule update --init'
-    print '-' * 74
+    print ('  git submodule update --init')
+    print ('-' * 74)
 
     pygments_style = 'tango'
     html_theme = 'default'
