diff --git PKGBUILD PKGBUILD
index fa4b5b3..f23a9bb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -40,6 +40,7 @@ b2sums=('33bacadd99e9c4d9fd7678dcddc5ea50462530bb877b2b38a4f839073be421a71d5d242
 prepare() {
   cd $pkgbase
   cargo fetch --locked --target "$(rustc --print host-tuple)"
+  patch -Np1 -i ../extend-develop_pyo3_ffi_pure-test-timeout.patch
 }
 
 build() {
@@ -89,3 +90,7 @@ package_python-maturin() {
   rm -vr "$pkgdir/usr/bin"
   install -vDm 644 -t "$pkgdir/usr/share/licenses/$pkgname" license-mit
 }
+
+source+=(extend-develop_pyo3_ffi_pure-test-timeout.patch)
+sha512sums+=('25950af759d4bf9359448fa5224be7e5bacbc9813cb86bb23562dbd61304349e4188d0978752ee82c3663ff6ca39550ec69df31f50b794ee2d9e24c59e8f6651')
+b2sums+=('f53437cabb81a9d1ba2068b7dbd41a4ef9547ae25754a260423bb68b0755e6ad364ed17aaec5499730f4939bc3392ad52c1ad85c183b9e8babf1d905557e5b0d')
