diff --git PKGBUILD PKGBUILD
index ee884de..62afbd0 100644
--- PKGBUILD
+++ PKGBUILD
@@ -38,7 +38,9 @@ b2sums=('98f7f159c28d7b773a8daa103c1d2280a6ceab979d3d564d3c3ae47f9540c6307cfbde0
 
 prepare() {
   cd $pkgbase
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  cargo update -p ring
+  cargo fetch --locked
 }
 
 build() {
