Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422467)
+++ PKGBUILD	(working copy)
@@ -11,7 +11,7 @@
 license=('GPL')
 url='https://www.r-project.org/'
 depends=('lapack' 'libtiff' 'pcre2' 'perl' 'libxmu' 'pango' 'zip' 'unzip' 'curl' 'icu' 'which')
-makedepends=('jdk-openjdk' 'gcc-fortran' 'tk' 'texlive-core')
+makedepends=('gcc-fortran' 'tk' 'texlive-core')
 optdepends=('tk: tcl/tk interface'
             'texlive-bin: latex sty files'
             'gcc-fortran: needed to compile some CRAN packages'
