--- PKGBUILD.orig	2021-10-19 17:29:02.090336845 +0800
+++ PKGBUILD	2021-10-19 17:29:06.797001453 +0800
@@ -14,8 +14,15 @@
 depends=('ncurses')
 optdepends=('words: for hangman')
 makedepends=()
-source=("https://github.com/msharov/$pkgname/archive/v$pkgver.tar.gz")
-sha256sums=('fe4e54fa56aff00ac3a45b2c4a5173bb2627c723ad7ae9d7548e43148b177be6')
+source=("https://github.com/msharov/$pkgname/archive/v$pkgver.tar.gz"
+        "link-atomic.patch")
+sha256sums=('fe4e54fa56aff00ac3a45b2c4a5173bb2627c723ad7ae9d7548e43148b177be6'
+            'f2c448684424ee5b15cc3651943000b61283e4d7e1d75005b1585da756d76d38')
+
+prepare() {
+    cd "$pkgname-$pkgver"
+    patch -Np1 < ${srcdir}/link-atomic.patch
+}
 
 build() {
     cd "$pkgname-$pkgver"
