Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 1014381)
+++ PKGBUILD	(working copy)
@@ -21,6 +21,12 @@
 
 build() {
   cd "$_pkgname-$pkgver"
+  echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
+  echo -e "\ntikv-jemallocator = { git = 'https://github.com/felixonmars/jemallocator' }" >> Cargo.toml
+  sed -i 's/jemallocator = "0.3.2"/tikv-jemallocator = "0.4.1"/' meilisearch-http/Cargo.toml
+  sed -i 's/jemallocator/tikv_jemallocator/g' meilisearch-http/src/main.rs
+
+  cargo update -p ring -p jemallocator
   cargo build --release --locked --all-features
 }
 
