--- a/crates/meilisearch/tests/vector/openai.rs
+++ b/crates/meilisearch/tests/vector/openai.rs
@@ -630,11 +630,11 @@
         }))
         .await;
     snapshot!(code, @"200 OK");
-    snapshot!(json_string!(response["hits"]), @r###"
+    snapshot!(json_string!(response["hits"], { ".**._rankingScore" => meili_snap::insta::rounded_redaction(6) }), @r###"
     [
       {
         "id": 0,
-        "_rankingScore": 0.07944583892822266
+        "_rankingScore": 0.079446
       }
     ]
     "###);
