--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,13 @@
   "${pkgname}::git+$url#commit=$_gitcommit"
 )
 sha256sums=('95a2061038060651b53479e9cd4df79d9401530fd94effdd0f55e5df4d117d08')
+source_riscv64=("$pkgname-zig-0.16.patch")
+sha256sums_riscv64=('12de2626573d5454b7a86e3f88494927a22221a1ebc4cb62d0bf3cefef7efccd')
+
+prepare() {
+  cd "$pkgname"
+  patch -Np1 -i "../$pkgname-zig-0.16.patch"
+}
 
 build() {
   cd "$pkgname"
