--- PKGBUILD
+++ PKGBUILD
@@ -51,6 +51,15 @@ prepare() {
   cd CddInterface-*
   sed -e 's|/usr/include/cdd|/usr/include/cddlib|' -i configure.ac # Fix build with cddlib 0.94k
   rm configure
+  cd ..
+
+# Update the config.guess file
+  vendors=( "xgap-4.30" "DeepThought-1.0.2" "ZeroMQInterface-0.12" )
+  for v in "${vendors[@]}"; do
+      pushd $v
+      autoreconf -fiv
+      popd
+  done
 }
 
 build() {
