--- PKGBUILD
+++ PKGBUILD
@@ -22,7 +22,6 @@ pkgname=(
   qemu-ui-{curses,dbus,egl-headless,gtk,opengl,sdl,spice-{app,core}}
   qemu-user{,-static}{,-binfmt}
   qemu-vhost-user-gpu
-  qemu-vmsr-helper
   qemu-{base,desktop,emulators-full,full}
 )
 pkgver=9.2.3
@@ -124,6 +123,7 @@ makedepends=(
   zlib
   zstd
 )
+options=(!lto)
 source=(
   https://download.qemu.org/qemu-$pkgver.tar.xz{,.sig}
   bridge.conf
@@ -435,13 +435,6 @@ package_qemu-common() {
   # remove unneeded files
   find "$pkgdir" -name .buildinfo -delete
 
-  # remove files provided by seabios
-  rm -fv "$pkgdir/usr/share/$pkgbase/"{bios,vgabios}*
-
-  # remove files provided by edk2-{aarch64,arm,ovmf}
-  rm -fv "$pkgdir/usr/share/$pkgbase/"edk2-*
-  rm -frv "$pkgdir/usr/share/$pkgbase/firmware"
-
   (
     # create man page symlinks for all system emulators
     cd "$pkgdir/usr/share/man/man1"
@@ -502,6 +495,9 @@ package_qemu-common() {
 
     _pick qemu-system-aarch64 usr/bin/qemu-system-aarch64
     _pick qemu-system-aarch64 usr/share/man/man1/qemu-system-aarch64.1*
+    # NOTE: needs to be replaced by edk2, not buildable on riscv64 yet
+    _pick qemu-system-aarch64 usr/share/qemu/firmware/*-aarch64*
+    _pick qemu-system-aarch64 usr/share/qemu/edk2-aarch64-*
 
     _pick qemu-system-alpha usr/bin/qemu-system-alpha
     _pick qemu-system-alpha usr/share/man/man1/qemu-system-alpha.1*
@@ -512,6 +508,9 @@ package_qemu-common() {
     _pick qemu-system-arm usr/share/man/man1/qemu-system-arm.1*
 
     _pick qemu-system-arm-firmware usr/share/qemu/npcm7xx_bootrom.bin
+    # NOTE: needs to be replaced by edk2, not buildable on riscv64 yet
+    _pick qemu-system-arm-firmware usr/share/qemu/firmware/*-arm*
+    _pick qemu-system-arm-firmware usr/share/qemu/edk2-arm-*
 
     _pick qemu-system-avr usr/bin/qemu-system-avr
     _pick qemu-system-avr usr/share/man/man1/qemu-system-avr.1*
@@ -583,6 +582,13 @@ package_qemu-common() {
 
     _pick qemu-system-x86-firmware usr/share/qemu/{kvmvapic,linuxboot,multiboot{,_dma},pvh}.bin
     _pick qemu-system-x86-firmware usr/share/qemu/qboot.rom
+    # NOTE: needs to be replaced by seabios/edk2, not buildable on riscv64 yet
+    _pick qemu-system-x86-firmware usr/share/qemu/firmware/*-i386*
+    _pick qemu-system-x86-firmware usr/share/qemu/firmware/*-x86_64*
+    _pick qemu-system-x86-firmware usr/share/qemu/edk2-i386-*
+    _pick qemu-system-x86-firmware usr/share/qemu/edk2-x86_64-*
+    _pick qemu-system-x86-firmware usr/share/qemu/bios*
+    _pick qemu-system-x86-firmware usr/share/qemu/vgabios*
 
     _pick qemu-system-xtensa usr/bin/qemu-system-xtensa{,eb}
     _pick qemu-system-xtensa usr/share/man/man1/qemu-system-xtensa{,eb}.1*
@@ -610,7 +616,6 @@ package_qemu-common() {
     _pick qemu-vhost-user-gpu usr/lib/qemu/vhost-user-gpu
     _pick qemu-vhost-user-gpu usr/share/qemu/vhost-user/50-qemu-gpu.json
 
-    _pick qemu-vmsr-helper usr/bin/qemu-vmsr-helper
   )
 }
 
@@ -807,7 +812,7 @@ package_qemu-hw-s390x-virtio-gpu-ccw() {
 
 package_qemu-system-aarch64() {
   pkgdesc="QEMU system emulator for AARCH64"
-  depends=("${_qemu_system_deps[@]}" dtc edk2-aarch64 systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" dtc systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -828,7 +833,7 @@ package_qemu-system-alpha-firmware() {
 
 package_qemu-system-arm() {
   pkgdesc="QEMU system emulator for ARM"
-  depends=("${_qemu_system_deps[@]}" dtc edk2-arm qemu-system-arm-firmware=$pkgver-$pkgrel systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" dtc qemu-system-arm-firmware=$pkgver-$pkgrel systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -983,7 +988,7 @@ package_qemu-system-tricore() {
 
 package_qemu-system-x86() {
   pkgdesc="QEMU system emulator for x86"
-  depends=("${_qemu_system_deps[@]}" dtc edk2-ovmf libcbor libcbor.so qemu-system-x86-firmware=$pkgver-$pkgrel seabios systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" dtc libcbor libcbor.so qemu-system-x86-firmware=$pkgver-$pkgrel systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -1227,7 +1232,6 @@ package_qemu-full() {
     qemu-tests=$pkgver-$pkgrel
     qemu-tools=$pkgver-$pkgrel
     qemu-user=$pkgver-$pkgrel
-    qemu-vmsr-helper=$pkgver-$pkgrel
   )
   optdepends=("${_qemu_full_optdepends[@]}")
   provides=(qemu=$pkgver)
