diff --git PKGBUILD PKGBUILD
index 768b1eb..500bd3d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -97,6 +97,7 @@ makedepends=(
   zlib
   zstd
 )
+options=(!lto)
 source=(
   https://download.qemu.org/qemu-$pkgver.tar.xz{,.sig}
   bridge.conf
@@ -362,13 +363,6 @@ package_qemu-common() {
   # remove unneeded files
   find "$pkgdir" -name .buildinfo -delete
 
-  # remove files provided by seabios
-  rm -fv "$pkgdir/usr/share/$pkgbase/"{bios,vgabios}*
-
-  # remove files provided by edk2-{aarch64,arm,ovmf}
-  rm -fv "$pkgdir/usr/share/$pkgbase/"edk2-*
-  rm -frv "$pkgdir/usr/share/$pkgbase/firmware"
-
   (
     # create man page symlinks for all system emulators
     cd "$pkgdir/usr/share/man/man1"
@@ -428,6 +422,9 @@ package_qemu-common() {
 
     _pick qemu-system-aarch64 usr/bin/qemu-system-aarch64
     _pick qemu-system-aarch64 usr/share/man/man1/qemu-system-aarch64.1*
+    # NOTE: needs to be replaced by edk2, not buildable on riscv64 yet
+    _pick qemu-system-aarch64 usr/share/qemu/firmware/*-aarch64*
+    _pick qemu-system-aarch64 usr/share/qemu/edk2-aarch64-*
 
     _pick qemu-system-alpha usr/bin/qemu-system-alpha
     _pick qemu-system-alpha usr/share/man/man1/qemu-system-alpha.1*
@@ -438,6 +435,9 @@ package_qemu-common() {
     _pick qemu-system-arm usr/share/man/man1/qemu-system-arm.1*
 
     _pick qemu-system-arm-firmware usr/share/qemu/npcm7xx_bootrom.bin
+    # NOTE: needs to be replaced by edk2, not buildable on riscv64 yet
+    _pick qemu-system-arm-firmware usr/share/qemu/firmware/*-arm*
+    _pick qemu-system-arm-firmware usr/share/qemu/edk2-arm-*
 
     _pick qemu-system-avr usr/bin/qemu-system-avr
     _pick qemu-system-avr usr/share/man/man1/qemu-system-avr.1*
@@ -517,6 +517,13 @@ package_qemu-common() {
     _pick qemu-system-x86-firmware usr/share/qemu/qboot.rom
     # NOTE: needs to be replaced by sgabios
     _pick qemu-system-x86-firmware usr/share/qemu/sgabios.bin
+    # NOTE: needs to be replaced by seabios/edk2, not buildable on riscv64 yet
+    _pick qemu-system-x86-firmware usr/share/qemu/firmware/*-i386*
+    _pick qemu-system-x86-firmware usr/share/qemu/firmware/*-x86_64*
+    _pick qemu-system-x86-firmware usr/share/qemu/edk2-i386-*
+    _pick qemu-system-x86-firmware usr/share/qemu/edk2-x86_64-*
+    _pick qemu-system-x86-firmware usr/share/qemu/bios*
+    _pick qemu-system-x86-firmware usr/share/qemu/vgabios*
 
     _pick qemu-system-xtensa usr/bin/qemu-system-xtensa{,eb}
     _pick qemu-system-xtensa usr/share/man/man1/qemu-system-xtensa{,eb}.1*
@@ -711,7 +718,7 @@ package_qemu-hw-s390x-virtio-gpu-ccw() {
 
 package_qemu-system-aarch64() {
   pkgdesc="QEMU system emulator for AARCH64"
-  depends=("${_qemu_system_deps[@]}" edk2-aarch64 systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
 }
 
@@ -874,7 +881,7 @@ package_qemu-system-tricore() {
 
 package_qemu-system-x86() {
   pkgdesc="QEMU system emulator for x86"
-  depends=("${_qemu_system_deps[@]}" edk2-ovmf qemu-system-x86-firmware=$pkgver-$pkgrel seabios systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" qemu-system-x86-firmware=$pkgver-$pkgrel systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
 }
 
