--- PKGBUILD
+++ PKGBUILD
@@ -203,7 +203,7 @@
   'qemu-system-mips: for MIPS system emulator'
   'qemu-system-or1k: for OpenRisc32 system emulator'
   'qemu-system-ppc: for PPC system emulator'
-  'qemu-system-riscv: for RISC-V system emulator'
+  'qemu-system-x86: for x86 system emulator'
   'qemu-system-rx: for RX system emulator'
   'qemu-system-s390x: for S390 system emulator'
   'qemu-system-sh4: for SH4 system emulator'
@@ -425,9 +425,6 @@
   # remove unneeded files
   find "$pkgdir" -name .buildinfo -delete
 
-  # remove files provided by seabios
-  rm -fv "$pkgdir/usr/share/$pkgbase/"{bios,vgabios}*
-
   (
     # create man page symlinks for all system emulators
     cd "$pkgdir/usr/share/man/man1"
@@ -580,6 +577,9 @@
 
     _pick qemu-system-x86-firmware usr/share/qemu/{kvmvapic,{linuxboot,multiboot}_dma,pvh}.bin
     _pick qemu-system-x86-firmware usr/share/qemu/qboot.rom
+    # NOTE: needs to be replaced by seabios, not available on riscv64 yet
+    _pick qemu-system-x86-firmware usr/share/qemu/bios*
+    _pick qemu-system-x86-firmware usr/share/qemu/vgabios*
 
     _pick qemu-system-xtensa usr/bin/qemu-system-xtensa{,eb}
     _pick qemu-system-xtensa usr/share/man/man1/qemu-system-xtensa{,eb}.1*
@@ -1010,7 +1010,7 @@
 
 package_qemu-system-x86() {
   pkgdesc="QEMU system emulator for x86"
-  depends=("${_qemu_system_deps[@]}" dtc edk2-ovmf libcbor libcbor.so qemu-system-x86-firmware=$pkgver-$pkgrel seabios systemd-libs libudev.so)
+  depends=("${_qemu_system_deps[@]}" dtc edk2-ovmf libcbor libcbor.so qemu-system-x86-firmware=$pkgver-$pkgrel systemd-libs libudev.so)
   mv -v $pkgname/* "$pkgdir"
   _install_licenses
 }
@@ -1215,7 +1215,7 @@
   depends=(
     qemu-common=$pkgver-$pkgrel
     qemu-img=$pkgver-$pkgrel
-    qemu-system-x86=$pkgver-$pkgrel
+    qemu-system-riscv=$pkgver-$pkgrel
     virtiofsd
   )
   optdepends=("${_qemu_base_optdepends[@]}")
