diff --unified --recursive --text a/Makefile b/Makefile
--- a/Makefile	2021-09-08 20:32:52.061254345 +0000
+++ b/Makefile	2021-09-08 20:33:13.004599560 +0000
@@ -51,7 +51,7 @@
 CPPFLAGS += -DNLSDIR=\"$(NLSDIR)\"
 
 CFLAGS ?= -O2
-CFLAGS += -Wall -Wno-format -Wno-deprecated-declarations -std=c99
+CFLAGS += -Wall -Wno-deprecated-declarations -std=c99
 CFLAGS += $(shell $(PKG_CONFIG) --cflags $(PKG_DEPS))
 
 ifeq ($(WITH_GTK3), 0)
