diff --git PKGBUILD PKGBUILD
index 8c619ed1..493a3115 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,14 +10,17 @@ license=('custom')
 url="https://www.zlib.net/"
 depends=('glibc')
 options=('staticlibs')  # needed by binutils testsuite
-source=("https://zlib.net/zlib-${pkgver}.tar.gz"{,.asc})
+source=("https://zlib.net/zlib-${pkgver}.tar.gz"{,.asc}
+        "fix-zlib-crc32.patch"::"https://github.com/madler/zlib/commit/ec3df00224d4b396e2ac6586ab5d25f673caa4c2.patch")
 sha256sums=('91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9'
-            'SKIP')
+            'SKIP'
+            '00e023c3ccb7b895ebb3421970b1b77f8a527b40190f35050b79fd0e817a7b0c')
 validpgpkeys=('5ED46A6721D365587791E2AA783FCD8E58BCAFBA')
 
 prepare() {
 	cd "${srcdir}/zlib-$pkgver"
 	grep -A 24 '^  Copyright' zlib.h > LICENSE
+	patch -Np1 -i "../fix-zlib-crc32.patch"
 }
 
 build() {
