diff --git PKGBUILD PKGBUILD
index bbfd5213..0f76d895 100644
--- PKGBUILD
+++ PKGBUILD
@@ -63,11 +63,11 @@ build() {
   python module/setup.py build
   python -m compileall renpy
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
-  PYTHONPATH=module/build/lib.linux-x86_64-cpython-${python_version} python renpy.py . compile
-  PYTHONPATH=module/build/lib.linux-x86_64-cpython-${python_version} python renpy.py the_question compile
-  PYTHONPATH=module/build/lib.linux-x86_64-cpython-${python_version} python renpy.py tutorial compile
+  PYTHONPATH=module/build/lib.linux-${CARCH}-cpython-${python_version} python renpy.py . compile
+  PYTHONPATH=module/build/lib.linux-${CARCH}-cpython-${python_version} python renpy.py the_question compile
+  PYTHONPATH=module/build/lib.linux-${CARCH}-cpython-${python_version} python renpy.py tutorial compile
   cd sphinx
-  PYTHONPATH=../module/build/lib.linux-x86_64-cpython-${python_version} ./build.sh
+  PYTHONPATH=../module/build/lib.linux-${CARCH}-cpython-${python_version} ./build.sh
 }
 
 package_renpy() {
