diff --git PKGBUILD PKGBUILD
index 1099ade76..56db12e4b 100644
--- PKGBUILD
+++ PKGBUILD
@@ -62,11 +62,11 @@ build() {
   cd renpy
   python module/setup.py build
   python -m compileall renpy
-  PYTHONPATH=module/build/lib.linux-x86_64-3.10 python renpy.py . compile
-  PYTHONPATH=module/build/lib.linux-x86_64-3.10 python renpy.py the_question compile
-  PYTHONPATH=module/build/lib.linux-x86_64-3.10 python renpy.py tutorial compile
+  PYTHONPATH=module/build/lib.linux-riscv64-cpython-310 python renpy.py . compile
+  PYTHONPATH=module/build/lib.linux-riscv64-cpython-310 python renpy.py the_question compile
+  PYTHONPATH=module/build/lib.linux-riscv64-cpython-310 python renpy.py tutorial compile
   cd sphinx
-  PYTHONPATH=../module/build/lib.linux-x86_64-3.10 ./build.sh
+  PYTHONPATH=../module/build/lib.linux-riscv64-cpython-310 ./build.sh
 }
 
 package_renpy() {
