diff --git PKGBUILD PKGBUILD
index 30f19ba..99e1784 100644
--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,14 @@
 source=(git+https://github.com/microsoft/debugpy#tag=v$pkgver)
 sha256sums=('be7e6727a746db30f5e309c56a90c5e788c2f0fae8826794b9b3bb3a06970977')
 
+# https://github.com/microsoft/debugpy/blob/v1.8.19/src/debugpy/_vendored/pydevd/pydevd_tracing.py#L161
+# Special handling for x86 and x86_64
+case $CARCH in
+    i*86) _SUFFIX=x86;;
+    x86_64) _SUFFIX=amd64;;
+    *) _SUFFIX=$CARCH;;
+esac
+
 build() {
   cd $_pipname
 # Recythonize sources
@@ -31,13 +39,13 @@
   python -m build --wheel --no-isolation
 # Compile attach libraries
   cd build/lib*/debugpy/_vendored/pydevd/pydevd_attach_to_process
-  g++ ${CXXFLAGS} -shared -o attach_linux_amd64.so -fPIC -nostartfiles linux_and_mac/attach.cpp ${LDFLAGS}
+  g++ ${CXXFLAGS} -shared -o attach_linux_$_SUFFIX.so -fPIC -nostartfiles linux_and_mac/attach.cpp ${LDFLAGS}
 }
 
 package() {
   cd $_pipname
   python -m installer --destdir="$pkgdir" dist/*.whl
-  install -Dm755 build/lib*/debugpy/_vendored/pydevd/pydevd_attach_to_process/attach_linux_amd64.so \
+  install -Dm755 build/lib*/debugpy/_vendored/pydevd/pydevd_attach_to_process/attach_linux_$_SUFFIX.so \
     -t "$pkgdir"`python -c "import site; print(site.getsitepackages()[0])"`/debugpy/_vendored/pydevd/pydevd_attach_to_process
   install -Dm644 LICENSE "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 }
