--- PKGBUILD
+++ PKGBUILD
@@ -32,9 +32,9 @@ build() {
   # Build options are derived from here https://github.com/WebAssembly/wasi-sdk/blob/main/Makefile
   # remove compiler options not supported by the wasm32-wasi target
   export CFLAGS="$(echo $CFLAGS | sed "s/-mtune=generic//;
-  s/-fstack-clash-protection//; s/-fcf-protection//; s/-fexceptions//; s/-march=x86-64//")"
+  s/-fstack-clash-protection//; s/-fcf-protection//; s/-fexceptions//; s/-march=rv64gc//; s/-mabi=lp64d//")"
   export CXXFLAGS="$(echo $CXXFLAGS | sed "s/-mtune=generic//;
-  s/-fstack-clash-protection//; s/-fcf-protection//; s/-fexceptions//; s/-march=x86-64//")"
+  s/-fstack-clash-protection//; s/-fcf-protection//; s/-fexceptions//; s/-march=rv64gc//; s/-mabi=lp64d//")"
 
   cmake -B build -G Ninja \
     -DCMAKE_C_COMPILER_WORKS=ON \
