diff --git PKGBUILD PKGBUILD
index 11882f8ac..b1120b3d7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -16,8 +16,10 @@ makedepends=(
 )
 provides=(libebml.so)
 _tag=6c59e5e1ce0087e4465f6d3f479449d3f6bcb167
-source=(git+https://github.com/Matroska-Org/libebml.git#tag=${_tag})
-sha256sums=(SKIP)
+source=(git+https://github.com/Matroska-Org/libebml.git#tag=${_tag}
+        f0bfd53647961e799a43d918c46cf3b6bff89806.patch)
+sha256sums=('SKIP'
+            '2bbe3b24310d33a459eb24b9022f8e488b564a6fdf04d331c7f960add0968974')
 
 pkgver() {
   cd libebml
@@ -30,6 +32,9 @@ prepare() {
     rm -rf build
   fi
   mkdir build
+
+  cd libebml
+  patch -Np1  < "${srcdir}/f0bfd53647961e799a43d918c46cf3b6bff89806.patch"
 }
 
 build() {
