diff --git PKGBUILD PKGBUILD
index b86c485..7cade20 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,14 +10,22 @@
 depends=(libhandy libpwquality python-gobject python-pykeepass)
 makedepends=(git gobject-introspection meson)
 _commit=868a3a9c25258033bc2859b80bea554838dbecf4
-source=("git+https://gitlab.gnome.org/World/PasswordSafe.git/#commit=$_commit")
-sha256sums=('SKIP')
+source=("git+https://gitlab.gnome.org/World/PasswordSafe.git/#commit=$_commit"
+        "remove-incorrect-i18n.merge_file-argument.patch")
+sha256sums=('SKIP'
+            '4a8ff344004c6e106c576fe9dd90e736e5a3225321a849f35a4853d6ab4fb7c4')
 
 pkgver() {
   cd PasswordSafe
   git describe --tags | sed 's/-/+/g'
 }
 
+prepare() {
+  cd PasswordSafe
+  # https://gitlab.gnome.org/World/secrets/-/merge_requests/604
+  patch -Np1 -i ../remove-incorrect-i18n.merge_file-argument.patch
+}
+
 build() {
   arch-meson PasswordSafe build
   ninja -C build
