--- PKGBUILD.orig	2021-12-14 16:54:40.977627353 +0000
+++ PKGBUILD	2021-12-14 18:02:27.157848818 +0000
@@ -11,11 +11,13 @@
 license=('MIT' 'APACHE')
 depends=('gcc-libs')
 makedepends=('rust')
-source=("fd-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('8ead7eedd8dac77f4f253cd70c788d7cbdbeb66369829d845ec2da1dc651b73753a23b44635e40a70084cbe2a2f2f4ba129738e8b4c8601c3c82836c1a338b89')
+source=("fd-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz" disable-jemallocator.patch)
+sha512sums=('8ead7eedd8dac77f4f253cd70c788d7cbdbeb66369829d845ec2da1dc651b73753a23b44635e40a70084cbe2a2f2f4ba129738e8b4c8601c3c82836c1a338b89'
+            'cebc40eef62fc8a8bddef3d816638c82d0037b187b930444a837d73d1d0c7084aa6344a4379fc5d60c3f078d1822395341bca9e8ab37759b019530101750538f')
 
 build() {
   cd fd-$pkgver
+  patch -Np1 -i ../disable-jemallocator.patch
   cargo build --release --locked
 }
 
