diff --git PKGBUILD PKGBUILD
index d80ad65..cff79bc 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,11 +11,13 @@ url=https://github.com/sharkdp/fd
 license=('MIT' 'APACHE')
 depends=('gcc-libs')
 makedepends=('rust')
-source=("fd-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz")
-sha512sums=('5e921cb3556899e086f0fbaae21842b532c486c75e91cbb8351e25f640467217ac89e8cfba044e855e2425222bfdbd01befa4f35251c8705b30b3503adf3a7bf')
+source=("fd-$pkgver.tar.gz::$url/archive/v$pkgver.tar.gz" disable-jemallocator.patch)
+sha512sums=('5e921cb3556899e086f0fbaae21842b532c486c75e91cbb8351e25f640467217ac89e8cfba044e855e2425222bfdbd01befa4f35251c8705b30b3503adf3a7bf'
+            'a60df4a361efdd088d709015ad056cb7f29a48732c6ae91bef40d56cbfa743ef7653c8db2138b35401d98a33aecfbb313848f675ca600f0dfd900dbed10cd84d')
 
 build() {
   cd fd-$pkgver
+  patch -Np1 -i ../disable-jemallocator.patch
   cargo build --release --locked
 }
 
