diff --git PKGBUILD PKGBUILD
index 27559b77..e0dde9f2 100644
--- PKGBUILD
+++ PKGBUILD
@@ -20,7 +20,9 @@ b2sums=('b41f0f0f22f0bef615ec6b013c2d95149ad4a13fde438b7b25064d706388531d25fafaa
 
 prepare() {
   cd skim-$pkgver
-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
+  echo -e "\n[patch.crates-io]\ntuikit = { git = 'https://github.com/felixonmars/tuikit' }" >> Cargo.toml
+  cargo update -p nix:0.14.1
+  cargo fetch --locked
 }
 
 build() {
