--- PKGBUILD
+++ PKGBUILD
@@ -13,6 +13,11 @@ makedepends=('pkg-config')
 source=(https://download.videolan.org/contrib/tiger/$pkgname-$pkgver.tar.gz)
 sha256sums=('7342b255fb476d75b08dd4c56d5ce580f39b1c71c42e99fb7af9d999725d4166')
 
+prepare() {
+  cd "${srcdir}/${pkgname}-${pkgver}"
+  cp -v /usr/share/autoconf/build-aux/config.{guess,sub} misc/autotools/
+}
+
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
   ./configure --prefix=/usr \
