--- PKGBUILD
+++ PKGBUILD
@@ -21,6 +21,7 @@
 
 prepare() {
   cd $pkgbase-$pkgver
+  patch -Np1 -i ../fix-riscv-conversions.patch
   autoreconf -fiv
 }
 
@@ -43,3 +44,6 @@
   cd $pkgbase-$pkgver
   make -C doc DESTDIR="$pkgdir" install
 }
+
+source+=(fix-riscv-conversions.patch)
+b2sums+=('8fbadb0489f9e896db1bb12866a9f4db781fb4973b9e850944c080853fca7c8878595053d57d5fd4dbce3c0af8c4c83a12bc85b8ea4109b65a44077bfc0d18ca')
