--- PKGBUILD
+++ PKGBUILD
@@ -14,6 +14,11 @@ source=(https://download.savannah.nongnu.org/releases/${pkgname}/${pkgname}-${pk
 #validpgpkeys=('EEFF99F6ADE26B135C85141D663843B80213D86E') # Dalibor Topic
 sha256sums=('f156abc5de8658f22ee8f08d7a72c88f9409ebd8c7933e9466b0842afeb2f145')
 
+prepare() {
+  cd ${pkgname}-${pkgver}
+  autoreconf -fi
+}
+
 build() {
   cd ${pkgname}-${pkgver}
   ./configure --prefix=/usr
