diff --git PKGBUILD PKGBUILD
index ab1317c..70ce6b3 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,6 +15,7 @@
 optdepends=('clang: Support for beautification of message filter scripts')
 source=("https://github.com/martinrotter/$pkgbase/archive/$pkgver/$pkgbase-$pkgver.tar.gz")
 sha256sums=('9154a9c674b35f68486ca7e024265e044b7c7971485bafbe9b6ebbdf27463489')
+options=(!lto)
 
 prepare() {
   mkdir build{,-nowebengine}
