diff --git PKGBUILD PKGBUILD
index 01c14cf4c..8dbb06667 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,7 +9,7 @@ arch=('any')
 url='https://github.com/kcsaff/getkey'
 license=('MIT')
 depends=('python')
-makedepends=('python-setuptools' 'flake8')
+makedepends=('python-pip' 'python-setuptools' 'flake8')
 checkdepends=('python-pytest-cov')
 source=("https://github.com/kcsaff/$_pkgname/archive/v$pkgver/$pkgname-$pkgver.tar.gz")
 sha256sums=('096792463ed0cd8ac92d7b2b6d23cff1c815b449403115034bed2029b378d54c')
