diff --git PKGBUILD PKGBUILD
index 8ef9e1d..8639bca 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,8 +12,10 @@ arch=(x86_64)
 url=https://commons.apache.org/proper/commons-daemon/
 license=(APACHE)
 _tag=466cc9a2442412dbb25dc5dad803982a7f864264
-source=(git+https://github.com/apache/commons-daemon.git#tag=${_tag})
-b2sums=(SKIP)
+source=(git+https://github.com/apache/commons-daemon.git#tag=${_tag}
+        "riscv64-support.patch")
+b2sums=('SKIP'
+        'ee4c59ef21052bf2a681ff42e4d1d29c158b4ecb93a533bb6e219dfff9038b9177c3f38ec72aa3ca123615c9c18fe8f3c8e8c15a87a06523c40196949909cd12')
 makedepends=(
   docbook2x
   docbook-utils
@@ -25,6 +27,7 @@ makedepends=(
 
 prepare() {
   cd commons-daemon/src/native/unix
+  patch -Np4 -i "${srcdir}/riscv64-support.patch"
   support/buildconf.sh
 }
 
