--- PKGBUILD	(revision 465349)
+++ PKGBUILD	(working copy)
@@ -12,8 +12,10 @@
 url=https://commons.apache.org/proper/commons-daemon/
 license=(APACHE)
 _tag=5ead75b56ce0e171931de808bf0529666c1c4cbb
-source=(git+https://github.com/apache/commons-daemon.git#tag=${_tag})
-b2sums=(SKIP)
+source=(git+https://github.com/apache/commons-daemon.git#tag=${_tag}
+        'https://github.com/apache/commons-daemon/commit/0015f5424f3dc63df5c600ce243ea2d52eb7cf88.patch')
+b2sums=('SKIP'
+        '29821ba55c3641cb932b6c67bbec528be656d157e31723c1f6a286cca4c84de542a517e9bb4c4cc4a3c5c184fdde3034519ba5905b91a49c38376e175832e513')
 makedepends=(
   docbook2x
   docbook-utils
@@ -25,6 +27,7 @@
 
 prepare() {
   cd commons-daemon/src/native/unix
+  patch -Np4 -i "${srcdir}/0015f5424f3dc63df5c600ce243ea2d52eb7cf88.patch"
   support/buildconf.sh
 }
 
