diff --git a/libstdc++-v3/po/Makefile.am b/libstdc++-v3/po/Makefile.am
index ab716ff04..296d3404f 100644
--- a/libstdc++-v3/po/Makefile.am
+++ b/libstdc++-v3/po/Makefile.am
@@ -38,7 +38,7 @@ MSGFMT = msgfmt
 EXTRA_DIST = string_literals.cc POTFILES.in $(PACKAGE).pot $(LOCALE_IN)
 
 .po.mo:
-	$(MSGFMT) -o $@ $<
+	LD_PRELOAD="/usr/lib/libstdc++.so" $(MSGFMT) -o $@ $<
 
 all-local: all-local-$(USE_NLS)
 all-local-no:
diff --git a/libstdc++-v3/po/Makefile.in b/libstdc++-v3/po/Makefile.in
index 8edd2e056..48f458502 100644
--- a/libstdc++-v3/po/Makefile.in
+++ b/libstdc++-v3/po/Makefile.in
@@ -553,7 +553,7 @@ uninstall-am:
 
 
 .po.mo:
-	$(MSGFMT) -o $@ $<
+	LD_PRELOAD="/usr/lib/libstdc++.so" $(MSGFMT) -o $@ $<
 
 all-local: all-local-$(USE_NLS)
 all-local-no:
