diff --git PKGBUILD PKGBUILD
index 4e4b59fa..aa9e140d 100644
--- PKGBUILD
+++ PKGBUILD
@@ -69,7 +69,6 @@ build() {
       --disable-libssp
       --disable-libstdcxx-pch
       --disable-werror
-      --with-build-config=bootstrap-lto
       --enable-link-serialization=1
       --program-suffix=-${_majorver}
       --enable-version-specific-runtime-libs
@@ -78,6 +77,9 @@ build() {
 
   cd gcc-build
 
+  CFLAGS=${CFLAGS/-Wp,-D_FORTIFY_SOURCE=2/}
+  CXXFLAGS=${CXXFLAGS/-Wp,-D_FORTIFY_SOURCE=2/}
+
   # Credits @allanmcrae
   # https://github.com/allanmcrae/toolchain/blob/f18604d70c5933c31b51a320978711e4e6791cf1/gcc/PKGBUILD
   # TODO: properly deal with the build issues resulting from this
@@ -98,6 +100,10 @@ build() {
 
   # make documentation
   make -O -C $CHOST/libstdc++-v3/doc doc-man-doxygen
+
+  # Patch spec strings embedded in binaries
+  sed -i 's/%{pthread:--push-state --as-needed -latomic --pop-state}/  --push-state --as-needed -latomic --pop-state         /' gcc/xgcc
+  sed -i 's/%{pthread:--push-state --as-needed -latomic --pop-state}/  --push-state --as-needed -latomic --pop-state         /' gcc/xg++
 }
 
 check() {
@@ -114,23 +120,19 @@ check() {
 package_gcc11-libs() {
   pkgdesc="Runtime libraries shipped by GCC (11.x.x)"
   depends=('glibc>=2.27')
-  options=(!emptydirs !strip)
+  options=(!emptydirs !strip staticlibs)
 
   cd gcc-build
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install-shared
-  mv "${pkgdir}/${_libdir}"/../lib/* "${pkgdir}/${_libdir}"
-  rmdir "${pkgdir}/${_libdir}"/../lib
   rm -f "$pkgdir/$_libdir/libgcc_eh.a"
 
   for lib in libasan.so \
-             libatomic.so \
+             libatomic.{so,a} \
              libgfortran.so \
              libgomp.so \
-             libitm.so \
              liblsan.so \
              libquadmath.so \
              libstdc++.so \
-             libtsan.so \
              libubsan.so; do
     ln -s /usr/lib/$lib "$pkgdir/$_libdir/$lib"
   done
@@ -159,6 +161,7 @@ package_gcc11() {
 
   make -C $CHOST/libgcc DESTDIR="$pkgdir" install
   rm -rf "${pkgdir}/${_libdir}"/../lib
+  rm -rf "${pkgdir}/${_libdir}"/libgcc_s.so*
 
   make -C $CHOST/libstdc++-v3/src DESTDIR="$pkgdir" install
   make -C $CHOST/libstdc++-v3/include DESTDIR="$pkgdir" install
@@ -175,11 +178,9 @@ package_gcc11() {
     "$pkgdir/${_libdir}/bfd-plugins/"
 
   make -C $CHOST/libgomp DESTDIR="$pkgdir" install-nodist_{libsubinclude,toolexeclib}HEADERS
-  make -C $CHOST/libitm DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libquadmath DESTDIR="$pkgdir" install-nodist_libsubincludeHEADERS
   make -C $CHOST/libsanitizer DESTDIR="$pkgdir" install-nodist_{saninclude,toolexeclib}HEADERS
   make -C $CHOST/libsanitizer/asan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
-  make -C $CHOST/libsanitizer/tsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
   make -C $CHOST/libsanitizer/lsan DESTDIR="$pkgdir" install-nodist_toolexeclibHEADERS
 
   make -C libcpp DESTDIR="$pkgdir" install
