diff --git PKGBUILD PKGBUILD
index 96179ae0b..6ab64b233 100644
--- PKGBUILD
+++ PKGBUILD
@@ -9,8 +9,16 @@ url="http://mp3splt.sourceforge.net/"
 license=('GPL')
 depends=('desktop-file-utils' 'gtk3' 'libmp3splt' 'gst-plugins-good' 'gst-plugins-ugly')
 changelog=$pkgname.changelog
-source=(https://downloads.sourceforge.net/sourceforge/mp3splt/$pkgname-$pkgver.tar.gz)
-sha256sums=('2084cc942b5d1aa6ff74848a41d05cb37876cbc74dfdc341ceb424c763887650')
+source=(https://downloads.sourceforge.net/sourceforge/mp3splt/$pkgname-$pkgver.tar.gz
+        https://sourceforge.net/p/mp3splt/bugs/_discuss/thread/6fa0946d/a0a9/attachment/mp3splt-gtk-format-security.patch)
+sha256sums=('2084cc942b5d1aa6ff74848a41d05cb37876cbc74dfdc341ceb424c763887650'
+            'f59c8ccffdbd1c9a605dc088a3e4b1b44684e94ae48a8073683cb194969085fd')
+
+prepare(){
+  cd "${srcdir}"/$pkgname-$pkgver
+
+  patch -Np1 -i "${srcdir}/mp3splt-gtk-format-security.patch"
+}
 
 build() {
   cd "${srcdir}"/$pkgname-$pkgver
