Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 391859)
+++ PKGBUILD	(working copy)
@@ -33,13 +33,14 @@
   ./bundle.sh
 
   cd nss-$pkgver/nss
-  ./build.sh \
-    --target x64 \
+  CFLAGS="$CFLAGS -DNS_PTR_GT_32" ./build.sh \
+    --target riscv64 \
     --opt \
     --system-sqlite \
     --system-nspr \
     --enable-libpkix \
-    --disable-tests
+    --disable-tests \
+    -DNSS_USE_64=1
 }
 
 package_nss() {
