Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 460997)
+++ PKGBUILD	(working copy)
@@ -9,7 +9,7 @@
 license=(GPL)
 url="https://github.com/jstedfast/gmime"
 depends=(glib2 gpgme zlib)
-makedepends=(gtk-sharp-2 gobject-introspection gtk-doc git vala docbook-utils)
+makedepends=(gobject-introspection gtk-doc git vala docbook-utils)
 _commit=91dcee38ea301463fb9c6bc936f08b9fd1c0969f  # gmime-2-6
 source=("git+https://github.com/jstedfast/gmime#commit=$_commit"
         0001-Avoid-non-UTF-8-code.patch)
@@ -37,7 +37,6 @@
   ./configure --prefix=/usr \
     --program-prefix=$pkgname \
     --enable-gtk-doc \
-    --enable-mono \
     --enable-smime \
     --disable-static
 
