diff --git PKGBUILD PKGBUILD
index 8c41ed9..ff6bc40 100644
--- PKGBUILD
+++ PKGBUILD
@@ -89,7 +89,6 @@ makedepends=(
 )
 optdepends=(
   'avisynthplus: AviSynthPlus support'
-  'intel-media-sdk: Intel QuickSync support (legacy)'
   'onevpl-intel-gpu: Intel QuickSync support'
   'ladspa: LADSPA filters'
   'nvidia-utils: Nvidia NVDEC/NVENC support'
@@ -187,8 +186,6 @@ build() {
     --enable-libxml2 \
     --enable-libxvid \
     --enable-libzimg \
-    --enable-nvdec \
-    --enable-nvenc \
     --enable-opencl \
     --enable-opengl \
     --enable-shared \
