diff --git PKGBUILD PKGBUILD
index 508810c..f3566c7 100644
--- PKGBUILD
+++ PKGBUILD
@@ -32,7 +32,6 @@
   libfreetype.so
   libgl
   libiec61883
-  libmfx
   libmodplug
   libpulse
   librav1e.so
@@ -87,9 +86,7 @@
 )
 optdepends=(
   'avisynthplus: AviSynthPlus support'
-  'intel-media-sdk: Intel QuickSync support'
   'ladspa: LADSPA filters'
-  'nvidia-utils: Nvidia NVDEC/NVENC support'
 )
 provides=(
   libavcodec.so
@@ -150,7 +147,6 @@
     --enable-libgsm \
     --enable-libiec61883 \
     --enable-libjack \
-    --enable-libmfx \
     --enable-libmodplug \
     --enable-libmp3lame \
     --enable-libopencore_amrnb \
@@ -178,8 +174,6 @@
     --enable-libxml2 \
     --enable-libxvid \
     --enable-libzimg \
-    --enable-nvdec \
-    --enable-nvenc \
     --enable-opencl \
     --enable-opengl \
     --enable-shared \
