--- PKGBUILD
+++ PKGBUILD
@@ -6,7 +6,7 @@
 
 pkgname=ffmpeg
 pkgver=4.4.1
-pkgrel=1
+pkgrel=1.1
 epoch=2
 pkgdesc='Complete solution to record, convert and stream audio and video'
 arch=(x86_64)
@@ -30,7 +30,6 @@ depends=(
   libdrm
   libfreetype.so
   libiec61883
-  libmfx
   libmodplug
   libpulse
   librav1e.so
@@ -82,13 +81,13 @@ optdepends=(
   'avisynthplus: AviSynthPlus support'
   'intel-media-sdk: Intel QuickSync support'
   'ladspa: LADSPA filters'
-  'nvidia-utils: Nvidia NVDEC/NVENC support'
 )
 provides=(
   libavcodec.so
   libavdevice.so
   libavfilter.so
   libavformat.so
+  libavresample.so
   libavutil.so
   libpostproc.so
   libswresample.so
@@ -135,6 +134,7 @@ build() {
     --enable-ladspa \
     --enable-libaom \
     --enable-libass \
+    --enable-avresample \
     --enable-libbluray \
     --enable-libdav1d \
     --enable-libdrm \
@@ -143,7 +143,6 @@ build() {
     --enable-libgsm \
     --enable-libiec61883 \
     --enable-libjack \
-    --enable-libmfx \
     --enable-libmodplug \
     --enable-libmp3lame \
     --enable-libopencore_amrnb \
@@ -171,8 +170,6 @@ build() {
     --enable-libxml2 \
     --enable-libxvid \
     --enable-libzimg \
-    --enable-nvdec \
-    --enable-nvenc \
     --enable-shared \
     --enable-version3
 
