diff --git PKGBUILD PKGBUILD
index a118625..7cdb5db 100644
--- PKGBUILD
+++ PKGBUILD
@@ -15,18 +15,22 @@ makedepends=('cmake' 'libxslt' 'qt5-base' 'jack' 'libvorbis' 'libxv' 'opus' 'des
              'libvdpau' 'libdca' 'fribidi' 'glu' 'qt5-tools' 'twolame' 'libfdk-aac' 'libass')
 options=('!emptydirs')
 source=(https://downloads.sourceforge.net/avidemux/avidemux_${pkgver}.tar.gz
-        log.diff qt-5.15.diff)
+        log.diff qt-5.15.diff
+        riscv64.diff::https://patch-diff.githubusercontent.com/raw/mean00/avidemux2/pull/331.patch)
 sha256sums=('628a404f521ff2812760700ae3e2aa78e5816b0ff3fb6fd05ac3e75248d97401'
             '8b751d137037f22fd4ae6750709014bcf61269a18f1c32e1a91f481285c807aa'
-            '255a2446d493503dc237482f6fe26e2186f0ea0ef43ac9e17b8059fe232aad43')
+            '255a2446d493503dc237482f6fe26e2186f0ea0ef43ac9e17b8059fe232aad43'
+            'a4102547b422b50e0c5bc673bf29626dae9e89123000fd3b40f893daada24dd0')
 sha512sums=('d936eb005a8418b8f3081fce599cd04bb1a256bb7cb4bf50bf2ed876658881ca76b8c6f3805da9f05614eea44eaf2d53da0f802b28ce76e754b738f2682df031'
             'bbae5b4a549113b12e728c4c0ac8c1c92a0b7500be8dcecc791c16c9a913406feaa9f7bc477985970a920d2df391cc9392457512d84b9c4b829981cc072f2b2e'
-            'ed78c2a3e4ede4fed1d3ea353c7dc350dc73050f76c8112cb24d97b3eda01f7c186d5db199f71f56953c5e5d8104de44dc0e84e1bfa01b6647c050c11fc375f2')
+            'ed78c2a3e4ede4fed1d3ea353c7dc350dc73050f76c8112cb24d97b3eda01f7c186d5db199f71f56953c5e5d8104de44dc0e84e1bfa01b6647c050c11fc375f2'
+            '27ffe6f97e7caad99497393e1f12372e46d0de7f7bf96c959f1e322ce8f30a6e18562bdfad9ce38d9d18d2b02f9d4d8ef3d4d88e5f1840074d5cff0b793d9873')
 
 prepare() {
   cd ${pkgbase}_${pkgver}
   patch -Np1 -i ../log.diff  # don't hide the build output
   patch -Np1 -i ../qt-5.15.diff  # fix build against Qt 5.15.0
+  patch -Np1 -i ../riscv64.diff # add riscv64 support
   sed -i 's|../avidemux/qt4|../avidemux/qt4 -DLRELEASE_EXECUTABLE=/usr/bin/lrelease-qt5|' bootStrap.bash
   sed -e 's|0.19|1.0|' -i avidemux_plugins/ADM_videoFilters6/ass/CMakeLists.txt
 }
