diff --git a/tests/test_hypothesis.py b/tests/test_hypothesis.py
index dc812db..d13aae0 100644
--- a/tests/test_hypothesis.py
+++ b/tests/test_hypothesis.py
@@ -213,7 +213,7 @@ class ParseTestCase(unittest.TestCase):
             xform([struct[:-1]])
 
     @given(dbus_signatures(blacklist="hbs", exclude_dicts=True))
-    @settings(max_examples=100)
+    @settings(max_examples=100, deadline=1000)
     @example(sig="v")
     def test_exceptions(self, sig):
         """
@@ -235,7 +235,7 @@ class SignatureTestCase(unittest.TestCase):
     """
 
     @given(STRATEGY_GENERATOR.parseString("v", parseAll=True)[0])
-    @settings(max_examples=50)
+    @settings(max_examples=50, deadline=1000)
     def test_unpacking(self, value):
         """
         Test that signature unpacking works.
