diff --git PKGBUILD PKGBUILD
index 7fe0ef1..87c9c3f 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,13 +12,16 @@ license=('BSD')
 depends=('libmagick')
 makedepends=('cmake')
 source=(https://launchpad.net/cuneiform-linux/${pkgver%.*}/${pkgver%.*}/+download/$pkgname-linux-$pkgver.tar.bz2
-	build-fix.patch)
+	build-fix.patch
+	build-fix2.patch)
 sha256sums=('577e0d054de72086c33b4e8ae15033657776509b9a7af6eb272888eefcbdbbad'
-            '3d9f4d923ec6ca5555550819096b7d66e364a8aeb74a9d1f0d961784d267642b')
+            '3d9f4d923ec6ca5555550819096b7d66e364a8aeb74a9d1f0d961784d267642b'
+            '025fabac7fbe85311296cbe04ff4aaf290ab4d15bf80ec3bf4986b5666d2dd8b')
 
 prepare() {
   cd $pkgname-linux-$pkgver
   patch -Np1 -i ../build-fix.patch
+  patch -Np1 -i ../build-fix2.patch
   sed -i 's/lib64/lib/' install_files.cmake
 # Don't override LDFLAGS
   sed -e '/CMAKE_SHARED_LINKER_FLAGS/d' -i cuneiform_src/CMakeLists.txt
