--- PKGBUILD
+++ PKGBUILD
@@ -11,17 +11,22 @@ arch=('any')
 url=https://github.com/readthedocs/sphinx_rtd_theme
 license=('MIT')
 depends=('python-sphinx')
-makedepends=('nodejs-lts-fermium' 'npm' 'python-setuptools-scm' 'python-build' 'python-installer' 'python-wheel')
-source=("$url/archive/$pkgver/sphinx_rtd_theme-$pkgver.tar.gz")
-b2sums=('1998051e814fcbee541ffe53ccaa5d1ebdeaa27bb95c2e68c840032660fc105621041f72a12c6f2ec77db400cfef21a57850e351921572a71b52dfa2e0b460fc')
+makedepends=('nodejs-lts-gallium' 'npm' 'python-setuptools-scm' 'python-build' 'python-installer' 'python-wheel')
+source=("$url/archive/$pkgver/sphinx_rtd_theme-$pkgver.tar.gz"
+        fix-for-new-nodejs.patch)
+b2sums=('1998051e814fcbee541ffe53ccaa5d1ebdeaa27bb95c2e68c840032660fc105621041f72a12c6f2ec77db400cfef21a57850e351921572a71b52dfa2e0b460fc'
+        '96d06e20679149010a5714310e1e40c0892a5a6f9cc8fe337fd3564295bcf635e543b2a4b26948e8982fd575895e80a7d888f61b9c2235d2284f256a64a22b61')
 
 prepare() {
   cd sphinx_rtd_theme-$pkgver
-  npm ci
+  patch -Np1 -i ../fix-for-new-nodejs.patch
+  rm package-lock.json
+  npm install
 }
 
 build() {
   cd sphinx_rtd_theme-$pkgver
+  export NODE_OPTIONS=--openssl-legacy-provider
   npm run build
   python -m build --wheel --no-isolation
 }
