--- PKGBUILD
+++ PKGBUILD
@@ -12,17 +12,22 @@ arch=('any')
 url=https://github.com/readthedocs/sphinx_rtd_theme
 license=('MIT')
 depends=('python-sphinx' 'python-sphinxcontrib-jquery')
-makedepends=('nodejs-lts-fermium' 'npm' 'python-setuptools-scm' 'python-build' 'python-installer' 'python-wheel')
-source=("$url/archive/$pkgver/sphinx_rtd_theme-$pkgver.tar.gz")
-b2sums=('25184f0b810ebe0f346a87e5651243c1b740bd1803ec57c5f842527f245cf60f614b7973e19a2c11f4482d84efaff9c1549923947d0597d36acf18200fa5193b')
+makedepends=('nodejs-lts-gallium' 'npm' 'python-setuptools-scm' 'python-build' 'python-installer' 'python-wheel')
+source=("$url/archive/$pkgver/sphinx_rtd_theme-$pkgver.tar.gz"
+        "fix-for-new-nodejs.patch")
+b2sums=('25184f0b810ebe0f346a87e5651243c1b740bd1803ec57c5f842527f245cf60f614b7973e19a2c11f4482d84efaff9c1549923947d0597d36acf18200fa5193b'
+        '96d06e20679149010a5714310e1e40c0892a5a6f9cc8fe337fd3564295bcf635e543b2a4b26948e8982fd575895e80a7d888f61b9c2235d2284f256a64a22b61')

 prepare() {
   cd sphinx_rtd_theme-$pkgver
-  npm ci
+  patch -Np1 -i ../fix-for-new-nodejs.patch
+  rm package-lock.json
+  npm install
 }

 build() {
   cd sphinx_rtd_theme-$pkgver
+  export NODE_OPTIONS=--openssl-legacy-provider
   npm run build
   python -m build --wheel --no-isolation
 }
