--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,11 @@ depends=('perl')
 install=$pkgname.install
 changelog=$pkgname.changelog
 backup=('etc/xdg/surfraw/conf' 'etc/xdg/surfraw/bookmarks')
-source=($pkgname-$pkgver.tar.bz2::https://gitlab.com/surfraw/Surfraw/repository/$pkgname-$pkgver/archive.tar.bz2)
-sha256sums=('d63c19c382b0e888e442f4106ee6babb5071ce076ab7834028db070814c42556')
+source=($pkgname-$pkgver.tar.bz2::https://gitlab.com/surfraw/Surfraw/-/archive/surfraw-$pkgver/Surfraw-surfraw-$pkgver.tar.bz2)
+sha256sums=('1859bc0e54080ae710511d9420af2ef24823ca3a141d0fe248c7bf0f5f18e6c5')
 
 build() {
-  cd "${srcdir}"/Surfraw-$pkgname-$pkgver-8b7e1d1f4f63d07370953ffc3abe492f943c90a2
+  cd "${srcdir}"/Surfraw-$pkgname-$pkgver
 
   ./prebuild
   ./configure --prefix=/usr --sysconfdir=/etc
@@ -24,7 +24,7 @@ build() {
 }
 
 package () {
-  cd "${srcdir}"/Surfraw-$pkgname-$pkgver-8b7e1d1f4f63d07370953ffc3abe492f943c90a2
+  cd "${srcdir}"/Surfraw-$pkgname-$pkgver
 
   make DESTDIR="${pkgdir}" install
 
