--- PKGBUILD
+++ PKGBUILD
@@ -88,6 +88,8 @@ check() {
   --ignore tests/test_benchmarks_cookiejar.py
   --ignore tests/test_benchmarks_http_websocket.py
   --ignore tests/test_benchmarks_http_writer.py
+  # skip import time test for slow riscv64
+  --deselect 'tests/test_imports.py::test_import_time'
   )
   local python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
 
