diff --git PKGBUILD PKGBUILD
index b776af4..bbe3adb 100644
--- PKGBUILD
+++ PKGBUILD
@@ -83,7 +83,8 @@ check() {
   local _python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
   mv tests/autobahn/test_autobahn.py{,.bak} # Docker tests
   # https://github.com/aio-libs/aiohttp/issues/8234
-  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${_python_version}" pytest -k 'not test_aiohttp_plugin'
+  # skip import time test
+  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${_python_version}" pytest -k 'not test_aiohttp_plugin and not test_import_time'
   mv tests/autobahn/test_autobahn.py{.bak,}
 }
 
