--- PKGBUILD
+++ PKGBUILD
@@ -87,7 +87,8 @@ check() {
   cd ${pkgname}
   local _python_version=$(python -c 'import sys; print("".join(map(str, sys.version_info[:2])))')
   mv tests/autobahn/test_autobahn.py{,.bak} # Docker tests
-  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${_python_version}" pytest
+  # skip import time test
+  PYTHONPATH="$PWD/build/lib.linux-$CARCH-cpython-${_python_version}" pytest -k 'not test_import_time'
   mv tests/autobahn/test_autobahn.py{.bak,}
 }
 
