diff --git a/test/localhost/authority.rb b/test/localhost/authority.rb
index 7a92287..30c8ef9 100644
--- a/test/localhost/authority.rb
+++ b/test/localhost/authority.rb
@@ -13,6 +13,10 @@ require 'async/io/shared_endpoint'
 require 'async/process'
 require 'fileutils'
 
+include Sus::Fixtures::Async::ReactorContext
+
+def timeout = 9
+
 describe Localhost::Authority do
 	let(:authority) {subject.new}
 	
diff --git a/test/localhost/protocol.rb b/test/localhost/protocol.rb
index 9a4752e..d8d6f52 100644
--- a/test/localhost/protocol.rb
+++ b/test/localhost/protocol.rb
@@ -13,6 +13,10 @@ require 'async/io/shared_endpoint'
 require 'async/process'
 require 'fileutils'
 
+include Sus::Fixtures::Async::ReactorContext
+
+def timeout = 9
+
 AValidProtocol = Sus::Shared("valid protocol") do |protocol, openssl_options, curl_options|
 	it "can connect using #{protocol} using openssl" do
 		status = Async::Process.spawn("openssl", "s_client", "-connect", "localhost:4040", *openssl_options)
