Index: PKGBUILD
===================================================================
--- PKGBUILD	(revision 422645)
+++ PKGBUILD	(working copy)
@@ -9,12 +9,15 @@
 license=('GPL2' 'FDL')
 makedepends=('intltool' 'espeak-ng' 'libltdl' 'python-xdg' 'dotconf' 'libpulse' 'libao')
 source=("https://github.com/brailcom/speechd/releases/download/${pkgver}/speech-dispatcher-${pkgver}.tar.gz"
-        'speech-dispatcherd.service')
+        'speech-dispatcherd.service'
+        $pkgname-glib.patch::https://github.com/brailcom/speechd/commit/a2faab416e42cbdf3d73f98578a89eb7a235e25a.patch)
 sha512sums=('9af240d9a15062489c170440911e9d148d0d113a0f22211f1496e9754345a5d079e160962d666801322dc2efa1134f8b5e209c0183d93a1f0896705dc6011122'
-            '73eb563003b03f4836b17af66335664050d1dc3bd8434be368cd09ff8933aa2adc4bd17c4ce5c1bbbd406a0b0880191426a679fa05eac0a30f4005eaeeb47340')
+            '73eb563003b03f4836b17af66335664050d1dc3bd8434be368cd09ff8933aa2adc4bd17c4ce5c1bbbd406a0b0880191426a679fa05eac0a30f4005eaeeb47340'
+            '098efb310446ebf25b80dac2a321bb1f5d74905d92a5f8a49838df687f5ef1e3bdae513f3b65451b8befcbaa485069730555c522ff96517adb74fcf41b2df85b')
 
 prepare() {
   cd ${pkgname}-${pkgver}
+  patch -p1 -i ../$pkgname-glib.patch
 
   # Disabled due to bug: https://bugs.archlinux.org/task/35700
   sed -i "s/cicero //g" configure.ac
