--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,12 @@ sha512sums=('c4fa90ca1f244c1b0096c2a2127e8c88ce92f9a0a0d57bfe18fa41941f201566281
 
 prepare() {
   cd cli-$pkgver
+
+  # fix: possible racing condition fails the build process, so we invoke this manually
+  node ./bin/npm-cli.js install --ignore-scripts --no-audit
+  # guard test: this should success, indicating a successful install
+  node ./bin/npm-cli.js ls
+
   mkdir -p node_modules/.bin
   ln -sf /usr/bin/marked{,-man} node_modules/.bin/
 
