--- PKGBUILD
+++ PKGBUILD
@@ -18,6 +18,12 @@ sha512sums=('309d233874cf2a92b7444c4785a49122f1cdbebac9eda9615f6b4b95f8b1221f16e
 
 prepare() {
   cd cli-$pkgver
+
+  # fix: possible racing condition fails the build process, so we invoke this manually
+  node ./bin/npm-cli.js install --ignore-scripts --no-audit
+  # guard test: this should success, indicating a successful install
+  node ./bin/npm-cli.js ls
+
   # 'deps' was added as a dep for many Makefile targets since 8.6 but it's not easy to fix.
   # Let's skip and do it ourselves instead.
   sed -i 's|node bin/npm-cli.js run resetdeps|true|' Makefile
